<!DOCTYPE html>
<html>
<head> <title>Collective Variables Module - Colvars Module - Reference Manual</title>
<meta charset="utf-8"/>
<meta content="width=device-width, initial-scale=1" name="viewport"/>
<meta content="TeX4ht (http://www.cse.ohio-state.edu/~gurari/TeX4ht/)" name="generator"/>
<link href="colvars-refman-namd.css" rel="stylesheet" type="text/css"/>
<script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML" type="text/javascript"></script>
<style type="text/css"> 
.MathJax_MathML {text-indent: 0;} 
</style>
</head><body>
<!--l. 125--><p class="noindent"></p><div id="container"> <div class="inner">
<!--l. 128--><p class="indent">
</p><!--l. 130--><p class="noindent"><span class="colorbox" id="colorbox1"> </span></p><div class="minipage"><!--l. 164--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace></math><br class="newline"/>
<!--l. 164--><p class="noindent"><span class="phvb8t-x-x-248">COLLECTIVE VARIABLES MODULE</span><br class="newline"/>
</p><!--l. 164--><p class="noindent"><span class="phvb8t-x-x-248">Reference manual for NAMD</span><br class="newline"/>
</p><!--l. 164--><p class="noindent"><br class="newline"/><span class="phvr8t-x-x-109">Code version: 2022-02-09</span><br class="newline"/>
</p><!--l. 164--><p class="noindent"><span class="phvr8t-x-x-109">Updated versions of this manual: [</span><a href="https://colvars.github.io/colvars-refman-gromacs/colvars-refman-gromacs.html"><span class="phvr8t-x-x-109">GROMACS</span></a><span class="phvr8t-x-x-109">] [</span><a href="https://colvars.github.io/colvars-refman-lammps/colvars-refman-lammps.html"><span class="phvr8t-x-x-109">LAMMPS</span></a><span class="phvr8t-x-x-109">] [</span><a href="https://colvars.github.io/colvars-refman-namd/colvars-refman-namd.html"><span class="phvr8t-x-x-109">NAMD</span></a><span class="phvr8t-x-x-109">] [</span><a href="https://colvars.github.io/colvars-refman-vmd/colvars-refman-vmd.html"><span class="phvr8t-x-x-109">VMD</span></a><span class="phvr8t-x-x-109">]</span></p></div>
<!--l. 168--><p class="indent"> <img alt="Colvars logo" class="cover_picture" src="cover-512px.jpg"/> <br class="newline"/><span class="phvr8t-x-x-120">Alejandro Bernardin, Haochuan Chen, Jeffrey R. Comer, Giacomo Fiorin, Haohao Fu,</span>
<span class="phvr8t-x-x-120">J</span><span class="phvr8t-x-x-120">é</span><span class="phvr8t-x-x-120">r</span><span class="phvr8t-x-x-120">ô</span><span class="phvr8t-x-x-120">me H</span><span class="phvr8t-x-x-120">é</span><span class="phvr8t-x-x-120">nin, Axel Kohlmeyer, Fabrizio Marinelli, Joshua V. Vermaas, Andrew D.</span>
<span class="phvr8t-x-x-120">White</span>
</p><!--l. 176--><p class="noindent">(<a href="https://colvars.github.io/pdf/colvars-refman-namd.pdf">PDF version</a>)
                                                                                         
                                                                                         
</p><!--l. 187--><p class="indent">
<a id="x1-1doc"></a>
</p><!--l. 189--><p class="indent"> 
</p>
<h3 class="likesectionHead"><a id="x1-1000"></a>Contents</h3>
<div class="tableofcontents">
<span class="sectionToc">1 <a href="#x1-20001" id="QQ2-1-2">Overview</a></span>
<br/> <span class="sectionToc">2 <a href="#x1-30002" id="QQ2-1-3">Writing a Colvars configuration: a crash course</a></span>
<br/> <span class="sectionToc">3 <a href="#x1-60003" id="QQ2-1-6">Enabling and controlling the Colvars module in NAMD</a></span>
<br/>    <span class="subsectionToc">3.1 <a href="#x1-70003.1" id="QQ2-1-7">Units in the Colvars module</a></span>
<br/>    <span class="subsectionToc">3.2 <a href="#x1-80003.2" id="QQ2-1-8">NAMD parameters</a></span>
<br/>    <span class="subsectionToc">3.3 <a href="#x1-90003.3" id="QQ2-1-9">Using the scripting interface to control the Colvars module</a></span>
<br/>     <span class="subsubsectionToc">3.3.1 <a href="#x1-100003.3.1" id="QQ2-1-10">Setting up the Colvars module</a></span>
<br/>     <span class="subsubsectionToc">3.3.2 <a href="#x1-110003.3.2" id="QQ2-1-11">Using the Colvars version in scripts</a></span>
<br/>     <span class="subsubsectionToc">3.3.3 <a href="#x1-120003.3.3" id="QQ2-1-12">Loading and saving the Colvars state and other information</a></span>
<br/>     <span class="subsubsectionToc">3.3.4 <a href="#x1-130003.3.4" id="QQ2-1-13">Accessing atomic data</a></span>
<br/>     <span class="subsubsectionToc">3.3.5 <a href="#x1-140003.3.5" id="QQ2-1-14">Managing collective variables</a></span>
<br/>     <span class="subsubsectionToc">3.3.6 <a href="#x1-150003.3.6" id="QQ2-1-15">Applying and analyzing forces on collective variables</a></span>
<br/>     <span class="subsubsectionToc">3.3.7 <a href="#x1-160003.3.7" id="QQ2-1-16">Managing collective variable biases</a></span>
<br/>     <span class="subsubsectionToc">3.3.8 <a href="#x1-170003.3.8" id="QQ2-1-17">Loading and saving the state of individual biases</a></span>
<br/>    <span class="subsectionToc">3.4 <a href="#x1-180003.4" id="QQ2-1-18">Configuration syntax used by the Colvars module</a></span>
<br/>    <span class="subsectionToc">3.5 <a href="#x1-190003.5" id="QQ2-1-19">Global keywords</a></span>
<br/>    <span class="subsectionToc">3.6 <a href="#x1-200003.6" id="QQ2-1-20">Input state file</a></span>
<br/>    <span class="subsectionToc">3.7 <a href="#x1-210003.7" id="QQ2-1-21">Output files</a></span>
<br/>    <span class="subsectionToc">3.8 <a href="#x1-220003.8" id="QQ2-1-22">File formats</a></span>
<br/>     <span class="subsubsectionToc">3.8.1 <a href="#x1-230003.8.1" id="QQ2-1-23">Configuration and state files.</a></span>
<br/>     <span class="subsubsectionToc">3.8.2 <a href="#x1-240003.8.2" id="QQ2-1-24">Index (NDX) files</a></span>
<br/>     <span class="subsubsectionToc">3.8.3 <a href="#x1-250003.8.3" id="QQ2-1-25">XYZ coordinate files</a></span>
<br/>     <span class="subsubsectionToc">3.8.4 <a href="#x1-260003.8.4" id="QQ2-1-26">PDB coordinate files</a></span>
<br/>     <span class="subsubsectionToc">3.8.5 <a href="#x1-270003.8.5" id="QQ2-1-27">Grid files: multicolumn text format</a></span>
<br/>     <span class="subsubsectionToc">3.8.6 <a href="#x1-280003.8.6" id="QQ2-1-28">Output trajectory files</a></span>
<br/> <span class="sectionToc">4 <a href="#x1-290004" id="QQ2-1-29">Defining collective variables</a></span>
<br/>    <span class="subsectionToc">4.1 <a href="#x1-300004.1" id="QQ2-1-30">Choosing a function</a></span>
<br/>    <span class="subsectionToc">4.2 <a href="#x1-310004.2" id="QQ2-1-31">Distances</a></span>
<br/>     <span class="subsubsectionToc">4.2.1 <a href="#x1-320004.2.1" id="QQ2-1-32"><span class="t1-zi4r-0-x-x-109">distance</span>: center-of-mass distance between two groups.</a></span>
<br/>     <span class="subsubsectionToc">4.2.2 <a href="#x1-330004.2.2" id="QQ2-1-33"><span class="t1-zi4r-0-x-x-109">distanceZ</span>: projection of a distance vector on an axis.</a></span>
<br/>     <span class="subsubsectionToc">4.2.3 <a href="#x1-340004.2.3" id="QQ2-1-34"><span class="t1-zi4r-0-x-x-109">distanceXY</span>: modulus of the projection of a distance vector on a plane.</a></span>
<br/>     <span class="subsubsectionToc">4.2.4 <a href="#x1-350004.2.4" id="QQ2-1-35"><span class="t1-zi4r-0-x-x-109">distanceVec</span>: distance vector between two groups.</a></span>
<br/>     <span class="subsubsectionToc">4.2.5 <a href="#x1-360004.2.5" id="QQ2-1-36"><span class="t1-zi4r-0-x-x-109">distanceDir</span>: distance unit vector between two groups.</a></span>
<br/>     <span class="subsubsectionToc">4.2.6 <a href="#x1-370004.2.6" id="QQ2-1-37"><span class="t1-zi4r-0-x-x-109">distanceInv</span>: mean distance between two groups of atoms.</a></span>
<br/>    <span class="subsectionToc">4.3 <a href="#x1-380004.3" id="QQ2-1-38">Angles</a></span>
<br/>     <span class="subsubsectionToc">4.3.1 <a href="#x1-390004.3.1" id="QQ2-1-39"><span class="t1-zi4r-0-x-x-109">angle</span>: angle between three groups.</a></span>
<br/>     <span class="subsubsectionToc">4.3.2 <a href="#x1-400004.3.2" id="QQ2-1-40"><span class="t1-zi4r-0-x-x-109">dipoleAngle</span>: angle between two groups and dipole of a third group.</a></span>
<br/>     <span class="subsubsectionToc">4.3.3 <a href="#x1-410004.3.3" id="QQ2-1-41"><span class="t1-zi4r-0-x-x-109">dihedral</span>: torsional angle between four groups.</a></span>
<br/>     <span class="subsubsectionToc">4.3.4 <a href="#x1-420004.3.4" id="QQ2-1-42"><span class="t1-zi4r-0-x-x-109">polarTheta</span>: polar angle in spherical coordinates.</a></span>
<br/>     <span class="subsubsectionToc">4.3.5 <a href="#x1-430004.3.5" id="QQ2-1-43"><span class="t1-zi4r-0-x-x-109">polarPhi</span>: azimuthal angle in spherical coordinates.</a></span>
<br/>    <span class="subsectionToc">4.4 <a href="#x1-440004.4" id="QQ2-1-44">Contacts</a></span>
<br/>     <span class="subsubsectionToc">4.4.1 <a href="#x1-450004.4.1" id="QQ2-1-45"><span class="t1-zi4r-0-x-x-109">coordNum</span>: coordination number between two groups.</a></span>
<br/>     <span class="subsubsectionToc">4.4.2 <a href="#x1-460004.4.2" id="QQ2-1-46"><span class="t1-zi4r-0-x-x-109">selfCoordNum</span>: coordination number between atoms within a group.</a></span>
<br/>     <span class="subsubsectionToc">4.4.3 <a href="#x1-470004.4.3" id="QQ2-1-47"><span class="t1-zi4r-0-x-x-109">hBond</span>: hydrogen bond between two atoms.</a></span>
<br/>    <span class="subsectionToc">4.5 <a href="#x1-480004.5" id="QQ2-1-48">Collective metrics</a></span>
<br/>     <span class="subsubsectionToc">4.5.1 <a href="#x1-490004.5.1" id="QQ2-1-49"><span class="t1-zi4r-0-x-x-109">rmsd</span>: root mean square displacement (RMSD) from reference positions.</a></span>
<br/>     <span class="subsubsectionToc">4.5.2 <a href="#x1-500004.5.2" id="QQ2-1-50">Advanced usage of the <span class="t1-zi4r-0-x-x-109">rmsd </span>component.</a></span>
<br/>     <span class="subsubsectionToc">4.5.3 <a href="#x1-510004.5.3" id="QQ2-1-51"><span class="t1-zi4r-0-x-x-109">eigenvector</span>: projection of the atomic coordinates on a vector.</a></span>
<br/>     <span class="subsubsectionToc">4.5.4 <a href="#x1-520004.5.4" id="QQ2-1-52"><span class="t1-zi4r-0-x-x-109">gyration</span>: radius of gyration of a group of atoms.</a></span>
<br/>     <span class="subsubsectionToc">4.5.5 <a href="#x1-530004.5.5" id="QQ2-1-53"><span class="t1-zi4r-0-x-x-109">inertia</span>: total moment of inertia of a group of atoms.</a></span>
<br/>     <span class="subsubsectionToc">4.5.6 <a href="#x1-540004.5.6" id="QQ2-1-54"><span class="t1-zi4r-0-x-x-109">dipoleMagnitude</span>: dipole magnitude of a group of atoms.</a></span>
<br/>     <span class="subsubsectionToc">4.5.7 <a href="#x1-550004.5.7" id="QQ2-1-55"><span class="t1-zi4r-0-x-x-109">inertiaZ</span>: total moment of inertia of a group of atoms around a chosen axis.</a></span>
<br/>    <span class="subsectionToc">4.6 <a href="#x1-560004.6" id="QQ2-1-56">Rotations</a></span>
<br/>     <span class="subsubsectionToc">4.6.1 <a href="#x1-570004.6.1" id="QQ2-1-57"><span class="t1-zi4r-0-x-x-109">orientation</span>: orientation from reference coordinates.</a></span>
<br/>     <span class="subsubsectionToc">4.6.2 <a href="#x1-580004.6.2" id="QQ2-1-58"><span class="t1-zi4r-0-x-x-109">orientationAngle</span>: angle of rotation from reference coordinates.</a></span>
<br/>     <span class="subsubsectionToc">4.6.3 <a href="#x1-590004.6.3" id="QQ2-1-59"><span class="t1-zi4r-0-x-x-109">orientationProj</span>: cosine of the angle of rotation from reference coordinates.</a></span>
<br/>     <span class="subsubsectionToc">4.6.4 <a href="#x1-600004.6.4" id="QQ2-1-60"><span class="t1-zi4r-0-x-x-109">spinAngle</span>: angle of rotation around a given axis.</a></span>
<br/>     <span class="subsubsectionToc">4.6.5 <a href="#x1-610004.6.5" id="QQ2-1-61"><span class="t1-zi4r-0-x-x-109">tilt</span>: cosine of the rotation orthogonal to a given axis.</a></span>
<br/>     <span class="subsubsectionToc">4.6.6 <a href="#x1-620004.6.6" id="QQ2-1-62"><span class="t1-zi4r-0-x-x-109">eulerPhi</span>: Roll angle from references coordinates.</a></span>
<br/>     <span class="subsubsectionToc">4.6.7 <a href="#x1-630004.6.7" id="QQ2-1-63"><span class="t1-zi4r-0-x-x-109">eulerTheta</span>: Pitch angle from references coordinates.</a></span>
<br/>     <span class="subsubsectionToc">4.6.8 <a href="#x1-640004.6.8" id="QQ2-1-64"><span class="t1-zi4r-0-x-x-109">eulerPsi</span>: Yaw angle from references coordinates.</a></span>
<br/>    <span class="subsectionToc">4.7 <a href="#x1-650004.7" id="QQ2-1-65">Protein structure descriptors</a></span>
<br/>     <span class="subsubsectionToc">4.7.1                                                                                         <a href="#x1-660004.7.1" id="QQ2-1-66"><span class="t1-zi4r-0-x-x-109">alpha</span>:
<!--l. 67--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>α</mi></math>-helix
content of a protein segment.</a></span>
<br/>     <span class="subsubsectionToc">4.7.2 <a href="#x1-670004.7.2" id="QQ2-1-67"><span class="t1-zi4r-0-x-x-109">dihedralPC</span>: protein dihedral principal component</a></span>
<br/>    <span class="subsectionToc">4.8 <a href="#x1-680004.8" id="QQ2-1-68">Raw data: building blocks for custom functions</a></span>
<br/>     <span class="subsubsectionToc">4.8.1 <a href="#x1-690004.8.1" id="QQ2-1-69"><span class="t1-zi4r-0-x-x-109">cartesian</span>: vector of atomic Cartesian coordinates.</a></span>
<br/>     <span class="subsubsectionToc">4.8.2 <a href="#x1-700004.8.2" id="QQ2-1-70"><span class="t1-zi4r-0-x-x-109">distancePairs</span>: set of pairwise distances between two groups.</a></span>
<br/>    <span class="subsectionToc">4.9 <a href="#x1-710004.9" id="QQ2-1-71">Geometric path collective variables</a></span>
<br/>     <span class="subsubsectionToc">4.9.1 <a href="#x1-720004.9.1" id="QQ2-1-72"><span class="t1-zi4r-0-x-x-109">gspath</span>: progress along a path defined in atomic Cartesian coordinate space.</a></span>
<br/>     <span class="subsubsectionToc">4.9.2 <a href="#x1-730004.9.2" id="QQ2-1-73"><span class="t1-zi4r-0-x-x-109">gzpath</span>: distance from a path defined in atomic Cartesian coordinate space.</a></span>
<br/>     <span class="subsubsectionToc">4.9.3 <a href="#x1-740004.9.3" id="QQ2-1-74"><span class="t1-zi4r-0-x-x-109">linearCombination</span>: Helper CV to define a linear combination of other CVs</a></span>
<br/>     <span class="subsubsectionToc">4.9.4 <a href="#x1-750004.9.4" id="QQ2-1-75"><span class="t1-zi4r-0-x-x-109">customColvar</span>: Helper CV to define a mathematical expression
as CV from other CVs</a></span>
<br/>     <span class="subsubsectionToc">4.9.5 <a href="#x1-760004.9.5" id="QQ2-1-76"><span class="t1-zi4r-0-x-x-109">gspathCV</span>: progress along a path defined in CV space.</a></span>
<br/>     <span class="subsubsectionToc">4.9.6 <a href="#x1-770004.9.6" id="QQ2-1-77"><span class="t1-zi4r-0-x-x-109">gzpathCV</span>: distance from a path defined in CV space.</a></span>
<br/>    <span class="subsectionToc">4.10 <a href="#x1-780004.10" id="QQ2-1-78">Arithmetic path collective variables</a></span>
<br/>     <span class="subsubsectionToc">4.10.1 <a href="#x1-790004.10.1" id="QQ2-1-79"><span class="t1-zi4r-0-x-x-109">aspathCV</span>: progress along a path defined in CV space.</a></span>
<br/>     <span class="subsubsectionToc">4.10.2 <a href="#x1-800004.10.2" id="QQ2-1-80"><span class="t1-zi4r-0-x-x-109">azpathCV</span>: distance from a path defined in CV space.</a></span>
<br/>     <span class="subsubsectionToc">4.10.3 <a href="#x1-810004.10.3" id="QQ2-1-81">Path collective variables in Cartesian coordinates</a></span>
<br/>    <span class="subsectionToc">4.11 <a href="#x1-820004.11" id="QQ2-1-82">Dense neural network in CV space</a></span>
<br/>     <span class="subsubsectionToc">4.11.1 <a href="#x1-830004.11.1" id="QQ2-1-84"><span class="t1-zi4r-0-x-x-109">NeuralNetwork</span>: Dense neural network.</a></span>
<br/>    <span class="subsectionToc">4.12 <a href="#x1-840004.12" id="QQ2-1-85">Volumetric map-based variables</a></span>
<br/>     <span class="subsubsectionToc">4.12.1 <a href="#x1-850004.12.1" id="QQ2-1-86"><span class="t1-zi4r-0-x-x-109">mapTotal</span>: total value of a volumetric map</a></span>
<br/>     <span class="subsubsectionToc">4.12.2 <a href="#x1-860004.12.2" id="QQ2-1-87">Multiple volumetric maps collective variables</a></span>
<br/>    <span class="subsectionToc">4.13 <a href="#x1-870004.13" id="QQ2-1-88">Shared keywords for all components</a></span>
<br/>    <span class="subsectionToc">4.14 <a href="#x1-880004.14" id="QQ2-1-89">Periodic components</a></span>
<br/>    <span class="subsectionToc">4.15 <a href="#x1-890004.15" id="QQ2-1-90">Non-scalar components</a></span>
<br/>     <span class="subsubsectionToc">4.15.1 <a href="#x1-900004.15.1" id="QQ2-1-91">Calculating total forces</a></span>
<br/>    <span class="subsectionToc">4.16 <a href="#x1-910004.16" id="QQ2-1-92">Linear and polynomial combinations of components</a></span>
<br/>    <span class="subsectionToc">4.17 <a href="#x1-920004.17" id="QQ2-1-93">Custom functions</a></span>
<br/>    <span class="subsectionToc">4.18 <a href="#x1-930004.18" id="QQ2-1-94">Scripted functions</a></span>
<br/>    <span class="subsectionToc">4.19 <a href="#x1-940004.19" id="QQ2-1-95">Defining grid parameters</a></span>
<br/>    <span class="subsectionToc">4.20 <a href="#x1-950004.20" id="QQ2-1-96">Trajectory output</a></span>
<br/>    <span class="subsectionToc">4.21 <a href="#x1-960004.21" id="QQ2-1-97">Extended Lagrangian</a></span>
<br/>    <span class="subsectionToc">4.22 <a href="#x1-970004.22" id="QQ2-1-98">Multiple time-step variables</a></span>
<br/>    <span class="subsectionToc">4.23 <a href="#x1-980004.23" id="QQ2-1-99">Backward-compatibility</a></span>
<br/>    <span class="subsectionToc">4.24 <a href="#x1-990004.24" id="QQ2-1-100">Statistical analysis</a></span>
<br/> <span class="sectionToc">5 <a href="#x1-1000005" id="QQ2-1-101">Selecting atoms</a></span>
<br/>    <span class="subsectionToc">5.1 <a href="#x1-1010005.1" id="QQ2-1-102">Atom selection keywords</a></span>
<br/>    <span class="subsectionToc">5.2 <a href="#x1-1020005.2" id="QQ2-1-103">Moving frame of reference.</a></span>
<br/>    <span class="subsectionToc">5.3 <a href="#x1-1050005.3" id="QQ2-1-106">Treatment of periodic boundary conditions.</a></span>
<br/>    <span class="subsectionToc">5.4 <a href="#x1-1060005.4" id="QQ2-1-107">Performance of a Colvars calculation based on group size.</a></span>
<br/> <span class="sectionToc">6 <a href="#x1-1070006" id="QQ2-1-108">Biasing and analysis methods</a></span>
<br/>    <span class="subsectionToc">6.1 <a href="#x1-1080006.1" id="QQ2-1-109">Thermodynamic integration</a></span>
<br/>    <span class="subsectionToc">6.2 <a href="#x1-1090006.2" id="QQ2-1-110">Adaptive Biasing Force</a></span>
<br/>     <span class="subsubsectionToc">6.2.1 <a href="#x1-1100006.2.1" id="QQ2-1-111">ABF requirements on collective variables</a></span>
<br/>     <span class="subsubsectionToc">6.2.2 <a href="#x1-1110006.2.2" id="QQ2-1-112">Parameters for ABF</a></span>
<br/>     <span class="subsubsectionToc">6.2.3 <a href="#x1-1120006.2.3" id="QQ2-1-113">Multiple-replica ABF</a></span>
<br/>     <span class="subsubsectionToc">6.2.4 <a href="#x1-1130006.2.4" id="QQ2-1-114">Output files</a></span>
<br/>     <span class="subsubsectionToc">6.2.5 <a href="#x1-1140006.2.5" id="QQ2-1-115">Multidimensional free energy surfaces</a></span>
<br/>    <span class="subsectionToc">6.3 <a href="#x1-1150006.3" id="QQ2-1-116">Extended-system Adaptive Biasing Force (eABF)</a></span>
<br/>     <span class="subsubsectionToc">6.3.1 <a href="#x1-1160006.3.1" id="QQ2-1-117">CZAR estimator of the free energy</a></span>
<br/>     <span class="subsubsectionToc">6.3.2 <a href="#x1-1170006.3.2" id="QQ2-1-118">Zheng/Yang estimator of the free energy</a></span>
<br/>    <span class="subsectionToc">6.4 <a href="#x1-1190006.4" id="QQ2-1-120">Metadynamics</a></span>
<br/>     <span class="subsubsectionToc">6.4.1 <a href="#x1-1200006.4.1" id="QQ2-1-121">Treatment of the PMF boundaries</a></span>
<br/>     <span class="subsubsectionToc">6.4.2 <a href="#x1-1210006.4.2" id="QQ2-1-122">Basic configuration keywords</a></span>
<br/>     <span class="subsubsectionToc">6.4.3 <a href="#x1-1220006.4.3" id="QQ2-1-123">Output files</a></span>
<br/>     <span class="subsubsectionToc">6.4.4 <a href="#x1-1230006.4.4" id="QQ2-1-124">Performance optimization</a></span>
<br/>     <span class="subsubsectionToc">6.4.5 <a href="#x1-1240006.4.5" id="QQ2-1-125">Ensemble-Biased Metadynamics</a></span>
<br/>     <span class="subsubsectionToc">6.4.6 <a href="#x1-1250006.4.6" id="QQ2-1-126">Well-tempered metadynamics</a></span>
<br/>     <span class="subsubsectionToc">6.4.7 <a href="#x1-1260006.4.7" id="QQ2-1-127">Multiple-walker metadynamics</a></span>
<br/>    <span class="subsectionToc">6.5 <a href="#x1-1270006.5" id="QQ2-1-128">Harmonic restraints</a></span>
<br/>     <span class="subsubsectionToc">6.5.1 <a href="#x1-1280006.5.1" id="QQ2-1-129">Moving restraints: steered molecular dynamics</a></span>
<br/>     <span class="subsubsectionToc">6.5.2 <a href="#x1-1290006.5.2" id="QQ2-1-130">Moving restraints: umbrella sampling</a></span>
<br/>     <span class="subsubsectionToc">6.5.3 <a href="#x1-1300006.5.3" id="QQ2-1-131">Changing force constant</a></span>
<br/>    <span class="subsectionToc">6.6 <a href="#x1-1310006.6" id="QQ2-1-132">Computing the work of a changing restraint</a></span>
<br/>    <span class="subsectionToc">6.7 <a href="#x1-1320006.7" id="QQ2-1-133">Harmonic wall restraints</a></span>
<br/>    <span class="subsectionToc">6.8 <a href="#x1-1330006.8" id="QQ2-1-134">Linear restraints</a></span>
<br/>    <span class="subsectionToc">6.9 <a href="#x1-1340006.9" id="QQ2-1-135">Adaptive Linear Bias/Experiment Directed Simulation</a></span>
<br/>    <span class="subsectionToc">6.10 <a href="#x1-1350006.10" id="QQ2-1-136">Multidimensional histograms</a></span>
<br/>     <span class="subsubsectionToc">6.10.1 <a href="#x1-1360006.10.1" id="QQ2-1-137">Grid definition for multidimensional histograms</a></span>
<br/>    <span class="subsectionToc">6.11 <a href="#x1-1370006.11" id="QQ2-1-138">Reweighted free-energy landscapes from accelerated MD</a></span>
<br/>    <span class="subsectionToc">6.12 <a href="#x1-1380006.12" id="QQ2-1-139">Probability distribution-restraints</a></span>
<br/>    <span class="subsectionToc">6.13 <a href="#x1-1390006.13" id="QQ2-1-140">Defining scripted biases</a></span>
<br/>    <span class="subsectionToc">6.14 <a href="#x1-1400006.14" id="QQ2-1-141">Performance of scripted biases</a></span>
<br/> <span class="sectionToc">7 <a href="#x1-1410007" id="QQ2-1-142">Scripting interface (Tcl): list of commands</a></span>
<br/>    <span class="subsectionToc">7.1 <a href="#x1-1420007.1" id="QQ2-1-143">Commands to manage the Colvars module</a></span>
<br/>    <span class="subsectionToc">7.2 <a href="#x1-1430007.2" id="QQ2-1-144">Commands to manage individual collective variables</a></span>
<br/>    <span class="subsectionToc">7.3 <a href="#x1-1440007.3" id="QQ2-1-145">Commands to manage individual biases</a></span>
<br/> <span class="sectionToc">8 <a href="#x1-1450008" id="QQ2-1-146">Syntax changes from older versions</a></span>
<br/> <span class="sectionToc">9 <a href="#x1-1460009" id="QQ2-1-147">Compilation notes</a></span>
</div>
<!--l. 65--><p class="indent">
</p>
<div class="flushright">
<!--l. 65--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 65--><p class="noindent"></p><div class="heading"><a id="sec:colvars_intro">
<h3 class="sectionHead"><span class="titlemark">1    </span> <a id="x1-20001"></a>Overview</h3>
<!--l. 65--><p class="noindent"></p></a> </div> 
<!--l. 67--><p class="indent">   In molecular dynamics simulations, it is often useful to reduce the large number of degrees of freedom of a
physical system into few parameters whose statistical distributions can be analyzed individually, or used to
define biasing potentials to alter the dynamics of the system in a controlled manner. These have been
called ‘order parameters', ‘collective variables', ‘(surrogate) reaction coordinates', and many other
terms.
</p><!--l. 70--><p class="indent">   Here we use primarily the term ‘collective variable', often shortened to <span class="ptmri8t-x-x-109">colvar</span>, to indicate any differentiable function of atomic
Cartesian coordinates, <!--l. 70--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>x</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mi>i</mi></mrow></msub></math>,
with <!--l. 70--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math>
between <!--l. 70--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>1</mn></math>
and <!--l. 70--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>N</mi></math>, the
total number of atoms: </p><table class="equation"><tr><td> <a id="x1-2001r1"></a>
<!--l. 72--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>ξ</mi><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow><mspace class="tmspace" width="3.04074pt"></mspace> <mo class="MathClass-rel">=</mo> <mi>ξ</mi><mrow><mo class="MathClass-open">(</mo><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>X</mi></mstyle><mtext></mtext></mstyle><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow></mrow><mo class="MathClass-close">)</mo></mrow><mspace class="tmspace" width="3.04074pt"></mspace> <mo class="MathClass-rel">=</mo> <mi>ξ</mi> <mfenced close=")" open="(" separators=""><mrow><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>x</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mi>i</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-punc">,</mo><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>x</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mi>j</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-punc">,</mo><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>x</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mi>k</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-punc">,</mo><mo class="MathClass-op">…</mo></mrow></mfenced><mspace class="tmspace" width="3.04074pt"></mspace><mo class="MathClass-punc">,</mo><mspace class="tmspace" width="3.04074pt"></mspace><mspace class="tmspace" width="3.04074pt"></mspace><mn>1</mn> <mo class="MathClass-rel">≤</mo> <mi>i</mi><mo class="MathClass-punc">,</mo><mi>j</mi><mo class="MathClass-punc">,</mo><mi>k</mi><mo class="MathClass-op">…</mo> <mo class="MathClass-rel">≤</mo> <mi>N</mi>
</math></td><td class="eq-no">(1)</td></tr></table>
<!--l. 80--><p class="indent">   This manual documents the collective variables module (<span class="ptmb8t-x-x-109">Colvars</span>), a software that provides an implementation for the
functions <!--l. 80--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi><mrow><mo class="MathClass-open">(</mo><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>X</mi></mstyle><mtext></mtext></mstyle></mrow><mo class="MathClass-close">)</mo></mrow></math>
with a focus on flexibility, robustness and high performance. The module is designed to perform multiple tasks
concurrently during or after a simulation, the most common of which are: </p>
<ul class="itemize1">
<li class="itemize">apply restraints or biasing potentials to multiple variables, tailored on the system by choosing from a
     wide set of basis functions, without limitations on their number or on the number of atoms involved;
     while this can in principle be done through a TclForces script, using the Colvars module is both easier
     and computationally more efficient;
     </li>
<li class="itemize">calculate  potentials  of  mean  force  (PMFs)  along  any  set  of  variables,  using  different  enhanced
     sampling methods, such as Adaptive Biasing Force (ABF), metadynamics, steered MD and umbrella
     sampling; variants of these methods that make use of an ensemble of replicas are supported as well;
                                                                                         
                                                                                         
     </li>
<li class="itemize">calculate  statistical  properties  of  the  variables,  such  as  running  averages  and  standard  deviations,
     correlation functions of pairs of variables, and multidimensional histograms: this can be done either at
     run-time without the need to save very large trajectory files, or after a simulation has been completed
     using VMD and the <span class="t1-zi4r-0-x-x-109">cv </span>command.
     </li></ul>
<!--l. 94--><p class="indent">   Detailed explanations of the design of the Colvars module are provided in reference <span class="cite">[<a href="#XFiorin2013">1</a>]</span>. Please cite this
reference whenever publishing work that makes use of this module.
</p><!--l. 106--><p class="indent">
</p>
<div class="flushright">
<!--l. 106--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 106--><p class="noindent"></p><div class="heading"><a id="sec:colvars_crash_course">
<h3 class="sectionHead"><span class="titlemark">2    </span> <a id="x1-30002"></a>Writing a Colvars configuration: a crash course</h3>
<!--l. 106--><p class="noindent"></p></a> </div> 
<!--l. 108--><p class="indent">   The Colvars configuration is a plain text file or string that defines collective variables, biases, and general
parameters of the Colvars module. It is passed to the module using back-end-specific commands documented in
section <a href="#x1-60003">3<!--tex4ht:ref: sec:colvarmodule --></a>.
</p>
<!--l. 114--><p class="noindent"><span class="likeparagraphHead"><a id="x1-40002"></a>Example: steering two groups of atoms away from each other.</span>
   Now let us look at a complete, non-trivial configuration. Suppose that we want to run a steered MD experiment
where a small molecule is pulled away from a protein binding site. In Colvars terms, this is done by applying a
moving restraint to the distance between the two objects. The configuration will contain two blocks, one
defining the distance variable (see section <a href="#x1-290004">4<!--tex4ht:ref: sec:colvar --></a> and <a href="#x1-320004.2.1">4.2.1<!--tex4ht:ref: sec:cvc_distance --></a>), and the other the moving harmonic restraint
(<a href="#x1-1270006.5">6.5<!--tex4ht:ref: sec:colvarbias_harmonic --></a>).</p>
<div class="mdframed" id="mdframed-1">
<!--l. 122--><p class="noindent"></p><pre>
<!--l. 124--><p class="noindent">colvar {<br class="newline"/>  name dist<br class="newline"/>  distance {<br class="newline"/>    group1 { atomNumbersRange 42-55 }<br class="newline"/>    group2 {<br class="newline"/>      psfSegID  PR<br class="newline"/>      atomNameResidueRange  CA 15-30<br class="newline"/>    }<br class="newline"/>  }<br class="newline"/>}<br class="newline"/><br class="newline"/>harmonic {<br class="newline"/>  colvars dist<br class="newline"/>  forceConstant 20.0<br class="newline"/>  centers 4.0         # initial distance<br class="newline"/>  targetCenters 15.0  # final distance<br class="newline"/>  targetNumSteps 500000<br class="newline"/>} </p></pre>
</div>
<!--l. 145--><p class="indent">   Reading this input in plain English: the variable here named <span class="ptmri8t-x-x-109">dist </span>consists in a
distance function between the centers of two groups: the ligand (atoms 42 to 55) and the
                                                                                         
                                                                                         
<!--l. 145--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>α</mi></math>-carbon atoms of
residues 15 to 30 in the protein (segment name PR). To the “<span class="ptmri8t-x-x-109">dist</span>" variable, we apply a <span class="t1-zi4r-0-x-x-109">harmonic </span>potential of force constant
20 kcal/mol/Å<!--l. 146--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow></mrow><mrow><mn>2</mn></mrow></msup></math>,
initially centered around a value of 4 Å, which will increase to 15 Å over 500,000 simulation steps.
</p><!--l. 148--><p class="indent">   The atom selection keywords are detailed in section <a href="#x1-1000005">5<!--tex4ht:ref: sec:colvar_atom_groups --></a>.
</p>
<!--l. 152--><p class="noindent"><span class="likeparagraphHead"><a id="x1-50002"></a>Example: using multiple variables and multiple biasing/analysis methods together.</span>
   A more complex example configuration is included below, showing how a variable may be constructed by combining
multiple existing functions, and how multiple variables or multiple biases may be used concurrently. The colvar indicated
below as “<!--l. 155--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>d</mi></math>"
is defined as the difference between two distances (see <a href="#x1-310004.2">4.2<!--tex4ht:ref: sec:cvc_distances --></a>): the first distance
(<!--l. 155--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>d</mi></mrow><mrow><mn>1</mn></mrow></msub></math>)
is taken between the center of mass of atoms 1 and 2 and that of atoms 3 to 5, the second
(<!--l. 155--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>d</mi></mrow><mrow><mn>2</mn></mrow></msub></math>)
between atom 7 and the center of mass of atoms 8 to 10 (see <a href="#x1-1000005">5<!--tex4ht:ref: sec:colvar_atom_groups --></a>). The difference
<!--l. 156--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>d</mi> <mo class="MathClass-rel">=</mo> <msub><mrow><mi>d</mi></mrow><mrow><mn>1</mn></mrow></msub> <mo class="MathClass-bin">−</mo><msub><mrow><mi>d</mi></mrow><mrow><mn>2</mn></mrow></msub></math> is obtained by multiplying
the two by a coefficient <!--l. 156--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>C</mi> <mo class="MathClass-rel">=</mo> <mo class="MathClass-bin">+</mo><mn>1</mn></math> or
<!--l. 156--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>C</mi> <mo class="MathClass-rel">=</mo> <mo class="MathClass-bin">−</mo><mn>1</mn></math>, respectively (see <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a>).
The colvar called “<!--l. 157--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>c</mi></math>"
is the coordination number calculated between atoms 1 to 10 and atoms 11 to 20. A harmonic restraint (see <a href="#x1-1270006.5">6.5<!--tex4ht:ref: sec:colvarbias_harmonic --></a>) is applied
to both <!--l. 157--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>d</mi></math> and
<!--l. 157--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>c</mi></math>: to allow using the
same force constant <!--l. 157--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>K</mi></math>,
both <!--l. 157--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>d</mi></math> and
<!--l. 157--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>c</mi></math> are scaled by their respective
fluctuation widths <!--l. 157--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>w</mi></mrow><mrow><mi>d</mi></mrow></msub></math> and
<!--l. 157--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>w</mi></mrow><mrow><mi>c</mi></mrow></msub></math>. A third colvar “alpha" is defined
as the <!--l. 158--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>α</mi></math>-helical content of residues 1
to 10 (see <a href="#x1-660004.7.1">4.7.1<!--tex4ht:ref: sec:cvc_alpha --></a>). The values of “<!--l. 159--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>c</mi></math>"
and “alpha" are also recorded throughout the simulation as a joint 2-dimensional histogram (see <a href="#x1-1350006.10">6.10<!--tex4ht:ref: sec:colvarbias_histogram --></a>).</p>
<div class="mdframed" id="mdframed-2">
<!--l. 161--><p class="noindent"></p><pre>colvar {<br class="newline"/>  # difference of two distances<br class="newline"/>  name d <br class="newline"/>  width 0.2 # 0.2 Å of estimated fluctuation width <br class="newline"/>  distance {<br class="newline"/>    componentCoeff 1.0<br class="newline"/>    group1 { atomNumbers 1 2 }<br class="newline"/>    group2 { atomNumbers 3 4 5 }<br class="newline"/>  }<br class="newline"/>  distance {<br class="newline"/>    componentCoeff -1.0<br class="newline"/>    group1 { atomNumbers 7 }<br class="newline"/>    group2 { atomNumbers 8 9 10 }<br class="newline"/>  }<br class="newline"/>}<br class="newline"/><br class="newline"/>colvar {<br class="newline"/>  name c<br class="newline"/>  coordNum {<br class="newline"/>    cutoff 6.0<br class="newline"/>    group1 { atomNumbersRange 1-10 }<br class="newline"/>    group2 { atomNumbersRange 11-20 }<br class="newline"/>    tolerance 1.0e-6<br class="newline"/>    pairListFrequency 1000<br class="newline"/>  }<br class="newline"/>}<br class="newline"/>colvar {<br class="newline"/>  name alpha<br class="newline"/>  alpha {<br class="newline"/>    psfSegID PROT<br class="newline"/>    residueRange 1-10<br class="newline"/>  }<br class="newline"/>} <br class="newline"/>harmonic {<br class="newline"/>  colvars d c<br class="newline"/>  centers 3.0 4.0<br class="newline"/>  forceConstant 5.0<br class="newline"/>}<br class="newline"/><br class="newline"/>histogram {<br class="newline"/>  colvars c alpha<br class="newline"/>} </pre>
</div>
<!--l. 432--><p class="indent">
</p>
<div class="flushright">
<!--l. 432--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 432--><p class="noindent"></p><div class="heading"><a id="sec:colvarmodule">
<h3 class="sectionHead"><span class="titlemark">3    </span> <a id="x1-60003"></a>Enabling and controlling the Colvars module in NAMD</h3>
<!--l. 432--><p class="noindent"></p></a> </div> 
<!--l. 434--><p class="indent">   Here, we document the syntax of the commands and parameters used to set up and use the Colvars module in
NAMD <span class="cite">[<a href="#XPhillips2020">2</a>]</span>. One of these parameters is the configuration file or the configuration text for the module itself, whose
syntax is described in <a href="#x1-180003.4">3.4<!--tex4ht:ref: sec:colvars_config_syntax --></a> and in the following sections.
</p>
<div class="flushright">
<!--l. 437--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 437--><p class="noindent"></p><div class="heading"><a id="sec:units">
<h4 class="subsectionHead"><span class="titlemark">3.1    </span> <a id="x1-70003.1"></a>Units in the Colvars module</h4>
<!--l. 437--><p class="noindent"></p></a> </div> 
<!--l. 439--><p class="indent">   The “internal units" of the Colvars module are the units in which values are expected to be in the configuration
file, and in which collective variable values, energies, etc. are expressed in the output and colvars trajectory files.
Generally <span class="ptmb8t-x-x-109">the Colvars module uses internally the same units as its back-end MD engine, with the exception of</span>
<span class="ptmb8t-x-x-109">VMD</span>, where different unit sets are supported to allow for easy setup, visualization and analysis of Colvars
simulations performed with any simulation engine.
</p><!--l. 442--><p class="indent">   Note that <span class="ptmb8t-x-x-109">angles </span>are expressed in degrees, and derived quantites such as force constants are based on degrees as
well. Atomic coordinates read from <span class="ptmb8t-x-x-109">XYZ files </span>(and PDB files where applicable) are expected to be
expressed in Ångström, no matter what unit system is in use by the back-end (NAMD) or the Colvars
Module.
</p><!--l. 445--><p class="indent">   To avoid errors due to reading configuration files written in a different unit system, it can be specified within the
input:
</p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-7001"></a><span class="colorbox" id="colorbox2"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">units</span></span> <!--l. 469--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Unit
     system to be used<!--l. 469--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>global <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>string <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>A string defining the units to be used internally by Colvars. In NAMD the only supported
     value is NAMD's native units: <span class="ptmri8t-x-x-109">real </span>(Å, kcal/mol). </li></ul>
<div class="flushright">
<!--l. 513--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 513--><p class="noindent"></p><div class="heading"><a id="sec:colvars_mdengine_parameters">
<h4 class="subsectionHead"><span class="titlemark">3.2    </span> <a id="x1-80003.2"></a>NAMD parameters</h4>
<!--l. 513--><p class="noindent"></p></a> </div> 
<!--l. 513--><p class="indent">   To enable a Colvars-based calculation, the <span class="t1-zi4r-0-x-x-109">colvars on </span>command must be added to the NAMD script. Two
optional commands, <span class="t1-zi4r-0-x-x-109">colvarsConfig </span>and <span class="t1-zi4r-0-x-x-109">colvarsInput </span>can be used to define the module's configuration or
continue a previous simulation. Because these are static parameters, it is typically more convenient to use the <span class="t1-zi4r-0-x-x-109">cv</span>
command in the rest of the NAMD script.
</p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-8001"></a><span class="colorbox" id="colorbox3"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">colvars</span></span> <!--l. 513--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Enable
     the Colvars module<!--l. 513--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>NAMD configuration file <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">off</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>If this flag is on, the Colvars module within NAMD is enabled.
     </li>
<li class="itemize"><a id="dx1-8002"></a><span class="colorbox" id="colorbox4"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">colvarsConfig</span></span> <!--l. 513--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Configuration
     file for the collective variables<!--l. 513--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>NAMD configuration file <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>UNIX filename <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Name of the Colvars configuration file (<a href="#x1-180003.4">3.4<!--tex4ht:ref: sec:colvars_config_syntax --></a>, <a href="#x1-190003.5">3.5<!--tex4ht:ref: sec:colvars_global --></a> and following sections). This file can
     also  be  provided  by  the  Tcl  command  <span class="t1-zi4r-0-x-x-109">cv configfile</span>.  Alternatively,  the  contents  of  the  file  (as
     opposed to the file itself) can be given as a string argument to the command <span class="t1-zi4r-0-x-x-109">cv config</span>.
     </li>
<li class="itemize"><a id="dx1-8003"></a><span class="colorbox" id="colorbox5"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">colvarsInput</span></span> <!--l. 513--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Input
     state file for the collective variables<!--l. 513--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>NAMD configuration file <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>UNIX filename <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Keyword used to specify the input state file's name (<a href="#x1-200003.6">3.6<!--tex4ht:ref: sec:colvars_input --></a>). If the input file is meant to be
     loaded within a Tcl script section, the <span class="t1-zi4r-0-x-x-109">cv load </span>command may be used instead.
     </li></ul>
<div class="flushright">
<!--l. 617--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 617--><p class="noindent"></p><div class="heading"><a id="sec:cv_scripting">
<h4 class="subsectionHead"><span class="titlemark">3.3    </span> <a id="x1-90003.3"></a>Using the scripting interface to control the Colvars module</h4>
<!--l. 617--><p class="noindent"></p></a> </div> 
<!--l. 620--><p class="indent">   After the first initialization of the Colvars module, the internal state of Colvars objects may be queried or
modified in a NAMD script:</p>
<div class="mdframed" id="mdframed-3">
<!--l. 621--><p class="noindent"></p><pre>cv <!--l. 621--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ptmri8t-x-x-109">method</span><!--l. 621--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math> arg1 arg2 ...
</pre>
</div>
<!--l. 622--><p class="noindent">where <!--l. 622--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="t1-zi4r-0-x-x-109">method</span><!--l. 622--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>
is the name of a specific procedure and <span class="t1-zi4r-0-x-x-109">arg1</span>, <span class="t1-zi4r-0-x-x-109">arg2</span>, …are its required and/or optional arguments.
</p><!--l. 632--><p class="indent">   The most frequent uses of the scripting interface are discussed and exemplified in this section. For the full
reference documentation of all procedures supported by the Colvars scripting interface, see section <a href="#x1-1410007">7<!--tex4ht:ref: sec:cvscript_tcl --></a> for the Tcl
command reference.
</p>
<div class="flushright">
<!--l. 636--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 636--><p class="noindent"></p><div class="heading"><a id="sec:cv_command_init">
<h5 class="subsubsectionHead"><span class="titlemark">3.3.1    </span> <a id="x1-100003.3.1"></a>Setting up the Colvars module</h5>
<!--l. 636--><p class="noindent"></p></a> </div> 
<!--l. 648--><p class="indent">   If the NAMD configuration parameter <span class="t1-zi4r-0-x-x-109">colvars </span>is on, the <span class="t1-zi4r-0-x-x-109">cv </span>Tcl command can be used anywhere in the NAMD
script, and will be invoked as soon as NAMD begins processing Tcl commands.
</p><!--l. 650--><p class="indent">   To define new collective variables and/or biases for immediate use in the current session, configuration can be
loaded from an external configuration file:</p>
<div class="mdframed" id="mdframed-4">
<!--l. 651--><p class="noindent"></p><pre>cv configfile "colvars-file.in" </pre>
</div>
<!--l. 652--><p class="indent">   This can in principle be called at any time, if only flags internal to Colvars are being modified. In practice, when
new atoms or any new atomic properties (e.g. total forces are being requested), initialization steps are required that
are not carried out during a simulation. Therefore, it is good practice to change the Colvars configuration outside the
scope between segments of the same simulation.
</p><!--l. 656--><p class="indent">   To load the configuration directly from a string the “<span class="t1-zi4r-0-x-x-109">config</span>" method may be used:</p>
<div class="mdframed" id="mdframed-5">
<!--l. 657--><p class="noindent"></p><pre>cv config "keyword { ... }" </pre>
</div>
<!--l. 658--><p class="noindent">This method is particularly useful to dynamically define the Colvars configuration within a NAMD script. For
example, when running an ensemble of umbrella sampling simulations in NAMD, it may be convenient to use an
identical NAMD script, and let the queuing system assist in defining the window. In this example, in a Slurm array
job an environment variable is used to define the window's numeric index (starting at zero), and the umbrella
                                                                                         
                                                                                         
restraint center (starting at 2 for the first window, and increasing in increments of 0.25 for all other
windows):</p>
<div class="mdframed" id="mdframed-6">
<!--l. 671--><p class="noindent"></p><pre>cv configfile colvar-definition.in<br class="newline"/>set window $env(SLURM_ARRAY_TASK_ID)<br class="newline"/>cv config "harmonic {<br class="newline"/>  name us_${window}<br class="newline"/>  colvars xi<br class="newline"/>  centers [expr 2.0 + 0.25 * ${window}]<br class="newline"/>  ...<br class="newline"/>}" </pre>
</div>
<div class="flushright">
<!--l. 676--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 676--><p class="noindent"></p><div class="heading"><a id="sec:cv_command_version">
<h5 class="subsubsectionHead"><span class="titlemark">3.3.2    </span> <a id="x1-110003.3.2"></a>Using the Colvars version in scripts</h5>
<!--l. 676--><p class="noindent"></p></a> </div> 
<!--l. 678--><p class="indent">   The vast majority of the syntax in Colvars is backward-compatible, adding keywords when new features are
introduced. However, when using multiple versions simultaneously it may be useful to test within the script whether
the version is recent enough to support the desired feature. The “<span class="t1-zi4r-0-x-x-109">version</span>" can be used to get the Colvars version for
this use:</p>
<div class="mdframed" id="mdframed-7">
<!--l. 681--><p class="noindent"></p><pre>if { [cv version] &gt;= "2020-02-25" } {<br class="newline"/>  cv config "(use a recent feature)"<br class="newline"/>} </pre>
</div>
<div class="flushright">
<!--l. 687--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 687--><p class="noindent"></p><div class="heading"><a id="sec:cv_command_loadsave">
<h5 class="subsubsectionHead"><span class="titlemark">3.3.3    </span> <a id="x1-120003.3.3"></a>Loading and saving the Colvars state and other information</h5>
<!--l. 687--><p class="noindent"></p></a> </div> 
<!--l. 689--><p class="indent">   After a configuration is fully defined, the “<span class="t1-zi4r-0-x-x-109">load</span>" method may be used to load a state file from a previous
simulation that contains e.g. data from history-dependent biases), to either continue that simulation or analyze its
results:
                                                                                         
                                                                                         
</p>
<div class="mdframed" id="mdframed-8">
<!--l. 690--><p class="noindent"></p><pre>cv load "<!--l. 690--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>oldjob<!--l. 690--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>.colvars.state"
</pre>
</div>
<!--l. 691--><p class="noindent">or more simply using the prefix of the state file itself:</p>
<div class="mdframed" id="mdframed-9">
<!--l. 692--><p class="noindent"></p><pre>cv load "<!--l. 692--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>oldjob<!--l. 692--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>"
</pre>
</div>
<!--l. 699--><p class="noindent">The latter version is particularly convenient in combination with the NAMD <span class="t1-zi4r-0-x-x-109">reinitatoms </span>command, for
example:<br class="newline"/></p>
<div class="mdframed" id="mdframed-10">
<!--l. 699--><p class="noindent"></p><pre>reinitatoms <!--l. 699--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>oldjob<!--l. 699--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><br class="newline"/>cv load <!--l. 699--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>oldjob<!--l. 699--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>
</pre>
</div>
<!--l. 699--><p class="indent">   The step number contained by the loaded file will be used internally by Colvars to control time-dependent
biases, unless <span class="t1-zi4r-0-x-x-109">firstTimestep </span>is issued, in which case that value will be used.
</p><!--l. 709--><p class="indent">   When the system's topology is changed during simulation via the <span class="t1-zi4r-0-x-x-109">structure </span>command (e.g. in constant-pH
simulations), it is generally best to reset and re-initalize the module from scratch before loading the corresponding
snapshot:</p>
<div class="mdframed" id="mdframed-11">
<!--l. 709--><p class="noindent"></p><pre>structure newsystem.psf<br class="newline"/>reinitatoms <!--l. 709--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>snapshot<!--l. 709--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><br class="newline"/>cv reset<br class="newline"/>cv configfile ...<br class="newline"/>cv load <!--l. 709--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>snapshot<!--l. 709--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>
</pre>
</div>
<!--l. 711--><p class="indent">   The “<span class="t1-zi4r-0-x-x-109">save</span>" method, analogous to “<span class="t1-zi4r-0-x-x-109">load</span>", allows to save all restart information to a state file. This is normally
not required during a simulation if <a href="#Colvars-global|colvarsRestartFrequency"><span class="t1-zi4r-0-x-x-109">colvarsRestartFrequency</span></a> is defined (either directly or indirectly by the
NAMD restart frequency). Because not only a state file (used to continue simulations) but also other data files (used
to analyze the trajectory) are written, it is generally recommended to call the <span class="t1-zi4r-0-x-x-109">save </span>method using a prefix, rather than
a complete file name:</p>
<div class="mdframed" id="mdframed-12">
<!--l. 714--><p class="noindent"></p><pre>cv save "<!--l. 714--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>job<!--l. 714--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>"
</pre>
</div>
<div class="flushright">
<!--l. 716--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 716--><p class="noindent"></p><div class="heading"><a id="sec:cv_command_atomic_data">
<h5 class="subsubsectionHead"><span class="titlemark">3.3.4    </span> <a id="x1-130003.3.4"></a>Accessing atomic data</h5>
<!--l. 716--><p class="noindent"></p></a> </div> 
<!--l. 718--><p class="indent">   For computational efficiency the Colvars module keeps internal copies of the numeric IDs, masses, charges,
positions, and optionally total forces of the atoms requested for a Colvars computation. At each simulation step,
up-to-date versions of these properties are copied from central memory of NAMD into the internal memory of the
Colvars module. In a post-processing workflow or outside a simulation (e.g. when using VMD), this copy can be
carried out as part of the <span class="t1-zi4r-0-x-x-109">update </span>method:</p>
<div class="mdframed" id="mdframed-13">
<!--l. 721--><p class="noindent"></p><pre>cv update   </pre>
</div>
<!--l. 722--><p class="noindent">which also performs the (re-)computation of all variables and biases defined in Colvars.
</p><!--l. 724--><p class="indent">   For example, the current sequence of numeric IDs of the atoms processed by Colvars can be obtained
as:</p>
<div class="mdframed" id="mdframed-14">
<!--l. 725--><p class="noindent"></p><pre>cv getatomids   </pre>
</div>
<!--l. 726--><p class="noindent">and their current positions as:</p>
<div class="mdframed" id="mdframed-15">
<!--l. 727--><p class="noindent"></p><pre>cv getatompositions   </pre>
</div>
<!--l. 728--><p class="noindent">This may prove useful to test the correctness of the coordinates passed to Colvars, particularly in regard to periodic
boundary conditions (<a href="#x1-1050005.3">5.3<!--tex4ht:ref: sec:colvar_atom_groups_wrapping --></a>). There is currently no mechanism to modify the above fields via the scripting interface,
but such capability will be added in the future.
</p><!--l. 732--><p class="indent">   A special case are collective variables that are based on <span class="ptmri8t-x-x-109">centers of mass</span>, or <span class="ptmri8t-x-x-109">volumetric maps </span>(see e.g. <a href="#colvar|mapTotal"><span class="t1-zi4r-0-x-x-109">mapTotal</span></a>).
These are often computed by NAMD itself, and any atoms that are used only for such variables will not be included
in the lists above.
</p><!--l. 734--><p class="indent">   While running a simulation, or when setting one up in VMD, it is possible to examine all the forces that were
last applied by Colvars to the atoms, or are about to be applied:</p>
<div class="mdframed" id="mdframed-16">
<!--l. 735--><p class="noindent"></p><pre>cv getatomappliedforces   </pre>
</div>
<!--l. 736--><p class="noindent">where the length and order of this sequence matches that provided by the <span class="t1-zi4r-0-x-x-109">getatomids </span>method. A simpler way of
testing the stability of a Colvars configuration before or during a simulation makes use of aggregated data, such as
the energy:</p>
<div class="mdframed" id="mdframed-17">
<!--l. 738--><p class="noindent"></p><pre>cv getenergy   </pre>
</div>
<!--l. 739--><p class="noindent">the root-mean-square of the Colvars applied forces:</p>
<div class="mdframed" id="mdframed-18">
<!--l. 740--><p class="noindent"></p><pre>cv getatomappliedforcesrms   </pre>
</div>
<!--l. 741--><p class="noindent">or the maximum norm of the applied forces:</p>
<div class="mdframed" id="mdframed-19">
<!--l. 742--><p class="noindent"></p><pre>cv getatomappliedforcesmax   </pre>
</div>
<!--l. 743--><p class="noindent">which can be matched to a specific atom via its numeric ID obtained as:</p>
<div class="mdframed" id="mdframed-20">
<!--l. 744--><p class="noindent"></p><pre>cv getatomappliedforcesmaxid   </pre>
</div>
<!--l. 746--><p class="indent">   See <a href="#x1-1420007.1">7.1<!--tex4ht:ref: sec:cvscript_tcl_cv --></a> for a complete list of scripting commands used to manage atomic data and runtime parameters of he
Colvars module.
</p>
<div class="flushright">
<!--l. 775--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 775--><p class="noindent"></p><div class="heading"><a id="sec:cv_command_colvar">
<h5 class="subsubsectionHead"><span class="titlemark">3.3.5    </span> <a id="x1-140003.3.5"></a>Managing collective variables</h5>
<!--l. 775--><p class="noindent"></p></a> </div> 
<!--l. 777--><p class="indent">   After one or more collective variables are defined, they can be accessed with the following syntax.</p>
<div class="mdframed" id="mdframed-21">
<!--l. 778--><p class="noindent"></p><pre>cv colvar "<!--l. 778--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>name<!--l. 778--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>" <!--l. 778--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ptmri8t-x-x-109">method</span><!--l. 778--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math> arg1 arg2 ...
</pre>
</div>
<!--l. 779--><p class="noindent">where “<!--l. 779--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="t1-zi4r-0-x-x-109">name</span><!--l. 779--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>"
is the name of the variable.
</p><!--l. 781--><p class="indent">   For example, to recompute the collective variable “<span class="t1-zi4r-0-x-x-109">xi</span>" after a change in its parameters, the following command
can be used:</p>
<div class="mdframed" id="mdframed-22">
<!--l. 782--><p class="noindent"></p><pre>cv colvar "xi" update  </pre>
</div>
<!--l. 783--><p class="noindent">This ordinarily is not needed during a simulation run, where all variables are recomputed at every step (along with
biasing forces acting on them). However, when analyzing an existing trajectory, e.g. in VMD, a call to <span class="t1-zi4r-0-x-x-109">update </span>is
generally required.
</p><!--l. 786--><p class="indent">   While in all typical cases all configuration of the variables is done with the “<span class="t1-zi4r-0-x-x-109">config</span>" or “<span class="t1-zi4r-0-x-x-109">configfile</span>" methods,
a limited set of changes can be enacted at runtime using:</p>
<div class="mdframed" id="mdframed-23">
<!--l. 787--><p class="noindent"></p><pre>cv colvar "<!--l. 787--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>name<!--l. 787--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>" modifycvcs arg1 arg2 ...
</pre>
</div>
<!--l. 788--><p class="noindent">where each argument is a string passed to the function or functions that are used to compute the variable,
and are called colvar components, or CVCs (<a href="#x1-300004.1">4.1<!--tex4ht:ref: sec:cvc_list --></a>). For example, a variable “<span class="t1-zi4r-0-x-x-109">DeltaZ</span>" made of a single
“<span class="t1-zi4r-0-x-x-109">distanceZ</span>" component can be made periodic with a period equal to the unit cell dimension along the
<!--l. 789--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>Z</mi></math>-axis:
                                                                                         
                                                                                         
</p>
<div class="mdframed" id="mdframed-24">
<!--l. 790--><p class="noindent"></p><pre>cv colvar "DeltaZ" modifycvcs "period <span class="ptmri8t-x-x-109">Lz</span>" </pre>
</div>
<!--l. 792--><p class="noindent">Please note that this option is currently limited to changing the values of the polynomial superposition parameters
<a href="#sec:cvc_superp"><span class="t1-zi4r-0-x-x-109">componentCoeff</span></a>, or of the <a href="#sec:cvc_superp"><span class="t1-zi4r-0-x-x-109">componentExp</span></a> to update on the fly, of <a href="#sec:cvc_periodic"><span class="t1-zi4r-0-x-x-109">period</span></a>, <a href="#sec:cvc_periodic"><span class="t1-zi4r-0-x-x-109">wrapAround</span></a> or <a href="#colvar|distance|forceNoPBC"><span class="t1-zi4r-0-x-x-109">forceNoPBC</span></a> options for
components that support them.
</p><!--l. 794--><p class="indent">   If the variable is computed using many components, it is possible to selectively turn some of them on or
off:</p>
<div class="mdframed" id="mdframed-25">
<!--l. 795--><p class="noindent"></p><pre>cv colvar "<!--l. 795--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>name<!--l. 795--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>" cvcflags <!--l. 795--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>flags<!--l. 795--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>
</pre>
</div>
<!--l. 796--><p class="noindent">where “<!--l. 796--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="t1-zi4r-0-x-x-109">flags</span><!--l. 796--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>"
is a list of 0/1 values, one per component. This is useful for example when script-based path collective variables in
Cartesian coordinates (<a href="#x1-810004.10.3">4.10.3<!--tex4ht:ref: sec:apath_scripted --></a>) are used, to minimize computational cost by disabling the computation of terms that
are very close to zero.
</p><!--l. 799--><p class="indent"> <span class="ptmb8t-x-x-109">Important: </span>None of the changes enacted by the “<span class="t1-zi4r-0-x-x-109">modifycvcs</span>" or “<span class="t1-zi4r-0-x-x-109">cvcflags</span>" methods will be saved to state
files, and will be lost when restarting a simulation, deleting the corresponding collective variable, or resetting the
module with the  “<span class="t1-zi4r-0-x-x-109">reset</span>" method.
</p>
<div class="flushright">
<!--l. 801--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 801--><p class="noindent"></p><div class="heading"><a id="sec:cv_command_colvar_force">
<h5 class="subsubsectionHead"><span class="titlemark">3.3.6    </span> <a id="x1-150003.3.6"></a>Applying and analyzing forces on collective variables</h5>
<!--l. 801--><p class="noindent"></p></a> </div> 
<!--l. 803--><p class="indent">   As soon as a collective variable is up to date (during a MD run or after its “<span class="t1-zi4r-0-x-x-109">update</span>" method has been called),
forces can be applied to it, e.g. as part of a custom restraint implemented by <a href="#Colvars-global|scriptedColvarForces"><span class="t1-zi4r-0-x-x-109">scriptedColvarForces</span></a>:</p>
<div class="mdframed" id="mdframed-26">
<!--l. 804--><p class="noindent"></p><pre>cv colvar "xi" addforce <!--l. 804--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>force<!--l. 804--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>
</pre>
</div>
<!--l. 805--><p class="noindent">where “<!--l. 805--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="t1-zi4r-0-x-x-109">force</span><!--l. 805--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>"
is a scalar or a vector (depending on the type of variable “<span class="t1-zi4r-0-x-x-109">xi</span>") and is defined by the user's function. The force will
be physically applied to the corresponding atoms during the simulation after Colvars communicates all
forces to the rest of NAMD.  Until then, the total force applied to “<span class="t1-zi4r-0-x-x-109">xi</span>" from all biases can be retrieved
by:</p>
<div class="mdframed" id="mdframed-27">
<!--l. 809--><p class="noindent"></p><pre>cv colvar "xi" getappliedforce  </pre>
</div>
<!--l. 810--><p class="noindent">(see also the use of the <a href="#colvar|outputAppliedForce"><span class="t1-zi4r-0-x-x-109">outputAppliedForce</span></a> option).
                                                                                         
                                                                                         
</p><!--l. 812--><p class="indent">   To obtain the total force projected on the variable “<span class="t1-zi4r-0-x-x-109">xi</span>":</p>
<div class="mdframed" id="mdframed-28">
<!--l. 813--><p class="noindent"></p><pre>cv colvar "xi" gettotalforce  </pre>
</div>
<!--l. 814--><p class="noindent">Note that not all types of variable support this option, and the value of the total force may not be available
immediately: see <a href="#colvar|outputTotalForce"><span class="t1-zi4r-0-x-x-109">outputTotalForce</span></a> for more details.
</p><!--l. 816--><p class="indent">   See <a href="#x1-1430007.2">7.2<!--tex4ht:ref: sec:cvscript_tcl_colvar --></a> for a complete list of scripting commands used to manage collective variables.
</p>
<div class="flushright">
<!--l. 819--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 819--><p class="noindent"></p><div class="heading"><a id="sec:cv_command_bias">
<h5 class="subsubsectionHead"><span class="titlemark">3.3.7    </span> <a id="x1-160003.3.7"></a>Managing collective variable biases</h5>
<!--l. 819--><p class="noindent"></p></a> </div> 
<!--l. 821--><p class="indent">   Because biases depend only upon data internal to the Colvars module (i.e. they do not need atomic coordinates
from NAMD), it is generally easy to create them or update their configuration at any time. For example, given the
most current value of the variable “<span class="t1-zi4r-0-x-x-109">xi</span>", an already-defined harmonic restraint on it named “<span class="t1-zi4r-0-x-x-109">h_xi</span>" can be updated
as:</p>
<div class="mdframed" id="mdframed-29">
<!--l. 823--><p class="noindent"></p><pre>cv bias "h_xi" update  </pre>
</div>
<!--l. 824--><p class="noindent">During a running simulation this step is not needed, because an automatic update of each bias is already carried
out.
</p><!--l. 849--><p class="indent">   Another circumstance when “<span class="t1-zi4r-0-x-x-109">update</span>" may be called e.g. as part of the function invoked by
<a href="#Colvars-global|scriptedColvarForces"><span class="t1-zi4r-0-x-x-109">scriptedColvarForces</span></a>, it is executed before any biasing forces are applied to the variables, thus allowing to
modify them. This use of “<span class="t1-zi4r-0-x-x-109">update</span>" is often used e.g. in the definition of custom bias-exchange algorithms as part of
the NAMD script. Because a restraint is a relatively light-weight object, the easiest way to change the configuration
of an existing bias is deleting it and re-creating it:</p>
<div class="mdframed" id="mdframed-30">
<!--l. 849--><p class="noindent"></p><pre><span class="ptmb8t-x-x-109">#</span><span class="ptmb8t-x-x-109"> Delete</span><span class="ptmb8t-x-x-109"> the</span><span class="ptmb8t-x-x-109"> restraint</span><span class="ptmb8t-x-x-109"> "harmonic_xi"</span><br class="newline"/>cv bias harmonic_xi delete<br class="newline"/><span class="ptmb8t-x-x-109">#</span><span class="ptmb8t-x-x-109"> Re-define</span><span class="ptmb8t-x-x-109"> it,</span><span class="ptmb8t-x-x-109"> but</span><span class="ptmb8t-x-x-109"> using</span><span class="ptmb8t-x-x-109"> an</span><span class="ptmb8t-x-x-109"> updated</span><span class="ptmb8t-x-x-109"> restraint</span><span class="ptmb8t-x-x-109"> center</span><br class="newline"/>cv config "harmonic {<br class="newline"/>  name harmonic_xi<br class="newline"/>  centers ${new_center}]<br class="newline"/>  ...<br class="newline"/>}"<br class="newline"/><span class="ptmb8t-x-x-109">#</span><span class="ptmb8t-x-x-109"> Now</span><span class="ptmb8t-x-x-109"> update</span><span class="ptmb8t-x-x-109"> it</span><span class="ptmb8t-x-x-109"> (based</span><span class="ptmb8t-x-x-109"> on</span><span class="ptmb8t-x-x-109"> the</span><span class="ptmb8t-x-x-109"> current</span><span class="ptmb8t-x-x-109"> value</span><span class="ptmb8t-x-x-109"> of</span><span class="ptmb8t-x-x-109"> "xi")</span><br class="newline"/>cv bias harmonic_xi update </pre>
</div>
<!--l. 849--><p class="indent">   It is also possible to make the change subject to a condition on the energy of the new bias:
                                                                                         
                                                                                         
</p>
<div class="mdframed" id="mdframed-31">
<!--l. 849--><p class="noindent"></p><pre>...<br class="newline"/>cv bias harmonic_xi update<br class="newline"/>if { [cv bias harmonic_xi energy] &lt; ${E_accept} } {<br class="newline"/>    ...<br class="newline"/>} </pre>
</div>
<div class="flushright">
<!--l. 851--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 851--><p class="noindent"></p><div class="heading"><a id="sec:cv_command_bias_loadsave">
<h5 class="subsubsectionHead"><span class="titlemark">3.3.8    </span> <a id="x1-170003.3.8"></a>Loading and saving the state of individual biases</h5>
<!--l. 851--><p class="noindent"></p></a> </div> 
<!--l. 853--><p class="indent">   Some types of bias are history-dependent, and the magnitude of their forces depends not only on the values of
their corresponding variables, but also on previous simulation history. It is thus useful to load information from a
state file that contains information specifically for one bias only, for example:</p>
<div class="mdframed" id="mdframed-32">
<!--l. 855--><p class="noindent"></p><pre>cv bias "metadynamics1" load "old.colvars.state" </pre>
</div>
<!--l. 856--><p class="noindent">or alternatively, using the prefix of the file instead of its full name:</p>
<div class="mdframed" id="mdframed-33">
<!--l. 857--><p class="noindent"></p><pre>cv bias "metadynamics1" load "old" </pre>
</div>
<!--l. 858--><p class="indent">   A corresponding “<span class="t1-zi4r-0-x-x-109">save</span>" function is also available:</p>
<div class="mdframed" id="mdframed-34">
<!--l. 859--><p class="noindent"></p><pre>cv bias "metadynamics1" save "new" </pre>
</div>
<!--l. 860--><p class="noindent">This pair of functions is also used internally by Colvars to implement e.g. multiple-walker metadynamics (<a href="#x1-1260006.4.7">6.4.7<!--tex4ht:ref: sec:colvarbias_meta_mr --></a>), but
they can be called from a scripted function to implement alternative coupling schemes.
</p><!--l. 862--><p class="indent">   See <a href="#x1-1440007.3">7.3<!--tex4ht:ref: sec:cvscript_tcl_bias --></a> for a complete list of scripting commands used to manage biases.
</p>
<div class="flushright">
<!--l. 868--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 868--><p class="noindent"></p><div class="heading"><a id="sec:colvars_config_syntax">
<h4 class="subsectionHead"><span class="titlemark">3.4    </span> <a id="x1-180003.4"></a>Configuration syntax used by the Colvars module</h4>
<!--l. 868--><p class="noindent"></p></a> </div> 
<!--l. 870--><p class="indent">   Configuration for the Colvars module is passed using an external file, or inlined as a string in a NAMD script
using the Tcl command <span class="t1-zi4r-0-x-x-109">cv config "..."</span>. Configuration lines follow the format “<span class="t1-zi4r-0-x-x-109">keyword value</span>" or “<span class="t1-zi4r-0-x-x-109">keyword {</span>
<span class="t1-zi4r-0-x-x-109">... }</span>", where the keyword and its value must be separated by one or more space characters. The following
formatting rules apply:
</p>
<ul class="itemize1">
<li class="itemize"><span class="ptmri8t-x-x-109">Keywords  are  case-insensitive</span>;  for  example,  <span class="t1-zi4r-0-x-x-109">upperBoundary </span>is  the  same  as  <span class="t1-zi4r-0-x-x-109">upperboundary </span>and
     <span class="t1-zi4r-0-x-x-109">UPPERBOUNDARY</span>); note that their string values are however still case-sensitive (e.g. names of variables,
     file names).
     </li>
<li class="itemize">A long value, or a list of multiple values, can be distributed across multiple lines by using <span class="ptmri8t-x-x-109">curly braces</span>,
     “<span class="t1-zi4r-0-x-x-109">{</span>" and “<span class="t1-zi4r-0-x-x-109">}</span>": the opening brace “<span class="t1-zi4r-0-x-x-109">{</span>" must occur on the same line as the keyword, following at least one
     space character; the closing brace “<span class="t1-zi4r-0-x-x-109">}</span>" may be at any position after that; any keywords following the
     closing brace on the same line are not valid (they should appear instead on a different line).
     </li>
<li class="itemize">Many keywords are nested, and are only meaningful within the specific context of another keyword;
     for example, the keyword <span class="t1-zi4r-0-x-x-109">name </span>is available inside the block of the keyword <span class="t1-zi4r-0-x-x-109">colvar {...}</span>, but not
     outside of it; for every keyword documented in the following, the <span class="ptmri8t-x-x-109">“parent" keyword </span>that defines such
     context is also indicated.
     </li>
<li class="itemize">If  a  keyword  requiring  a  boolean  value  (<span class="t1-zi4r-0-x-x-109">yes|on|true </span>or  <span class="t1-zi4r-0-x-x-109">no|off|false</span>)  is  provided  without  an
     explicit value, it defaults to ‘<span class="t1-zi4r-0-x-x-109">yes|on|true</span>'; for example, ‘<span class="t1-zi4r-0-x-x-109">outputAppliedForce</span>' may be used as
     shorthand for ‘<span class="t1-zi4r-0-x-x-109">outputAppliedForce on</span>'.
     </li>
<li class="itemize">The hash character “<span class="t1-zi4r-0-x-x-109">#</span>" indicates a <span class="ptmri8t-x-x-109">comment</span>: all text in the same line following this character will be
     ignored.
     </li>
<li class="itemize">Outside  of  comments,  <span class="ptmri8t-x-x-109">only  ASCII  characters  </span>are  allowed  for  defining  keywords,  and  the  only
     white-space  characters  supported  are  spaces,  tabs  and  newlines:  a  warning  will  be  printed  upon
     detection  of  non-ASCII  characters  in  a  configuration  line,  which  include  both  characters  that  are
     visibly  “special",  as  well  as  those  with  a  very  similar  appearance  to  ASCII  ones  (for  instance,
     <a href="https://en.wikipedia.org/wiki/Non-breaking_space">non-breaking spaces</a>); common ways to identify/remove non-ASCII characters are using the Emacs
     text editor, or using <span class="t1-zi4r-0-x-x-109">LC_ALL=C vi</span>.
     </li>
<li class="itemize">Tcl syntax is generally not available inside a Colvars configuration file/string, but it is possible to
     use Tcl variables or bracket expansion of commands when configuration is passed via the command
     <span class="t1-zi4r-0-x-x-109">cv config "..."</span>; for example, this is particularly useful when combined with NAMD's parameter
     introspection,  e.g.  <span class="t1-zi4r-0-x-x-109">cv  config  "colvarsTrajFrequency  [DCDFreq]" </span>allows  synchronizing  the
     output frequencies of NAMD and Colvars trajectory files. .
                                                                                         
                                                                                         
     </li></ul>
<div class="flushright">
<!--l. 895--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 895--><p class="noindent"></p><div class="heading"><a id="sec:colvars_global">
<h4 class="subsectionHead"><span class="titlemark">3.5    </span> <a id="x1-190003.5"></a>Global keywords</h4>
<!--l. 895--><p class="noindent"></p></a> </div> 
<!--l. 897--><p class="indent">   The following keywords are available in the global context of the Colvars configuration, i.e. they are not nested
inside other keywords: </p>
<ul class="itemize1">
<li class="itemize"><a id="Colvars-global|colvarsTrajFrequency"> </a> <a id="dx1-19001"></a><span class="colorbox" id="colorbox6"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">colvarsTrajFrequency</span></span> <!--l. 910--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Colvar
     value trajectory frequency<!--l. 910--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>global <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive integer <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span>100 <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>The values of each colvar (and of other related quantities, if requested) are written to
     the file <span class="ptmri8t-x-x-109">outputName</span><span class="t1-zi4r-0-x-x-109">.colvars.traj </span>every these many steps throughout the simulation. If the value is
     0, such trajectory file is not written. For optimization the output is buffered, and synchronized with
     the disk only when the restart file is being written.
     </li>
<li class="itemize"><a id="Colvars-global|colvarsRestartFrequency"> </a> <a id="dx1-19002"></a><span class="colorbox" id="colorbox7"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">colvarsRestartFrequency</span></span> <!--l. 923--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Colvar
     module restart frequency<!--l. 923--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>global <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive integer <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span>NAMD parameter <span class="t1-zi4r-0-x-x-109">restartFreq</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>The state file and any other output files produced by Colvars are written every these
     many steps (the trajectory file is still written every <a href="#Colvars-global|colvarsTrajFrequency"><span class="t1-zi4r-0-x-x-109">colvarsTrajFrequency</span></a> steps).  It is generally a
     good idea to leave this parameter at its default value, unless needed for special cases or to disable
     automatic writing of output files altogether. Writing can still be invoked at any time via the command
     <span class="t1-zi4r-0-x-x-109">cv save</span>.
     </li>
<li class="itemize"><a id="Colvars-global|indexFile"> </a> <a id="dx1-19003"></a><span class="colorbox" id="colorbox8"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">indexFile</span></span> <!--l. 942--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Index
     file for atom selection (GROMACS “ndx" format)<!--l. 942--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>global <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>UNIX filename <br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>This option reads an index file (usually with a <span class="t1-zi4r-0-x-x-109">.ndx </span>extension) as produced by the
     <span class="t1-zi4r-0-x-x-109">make_ndx </span>tool of GROMACS. This keyword may be repeated to load multiple index files. A group
     with the same name may appear multiple times, as long as it contains the same indices in identical
     order each time: an error is raised otherwise.   The names of index groups contained in this file can
     then be used to define atom groups with the <span class="t1-zi4r-0-x-x-109">indexGroup </span>keyword. Other supported methods to select
     atoms are described in <a href="#x1-1000005">5<!--tex4ht:ref: sec:colvar_atom_groups --></a>.
                                                                                         
                                                                                         
     </li>
<li class="itemize"><a id="Colvars-global|smp"> </a> <a id="dx1-19004"></a><span class="colorbox" id="colorbox9"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">smp</span></span> <!--l. 952--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Whether
     SMP parallelism should be used<!--l. 952--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>global <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">on</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>If this flag is enabled (default), SMP parallelism over threads will be used to compute
     variables and biases, provided that this is supported by the NAMD build in use.
     </li></ul>
<div class="flushright">
<!--l. 957--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 957--><p class="noindent"></p><div class="heading"><a id="sec:colvars_input">
<h4 class="subsectionHead"><span class="titlemark">3.6    </span> <a id="x1-200003.6"></a>Input state file</h4>
<!--l. 957--><p class="noindent"></p></a> </div> 
<!--l. 959--><p class="indent">   Because many of the methods implemented in Colvars are history-dependent, a <span class="ptmri8t-x-x-109">state file </span>is often needed to
continue a long simulation over consecutive runs. Such state file is written automatically at the end of any simulation
with Colvars, and contains data accumulated during that simulation along with the step number at the end of it. The
step number read from the state file is then used to control such time-dependent biases: because of this
essential role, the step number internal to Colvars may not always match the step number reported by the
MD program that carried during the simulation (which may instead restart from zero each time). If a
state file is not given, the NAMD command <span class="t1-zi4r-0-x-x-109">firstTimestep </span>may be used to control the Colvars step
number.
</p><!--l. 964--><p class="indent">   Depending on the configuration, a state file may need to be loaded issued at the beginning of a new simulation
when time-dependent biasing methods are applied (moving restraints, metadynamics, ABF, ...). When the
Colvars module is initialized in NAMD, the <span class="t1-zi4r-0-x-x-109">colvarsInput </span>keyword can be used to give the name of
the state file.  After initialization, a state file may be loaded at any time with the Tcl command <span class="t1-zi4r-0-x-x-109">cv</span>
<span class="t1-zi4r-0-x-x-109">load</span>.
</p><!--l. 971--><p class="indent">   It is possible to load a state file even if the configuration has changed: for example, new variables
may be defined or restraints be added in between consecutive runs. For each newly defined variable or
bias, no information will be read from the state file if this is unavailable: such new objects will remain
uninitialized until the first compute step. Conversely, any information that the state file has about variables
or biases that are not defined any longer is silently ignored. <span class="ptmri8t-x-x-109">Because these checks are done by the</span>
<span class="ptmri8t-x-x-109">names of variables or biases, it is the user's responsibility to ensure that these are consistent between</span>
<span class="ptmri8t-x-x-109">runs.</span>
</p>
<div class="flushright">
<!--l. 979--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 979--><p class="noindent"></p><div class="heading"><a id="sec:colvars_output">
<h4 class="subsectionHead"><span class="titlemark">3.7    </span> <a id="x1-210003.7"></a>Output files</h4>
<!--l. 979--><p class="noindent"></p></a> </div> 
<!--l. 981--><p class="indent">   During a simulation with collective variables defined, the following three output files are written:
</p>
<ul class="itemize1">
<li class="itemize">A <span class="ptmri8t-x-x-109">state file</span>, named <span class="ptmri8t-x-x-109">outputName</span><span class="t1-zi4r-0-x-x-109">.colvars.state</span>; this file is in ASCII (plain text) format, regardless of
     the value of <span class="t1-zi4r-0-x-x-109">binaryOutput </span>in the NAMD configuration. This file is written at the end of the specified
     run, but can also be written at any time with the command <span class="t1-zi4r-0-x-x-109">cv save </span>(<a href="#x1-120003.3.3">3.3.3<!--tex4ht:ref: sec:cv_command_loadsave --></a>).<br class="newline"/><span class="ptmri8t-x-x-109">This is the only Colvars output file needed to continue a simulation.</span>
</li>
<li class="itemize">If  the  parameter  <a href="#Colvars-global|colvarsRestartFrequency"><span class="t1-zi4r-0-x-x-109">colvarsRestartFrequency</span></a> is  larger  than  zero,  a  <span class="ptmri8t-x-x-109">restart  file  </span>is  written  every
     that  many  steps:  this  file  is  fully  equivalent  to  the  final  state  file.  The  name  of  this  file  is
     <span class="ptmri8t-x-x-109">restartName</span><span class="t1-zi4r-0-x-x-109">.colvars.state</span>.
     </li>
<li class="itemize">If the parameter <a href="#Colvars-global|colvarsTrajFrequency"><span class="t1-zi4r-0-x-x-109">colvarsTrajFrequency</span></a> is greater than 0 (default value: 100 steps), a <span class="ptmri8t-x-x-109">trajectory file</span>,
     named <span class="ptmri8t-x-x-109">outputName</span><span class="t1-zi4r-0-x-x-109">.colvars.traj</span>, is written during the simulation. Unlike the state file, this file is
     not needed to restart a simulation, but can be used for post-processing and analysis. The format of this
     file is described in sec. <a href="#x1-280003.8.6">3.8.6<!--tex4ht:ref: sec:colvars_traj_format --></a>.
     </li>
<li class="itemize">Additionally,  certain  features,  when  enabled,  can  emit  output  files  with  a  specific  purpose:  for
     example, potentials of mean force can be written to file to be analyzed or plotted. These files are
     described  in  the  respective  sections,  but  as  a  general  rule  they  all  use  names  beginning  with  the
     <span class="ptmri8t-x-x-109">outputName</span>prefix.
     </li></ul>
<!--l. 997--><p class="indent">   Other output files may also be written by specific methods, e.g. the ABF or metadynamics methods (<a href="#x1-1090006.2">6.2<!--tex4ht:ref: sec:colvarbias_abf --></a>, <a href="#x1-1190006.4">6.4<!--tex4ht:ref: sec:colvarbias_meta --></a>).
Like the trajectory file, they are needed only for analyzing, not continuing a simulation. All such files' names also
begin with the prefix <span class="ptmri8t-x-x-109">outputName</span>.
</p><!--l. 1001--><p class="indent">   Lastly, the total energy of all biases or restraints applied to the colvars appears under the NAMD standard
output, under the MISC column.
</p>
<div class="flushright">
<!--l. 1004--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1004--><p class="noindent"></p><div class="heading"><a id="sec:colvars_file_formats">
<h4 class="subsectionHead"><span class="titlemark">3.8    </span> <a id="x1-220003.8"></a>File formats</h4>
<!--l. 1004--><p class="noindent"></p></a> </div> 
<div class="flushright">
<!--l. 1006--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1006--><p class="noindent"></p><div class="heading"><a id="sec:colvars_text_format">
<h5 class="subsubsectionHead"><span class="titlemark">3.8.1    </span> <a id="x1-230003.8.1"></a>Configuration and state files.</h5>
<!--l. 1006--><p class="noindent"></p></a> </div> 
<!--l. 1008--><p class="indent"> <span class="ptmri8t-x-x-109">Configuration files </span>are text files that are generally read as input by NAMD, and may be optionally inlined in a
NAMD script (see <a href="#x1-100003.3.1">3.3.1<!--tex4ht:ref: sec:cv_command_init --></a>). Starting from version 2017-02-01, changes in newline encodings are handled
transparently, i.e. it is possible to typeset a configuration file in Windows (CR-LF newlines) and then use it with
Linux or macOS (LF-only newlines).
</p><!--l. 1011--><p class="indent"> <span class="ptmri8t-x-x-109">State files</span>, although not written manually, follow otherwise the same text format as configuration
files.
</p>
<div class="flushright">
<!--l. 1014--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1014--><p class="noindent"></p><div class="heading"><a id="sec:colvars_ndx_format">
<h5 class="subsubsectionHead"><span class="titlemark">3.8.2    </span> <a id="x1-240003.8.2"></a>Index (NDX) files</h5>
<!--l. 1014--><p class="noindent"></p></a> </div> 
<!--l. 1016--><p class="indent">   For atom selections that cannot be specified only by using internal Colvars keywords, external <span class="ptmri8t-x-x-109">index files </span>may
also be used following the <a href="http://manual.gromacs.org/documentation/current/reference-manual/file-formats.html#ndx">NDX format</a> used in GROMACS:</p>
<div class="mdframed" id="mdframed-35">
<!--l. 1017--><p class="noindent"></p><pre>[ group_1_name ]<br class="newline"/>  i1  i2  i3  i4  ...<br class="newline"/>  ...             ...  iN<br class="newline"/>[ group_2_name ]<br class="newline"/>  ... </pre>
</div>
<!--l. 1024--><p class="noindent">where <span class="ptmri8t-x-x-109">i1 </span>through <span class="ptmri8t-x-x-109">iN </span>are 1-based indices. Each group name may not contain spaces or tabs: otherwise, a parsing error
will be raised.
</p><!--l. 1026--><p class="indent">   Multiple index files may be provided to Colvars, each using the keyword <a href="#Colvars-global|indexFile"><span class="t1-zi4r-0-x-x-109">indexFile</span></a>. Two index files may
contain groups with the same names, however these must also represent identical atom selections, i.e. the same
sequence of indices including order.
</p><!--l. 1030--><p class="indent">   Other than with GROMACS, an index group may also be generated from the VMD command-line interface,
using the helper function <span class="t1-zi4r-0-x-x-109">write_index_group </span>provided in the <span class="t1-zi4r-0-x-x-109">colvartools </span>folder:
                                                                                         
                                                                                         
</p>
<div class="mdframed" id="mdframed-36">
<!--l. 1031--><p class="noindent"></p><pre>source colvartools/write_index_group.tcl<br class="newline"/>set sel [atomselect top "resname XXX and not hydrogen"]<br class="newline"/>write_index_group indexfile.ndx $sel "Ligand" </pre>
</div>
<div class="flushright">
<!--l. 1039--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1039--><p class="noindent"></p><div class="heading"><a id="sec:colvars_xyz_format">
<h5 class="subsubsectionHead"><span class="titlemark">3.8.3    </span> <a id="x1-250003.8.3"></a>XYZ coordinate files</h5>
<!--l. 1039--><p class="noindent"></p></a> </div>  XYZ coordinate files are text files, which are read by the Colvars module using an internal reader, and expect
the following format:
<!--l. 1042--><p class="noindent"><!--tex4ht:inline--></p><div class="tabular"> <table cellpadding="0" cellspacing="0" class="tabular" id="TBL-1"><colgroup id="TBL-1-1g"><col id="TBL-1-1"/><col id="TBL-1-2"/><col id="TBL-1-3"/><col id="TBL-1-4"/></colgroup><tr id="TBL-1-1-" style="vertical-align:baseline;"><td class="td11" id="TBL-1-1-1" style="text-align:right; white-space:nowrap;"> <!--l. 1043--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>N</mi></math> </td></tr><tr id="TBL-1-2-" style="vertical-align:baseline;"><td class="td11" id="TBL-1-2-1" style="text-align:right; white-space:nowrap;"></td><td class="td11" id="TBL-1-2-2" style="text-align:right; white-space:nowrap;"> <span class="ptmri8t-x-x-109">Comment  </span></td><td class="td11" id="TBL-1-2-3" style="text-align:right; white-space:nowrap;"> <span class="ptmri8t-x-x-109">line  </span></td><td class="td11" id="TBL-1-2-4" style="text-align:right; white-space:nowrap;"> </td>
</tr><tr id="TBL-1-3-" style="vertical-align:baseline;"><td class="td11" id="TBL-1-3-1" style="text-align:right; white-space:nowrap;"> <!--l. 1045--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>E</mi></mrow><mrow><mn>1</mn></mrow></msub></math> </td><td class="td11" id="TBL-1-3-2" style="text-align:right; white-space:nowrap;"> <!--l. 1045--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>x</mi></mrow><mrow><mn>1</mn></mrow></msub></math> </td><td class="td11" id="TBL-1-3-3" style="text-align:right; white-space:nowrap;"> <!--l. 1045--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>y</mi></mrow><mrow><mn>1</mn></mrow></msub></math> </td><td class="td11" id="TBL-1-3-4" style="text-align:right; white-space:nowrap;"> <!--l. 1045--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>z</mi></mrow><mrow><mn>1</mn></mrow></msub></math> </td>
</tr><tr id="TBL-1-4-" style="vertical-align:baseline;"><td class="td11" id="TBL-1-4-1" style="text-align:right; white-space:nowrap;"> <!--l. 1046--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>E</mi></mrow><mrow><mn>2</mn></mrow></msub></math> </td><td class="td11" id="TBL-1-4-2" style="text-align:right; white-space:nowrap;"> <!--l. 1046--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>x</mi></mrow><mrow><mn>2</mn></mrow></msub></math> </td><td class="td11" id="TBL-1-4-3" style="text-align:right; white-space:nowrap;"> <!--l. 1046--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>y</mi></mrow><mrow><mn>2</mn></mrow></msub></math> </td><td class="td11" id="TBL-1-4-4" style="text-align:right; white-space:nowrap;"> <!--l. 1046--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>z</mi></mrow><mrow><mn>2</mn></mrow></msub></math> </td>
</tr><tr id="TBL-1-5-" style="vertical-align:baseline;"><td class="td11" id="TBL-1-5-1" style="text-align:right; white-space:nowrap;">                                                                                      …  </td>
</tr><tr id="TBL-1-6-" style="vertical-align:baseline;"><td class="td11" id="TBL-1-6-1" style="text-align:right; white-space:nowrap;"> <!--l. 1048--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>E</mi></mrow><mrow><mi>N</mi></mrow></msub></math> </td><td class="td11" id="TBL-1-6-2" style="text-align:right; white-space:nowrap;"> <!--l. 1048--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>x</mi></mrow><mrow><mi>N</mi></mrow></msub></math> </td><td class="td11" id="TBL-1-6-3" style="text-align:right; white-space:nowrap;"> <!--l. 1048--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>y</mi></mrow><mrow><mi>N</mi></mrow></msub></math> </td><td class="td11" id="TBL-1-6-4" style="text-align:right; white-space:nowrap;"> <!--l. 1048--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>z</mi></mrow><mrow><mi>N</mi></mrow></msub></math> </td>
</tr><tr id="TBL-1-7-" style="vertical-align:baseline;"><td class="td11" id="TBL-1-7-1" style="text-align:right; white-space:nowrap;"> </td></tr></table>
</div>
<!--l. 1051--><p class="noindent">where <!--l. 1051--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>N</mi></math> is the number of atomic
coordinates in the file and <!--l. 1051--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>E</mi></mrow><mrow><mi>i</mi></mrow></msub></math> is
the chemical element of the <!--l. 1051--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math>-th
atom. Because <!--l. 1052--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>E</mi></mrow><mrow><mi>i</mi></mrow></msub></math>
is not used in Colvars, any string that does not contain tabs or spaces is acceptable.
</p><!--l. 1054--><p class="indent">   An XYZ file may contain either one of the following scenarios:
     </p><ol class="enumerate1">
<li class="enumerate" id="x1-25001x3.8.3">The file contains as many coordinates as the atoms that they are being read for: all coordinates will
     be read from the file following the same order as the atoms appear in the selection generated using
     the keywords listed in section <a href="#x1-1000005">5<!--tex4ht:ref: sec:colvar_atom_groups --></a>. (Note that the order is guaranteed only if a single type of selection
     keyword is used one or more times, and not guaranteed when different types of selection keywords
     are used.)
     </li>
<li class="enumerate" id="x1-25002x3.8.3">The file contains more coordinates than needed, and it is assumed to contain coordinates for the entire
     system: only coordinates that match the numeric indices of the selected atoms are read, in order of
     increasing number.</li></ol>
<!--l. 1061--><p class="indent">   XYZ-file coordinates are read directly by Colvars and stored internally as double-precision floating point
numbers.
</p>
<div class="flushright">
<!--l. 1090--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1090--><p class="noindent"></p><div class="heading"><a id="sec:colvars_pdb_format">
<h5 class="subsubsectionHead"><span class="titlemark">3.8.4    </span> <a id="x1-260003.8.4"></a>PDB coordinate files</h5>
<!--l. 1090--><p class="noindent"></p></a> </div>  PDB coordinate files are read by the Colvars module using existing functionality in NAMD, and therefore
follow the same format. The values of the atomic coordinates and other fields, such as occupancy or temperature
factors, are then communicated to Colvars by NAMD.
<!--l. 1090--><p class="indent">   PDB files may be used either as one of the available mechanisms to select atoms (see the <a href="#atom-group|atomsFile"><span class="t1-zi4r-0-x-x-109">atomsFile</span></a> keyword),
or more frequently to read reference coordinates for least-squares fit alignment (see the <a href="#atom-group|refPositionsFile"><span class="t1-zi4r-0-x-x-109">refPositionsFile</span></a>
keyword).
</p><!--l. 1090--><p class="indent">   To select atoms via the <a href="#atom-group|atomsFile"><span class="t1-zi4r-0-x-x-109">atomsFile</span></a> keyword, the option <a href="#atom-group|atomsCol"><span class="t1-zi4r-0-x-x-109">atomsFile</span></a> is required, and atoms are selected based on
either one of the following cases.
     </p><ol class="enumerate1">
<li class="enumerate" id="x1-26001x3.8.4">All atoms with a non-zero value of the corresponding column are selected.
     </li>
<li class="enumerate" id="x1-26002x3.8.4">If and only if <a href="#atom-group|atomsColValue"><span class="t1-zi4r-0-x-x-109">atomsFile</span></a> is provided, only atoms with a value of <span class="t1-zi4r-0-x-x-109">atomsCol </span>equaling <span class="t1-zi4r-0-x-x-109">atomsColValue</span>
     are selected. This can be useful to reuse the same PDB file for generating multiple selections.</li></ol>
<!--l. 1090--><p class="indent">   To read coordinates via the <a href="#atom-group|refPositionsFile"><span class="t1-zi4r-0-x-x-109">refPositionsFile</span></a> keyword, there are four possible scenarios.
     </p><ol class="enumerate1">
<li class="enumerate" id="x1-26003x3.8.4">The file contains as many coordinates as the atoms that they are being read for: all coordinates will
     be read from the file following the same order as the atoms appear in the selection generated using
     the keywords listed in section <a href="#x1-1000005">5<!--tex4ht:ref: sec:colvar_atom_groups --></a>. (Note that the order is guaranteed only if a single type of selection
     keyword is used one or more times, and not guaranteed when different types of selection keywords
     are used.)
     </li>
<li class="enumerate" id="x1-26004x3.8.4">The file contains more coordinates than needed, and it is assumed to contain coordinates for the entire
     system: only coordinates that match the numeric indices of the selected atoms are read, in order of
     increasing number.
     </li>
<li class="enumerate" id="x1-26005x3.8.4">If the corresponding <a href="#atom-group|refPositionsCol"><span class="t1-zi4r-0-x-x-109">refPositionsCol</span></a> keyword is specified, only positions with a non-zero value of
     the column <span class="t1-zi4r-0-x-x-109">refPositionsCol </span>are read. This is particularly useful when all atoms were selected via
     <a href="#atom-group|atomsFile"><span class="t1-zi4r-0-x-x-109">atomsFile</span></a> and <a href="#atom-group|atomsCol"><span class="t1-zi4r-0-x-x-109">atomsCol</span></a>, which can then be the same as <span class="t1-zi4r-0-x-x-109">refPositionsCol </span>for loading coordinates.
     </li>
<li class="enumerate" id="x1-26006x3.8.4">This                     is                     a                     special                     case                     of                     the
     previous point: if both <a href="#atom-group|refPositionsCol"><span class="t1-zi4r-0-x-x-109">refPositionsCol</span></a> and <a href="#atom-group|refPositionsColValue"><span class="t1-zi4r-0-x-x-109">refPositionsColValue</span></a> are specified, only atoms for
     which the <span class="t1-zi4r-0-x-x-109">refPositionsCol </span>column as the value <span class="t1-zi4r-0-x-x-109">refPositionsColValue </span>are read.</li></ol>
<!--l. 1090--><p class="indent">   Due to the fixed-precision PDB format, it is not recommended to use PDB files to read coordinates when
precision is a concern, and the XYZ format (see <a href="#x1-240003.8.2">3.8.2<!--tex4ht:ref: sec:colvars_ndx_format --></a>) is recommended instead.
                                                                                         
                                                                                         
</p>
<div class="flushright">
<!--l. 1093--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1093--><p class="noindent"></p><div class="heading"><a id="sec:colvar_multicolumn_grid">
<h5 class="subsubsectionHead"><span class="titlemark">3.8.5    </span> <a id="x1-270003.8.5"></a>Grid files: multicolumn text format</h5>
<!--l. 1093--><p class="noindent"></p></a> </div> 
<!--l. 1095--><p class="indent">   Many simulation methods and analysis tools write files that contain functions of the collective variables
tabulated on a grid (e.g. potentials of mean force or multidimentional histograms) for the purpose of analyzing
results. Such files are produced by ABF (<a href="#x1-1090006.2">6.2<!--tex4ht:ref: sec:colvarbias_abf --></a>), metadynamics (<a href="#x1-1190006.4">6.4<!--tex4ht:ref: sec:colvarbias_meta --></a>), multidimensional histograms (<a href="#x1-1350006.10">6.10<!--tex4ht:ref: sec:colvarbias_histogram --></a>), as well as
any restraint with optional thermodynamic integration support (<a href="#x1-1080006.1">6.1<!--tex4ht:ref: sec:colvarbias_ti --></a>).
</p><!--l. 1098--><p class="indent">   In some cases, these files may also be read as input of a new simulation. Suitable input files for this purpose are
typically generated as output files of previous simulations, or directly by the user in the specific case of ensemble-biased
metadynamics (<a href="#x1-1240006.4.5">6.4.5<!--tex4ht:ref: sec:colvarbias_ebmeta --></a>). This section explains the “multicolumn" format used by these files. For a multidimensional
function <!--l. 1101--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>f</mi><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mn>1</mn></mrow></msub></mrow></mrow></math>,
<!--l. 1101--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mn>2</mn></mrow></msub></math>,
…<!--l. 1101--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">)</mo></math> the
multicolumn grid format is defined as follows:<br class="newline"/>
</p>
<div class="tabular"> <table cellpadding="0" cellspacing="0" class="tabular" id="TBL-2"><colgroup id="TBL-2-1g"><col id="TBL-2-1"/><col id="TBL-2-2"/><col id="TBL-2-3"/><col id="TBL-2-4"/><col id="TBL-2-5"/><col id="TBL-2-6"/><col id="TBL-2-7"/></colgroup><tr id="TBL-2-1-" style="vertical-align:baseline;"><td class="td11" id="TBL-2-1-1" style="text-align:left; white-space:nowrap;"> #  </td><td class="td11" id="TBL-2-1-2" style="text-align:left; white-space:nowrap;"> <!--l. 1104--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>N</mi></mrow><mrow><mi mathvariant="normal">cv</mi></mrow></msub></math> </td><td class="td11" id="TBL-2-1-3" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-2-1-4" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-2-1-5" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-2-1-6" style="text-align:left; white-space:nowrap;"> </td>
</tr><tr id="TBL-2-2-" style="vertical-align:baseline;"><td class="td11" id="TBL-2-2-1" style="text-align:left; white-space:nowrap;"> #   </td><td class="td11" id="TBL-2-2-2" style="text-align:left; white-space:nowrap;"> <!--l. 1105--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="monospace">min</mi><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mn>1</mn></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math> </td><td class="td11" id="TBL-2-2-3" style="text-align:left; white-space:nowrap;"> <!--l. 1105--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="monospace">width</mi><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mn>1</mn></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math> </td><td class="td11" id="TBL-2-2-4" style="text-align:left; white-space:nowrap;"> <!--l. 1105--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="monospace">npoints</mi><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mn>1</mn></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math> </td><td class="td11" id="TBL-2-2-5" style="text-align:left; white-space:nowrap;"> <!--l. 1105--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="monospace">periodic</mi><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mn>1</mn></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math> </td>
</tr><tr id="TBL-2-3-" style="vertical-align:baseline;"><td class="td11" id="TBL-2-3-1" style="text-align:left; white-space:nowrap;"> #   </td><td class="td11" id="TBL-2-3-2" style="text-align:left; white-space:nowrap;"> <!--l. 1106--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="monospace">min</mi><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mn>2</mn></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math> </td><td class="td11" id="TBL-2-3-3" style="text-align:left; white-space:nowrap;"> <!--l. 1106--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="monospace">width</mi><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mn>2</mn></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math> </td><td class="td11" id="TBL-2-3-4" style="text-align:left; white-space:nowrap;"> <!--l. 1106--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="monospace">npoints</mi><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mn>2</mn></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math> </td><td class="td11" id="TBL-2-3-5" style="text-align:left; white-space:nowrap;"> <!--l. 1106--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="monospace">periodic</mi><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mn>2</mn></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math> </td>
</tr><tr id="TBL-2-4-" style="vertical-align:baseline;"><td class="td11" id="TBL-2-4-1" style="text-align:left; white-space:nowrap;"> #   </td><td class="td11" id="TBL-2-4-2" style="text-align:left; white-space:nowrap;"> …                                                                                                                                                                            </td><td class="td11" id="TBL-2-4-3" style="text-align:left; white-space:nowrap;"> …                                                                                                                                                                            </td><td class="td11" id="TBL-2-4-4" style="text-align:left; white-space:nowrap;"> …                                                                                                                                                                            </td><td class="td11" id="TBL-2-4-5" style="text-align:left; white-space:nowrap;"> …                                                                                                                                                                            </td></tr><tr id="TBL-2-5-" style="vertical-align:baseline;"><td class="td11" id="TBL-2-5-1" style="text-align:left; white-space:nowrap;"> # </td> <td class="td11" id="TBL-2-5-2" style="text-align:left; white-space:nowrap;"> <!--l. 1108--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="monospace">
min</mi><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><msub><mrow><mi>N</mi></mrow><mrow><mi mathvariant="normal">cv</mi></mrow></msub></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math> </td><td class="td11" id="TBL-2-5-3" style="text-align:left; white-space:nowrap;"> <!--l. 1108--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="monospace">width</mi><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><msub><mrow><mi>N</mi></mrow><mrow><mi mathvariant="normal">cv</mi></mrow></msub></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math> </td><td class="td11" id="TBL-2-5-4" style="text-align:left; white-space:nowrap;"> <!--l. 1108--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="monospace">npoints</mi><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><msub><mrow><mi>N</mi></mrow><mrow><mi mathvariant="normal">cv</mi></mrow></msub></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math> </td><td class="td11" id="TBL-2-5-5" style="text-align:left; white-space:nowrap;"> <!--l. 1108--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="monospace">periodic</mi><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><msub><mrow><mi>N</mi></mrow><mrow><mi mathvariant="normal">cv</mi></mrow></msub></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math> </td><td class="td11" id="TBL-2-5-6" style="text-align:left; white-space:nowrap;"> </td>
</tr><tr id="TBL-2-6-" style="vertical-align:baseline;"><td class="td11" id="TBL-2-6-1" style="text-align:left; white-space:nowrap;"> </td>
</tr><tr id="TBL-2-7-" style="vertical-align:baseline;"><td class="td11" id="TBL-2-7-1" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-2-7-2" style="text-align:left; white-space:nowrap;"> <!--l. 1110--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msubsup><mrow><mi>ξ</mi></mrow><mrow><mn>1</mn></mrow><mrow><mn>1</mn></mrow></msubsup></math> </td><td class="td11" id="TBL-2-7-3" style="text-align:left; white-space:nowrap;"> <!--l. 1110--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msubsup><mrow><mi>ξ</mi></mrow><mrow><mn>2</mn></mrow><mrow><mn>1</mn></mrow></msubsup></math> </td><td class="td11" id="TBL-2-7-4" style="text-align:left; white-space:nowrap;"> …                                                                                                                                                                            </td><td class="td11" id="TBL-2-7-5" style="text-align:left; white-space:nowrap;"> <!--l. 1110--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msubsup><mrow><mi>ξ</mi></mrow><mrow><msub><mrow><mi>N</mi></mrow><mrow><mi mathvariant="normal">cv</mi></mrow></msub></mrow><mrow><mn>1</mn></mrow></msubsup></math> </td><td class="td11" id="TBL-2-7-6" style="text-align:left; white-space:nowrap;"> f(<!--l. 1110--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msubsup><mrow><mi>ξ</mi></mrow><mrow><mn>1</mn></mrow><mrow><mn>1</mn></mrow></msubsup></math>, <!--l. 1110--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msubsup><mrow><mi>ξ</mi></mrow><mrow><mn>2</mn></mrow><mrow><mn>1</mn></mrow></msubsup></math>, …, <!--l. 1110--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msubsup><mrow><mi>ξ</mi></mrow><mrow><msub><mrow><mi>N</mi></mrow><mrow><mi mathvariant="normal">cv</mi></mrow></msub></mrow><mrow><mn>1</mn></mrow></msubsup></math>)  </td><td class="td11" id="TBL-2-7-7" style="text-align:left; white-space:nowrap;"> </td>
</tr><tr id="TBL-2-8-" style="vertical-align:baseline;"><td class="td11" id="TBL-2-8-1" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-2-8-2" style="text-align:left; white-space:nowrap;"> <!--l. 1111--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msubsup><mrow><mi>ξ</mi></mrow><mrow><mn>1</mn></mrow><mrow><mn>1</mn></mrow></msubsup></math> </td><td class="td11" id="TBL-2-8-3" style="text-align:left; white-space:nowrap;"> <!--l. 1111--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msubsup><mrow><mi>ξ</mi></mrow><mrow><mn>2</mn></mrow><mrow><mn>1</mn></mrow></msubsup></math> </td><td class="td11" id="TBL-2-8-4" style="text-align:left; white-space:nowrap;"> …                                                                                                                                                                            </td><td class="td11" id="TBL-2-8-5" style="text-align:left; white-space:nowrap;"> <!--l. 1111--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msubsup><mrow><mi>ξ</mi></mrow><mrow><msub><mrow><mi>N</mi></mrow><mrow><mi mathvariant="normal">cv</mi></mrow></msub></mrow><mrow><mn>2</mn></mrow></msubsup></math> </td><td class="td11" id="TBL-2-8-6" style="text-align:left; white-space:nowrap;"> f(<!--l. 1111--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msubsup><mrow><mi>ξ</mi></mrow><mrow><mn>1</mn></mrow><mrow><mn>1</mn></mrow></msubsup></math>, <!--l. 1111--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msubsup><mrow><mi>ξ</mi></mrow><mrow><mn>2</mn></mrow><mrow><mn>1</mn></mrow></msubsup></math>, …, <!--l. 1111--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msubsup><mrow><mi>ξ</mi></mrow><mrow><msub><mrow><mi>N</mi></mrow><mrow><mi mathvariant="normal">cv</mi></mrow></msub></mrow><mrow><mn>2</mn></mrow></msubsup></math>)  </td>
</tr><tr id="TBL-2-9-" style="vertical-align:baseline;"><td class="td11" id="TBL-2-9-1" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-2-9-2" style="text-align:left; white-space:nowrap;"> …                                                                                                                                                                            </td><td class="td11" id="TBL-2-9-3" style="text-align:left; white-space:nowrap;"> …                                                                                                                                                                            </td><td class="td11" id="TBL-2-9-4" style="text-align:left; white-space:nowrap;"> …                                                                                                                                                                            </td><td class="td11" id="TBL-2-9-5" style="text-align:left; white-space:nowrap;"> …                                                                                                                                                                            </td><td class="td11" id="TBL-2-9-6" style="text-align:left; white-space:nowrap;"> …                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               </td></tr><tr id="TBL-2-10-" style="vertical-align:baseline;"><td class="td11" id="TBL-2-10-1" style="text-align:left; white-space:nowrap;"> </td>
</tr><tr id="TBL-2-11-" style="vertical-align:baseline;"><td class="td11" id="TBL-2-11-1" style="text-align:left; white-space:nowrap;"> </td> </tr></table>
</div>
<!--l. 1116--><p class="noindent">Lines beginning with the character “<span class="t1-zi4r-0-x-x-109">#</span>" are the header of the file.
<!--l. 1117--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>N</mi></mrow><mrow><mi mathvariant="normal">cv</mi></mrow></msub></math>
is the number of collective variables sampled by the grid. For each variable
<!--l. 1118--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></math>,
<!--l. 1118--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="monospace">min</mi><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math>
is the lowest value sampled by the grid (i.e. the left-most boundary of the grid along
<!--l. 1118--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></math>),
<!--l. 1118--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="monospace">width</mi><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math> is the width of each
grid step along <!--l. 1118--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></math>,
<!--l. 1118--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="monospace">npoints</mi><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math> is the number
of points and <!--l. 1118--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="monospace">periodic</mi><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math>
is a flag whose value is 1 or 0 depending on whether the grid is periodic along
<!--l. 1118--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></math>. In most
situations: </p>
<ul class="itemize1">
<li class="itemize"><!--l. 1121--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="monospace">min</mi><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math>
     is given by the <a href="#colvar|lowerBoundary"><span class="t1-zi4r-0-x-x-109">lowerBoundary</span></a> keyword of the variable <!--l. 1121--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></math>;
     </li>
<li class="itemize"><!--l. 1122--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="monospace">width</mi><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math>
     is given by the <a href="#colvar|width"><span class="t1-zi4r-0-x-x-109">width</span></a> keyword;
     </li>
<li class="itemize"><!--l. 1123--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="monospace">npoints</mi><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math>
     is calculated from the two above numbers and the <a href="#colvar|upperBoundary"><span class="t1-zi4r-0-x-x-109">upperBoundary</span></a> keyword;
     </li>
<li class="itemize"><!--l. 1124--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="monospace">periodic</mi><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math>
     is set to 1 if and only if <!--l. 1124--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></math>
     is periodic and the grids' boundaries cover its period.</li></ul>
<!--l. 1126--><p class="noindent"><span class="ptmb8t-x-x-109">Exception: </span>there is a slightly different header in PMF files computed by ABF (<a href="#x1-1090006.2">6.2<!--tex4ht:ref: sec:colvarbias_abf --></a>) or by other biases with an optional
thermodynamic integration (TI) estimator (<a href="#x1-1080006.1">6.1<!--tex4ht:ref: sec:colvarbias_ti --></a>). In this case, free-energy gradients are accumulated on an <span class="t1-zi4r-0-x-x-109">(npoints)</span>-long grid
along each variable <!--l. 1127--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>:
after these gradients are integrated, the resulting PMF is discretized on a grid with <span class="t1-zi4r-0-x-x-109">(npoints+1) </span>points along
<!--l. 1127--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>. Therefore, the edges of
the PMF's grid extend <!--l. 1128--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="monospace">width</mi><mo class="MathClass-bin">∕</mo><mn>2</mn></math>
above and below the original boundaries (unless these are periodic). The format of the file's header is otherwise
unchanged.
</p><!--l. 1130--><p class="indent">   After the header, the rest of the file contains values of the tabulated function
<!--l. 1131--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>f</mi><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mn>1</mn></mrow></msub></mrow></mrow></math>,
<!--l. 1131--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mn>2</mn></mrow></msub></math>,
…<!--l. 1131--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><msub><mrow><mi>N</mi></mrow><mrow><mi mathvariant="normal">cv</mi></mrow></msub></mrow></msub><mo class="MathClass-close">)</mo></math>, one for each line.
The first <!--l. 1132--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>N</mi></mrow><mrow><mi mathvariant="normal">cv</mi></mrow></msub></math> columns
contain values of <!--l. 1132--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mn>1</mn></mrow></msub></math>,
<!--l. 1132--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mn>2</mn></mrow></msub></math>,
…<!--l. 1132--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><msub><mrow><mi>N</mi></mrow><mrow><mi mathvariant="normal">cv</mi></mrow></msub></mrow></msub></math> and the last column contains
the value of the function <!--l. 1132--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>f</mi></math>.
Points are sorted in ascending order with the fastest-changing values at the right (“C-style" order). Each sweep of the right-most
variable <!--l. 1134--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><msub><mrow><mi>N</mi></mrow><mrow><mi mathvariant="normal">cv</mi></mrow></msub></mrow></msub></math>
is terminated by an empty line. For two dimensional grid files, this allows quick visualization by programs such as
GNUplot. <br class="newline"/>
</p><!--l. 1138--><p class="noindent"><span class="ptmb8t-x-x-109">Example 1: </span>multicolumn text file for a one-dimensional histogram with <span class="t1-zi4r-0-x-x-109">lowerBoundary </span>= 15, <span class="t1-zi4r-0-x-x-109">upperBoundary </span>= 48
and <span class="t1-zi4r-0-x-x-109">width </span>= 0.1.
</p>
<div class="tabular"> <table cellpadding="0" cellspacing="0" class="tabular" id="TBL-3"><colgroup id="TBL-3-1g"><col id="TBL-3-1"/><col id="TBL-3-2"/><col id="TBL-3-3"/><col id="TBL-3-4"/><col id="TBL-3-5"/><col id="TBL-3-6"/></colgroup><tr id="TBL-3-1-" style="vertical-align:baseline;"><td class="td11" id="TBL-3-1-1" style="text-align:left; white-space:nowrap;"> </td></tr><tr id="TBL-3-2-" style="vertical-align:baseline;"><td class="td11" id="TBL-3-2-1" style="text-align:left; white-space:nowrap;"> #</td><td class="td11" id="TBL-3-2-2" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">1     </span></td><td class="td11" id="TBL-3-2-3" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-3-2-4" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-3-2-5" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-3-2-6" style="text-align:left; white-space:nowrap;"> </td>
</tr><tr id="TBL-3-3-" style="vertical-align:baseline;"><td class="td11" id="TBL-3-3-1" style="text-align:left; white-space:nowrap;"> #  </td><td class="td11" id="TBL-3-3-2" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">15    </span></td><td class="td11" id="TBL-3-3-3" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">0.1         </span></td><td class="td11" id="TBL-3-3-4" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">330 </span></td><td class="td11" id="TBL-3-3-5" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">0 </span></td></tr><tr id="TBL-3-4-" style="vertical-align:baseline;"><td class="td11" id="TBL-3-4-1" style="text-align:left; white-space:nowrap;"> </td>
</tr><tr id="TBL-3-5-" style="vertical-align:baseline;"><td class="td11" id="TBL-3-5-1" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-3-5-2" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">15.05 </span></td><td class="td11" id="TBL-3-5-3" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">6.14012e-07 </span></td><td class="td11" id="TBL-3-5-4" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-3-5-5" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-3-5-6" style="text-align:left; white-space:nowrap;"> </td></tr><tr id="TBL-3-6-" style="vertical-align:baseline;"><td class="td11" id="TBL-3-6-1" style="text-align:left; white-space:nowrap;"> </td> <td class="td11" id="TBL-3-6-2" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">15.15 </span></td> <td class="td11" id="TBL-3-6-3" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">7.47644e-07 </span></td> <td class="td11" id="TBL-3-6-4" style="text-align:left; white-space:nowrap;"> </td> <td class="td11" id="TBL-3-6-5" style="text-align:left; white-space:nowrap;"> </td> <td class="td11" id="TBL-3-6-6" style="text-align:left; white-space:nowrap;"></td>
</tr><tr id="TBL-3-7-" style="vertical-align:baseline;"><td class="td11" id="TBL-3-7-1" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-3-7-2" style="text-align:left; white-space:nowrap;"> …          </td><td class="td11" id="TBL-3-7-3" style="text-align:left; white-space:nowrap;"> …                      </td><td class="td11" id="TBL-3-7-4" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-3-7-5" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-3-7-6" style="text-align:left; white-space:nowrap;"> </td></tr><tr id="TBL-3-8-" style="vertical-align:baseline;"><td class="td11" id="TBL-3-8-1" style="text-align:left; white-space:nowrap;"> </td> <td class="td11" id="TBL-3-8-2" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">47.85 </span></td> <td class="td11" id="TBL-3-8-3" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">1.65944e-06 </span></td> <td class="td11" id="TBL-3-8-4" style="text-align:left; white-space:nowrap;"> </td> <td class="td11" id="TBL-3-8-5" style="text-align:left; white-space:nowrap;"> </td> <td class="td11" id="TBL-3-8-6" style="text-align:left; white-space:nowrap;"></td>
</tr><tr id="TBL-3-9-" style="vertical-align:baseline;"><td class="td11" id="TBL-3-9-1" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-3-9-2" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">47.95 </span></td><td class="td11" id="TBL-3-9-3" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">1.46712e-06 </span></td><td class="td11" id="TBL-3-9-4" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-3-9-5" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-3-9-6" style="text-align:left; white-space:nowrap;"> </td></tr><tr id="TBL-3-10-" style="vertical-align:baseline;"><td class="td11" id="TBL-3-10-1" style="text-align:left; white-space:nowrap;"> </td>
</tr><tr id="TBL-3-11-" style="vertical-align:baseline;"><td class="td11" id="TBL-3-11-1" style="text-align:left; white-space:nowrap;"> </td> </tr></table></div>
<!--l. 1153--><p class="noindent"><span class="ptmb8t-x-x-109">Example 2: </span>multicolumn text file for a two-dimensional histogram of two dihedral angles (periodic interval with
                                                                                         
                                                                                         
6<!--l. 1153--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></math>
bins):
</p>
<div class="tabular"> <table cellpadding="0" cellspacing="0" class="tabular" id="TBL-4"><colgroup id="TBL-4-1g"><col id="TBL-4-1"/><col id="TBL-4-2"/><col id="TBL-4-3"/><col id="TBL-4-4"/><col id="TBL-4-5"/><col id="TBL-4-6"/></colgroup><tr id="TBL-4-1-" style="vertical-align:baseline;"><td class="td11" id="TBL-4-1-1" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-4-1-2" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-4-1-3" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-4-1-4" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-4-1-5" style="text-align:left; white-space:nowrap;"> </td></tr><tr id="TBL-4-2-" style="vertical-align:baseline;"><td class="td11" id="TBL-4-2-1" style="text-align:left; white-space:nowrap;"> # </td> <td class="td11" id="TBL-4-2-2" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">2 </span></td> <td class="td11" id="TBL-4-2-3" style="text-align:left; white-space:nowrap;"> </td> <td class="td11" id="TBL-4-2-4" style="text-align:left; white-space:nowrap;"> </td> <td class="td11" id="TBL-4-2-5" style="text-align:left; white-space:nowrap;"></td><td class="td11" id="TBL-4-2-6" style="text-align:left; white-space:nowrap;"> </td>
</tr><tr id="TBL-4-3-" style="vertical-align:baseline;"><td class="td11" id="TBL-4-3-1" style="text-align:left; white-space:nowrap;"> #  </td><td class="td11" id="TBL-4-3-2" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">-180.0 </span></td><td class="td11" id="TBL-4-3-3" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">6.0    </span></td><td class="td11" id="TBL-4-3-4" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">30          </span></td><td class="td11" id="TBL-4-3-5" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">1 </span></td></tr><tr id="TBL-4-4-" style="vertical-align:baseline;"><td class="td11" id="TBL-4-4-1" style="text-align:left; white-space:nowrap;"> # </td> <td class="td11" id="TBL-4-4-2" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">-180.0 </span></td> <td class="td11" id="TBL-4-4-3" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">6.0 </span></td> <td class="td11" id="TBL-4-4-4" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">30 </span></td> <td class="td11" id="TBL-4-4-5" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">1</span></td>
</tr><tr id="TBL-4-5-" style="vertical-align:baseline;"><td class="td11" id="TBL-4-5-1" style="text-align:left; white-space:nowrap;"> </td></tr><tr id="TBL-4-6-" style="vertical-align:baseline;"><td class="td11" id="TBL-4-6-1" style="text-align:left; white-space:nowrap;"></td><td class="td11" id="TBL-4-6-2" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">-177.0 </span></td><td class="td11" id="TBL-4-6-3" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">-177.0 </span></td><td class="td11" id="TBL-4-6-4" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">8.97117e-06 </span></td><td class="td11" id="TBL-4-6-5" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-4-6-6" style="text-align:left; white-space:nowrap;"> </td>
</tr><tr id="TBL-4-7-" style="vertical-align:baseline;"><td class="td11" id="TBL-4-7-1" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-4-7-2" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">-177.0 </span></td><td class="td11" id="TBL-4-7-3" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">-171.0 </span></td><td class="td11" id="TBL-4-7-4" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">1.53525e-06 </span></td><td class="td11" id="TBL-4-7-5" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-4-7-6" style="text-align:left; white-space:nowrap;"> </td></tr><tr id="TBL-4-8-" style="vertical-align:baseline;"><td class="td11" id="TBL-4-8-1" style="text-align:left; white-space:nowrap;"> </td> <td class="td11" id="TBL-4-8-2" style="text-align:left; white-space:nowrap;"> … </td> <td class="td11" id="TBL-4-8-3" style="text-align:left; white-space:nowrap;"> … </td> <td class="td11" id="TBL-4-8-4" style="text-align:left; white-space:nowrap;"> … </td> <td class="td11" id="TBL-4-8-5" style="text-align:left; white-space:nowrap;"> </td> <td class="td11" id="TBL-4-8-6" style="text-align:left; white-space:nowrap;"></td>
</tr><tr id="TBL-4-9-" style="vertical-align:baseline;"><td class="td11" id="TBL-4-9-1" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-4-9-2" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">-177.0 </span></td><td class="td11" id="TBL-4-9-3" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">177.0 </span></td><td class="td11" id="TBL-4-9-4" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">2.442956-06 </span></td><td class="td11" id="TBL-4-9-5" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-4-9-6" style="text-align:left; white-space:nowrap;"> </td></tr><tr id="TBL-4-10-" style="vertical-align:baseline;"><td class="td11" id="TBL-4-10-1" style="text-align:left; white-space:nowrap;"> </td>
</tr><tr id="TBL-4-11-" style="vertical-align:baseline;"><td class="td11" id="TBL-4-11-1" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-4-11-2" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">-171.0 </span></td><td class="td11" id="TBL-4-11-3" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">-177.0 </span></td><td class="td11" id="TBL-4-11-4" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">2.04702e-05 </span></td><td class="td11" id="TBL-4-11-5" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-4-11-6" style="text-align:left; white-space:nowrap;"> </td></tr><tr id="TBL-4-12-" style="vertical-align:baseline;"><td class="td11" id="TBL-4-12-1" style="text-align:left; white-space:nowrap;"> </td> <td class="td11" id="TBL-4-12-2" style="text-align:left; white-space:nowrap;"> … </td> <td class="td11" id="TBL-4-12-3" style="text-align:left; white-space:nowrap;"> … </td> <td class="td11" id="TBL-4-12-4" style="text-align:left; white-space:nowrap;"> … </td> <td class="td11" id="TBL-4-12-5" style="text-align:left; white-space:nowrap;"> </td> <td class="td11" id="TBL-4-12-6" style="text-align:left; white-space:nowrap;"></td>
</tr><tr id="TBL-4-13-" style="vertical-align:baseline;"><td class="td11" id="TBL-4-13-1" style="text-align:left; white-space:nowrap;"> </td> </tr></table></div>
<div class="flushright">
<!--l. 1171--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1171--><p class="noindent"></p><div class="heading"><a id="sec:colvars_traj_format">
<h5 class="subsubsectionHead"><span class="titlemark">3.8.6    </span> <a id="x1-280003.8.6"></a>Output trajectory files</h5>
<!--l. 1171--><p class="noindent"></p></a> </div> 
<!--l. 1173--><p class="indent">   The Colvars trajectory file (with a suffix <span class="t1-zi4r-0-x-x-109">.colvars.traj</span>) is a plain text file (scientific notation with 14-digit
precision) whose columns represent quantities such as colvar values, applied forces, or individual restraints' energies.
Under most scenarios, plotting or analyzing this file is straightforward: for example, the following contains a variable
“<!--l. 1173--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>A</mi></math>" and the energy
of a restraint “<!--l. 1173--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>r</mi><mi>A</mi></math>":</p>
<div class="mdframed" id="mdframed-37">
<!--l. 1174--><p class="noindent"></p><pre>#       step   A                     E_rA<br class="newline"/>           0    1.42467449615693e+01  6.30982865292123e+02<br class="newline"/>         100    1.42282559728026e+01  6.20640585041317e+02<br class="newline"/>…</pre>
</div>
<!--l. 1181--><p class="indent">   Occasionally, if the Colvars configuration is changed mid-run certain quantities may be
added or removed, changing the column layout. Labels in comment lines can assist in such
cases: for example, consider the trajectory above with the addition of a second variable,
“<!--l. 1181--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>B</mi></math>", after
10,000 steps:</p>
<div class="mdframed" id="mdframed-38">
<!--l. 1182--><p class="noindent"></p><pre>#       step   A                     E_rA<br class="newline"/>           0    1.42467449615693e+01  6.30982865292123e+02<br class="newline"/>         100    1.42282559728026e+01  6.20640585041317e+02<br class="newline"/>…<br class="newline"/>#       step   A                     B                     E_rA              <br class="newline"/>       10000    1.38136915830229e+01  9.99574098859265e-01  4.11184644791030e+02<br class="newline"/>       10100    1.36437184346326e+01  9.99574091957314e-01  3.37726286543895e+02 </pre>
</div>
<!--l. 1192--><p class="indent">   Analyzing the above file with standard tools is possible, but laborious: a <a href="https://github.com/Colvars/colvars/blob/master/colvartools/plot_colvars_traj.py?raw=true">convenience script</a> is
                                                                                         
                                                                                         
provided for this and related purposes. It may be used either as a command-line tool or as a Python
module:</p>
<div class="mdframed" id="mdframed-39">
<!--l. 1193--><p class="noindent"></p><pre>&gt;&gt;&gt; from plot_colvars_traj import Colvars_traj<br class="newline"/>&gt;&gt;&gt; traj = Colvars_traj('test.colvars.traj')<br class="newline"/>&gt;&gt;&gt; print(traj['A'].steps, traj['A'].values)<br class="newline"/>[    0   100  ...  10000 10100] [14.246745 14.228256 ... 13.813692 13.643718]<br class="newline"/>&gt;&gt;&gt; print(traj['B'].steps, traj['B'].values)<br class="newline"/>[10000 10100] [0.999574  0.9995741] </pre>
</div>
<!--l. 1203--><p class="indent">
</p>
<div class="flushright">
<!--l. 1203--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1203--><p class="noindent"></p><div class="heading"><a id="sec:colvar">
<h3 class="sectionHead"><span class="titlemark">4    </span> <a id="x1-290004"></a>Defining collective variables</h3>
<!--l. 1203--><p class="noindent"></p></a> </div> 
<!--l. 1205--><p class="indent">   A collective variable is defined by the keyword <span class="t1-zi4r-0-x-x-109">colvar </span>followed by its configuration options contained within
curly braces:</p>
<div class="mdframed" id="mdframed-40">
<!--l. 1207--><p class="noindent"></p><pre>colvar {<br class="newline"/>  name xi<br class="newline"/>  <!--l. 1210--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>other options<!--l. 1210--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><br class="newline"/>  function_name {<br class="newline"/>    <!--l. 1212--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>parameters<!--l. 1212--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><br class="newline"/>    <!--l. 1213--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>atom selection<!--l. 1213--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><br class="newline"/>  }<br class="newline"/>} </pre>
</div>
<!--l. 1219--><p class="noindent">There are multiple ways of defining a variable: </p>
<ul class="itemize1">
<li class="itemize">The  <span class="ptmri8t-x-x-109">simplest  and  most  common  way  </span>way  is  using  one  of  the  precompiled  functions  (here  called
     “components"), which are listed in section <a href="#x1-300004.1">4.1<!--tex4ht:ref: sec:cvc_list --></a>. For example, using the keyword <span class="t1-zi4r-0-x-x-109">rmsd </span>(section <a href="#x1-490004.5.1">4.5.1<!--tex4ht:ref: sec:cvc_rmsd --></a>)
     defines the variable as the root mean squared deviation (RMSD) of the selected atoms.
     </li>
<li class="itemize">A new variable may also be constructed as a linear or polynomial combination of the components
     listed in section <a href="#x1-300004.1">4.1<!--tex4ht:ref: sec:cvc_list --></a> (see <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a> for details).
     </li>
<li class="itemize">A user-defined mathematical function of the existing components (see list in section <a href="#x1-300004.1">4.1<!--tex4ht:ref: sec:cvc_list --></a>), or of the
     atomic coordinates directly (see the <span class="t1-zi4r-0-x-x-109">cartesian </span>keyword in <a href="#x1-690004.8.1">4.8.1<!--tex4ht:ref: sec:cvc_cartesian --></a>). The function is defined through
     the keyword <a href="#colvar|customFunction"><span class="t1-zi4r-0-x-x-109">customFunction</span></a> (see <a href="#x1-920004.17">4.17<!--tex4ht:ref: sec:colvar_custom_function --></a> for details).
     </li>
<li class="itemize">A  user-defined  Tcl  function  of  the  existing  components  (see  list  in  section <a href="#x1-300004.1">4.1<!--tex4ht:ref: sec:cvc_list --></a>),  or  of  the  atomic
     coordinates directly (see the <span class="t1-zi4r-0-x-x-109">cartesian </span>keyword in <a href="#x1-690004.8.1">4.8.1<!--tex4ht:ref: sec:cvc_cartesian --></a>). The function is provided by a separate Tcl
     script, and referenced through the keyword <a href="#colvar|scriptedFunction"><span class="t1-zi4r-0-x-x-109">scriptedFunction</span></a> (see <a href="#x1-930004.18">4.18<!--tex4ht:ref: sec:colvar_scripted --></a> for details). </li></ul>
<!--l. 1232--><p class="noindent">Choosing a component (function) is the only parameter strictly required to define a collective variable. It is also highly
recommended to specify a name for the variable: </p>
<ul class="itemize1">
<li class="itemize"><a id="colvar|name"> </a> <a id="dx1-29001"></a><span class="colorbox" id="colorbox10"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">name</span></span> <!--l. 1247--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Name
     of this colvar<!--l. 1247--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">colvar</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>string <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span>“<span class="t1-zi4r-0-x-x-109">colvar</span>" + numeric id <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>The name is an unique case-sensitive string which allows the Colvars module to identify
     this colvar unambiguously; it is also used in the trajectory file to label to the columns corresponding
     to this colvar.
     </li></ul>
<div class="flushright">
<!--l. 1252--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1252--><p class="noindent"></p><div class="heading"><a id="sec:cvc_list">
<h4 class="subsectionHead"><span class="titlemark">4.1    </span> <a id="x1-300004.1"></a>Choosing a function</h4>
<!--l. 1252--><p class="noindent"></p></a> </div> 
<!--l. 1254--><p class="indent">   In this context, the function that computes a colvar is called a <span class="ptmri8t-x-x-109">component</span>. A component's choice and definition
consists of including in the variable's configuration a keyword indicating the type of function (e.g. <span class="t1-zi4r-0-x-x-109">rmsd</span>), followed
by a definition block specifying the atoms involved (see <a href="#x1-1000005">5<!--tex4ht:ref: sec:colvar_atom_groups --></a>) and any additional parameters (cutoffs, “reference"
values, …). <span class="ptmri8t-x-x-109">At least one component must be chosen to define a variable: </span>if none of the keywords listed below is
found, an error is raised.
</p><!--l. 1258--><p class="indent">   The following components implement functions with a scalar value (i.e. a real number): </p>
<ul class="itemize1">
<li class="itemize"><a href="#colvar|distance"><span class="t1-zi4r-0-x-x-109">distance</span></a>: distance between two groups;
     </li>
<li class="itemize"><a href="#colvar|distanceZ"><span class="t1-zi4r-0-x-x-109">distanceZ</span></a>: projection of a distance vector on an axis;
     </li>
<li class="itemize"><a href="#colvar|distanceXY"><span class="t1-zi4r-0-x-x-109">distanceXY</span></a>: projection of a distance vector on a plane;
     </li>
<li class="itemize"><a href="#colvar|distanceInv"><span class="t1-zi4r-0-x-x-109">distanceInv</span></a>: mean distance between two groups of atoms (e.g. NOE-based distance);
     </li>
<li class="itemize"><a href="#colvar|angle"><span class="t1-zi4r-0-x-x-109">angle</span></a>: angle between three groups;
     </li>
<li class="itemize"><a href="#colvar|dihedral"><span class="t1-zi4r-0-x-x-109">dihedral</span></a>: torsional (dihedral) angle between four groups;
     </li>
<li class="itemize"><a href="#colvar|dipoleAngle"><span class="t1-zi4r-0-x-x-109">dipoleAngle</span></a>: angle between two groups and dipole of a third group;
                                                                                         
                                                                                         
     </li>
<li class="itemize"><a href="#colvar|dipoleMagnitude"><span class="t1-zi4r-0-x-x-109">dipoleMagnitude</span></a>: magnitude of the dipole of a group of atoms;
     </li>
<li class="itemize"><a href="#colvar|polarTheta"><span class="t1-zi4r-0-x-x-109">polarTheta</span></a>: polar angle of a group in spherical coordinates;
     </li>
<li class="itemize"><a href="#colvar|polarPhi"><span class="t1-zi4r-0-x-x-109">polarPhi</span></a>: azimuthal angle of a group in spherical coordinates;
     </li>
<li class="itemize"><a href="#colvar|coordNum"><span class="t1-zi4r-0-x-x-109">coordNum</span></a>: coordination number between two groups;
     </li>
<li class="itemize"><a href="#colvar|selfCoordNum"><span class="t1-zi4r-0-x-x-109">selfCoordNum</span></a>: coordination number of atoms within a group;
     </li>
<li class="itemize"><a href="#colvar|hBond"><span class="t1-zi4r-0-x-x-109">hBond</span></a>: hydrogen bond between two atoms;
     </li>
<li class="itemize"><a href="#colvar|rmsd"><span class="t1-zi4r-0-x-x-109">rmsd</span></a>: root mean square deviation (RMSD) from a set of reference coordinates;
     </li>
<li class="itemize"><a href="#colvar|eigenvector"><span class="t1-zi4r-0-x-x-109">eigenvector</span></a>: projection of the atomic coordinates on a vector;
     </li>
<li class="itemize"><a href="#colvar|mapTotal"><span class="t1-zi4r-0-x-x-109">mapTotal</span></a>: total value of a volumetric map;
     </li>
<li class="itemize"><a href="#colvar|orientationAngle"><span class="t1-zi4r-0-x-x-109">orientationAngle</span></a>: angle of the best-fit rotation from a set of reference coordinates;
     </li>
<li class="itemize"><a href="#colvar|orientationProj"><span class="t1-zi4r-0-x-x-109">orientationProj</span></a>: cosine of <a href="#colvar|orientationProj"><span class="t1-zi4r-0-x-x-109">orientationProj</span></a>;
     </li>
<li class="itemize"><a href="#colvar|spinAngle"><span class="t1-zi4r-0-x-x-109">spinAngle</span></a>: projection orthogonal to an axis of the best-fit rotation from a set of reference coordinates;
     </li>
<li class="itemize"><a href="#colvar|tilt"><span class="t1-zi4r-0-x-x-109">tilt</span></a>: projection on an axis of the best-fit rotation from a set of reference coordinates;
     </li>
<li class="itemize"><a href="#colvar|gyration"><span class="t1-zi4r-0-x-x-109">gyration</span></a>: radius of gyration of a group of atoms;
     </li>
<li class="itemize"><a href="#colvar|inertia"><span class="t1-zi4r-0-x-x-109">inertia</span></a>: moment of inertia of a group of atoms;
     </li>
<li class="itemize"><a href="#colvar|inertiaZ"><span class="t1-zi4r-0-x-x-109">inertiaZ</span></a>: moment of inertia of a group of atoms around a chosen axis;
     </li>
<li class="itemize"><a href="#colvar|alpha"><span class="t1-zi4r-0-x-x-109">alpha</span></a>: <!--l. 1292--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>α</mi></math>-helix
     content of a protein segment.
     </li>
<li class="itemize"><a href="#colvar|dihedralPC"><span class="t1-zi4r-0-x-x-109">dihedralPC</span></a>: projection of protein backbone dihedrals onto a dihedral principal component.  </li></ul>
<!--l. 1299--><p class="indent">   Some components do not return scalar, but vector values: </p>
<ul class="itemize1">
<li class="itemize"><a href="#colvar|distanceVec"><span class="t1-zi4r-0-x-x-109">distanceVec</span></a>: distance vector between two groups (length: 3);
     </li>
<li class="itemize"><a href="#colvar|distanceDir"><span class="t1-zi4r-0-x-x-109">distanceDir</span></a>: unit vector parallel to distanceVec (length: 3);
     </li>
<li class="itemize"><a href="#colvar|cartesian"><span class="t1-zi4r-0-x-x-109">cartesian</span></a>: vector of atomic Cartesian coordinates (length: <!--l. 1303--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>N</mi></math>
     times the number of Cartesian components requested, X, Y or Z);
     </li>
<li class="itemize"><a href="#colvar|distancePairs"><span class="t1-zi4r-0-x-x-109">distancePairs</span></a>: vector of mutual distances (length: <!--l. 1304--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>N</mi></mrow><mrow><mi mathvariant="normal">1</mi></mrow></msub> <mo class="MathClass-bin">×</mo><msub><mrow><mi>N</mi></mrow><mrow><mi mathvariant="normal">2</mi></mrow></msub></math>);
     </li>
<li class="itemize"><a href="#colvar|orientation"><span class="t1-zi4r-0-x-x-109">orientation</span></a>: best-fit rotation, expressed as a unit quaternion (length: 4).</li></ul>
<!--l. 1308--><p class="indent">   The types of components used in a colvar (scalar or not) determine the properties of that colvar, and particularly
which biasing or analysis methods can be applied.
</p><!--l. 1312--><p class="indent"> <span class="ptmb8t-x-x-109">What if “X" is not listed? </span>If a function type is not available on this list, it may be possible to define it as a
polynomial superposition of existing ones (see <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a>), a custom function (see <a href="#x1-920004.17">4.17<!--tex4ht:ref: sec:colvar_custom_function --></a>), or a scripted function (see
<a href="#x1-930004.18">4.18<!--tex4ht:ref: sec:colvar_scripted --></a>).
</p><!--l. 1314--><p class="indent">   In the rest of this section, all available component types are listed, along with their physical units and the ranges
of values, if limited. Such limiting values can be used to define <a href="#colvar|lowerBoundary"><span class="t1-zi4r-0-x-x-109">lowerBoundary</span></a> and <a href="#colvar|upperBoundary"><span class="t1-zi4r-0-x-x-109">upperBoundary</span></a> in the parent
colvar.
</p><!--l. 1317--><p class="indent">   For each type of component, the available configurations keywords are listed: when two components share
certain keywords, the second component references to the documentation of the first one that uses that keyword.
The very few keywords that are available for all types of components are listed in a separate section
<a href="#x1-870004.13">4.13<!--tex4ht:ref: sec:cvc_common --></a>.
</p>
<div class="flushright">
<!--l. 1323--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1323--><p class="noindent"></p><div class="heading"><a id="sec:cvc_distances">
<h4 class="subsectionHead"><span class="titlemark">4.2    </span> <a id="x1-310004.2"></a>Distances</h4>
<!--l. 1323--><p class="noindent"></p></a> </div> 
<div class="flushright">
<!--l. 1326--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1326--><p class="noindent"></p><div class="heading"><a id="sec:cvc_distance">
<h5 class="subsubsectionHead"><span class="titlemark">4.2.1    </span> <a id="x1-320004.2.1"></a><span class="t1-zi4r-0-x-x-109">distance</span>: center-of-mass distance between two groups.</h5>
<!--l. 1326--><p class="noindent"></p></a> </div>  <a id="colvar|distance"> </a>
<!--l. 1329--><p class="indent">   The <span class="t1-zi4r-0-x-x-109">distance {...} </span>block defines a distance component between the two atom groups, <span class="t1-zi4r-0-x-x-109">group1 </span>and
<span class="t1-zi4r-0-x-x-109">group2</span>.
</p><!--l. 1331--><p class="noindent"><span class="ptmb8t-x-x-109">List of keywords </span>(see also <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="colvar|distance|group1"> </a> <a id="dx1-32001"></a><span class="colorbox" id="colorbox11"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">group1</span></span> <!--l. 1339--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>First
     group of atoms<!--l. 1339--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">distance</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>Block <span class="t1-zi4r-0-x-x-109">group1 {...}</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>First group of atoms.
     </li>
<li class="itemize"><a id="colvar|distance|group2"> </a> <a id="dx1-32002"></a><span class="colorbox" id="colorbox12"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">group2:</span></span> analogous to <span class="t1-zi4r-0-x-x-109">group1</span>
</li>
<li class="itemize"><a id="colvar|distance|forceNoPBC"> </a> <a id="dx1-32003"></a><span class="colorbox" id="colorbox13"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">forceNoPBC</span></span> <!--l. 1361--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Calculate
     absolute rather than minimum-image distance?<!--l. 1361--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">distance</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">no</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>By default, in calculations with periodic boundary conditions, the <span class="t1-zi4r-0-x-x-109">distance </span>component
     returns the distance according to the minimum-image convention. If this parameter is set to <span class="t1-zi4r-0-x-x-109">yes</span>, PBC
     will be ignored and the distance between the coordinates as maintained internally will be used. This is
     only useful in a limited number of special cases, e.g. to describe the distance between remote points
     of a single macromolecule, which cannot be split across periodic cell boundaries, and for which the
     minimum-image distance might give the wrong result because of a relatively small periodic cell.
     </li>
<li class="itemize"><a id="colvar|distance|oneSiteTotalForce"> </a> <a id="dx1-32004"></a><span class="colorbox" id="colorbox14"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">oneSiteTotalForce</span></span> <!--l. 1376--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Measure
     total force on group 1 only?<!--l. 1376--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">angle</span>, <span class="t1-zi4r-0-x-x-109">dipoleAngle</span>, <span class="t1-zi4r-0-x-x-109">dihedral</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">no</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>If this is set to <span class="t1-zi4r-0-x-x-109">yes</span>, the total force is measured along a vector field (see equation (<a href="#x1-109004r28">28<!--tex4ht:ref: eq:gradient_vector --></a>) in
     section <a href="#x1-1090006.2">6.2<!--tex4ht:ref: sec:colvarbias_abf --></a>) that only involves atoms of <span class="t1-zi4r-0-x-x-109">group1</span>. This option is only useful for ABF, or custom biases
     that compute total forces. See section <a href="#x1-1090006.2">6.2<!--tex4ht:ref: sec:colvarbias_abf --></a> for details.
     </li></ul>
<!--l. 1380--><p class="indent">   The value returned is a positive number (in Å), ranging from
<!--l. 1380--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>0</mn></math> to the
largest possible interatomic distance within the chosen boundary conditions (with PBCs, the minimum image
convention is used unless the <span class="t1-zi4r-0-x-x-109">forceNoPBC </span>option is set).
</p>
<div class="flushright">
<!--l. 1386--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1386--><p class="noindent"></p><div class="heading"><a id="sec:cvc_distanceZ">
<h5 class="subsubsectionHead"><span class="titlemark">4.2.2    </span> <a id="x1-330004.2.2"></a><span class="t1-zi4r-0-x-x-109">distanceZ</span>: projection of a distance vector on an axis.</h5>
<!--l. 1386--><p class="noindent"></p></a> </div>  <a id="colvar|distanceZ"> </a>
<!--l. 1389--><p class="indent">   The <span class="t1-zi4r-0-x-x-109">distanceZ</span><span class="t1-zi4r-0-x-x-109"> {...} </span>block defines a distance projection component, which can be seen as measuring the
distance between two groups projected onto an axis, or the position of a group along such an axis. The axis
can be defined using either one reference group and a constant vector, or dynamically based on two
reference groups. One of the groups can be set to a dummy atom to allow the use of an absolute Cartesian
coordinate.
</p><!--l. 1396--><p class="noindent"><span class="ptmb8t-x-x-109">List of keywords </span>(see also <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="colvar|distanceZ|main"> </a> <a id="dx1-33001"></a><span class="colorbox" id="colorbox15"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">main</span></span> <!--l. 1404--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Main
     group of atoms<!--l. 1404--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">distanceZ</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>Block <span class="t1-zi4r-0-x-x-109">main {...}</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Group of atoms whose position <!--l. 1404--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>r</mi></mstyle><mtext></mtext></mstyle></math>
     is measured.
     </li>
<li class="itemize"><a id="colvar|distanceZ|ref"> </a> <a id="dx1-33002"></a><span class="colorbox" id="colorbox16"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">ref</span></span> <!--l. 1415--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Reference
     group of atoms<!--l. 1415--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">distanceZ</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>Block <span class="t1-zi4r-0-x-x-109">ref {...}</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Reference group of atoms. The position of its center of mass is noted <!--l. 1415--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>r</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mn>1</mn></mrow></msub></math>
     below.
     </li>
<li class="itemize"><a id="colvar|distanceZ|ref2"> </a> <a id="dx1-33003"></a><span class="colorbox" id="colorbox17"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">ref2</span></span> <!--l. 1430--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Secondary
     reference group<!--l. 1430--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">distanceZ</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>Block <span class="t1-zi4r-0-x-x-109">ref2 {...}</span><br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span>none <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Optional group of reference atoms, whose position <!--l. 1430--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>r</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mn>2</mn></mrow></msub></math>
     can be used to define a dynamic projection axis: <!--l. 1430--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>e</mi></mstyle><mtext></mtext></mstyle> <mo class="MathClass-rel">=</mo> <msup><mrow><mrow><mo class="MathClass-open">(</mo><mrow><mo class="MathClass-rel">∥</mo><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>r</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mn>2</mn></mrow></msub> <mo class="MathClass-bin">−</mo><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>r</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-rel">∥</mo></mrow><mo class="MathClass-close">)</mo></mrow></mrow><mrow><mo class="MathClass-bin">−</mo><mn>1</mn></mrow></msup> <mo class="MathClass-bin">×</mo><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>r</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mn>2</mn></mrow></msub> <mo class="MathClass-bin">−</mo><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>r</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mn>1</mn></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math>.
     In this case, the origin is <!--l. 1430--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>r</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mi>m</mi></mrow></msub> <mo class="MathClass-rel">=</mo> <mn>1</mn><mo class="MathClass-bin">∕</mo><mn>2</mn><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>r</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mn>1</mn></mrow></msub> <mo class="MathClass-bin">+</mo><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>r</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mn>2</mn></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math>,
     and the value of the component is <!--l. 1430--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>e</mi></mstyle><mtext></mtext></mstyle><mo class="MathClass-bin">⋅</mo><mrow><mo class="MathClass-open">(</mo><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>r</mi></mstyle><mtext></mtext></mstyle><mo class="MathClass-bin">−</mo><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>r</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mi>m</mi></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math>.
     </li>
<li class="itemize"><a id="colvar|distanceZ|axis"> </a> <a id="dx1-33004"></a><span class="colorbox" id="colorbox18"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">axis</span></span> <!--l. 1445--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Projection
     axis (Å)<!--l. 1445--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">distanceZ</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span><span class="t1-zi4r-0-x-x-109">(x, y, z) </span>triplet <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">(0.0, 0.0, 1.0)</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>The three components of this vector define a projection axis <!--l. 1445--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>e</mi></mstyle><mtext></mtext></mstyle></math>
     for the distance vector <!--l. 1445--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>r</mi></mstyle><mtext></mtext></mstyle><mo class="MathClass-bin">−</mo><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>r</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mn>1</mn></mrow></msub></math>
     joining the centers of groups <span class="t1-zi4r-0-x-x-109">ref </span>and <span class="t1-zi4r-0-x-x-109">main</span>. The value of the component is then <!--l. 1445--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>e</mi></mstyle><mtext></mtext></mstyle><mo class="MathClass-bin">⋅</mo><mrow><mo class="MathClass-open">(</mo><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>r</mi></mstyle><mtext></mtext></mstyle><mo class="MathClass-bin">−</mo><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>r</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mn>1</mn></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math>.
     The vector should be written as three components separated by commas and enclosed in parentheses.
     </li>
<li class="itemize"><a id="dx1-33005"></a><span class="colorbox" id="colorbox19"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">forceNoPBC:</span></span> see definition of <a href="#colvar|distance|forceNoPBC"><span class="t1-zi4r-0-x-x-109">forceNoPBC</span></a> (<span class="t1-zi4r-0-x-x-109">distance </span>component)
     </li>
<li class="itemize"><a id="dx1-33006"></a><span class="colorbox" id="colorbox20"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">oneSiteTotalForce:</span></span> see definition of <a href="#colvar|distance|oneSiteTotalForce"><span class="t1-zi4r-0-x-x-109">oneSiteTotalForce</span></a> (<span class="t1-zi4r-0-x-x-109">distance </span>component)</li></ul>
<!--l. 1453--><p class="noindent">This component returns a number (in Å) whose range is determined by the chosen boundary conditions. For instance, if
the <!--l. 1454--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math>
axis is used in a simulation with periodic boundaries, the returned value ranges between
<!--l. 1456--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-bin">−</mo><msub><mrow><mi>b</mi></mrow><mrow><mi>z</mi></mrow></msub><mo class="MathClass-bin">∕</mo><mn>2</mn></math> and
<!--l. 1456--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>b</mi></mrow><mrow><mi>z</mi></mrow></msub><mo class="MathClass-bin">∕</mo><mn>2</mn></math>, where
<!--l. 1456--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>b</mi></mrow><mrow><mi>z</mi></mrow></msub></math> is the box
length along <!--l. 1457--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math>
(this behavior is disabled if <span class="t1-zi4r-0-x-x-109">forceNoPBC </span>is set).
</p>
<div class="flushright">
<!--l. 1460--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1460--><p class="noindent"></p><div class="heading"><a id="sec:cvc_distanceXY">
<h5 class="subsubsectionHead"><span class="titlemark">4.2.3    </span> <a id="x1-340004.2.3"></a><span class="t1-zi4r-0-x-x-109">distanceXY</span>: modulus of the projection of a distance vector on a plane.</h5>
<!--l. 1460--><p class="noindent"></p></a> </div>  <a id="colvar|distanceXY"> </a>
<!--l. 1463--><p class="indent">   The <span class="t1-zi4r-0-x-x-109">distanceXY</span><span class="t1-zi4r-0-x-x-109"> {...} </span>block defines a distance projected on a plane, and accepts the same keywords as the
component <span class="t1-zi4r-0-x-x-109">distanceZ</span>, i.e. <span class="t1-zi4r-0-x-x-109">main</span>, <span class="t1-zi4r-0-x-x-109">ref</span>, either <span class="t1-zi4r-0-x-x-109">ref2 </span>or <span class="t1-zi4r-0-x-x-109">axis</span>, and <span class="t1-zi4r-0-x-x-109">oneSiteTotalForce</span>. It returns the norm
of the projection of the distance vector between <span class="t1-zi4r-0-x-x-109">main </span>and <span class="t1-zi4r-0-x-x-109">ref </span>onto the plane orthogonal to the axis.
The axis is defined using the <span class="t1-zi4r-0-x-x-109">axis </span>parameter or as the vector joining <span class="t1-zi4r-0-x-x-109">ref </span>and <span class="t1-zi4r-0-x-x-109">ref2 </span>(see <span class="t1-zi4r-0-x-x-109">distanceZ</span>
above).
</p><!--l. 1472--><p class="noindent"><span class="ptmb8t-x-x-109">List of keywords </span>(see also <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-34001"></a><span class="colorbox" id="colorbox21"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">main:</span></span> see definition of <a href="#colvar|distanceZ|main"><span class="t1-zi4r-0-x-x-109">main</span></a> (<span class="t1-zi4r-0-x-x-109">distanceZ </span>component)
     </li>
<li class="itemize"><a id="dx1-34002"></a><span class="colorbox" id="colorbox22"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">ref:</span></span> see definition of <a href="#colvar|distanceZ|ref"><span class="t1-zi4r-0-x-x-109">ref</span></a> (<span class="t1-zi4r-0-x-x-109">distanceZ </span>component)
     </li>
<li class="itemize"><a id="dx1-34003"></a><span class="colorbox" id="colorbox23"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">ref2:</span></span> see definition of <a href="#colvar|distanceZ|ref2"><span class="t1-zi4r-0-x-x-109">ref2</span></a> (<span class="t1-zi4r-0-x-x-109">distanceZ </span>component)
                                                                                         
                                                                                         
     </li>
<li class="itemize"><a id="dx1-34004"></a><span class="colorbox" id="colorbox24"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">axis:</span></span> see definition of <a href="#colvar|distanceZ|axis"><span class="t1-zi4r-0-x-x-109">axis</span></a> (<span class="t1-zi4r-0-x-x-109">distanceZ </span>component)
     </li>
<li class="itemize"><a id="dx1-34005"></a><span class="colorbox" id="colorbox25"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">forceNoPBC:</span></span> see definition of <a href="#colvar|distance|forceNoPBC"><span class="t1-zi4r-0-x-x-109">forceNoPBC</span></a> (<span class="t1-zi4r-0-x-x-109">distance </span>component)
     </li>
<li class="itemize"><a id="dx1-34006"></a><span class="colorbox" id="colorbox26"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">oneSiteTotalForce:</span></span> see definition of <a href="#colvar|distance|oneSiteTotalForce"><span class="t1-zi4r-0-x-x-109">oneSiteTotalForce</span></a> (<span class="t1-zi4r-0-x-x-109">distance </span>component)</li></ul>
<div class="flushright">
<!--l. 1489--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1489--><p class="noindent"></p><div class="heading"><a id="sec:cvc_distanceVec">
<h5 class="subsubsectionHead"><span class="titlemark">4.2.4    </span> <a id="x1-350004.2.4"></a><span class="t1-zi4r-0-x-x-109">distanceVec</span>: distance vector between two groups.</h5>
<!--l. 1489--><p class="noindent"></p></a> </div>  <a id="colvar|distanceVec"> </a>
<!--l. 1492--><p class="indent">   The <span class="t1-zi4r-0-x-x-109">distanceVec</span><span class="t1-zi4r-0-x-x-109"> {...} </span>block defines a distance vector component, which accepts the same keywords as the
component <span class="t1-zi4r-0-x-x-109">distance</span>: <span class="t1-zi4r-0-x-x-109">group1</span>, <span class="t1-zi4r-0-x-x-109">group2</span>, and <span class="t1-zi4r-0-x-x-109">forceNoPBC</span>. Its value is the 3-vector joining the centers of mass of
<span class="t1-zi4r-0-x-x-109">group1 </span>and <span class="t1-zi4r-0-x-x-109">group2</span>.
</p><!--l. 1498--><p class="noindent"><span class="ptmb8t-x-x-109">List of keywords </span>(see also <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-35001"></a><span class="colorbox" id="colorbox27"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">group1:</span></span> see definition of <a href="#colvar|distance|group1"><span class="t1-zi4r-0-x-x-109">group1</span></a> (<span class="t1-zi4r-0-x-x-109">distance </span>component)
     </li>
<li class="itemize"><a id="dx1-35002"></a><span class="colorbox" id="colorbox28"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">group2:</span></span> analogous to <span class="t1-zi4r-0-x-x-109">group1</span>
</li>
<li class="itemize"><a id="dx1-35003"></a><span class="colorbox" id="colorbox29"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">forceNoPBC:</span></span> see definition of <a href="#colvar|distance|forceNoPBC"><span class="t1-zi4r-0-x-x-109">forceNoPBC</span></a> (<span class="t1-zi4r-0-x-x-109">distance </span>component)
     </li>
<li class="itemize"><a id="dx1-35004"></a><span class="colorbox" id="colorbox30"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">oneSiteTotalForce:</span></span> see definition of <a href="#colvar|distance|oneSiteTotalForce"><span class="t1-zi4r-0-x-x-109">oneSiteTotalForce</span></a> (<span class="t1-zi4r-0-x-x-109">distance </span>component)</li></ul>
<div class="flushright">
<!--l. 1511--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1511--><p class="noindent"></p><div class="heading"><a id="sec:cvc_distanceDir">
<h5 class="subsubsectionHead"><span class="titlemark">4.2.5    </span> <a id="x1-360004.2.5"></a><span class="t1-zi4r-0-x-x-109">distanceDir</span>: distance unit vector between two groups.</h5>
<!--l. 1511--><p class="noindent"></p></a> </div>  <a id="colvar|distanceDir"> </a>
<!--l. 1514--><p class="indent">   The <span class="t1-zi4r-0-x-x-109">distanceDir</span><span class="t1-zi4r-0-x-x-109"> {...} </span>block defines a distance unit vector component, which accepts the same keywords as
the component <span class="t1-zi4r-0-x-x-109">distance</span>: <span class="t1-zi4r-0-x-x-109">group1</span>, <span class="t1-zi4r-0-x-x-109">group2</span>, and <span class="t1-zi4r-0-x-x-109">forceNoPBC</span>. It returns a 3-dimensional unit vector
<!--l. 1518--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="bold">d</mi> <mo class="MathClass-rel">=</mo> <mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>d</mi></mrow><mrow><mi>x</mi></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>d</mi></mrow><mrow><mi>y</mi></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>d</mi></mrow><mrow><mi>z</mi></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math>, with
<!--l. 1519--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-rel">|</mo><mi mathvariant="bold">d</mi><mo class="MathClass-rel">|</mo> <mo class="MathClass-rel">=</mo> <mn>1</mn></math>.
</p><!--l. 1521--><p class="noindent"><span class="ptmb8t-x-x-109">List of keywords </span>(see also <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-36001"></a><span class="colorbox" id="colorbox31"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">group1:</span></span> see definition of <a href="#colvar|distance|group1"><span class="t1-zi4r-0-x-x-109">group1</span></a> (<span class="t1-zi4r-0-x-x-109">distance </span>component)
     </li>
<li class="itemize"><a id="dx1-36002"></a><span class="colorbox" id="colorbox32"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">group2:</span></span> analogous to <span class="t1-zi4r-0-x-x-109">group1</span>
</li>
<li class="itemize"><a id="dx1-36003"></a><span class="colorbox" id="colorbox33"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">forceNoPBC:</span></span> see definition of <a href="#colvar|distance|forceNoPBC"><span class="t1-zi4r-0-x-x-109">forceNoPBC</span></a> (<span class="t1-zi4r-0-x-x-109">distance </span>component)
     </li>
<li class="itemize"><a id="dx1-36004"></a><span class="colorbox" id="colorbox34"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">oneSiteTotalForce:</span></span> see definition of <a href="#colvar|distance|oneSiteTotalForce"><span class="t1-zi4r-0-x-x-109">oneSiteTotalForce</span></a> (<span class="t1-zi4r-0-x-x-109">distance </span>component)</li></ul>
<div class="flushright">
<!--l. 1533--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1533--><p class="noindent"></p><div class="heading"><a id="sec:cvc_distanceInv">
<h5 class="subsubsectionHead"><span class="titlemark">4.2.6    </span> <a id="x1-370004.2.6"></a><span class="t1-zi4r-0-x-x-109">distanceInv</span>: mean distance between two groups of atoms.</h5>
<!--l. 1533--><p class="noindent"></p></a> </div>  <a id="colvar|distanceInv"> </a>
<!--l. 1536--><p class="indent">   The <span class="t1-zi4r-0-x-x-109">distanceInv</span><span class="t1-zi4r-0-x-x-109"> {...} </span>block defines a generalized mean distance between two groups of atoms 1 and 2, weighted
with exponent <!--l. 1536--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>1</mn><mo class="MathClass-bin">∕</mo><mi>n</mi></math>:
</p><table class="equation"><tr><td> <a id="x1-37001r2"></a>
<!--l. 1537--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<msubsup><mrow><mi>d</mi></mrow><mrow><mi mathvariant="normal">1,2</mi></mrow><mrow><mrow><mo class="MathClass-open">[</mo><mrow><mi>n</mi></mrow><mo class="MathClass-close">]</mo></mrow></mrow></msubsup><mspace class="tmspace" width="3.04074pt"></mspace> <mo class="MathClass-rel">=</mo> <mspace class="tmspace" width="3.04074pt"></mspace><msup><mrow> <mfenced close=")" open="(" separators=""><mrow> <mfrac><mrow><mn>1</mn></mrow>
<mrow><msub><mrow><mi>N</mi></mrow><mrow><mi mathvariant="normal">1</mi></mrow></msub><msub><mrow><mi>N</mi></mrow><mrow><mi mathvariant="normal">2</mi></mrow></msub></mrow></mfrac><munder class="msub"><mrow><mo mathsize="big">∑</mo></mrow><mrow><mi>i</mi><mo class="MathClass-punc">,</mo><mi>j</mi></mrow></munder><msup><mrow> <mfenced close=")" open="(" separators=""><mrow> <mfrac><mrow><mn>1</mn></mrow>
<mrow><mo class="MathClass-rel">∥</mo><msup><mrow><mi mathvariant="bold">d</mi></mrow><mrow><mi>i</mi><mi>j</mi></mrow></msup><mo class="MathClass-rel">∥</mo></mrow></mfrac></mrow></mfenced></mrow><mrow><mi>n</mi></mrow></msup></mrow></mfenced></mrow><mrow><mo class="MathClass-bin">−</mo><mn>1</mn><mo class="MathClass-bin">∕</mo><mi>n</mi></mrow></msup>
</math></td><td class="eq-no">(2)</td></tr></table>
<!--l. 1541--><p class="indent">   where <!--l. 1541--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-rel">∥</mo><msup><mrow><mi mathvariant="bold">d</mi></mrow><mrow><mi>i</mi><mi>j</mi></mrow></msup><mo class="MathClass-rel">∥</mo></math> is the
distance between atoms <!--l. 1541--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math>
and <!--l. 1541--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>j</mi></math> in groups 1 and
2 respectively, and <!--l. 1541--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>n</mi></math>
is an even integer.
</p><!--l. 1543--><p class="noindent"><span class="ptmb8t-x-x-109">List of keywords </span>(see also <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-37002"></a><span class="colorbox" id="colorbox35"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">group1:</span></span> see definition of <a href="#colvar|distance|group1"><span class="t1-zi4r-0-x-x-109">group1</span></a> (<span class="t1-zi4r-0-x-x-109">distance </span>component)
     </li>
<li class="itemize"><a id="dx1-37003"></a><span class="colorbox" id="colorbox36"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">group2:</span></span> analogous to <span class="t1-zi4r-0-x-x-109">group1</span>
</li>
<li class="itemize"><a id="dx1-37004"></a><span class="colorbox" id="colorbox37"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">oneSiteTotalForce:</span></span> see definition of <a href="#colvar|distance|oneSiteTotalForce"><span class="t1-zi4r-0-x-x-109">oneSiteTotalForce</span></a> (<span class="t1-zi4r-0-x-x-109">distance </span>component)
     </li>
<li class="itemize"><a id="dx1-37005"></a><span class="colorbox" id="colorbox38"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">exponent</span></span> <!--l. 1556--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Exponent
     <!--l. 1556--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>n</mi></math>
     in equation <a href="#x1-37001r2">2<!--tex4ht:ref: eq:distanceInv --></a><!--l. 1556--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">distanceInv</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive even integer <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span>6 <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Defines the exponent to which the individual distances are elevated before averaging.
     The default value of 6 is useful for example to applying restraints based on NOE-measured distances.
     </li></ul>
<!--l. 1558--><p class="noindent">This component returns a number in Å, ranging from <!--l. 1558--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>0</mn></math>
to the largest possible distance within the chosen boundary conditions.
</p>
<div class="flushright">
<!--l. 1561--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1561--><p class="noindent"></p><div class="heading"><a id="sec:cvc_angles">
<h4 class="subsectionHead"><span class="titlemark">4.3    </span> <a id="x1-380004.3"></a>Angles</h4>
<!--l. 1561--><p class="noindent"></p></a> </div> 
<div class="flushright">
<!--l. 1564--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1564--><p class="noindent"></p><div class="heading"><a id="sec:cvc_angle">
<h5 class="subsubsectionHead"><span class="titlemark">4.3.1    </span> <a id="x1-390004.3.1"></a><span class="t1-zi4r-0-x-x-109">angle</span>: angle between three groups.</h5>
<!--l. 1564--><p class="noindent"></p></a> </div>  <a id="colvar|angle"> </a>
<!--l. 1567--><p class="indent">   The <span class="t1-zi4r-0-x-x-109">angle</span><span class="t1-zi4r-0-x-x-109"> {...} </span>block defines an angle, and contains the three blocks <span class="t1-zi4r-0-x-x-109">group1</span>, <span class="t1-zi4r-0-x-x-109">group2</span>
and <span class="t1-zi4r-0-x-x-109">group3</span>, defining the three groups. It returns an angle (in degrees) within the interval
<!--l. 1570--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">[</mo><mrow><mn>0</mn> <mo class="MathClass-punc">:</mo> <mn>1</mn><mn>8</mn><mn>0</mn></mrow><mo class="MathClass-close">]</mo></mrow></math>.
</p><!--l. 1572--><p class="noindent"><span class="ptmb8t-x-x-109">List of keywords </span>(see also <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-39001"></a><span class="colorbox" id="colorbox39"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">group1:</span></span> see definition of <a href="#colvar|distance|group1"><span class="t1-zi4r-0-x-x-109">group1</span></a> (<span class="t1-zi4r-0-x-x-109">distance </span>component)
     </li>
<li class="itemize"><a id="dx1-39002"></a><span class="colorbox" id="colorbox40"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">group2:</span></span> analogous to <span class="t1-zi4r-0-x-x-109">group1</span>
</li>
<li class="itemize"><a id="dx1-39003"></a><span class="colorbox" id="colorbox41"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">group3:</span></span> analogous to <span class="t1-zi4r-0-x-x-109">group1</span>
</li>
<li class="itemize"><a id="dx1-39004"></a><span class="colorbox" id="colorbox42"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">forceNoPBC:</span></span> see definition of <a href="#colvar|distance|forceNoPBC"><span class="t1-zi4r-0-x-x-109">forceNoPBC</span></a> (<span class="t1-zi4r-0-x-x-109">distance </span>component)
     </li>
<li class="itemize"><a id="dx1-39005"></a><span class="colorbox" id="colorbox43"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">oneSiteTotalForce:</span></span> see definition of <a href="#colvar|distance|oneSiteTotalForce"><span class="t1-zi4r-0-x-x-109">oneSiteTotalForce</span></a> (<span class="t1-zi4r-0-x-x-109">distance </span>component)</li></ul>
<div class="flushright">
<!--l. 1587--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1587--><p class="noindent"></p><div class="heading"><a id="sec:cvc_dipoleAngle">
<h5 class="subsubsectionHead"><span class="titlemark">4.3.2    </span> <a id="x1-400004.3.2"></a><span class="t1-zi4r-0-x-x-109">dipoleAngle</span>: angle between two groups and dipole of a third group.</h5>
<!--l. 1587--><p class="noindent"></p></a> </div>  <a id="colvar|dipoleAngle"> </a>
<!--l. 1590--><p class="indent">   The <span class="t1-zi4r-0-x-x-109">dipoleAngle</span><span class="t1-zi4r-0-x-x-109"> {...} </span>block defines an angle, and contains the three blocks <span class="t1-zi4r-0-x-x-109">group1</span>, <span class="t1-zi4r-0-x-x-109">group2 </span>and <span class="t1-zi4r-0-x-x-109">group3</span>,
defining the three groups, being <span class="t1-zi4r-0-x-x-109">group1 </span>the group where dipole is calculated. It returns an angle (in degrees) within the
interval <!--l. 1593--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">[</mo><mrow><mn>0</mn> <mo class="MathClass-punc">:</mo> <mn>1</mn><mn>8</mn><mn>0</mn></mrow><mo class="MathClass-close">]</mo></mrow></math>.
</p><!--l. 1595--><p class="noindent"><span class="ptmb8t-x-x-109">List of keywords </span>(see also <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-40001"></a><span class="colorbox" id="colorbox44"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">group1:</span></span> see definition of <a href="#colvar|distance|group1"><span class="t1-zi4r-0-x-x-109">group1</span></a> (<span class="t1-zi4r-0-x-x-109">distance </span>component)
     </li>
<li class="itemize"><a id="dx1-40002"></a><span class="colorbox" id="colorbox45"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">group2:</span></span> analogous to <span class="t1-zi4r-0-x-x-109">group1</span>
</li>
<li class="itemize"><a id="dx1-40003"></a><span class="colorbox" id="colorbox46"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">group3:</span></span> analogous to <span class="t1-zi4r-0-x-x-109">group1</span>
</li>
<li class="itemize"><a id="dx1-40004"></a><span class="colorbox" id="colorbox47"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">forceNoPBC:</span></span> see definition of <a href="#colvar|distance|forceNoPBC"><span class="t1-zi4r-0-x-x-109">forceNoPBC</span></a> (<span class="t1-zi4r-0-x-x-109">distance </span>component)
     </li>
<li class="itemize"><a id="dx1-40005"></a><span class="colorbox" id="colorbox48"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">oneSiteTotalForce:</span></span> see definition of <a href="#colvar|distance|oneSiteTotalForce"><span class="t1-zi4r-0-x-x-109">oneSiteTotalForce</span></a> (<span class="t1-zi4r-0-x-x-109">distance </span>component)</li></ul>
<div class="flushright">
<!--l. 1609--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1609--><p class="noindent"></p><div class="heading"><a id="sec:cvc_dihedral">
<h5 class="subsubsectionHead"><span class="titlemark">4.3.3    </span> <a id="x1-410004.3.3"></a><span class="t1-zi4r-0-x-x-109">dihedral</span>: torsional angle between four groups.</h5>
<!--l. 1609--><p class="noindent"></p></a> </div>  <a id="colvar|dihedral"> </a>
<!--l. 1612--><p class="indent">   The <span class="t1-zi4r-0-x-x-109">dihedral</span><span class="t1-zi4r-0-x-x-109"> {...} </span>block defines a torsional angle, and contains the blocks <span class="t1-zi4r-0-x-x-109">group1</span>, <span class="t1-zi4r-0-x-x-109">group2</span>,
<span class="t1-zi4r-0-x-x-109">group3 </span>and <span class="t1-zi4r-0-x-x-109">group4</span>, defining the four groups. It returns an angle (in degrees) within the interval
<!--l. 1615--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">[</mo><mrow><mo class="MathClass-bin">−</mo><mn>1</mn><mn>8</mn><mn>0</mn> <mo class="MathClass-punc">:</mo> <mn>1</mn><mn>8</mn><mn>0</mn></mrow><mo class="MathClass-close">]</mo></mrow></math>. The
Colvars module calculates all the distances between two angles taking into account periodicity. For
instance, reference values for restraints or range boundaries can be defined by using any real number of
choice.
</p><!--l. 1620--><p class="noindent"><span class="ptmb8t-x-x-109">List of keywords </span>(see also <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-41001"></a><span class="colorbox" id="colorbox49"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">group1:</span></span> see definition of <a href="#colvar|distance|group1"><span class="t1-zi4r-0-x-x-109">group1</span></a> (<span class="t1-zi4r-0-x-x-109">distance </span>component)
     </li>
<li class="itemize"><a id="dx1-41002"></a><span class="colorbox" id="colorbox50"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">group2:</span></span> analogous to <span class="t1-zi4r-0-x-x-109">group1</span>
</li>
<li class="itemize"><a id="dx1-41003"></a><span class="colorbox" id="colorbox51"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">group3:</span></span> analogous to <span class="t1-zi4r-0-x-x-109">group1</span>
</li>
<li class="itemize"><a id="dx1-41004"></a><span class="colorbox" id="colorbox52"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">group4:</span></span> analogous to <span class="t1-zi4r-0-x-x-109">group1</span>
</li>
<li class="itemize"><a id="dx1-41005"></a><span class="colorbox" id="colorbox53"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">forceNoPBC:</span></span> see definition of <a href="#colvar|distance|forceNoPBC"><span class="t1-zi4r-0-x-x-109">forceNoPBC</span></a> (<span class="t1-zi4r-0-x-x-109">distance </span>component)
     </li>
<li class="itemize"><a id="dx1-41006"></a><span class="colorbox" id="colorbox54"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">oneSiteTotalForce:</span></span> see definition of <a href="#colvar|distance|oneSiteTotalForce"><span class="t1-zi4r-0-x-x-109">oneSiteTotalForce</span></a> (<span class="t1-zi4r-0-x-x-109">distance </span>component)</li></ul>
<div class="flushright">
<!--l. 1636--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1636--><p class="noindent"></p><div class="heading"><a id="sec:cvc_polarTheta">
<h5 class="subsubsectionHead"><span class="titlemark">4.3.4    </span> <a id="x1-420004.3.4"></a><span class="t1-zi4r-0-x-x-109">polarTheta</span>: polar angle in spherical coordinates.</h5>
<!--l. 1636--><p class="noindent"></p></a> </div>  <a id="colvar|polarTheta"> </a>
<!--l. 1639--><p class="indent">   The <span class="t1-zi4r-0-x-x-109">polarTheta</span><span class="t1-zi4r-0-x-x-109"> {...} </span>block defines the polar angle in spherical coordinates, for the center of mass of
a group of atoms described by the block <span class="t1-zi4r-0-x-x-109">atoms</span>. It returns an angle (in degrees) within the interval
<!--l. 1642--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">[</mo><mrow><mn>0</mn> <mo class="MathClass-punc">:</mo> <mn>1</mn><mn>8</mn><mn>0</mn></mrow><mo class="MathClass-close">]</mo></mrow></math>. To
obtain spherical coordinates in a frame of reference tied to another group of atoms, use the <span class="t1-zi4r-0-x-x-109">fittingGroup </span>(<a href="#x1-1020005.2">5.2<!--tex4ht:ref: sec:colvar_atom_groups_ref_frame --></a>)
option within the <span class="t1-zi4r-0-x-x-109">atoms </span>block. An example is provided in file <span class="t1-zi4r-0-x-x-109">examples/11_polar_angles.in </span>of the Colvars
public repository.
</p><!--l. 1648--><p class="noindent"><span class="ptmb8t-x-x-109">List of keywords </span>(see also <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="colvar|polarTheta|atoms"> </a> <a id="dx1-42001"></a><span class="colorbox" id="colorbox55"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">atoms</span></span> <!--l. 1657--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Atom
     group<!--l. 1657--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">polarPhi</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span><span class="t1-zi4r-0-x-x-109">atoms</span><span class="t1-zi4r-0-x-x-109"> {...} </span>block <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Defines the group of atoms for the COM of which the angle should be calculated. </li></ul>
<div class="flushright">
<!--l. 1661--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1661--><p class="noindent"></p><div class="heading"><a id="sec:cvc_polarPhi">
<h5 class="subsubsectionHead"><span class="titlemark">4.3.5    </span> <a id="x1-430004.3.5"></a><span class="t1-zi4r-0-x-x-109">polarPhi</span>: azimuthal angle in spherical coordinates.</h5>
<!--l. 1661--><p class="noindent"></p></a> </div>  <a id="colvar|polarPhi"> </a>
<!--l. 1664--><p class="indent">   The <span class="t1-zi4r-0-x-x-109">polarPhi</span><span class="t1-zi4r-0-x-x-109"> {...} </span>block defines the azimuthal angle in spherical coordinates, for the center of mass of
a group of atoms described by the block <span class="t1-zi4r-0-x-x-109">atoms</span>. It returns an angle (in degrees) within the interval
<!--l. 1667--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">[</mo><mrow><mo class="MathClass-bin">−</mo><mn>1</mn><mn>8</mn><mn>0</mn> <mo class="MathClass-punc">:</mo> <mn>1</mn><mn>8</mn><mn>0</mn></mrow><mo class="MathClass-close">]</mo></mrow></math>. The
Colvars module calculates all the distances between two angles taking into account periodicity. For instance,
reference values for restraints or range boundaries can be defined by using any real number of choice. To obtain
spherical coordinates in a frame of reference tied to another group of atoms, use the <span class="t1-zi4r-0-x-x-109">fittingGroup </span>(<a href="#x1-1020005.2">5.2<!--tex4ht:ref: sec:colvar_atom_groups_ref_frame --></a>) option
within the <span class="t1-zi4r-0-x-x-109">atoms </span>block. An example is provided in file <span class="t1-zi4r-0-x-x-109">examples/11_polar_angles.in </span>of the Colvars public
repository.
</p><!--l. 1677--><p class="noindent"><span class="ptmb8t-x-x-109">List of keywords </span>(see also <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="colvar|polarPhi|atoms"> </a> <a id="dx1-43001"></a><span class="colorbox" id="colorbox56"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">atoms</span></span> <!--l. 1686--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Atom
     group<!--l. 1686--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">polarPhi</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span><span class="t1-zi4r-0-x-x-109">atoms</span><span class="t1-zi4r-0-x-x-109"> {...} </span>block <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Defines the group of atoms for the COM of which the angle should be calculated. </li></ul>
<div class="flushright">
<!--l. 1690--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1690--><p class="noindent"></p><div class="heading"><a id="sec:cvc_contacts">
<h4 class="subsectionHead"><span class="titlemark">4.4    </span> <a id="x1-440004.4"></a>Contacts</h4>
<!--l. 1690--><p class="noindent"></p></a> </div> 
<div class="flushright">
<!--l. 1693--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1693--><p class="noindent"></p><div class="heading"><a id="sec:cvc_coordNum">
<h5 class="subsubsectionHead"><span class="titlemark">4.4.1    </span> <a id="x1-450004.4.1"></a><span class="t1-zi4r-0-x-x-109">coordNum</span>: coordination number between two groups.</h5>
<!--l. 1693--><p class="noindent"></p></a> </div>  <a id="colvar|coordNum"> </a>
<!--l. 1696--><p class="indent">   The <span class="t1-zi4r-0-x-x-109">coordNum {...} </span>block defines a coordination number (or number of contacts), which calculates the function
<!--l. 1698--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">(</mo><mrow><mn>1</mn><mo class="MathClass-bin">−</mo><msup><mrow><mrow><mo class="MathClass-open">(</mo><mrow><mi>d</mi><mo class="MathClass-bin">∕</mo><msub><mrow><mi>d</mi></mrow><mrow><mn>0</mn></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></mrow><mrow><mi>n</mi></mrow></msup></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-bin">∕</mo><mrow><mo class="MathClass-open">(</mo><mrow><mn>1</mn><mo class="MathClass-bin">−</mo><msup><mrow><mrow><mo class="MathClass-open">(</mo><mrow><mi>d</mi><mo class="MathClass-bin">∕</mo><msub><mrow><mi>d</mi></mrow><mrow><mn>0</mn></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></mrow><mrow><mi>m</mi></mrow></msup></mrow><mo class="MathClass-close">)</mo></mrow></math>, where
<!--l. 1698--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>d</mi></mrow><mrow><mn>0</mn></mrow></msub></math> is the “cutoff"
distance, and <!--l. 1699--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>n</mi></math>
and <!--l. 1699--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>m</mi></math> are
exponents that can control its long range behavior and stiffness <span class="cite">[<a href="#XIannuzzi2003">3</a>]</span>. This function is summed over all pairs of atoms
in <span class="t1-zi4r-0-x-x-109">group1 </span>and <span class="t1-zi4r-0-x-x-109">group2</span>: </p><table class="equation"><tr><td> <a id="x1-45001r3"></a>
<!--l. 1703--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>C</mi><mrow><mo class="MathClass-open">(</mo><mrow><mi mathvariant="monospace">group1</mi><mo class="MathClass-punc">,</mo><mi mathvariant="monospace">group2</mi></mrow><mo class="MathClass-close">)</mo></mrow><mspace class="tmspace" width="3.04074pt"></mspace> <mo class="MathClass-rel">=</mo> <mspace class="tmspace" width="3.04074pt"></mspace><munder class="msub"><mrow><mo mathsize="big">∑</mo></mrow><mrow><mi>i</mi><mo class="MathClass-rel">∈</mo><mi mathvariant="monospace">group1</mi></mrow></munder><munder class="msub"><mrow><mo mathsize="big">∑</mo></mrow><mrow><mi>j</mi><mo class="MathClass-rel">∈</mo><mi mathvariant="monospace">group2</mi></mrow></munder><mfrac><mrow><mn>1</mn><mo class="MathClass-bin">−</mo><msup><mrow><mrow><mo class="MathClass-open">(</mo><mrow><mo class="MathClass-rel">|</mo><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-bin">−</mo><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>j</mi></mrow></msub><mo class="MathClass-rel">|</mo><mo class="MathClass-bin">∕</mo><msub><mrow><mi>d</mi></mrow><mrow><mn>0</mn></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></mrow><mrow><mi>n</mi></mrow></msup></mrow>
<mrow><mn>1</mn><mo class="MathClass-bin">−</mo><msup><mrow><mrow><mo class="MathClass-open">(</mo><mrow><mo class="MathClass-rel">|</mo><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-bin">−</mo><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>j</mi></mrow></msub><mo class="MathClass-rel">|</mo><mo class="MathClass-bin">∕</mo><msub><mrow><mi>d</mi></mrow><mrow><mn>0</mn></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></mrow><mrow><mi>m</mi></mrow></msup></mrow></mfrac>
</math></td><td class="eq-no">(3)</td></tr></table>
<!--l. 1712--><p class="noindent"><span class="ptmb8t-x-x-109">List of keywords </span>(see also <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="colvar|coordNum|group1"> </a> <a id="dx1-45002"></a><span class="colorbox" id="colorbox57"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">group1:</span></span> see definition of <a href="#colvar|distance|group1"><span class="t1-zi4r-0-x-x-109">group1</span></a> (<span class="t1-zi4r-0-x-x-109">distance </span>component)
     </li>
<li class="itemize"><a id="colvar|coordNum|group2"> </a> <a id="dx1-45003"></a><span class="colorbox" id="colorbox58"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">group2:</span></span> analogous to <span class="t1-zi4r-0-x-x-109">group1</span>
</li>
<li class="itemize"><a id="colvar|coordNum|cutoff"> </a> <a id="dx1-45004"></a><span class="colorbox" id="colorbox59"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">cutoff</span></span> <!--l. 1735--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>“Interaction"
     distance (Å)<!--l. 1735--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">coordNum</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive decimal <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span>4.0 <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>This number defines the switching distance to define an interatomic contact: for <!--l. 1735--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>d</mi> <mo class="MathClass-rel">≪</mo> <msub><mrow><mi>d</mi></mrow><mrow><mn>0</mn></mrow></msub></math>,
     the switching function <!--l. 1735--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">(</mo><mrow><mn>1</mn><mo class="MathClass-bin">−</mo><msup><mrow><mrow><mo class="MathClass-open">(</mo><mrow><mi>d</mi><mo class="MathClass-bin">∕</mo><msub><mrow><mi>d</mi></mrow><mrow><mn>0</mn></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></mrow><mrow><mi>n</mi></mrow></msup></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-bin">∕</mo><mrow><mo class="MathClass-open">(</mo><mrow><mn>1</mn><mo class="MathClass-bin">−</mo><msup><mrow><mrow><mo class="MathClass-open">(</mo><mrow><mi>d</mi><mo class="MathClass-bin">∕</mo><msub><mrow><mi>d</mi></mrow><mrow><mn>0</mn></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></mrow><mrow><mi>m</mi></mrow></msup></mrow><mo class="MathClass-close">)</mo></mrow></math>
     is close to 1, at <!--l. 1735--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>d</mi> <mo class="MathClass-rel">=</mo> <msub><mrow><mi>d</mi></mrow><mrow><mn>0</mn></mrow></msub></math>
     it has a value of <!--l. 1735--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>n</mi><mo class="MathClass-bin">∕</mo><mi>m</mi></math>
     (<!--l. 1735--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>1</mn><mo class="MathClass-bin">∕</mo><mn>2</mn></math>
     with the default <!--l. 1735--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>n</mi></math>
     and <!--l. 1735--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>m</mi></math>),
                                                                                         
                                                                                         
     and at <!--l. 1735--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>d</mi> <mo class="MathClass-rel">≫</mo> <msub><mrow><mi>d</mi></mrow><mrow><mn>0</mn></mrow></msub></math>
     it goes to zero approximately like <!--l. 1735--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow><mi>d</mi></mrow><mrow><mi>m</mi><mo class="MathClass-bin">−</mo><mi>n</mi></mrow></msup></math>.
     Hence, for a proper behavior, <!--l. 1735--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>m</mi></math>
     must be larger than <!--l. 1735--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>n</mi></math>.
     </li>
<li class="itemize"><a id="colvar|coordNum|cutoff3"> </a> <a id="dx1-45005"></a><span class="colorbox" id="colorbox60"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">cutoff3</span></span> <!--l. 1747--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Reference
     distance vector (Å)<!--l. 1747--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">coordNum</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>“<span class="t1-zi4r-0-x-x-109">(x, y, z)</span>" triplet of positive decimals <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">(4.0, 4.0, 4.0)</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>The three components of this vector define three different cutoffs <!--l. 1747--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>d</mi></mrow><mrow><mn>0</mn></mrow></msub></math>
     for each direction. This option is mutually exclusive with <span class="t1-zi4r-0-x-x-109">cutoff</span>.
     </li>
<li class="itemize"><a id="colvar|coordNum|expNumer"> </a> <a id="dx1-45006"></a><span class="colorbox" id="colorbox61"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">expNumer</span></span> <!--l. 1757--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Numerator
     exponent<!--l. 1757--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">coordNum</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive even integer <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span>6 <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>This number defines the <!--l. 1757--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>n</mi></math>
     exponent for the switching function.
     </li>
<li class="itemize"><a id="colvar|coordNum|expDenom"> </a> <a id="dx1-45007"></a><span class="colorbox" id="colorbox62"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">expDenom</span></span> <!--l. 1767--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Denominator
     exponent<!--l. 1767--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">coordNum</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive even integer <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span>12 <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>This number defines the <!--l. 1767--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>m</mi></math>
     exponent for the switching function.
     </li>
<li class="itemize"><a id="colvar|coordNum|group2CenterOnly"> </a> <a id="dx1-45008"></a><span class="colorbox" id="colorbox63"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">group2CenterOnly</span></span> <!--l. 1780--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Use
     only <span class="t1-zi4r-0-x-x-109">group2</span>'s center of mass<!--l. 1780--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">coordNum</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">off</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>If this option is <span class="t1-zi4r-0-x-x-109">on</span>, only contacts between each atoms in <span class="t1-zi4r-0-x-x-109">group1 </span>and the center of mass
     of <span class="t1-zi4r-0-x-x-109">group2 </span>are calculated (by default, the sum extends over all pairs of atoms in <span class="t1-zi4r-0-x-x-109">group1 </span>and <span class="t1-zi4r-0-x-x-109">group2</span>).
     If <span class="t1-zi4r-0-x-x-109">group2 </span>is a <span class="t1-zi4r-0-x-x-109">dummyAtom</span>, this option is set to <span class="t1-zi4r-0-x-x-109">yes </span>by default.
     </li>
<li class="itemize"><a id="colvar|coordNum|tolerance"> </a> <a id="dx1-45009"></a><span class="colorbox" id="colorbox64"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">tolerance</span></span> <!--l. 1790--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Pairlist
     control<!--l. 1790--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">coordNum</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>decimal <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span>0.0 <br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>This  controls  the  pairlist  feature,  dictating  the  minimum  value  for  each  summation
                                                                                         
                                                                                         
     element in Eq. <a href="#x1-45001r3">3<!--tex4ht:ref: eq:cvc_coordNum --></a> such that the pair that contributed the summation element is included in subsequent
     simulation timesteps until the next pairlist recalculation. For most applications, this value should be
     small  (eg.  0.001)  to  avoid  missing  important  contributions  to  the  overall  sum.  Higher  values  will
     improve performance by reducing the number of pairs that contribute to the sum. Values above 1
     will exclude all possible pair interactions. Similarly, values below 0 will never exclude a pair from
     consideration. To ensure continuous forces, Eq. <a href="#x1-45001r3">3<!--tex4ht:ref: eq:cvc_coordNum --></a> is further modified by subtracting the tolerance and
     then rescaling so that each pair covers the range <!--l. 1790--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mfenced close="]" open="[" separators=""><mrow><mn>0</mn><mo class="MathClass-punc">,</mo><mn>1</mn></mrow></mfenced></math>.
     </li>
<li class="itemize"><a id="colvar|coordNum|pairListFrequency"> </a> <a id="dx1-45010"></a><span class="colorbox" id="colorbox65"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">pairListFrequency</span></span> <!--l. 1800--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Pairlist
     regeneration frequency<!--l. 1800--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">coordNum</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive integer <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span>100 <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>This controls the pairlist feature, dictating how many steps are taken between regenerating
     pairlists if the tolerance is greater than 0.  </li></ul>
<!--l. 1803--><p class="indent">   This component returns a dimensionless number, which ranges from approximately 0 (all interatomic distances are much larger than
the cutoff) to <!--l. 1805--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>N</mi></mrow><mrow><mi mathvariant="monospace">group1</mi></mrow></msub><mo class="MathClass-bin">×</mo><msub><mrow><mi>N</mi></mrow><mrow><mi mathvariant="monospace">group2</mi></mrow></msub></math> (all distances
are less than the cutoff), or <!--l. 1806--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>N</mi></mrow><mrow><mi mathvariant="monospace">group1</mi></mrow></msub></math>
if <span class="t1-zi4r-0-x-x-109">group2CenterOnly </span>is used. For performance reasons, at least one of <span class="t1-zi4r-0-x-x-109">group1 </span>and <span class="t1-zi4r-0-x-x-109">group2 </span>should
be of limited size or <span class="t1-zi4r-0-x-x-109">group2CenterOnly </span>should be used: the cost of the loop over all pairs grows as
<!--l. 1808--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>N</mi></mrow><mrow><mi mathvariant="monospace">group1</mi></mrow></msub><mo class="MathClass-bin">×</mo><msub><mrow><mi>N</mi></mrow><mrow><mi mathvariant="monospace">group2</mi></mrow></msub></math>. Setting
<!--l. 1809--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="monospace">tolerance</mi> <mo class="MathClass-rel">&gt;</mo> <mn>0</mn></math>
ameliorates this to some degree, although every pair is still checked to regenerate the pairlist.
</p>
<div class="flushright">
<!--l. 1813--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1813--><p class="noindent"></p><div class="heading"><a id="sec:cvc_selfCoordNum">
<h5 class="subsubsectionHead"><span class="titlemark">4.4.2    </span> <a id="x1-460004.4.2"></a><span class="t1-zi4r-0-x-x-109">selfCoordNum</span>: coordination number between atoms within a group.</h5>
<!--l. 1813--><p class="noindent"></p></a> </div>  <a id="colvar|selfCoordNum"> </a>
<!--l. 1816--><p class="indent">   The <span class="t1-zi4r-0-x-x-109">selfCoordNum {...} </span>block defines a coordination number similarly to the component <span class="t1-zi4r-0-x-x-109">coordNum</span>, but the
function is summed over atom pairs within <span class="t1-zi4r-0-x-x-109">group1</span>: </p><table class="equation"><tr><td> <a id="x1-46001r4"></a>
<!--l. 1819--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>C</mi><mrow><mo class="MathClass-open">(</mo><mrow><mi mathvariant="monospace">group1</mi></mrow><mo class="MathClass-close">)</mo></mrow><mspace class="tmspace" width="3.04074pt"></mspace> <mo class="MathClass-rel">=</mo> <mspace class="tmspace" width="3.04074pt"></mspace><munder class="msub"><mrow><mo mathsize="big">∑</mo></mrow><mrow><mi>i</mi><mo class="MathClass-rel">∈</mo><mi mathvariant="monospace">group1</mi></mrow></munder><munder class="msub"><mrow><mo mathsize="big">∑</mo></mrow><mrow><mi>j</mi><mo class="MathClass-rel">&gt;</mo><mi>i</mi></mrow></munder> <mfrac><mrow><mn>1</mn><mo class="MathClass-bin">−</mo><msup><mrow><mrow><mo class="MathClass-open">(</mo><mrow><mo class="MathClass-rel">|</mo><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-bin">−</mo><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>j</mi></mrow></msub><mo class="MathClass-rel">|</mo><mo class="MathClass-bin">∕</mo><msub><mrow><mi>d</mi></mrow><mrow><mn>0</mn></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></mrow><mrow><mi>n</mi></mrow></msup></mrow>
<mrow><mn>1</mn><mo class="MathClass-bin">−</mo><msup><mrow><mrow><mo class="MathClass-open">(</mo><mrow><mo class="MathClass-rel">|</mo><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-bin">−</mo><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>j</mi></mrow></msub><mo class="MathClass-rel">|</mo><mo class="MathClass-bin">∕</mo><msub><mrow><mi>d</mi></mrow><mrow><mn>0</mn></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></mrow><mrow><mi>m</mi></mrow></msup></mrow></mfrac>
</math></td><td class="eq-no">(4)</td></tr></table>
<!--l. 1827--><p class="indent">   The keywords accepted by <span class="t1-zi4r-0-x-x-109">selfCoordNum </span>are a subset of those accepted by <span class="t1-zi4r-0-x-x-109">coordNum</span>, namely <span class="t1-zi4r-0-x-x-109">group1 </span>(here
defining <span class="ptmri8t-x-x-109">all </span>of the atoms to be considered), <span class="t1-zi4r-0-x-x-109">cutoff</span>, <span class="t1-zi4r-0-x-x-109">expNumer</span>, and <span class="t1-zi4r-0-x-x-109">expDenom</span>.
</p><!--l. 1832--><p class="noindent"><span class="ptmb8t-x-x-109">List of keywords </span>(see also <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-46002"></a><span class="colorbox" id="colorbox66"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">group1:</span></span> see definition of <a href="#colvar|coordNum|group1"><span class="t1-zi4r-0-x-x-109">group1</span></a> (<span class="t1-zi4r-0-x-x-109">coordNum </span>component)
     </li>
<li class="itemize"><a id="dx1-46003"></a><span class="colorbox" id="colorbox67"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">cutoff:</span></span> see definition of <a href="#colvar|coordNum|cutoff"><span class="t1-zi4r-0-x-x-109">cutoff</span></a> (<span class="t1-zi4r-0-x-x-109">coordNum </span>component)
     </li>
<li class="itemize"><a id="dx1-46004"></a><span class="colorbox" id="colorbox68"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">cutoff3:</span></span> see definition of <a href="#colvar|coordNum|cutoff3"><span class="t1-zi4r-0-x-x-109">cutoff3</span></a> (<span class="t1-zi4r-0-x-x-109">coordNum </span>component)
     </li>
<li class="itemize"><a id="dx1-46005"></a><span class="colorbox" id="colorbox69"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">expNumer:</span></span> see definition of <a href="#colvar|coordNum|expNumer"><span class="t1-zi4r-0-x-x-109">expNumer</span></a> (<span class="t1-zi4r-0-x-x-109">coordNum </span>component)
     </li>
<li class="itemize"><a id="dx1-46006"></a><span class="colorbox" id="colorbox70"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">expDenom:</span></span> see definition of <a href="#colvar|coordNum|expDenom"><span class="t1-zi4r-0-x-x-109">expDenom</span></a> (<span class="t1-zi4r-0-x-x-109">coordNum </span>component)
     </li>
<li class="itemize"><a id="dx1-46007"></a><span class="colorbox" id="colorbox71"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">tolerance:</span></span> see definition of <a href="#colvar|coordNum|tolerance"><span class="t1-zi4r-0-x-x-109">tolerance</span></a> (<span class="t1-zi4r-0-x-x-109">coordNum </span>component)
     </li>
<li class="itemize"><a id="dx1-46008"></a><span class="colorbox" id="colorbox72"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">pairListFrequency:</span></span> see definition of <a href="#colvar|coordNum|pairListFrequency"><span class="t1-zi4r-0-x-x-109">pairListFrequency</span></a> (<span class="t1-zi4r-0-x-x-109">coordNum </span>component)</li></ul>
<!--l. 1849--><p class="indent">   This component returns a dimensionless number, which ranges from approximately 0 (all interatomic distances much larger
than the cutoff) to <!--l. 1851--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>N</mi></mrow><mrow><mi mathvariant="monospace">group1</mi></mrow></msub><mo class="MathClass-bin">×</mo><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>N</mi></mrow><mrow><mi mathvariant="monospace">group1</mi></mrow></msub><mo class="MathClass-bin">−</mo><mn>1</mn></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-bin">∕</mo><mn>2</mn></math>
(all distances within the cutoff). For performance reasons, <span class="t1-zi4r-0-x-x-109">group1 </span>should be of limited size, because the cost of the loop over
all pairs grows as <!--l. 1854--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msubsup><mrow><mi>N</mi></mrow><mrow><mi mathvariant="monospace">group1</mi></mrow><mrow><mn>2</mn></mrow></msubsup></math>.
</p>
<div class="flushright">
<!--l. 1858--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1858--><p class="noindent"></p><div class="heading"><a id="sec:cvc_hBond">
<h5 class="subsubsectionHead"><span class="titlemark">4.4.3    </span> <a id="x1-470004.4.3"></a><span class="t1-zi4r-0-x-x-109">hBond</span>: hydrogen bond between two atoms.</h5>
<!--l. 1858--><p class="noindent"></p></a> </div>  <a id="colvar|hBond"> </a>
<!--l. 1861--><p class="indent">   The <span class="t1-zi4r-0-x-x-109">hBond {...} </span>block defines a hydrogen bond, implemented as a coordination number (eq. <a href="#x1-45001r3">3<!--tex4ht:ref: eq:cvc_coordNum --></a>) between the
donor and the acceptor atoms. Therefore, it accepts the same options <span class="t1-zi4r-0-x-x-109">cutoff </span>(with a different default value of
3.3 Å), <span class="t1-zi4r-0-x-x-109">expNumer </span>(with a default value of 6) and <span class="t1-zi4r-0-x-x-109">expDenom </span>(with a default value of 8). Unlike <span class="t1-zi4r-0-x-x-109">coordNum</span>, it requires
two atom numbers, <span class="t1-zi4r-0-x-x-109">acceptor </span>and <span class="t1-zi4r-0-x-x-109">donor</span>, to be defined. It returns an adimensional number, with values between 0
                                                                                         
                                                                                         
(acceptor and donor far outside the cutoff distance) and 1 (acceptor and donor much closer than the
cutoff).
</p><!--l. 1872--><p class="noindent"><span class="ptmb8t-x-x-109">List of keywords </span>(see also <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-47001"></a><span class="colorbox" id="colorbox73"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">acceptor</span></span> <!--l. 1879--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Number
     of the acceptor atom<!--l. 1879--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">hBond</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive integer <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Number that uses the same convention as <span class="t1-zi4r-0-x-x-109">atomNumbers</span>.
     </li>
<li class="itemize"><a id="dx1-47002"></a><span class="colorbox" id="colorbox74"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">donor:</span></span> analogous to <span class="t1-zi4r-0-x-x-109">acceptor</span>
</li>
<li class="itemize"><a id="dx1-47003"></a><span class="colorbox" id="colorbox75"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">cutoff:</span></span> see definition of <a href="#colvar|coordNum|cutoff"><span class="t1-zi4r-0-x-x-109">cutoff</span></a> (<span class="t1-zi4r-0-x-x-109">coordNum </span>component)<br class="newline"/><span class="ptmb8t-x-x-109">Note: </span>default value is 3.3 Å.
     </li>
<li class="itemize"><a id="dx1-47004"></a><span class="colorbox" id="colorbox76"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">expNumer:</span></span> see definition of <a href="#colvar|coordNum|expNumer"><span class="t1-zi4r-0-x-x-109">expNumer</span></a> (<span class="t1-zi4r-0-x-x-109">coordNum </span>component)<br class="newline"/><span class="ptmb8t-x-x-109">Note: </span>default value is 6.
     </li>
<li class="itemize"><a id="dx1-47005"></a><span class="colorbox" id="colorbox77"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">expDenom:</span></span> see definition of <a href="#colvar|coordNum|expDenom"><span class="t1-zi4r-0-x-x-109">expDenom</span></a> (<span class="t1-zi4r-0-x-x-109">coordNum </span>component)<br class="newline"/><span class="ptmb8t-x-x-109">Note: </span>default value is 8.</li></ul>
<div class="flushright">
<!--l. 1894--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1894--><p class="noindent"></p><div class="heading"><a id="sec:cvc_collective">
<h4 class="subsectionHead"><span class="titlemark">4.5    </span> <a id="x1-480004.5"></a>Collective metrics</h4>
<!--l. 1894--><p class="noindent"></p></a> </div> 
<div class="flushright">
<!--l. 1897--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1897--><p class="noindent"></p><div class="heading"><a id="sec:cvc_rmsd">
<h5 class="subsubsectionHead"><span class="titlemark">4.5.1    </span> <a id="x1-490004.5.1"></a><span class="t1-zi4r-0-x-x-109">rmsd</span>: root mean square displacement (RMSD) from reference positions.</h5>
<!--l. 1897--><p class="noindent"></p></a> </div>  <a id="colvar|rmsd"> </a>
<!--l. 1900--><p class="indent">   The block <span class="t1-zi4r-0-x-x-109">rmsd</span><span class="t1-zi4r-0-x-x-109"> {...} </span>defines the root mean square replacement (RMSD) of
a group of atoms with respect to a reference structure. For each set of coordinates
<!--l. 1902--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">{</mo><mrow><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mn>1</mn></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-punc">,</mo><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mn>2</mn></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-punc">,</mo><mo class="MathClass-op">…</mo><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>N</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow></mrow><mo class="MathClass-close">}</mo></mrow></math>, the colvar component <span class="t1-zi4r-0-x-x-109">rmsd</span>
calculates the optimal rotation <!--l. 1905--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow><mi>U</mi></mrow><mrow><mrow><mo class="MathClass-open">{</mo><mrow><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>i</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow></mrow><mo class="MathClass-close">}</mo></mrow><mo class="MathClass-rel">→</mo><mrow><mo class="MathClass-open">{</mo><mrow><msubsup><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>i</mi></mrow><mrow><mi mathvariant="normal">(ref)</mi></mrow></msubsup></mrow><mo class="MathClass-close">}</mo></mrow></mrow></msup></math> that
best superimposes the coordinates <!--l. 1906--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">{</mo><mrow><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>i</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow></mrow><mo class="MathClass-close">}</mo></mrow></math>
onto a set of reference coordinates <!--l. 1907--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">{</mo><mrow><msubsup><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>i</mi></mrow><mrow><mi mathvariant="normal">(ref)</mi></mrow></msubsup></mrow><mo class="MathClass-close">}</mo></mrow></math>.
Both the current and the reference coordinates are centered on their centers of geometry,
<!--l. 1909--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi mathvariant="normal">cog</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow></math> and
<!--l. 1910--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msubsup><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi mathvariant="normal">cog</mi></mrow><mrow><mi mathvariant="normal">(ref)</mi></mrow></msubsup></math>. The
root mean square displacement is then defined as: </p><table class="equation"><tr><td> <a id="x1-49001r5"></a>
<!--l. 1912--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi mathvariant="normal">RMSD</mi><mrow><mo class="MathClass-open">(</mo><mrow><mrow><mo class="MathClass-open">{</mo><mrow><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>i</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow></mrow><mo class="MathClass-close">}</mo></mrow><mo class="MathClass-punc">,</mo><mrow><mo class="MathClass-open">{</mo><mrow><msubsup><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>i</mi></mrow><mrow><mi mathvariant="normal">(ref)</mi></mrow></msubsup></mrow><mo class="MathClass-close">}</mo></mrow></mrow><mo class="MathClass-close">)</mo></mrow><mspace class="tmspace" width="3.04074pt"></mspace> <mo class="MathClass-rel">=</mo> <mspace class="tmspace" width="3.04074pt"></mspace><msqrt><mrow> <mfrac> <mrow> <mn>1</mn></mrow>
<mrow><mi>N</mi></mrow></mfrac><munderover accent="false" accentunder="false"><mrow><mo mathsize="big">∑</mo></mrow><mrow><mi>i</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><mi>N</mi></mrow></munderover><msup><mrow> <mfenced close="|" open="|" separators=""><mrow><mi>U</mi> <mfenced close=")" open="(" separators=""><mrow><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>i</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-bin">−</mo><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi mathvariant="normal">cog</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow></mrow></mfenced><mo class="MathClass-bin">−</mo><mfenced close=")" open="(" separators=""><mrow><msubsup><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>i</mi></mrow><mrow><mi mathvariant="normal">(ref)</mi></mrow></msubsup> <mo class="MathClass-bin">−</mo><msubsup><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi mathvariant="normal">cog</mi></mrow><mrow><mi mathvariant="normal">(ref)</mi></mrow></msubsup></mrow></mfenced></mrow></mfenced></mrow><mrow><mn>2</mn></mrow></msup></mrow></msqrt>
</math></td><td class="eq-no">(5)</td></tr></table>
<!--l. 1922--><p class="indent">   The optimal rotation <!--l. 1923--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow><mi>U</mi></mrow><mrow><mrow><mo class="MathClass-open">{</mo><mrow><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>i</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow></mrow><mo class="MathClass-close">}</mo></mrow><mo class="MathClass-rel">→</mo><mrow><mo class="MathClass-open">{</mo><mrow><msubsup><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>i</mi></mrow><mrow><mi mathvariant="normal">(ref)</mi></mrow></msubsup></mrow><mo class="MathClass-close">}</mo></mrow></mrow></msup></math>
is calculated within the formalism developed in reference <span class="cite">[<a href="#XCoutsias2004">4</a>]</span>, which guarantees a continuous dependence of
<!--l. 1927--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow><mi>U</mi></mrow><mrow><mrow><mo class="MathClass-open">{</mo><mrow><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>i</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow></mrow><mo class="MathClass-close">}</mo></mrow><mo class="MathClass-rel">→</mo><mrow><mo class="MathClass-open">{</mo><mrow><msubsup><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>i</mi></mrow><mrow><mi mathvariant="normal">(ref)</mi></mrow></msubsup></mrow><mo class="MathClass-close">}</mo></mrow></mrow></msup></math> with respect
to <!--l. 1928--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">{</mo><mrow><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>i</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow></mrow><mo class="MathClass-close">}</mo></mrow></math>.
</p><!--l. 1930--><p class="noindent"><span class="ptmb8t-x-x-109">List of keywords </span>(see also <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="colvar|rmsd|atoms"> </a> <a id="dx1-49002"></a><span class="colorbox" id="colorbox78"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">atoms</span></span> <!--l. 1944--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Atom
     group<!--l. 1944--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">rmsd</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span><span class="t1-zi4r-0-x-x-109">atoms</span><span class="t1-zi4r-0-x-x-109"> {...} </span>block <br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>Defines  the  group  of  atoms  of  which  the  RMSD  should  be  calculated.  Optimal  fit
     options (such as <span class="t1-zi4r-0-x-x-109">refPositions </span>and <span class="t1-zi4r-0-x-x-109">rotateToReference</span>) should typically NOT be set within this
     block. Exceptions to this rule are the special cases discussed in the <span class="ptmri8t-x-x-109">Advanced usage </span>paragraph below.
     </li>
<li class="itemize"><a id="colvar|rmsd|refPositions"> </a> <a id="dx1-49003"></a><span class="colorbox" id="colorbox79"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositions</span></span> <!--l. 1955--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Reference
     coordinates<!--l. 1955--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">rmsd</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>space-separated list of <span class="t1-zi4r-0-x-x-109">(x, y, z) </span>triplets <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>This option (mutually exclusive with <span class="t1-zi4r-0-x-x-109">refPositionsFile</span>) sets the reference coordinates
     for RMSD calculation, and uses these to compute the roto-translational fit. See the equivalent option
     <a href="#atom-group|refPositions"><span class="t1-zi4r-0-x-x-109">refPositions</span></a> within the atom group definition for details on acceptable formats and other features.
                                                                                         
                                                                                         
     </li>
<li class="itemize"><a id="colvar|rmsd|refPositionsFile"> </a> <a id="dx1-49004"></a><span class="colorbox" id="colorbox80"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositionsFile</span></span> <!--l. 1966--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Reference
     coordinates file<!--l. 1966--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">rmsd</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>UNIX filename <br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>This option (mutually exclusive with <span class="t1-zi4r-0-x-x-109">refPositions</span>) sets the reference coordinates
     for RMSD calculation, and uses these to compute the roto-translational fit. See the equivalent option
     <a href="#atom-group|refPositionsFile"><span class="t1-zi4r-0-x-x-109">refPositionsFile</span></a> within the atom group definition for details on acceptable file formats and other
     features.
     </li>
<li class="itemize"><a id="colvar|rmsd|refPositionsCol"> </a> <a id="dx1-49005"></a><span class="colorbox" id="colorbox81"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositionsCol</span></span> <!--l. 1991--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>PDB
     column containing atom flags<!--l. 1991--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">rmsd</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span><span class="t1-zi4r-0-x-x-109">O</span>, <span class="t1-zi4r-0-x-x-109">B</span>, <span class="t1-zi4r-0-x-x-109">X</span>, <span class="t1-zi4r-0-x-x-109">Y</span>, or <span class="t1-zi4r-0-x-x-109">Z</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>If <span class="t1-zi4r-0-x-x-109">refPositionsFile </span>is a PDB file that contains all the atoms in the topology, this
     option may be provided to set which PDB field is used to flag the reference coordinates for <span class="t1-zi4r-0-x-x-109">atoms</span>.
     </li>
<li class="itemize"><a id="colvar|rmsd|refPositionsColValue"> </a> <a id="dx1-49006"></a><span class="colorbox" id="colorbox82"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositionsColValue</span></span> <!--l. 1991--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Atom
     selection flag in the PDB column<!--l. 1991--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">rmsd</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive decimal <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>If defined, this value identifies in the PDB column <span class="t1-zi4r-0-x-x-109">refPositionsCol </span>of the file <span class="t1-zi4r-0-x-x-109">refPositionsFile</span>
     which atom positions are to be read. Otherwise, all positions with a non-zero value are read.
     </li>
<li class="itemize"><a id="colvar|rmsd|atomPermutation"> </a> <a id="dx1-49007"></a><span class="colorbox" id="colorbox83"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">atomPermutation</span></span> <!--l. 2015--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Alternate
     ordering of atoms for RMSD computation<!--l. 2015--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">rmsd</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>List of atom numbers <br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>If  defined,  this  parameter  defines  a  re-ordering  (permutation)  of  the  1-based  atom
     numbers that can be used to compute the RMSD, typically due to molecular symmetry. This parameter
     can be specified multiple times, each one defining a new permutation: the returned RMSD value is
     the minimum over the set of permutations. For example, if the atoms making up the group are 6, 7, 8,
     9, and atoms 7, 8, and 9 are invariant by circular permutation (as the hydrogens in a CH3 group), a
     symmetry-adapted RMSD would be obtained by adding:<br class="newline"/><span class="t1-zi4r-0-x-x-109">atomPermutation 6 8 9 7</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">atomPermutation 6 9 7 8</span><br class="newline"/>Note that <span class="ptmri8t-x-x-109">this does not affect the least-squares roto-translational fit</span>, which is done using the topology
     ordering of atoms, and the reference positions in the order provided. Therefore, this feature is mostly
     useful when using custom fitting parameters within the atom group, such as <a href="#atom-group|fittingGroup"><span class="t1-zi4r-0-x-x-109">fittingGroup</span></a>, or when
     fitting is disabled altogether. </li></ul>
<!--l. 2017--><p class="noindent">This component returns a positive real number (in Å).
</p>
<div class="flushright">
<!--l. 2020--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2020--><p class="noindent"></p><div class="heading"><a id="sec:cvc_rmsd_advanced">
<h5 class="subsubsectionHead"><span class="titlemark">4.5.2    </span> <a id="x1-500004.5.2"></a>Advanced usage of the <span class="t1-zi4r-0-x-x-109">rmsd </span>component.</h5>
<!--l. 2020--><p class="noindent"></p></a> </div>  In the standard usage as described above, the <span class="t1-zi4r-0-x-x-109">rmsd </span>component calculates a minimum RMSD, that is, current
coordinates are optimally fitted onto the same reference coordinates that are used to compute the RMSD value. The
fit itself is handled by the atom group object, whose parameters are automatically set by the <span class="t1-zi4r-0-x-x-109">rmsd </span>component. For
very specific applications, however, it may be useful to control the fitting process separately from the definition of
the reference coordinates, to evaluate various types of non-minimal RMSD values. This can be achieved by setting
the related options (<span class="t1-zi4r-0-x-x-109">refPositions</span>, etc.) explicitly in the atom group block. This allows for the following
non-standard cases:
<!--l. 2034--><p class="indent">
</p><ol class="enumerate1">
<li class="enumerate" id="x1-50002x1">applying the optimal translation, but no rotation (<span class="t1-zi4r-0-x-x-109">rotateToReference off</span>), to bias or restrain the
     shape and orientation, but not the position of the atom group;
     </li>
<li class="enumerate" id="x1-50004x2">applying the optimal rotation, but no translation (<span class="t1-zi4r-0-x-x-109">centerToReference off</span>), to bias or restrain the
     shape and position, but not the orientation of the atom group;
     </li>
<li class="enumerate" id="x1-50006x3">disabling the application of optimal roto-translations, which lets the RMSD component describe the
     deviation of atoms from fixed positions in the laboratory frame: this allows for custom positional
     restraints within the Colvars module;
     </li>
<li class="enumerate" id="x1-50008x4">fitting the atomic positions to different reference coordinates than those used in the RMSD calculation
     itself (by specifying <a href="#atom-group|refPositions"><span class="t1-zi4r-0-x-x-109">refPositions</span></a> or <a href="#atom-group|refPositionsFile"><span class="t1-zi4r-0-x-x-109">refPositionsFile</span></a> within the atom group as well as within
     the <span class="t1-zi4r-0-x-x-109">rmsd </span>block);
     </li>
<li class="enumerate" id="x1-50010x5">applying  the  optimal  rotation  and/or  translation  from  a  separate  atom  group,  defined  through
     <span class="t1-zi4r-0-x-x-109">fittingGroup</span>: the RMSD then reflects the deviation from reference coordinates in a separate, moving
     reference frame (see example in the section on <a href="#atom-group|fittingGroup"><span class="t1-zi4r-0-x-x-109">fittingGroup</span></a>).</li></ol>
<div class="flushright">
<!--l. 2057--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2057--><p class="noindent"></p><div class="heading"><a id="sec:cvc_eigenvector">
<h5 class="subsubsectionHead"><span class="titlemark">4.5.3    </span> <a id="x1-510004.5.3"></a><span class="t1-zi4r-0-x-x-109">eigenvector</span>: projection of the atomic coordinates on a vector.</h5>
<!--l. 2057--><p class="noindent"></p></a> </div>  <a id="colvar|eigenvector"> </a>
<!--l. 2060--><p class="indent">   The block <span class="t1-zi4r-0-x-x-109">eigenvector</span><span class="t1-zi4r-0-x-x-109"> {...} </span>defines the projection of the coordinates of a group of
atoms (or more precisely, their deviations from the reference coordinates) onto a vector in
<!--l. 2062--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow><mi>ℝ</mi></mrow><mrow><mn>3</mn><mi>n</mi></mrow></msup></math>, where
<!--l. 2062--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>n</mi></math> is the
number of atoms in the group. The computed quantity is the total projection: </p><table class="equation"><tr><td> <a id="x1-51001r6"></a>
<!--l. 2065--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>p</mi><mrow><mo class="MathClass-open">(</mo><mrow><mrow><mo class="MathClass-open">{</mo><mrow><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>i</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow></mrow><mo class="MathClass-close">}</mo></mrow><mo class="MathClass-punc">,</mo><mrow><mo class="MathClass-open">{</mo><mrow><msubsup><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>i</mi></mrow><mrow><mi mathvariant="normal">(ref)</mi></mrow></msubsup></mrow><mo class="MathClass-close">}</mo></mrow></mrow><mo class="MathClass-close">)</mo></mrow><mspace class="tmspace" width="3.04074pt"></mspace> <mo class="MathClass-rel">=</mo> <mspace class="tmspace" width="3.04074pt"></mspace><munderover accent="false" accentunder="false"><mrow><mo mathsize="big">∑</mo></mrow><mrow>
<mi>i</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><mi>n</mi></mrow></munderover><msub><mrow><mi mathvariant="bold">v</mi></mrow><mrow>
<mi>i</mi></mrow></msub><mo class="MathClass-bin">⋅</mo><mfenced close=")" open="(" separators=""><mrow><mi>U</mi><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>i</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-bin">−</mo><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi mathvariant="normal">cog</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-bin">−</mo><mrow><mo class="MathClass-open">(</mo><mrow><msubsup><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>i</mi></mrow><mrow><mi mathvariant="normal">(ref)</mi></mrow></msubsup> <mo class="MathClass-bin">−</mo><msubsup><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi mathvariant="normal">
cog</mi></mrow><mrow><mi mathvariant="normal">(ref)</mi></mrow></msubsup></mrow><mo class="MathClass-close">)</mo></mrow></mrow></mfenced><mi mathvariant="normal">,</mi>
</math></td><td class="eq-no">(6)</td></tr></table>
<!--l. 2074--><p class="indent">   where, as in the <span class="t1-zi4r-0-x-x-109">rmsd </span>component, <!--l. 2074--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>U</mi></math>
is the optimal rotation matrix, <!--l. 2075--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi mathvariant="normal">cog</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow></math>
and <!--l. 2076--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msubsup><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi mathvariant="normal">cog</mi></mrow><mrow><mi mathvariant="normal">(ref)</mi></mrow></msubsup></math>
are the centers of geometry of the current and reference positions respectively, and
<!--l. 2078--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="bold">v</mi></mrow><mrow><mi>i</mi></mrow></msub></math>
are the components of the vector for each atom. Example choices for
<!--l. 2079--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi mathvariant="bold">v</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math> are an
eigenvector of the covariance matrix (essential mode), or a normal mode of the system. It is assumed that
<!--l. 2081--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mo class="MathClass-op">∑</mo><!--nolimits--></mrow><mrow><mi>i</mi></mrow></msub><msub><mrow><mi mathvariant="bold">v</mi></mrow><mrow><mi>i</mi></mrow></msub> <mo class="MathClass-rel">=</mo> <mn>0</mn></math>: otherwise, the Colvars
module centers the <!--l. 2082--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="bold">v</mi></mrow><mrow><mi>i</mi></mrow></msub></math>
automatically when reading them from the configuration.
</p><!--l. 2085--><p class="noindent"><span class="ptmb8t-x-x-109">List of keywords </span>(see also <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-51002"></a><span class="colorbox" id="colorbox84"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">atoms:</span></span> see definition of <a href="#colvar|rmsd|atoms"><span class="t1-zi4r-0-x-x-109">atoms</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
<li class="itemize"><a id="dx1-51003"></a><span class="colorbox" id="colorbox85"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositions:</span></span> see definition of <a href="#colvar|rmsd|refPositions"><span class="t1-zi4r-0-x-x-109">refPositions</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
<li class="itemize"><a id="dx1-51004"></a><span class="colorbox" id="colorbox86"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositionsFile:</span></span> see definition of <a href="#colvar|rmsd|refPositionsFile"><span class="t1-zi4r-0-x-x-109">refPositionsFile</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
<li class="itemize"><a id="dx1-51005"></a><span class="colorbox" id="colorbox87"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositionsCol:</span></span> see definition of <a href="#colvar|rmsd|refPositionsCol"><span class="t1-zi4r-0-x-x-109">refPositionsCol</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
<li class="itemize"><a id="dx1-51006"></a><span class="colorbox" id="colorbox88"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositionsColValue:</span></span> see definition of <a href="#colvar|rmsd|refPositionsColValue"><span class="t1-zi4r-0-x-x-109">refPositionsColValue</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
                                                                                         
                                                                                         
     </li>
<li class="itemize"><a id="dx1-51007"></a><span class="colorbox" id="colorbox89"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">vector</span></span> <!--l. 2105--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Vector
     components<!--l. 2105--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">eigenvector</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>space-separated list of <span class="t1-zi4r-0-x-x-109">(x, y, z) </span>triplets <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>This option (mutually exclusive with <span class="t1-zi4r-0-x-x-109">vectorFile</span>) sets the values of the vector components.
     </li>
<li class="itemize"><a id="dx1-51008"></a><span class="colorbox" id="colorbox90"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">vectorFile</span></span> <!--l. 2116--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>file
     containing vector components<!--l. 2116--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">eigenvector</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>UNIX filename <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>This option (mutually exclusive with <span class="t1-zi4r-0-x-x-109">vector</span>) sets the name of a coordinate file containing
     the vector components; the file is read according to the same format used for <span class="t1-zi4r-0-x-x-109">refPositionsFile</span>.
     For a PDB file specifically, the components are read from the X, Y and Z fields. <span class="ptmb8t-x-x-109">Note: </span><span class="ptmri8t-x-x-109">The PDB</span>
<span class="ptmri8t-x-x-109">file has limited precision and fixed-point numbers: in some cases, the vector components may not be</span>
<span class="ptmri8t-x-x-109">accurately represented; a XYZ file should be used instead, containing floating-point numbers.</span>
</li>
<li class="itemize"><a id="dx1-51009"></a><span class="colorbox" id="colorbox91"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">vectorCol</span></span> <!--l. 2134--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>PDB
     column used to flag participating atoms<!--l. 2134--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">eigenvector</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span><span class="t1-zi4r-0-x-x-109">O </span>or <span class="t1-zi4r-0-x-x-109">B</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Analogous to <span class="t1-zi4r-0-x-x-109">atomsCol</span>.
     </li>
<li class="itemize"><a id="dx1-51010"></a><span class="colorbox" id="colorbox92"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">vectorColValue</span></span> <!--l. 2134--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Value
     used to flag participating atoms in the PDB file<!--l. 2134--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">eigenvector</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive decimal <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Analogous to <span class="t1-zi4r-0-x-x-109">atomsColValue</span>.
     </li>
<li class="itemize"><a id="dx1-51011"></a><span class="colorbox" id="colorbox93"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">differenceVector</span></span> <!--l. 2146--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>The
     <!--l. 2146--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>3</mn><mi>n</mi></math>-dimensional
     vector is the difference between <span class="t1-zi4r-0-x-x-109">vector </span>and <span class="t1-zi4r-0-x-x-109">refPositions</span><!--l. 2146--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">eigenvector</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">off</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>If this option is <span class="t1-zi4r-0-x-x-109">on</span>, the numbers provided by <span class="t1-zi4r-0-x-x-109">vector </span>or <span class="t1-zi4r-0-x-x-109">vectorFile </span>are interpreted as
     another set of positions, <!--l. 2146--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msubsup><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>i</mi></mrow><mrow><mi>′</mi></mrow></msubsup></math>:
     the vector <!--l. 2146--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="bold">v</mi></mrow><mrow><mi>i</mi></mrow></msub></math>
     is then defined as <!--l. 2146--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="bold">v</mi></mrow><mrow><mi>i</mi></mrow></msub> <mo class="MathClass-rel">=</mo> <mfenced close=")" open="(" separators=""><mrow><msubsup><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>i</mi></mrow><mrow><mi>′</mi></mrow></msubsup><mo class="MathClass-bin">−</mo><msubsup><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>i</mi></mrow><mrow><mi mathvariant="normal">(ref)</mi></mrow></msubsup></mrow></mfenced></math>.
     This allows to conveniently define a colvar <!--l. 2146--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>
     as a projection on the linear transformation between two sets of positions, “A" and “B". For convenience,
     the vector is also normalized so that <!--l. 2146--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi> <mo class="MathClass-rel">=</mo> <mn>0</mn></math>
     when the atoms are at the set of positions “A" and <!--l. 2146--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi> <mo class="MathClass-rel">=</mo> <mn>1</mn></math>
     at the set of positions “B".  </li></ul>
<!--l. 2148--><p class="noindent">This component returns a number (in Å), whose value ranges between the smallest and largest absolute positions in the
                                                                                         
                                                                                         
unit cell during the simulations (see also <span class="t1-zi4r-0-x-x-109">distanceZ</span>). Due to the normalization in eq. <a href="#x1-51001r6">6<!--tex4ht:ref: eq:cvc_eigenvector --></a>, this range does not depend
on the number of atoms involved.
</p>
<div class="flushright">
<!--l. 2155--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2155--><p class="noindent"></p><div class="heading"><a id="sec:cvc_gyration">
<h5 class="subsubsectionHead"><span class="titlemark">4.5.4    </span> <a id="x1-520004.5.4"></a><span class="t1-zi4r-0-x-x-109">gyration</span>: radius of gyration of a group of atoms.</h5>
<!--l. 2155--><p class="noindent"></p></a> </div>  <a id="colvar|gyration"> </a>
<!--l. 2158--><p class="indent">   The block <span class="t1-zi4r-0-x-x-109">gyration</span><span class="t1-zi4r-0-x-x-109"> {...} </span>defines the parameters for calculating the radius of gyration of a group of atomic positions
<!--l. 2160--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">{</mo><mrow><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mn>1</mn></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-punc">,</mo><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mn>2</mn></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-punc">,</mo><mo class="MathClass-op">…</mo><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>N</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow></mrow><mo class="MathClass-close">}</mo></mrow></math> with respect to their
center of geometry, <!--l. 2162--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi mathvariant="normal">cog</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow></math>: </p><table class="equation"><tr><td>
<a id="x1-52001r7"></a>
<!--l. 2163--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<msub><mrow><mi>R</mi></mrow><mrow><mi mathvariant="normal">gyr</mi></mrow></msub><mspace class="tmspace" width="3.04074pt"></mspace> <mo class="MathClass-rel">=</mo> <mspace class="tmspace" width="3.04074pt"></mspace><msqrt><mrow> <mfrac> <mrow> <mn>1</mn></mrow>
<mrow><mi>N</mi></mrow></mfrac><munderover accent="false" accentunder="false"><mrow><mo mathsize="big">∑</mo></mrow><mrow><mi>i</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><mi>N</mi></mrow></munderover><msup><mrow> <mfenced close="|" open="|" separators=""><mrow><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>i</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-bin">−</mo><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi mathvariant="normal">cog</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow></mrow></mfenced></mrow><mrow><mn>2</mn></mrow></msup></mrow></msqrt>
</math></td><td class="eq-no">(7)</td></tr></table>
<!--l. 2169--><p class="indent">   This component must contain one <span class="t1-zi4r-0-x-x-109">atoms</span><span class="t1-zi4r-0-x-x-109"> {...} </span>block to define the atom group, and returns a positive number,
expressed in Å.
</p><!--l. 2173--><p class="noindent"><span class="ptmb8t-x-x-109">List of keywords </span>(see also <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-52002"></a><span class="colorbox" id="colorbox94"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">atoms:</span></span> see definition of <a href="#colvar|rmsd|atoms"><span class="t1-zi4r-0-x-x-109">atoms</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)</li></ul>
<div class="flushright">
<!--l. 2179--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2179--><p class="noindent"></p><div class="heading"><a id="sec:cvc_inertia">
<h5 class="subsubsectionHead"><span class="titlemark">4.5.5    </span> <a id="x1-530004.5.5"></a><span class="t1-zi4r-0-x-x-109">inertia</span>: total moment of inertia of a group of atoms.</h5>
<!--l. 2179--><p class="noindent"></p></a> </div>  <a id="colvar|inertia"> </a>
<!--l. 2182--><p class="indent">   The block <span class="t1-zi4r-0-x-x-109">inertia</span><span class="t1-zi4r-0-x-x-109"> {...} </span>defines the parameters for calculating the total moment of inertia of a group of atomic positions
<!--l. 2184--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">{</mo><mrow><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mn>1</mn></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-punc">,</mo><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mn>2</mn></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-punc">,</mo><mo class="MathClass-op">…</mo><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>N</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow></mrow><mo class="MathClass-close">}</mo></mrow></math> with respect to their
center of geometry, <!--l. 2186--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi mathvariant="normal">cog</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow></math>: </p><table class="equation"><tr><td>
<a id="x1-53001r8"></a>
<!--l. 2187--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>I</mi><mspace class="tmspace" width="3.04074pt"></mspace> <mo class="MathClass-rel">=</mo> <mspace class="tmspace" width="3.04074pt"></mspace><munderover accent="false" accentunder="false"><mrow><mo mathsize="big">∑</mo></mrow><mrow><mi>i</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><mi>N</mi></mrow></munderover><msup><mrow> <mfenced close="|" open="|" separators=""><mrow><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow>
<mi>i</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-bin">−</mo><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi mathvariant="normal">cog</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow></mrow></mfenced></mrow><mrow><mn>2</mn></mrow></msup>
</math></td><td class="eq-no">(8)</td></tr></table>
<!--l. 2192--><p class="indent"> <span class="ptmri8t-x-x-109">Note that all atomic masses are set to 1 for simplicity. </span>This component must contain
one <span class="t1-zi4r-0-x-x-109">atoms</span><span class="t1-zi4r-0-x-x-109"> {...} </span>block to define the atom group, and returns a positive number, expressed in
Å<!--l. 2195--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow></mrow><mrow><mn>2</mn></mrow></msup></math>.
</p><!--l. 2197--><p class="noindent"><span class="ptmb8t-x-x-109">List of keywords </span>(see also <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-53002"></a><span class="colorbox" id="colorbox95"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">atoms:</span></span> see definition of <a href="#colvar|rmsd|atoms"><span class="t1-zi4r-0-x-x-109">atoms</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)</li></ul>
<div class="flushright">
<!--l. 2203--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2203--><p class="noindent"></p><div class="heading"><a id="sec:cvc_dipoleMagnitude">
<h5 class="subsubsectionHead"><span class="titlemark">4.5.6    </span> <a id="x1-540004.5.6"></a><span class="t1-zi4r-0-x-x-109">dipoleMagnitude</span>: dipole magnitude of a group of atoms.</h5>
<!--l. 2203--><p class="noindent"></p></a> </div>  <a id="colvar|dipoleMagnitude"> </a> The <span class="t1-zi4r-0-x-x-109">dipoleMagnitude</span><span class="t1-zi4r-0-x-x-109"> {...} </span>block defines the dipole magnitude of a group of atoms (norm of the dipole
moment's vector), being <span class="t1-zi4r-0-x-x-109">atoms </span>the group where dipole magnitude is calculated. It returns the magnitude in elementary
charge <!--l. 2206--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>e</mi></math>
times Å.
<!--l. 2208--><p class="noindent"><span class="ptmb8t-x-x-109">List of keywords </span>(see also <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-54001"></a><span class="colorbox" id="colorbox96"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">atoms:</span></span> see definition of <a href="#colvar|rmsd|atoms"><span class="t1-zi4r-0-x-x-109">atoms</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)</li></ul>
<div class="flushright">
<!--l. 2214--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2214--><p class="noindent"></p><div class="heading"><a id="sec:cvc_inertiaZ">
<h5 class="subsubsectionHead"><span class="titlemark">4.5.7    </span> <a id="x1-550004.5.7"></a><span class="t1-zi4r-0-x-x-109">inertiaZ</span>: total moment of inertia of a group of atoms around a chosen axis.</h5>
<!--l. 2214--><p class="noindent"></p></a> </div>  <a id="colvar|inertiaZ"> </a>
<!--l. 2217--><p class="indent">   The block <span class="t1-zi4r-0-x-x-109">inertiaZ</span><span class="t1-zi4r-0-x-x-109"> {...} </span>defines the parameters for calculating the component along the axis
<!--l. 2218--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="bold">e</mi></math> of the moment of inertia of a group
of atomic positions <!--l. 2219--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">{</mo><mrow><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mn>1</mn></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-punc">,</mo><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mn>2</mn></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-punc">,</mo><mo class="MathClass-op">…</mo><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>N</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow></mrow><mo class="MathClass-close">}</mo></mrow></math> with respect
to their center of geometry, <!--l. 2221--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi mathvariant="normal">cog</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow></math>:
</p><table class="equation"><tr><td> <a id="x1-55001r9"></a>
<!--l. 2222--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<msub><mrow><mi>I</mi></mrow><mrow><mi mathvariant="bold">e</mi></mrow></msub><mspace class="tmspace" width="3.04074pt"></mspace> <mo class="MathClass-rel">=</mo> <mspace class="tmspace" width="3.04074pt"></mspace><munderover accent="false" accentunder="false"><mrow><mo mathsize="big">∑</mo></mrow><mrow><mi>i</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><mi>N</mi></mrow></munderover><msup><mrow> <mfenced close=")" open="(" separators=""><mrow><mfenced close=")" open="(" separators=""><mrow><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow>
<mi>i</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-bin">−</mo><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi mathvariant="normal">cog</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow></mrow></mfenced><mo class="MathClass-bin">⋅</mo><mi mathvariant="bold">e</mi></mrow></mfenced></mrow><mrow><mn>2</mn></mrow></msup>
</math></td><td class="eq-no">(9)</td></tr></table>
<!--l. 2227--><p class="indent"> <span class="ptmri8t-x-x-109">Note that all atomic masses are set to 1 for simplicity. </span>This component must contain
one <span class="t1-zi4r-0-x-x-109">atoms</span><span class="t1-zi4r-0-x-x-109"> {...} </span>block to define the atom group, and returns a positive number, expressed in
Å<!--l. 2230--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow></mrow><mrow><mn>2</mn></mrow></msup></math>.
</p><!--l. 2233--><p class="noindent"><span class="ptmb8t-x-x-109">List of keywords </span>(see also <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-55002"></a><span class="colorbox" id="colorbox97"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">atoms:</span></span> see definition of <a href="#colvar|rmsd|atoms"><span class="t1-zi4r-0-x-x-109">atoms</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
<li class="itemize"><a id="dx1-55003"></a><span class="colorbox" id="colorbox98"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">axis</span></span> <!--l. 2244--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Projection
     axis (Å)<!--l. 2244--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">inertiaZ</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span><span class="t1-zi4r-0-x-x-109">(x, y, z) </span>triplet <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">(0.0, 0.0, 1.0)</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>The  three  components  of  this  vector  define  (when  normalized)  the  projection  axis
     <!--l. 2244--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="bold">e</mi></math>.
     </li></ul>
<div class="flushright">
<!--l. 2248--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2248--><p class="noindent"></p><div class="heading"><a id="sec:cvc_rotations">
<h4 class="subsectionHead"><span class="titlemark">4.6    </span> <a id="x1-560004.6"></a>Rotations</h4>
<!--l. 2248--><p class="noindent"></p></a> </div> 
<!--l. 2250--><p class="indent">   The variables discussed in this section quantify the rotations of macromolecules (or other quasi-rigid objects)
from a given set of reference coordinates to the current coordinates. Such rotations are computed following the same
method used for best-fit RMSDs (see <a href="#colvar|rmsd"><span class="t1-zi4r-0-x-x-109">rmsd</span></a> and <a href="#atom-group|fittingGroup"><span class="t1-zi4r-0-x-x-109">fittingGroup</span></a>). The underlying mathematical formalism is
described in reference <span class="cite">[<a href="#XCoutsias2004">4</a>]</span>, and the implementation in reference <span class="cite">[<a href="#XFiorin2013">1</a>]</span>.
</p><!--l. 2254--><p class="indent">   The first of the functions described is the <a href="#colvar|orientation"><span class="t1-zi4r-0-x-x-109">orientation</span></a>, which describes the full rotation as a unit quaternion
<!--l. 2254--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="sans-serif">q</mi> <mo class="MathClass-rel">=</mo> <mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>q</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>q</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>q</mi></mrow><mrow><mn>2</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>q</mi></mrow><mrow><mn>3</mn></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math>,
i.e. 4 numbers with one constraint (3 degrees of freedom). The quaternion
<!--l. 2255--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="sans-serif">q</mi></math>
is <span class="ptmri8t-x-x-109">one of only two representations that are both complete and accurate</span>, the other being a
<!--l. 2255--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>3</mn><mo class="MathClass-bin">×</mo><mn>3</mn></math> unit matrix with 3 independent
parameters. Although <!--l. 2256--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="sans-serif">q</mi></math>
is used internally in the Colvars module for features such as the <a href="#colvar|rmsd"><span class="t1-zi4r-0-x-x-109">rmsd</span></a> function and the <a href="#atom-group|fittingGroup"><span class="t1-zi4r-0-x-x-109">fittingGroup</span></a>
option, its direct use as a collective variable is more difficult, and mostly limited to fixed or moving
restraints.
</p><!--l. 2258--><p class="indent">   The two functions <a href="#colvar|orientationAngle"><span class="t1-zi4r-0-x-x-109">orientationAngle</span></a> and <a href="#colvar|orientationProj"><span class="t1-zi4r-0-x-x-109">orientationProj</span></a>, with the latter being
the cosine of the former, represent the amplitude of the full rotation described by
<!--l. 2258--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="sans-serif">q</mi></math>,
regardless of the direction of its axis. As one-dimensional scalar variables, both <a href="#colvar|orientationAngle"><span class="t1-zi4r-0-x-x-109">orientationAngle</span></a> and
<a href="#colvar|orientationProj"><span class="t1-zi4r-0-x-x-109">orientationProj</span></a> are a much reduced simplification of the full rotation. However, they can be used in a variety of
methods including both restraints and PMF computations.
</p><!--l. 2262--><p class="indent">   A slightly more complete parametrization is achieved by decomposing the full rotation into the two parameters,
<a href="#colvar|tilt"><span class="t1-zi4r-0-x-x-109">tilt</span></a> and <a href="#colvar|spinAngle"><span class="t1-zi4r-0-x-x-109">spinAngle</span></a>. These quantify the amplitudes of two independent sub-rotations away from a certain axis
<!--l. 2263--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="bold">e</mi></math>, and around the same
axis <!--l. 2263--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="bold">e</mi></math>, respectively.
The axis <!--l. 2264--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="bold">e</mi></math> is
chosen by the user, and is by default the Z axis: under that choice, <a href="#colvar|tilt"><span class="t1-zi4r-0-x-x-109">tilt</span></a> is equivalent to the sine of the Euler “pitch" angle
<!--l. 2265--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>𝜃</mi></math>, and <a href="#colvar|spinAngle"><span class="t1-zi4r-0-x-x-109">spinAngle</span></a> to the sum
of the other two angles, <!--l. 2265--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">(</mo><mrow><mi>ϕ</mi> <mo class="MathClass-bin">+</mo><mi>ψ</mi></mrow><mo class="MathClass-close">)</mo></mrow></math>.
This parameterization is mathematically well defined for almost all full rotations, including small ones when the
current coordinates are almost completely aligned with the reference ones. However, a mathematical singularity
prevents using the <a href="#colvar|spinAngle"><span class="t1-zi4r-0-x-x-109">spinAngle</span></a> function near configurations where the value of tilt <a href="#colvar|tilt"><span class="t1-zi4r-0-x-x-109">tilt</span></a> is -1 (i.e. a
-180<!--l. 2267--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></math> rotation around an
axis orthogonal to <!--l. 2267--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="bold">e</mi></math>).
For these reasons, <a href="#colvar|tilt"><span class="t1-zi4r-0-x-x-109">tilt</span></a> and <a href="#colvar|spinAngle"><span class="t1-zi4r-0-x-x-109">spinAngle</span></a> are useful when the allowed rotations are known to
have approximately the same axis, and differ only in the magnitude of the corresponding angle.
In this use case, <a href="#colvar|spinAngle"><span class="t1-zi4r-0-x-x-109">spinAngle</span></a> measures the angle of the sub-rotation around the chosen axis
<!--l. 2269--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="bold">e</mi></math>, whereas <a href="#colvar|tilt"><span class="t1-zi4r-0-x-x-109">tilt</span></a> measures
the dot product between <!--l. 2269--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="bold">e</mi></math>
and the actual axis of the full rotation.
</p><!--l. 2271--><p class="indent">   Lastly, the traditional Euler angles are also available as the functions <a href="#colvar|eulerPhi"><span class="t1-zi4r-0-x-x-109">eulerPhi</span></a>, <a href="#colvar|eulerTheta"><span class="t1-zi4r-0-x-x-109">eulerTheta</span></a> and
<a href="#colvar|eulerPsi"><span class="t1-zi4r-0-x-x-109">eulerPsi</span></a>. Altogether, these are sufficient to represent all three degrees of freedom of a full rotation.
However, they also suffer from the potential “gimbal lock" problem, which emerges whenever
<!--l. 2273--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>𝜃</mi> <mo class="MathClass-rel">≃</mo><mo class="MathClass-bin">±</mo><mn>9</mn><msup><mrow><mn>0</mn></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></math>,
which includes also the case where the full rotation is small. Under such conditions, the angles
                                                                                         
                                                                                         
<!--l. 2274--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ϕ</mi></math> and
<!--l. 2274--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ψ</mi></math> are both
ill-defined and cannot be used as collective variables. For these reasons, it is highly recommmended that Euler angles are
used only in simulations where their range of applicability is <span class="ptmri8t-x-x-109">known ahead of time</span>, and excludes configurations
where <!--l. 2275--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>𝜃</mi> <mo class="MathClass-rel">≃</mo><mo class="MathClass-bin">±</mo><mn>9</mn><msup><mrow><mn>0</mn></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></math>
altogether.
</p>
<div class="flushright">
<!--l. 2277--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2277--><p class="noindent"></p><div class="heading"><a id="sec:cvc_orientation">
<h5 class="subsubsectionHead"><span class="titlemark">4.6.1    </span> <a id="x1-570004.6.1"></a><span class="t1-zi4r-0-x-x-109">orientation</span>: orientation from reference coordinates.</h5>
<!--l. 2277--><p class="noindent"></p></a> </div>  <a id="colvar|orientation"> </a>
<!--l. 2280--><p class="indent">   The block <span class="t1-zi4r-0-x-x-109">orientation</span><span class="t1-zi4r-0-x-x-109"> {...} </span>returns the same optimal rotation used in the <span class="t1-zi4r-0-x-x-109">rmsd </span>component to superimpose the coordinates
<!--l. 2282--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">{</mo><mrow><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>i</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow></mrow><mo class="MathClass-close">}</mo></mrow></math> onto a set of reference coordinates
<!--l. 2283--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">{</mo><mrow><msubsup><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>i</mi></mrow><mrow><mi mathvariant="normal">(ref)</mi></mrow></msubsup></mrow><mo class="MathClass-close">}</mo></mrow></math>. Such component returns a
four dimensional vector <!--l. 2284--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="sans-serif">q</mi> <mo class="MathClass-rel">=</mo> <mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>q</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>q</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>q</mi></mrow><mrow><mn>2</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>q</mi></mrow><mrow><mn>3</mn></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math>, with
<!--l. 2285--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mo class="MathClass-op">∑</mo><!--nolimits--></mrow><mrow><mi>i</mi></mrow></msub><msubsup><mrow><mi>q</mi></mrow><mrow><mi>i</mi></mrow><mrow><mn>2</mn></mrow></msubsup> <mo class="MathClass-rel">=</mo> <mn>1</mn></math>; this <span class="ptmri8t-x-x-109">quaternion </span>expresses the optimal
rotation <!--l. 2286--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">{</mo><mrow><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>i</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow></mrow><mo class="MathClass-close">}</mo></mrow><mo class="MathClass-rel">→</mo><mrow><mo class="MathClass-open">{</mo><mrow><msubsup><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>i</mi></mrow><mrow><mi mathvariant="normal">(ref)</mi></mrow></msubsup></mrow><mo class="MathClass-close">}</mo></mrow></math> according to the formalism
in reference <span class="cite">[<a href="#XCoutsias2004">4</a>]</span>. The quaternion <!--l. 2288--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>q</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>q</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>q</mi></mrow><mrow><mn>2</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>q</mi></mrow><mrow><mn>3</mn></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math>
can also be written as <!--l. 2289--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mfenced close=")" open="(" separators=""><mrow><mo class="qopname">cos</mo><!--nolimits--><mrow><mo class="MathClass-open">(</mo><mrow><mi>𝜃</mi><mo class="MathClass-bin">∕</mo><mn>2</mn></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-punc">,</mo><mspace class="thinspace" width="0.3em"></mspace><mo class="qopname">sin</mo><!--nolimits--><mrow><mo class="MathClass-open">(</mo><mrow><mi>𝜃</mi><mo class="MathClass-bin">∕</mo><mn>2</mn></mrow><mo class="MathClass-close">)</mo></mrow><mi mathvariant="bold">u</mi></mrow></mfenced></math>,
where <!--l. 2290--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>𝜃</mi></math> is the angle and
<!--l. 2291--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="bold">u</mi></math> the normalized axis of rotation;
for example, a rotation of 90<!--l. 2292--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></math>
around the <!--l. 2292--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math> axis
is expressed as “<span class="t1-zi4r-0-x-x-109">(0.707, 0.0, 0.0, 0.707)</span>". The script <a href="https://raw.githubusercontent.com/Colvars/colvars/master/colvartools/quaternion2rmatrix.tcl"><span class="t1-zi4r-0-x-x-109">quaternion2rmatrix.tcl</span></a> provides Tcl functions for converting
to and from a <!--l. 2295--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>4</mn><mo class="MathClass-bin">×</mo><mn>4</mn></math>
rotation matrix in a format suitable for usage in VMD.
</p><!--l. 2298--><p class="indent">   As for the component <span class="t1-zi4r-0-x-x-109">rmsd</span>, the available options are <span class="t1-zi4r-0-x-x-109">atoms</span>, <span class="t1-zi4r-0-x-x-109">refPositionsFile</span>, <span class="t1-zi4r-0-x-x-109">refPositionsCol </span>and
<span class="t1-zi4r-0-x-x-109">refPositionsColValue</span>,  and <span class="t1-zi4r-0-x-x-109">refPositions</span>.
</p><!--l. 2300--><p class="indent"> <span class="ptmb8t-x-x-109">Note: </span><span class="t1-zi4r-0-x-x-109">refPositions</span>and <span class="t1-zi4r-0-x-x-109">refPositionsFile </span>define the set of positions <span class="ptmri8t-x-x-109">from which </span>the optimal rotation is
calculated, but this rotation is not applied to the coordinates of the atoms involved: it is used instead to define the
variable itself.
</p><!--l. 2302--><p class="noindent"><span class="ptmb8t-x-x-109">List of keywords </span>(see also <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-57001"></a><span class="colorbox" id="colorbox99"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">atoms:</span></span> see definition of <a href="#colvar|rmsd|atoms"><span class="t1-zi4r-0-x-x-109">atoms</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
<li class="itemize"><a id="dx1-57002"></a><span class="colorbox" id="colorbox100"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositions:</span></span> see definition of <a href="#colvar|rmsd|refPositions"><span class="t1-zi4r-0-x-x-109">refPositions</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
<li class="itemize"><a id="dx1-57003"></a><span class="colorbox" id="colorbox101"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositionsFile:</span></span> see definition of <a href="#colvar|rmsd|refPositionsFile"><span class="t1-zi4r-0-x-x-109">refPositionsFile</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
                                                                                         
                                                                                         
     </li>
<li class="itemize"><a id="dx1-57004"></a><span class="colorbox" id="colorbox102"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositionsCol:</span></span> see definition of <a href="#colvar|rmsd|refPositionsCol"><span class="t1-zi4r-0-x-x-109">refPositionsCol</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
<li class="itemize"><a id="dx1-57005"></a><span class="colorbox" id="colorbox103"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositionsColValue:</span></span> see definition of <a href="#colvar|rmsd|refPositionsColValue"><span class="t1-zi4r-0-x-x-109">refPositionsColValue</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
<li class="itemize"><a id="dx1-57006"></a><span class="colorbox" id="colorbox104"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">closestToQuaternion</span></span> <!--l. 2329--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Reference
     rotation<!--l. 2329--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">orientation</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>“<span class="t1-zi4r-0-x-x-109">(q0, q1, q2, q3)</span>" quadruplet <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">(1.0, 0.0, 0.0, 0.0) </span>(“null" rotation) <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Between the two equivalent quaternions <!--l. 2329--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>q</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>q</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>q</mi></mrow><mrow><mn>2</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>q</mi></mrow><mrow><mn>3</mn></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math>
     and <!--l. 2329--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">(</mo><mrow><mo class="MathClass-bin">−</mo><msub><mrow><mi>q</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-punc">,</mo><mo class="MathClass-bin">−</mo><msub><mrow><mi>q</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-punc">,</mo><mo class="MathClass-bin">−</mo><msub><mrow><mi>q</mi></mrow><mrow><mn>2</mn></mrow></msub><mo class="MathClass-punc">,</mo><mo class="MathClass-bin">−</mo><msub><mrow><mi>q</mi></mrow><mrow><mn>3</mn></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math>,
     the  closer  to  <span class="t1-zi4r-0-x-x-109">(1.0,  0.0,  0.0,  0.0) </span>is  chosen.  This  simplifies  the  visualization  of  the  colvar
     trajectory when sampled values are a smaller subset of all possible rotations. <span class="ptmb8t-x-x-109">Note: </span><span class="ptmri8t-x-x-109">this only affects</span>
<span class="ptmri8t-x-x-109">the output, never the dynamics</span>.
     </li></ul>
<!--l. 2333--><p class="indent"> <span class="ptmb8t-x-x-109">Tip: stopping the rotation of a protein. </span>To stop the rotation of an elongated macromolecule in solution (and
use an anisotropic box to save water molecules), it is possible to define a colvar with an <span class="t1-zi4r-0-x-x-109">orientation </span>component,
and restrain it through the <span class="t1-zi4r-0-x-x-109">harmonic </span>bias around the identity rotation, <span class="t1-zi4r-0-x-x-109">(1.0, 0.0, 0.0, 0.0)</span>. Only
the overall orientation of the macromolecule is affected, and <span class="ptmri8t-x-x-109">not </span>its internal degrees of freedom. The
user should also take care that the macromolecule is composed by a single chain, or disable <span class="t1-zi4r-0-x-x-109">wrapAll</span>
otherwise.
</p>
<div class="flushright">
<!--l. 2345--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2345--><p class="noindent"></p><div class="heading"><a id="sec:cvc_orientationAngle">
<h5 class="subsubsectionHead"><span class="titlemark">4.6.2    </span> <a id="x1-580004.6.2"></a><span class="t1-zi4r-0-x-x-109">orientationAngle</span>: angle of rotation from reference coordinates.</h5>
<!--l. 2345--><p class="noindent"></p></a> </div>  <a id="colvar|orientationAngle"> </a>
<!--l. 2348--><p class="indent">   The block <span class="t1-zi4r-0-x-x-109">orientationAngle</span><span class="t1-zi4r-0-x-x-109"> {...} </span>accepts the same base options as the component <span class="t1-zi4r-0-x-x-109">orientation</span>: <span class="t1-zi4r-0-x-x-109">atoms</span>,
<span class="t1-zi4r-0-x-x-109">refPositions</span>, <span class="t1-zi4r-0-x-x-109">refPositionsFile</span>, <span class="t1-zi4r-0-x-x-109">refPositionsCol </span>and <span class="t1-zi4r-0-x-x-109">refPositionsColValue</span>. The returned value is the angle of
rotation <!--l. 2350--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>𝜃</mi></math>
between the current and the reference positions. This angle is expressed in degrees within the range
[0<!--l. 2351--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></math>:180<!--l. 2351--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></math>].
</p><!--l. 2353--><p class="noindent"><span class="ptmb8t-x-x-109">List of keywords </span>(see also <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-58001"></a><span class="colorbox" id="colorbox105"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">atoms:</span></span> see definition of <a href="#colvar|rmsd|atoms"><span class="t1-zi4r-0-x-x-109">atoms</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
<li class="itemize"><a id="dx1-58002"></a><span class="colorbox" id="colorbox106"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositions:</span></span> see definition of <a href="#colvar|rmsd|refPositions"><span class="t1-zi4r-0-x-x-109">refPositions</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
                                                                                         
                                                                                         
     </li>
<li class="itemize"><a id="dx1-58003"></a><span class="colorbox" id="colorbox107"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositionsFile:</span></span> see definition of <a href="#colvar|rmsd|refPositionsFile"><span class="t1-zi4r-0-x-x-109">refPositionsFile</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
<li class="itemize"><a id="dx1-58004"></a><span class="colorbox" id="colorbox108"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositionsCol:</span></span> see definition of <a href="#colvar|rmsd|refPositionsCol"><span class="t1-zi4r-0-x-x-109">refPositionsCol</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
<li class="itemize"><a id="dx1-58005"></a><span class="colorbox" id="colorbox109"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositionsColValue:</span></span> see definition of <a href="#colvar|rmsd|refPositionsColValue"><span class="t1-zi4r-0-x-x-109">refPositionsColValue</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
     </li></ul>
<div class="flushright">
<!--l. 2370--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2370--><p class="noindent"></p><div class="heading"><a id="sec:cvc_orientationProj">
<h5 class="subsubsectionHead"><span class="titlemark">4.6.3    </span> <a id="x1-590004.6.3"></a><span class="t1-zi4r-0-x-x-109">orientationProj</span>: cosine of the angle of rotation from reference coordinates.</h5>
<!--l. 2370--><p class="noindent"></p></a> </div>  <a id="colvar|orientationProj"> </a>
<!--l. 2373--><p class="indent">   The block <span class="t1-zi4r-0-x-x-109">orientationProj</span><span class="t1-zi4r-0-x-x-109"> {...} </span>accepts the same base options as the component <span class="t1-zi4r-0-x-x-109">orientation</span>: <span class="t1-zi4r-0-x-x-109">atoms</span>,
<span class="t1-zi4r-0-x-x-109">refPositions</span>, <span class="t1-zi4r-0-x-x-109">refPositionsFile</span>, <span class="t1-zi4r-0-x-x-109">refPositionsCol </span>and <span class="t1-zi4r-0-x-x-109">refPositionsColValue</span>. The returned value is the cosine of the
angle of rotation <!--l. 2375--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>𝜃</mi></math>
between the current and the reference positions. The range of values is [-1:1].
</p><!--l. 2378--><p class="noindent"><span class="ptmb8t-x-x-109">List of keywords </span>(see also <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-59001"></a><span class="colorbox" id="colorbox110"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">atoms:</span></span> see definition of <a href="#colvar|rmsd|atoms"><span class="t1-zi4r-0-x-x-109">atoms</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
<li class="itemize"><a id="dx1-59002"></a><span class="colorbox" id="colorbox111"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositions:</span></span> see definition of <a href="#colvar|rmsd|refPositions"><span class="t1-zi4r-0-x-x-109">refPositions</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
<li class="itemize"><a id="dx1-59003"></a><span class="colorbox" id="colorbox112"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositionsFile:</span></span> see definition of <a href="#colvar|rmsd|refPositionsFile"><span class="t1-zi4r-0-x-x-109">refPositionsFile</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
<li class="itemize"><a id="dx1-59004"></a><span class="colorbox" id="colorbox113"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositionsCol:</span></span> see definition of <a href="#colvar|rmsd|refPositionsCol"><span class="t1-zi4r-0-x-x-109">refPositionsCol</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
<li class="itemize"><a id="dx1-59005"></a><span class="colorbox" id="colorbox114"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositionsColValue:</span></span> see definition of <a href="#colvar|rmsd|refPositionsColValue"><span class="t1-zi4r-0-x-x-109">refPositionsColValue</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
     </li></ul>
<div class="flushright">
<!--l. 2394--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2394--><p class="noindent"></p><div class="heading"><a id="sec:cvc_spinAngle">
<h5 class="subsubsectionHead"><span class="titlemark">4.6.4    </span> <a id="x1-600004.6.4"></a><span class="t1-zi4r-0-x-x-109">spinAngle</span>: angle of rotation around a given axis.</h5>
<!--l. 2394--><p class="noindent"></p></a> </div>  <a id="colvar|spinAngle"> </a>
<!--l. 2397--><p class="indent">   The complete rotation described by <span class="t1-zi4r-0-x-x-109">orientation </span>can optionally be decomposed into two sub-rotations: one is a
“<span class="ptmri8t-x-x-109">spin</span>" rotation around <span class="ptmb8t-x-x-109">e</span>, and the other a “<span class="ptmri8t-x-x-109">tilt</span>" rotation around an axis orthogonal to <span class="ptmb8t-x-x-109">e</span>. The component <span class="t1-zi4r-0-x-x-109">spinAngle</span>
measures the angle of the “spin" sub-rotation around <span class="ptmb8t-x-x-109">e</span>.
</p><!--l. 2400--><p class="noindent"><span class="ptmb8t-x-x-109">List of keywords </span>(see also <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-60001"></a><span class="colorbox" id="colorbox115"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">atoms:</span></span> see definition of <a href="#colvar|rmsd|atoms"><span class="t1-zi4r-0-x-x-109">atoms</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
<li class="itemize"><a id="dx1-60002"></a><span class="colorbox" id="colorbox116"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositions:</span></span> see definition of <a href="#colvar|rmsd|refPositions"><span class="t1-zi4r-0-x-x-109">refPositions</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
<li class="itemize"><a id="dx1-60003"></a><span class="colorbox" id="colorbox117"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositionsFile:</span></span> see definition of <a href="#colvar|rmsd|refPositionsFile"><span class="t1-zi4r-0-x-x-109">refPositionsFile</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
<li class="itemize"><a id="dx1-60004"></a><span class="colorbox" id="colorbox118"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositionsCol:</span></span> see definition of <a href="#colvar|rmsd|refPositionsCol"><span class="t1-zi4r-0-x-x-109">refPositionsCol</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
<li class="itemize"><a id="dx1-60005"></a><span class="colorbox" id="colorbox119"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositionsColValue:</span></span> see definition of <a href="#colvar|rmsd|refPositionsColValue"><span class="t1-zi4r-0-x-x-109">refPositionsColValue</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
<li class="itemize"><a id="colvar|spinAngle|axis"> </a> <a id="dx1-60006"></a><span class="colorbox" id="colorbox120"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">axis</span></span> <!--l. 2421--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Special
     rotation axis (Å)<!--l. 2421--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">tilt</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span><span class="t1-zi4r-0-x-x-109">(x, y, z) </span>triplet <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">(0.0, 0.0, 1.0)</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>The three components of this vector define (when normalized) the special rotation axis
     used to calculate the <span class="t1-zi4r-0-x-x-109">tilt </span>and <span class="t1-zi4r-0-x-x-109">spinAngle </span>components. </li></ul>
<!--l. 2423--><p class="noindent">The component <span class="t1-zi4r-0-x-x-109">spinAngle </span>returns an angle (in degrees) within the periodic interval
<!--l. 2423--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">[</mo><mrow><mo class="MathClass-bin">−</mo><mn>1</mn><mn>8</mn><mn>0</mn> <mo class="MathClass-punc">:</mo> <mn>1</mn><mn>8</mn><mn>0</mn></mrow><mo class="MathClass-close">]</mo></mrow></math>.
</p><!--l. 2425--><p class="indent"> <span class="ptmb8t-x-x-109">Note: </span>the value of <span class="t1-zi4r-0-x-x-109">spinAngle </span>is a continuous function almost everywhere,
with the exception of configurations with the corresponding “tilt" angle equal to
180<!--l. 2425--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></math> (i.e. the <span class="t1-zi4r-0-x-x-109">tilt</span>
component is equal to <!--l. 2425--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-bin">−</mo><mn>1</mn></math>):
in those cases, <span class="t1-zi4r-0-x-x-109">spinAngle </span>is undefined. If such configurations are expected, consider
defining a <span class="t1-zi4r-0-x-x-109">tilt </span>colvar using the same axis <span class="ptmb8t-x-x-109">e</span>, and restraining it with a lower wall away from
<!--l. 2425--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-bin">−</mo><mn>1</mn></math>.
</p>
<div class="flushright">
<!--l. 2428--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2428--><p class="noindent"></p><div class="heading"><a id="sec:cvc_tilt">
<h5 class="subsubsectionHead"><span class="titlemark">4.6.5    </span> <a id="x1-610004.6.5"></a><span class="t1-zi4r-0-x-x-109">tilt</span>: cosine of the rotation orthogonal to a given axis.</h5>
<!--l. 2428--><p class="noindent"></p></a> </div>  <a id="colvar|tilt"> </a>
<!--l. 2431--><p class="indent">   The component <span class="t1-zi4r-0-x-x-109">tilt </span>measures the cosine of the angle of the “tilt" sub-rotation, which combined with the “spin"
sub-rotation provides the complete rotation of a group of atoms. The cosine of the tilt angle rather than the tilt angle
itself is implemented, because the latter is unevenly distributed even for an isotropic system: consider as an analogy the
angle <!--l. 2432--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>𝜃</mi></math>
in the spherical coordinate system. The component <span class="t1-zi4r-0-x-x-109">tilt </span>relies on the same options as <span class="t1-zi4r-0-x-x-109">spinAngle</span>,
including the definition of the axis <span class="ptmb8t-x-x-109">e</span>. The values of <span class="t1-zi4r-0-x-x-109">tilt </span>are real numbers in the interval
<!--l. 2434--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">[</mo><mrow><mo class="MathClass-bin">−</mo><mn>1</mn> <mo class="MathClass-punc">:</mo> <mn>1</mn></mrow><mo class="MathClass-close">]</mo></mrow></math>: the value
<!--l. 2434--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>1</mn></math> represents an orientation fully
parallel to <span class="ptmb8t-x-x-109">e </span>(tilt angle = 0<!--l. 2434--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></math>),
and the value <!--l. 2434--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-bin">−</mo><mn>1</mn></math>
represents an anti-parallel orientation.
</p><!--l. 2436--><p class="noindent"><span class="ptmb8t-x-x-109">List of keywords </span>(see also <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-61001"></a><span class="colorbox" id="colorbox121"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">atoms:</span></span> see definition of <a href="#colvar|rmsd|atoms"><span class="t1-zi4r-0-x-x-109">atoms</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
<li class="itemize"><a id="dx1-61002"></a><span class="colorbox" id="colorbox122"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositions:</span></span> see definition of <a href="#colvar|rmsd|refPositions"><span class="t1-zi4r-0-x-x-109">refPositions</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
<li class="itemize"><a id="dx1-61003"></a><span class="colorbox" id="colorbox123"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositionsFile:</span></span> see definition of <a href="#colvar|rmsd|refPositionsFile"><span class="t1-zi4r-0-x-x-109">refPositionsFile</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
<li class="itemize"><a id="dx1-61004"></a><span class="colorbox" id="colorbox124"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositionsCol:</span></span> see definition of <a href="#colvar|rmsd|refPositionsCol"><span class="t1-zi4r-0-x-x-109">refPositionsCol</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
<li class="itemize"><a id="dx1-61005"></a><span class="colorbox" id="colorbox125"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositionsColValue:</span></span> see definition of <a href="#colvar|rmsd|refPositionsColValue"><span class="t1-zi4r-0-x-x-109">refPositionsColValue</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
<li class="itemize"><a id="dx1-61006"></a><span class="colorbox" id="colorbox126"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">axis:</span></span> see definition of <a href="#colvar|spinAngle|axis"><span class="t1-zi4r-0-x-x-109">axis</span></a> (<span class="t1-zi4r-0-x-x-109">spinAngle </span>component)</li></ul>
<div class="flushright">
<!--l. 2455--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2455--><p class="noindent"></p><div class="heading"><a id="sec:cvc_eulerPhi">
<h5 class="subsubsectionHead"><span class="titlemark">4.6.6    </span> <a id="x1-620004.6.6"></a><span class="t1-zi4r-0-x-x-109">eulerPhi</span>: Roll angle from references coordinates.</h5>
<!--l. 2455--><p class="noindent"></p></a> </div>  <a id="colvar|eulerPhi"> </a>
<!--l. 2458--><p class="indent">   Assuming the axes of the original frame are denoted as <span class="ptmri8t-x-x-109">x, y, z </span>and the axes of the rotated frame as <span class="ptmri8t-x-x-109">X, Y, Z</span>, the
line of nodes, <span class="ptmri8t-x-x-109">N</span>, can be defined as the intersection of the plane <span class="ptmri8t-x-x-109">xy </span>and <span class="ptmri8t-x-x-109">XY</span>. The axis perpendicular to <span class="ptmri8t-x-x-109">N </span>and <span class="ptmri8t-x-x-109">z </span>is
defined as <span class="ptmri8t-x-x-109">P</span>. In this case, as illustrated in the figure below, the complete rotation described by <span class="t1-zi4r-0-x-x-109">orientation </span>can
optionally be decomposed into three Euler angles: </p>
<ul class="itemize1">
<li class="itemize">the “roll" angle <!--l. 2466--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ϕ</mi></math>,
     i.e. the rotation angle from the <span class="ptmri8t-x-x-109">x </span>axis to the <span class="ptmri8t-x-x-109">N </span>axis;
     </li>
<li class="itemize">the “pitch" angle <!--l. 2467--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>𝜃</mi></math>,
     i.e. the rotation angle from the <span class="ptmri8t-x-x-109">P </span>axis to the <span class="ptmri8t-x-x-109">Z </span>axis, and
     </li>
<li class="itemize">the “yaw" angle <!--l. 2469--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ψ</mi></math>,
     i.e. the rotation angle from the <span class="ptmri8t-x-x-109">N </span>axis to the <span class="ptmri8t-x-x-109">X </span>axis.</li></ul>
<div class="center">
<!--l. 2472--><p class="noindent">
</p><!--l. 2474--><p class="noindent"><img alt="PIC" src="eulerangles-512px.png"/></p></div>
<!--l. 2480--><p class="indent">   Although Euler angles are more straightforward to use than quaternions, they are also potentially subject to the
“gimbal lock" problem:<br class="newline"/><a class="url" href="https://en.wikipedia.org/wiki/Gimbal_lock"><span class="t1-zi4r-0-x-x-109">https://en.wikipedia.org/wiki/Gimbal_lock</span></a><br class="newline"/>which arises whenever <!--l. 2482--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>𝜃</mi> <mo class="MathClass-rel">≃</mo><mo class="MathClass-bin">±</mo><mn>9</mn><msup><mrow><mn>0</mn></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></math>,
including the common case when <span class="ptmri8t-x-x-109">the simulated coordinates are near the reference coordinates</span>. Therefore, a safe
use of Euler angles as collective variables requires the use of restraints to avoid such singularities, such as done in
reference <span class="cite">[<a href="#XFu2017">5</a>]</span> and in the <a href="http://www.ks.uiuc.edu/Training/Tutorials/namd/PLB/tutorial-protein-ligand.pdf">protein-ligand binding NAMD tutorial</a>.
</p><!--l. 2485--><p class="indent">   The <span class="t1-zi4r-0-x-x-109">eulerPhi </span>component accepts exactly the same options as <span class="t1-zi4r-0-x-x-109">orientation</span>, and measures the
rotation angle from the <span class="ptmri8t-x-x-109">x </span>axis to the <span class="ptmri8t-x-x-109">N </span>axis. This angle is expressed in degrees within the periodic range
<!--l. 2488--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">[</mo><mrow><mo class="MathClass-bin">−</mo><mn>1</mn><mn>8</mn><msup><mrow><mn>0</mn></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup> <mo class="MathClass-punc">:</mo> <mn>1</mn><mn>8</mn><msup><mrow><mn>0</mn></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></mrow><mo class="MathClass-close">]</mo></mrow></math>.
</p><!--l. 2490--><p class="noindent"><span class="ptmb8t-x-x-109">List of keywords </span>(see also <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-62001"></a><span class="colorbox" id="colorbox127"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">atoms:</span></span> see definition of <a href="#colvar|rmsd|atoms"><span class="t1-zi4r-0-x-x-109">atoms</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
<li class="itemize"><a id="dx1-62002"></a><span class="colorbox" id="colorbox128"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositions:</span></span> see definition of <a href="#colvar|rmsd|refPositions"><span class="t1-zi4r-0-x-x-109">refPositions</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
<li class="itemize"><a id="dx1-62003"></a><span class="colorbox" id="colorbox129"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositionsFile:</span></span> see definition of <a href="#colvar|rmsd|refPositionsFile"><span class="t1-zi4r-0-x-x-109">refPositionsFile</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
<li class="itemize">
<a id="dx1-62004"></a>
<div class="colorbox" id="colorbox130"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositionsCol:</span></div>
<!--l. 2509--><p class="noindent"> see definition of <a href="#colvar|rmsd|refPositionsCol"><span class="t1-zi4r-0-x-x-109">refPositionsCol</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
     </p></li>
<li class="itemize">
<a id="dx1-62005"></a>
<div class="colorbox" id="colorbox131"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositionsColValue:</span></div>
<!--l. 2509--><p class="noindent"> see definition of <a href="#colvar|rmsd|refPositionsColValue"><span class="t1-zi4r-0-x-x-109">refPositionsColValue</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component) </p></li></ul>
<div class="flushright">
<!--l. 2514--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2514--><p class="noindent"></p><div class="heading"><a id="sec:cvc_eulerTheta">
<h5 class="subsubsectionHead"><span class="titlemark">4.6.7    </span> <a id="x1-630004.6.7"></a><span class="t1-zi4r-0-x-x-109">eulerTheta</span>: Pitch angle from references coordinates.</h5>
<!--l. 2514--><p class="noindent"></p></a> </div>  <a id="colvar|eulerTheta"> </a>
<!--l. 2517--><p class="indent">   This component accepts exactly the same options as <span class="t1-zi4r-0-x-x-109">orientation</span>, and measures the
rotation angle from the <span class="ptmri8t-x-x-109">P </span>axis to the <span class="ptmri8t-x-x-109">Z </span>axis. This angle is expressed in degrees within the range
<!--l. 2519--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">[</mo><mrow><mo class="MathClass-bin">−</mo><mn>9</mn><msup><mrow><mn>0</mn></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup> <mo class="MathClass-punc">:</mo> <mn>9</mn><msup><mrow><mn>0</mn></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></mrow><mo class="MathClass-close">]</mo></mrow></math>.
</p><!--l. 2521--><p class="indent"> <span class="ptmb8t-x-x-109">Warning: </span>When this angle reaches <!--l. 2521--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-bin">−</mo><mn>9</mn><msup><mrow><mn>0</mn></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></math>
or <!--l. 2521--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>9</mn><msup><mrow><mn>0</mn></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></math>, the
definition of orientation by euler angles suffers from the gimbal lock issue. You may need to apply a restraint to keep
<span class="t1-zi4r-0-x-x-109">eulerTheta </span>away from the singularities.
</p><!--l. 2526--><p class="noindent"><span class="ptmb8t-x-x-109">List of keywords </span>(see also <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-63001"></a><span class="colorbox" id="colorbox132"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">atoms:</span></span> see definition of <a href="#colvar|rmsd|atoms"><span class="t1-zi4r-0-x-x-109">atoms</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
<li class="itemize"><a id="dx1-63002"></a><span class="colorbox" id="colorbox133"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositions:</span></span> see definition of <a href="#colvar|rmsd|refPositions"><span class="t1-zi4r-0-x-x-109">refPositions</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
<li class="itemize"><a id="dx1-63003"></a><span class="colorbox" id="colorbox134"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositionsFile:</span></span> see definition of <a href="#colvar|rmsd|refPositionsFile"><span class="t1-zi4r-0-x-x-109">refPositionsFile</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
<li class="itemize">
<a id="dx1-63004"></a>
<div class="colorbox" id="colorbox135"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositionsCol:</span></div>
<!--l. 2545--><p class="noindent"> see definition of <a href="#colvar|rmsd|refPositionsCol"><span class="t1-zi4r-0-x-x-109">refPositionsCol</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
     </p></li>
<li class="itemize">
<a id="dx1-63005"></a>
<div class="colorbox" id="colorbox136"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositionsColValue:</span></div>
<!--l. 2545--><p class="noindent"> see definition of <a href="#colvar|rmsd|refPositionsColValue"><span class="t1-zi4r-0-x-x-109">refPositionsColValue</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component) </p></li></ul>
<div class="flushright">
<!--l. 2550--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2550--><p class="noindent"></p><div class="heading"><a id="sec:cvc_eulerPsi">
<h5 class="subsubsectionHead"><span class="titlemark">4.6.8    </span> <a id="x1-640004.6.8"></a><span class="t1-zi4r-0-x-x-109">eulerPsi</span>: Yaw angle from references coordinates.</h5>
<!--l. 2550--><p class="noindent"></p></a> </div>  <a id="colvar|eulerPsi"> </a>
<!--l. 2553--><p class="indent">   This component accepts exactly the same options as <span class="t1-zi4r-0-x-x-109">orientation</span>, and measures the rotation
angle from the <span class="ptmri8t-x-x-109">N </span>axis to the <span class="ptmri8t-x-x-109">X </span>axis. This angle is expressed in degrees within the periodic range
<!--l. 2556--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">[</mo><mrow><mo class="MathClass-bin">−</mo><mn>1</mn><mn>8</mn><msup><mrow><mn>0</mn></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup> <mo class="MathClass-punc">:</mo> <mn>1</mn><mn>8</mn><msup><mrow><mn>0</mn></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></mrow><mo class="MathClass-close">]</mo></mrow></math>.
</p><!--l. 2558--><p class="noindent"><span class="ptmb8t-x-x-109">List of keywords </span>(see also <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-64001"></a><span class="colorbox" id="colorbox137"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">atoms:</span></span> see definition of <a href="#colvar|rmsd|atoms"><span class="t1-zi4r-0-x-x-109">atoms</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
<li class="itemize"><a id="dx1-64002"></a><span class="colorbox" id="colorbox138"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositions:</span></span> see definition of <a href="#colvar|rmsd|refPositions"><span class="t1-zi4r-0-x-x-109">refPositions</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
<li class="itemize"><a id="dx1-64003"></a><span class="colorbox" id="colorbox139"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositionsFile:</span></span> see definition of <a href="#colvar|rmsd|refPositionsFile"><span class="t1-zi4r-0-x-x-109">refPositionsFile</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
<li class="itemize">
<a id="dx1-64004"></a>
<div class="colorbox" id="colorbox140"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositionsCol:</span></div>
<!--l. 2577--><p class="noindent"> see definition of <a href="#colvar|rmsd|refPositionsCol"><span class="t1-zi4r-0-x-x-109">refPositionsCol</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component)
     </p></li>
<li class="itemize">
<a id="dx1-64005"></a>
<div class="colorbox" id="colorbox141"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositionsColValue:</span></div>
<!--l. 2577--><p class="noindent"> see definition of <a href="#colvar|rmsd|refPositionsColValue"><span class="t1-zi4r-0-x-x-109">refPositionsColValue</span></a> (<span class="t1-zi4r-0-x-x-109">rmsd </span>component) </p></li></ul>
<div class="flushright">
<!--l. 2810--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2810--><p class="noindent"></p><div class="heading"><a id="sec:cvc_protein">
<h4 class="subsectionHead"><span class="titlemark">4.7    </span> <a id="x1-650004.7"></a>Protein structure descriptors</h4>
<!--l. 2810--><p class="noindent"></p></a> </div> 
<div class="flushright">
<!--l. 2810--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2810--><p class="noindent"></p><div class="heading"><a id="sec:cvc_alpha">
<h5 class="subsubsectionHead"><span class="titlemark">4.7.1    </span> <a id="x1-660004.7.1"></a><span class="t1-zi4r-0-x-x-109">alpha</span>: <!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>α</mi></math>-helix
content of a protein segment.</h5>
<!--l. 2810--><p class="noindent"></p></a> </div>  <a id="colvar|alpha"> </a>
<!--l. 2810--><p class="indent">   The block <span class="t1-zi4r-0-x-x-109">alpha</span><span class="t1-zi4r-0-x-x-109"> {...} </span>defines the parameters to calculate the helical content of a segment of protein residues. The
<!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>α</mi></math>-helical content
across the <!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>N</mi> <mo class="MathClass-bin">+</mo><mn>1</mn></math>
residues <!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>N</mi></mrow><mrow><mn>0</mn></mrow></msub></math>
to <!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>N</mi></mrow><mrow><mn>0</mn></mrow></msub> <mo class="MathClass-bin">+</mo><mi>N</mi></math> is
calculated by the formula: </p>
<!--tex4ht:inline--><!--l. 2810--><math display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mtable class="eqnarray" columnalign="right center left">
<mtr><mtd class="eqnarray-1"> <mi>α</mi> <mfenced close=")" open="(" separators=""><mrow><msubsup><mrow><mi mathvariant="normal">C</mi></mrow><mrow><mi>α</mi></mrow><mrow><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>N</mi></mrow><mrow><mn>0</mn></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></mrow></msubsup><mo class="MathClass-punc">,</mo><msup><mrow><mi mathvariant="normal">O</mi></mrow><mrow><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>N</mi></mrow><mrow><mn>0</mn></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></mrow></msup><mo class="MathClass-punc">,</mo><msubsup><mrow><mi mathvariant="normal">C</mi></mrow><mrow>
<mi>α</mi></mrow><mrow><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>N</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-bin">+</mo><mn>1</mn></mrow><mo class="MathClass-close">)</mo></mrow></mrow></msubsup><mo class="MathClass-punc">,</mo><msup><mrow><mi mathvariant="normal">O</mi></mrow><mrow><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>N</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-bin">+</mo><mn>1</mn></mrow><mo class="MathClass-close">)</mo></mrow></mrow></msup><mo class="MathClass-punc">,</mo><mo class="MathClass-op">…</mo><msup><mrow><mi mathvariant="normal">N</mi></mrow><mrow><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>N</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-bin">+</mo><mn>5</mn></mrow><mo class="MathClass-close">)</mo></mrow></mrow></msup><mo class="MathClass-punc">,</mo><msubsup><mrow><mi mathvariant="normal">C</mi></mrow><mrow>
<mi>α</mi></mrow><mrow><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>N</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-bin">+</mo><mn>5</mn></mrow><mo class="MathClass-close">)</mo></mrow></mrow></msubsup><mo class="MathClass-punc">,</mo><msup><mrow><mi mathvariant="normal">O</mi></mrow><mrow><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>N</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-bin">+</mo><mn>5</mn></mrow><mo class="MathClass-close">)</mo></mrow></mrow></msup><mo class="MathClass-punc">,</mo><mo class="MathClass-op">…</mo><msup><mrow><mi mathvariant="normal">N</mi></mrow><mrow><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>N</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-bin">+</mo><mi>N</mi></mrow><mo class="MathClass-close">)</mo></mrow></mrow></msup><mo class="MathClass-punc">,</mo><msubsup><mrow><mi mathvariant="normal">C</mi></mrow><mrow>
<mi>α</mi></mrow><mrow><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>N</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-bin">+</mo><mi>N</mi></mrow><mo class="MathClass-close">)</mo></mrow></mrow></msubsup></mrow></mfenced><mspace class="tmspace" width="3.04074pt"></mspace> <mo class="MathClass-rel">=</mo> <mspace class="tmspace" width="3.04074pt"></mspace><mspace class="tmspace" width="3.04074pt"></mspace><mspace class="tmspace" width="3.04074pt"></mspace><mspace class="tmspace" width="3.04074pt"></mspace></mtd><mtd class="eqnarray-2"> </mtd><mtd class="eqnarray-3"> </mtd><mtd class="eqnarray-4"> <mtext class="eqnarray">(10)</mtext><mtext class="label" id="x1-66001r10"></mtext><mtext class="endlabel"></mtext></mtd>
</mtr><mtr><mtd class="eqnarray-1"> <mspace class="tmspace" width="3.04074pt"></mspace><mspace class="tmspace" width="3.04074pt"></mspace><mspace class="tmspace" width="3.04074pt"></mspace><mspace class="tmspace" width="3.04074pt"></mspace> <mfrac><mrow><mn>1</mn></mrow>
<mrow><mn>2</mn><mrow><mo class="MathClass-open">(</mo><mrow><mi>N</mi> <mo class="MathClass-bin">−</mo><mn>2</mn></mrow><mo class="MathClass-close">)</mo></mrow></mrow></mfrac><munderover accent="false" accentunder="false"><mrow><mo mathsize="big">∑</mo></mrow><mrow><mi>n</mi><mo class="MathClass-rel">=</mo><msub><mrow><mi>N</mi></mrow><mrow><mn>0</mn></mrow></msub></mrow><mrow><msub><mrow><mi>N</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-bin">+</mo><mi>N</mi><mo class="MathClass-bin">−</mo><mn>2</mn></mrow></munderover><mi mathvariant="normal">angf</mi> <mfenced close=")" open="(" separators=""><mrow><msubsup><mrow><mi mathvariant="normal">C</mi></mrow><mrow>
<mi>α</mi></mrow><mrow><mrow><mo class="MathClass-open">(</mo><mrow><mi>n</mi></mrow><mo class="MathClass-close">)</mo></mrow></mrow></msubsup><mo class="MathClass-punc">,</mo><msubsup><mrow><mi mathvariant="normal">C</mi></mrow><mrow>
<mi>α</mi></mrow><mrow><mrow><mo class="MathClass-open">(</mo><mrow><mi>n</mi><mo class="MathClass-bin">+</mo><mn>1</mn></mrow><mo class="MathClass-close">)</mo></mrow></mrow></msubsup><mo class="MathClass-punc">,</mo><msubsup><mrow><mi mathvariant="normal">C</mi></mrow><mrow>
<mi>α</mi></mrow><mrow><mrow><mo class="MathClass-open">(</mo><mrow><mi>n</mi><mo class="MathClass-bin">+</mo><mn>2</mn></mrow><mo class="MathClass-close">)</mo></mrow></mrow></msubsup></mrow></mfenced><mspace class="tmspace" width="3.04074pt"></mspace> <mo class="MathClass-bin">+</mo><mspace class="tmspace" width="3.04074pt"></mspace> <mfrac><mrow><mn>1</mn></mrow>
<mrow><mn>2</mn><mrow><mo class="MathClass-open">(</mo><mrow><mi>N</mi> <mo class="MathClass-bin">−</mo><mn>4</mn></mrow><mo class="MathClass-close">)</mo></mrow></mrow></mfrac><munderover accent="false" accentunder="false"><mrow><mo mathsize="big">∑</mo></mrow><mrow><mi>n</mi><mo class="MathClass-rel">=</mo><msub><mrow><mi>N</mi></mrow><mrow><mn>0</mn></mrow></msub></mrow><mrow><msub><mrow><mi>N</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-bin">+</mo><mi>N</mi><mo class="MathClass-bin">−</mo><mn>4</mn></mrow></munderover><mi mathvariant="normal">hbf</mi> <mfenced close=")" open="(" separators=""><mrow><msup><mrow><mi mathvariant="normal">O</mi></mrow><mrow><mrow><mo class="MathClass-open">(</mo><mrow><mi>n</mi></mrow><mo class="MathClass-close">)</mo></mrow></mrow></msup><mo class="MathClass-punc">,</mo><msup><mrow><mi mathvariant="normal">N</mi></mrow><mrow><mrow><mo class="MathClass-open">(</mo><mrow><mi>n</mi><mo class="MathClass-bin">+</mo><mn>4</mn></mrow><mo class="MathClass-close">)</mo></mrow></mrow></msup></mrow></mfenced><mi mathvariant="normal">,</mi></mtd><mtd class="eqnarray-2"> </mtd><mtd class="eqnarray-3"> </mtd><mtd class="eqnarray-4"> <mtext class="eqnarray"></mtext></mtd>
</mtr><mtr><mtd class="eqnarray-1"> </mtd><mtd class="eqnarray-2"> </mtd><mtd class="eqnarray-3"> </mtd><mtd class="eqnarray-4"> <mtext class="eqnarray">(11)</mtext><mtext class="label" id="x1-66001r11"></mtext><mtext class="endlabel"></mtext></mtd> </mtr></mtable>
</math>
<!--l. 2810--><p class="nopar">
</p>
<!--l. 2810--><p class="indent">   where the score function for the <!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="normal">C</mi></mrow><mrow><mi>α</mi></mrow></msub><mo class="MathClass-bin">−</mo><msub><mrow><mi mathvariant="normal">C</mi></mrow><mrow><mi>α</mi></mrow></msub><mo class="MathClass-bin">−</mo><msub><mrow><mi mathvariant="normal">C</mi></mrow><mrow><mi>α</mi></mrow></msub></math>
angle is defined as: </p><table class="equation"><tr><td> <a id="x1-66002r12"></a>
<!--l. 2810--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi mathvariant="normal">angf</mi> <mfenced close=")" open="(" separators=""><mrow><msubsup><mrow><mi mathvariant="normal">C</mi></mrow><mrow><mi>α</mi></mrow><mrow><mrow><mo class="MathClass-open">(</mo><mrow><mi>n</mi></mrow><mo class="MathClass-close">)</mo></mrow></mrow></msubsup><mo class="MathClass-punc">,</mo><msubsup><mrow><mi mathvariant="normal">C</mi></mrow><mrow>
<mi>α</mi></mrow><mrow><mrow><mo class="MathClass-open">(</mo><mrow><mi>n</mi><mo class="MathClass-bin">+</mo><mn>1</mn></mrow><mo class="MathClass-close">)</mo></mrow></mrow></msubsup><mo class="MathClass-punc">,</mo><msubsup><mrow><mi mathvariant="normal">C</mi></mrow><mrow>
<mi>α</mi></mrow><mrow><mrow><mo class="MathClass-open">(</mo><mrow><mi>n</mi><mo class="MathClass-bin">+</mo><mn>2</mn></mrow><mo class="MathClass-close">)</mo></mrow></mrow></msubsup></mrow></mfenced><mspace class="tmspace" width="3.04074pt"></mspace> <mo class="MathClass-rel">=</mo> <mspace class="tmspace" width="3.04074pt"></mspace><mfrac><mrow><mn>1</mn><mo class="MathClass-bin">−</mo><msup><mrow><mfenced close=")" open="(" separators=""><mrow><mi>𝜃</mi><mrow><mo class="MathClass-open">(</mo><mrow><msubsup><mrow><mi mathvariant="normal">C</mi></mrow><mrow><mi>α</mi></mrow><mrow><mrow><mo class="MathClass-open">(</mo><mrow><mi>n</mi></mrow><mo class="MathClass-close">)</mo></mrow></mrow></msubsup><mo class="MathClass-punc">,</mo><msubsup><mrow><mi mathvariant="normal">C</mi></mrow><mrow><mi>α</mi></mrow><mrow><mrow><mo class="MathClass-open">(</mo><mrow><mi>n</mi><mo class="MathClass-bin">+</mo><mn>1</mn></mrow><mo class="MathClass-close">)</mo></mrow></mrow></msubsup><mo class="MathClass-punc">,</mo><msubsup><mrow><mi mathvariant="normal">C</mi></mrow><mrow><mi>α</mi></mrow><mrow><mrow><mo class="MathClass-open">(</mo><mrow><mi>n</mi><mo class="MathClass-bin">+</mo><mn>2</mn></mrow><mo class="MathClass-close">)</mo></mrow></mrow></msubsup></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-bin">−</mo><msub><mrow><mi>𝜃</mi></mrow><mrow><mn>0</mn></mrow></msub></mrow></mfenced></mrow><mrow><mn>2</mn></mrow></msup><mo class="MathClass-bin">∕</mo><msup><mrow> <mfenced close=")" open="(" separators=""><mrow><mi>Δ</mi><msub><mrow><mi>𝜃</mi></mrow><mrow><mi mathvariant="normal">tol</mi></mrow></msub></mrow></mfenced></mrow><mrow><mn>2</mn></mrow></msup></mrow>
<mrow><mn>1</mn><mo class="MathClass-bin">−</mo><msup><mrow><mfenced close=")" open="(" separators=""><mrow><mi>𝜃</mi><mrow><mo class="MathClass-open">(</mo><mrow><msubsup><mrow><mi mathvariant="normal">C</mi></mrow><mrow><mi>α</mi></mrow><mrow><mrow><mo class="MathClass-open">(</mo><mrow><mi>n</mi></mrow><mo class="MathClass-close">)</mo></mrow></mrow></msubsup><mo class="MathClass-punc">,</mo><msubsup><mrow><mi mathvariant="normal">C</mi></mrow><mrow><mi>α</mi></mrow><mrow><mrow><mo class="MathClass-open">(</mo><mrow><mi>n</mi><mo class="MathClass-bin">+</mo><mn>1</mn></mrow><mo class="MathClass-close">)</mo></mrow></mrow></msubsup><mo class="MathClass-punc">,</mo><msubsup><mrow><mi mathvariant="normal">C</mi></mrow><mrow><mi>α</mi></mrow><mrow><mrow><mo class="MathClass-open">(</mo><mrow><mi>n</mi><mo class="MathClass-bin">+</mo><mn>2</mn></mrow><mo class="MathClass-close">)</mo></mrow></mrow></msubsup></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-bin">−</mo><msub><mrow><mi>𝜃</mi></mrow><mrow><mn>0</mn></mrow></msub></mrow></mfenced></mrow><mrow><mn>4</mn></mrow></msup><mo class="MathClass-bin">∕</mo><msup><mrow> <mfenced close=")" open="(" separators=""><mrow><mi>Δ</mi><msub><mrow><mi>𝜃</mi></mrow><mrow><mi mathvariant="normal">tol</mi></mrow></msub></mrow></mfenced></mrow><mrow><mn>4</mn></mrow></msup></mrow></mfrac><mi mathvariant="normal">,</mi>
</math></td><td class="eq-no">(12)</td></tr></table>
<!--l. 2810--><p class="indent">   and the score function for the <!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow><mi mathvariant="normal">O</mi></mrow><mrow><mrow><mo class="MathClass-open">(</mo><mrow><mi>n</mi></mrow><mo class="MathClass-close">)</mo></mrow></mrow></msup> <mo class="MathClass-rel">↔</mo><msup><mrow><mi mathvariant="normal">N</mi></mrow><mrow><mrow><mo class="MathClass-open">(</mo><mrow><mi>n</mi><mo class="MathClass-bin">+</mo><mn>4</mn></mrow><mo class="MathClass-close">)</mo></mrow></mrow></msup></math>
hydrogen bond is defined through a <span class="t1-zi4r-0-x-x-109">hBond </span>colvar component on the same atoms.
                                                                                         
                                                                                         
</p><!--l. 2810--><p class="noindent"><span class="ptmb8t-x-x-109">List of keywords </span>(see also <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="colvar|alpha|residueRange"> </a> <a id="dx1-66003"></a><span class="colorbox" id="colorbox142"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">residueRange</span></span> <!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Potential
     <!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>α</mi></math>-helical
     residues<!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">alpha</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>“<!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>Initial
     residue number<!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>-<!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>Final
     residue number<!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>"
     <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>This option specifies the range of residues on which this component should be defined.
     The Colvars module looks for the atoms within these residues named “<span class="t1-zi4r-0-x-x-109">CA</span>", “<span class="t1-zi4r-0-x-x-109">N</span>" and “<span class="t1-zi4r-0-x-x-109">O</span>", and raises an
     error if any of those atoms is not found.
     </li>
<li class="itemize"><a id="colvar|alpha|psfSegID"> </a> <a id="dx1-66004"></a><span class="colorbox" id="colorbox143"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">psfSegID</span></span> <!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>PSF
     segment identifier<!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">alpha</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>string (max 4 characters) <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>This option sets the PSF segment identifier for the residues specified in <span class="t1-zi4r-0-x-x-109">residueRange</span>.
     This option is only required when PSF topologies are used.
     </li>
<li class="itemize"><a id="colvar|alpha|hBondCoeff"> </a> <a id="dx1-66005"></a><span class="colorbox" id="colorbox144"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">hBondCoeff</span></span> <!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Coefficient
     for the hydrogen bond term<!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">alpha</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive between 0 and 1 <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span>0.5 <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>This number specifies the contribution to the total value from the hydrogen bond terms.
     0 disables the hydrogen bond terms, 1 disables the angle terms.
     </li>
<li class="itemize"><a id="colvar|alpha|angleRef"> </a> <a id="dx1-66006"></a><span class="colorbox" id="colorbox145"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">angleRef</span></span> <!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Reference
     <!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="normal">C</mi></mrow><mrow><mi>α</mi></mrow></msub><mo class="MathClass-bin">−</mo><msub><mrow><mi mathvariant="normal">C</mi></mrow><mrow><mi>α</mi></mrow></msub><mo class="MathClass-bin">−</mo><msub><mrow><mi mathvariant="normal">C</mi></mrow><mrow><mi>α</mi></mrow></msub></math>
     angle<!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">alpha</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive decimal <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span>88<!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></math>
<br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>This option sets the reference angle used in the score function (<a href="#x1-66002r12">12<!--tex4ht:ref: eq:colvars_alpha_Calpha --></a>).
     </li>
<li class="itemize"><a id="colvar|alpha|angleTol"> </a> <a id="dx1-66007"></a><span class="colorbox" id="colorbox146"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">angleTol</span></span> <!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Tolerance
     in the <!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="normal">C</mi></mrow><mrow><mi>α</mi></mrow></msub><mo class="MathClass-bin">−</mo><msub><mrow><mi mathvariant="normal">C</mi></mrow><mrow><mi>α</mi></mrow></msub><mo class="MathClass-bin">−</mo><msub><mrow><mi mathvariant="normal">C</mi></mrow><mrow><mi>α</mi></mrow></msub></math>
     angle<!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">alpha</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive decimal <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span>15<!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></math>
<br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>This option sets the angle tolerance used in the score function (<a href="#x1-66002r12">12<!--tex4ht:ref: eq:colvars_alpha_Calpha --></a>).
     </li>
<li class="itemize"><a id="colvar|alpha|hBondCutoff"> </a> <a id="dx1-66008"></a><span class="colorbox" id="colorbox147"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">hBondCutoff</span></span> <!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Hydrogen
     bond cutoff<!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">alpha</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive decimal <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span>3.3 Å <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Equivalent to the <span class="t1-zi4r-0-x-x-109">cutoff </span>option in the <span class="t1-zi4r-0-x-x-109">hBond </span>component.
     </li>
<li class="itemize"><a id="colvar|alpha|hBondExpNumer"> </a> <a id="dx1-66009"></a><span class="colorbox" id="colorbox148"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">hBondExpNumer</span></span> <!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Hydrogen
     bond numerator exponent<!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">alpha</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive integer <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span>6 <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Equivalent to the <span class="t1-zi4r-0-x-x-109">expNumer </span>option in the <span class="t1-zi4r-0-x-x-109">hBond </span>component.
     </li>
<li class="itemize"><a id="colvar|alpha|hBondExpDenom"> </a> <a id="dx1-66010"></a><span class="colorbox" id="colorbox149"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">hBondExpDenom</span></span> <!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Hydrogen
     bond denominator exponent<!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">alpha</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive integer <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span>8 <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Equivalent to the <span class="t1-zi4r-0-x-x-109">expDenom </span>option in the <span class="t1-zi4r-0-x-x-109">hBond </span>component.
     </li></ul>
<!--l. 2810--><p class="indent">   This component returns positive values, always comprised between 0 (lowest
<!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>α</mi></math>-helical score) and
1 (highest <!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>α</mi></math>-helical
score).
</p>
<div class="flushright">
<!--l. 2810--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2810--><p class="noindent"></p><div class="heading"><a id="sec:cvc_dihedralPC">
<h5 class="subsubsectionHead"><span class="titlemark">4.7.2    </span> <a id="x1-670004.7.2"></a><span class="t1-zi4r-0-x-x-109">dihedralPC</span>: protein dihedral principal component</h5>
<!--l. 2810--><p class="noindent"></p></a> </div>  <a id="colvar|dihedralPC"> </a>
<!--l. 2810--><p class="indent">   The block <span class="t1-zi4r-0-x-x-109">dihedralPC</span><span class="t1-zi4r-0-x-x-109"> {...} </span>defines the parameters to calculate the projection of backbone dihedral angles within a
protein segment onto a <span class="ptmri8t-x-x-109">dihedral principal component</span>, following the formalism of dihedral principal component analysis
(dPCA) proposed by Mu et al.<span class="cite">[<a href="#XMu2005">6</a>]</span> and documented in detail by Altis et al.<span class="cite">[<a href="#XAltis2007">7</a>]</span>. Given a peptide or protein segment of
<!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>N</mi></math> residues, each with
Ramachandran angles <!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ϕ</mi></mrow><mrow><mi>i</mi></mrow></msub></math>
                                                                                         
                                                                                         
and <!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ψ</mi></mrow><mrow><mi>i</mi></mrow></msub></math>, dPCA rests on a
variance/covariance analysis of the <!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>4</mn><mrow><mo class="MathClass-open">(</mo><mrow><mi>N</mi> <mo class="MathClass-bin">−</mo><mn>1</mn></mrow><mo class="MathClass-close">)</mo></mrow></math>
variables <!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="qopname">cos</mo><!--nolimits--><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ψ</mi></mrow><mrow><mn>1</mn></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-punc">,</mo><mo class="qopname">sin</mo><!--nolimits--><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ψ</mi></mrow><mrow><mn>1</mn></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-punc">,</mo><mo class="qopname">cos</mo><!--nolimits--><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ϕ</mi></mrow><mrow><mn>2</mn></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-punc">,</mo><mo class="qopname">sin</mo><!--nolimits--><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ϕ</mi></mrow><mrow><mn>2</mn></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-rel">⋯</mo><mo class="qopname">cos</mo><!--nolimits--><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ϕ</mi></mrow><mrow><mi>N</mi></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-punc">,</mo><mo class="qopname">sin</mo><!--nolimits--><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ϕ</mi></mrow><mrow><mi>N</mi></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math>. Note
that angles <!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ϕ</mi></mrow><mrow><mn>1</mn></mrow></msub></math>
and <!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ψ</mi></mrow><mrow><mi>N</mi></mrow></msub></math>
have little impact on chain conformation, and are therefore discarded, following the implementation of dPCA in the
analysis software Carma.<span class="cite">[<a href="#XGlykos2006">8</a>]</span>
</p><!--l. 2810--><p class="indent">   For a given principal component (eigenvector) of coefficients
<!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>k</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></mrow><mrow><mn>1</mn><mo class="MathClass-rel">≤</mo><mi>i</mi><mo class="MathClass-rel">≤</mo><mn>4</mn><mrow><mo class="MathClass-open">(</mo><mrow><mi>N</mi><mo class="MathClass-bin">−</mo><mn>1</mn></mrow><mo class="MathClass-close">)</mo></mrow></mrow></msub></math>, the
projection of the current backbone conformation is: </p><table class="equation"><tr><td> <a id="x1-67001r13"></a>
<!--l. 2810--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>ξ</mi> <mo class="MathClass-rel">=</mo><munderover accent="false" accentunder="false"><mrow><mo mathsize="big"> ∑</mo></mrow><mrow><mi>n</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><mi>N</mi><mo class="MathClass-bin">−</mo><mn>1</mn></mrow></munderover><msub><mrow><mi>k</mi></mrow><mrow>
<mn>4</mn><mi>n</mi><mo class="MathClass-bin">−</mo><mn>3</mn></mrow></msub><mo class="qopname">cos</mo><!--nolimits--><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ψ</mi></mrow><mrow><mi>n</mi></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-bin">+</mo><msub><mrow><mi>k</mi></mrow><mrow><mn>4</mn><mi>n</mi><mo class="MathClass-bin">−</mo><mn>2</mn></mrow></msub><mo class="qopname">sin</mo><!--nolimits--><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ψ</mi></mrow><mrow><mi>n</mi></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-bin">+</mo><msub><mrow><mi>k</mi></mrow><mrow><mn>4</mn><mi>n</mi><mo class="MathClass-bin">−</mo><mn>1</mn></mrow></msub><mo class="qopname">cos</mo><!--nolimits--><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ϕ</mi></mrow><mrow><mi>n</mi><mo class="MathClass-bin">+</mo><mn>1</mn></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-bin">+</mo><msub><mrow><mi>k</mi></mrow><mrow><mn>4</mn><mi>n</mi></mrow></msub><mo class="qopname">sin</mo><!--nolimits--><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ϕ</mi></mrow><mrow><mi>n</mi><mo class="MathClass-bin">+</mo><mn>1</mn></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow>
</math></td><td class="eq-no">(13)</td></tr></table>
<!--l. 2810--><p class="indent"> <span class="t1-zi4r-0-x-x-109">dihedralPC </span>expects the same parameters as the <span class="t1-zi4r-0-x-x-109">alpha </span>component for defining the relevant residues
(<span class="t1-zi4r-0-x-x-109">residueRange </span>and <span class="t1-zi4r-0-x-x-109">psfSegID</span>) in addition to the following:
</p><!--l. 2810--><p class="noindent"><span class="ptmb8t-x-x-109">List of keywords </span>(see also <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-67002"></a><span class="colorbox" id="colorbox150"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">residueRange:</span></span> see definition of <a href="#colvar|alpha|residueRange"><span class="t1-zi4r-0-x-x-109">residueRange</span></a> (<span class="t1-zi4r-0-x-x-109">alpha </span>component)
     </li>
<li class="itemize"><a id="dx1-67003"></a><span class="colorbox" id="colorbox151"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">psfSegID:</span></span> see definition of <a href="#colvar|alpha|psfSegID"><span class="t1-zi4r-0-x-x-109">psfSegID</span></a> (<span class="t1-zi4r-0-x-x-109">alpha </span>component)
     </li>
<li class="itemize"><a id="dx1-67004"></a><span class="colorbox" id="colorbox152"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">vectorFile</span></span> <!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>File
     containing dihedral PCA eigenvector(s)<!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">dihedralPC</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>file name <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>A text file containing the coefficients of dihedral PCA eigenvectors on the cosine and
     sine coordinates. The vectors should be arranged in columns, as in the files output by Carma.<span class="cite">[<a href="#XGlykos2006">8</a>]</span>
</li>
<li class="itemize"><a id="dx1-67005"></a><span class="colorbox" id="colorbox153"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">vectorNumber</span></span> <!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>File
     containing dihedralPCA eigenvector(s)<!--l. 2810--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">dihedralPC</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive integer <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Number of the eigenvector to be used for this component.</li></ul>
<div class="flushright">
<!--l. 2847--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2847--><p class="noindent"></p><div class="heading"><a id="sec:cvc_raw">
<h4 class="subsectionHead"><span class="titlemark">4.8    </span> <a id="x1-680004.8"></a>Raw data: building blocks for custom functions</h4>
<!--l. 2847--><p class="noindent"></p></a> </div> 
<div class="flushright">
<!--l. 2849--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2849--><p class="noindent"></p><div class="heading"><a id="sec:cvc_cartesian">
<h5 class="subsubsectionHead"><span class="titlemark">4.8.1    </span> <a id="x1-690004.8.1"></a><span class="t1-zi4r-0-x-x-109">cartesian</span>: vector of atomic Cartesian coordinates.</h5>
<!--l. 2849--><p class="noindent"></p></a> </div>  <a id="colvar|cartesian"> </a>
<!--l. 2852--><p class="indent">   The <span class="t1-zi4r-0-x-x-109">cartesian</span><span class="t1-zi4r-0-x-x-109"> {...} </span>block defines a component returning a flat vector containing the Cartesian coordinates of all participating
atoms, in the order <!--l. 2854--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>x</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>y</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>z</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-punc">,</mo><mo class="MathClass-rel">⋯</mo><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-punc">,</mo><msub><mrow><mi>x</mi></mrow><mrow><mi>n</mi></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>y</mi></mrow><mrow><mi>n</mi></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>z</mi></mrow><mrow><mi>n</mi></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math>.
</p><!--l. 2856--><p class="noindent"><span class="ptmb8t-x-x-109">List of keywords </span>(see also <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-69001"></a><span class="colorbox" id="colorbox154"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">atoms</span></span> <!--l. 2865--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Group
     of atoms<!--l. 2865--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">cartesian</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>Block <span class="t1-zi4r-0-x-x-109">atoms {...}</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Defines the atoms whose coordinates make up the value of the component. If <span class="t1-zi4r-0-x-x-109">rotateToReference</span>,
     <span class="t1-zi4r-0-x-x-109">centerToReference</span>, or <span class="t1-zi4r-0-x-x-109">centerToOrigin </span>are defined, coordinates are evaluated within the moving
     frame of reference.</li></ul>
<div class="flushright">
<!--l. 2869--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2869--><p class="noindent"></p><div class="heading"><a id="sec:cvc_distancePairs">
<h5 class="subsubsectionHead"><span class="titlemark">4.8.2    </span> <a id="x1-700004.8.2"></a><span class="t1-zi4r-0-x-x-109">distancePairs</span>: set of pairwise distances between two groups.</h5>
<!--l. 2869--><p class="noindent"></p></a> </div>  <a id="colvar|distancePairs"> </a>
<!--l. 2872--><p class="indent">   The  <span class="t1-zi4r-0-x-x-109">distancePairs</span><span class="t1-zi4r-0-x-x-109"> {...} </span>block  defines  a
<!--l. 2872--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>N</mi></mrow><mrow><mi mathvariant="normal">1</mi></mrow></msub> <mo class="MathClass-bin">×</mo><msub><mrow><mi>N</mi></mrow><mrow><mi mathvariant="normal">2</mi></mrow></msub></math>-dimensional
variable that includes all mutual distances between the atoms of two groups. This can be useful, for example, to
develop a new variable defined over two groups, by using the <span class="t1-zi4r-0-x-x-109">scriptedFunction </span>feature.
</p><!--l. 2875--><p class="noindent"><span class="ptmb8t-x-x-109">List of keywords </span>(see also <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-70001"></a><span class="colorbox" id="colorbox155"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">group1:</span></span> see definition of <a href="#colvar|distance|group1"><span class="t1-zi4r-0-x-x-109">group1</span></a> (<span class="t1-zi4r-0-x-x-109">distance </span>component)
     </li>
<li class="itemize"><a id="dx1-70002"></a><span class="colorbox" id="colorbox156"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">group2:</span></span> analogous to <span class="t1-zi4r-0-x-x-109">group1</span>
</li>
<li class="itemize"><a id="dx1-70003"></a><span class="colorbox" id="colorbox157"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">forceNoPBC:</span></span> see definition of <a href="#colvar|distance|forceNoPBC"><span class="t1-zi4r-0-x-x-109">forceNoPBC</span></a> (<span class="t1-zi4r-0-x-x-109">distance </span>component)</li></ul>
<!--l. 2883--><p class="noindent">This component returns a <!--l. 2883--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>N</mi></mrow><mrow><mi mathvariant="normal">1</mi></mrow></msub> <mo class="MathClass-bin">×</mo><msub><mrow><mi>N</mi></mrow><mrow><mi mathvariant="normal">2</mi></mrow></msub></math>-dimensional
vector of numbers, each ranging from <!--l. 2883--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>0</mn></math>
to the largest possible distance within the chosen boundary conditions.
</p>
<div class="flushright">
<!--l. 2886--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2886--><p class="noindent"></p><div class="heading"><a id="sec:cvc_gpath">
<h4 class="subsectionHead"><span class="titlemark">4.9    </span> <a id="x1-710004.9"></a>Geometric path collective variables</h4>
<!--l. 2886--><p class="noindent"></p></a> </div> 
<!--l. 2888--><p class="indent">   The geometric path collective variables define the progress along a path,
<!--l. 2888--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>s</mi></math>, and the distance
from the path, <!--l. 2888--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math>.
These CVs are proposed by Leines and Ensing<span class="cite">[<a href="#XLeines2012">9</a>]</span> , which differ from that<span class="cite">[<a href="#XBranduardi2007">10</a>]</span> proposed by Branduardi et al., and utilize
a set of geometric algorithms. The path is defined as a series of frames in the atomic Cartesian coordinate space or the
CV space. <!--l. 2888--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>s</mi></math>
and <!--l. 2888--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math> are
computed as
</p>
<table class="equation"><tr><td> <a id="x1-71001r14"></a>
<!--l. 2890--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>s</mi> <mo class="MathClass-rel">=</mo> <mfrac><mrow><mi>m</mi></mrow>
<mrow><mi>M</mi></mrow></mfrac> <mo class="MathClass-bin">±</mo> <mfrac><mrow><mn>1</mn></mrow>
<mrow><mn>2</mn><mi>M</mi></mrow></mfrac> <mfenced close=")" open="(" separators=""><mrow><mfrac><mrow><msqrt><mrow><msup><mrow><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi mathvariant="bold">v</mi> </mrow><mrow><mn>1</mn> </mrow> </msub> <mo class="MathClass-bin">⋅</mo><msub><mrow><mi mathvariant="bold">v</mi> </mrow><mrow><mn>3</mn> </mrow> </msub> </mrow><mo class="MathClass-close">)</mo></mrow></mrow><mrow><mn>2</mn> </mrow> </msup> <mo class="MathClass-bin">−</mo><mo class="MathClass-rel">|</mo><msub><mrow><mi mathvariant="bold">v</mi> </mrow><mrow><mn>3</mn> </mrow> </msub> <msup><mrow><mo class="MathClass-rel">|</mo></mrow><mrow><mn>2</mn> </mrow> </msup> <mrow><mo class="MathClass-open">(</mo><mrow><mo class="MathClass-rel">|</mo><msub><mrow><mi mathvariant="bold">v</mi> </mrow><mrow><mn>1</mn> </mrow> </msub> <msup><mrow><mo class="MathClass-rel">|</mo></mrow><mrow><mn>2</mn> </mrow> </msup> <mo class="MathClass-bin">−</mo><mo class="MathClass-rel">|</mo><msub><mrow><mi mathvariant="bold">v</mi> </mrow><mrow><mn>2</mn> </mrow> </msub> <msup><mrow><mo class="MathClass-rel">|</mo></mrow><mrow><mn>2</mn> </mrow> </msup> </mrow><mo class="MathClass-close">)</mo></mrow></mrow></msqrt><mo class="MathClass-bin">−</mo><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi mathvariant="bold">v</mi></mrow><mrow><mn>1</mn></mrow></msub> <mo class="MathClass-bin">⋅</mo><msub><mrow><mi mathvariant="bold">v</mi></mrow><mrow><mn>3</mn></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></mrow>
<mrow><mo class="MathClass-rel">|</mo><msub><mrow><mi mathvariant="bold">v</mi></mrow><mrow><mn>3</mn></mrow></msub><msup><mrow><mo class="MathClass-rel">|</mo></mrow><mrow><mn>2</mn></mrow></msup></mrow></mfrac> <mo class="MathClass-bin">−</mo><mn>1</mn></mrow></mfenced>
</math></td><td class="eq-no">(14)</td></tr></table>
<table class="equation"><tr><td> <a id="x1-71002r15"></a>
<!--l. 2894--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>z</mi> <mo class="MathClass-rel">=</mo> <msqrt><mrow><msup><mrow><mfenced close=")" open="(" separators=""><mrow><msub><mrow><mi mathvariant="bold">v</mi> </mrow><mrow><mn>1</mn> </mrow> </msub> <mo class="MathClass-bin">+</mo> <mfrac> <mrow> <mn>1</mn></mrow>
<mrow><mn>2</mn></mrow></mfrac> <mfenced close=")" open="(" separators=""><mrow><mfrac><mrow><msqrt><mrow><msup><mrow><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi mathvariant="bold">v</mi> </mrow><mrow><mn>1</mn> </mrow> </msub> <mo class="MathClass-bin">⋅</mo><msub><mrow><mi mathvariant="bold">v</mi> </mrow><mrow><mn>3</mn> </mrow> </msub> </mrow><mo class="MathClass-close">)</mo></mrow></mrow><mrow><mn>2</mn> </mrow> </msup> <mo class="MathClass-bin">−</mo><mo class="MathClass-rel">|</mo><msub><mrow><mi mathvariant="bold">v</mi> </mrow><mrow><mn>3</mn> </mrow> </msub> <msup><mrow><mo class="MathClass-rel">|</mo></mrow><mrow><mn>2</mn> </mrow> </msup> <mrow><mo class="MathClass-open">(</mo><mrow><mo class="MathClass-rel">|</mo><msub><mrow><mi mathvariant="bold">v</mi> </mrow><mrow><mn>1</mn> </mrow> </msub> <msup><mrow><mo class="MathClass-rel">|</mo></mrow><mrow><mn>2</mn> </mrow> </msup> <mo class="MathClass-bin">−</mo><mo class="MathClass-rel">|</mo><msub><mrow><mi mathvariant="bold">v</mi> </mrow><mrow><mn>2</mn> </mrow> </msub> <msup><mrow><mo class="MathClass-rel">|</mo></mrow><mrow><mn>2</mn> </mrow> </msup> </mrow><mo class="MathClass-close">)</mo></mrow></mrow></msqrt><mo class="MathClass-bin">−</mo><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi mathvariant="bold">v</mi></mrow><mrow><mn>1</mn></mrow></msub> <mo class="MathClass-bin">⋅</mo><msub><mrow><mi mathvariant="bold">v</mi></mrow><mrow><mn>3</mn></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></mrow>
<mrow><mo class="MathClass-rel">|</mo><msub><mrow><mi mathvariant="bold">v</mi></mrow><mrow><mn>3</mn></mrow></msub><msup><mrow><mo class="MathClass-rel">|</mo></mrow><mrow><mn>2</mn></mrow></msup></mrow></mfrac> <mo class="MathClass-bin">−</mo><mn>1</mn></mrow></mfenced><msub><mrow><mi mathvariant="bold">v</mi></mrow><mrow><mn>4</mn></mrow></msub></mrow></mfenced></mrow><mrow><mn>2</mn></mrow></msup></mrow></msqrt>
</math></td><td class="eq-no">(15)</td></tr></table>
<!--l. 2898--><p class="indent">   where <!--l. 2898--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="bold">v</mi></mrow><mrow><mn>1</mn></mrow></msub> <mo class="MathClass-rel">=</mo><msub><mrow> <mi mathvariant="bold">s</mi></mrow><mrow><mi>m</mi></mrow></msub><mo class="MathClass-bin">−</mo><mi mathvariant="bold">z</mi></math>
is the vector connecting the current position to the closest frame,
<!--l. 2898--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="bold">v</mi></mrow><mrow><mn>2</mn></mrow></msub> <mo class="MathClass-rel">=</mo> <mi mathvariant="bold">z</mi> <mo class="MathClass-bin">−</mo><msub><mrow><mi mathvariant="bold">s</mi></mrow><mrow><mi>m</mi><mo class="MathClass-bin">−</mo><mn>1</mn></mrow></msub></math>
is the vector connecting the second closest frame to the current position,
<!--l. 2898--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="bold">v</mi></mrow><mrow><mn>3</mn></mrow></msub> <mo class="MathClass-rel">=</mo><msub><mrow> <mi mathvariant="bold">s</mi></mrow><mrow><mi>m</mi><mo class="MathClass-bin">+</mo><mn>1</mn></mrow></msub> <mo class="MathClass-bin">−</mo><msub><mrow><mi mathvariant="bold">s</mi></mrow><mrow><mi>m</mi></mrow></msub></math>
is the vector connecting the closest frame to the third closest frame, and
<!--l. 2898--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="bold">v</mi></mrow><mrow><mn>4</mn></mrow></msub> <mo class="MathClass-rel">=</mo><msub><mrow> <mi mathvariant="bold">s</mi></mrow><mrow><mi>m</mi></mrow></msub><mo class="MathClass-bin">−</mo><msub><mrow><mi mathvariant="bold">s</mi></mrow><mrow><mi>m</mi><mo class="MathClass-bin">−</mo><mn>1</mn></mrow></msub></math>
is the vector connecting the second closest frame to the closest frame.
<!--l. 2898--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>m</mi></math> and
<!--l. 2898--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>M</mi></math> are the current index
of the closest frame and the total number of frames, respectively. If the current position is on the left of the closest reference
frame, the <!--l. 2898--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-bin">±</mo></math> in
<!--l. 2898--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>s</mi></math> turns to
the positive sign. Otherwise it turns to the negative sign.
</p><!--l. 2900--><p class="indent">   The equations above assume: (i) the frames are equidistant and (ii) the second and the third closest frames are
neighbouring to the closest frame. When these assumptions are not satisfied, this set of path CV should be used
carefully.
</p>
<div class="flushright">
<!--l. 2902--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2902--><p class="noindent"></p><div class="heading"><a id="sec:cvc_gspath">
<h5 class="subsubsectionHead"><span class="titlemark">4.9.1    </span> <a id="x1-720004.9.1"></a><span class="t1-zi4r-0-x-x-109">gspath</span>: progress along a path defined in atomic Cartesian coordinate space.</h5>
<!--l. 2902--><p class="noindent"></p></a> </div>  <a id="colvar|gspath"> </a>
<!--l. 2905--><p class="indent">   In the <span class="t1-zi4r-0-x-x-109">gspath</span><span class="t1-zi4r-0-x-x-109"> {...} </span>and the <span class="t1-zi4r-0-x-x-109">gzpath</span><span class="t1-zi4r-0-x-x-109"> {...} </span>block all vectors, namely
<!--l. 2905--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="bold">z</mi></math> and
<!--l. 2905--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="bold">s</mi></mrow><mrow><mi>k</mi></mrow></msub></math>
are defined in atomic Cartesian coordinate space. More specifically,
<!--l. 2905--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="bold">z</mi> <mo class="MathClass-rel">=</mo> <mfenced close="]" open="[" separators=""><mrow><msub><mrow><mi mathvariant="bold">r</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-punc">,</mo><mo class="MathClass-rel">⋯</mo><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-punc">,</mo><msub><mrow><mi mathvariant="bold">r</mi></mrow><mrow><mi>n</mi></mrow></msub></mrow></mfenced></math>, where
<!--l. 2905--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="bold">r</mi></mrow><mrow><mi>i</mi></mrow></msub></math> is the
<!--l. 2905--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math>-th atom specified in
the <span class="t1-zi4r-0-x-x-109">atoms </span>block. <!--l. 2905--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="bold">s</mi></mrow><mrow><mi>k</mi></mrow></msub> <mo class="MathClass-rel">=</mo> <mfenced close="]" open="[" separators=""><mrow><msub><mrow><mi mathvariant="bold">r</mi></mrow><mrow><mi>k</mi><mo class="MathClass-punc">,</mo><mn>1</mn></mrow></msub><mo class="MathClass-punc">,</mo><mo class="MathClass-rel">⋯</mo><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-punc">,</mo><msub><mrow><mi mathvariant="bold">r</mi></mrow><mrow><mi>k</mi><mo class="MathClass-punc">,</mo><mi>n</mi></mrow></msub></mrow></mfenced></math>,
where <!--l. 2905--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="bold">r</mi></mrow><mrow><mi>k</mi><mo class="MathClass-punc">,</mo><mi>i</mi></mrow></msub></math> means
                                                                                         
                                                                                         
the <!--l. 2905--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math>-th atom
of the <!--l. 2905--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>k</mi></math>-th
reference frame.
</p><!--l. 2907--><p class="noindent"><span class="ptmb8t-x-x-109">List of keywords </span>(see also <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-72001"></a><span class="colorbox" id="colorbox158"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">atoms</span></span> <!--l. 2914--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Group
     of atoms<!--l. 2914--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">gspath </span>and <span class="t1-zi4r-0-x-x-109">gzpath</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>Block <span class="t1-zi4r-0-x-x-109">atoms {...}</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Defines the atoms whose coordinates make up the value of the component.
     </li>
<li class="itemize"><a id="dx1-72002"></a><span class="colorbox" id="colorbox159"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositionsCol</span></span> <!--l. 2922--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>PDB
     column containing atom flags<!--l. 2922--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">gspath </span>and <span class="t1-zi4r-0-x-x-109">gzpath</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span><span class="t1-zi4r-0-x-x-109">O</span>, <span class="t1-zi4r-0-x-x-109">B</span>, <span class="t1-zi4r-0-x-x-109">X</span>, <span class="t1-zi4r-0-x-x-109">Y</span>, or <span class="t1-zi4r-0-x-x-109">Z</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>If <span class="t1-zi4r-0-x-x-109">refPositionsFileN </span>is a PDB file that contains all the atoms in the topology, this
     option may be provided to set which PDB field is used to flag the reference coordinates for <span class="t1-zi4r-0-x-x-109">atoms</span>.
     </li>
<li class="itemize"><a id="dx1-72003"></a><span class="colorbox" id="colorbox160"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositionsFileN</span></span> <!--l. 2931--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>File
     containing the reference positions for fitting<!--l. 2931--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">gspath </span>and <span class="t1-zi4r-0-x-x-109">gzpath</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>UNIX filename <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>The path is defined by multiple <span class="t1-zi4r-0-x-x-109">refPositionsFile</span>s which are similiar to <span class="t1-zi4r-0-x-x-109">refPositionsFile</span>
     in the <span class="t1-zi4r-0-x-x-109">rmsd </span>CV. If your path consists of <!--l. 2931--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>1</mn><mn>0</mn></math>
     nodes, you can list the coordinate file (in PDB or XYZ format) from <span class="t1-zi4r-0-x-x-109">refPositionsFile1 </span>to <span class="t1-zi4r-0-x-x-109">refPositionsFile10</span>.
     </li>
<li class="itemize"><a id="dx1-72004"></a><span class="colorbox" id="colorbox161"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">useSecondClosestFrame</span></span> <!--l. 2941--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Define
     <!--l. 2941--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="bold">s</mi></mrow><mrow><mi>m</mi><mo class="MathClass-bin">−</mo><mn>1</mn></mrow></msub></math>
     as the second closest frame?<!--l. 2941--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">gspath </span>and <span class="t1-zi4r-0-x-x-109">gzpath</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">on</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>The definition assumes the second closest frame is neighbouring to the closest frame.
     This  is  not  always  true  especially  when  the  path  is  crooked.  If  this  option  is  set  to  <span class="t1-zi4r-0-x-x-109">on </span>(default),
     <!--l. 2941--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="bold">s</mi></mrow><mrow><mi>m</mi><mo class="MathClass-bin">−</mo><mn>1</mn></mrow></msub></math>
     is defined as the second closest frame. If this option is set to <span class="t1-zi4r-0-x-x-109">off</span>, <!--l. 2941--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="bold">s</mi></mrow><mrow><mi>m</mi><mo class="MathClass-bin">−</mo><mn>1</mn></mrow></msub></math>
     is defined as the left or right neighbouring frame of the closest frame.
     </li>
<li class="itemize"><a id="dx1-72005"></a><span class="colorbox" id="colorbox162"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">useThirdClosestFrame</span></span> <!--l. 2951--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Define
     <!--l. 2951--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="bold">s</mi></mrow><mrow><mi>m</mi><mo class="MathClass-bin">+</mo><mn>1</mn></mrow></msub></math>
     as the third closest frame?<!--l. 2951--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">gspath </span>and <span class="t1-zi4r-0-x-x-109">gzpath</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">off</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>The definition assumes the third closest frame is neighbouring to the closest frame. This
     is not always true especially when the path is crooked. If this option is set to <span class="t1-zi4r-0-x-x-109">on</span>, <!--l. 2951--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="bold">s</mi></mrow><mrow><mi>m</mi><mo class="MathClass-bin">+</mo><mn>1</mn></mrow></msub></math>
     is defined as the third closest frame. If this option is set to <span class="t1-zi4r-0-x-x-109">off </span>(default), <!--l. 2951--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="bold">s</mi></mrow><mrow><mi>m</mi><mo class="MathClass-bin">+</mo><mn>1</mn></mrow></msub></math>
     is defined as the left or right neighbouring frame of the closest frame.
     </li>
<li class="itemize"><a id="dx1-72006"></a><span class="colorbox" id="colorbox163"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">fittingAtoms</span></span> <!--l. 2960--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>The
     atoms that are used for alignment<!--l. 2960--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">gspath </span>and <span class="t1-zi4r-0-x-x-109">gspath</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>Group of atoms <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Before calculating <!--l. 2960--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="bold">v</mi></mrow><mrow><mn>1</mn></mrow></msub></math>,
     <!--l. 2960--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="bold">v</mi></mrow><mrow><mn>2</mn></mrow></msub></math>,
     <!--l. 2960--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="bold">v</mi></mrow><mrow><mn>3</mn></mrow></msub></math>
     and <!--l. 2960--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="bold">v</mi></mrow><mrow><mn>4</mn></mrow></msub></math>,
     the  current  frame  need  to  be  aligned  to  the  corresponding  reference  frames.  This  option  specifies
     which atoms are used to do alignment.
     </li></ul>
<div class="flushright">
<!--l. 2964--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2964--><p class="noindent"></p><div class="heading"><a id="sec:cvc_gzpath">
<h5 class="subsubsectionHead"><span class="titlemark">4.9.2    </span> <a id="x1-730004.9.2"></a><span class="t1-zi4r-0-x-x-109">gzpath</span>: distance from a path defined in atomic Cartesian coordinate space.</h5>
<!--l. 2964--><p class="noindent"></p></a> </div>  <a id="colvar|gzpath"> </a>
<!--l. 2967--><p class="noindent"><span class="ptmb8t-x-x-109">List of keywords </span>(see also <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-73001"></a><span class="colorbox" id="colorbox164"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">useZsquare</span></span> <!--l. 2977--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Compute
     <!--l. 2977--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow><mi>z</mi></mrow><mrow><mn>2</mn></mrow></msup></math>
     instead of <!--l. 2977--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math><!--l. 2977--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">gzpath</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">off</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span><!--l. 2977--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math>
     is not differentiable when it is zero. This implementation workarounds it by setting the derivative of
     <!--l. 2977--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math>
     to zero when <!--l. 2977--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi> <mo class="MathClass-rel">=</mo> <mn>0</mn></math>.
     Another workaround is set this option to <span class="t1-zi4r-0-x-x-109">on</span>, which computes <!--l. 2977--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow><mi>z</mi></mrow><mrow><mn>2</mn></mrow></msup></math>
     instead of <!--l. 2977--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math>,
     and then <!--l. 2977--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow><mi>z</mi></mrow><mrow><mn>2</mn></mrow></msup></math>
     is differentiable when it is zero.
     </li></ul>
<!--l. 2981--><p class="indent">   The usage of <span class="t1-zi4r-0-x-x-109">gzpath </span>and <span class="t1-zi4r-0-x-x-109">gspath </span>is illustrated below:
                                                                                         
                                                                                         
</p>
<div class="mdframed" id="mdframed-41">
<!--l. 2983--><p class="noindent"></p><pre>colvar {<br class="newline"/>  # Progress along the path<br class="newline"/>  name gs<br class="newline"/>  # The path is defined by 5 reference frames (from string-00.pdb to string-04.pdb)<br class="newline"/>  # Use atomic coordinate from atoms 1, 2 and 3 to compute the path<br class="newline"/>  gspath {<br class="newline"/>    atoms {atomnumbers { 1 2 3 }}<br class="newline"/>    refPositionsFile1 string-00.pdb<br class="newline"/>    refPositionsFile2 string-01.pdb<br class="newline"/>    refPositionsFile3 string-02.pdb<br class="newline"/>    refPositionsFile4 string-03.pdb<br class="newline"/>    refPositionsFile5 string-04.pdb<br class="newline"/>  }<br class="newline"/>}<br class="newline"/><span class="t1-zi4r-0-x-x-109">colvar {</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> # Distance from the path</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> name gz</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> # The path is defined by 5 reference frames (from string-00.pdb to string-04.pdb)</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> # Use atomic coordinate from atoms 1, 2 and 3 to compute the path</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> gzpath {</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> atoms {atomnumbers { 1 2 3 }}</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> refPositionsFile1 string-00.pdb</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> refPositionsFile2 string-01.pdb</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> refPositionsFile3 string-02.pdb</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> refPositionsFile4 string-03.pdb</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> refPositionsFile5 string-04.pdb</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> }</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">} </span></pre>
</div>
<div class="flushright">
<!--l. 3015--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3015--><p class="noindent"></p><div class="heading"><a id="sec:cvc_linearCombination">
<h5 class="subsubsectionHead"><span class="titlemark">4.9.3    </span> <a id="x1-740004.9.3"></a><span class="t1-zi4r-0-x-x-109">linearCombination</span>: Helper CV to define a linear combination of other CVs</h5>
<!--l. 3015--><p class="noindent"></p></a> </div>  <a id="colvar|linearCombination"> </a>
<!--l. 3018--><p class="indent">   This is a helper CV which can be defined as a linear combination of other CVs. It maybe useful
when you want to define the <span class="t1-zi4r-0-x-x-109">gspathCV</span><span class="t1-zi4r-0-x-x-109"> {...} </span>and the <span class="t1-zi4r-0-x-x-109">gzpathCV</span><span class="t1-zi4r-0-x-x-109"> {...} </span>as combinations of other
CVs.
</p>
<div class="flushright">
<!--l. 3022--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3022--><p class="noindent"></p><div class="heading"><a id="sec:cvc_customColvar">
<h5 class="subsubsectionHead"><span class="titlemark">4.9.4    </span> <a id="x1-750004.9.4"></a><span class="t1-zi4r-0-x-x-109">customColvar</span>: Helper CV to define a mathematical expression as CV from other CVs</h5>
<!--l. 3022--><p class="noindent"></p></a> </div>  <a id="colvar|customColvar"> </a>
<!--l. 3026--><p class="indent">   This is a helper CV which can be defined as a mathematical expression (see <a href="#x1-920004.17">4.17<!--tex4ht:ref: sec:colvar_custom_function --></a>) of other CVs. It maybe useful
when you want to define the <span class="t1-zi4r-0-x-x-109">gspathCV</span><span class="t1-zi4r-0-x-x-109"> {...}</span>, the <span class="t1-zi4r-0-x-x-109">gzpathCV</span><span class="t1-zi4r-0-x-x-109"> {...} </span>and <span class="t1-zi4r-0-x-x-109">NeuralNetwork</span><span class="t1-zi4r-0-x-x-109"> {...} </span>as combinations
of other CVs.
</p>
<div class="flushright">
<!--l. 3028--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3028--><p class="noindent"></p><div class="heading"><a id="sec:cvc_gspathCV">
<h5 class="subsubsectionHead"><span class="titlemark">4.9.5    </span> <a id="x1-760004.9.5"></a><span class="t1-zi4r-0-x-x-109">gspathCV</span>: progress along a path defined in CV space.</h5>
<!--l. 3028--><p class="noindent"></p></a> </div>  <a id="colvar|gspathCV"> </a>
<!--l. 3031--><p class="indent">   In the <span class="t1-zi4r-0-x-x-109">gspathCV</span><span class="t1-zi4r-0-x-x-109"> {...} </span>and the <span class="t1-zi4r-0-x-x-109">gzpathCV</span><span class="t1-zi4r-0-x-x-109"> {...} </span>block all vectors, namely
<!--l. 3031--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="bold">z</mi></math> and
<!--l. 3031--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="bold">s</mi></mrow><mrow><mi>k</mi></mrow></msub></math> are defined in CV space.
More specifically, <!--l. 3031--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="bold">z</mi> <mo class="MathClass-rel">=</mo> <mfenced close="]" open="[" separators=""><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-punc">,</mo><mo class="MathClass-rel">⋯</mo><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-punc">,</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>n</mi></mrow></msub></mrow></mfenced></math>,
where <!--l. 3031--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></math> is the
<!--l. 3031--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math>-th CV.
<!--l. 3031--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="bold">s</mi></mrow><mrow><mi>k</mi></mrow></msub> <mo class="MathClass-rel">=</mo> <mfenced close="]" open="[" separators=""><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>k</mi><mo class="MathClass-punc">,</mo><mn>1</mn></mrow></msub><mo class="MathClass-punc">,</mo><mo class="MathClass-rel">⋯</mo><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-punc">,</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>k</mi><mo class="MathClass-punc">,</mo><mi>n</mi></mrow></msub></mrow></mfenced></math>, where
<!--l. 3031--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>k</mi><mo class="MathClass-punc">,</mo><mi>i</mi></mrow></msub></math> means
the <!--l. 3031--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math>-th CV of
the <!--l. 3031--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>k</mi></math>-th
reference frame. It should be note that these two CVs requires the <span class="t1-zi4r-0-x-x-109">pathFile </span>option, which specifies a path file. Each
line in the path file contains a set of space-seperated CV value of the reference frame. The sequence of reference
frames matches the sequence of the lines.
</p><!--l. 3034--><p class="noindent"><span class="ptmb8t-x-x-109">List of keywords </span>(see also <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-76001"></a><span class="colorbox" id="colorbox165"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">useSecondClosestFrame</span></span> <!--l. 3043--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Define
     <!--l. 3043--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="bold">s</mi></mrow><mrow><mi>m</mi><mo class="MathClass-bin">−</mo><mn>1</mn></mrow></msub></math>
     as the second closest frame?<!--l. 3043--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">gspathCV </span>and <span class="t1-zi4r-0-x-x-109">gzpathCV</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">on</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>The definition assumes the second closest frame is neighbouring to the closest frame.
                                                                                         
                                                                                         
     This  is  not  always  true  especially  when  the  path  is  crooked.  If  this  option  is  set  to  <span class="t1-zi4r-0-x-x-109">on </span>(default),
     <!--l. 3043--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="bold">s</mi></mrow><mrow><mi>m</mi><mo class="MathClass-bin">−</mo><mn>1</mn></mrow></msub></math>
     is defined as the second closest frame. If this option is set to <span class="t1-zi4r-0-x-x-109">off</span>, <!--l. 3043--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="bold">s</mi></mrow><mrow><mi>m</mi><mo class="MathClass-bin">−</mo><mn>1</mn></mrow></msub></math>
     is defined as the left or right neighbouring frame of the closest frame.
     </li>
<li class="itemize"><a id="dx1-76002"></a><span class="colorbox" id="colorbox166"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">useThirdClosestFrame</span></span> <!--l. 3053--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Define
     <!--l. 3053--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="bold">s</mi></mrow><mrow><mi>m</mi><mo class="MathClass-bin">+</mo><mn>1</mn></mrow></msub></math>
     as the third closest frame?<!--l. 3053--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">gspathCV </span>and <span class="t1-zi4r-0-x-x-109">gzpathCV</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">off</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>The definition assumes the third closest frame is neighbouring to the closest frame. This
     is not always true especially when the path is crooked. If this option is set to <span class="t1-zi4r-0-x-x-109">on</span>, <!--l. 3053--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="bold">s</mi></mrow><mrow><mi>m</mi><mo class="MathClass-bin">+</mo><mn>1</mn></mrow></msub></math>
     is defined as the third closest frame. If this option is set to <span class="t1-zi4r-0-x-x-109">off </span>(default), <!--l. 3053--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="bold">s</mi></mrow><mrow><mi>m</mi><mo class="MathClass-bin">+</mo><mn>1</mn></mrow></msub></math>
     is defined as the left or right neighbouring frame of the closest frame.
     </li>
<li class="itemize"><a id="dx1-76003"></a><span class="colorbox" id="colorbox167"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">pathFile</span></span> <!--l. 3062--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>The
     file name of the path file.<!--l. 3062--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">gspathCV </span>and <span class="t1-zi4r-0-x-x-109">gzpathCV</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>UNIX filename <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Defines the nodes or images that constitutes the path in CV space. The CVs of an image
     are listed in a line of <span class="t1-zi4r-0-x-x-109">pathFile </span>using space-seperated format. Lines from top to button in <span class="t1-zi4r-0-x-x-109">pathFile</span>
     corresponds images from initial to last.
     </li></ul>
<div class="flushright">
<!--l. 3066--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3066--><p class="noindent"></p><div class="heading"><a id="sec:cvc_gzpathCV">
<h5 class="subsubsectionHead"><span class="titlemark">4.9.6    </span> <a id="x1-770004.9.6"></a><span class="t1-zi4r-0-x-x-109">gzpathCV</span>: distance from a path defined in CV space.</h5>
<!--l. 3066--><p class="noindent"></p></a> </div>  <a id="colvar|gzpathCV"> </a>
<!--l. 3069--><p class="noindent"><span class="ptmb8t-x-x-109">List of keywords </span>(see also <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-77001"></a><span class="colorbox" id="colorbox168"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">useZsquare</span></span> <!--l. 3079--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Compute
     <!--l. 3079--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow><mi>z</mi></mrow><mrow><mn>2</mn></mrow></msup></math>
     instead of <!--l. 3079--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math><!--l. 3079--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">gzpathCV</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">off</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span><!--l. 3079--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math>
     is not differentiable when it is zero. This implementation workarounds it by setting the derivative of
                                                                                         
                                                                                         
     <!--l. 3079--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math>
     to zero when <!--l. 3079--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi> <mo class="MathClass-rel">=</mo> <mn>0</mn></math>.
     Another workaround is set this option to <span class="t1-zi4r-0-x-x-109">on</span>, which computes <!--l. 3079--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow><mi>z</mi></mrow><mrow><mn>2</mn></mrow></msup></math>
     instead of <!--l. 3079--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math>,
     and then <!--l. 3079--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow><mi>z</mi></mrow><mrow><mn>2</mn></mrow></msup></math>
     is differentiable when it is zero.
     </li></ul>
<!--l. 3083--><p class="indent">   The usage of <span class="t1-zi4r-0-x-x-109">gzpathCV </span>and <span class="t1-zi4r-0-x-x-109">gspathCV </span>is illustrated below:</p>
<div class="mdframed" id="mdframed-42">
<!--l. 3085--><p class="noindent"></p><pre>colvar {<br class="newline"/>  # Progress along the path<br class="newline"/>  name gs<br class="newline"/>  # Path defined by the CV space of two dihedral angles<br class="newline"/>  gspathCV {<br class="newline"/>    pathFile ./path.txt<br class="newline"/>    dihedral {<br class="newline"/>      name 001<br class="newline"/>      group1 {atomNumbers {5}}<br class="newline"/>      group2 {atomNumbers {7}}<br class="newline"/>      group3 {atomNumbers {9}}<br class="newline"/>      group4 {atomNumbers {15}}<br class="newline"/>    }<br class="newline"/>    dihedral {<br class="newline"/>      name 002<br class="newline"/>      group1 {atomNumbers {7}}<br class="newline"/>      group2 {atomNumbers {9}}<br class="newline"/>      group3 {atomNumbers {15}}<br class="newline"/>      group4 {atomNumbers {17}}<br class="newline"/>    }<br class="newline"/>  }<br class="newline"/>}<br class="newline"/><br class="newline"/>colvar {<br class="newline"/>  # Distance from the path<br class="newline"/>  name gz<br class="newline"/>  gzpathCV {<br class="newline"/>    pathFile ./path.txt<br class="newline"/>    dihedral {<br class="newline"/>      name 001<br class="newline"/>      group1 {atomNumbers {5}}<br class="newline"/>      group2 {atomNumbers {7}}<br class="newline"/>      group3 {atomNumbers {9}}<br class="newline"/>      group4 {atomNumbers {15}}<br class="newline"/>    }<br class="newline"/>    dihedral {<br class="newline"/>      name 002<br class="newline"/>      group1 {atomNumbers {7}}<br class="newline"/>      group2 {atomNumbers {9}}<br class="newline"/>      group3 {atomNumbers {15}}<br class="newline"/>      group4 {atomNumbers {17}}<br class="newline"/>    }<br class="newline"/>  }<br class="newline"/>} </pre>
</div>
<div class="flushright">
<!--l. 3133--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3133--><p class="noindent"></p><div class="heading"><a id="sec:cvc_apathCV">
<h4 class="subsectionHead"><span class="titlemark">4.10    </span> <a id="x1-780004.10"></a>Arithmetic path collective variables</h4>
<!--l. 3133--><p class="noindent"></p></a> </div> 
<!--l. 3135--><p class="indent">   The arithmetic path collective variable in CV space uses a similar formula as the one proposed by Branduardi<span class="cite">[<a href="#XBranduardi2007">10</a>]</span> et al., except
that it computes <!--l. 3137--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>s</mi></math>
and <!--l. 3137--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math>
in CV space instead of RMSDs in Cartesian space. Moreover, this implementation allows different
coefficients for each CV components as described in <span class="cite">[<a href="#XHovan2019">11</a>]</span>. Assuming a path is composed of
<!--l. 3139--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>N</mi></math> reference frames and defined
in an <!--l. 3140--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>M</mi></math>-dimensional CV
space, then the equations of <!--l. 3140--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>s</mi></math>
and <!--l. 3141--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math> of
the path are
</p>
<table class="equation"><tr><td> <a id="x1-78001r16"></a>
<!--l. 3143--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>s</mi> <mo class="MathClass-rel">=</mo> <mfrac><mrow><mn>1</mn></mrow>
<mrow><mi>N</mi> <mo class="MathClass-bin">−</mo><mn>1</mn></mrow></mfrac> <mfrac><mrow><munderover accent="false" accentunder="false"><mrow><mo mathsize="big">∑</mo></mrow><mrow><mi>i</mi><mo class="MathClass-rel">=</mo><mn>0</mn></mrow><mrow><mi>N</mi><mo class="MathClass-bin">−</mo><mn>1</mn></mrow></munderover><mi>i</mi><mo class="qopname">exp</mo><!--nolimits--> <mfenced close=")" open="(" separators=""><mrow><mo class="MathClass-bin">−</mo><mi>λ</mi><munderover accent="false" accentunder="false"><mrow><mo mathsize="big">∑</mo></mrow><mrow><mi>j</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><mi>M</mi></mrow></munderover><msubsup><mrow><mi>c</mi></mrow><mrow><mi>j</mi></mrow><mrow><mn>2</mn></mrow></msubsup><msup><mrow> <mfenced close=")" open="(" separators=""><mrow><msub><mrow><mi>x</mi></mrow><mrow><mi>j</mi></mrow></msub><mo class="MathClass-bin">−</mo><msub><mrow><mi>x</mi></mrow><mrow><mi>i</mi><mo class="MathClass-punc">,</mo><mi>j</mi></mrow></msub></mrow></mfenced></mrow><mrow><mn>2</mn></mrow></msup></mrow></mfenced></mrow>
<mrow><munderover accent="false" accentunder="false"><mrow><mo mathsize="big">∑</mo></mrow><mrow><mi>i</mi><mo class="MathClass-rel">=</mo><mn>0</mn></mrow><mrow><mi>N</mi><mo class="MathClass-bin">−</mo><mn>1</mn></mrow></munderover><mo class="qopname">exp</mo><!--nolimits--> <mfenced close=")" open="(" separators=""><mrow><mo class="MathClass-bin">−</mo><mi>λ</mi><munderover accent="false" accentunder="false"><mrow><mo mathsize="big">∑</mo></mrow><mrow><mi>j</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><mi>M</mi></mrow></munderover><msubsup><mrow><mi>c</mi></mrow><mrow><mi>j</mi></mrow><mrow><mn>2</mn></mrow></msubsup><msup><mrow> <mfenced close=")" open="(" separators=""><mrow><msub><mrow><mi>x</mi></mrow><mrow><mi>j</mi></mrow></msub><mo class="MathClass-bin">−</mo><msub><mrow><mi>x</mi></mrow><mrow><mi>i</mi><mo class="MathClass-punc">,</mo><mi>j</mi></mrow></msub></mrow></mfenced></mrow><mrow><mn>2</mn></mrow></msup></mrow></mfenced></mrow></mfrac>
</math></td><td class="eq-no">(16)</td></tr></table>
<table class="equation"><tr><td> <a id="x1-78002r17"></a>
<!--l. 3150--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>z</mi> <mo class="MathClass-rel">=</mo> <mo class="MathClass-bin">−</mo><mfrac><mrow><mn>1</mn></mrow>
<mrow><mi>λ</mi></mrow></mfrac><mo class="qopname">ln</mo><!--nolimits--> <mfenced close=")" open="(" separators=""><mrow><munderover accent="false" accentunder="false"><mrow><mo mathsize="big">∑</mo></mrow><mrow><mi>i</mi><mo class="MathClass-rel">=</mo><mn>0</mn></mrow><mrow><mi>N</mi><mo class="MathClass-bin">−</mo><mn>1</mn></mrow></munderover><mo class="qopname">exp</mo><!--nolimits--> <mfenced close=")" open="(" separators=""><mrow><mo class="MathClass-bin">−</mo><mi>λ</mi><munderover accent="false" accentunder="false"><mrow><mo mathsize="big">∑</mo></mrow><mrow>
<mi>j</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><mi>M</mi></mrow></munderover><msubsup><mrow><mi>c</mi></mrow><mrow>
<mi>j</mi></mrow><mrow><mn>2</mn></mrow></msubsup><msup><mrow> <mfenced close=")" open="(" separators=""><mrow><msub><mrow><mi>x</mi></mrow><mrow>
<mi>j</mi></mrow></msub><mo class="MathClass-bin">−</mo><msub><mrow><mi>x</mi></mrow><mrow><mi>i</mi><mo class="MathClass-punc">,</mo><mi>j</mi></mrow></msub></mrow></mfenced></mrow><mrow><mn>2</mn></mrow></msup></mrow></mfenced></mrow></mfenced>
</math></td><td class="eq-no">(17)</td></tr></table>
<!--l. 3155--><p class="indent">   where <!--l. 3155--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>c</mi></mrow><mrow><mi>j</mi></mrow></msub></math> is the
coefficient(weight) of the <!--l. 3155--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>j</mi></math>-th
CV, <!--l. 3155--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>x</mi></mrow><mrow><mi>i</mi><mo class="MathClass-punc">,</mo><mi>j</mi></mrow></msub></math> is the value
of <!--l. 3156--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>j</mi></math>-th CV of
<!--l. 3156--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math>-th reference frame
and <!--l. 3156--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>x</mi></mrow><mrow><mi>j</mi></mrow></msub></math> is the value of
<!--l. 3156--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>j</mi></math>-th CV of current frame.
<!--l. 3157--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>λ</mi></math> is a parameter to
smooth the variation of <!--l. 3157--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>s</mi></math>
and <!--l. 3157--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math>. It should be
noted that the index <!--l. 3158--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math>
ranges from <!--l. 3158--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>0</mn></math> to
<!--l. 3158--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>N</mi> <mo class="MathClass-bin">−</mo><mn>1</mn></math>, and the definition
of <!--l. 3159--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>s</mi></math> is normalized
by <!--l. 3159--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>1</mn><mo class="MathClass-bin">∕</mo><mrow><mo class="MathClass-open">(</mo><mrow><mi>N</mi> <mo class="MathClass-bin">−</mo><mn>1</mn></mrow><mo class="MathClass-close">)</mo></mrow></math>. Consequently,
the scope of <!--l. 3159--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>s</mi></math>
is <!--l. 3160--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">[</mo><mrow><mn>0</mn> <mo class="MathClass-punc">:</mo> <mn>1</mn></mrow><mo class="MathClass-close">]</mo></mrow></math>.
</p>
<div class="flushright">
<!--l. 3162--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3162--><p class="noindent"></p><div class="heading"><a id="sec:cvc_aspathCV">
<h5 class="subsubsectionHead"><span class="titlemark">4.10.1    </span> <a id="x1-790004.10.1"></a><span class="t1-zi4r-0-x-x-109">aspathCV</span>: progress along a path defined in CV space.</h5>
<!--l. 3162--><p class="noindent"></p></a> </div> 
<!--l. 3164--><p class="indent">   This colvar component computes the <!--l. 3164--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>s</mi></math>
variable.
</p><!--l. 3166--><p class="indent"> <a id="colvar|aspathCV"> </a>
</p><!--l. 3168--><p class="noindent"><span class="ptmb8t-x-x-109">List of keywords </span>(see also <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-79001"></a><span class="colorbox" id="colorbox169"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">weights</span></span> <!--l. 3177--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Coefficients
     of the collective variables<!--l. 3177--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">aspathCV </span>and <span class="t1-zi4r-0-x-x-109">azpathCV</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>space-separated numbers in a <span class="t1-zi4r-0-x-x-109">{...} </span>block <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">{1.0 ...}</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Define the coefficients. The <!--l. 3177--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>j</mi></math>-th
     value in the <span class="t1-zi4r-0-x-x-109">{...} </span>block corresponds the <!--l. 3177--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>c</mi></mrow><mrow><mi>j</mi></mrow></msub></math>
     in the equations.
     </li>
<li class="itemize"><a id="dx1-79002"></a><span class="colorbox" id="colorbox170"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">lambda</span></span> <!--l. 3187--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Smoothness
     of the variation of <!--l. 3187--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>s</mi></math>
     and <!--l. 3187--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math><!--l. 3187--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">aspathCV </span>and <span class="t1-zi4r-0-x-x-109">azpathCV</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>decimal <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span>inverse of the mean square displacements of successive reference frames <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>The value of <!--l. 3187--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>λ</mi></math>
     in the equations.
     </li>
<li class="itemize"><a id="dx1-79003"></a><span class="colorbox" id="colorbox171"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">pathFile</span></span> <!--l. 3196--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>The
     file name of the path file.<!--l. 3196--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">aspathCV </span>and <span class="t1-zi4r-0-x-x-109">azpathCV</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>UNIX filename <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Defines the nodes or images that constitutes the path in CV space. The CVs of an image
     are listed in a line of <span class="t1-zi4r-0-x-x-109">pathFile </span>using space-seperated format. Lines from top to button in <span class="t1-zi4r-0-x-x-109">pathFile</span>
     corresponds images from initial to last.
     </li></ul>
<div class="flushright">
<!--l. 3200--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3200--><p class="noindent"></p><div class="heading"><a id="sec:cvc_azpathCV">
<h5 class="subsubsectionHead"><span class="titlemark">4.10.2    </span> <a id="x1-800004.10.2"></a><span class="t1-zi4r-0-x-x-109">azpathCV</span>: distance from a path defined in CV space.</h5>
<!--l. 3200--><p class="noindent"></p></a> </div> 
<!--l. 3202--><p class="indent">   This colvar component computes the <!--l. 3202--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math>
variable. Options are the same as in <a href="#x1-790004.10.1">4.10.1<!--tex4ht:ref: sec:cvc_aspathCV --></a>.
</p><!--l. 3204--><p class="indent">   The usage of <span class="t1-zi4r-0-x-x-109">azpathCV </span>and <span class="t1-zi4r-0-x-x-109">aspathCV </span>is illustrated below:</p>
<div class="mdframed" id="mdframed-43">
<!--l. 3206--><p class="noindent"></p><pre>colvar {<br class="newline"/>  # Progress along the path<br class="newline"/>  name as<br class="newline"/>  # Path defined by the CV space of two dihedral angles<br class="newline"/>  aspathCV {<br class="newline"/>    pathFile ./path.txt<br class="newline"/>    weights {1.0 1.0}<br class="newline"/>    lambda 0.005<br class="newline"/>    dihedral {<br class="newline"/>      name 001<br class="newline"/>      group1 {atomNumbers {5}}<br class="newline"/>      group2 {atomNumbers {7}}<br class="newline"/>      group3 {atomNumbers {9}}<br class="newline"/>      group4 {atomNumbers {15}}<br class="newline"/>    }<br class="newline"/>    dihedral {<br class="newline"/>      name 002<br class="newline"/>      group1 {atomNumbers {7}}<br class="newline"/>      group2 {atomNumbers {9}}<br class="newline"/>      group3 {atomNumbers {15}}<br class="newline"/>      group4 {atomNumbers {17}}<br class="newline"/>    }<br class="newline"/>  }<br class="newline"/>}<br class="newline"/><br class="newline"/>colvar {<br class="newline"/>  # Distance from the path<br class="newline"/>  name az<br class="newline"/>  azpathCV {<br class="newline"/>    pathFile ./path.txt<br class="newline"/>    weights {1.0 1.0}<br class="newline"/>    lambda 0.005<br class="newline"/>    dihedral {<br class="newline"/>      name 001<br class="newline"/>      group1 {atomNumbers {5}}<br class="newline"/>      group2 {atomNumbers {7}}<br class="newline"/>      group3 {atomNumbers {9}}<br class="newline"/>      group4 {atomNumbers {15}}<br class="newline"/>    }<br class="newline"/>    dihedral {<br class="newline"/>      name 002<br class="newline"/>      group1 {atomNumbers {7}}<br class="newline"/>      group2 {atomNumbers {9}}<br class="newline"/>      group3 {atomNumbers {15}}<br class="newline"/>      group4 {atomNumbers {17}}<br class="newline"/>    }<br class="newline"/>  }<br class="newline"/>} </pre>
</div>
<div class="flushright">
<!--l. 3259--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3259--><p class="noindent"></p><div class="heading"><a id="sec:apath_scripted">
<h5 class="subsubsectionHead"><span class="titlemark">4.10.3    </span> <a id="x1-810004.10.3"></a>Path collective variables in Cartesian coordinates</h5>
<!--l. 3259--><p class="noindent"></p></a> </div> 
<!--l. 3261--><p class="indent">   The path collective variables defined by Branduardi et al. <span class="cite">[<a href="#XBranduardi2007">10</a>]</span> are based on RMSDs in Cartesian coordinates.
Noting <!--l. 3263--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>d</mi></mrow><mrow><mi>i</mi></mrow></msub></math>
the RMSD between the current set of Cartesian coordinates and those of image number
<!--l. 3263--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math> of the
path:
</p>
<table class="equation"><tr><td> <a id="x1-81001r18"></a>
<!--l. 3265--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>s</mi> <mo class="MathClass-rel">=</mo> <mfrac><mrow><mn>1</mn></mrow>
<mrow><mi>N</mi> <mo class="MathClass-bin">−</mo><mn>1</mn></mrow></mfrac> <mfrac><mrow><munderover accent="false" accentunder="false"><mrow><mo mathsize="big">∑</mo></mrow><mrow><mi>i</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><mi>N</mi></mrow></munderover><mrow><mo class="MathClass-open">(</mo><mrow><mi>i</mi><mo class="MathClass-bin">−</mo><mn>1</mn></mrow><mo class="MathClass-close">)</mo></mrow><mo class="qopname">exp</mo><!--nolimits--> <mfenced close=")" open="(" separators=""><mrow><mo class="MathClass-bin">−</mo><mi>λ</mi><msubsup><mrow><mi>d</mi></mrow><mrow><mi>i</mi></mrow><mrow><mn>2</mn></mrow></msubsup></mrow></mfenced></mrow>
<mrow><munderover accent="false" accentunder="false"><mrow><mo mathsize="big">∑</mo></mrow><mrow><mi>i</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><mi>N</mi></mrow></munderover><mo class="qopname">exp</mo><!--nolimits--> <mfenced close=")" open="(" separators=""><mrow><mo class="MathClass-bin">−</mo><mi>λ</mi><msubsup><mrow><mi>d</mi></mrow><mrow><mi>i</mi></mrow><mrow><mn>2</mn></mrow></msubsup></mrow></mfenced></mrow></mfrac>
</math></td><td class="eq-no">(18)</td></tr></table>
<table class="equation"><tr><td> <a id="x1-81002r19"></a>
<!--l. 3270--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>z</mi> <mo class="MathClass-rel">=</mo> <mo class="MathClass-bin">−</mo><mfrac><mrow><mn>1</mn></mrow>
<mrow><mi>λ</mi></mrow></mfrac><mo class="qopname">ln</mo><!--nolimits--> <mfenced close=")" open="(" separators=""><mrow><munderover accent="false" accentunder="false"><mrow><mo mathsize="big">∑</mo></mrow><mrow><mi>i</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><mi>N</mi></mrow></munderover><mo class="qopname">exp</mo><!--nolimits--><mrow><mo class="MathClass-open">(</mo><mrow><mo class="MathClass-bin">−</mo><mi>λ</mi><msubsup><mrow><mi>d</mi></mrow><mrow>
<mi>i</mi></mrow><mrow><mn>2</mn></mrow></msubsup></mrow><mo class="MathClass-close">)</mo></mrow></mrow></mfenced>
</math></td><td class="eq-no">(19)</td></tr></table>
<!--l. 3274--><p class="indent">   where <!--l. 3274--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>λ</mi></math> is
the smoothing parameter.
</p><!--l. 3277--><p class="indent">   These coordinates are implemented as Tcl-scripted combinations of <span class="t1-zi4r-0-x-x-109">rmsd </span>components. The implementation is
available as file <span class="t1-zi4r-0-x-x-109">colvartools/pathCV.tcl</span>, and an example is provided in file <span class="t1-zi4r-0-x-x-109">examples/10_pathCV.namd </span>of the
Colvars public repository. It implements an optimization procedure, whereby the distance to a given image is only
calculated if its contribution to the sum is larger than a user-defined <span class="t1-zi4r-0-x-x-109">tolerance </span>parameter. All distances are
calculated every <span class="t1-zi4r-0-x-x-109">freq </span>timesteps to update the list of nearby images.
</p>
<div class="flushright">
<!--l. 3286--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3286--><p class="noindent"></p><div class="heading"><a id="sec:cvc_neuralNetwork">
<h4 class="subsectionHead"><span class="titlemark">4.11    </span> <a id="x1-820004.11"></a>Dense neural network in CV space</h4>
<!--l. 3286--><p class="noindent"></p></a> </div> 
<!--l. 3288--><p class="indent">   This colvar computes a non-linear combination of other scalar colvar components, and the non-linear
transformation is defined by a dense neural network.<span class="cite">[<a href="#XChen2022">12</a>]</span>
</p>
<hr class="figure"/><div class="figure">
<a id="x1-820011"></a>
<!--l. 3294--><p class="noindent"><img alt="Dense neural network" class="neural_network" src="dense_nn.png"/>
<br/> </p><div class="caption"><span class="id">Figure 1:  </span><span class="content">Graphical  representation  of  an  example  dense  neural  network  with  two  hidden  layers
<!--l. 3296--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>y</mi></mrow><mrow><mn>1</mn></mrow></msub></math>
and <!--l. 3296--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>y</mi></mrow><mrow><mn>2</mn></mrow></msub></math>.
Both the input layer <!--l. 3296--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>x</mi></math>
and the output layer <!--l. 3296--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math>
have two nodes. The input nodes can be any existing scalar CVs. </span></div><!--tex4ht:label?: x1-820011 -->
</div><hr class="endfigure"/>
<!--l. 3300--><p class="indent">   The output of the <!--l. 3300--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>j</mi></math>-th
node of a <!--l. 3300--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>k</mi></math>-th
layer that has <!--l. 3300--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>N</mi></mrow><mrow><mi>k</mi></mrow></msub></math>
nodes is computed by
</p>
<table class="equation"><tr><td> <a id="x1-82002r20"></a>
<!--l. 3302--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<msub><mrow><mi>y</mi></mrow><mrow><mi>k</mi><mo class="MathClass-punc">,</mo><mi>j</mi></mrow></msub> <mo class="MathClass-rel">=</mo> <msub><mrow><mi>f</mi></mrow><mrow><mi>k</mi></mrow></msub> <mfenced close=")" open="(" separators=""><mrow><munderover accent="false" accentunder="false"><mrow><mo mathsize="big">∑</mo></mrow><mrow><mi>i</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><msub><mrow><mi>N</mi></mrow><mrow><mi>k</mi><mo class="MathClass-bin">−</mo><mn>1</mn></mrow></msub></mrow></munderover><msub><mrow><mi>w</mi></mrow><mrow>
<mrow><mo class="MathClass-open">(</mo><mrow><mi>k</mi><mo class="MathClass-punc">,</mo><mi>j</mi></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-punc">,</mo><mrow><mo class="MathClass-open">(</mo><mrow><mi>k</mi><mo class="MathClass-bin">−</mo><mn>1</mn><mo class="MathClass-punc">,</mo><mi>i</mi></mrow><mo class="MathClass-close">)</mo></mrow></mrow></msub><msub><mrow><mi>y</mi></mrow><mrow><mi>k</mi><mo class="MathClass-bin">−</mo><mn>1</mn><mo class="MathClass-punc">,</mo><mi>i</mi></mrow></msub><mo class="MathClass-bin">+</mo><msub><mrow><mi>b</mi></mrow><mrow><mi>k</mi><mo class="MathClass-punc">,</mo><mi>j</mi></mrow></msub></mrow></mfenced><mo class="MathClass-punc">,</mo>
</math></td><td class="eq-no">(20)</td></tr></table>
<!--l. 3306--><p class="indent">   where <!--l. 3306--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>f</mi></mrow><mrow><mi>k</mi></mrow></msub></math> is the activation
function of the <!--l. 3306--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>k</mi></math>-th
layer, <!--l. 3306--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>w</mi></mrow><mrow><mrow><mo class="MathClass-open">(</mo><mrow><mi>k</mi><mo class="MathClass-punc">,</mo><mi>j</mi></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-punc">,</mo><mrow><mo class="MathClass-open">(</mo><mrow><mi>k</mi><mo class="MathClass-bin">−</mo><mn>1</mn><mo class="MathClass-punc">,</mo><mi>i</mi></mrow><mo class="MathClass-close">)</mo></mrow></mrow></msub></math> is the weight of
<!--l. 3306--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>j</mi></math>-th node with respect
to the <!--l. 3306--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math>-th output of
previous layer, and <!--l. 3306--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>b</mi></mrow><mrow><mi>k</mi><mo class="MathClass-punc">,</mo><mi>j</mi></mrow></msub></math>
is the bias of <!--l. 3306--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>j</mi></math>-th
node of <!--l. 3306--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>k</mi></math>-th
layer.
</p>
<div class="flushright">
<!--l. 3308--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3308--><p class="noindent"></p><div class="heading"><a id="sec:cvc_neuralNetwork">
<h5 class="subsubsectionHead"><span class="titlemark">4.11.1    </span> <a id="x1-830004.11.1"></a><span class="t1-zi4r-0-x-x-109">NeuralNetwork</span>: Dense neural network.</h5>
<!--l. 3308--><p class="noindent"></p></a> </div> 
<!--l. 3310--><p class="indent"> <a id="colvar|NeuralNetwork"> </a>
</p><!--l. 3312--><p class="noindent"><span class="ptmb8t-x-x-109">List of keywords </span>(see also <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="colvar|NeuralNetwork|output_component"> </a> <a id="dx1-83001"></a><span class="colorbox" id="colorbox172"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">output_component</span></span> <!--l. 3322--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>The
     <!--l. 3322--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>j</mi></math>-th
     node of the output or the last layer<!--l. 3322--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">NeuralNetwork</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>integer starting from 0 <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>The value of this option specifies the output node to be used as the value of this CV.
                                                                                         
                                                                                         
     </li>
<li class="itemize"><a id="colvar|NeuralNetwork|layeri_WeightsFile"> </a> <a id="dx1-83002"></a><span class="colorbox" id="colorbox173"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">layer</span><!--l. 3336--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math><span class="t1-zi4r-0-x-x-120">_WeightsFile</span></span> <!--l. 3336--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>The
     weights from layer <!--l. 3336--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi><mo class="MathClass-bin">−</mo><mn>1</mn></math>
     to layer <!--l. 3336--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math><!--l. 3336--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">NeuralNetwork</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>UNIX filename <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>The letter <!--l. 3336--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math>
     in this option needs to be replaced with the indexing number starting from 1, for example, <span class="t1-zi4r-0-x-x-109">layer1_WeightsFile</span>
     and <span class="t1-zi4r-0-x-x-109">layer2_WeightsFile</span>. The value of this option specifies a plain text file containing the weights
     from layer <!--l. 3336--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi><mo class="MathClass-bin">−</mo><mn>1</mn></math>
     to layer <!--l. 3336--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math>.
     In the file, the number at <!--l. 3336--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>k</mi></math>-th
     column and <!--l. 3336--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>l</mi></math>-th
     row represents the weight from node <!--l. 3336--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>k</mi></math>
     at layer <!--l. 3336--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi><mo class="MathClass-bin">−</mo><mn>1</mn></math>
     to node <!--l. 3336--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>l</mi></math>
     at layer <!--l. 3336--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math>.
     </li>
<li class="itemize"><a id="colvar|NeuralNetwork|layeri_BiasesFile"> </a> <a id="dx1-83003"></a><span class="colorbox" id="colorbox174"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">layer</span><!--l. 3350--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math><span class="t1-zi4r-0-x-x-120">_BiasesFile</span></span> <!--l. 3350--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>The
     biases from layer <!--l. 3350--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi><mo class="MathClass-bin">−</mo><mn>1</mn></math>
     to layer <!--l. 3350--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math><!--l. 3350--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">NeuralNetwork</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>UNIX filename <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>The letter <!--l. 3350--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math>
     in this option needs to be replaced with the indexing number starting from 1, for example, <span class="t1-zi4r-0-x-x-109">layer1_BiasesFile</span>
     and <span class="t1-zi4r-0-x-x-109">layer2_BiasesFile</span>. The value of this option specifies a plain text file containing the weights of
     layer <!--l. 3350--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math>.
     The file should have only one column, where the number at <!--l. 3350--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>l</mi></math>-th
     row represents the bias of node <!--l. 3350--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>l</mi></math>
     from layer <!--l. 3350--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi><mo class="MathClass-bin">−</mo><mn>1</mn></math>
     to layer <!--l. 3350--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math>.
     </li>
<li class="itemize"><a id="colvar|NeuralNetwork|layeri_activation"> </a> <a id="dx1-83004"></a><span class="colorbox" id="colorbox175"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">layer</span><!--l. 3364--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math><span class="t1-zi4r-0-x-x-120">_activation</span></span> <!--l. 3364--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>The
     activation function from layer <!--l. 3364--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi><mo class="MathClass-bin">−</mo><mn>1</mn></math>
     to layer <!--l. 3364--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math><!--l. 3364--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">NeuralNetwork</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span><span class="t1-zi4r-0-x-x-109">tanh</span>, <span class="t1-zi4r-0-x-x-109">sigmoid</span>, <span class="t1-zi4r-0-x-x-109">linear</span>, <span class="t1-zi4r-0-x-x-109">relu</span>, <span class="t1-zi4r-0-x-x-109">lrelu100</span>, <span class="t1-zi4r-0-x-x-109">elu </span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>The letter <!--l. 3364--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math>
     in this option needs to be replaced with the indexing number starting from 1, for example, <span class="t1-zi4r-0-x-x-109">layer1_activation</span>
     and <span class="t1-zi4r-0-x-x-109">layer2_activation</span>. The activation function from layer <!--l. 3364--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi><mo class="MathClass-bin">−</mo><mn>1</mn></math>
     to layer <!--l. 3364--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math>.
     Available choices are <span class="t1-zi4r-0-x-x-109">tanh</span>, <span class="t1-zi4r-0-x-x-109">sigmoid</span>, <span class="t1-zi4r-0-x-x-109">linear </span>(identity), <span class="t1-zi4r-0-x-x-109">relu</span>, <span class="t1-zi4r-0-x-x-109">lrelu100 </span>(a leaky rely with coefficients
     <!--l. 3364--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>1</mn><msup><mrow><mn>0</mn></mrow><mrow><mo class="MathClass-bin">−</mo><mn>2</mn></mrow></msup></math>
     and 1), and <span class="t1-zi4r-0-x-x-109">elu </span>(with coefficient 1).
     </li>
<li class="itemize"><a id="colvar|NeuralNetwork|layeri_custom_activation"> </a> <a id="dx1-83005"></a><span class="colorbox" id="colorbox176"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">layer</span><!--l. 3379--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math><span class="t1-zi4r-0-x-x-120">_custom_activation</span></span> <!--l. 3379--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>An
                                                                                         
                                                                                         
     alternative custom expression as the activation function from layer <!--l. 3379--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi><mo class="MathClass-bin">−</mo><mn>1</mn></math>
     to layer <!--l. 3379--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math><!--l. 3379--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">NeuralNetwork</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>string <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Mathematical expression to define the activation function from layer <!--l. 3379--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi><mo class="MathClass-bin">−</mo><mn>1</mn></math>
     to layer <!--l. 3379--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math>.
     The input value must be written as <span class="t1-zi4r-0-x-x-109">x</span>. For example, the ELU activation function can be expressed as
     <span class="t1-zi4r-0-x-x-109">select(step(x), alpha*(exp(x)-1), x)</span>. It should be noted that this option is mutually exclusive
     with the option <a href="#."><span class="t1-zi4r-0-x-x-109">colvar|NeuralNetwork|layeri_activation</span></a>
</li></ul>
<!--l. 3383--><p class="indent">   An example of configuration using <span class="t1-zi4r-0-x-x-109">NeuralNetwork </span>is shown below:
</p><!--l. 3388--><p class="noindent"><span class="t1-zi4r-0-x-x-109">colvar {</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> # Define a neural network with 2 layers</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> # The inputs are two torsion angles</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> # and the first node at the output layer is used as the final CV</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> name nn_output_1</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> NeuralNetwork {</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> output_component 0</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> layer1_WeightsFile dense_1_weights.txt</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> layer1_BiasesFile dense_1_biases.txt</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> layer1_activation tanh</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> layer2_WeightsFile dense_2_weights.txt</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> layer2_BiasesFile dense_2_biases.txt</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> layer2_activation tanh</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> # The component coefficient is used for normalization</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> componentCoeff 180.0</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> dihedral {</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> name 001</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> # normalization factor 1.0/180.0</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> componentCoeff 0.00555555555555555556</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> group1 {atomNumbers {5}}</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> group2 {atomNumbers {7}}</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> group3 {atomNumbers {9}}</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> group4 {atomNumbers {15}}</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> }</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> dihedral {</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> name 002</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> # normalization factor 1.0/180.0</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> componentCoeff 0.00555555555555555556</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> group1 {atomNumbers {7}}</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> group2 {atomNumbers {9}}</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> group3 {atomNumbers {15}}</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> group4 {atomNumbers {17}}</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> }</span><br class="newline"/><span class="t1-zi4r-0-x-x-109"> </span><span class="t1-zi4r-0-x-x-109"> }</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">}</span>
</p>
<div class="flushright">
<!--l. 3427--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3427--><p class="noindent"></p><div class="heading"><a id="sec:cvc_volmaps">
<h4 class="subsectionHead"><span class="titlemark">4.12    </span> <a id="x1-840004.12"></a>Volumetric map-based variables</h4>
<!--l. 3427--><p class="noindent"></p></a> </div> 
<!--l. 3429--><p class="indent">   Volumetric maps of the Cartesian coordinates, typically defined as mesh grid along the three Cartesian axes, may
be used to define collective variables. In NAMD this feature implemented as an interface between Colvars and
<a href="https://www.ks.uiuc.edu/Research/namd/2.14/ug/node43.html">GridForces</a>. Please cite <span class="cite">[<a href="#XFiorin2020">13</a>]</span> when using this implementation of collective variables based on volumetric
maps.
</p>
<div class="flushright">
<!--l. 3433--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3433--><p class="noindent"></p><div class="heading"><a id="sec:cvc_map_total">
<h5 class="subsubsectionHead"><span class="titlemark">4.12.1    </span> <a id="x1-850004.12.1"></a><span class="t1-zi4r-0-x-x-109">mapTotal</span>: total value of a volumetric map</h5>
<!--l. 3433--><p class="noindent"></p></a> </div>  <a id="colvar|mapTotal"> </a>
<!--l. 3436--><p class="indent">   Given  a  function  of  the  Cartesian  coordinates
<!--l. 3436--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ϕ</mi><mrow><mo class="MathClass-open">(</mo><mrow><mi mathvariant="bold">x</mi></mrow><mo class="MathClass-close">)</mo></mrow> <mo class="MathClass-rel">=</mo> <mi>ϕ</mi><mrow><mo class="MathClass-open">(</mo><mrow><mi>x</mi><mo class="MathClass-punc">,</mo><mi>y</mi><mo class="MathClass-punc">,</mo><mi>z</mi></mrow><mo class="MathClass-close">)</mo></mrow></math>, a <span class="t1-zi4r-0-x-x-109">mapTotal </span>collective variable
component <!--l. 3436--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>Φ</mi><mrow><mo class="MathClass-open">(</mo><mrow><mi mathvariant="bold">X</mi></mrow><mo class="MathClass-close">)</mo></mrow></math> is defined as the sum of
the values of the function <!--l. 3436--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ϕ</mi><mrow><mo class="MathClass-open">(</mo><mrow><mi mathvariant="bold">x</mi></mrow><mo class="MathClass-close">)</mo></mrow></math> evaluated
at the coordinates of each atom, <!--l. 3436--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>i</mi></mrow></msub></math>:
</p><table class="equation"><tr><td> <a id="x1-85001r21"></a>
<!--l. 3437--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>Φ</mi><mrow><mo class="MathClass-open">(</mo><mrow><mi mathvariant="bold">X</mi></mrow><mo class="MathClass-close">)</mo></mrow> <mo class="MathClass-rel">=</mo><munderover accent="false" accentunder="false"><mrow><mo mathsize="big"> ∑</mo></mrow><mrow><mi>i</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><mi>N</mi></mrow></munderover><msub><mrow><mi>w</mi></mrow><mrow>
<mi>i</mi></mrow></msub><mi>ϕ</mi><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow>
</math></td><td class="eq-no">(21)</td></tr></table>
<!--l. 3441--><p class="noindent">where <!--l. 3441--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>w</mi></mrow><mrow><mi>i</mi></mrow></msub></math>
are weights assigned to each variable (1 by default). This formulation allows, for example,
to “count" the number of atoms within a region of space by using a positive-valued function
<!--l. 3442--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ϕ</mi><mrow><mo class="MathClass-open">(</mo><mrow><mi mathvariant="bold">x</mi></mrow><mo class="MathClass-close">)</mo></mrow></math>, such as
for example the number of water molecules in a hydrophobic cavity <span class="cite">[<a href="#XFiorin2020">13</a>]</span>.<br class="newline"/>
</p><!--l. 3444--><p class="indent">   Because the volumetric map itself and the atoms affected by it are defined externally to Colvars, this component
has a very limited number of keywords.
</p><!--l. 3446--><p class="noindent"><span class="ptmb8t-x-x-109">List of keywords </span>(see also <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="colvar|mapTotal|mapName"> </a> <a id="dx1-85002"></a><span class="colorbox" id="colorbox177"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">mapName</span></span> <!--l. 3458--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Specify
     the name of the volumetric map to use as a colvar<!--l. 3458--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">mapTotal</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>string <br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>The  value  of  this  option  specifies  the  label  of  the  volumetric  map  to  use  for  this
     collective variable component. This label must identify a map already loaded in NAMD via <span class="t1-zi4r-0-x-x-109">mGridForcePotFile</span>,
     and the corresponding value of <span class="t1-zi4r-0-x-x-109">mGridForceScale </span>must be set to (0, 0, 0), so that its collective force
     is computed dynamically.
     </li>
<li class="itemize"><a id="colvar|mapTotal|mapID"> </a> <a id="dx1-85003"></a><span class="colorbox" id="colorbox178"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">mapID</span></span> <!--l. 3470--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Specify
     the numeric index of the volumetric map to use as a colvar<!--l. 3470--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">mapTotal</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>non-negative integer <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>The value of this option specifies the numeric index of the volumetric map to use for this
     collective variable component. The map referred to must be already loaded before using this option.
     This option is mutually exclusive with <a href="#;"><span class="t1-zi4r-0-x-x-109">colvar|mapTotal|mapName</span></a> if the latter is given, the numeric
     ID is obtained internally from NAMD.
     </li>
<li class="itemize"><a id="colvar|mapTotal|atoms"> </a> <a id="dx1-85004"></a><span class="colorbox" id="colorbox179"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">atoms</span></span> <!--l. 3483--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Atom
     group<!--l. 3483--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">mapTotal</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span><span class="t1-zi4r-0-x-x-109">atoms</span><span class="t1-zi4r-0-x-x-109"> {...} </span>block <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>If defined, this option allows defining the set of atoms to be aligned onto the volumetric
     map following the syntax described in section <a href="#x1-1000005">5<!--tex4ht:ref: sec:colvar_atom_groups --></a>. This also allows, for instance, to express the volumetric
     map in a rotated frame of reference (see <a href="#x1-1020005.2">5.2<!--tex4ht:ref: sec:colvar_atom_groups_ref_frame --></a>).   In NAMD, this keyword is optional; unless a frame
     of reference other than the laboratory is needed, it is computationally much more efficient to select
     atoms via <a href="https://www.ks.uiuc.edu/Research/namd/2.14/ug/node43.html">GridForces</a> keywords.
     </li>
<li class="itemize"><a id="colvar|mapTotal|atomWeights"> </a> <a id="dx1-85005"></a><span class="colorbox" id="colorbox180"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">atoms</span></span> <!--l. 3495--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Weights
     to assign to each atom<!--l. 3495--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">mapTotal</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>list of space-separated decimals <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span>all weights equal to 1 <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>If defined, the value of this option assigns weights <!--l. 3495--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>w</mi></mrow><mrow><mi>i</mi></mrow></msub></math>
     to the individual atoms in a <span class="t1-zi4r-0-x-x-109">mapTotal </span>variable. This option requires defining <span class="t1-zi4r-0-x-x-109">atoms </span>explicitly, and the
     number of weights must match the number of atoms.
                                                                                         
                                                                                         
     </li></ul>
<!--l. 3533--><p class="noindent"><span class="ptmb8t-x-x-109">Example: biasing the number of molecules inside a cavity using a volumetric map.</span>
</p><!--l. 3533--><p class="indent">   Firstly, a volumetric map that has a value of 1 inside the cavity and 0 outside should be prepared. A reasonable
starting point may be obtained, for example, with VMD: using an existing trajectory where the cavity is occupied by
solvent and a spatial selection that identifies all the molecules within the cavity, <span class="t1-zi4r-0-x-x-109">volmap occupancy -allframes</span>
<span class="t1-zi4r-0-x-x-109">-combine max </span>computes the occupancy map as a step function (values 1 or 0), and <span class="t1-zi4r-0-x-x-109">volutil -smooth </span><span class="t1-zi4r-0-x-x-109">… </span>makes it a
continuous map, suitable for use as a MD simulation bias. A PDB file defining the selection (for example, where all
water oxygens and ions have an occupancy of 1 and other atoms 0) is also prepared using VMD. Both the
map file and the atom selection file are then loaded via the <span class="t1-zi4r-0-x-x-109">mGridForcePotFile </span>and related NAMD
commands:</p>
<div class="mdframed" id="mdframed-44">
<!--l. 3533--><p class="noindent"></p><pre>mGridForce yes<br class="newline"/>mGridForcePotFile Cavity cavity.dx   # OpenDX map file<br class="newline"/>mGridForceFile Cavity water-sel.pdb  # PDB file used for atom selection<br class="newline"/>mGridForceCol Cavity O               # Use the occupancy column of the PDB file<br class="newline"/>mGridForceChargeCol Cavity B         # Use beta as “charge" (default: electric charge)<br class="newline"/>mGridForceScale Cavity 0.0 0.0 0.0   # Do not use GridForces for this map<br class="newline"/></pre>
</div>
<!--l. 3533--><p class="indent">   The value of <span class="t1-zi4r-0-x-x-109">mGridForceScale </span>is particularly important, because it determines the GridForces force constant
for the “Cavity" map. A non-zero value enables a conventional GridForces calculation, where the force constant
remains fixed within each <span class="t1-zi4r-0-x-x-109">run </span>command and the forces on the atoms depend only on their positions in space.
However, setting <span class="t1-zi4r-0-x-x-109">mGridForceScale </span>to zero signals to NAMD that the force acting through the volumetric map may
be computed dynamically, as part of a collective-variable biasing scheme. To do so, the map labeled “Cavity" needs
to be referred to in the Colvars configuration:<br class="newline"/></p>
<div class="mdframed" id="mdframed-45">
<!--l. 3533--><p class="noindent"></p><pre>cv config "<br class="newline"/>colvar {<br class="newline"/>  name n_waters<br class="newline"/>  mapTotal {<br class="newline"/>    mapName Cavity # Same label as the GridForce map<br class="newline"/>  }<br class="newline"/>}" </pre>
</div>
<!--l. 3533--><p class="indent">   The variable “<span class="t1-zi4r-0-x-x-109">n_waters</span>" may then be used with any of the enhanced sampling methods available (<a href="#x1-1070006">6<!--tex4ht:ref: sec:colvarbias --></a>): new
forces applied to it at each simulation step will be transmitted to the corresponding atoms within the same
step.
</p>
<div class="flushright">
<!--l. 3536--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3536--><p class="noindent"></p><div class="heading"><a id="sec:cvc_multimap">
<h5 class="subsubsectionHead"><span class="titlemark">4.12.2    </span> <a id="x1-860004.12.2"></a>Multiple volumetric maps collective variables</h5>
<!--l. 3536--><p class="noindent"></p></a> </div> 
<!--l. 3538--><p class="indent">   To study processes that involve changes in shape of a macromolecular aggregate (for example, deformations of
lipid membranes) it is useful to define collective variables based on more than one volumetric map at a time,
measuring the relative similarity with each map while still achieving correct thermodynamic sampling of each
state.
</p><!--l. 3540--><p class="indent">   This is achieved by combining multiple <span class="t1-zi4r-0-x-x-109">mapTotal </span>components, each based on a differently-shaped volumetric map, into a single
collective variable <!--l. 3540--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>.
To track transitions between states, the contribution of each map to
<!--l. 3541--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math> should
be discriminated from the others, for example by assigning to it a different weight. The “Multi-Map"
progress variable <span class="cite">[<a href="#XFiorin2020">13</a>]</span> uses a weight sum of these components, using linearly-increasing weights: </p><table class="equation"><tr><td>
<a id="x1-86001r22"></a>
<!--l. 3543--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>ξ</mi><mrow><mo class="MathClass-open">(</mo><mrow><mi mathvariant="bold">X</mi></mrow><mo class="MathClass-close">)</mo></mrow> <mo class="MathClass-rel">=</mo><munderover accent="false" accentunder="false"><mrow><mo mathsize="big"> ∑</mo></mrow><mrow><mi>k</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><mi>K</mi></mrow></munderover><msub><mrow><mi>Φ</mi></mrow><mrow>
<mi>k</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi mathvariant="bold">X</mi></mrow><mo class="MathClass-close">)</mo></mrow> <mo class="MathClass-rel">=</mo><munderover accent="false" accentunder="false"><mrow><mo mathsize="big"> ∑</mo></mrow><mrow><mi>k</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><mi>K</mi></mrow></munderover><mi>k</mi><munderover accent="false" accentunder="false"><mrow><mo mathsize="big">∑</mo></mrow><mrow>
<mi>i</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><mi>N</mi></mrow></munderover><msub><mrow><mi>ϕ</mi></mrow><mrow>
<mi>k</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow>
</math></td><td class="eq-no">(22)</td></tr></table>
<!--l. 3547--><p class="noindent">where <!--l. 3547--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>K</mi></math> is the number of
maps employed and each <!--l. 3547--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>Φ</mi></mrow><mrow><mi>k</mi></mrow></msub></math>
is a <span class="t1-zi4r-0-x-x-109">mapTotal </span>component.
</p><!--l. 3549--><p class="indent">   Here is a link to the <span class="ptmb8t-x-x-109">Multi-Map tutorial </span>page: <a class="url" href="https://colvars.github.io/multi-map/multi-map.html"><span class="t1-zi4r-0-x-x-109">https://colvars.github.io/multi-map/multi-map.html</span></a><br class="newline"/>An example configuration for illustration purposes is also included below.
</p><!--l. 3552--><p class="noindent"><span class="ptmb8t-x-x-109">Example: transitions between macromolecular shapes using volumetric maps.</span><br class="newline"/>A series of map files, each representing a different shape, is loaded into NAMD:<br class="newline"/></p>
<div class="mdframed" id="mdframed-46">
<!--l. 3554--><p class="noindent"></p><pre>mGridForce yes<br class="newline"/>for { set k 1 } { $k &lt;= $K } { incr k } {<br class="newline"/>  mGridForcePotFile Shape_$k map_$k.dx   # Density map of the k-th state<br class="newline"/>  mGridForceFile Shape_$k atoms.pdb   # PDB file used for atom selection<br class="newline"/>  mGridForceCol Shape_$k O   # Use the occupancy column of the PDB file atoms.pdb<br class="newline"/>  mGridForceChargeCol Shape_$k B   # Use beta as “charge" (default: electric charge)<br class="newline"/>  mGridForceScale Shape_$k 0.0 0.0 0.0  # Do not use GridForces for this map<br class="newline"/>} </pre>
</div>
<!--l. 3565--><p class="indent">   The GridForces maps thus loaded are then used to define the Multi-Map collective variable, with coefficients
<!--l. 3565--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>k</mi></mrow></msub> <mo class="MathClass-rel">=</mo> <mi>k</mi></math>
<span class="cite">[<a href="#XFiorin2020">13</a>]</span>:<br class="newline"/>
</p>
<div class="mdframed" id="mdframed-47">
<!--l. 3566--><p class="noindent"></p><pre># Collect the definition of all components into one string<br class="newline"/>set components "<br class="newline"/>for { set k 1 } { $k &lt;= $K } { incr k } {<br class="newline"/>  set components "${components}<br class="newline"/>  mapTotal {<br class="newline"/>    mapName Shape_$k<br class="newline"/>    componentCoeff $k<br class="newline"/>  }<br class="newline"/>}<br class="newline"/>"<br class="newline"/># Use this string to define the variable<br class="newline"/>cv config "<br class="newline"/>colvar {<br class="newline"/>  name shapes<br class="newline"/>  ${components}<br class="newline"/>}" </pre>
</div>
<!--l. 3585--><p class="indent">   The above example illustrates a use case where a weighted sum (i.e. a linear combination) is used to define a
single variable from multiple components. Depending on the problem under study, non-linear functions may be
more appropriate. These may be defined a custom functions if implemented (see <a href="#x1-920004.17">4.17<!--tex4ht:ref: sec:colvar_custom_function --></a>), or scripted functions (see
<a href="#x1-930004.18">4.18<!--tex4ht:ref: sec:colvar_scripted --></a>).
</p>
<div class="flushright">
<!--l. 3592--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3592--><p class="noindent"></p><div class="heading"><a id="sec:cvc_common">
<h4 class="subsectionHead"><span class="titlemark">4.13    </span> <a id="x1-870004.13"></a>Shared keywords for all components</h4>
<!--l. 3592--><p class="noindent"></p></a> </div> 
<!--l. 3594--><p class="indent">   The following options can be used for any of the above colvar components in order to obtain a polynomial
combination or any user-supplied function provided by <a href="#sec:cvc_superp"><span class="t1-zi4r-0-x-x-109">scriptedFunction</span></a>. </p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-87001"></a><span class="colorbox" id="colorbox181"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">name</span></span> <!--l. 3606--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Name
     of this component<!--l. 3606--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>any component <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>string <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span>type of component + numeric id <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>The name is an unique case-sensitive string which allows the Colvars module to identify
     this component. This is useful, for example, when combining multiple components via a <span class="t1-zi4r-0-x-x-109">scriptedFunction</span>.
     It also defines the variable name representing the component's value in a <a href="#colvar|customFunction"><span class="t1-zi4r-0-x-x-109">customFunction</span></a> expression.
                                                                                         
                                                                                         
     </li>
<li class="itemize"><a id="dx1-87002"></a><span class="colorbox" id="colorbox182"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">scalable</span></span> <!--l. 3618--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Attempt
     to calculate this component in parallel?<!--l. 3618--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>any component <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">on</span>, if available <br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>If  set  to  <span class="t1-zi4r-0-x-x-109">on </span>(default),  the  Colvars  module  will  attempt  to  calculate  this  component
     in parallel to reduce overhead. Whether this option is available depends on the type of component:
     currently supported are <span class="t1-zi4r-0-x-x-109">distance</span>, <span class="t1-zi4r-0-x-x-109">distanceZ</span>, <span class="t1-zi4r-0-x-x-109">distanceXY</span>, <span class="t1-zi4r-0-x-x-109">distanceVec</span>, <span class="t1-zi4r-0-x-x-109">distanceDir</span>, <span class="t1-zi4r-0-x-x-109">angle </span>and
     <span class="t1-zi4r-0-x-x-109">dihedral</span>. This flag influences computational cost, but does not affect numerical results: therefore, it
     should only be turned off for debugging or testing purposes.  </li></ul>
<div class="flushright">
<!--l. 3622--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3622--><p class="noindent"></p><div class="heading"><a id="sec:cvc_periodic">
<h4 class="subsectionHead"><span class="titlemark">4.14    </span> <a id="x1-880004.14"></a>Periodic components</h4>
<!--l. 3622--><p class="noindent"></p></a> </div>  The following components returns real numbers that lie in a periodic interval: 
     <ul class="itemize1">
<li class="itemize"><span class="t1-zi4r-0-x-x-109">dihedral</span>: torsional angle between four groups;
     </li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">spinAngle</span>:  angle  of  rotation  around  a  predefined  axis  in  the  best-fit  from  a  set  of  reference
     coordinates.</li></ul>
<!--l. 3630--><p class="noindent">In certain conditions, <span class="t1-zi4r-0-x-x-109">distanceZ </span>can also be periodic, namely when periodic boundary conditions (PBCs) are defined in
the simulation and <span class="t1-zi4r-0-x-x-109">distanceZ</span>'s axis is parallel to a unit cell vector.
</p><!--l. 3634--><p class="indent">   In addition, a custom or scripted scalar colvar may be periodic depending on its user-defined expression. It will
only be treated as such by the Colvars module if the period is specified using the <span class="t1-zi4r-0-x-x-109">period </span>keyword, while
<span class="t1-zi4r-0-x-x-109">wrapAround </span>is optional.
</p><!--l. 3639--><p class="indent">   The following keywords can be used within periodic components, or within custom variables (<a href="#x1-920004.17">4.17<!--tex4ht:ref: sec:colvar_custom_function --></a>), or wthin
scripted variables <a href="#x1-930004.18">4.18<!--tex4ht:ref: sec:colvar_scripted --></a>).
</p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-88001"></a><span class="colorbox" id="colorbox183"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">period</span></span> <!--l. 3653--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Period
     of the component<!--l. 3653--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">distanceZ</span>, custom colvars <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive decimal <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span>0.0 <br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>Setting  this  number  enables  the  treatment  of  <span class="t1-zi4r-0-x-x-109">distanceZ </span>as  a  periodic  component:
     by default, <span class="t1-zi4r-0-x-x-109">distanceZ </span>is not considered periodic. The keyword is supported, but irrelevant within
     <span class="t1-zi4r-0-x-x-109">dihedral </span>or <span class="t1-zi4r-0-x-x-109">spinAngle</span>, because their period is always 360 degrees.
                                                                                         
                                                                                         
     </li>
<li class="itemize"><a id="dx1-88002"></a><span class="colorbox" id="colorbox184"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">wrapAround</span></span> <!--l. 3664--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Center
     of the wrapping interval for periodic variables<!--l. 3664--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">distanceZ</span>, <span class="t1-zi4r-0-x-x-109">dihedral</span>, <span class="t1-zi4r-0-x-x-109">spinAngle</span>, custom colvars <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>decimal <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span>0.0 <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>By default, values of the periodic components are centered around zero, ranging from
     <!--l. 3664--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-bin">−</mo><mi>P</mi><mo class="MathClass-bin">∕</mo><mn>2</mn></math>
     to <!--l. 3664--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>P</mi><mo class="MathClass-bin">∕</mo><mn>2</mn></math>,
     where <!--l. 3664--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>P</mi></math>
     is  the  period.  Setting  this  number  centers  the  interval  around  this  value.  This  can  be  useful  for
     convenience of output, or to set the walls for a <span class="t1-zi4r-0-x-x-109">harmonicWalls </span>in an order that would not otherwise
     be allowed. </li></ul>
<!--l. 3667--><p class="indent">   Internally, all differences between two values of a periodic colvar follow the minimum image convention: they
are calculated based on the two periodic images that are closest to each other.
</p><!--l. 3671--><p class="indent"> <span class="ptmri8t-x-x-109">Note: linear or polynomial combinations of periodic components (see </span><a href="#x1-910004.16"><span class="ptmri8t-x-x-109">4.16</span><!--tex4ht:ref: sec:cvc_superp --></a><span class="ptmri8t-x-x-109">) may become meaningless when</span>
<span class="ptmri8t-x-x-109">components cross the periodic boundary. Use such combinations carefully: estimate the range of possible values of</span>
<span class="ptmri8t-x-x-109">each component in a given simulation, and make use of  </span><span class="t1-zi4r-0-x-x-109">wrapAround </span><span class="ptmri8t-x-x-109">to limit this problem whenever</span>
<span class="ptmri8t-x-x-109">possible.</span>
</p>
<div class="flushright">
<!--l. 3674--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3674--><p class="noindent"></p><div class="heading"><a id="sec:cvc_non_scalar">
<h4 class="subsectionHead"><span class="titlemark">4.15    </span> <a id="x1-890004.15"></a>Non-scalar components</h4>
<!--l. 3674--><p class="noindent"></p></a> </div> 
<!--l. 3676--><p class="indent">   When one of the following components are used, the defined colvar returns a value that is not a scalar number:
</p>
<ul class="itemize1">
<li class="itemize"><span class="t1-zi4r-0-x-x-109">distanceVec</span>: 3-dimensional vector of the distance between two groups;
     </li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">distanceDir</span>: 3-dimensional unit vector of the distance between two groups;
     </li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">orientation</span>: 4-dimensional unit quaternion representing the best-fit rotation from a set of reference
     coordinates.</li></ul>
<!--l. 3685--><p class="noindent">The distance between two 3-dimensional unit vectors is computed as the angle between them. The distance between two
quaternions is computed as the angle between the two 4-dimensional unit vectors: because the orientation represented by
<!--l. 3688--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="sans-serif">q</mi></math> is the same as the
one represented by <!--l. 3689--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-bin">−</mo><mi mathvariant="sans-serif">q</mi></math>,
distances between two quaternions are computed considering the closest of the two symmetric images.
</p><!--l. 3692--><p class="indent">   Non-scalar components carry the following restrictions: </p>
<ul class="itemize1">
<li class="itemize">Calculation of total forces (<span class="t1-zi4r-0-x-x-109">outputTotalForce </span>option) is currently not implemented.
     </li>
<li class="itemize">Each colvar can only contain one non-scalar component.
     </li>
<li class="itemize">Binning  on  a  grid  (<span class="t1-zi4r-0-x-x-109">abf</span>,  <span class="t1-zi4r-0-x-x-109">histogram </span>and  <span class="t1-zi4r-0-x-x-109">metadynamics </span>with  <span class="t1-zi4r-0-x-x-109">useGrids </span>enabled)  is  currently  not
     implemented for colvars based on such components.</li></ul>
<!--l. 3705--><p class="indent"> <span class="ptmri8t-x-x-109">Note: while these restrictions apply to individual colvars based on non-scalar components, no limit is set to the</span>
<span class="ptmri8t-x-x-109">number of scalar colvars. To compute multi-dimensional histograms and PMFs, use sets of scalar colvars of</span>
<span class="ptmri8t-x-x-109">arbitrary size.</span>
</p>
<div class="flushright">
<!--l. 3708--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3708--><p class="noindent"></p><div class="heading"><a id="sec:cvc_sys_forces">
<h5 class="subsubsectionHead"><span class="titlemark">4.15.1    </span> <a id="x1-900004.15.1"></a>Calculating total forces</h5>
<!--l. 3708--><p class="noindent"></p></a> </div>  In addition to the restrictions due to the type of value computed (scalar or non-scalar), a final restriction can
arise when calculating total force (<span class="t1-zi4r-0-x-x-109">outputTotalForce </span>option or application of a <span class="t1-zi4r-0-x-x-109">abf </span>bias). total forces are available
currently only for the following components: <span class="t1-zi4r-0-x-x-109">distance</span>, <span class="t1-zi4r-0-x-x-109">distanceZ</span>, <span class="t1-zi4r-0-x-x-109">distanceXY</span>, <span class="t1-zi4r-0-x-x-109">angle</span>, <span class="t1-zi4r-0-x-x-109">dihedral</span>, <span class="t1-zi4r-0-x-x-109">rmsd</span>,
<span class="t1-zi4r-0-x-x-109">eigenvector </span>and <span class="t1-zi4r-0-x-x-109">gyration</span>.

                                                                                         <div class="flushright">
<!--l. 3719--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3719--><p class="noindent"></p><div class="heading"><a id="sec:cvc_superp">
<h4 class="subsectionHead"><span class="titlemark">4.16    </span> <a id="x1-910004.16"></a>Linear and polynomial combinations of components</h4>
<!--l. 3719--><p class="noindent"></p></a> </div> 
<!--l. 3721--><p class="indent">   To  extend  the  set  of  possible  definitions  of  colvars
<!--l. 3721--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi><mrow><mo class="MathClass-open">(</mo><mrow><mi mathvariant="bold">r</mi></mrow><mo class="MathClass-close">)</mo></mrow></math>, multiple
components <!--l. 3722--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>q</mi></mrow><mrow><mi>i</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi mathvariant="bold">r</mi></mrow><mo class="MathClass-close">)</mo></mrow></math>
can be summed with the formula: </p><table class="equation"><tr><td> <a id="x1-91001r23"></a>
<!--l. 3723--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>ξ</mi><mrow><mo class="MathClass-open">(</mo><mrow><mi mathvariant="bold">r</mi></mrow><mo class="MathClass-close">)</mo></mrow> <mo class="MathClass-rel">=</mo><munder class="msub"><mrow><mo mathsize="big"> ∑</mo></mrow><mrow><mi>i</mi></mrow></munder><msub><mrow><mi>c</mi></mrow><mrow><mi>i</mi></mrow></msub><msup><mrow><mrow><mo class="MathClass-open">[</mo><mrow><msub><mrow><mi>q</mi></mrow><mrow><mi>i</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi mathvariant="bold">r</mi></mrow><mo class="MathClass-close">)</mo></mrow></mrow><mo class="MathClass-close">]</mo></mrow></mrow><mrow><msub><mrow><mi>n</mi></mrow><mrow><mi>i</mi></mrow></msub>
</mrow></msup>
</math></td><td class="eq-no">(23)</td></tr></table>
<!--l. 3727--><p class="indent">   where each component appears with a unique coefficient
<!--l. 3727--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>c</mi></mrow><mrow><mi>i</mi></mrow></msub></math> (1.0 by default) the
positive integer exponent <!--l. 3728--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>n</mi></mrow><mrow><mi>i</mi></mrow></msub></math>
(1 by default).
</p><!--l. 3730--><p class="indent">   Any set of components can be combined within a colvar, provided that they return the same type of values
(scalar, unit vector, vector, or quaternion). By default, the colvar is the sum of its components. Linear or polynomial
combinations (following equation (<a href="#x1-91001r23">23<!--tex4ht:ref: eq:colvar_combination --></a>)) can be obtained by setting the following parameters, which are common to
all components: </p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-91002"></a><span class="colorbox" id="colorbox185"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">componentCoeff</span></span> <!--l. 3746--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Coefficient
     of this component in the colvar<!--l. 3746--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>any component <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>decimal <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">1.0</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>Defines the coefficient by which this component is multiplied (after being raised to
     <span class="t1-zi4r-0-x-x-109">componentExp</span>) before being added to the sum.
     </li>
<li class="itemize"><a id="dx1-91003"></a><span class="colorbox" id="colorbox186"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">componentExp</span></span> <!--l. 3758--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Exponent
     of this component in the colvar<!--l. 3758--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>any component <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>integer <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">1</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Defines the power at which the value of this component is raised before being added to
     the sum. When this exponent is different than 1 (non-linear sum), total forces and the Jacobian force
     are not available, making the colvar unsuitable for ABF calculations. </li></ul>
<!--l. 3761--><p class="indent"> <span class="ptmb8t-x-x-109">Example: </span>To define the <span class="ptmri8t-x-x-109">average </span>of a colvar across different parts of the system, simply define within the same colvar block
a series of components of the same type (applied to different atom groups), and assign to each component a <span class="t1-zi4r-0-x-x-109">componentCoeff</span>
of <!--l. 3765--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>1</mn><mo class="MathClass-bin">∕</mo><mi>N</mi></math>.
</p>
<div class="flushright">
<!--l. 3882--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3882--><p class="noindent"></p><div class="heading"><a id="sec:colvar_custom_function">
<h4 class="subsectionHead"><span class="titlemark">4.17    </span> <a id="x1-920004.17"></a>Custom functions</h4>
<!--l. 3882--><p class="noindent"></p></a> </div> 
<!--l. 3882--><p class="indent">   Collective variables may be defined by specifying a custom function as an analytical expression.
The expression is parsed by Lepton, the lightweight expression parser written by Peter Eastman
(<a class="url" href="https://simtk.org/projects/lepton"><span class="t1-zi4r-0-x-x-109">https://simtk.org/projects/lepton</span></a>). Lepton produces efficient evaluation routines for the function and its
derivatives.
</p>
<ul class="itemize1">
<li class="itemize"><a id="colvar|customFunction"> </a> <a id="dx1-92001"></a><span class="colorbox" id="colorbox187"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">customFunction</span></span> <!--l. 3882--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Compute
     colvar as a custom function of its components<!--l. 3882--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">colvar</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>string <br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>Mathematical expression to define the colvar as a closed-form function of its colvar
     components.  See  below  for  the  detailed  syntax  of  Lepton  expressions.  Multiple  mentions  of  this
     keyword can be used to define a vector variable (as in the example below).
     </li>
<li class="itemize"><a id="dx1-92002"></a><span class="colorbox" id="colorbox188"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">customFunctionType</span></span> <!--l. 3882--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Type
     of value returned by the scripted colvar<!--l. 3882--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">colvar</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>string <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">scalar</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>With this flag, the user may specify whether the colvar is a scalar or one of the following
     vector types: <span class="t1-zi4r-0-x-x-109">vector3 </span>(a 3D vector), <span class="t1-zi4r-0-x-x-109">unit_vector3 </span>(a normalized 3D vector), or <span class="t1-zi4r-0-x-x-109">unit_quaternion</span>
     (a normalized quaternion), or <span class="t1-zi4r-0-x-x-109">vector</span>. Note that the scalar and vector cases are not necessary, as they
     are detected automatically. </li></ul>
<!--l. 3882--><p class="indent">   The expression may use the collective variable components as variables, referred to by their user-defined <span class="t1-zi4r-0-x-x-109">name</span>.
Scalar elements of vector components may be accessed by appending a 1-based index to their <span class="t1-zi4r-0-x-x-109">name</span>, as in the
example below. When implementing generic functions of Cartesian coordinates rather than functions of existing
components, the <span class="t1-zi4r-0-x-x-109">cartesian </span>component may be particularly useful. A scalar-valued custom variable may be
manually defined as periodic by providing the keyword <span class="t1-zi4r-0-x-x-109">period</span>, and the optional keyword <span class="t1-zi4r-0-x-x-109">wrapAround</span>, with the
same meaning as in periodic components (see <a href="#x1-880004.14">4.14<!--tex4ht:ref: sec:cvc_periodic --></a> for details). A vector variable may be defined by specifying the
<span class="t1-zi4r-0-x-x-109">customFunction </span>parameter several times: each expression defines one scalar element of the vector colvar, as in this
example:</p>
<div class="mdframed" id="mdframed-48">
<!--l. 3882--><p class="noindent"></p><pre>colvar {<br class="newline"/>  name custom<br class="newline"/><br class="newline"/>  # A 2-dimensional vector function of a scalar x and a 3-vector r<br class="newline"/>  customFunction cos(x) * (r1 + r2 + r3)<br class="newline"/>  customFunction sqrt(r1 * r2)<br class="newline"/><br class="newline"/>  distance {<br class="newline"/>    name x<br class="newline"/>    group1 { atomNumbers 1 }<br class="newline"/>    group2 { atomNumbers 50 }<br class="newline"/>  }<br class="newline"/>  distanceVec {<br class="newline"/>    name r<br class="newline"/>    group1 { atomNumbers 10 11 12 }<br class="newline"/>    group2 { atomNumbers 20 21 22 }<br class="newline"/>  }<br class="newline"/>} </pre>
</div>
<!--l. 3882--><p class="indent">   Numeric constants may be given in either decimal or exponential form (e.g. 3.12e-2). An expression may be
followed by definitions for intermediate values that appear in the expression, separated by semicolons. For example,
the expression:<br class="newline"/><span class="t1-zi4r-0-x-x-109">a</span><span class="t1-zi4r-0-x-x-109">^2 + a*b + b</span><span class="t1-zi4r-0-x-x-109">^2; a = a1 + a2; b = b1 + b2</span><br class="newline"/>is exactly equivalent to:<br class="newline"/><span class="t1-zi4r-0-x-x-109">(a1 + a2)</span><span class="t1-zi4r-0-x-x-109">^2 + (a1 + a2) * (b1 + b2) + (b1 + b2)</span><span class="t1-zi4r-0-x-x-109">^2</span>.<br class="newline"/>The definition of an intermediate value may itself involve other intermediate values. All uses of a value must appear
<span class="ptmri8t-x-x-109">before </span>that value's definition.
</p><!--l. 3882--><p class="indent">   Lepton supports the usual arithmetic operators +, -, *, /, and ^ (power), as well as the following
functions:
</p>
<div class="center">
<!--l. 3882--><p class="noindent">
</p>
<div class="tabular"> <table cellpadding="0" cellspacing="0" class="tabular" id="TBL-5" rules="groups"><colgroup id="TBL-5-1g"><col id="TBL-5-1"/></colgroup><colgroup id="TBL-5-2g"><col id="TBL-5-2"/></colgroup><tr class="hline"><td><hr/></td><td><hr/></td></tr><tr id="TBL-5-1-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-1-1" style="text-align:left; white-space:nowrap;"> sqrt    </td><td class="td11" id="TBL-5-1-2" style="text-align:left; white-space:nowrap;"> Square root                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         </td>
</tr><tr id="TBL-5-2-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-2-1" style="text-align:left; white-space:nowrap;"> exp      </td><td class="td11" id="TBL-5-2-2" style="text-align:left; white-space:nowrap;"> Exponential                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         </td>
</tr><tr id="TBL-5-3-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-3-1" style="text-align:left; white-space:nowrap;"> log      </td><td class="td11" id="TBL-5-3-2" style="text-align:left; white-space:nowrap;"> Natural logarithm                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               </td>
</tr><tr id="TBL-5-4-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-4-1" style="text-align:left; white-space:nowrap;"> erf       </td><td class="td11" id="TBL-5-4-2" style="text-align:left; white-space:nowrap;"> Error function                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     </td></tr><tr id="TBL-5-5-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-5-1" style="text-align:left; white-space:nowrap;"> erfc </td><td class="td11" id="TBL-5-5-2" style="text-align:left; white-space:nowrap;"> Complementary error function</td>
</tr><tr class="hline"><td><hr/></td><td><hr/></td></tr><tr id="TBL-5-6-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-6-1" style="text-align:left; white-space:nowrap;"> sin       </td><td class="td11" id="TBL-5-6-2" style="text-align:left; white-space:nowrap;"> Sine (angle in radians)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        </td>
</tr><tr id="TBL-5-7-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-7-1" style="text-align:left; white-space:nowrap;"> cos      </td><td class="td11" id="TBL-5-7-2" style="text-align:left; white-space:nowrap;"> Cosine (angle in radians)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    </td>
</tr><tr id="TBL-5-8-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-8-1" style="text-align:left; white-space:nowrap;"> sec      </td><td class="td11" id="TBL-5-8-2" style="text-align:left; white-space:nowrap;"> Secant (angle in radians)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    </td>
</tr><tr id="TBL-5-9-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-9-1" style="text-align:left; white-space:nowrap;"> csc      </td><td class="td11" id="TBL-5-9-2" style="text-align:left; white-space:nowrap;"> Cosecant (angle in radians)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                </td>
</tr><tr id="TBL-5-10-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-10-1" style="text-align:left; white-space:nowrap;"> tan      </td><td class="td11" id="TBL-5-10-2" style="text-align:left; white-space:nowrap;"> Tangent (angle in radians)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  </td>
</tr><tr id="TBL-5-11-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-11-1" style="text-align:left; white-space:nowrap;"> cot      </td><td class="td11" id="TBL-5-11-2" style="text-align:left; white-space:nowrap;"> Cotangent (angle in radians)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               </td>
</tr><tr id="TBL-5-12-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-12-1" style="text-align:left; white-space:nowrap;"> asin     </td><td class="td11" id="TBL-5-12-2" style="text-align:left; white-space:nowrap;"> Inverse sine (in radians)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      </td>
</tr><tr id="TBL-5-13-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-13-1" style="text-align:left; white-space:nowrap;"> acos    </td><td class="td11" id="TBL-5-13-2" style="text-align:left; white-space:nowrap;"> Inverse cosine (in radians)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  </td>
</tr><tr id="TBL-5-14-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-14-1" style="text-align:left; white-space:nowrap;"> atan     </td><td class="td11" id="TBL-5-14-2" style="text-align:left; white-space:nowrap;"> Inverse tangent (in radians)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                </td>
</tr><tr id="TBL-5-15-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-15-1" style="text-align:left; white-space:nowrap;"> atan2   </td><td class="td11" id="TBL-5-15-2" style="text-align:left; white-space:nowrap;"> Two-argument inverse tangent (in radians)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        </td>
</tr><tr class="hline"><td><hr/></td><td><hr/></td></tr><tr id="TBL-5-16-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-16-1" style="text-align:left; white-space:nowrap;"> sinh     </td><td class="td11" id="TBL-5-16-2" style="text-align:left; white-space:nowrap;"> Hyperbolic sine                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   </td>
</tr><tr id="TBL-5-17-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-17-1" style="text-align:left; white-space:nowrap;"> cosh    </td><td class="td11" id="TBL-5-17-2" style="text-align:left; white-space:nowrap;"> Hyperbolic cosine                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               </td>
</tr><tr id="TBL-5-18-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-18-1" style="text-align:left; white-space:nowrap;"> tanh    </td><td class="td11" id="TBL-5-18-2" style="text-align:left; white-space:nowrap;"> Hyperbolic tangent                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             </td>
</tr><tr class="hline"><td><hr/></td><td><hr/></td></tr><tr id="TBL-5-19-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-19-1" style="text-align:left; white-space:nowrap;"> abs      </td><td class="td11" id="TBL-5-19-2" style="text-align:left; white-space:nowrap;"> Absolute value                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    </td>
</tr><tr id="TBL-5-20-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-20-1" style="text-align:left; white-space:nowrap;"> floor    </td><td class="td11" id="TBL-5-20-2" style="text-align:left; white-space:nowrap;"> Floor                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   </td>
</tr><tr id="TBL-5-21-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-21-1" style="text-align:left; white-space:nowrap;"> ceil      </td><td class="td11" id="TBL-5-21-2" style="text-align:left; white-space:nowrap;"> Ceiling                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                </td>
</tr><tr id="TBL-5-22-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-22-1" style="text-align:left; white-space:nowrap;"> min     </td><td class="td11" id="TBL-5-22-2" style="text-align:left; white-space:nowrap;"> Minimum of two values                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      </td>
</tr><tr id="TBL-5-23-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-23-1" style="text-align:left; white-space:nowrap;"> max    </td><td class="td11" id="TBL-5-23-2" style="text-align:left; white-space:nowrap;"> Maximum of two values                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     </td>
</tr><tr id="TBL-5-24-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-24-1" style="text-align:left; white-space:nowrap;"> delta    </td><td class="td11" id="TBL-5-24-2" style="text-align:left; white-space:nowrap;"> <!--l. 3882--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="normal">delta</mi><mrow><mo class="MathClass-open">(</mo><mrow><mi>x</mi></mrow><mo class="MathClass-close">)</mo></mrow> <mo class="MathClass-rel">=</mo> <mn>1</mn></math> if <!--l. 3882--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>x</mi> <mo class="MathClass-rel">=</mo> <mn>0</mn></math>, 0 otherwise                                                                                                                                                                            </td>
</tr><tr id="TBL-5-25-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-25-1" style="text-align:left; white-space:nowrap;"> step     </td><td class="td11" id="TBL-5-25-2" style="text-align:left; white-space:nowrap;"> <!--l. 3882--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="normal">step</mi><mrow><mo class="MathClass-open">(</mo><mrow><mi>x</mi></mrow><mo class="MathClass-close">)</mo></mrow> <mo class="MathClass-rel">=</mo> <mn>0</mn></math> if <!--l. 3882--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>x</mi> <mo class="MathClass-rel">&lt;</mo> <mn>0</mn></math>, 1 if <!--l. 3882--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>x</mi> <mo class="MathClass-rel">&gt;</mo><mo class="MathClass-rel">=</mo> <mn>0</mn></math> </td>
</tr><tr id="TBL-5-26-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-26-1" style="text-align:left; white-space:nowrap;"> select  </td><td class="td11" id="TBL-5-26-2" style="text-align:left; white-space:nowrap;"> <!--l. 3882--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="normal">select</mi><mrow><mo class="MathClass-open">(</mo><mrow><mi>x</mi><mo class="MathClass-punc">,</mo><mi>y</mi><mo class="MathClass-punc">,</mo><mi>z</mi></mrow><mo class="MathClass-close">)</mo></mrow> <mo class="MathClass-rel">=</mo> <mi>z</mi></math> if <!--l. 3882--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>x</mi> <mo class="MathClass-rel">=</mo> <mn>0</mn></math>, <!--l. 3882--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>y</mi></math> otherwise  </td>
</tr><tr class="hline"><td><hr/></td><td><hr/></td></tr><tr id="TBL-5-27-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-27-1" style="text-align:left; white-space:nowrap;"> </td></tr></table>
</div></div>
<div class="flushright">
<!--l. 3954--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3954--><p class="noindent"></p><div class="heading"><a id="sec:colvar_scripted">
<h4 class="subsectionHead"><span class="titlemark">4.18    </span> <a id="x1-930004.18"></a>Scripted functions</h4>
<!--l. 3954--><p class="noindent"></p></a> </div>  When scripting is supported (default in NAMD), a colvar may be defined as a scripted function of its
components, rather than a linear or polynomial combination. When implementing generic functions of Cartesian
coordinates rather than functions of existing components, the <span class="t1-zi4r-0-x-x-109">cartesian </span>component may be particularly useful. A
scalar-valued scripted variable may be manually defined as periodic by providing the keyword <span class="t1-zi4r-0-x-x-109">period</span>, and
the optional keyword <span class="t1-zi4r-0-x-x-109">wrapAround</span>, with the same meaning as in periodic components (see <a href="#x1-880004.14">4.14<!--tex4ht:ref: sec:cvc_periodic --></a> for
details).
<!--l. 3954--><p class="indent">   An example of elaborate scripted colvar is given in example 10, in the form of path-based collective variables as
defined by Branduardi et al<span class="cite">[<a href="#XBranduardi2007">10</a>]</span> (Section <a href="#x1-810004.10.3">4.10.3<!--tex4ht:ref: sec:apath_scripted --></a>).
</p>
<ul class="itemize1">
<li class="itemize"><a id="colvar|scriptedFunction"> </a> <a id="dx1-93001"></a><span class="colorbox" id="colorbox189"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">scriptedFunction</span></span> <!--l. 3954--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Compute
     colvar as a scripted function of its components<!--l. 3954--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">colvar</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>string <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>If this option is specified, the colvar will be computed as a scripted function of the values
     of its components. To that effect, the user should define two Tcl procedures: <span class="t1-zi4r-0-x-x-109">calc_</span><!--l. 3954--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="t1-zi4r-0-x-x-109">scriptedFunction</span><!--l. 3954--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>
     and <span class="t1-zi4r-0-x-x-109">calc_</span><!--l. 3954--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="t1-zi4r-0-x-x-109">scriptedFunction</span><!--l. 3954--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><span class="t1-zi4r-0-x-x-109">_gradient</span>,
     both  accepting  as  many  parameters  as  the  colvar  has  components.  Values  of  the  components  will
     be  passed  to  those  procedures  in  the  order  defined  by  their  sorted  <span class="t1-zi4r-0-x-x-109">name </span>strings.  Note  that  if  all
     components  are  of  the  same  type,  their  default  names  are  sorted  in  the  order  in  which  they  are
     defined,  so  that  names  need  only  be  specified  for  combinations  of  components  of  different  types.
     <span class="t1-zi4r-0-x-x-109">calc_</span><!--l. 3954--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="t1-zi4r-0-x-x-109">scriptedFunction</span><!--l. 3954--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>
     should return one value of type <!--l. 3954--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>scriptedFunctionType<!--l. 3954--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>,
     corresponding to the colvar value. <span class="t1-zi4r-0-x-x-109">calc_</span><!--l. 3954--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="t1-zi4r-0-x-x-109">scriptedFunction</span><!--l. 3954--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><span class="t1-zi4r-0-x-x-109">_gradient</span>
     should return a Tcl list containing the derivatives of the function with respect to each component. If
     both the function and some of the components are vectors, the gradient is really a Jacobian matrix that
     should be passed as a linear vector in row-major order, i.e. for a function <!--l. 3954--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>f</mi></mrow><mrow><mi>i</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>x</mi></mrow><mrow><mi>j</mi></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math>:
     <!--l. 3954--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mo class="MathClass-op">∇</mo></mrow><mrow><mi>x</mi></mrow></msub><msub><mrow><mi>f</mi></mrow><mrow><mn>1</mn></mrow></msub><msub><mrow><mo class="MathClass-op">∇</mo></mrow><mrow><mi>x</mi></mrow></msub><msub><mrow><mi>f</mi></mrow><mrow><mn>2</mn></mrow></msub><mo class="MathClass-rel">⋯</mo><mspace class="thinspace" width="0.3em"></mspace></math>.
     </li>
<li class="itemize"><a id="dx1-93002"></a><span class="colorbox" id="colorbox190"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">scriptedFunctionType</span></span> <!--l. 3954--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Type
     of value returned by the scripted colvar<!--l. 3954--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">colvar</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>string <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">scalar</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>If a colvar is defined as a scripted function, its type is not constrained by the types of its
     components. With this flag, the user may specify whether the colvar is a scalar or one of the following
                                                                                         
                                                                                         
     vector types: <span class="t1-zi4r-0-x-x-109">vector3 </span>(a 3D vector), <span class="t1-zi4r-0-x-x-109">unit_vector3 </span>(a normalized 3D vector), or <span class="t1-zi4r-0-x-x-109">unit_quaternion</span>
     (a normalized quaternion), or <span class="t1-zi4r-0-x-x-109">vector </span>(a vector whose size is specified by <span class="t1-zi4r-0-x-x-109">scriptedFunctionVectorSize</span>).
     Non-scalar values should be passed as space-separated lists.
     </li>
<li class="itemize"><a id="dx1-93003"></a><span class="colorbox" id="colorbox191"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">scriptedFunctionVectorSize</span></span> <!--l. 3954--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Dimension
     of the vector value of a scripted colvar<!--l. 3954--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">colvar</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive integer <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>This parameter is only valid when <span class="t1-zi4r-0-x-x-109">scriptedFunctionType </span>is set to <span class="t1-zi4r-0-x-x-109">vector</span>. It defines
     the vector length of the colvar value returned by the function.</li></ul>
<div class="flushright">
<!--l. 3958--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3958--><p class="noindent"></p><div class="heading"><a id="sec:colvar_grid_params">
<h4 class="subsectionHead"><span class="titlemark">4.19    </span> <a id="x1-940004.19"></a>Defining grid parameters</h4>
<!--l. 3958--><p class="noindent"></p></a> </div> 
<!--l. 3960--><p class="indent">   Many algorithms require the definition of boundaries and/or characteristic spacings that can be used to define
discrete “states" in the collective variable, or to combine variables with very different units. The parameters
described below offer a way to specify these parameters only once for each variable, while using them multiple
times in restraints, time-dependent biases or analysis methods.
</p>
<ul class="itemize1">
<li class="itemize"><a id="colvar|width"> </a> <a id="dx1-94001"></a><span class="colorbox" id="colorbox192"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">width</span></span> <!--l. 3978--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Unit
     of the variable, or grid spacing<!--l. 3978--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">colvar</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive decimal <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span>1.0 <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>This number defines the effective unit of measurement for the collective variable, and
     is  used  by  the  biasing  methods  for  the  following  purposes.  Harmonic  (<a href="#x1-1270006.5">6.5<!--tex4ht:ref: sec:colvarbias_harmonic --></a>),  harmonic  walls  (<a href="#x1-1320006.7">6.7<!--tex4ht:ref: sec:colvarbias_harmonic_walls --></a>)
     and  linear  restraints  (<a href="#x1-1330006.8">6.8<!--tex4ht:ref: sec:colvarbias_linear --></a>)  use  it  to  set  the  physical  unit  of  the  force  constant,  which  is  useful  for
     multidimensional restraints involving multiple variables with very different units (for examples, Å or
     degrees <!--l. 3978--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></math>)
     with a single, scaled force constant. The values of the scaled force constant in the units of each variable
     are printed at initialization time. Histograms (<a href="#x1-1350006.10">6.10<!--tex4ht:ref: sec:colvarbias_histogram --></a>), ABF (<a href="#x1-1090006.2">6.2<!--tex4ht:ref: sec:colvarbias_abf --></a>) and metadynamics (<a href="#x1-1190006.4">6.4<!--tex4ht:ref: sec:colvarbias_meta --></a>) all use this
     number as the initial choice for the grid spacing along this variable: for this reason, <span class="t1-zi4r-0-x-x-109">width </span>should
     generally be no larger than the standard deviation of the colvar in an unbiased simulation. Unless it is
     required to control the spacing, it is usually simplest to keep the default value of 1, so that restraint
     force constants are provided with their full physical unit.
                                                                                         
                                                                                         
     </li>
<li class="itemize"><a id="colvar|lowerBoundary"> </a> <a id="dx1-94002"></a><span class="colorbox" id="colorbox193"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">lowerBoundary</span></span> <!--l. 3992--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Lower
     boundary of the colvar<!--l. 3992--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">colvar</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>decimal <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span>natural boundary of the function <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Defines the lowest end of the interval of “relevant" values for the variable. This number
     can be, for example, a true physical boundary imposed by the choice of function (e.g. the <span class="t1-zi4r-0-x-x-109">distance</span>
     function is always larger than zero): if this is the case, and only one function is used to define the
     variable, the default value of this number is set to the lowest end of the range of values of that function,
     if available (see Section <a href="#x1-300004.1">4.1<!--tex4ht:ref: sec:cvc_list --></a>). Alternatively, this value may be provided by the user, to represent for
     example the left-most point of a PMF calculation along this variable. In the latter case, it is the user's
     responsibility to either <span class="ptmri8t-x-x-109">(a) </span>ensure the variable does not go significantly beyond the boundary (for
     example by adding a <span class="t1-zi4r-0-x-x-109">harmonicWalls </span>restraint, <a href="#x1-1320006.7">6.7<!--tex4ht:ref: sec:colvarbias_harmonic_walls --></a>), or <span class="ptmri8t-x-x-109">(b) </span>instruct the code that this is a true physical
     boundary by setting <a href="#colvar|hardLowerBoundary"><span class="t1-zi4r-0-x-x-109">hardLowerBoundary</span></a>.
     </li>
<li class="itemize"><a id="colvar|upperBoundary"> </a> <a id="dx1-94003"></a><span class="colorbox" id="colorbox194"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">upperBoundary</span></span> <!--l. 4002--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Upper
     boundary of the colvar<!--l. 4002--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">colvar</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>decimal <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span>natural boundary of the function <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Similarly to <span class="t1-zi4r-0-x-x-109">lowerBoundary</span>, defines the highest of the “relevant" values of the variable.
     </li>
<li class="itemize"><a id="colvar|hardLowerBoundary"> </a> <a id="dx1-94004"></a><span class="colorbox" id="colorbox195"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">hardLowerBoundary</span></span> <!--l. 4015--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Whether
     the lower boundary is the physical lower limit<!--l. 4015--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">colvar</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span>provided by the component <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>When the colvar has a “natural" boundary (for example, a <span class="t1-zi4r-0-x-x-109">distance </span>colvar cannot go
     below 0) this flag is automatically enabled. For more complex variable definitions, or when <a href="#colvar|lowerBoundary"><span class="t1-zi4r-0-x-x-109">lowerBoundary</span></a>
     is provided directly by the user, it may be useful to set this flag explicitly. This option does not affect
     simulation results, but enables some internal optimizations by letting the code know that the variable
     is unable to cross the lower boundary, regardless of whether restraints are applied to it.
     </li>
<li class="itemize"><a id="colvar|hardUpperBoundary"> </a> <a id="dx1-94005"></a><span class="colorbox" id="colorbox196"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">hardUpperBoundary</span></span> <!--l. 4025--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Whether
     the upper boundary is the physical upper limit of the colvar's values<!--l. 4025--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">colvar</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span>provided by the component <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Analogous to <span class="t1-zi4r-0-x-x-109">hardLowerBoundary</span>.
     </li>
<li class="itemize"><a id="colvar|expandBoundaries"> </a> <a id="dx1-94006"></a><span class="colorbox" id="colorbox197"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">expandBoundaries</span></span> <!--l. 4040--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Allow
     to expand the two boundaries if needed<!--l. 4040--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">colvar</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">off</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>If defined, <span class="t1-zi4r-0-x-x-109">lowerBoundary </span>and <span class="t1-zi4r-0-x-x-109">upperBoundary </span>may be automatically expanded to
     accommodate colvar values that do not fit in the initial range. Currently, this option is used by the
     metadynamics bias (<a href="#x1-1190006.4">6.4<!--tex4ht:ref: sec:colvarbias_meta --></a>) to keep all of its hills fully within the grid. This option cannot be used when
     the initial boundaries already span the full period of a periodic colvar.
     </li></ul>
<div class="flushright">
<!--l. 4045--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 4045--><p class="noindent"></p><div class="heading"><a id="sec:colvar_traj_output">
<h4 class="subsectionHead"><span class="titlemark">4.20    </span> <a id="x1-950004.20"></a>Trajectory output</h4>
<!--l. 4045--><p class="noindent"></p></a> </div> 
<ul class="itemize1">
<li class="itemize"><a id="dx1-95001"></a><span class="colorbox" id="colorbox198"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">outputValue</span></span> <!--l. 4058--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Output
     a trajectory for this colvar<!--l. 4058--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">colvar</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">on</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>If <span class="t1-zi4r-0-x-x-109">colvarsTrajFrequency </span>is non-zero, the value of this colvar is written to the trajectory
     file every <span class="t1-zi4r-0-x-x-109">colvarsTrajFrequency </span>steps in the column labeled “<!--l. 4058--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="t1-zi4r-0-x-x-109">name</span><!--l. 4058--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>".
     </li>
<li class="itemize"><a id="dx1-95002"></a><span class="colorbox" id="colorbox199"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">outputVelocity</span></span> <!--l. 4070--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Output
     a velocity trajectory for this colvar<!--l. 4070--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">colvar</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">off</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>If <span class="t1-zi4r-0-x-x-109">colvarsTrajFrequency </span>is defined, the finite-difference calculated velocity of this
     colvar are written to the trajectory file under the label “<span class="t1-zi4r-0-x-x-109">v_</span><!--l. 4070--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="t1-zi4r-0-x-x-109">name</span><!--l. 4070--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>".
     </li>
<li class="itemize"><a id="dx1-95003"></a><span class="colorbox" id="colorbox200"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">outputEnergy</span></span> <!--l. 4084--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Output
     an energy trajectory for this colvar<!--l. 4084--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">colvar</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">off</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>This option applies only to extended Lagrangian colvars. If <span class="t1-zi4r-0-x-x-109">colvarsTrajFrequency</span>
     is defined, the kinetic energy of the extended degree and freedom and the potential energy of the
     restraining spring are are written to the trajectory file under the labels “<span class="t1-zi4r-0-x-x-109">Ek_</span><!--l. 4084--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="t1-zi4r-0-x-x-109">name</span><!--l. 4084--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>"
     and “<span class="t1-zi4r-0-x-x-109">Ep_</span><!--l. 4084--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="t1-zi4r-0-x-x-109">name</span><!--l. 4084--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>".
     </li>
<li class="itemize"><a id="colvar|outputTotalForce"> </a> <a id="dx1-95004"></a><span class="colorbox" id="colorbox201"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">outputTotalForce</span></span> <!--l. 4106--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Output
     a total force trajectory for this colvar<!--l. 4106--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">colvar</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">off</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>If <span class="t1-zi4r-0-x-x-109">colvarsTrajFrequency </span>is defined, the total force on this colvar (i.e. the projection
     of  all  atomic  total  forces  onto  this  colvar  —  see  equation (<a href="#x1-109004r28">28<!--tex4ht:ref: eq:gradient_vector --></a>)  in  section <a href="#x1-1090006.2">6.2<!--tex4ht:ref: sec:colvarbias_abf --></a>)  are  written  to  the
     trajectory file under the label “<span class="t1-zi4r-0-x-x-109">fs_</span><!--l. 4106--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="t1-zi4r-0-x-x-109">name</span><!--l. 4106--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>".
     For extended Lagrangian colvars, the “total force" felt by the extended degree of freedom is simply
     the force from the harmonic spring. Due to design constraints, the total force reported by NAMD to
     Colvars was computed at the previous simulation step.  <span class="ptmb8t-x-x-109">Note: </span>not all components support this option.
     The physical unit for this force is kcal/mol, divided by the colvar unit U.
     </li>
<li class="itemize"><a id="colvar|outputAppliedForce"> </a> <a id="dx1-95005"></a><span class="colorbox" id="colorbox202"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">outputAppliedForce</span></span> <!--l. 4124--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Output
     an applied force trajectory for this colvar<!--l. 4124--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">colvar</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">off</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>If <span class="t1-zi4r-0-x-x-109">colvarsTrajFrequency </span>is defined, the total force applied on this colvar by Colvars
     biases are written to the trajectory under the label “<span class="t1-zi4r-0-x-x-109">fa_</span><!--l. 4124--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="t1-zi4r-0-x-x-109">name</span><!--l. 4124--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>".
     For extended Lagrangian colvars, this force is actually applied to the extended degree of freedom
     rather than the geometric colvar itself. The physical unit for this force is kcal/mol divided by the colvar
     unit.
     </li></ul>
<div class="flushright">
<!--l. 4129--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 4129--><p class="noindent"></p><div class="heading"><a id="sec:colvar_extended">
<h4 class="subsectionHead"><span class="titlemark">4.21    </span> <a id="x1-960004.21"></a>Extended Lagrangian</h4>
<!--l. 4129--><p class="noindent"></p></a> </div> 
<!--l. 4131--><p class="indent">   The following options enable extended-system dynamics, where a colvar is coupled to an additional degree of
freedom (fictitious particle) by a harmonic spring. This extended coordinate masks the colvar and replaces it
transparently from the perspective of biasing and analysis methods. Biasing forces are then applied to the extended
degree of freedom, and the actual geometric colvar (function of Cartesian coordinates) only feels the force from the
                                                                                         
                                                                                         
harmonic spring. This is particularly useful when combined with an <a href="#sec:colvarbias_abf"><span class="t1-zi4r-0-x-x-109">abf</span></a> bias to perform eABF simulations
(<a href="#x1-1150006.3">6.3<!--tex4ht:ref: sec:eABF --></a>).
</p><!--l. 4142--><p class="indent">   Note that for some biases (<a href="#sec:colvarbias_harmonic_walls"><span class="t1-zi4r-0-x-x-109">harmonicWalls</span></a>, <a href="#sec:colvarbias_histogram"><span class="t1-zi4r-0-x-x-109">histogram</span></a>), this masking behavior is controlled by the keyword
<a href="#colvarbias|bypassExtendedLagrangian"><span class="t1-zi4r-0-x-x-109">bypassExtendedLagrangian</span></a>. Specifically for <span class="t1-zi4r-0-x-x-109">harmonicWalls</span>, the default behavior is to bypass extended
Lagrangian coordinates and act directly on the actual colvars.
</p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-96001"></a><span class="colorbox" id="colorbox203"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">extendedLagrangian</span></span> <!--l. 4166--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Add
     extended degree of freedom<!--l. 4166--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">colvar</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">off</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Adds a fictitious particle to be coupled to the colvar by a harmonic spring. The fictitious
     mass and the force constant of the coupling potential are derived from the parameters <span class="t1-zi4r-0-x-x-109">extendedTimeConstant</span>
     and <span class="t1-zi4r-0-x-x-109">extendedFluctuation</span>, described below. Biasing forces on the colvar are applied to this fictitious
     particle, rather than to the atoms directly. This implements the extended Lagrangian formalism used
     in some metadynamics simulations <span class="cite">[<a href="#XIannuzzi2003">3</a>]</span>. The energy associated with the extended degree of freedom
     is reported along with bias energies under the MISC title in NAMD's energy output.
     </li>
<li class="itemize"><a id="dx1-96002"></a><span class="colorbox" id="colorbox204"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">extendedFluctuation</span></span> <!--l. 4179--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Standard
     deviation between the colvar and the fictitious particle (colvar unit)<!--l. 4179--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">colvar</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive decimal <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Defines the spring stiffness for the <span class="t1-zi4r-0-x-x-109">extendedLagrangian </span>mode, by setting the typical
     deviation between the colvar and the extended degree of freedom due to thermal fluctuation. The
     spring force constant is calculated internally as <!--l. 4179--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>k</mi></mrow><mrow><mi>B</mi></mrow></msub><mi>T</mi> <mo class="MathClass-bin">∕</mo><msup><mrow><mi>σ</mi></mrow><mrow><mn>2</mn></mrow></msup></math>,
     where <!--l. 4179--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>σ</mi></math>
     is the value of <span class="t1-zi4r-0-x-x-109">extendedFluctuation</span>.
     </li>
<li class="itemize"><a id="dx1-96003"></a><span class="colorbox" id="colorbox205"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">extendedTimeConstant</span></span> <!--l. 4194--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Oscillation
     period of the fictitious particle (fs)<!--l. 4194--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">colvar</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive decimal <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">200</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Defines the inertial mass of the fictitious particle, by setting the oscillation period of the
     harmonic oscillator formed by the fictitious particle and the spring. The period should be much larger
     than the MD time step to ensure accurate integration of the extended particle's equation of motion. The
     fictitious mass is calculated internally as <!--l. 4194--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>k</mi></mrow><mrow><mi>B</mi></mrow></msub><mi>T</mi> <msup><mrow><mrow><mo class="MathClass-open">(</mo><mrow><mi>τ</mi><mo class="MathClass-bin">∕</mo><mn>2</mn><mi>π</mi><mi>σ</mi></mrow><mo class="MathClass-close">)</mo></mrow></mrow><mrow><mn>2</mn></mrow></msup></math>,
     where <!--l. 4194--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>τ</mi></math>
     is the period and <!--l. 4194--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>σ</mi></math>
     is the typical fluctuation (see above).
     </li>
<li class="itemize"><a id="dx1-96004"></a><span class="colorbox" id="colorbox206"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">extendedTemp</span></span> <!--l. 4221--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Temperature
     for the extended degree of freedom (K)<!--l. 4221--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">colvar</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive decimal <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span>thermostat temperature <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Temperature used for calculating the coupling force constant of the extended variable
     (see <span class="t1-zi4r-0-x-x-109">extendedFluctuation</span>) and, if needed, as a target temperature for extended Langevin dynamics
     (see <span class="t1-zi4r-0-x-x-109">extendedLangevinDamping</span>). This should normally be left at its default value.
     </li>
<li class="itemize"><a id="dx1-96005"></a><span class="colorbox" id="colorbox207"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">extendedLangevinDamping</span></span> <!--l. 4238--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Damping
     factor for extended Langevin dynamics (ps<!--l. 4238--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow></mrow><mrow><mo class="MathClass-bin">−</mo><mn>1</mn></mrow></msup></math>)<!--l. 4238--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">colvar</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive decimal <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">1.0</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>If  this  is  non-zero,  the  extended  degree  of  freedom  undergoes  Langevin  dynamics
     at  temperature  <span class="t1-zi4r-0-x-x-109">extendedTemp</span>.  The  friction  force  is  minus  <span class="t1-zi4r-0-x-x-109">extendedLangevinDamping </span>times  the
     velocity. This is useful because the extended dynamics coordinate may heat up in the transient non-equilibrium
     regime of ABF. Use moderate damping values, to limit viscous friction (potentially slowing down
     diffusive  sampling)  and  stochastic  noise  (increasing  the  variance  of  statistical  measurements).  In
     doubt, use the default value. </li></ul>
<div class="flushright">
<!--l. 4242--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 4242--><p class="noindent"></p><div class="heading"><a id="sec:mts_colvar">
<h4 class="subsectionHead"><span class="titlemark">4.22    </span> <a id="x1-970004.22"></a>Multiple time-step variables</h4>
<!--l. 4242--><p class="noindent"></p></a> </div> 
<ul class="itemize1">
<li class="itemize"><a id="dx1-97001"></a><span class="colorbox" id="colorbox208"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">timeStepFactor</span></span> <!--l. 4259--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Compute
     this colvar once in a certain number of timesteps<!--l. 4259--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">colvar</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive integer <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">1</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Instructs this colvar to activate at a time interval equal to the base (MD) timestep times
     <span class="t1-zi4r-0-x-x-109">timeStepFactor</span>.<span class="cite">[<a href="#XFerrarotti2015">14</a>]</span> At other time steps, the value of the variable is not updated, and no biasing
     forces are applied. Any forces exerted by biases are accumulated over the given time interval, then
     applied as an impulse at the next update.  </li></ul>
<div class="flushright">
<!--l. 4262--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 4262--><p class="noindent"></p><div class="heading"><a id="sec:colvar_compatibility">
<h4 class="subsectionHead"><span class="titlemark">4.23    </span> <a id="x1-980004.23"></a>Backward-compatibility</h4>
<!--l. 4262--><p class="noindent"></p></a> </div> 
<ul class="itemize1">
<li class="itemize"><a id="dx1-98001"></a><span class="colorbox" id="colorbox209"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">subtractAppliedForce</span></span> <!--l. 4278--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Do
     not include biasing forces in the total force for this colvar<!--l. 4278--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">colvar</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">off</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>If  the  colvar  supports  total  force  calculation  (see  <a href="#x1-900004.15.1">4.15.1<!--tex4ht:ref: sec:cvc_sys_forces --></a>),  all  forces  applied  to  this
     colvar by biases will be removed from the total force. This keyword allows to recover some of the
     “system force" calculation available in the Colvars module before version 2016-08-10. Please note
     that removal of <span class="ptmri8t-x-x-109">all </span>other external forces (including biasing forces applied to a different colvar) is
     <span class="ptmri8t-x-x-109">no longer supported</span>, due to changes in the underlying simulation engines (primarily NAMD). This
     option may be useful when continuing a previous simulation where the removal of external/applied
     forces is essential. <span class="ptmri8t-x-x-109">For all new simulations, the use of this option is not recommended.</span>
</li></ul>
<div class="flushright">
<!--l. 4283--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 4283--><p class="noindent"></p><div class="heading"><a id="sec:colvar_acf">
<h4 class="subsectionHead"><span class="titlemark">4.24    </span> <a id="x1-990004.24"></a>Statistical analysis</h4>
<!--l. 4283--><p class="noindent"></p></a> </div> 
<!--l. 4285--><p class="indent">   Run-time calculations of statistical properties that depend explicitly on time can be performed for individual
collective variables. Currently, several types of time correlation functions, running averages and running standard
deviations are implemented. For run-time computation of histograms, please see the histogram bias
(<a href="#x1-1350006.10">6.10<!--tex4ht:ref: sec:colvarbias_histogram --></a>).
</p>
<ul class="itemize1">
<li class="itemize"><a id="colvar|corrFunc"> </a> <a id="dx1-99001"></a><span class="colorbox" id="colorbox210"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">corrFunc</span></span> <!--l. 4300--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Calculate
     a time correlation function?<!--l. 4300--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">colvar</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">off</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Whether or not a time correlaction function should be calculated for this colvar.
                                                                                         
                                                                                         
     </li>
<li class="itemize"><a id="dx1-99002"></a><span class="colorbox" id="colorbox211"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">corrFuncWithColvar</span></span> <!--l. 4312--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Colvar
     name for the correlation function<!--l. 4312--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">colvar</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>string <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>By default, the auto-correlation function (ACF) of this colvar, <!--l. 4312--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></math>,
     is calculated. When this option is specified, the correlation function is calculated instead with another
     colvar, <!--l. 4312--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>j</mi></mrow></msub></math>,
     which must be of the same type (scalar, vector, or quaternion) as <!--l. 4312--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></math>.
     </li>
<li class="itemize"><a id="dx1-99003"></a><span class="colorbox" id="colorbox212"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">corrFuncType</span></span> <!--l. 4330--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Type
     of the correlation function<!--l. 4330--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">colvar</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span><span class="t1-zi4r-0-x-x-109">velocity</span>, <span class="t1-zi4r-0-x-x-109">coordinate </span>or <span class="t1-zi4r-0-x-x-109">coordinate_p2</span><br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">velocity</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>With <span class="t1-zi4r-0-x-x-109">coordinate </span>or <span class="t1-zi4r-0-x-x-109">velocity</span>, the correlation function <!--l. 4330--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>C</mi></mrow><mrow><mi>i</mi><mo class="MathClass-punc">,</mo><mi>j</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow></math> =
     <!--l. 4330--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mfenced close="⟩" open="⟨" separators=""><mrow><mi>Π</mi> <mfenced close=")" open="(" separators=""><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>t</mi></mrow><mrow><mn>0</mn></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-punc">,</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>j</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>t</mi></mrow><mrow><mn>0</mn></mrow></msub> <mo class="MathClass-bin">+</mo><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow></mrow></mfenced></mrow></mfenced></math>
     is calculated between the variables <!--l. 4330--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></math>
     and <!--l. 4330--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>j</mi></mrow></msub></math>,
     or their velocities. <!--l. 4330--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>Π</mi><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>j</mi></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math>
     is the scalar product when calculated between scalar or vector values, whereas for quaternions it is the
     cosine between the two corresponding rotation axes. With <span class="t1-zi4r-0-x-x-109">coordinate_p2</span>, the second order Legendre
     polynomial, <!--l. 4330--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">(</mo><mrow><mn>3</mn><mo class="qopname">cos</mo><!--nolimits--><msup><mrow><mrow><mo class="MathClass-open">(</mo><mrow><mi>𝜃</mi></mrow><mo class="MathClass-close">)</mo></mrow></mrow><mrow><mn>2</mn></mrow></msup> <mo class="MathClass-bin">−</mo><mn>1</mn></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-bin">∕</mo><mn>2</mn></math>,
     is used instead of the cosine.
     </li>
<li class="itemize"><a id="dx1-99004"></a><span class="colorbox" id="colorbox213"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">corrFuncNormalize</span></span> <!--l. 4341--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Normalize
     the time correlation function?<!--l. 4341--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">colvar</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">on</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>If enabled, the value of the correlation function at <!--l. 4341--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>t</mi></math> =
     0 is normalized to 1; otherwise, it equals to <!--l. 4341--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mfenced close="⟩" open="⟨" separators=""><mrow><mi>O</mi> <mfenced close=")" open="(" separators=""><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>j</mi></mrow></msub></mrow></mfenced></mrow></mfenced></math>.
     </li>
<li class="itemize"><a id="dx1-99005"></a><span class="colorbox" id="colorbox214"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">corrFuncLength</span></span> <!--l. 4350--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Length
     of the time correlation function<!--l. 4350--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">colvar</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive integer <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">1000</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Length (in number of points) of the time correlation function.
     </li>
<li class="itemize"><a id="dx1-99006"></a><span class="colorbox" id="colorbox215"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">corrFuncStride</span></span> <!--l. 4359--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Stride
     of the time correlation function<!--l. 4359--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">colvar</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive integer <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">1</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Number of steps between two values of the time correlation function.
     </li>
<li class="itemize"><a id="dx1-99007"></a><span class="colorbox" id="colorbox216"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">corrFuncOffset</span></span> <!--l. 4370--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Offset
     of the time correlation function<!--l. 4370--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">colvar</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive integer <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">0</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>The starting time (in number of steps) of the time correlation function (default: <!--l. 4370--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>t</mi></math> =
     0). <span class="ptmb8t-x-x-109">Note: </span><span class="ptmri8t-x-x-109">the value at </span><!--l. 4370--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>t</mi></math><span class="ptmri8t-x-x-109"> =</span>
<span class="ptmri8t-x-x-109">0 is always used for the normalization</span>.
     </li>
<li class="itemize"><a id="dx1-99008"></a><span class="colorbox" id="colorbox217"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">corrFuncOutputFile</span></span> <!--l. 4379--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Output
     file for the time correlation function<!--l. 4379--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">colvar</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>UNIX filename <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="ptmri8t-x-x-109">outputName</span><span class="t1-zi4r-0-x-x-109">.</span><!--l. 4379--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="t1-zi4r-0-x-x-109">name</span><!--l. 4379--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><span class="t1-zi4r-0-x-x-109">.corrfunc.dat</span>
<br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>The time correlation function is saved in this file.
     </li>
<li class="itemize"><a id="colvar|runAve"> </a> <a id="dx1-99009"></a><span class="colorbox" id="colorbox218"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">runAve</span></span> <!--l. 4390--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Calculate
     the running average and standard deviation<!--l. 4390--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">colvar</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">off</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Whether or not the running average and standard deviation should be calculated for this
     colvar.
     </li>
<li class="itemize"><a id="dx1-99010"></a><span class="colorbox" id="colorbox219"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">runAveLength</span></span> <!--l. 4399--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Length
     of the running average window<!--l. 4399--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">colvar</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive integer <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">1000</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Length (in number of points) of the running average window.
     </li>
<li class="itemize"><a id="dx1-99011"></a><span class="colorbox" id="colorbox220"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">runAveStride</span></span> <!--l. 4408--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Stride
     of the running average window values<!--l. 4408--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">colvar</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive integer <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">1</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Number of steps between two values within the running average window.
     </li>
<li class="itemize"><a id="dx1-99012"></a><span class="colorbox" id="colorbox221"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">runAveOutputFile</span></span> <!--l. 4417--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Output
     file for the running average and standard deviation<!--l. 4417--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">colvar</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>UNIX filename <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="ptmri8t-x-x-109">outputName</span><span class="t1-zi4r-0-x-x-109">.</span><!--l. 4417--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="t1-zi4r-0-x-x-109">name</span><!--l. 4417--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><span class="t1-zi4r-0-x-x-109">.runave.traj</span>
<br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>The running average and standard deviation are saved in this file.
     </li></ul>
<!--l. 4422--><p class="indent">
</p>
<div class="flushright">
<!--l. 4422--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 4422--><p class="noindent"></p><div class="heading"><a id="sec:colvar_atom_groups">
<h3 class="sectionHead"><span class="titlemark">5    </span> <a id="x1-1000005"></a>Selecting atoms</h3>
<!--l. 4422--><p class="noindent"></p></a> </div> 
<!--l. 4424--><p class="indent">   To define collective variables, atoms are usually selected as groups. Each group is defined using an identifier that
is unique in the context of the specific colvar component (e.g. for a distance component, the two groups are <span class="t1-zi4r-0-x-x-109">group1</span>
and <span class="t1-zi4r-0-x-x-109">group2</span>). The identifier is followed by a brace-delimited block containing selection keywords and other
parameters, including an optional <span class="t1-zi4r-0-x-x-109">name</span>:
</p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-100001"></a><span class="colorbox" id="colorbox222"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">name</span></span> <!--l. 4435--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Unique
     name for the atom group<!--l. 4435--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>atom group <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>string <br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>This parameter defines a unique name for this atom group, which can be referred to
     in the definition of other atom groups (including in other colvars) by invoking <span class="t1-zi4r-0-x-x-109">atomsOfGroup </span>as a
     selection keyword.</li></ul>
<div class="flushright">
<!--l. 4439--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 4439--><p class="noindent"></p><div class="heading"><a id="sec:colvar_atom_groups_sel">
<h4 class="subsectionHead"><span class="titlemark">5.1    </span> <a id="x1-1010005.1"></a>Atom selection keywords</h4>
<!--l. 4439--><p class="noindent"></p></a> </div> 
<!--l. 4441--><p class="indent">   Selection keywords may be used individually or in combination with each other, and each can be repeated any
number of times. Selection is incremental: each keyword adds the corresponding atoms to the selection, so that
different sets of atoms can be combined. However, atoms included by multiple keywords are only counted once.
Below is an example configuration for an atom group called “<span class="t1-zi4r-0-x-x-109">atoms</span>". <span class="ptmb8t-x-x-109">Note: </span><span class="ptmri8t-x-x-109">this is an unusually varied combination</span>
<span class="ptmri8t-x-x-109">of selection keywords, demonstrating how they can be combined together: most simulations only use one of</span>
<span class="ptmri8t-x-x-109">them.</span><br class="newline"/>
</p>
<div class="mdframed" id="mdframed-49">
<!--l. 4447--><p class="noindent"></p><pre>atoms {<br class="newline"/><br class="newline"/>  # add atoms 1 and 3 to this group (note: first atom in the system is 1)<br class="newline"/>  atomNumbers { <br class="newline"/>    1 3<br class="newline"/>  }<br class="newline"/><br class="newline"/>  # add atoms starting from 20 up to and including 50<br class="newline"/>  atomNumbersRange 20-50<br class="newline"/>  # add all the atoms with occupancy 2 in the file atoms.pdb<br class="newline"/>  atomsFile atoms.pdb<br class="newline"/>  atomsCol O<br class="newline"/>  atomsColValue 2.0<br class="newline"/><br class="newline"/>  # add all the C-alphas within residues 11 to 20 of segments "PR1" and "PR2"<br class="newline"/>  psfSegID PR1 PR2<br class="newline"/>  atomNameResidueRange CA 11-20<br class="newline"/>  atomNameResidueRange CA 11-20<br class="newline"/>  # add index group (requires a .ndx file to be provided globally)<br class="newline"/>  indexGroup Water<br class="newline"/>} </pre>
</div>
<!--l. 4474--><p class="indent">   The resulting selection includes atoms 1 and 3, those between 20 and 50, the
<!--l. 4474--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="normal">C</mi></mrow><mrow><mi>α</mi></mrow></msub></math> atoms
between residues 11 and 20 of the two segments <span class="t1-zi4r-0-x-x-109">PR1 </span>and <span class="t1-zi4r-0-x-x-109">PR2</span>, and those in the index group called “Water". The
indices of this group are read from the file provided by the global keyword <a href="#Colvars-global|indexFile"><span class="t1-zi4r-0-x-x-109">indexFile</span></a>.
</p><!--l. 4478--><p class="indent">   The complete list of selection keywords available in NAMD is:
</p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-101001"></a><span class="colorbox" id="colorbox223"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">atomNumbers</span></span> <!--l. 4490--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>List
     of atom numbers<!--l. 4490--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>atom group <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>space-separated list of positive integers <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>This option adds to the group all the atoms whose numbers are in the list. <span class="ptmri8t-x-x-109">The number</span>
<span class="ptmri8t-x-x-109">of the first atom in the system is 1: to convert from a VMD selection, use “atomselect get serial".</span>
</li>
<li class="itemize"><a id="dx1-101002"></a><span class="colorbox" id="colorbox224"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">indexGroup</span></span> <!--l. 4498--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Name
     of index group to be used (GROMACS format)<!--l. 4498--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>atom group <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>string <br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>If the name of an index file has been provided by <span class="t1-zi4r-0-x-x-109">indexFile</span>, this option allows to
     select one index group from that file: the atoms from that index group will be used to define the current
     group.
                                                                                         
                                                                                         
     </li>
<li class="itemize"><a id="dx1-101003"></a><span class="colorbox" id="colorbox225"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">atomsOfGroup</span></span> <!--l. 4507--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Name
     of group defined previously<!--l. 4507--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>atom group <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>string <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Refers to a group defined previously using its user-defined <span class="t1-zi4r-0-x-x-109">name</span>. This adds all atoms of
     that named group to the current group.
     </li>
<li class="itemize"><a id="dx1-101004"></a><span class="colorbox" id="colorbox226"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">atomNumbersRange</span></span> <!--l. 4516--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Atoms
     within a number range<!--l. 4516--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>atom group <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span><!--l. 4516--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>Starting
     number<!--l. 4516--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>-<!--l. 4516--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>Ending
     number<!--l. 4516--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>
<br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>This  option  includes  in  the  group  all  atoms  whose  numbers  are  within  the  range
     specified. <span class="ptmri8t-x-x-109">The number of the first atom in the system is 1.</span>
</li>
<li class="itemize"><a id="dx1-101005"></a><span class="colorbox" id="colorbox227"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">atomNameResidueRange</span></span> <!--l. 4571--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Named
     atoms within a range of residue numbers<!--l. 4571--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>atom group <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span><!--l. 4571--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>Atom
     name<!--l. 4571--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>
<!--l. 4571--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>Starting
     residue<!--l. 4571--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>-<!--l. 4571--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>Ending
     residue<!--l. 4571--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>
<br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>This option adds to the group all the atoms with the provided name, within residues in
     the given range.
     </li>
<li class="itemize"><a id="dx1-101006"></a><span class="colorbox" id="colorbox228"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">psfSegID</span></span> <!--l. 4571--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>PSF
     segment identifier<!--l. 4571--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>atom group <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>space-separated list of strings (max 4 characters) <br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>This  option  sets  the  PSF  segment  identifier  for  <span class="t1-zi4r-0-x-x-109">atomNameResidueRange</span>.  Multiple
     values may be provided, which correspond to multiple instances of <span class="t1-zi4r-0-x-x-109">atomNameResidueRange</span>, in order
     of their occurrence. This option is only necessary if a PSF topology file is used.
     </li>
<li class="itemize"><a id="atom-group|atomsFile"> </a><a id="dx1-101007"></a><span class="colorbox" id="colorbox229"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">atomsFile</span></span> <!--l. 4571--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>PDB
     file name for atom selection<!--l. 4571--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>atom group <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>UNIX filename <br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>This  option  selects  atoms  from  the  PDB  file  provided  and  adds  them  to  the  group
     according to numerical flags in the column <span class="t1-zi4r-0-x-x-109">atomsCol</span>. <span class="ptmb8t-x-x-109">Note: </span><span class="ptmri8t-x-x-109">the sequence of atoms in the PDB file</span>
<span class="ptmri8t-x-x-109">provided must match that in the system's topology</span>.
                                                                                         
                                                                                         
     </li>
<li class="itemize"><a id="atom-group|atomsCol"> </a><a id="dx1-101008"></a><span class="colorbox" id="colorbox230"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">atomsCol</span></span> <!--l. 4571--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>PDB
     column to use for atom selection flags<!--l. 4571--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>atom group <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span><span class="t1-zi4r-0-x-x-109">O</span>, <span class="t1-zi4r-0-x-x-109">B</span>, <span class="t1-zi4r-0-x-x-109">X</span>, <span class="t1-zi4r-0-x-x-109">Y</span>, or <span class="t1-zi4r-0-x-x-109">Z</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>This option specifies which PDB column in <span class="t1-zi4r-0-x-x-109">atomsFile </span>is used to determine which
     atoms are to be included in the group.
     </li>
<li class="itemize"><a id="atom-group|atomsColValue"> </a><a id="dx1-101009"></a><span class="colorbox" id="colorbox231"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">atomsColValue</span></span> <!--l. 4571--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Atom
     selection flag in the PDB column<!--l. 4571--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>atom group <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive decimal <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>If defined, this value in <span class="t1-zi4r-0-x-x-109">atomsCol </span>identifies atoms in <span class="t1-zi4r-0-x-x-109">atomsFile </span>that are included in
     the group. If undefined, all atoms with a non-zero value in <span class="t1-zi4r-0-x-x-109">atomsCol </span>are included.
     </li>
<li class="itemize"><a id="dx1-101010"></a><span class="colorbox" id="colorbox232"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">dummyAtom</span></span> <!--l. 4579--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Dummy
     atom position (Å)<!--l. 4579--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>atom group <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span><span class="t1-zi4r-0-x-x-109">(x, y, z) </span>triplet <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Instead of selecting any atom, this option makes the group a virtual particle at a fixed
     position  in  space.  This  is  useful  e.g. to  replace  a  group's  center  of  geometry  with  a  user-defined
     position.
     </li></ul>
<div class="flushright">
<!--l. 4583--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 4583--><p class="noindent"></p><div class="heading"><a id="sec:colvar_atom_groups_ref_frame">
<h4 class="subsectionHead"><span class="titlemark">5.2    </span> <a id="x1-1020005.2"></a>Moving frame of reference.</h4>
<!--l. 4583--><p class="noindent"></p></a> </div> 
<!--l. 4585--><p class="indent">   The following options define an automatic calculation of an optimal translation (<span class="t1-zi4r-0-x-x-109">centerToReference</span>) or
optimal rotation (<span class="t1-zi4r-0-x-x-109">rotateToReference</span>), that superimposes the positions of this group to a provided
set of reference coordinates. Alternately, <span class="t1-zi4r-0-x-x-109">centerToOrigin </span>applies a translation to place the
geometric center of the group at (0, 0, 0). This can allow, for example, to effectively remove from
certain colvars the effects of molecular tumbling and of diffusion. Given the set of atomic positions
<!--l. 4588--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>i</mi></mrow></msub></math>, the colvar
<!--l. 4588--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math> can be defined on a set of
roto-translated positions <!--l. 4588--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msubsup><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>i</mi></mrow><mrow><mi>′</mi></mrow></msubsup> <mo class="MathClass-rel">=</mo> <mi>R</mi><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-bin">−</mo><msup><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi mathvariant="normal">C</mi></mrow></msup></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-bin">+</mo><msup><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi mathvariant="normal">ref</mi></mrow></msup></math>.
<!--l. 4589--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi mathvariant="normal">C</mi></mrow></msup></math> is the geometric
center of the <!--l. 4589--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>i</mi></mrow></msub></math>,
<!--l. 4589--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>R</mi></math> is the optimal rotation matrix
                                                                                         
                                                                                         
to the reference positions and <!--l. 4589--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi mathvariant="normal">ref</mi></mrow></msup></math>
is the geometric center of the reference positions.
</p><!--l. 4591--><p class="indent">   Components that are defined based on pairwise distances are naturally invariant under global roto-translations.
Other components are instead affected by global rotations or translations: however, they can be made invariant if
they are expressed in the frame of reference of a chosen group of atoms, using the <span class="t1-zi4r-0-x-x-109">centerToReference </span>and
<span class="t1-zi4r-0-x-x-109">rotateToReference </span>options. Finally, a few components are defined by convention using a roto-translated frame
(e.g. the minimal RMSD): for these components, <span class="t1-zi4r-0-x-x-109">centerToReference </span>and <span class="t1-zi4r-0-x-x-109">rotateToReference </span>are enabled by
default. In typical applications, the default settings result in the expected behavior.
</p>
<!--l. 4596--><p class="noindent"><span class="likeparagraphHead"><a id="x1-1030005.2"></a>Warning on rotating frames of reference and periodic boundary conditions.</span>
<span class="t1-zi4r-0-x-x-109">rotateToReference </span>affects coordinates that depend on minimum-image distances in periodic boundary
conditions (PBC). After rotation of the coordinates, the periodic cell vectors become irrelevant: the rotated system is
effectively non-periodic. A safe way to handle this is to ensure that the relevant inter-group distance
vectors remain smaller than the half-size of the periodic cell. If this is not desirable, one should avoid
the rotating frame of reference, and apply orientational restraints to the reference group instead, in
order to keep the orientation of the reference group consistent with the orientation of the periodic
cell.
</p>
<!--l. 4602--><p class="noindent"><span class="likeparagraphHead"><a id="x1-1040005.2"></a>Warning on rotating frames of reference and ABF.</span>
   Note that <span class="t1-zi4r-0-x-x-109">centerToReference </span>and <span class="t1-zi4r-0-x-x-109">rotateToReference </span>may affect the Jacobian derivative of colvar
components in a way that is not taken into account by default. Be careful when using these options in ABF
simulations or when using total force values.
</p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-104001"></a><span class="colorbox" id="colorbox233"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">centerToReference</span></span> <!--l. 4618--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Implicitly
     remove translations for this group<!--l. 4618--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>atom group <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">off</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>If this option is <span class="t1-zi4r-0-x-x-109">on</span>, the center of geometry of the group will be aligned with that of the
     reference positions provided by either <span class="t1-zi4r-0-x-x-109">refPositions </span>or <span class="t1-zi4r-0-x-x-109">refPositionsFile</span>. Colvar components will
     only have access to the aligned positions. <span class="ptmb8t-x-x-109">Note</span>: unless otherwise specified, <span class="t1-zi4r-0-x-x-109">rmsd </span>and <span class="t1-zi4r-0-x-x-109">eigenvector</span>
     set this option to <span class="t1-zi4r-0-x-x-109">on </span><span class="ptmri8t-x-x-109">by default</span>.
     </li>
<li class="itemize"><a id="dx1-104002"></a><span class="colorbox" id="colorbox234"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">centerToOrigin</span></span> <!--l. 4631--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Implicitly
     remove translations for this group by keeping its center at the origin<!--l. 4631--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>atom group <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">off</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>This option implies <span class="t1-zi4r-0-x-x-109">centerToReference</span>. If this option is <span class="t1-zi4r-0-x-x-109">on</span>, coordinates from the
     group will be translated so that the center of geometry of the group remains at (0, 0, 0), except if
                                                                                         
                                                                                         
     <span class="t1-zi4r-0-x-x-109">fittingGroup </span>is enabled. In that case, the translation applied is the translation that brings the center
     of geometry of the fitting group to (0, 0, 0).
     </li>
<li class="itemize"><a id="dx1-104003"></a><span class="colorbox" id="colorbox235"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">rotateToReference</span></span> <!--l. 4645--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Implicitly
     remove rotations for this group<!--l. 4645--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>atom group <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">off</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>If this option is <span class="t1-zi4r-0-x-x-109">on</span>, the coordinates of this group will be optimally superimposed to
     the reference positions provided by either <span class="t1-zi4r-0-x-x-109">refPositions </span>or <span class="t1-zi4r-0-x-x-109">refPositionsFile</span>. The rotation will be
     performed around the center of geometry if <span class="t1-zi4r-0-x-x-109">centerToReference </span>is <span class="t1-zi4r-0-x-x-109">on</span>, or around the origin otherwise.
     The algorithm used is the same employed by the <span class="t1-zi4r-0-x-x-109">orientation </span>colvar component <span class="cite">[<a href="#XCoutsias2004">4</a>]</span>. Forces applied
     to the atoms of this group will also be implicitly rotated back to the original frame. <span class="ptmb8t-x-x-109">Note</span>: unless
     otherwise specified, <span class="t1-zi4r-0-x-x-109">rmsd </span>and <span class="t1-zi4r-0-x-x-109">eigenvector </span>set this option to <span class="t1-zi4r-0-x-x-109">on </span><span class="ptmri8t-x-x-109">by default</span>.
     </li>
<li class="itemize"><a id="atom-group|refPositions"> </a> <a id="dx1-104004"></a><span class="colorbox" id="colorbox236"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositions</span></span> <!--l. 4657--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Reference
     positions for fitting (Å)<!--l. 4657--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>atom group <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>space-separated list of <span class="t1-zi4r-0-x-x-109">(x, y, z) </span>triplets <br class="newline"/><span class="ptmb8t-x-x-109">Description:    </span>This option provides a list of reference coordinates for <span class="t1-zi4r-0-x-x-109">centerToReference </span>and/or
     <span class="t1-zi4r-0-x-x-109">rotateToReference</span>, and is mutually exclusive with <span class="t1-zi4r-0-x-x-109">refPositionsFile</span>. If only <span class="t1-zi4r-0-x-x-109">centerToReference</span>
     is <span class="t1-zi4r-0-x-x-109">on</span>, the list may contain a single (x, y, z) triplet; if also <span class="t1-zi4r-0-x-x-109">rotateToReference </span>is <span class="t1-zi4r-0-x-x-109">on</span>, the list should
     be as long as the atom group, and <span class="ptmri8t-x-x-109">its order must match the order in which atoms were defined</span>.
     </li>
<li class="itemize"><a id="atom-group|refPositionsFile"> </a> <a id="dx1-104005"></a><span class="colorbox" id="colorbox237"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositionsFile</span></span> <!--l. 4670--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>File
     containing the reference positions for fitting<!--l. 4670--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>atom group <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>UNIX filename <br class="newline"/><span class="ptmb8t-x-x-109">Description:    </span>This option provides a list of reference coordinates for <span class="t1-zi4r-0-x-x-109">centerToReference </span>and/or
     <span class="t1-zi4r-0-x-x-109">rotateToReference</span>, and is mutually exclusive with <span class="t1-zi4r-0-x-x-109">refPositions</span>. The acceptable file format is
     XYZ (see <a href="#x1-250003.8.3">3.8.3<!--tex4ht:ref: sec:colvars_xyz_format --></a>), which is read in double precision. Alternatively, a PDB file may be read (see <a href="#x1-260003.8.4">3.8.4<!--tex4ht:ref: sec:colvars_pdb_format --></a>)
     using NAMD's reader; however, due to the constraints of the PDB format <span class="ptmri8t-x-x-109">PDB files are discouraged</span>
<span class="ptmri8t-x-x-109">if the precision of the reference coordinates is a concern</span>.
     </li>
<li class="itemize"><a id="atom-group|refPositionsCol"> </a> <a id="dx1-104006"></a><span class="colorbox" id="colorbox238"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositionsCol</span></span> <!--l. 4691--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>PDB
     column containing atom flags<!--l. 4691--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>atom group <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span><span class="t1-zi4r-0-x-x-109">O</span>, <span class="t1-zi4r-0-x-x-109">B</span>, <span class="t1-zi4r-0-x-x-109">X</span>, <span class="t1-zi4r-0-x-x-109">Y</span>, or <span class="t1-zi4r-0-x-x-109">Z</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Like <span class="t1-zi4r-0-x-x-109">atomsCol </span>for <span class="t1-zi4r-0-x-x-109">atomsFile</span>, indicates which column to use to identify the atoms in
     <span class="t1-zi4r-0-x-x-109">refPositionsFile </span>(if this is a PDB file).
     </li>
<li class="itemize"><a id="atom-group|refPositionsColValue"> </a> <a id="dx1-104007"></a><span class="colorbox" id="colorbox239"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refPositionsColValue</span></span> <!--l. 4691--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Atom
     selection flag in the PDB column<!--l. 4691--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>atom group <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive decimal <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Analogous to <span class="t1-zi4r-0-x-x-109">atomsColValue</span>, but applied to <span class="t1-zi4r-0-x-x-109">refPositionsCol</span>.
     </li>
<li class="itemize"><a id="atom-group|fittingGroup"> </a> <a id="dx1-104008"></a><span class="colorbox" id="colorbox240"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">fittingGroup</span></span> <!--l. 4704--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Use
     an alternate set of atoms to define the roto-translation<!--l. 4704--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>atom group <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>Block <span class="t1-zi4r-0-x-x-109">fittingGroup { ... }</span><br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span>This atom group itself <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>If either <span class="t1-zi4r-0-x-x-109">centerToReference </span>or <span class="t1-zi4r-0-x-x-109">rotateToReference </span>is defined, this keyword defines
     an alternate atom group to calculate the optimal roto-translation. Use this option to define a continuous
     rotation if the structure of the group involved changes significantly (a typical symptom would be the
     message “Warning: discontinuous rotation!"). <span class="ptmb8t-x-x-109">Performance considerations: </span>note that enabling this
     option will result in projecting each of the atomic gradients of the colvar (e.g. the RMSD) onto each
     the gradients of the roto-translation, which may be a computationally expensive operation: see the
     closely related <a href="#atom-group|enableFitGradients"><span class="t1-zi4r-0-x-x-109">enableFitGradients</span></a> for details.
     </li></ul>
<!--l. 4735--><p class="indent">   The following example illustrates the use of <span class="t1-zi4r-0-x-x-109">fittingGroup </span>as part of a Distance to Bound Configuration
(DBC) coordinate for use in ligand restraints for binding affinity calculations.<span class="cite">[<a href="#XSalari2018">15</a>]</span> The group called
“<span class="t1-zi4r-0-x-x-109">atoms</span>" describes coordinates of a ligand's atoms, expressed in a moving frame of reference tied to a
binding site (here within a protein). An optimal roto-translation is calculated automatically by fitting the
C<!--l. 4735--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow></mrow><mrow><mi>α</mi></mrow></msub></math> trace
of the rest of the protein onto the coordinates provided by a PDB file. To define a DBC coordinate, this atom group
would be used within an <a href="#sec:cvc_rmsd"><span class="t1-zi4r-0-x-x-109">rmsd</span></a> function.</p>
<div class="mdframed" id="mdframed-50">
<!--l. 4735--><p class="noindent"></p><pre># Example: defining a group "atoms" (the ligand) whose coordinates are expressed <br class="newline"/># in a roto-translated frame of reference defined by a second group (the receptor)<br class="newline"/>atoms {<br class="newline"/><br class="newline"/>  atomNumbers 1 2 3 4 5 6 7 # atoms of the ligand (1-based)<br class="newline"/><br class="newline"/>  centerToReference yes<br class="newline"/>  rotateToReference yes<br class="newline"/>  fittingGroup {<br class="newline"/>    # define the frame by fitting alpha carbon atoms<br class="newline"/>    # in 2 protein segments close to the site<br class="newline"/>    psfSegID PROT PROT<br class="newline"/>    atomNameResidueRange CA 1-40<br class="newline"/>    atomNameResidueRange CA 59-100<br class="newline"/>  } <br class="newline"/>  refPositionsFile all.pdb # can be the entire system<br class="newline"/>} </pre>
</div>
<!--l. 4737--><p class="indent">   The following two options have default values appropriate for the vast majority of applications, and are only
provided to support rare, special cases. </p>
<ul class="itemize1">
<li class="itemize"><a id="atom-group|enableFitGradients"> </a> <a id="dx1-104009"></a><span class="colorbox" id="colorbox241"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">enableFitGradients</span></span> <!--l. 4756--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Include
     the roto-translational contribution to colvar gradients<!--l. 4756--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>atom group <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">on</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>When either <span class="t1-zi4r-0-x-x-109">centerToReference </span>or <span class="t1-zi4r-0-x-x-109">rotateToReference </span>is on, the gradients of some
     colvars include terms proportional to <!--l. 4756--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>∂</mi><mi>R</mi><mo class="MathClass-bin">∕</mo><mi>∂</mi><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>i</mi></mrow></msub></math>
     (rotational gradients) and <!--l. 4756--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>∂</mi><msup><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi mathvariant="normal">C</mi></mrow></msup><mo class="MathClass-bin">∕</mo><mi>∂</mi><msub><mrow><mi mathvariant="bold">x</mi></mrow><mrow><mi>i</mi></mrow></msub></math>
     (translational gradients). By default, these terms are calculated and included in the total gradients; if
     this option is set to <span class="t1-zi4r-0-x-x-109">off</span>, they are neglected. In the case of a minimum RMSD component, this flag is
     automatically disabled because the contributions of those derivatives to the gradients cancel out; other
     types of variable will require projecting each of the gradients of the variable onto each of the gradients
     of the roto-translation (i.e. a <!--l. 4756--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>O</mi><mrow><mo class="MathClass-open">(</mo><mrow><msup><mrow><mi>N</mi></mrow><mrow><mn>2</mn></mrow></msup></mrow><mo class="MathClass-close">)</mo></mrow></math>
     loop). When <a href="#atom-group|fittingGroup"><span class="t1-zi4r-0-x-x-109">fittingGroup</span></a> is enabled, the computation is a <!--l. 4756--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>O</mi><mrow><mo class="MathClass-open">(</mo><mrow><mi>N</mi> <mo class="MathClass-bin">×</mo><mi>M</mi></mrow><mo class="MathClass-close">)</mo></mrow></math>
     loop for all variables, including RMSDs.
     </li></ul>
<div class="flushright">
<!--l. 4761--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 4761--><p class="noindent"></p><div class="heading"><a id="sec:colvar_atom_groups_wrapping">
<h4 class="subsectionHead"><span class="titlemark">5.3    </span> <a id="x1-1050005.3"></a>Treatment of periodic boundary conditions.</h4>
<!--l. 4761--><p class="noindent"></p></a> </div> 
<!--l. 4777--><p class="indent">   In simulations with periodic boundary conditions, NAMD maintains the coordinates of all the atoms within a
molecule contiguous to each other (i.e. there are no spurious “jumps" in the molecular bonds). The Colvars module
relies on this when calculating a group's center of geometry, but this condition may fail if the group spans different
molecules. In that case, writing the NAMD output and restart files using <span class="t1-zi4r-0-x-x-109">wrapAll </span>or <span class="t1-zi4r-0-x-x-109">wrapWater </span>could
produce wrong results when a simulation run is continued from a previous one. The user should then
determine, according to which type of colvars are being calculated, whether <span class="t1-zi4r-0-x-x-109">wrapAll </span>or <span class="t1-zi4r-0-x-x-109">wrapWater </span>can be
enabled.
</p><!--l. 4777--><p class="indent">   In general, internal coordinate wrapping by NAMD does not affect the calculation of colvars if each atom group
satisfies one or more of the following:
</p><!--l. 4792--><p class="indent">
</p><ol class="enumerate1">
<li class="enumerate" id="x1-105001x5.3">it is composed by only one atom;
     </li>
<li class="enumerate" id="x1-105002x5.3">it is used by a colvar component which does not make use of its center of geometry, but only of
     pairwise distances (<span class="t1-zi4r-0-x-x-109">distanceInv</span>, <span class="t1-zi4r-0-x-x-109">coordNum</span>, <span class="t1-zi4r-0-x-x-109">hBond</span>, <span class="t1-zi4r-0-x-x-109">alpha</span>, <span class="t1-zi4r-0-x-x-109">dihedralPC</span>);
     </li>
<li class="enumerate" id="x1-105003x5.3">it is used by a colvar component that ignores the ill-defined Cartesian components of its center of mass
     (such as the <!--l. 4795--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>x</mi></math>
     and <!--l. 4795--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>y</mi></math>
     components of a membrane's center of mass modeled with <span class="t1-zi4r-0-x-x-109">distanceZ</span>);
                                                                                         
                                                                                         
     </li>
<li class="enumerate" id="x1-105004x5.3">it has all of its atoms within the same molecular fragment.</li></ol>
<div class="flushright">
<!--l. 4802--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 4802--><p class="noindent"></p><div class="heading"><a id="sec:colvar_atom_groups_scaling">
<h4 class="subsectionHead"><span class="titlemark">5.4    </span> <a id="x1-1060005.4"></a>Performance of a Colvars calculation based on group size.</h4>
<!--l. 4802--><p class="noindent"></p></a> </div> 
<!--l. 4804--><p class="indent">   In simulations performed with message-passing programs (such as NAMD or LAMMPS), the calculation of
energy and forces is distributed (i.e., parallelized) across multiple nodes, as well as over the processor cores of each
node. When Colvars is enabled, certain atomic coordinates are collected on a single node, where the
calculation of collective variables and of their biases is executed. This means that for simulations over large
numbers of nodes, a Colvars calculation may produce a significant overhead, coming from the costs of
transmitting atomic coordinates to one node and of processing them. The latency-tolerant design and dynamic
load balancing of NAMD may alleviate both factors, but a noticeable performance impact may be
observed.
</p><!--l. 4809--><p class="indent">   Performance can be improved in multiple ways: </p>
<ul class="itemize1">
<li class="itemize">The calculation of variables, components and biases can be distributed over the processor cores of the
     node where the Colvars module is executed. Currently, an equal weight is assigned to each colvar,
     or to each component of those colvars that include more than one component. The performance of
     simulations that use many colvars or components is improved automatically. For simulations that use
     a single large colvar, it may be advisable to partition it in multiple components, which will be then
     distributed across the available cores. In NAMD, this feature is enabled in all binaries compiled using
     SMP builds of Charm++ with the CkLoop extension.   If printed, the message “SMP parallelism is
     available." indicates the availability of the option. If available, the option is turned on by default, but
     may be disabled using the keyword <a href="#Colvars-global|smp"><span class="t1-zi4r-0-x-x-109">smp</span></a> if required for debugging.
     </li>
<li class="itemize">NAMD also offers a parallelized calculation of the centers of mass of groups of atoms. This option is
     on by default for all components that are simple functions of centers of mass, and is controlled by the
     keyword <a href="#sec:cvc_common"><span class="t1-zi4r-0-x-x-109">scalable</span></a>. When supported, the message “Will enable scalable calculation for group …" is
     printed for each group.
     </li>
<li class="itemize">As a general rule, the size of atom groups should be kept relatively small (up to a few thousands
     of  atoms,  depending  on  the  size  of  the  entire  system  in  comparison).  To  gain  an  estimate  of  the
     computational cost of a large colvar, one can use a test calculation of the same colvar in VMD (hint:
     use the <span class="t1-zi4r-0-x-x-109">time </span>Tcl command to measure the cost of running <span class="t1-zi4r-0-x-x-109">cv update</span>).</li></ul>
<!--l. 4833--><p class="indent">
</p>
<div class="flushright">
<!--l. 4833--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 4833--><p class="noindent"></p><div class="heading"><a id="sec:colvarbias">
<h3 class="sectionHead"><span class="titlemark">6    </span> <a id="x1-1070006"></a>Biasing and analysis methods</h3>
<!--l. 4833--><p class="noindent"></p></a> </div> 
<!--l. 4835--><p class="indent">   A biasing or analysis method can be applied to existing collective variables by using the following
configuration:</p>
<div class="mdframed" id="mdframed-51">
<!--l. 4837--><p class="noindent"></p><pre>
<!--l. 4839--><p class="noindent"><!--l. 4839--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>biastype<!--l. 4839--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math> {<br class="newline"/>  name <!--l. 4840--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>name<!--l. 4840--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>
<br class="newline"/>  colvars <!--l. 4841--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>xi1<!--l. 4841--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math> <!--l. 4841--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>xi2<!--l. 4841--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math> ...<br class="newline"/>  <!--l. 4842--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>parameters<!--l. 4842--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><br class="newline"/>} </p></pre>
</div>
<!--l. 4846--><p class="noindent">The keyword <!--l. 4846--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="t1-zi4r-0-x-x-109">biastype</span><!--l. 4846--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>
indicates the method of choice. There can be multiple instances of the same method, e.g. using multiple <span class="t1-zi4r-0-x-x-109">harmonic</span>
blocks allows defining multiple restraints.
</p><!--l. 4849--><p class="indent">   All biasing and analysis methods implemented recognize the following options: </p>
<ul class="itemize1">
<li class="itemize"><a id="colvarbias|name"> </a> <a id="dx1-107001"></a><span class="colorbox" id="colorbox242"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">name</span></span> <!--l. 4862--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Identifier
     for the bias<!--l. 4862--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>colvar bias <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>string <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><!--l. 4862--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="t1-zi4r-0-x-x-109">type</span>
<span class="t1-zi4r-0-x-x-109">of bias</span><!--l. 4862--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo><mo class="MathClass-rel">&lt;</mo></math><span class="t1-zi4r-0-x-x-109">bias</span>
<span class="t1-zi4r-0-x-x-109">index</span><!--l. 4862--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>
<br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>This string is used to identify the bias or analysis method in the output, and to name
     some output files. <span class="ptmb8t-x-x-109">Tip: </span>although a unique name is assigned automatically, <span class="ptmri8t-x-x-109">you are strongly encouraged</span>
<span class="ptmri8t-x-x-109">to give a name to a bias that you may want to analyze later</span>. For example, processing a trajectory
     file (sec. <a href="#x1-280003.8.6">3.8.6<!--tex4ht:ref: sec:colvars_traj_format --></a>) containing a harmonic restraint is much simpler with a clearly identifiable name, like
     “<span class="t1-zi4r-0-x-x-109">smd</span>" for a moving harmonic restraint or “<span class="t1-zi4r-0-x-x-109">us</span>" for a static one.
     </li>
<li class="itemize"><a id="colvarbias|colvars"> </a> <a id="dx1-107002"></a><span class="colorbox" id="colorbox243"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">colvars</span></span> <!--l. 4871--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Collective
     variables involved<!--l. 4871--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>colvar bias <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>space-separated list of colvar names <br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>This option selects by name all the variables to which this bias or analysis will be
     applied.
     </li>
<li class="itemize"><a id="colvarbias|outputEnergy"> </a> <a id="dx1-107003"></a><span class="colorbox" id="colorbox244"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">outputEnergy</span></span> <!--l. 4883--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Write
     the current bias energy to the trajectory file<!--l. 4883--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>colvar bias <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">off</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>If this option is chosen and <span class="t1-zi4r-0-x-x-109">colvarsTrajFrequency </span>is not zero, the current value of
     the biasing energy will be written to the trajectory file during the simulation. The total energy of all
     Colvars biases is also reported by NAMD, as part of the MISC title.
     </li>
<li class="itemize"><a id="colvarbias|outputFreq"> </a> <a id="dx1-107004"></a><span class="colorbox" id="colorbox245"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">outputFreq</span></span> <!--l. 4895--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Frequency
     (number of steps) at which output files are written<!--l. 4895--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>colvar bias <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive integer <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><a href="#Colvars-global|colvarsRestartFrequency"><span class="t1-zi4r-0-x-x-109">colvarsRestartFrequency</span></a><br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>If this bias produces aggregated data that needs to be written to disk (for example, a
     PMF), this number specifies the number of steps after which these data are written to files. A value
     of zero disables writing files for this bias during the simulation (except for <a href="#colvarbias|outputEnergy"><span class="t1-zi4r-0-x-x-109">outputEnergy</span></a>, which is
     controlled by <a href="#Colvars-global|colvarsTrajFrequency"><span class="t1-zi4r-0-x-x-109">colvarsTrajFrequency</span></a>). All output files are also written at the end of a simulation
     run, regardless of the value of this number.
     </li>
<li class="itemize"><a id="colvarbias|bypassExtendedLagrangian"> </a> <a id="dx1-107005"></a><span class="colorbox" id="colorbox246"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">bypassExtendedLagrangian</span></span> <!--l. 4910--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Apply
     bias to actual colvars, bypassing extended coordinates<!--l. 4910--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>colvar bias <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">off</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>This option is implemented by the <a href="#sec:colvarbias_harmonic_walls"><span class="t1-zi4r-0-x-x-109">harmonicWalls</span></a> and <a href="#sec:colvarbias_histogram"><span class="t1-zi4r-0-x-x-109">histogram</span></a> biases. It is only
     relevant  if  the  bias  is  applied  to  one  or  several  extended-Lagrangian  colvars  (<a href="#x1-960004.21">4.21<!--tex4ht:ref: sec:colvar_extended --></a>),  for  example
     within an eABF (<a href="#x1-1150006.3">6.3<!--tex4ht:ref: sec:eABF --></a>) simulation. Usually, biases use the value of the extended coordinate as a proxy
     for  the  actual  colvar,  and  their  biasing  forces  are  applied  to  the  extended  coordinates  as  well.  If
     <span class="t1-zi4r-0-x-x-109">bypassExtendedLagrangian </span>is enabled, the bias behaves as if there were no extended coordinates,
     and accesses the value of the underlying colvars, applying any biasing forces along the gradients of
     those variables.
     </li>
<li class="itemize"><a id="colvarbias|stepZeroData"> </a> <a id="dx1-107006"></a><span class="colorbox" id="colorbox247"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">stepZeroData</span></span> <!--l. 4925--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Accumulate
     data starting at step 0 of a simulation run<!--l. 4925--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>colvar bias <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">off</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>This option is meaningful for biases that record and accumulate data during a simulation,
     such as ABF (<a href="#x1-1090006.2">6.2<!--tex4ht:ref: sec:colvarbias_abf --></a>), metadynamics (<a href="#x1-1190006.4">6.4<!--tex4ht:ref: sec:colvarbias_meta --></a>), histograms (<a href="#x1-1350006.10">6.10<!--tex4ht:ref: sec:colvarbias_histogram --></a>) and in general any bias that accumulates
     free-energy samples with thermodynamic integration, or TI (<a href="#x1-1080006.1">6.1<!--tex4ht:ref: sec:colvarbias_ti --></a>). When this option is disabled (default),
     data will only be recorded into the bias after the first coordinate update: this is generally the correct
                                                                                         
                                                                                         
     choice in simulation runs. Biasing energy and forces will always be computed for all active biases,
     regardless of this option. Note that in some cases the bias may require data from previous simulation
     steps: for example, TI requires total atomic forces (see <a href="#colvar|outputTotalForce"><span class="t1-zi4r-0-x-x-109">outputTotalForce</span></a>) which are only available
     at the following step in NAMD; turning on this flag in those cases will raise an error.
     </li>
<li class="itemize"><a id="colvarbias|scaledBiasingForce"> </a> <a id="dx1-107007"></a><span class="colorbox" id="colorbox248"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">scaledBiasingForce</span></span> <!--l. 4931--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Scale
     biasing force by a factor in an external histogram<!--l. 4931--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>colvar bias <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">off</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>If this option is set to <span class="t1-zi4r-0-x-x-109">on</span>, the biasing force at each step will be scaled by a factor provided
     in the grid of an external histogram. The histogram file is provided by <span class="t1-zi4r-0-x-x-109">scaledBiasingForceFactorsGrid</span>.
     It ought to be noted that if the variables are not in any grids of the histogram, the scaling factor is 1.0.
     </li>
<li class="itemize"><a id="colvarbias|scaledBiasingForceFactorsGrid"> </a> <a id="dx1-107008"></a><span class="colorbox" id="colorbox249"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">scaledBiasingForceFactorsGrid</span></span> <!--l. 4937--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>A
     histogram file with the scaling factor of biasing force in each bin<!--l. 4937--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>colvar bias <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>string <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>If <span class="t1-zi4r-0-x-x-109">scaledBiasingForce </span>is set to <span class="t1-zi4r-0-x-x-109">on</span>, this option accepts the filename of the histogram
     file that contains the scaling factors. The format of the histogram file is the same as the PMF files
     written by <span class="t1-zi4r-0-x-x-109">metadynamics </span>or <span class="t1-zi4r-0-x-x-109">ABF</span>.
     </li></ul>
<div class="flushright">
<!--l. 4942--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 4942--><p class="noindent"></p><div class="heading"><a id="sec:colvarbias_ti">
<h4 class="subsectionHead"><span class="titlemark">6.1    </span> <a id="x1-1080006.1"></a>Thermodynamic integration</h4>
<!--l. 4942--><p class="noindent"></p></a> </div> 
<!--l. 4944--><p class="indent">   The methods implemented here provide a variety of estimators of conformational free-energies.
These are carried out at run-time, or with the use of post-processing tools over the generated output
files. The specifics of each estimator are discussed in the documentation of each biasing or analysis
method.
</p><!--l. 4948--><p class="indent">   A special case is the traditional thermodynamic integration (TI) method, used for example to compute potentials
of mean force (PMFs). Most types of restraints (<a href="#x1-1270006.5">6.5<!--tex4ht:ref: sec:colvarbias_harmonic --></a>, <a href="#x1-1320006.7">6.7<!--tex4ht:ref: sec:colvarbias_harmonic_walls --></a>, <a href="#x1-1330006.8">6.8<!--tex4ht:ref: sec:colvarbias_linear --></a>, ...) as well as metadynamics (<a href="#x1-1190006.4">6.4<!--tex4ht:ref: sec:colvarbias_meta --></a>) can optionally use TI
alongside their own estimator, based on the keywords documented below.
</p>
<ul class="itemize1">
<li class="itemize"><a id="colvarbias|writeTIPMF"> </a> <a id="dx1-108001"></a><span class="colorbox" id="colorbox250"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">writeTIPMF</span></span> <!--l. 4964--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Write
                                                                                         
                                                                                         
     the PMF computed by thermodynamic integration<!--l. 4964--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>colvar bias <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">off</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>If  the  bias  is  applied  to  a  variable  that  supports  the  calculation  of  total  forces  (see
     <a href="#sec:colvar"><span class="t1-zi4r-0-x-x-109">outputotalForce</span></a> and <a href="#x1-900004.15.1">4.15.1<!--tex4ht:ref: sec:cvc_sys_forces --></a>), this option allows calculating the corresponding PMF by thermodynamic
     integration, and writing it to the file <span class="ptmri8t-x-x-109">outputName</span><span class="t1-zi4r-0-x-x-109">.</span><!--l. 4964--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="t1-zi4r-0-x-x-109">name</span><!--l. 4964--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><span class="t1-zi4r-0-x-x-109">.ti.pmf</span>,
     where <!--l. 4964--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="t1-zi4r-0-x-x-109">name</span><!--l. 4964--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>
     is the name of the bias and the contents of the file are in multicolumn text format (<a href="#x1-270003.8.5">3.8.5<!--tex4ht:ref: sec:colvar_multicolumn_grid --></a>). The total
     force includes the forces applied to the variable by all bias, except those from this bias itself. If any
     bias applies time-dependent forces besides the one using this option, an error is raised.
     </li>
<li class="itemize"><a id="colvarbias|writeTISamples"> </a> <a id="dx1-108002"></a><span class="colorbox" id="colorbox251"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">writeTISamples</span></span> <!--l. 4980--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Write
     the free-energy gradient samples<!--l. 4980--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>colvar bias <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">off</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>This option allows to compute total forces for use with thermodynamic integration as
     done by the keyword <a href="#sec:colvarbias"><span class="t1-zi4r-0-x-x-109">writeTIPMF</span></a>. The names of the files containing the variables' histogram and
     mean thermodynamic forces are <span class="ptmri8t-x-x-109">outputName</span><span class="t1-zi4r-0-x-x-109">.</span><!--l. 4980--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="t1-zi4r-0-x-x-109">name</span><!--l. 4980--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><span class="t1-zi4r-0-x-x-109">.ti.count</span>
     and <span class="ptmri8t-x-x-109">outputName</span><span class="t1-zi4r-0-x-x-109">.</span><!--l. 4980--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="t1-zi4r-0-x-x-109">name</span><!--l. 4980--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><span class="t1-zi4r-0-x-x-109">.ti.force</span>,
     respectively: these can be used by <span class="t1-zi4r-0-x-x-109">abf_integrate </span>(see <a href="#x1-1140006.2.5">6.2.5<!--tex4ht:ref: sec:colvarbias_abf_post --></a>) or similar utility. Note that because the
     <span class="t1-zi4r-0-x-x-109">.force </span>file contains mean forces instead of free-energy gradients, <span class="t1-zi4r-0-x-x-109">abf_integrate </span><!--l. 4980--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="t1-zi4r-0-x-x-109">filename</span><!--l. 4980--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>
<span class="t1-zi4r-0-x-x-109">-s -1.0 </span>should be used. This option is on by default when <span class="t1-zi4r-0-x-x-109">writeTIPMF </span>is on, but can be enabled
     separately if the bias is applied to more than one variable, making not possible the direct integration
     of the PMF at runtime. If any bias applies time-dependent forces besides the one using this option, an
     error is raised.
     </li></ul>
<!--l. 4984--><p class="indent">   In adaptive biasing force (ABF) (<a href="#x1-1090006.2">6.2<!--tex4ht:ref: sec:colvarbias_abf --></a>) the above keywords are not recognized, because their functionality is
either included already (conventional ABF) or not available (extended-system ABF).
</p>
<div class="flushright">
<!--l. 4988--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 4988--><p class="noindent"></p><div class="heading"><a id="sec:colvarbias_abf">
<h4 class="subsectionHead"><span class="titlemark">6.2    </span> <a id="x1-1090006.2"></a>Adaptive Biasing Force</h4>
<!--l. 4988--><p class="noindent"></p></a> </div> 
<!--l. 4990--><p class="indent">   For a full description of the Adaptive Biasing Force method, see reference <span class="cite">[<a href="#XDarve2008">16</a>]</span>. For details about this
implementation, see references <span class="cite">[<a href="#XHenin2004">17</a>]</span> and <span class="cite">[<a href="#XHenin2010">18</a>]</span>. <span class="ptmb8t-x-x-109">When publishing research that makes use of this functionality,</span>
<span class="ptmb8t-x-x-109">please cite references</span><span class="ptmb8t-x-x-109"> </span><span class="cite"><span class="ptmb8t-x-x-109">[</span><a href="#XDarve2008"><span class="ptmb8t-x-x-109">16</span></a><span class="ptmb8t-x-x-109">]</span></span> <span class="ptmb8t-x-x-109">and </span><span class="cite"><span class="ptmb8t-x-x-109">[</span><a href="#XHenin2010"><span class="ptmb8t-x-x-109">18</span></a><span class="ptmb8t-x-x-109">]</span></span><span class="ptmb8t-x-x-109">.</span>
</p><!--l. 4996--><p class="indent">   An alternate usage of this feature is the application of custom tabulated biasing potentials to one or more colvars.
See <span class="t1-zi4r-0-x-x-109">inputPrefix </span>and <span class="t1-zi4r-0-x-x-109">updateBias </span>below.
</p><!--l. 5000--><p class="indent">   Combining ABF with the extended Lagrangian feature (<a href="#x1-960004.21">4.21<!--tex4ht:ref: sec:colvar_extended --></a>) of the variables produces the extended-system
ABF variant of the method (<a href="#x1-1150006.3">6.3<!--tex4ht:ref: sec:eABF --></a>).
</p><!--l. 5004--><p class="indent">   ABF is based on the thermodynamic integration (TI) scheme for computing
free energy profiles. The free energy as a function of a set of collective variables
<!--l. 5006--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle> <mo class="MathClass-rel">=</mo> <msub><mrow><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></mrow><mrow><mi>i</mi><mo class="MathClass-rel">∈</mo><mrow><mo class="MathClass-open">[</mo><mrow><mn>1</mn><mo class="MathClass-punc">,</mo><mi>n</mi></mrow><mo class="MathClass-close">]</mo></mrow></mrow></msub></math> is defined from the
canonical distribution of <!--l. 5007--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></math>,
<!--l. 5007--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="bold-script">𝒫</mi><mrow><mo class="MathClass-open">(</mo><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></mrow><mo class="MathClass-close">)</mo></mrow></math>:
</p>
<table class="equation"><tr><td> <a id="x1-109001r24"></a>
<!--l. 5009--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>A</mi><mrow><mo class="MathClass-open">(</mo><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></mrow><mo class="MathClass-close">)</mo></mrow> <mo class="MathClass-rel">=</mo> <mo class="MathClass-bin">−</mo><mfrac><mrow><mn>1</mn></mrow>
<mrow><mi>β</mi></mrow></mfrac><mo class="qopname">ln</mo><!--nolimits--><mi mathvariant="bold-script">𝒫</mi><mrow><mo class="MathClass-open">(</mo><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-bin">+</mo><msub><mrow><mi>A</mi></mrow><mrow><mn>0</mn></mrow></msub>
</math></td><td class="eq-no">(24)</td></tr></table>
<!--l. 5014--><p class="indent">   In the TI formalism, the free energy is obtained from its gradient, which is generally calculated in the form of the average
of a force <!--l. 5016--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>F</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mi>ξ</mi></mrow></msub></math>
exerted on <!--l. 5016--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></math>, taken
over an iso-<!--l. 5016--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></math>
surface:
</p>
<table class="equation"><tr><td> <a id="x1-109002r25"></a>
<!--l. 5018--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow>
<mstyle class="mbox"><mtext></mtext><mstyle class="math"><mo class="MathClass-op">∇</mo></mstyle><mtext></mtext></mstyle></mrow><mrow><mi>ξ</mi></mrow></msub><mi>A</mi><mrow><mo class="MathClass-open">(</mo><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></mrow><mo class="MathClass-close">)</mo></mrow> <mo class="MathClass-rel">=</mo><msub><mrow> <mfenced close="⟩" open="⟨" separators=""><mrow><mo class="MathClass-bin">−</mo><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>F</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mi>ξ</mi></mrow></msub></mrow></mfenced></mrow><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></mrow></msub>
</math></td><td class="eq-no">(25)</td></tr></table>
<!--l. 5023--><p class="indent">   Several formulae that take the form of (<a href="#x1-109002r25">25<!--tex4ht:ref: eq:gradient --></a>) have been proposed. This implementation relies partly on the classic
formulation <span class="cite">[<a href="#XCarter1989">19</a>]</span>, and partly on a more versatile scheme originating in a work by Ruiz-Montero et al. <span class="cite">[<a href="#XRuiz-Montero1997">20</a>]</span>, generalized by
den Otter <span class="cite">[<a href="#XdenOtter2000">21</a>]</span> and extended to multiple variables by Ciccotti et al. <span class="cite">[<a href="#XCiccotti2005">22</a>]</span>. Consider a system subject to constraints of the
form <!--l. 5029--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>σ</mi></mrow><mrow><mi>k</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>x</mi></mstyle><mtext></mtext></mstyle></mrow><mo class="MathClass-close">)</mo></mrow> <mo class="MathClass-rel">=</mo> <mn>0</mn></math>. Let
<!--l. 5030--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>v</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mi>i</mi></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></mrow><mrow><mi>i</mi><mo class="MathClass-rel">∈</mo><mrow><mo class="MathClass-open">[</mo><mrow><mn>1</mn><mo class="MathClass-punc">,</mo><mi>n</mi></mrow><mo class="MathClass-close">]</mo></mrow></mrow></msub></math> be arbitrarily chosen
vector fields (<!--l. 5031--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow><mi>ℝ</mi></mrow><mrow><mn>3</mn><mi>N</mi></mrow></msup> <mo class="MathClass-rel">→</mo> <msup><mrow><mi>ℝ</mi></mrow><mrow><mn>3</mn><mi>N</mi></mrow></msup></math>)
verifying, for all <!--l. 5031--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math>,
<!--l. 5032--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>j</mi></math>, and
                                                                                         
                                                                                         
<!--l. 5032--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>k</mi></math>:
</p>
<!--tex4ht:inline--><!--l. 5034--><math display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mtable class="eqnarray" columnalign="right center left">
<mtr><mtd class="eqnarray-1"><msub><mrow> <mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>v</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-bin">⋅</mo><mstyle class="mbox"><mtext></mtext><mstyle class="math"><msub><mrow><mo class="MathClass-op">∇</mo></mrow><mrow><mspace class="thinspace" width="0.3em"></mspace><mspace class="thinspace" width="0.3em"></mspace><mi>x</mi></mrow></msub><mspace class="thinspace" width="0.3em"></mspace></mstyle><mtext></mtext></mstyle><msub><mrow><mi>ξ</mi></mrow><mrow><mi>j</mi></mrow></msub></mtd><mtd class="eqnarray-2"> <mo class="MathClass-rel">=</mo></mtd><mtd class="eqnarray-3"> <msub><mrow><mi>δ</mi></mrow><mrow><mi>i</mi><mi>j</mi></mrow></msub></mtd><mtd class="eqnarray-4"> <mtext class="eqnarray">(26)</mtext><mtext class="label" id="x1-109003r26"></mtext><mtext class="endlabel"></mtext></mtd>
</mtr><mtr><mtd class="eqnarray-1"><msub><mrow> <mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>v</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-bin">⋅</mo><mstyle class="mbox"><mtext></mtext><mstyle class="math"><msub><mrow><mo class="MathClass-op">∇</mo></mrow><mrow><mspace class="thinspace" width="0.3em"></mspace><mspace class="thinspace" width="0.3em"></mspace><mi>x</mi></mrow></msub><mspace class="thinspace" width="0.3em"></mspace></mstyle><mtext></mtext></mstyle><msub><mrow><mi>σ</mi></mrow><mrow><mi>k</mi></mrow></msub></mtd><mtd class="eqnarray-2"> <mo class="MathClass-rel">=</mo></mtd><mtd class="eqnarray-3"> <mn>0</mn> </mtd><mtd class="eqnarray-4"> <mtext class="eqnarray">(27)</mtext><mtext class="label" id="x1-109003r27"></mtext><mtext class="endlabel"></mtext></mtd> </mtr></mtable>
</math>
<!--l. 5039--><p class="nopar">
</p>
<!--l. 5041--><p class="indent">   then the following holds <span class="cite">[<a href="#XCiccotti2005">22</a>]</span>:
</p>
<table class="equation"><tr><td> <a id="x1-109004r28"></a>
<!--l. 5043--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mfrac><mrow><mi>∂</mi><mi>A</mi></mrow>
<mrow><mi>∂</mi><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow></mfrac> <mo class="MathClass-rel">=</mo><msub><mrow> <mfenced close="⟩" open="⟨" separators=""><mrow><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>v</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-bin">⋅</mo><mstyle class="mbox"><mtext></mtext><mstyle class="math"><msub><mrow><mo class="MathClass-op">∇</mo></mrow><mrow><mspace class="thinspace" width="0.3em"></mspace><mspace class="thinspace" width="0.3em"></mspace><mi>x</mi></mrow></msub><mspace class="thinspace" width="0.3em"></mspace></mstyle><mtext></mtext></mstyle><mi>V</mi> <mo class="MathClass-bin">−</mo><msub><mrow><mi>k</mi></mrow><mrow><mi>B</mi></mrow></msub><mi>T</mi> <mstyle class="mbox"><mtext></mtext><mstyle class="math"><msub><mrow><mo class="MathClass-op">∇</mo></mrow><mrow><mspace class="thinspace" width="0.3em"></mspace><mspace class="thinspace" width="0.3em"></mspace><mi>x</mi></mrow></msub><mspace class="thinspace" width="0.3em"></mspace></mstyle><mtext></mtext></mstyle><mo class="MathClass-bin">⋅</mo><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>v</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mi>i</mi></mrow></msub></mrow></mfenced></mrow><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></mrow></msub>
</math></td><td class="eq-no">(28)</td></tr></table>
<!--l. 5049--><p class="indent">   where <!--l. 5049--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>V</mi></math> is the potential
energy function. <!--l. 5050--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>v</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mi>i</mi></mrow></msub></math>
can be interpreted as the direction along which the force acting on variable
<!--l. 5051--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></math> is measured,
whereas the second term in the average corresponds to the geometric entropy contribution that appears as a Jacobian
correction in the classic formalism <span class="cite">[<a href="#XCarter1989">19</a>]</span>. Condition (<a href="#x1-109003r26">26<!--tex4ht:ref: eq:ortho_gradient --></a>) states that the direction along which the total force on
<!--l. 5055--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></math> is measured is orthogonal to
the gradient of <!--l. 5056--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>j</mi></mrow></msub></math>, which means
that the force measured on <!--l. 5056--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></math>
does not act on <!--l. 5057--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>j</mi></mrow></msub></math>.
                                                                                         
                                                                                         
</p><!--l. 5059--><p class="indent">   Equation (<a href="#x1-109003r27">27<!--tex4ht:ref: eq:ortho_constraints --></a>) implies that constraint forces are orthogonal to the directions along which the free energy
gradient is measured, so that the measurement is effectively performed on unconstrained degrees of freedom. In
NAMD, constraints are typically applied to the lengths of bonds involving hydrogen atoms, for example in TIP3P
water molecules (parameter <span class="t1-zi4r-0-x-x-109">rigidBonds</span>).
</p><!--l. 5066--><p class="indent">   In the framework of ABF, <!--l. 5067--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>F</mi></mrow><mrow><mi>ξ</mi></mrow></msub></math> is
accumulated in bins of finite size <!--l. 5067--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>δ</mi><mi>ξ</mi></math>,
thereby providing an estimate of the free energy gradient according to equation (<a href="#x1-109002r25">25<!--tex4ht:ref: eq:gradient --></a>). The biasing force applied
along the collective variables to overcome free energy barriers is calculated as:
</p>
<table class="equation"><tr><td> <a id="x1-109005r29"></a>
<!--l. 5073--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<msup><mrow><mi>F</mi></mrow><mrow><mi>A</mi><mi>B</mi><mi>F</mi> </mrow></msup> <mo class="MathClass-rel">=</mo> <mi>α</mi><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>N</mi></mrow><mrow>
<mi>ξ</mi></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-bin">×</mo><mstyle class="mbox"><mtext></mtext><mstyle class="math"><msub><mrow><mo class="MathClass-op">∇</mo></mrow><mrow><mspace class="thinspace" width="0.3em"></mspace><mspace class="thinspace" width="0.3em"></mspace><mi>x</mi></mrow></msub><mspace class="thinspace" width="0.3em"></mspace></mstyle><mtext></mtext></mstyle><mover accent="false"><mrow><mi>A</mi></mrow><mo class="MathClass-op">̃</mo></mover><mrow><mo class="MathClass-open">(</mo><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></mrow><mo class="MathClass-close">)</mo></mrow>
</math></td><td class="eq-no">(29)</td></tr></table>
<!--l. 5078--><p class="indent">   where <!--l. 5078--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle class="mbox"><mtext></mtext><mstyle class="math"><msub><mrow><mo class="MathClass-op">∇</mo></mrow><mrow><mspace class="thinspace" width="0.3em"></mspace><mspace class="thinspace" width="0.3em"></mspace><mi>x</mi></mrow></msub><mspace class="thinspace" width="0.3em"></mspace></mstyle><mtext></mtext></mstyle><mover accent="false"><mrow><mi>A</mi></mrow><mo class="MathClass-op">̃</mo></mover></math>
denotes the current estimate of the free energy gradient at the current point
<!--l. 5079--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></math> in the collective
variable subspace, and <!--l. 5080--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>α</mi><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>N</mi></mrow><mrow><mi>ξ</mi></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math>
is a scaling factor that is ramped from 0 to 1 as the local number of samples
<!--l. 5081--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>N</mi></mrow><mrow><mi>ξ</mi></mrow></msub></math>
increases to prevent nonequilibrium effects in the early phase of the simulation, when the gradient estimate has a
large variance. See the <span class="t1-zi4r-0-x-x-109">fullSamples </span>parameter below for details.
</p><!--l. 5086--><p class="indent">   As  sampling  of  the  phase  space  proceeds,  the  estimate
<!--l. 5087--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle class="mbox"><mtext></mtext><mstyle class="math"><msub><mrow><mo class="MathClass-op">∇</mo></mrow><mrow><mspace class="thinspace" width="0.3em"></mspace><mspace class="thinspace" width="0.3em"></mspace><mi>x</mi></mrow></msub><mspace class="thinspace" width="0.3em"></mspace></mstyle><mtext></mtext></mstyle><mover accent="false"><mrow><mi>A</mi></mrow><mo class="MathClass-op">̃</mo></mover></math> is
progressively refined. The biasing force introduced in the equations of motion guarantees that in the bin centered
around <!--l. 5089--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></math>,
the forces acting along the selected collective variables average to zero over time. Eventually,
as the undelying free energy surface is canceled by the adaptive bias, evolution of the system along
<!--l. 5092--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></math> is
governed mainly by diffusion. Although this implementation of ABF can in principle be used in arbitrary dimension,
a higher-dimension collective variable space is likely to be difficult to sample and visualize. Most commonly, the
number of variables is one or two, sometimes three.
</p>
<div class="flushright">
<!--l. 5100--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 5100--><p class="noindent"></p><div class="heading"><a id="sec:colvarbias_abf_req">
<h5 class="subsubsectionHead"><span class="titlemark">6.2.1    </span> <a id="x1-1100006.2.1"></a>ABF requirements on collective variables</h5>
<!--l. 5100--><p class="noindent"></p></a> </div> 
<!--l. 5102--><p class="indent">   The following conditions must be met for an ABF simulation to be possible and to produce an accurate estimate
of the free energy profile. Note that these requirements do not apply when using the extended-system ABF method
(<a href="#x1-1150006.3">6.3<!--tex4ht:ref: sec:eABF --></a>).
</p><!--l. 5107--><p class="indent">
</p><ol class="enumerate1">
<li class="enumerate" id="x1-110002x1"><span class="ptmri8t-x-x-109">Only linear combinations </span>of colvar components can be used in ABF calculations.
     </li>
<li class="enumerate" id="x1-110004x2"><span class="ptmri8t-x-x-109">Availability   of   total   forces   </span>is   necessary.   The   following   colvar   components   can   be   used   in
     ABF  calculations:  <span class="t1-zi4r-0-x-x-109">distance</span>,  <span class="t1-zi4r-0-x-x-109">distance_xy</span>,  <span class="t1-zi4r-0-x-x-109">distance_z</span>,  <span class="t1-zi4r-0-x-x-109">angle</span>,  <span class="t1-zi4r-0-x-x-109">dihedral</span>,  <span class="t1-zi4r-0-x-x-109">gyration</span>,  <span class="t1-zi4r-0-x-x-109">rmsd </span>and
     <span class="t1-zi4r-0-x-x-109">eigenvector</span>. Atom groups may not be replaced by dummy atoms, unless they are excluded from the
     force measurement by specifying <span class="t1-zi4r-0-x-x-109">oneSiteTotalForce</span>, if available.
     </li>
<li class="enumerate" id="x1-110006x3"><span class="ptmri8t-x-x-109">Mutual orthogonality of colvars</span>. In a multidimensional ABF calculation, equation (<a href="#x1-109003r26">26<!--tex4ht:ref: eq:ortho_gradient --></a>) must be satisfied for any two
     colvars <!--l. 5116--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></math>
     and <!--l. 5116--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>j</mi></mrow></msub></math>.
     Various cases fulfill this orthogonality condition:
          <ul class="itemize1">
<li class="itemize"><!--l. 5119--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></math>
          and <!--l. 5119--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>j</mi></mrow></msub></math>
          are based on non-overlapping sets of atoms.
          </li>
<li class="itemize">atoms involved in the force measurement on <!--l. 5120--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></math>
          do not participate in the definition of <!--l. 5121--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>j</mi></mrow></msub></math>.
          This can be obtained using the option <span class="t1-zi4r-0-x-x-109">oneSiteTotalForce </span>of the <span class="t1-zi4r-0-x-x-109">distance</span>, <span class="t1-zi4r-0-x-x-109">angle</span>, and <span class="t1-zi4r-0-x-x-109">dihedral</span>
          components (example: Ramachandran angles <!--l. 5123--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ϕ</mi></math>,
          <!--l. 5123--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ψ</mi></math>).
          </li>
<li class="itemize"><!--l. 5124--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></math>
          and <!--l. 5124--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>j</mi></mrow></msub></math>
          are orthogonal by construction. Useful cases are the sum and difference of two components, or
          <span class="t1-zi4r-0-x-x-109">distance_z </span>and <span class="t1-zi4r-0-x-x-109">distance_xy </span>using the same axis.</li></ul>
</li>
<li class="enumerate" id="x1-110008x4"><span class="ptmri8t-x-x-109">Mutual orthogonality of components</span>: when several components are combined into a colvar, it is assumed that their
     vectors <!--l. 5128--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>v</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mi>i</mi></mrow></msub></math>
     (equation (<a href="#x1-109004r28">28<!--tex4ht:ref: eq:gradient_vector --></a>)) are mutually orthogonal. The cases described for colvars in the previous paragraph
     apply.
     </li>
<li class="enumerate" id="x1-110010x5"><span class="ptmri8t-x-x-109">Orthogonality of colvars and constraints</span>: equation <a href="#x1-109003r27">27<!--tex4ht:ref: eq:ortho_constraints --></a> can be satisfied in two simple ways, if
     either no constrained atoms are involved in the force measurement (see point 3 above) or pairs of
                                                                                         
                                                                                         
     atoms joined by a constrained bond are part of an <span class="ptmri8t-x-x-109">atom group </span>which only intervenes through its
     center (center of mass or geometric center) in the force measurement. In the latter case, the
     contributions of the two atoms to the left-hand side of equation <a href="#x1-109003r27">27<!--tex4ht:ref: eq:ortho_constraints --></a> cancel out. For example, all atoms
     of a rigid TIP3P water molecule can safely be included in an atom group used in a <span class="t1-zi4r-0-x-x-109">distance</span>
     component.</li></ol>
<div class="flushright">
<!--l. 5141--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 5141--><p class="noindent"></p><div class="heading"><a id="sec:colvarbias_abf_params">
<h5 class="subsubsectionHead"><span class="titlemark">6.2.2    </span> <a id="x1-1110006.2.2"></a>Parameters for ABF</h5>
<!--l. 5141--><p class="noindent"></p></a> </div> 
<!--l. 5143--><p class="indent">   ABF depends on parameters from collective variables to define the grid on which free energy gradients are
computed. In the direction of each colvar, the grid ranges from <span class="t1-zi4r-0-x-x-109">lowerBoundary </span>to <span class="t1-zi4r-0-x-x-109">upperBoundary</span>, and the bin
width (grid spacing) is set by the <a href="#colvar|width"><span class="t1-zi4r-0-x-x-109">width</span></a> parameter. The following specific parameters can be set in the ABF
configuration block:
</p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-111001"></a><span class="colorbox" id="colorbox252"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">name:</span></span> see definition of <a href="#sec:colvarbias"><span class="t1-zi4r-0-x-x-109">name</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="dx1-111002"></a><span class="colorbox" id="colorbox253"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">colvars:</span></span> see definition of <a href="#sec:colvarbias"><span class="t1-zi4r-0-x-x-109">colvars</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="dx1-111003"></a><span class="colorbox" id="colorbox254"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">outputEnergy:</span></span> see definition of <a href="#sec:colvarbias"><span class="t1-zi4r-0-x-x-109">outputEnergy</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="dx1-111004"></a><span class="colorbox" id="colorbox255"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">outputFreq:</span></span> see definition of <a href="#colvarbias|outputFreq"><span class="t1-zi4r-0-x-x-109">outputFreq</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="dx1-111005"></a><span class="colorbox" id="colorbox256"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">stepZeroData:</span></span> see definition of <a href="#sec:colvarbias"><span class="t1-zi4r-0-x-x-109">stepZeroData</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="dx1-111006"></a><span class="colorbox" id="colorbox257"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">fullSamples</span></span> <!--l. 5169--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Number
     of samples in a bin prior to application of the ABF<!--l. 5169--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">abf</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive integer <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span>200 <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>To avoid nonequilibrium effects due to large fluctuations of the force exerted along the
     colvars, it is recommended to apply a biasing force only after a the estimate has started converging. If
     <span class="t1-zi4r-0-x-x-109">fullSamples </span>is non-zero, the applied biasing force is scaled by a factor <!--l. 5169--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>α</mi><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>N</mi></mrow><mrow><mi>ξ</mi></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math>
     between 0 and 1. If the number of samples <!--l. 5169--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>N</mi></mrow><mrow><mi>ξ</mi></mrow></msub></math>
     in the current bin is higher than <span class="t1-zi4r-0-x-x-109">fullSamples</span>, the factor is one. If it is less than half of <span class="t1-zi4r-0-x-x-109">fullSamples</span>,
                                                                                         
                                                                                         
     the factor is zero and no bias is applied. Between those two thresholds, the factor follows a linear ramp
     from 0 to 1: <!--l. 5169--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>α</mi><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>N</mi></mrow><mrow><mi>ξ</mi></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow> <mo class="MathClass-rel">=</mo> <mrow><mo class="MathClass-open">(</mo><mrow><mn>2</mn><msub><mrow><mi>N</mi></mrow><mrow><mi>ξ</mi></mrow></msub><mo class="MathClass-bin">∕</mo><mi mathvariant="normal">fullSamples</mi></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-bin">−</mo><mn>1</mn></math>
     .
     </li>
<li class="itemize"><a id="dx1-111007"></a><span class="colorbox" id="colorbox258"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">maxForce</span></span> <!--l. 5180--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Maximum
     magnitude of the ABF force<!--l. 5180--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">abf</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive decimals (one per colvar) <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span>disabled <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>This option enforces a cap on the magnitude of the biasing force effectively applied by
     this ABF bias on each colvar. This can be useful in the presence of singularities in the PMF such as
     hard walls, where the discretization of the average force becomes very inaccurate, causing the colvar's
     diffusion to get “stuck" at the singularity. To enable this cap, provide one non-negative value for each
     colvar. The unit of force is kcal/mol divided by the colvar unit.
     </li>
<li class="itemize"><a id="dx1-111008"></a><span class="colorbox" id="colorbox259"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">hideJacobian</span></span> <!--l. 5199--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Remove
     geometric entropy term from calculated free energy gradient?<!--l. 5199--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">abf</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">no</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>In a few special cases, most notably distance-based variables, an alternate definition of
     the potential of mean force is traditionally used, which excludes the Jacobian term describing the effect
     of geometric entropy on the distribution of the variable. This results, for example, in particle-particle
     potentials of mean force being flat at large separations. The Jacobian term is exactly represented in
     equation (<a href="#x1-109004r28">28<!--tex4ht:ref: eq:gradient_vector --></a>) by the second term of the average, <!--l. 5199--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-bin">−</mo><msub><mrow><mi>k</mi></mrow><mrow><mi>B</mi></mrow></msub><mi>T</mi> <mstyle class="mbox"><mtext></mtext><mstyle class="math"><msub><mrow><mo class="MathClass-op">∇</mo></mrow><mrow><mspace class="thinspace" width="0.3em"></mspace><mspace class="thinspace" width="0.3em"></mspace><mi>x</mi></mrow></msub><mspace class="thinspace" width="0.3em"></mspace></mstyle><mtext></mtext></mstyle><mo class="MathClass-bin">⋅</mo><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>v</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mi>i</mi></mrow></msub></math>.
     Enabling the <span class="t1-zi4r-0-x-x-109">hideJacobian </span>option causes the output data to follow the traditional potential of mean
     force convention, by omitting this contribution from the measured free energy gradients. To ensure
     uniform sampling despite the incomplete description of the free energy, an additional biasing force
     that counteracts the Jacobian force is applied internally by the colvar.
     </li>
<li class="itemize"><a id="dx1-111009"></a><span class="colorbox" id="colorbox260"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">historyFreq</span></span> <!--l. 5210--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Frequency
     (in timesteps) at which ABF history files are accumulated<!--l. 5210--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">abf</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive integer <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span>0 <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>If this number is non-zero, the free energy gradient estimate and sampling histogram
     (and the PMF in one-dimensional calculations) are written to files on disk at the given time interval.
     History file names use the same prefix as output files, with “<span class="t1-zi4r-0-x-x-109">.hist</span>" appended (<span class="ptmri8t-x-x-109">outputName</span><span class="t1-zi4r-0-x-x-109">.hist.pmf</span>).
     <span class="t1-zi4r-0-x-x-109">historyFreq </span>must be a multiple of <a href="#colvarbias|outputFreq"><span class="t1-zi4r-0-x-x-109">outputFreq</span></a>.
     </li>
<li class="itemize"><a id="abf|inputPrefix"> </a> <a id="dx1-111010"></a><span class="colorbox" id="colorbox261"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">inputPrefix</span></span> <!--l. 5228--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Filename
     prefix for reading ABF data<!--l. 5228--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">abf</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>list of strings <br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>If this parameter is set, for each item in the list, ABF tries to read a gradient and a
                                                                                         
                                                                                         
     sampling files named <!--l. 5228--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="t1-zi4r-0-x-x-109">inputPrefix</span><!--l. 5228--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><span class="t1-zi4r-0-x-x-109">.grad</span>
     and <!--l. 5228--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="t1-zi4r-0-x-x-109">inputPrefix</span><!--l. 5228--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><span class="t1-zi4r-0-x-x-109">.count</span>.
     This is done at startup and sets the initial state of the ABF algorithm. The data from all provided files
     is combined appropriately. Also, the grid definition (min and max values, width) need not be the same
     that for the current run. This command is useful to piece together data from simulations in different
     regions of collective variable space, or change the colvar boundary values and widths. Note that it is
     not recommended to use it to switch to a smaller width, as that will leave some bins empty in the finer
     data grid. This option is NOT compatible with reading the data from a restart file (<span class="t1-zi4r-0-x-x-109">colvarsInput</span>
     option of the NAMD config file).
     </li>
<li class="itemize"><a id="dx1-111011"></a><span class="colorbox" id="colorbox262"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">applyBias</span></span> <!--l. 5238--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Apply
     the ABF bias?<!--l. 5238--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">abf</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">yes</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>If this is set to no, the calculation proceeds normally but the adaptive biasing force is
     not applied. Data is still collected to compute the free energy gradient. This is mostly intended for
     testing purposes, and should not be used in routine simulations.
     </li>
<li class="itemize"><a id="dx1-111012"></a><span class="colorbox" id="colorbox263"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">updateBias</span></span> <!--l. 5251--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Update
     the ABF bias?<!--l. 5251--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">abf</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">yes</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:    </span>If this is set to no, the initial biasing force (e.g. read from a restart file or through
     <span class="t1-zi4r-0-x-x-109">inputPrefix</span>) is not updated during the simulation. As a result, a constant bias is applied. This can
     be used to apply a custom, tabulated biasing potential to any combination of colvars. To that effect,
     one should prepare a gradient file containing the gradient of the potential to be applied (negative of
     the bias force), and a count file containing only values greater than <span class="t1-zi4r-0-x-x-109">fullSamples</span>. These files must
     match the grid parameters of the colvars.  </li></ul>
<div class="flushright">
<!--l. 5286--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 5286--><p class="noindent"></p><div class="heading"><a id="sec:colvarbias_abf_shared">
<h5 class="subsubsectionHead"><span class="titlemark">6.2.3    </span> <a id="x1-1120006.2.3"></a>Multiple-replica ABF</h5>
<!--l. 5286--><p class="noindent"></p></a> </div> 
<ul class="itemize1">
<li class="itemize"><a id="dx1-112001"></a><span class="colorbox" id="colorbox264"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">shared</span></span> <!--l. 5286--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Apply
                                                                                         
                                                                                         
     multiple-replica ABF, sharing force samples among the replicas?<!--l. 5286--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">abf</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">no</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>This is command requires that NAMD be compiled and executed with multiple-replica
     support. If <span class="t1-zi4r-0-x-x-109">shared </span>is set to yes, the total force samples will be synchronized among all replicas at
     intervals  defined  by  <span class="t1-zi4r-0-x-x-109">sharedFreq</span>.  This  implements  the  multiple-walker  ABF  scheme  described  in
     <span class="cite">[<a href="#XMinoukadeh2010">23</a>]</span>; this implementation is documented in <span class="cite">[<a href="#XComer2014c">24</a>]</span>. Thus, it is as if total force samples among all replicas
     are gathered in a single shared buffer, which why the algorithm is referred to as shared ABF. Shared
     ABF allows all replicas to benefit from the sampling done by other replicas and can lead to faster
     convergence of the biasing force.
     </li>
<li class="itemize"><a id="dx1-112002"></a><span class="colorbox" id="colorbox265"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">sharedFreq</span></span> <!--l. 5286--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Frequency
     (in timesteps) at which force samples are synchronized among the replicas<!--l. 5286--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">abf</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive integer <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><a href="#colvarbias|outputFreq"><span class="t1-zi4r-0-x-x-109">outputFreq</span></a><br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>In the current implementation of shared ABF, each replica maintains a separate buffer
     of total force samples that determine the biasing force. Every <span class="t1-zi4r-0-x-x-109">sharedFreq </span>steps, the replicas communicate
     the samples that have been gathered since the last synchronization time, ensuring all replicas apply a
     similar biasing force.  </li></ul>
<div class="flushright">
<!--l. 5288--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 5288--><p class="noindent"></p><div class="heading"><a id="sec:colvarbias_abf_output">
<h5 class="subsubsectionHead"><span class="titlemark">6.2.4    </span> <a id="x1-1130006.2.4"></a>Output files</h5>
<!--l. 5288--><p class="noindent"></p></a> </div> 
<!--l. 5290--><p class="indent">   The ABF bias produces the following files, all in multicolumn text format (<a href="#x1-270003.8.5">3.8.5<!--tex4ht:ref: sec:colvar_multicolumn_grid --></a>): </p>
<ul class="itemize1">
<li class="itemize"><span class="ptmri8t-x-x-109">outputName</span><span class="t1-zi4r-0-x-x-109">.grad</span>: current estimate of the free energy gradient (grid), in multicolumn;
     </li>
<li class="itemize"><span class="ptmri8t-x-x-109">outputName</span><span class="t1-zi4r-0-x-x-109">.count</span>: histogram of samples collected, on the same grid;
     </li>
<li class="itemize"><span class="ptmri8t-x-x-109">outputName</span><span class="t1-zi4r-0-x-x-109">.pmf</span>: integrated free energy profile or PMF (for dimensions 1, 2 or 3).</li></ul>
<!--l. 5298--><p class="indent">   Also in the case of one-dimensional calculations, the ABF bias can report its current energy via
<span class="t1-zi4r-0-x-x-109">outputEnergy</span>; in higher dimensions, such computation is not implemented and the energy reported is
zero.
                                                                                         
                                                                                         
</p><!--l. 5300--><p class="indent">   If several ABF biases are defined concurrently, their name is inserted to produce unique filenames for output, as
in <span class="ptmri8t-x-x-109">outputName</span><span class="t1-zi4r-0-x-x-109">.abf1.grad</span>. This should not be done routinely and could lead to meaningless results: only do it if
you know what you are doing!
</p><!--l. 5305--><p class="indent">   If the colvar space has been partitioned into sections (<span class="ptmri8t-x-x-109">windows</span>) in which independent ABF simulations have
been run, the resulting data can be merged using the <span class="t1-zi4r-0-x-x-109">inputPrefix </span>option described above (a run of 0 steps is
enough).
</p>
<div class="flushright">
<!--l. 5311--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 5311--><p class="noindent"></p><div class="heading"><a id="sec:colvarbias_abf_post">
<h5 class="subsubsectionHead"><span class="titlemark">6.2.5    </span> <a id="x1-1140006.2.5"></a>Multidimensional free energy surfaces</h5>
<!--l. 5311--><p class="noindent"></p></a> </div> 
<!--l. 5313--><p class="indent">   If a one-dimensional calculation is performed, the estimated free energy gradient is integrated using
a simple rectangle rule. In dimension 2 or 3, it is calculated as the solution of a Poisson equation: </p><table class="equation"><tr><td>
<a id="x1-114001r30"></a>
<!--l. 5316--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>Δ</mi><mi>A</mi><mrow><mo class="MathClass-open">(</mo><mrow><mi>ξ</mi></mrow><mo class="MathClass-close">)</mo></mrow> <mo class="MathClass-rel">=</mo> <mo class="MathClass-bin">−</mo><mo class="MathClass-op">∇</mo><mo class="MathClass-bin">⋅</mo><mrow><mo class="MathClass-open">⟨</mo><mrow><msub><mrow><mi>F</mi></mrow><mrow><mi>ξ</mi></mrow></msub></mrow><mo class="MathClass-close">⟩</mo></mrow>
</math></td><td class="eq-no">(30)</td></tr></table>
<!--l. 5319--><p class="indent">   wehere <!--l. 5319--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>Δ</mi><mi>A</mi></math>
is the Laplacian of the free energy. The potential of mean force is written under the file name <span class="t1-zi4r-0-x-x-109"><outputname>.pmf</outputname></span>, in
a plain text format (see <a href="#x1-270003.8.5">3.8.5<!--tex4ht:ref: sec:colvar_multicolumn_grid --></a>) that can be read by most data plotting and analysis programs (e.g. Gnuplot).
This applies periodic boundary conditions to periodic coordinates, and Neumann boundary conditions
otherwise (imposed free energy gradient at the boundary of the domain). Note that the grid used for free
energy discretization is extended by one point along non-periodic coordinates, but not along periodic
coordinates.
</p><!--l. 5327--><p class="indent">   In dimension 4 or greater, integrating the discretized gradient becomes non-trivial. The standalone utility
<span class="t1-zi4r-0-x-x-109">abf_integrate </span>is provided to perform that task. Because 4D ABF calculations are uncommon, <span class="ptmb8t-x-x-109">this tool is</span>
<span class="ptmb8t-x-x-109">practically deprecated </span>by the Poisson integration described above.
</p><!--l. 5332--><p class="indent"> <span class="t1-zi4r-0-x-x-109">abf_integrate </span>reads the gradient data and uses it to perform a Monte-Carlo (M-C) simulation in discretized
collective variable space (specifically, on the same grid used by ABF to discretize the free energy gradient). By
default, a history-dependent bias (similar in spirit to metadynamics) is used: at each M-C step, the bias at
the current position is incremented by a preset amount (the <span class="ptmri8t-x-x-109">hill height</span>). Upon convergence, this bias
counteracts optimally the underlying gradient; it is negated to obtain the estimate of the free energy
surface.
                                                                                         
                                                                                         
</p><!--l. 5341--><p class="indent"> <span class="t1-zi4r-0-x-x-109">abf_integrate </span>is invoked using the command-line:<br class="newline"/><span class="t1-zi4r-0-">abf_integrate <gradient_file> [-n <nsteps>] [-t <temp>] [-m (0|1)] [-h <hill_height>] [-f</hill_height></temp></nsteps></gradient_file></span>
<span class="t1-zi4r-0-"><factor>]</factor></span>
</p><!--l. 5346--><p class="indent">   The gradient file name is provided first, followed by other parameters in any order. They are described below,
with their default value in square brackets: </p>
<ul class="itemize1">
<li class="itemize"><span class="t1-zi4r-0-x-x-109">-n</span>: number of M-C steps to be performed; by default, a minimal number of steps is chosen based on
     the size of the grid, and the integration runs until a convergence criterion is satisfied (based on the
     RMSD between the target gradient and the real PMF gradient)
     </li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">-t</span>: temperature for M-C sampling (unrelated to the simulation temperature) [500 K]
     </li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">-s</span>:  scaling  factor  for  the  gradients;  when  using  a  histogram  of  total  forces  obtained  from
     <a href="#colvar|outputTotalForce"><span class="t1-zi4r-0-x-x-109">outputTotalForce</span></a> or the <span class="t1-zi4r-0-x-x-109">.force </span>file written by <a href="#colvarbias|writeTISamples"><span class="t1-zi4r-0-x-x-109">writeTISamples</span></a>, a scaling factor of -1 should be
     used [1.0]
     </li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">-m</span>: use metadynamics-like biased sampling? (0 = false) [1]
     </li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">-h</span>: increment for the history-dependent bias (“hill height") [0.01 kcal/mol]
     </li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">-f</span>: if non-zero, this factor is used to scale the increment stepwise in the second half of the M-C
     sampling to refine the free energy estimate [0.5]</li></ul>
<!--l. 5362--><p class="indent">   Using the default values of all parameters should give reasonable results in most cases.
</p><!--l. 5365--><p class="indent"> <span class="t1-zi4r-0-x-x-109">abf_integrate </span>produces the following output files: </p>
<ul class="itemize1">
<li class="itemize"><span class="t1-zi4r-0-x-x-109"><gradient_file>.pmf</gradient_file></span>: computed free energy surface
     </li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109"><gradient_file>.histo</gradient_file></span>: histogram of M-C sampling (not usable in a straightforward way if the
     history-dependent bias has been applied)
     </li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109"><gradient_file>.est</gradient_file></span>:  estimated  gradient  of  the  calculated  free  energy  surface  (from  finite
     differences)
     </li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109"><gradient_file>.dev</gradient_file></span>:  deviation  between  the  user-provided  numerical  gradient  and  the  actual
     gradient  of  the  calculated  free  energy  surface.  The  RMS  norm  of  this  vector  field  is  used  as  a
     convergence criteria and displayed periodically during the integration.</li></ul>
<!--l. 5378--><p class="indent"> <span class="ptmb8t-x-x-109">Note: </span>Typically, the “deviation" vector field does not vanish as the integration converges. This happens because
the numerical estimate of the gradient does not exactly derive from a potential, due to numerical approximations
used to obtain it (finite sampling and discretization on a grid).
</p>
<div class="flushright">
<!--l. 5386--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 5386--><p class="noindent"></p><div class="heading"><a id="sec:ecolvarbias_abf_extended">
<h4 class="subsectionHead"><span class="titlemark">6.3    </span> <a id="x1-1150006.3"></a>Extended-system Adaptive Biasing Force (eABF)</h4>
<!--l. 5386--><p class="noindent"></p></a> </div> 
<!--l. 5389--><p class="indent">   Extended-system ABF (eABF) is a variant of ABF (<a href="#x1-1090006.2">6.2<!--tex4ht:ref: sec:colvarbias_abf --></a>) where the bias is not applied
directly to the collective variable, but to an extended coordinate (“fictitious variable")
<!--l. 5392--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>λ</mi></math> that
evolves dynamically according to Newtonian or Langevin dynamics. Such an extended coordinate is enabled for a
given colvar using the <span class="t1-zi4r-0-x-x-109">extendedLagrangian </span>and associated keywords (<a href="#x1-960004.21">4.21<!--tex4ht:ref: sec:colvar_extended --></a>). The theory of eABF and the present
implementation are documented in detail in reference <span class="cite">[<a href="#XLesage2017">25</a>]</span>.
</p><!--l. 5398--><p class="indent">   Defining an ABF bias on a colvar wherein the <span class="t1-zi4r-0-x-x-109">extendedLagrangian </span>option is active will perform eABF
automatically; there is no dedicated option.
</p><!--l. 5401--><p class="indent">   The extended variable <!--l. 5401--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>λ</mi></math> is
coupled to the colvar <!--l. 5401--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi> <mo class="MathClass-rel">=</mo> <mi>ξ</mi><mrow><mo class="MathClass-open">(</mo><mrow><mi>q</mi></mrow><mo class="MathClass-close">)</mo></mrow></math> by the
harmonic potential <!--l. 5402--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">(</mo><mrow><mi>k</mi><mo class="MathClass-bin">∕</mo><mn>2</mn></mrow><mo class="MathClass-close">)</mo></mrow><msup><mrow><mrow><mo class="MathClass-open">(</mo><mrow><mi>z</mi><mo class="MathClass-bin">−</mo><mi>λ</mi></mrow><mo class="MathClass-close">)</mo></mrow></mrow><mrow><mn>2</mn></mrow></msup></math>. Under eABF
dynamics, the adaptive bias on <!--l. 5403--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>λ</mi></math>
is the running estimate of the average spring force: </p><table class="equation"><tr><td> <a id="x1-115001r31"></a>
<!--l. 5405--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<msup><mrow><mi>F</mi></mrow><mrow><mi mathvariant="normal">bias</mi></mrow></msup><mrow><mo class="MathClass-open">(</mo><mrow><msup><mrow><mi>λ</mi></mrow><mrow><mo class="MathClass-bin">∗</mo></mrow></msup></mrow><mo class="MathClass-close">)</mo></mrow> <mo class="MathClass-rel">=</mo><msub><mrow> <mfenced close="⟩" open="⟨" separators=""><mrow><mi>k</mi><mrow><mo class="MathClass-open">(</mo><mrow><mi>λ</mi> <mo class="MathClass-bin">−</mo><mi>z</mi></mrow><mo class="MathClass-close">)</mo></mrow></mrow></mfenced></mrow><mrow><msup><mrow>
<mi>λ</mi></mrow><mrow><mo class="MathClass-bin">∗</mo></mrow></msup></mrow></msub>
</math></td><td class="eq-no">(31)</td></tr></table>
<!--l. 5408--><p class="indent">   where the angle brackets indicate a canonical average conditioned by
<!--l. 5408--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>λ</mi> <mo class="MathClass-rel">=</mo> <msup><mrow><mi>λ</mi></mrow><mrow><mo class="MathClass-bin">∗</mo></mrow></msup></math>.
At long simulation times, eABF produces a flat histogram of the extended variable
<!--l. 5409--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>λ</mi></math>, and a flattened
histogram of <!--l. 5410--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>,
whose exact shape depends on the strength of the coupling as defined by <span class="t1-zi4r-0-x-x-109">extendedFluctuation </span>in the colvar. Coupling should
be somewhat loose for faster exploration and convergence, but strong enough that the bias does help overcome barriers along
the colvar <!--l. 5413--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>.<span class="cite">[<a href="#XLesage2017">25</a>]</span>
Distribution of the colvar may be assessed by plotting its histogram, which is written to
the <span class="ptmri8t-x-x-109">outputName</span><span class="t1-zi4r-0-x-x-109">.zcount </span>file in every eABF simulation. Note that a <span class="t1-zi4r-0-x-x-109">histogram </span>bias (<a href="#x1-1350006.10">6.10<!--tex4ht:ref: sec:colvarbias_histogram --></a>)
applied to an extended-Lagrangian colvar will access the extended degree of freedom
<!--l. 5418--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>λ</mi></math>, not the original
colvar <!--l. 5418--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>;
however, the joint histogram may be explicitly requested by listing the name of the colvar twice in a row within the
                                                                                         
                                                                                         
<span class="t1-zi4r-0-x-x-109">colvars </span>parameter of the <span class="t1-zi4r-0-x-x-109">histogram </span>block.
</p><!--l. 5422--><p class="indent">   The eABF PMF is that of the coordinate <!--l. 5422--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>λ</mi></math>,
it is not exactly the free energy profile of <!--l. 5422--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>.
That quantity can be calculated based on either the CZAR estimator or the Zheng/Yang estimator.
</p>
<div class="flushright">
<!--l. 5427--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 5427--><p class="noindent"></p><div class="heading"><a id="sec:colvarbias_ti_ext_czar">
<h5 class="subsubsectionHead"><span class="titlemark">6.3.1    </span> <a id="x1-1160006.3.1"></a>CZAR estimator of the free energy</h5>
<!--l. 5427--><p class="noindent"></p></a> </div> 
<!--l. 5429--><p class="indent">   The <span class="ptmri8t-x-x-109">corrected z-averaged restraint </span>(CZAR) estimator is described in detail in reference <span class="cite">[<a href="#XLesage2017">25</a>]</span>. It is computed
automatically in eABF simulations, regardless of the number of colvars involved. Note that ABF may also be
applied on a combination of extended and non-extended colvars; in that case, CZAR still provides an unbiased
estimate of the free energy gradient.
</p><!--l. 5436--><p class="indent">   CZAR estimates the free energy gradient as: </p><table class="equation"><tr><td> <a id="x1-116001r32"></a>
<!--l. 5437--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<msup><mrow><mi>A</mi></mrow><mrow><mi>′</mi></mrow></msup><mrow><mo class="MathClass-open">(</mo><mrow><mi>z</mi></mrow><mo class="MathClass-close">)</mo></mrow> <mo class="MathClass-rel">=</mo> <mo class="MathClass-bin">−</mo><mfrac><mrow><mn>1</mn></mrow>
<mrow><mi>β</mi></mrow></mfrac> <mfrac><mrow><mi>d</mi><mo class="qopname">ln</mo><!--nolimits--><mover accent="true"><mrow><mi>ρ</mi></mrow><mo class="qopname">̃</mo></mover><mrow><mo class="MathClass-open">(</mo><mrow><mi>z</mi></mrow><mo class="MathClass-close">)</mo></mrow></mrow>
<mrow><mi>d</mi><mi>z</mi></mrow></mfrac> <mo class="MathClass-bin">+</mo><mi>k</mi><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mrow><mo class="MathClass-open">⟨</mo><mrow><mi>λ</mi></mrow><mo class="MathClass-close">⟩</mo></mrow></mrow><mrow><mi>z</mi></mrow></msub><mo class="MathClass-bin">−</mo><mi>z</mi></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-punc">.</mo>
</math></td><td class="eq-no">(32)</td></tr></table>
<!--l. 5441--><p class="indent">   where <!--l. 5441--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi> <mo class="MathClass-rel">=</mo> <mi>ξ</mi><mrow><mo class="MathClass-open">(</mo><mrow><mi>q</mi></mrow><mo class="MathClass-close">)</mo></mrow></math> is the colvar,
<!--l. 5441--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>λ</mi></math> is the extended variable
harmonically coupled to <!--l. 5442--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math>
with a force constant <!--l. 5442--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>k</mi></math>,
and <!--l. 5442--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mover accent="true"><mrow><mi>ρ</mi></mrow><mo class="MathClass-op">̃</mo></mover><mrow><mo class="MathClass-open">(</mo><mrow><mi>z</mi></mrow><mo class="MathClass-close">)</mo></mrow></math> is the observed
distribution (histogram) of <!--l. 5443--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math>,
affected by the eABF bias.
</p><!--l. 5445--><p class="indent">   Parameters for the CZAR estimator are: </p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-116002"></a><span class="colorbox" id="colorbox266"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">CZARestimator</span></span> <!--l. 5452--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Calculate
     CZAR estimator of the free energy?<!--l. 5452--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">abf</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">yes</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>This option is only available when ABF is performed on extended-Lagrangian colvars.
     When enabled, it triggers calculation of the free energy following the CZAR estimator.
                                                                                         
                                                                                         
     </li>
<li class="itemize"><a id="dx1-116003"></a><span class="colorbox" id="colorbox267"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">writeCZARwindowFile</span></span> <!--l. 5473--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Write
     internal data from CZAR to a separate file?<!--l. 5473--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">abf</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">no</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>When this option is enabled, eABF simulations will write a file containing the <!--l. 5473--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math>-averaged
     restraint force under the name <span class="ptmri8t-x-x-109">outputName</span><span class="t1-zi4r-0-x-x-109">.zgrad</span>. The same information is always included in the
     colvars state file, which is sufficient for restarting an eABF simulation. These separate file is only
     useful  when  joining  adjacent  windows  from  a  stratified  eABF  simulation,  either  to  continue  the
     simulation in a broader window or to compute a CZAR estimate of the PMF over the full range of
     the coordinate(s). <span class="ptmb8t-x-x-109">Important warning. </span>Unbiased free-energy estimators from eABF dynamics rely
     on some form of sampling histogram. When running stratified (windowed) calculations this histogram
     becomes discontinuous, and as a result the free energy gradient estimated by CZAR is inaccurate
     at the window boundary, resulting in visible "blips" in the PMF. As a workaround, we recommend
     manually replacing the two free energy gradient values at the boundary, either with the ABF values
     from .grad files (accurate in the limit of tight coupling), or with values interpolated for the neighboring
     values of the CZAR gradient.  </li></ul>
<!--l. 5476--><p class="indent">   Similar to ABF, the CZAR estimator produces two output files in multicolumn text format (<a href="#x1-270003.8.5">3.8.5<!--tex4ht:ref: sec:colvar_multicolumn_grid --></a>):
</p>
<ul class="itemize1">
<li class="itemize"><span class="ptmri8t-x-x-109">outputName</span><span class="t1-zi4r-0-x-x-109">.czar.grad</span>: current estimate of the free energy gradient (grid), in multicolumn;
     </li>
<li class="itemize"><span class="ptmri8t-x-x-109">outputName</span><span class="t1-zi4r-0-x-x-109">.czar.pmf</span>: only for one-dimensional calculations, integrated free energy profile or PMF.</li></ul>
<!--l. 5483--><p class="noindent">The sampling histogram associated with the CZAR estimator is the
<!--l. 5483--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math>-histogram,
which is written in the file <span class="ptmri8t-x-x-109">outputName</span><span class="t1-zi4r-0-x-x-109">.zcount</span>.
</p>
<div class="flushright">
<!--l. 5554--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 5554--><p class="noindent"></p><div class="heading"><a id="sec:colvarbias_ti_ext_zheng_yang">
<h5 class="subsubsectionHead"><span class="titlemark">6.3.2    </span> <a id="x1-1170006.3.2"></a>Zheng/Yang estimator of the free energy</h5>
<!--l. 5554--><p class="noindent"></p></a> </div>  This feature has been contributed to NAMD by the following authors:
<!--l. 5554--><p class="indent">
</p><div class="quote">
<!--l. 5554--><p class="noindent">Haohao Fu and Christophe Chipot<br class="newline"/>Laboratoire International Associé Centre National de la Recherche Scientifique et University
     of Illinois at Urbana--Champaign, <br class="newline"/>Unité Mixte de Recherche No. 7565, Université de Lorraine, <br class="newline"/>B.P. 70239, 54506 Vandœuvre-lès-Nancy cedex, France</p></div>
<!--l. 5554--><p class="indent"> <span class="ptmr8c-x-x-109">©</span> 2016, <span class="ptmrc8t-x-x-109">C<span class="small-caps">E</span><span class="small-caps">N</span><span class="small-caps">T</span><span class="small-caps">R</span><span class="small-caps">E</span> N<span class="small-caps">A</span><span class="small-caps">T</span><span class="small-caps">I</span><span class="small-caps">O</span><span class="small-caps">N</span><span class="small-caps">A</span><span class="small-caps">L</span> <span class="small-caps">D</span><span class="small-caps">E</span> <span class="small-caps">L</span><span class="small-caps">A</span> R<span class="small-caps">E</span><span class="small-caps">C</span><span class="small-caps">H</span><span class="small-caps">E</span><span class="small-caps">R</span><span class="small-caps">C</span><span class="small-caps">H</span><span class="small-caps">E</span> S<span class="small-caps">C</span><span class="small-caps">I</span><span class="small-caps">E</span><span class="small-caps">N</span><span class="small-caps">T</span><span class="small-caps">I</span><span class="small-caps">fi</span><span class="small-caps">Q</span><span class="small-caps">U</span><span class="small-caps">E</span></span>
</p><!--l. 5554--><p class="indent">   This implementation is fully documented in <span class="cite">[<a href="#XFu2016">26</a>]</span>. The Zheng and Yang estimator <span class="cite">[<a href="#XZheng2012">27</a>]</span> is based on Umbrella
Integration <span class="cite">[<a href="#XKastner2005">28</a>]</span>. The free energy gradient is estimated as :
</p>
<table class="equation"><tr><td> <a id="x1-117001r33"></a>
<!--l. 5554--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<msup><mrow><mi>A</mi></mrow><mrow><mi>′</mi></mrow></msup><mrow><mo class="MathClass-open">(</mo><mrow><msup><mrow><mi>ξ</mi></mrow><mrow><mo class="MathClass-bin">∗</mo></mrow></msup></mrow><mo class="MathClass-close">)</mo></mrow> <mo class="MathClass-rel">=</mo> <mfrac><mrow><munder class="msub"><mrow><mo mathsize="big">∑</mo></mrow><mrow><mi>λ</mi></mrow></munder><mi>N</mi><mrow><mo class="MathClass-open">(</mo><mrow><msup><mrow><mi>ξ</mi></mrow><mrow><mo class="MathClass-bin">∗</mo></mrow></msup><mo class="MathClass-punc">,</mo><mi>λ</mi></mrow><mo class="MathClass-close">)</mo></mrow> <mfenced close="]" open="[" separators=""><mrow><mfrac><mrow><mrow><mo class="MathClass-open">(</mo><mrow><msup><mrow><mi>ξ</mi></mrow><mrow><mo class="MathClass-bin">∗</mo></mrow></msup><mo class="MathClass-bin">−</mo><msub><mrow><mrow><mo class="MathClass-open">⟨</mo><mrow><mi>ξ</mi></mrow><mo class="MathClass-close">⟩</mo></mrow></mrow><mrow>
<mi>λ</mi></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></mrow>
<mrow><mi>β</mi><msubsup><mrow><mi>σ</mi></mrow><mrow><mi>λ</mi></mrow><mrow><mn>2</mn></mrow></msubsup></mrow></mfrac> <mo class="MathClass-bin">−</mo><mi>k</mi><mrow><mo class="MathClass-open">(</mo><mrow><msup><mrow><mi>ξ</mi></mrow><mrow><mo class="MathClass-bin">∗</mo></mrow></msup><mo class="MathClass-bin">−</mo><mi>λ</mi></mrow><mo class="MathClass-close">)</mo></mrow></mrow></mfenced></mrow>
<mrow><munder class="msub"><mrow><mo mathsize="big">∑</mo></mrow><mrow><mi>λ</mi></mrow></munder><mi>N</mi><mrow><mo class="MathClass-open">(</mo><mrow><msup><mrow><mi>ξ</mi></mrow><mrow><mo class="MathClass-bin">∗</mo></mrow></msup><mo class="MathClass-punc">,</mo><mi>λ</mi></mrow><mo class="MathClass-close">)</mo></mrow></mrow></mfrac>
</math></td><td class="eq-no">(33)</td></tr></table>
<!--l. 5554--><p class="indent">   where <!--l. 5554--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math> is the colvar,
<!--l. 5554--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>λ</mi></math> is the extended variable
harmonically coupled to <!--l. 5554--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>
with a force constant <!--l. 5554--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>k</mi></math>,
<!--l. 5554--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>N</mi><mrow><mo class="MathClass-open">(</mo><mrow><mi>ξ</mi><mo class="MathClass-punc">,</mo><mi>λ</mi></mrow><mo class="MathClass-close">)</mo></mrow></math> is the number of samples collected
in a <!--l. 5554--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">(</mo><mrow><mi>ξ</mi><mo class="MathClass-punc">,</mo><mi>λ</mi></mrow><mo class="MathClass-close">)</mo></mrow></math> bin, which is assumed
to be a Gaussian function of <!--l. 5554--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>
with mean <!--l. 5554--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mrow><mo class="MathClass-open">⟨</mo><mrow><mi>ξ</mi></mrow><mo class="MathClass-close">⟩</mo></mrow></mrow><mrow><mi>λ</mi></mrow></msub></math> and
standard deviation <!--l. 5554--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>σ</mi></mrow><mrow><mi>λ</mi></mrow></msub></math>.
</p><!--l. 5554--><p class="indent">   The estimator is enabled through the following option: </p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-117002"></a><span class="colorbox" id="colorbox268"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">UIestimator</span></span> <!--l. 5554--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Calculate
     UI estimator of the free energy?<!--l. 5554--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">abf</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">no</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>This option is only available when ABF is performed on extended-Lagrangian colvars.
     When enabled, it triggers calculation of the free energy following the UI estimator. </li></ul>
<!--l. 5554--><p class="noindent"><span class="likeparagraphHead"><a id="x1-1180006.3.2"></a>Usage for multiple--replica eABF.</span>
   The eABF algorithm can be associated with a multiple--walker strategy <span class="cite">[<a href="#XMinoukadeh2010">23</a>, <a href="#XComer2014c">24</a>]</span> (<a href="#x1-1120006.2.3">6.2.3<!--tex4ht:ref: sec:mw-ABF --></a>). To run a
multiple--replica eABF simulation, start a multiple-replica NAMD run (option <span class="t1-zi4r-0-x-x-109">+replicas</span>) and set <span class="t1-zi4r-0-x-x-109">shared on </span>in the
Colvars config file to enable the multiple--walker ABF algorithm. It should be noted that in contrast with classical
                                                                                         
                                                                                         
MW--ABF simulations, the output files of an MW--eABF simulation only show the free energy estimate of the
corresponding replica.
</p><!--l. 5554--><p class="indent">   One can merge the results, using <span class="t1-zi4r-0-x-x-109">./eabf.tcl -mergemwabf [merged_filename] [eabf_output1]</span>
<span class="t1-zi4r-0-x-x-109">[eabf_output2] ...</span>, e.g., <span class="t1-zi4r-0-x-x-109">./eabf.tcl -mergemwabf merge.eabf eabf.0.UI eabf.1.UI eabf.2.UI</span>
<span class="t1-zi4r-0-x-x-109">eabf.3.UI</span>.
</p><!--l. 5554--><p class="indent">   If one runs an ABF--based calculation, breaking the reaction pathway into several non--overlapping windows,
one can use <span class="t1-zi4r-0-x-x-109">./eabf.tcl -mergesplitwindow [merged_fileprefix] [eabf_output] [eabf_output2]</span>
<span class="t1-zi4r-0-x-x-109">... </span>to merge the data accrued in these non--overlapping windows. This option can be utilized in both
eABF and classical ABF simulations, e.g., <span class="t1-zi4r-0-x-x-109">./eabf.tcl -mergesplitwindow merge window0.czar</span>
<span class="t1-zi4r-0-x-x-109">window1.czar window2.czar window3.czar</span>, <span class="t1-zi4r-0-x-x-109">./eabf.tcl -mergesplitwindow merge window0.UI</span>
<span class="t1-zi4r-0-x-x-109">window1.UI window2.UI window3.UI </span>or <span class="t1-zi4r-0-x-x-109">./eabf.tcl -mergesplitwindow merge abf0 abf1 abf2</span>
<span class="t1-zi4r-0-x-x-109">abf3</span>.
</p>
<div class="flushright">
<!--l. 5557--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 5557--><p class="noindent"></p><div class="heading"><a id="sec:colvarbias_meta">
<h4 class="subsectionHead"><span class="titlemark">6.4    </span> <a id="x1-1190006.4"></a>Metadynamics</h4>
<!--l. 5557--><p class="noindent"></p></a> </div> 
<!--l. 5559--><p class="indent">   The metadynamics method uses a history-dependent potential <span class="cite">[<a href="#XLaio2002">29</a>]</span> that generalizes to any type of colvars the
conformational flooding <span class="cite">[<a href="#XGrubmuller1995">30</a>]</span> and local elevation <span class="cite">[<a href="#XHuber1994">31</a>]</span> methods, originally formulated to use as colvars the principal
components of a covariance matrix or a set of dihedral angles, respectively. The metadynamics potential on the
colvars <!--l. 5560--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle> <mo class="MathClass-rel">=</mo> <mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mn>2</mn></mrow></msub><mo class="MathClass-punc">,</mo><mo class="MathClass-op">…</mo><mo class="MathClass-punc">,</mo><msub><mrow><mi>ξ</mi></mrow><mrow><msub><mrow><mi>N</mi></mrow><mrow><mi mathvariant="normal">cv</mi></mrow></msub></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math>
is defined as: </p><table class="equation"><tr><td> <a id="x1-119001r34"></a>
<!--l. 5561--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<msub><mrow><mi>V</mi></mrow><mrow><mi mathvariant="normal">meta</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow></mrow><mo class="MathClass-close">)</mo></mrow><mspace class="tmspace" width="3.04074pt"></mspace> <mo class="MathClass-rel">=</mo> <mspace class="tmspace" width="3.04074pt"></mspace><munderover accent="false" accentunder="false"><mrow><mo mathsize="big">∑</mo></mrow><mrow><msup><mrow><mi>t</mi></mrow><mrow><mi>′</mi></mrow></msup><mo class="MathClass-rel">=</mo><mi>δ</mi><mi>t</mi><mo class="MathClass-punc">,</mo><mn>2</mn><mi>δ</mi><mi>t</mi><mo class="MathClass-punc">,</mo><mo>…</mo></mrow><mrow><msup><mrow><mi>t</mi></mrow><mrow><mi>′</mi></mrow></msup><mo class="MathClass-rel">&lt;</mo><mi>t</mi></mrow></munderover><mi>W</mi><mspace class="tmspace" width="2.43306pt"></mspace><munderover accent="false" accentunder="false"><mrow><mo mathsize="big">∏</mo></mrow><mrow>
<mi>i</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><msub><mrow><mi>N</mi></mrow><mrow><mi mathvariant="normal">cv</mi></mrow></msub></mrow></munderover><mo class="qopname"> exp</mo><!--nolimits--> <mfenced close=")" open="(" separators=""><mrow><mo class="MathClass-bin">−</mo><mfrac><mrow><msup><mrow><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-bin">−</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><msup><mrow><mi>t</mi></mrow><mrow><mi>′</mi></mrow></msup></mrow><mo class="MathClass-close">)</mo></mrow></mrow><mo class="MathClass-close">)</mo></mrow></mrow><mrow><mn>2</mn></mrow></msup></mrow>
<mrow><mn>2</mn><msubsup><mrow><mi>σ</mi></mrow><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow><mrow><mn>2</mn></mrow></msubsup></mrow></mfrac> </mrow></mfenced><mi mathvariant="normal">,</mi>
</math></td><td class="eq-no">(34)</td></tr></table>
<!--l. 5570--><p class="indent">   where <!--l. 5570--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>V</mi></mrow><mrow><mi mathvariant="normal">meta</mi></mrow></msub></math>
is the history-dependent potential acting on the <span class="ptmri8t-x-x-109">current </span>values of the colvars
<!--l. 5570--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></math>,
and depends only parametrically on the <span class="ptmri8t-x-x-109">previous </span>values of the colvars.
<!--l. 5571--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>V</mi></mrow><mrow><mi mathvariant="normal">meta</mi></mrow></msub></math> is constructed as a
sum of <!--l. 5571--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>N</mi></mrow><mrow><mi mathvariant="normal">cv</mi></mrow></msub></math>-dimensional
repulsive Gaussian “hills", whose height is a chosen energy constant
                                                                                         
                                                                                         
<!--l. 5571--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>W</mi></math>, and whose centers are the
previously explored configurations <!--l. 5571--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mfenced close=")" open="(" separators=""><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle><mrow><mo class="MathClass-open">(</mo><mrow><mi>δ</mi><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-punc">,</mo><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle><mrow><mo class="MathClass-open">(</mo><mrow><mn>2</mn><mi>δ</mi><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-punc">,</mo><mo class="MathClass-op">…</mo></mrow></mfenced></math>.
</p><!--l. 5573--><p class="indent">   During the simulation, the system evolves towards the nearest minimum of the “effective" potential of mean
force <!--l. 5573--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>Ã</mi><mrow><mo class="MathClass-open">(</mo><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></mrow><mo class="MathClass-close">)</mo></mrow></math>,
which is the sum of the “real" underlying potential of mean force
<!--l. 5573--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>A</mi><mrow><mo class="MathClass-open">(</mo><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></mrow><mo class="MathClass-close">)</mo></mrow></math> and the the
metadynamics potential, <!--l. 5573--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>V</mi></mrow><mrow><mi mathvariant="normal">meta</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></mrow><mo class="MathClass-close">)</mo></mrow></math>.
Therefore, at any given time the probability of observing the configuration
<!--l. 5574--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle class="mbox"><mtext></mtext><mstyle class="math"><msup><mrow><mi>ξ</mi></mrow><mrow><mo class="MathClass-bin">∗</mo></mrow></msup></mstyle><mtext></mtext></mstyle></math> is proportional
to <!--l. 5574--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="qopname">exp</mo><!--nolimits--> <mfenced close=")" open="(" separators=""><mrow><mo class="MathClass-bin">−</mo><mi>Ã</mi><mrow><mo class="MathClass-open">(</mo><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><msup><mrow><mi>ξ</mi></mrow><mrow><mo class="MathClass-bin">∗</mo></mrow></msup></mstyle><mtext></mtext></mstyle></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-bin">∕</mo><msub><mrow><mi>κ</mi></mrow><mrow><mi mathvariant="normal">B</mi></mrow></msub><mi>T</mi> </mrow></mfenced></math>: this is also
the probability that a new Gaussian “hill" is added at that configuration. If the simulation is run for a sufficiently long time,
each local minimum is canceled out by the sum of the Gaussian “hills". At that stage the “effective" potential of mean force
<!--l. 5576--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>Ã</mi><mrow><mo class="MathClass-open">(</mo><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></mrow><mo class="MathClass-close">)</mo></mrow></math> is constant, and
<!--l. 5576--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-bin">−</mo><msub><mrow><mi>V</mi></mrow><mrow><mi mathvariant="normal">meta</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></mrow><mo class="MathClass-close">)</mo></mrow></math> is an estimator of the “real"
potential of mean force <!--l. 5576--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>A</mi><mrow><mo class="MathClass-open">(</mo><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></mrow><mo class="MathClass-close">)</mo></mrow></math>,
save for an additive constant: </p><table class="equation"><tr><td> <a id="x1-119002r35"></a>
<!--l. 5577--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>A</mi><mrow><mo class="MathClass-open">(</mo><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></mrow><mo class="MathClass-close">)</mo></mrow><mspace class="tmspace" width="3.04074pt"></mspace> <mo class="MathClass-rel">≃</mo><mspace class="tmspace" width="3.04074pt"></mspace><mo class="MathClass-bin">−</mo><msub><mrow><mi>V</mi></mrow><mrow><mi mathvariant="normal">meta</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-bin">+</mo><mi>K</mi>
</math></td><td class="eq-no">(35)</td></tr></table>
<!--l. 5584--><p class="indent">   Such estimate of the free energy can be provided by enabling <span class="t1-zi4r-0-x-x-109">writeFreeEnergyFile</span>.
Assuming that the set of collective variables includes all relevant degrees of freedom, the
predicted error of the estimate is a simple function of the correlation times of the colvars
<!--l. 5585--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>τ</mi></mrow><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow></msub></math>, and of the user-defined
parameters <!--l. 5585--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>W</mi></math>,
<!--l. 5585--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>σ</mi></mrow><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow></msub></math> and
<!--l. 5585--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>δ</mi><mi>t</mi></math>
<span class="cite">[<a href="#XBussi2006">32</a>]</span>. In typical applications, a good rule of thumb can be to choose the ratio
<!--l. 5586--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>W</mi><mo class="MathClass-bin">∕</mo><mi>δ</mi><mi>t</mi></math> much smaller
than <!--l. 5586--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>κ</mi></mrow><mrow><mi mathvariant="normal">B</mi></mrow></msub><mi>T</mi> <mo class="MathClass-bin">∕</mo><msub><mrow><mi>τ</mi></mrow><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></mrow></msub></math>, where
<!--l. 5586--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>τ</mi></mrow><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></mrow></msub></math> is the longest
among <!--l. 5586--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></math>'s
correlation times: <!--l. 5586--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>σ</mi></mrow><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow></msub></math>
then dictates the resolution of the calculated PMF.
</p><!--l. 5588--><p class="indent">   If the metadynamics parameters are chosen correctly, after an equilibration time,
<!--l. 5588--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>t</mi></mrow><mrow><mi>e</mi></mrow></msub></math>,
the estimator provided by eq. <a href="#x1-119002r35">35<!--tex4ht:ref: eq:colvars_meta_fes --></a> oscillates on time around the “real" free energy, thereby a
better estimate of the latter can be obtained as the time average of the bias potential after
<!--l. 5589--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>t</mi></mrow><mrow><mi>e</mi></mrow></msub></math>
<span class="cite">[<a href="#XMarinelli2009">33</a>, <a href="#XCrespo2010">34</a>]</span>: </p><table class="equation"><tr><td> <a id="x1-119003r36"></a>
<!--l. 5590--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>A</mi><mrow><mo class="MathClass-open">(</mo><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></mrow><mo class="MathClass-close">)</mo></mrow><mspace class="tmspace" width="3.04074pt"></mspace> <mo class="MathClass-rel">=</mo> <mspace class="tmspace" width="3.04074pt"></mspace><mo class="MathClass-bin">−</mo> <mfrac><mrow><mn>1</mn></mrow>
<mrow><msub><mrow><mi>t</mi></mrow><mrow><mi>t</mi><mi>o</mi><mi>t</mi></mrow></msub><mo class="MathClass-bin">−</mo><msub><mrow><mi>t</mi></mrow><mrow><mi>e</mi></mrow></msub></mrow></mfrac><msubsup><mrow><mo class="MathClass-op">∫
 <!--nolimits--></mo><!--nolimits--></mrow><mrow><msub><mrow><mi>t</mi></mrow><mrow><mi>e</mi></mrow></msub></mrow><mrow><msub><mrow><mi>t</mi></mrow><mrow><mi>t</mi><mi>o</mi><mi>t</mi></mrow></msub>
</mrow></msubsup><msub><mrow><mi>V</mi></mrow><mrow><mi mathvariant="normal">meta</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle><mo class="MathClass-punc">,</mo><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow><mi>d</mi><mi>t</mi>
</math></td><td class="eq-no">(36)</td></tr></table>
<!--l. 5596--><p class="indent">   where <!--l. 5596--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>t</mi></mrow><mrow><mi>e</mi></mrow></msub></math> is
the time after which the bias potential grows (approximately) evenly during the simulation and
<!--l. 5596--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>t</mi></mrow><mrow><mi>t</mi><mi>o</mi><mi>t</mi></mrow></msub></math> is the
total simulation time. The free energy calculated according to eq. <a href="#x1-119003r36">36<!--tex4ht:ref: eq:colvars_meta_fes_av --></a> can thus be obtained averaging on time mutiple
time-dependent free energy estimates, that can be printed out through the keyword <span class="t1-zi4r-0-x-x-109">keepFreeEnergyFiles</span>. An
alternative is to obtain the free energy profiles by summing the hills added during the simulation; the hills trajectory
can be printed out by enabling the option <span class="t1-zi4r-0-x-x-109">writeHillsTrajectory</span>.
</p>
<div class="flushright">
<!--l. 5598--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 5598--><p class="noindent"></p><div class="heading"><a id="sec:colvarbias_meta_boundaries">
<h5 class="subsubsectionHead"><span class="titlemark">6.4.1    </span> <a id="x1-1200006.4.1"></a>Treatment of the PMF boundaries</h5>
<!--l. 5598--><p class="noindent"></p></a> </div> 
<!--l. 5600--><p class="indent">   In typical scenarios the Gaussian hills of a metadynamics potential are interpolated and summed together onto a
grid, which is much more efficient than computing each hill independently at every step (the keyword <a href="#metadynamics|useGrids"><span class="t1-zi4r-0-x-x-109">useGrids</span></a> is
<span class="t1-zi4r-0-x-x-109">on </span>by default). This numerical approximation typically yields neglibile errors in the resulting PMF <span class="cite">[<a href="#XFiorin2013">1</a>]</span>. However,
due to the finite thickness of the Gaussian function, the metadynamics potential would suddenly vanish each time a
variable exceeds its grid boundaries.
</p><!--l. 5604--><p class="indent">   To avoid such discontinuity the Colvars metadynamics code will keep an explicit copy of each hill that straddles
a grid's boundary, and will use it to compute metadynamics forces outside the grid. This measure is taken to protect
the accuracy and stability of a metadynamics simulation, except in cases of “natural" boundaries (for example, the
<!--l. 5605--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">[</mo><mrow><mn>0</mn> <mo class="MathClass-punc">:</mo> <mn>1</mn><mn>8</mn><mn>0</mn></mrow><mo class="MathClass-close">]</mo></mrow></math> interval
of an <span class="t1-zi4r-0-x-x-109">angle </span>colvar) or when the flags <a href="#colvar|hardLowerBoundary"><span class="t1-zi4r-0-x-x-109">hardLowerBoundary</span></a> and <a href="#colvar|hardUpperBoundary"><span class="t1-zi4r-0-x-x-109">hardUpperBoundary</span></a> are explicitly set by the user.
Unfortunately, processing explicit hills alongside the potential and force grids could easily become inefficient,
slowing down the simulation and increasing the state file's size.
</p><!--l. 5608--><p class="indent">   In general, it is a good idea to <span class="ptmri8t-x-x-109">define a repulsive potential to avoid hills from coming too close to the grid's</span>
<span class="ptmri8t-x-x-109">boundaries</span>, for example as a <span class="t1-zi4r-0-x-x-109">harmonicWalls </span>restraint (see <a href="#x1-1320006.7">6.7<!--tex4ht:ref: sec:colvarbias_harmonic_walls --></a>).<br class="newline"/>
</p><!--l. 5610--><p class="noindent"><span class="ptmb8t-x-x-109">Example: </span>Using harmonic walls to protect the grid's boundaries.</p>
<div class="mdframed" id="mdframed-52">
<!--l. 5611--><p class="noindent"></p><pre>colvar {<br class="newline"/>  name r<br class="newline"/>  distance { ... }<br class="newline"/>  upperBoundary 15.0<br class="newline"/>  width 0.2<br class="newline"/>}<br class="newline"/><br class="newline"/>metadynamics {<br class="newline"/>  name meta_r<br class="newline"/>  colvars r<br class="newline"/>  hillWeight 0.001<br class="newline"/>  hillWidth 2.0<br class="newline"/>}<br class="newline"/><br class="newline"/>harmonicWalls {<br class="newline"/>  name wall_r<br class="newline"/>  colvars r<br class="newline"/>  upperWalls 13.0<br class="newline"/>  upperWallConstant 2.0<br class="newline"/>} </pre>
</div>
<!--l. 5634--><p class="indent">   In the colvar <span class="t1-zi4r-0-x-x-109">r</span>, the <span class="t1-zi4r-0-x-x-109">distance </span>function used has a <span class="t1-zi4r-0-x-x-109">lowerBoundary </span>automatically set to 0 Å by default, thus the
keyword <span class="t1-zi4r-0-x-x-109">lowerBoundary </span>itself is not mandatory and <span class="t1-zi4r-0-x-x-109">hardLowerBoundary </span>is set to <span class="t1-zi4r-0-x-x-109">yes </span>internally. However,
<span class="t1-zi4r-0-x-x-109">upperBoundary </span>does not have such a “natural" choice of value. The metadynamics potential <span class="t1-zi4r-0-x-x-109">meta_r </span>will individually
process any hill whose center is too close to the <span class="t1-zi4r-0-x-x-109">upperBoundary</span>, more precisely within fewer grid points than 6 times the
Gaussian <!--l. 5636--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>σ</mi></math>
parameter plus one. It goes without saying that if the colvar <span class="t1-zi4r-0-x-x-109">r </span>represents a distance between two freely-moving
molecules, it will cross this “threshold" rather frequently.
</p><!--l. 5639--><p class="indent">   In this example, where the value of <span class="t1-zi4r-0-x-x-109">hillWidth </span>(<!--l. 5639--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>2</mn><mi>σ</mi></math>)
amounts to 2 grid points, the threshold is 6+1 = 7 grid points away from <span class="t1-zi4r-0-x-x-109">upperBoundary</span>. In explicit units, the <span class="t1-zi4r-0-x-x-109">width</span>
of <!--l. 5640--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>r</mi></math> is
<!--l. 5640--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>w</mi></mrow><mrow><mi>r</mi></mrow></msub> <mo class="MathClass-rel">=</mo></math> 0.2 Å, and the
threshold is 15.0 - 7<!--l. 5640--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-bin">×</mo></math>0.2
= 13.6 Å.
</p><!--l. 5642--><p class="indent">   The <span class="t1-zi4r-0-x-x-109">wall_r </span>restraint included in the example prevents this: the position of its <span class="t1-zi4r-0-x-x-109">upperWall </span>is 13 Å, i.e. 3 grid points
below the buffer's threshold (13.6 Å). For the chosen value of <span class="t1-zi4r-0-x-x-109">upperWallConstant</span>, the energy of the <span class="t1-zi4r-0-x-x-109">wall_r </span>bias
at <span class="t1-zi4r-0-x-x-109">r </span>= <!--l. 5643--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>r</mi></mrow><mrow><mi mathvariant="normal">upper</mi></mrow></msub></math> =
13.6 Å is: </p><table class="equation-star"><tr><td>
<!--l. 5644--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<msup><mrow><mi>E</mi></mrow><mrow><mo class="MathClass-bin">∗</mo></mrow></msup> <mo class="MathClass-rel">=</mo> <mfrac><mrow><mn>1</mn></mrow>
<mrow><mn>2</mn></mrow></mfrac><mi>k</mi><msup><mrow> <mfenced close=")" open="(" separators=""><mrow><mfrac><mrow><mi>r</mi><mo class="MathClass-bin">−</mo><msub><mrow><mi>r</mi></mrow><mrow><mi mathvariant="normal">upper</mi></mrow></msub></mrow>
<mrow><msub><mrow><mi>w</mi></mrow><mrow><mi>r</mi></mrow></msub></mrow></mfrac> </mrow></mfenced></mrow><mrow><mn>2</mn></mrow></msup> <mo class="MathClass-rel">=</mo> <mfrac><mrow><mn>1</mn></mrow>
<mrow><mn>2</mn></mrow></mfrac><mn>2</mn><mo class="MathClass-punc">.</mo><mn>0</mn><msup><mrow> <mfenced close=")" open="(" separators=""><mrow><mo class="MathClass-bin">−</mo><mn>3</mn></mrow></mfenced></mrow><mrow><mn>2</mn></mrow></msup> <mo class="MathClass-rel">=</mo> <mn>9</mn><mspace class="nbsp" width="1em"></mspace><mi mathvariant="normal">kcal∕mol</mi>
</math></td></tr></table>
<!--l. 5647--><p class="indent">   which results in a relative probability <!--l. 5647--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="qopname">exp</mo><!--nolimits--><mrow><mo class="MathClass-open">(</mo><mrow><mo class="MathClass-bin">−</mo><msup><mrow><mi>E</mi></mrow><mrow><mo class="MathClass-bin">∗</mo></mrow></msup><mo class="MathClass-bin">∕</mo><msub><mrow><mi>κ</mi></mrow><mrow><mi mathvariant="normal">B</mi></mrow></msub><mi>T</mi> </mrow><mo class="MathClass-close">)</mo></mrow> <mo class="MathClass-rel">≃</mo></math>
<!--l. 5647--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>3</mn><mo class="MathClass-bin">×</mo><mn>1</mn><msup><mrow><mn>0</mn></mrow><mrow><mo class="MathClass-bin">−</mo><mn>7</mn></mrow></msup></math> that <span class="t1-zi4r-0-x-x-109">r</span>
crosses the threshold. The probability that <span class="t1-zi4r-0-x-x-109">r </span>exceeds <span class="t1-zi4r-0-x-x-109">upperBoundary</span>, which is further away, has also become
vanishingly small. At that point, you may want to set <span class="t1-zi4r-0-x-x-109">hardUpperBoundary </span>to <span class="t1-zi4r-0-x-x-109">yes </span>for <span class="t1-zi4r-0-x-x-109">r</span>, and let <span class="t1-zi4r-0-x-x-109">meta_r </span>know that no
special treatment near the grid's boundaries will be needed.
</p><!--l. 5651--><p class="indent"> <span class="ptmri8t-x-x-109">What is the impact of the wall restraint onto the PMF? </span>Not a very complicated one: the PMF reconstructed
by metadynamics will simply show a sharp increase in free-energy where the wall potential kicks in
(<span class="t1-zi4r-0-x-x-109">r</span> <!--l. 5651--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>
13 Å). You may then choose between using the PMF only up until that point and discard the rest, or subtracting the
energy of the <span class="t1-zi4r-0-x-x-109">harmonicWalls </span>restraint from the PMF itself. Keep in mind, however, that the statistical convergence
of metadynamics may be less accurate where the wall potential is strong.
</p><!--l. 5655--><p class="indent">   In summary, although it would be simpler to set the wall's position <span class="t1-zi4r-0-x-x-109">upperWall </span>and the grid's boundary
<span class="t1-zi4r-0-x-x-109">upperBoundary </span>to the same number, the finite width of the Gaussian hills calls for setting the former strictly within
the latter.
</p>
<div class="flushright">
<!--l. 5658--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 5658--><p class="noindent"></p><div class="heading"><a id="sec:colvarbias_meta_basics">
<h5 class="subsubsectionHead"><span class="titlemark">6.4.2    </span> <a id="x1-1210006.4.2"></a>Basic configuration keywords</h5>
<!--l. 5658--><p class="noindent"></p></a> </div> 
<!--l. 5660--><p class="indent">   To enable a metadynamics calculation, a <span class="t1-zi4r-0-x-x-109">metadynamics {...} </span>block must be defined in the Colvars
configuration file. Its mandatory keywords are <a href="#colvarbias|colvars"><span class="t1-zi4r-0-x-x-109">colvars</span></a>, the variables involved, <a href="#metadynamics|hillWeight"><span class="t1-zi4r-0-x-x-109">hillWeight</span></a>, the weight parameter
<!--l. 5661--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>W</mi></math>, and the
widths <!--l. 5661--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>2</mn><mi>σ</mi></math>
of the Gaussian hills in each dimension given by the single dimensionless parameter <a href="#metadynamics|hillWidth"><span class="t1-zi4r-0-x-x-109">hillWidth</span></a>, or more explicitly
by the <a href="#metadynamics|gaussianSigmas"><span class="t1-zi4r-0-x-x-109">gaussianSigmas</span></a>.
</p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-121001"></a><span class="colorbox" id="colorbox269"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">name:</span></span> see definition of <a href="#sec:colvarbias"><span class="t1-zi4r-0-x-x-109">name</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="dx1-121002"></a><span class="colorbox" id="colorbox270"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">colvars:</span></span> see definition of <a href="#sec:colvarbias"><span class="t1-zi4r-0-x-x-109">colvars</span></a> (biasing and analysis methods)
                                                                                         
                                                                                         
     </li>
<li class="itemize"><a id="dx1-121003"></a><span class="colorbox" id="colorbox271"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">outputEnergy:</span></span> see definition of <a href="#sec:colvarbias"><span class="t1-zi4r-0-x-x-109">outputEnergy</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="dx1-121004"></a><span class="colorbox" id="colorbox272"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">outputFreq:</span></span> see definition of <a href="#colvarbias|outputFreq"><span class="t1-zi4r-0-x-x-109">outputFreq</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="dx1-121005"></a><span class="colorbox" id="colorbox273"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">writeTIPMF:</span></span> see definition of <a href="#sec:colvarbias"><span class="t1-zi4r-0-x-x-109">writeTIPMF</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="dx1-121006"></a><span class="colorbox" id="colorbox274"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">writeTISamples:</span></span> see definition of <a href="#sec:colvarbias"><span class="t1-zi4r-0-x-x-109">writeTISamples</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="dx1-121007"></a><span class="colorbox" id="colorbox275"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">stepZeroData:</span></span> see definition of <a href="#sec:colvarbias"><span class="t1-zi4r-0-x-x-109">stepZeroData</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="metadynamics|hillWeight"> </a> <a id="dx1-121008"></a><span class="colorbox" id="colorbox276"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">hillWeight</span></span> <!--l. 5681--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Height
     of each hill (kcal/mol)<!--l. 5681--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">metadynamics</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive decimal <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>This option sets the height <!--l. 5681--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>W</mi></math>
     of the Gaussian hills that are added during this run. Lower values provide more accurate sampling of
     the system's degrees of freedom at the price of longer simulation times to complete a PMF calculation
     based on metadynamics.
     </li>
<li class="itemize"><a id="metadynamics|hillWidth"> </a> <a id="dx1-121009"></a><span class="colorbox" id="colorbox277"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">hillWidth</span></span> <!--l. 5693--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Width
     <!--l. 5693--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>2</mn><mi>σ</mi></math>
     of a Gaussian hill, measured in number of grid points<!--l. 5693--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">metadynamics</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive decimal <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>This keyword sets the Gaussian width <!--l. 5693--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>2</mn><msub><mrow><mi>σ</mi></mrow><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow></msub></math>
     for all colvars, expressed in <span class="ptmri8t-x-x-109">number of grid points</span>, with the grid spacing along each colvar <!--l. 5693--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>
     determined  by  the  respective  value  of  <a href="#colvar|width"><span class="t1-zi4r-0-x-x-109">width</span></a>.  Values  between  1  and  3  are  recommended  for  this
     option: smaller numbers will fail to adequately interpolate each Gaussian function <span class="cite">[<a href="#XFiorin2013">1</a>]</span>, while larger
     values  may  be  unable  to  account  for  steep  free-energy  gradients.  The  values  of  each  half-width
     <!--l. 5693--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>σ</mi></mrow><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow></msub></math>
     in the physical units of <!--l. 5693--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></math>
     are also printed by NAMD at initialization time; alternatively, they may be set explicitly via <a href="#metadynamics|gaussianSigmas"><span class="t1-zi4r-0-x-x-109">gaussianSigmas</span></a>.
     </li>
<li class="itemize"><a id="metadynamics|gaussianSigmas"> </a> <a id="dx1-121010"></a><span class="colorbox" id="colorbox278"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">gaussianSigmas</span></span> <!--l. 5705--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Half-widths
     <!--l. 5705--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>σ</mi></math>
     of the Gaussian hill (one for each colvar)<!--l. 5705--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">metadynamics</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>space-separated list of decimals <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>This option sets the parameters <!--l. 5705--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>σ</mi></mrow><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow></msub></math>
     of the Gaussian hills along each colvar <!--l. 5705--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></math>,
     expressed in <span class="ptmri8t-x-x-109">the same unit of </span><!--l. 5705--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></math>.
                                                                                         
                                                                                         
     No restrictions are placed on each value, but a warning will be printed if <a href="#metadynamics|useGrids"><span class="t1-zi4r-0-x-x-109">useGrids</span></a> is <span class="t1-zi4r-0-x-x-109">on </span>and the
     Gaussian width <!--l. 5705--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>2</mn><msub><mrow><mi>σ</mi></mrow><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow></msub></math>
     is smaller than the corresponding grid spacing, <!--l. 5705--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="monospace">width</mi><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math>.
     If not given, default values will be computed from the dimensionless number <a href="#metadynamics|hillWidth"><span class="t1-zi4r-0-x-x-109">hillWidth</span></a>.
     </li>
<li class="itemize"><a id="metadynamics|newHillFrequency"> </a> <a id="dx1-121011"></a><span class="colorbox" id="colorbox279"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">newHillFrequency</span></span> <!--l. 5718--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Frequency
     of hill creation<!--l. 5718--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">metadynamics</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive integer <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">1000</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>This  option  sets  the  number  of  steps  after  which  a  new  Gaussian  hill  is  added  to
     the  metadynamics  potential.  The  product  of  this  number  and  the  integration  time-step  defines  the
     parameter <!--l. 5718--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>δ</mi><mi>t</mi></math>
     in eq. <a href="#x1-119001r34">34<!--tex4ht:ref: eq:colvars_meta_pot --></a>. Higher values provide more accurate statistical sampling, at the price of longer simulation
     times to complete a PMF calculation.
     </li></ul>
<div class="flushright">
<!--l. 5723--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 5723--><p class="noindent"></p><div class="heading"><a id="sec:colvarbias_meta_output">
<h5 class="subsubsectionHead"><span class="titlemark">6.4.3    </span> <a id="x1-1220006.4.3"></a>Output files</h5>
<!--l. 5723--><p class="noindent"></p></a> </div> 
<!--l. 5725--><p class="indent">   When interpolating grids are enabled (default behavior), the PMF is written by default every
<span class="t1-zi4r-0-x-x-109">colvarsRestartFrequency </span>steps to the file <span class="ptmri8t-x-x-109">outputName</span><span class="t1-zi4r-0-x-x-109">.pmf </span>in multicolumn text format (<a href="#x1-270003.8.5">3.8.5<!--tex4ht:ref: sec:colvar_multicolumn_grid --></a>). The following
two options allow to disable or control this behavior and to track statistical convergence:
</p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-122001"></a><span class="colorbox" id="colorbox280"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">writeFreeEnergyFile</span></span> <!--l. 5737--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Periodically
     write the PMF for visualization<!--l. 5737--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">metadynamics</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">on</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>When <span class="t1-zi4r-0-x-x-109">useGrids </span>and this option are <span class="t1-zi4r-0-x-x-109">on</span>, the PMF is written every <a href="#colvarbias|outputFreq"><span class="t1-zi4r-0-x-x-109">outputFreq</span></a> steps.
     </li>
<li class="itemize"><a id="dx1-122002"></a><span class="colorbox" id="colorbox281"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">keepFreeEnergyFiles</span></span> <!--l. 5747--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Keep
     all the PMF files<!--l. 5747--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">metadynamics</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">off</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>When <span class="t1-zi4r-0-x-x-109">writeFreeEnergyFile </span>and this option are <span class="t1-zi4r-0-x-x-109">on</span>, the step number is included in the
     file name, thus generating a series of PMF files. Activating this option can be useful to follow more
     closely the convergence of the simulation, by comparing PMFs separated by short times.
     </li>
<li class="itemize"><a id="metadynamics|writeHillsTrajectory"> </a> <a id="dx1-122003"></a><span class="colorbox" id="colorbox282"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">writeHillsTrajectory</span></span> <!--l. 5765--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Write
     a log of new hills<!--l. 5765--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">metadynamics</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">off</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>If this option is <span class="t1-zi4r-0-x-x-109">on</span>, a file containing the Gaussian hills written by the <span class="t1-zi4r-0-x-x-109">metadynamics</span>
     bias, with the name:<br class="newline"/>“<span class="ptmri8t-x-x-109">outputName</span><span class="t1-zi4r-0-x-x-109">.colvars.</span><!--l. 5765--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="t1-zi4r-0-x-x-109">name</span><!--l. 5765--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><span class="t1-zi4r-0-x-x-109">.hills.traj</span>",<br class="newline"/>which can be useful to post-process the time series of the Gassian hills. Each line is written every
     <span class="t1-zi4r-0-x-x-109">newHillFrequency</span>, regardless of the value of <a href="#colvarbias|outputFreq"><span class="t1-zi4r-0-x-x-109">outputFreq</span></a>. When <span class="t1-zi4r-0-x-x-109">multipleReplicas </span>is <span class="t1-zi4r-0-x-x-109">on</span>, its name
     is changed to:<br class="newline"/>“<span class="ptmri8t-x-x-109">outputName</span><span class="t1-zi4r-0-x-x-109">.colvars.</span><!--l. 5765--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="t1-zi4r-0-x-x-109">name</span><!--l. 5765--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><span class="t1-zi4r-0-x-x-109">.</span><!--l. 5765--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="t1-zi4r-0-x-x-109">replicaID</span><!--l. 5765--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><span class="t1-zi4r-0-x-x-109">.hills.traj</span>".<br class="newline"/>The columns of this file are the centers of the hills, <!--l. 5765--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><msup><mrow><mi>t</mi></mrow><mrow><mi>′</mi></mrow></msup></mrow><mo class="MathClass-close">)</mo></mrow></math>,
     followed by the half-widths, <!--l. 5765--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>σ</mi></mrow><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow></msub></math>,
     and the weight, <!--l. 5765--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>W</mi></math>.
     <span class="ptmb8t-x-x-109">Note: </span>prior to version 2020-02-24, the full-width <!--l. 5765--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>2</mn><mi>σ</mi></math>
     of the Gaussian was reported in lieu of <!--l. 5765--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>σ</mi></math>.
     </li></ul>
<div class="flushright">
<!--l. 5770--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 5770--><p class="noindent"></p><div class="heading"><a id="sec:colvarbias_meta_performance">
<h5 class="subsubsectionHead"><span class="titlemark">6.4.4    </span> <a id="x1-1230006.4.4"></a>Performance optimization</h5>
<!--l. 5770--><p class="noindent"></p></a> </div>  The following options control the computational cost of metadynamics calculations, but do not affect results.
Default values are chosen to minimize such cost with no loss of accuracy.

     <ul class="itemize1">
<li class="itemize"><a id="metadynamics|useGrids"> </a> <a id="dx1-123001"></a><span class="colorbox" id="colorbox283"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">useGrids</span></span> <!--l. 5792--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Interpolate
     the hills with grids<!--l. 5792--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">metadynamics</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">on</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>This option discretizes all hills for improved performance, accumulating their energy
     and their gradients on two separate grids of equal spacing. Grids are defined by the values of <span class="t1-zi4r-0-x-x-109">lowerBoundary</span>,
                                                                                         
                                                                                         
     <span class="t1-zi4r-0-x-x-109">upperBoundary </span>and  <span class="t1-zi4r-0-x-x-109">width </span>for  each  colvar.  Currently,  this  option  is  implemented  for  all  types  of
     variables except the non-scalar types (<span class="t1-zi4r-0-x-x-109">distanceDir </span>or <span class="t1-zi4r-0-x-x-109">orientation</span>). If <span class="t1-zi4r-0-x-x-109">expandBoundaries </span>is defined
     in one of the colvars, grids are automatically expanded along the direction of that colvar.
     </li>
<li class="itemize"><a id="dx1-123002"></a><span class="colorbox" id="colorbox284"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">rebinGrids</span></span> <!--l. 5805--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Recompute
     the grids when reading a state file<!--l. 5805--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">metadynamics</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">off</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>When restarting from a state file, the grid's parameters (boundaries and widths) saved
     in the state file override those in the configuration file. Enabling this option forces the grids to match
     those in the current configuration file.
     </li>
<li class="itemize"><a id="dx1-123003"></a><span class="colorbox" id="colorbox285"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">keepHills</span></span> <!--l. 5819--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Write
     each individual hill to the state file<!--l. 5819--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">metadynamics</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">off</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>When  <span class="t1-zi4r-0-x-x-109">useGrids </span>and  this  option  are  <span class="t1-zi4r-0-x-x-109">on</span>,  all  hills  are  saved  to  the  state  file  in  their
     analytic  form,  alongside  their  grids.  This  makes  it  possible  to  later  use  exact  analytic  Gaussians
     for  <span class="t1-zi4r-0-x-x-109">rebinGrids</span>.  To  only  keep  track  of  the  history  of  the  added  hills,  <span class="t1-zi4r-0-x-x-109">writeHillsTrajectory </span>is
     preferable.
     </li></ul>
<div class="flushright">
<!--l. 5824--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 5824--><p class="noindent"></p><div class="heading"><a id="sec:colvarbias_ebmeta">
<h5 class="subsubsectionHead"><span class="titlemark">6.4.5    </span> <a id="x1-1240006.4.5"></a>Ensemble-Biased Metadynamics</h5>
<!--l. 5824--><p class="noindent"></p></a> </div> 
<!--l. 5826--><p class="indent">   The ensemble-biased metadynamics (EBMetaD) approach <span class="cite">[<a href="#XMarinelli2015">35</a>]</span> is designed to reproduce a target
probability distribution along selected collective variables. Standard metadynamics can be seen as a
special case of EBMetaD with a flat distribution as target. This is achieved by weighing the Gaussian
functions used in the metadynamics approach by the inverse of the target probability distribution: </p><table class="equation"><tr><td>
<a id="x1-124001r37"></a>
<!--l. 5829--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<msub><mrow><mi>V</mi></mrow><mrow><mi mathvariant="normal">EBmetaD</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow></mrow><mo class="MathClass-close">)</mo></mrow><mspace class="tmspace" width="3.04074pt"></mspace> <mo class="MathClass-rel">=</mo> <mspace class="tmspace" width="3.04074pt"></mspace><munderover accent="false" accentunder="false"><mrow><mo mathsize="big">∑</mo></mrow><mrow><msup><mrow><mi>t</mi></mrow><mrow><mi>′</mi></mrow></msup><mo class="MathClass-rel">=</mo><mi>δ</mi><mi>t</mi><mo class="MathClass-punc">,</mo><mn>2</mn><mi>δ</mi><mi>t</mi><mo class="MathClass-punc">,</mo><mo>…</mo></mrow><mrow><msup><mrow><mi>t</mi></mrow><mrow><mi>′</mi></mrow></msup><mo class="MathClass-rel">&lt;</mo><mi>t</mi></mrow></munderover> <mfrac><mrow><mi>W</mi></mrow>
<mrow><mo class="qopname">exp</mo><!--nolimits--> <mfenced close=")" open="(" separators=""><mrow><msub><mrow><mi>S</mi></mrow><mrow><mi>ρ</mi></mrow></msub></mrow></mfenced><msub><mrow><mi>ρ</mi></mrow><mrow><mi>e</mi><mi>x</mi><mi>p</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle><mrow><mo class="MathClass-open">(</mo><mrow><msup><mrow><mi>t</mi></mrow><mrow><mi>′</mi></mrow></msup></mrow><mo class="MathClass-close">)</mo></mrow></mrow><mo class="MathClass-close">)</mo></mrow></mrow></mfrac><mspace class="tmspace" width="2.43306pt"></mspace><munderover accent="false" accentunder="false"><mrow><mo mathsize="big">∏</mo></mrow><mrow><mi>i</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><msub><mrow><mi>N</mi></mrow><mrow><mi mathvariant="normal">cv</mi></mrow></msub></mrow></munderover><mo class="qopname"> exp</mo><!--nolimits--> <mfenced close=")" open="(" separators=""><mrow><mo class="MathClass-bin">−</mo><mfrac><mrow><msup><mrow><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>t</mi></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-bin">−</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><msup><mrow><mi>t</mi></mrow><mrow><mi>′</mi></mrow></msup></mrow><mo class="MathClass-close">)</mo></mrow></mrow><mo class="MathClass-close">)</mo></mrow></mrow><mrow><mn>2</mn></mrow></msup></mrow>
<mrow><mn>2</mn><msubsup><mrow><mi>σ</mi></mrow><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow><mrow><mn>2</mn></mrow></msubsup></mrow></mfrac> </mrow></mfenced><mi mathvariant="normal">,</mi>
</math></td><td class="eq-no">(37)</td></tr></table>
<!--l. 5838--><p class="indent">   where <!--l. 5838--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ρ</mi></mrow><mrow><mi>e</mi><mi>x</mi><mi>p</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></mrow><mo class="MathClass-close">)</mo></mrow></math> is the target
probability distribution and <!--l. 5838--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>S</mi></mrow><mrow><mi>ρ</mi></mrow></msub> <mo class="MathClass-rel">=</mo> <mo class="MathClass-bin">−</mo><mo class="MathClass-op">∫
 <!--nolimits--></mo><!--nolimits--><msub><mrow><mi>ρ</mi></mrow><mrow><mi>e</mi><mi>x</mi><mi>p</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></mrow><mo class="MathClass-close">)</mo></mrow><mo class="qopname">log</mo><!--nolimits--><msub><mrow><mi>ρ</mi></mrow><mrow><mi>e</mi><mi>x</mi><mi>p</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></mrow><mo class="MathClass-close">)</mo></mrow><mspace class="thinspace" width="0.3em"></mspace><mi mathvariant="normal">d</mi><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></math>
its corresponding differential entropy. The method is designed so that during the simulation the resulting distribution of the
collective variable <!--l. 5839--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></math>
converges to <!--l. 5839--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ρ</mi></mrow><mrow><mi>e</mi><mi>x</mi><mi>p</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></mrow><mo class="MathClass-close">)</mo></mrow></math>.
A practical application of EBMetaD is to reproduce an “experimental" probability distribution, for example the
distance distribution between spectroscopic labels inferred from Förster resonance energy transfer (FRET) or
double electron-electron resonance (DEER) experiments <span class="cite">[<a href="#XMarinelli2015">35</a>]</span>.
</p><!--l. 5842--><p class="indent">   The PMF along <!--l. 5842--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>
can be estimated from the bias potential and the target ditribution <span class="cite">[<a href="#XMarinelli2015">35</a>]</span>: </p><table class="equation"><tr><td> <a id="x1-124002r38"></a>
<!--l. 5843--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>A</mi><mrow><mo class="MathClass-open">(</mo><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></mrow><mo class="MathClass-close">)</mo></mrow><mspace class="tmspace" width="3.04074pt"></mspace> <mo class="MathClass-rel">≃</mo><mspace class="tmspace" width="3.04074pt"></mspace><mo class="MathClass-bin">−</mo><msub><mrow><mi>V</mi></mrow><mrow><mi mathvariant="normal">EBmetaD</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-bin">−</mo><msub><mrow><mi>κ</mi></mrow><mrow><mi mathvariant="normal">B</mi></mrow></msub><mi>T</mi> <mo class="qopname">log</mo><!--nolimits--><msub><mrow><mi>ρ</mi></mrow><mrow><mi>e</mi><mi>x</mi><mi>p</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></mrow><mo class="MathClass-close">)</mo></mrow>
</math></td><td class="eq-no">(38)</td></tr></table>
<!--l. 5849--><p class="indent">   and obtained by enabling <span class="t1-zi4r-0-x-x-109">writeFreeEnergyFile</span>. Similarly to eq. <a href="#x1-119003r36">36<!--tex4ht:ref: eq:colvars_meta_fes_av --></a>, a more accurate estimate of the free
energy can be obtained by averaging (after an equilibration time) multiple time-dependent free energy estimates (see
<span class="t1-zi4r-0-x-x-109">keepFreeEnergyFiles</span>).
</p><!--l. 5852--><p class="indent">   The following additional options define the configuration for the ensemble-biased metadynamics approach:
</p>
<ul class="itemize1">
<li class="itemize"><a id="metadynamics|ebMeta"> </a> <a id="dx1-124003"></a><span class="colorbox" id="colorbox286"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">ebMeta</span></span> <!--l. 5865--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Perform
     ensemble-biased metadynamics<!--l. 5865--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">metadynamics</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">off</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>If enabled, this flag activates the ensemble-biased metadynamics as described by Marinelli
     et al.<span class="cite">[<a href="#XMarinelli2015">35</a>]</span>. The target distribution file, <span class="t1-zi4r-0-x-x-109">targetdistfile</span>, is then required. The keywords <span class="t1-zi4r-0-x-x-109">lowerBoundary</span>,
     <span class="t1-zi4r-0-x-x-109">upperBoundary </span>and <span class="t1-zi4r-0-x-x-109">width </span>for the respective variables are also needed to set the binning (grid) of the
     target distribution file.
     </li>
<li class="itemize"><a id="metadynamics|targetDistFile"> </a> <a id="dx1-124004"></a><span class="colorbox" id="colorbox287"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">targetDistFile</span></span> <!--l. 5876--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Target
     probability distribution file for ensemble-biased metadynamics<!--l. 5876--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">metadynamics</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>multicolumn text file <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>This file provides the target probability distribution, <!--l. 5876--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ρ</mi></mrow><mrow><mi>e</mi><mi>x</mi><mi>p</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></mrow><mo class="MathClass-close">)</mo></mrow></math>,
     reported in eq. <a href="#x1-124001r37">37<!--tex4ht:ref: eq:colvars_ebmeta_pot --></a>. The latter distribution must be a tabulated function provided in a multicolumn text
     format (see <a href="#x1-270003.8.5">3.8.5<!--tex4ht:ref: sec:colvar_multicolumn_grid --></a>). The provided distribution is then normalized.
     </li>
<li class="itemize"><a id="dx1-124005"></a><span class="colorbox" id="colorbox288"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">ebMetaEquilSteps</span></span> <!--l. 5887--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Number
     of equilibration steps for ensemble-biased metadynamics<!--l. 5887--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">metadynamics</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive integer <br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>The  EBMetaD  approach  may  introduce  large  hills  in  regions  with  small  values  of
     the target probability distribution (eq. <a href="#x1-124001r37">37<!--tex4ht:ref: eq:colvars_ebmeta_pot --></a>). This happens, for example, if the probability distribution
     sampled by a conventional molecular dynamics simulation is significantly different from the target
     distribution. This may lead to instabilities at the beginning of the simulation related to large biasing
     forces. In this case, it is useful to introduce an equilibration stage in which the bias potential gradually
     switches from standard metadynamics (eq. <a href="#x1-119001r34">34<!--tex4ht:ref: eq:colvars_meta_pot --></a>) to EBmetaD (eq. <a href="#x1-124001r37">37<!--tex4ht:ref: eq:colvars_ebmeta_pot --></a>) as <!--l. 5887--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>λ</mi><msub><mrow><mi>V</mi></mrow><mrow><mi mathvariant="normal">meta</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-bin">+</mo><mrow><mo class="MathClass-open">(</mo><mrow><mn>1</mn><mo class="MathClass-bin">−</mo><mi>λ</mi></mrow><mo class="MathClass-close">)</mo></mrow><msub><mrow><mi>V</mi></mrow><mrow><mi mathvariant="normal">EBmetaD</mi></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></mrow><mo class="MathClass-close">)</mo></mrow></math>,
     where <!--l. 5887--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>λ</mi> <mo class="MathClass-rel">=</mo> <mrow><mo class="MathClass-open">(</mo><mrow><mi mathvariant="monospace">ebMetaEquilSteps</mi><mo class="MathClass-bin">−</mo><mi mathvariant="monospace">step</mi></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-bin">∕</mo><mi mathvariant="monospace">ebMetaEquilSteps</mi></math>
     and <span class="t1-zi4r-0-x-x-109">step </span>is the current simulation step number.
     </li>
<li class="itemize"><a id="dx1-124006"></a><span class="colorbox" id="colorbox289"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">targetDistMinVal</span></span> <!--l. 5900--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Minimum
     value of the target distribution in reference to its <span class="t1-zi4r-0-x-x-109">maximum value</span><!--l. 5900--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">metadynamics</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive decimal <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>It is useful to set a minimum value of the target probability distribution to avoid values
     of the latter that are nearly zero, leading to very large hills. This parameter sets the minimum value of
     the target probability distribution that is expressed as a fraction of its <span class="t1-zi4r-0-x-x-109">maximum value</span>: <span class="t1-zi4r-0-x-x-109">minimum value</span>
<span class="t1-zi4r-0-x-x-109">= maximum value X targetDistMinVal</span>. This implies that 0 <span class="t1-zi4r-0-x-x-109"><targetdistminval>1 and its default
     value is set to 1/1000000. To avoid divisions by zero (see eq. <a href="#x1-124001r37">37<!--tex4ht:ref: eq:colvars_ebmeta_pot --></a>), if <span class="t1-zi4r-0-x-x-109">targetDistMinVal </span>is set as zero,
     values of <!--l. 5900--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ρ</mi></mrow><mrow><mi>e</mi><mi>x</mi><mi>p</mi></mrow></msub></math>
     equal to zero are replaced by the smallest positive value read in the same file.
     </targetdistminval></span></li></ul>
<!--l. 5904--><p class="indent">   As with standard metadynamics, multidimensional probability distributions can be targeted using a single
<span class="t1-zi4r-0-x-x-109">metadynamics </span>block using multiple <span class="t1-zi4r-0-x-x-109">colvars </span>and a multidimensional target distribution file (see <a href="#x1-270003.8.5">3.8.5<!--tex4ht:ref: sec:colvar_multicolumn_grid --></a>). Instead,
multiple probability distributions on different variables can be targeted separately in the same simulation by
introducing multiple <span class="t1-zi4r-0-x-x-109">metadynamics </span>blocks with the <span class="t1-zi4r-0-x-x-109">ebMeta </span>option. <br class="newline"/>
</p><!--l. 5908--><p class="noindent"><span class="ptmb8t-x-x-109">Example: </span>EBmetaD configuration for a single variable.</p>
<div class="mdframed" id="mdframed-53">
<!--l. 5909--><p class="noindent"></p><pre>colvar {<br class="newline"/>  name r <br class="newline"/>  distance {<br class="newline"/>    group1 { atomNumbers 991 992 }<br class="newline"/>    group2 { atomNumbers 1762 1763 }<br class="newline"/>  }<br class="newline"/>  upperBoundary 100.0 <br class="newline"/>  width 0.1 <br class="newline"/>}<br class="newline"/><br class="newline"/>metadynamics {<br class="newline"/>  name ebmeta<br class="newline"/>  colvars r<br class="newline"/>  hillWeight 0.01<br class="newline"/>  hillWidth 3.0<br class="newline"/>  ebMeta on<br class="newline"/>  targetDistFile targetdist1.dat<br class="newline"/>  ebMetaEquilSteps 500000<br class="newline"/>} </pre>
</div>
<!--l. 5931--><p class="noindent">where <span class="t1-zi4r-0-x-x-109">targetdist1.dat </span>is a text file in “multicolumn" format (<a href="#x1-270003.8.5">3.8.5<!--tex4ht:ref: sec:colvar_multicolumn_grid --></a>) with the same <span class="t1-zi4r-0-x-x-109">width </span>as the variable <span class="t1-zi4r-0-x-x-109">r </span>(0.1 in
this case):<br class="newline"/><!--tex4ht:inline--></p><div class="tabular"> <table cellpadding="0" cellspacing="0" class="tabular" id="TBL-6"><colgroup id="TBL-6-1g"><col id="TBL-6-1"/><col id="TBL-6-2"/><col id="TBL-6-3"/><col id="TBL-6-4"/><col id="TBL-6-5"/><col id="TBL-6-6"/></colgroup><tr id="TBL-6-1-" style="vertical-align:baseline;"><td class="td11" id="TBL-6-1-1" style="text-align:left; white-space:nowrap;"> </td></tr><tr id="TBL-6-2-" style="vertical-align:baseline;"><td class="td11" id="TBL-6-2-1" style="text-align:left; white-space:nowrap;"> #</td><td class="td11" id="TBL-6-2-2" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">1     </span></td><td class="td11" id="TBL-6-2-3" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-6-2-4" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-6-2-5" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-6-2-6" style="text-align:left; white-space:nowrap;"> </td>
</tr><tr id="TBL-6-3-" style="vertical-align:baseline;"><td class="td11" id="TBL-6-3-1" style="text-align:left; white-space:nowrap;"> #  </td><td class="td11" id="TBL-6-3-2" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">0.0   </span></td><td class="td11" id="TBL-6-3-3" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">0.1    </span></td><td class="td11" id="TBL-6-3-4" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">1000 </span></td><td class="td11" id="TBL-6-3-5" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">0 </span></td></tr><tr id="TBL-6-4-" style="vertical-align:baseline;"><td class="td11" id="TBL-6-4-1" style="text-align:left; white-space:nowrap;"> </td>
</tr><tr id="TBL-6-5-" style="vertical-align:baseline;"><td class="td11" id="TBL-6-5-1" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-6-5-2" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">0.05 </span></td><td class="td11" id="TBL-6-5-3" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">0.0012 </span></td>
</tr><tr id="TBL-6-6-" style="vertical-align:baseline;"><td class="td11" id="TBL-6-6-1" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-6-6-2" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">0.15 </span></td><td class="td11" id="TBL-6-6-3" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">0.0014 </span></td>
</tr><tr id="TBL-6-7-" style="vertical-align:baseline;"><td class="td11" id="TBL-6-7-1" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-6-7-2" style="text-align:left; white-space:nowrap;"> …          </td><td class="td11" id="TBL-6-7-3" style="text-align:left; white-space:nowrap;"> …            </td>
</tr><tr id="TBL-6-8-" style="vertical-align:baseline;"><td class="td11" id="TBL-6-8-1" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-6-8-2" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">99.95 </span></td><td class="td11" id="TBL-6-8-3" style="text-align:left; white-space:nowrap;"> <span class="t1-zi4r-0-x-x-109">0.0010 </span></td>
</tr><tr id="TBL-6-9-" style="vertical-align:baseline;"><td class="td11" id="TBL-6-9-1" style="text-align:left; white-space:nowrap;"> </td>
</tr><tr id="TBL-6-10-" style="vertical-align:baseline;"><td class="td11" id="TBL-6-10-1" style="text-align:left; white-space:nowrap;"> </td></tr></table></div>
<!--l. 5944--><p class="indent"> <span class="ptmb8t-x-x-109">Tip: </span>Besides setting a meaninful value for <span class="t1-zi4r-0-x-x-109">targetDistMinVal</span>, the exploration of unphysically low values
of the target distribution (which would lead to very large hills and possibly numerical instabilities)
can be also prevented by restricting sampling to a given interval, using e.g. <span class="t1-zi4r-0-x-x-109">harmonicWalls </span>restraint
(<a href="#x1-1320006.7">6.7<!--tex4ht:ref: sec:colvarbias_harmonic_walls --></a>).
</p>
<div class="flushright">
<!--l. 5949--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 5949--><p class="noindent"></p><div class="heading"><a id="sec:colvarbias_well_tempered">
<h5 class="subsubsectionHead"><span class="titlemark">6.4.6    </span> <a id="x1-1250006.4.6"></a>Well-tempered metadynamics</h5>
<!--l. 5949--><p class="noindent"></p></a> </div>  The following options define the configuration for the “well-tempered" metadynamics approach
<span class="cite">[<a href="#XBarducci2008">36</a>]</span>:

     <ul class="itemize1">
<li class="itemize"><a id="dx1-125001"></a><span class="colorbox" id="colorbox290"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">wellTempered</span></span> <!--l. 5964--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Perform
     well-tempered metadynamics<!--l. 5964--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">metadynamics</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">off</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>If enabled, this flag causes well-tempered metadynamics as described by Barducci et
     al.<span class="cite">[<a href="#XBarducci2008">36</a>]</span> to be performed, rather than standard metadynamics. The parameter <span class="t1-zi4r-0-x-x-109">biasTemperature </span>is then
     required. This feature was contributed by Li Li (Luthey-Schulten group, Department of Chemistry,
     UIUC).
                                                                                         
                                                                                         
     </li>
<li class="itemize"><a id="dx1-125002"></a><span class="colorbox" id="colorbox291"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">biasTemperature</span></span> <!--l. 5978--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Temperature
     bias for well-tempered metadynamics<!--l. 5978--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">metadynamics</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive decimal <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>When running metadynamics in the long time limit, collective variable space is sampled
     to a modified temperature <!--l. 5978--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>T</mi> <mo class="MathClass-bin">+</mo><mi>Δ</mi><mi>T</mi> </math>.
     In conventional metadynamics, the temperature “boost" <!--l. 5978--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>Δ</mi><mi>T</mi> </math>
     would constantly increases with time. Instead, in well-tempered metadynamics <!--l. 5978--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>Δ</mi><mi>T</mi> </math>
     must  be  defined  by  the  user  via  <span class="t1-zi4r-0-x-x-109">biasTemperature</span>.  The  written  PMF  includes  the  scaling  factor
     <!--l. 5978--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">(</mo><mrow><mi>T</mi> <mo class="MathClass-bin">+</mo><mi>Δ</mi><mi>T</mi> </mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-bin">∕</mo><mi>Δ</mi><mi>T</mi> </math>
<span class="cite">[<a href="#XBarducci2008">36</a>]</span>. A careful choice of <!--l. 5978--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>Δ</mi><mi>T</mi> </math>
     determines the sampling and convergence rate, and is hence crucial to the success of a well-tempered
     metadynamics simulation.</li></ul>
<div class="flushright">
<!--l. 5982--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 5982--><p class="noindent"></p><div class="heading"><a id="sec:colvarbias_meta_mr">
<h5 class="subsubsectionHead"><span class="titlemark">6.4.7    </span> <a id="x1-1260006.4.7"></a>Multiple-walker metadynamics</h5>
<!--l. 5982--><p class="noindent"></p></a> </div>  Metadynamics calculations can be performed concurrently by multiple replicas that share a common
history. This variant of the method is called multiple-walker metadynamics <span class="cite">[<a href="#XRaiteri2006">37</a>]</span>: the Gaussian hills of all
replicas are periodically combined into a single biasing potential, intended to converge to a single
PMF.
<!--l. 5986--><p class="indent">   In the implementation here described <span class="cite">[<a href="#XFiorin2013">1</a>]</span>, replicas communicate through files. This arrangement allows
launching the replicas either (1) as a bundle (i.e. a single job in a cluster's queueing system) or (2) as fully
independent runs (i.e. as separate jobs for the queueing system). One advantage of the use case (1) is that an
identical Colvars configuration can be used for all replicas (otherwise, <span class="t1-zi4r-0-x-x-109">replicaID </span>needs to be manually set to a
different string for each replica). However, the use case (2) is less demanding in terms of high-performance
computing resources: a typical scenario would be a computer cluster (including virtual servers from a cloud
provider) where not all nodes are connected to each other at high speed, and thus each replica runs on a small group
of nodes or a single node.
</p><!--l. 5991--><p class="indent">   Whichever way the replicas are started (coupled or not), a shared filesystem is needed so that each replica can
read the files created by the others: paths to these files are stored in the shared file <span class="t1-zi4r-0-x-x-109">replicasRegistry</span>. This file, and
those listed in it, are read every <span class="t1-zi4r-0-x-x-109">replicaUpdateFrequency </span>steps. Each time the Colvars state file is written (for
example, <span class="t1-zi4r-0-x-x-109">colvarsRestartFrequency </span>steps), the file named:<br class="newline"/><span class="ptmri8t-x-x-109">outputName</span><span class="t1-zi4r-0-x-x-109">.colvars.</span><span class="ptmri8t-x-x-109">name</span><span class="t1-zi4r-0-x-x-109">.</span><span class="ptmri8t-x-x-109">replicaID</span><span class="t1-zi4r-0-x-x-109">.state</span><br class="newline"/>is written as well; this file contains only the state of the metadynamics bias, which the other replicas will read in
turn. In between the times when this file is modified/replaced, new hills are also temporarily written to the file
named:<br class="newline"/><span class="ptmri8t-x-x-109">outputName</span><span class="t1-zi4r-0-x-x-109">.colvars.</span><span class="ptmri8t-x-x-109">name</span><span class="t1-zi4r-0-x-x-109">.</span><span class="ptmri8t-x-x-109">replicaID</span><span class="t1-zi4r-0-x-x-109">.hills</span><br class="newline"/>Both files are only used for communication, and may be deleted after the replica begins writing files with a new
<span class="ptmri8t-x-x-109">outputName</span>.<br class="newline"/>
</p><!--l. 6000--><p class="noindent"><span class="ptmb8t-x-x-109">Example: </span>Multiple-walker metadynamics with file-based communication.<br class="newline"/></p>
<div class="mdframed" id="mdframed-54">
<!--l. 6001--><p class="noindent"></p><pre>metadynamics {<br class="newline"/>  name mymtd<br class="newline"/>  colvars x<br class="newline"/>  hillWeight 0.001<br class="newline"/>  newHillFrequency 1000<br class="newline"/>  hillWidth 3.0<br class="newline"/>  <br class="newline"/>  multipleReplicas on<br class="newline"/>  replicasRegistry /shared-folder/mymtd-replicas.txt<br class="newline"/>  replicaUpdateFrequency 50000 # Best if larger than newHillFrequency<br class="newline"/>} </pre>
</div>
<!--l. 6015--><p class="indent">   The following are the multiple-walkers related options:
</p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-126001"></a><span class="colorbox" id="colorbox292"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">multipleReplicas</span></span> <!--l. 6026--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Enable
     multiple-walker metadynamics<!--l. 6026--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">metadynamics</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">off</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>This option turns on multiple-walker communication between replicas.
     </li>
<li class="itemize"><a id="dx1-126002"></a><span class="colorbox" id="colorbox293"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">replicasRegistry</span></span> <!--l. 6036--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Multiple
     replicas database file<!--l. 6036--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">metadynamics</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>UNIX filename <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>If <span class="t1-zi4r-0-x-x-109">multipleReplicas </span>is <span class="t1-zi4r-0-x-x-109">on</span>, this option sets the path to the replicas' shared database
     file. It is best to use an absolute path (especially when running individual replicas in separate folders).
     </li>
<li class="itemize"><a id="dx1-126003"></a><span class="colorbox" id="colorbox294"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">replicaUpdateFrequency</span></span> <!--l. 6046--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>How
     often hills are shared between replicas<!--l. 6046--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">metadynamics</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive integer <br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>If  <span class="t1-zi4r-0-x-x-109">multipleReplicas </span>is  <span class="t1-zi4r-0-x-x-109">on</span>,  this  option  sets  the  number  of  steps  after  which  each
     replica tries to read the other replicas' files. On a networked file system, it is best to use a number of
     steps that corresponds to at least a minute of wall time.
     </li>
<li class="itemize"><a id="dx1-126004"></a><span class="colorbox" id="colorbox295"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">replicaID</span></span> <!--l. 6058--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Set
     the identifier for this replica<!--l. 6058--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">metadynamics</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>string <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span>replica index (only if a shared communicator is used) <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>If <span class="t1-zi4r-0-x-x-109">multipleReplicas </span>is <span class="t1-zi4r-0-x-x-109">on</span>, this option sets a unique identifier for this replicas. When
     the replicas are launched in a single command (i.e. they share a parallel communicator and are tightly
     synchronized) this value is optional, and defaults to the replica's numeric index (starting at zero).
     However, when the replicas are launched as independent runs this option is required.
     </li>
<li class="itemize"><a id="dx1-126005"></a><span class="colorbox" id="colorbox296"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">writePartialFreeEnergyFile</span></span> <!--l. 6070--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Periodically
     write the contribution to the PMF from this replica<!--l. 6070--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">metadynamics</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">off</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>If <span class="t1-zi4r-0-x-x-109">multipleReplicas </span>is <span class="t1-zi4r-0-x-x-109">on</span>, enabling this option produces an additional file <span class="ptmri8t-x-x-109">outputName</span><span class="t1-zi4r-0-x-x-109">.partial.pmf</span>,
     which can be useful to monitor the contribution of each replica to the total PMF (which is written to
     the file <span class="ptmri8t-x-x-109">outputName</span><span class="t1-zi4r-0-x-x-109">.pmf</span>). <span class="ptmb8t-x-x-109">Note: </span>the name of this file is chosen for consistency and convenience, <span class="ptmri8t-x-x-109">but</span>
<span class="ptmri8t-x-x-109">its content is not a PMF </span>and it is not expected to converge, even if the total PMF does.
     </li></ul>
<div class="flushright">
<!--l. 6076--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 6076--><p class="noindent"></p><div class="heading"><a id="sec:colvarbias_harmonic">
<h4 class="subsectionHead"><span class="titlemark">6.5    </span> <a id="x1-1270006.5"></a>Harmonic restraints</h4>
<!--l. 6076--><p class="noindent"></p></a> </div> 
<!--l. 6078--><p class="indent">   The harmonic biasing method may be used to enforce fixed or moving restraints, including variants of Steered
and Targeted MD. Within energy minimization runs, it allows for restrained minimization, e.g. to calculate relaxed
potential energy surfaces. In the context of the Colvars module, harmonic potentials are meant according to their
textbook definition: </p><table class="equation"><tr><td> <a id="x1-127001r39"></a>
<!--l. 6083--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>V</mi> <mrow><mo class="MathClass-open">(</mo><mrow><mi>ξ</mi></mrow><mo class="MathClass-close">)</mo></mrow> <mo class="MathClass-rel">=</mo> <mfrac><mrow><mn>1</mn></mrow>
<mrow><mn>2</mn></mrow></mfrac><mi>k</mi><msup><mrow> <mfenced close=")" open="(" separators=""><mrow><mfrac><mrow><mi>ξ</mi> <mo class="MathClass-bin">−</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mn>0</mn></mrow></msub></mrow>
<mrow><msub><mrow><mi>w</mi></mrow><mrow><mi>ξ</mi></mrow></msub></mrow></mfrac> </mrow></mfenced></mrow><mrow><mn>2</mn></mrow></msup>
</math></td><td class="eq-no">(39)</td></tr></table>
<!--l. 6087--><p class="indent">   There are two noteworthy aspects of this expression:
                                                                                         
                                                                                         
     </p><ol class="enumerate1">
<li class="enumerate" id="x1-127003x1">Because the <span class="ptmri8t-x-x-109">standard coefficient of </span><!--l. 6089--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>1</mn><mo class="MathClass-bin">∕</mo><mn>2</mn></math>
     of the harmonic potential is included, this expression differs from harmonic bond and angle potentials
     historically used in common force fields, where the factor was typically omitted resulting in a non-standard
     definition of the force constant.
     </li>
<li class="enumerate" id="x1-127005x2">The variable <!--l. 6090--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>
     is not only centered at <!--l. 6090--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mn>0</mn></mrow></msub></math>,
     but is also <span class="ptmri8t-x-x-109">scaled by its characteristic length scale </span><!--l. 6090--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>w</mi></mrow><mrow><mi>ξ</mi></mrow></msub></math>
     (keyword <a href="#colvar|width"><span class="t1-zi4r-0-x-x-109">width</span></a>). The resulting dimensionless variable <!--l. 6091--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi> <mo class="MathClass-rel">=</mo> <mrow><mo class="MathClass-open">(</mo><mrow><mi>ξ</mi> <mo class="MathClass-bin">−</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mn>0</mn></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-bin">∕</mo><msub><mrow><mi>w</mi></mrow><mrow><mi>ξ</mi></mrow></msub></math>
     is typically easier to treat numerically: for example, when the forces typically experienced by <!--l. 6091--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>
     are much smaller than <!--l. 6091--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>k</mi><mo class="MathClass-bin">∕</mo><msub><mrow><mi>w</mi></mrow><mrow><mi>ξ</mi></mrow></msub></math>
     and <!--l. 6091--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>k</mi></math>
     is chosen equal to <!--l. 6091--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>κ</mi></mrow><mrow><mi mathvariant="normal">B</mi></mrow></msub><mi>T</mi> </math>
     (thermal energy), the resulting probability distribution of <!--l. 6091--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math>
     is approximately a Gaussian with mean equal to 0 and standard deviation equal to 1.
     <!--l. 6093--><p class="noindent">This property can be used for setting the force constant in umbrella-sampling ensemble runs: if the
     restraint centers are chosen in increments of <!--l. 6093--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>w</mi></mrow><mrow><mi>ξ</mi></mrow></msub></math>,
     the resulting distributions of <!--l. 6093--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>
     are most often optimally overlapped. In regions where the underlying free-energy landscape induces
     highly skewed distributions of <!--l. 6094--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>,
     additional windows may be added as needed, with spacings finer than <!--l. 6094--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>w</mi></mrow><mrow><mi>ξ</mi></mrow></msub></math>.</p></li></ol>
<!--l. 6097--><p class="indent">   Beyond one dimension, the use of a scaled harmonic potential also allows a standard definition of a
multi-dimensional restraint with a unified force constant: </p><table class="equation"><tr><td> <a id="x1-127006r40"></a>
<!--l. 6098--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>V</mi> <mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-punc">,</mo><mo class="MathClass-op">…</mo><mo class="MathClass-punc">,</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>M</mi></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow> <mo class="MathClass-rel">=</mo> <mfrac><mrow><mn>1</mn></mrow>
<mrow><mn>2</mn></mrow></mfrac><mi>k</mi><munderover accent="false" accentunder="false"><mrow><mo mathsize="big">∑</mo></mrow><mrow><mi>i</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><mi>M</mi></mrow></munderover><msup><mrow> <mfenced close=")" open="(" separators=""><mrow><mfrac><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-bin">−</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mn>0</mn></mrow></msub></mrow>
<mrow><msub><mrow><mi>w</mi></mrow><mrow><mi>ξ</mi></mrow></msub></mrow></mfrac> </mrow></mfenced></mrow><mrow><mn>2</mn></mrow></msup>
</math></td><td class="eq-no">(40)</td></tr></table>
<!--l. 6103--><p class="indent">   If one-dimensional or homogeneous multi-dimensional restraints are defined, and there are no other uses for the parameter
<!--l. 6103--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>w</mi></mrow><mrow><mi>ξ</mi></mrow></msub></math>, <span class="t1-zi4r-0-x-x-109">width </span><span class="ptmri8t-x-x-109">can be left at</span>
<span class="ptmri8t-x-x-109">its default value of </span><!--l. 6103--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>1</mn></math>.
</p><!--l. 6105--><p class="indent">   A harmonic restraint is defined by a <span class="t1-zi4r-0-x-x-109">harmonic</span><span class="t1-zi4r-0-x-x-109"> {...} </span>block, which may contain the following keywords:
</p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-127007"></a><span class="colorbox" id="colorbox297"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">name:</span></span> see definition of <a href="#sec:colvarbias"><span class="t1-zi4r-0-x-x-109">name</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="dx1-127008"></a><span class="colorbox" id="colorbox298"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">colvars:</span></span> see definition of <a href="#sec:colvarbias"><span class="t1-zi4r-0-x-x-109">colvars</span></a> (biasing and analysis methods)
                                                                                         
                                                                                         
     </li>
<li class="itemize"><a id="dx1-127009"></a><span class="colorbox" id="colorbox299"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">outputEnergy:</span></span> see definition of <a href="#sec:colvarbias"><span class="t1-zi4r-0-x-x-109">outputEnergy</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="dx1-127010"></a><span class="colorbox" id="colorbox300"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">writeTIPMF:</span></span> see definition of <a href="#sec:colvarbias"><span class="t1-zi4r-0-x-x-109">writeTIPMF</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="dx1-127011"></a><span class="colorbox" id="colorbox301"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">writeTISamples:</span></span> see definition of <a href="#sec:colvarbias"><span class="t1-zi4r-0-x-x-109">writeTISamples</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="dx1-127012"></a><span class="colorbox" id="colorbox302"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">stepZeroData:</span></span> see definition of <a href="#sec:colvarbias"><span class="t1-zi4r-0-x-x-109">stepZeroData</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="harmonic|forceConstant"> </a> <a id="dx1-127013"></a><span class="colorbox" id="colorbox303"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">forceConstant</span></span> <!--l. 6129--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Scaled
     force constant (kcal/mol)<!--l. 6129--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">harmonic</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive decimal <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">1.0</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>This option defines a <span class="ptmri8t-x-x-109">scaled </span>force constant <!--l. 6129--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>k</mi></math>
     for the harmonic potential (eq. <a href="#x1-127006r40">40<!--tex4ht:ref: eq:colvarbias_harmonic_multi --></a>). To ensure consistency for multidimensional restraints, it is divided
     internally by the square of the specific <span class="t1-zi4r-0-x-x-109">width </span>of each variable (which is 1 by default). This makes all
     values effectively dimensionless and of commensurate size. For instance, if this force constant is set to
     the thermal energy <!--l. 6129--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>κ</mi></mrow><mrow><mi mathvariant="normal">B</mi></mrow></msub><mi>T</mi> </math>
     (equal to <!--l. 6129--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>R</mi><mi>T</mi> </math>
     if molar units are used), then the amplitude of the thermal fluctuations of each variable <!--l. 6129--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>
     will be on the order of its <span class="t1-zi4r-0-x-x-109">width</span>, <!--l. 6129--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>w</mi></mrow><mrow><mi>ξ</mi></mrow></msub></math>.
     This can be used to estimate the optimal spacing of umbrella-sampling windows (under the assumption
     that the force constant is larger than the curvature of the underlying free energy). <span class="ptmri8t-x-x-109">The values of the</span>
<span class="ptmri8t-x-x-109">actual force constants </span><!--l. 6129--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>k</mi><mo class="MathClass-bin">∕</mo><msubsup><mrow><mi>w</mi></mrow><mrow><mi>ξ</mi></mrow><mrow><mn>2</mn></mrow></msubsup></math>
<span class="ptmri8t-x-x-109">are always printed when the restraint is defined.</span>
</li>
<li class="itemize"><a id="harmonic|centers"> </a> <a id="dx1-127014"></a><span class="colorbox" id="colorbox304"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">centers</span></span> <!--l. 6145--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Initial
     harmonic restraint centers<!--l. 6145--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">harmonic</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>space-separated list of colvar values <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>The centers (equilibrium values) of the restraint, <!--l. 6145--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mn>0</mn></mrow></msub></math>,
     are entered here. The number of values must be the number of requested colvars. Each value is a
     decimal number if the corresponding colvar returns a scalar, a “<span class="t1-zi4r-0-x-x-109">(x, y, z)</span>" triplet if it returns a unit
     vector or a vector, and a “<span class="t1-zi4r-0-x-x-109">(q0, q1, q2, q3)</span>" quadruplet if it returns a rotational quaternion. If
     a colvar has periodicities or symmetries, its closest image to the restraint center is considered when
     calculating the harmonic potential.</li></ul>
<!--l. 6148--><p class="indent"> <span class="ptmb8t-x-x-109">Tip: </span>A complex set of restraints can be applied to a system, by defining several colvars, and applying one
or more harmonic restraints to different groups of colvars. In some cases, dozens of colvars can be
defined, but their value may not be relevant: to limit the size of the colvars trajectory file, it may be
wise to disable <span class="t1-zi4r-0-x-x-109">outputValue </span>for such “ancillary" variables, and leave it enabled only for “relevant"
ones.
</p>
<div class="flushright">
<!--l. 6157--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 6157--><p class="noindent"></p><div class="heading"><a id="sec:colvarbias_smd">
<h5 class="subsubsectionHead"><span class="titlemark">6.5.1    </span> <a id="x1-1280006.5.1"></a>Moving restraints: steered molecular dynamics</h5>
<!--l. 6157--><p class="noindent"></p></a> </div> 
<!--l. 6159--><p class="indent">   The following options allow to change gradually the centers of the harmonic restraints during a simulations.
When the centers are changed continuously, a steered MD in a collective variable space is carried
out.
</p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-128001"></a><span class="colorbox" id="colorbox305"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">targetCenters</span></span> <!--l. 6187--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Steer
     the restraint centers towards these targets<!--l. 6187--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">harmonic</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>space-separated list of colvar values <br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>When defined, the current <span class="t1-zi4r-0-x-x-109">centers </span>will be moved towards these values during the
     simulation.  By  default,  the  centers  are  moved  over  a  total  of  <span class="t1-zi4r-0-x-x-109">targetNumSteps </span>steps  by  a  linear
     interpolation, in the spirit of Steered MD. If <span class="t1-zi4r-0-x-x-109">targetNumStages </span>is set to a nonzero value, the change
     is performed in discrete stages, lasting <span class="t1-zi4r-0-x-x-109">targetNumSteps </span>steps <span class="ptmri8t-x-x-109">each</span>. This second mode may be used
     to sample successive windows in the context of an Umbrella Sampling simulation. When continuing a
     simulation run, the <span class="t1-zi4r-0-x-x-109">centers </span>specified in the configuration file <!--l. 6187--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="t1-zi4r-0-x-x-109">colvarsConfig</span><!--l. 6187--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>
     are overridden by those saved in the restart file <!--l. 6187--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="t1-zi4r-0-x-x-109">colvarsInput</span><!--l. 6187--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>.
     To perform Steered MD in an arbitrary space of colvars, it is sufficient to use this option and enable
     <span class="t1-zi4r-0-x-x-109">outputAccumulatedWork </span>and/or <span class="t1-zi4r-0-x-x-109">outputAppliedForce </span>within each of the colvars involved.
     </li>
<li class="itemize"><a id="dx1-128002"></a><span class="colorbox" id="colorbox306"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">targetNumSteps</span></span> <!--l. 6201--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Number
     of steps for steering<!--l. 6201--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">harmonic</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive integer <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>In single-stage (continuous) transformations, defines the number of MD steps required
     to move the restraint centers (or force constant) towards the values specified with <span class="t1-zi4r-0-x-x-109">targetCenters </span>or
     <span class="t1-zi4r-0-x-x-109">targetForceConstant</span>. After the target values have been reached, the centers (resp. force constant)
     are kept fixed. In multi-stage transformations, this sets the number of MD steps <span class="ptmri8t-x-x-109">per stage</span>.
     </li>
<li class="itemize"><a id="dx1-128003"></a><span class="colorbox" id="colorbox307"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">outputCenters</span></span> <!--l. 6212--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Write
     the current centers to the trajectory file<!--l. 6212--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">harmonic</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">off</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>If this option is chosen and <span class="t1-zi4r-0-x-x-109">colvarsTrajFrequency </span>is not zero, the positions of the
     restraint  centers  will  be  written  to  the  trajectory  file  during  the  simulation.  This  option  allows  to
     conveniently extract the PMF from the colvars trajectory files in a steered MD calculation.
                                                                                         
                                                                                         
     </li></ul>
<!--l. 6216--><p class="indent"> <span class="ptmb8t-x-x-109">Note on restarting moving restraint simulations: </span>Information about the current step and stage of a
simulation with moving restraints is stored in the restart file (state file). Thus, such simulations can be
run in several chunks, and restarted directly using the same colvars configuration file. In case of a
restart, the values of parameters such as <span class="t1-zi4r-0-x-x-109">targetCenters</span>, <span class="t1-zi4r-0-x-x-109">targetNumSteps</span>, etc. should not be changed
manually.
</p>
<div class="flushright">
<!--l. 6225--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 6225--><p class="noindent"></p><div class="heading"><a id="sec:colvarbias_us">
<h5 class="subsubsectionHead"><span class="titlemark">6.5.2    </span> <a id="x1-1290006.5.2"></a>Moving restraints: umbrella sampling</h5>
<!--l. 6225--><p class="noindent"></p></a> </div> 
<!--l. 6227--><p class="indent">   The centers of the harmonic restraints can also be changed in discrete stages: in this cases a one-dimensional
umbrella sampling simulation is performed. The sampling windows in simulation are calculated in sequence. The
colvars trajectory file may then be used both to evaluate the correlation times between consecutive windows, and to
calculate the frequency distribution of the colvar of interest in each window. Furthermore, frequency
distributions on a predefined grid can be automatically obtained by using the <span class="t1-zi4r-0-x-x-109">histogram </span>bias (see
<a href="#x1-1350006.10">6.10<!--tex4ht:ref: sec:colvarbias_histogram --></a>).
</p><!--l. 6232--><p class="indent">   To activate an umbrella sampling simulation, the same keywords as in the previous section can be used, with the
addition of the following: </p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-129001"></a><span class="colorbox" id="colorbox308"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">targetNumStages</span></span> <!--l. 6246--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Number
     of stages for steering<!--l. 6246--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">harmonic</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>non-negative integer <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">0</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>If non-zero, sets the number of stages in which the restraint centers or force constant are
     changed to their target values. If zero, the change is continuous. Each stage lasts <span class="t1-zi4r-0-x-x-109">targetNumSteps </span>MD
     steps. To sample both ends of the transformation, the simulation should be run for <span class="t1-zi4r-0-x-x-109">targetNumSteps</span>
<!--l. 6246--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-bin">×</mo></math>
     (<span class="t1-zi4r-0-x-x-109">targetNumStages </span>+ 1).
     </li></ul>
<div class="flushright">
<!--l. 6251--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 6251--><p class="noindent"></p><div class="heading"><a id="sec:colvarbias_force_k_ti">
<h5 class="subsubsectionHead"><span class="titlemark">6.5.3    </span> <a id="x1-1300006.5.3"></a>Changing force constant</h5>
<!--l. 6251--><p class="noindent"></p></a> </div> 
<!--l. 6253--><p class="indent">   The force constant of the harmonic restraint may also be changed to equilibrate <span class="cite">[<a href="#XDeng2009">38</a>]</span>.
</p>
<ul class="itemize1">
<li class="itemize"><a id="harmonic|targetForceConstant"> </a> <a id="dx1-130001"></a><span class="colorbox" id="colorbox309"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">targetForceConstant</span></span> <!--l. 6284--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Change
     the force constant towards this value<!--l. 6284--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">harmonic</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive decimal <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>When defined, the current <span class="t1-zi4r-0-x-x-109">forceConstant </span>will be moved towards this value during the
     simulation. Time evolution of the force constant is dictated by the <span class="t1-zi4r-0-x-x-109">targetForceExponent </span>parameter
     (see below). By default, the force constant is changed smoothly over a total of <span class="t1-zi4r-0-x-x-109">targetNumSteps </span>steps.
     This is useful to introduce or remove restraints in a progressive manner. If <span class="t1-zi4r-0-x-x-109">targetNumStages </span>is set
     to a nonzero value, the change is performed in discrete stages, lasting <span class="t1-zi4r-0-x-x-109">targetNumSteps </span>steps <span class="ptmri8t-x-x-109">each</span>.
     This second mode may be used to compute the conformational free energy change associated with the
     restraint, within the FEP or TI formalisms. For convenience, the code provides an estimate of the free
     energy derivative for use in TI, with the format:<br class="newline"/><span class="t1-zi4r-0-x-x-109">colvars: Lambda= ***.** dA/dLambda= ***.**</span><br class="newline"/>A more complete free energy calculation (particularly with regard to convergence analysis), while
     not handled by the Colvars module, can be performed by post-processing the colvars trajectory, if
     <span class="t1-zi4r-0-x-x-109">colvarsTrajFrequency </span>is set to a suitably small value. It should be noted, however, that restraint free
     energy calculations may be handled more efficiently by an indirect route, through the determination
     of a PMF for the restrained coordinate.<span class="cite">[<a href="#XDeng2009">38</a>]</span>
</li>
<li class="itemize"><a id="dx1-130002"></a><span class="colorbox" id="colorbox310"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">targetForceExponent</span></span> <!--l. 6307--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Exponent
     in the time-dependence of the force constant<!--l. 6307--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">harmonic</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>decimal equal to or greater than 1.0 <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">1.0</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Sets the exponent, <!--l. 6307--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>α</mi></math>,
     in the function used to vary the force constant as a function of time. The force is varied according to a
     coupling parameter <!--l. 6307--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>λ</mi></math>,
     raised to the power <!--l. 6307--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>α</mi></math>:
     <!--l. 6307--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>k</mi></mrow><mrow><mi>λ</mi></mrow></msub> <mo class="MathClass-rel">=</mo> <msub><mrow><mi>k</mi></mrow><mrow><mn>0</mn></mrow></msub> <mo class="MathClass-bin">+</mo><msup><mrow><mi>λ</mi></mrow><mrow><mi>α</mi></mrow></msup><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>k</mi></mrow><mrow><mn>1</mn></mrow></msub> <mo class="MathClass-bin">−</mo><msub><mrow><mi>k</mi></mrow><mrow><mn>0</mn></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math>,
     where <!--l. 6307--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>k</mi></mrow><mrow><mn>0</mn></mrow></msub></math>,
     <!--l. 6307--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>k</mi></mrow><mrow><mi>λ</mi></mrow></msub></math>,
     and <!--l. 6307--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>k</mi></mrow><mrow><mn>1</mn></mrow></msub></math>
     are the initial, current, and final values of the force constant. The parameter <!--l. 6307--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>λ</mi></math>
     evolves linearly from 0 to 1, either smoothly, or in <span class="t1-zi4r-0-x-x-109">targetNumStages </span>equally spaced discrete stages,
     or according to an arbitrary schedule set with <span class="t1-zi4r-0-x-x-109">lambdaSchedule</span>. When the initial value of the force
     constant is zero, an exponent greater than 1.0 distributes the effects of introducing the restraint more
     smoothly over time than a linear dependence, and ensures that there is no singularity in the derivative
     of the restraint free energy with respect to lambda. A value of 4 has been found to give good results
     in some tests.
                                                                                         
                                                                                         
     </li>
<li class="itemize"><a id="dx1-130003"></a><span class="colorbox" id="colorbox311"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">targetEquilSteps</span></span> <!--l. 6317--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Number
     of steps discarded from TI estimate<!--l. 6317--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">harmonic</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive integer <br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>Defines the number of steps within each stage that are considered equilibration and
     discarded from the restraint free energy derivative estimate reported reported in the output.
     </li>
<li class="itemize"><a id="dx1-130004"></a><span class="colorbox" id="colorbox312"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">lambdaSchedule</span></span> <!--l. 6327--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Schedule
     of lambda-points for changing force constant<!--l. 6327--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">harmonic</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>list of real numbers between 0 and 1 <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>If specified together with targetForceConstant, sets the sequence of discrete <!--l. 6327--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>λ</mi></math>
     values that will be used for different stages.
     </li></ul>
<div class="flushright">
<!--l. 6332--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 6332--><p class="noindent"></p><div class="heading"><a id="sec:colvarbias_acc_work">
<h4 class="subsectionHead"><span class="titlemark">6.6    </span> <a id="x1-1310006.6"></a>Computing the work of a changing restraint</h4>
<!--l. 6332--><p class="noindent"></p></a> </div> 
<!--l. 6334--><p class="indent">   If the restraint centers or force constant are changed continuosly (<span class="t1-zi4r-0-x-x-109">targetNumStages </span>undefined) it is possible to
record the net work performed by the changing restraint: </p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-131001"></a><span class="colorbox" id="colorbox313"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">outputAccumulatedWork</span></span> <!--l. 6346--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Write
     the accumulated work of the changing restraint to the Colvars trajectory file<!--l. 6346--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">harmonic</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">off</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>If <span class="t1-zi4r-0-x-x-109">targetCenters </span>or <span class="t1-zi4r-0-x-x-109">targetForceConstant </span>are defined and this option is enabled,
     the  accumulated  work  from  the  beginning  of  the  simulation  will  be  written  to  the  trajectory  file
     (<span class="t1-zi4r-0-x-x-109">colvarsTrajFrequency </span>must be non-zero). When the simulation is continued from a state file, the
     previously accumulated work is included in the integral. This option allows to conveniently extract
     the estimated PMF of a steered MD calculation (when <span class="t1-zi4r-0-x-x-109">targetCenters </span>is used), or of other simulation
     protocols.  </li></ul>
<div class="flushright">
<!--l. 6350--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 6350--><p class="noindent"></p><div class="heading"><a id="sec:colvarbias_harmonic_walls">
<h4 class="subsectionHead"><span class="titlemark">6.7    </span> <a id="x1-1320006.7"></a>Harmonic wall restraints</h4>
<!--l. 6350--><p class="noindent"></p></a> </div> 
<!--l. 6352--><p class="indent">   The <span class="t1-zi4r-0-x-x-109">harmonicWalls</span><span class="t1-zi4r-0-x-x-109"> {...} </span>bias is closely related to the harmonic bias (see <a href="#x1-1270006.5">6.5<!--tex4ht:ref: sec:colvarbias_harmonic --></a>), with the following two
differences: <span class="ptmri8t-x-x-109">(i) </span>instead of a center a <span class="ptmri8t-x-x-109">lower wall </span>and/or an <span class="ptmri8t-x-x-109">upper wall </span>are defined, outside of which the bias
implements a half-harmonic potential; </p><table class="equation"><tr><td> <a id="x1-132001r41"></a>
<!--l. 6353--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>V</mi> <mrow><mo class="MathClass-open">(</mo><mrow><mi>ξ</mi></mrow><mo class="MathClass-close">)</mo></mrow> <mo class="MathClass-rel">=</mo> <mfenced close="" open="{" separators=""><mrow> <mtable align="axis" class="array" columnlines="none" equalcolumns="false" equalrows="false" style=""> <mtr><mtd class="array" columnalign="left"> <mfrac><mrow><mn>1</mn></mrow> <mrow><mn>2</mn></mrow></mfrac><mi>k</mi><msup><mrow> <mfenced close=")" open="(" separators=""><mrow><mfrac><mrow><mi>ξ</mi><mo class="MathClass-bin">−</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi mathvariant="normal">upper</mi></mrow></msub></mrow>
<mrow><msub><mrow><mi>w</mi></mrow><mrow><mi>ξ</mi> </mrow></msub></mrow></mfrac> </mrow></mfenced> </mrow><mrow><mn>2</mn></mrow></msup> </mtd><mtd class="array" columnalign="left"> <mi mathvariant="normal">if</mi><mspace class="nbsp" width="1em"></mspace><mi>ξ</mi> <mo class="MathClass-rel">&gt;</mo> <msub><mrow><mi>ξ</mi></mrow><mrow><mi mathvariant="normal">upper</mi></mrow></msub> </mtd>
</mtr> <mtr><mtd class="array" columnalign="left"> <mn>0</mn> </mtd><mtd class="array" columnalign="left"> <mi mathvariant="normal">if</mi><mspace class="nbsp" width="1em"></mspace><msub><mrow><mi>ξ</mi></mrow><mrow><mi mathvariant="normal">lower</mi></mrow></msub> <mo class="MathClass-rel">≤</mo> <mi>ξ</mi> <mo class="MathClass-rel">≥</mo> <msub><mrow><mi>ξ</mi></mrow><mrow><mi mathvariant="normal">upper</mi></mrow></msub> </mtd>
</mtr> <mtr><mtd class="array" columnalign="left"> <mfrac><mrow><mn>1</mn></mrow> <mrow><mn>2</mn></mrow></mfrac><mi>k</mi><msup><mrow> <mfenced close=")" open="(" separators=""><mrow><mfrac><mrow><mi>ξ</mi><mo class="MathClass-bin">−</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi mathvariant="normal">lower</mi></mrow></msub></mrow>
<mrow><msub><mrow><mi>w</mi></mrow><mrow><mi>ξ</mi> </mrow></msub></mrow></mfrac> </mrow></mfenced> </mrow><mrow><mn>2</mn></mrow></msup> </mtd><mtd class="array" columnalign="left"> <mi mathvariant="normal">if</mi><mspace class="nbsp" width="1em"></mspace><mi>ξ</mi> <mo class="MathClass-rel">&lt;</mo> <msub><mrow><mi>ξ</mi></mrow><mrow><mi mathvariant="normal">lower</mi></mrow></msub> </mtd>
</mtr> <!--l l--></mtable> </mrow></mfenced>
</math></td><td class="eq-no">(41)</td></tr></table>
<!--l. 6363--><p class="indent">   where <!--l. 6363--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi mathvariant="normal">lower</mi></mrow></msub></math>
and <!--l. 6363--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi mathvariant="normal">upper</mi></mrow></msub></math> are
the lower and upper wall thresholds, respectively; <span class="ptmri8t-x-x-109">(ii) </span>because an interval between two walls is defined, only scalar
variables can be used (but any number of variables can be defined, and the wall bias is intrinsically
multi-dimensional).<br class="newline"/>
</p><!--l. 6365--><p class="indent"> <span class="ptmb8t-x-x-109">Note: </span>this bias replaces the keywords <span class="t1-zi4r-0-x-x-109">lowerWall</span>, <span class="t1-zi4r-0-x-x-109">lowerWallConstant</span>, <span class="t1-zi4r-0-x-x-109">upperWall </span>and <span class="t1-zi4r-0-x-x-109">upperWallConstant</span>
defined in the <span class="t1-zi4r-0-x-x-109">colvar </span>context. Those keywords are deprecated.
</p><!--l. 6368--><p class="indent">   The <span class="t1-zi4r-0-x-x-109">harmonicWalls </span>bias implements the following options: </p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-132002"></a><span class="colorbox" id="colorbox314"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">name:</span></span> see definition of <a href="#sec:colvarbias"><span class="t1-zi4r-0-x-x-109">name</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="dx1-132003"></a><span class="colorbox" id="colorbox315"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">colvars:</span></span> see definition of <a href="#sec:colvarbias"><span class="t1-zi4r-0-x-x-109">colvars</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="dx1-132004"></a><span class="colorbox" id="colorbox316"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">outputEnergy:</span></span> see definition of <a href="#sec:colvarbias"><span class="t1-zi4r-0-x-x-109">outputEnergy</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="dx1-132005"></a><span class="colorbox" id="colorbox317"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">writeTIPMF:</span></span> see definition of <a href="#sec:colvarbias"><span class="t1-zi4r-0-x-x-109">writeTIPMF</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="dx1-132006"></a><span class="colorbox" id="colorbox318"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">writeTISamples:</span></span> see definition of <a href="#sec:colvarbias"><span class="t1-zi4r-0-x-x-109">writeTISamples</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="dx1-132007"></a><span class="colorbox" id="colorbox319"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">stepZeroData:</span></span> see definition of <a href="#sec:colvarbias"><span class="t1-zi4r-0-x-x-109">stepZeroData</span></a> (biasing and analysis methods)
                                                                                         
                                                                                         
     </li>
<li class="itemize"><a id="dx1-132008"></a><span class="colorbox" id="colorbox320"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">lowerWalls</span></span> <!--l. 6383--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Position
     of the lower wall(s)<!--l. 6383--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">colvar</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>Space-separated list of decimals <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Defines the values <!--l. 6383--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi mathvariant="normal">lower</mi></mrow></msub></math>
     below which a confining restraint on the colvar is applied to each colvar <!--l. 6383--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>.
     </li>
<li class="itemize"><a id="dx1-132009"></a><span class="colorbox" id="colorbox321"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">upperWalls</span></span> <!--l. 6391--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Position
     of the upper wall(s)<!--l. 6391--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">colvar</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>Space-separated list of decimals <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Defines the values <!--l. 6391--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi mathvariant="normal">upper</mi></mrow></msub></math>
     above which a confining restraint on the colvar is applied to each colvar <!--l. 6391--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>.
     </li>
<li class="itemize"><a id="dx1-132010"></a><span class="colorbox" id="colorbox322"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">forceConstant:</span></span> see definition of <a href="#sec:colvarbias_harmonic"><span class="t1-zi4r-0-x-x-109">forceConstant</span></a> (Harmonic restraints)
     </li>
<li class="itemize"><a id="dx1-132011"></a><span class="colorbox" id="colorbox323"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">lowerWallConstant</span></span> <!--l. 6405--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Force
     constant for the lower wall<!--l. 6405--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">harmonicWalls</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive decimal <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">forceConstant</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>When both sets of walls are defined (lower and upper), this keyword allows setting
     different force constants for them. As with <span class="t1-zi4r-0-x-x-109">forceConstant</span>, the specified constant is divided internally
     by the square of the specific <span class="t1-zi4r-0-x-x-109">width </span>of each variable (see also the equivalent keyword for the harmonic
     restraint, <a href="#harmonic|forceConstant"><span class="t1-zi4r-0-x-x-109">forceConstant</span></a>). The force constant reported in the output as “<!--l. 6405--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>k</mi></math>",
     and used in the change of force constant scheme, is the geometric mean of <span class="t1-zi4r-0-x-x-109">upperWallConstant </span>and
     <span class="t1-zi4r-0-x-x-109">upperWallConstant</span>.
     </li>
<li class="itemize"><a id="dx1-132012"></a><span class="colorbox" id="colorbox324"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">upperWallConstant:</span></span> analogous to <span class="t1-zi4r-0-x-x-109">lowerWallConstant</span>
</li>
<li class="itemize"><a id="dx1-132013"></a><span class="colorbox" id="colorbox325"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">targetForceConstant:</span></span> see definition of <a href="#sec:colvarbias_harmonic"><span class="t1-zi4r-0-x-x-109">targetForceConstant</span></a> (harmonic restraints)
     </li>
<li class="itemize"><a id="dx1-132014"></a><span class="colorbox" id="colorbox326"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">targetForceConstant</span></span> <!--l. 6422--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Change
     the force constant(s) towards this value<!--l. 6422--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">harmonicWalls</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive decimal <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>This keyword allows changing either one or both of the wall force constants over time.
     In the case that <span class="t1-zi4r-0-x-x-109">lowerWallConstant </span>and <span class="t1-zi4r-0-x-x-109">upperWallConstant </span>have the same value, the behavior
     of this keyword is identical to the corresponding keyword in the <span class="t1-zi4r-0-x-x-109">harmonic </span>restraint; otherwise, the
     change schedule is applied to the geometric mean of the two constant. When only one set of walls
     is defined (<span class="t1-zi4r-0-x-x-109">lowerWall </span>or <span class="t1-zi4r-0-x-x-109">upperWalls</span>), only the respective force constant is changed. <span class="ptmb8t-x-x-109">Note: </span>if only
     one of the two force constants is meant to change over time, it is possible to use two instances of
     <span class="t1-zi4r-0-x-x-109">harmonicWalls</span>, and apply the changing schedule only to one of them.
                                                                                         
                                                                                         
     </li>
<li class="itemize"><a id="dx1-132015"></a><span class="colorbox" id="colorbox327"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">targetNumSteps:</span></span> see definition of <a href="#sec:colvarbias_harmonic"><span class="t1-zi4r-0-x-x-109">targetNumSteps</span></a> (harmonic restraints)
     </li>
<li class="itemize"><a id="dx1-132016"></a><span class="colorbox" id="colorbox328"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">targetForceExponent:</span></span> see definition of <a href="#sec:colvarbias_harmonic"><span class="t1-zi4r-0-x-x-109">targetForceExponent</span></a> (harmonic restraints)
     </li>
<li class="itemize"><a id="dx1-132017"></a><span class="colorbox" id="colorbox329"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">targetEquilSteps:</span></span> see definition of <a href="#sec:colvarbias_harmonic"><span class="t1-zi4r-0-x-x-109">targetEquilSteps</span></a> (harmonic restraints)
     </li>
<li class="itemize"><a id="dx1-132018"></a><span class="colorbox" id="colorbox330"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">targetNumStages:</span></span> see definition of <a href="#sec:colvarbias_harmonic"><span class="t1-zi4r-0-x-x-109">targetNumStages</span></a> (harmonic restraints)
     </li>
<li class="itemize"><a id="dx1-132019"></a><span class="colorbox" id="colorbox331"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">lambdaSchedule:</span></span> see definition of <a href="#sec:colvarbias_harmonic"><span class="t1-zi4r-0-x-x-109">lambdaSchedule</span></a> (harmonic restraints)
     </li>
<li class="itemize"><a id="dx1-132020"></a><span class="colorbox" id="colorbox332"><span class="ptmb8t-x-x-109">Keyword   </span><span class="t1-zi4r-0-x-x-120">outputAccumulatedWork:</span></span> see   definition   of   <a href="#sec:colvarbias_harmonic"><span class="t1-zi4r-0-x-x-109">outputAccumulatedWork</span></a> (harmonic
     restraints)
     </li>
<li class="itemize"><a id="harmonicWalls|bypassExtendedLagrangian"> </a> <a id="dx1-132021"></a><span class="colorbox" id="colorbox333"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">bypassExtendedLagrangian</span></span> <!--l. 6442--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Apply
     bias to actual colvars, bypassing extended coordinates<!--l. 6442--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">harmonicWalls</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">on</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>This option behaves as <a href="#colvarbias|bypassExtendedLagrangian"><span class="t1-zi4r-0-x-x-109">bypassExtendedLagrangian</span></a> for other biases, but <span class="ptmb8t-x-x-109">it defaults</span>
<span class="ptmb8t-x-x-109">to </span><span class="t1-zi4b-0-x-x-109">on</span><span class="ptmb8t-x-x-109">, unlike in the general case</span>. Thus, by default, the <span class="t1-zi4r-0-x-x-109">harmonicWalls </span>bias applies to the actual
     colvars, so that the distribution of the colvar between the walls is unaffected by the bias, which then
     applies a flat-bottom potential <span class="ptmri8t-x-x-109">as a function of the colvar value</span>. This bias will affect the extended
     coordinate distribution near the walls. If <span class="t1-zi4r-0-x-x-109">bypassExtendedLagrangian </span>is disabled, <span class="t1-zi4r-0-x-x-109">harmonicWalls</span>
     applies a flat-bottom potential <span class="ptmri8t-x-x-109">as a function of the extended coordinate</span>. Conversely, this bias will then
     modify the distribution of the actual colvar value near the walls.
     </li></ul>
<!--l. 6446--><p class="noindent"><span class="ptmb8t-x-x-109">Example 1: </span>harmonic walls for one variable with two different force constants.</p>
<div class="mdframed" id="mdframed-55">
<!--l. 6447--><p class="noindent"></p><pre>harmonicWalls {<br class="newline"/>  name mywalls<br class="newline"/>  colvars dist<br class="newline"/>  lowerWalls 22.0 <br class="newline"/>  upperWalls 38.0 <br class="newline"/>  lowerWallConstant 2.0 <br class="newline"/>  upperWallConstant 10.0 <br class="newline"/>} </pre>
</div>
<!--l. 6458--><p class="noindent"><span class="ptmb8t-x-x-109">Example 2: </span>harmonic walls for two variables with a single force constant.</p>
<div class="mdframed" id="mdframed-56">
<!--l. 6459--><p class="noindent"></p><pre>harmonicWalls {<br class="newline"/>  name mywalls<br class="newline"/>  colvars phi psi<br class="newline"/>  lowerWalls -180.0 0.0<br class="newline"/>  upperWalls 0.0 180.0<br class="newline"/>  forceConstant 5.0 <br class="newline"/>} </pre>
</div>
<div class="flushright">
<!--l. 6470--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 6470--><p class="noindent"></p><div class="heading"><a id="sec:colvarbias_linear">
<h4 class="subsectionHead"><span class="titlemark">6.8    </span> <a id="x1-1330006.8"></a>Linear restraints</h4>
<!--l. 6470--><p class="noindent"></p></a> </div> 
<!--l. 6472--><p class="indent">   The linear restraint biasing method is used to minimally bias a simulation. There is generally a unique strength
of bias for each CV center, which means you must know the bias force constant specifically for the center of the CV.
This force constant may be found by using experiment directed simulation described in section <a href="#x1-1340006.9">6.9<!--tex4ht:ref: sec:colvarbias_alb --></a>. Please cite
Pitera and Chodera when using <span class="cite">[<a href="#XPitera2012">39</a>]</span>.
</p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-133001"></a><span class="colorbox" id="colorbox334"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">name:</span></span> see definition of <a href="#sec:colvarbias"><span class="t1-zi4r-0-x-x-109">name</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="dx1-133002"></a><span class="colorbox" id="colorbox335"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">colvars:</span></span> see definition of <a href="#sec:colvarbias"><span class="t1-zi4r-0-x-x-109">colvars</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="dx1-133003"></a><span class="colorbox" id="colorbox336"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">outputEnergy:</span></span> see definition of <a href="#sec:colvarbias"><span class="t1-zi4r-0-x-x-109">outputEnergy</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="dx1-133004"></a><span class="colorbox" id="colorbox337"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">forceConstant</span></span> <!--l. 6499--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Scaled
     force constant (kcal/mol)<!--l. 6499--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">linear</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive decimal <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">1.0</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>This option defines a <span class="ptmri8t-x-x-109">scaled </span>force constant for the linear bias. To ensure consistency
     for multidimensional restraints, it is divided internally by the specific <span class="t1-zi4r-0-x-x-109">width </span>of each variable (which
     is 1 by default), so that all variables are effectively dimensionless and of commensurate size. See also
     the equivalent keyword for the harmonic restraint, <a href="#harmonic|forceConstant"><span class="t1-zi4r-0-x-x-109">forceConstant</span></a>. <span class="ptmri8t-x-x-109">The values of the actual force</span>
<span class="ptmri8t-x-x-109">constants </span><!--l. 6499--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>k</mi><mo class="MathClass-bin">∕</mo><msub><mrow><mi>w</mi></mrow><mrow><mi>ξ</mi></mrow></msub></math>
<span class="ptmri8t-x-x-109">are always printed when the restraint is defined.</span>
</li>
<li class="itemize"><a id="dx1-133005"></a><span class="colorbox" id="colorbox338"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">centers</span></span> <!--l. 6508--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Initial
     linear restraint centers<!--l. 6508--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">linear</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>space-separated list of colvar values <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>These are analogous to the <a href="#harmonic|centers"><span class="t1-zi4r-0-x-x-109">centers</span></a> keyword of the harmonic restraint. Although they
     do not affect dynamics, they are here necessary to ensure a well-defined energy for the linear bias.
     </li>
<li class="itemize"><a id="dx1-133006"></a><span class="colorbox" id="colorbox339"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">writeTIPMF:</span></span> see definition of <a href="#sec:colvarbias"><span class="t1-zi4r-0-x-x-109">writeTIPMF</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="dx1-133007"></a><span class="colorbox" id="colorbox340"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">writeTISamples:</span></span> see definition of <a href="#sec:colvarbias"><span class="t1-zi4r-0-x-x-109">writeTISamples</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="dx1-133008"></a><span class="colorbox" id="colorbox341"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">targetForceConstant:</span></span> see definition of <a href="#sec:colvarbias_harmonic"><span class="t1-zi4r-0-x-x-109">targetForceConstant</span></a> (Harmonic restraints)
     </li>
<li class="itemize"><a id="dx1-133009"></a><span class="colorbox" id="colorbox342"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">targetNumSteps:</span></span> see definition of <a href="#sec:colvarbias_harmonic"><span class="t1-zi4r-0-x-x-109">targetNumSteps</span></a> (Harmonic restraints)
     </li>
<li class="itemize"><a id="dx1-133010"></a><span class="colorbox" id="colorbox343"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">targetForceExponent:</span></span> see definition of <a href="#sec:colvarbias_harmonic"><span class="t1-zi4r-0-x-x-109">targetForceExponent</span></a> (Harmonic restraints)
     </li>
<li class="itemize"><a id="dx1-133011"></a><span class="colorbox" id="colorbox344"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">targetEquilSteps:</span></span> see definition of <a href="#sec:colvarbias_harmonic"><span class="t1-zi4r-0-x-x-109">targetEquilSteps</span></a> (Harmonic restraints)
     </li>
<li class="itemize"><a id="dx1-133012"></a><span class="colorbox" id="colorbox345"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">targetNumStages:</span></span> see definition of <a href="#sec:colvarbias_harmonic"><span class="t1-zi4r-0-x-x-109">targetNumStages</span></a> (Harmonic restraints)
     </li>
<li class="itemize"><a id="dx1-133013"></a><span class="colorbox" id="colorbox346"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">lambdaSchedule:</span></span> see definition of <a href="#sec:colvarbias_harmonic"><span class="t1-zi4r-0-x-x-109">lambdaSchedule</span></a> (Harmonic restraints)
     </li>
<li class="itemize"><a id="dx1-133014"></a><span class="colorbox" id="colorbox347"><span class="ptmb8t-x-x-109">Keyword   </span><span class="t1-zi4r-0-x-x-120">outputAccumulatedWork:</span></span> see   definition   of   <a href="#sec:colvarbias_harmonic"><span class="t1-zi4r-0-x-x-109">outputAccumulatedWork</span></a> (Harmonic
     restraints)
     </li></ul>
<div class="flushright">
<!--l. 6523--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 6523--><p class="noindent"></p><div class="heading"><a id="sec:colvarbias_alb">
<h4 class="subsectionHead"><span class="titlemark">6.9    </span> <a id="x1-1340006.9"></a>Adaptive Linear Bias/Experiment Directed Simulation</h4>
<!--l. 6523--><p class="noindent"></p></a> </div> 
<!--l. 6525--><p class="indent">   Experiment directed simulation applies a linear bias with a changing force constant. Please cite White and Voth
<span class="cite">[<a href="#XWhite2014">40</a>]</span> when using this feature. As opposed to that reference, the force constant here is scaled by the <span class="t1-zi4r-0-x-x-109">width</span>
corresponding to the biased colvar. In White and Voth, each force constant is scaled by the colvars set center. The
bias converges to a linear bias, after which it will be the minimal possible bias. You may also stop the simulation,
take the median of the force constants (ForceConst) found in the colvars trajectory file, and then apply a linear bias
                                                                                         
                                                                                         
with that constant. All the notes about units described in sections <a href="#x1-1330006.8">6.8<!--tex4ht:ref: sec:colvarbias_linear --></a> and <a href="#x1-1270006.5">6.5<!--tex4ht:ref: sec:colvarbias_harmonic --></a> apply here as well. <span class="ptmb8t-x-x-109">This is not a valid</span>
<span class="ptmb8t-x-x-109">simulation of any particular statistical ensemble and is only an optimization algorithm until the bias has</span>
<span class="ptmb8t-x-x-109">converged</span>.
</p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-134001"></a><span class="colorbox" id="colorbox348"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">name:</span></span> see definition of <a href="#sec:colvarbias"><span class="t1-zi4r-0-x-x-109">name</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="dx1-134002"></a><span class="colorbox" id="colorbox349"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">colvars:</span></span> see definition of <a href="#sec:colvarbias"><span class="t1-zi4r-0-x-x-109">colvars</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="dx1-134003"></a><span class="colorbox" id="colorbox350"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">centers</span></span> <!--l. 6557--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Collective
     variable centers<!--l. 6557--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">alb</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>space-separated list of colvar values <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>The desired center (equilibrium values) which will be sought during the adaptive linear
     biasing. The number of values must be the number of requested colvars. Each value is a decimal
     number if the corresponding colvar returns a scalar, a “<span class="t1-zi4r-0-x-x-109">(x, y, z)</span>" triplet if it returns a unit vector
     or a vector, and a “<span class="t1-zi4r-0-x-x-109">q0, q1, q2, q3)</span>" quadruplet if it returns a rotational quaternion. If a colvar has
     periodicities or symmetries, its closest image to the restraint center is considered when calculating the
     linear potential.
     </li>
<li class="itemize"><a id="dx1-134004"></a><span class="colorbox" id="colorbox351"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">updateFrequency</span></span> <!--l. 6575--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>The
     duration of updates<!--l. 6575--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">alb</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>An integer <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>This is, <!--l. 6575--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>N</mi></math>,
     the number of simulation steps to use for each update to the bias. This determines how long the system
     requires to equilibrate after a change in force constant (<!--l. 6575--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>N</mi><mo class="MathClass-bin">∕</mo><mn>2</mn></math>),
     how long statistics are collected for an iteration (<!--l. 6575--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>N</mi><mo class="MathClass-bin">∕</mo><mn>2</mn></math>),
     and how quickly energy is added to the system (at most, <!--l. 6575--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>A</mi><mo class="MathClass-bin">∕</mo><mn>2</mn><mi>N</mi></math>,
     where <!--l. 6575--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>A</mi></math>
     is the <span class="t1-zi4r-0-x-x-109">forceRange</span>). Until the force constant has converged, the method as described is an optimization
     procedure and not an integration of a particular statistical ensemble. It is important that each step
     should be uncorrelated from the last so that iterations are independent. Therefore, <!--l. 6575--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>N</mi></math>
     should be at least twice the autocorrelation time of the collective variable. The system should also be
     able to dissipate energy as fast as <!--l. 6575--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>N</mi><mo class="MathClass-bin">∕</mo><mn>2</mn></math>,
     which can be done by adjusting thermostat parameters. Practically, <!--l. 6575--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>N</mi></math>
     has been tested successfully at significantly shorter than the autocorrelation time of the collective
     variables being biased and still converge correctly.
     </li>
<li class="itemize"><a id="dx1-134005"></a><span class="colorbox" id="colorbox352"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">forceRange</span></span> <!--l. 6590--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>The
     expected range of the force constant in units of energy<!--l. 6590--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">alb</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>A space-separated list of decimal numbers <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span>3 <!--l. 6590--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>k</mi></mrow><mrow><mi>b</mi></mrow></msub><mi>T</mi> </math>
<br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>This is largest magnitude of the force constant which one expects. If this parameter is
     too low, the simulation will not converge. If it is too high the simulation will waste time exploring
     values that are too large. A value of 3 <!--l. 6590--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>k</mi></mrow><mrow><mi>b</mi></mrow></msub><mi>T</mi> </math>
     has worked well in the systems presented as a first choice. This parameter is dynamically adjusted
     over the course of a simulation. The benefit is that a bad guess for the forceRange can be corrected.
     However, this can lead to large amounts of energy being added over time to the system. To prevent
     this dynamic update, add <span class="t1-zi4r-0-x-x-109">hardForceRange yes </span>as a parameter
     </li>
<li class="itemize"><a id="dx1-134006"></a><span class="colorbox" id="colorbox353"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">rateMax</span></span> <!--l. 6600--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>The
     maximum rate of change of force constant<!--l. 6600--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">alb</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>A list of space-separated real numbers <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>This optional parameter controls how much energy is added to the system from this bias.
     Tuning this separately from the <span class="t1-zi4r-0-x-x-109">updateFrequency </span>and <span class="t1-zi4r-0-x-x-109">forceRange </span>can allow for large bias changes
     but with a low <span class="t1-zi4r-0-x-x-109">rateMax </span>prevents large energy changes that can lead to instability in the simulation.
     </li></ul>
<div class="flushright">
<!--l. 6605--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 6605--><p class="noindent"></p><div class="heading"><a id="sec:colvarbias_histogram">
<h4 class="subsectionHead"><span class="titlemark">6.10    </span> <a id="x1-1350006.10"></a>Multidimensional histograms</h4>
<!--l. 6605--><p class="noindent"></p></a> </div> 
<!--l. 6607--><p class="indent">   The <span class="t1-zi4r-0-x-x-109">histogram </span>feature is used to record the distribution of a set of collective variables in the form of a
N-dimensional histogram. A <span class="t1-zi4r-0-x-x-109">histogram </span>block may define the following parameters:
</p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-135001"></a><span class="colorbox" id="colorbox354"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">name:</span></span> see definition of <a href="#sec:colvarbias"><span class="t1-zi4r-0-x-x-109">name</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="dx1-135002"></a><span class="colorbox" id="colorbox355"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">colvars:</span></span> see definition of <a href="#sec:colvarbias"><span class="t1-zi4r-0-x-x-109">colvars</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="dx1-135003"></a><span class="colorbox" id="colorbox356"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">outputFreq:</span></span> see definition of <a href="#sec:colvarbias"><span class="t1-zi4r-0-x-x-109">outputFreq</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="dx1-135004"></a><span class="colorbox" id="colorbox357"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">stepZeroData:</span></span> see definition of <a href="#sec:colvarbias"><span class="t1-zi4r-0-x-x-109">stepZeroData</span></a> (biasing and analysis methods)
                                                                                         
                                                                                         
     </li>
<li class="itemize"><a id="dx1-135005"></a><span class="colorbox" id="colorbox358"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">outputFile</span></span> <!--l. 6628--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Write
     the histogram to a file<!--l. 6628--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">histogram</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>UNIX filename <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="ptmri8t-x-x-109">outputName</span><span class="t1-zi4r-0-x-x-109">.</span><!--l. 6628--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="t1-zi4r-0-x-x-109">name</span><!--l. 6628--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><span class="t1-zi4r-0-x-x-109">.dat</span>
<br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>Name  of  the  file  containing  histogram  data  (multicolumn  format),  which  is  written
     every <a href="#colvarbias|outputFreq"><span class="t1-zi4r-0-x-x-109">outputFreq</span></a> steps. For the special case of 2 variables, Gnuplot may be used to visualize this
     file. If <span class="t1-zi4r-0-x-x-109">outputFile </span>is set to <span class="t1-zi4r-0-x-x-109">none</span>, the file is not written.
     </li>
<li class="itemize"><a id="dx1-135006"></a><span class="colorbox" id="colorbox359"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">outputFileDX</span></span> <!--l. 6641--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Write
     the histogram to a file<!--l. 6641--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">histogram</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>UNIX filename <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="ptmri8t-x-x-109">outputName</span><span class="t1-zi4r-0-x-x-109">.</span><!--l. 6641--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="t1-zi4r-0-x-x-109">name</span><!--l. 6641--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><span class="t1-zi4r-0-x-x-109">.dx</span>
<br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Name of the file containing histogram data (OpenDX format), which is written every
     <a href="#colvarbias|outputFreq"><span class="t1-zi4r-0-x-x-109">outputFreq</span></a> steps. For the special case of 3 variables, VMD may be used to visualize this file. This
     file is written by default if the dimension is 3 or more. If <span class="t1-zi4r-0-x-x-109">outputFileDX </span>is set to <span class="t1-zi4r-0-x-x-109">none</span>, the file is not
     written.
     </li>
<li class="itemize"><a id="dx1-135007"></a><span class="colorbox" id="colorbox360"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">gatherVectorColvars</span></span> <!--l. 6655--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Treat
     vector variables as multiple observations of a scalar variable?<!--l. 6655--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">histogram</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>UNIX filename <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">off</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>When this is set to <span class="t1-zi4r-0-x-x-109">on</span>, the components of a multi-dimensional colvar (e.g. one based
     on  <span class="t1-zi4r-0-x-x-109">cartesian</span>,  <span class="t1-zi4r-0-x-x-109">distancePairs</span>,  or  a  vector  of  scalar  numbers  given  by  <span class="t1-zi4r-0-x-x-109">scriptedFunction</span>)  are
     treated as multiple observations of a scalar variable. This results in the histogram being accumulated
     multiple times for each simulation step). When multiple vector variables are included in <span class="t1-zi4r-0-x-x-109">histogram</span>,
     these must have the same length because their components are accumulated together. For example, if
     <!--l. 6655--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>,
     <!--l. 6655--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>λ</mi></math>
     and <!--l. 6655--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>τ</mi></math>
     are three variables of dimensions 5, 5 and 1, respectively, for each iteration 5 triplets <!--l. 6655--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>λ</mi></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-punc">,</mo><mi>τ</mi></mrow><mo class="MathClass-close">)</mo></mrow></math>
     (<!--l. 6655--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi> <mo class="MathClass-rel">=</mo> <mn>1</mn><mo class="MathClass-punc">,</mo><mo class="MathClass-op">…</mo><mn>5</mn></math>)
     are accumulated into a 3-dimensional histogram.
     </li>
<li class="itemize"><a id="dx1-135008"></a><span class="colorbox" id="colorbox361"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">weights</span></span> <!--l. 6666--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Relative
     contributions of each vector component to the histogram<!--l. 6666--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">histogram</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>list of space-separated decimals <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span>all weights equal to 1 <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>When <span class="t1-zi4r-0-x-x-109">gatherVectorColvars </span>is <span class="t1-zi4r-0-x-x-109">on</span>, the components of each multi-dimensional colvar
     are accumulated with a different weight. For example, if <!--l. 6666--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>x</mi></math>
                                                                                         
                                                                                         
     and <!--l. 6666--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>y</mi></math>
     are  two  distinct  <span class="t1-zi4r-0-x-x-109">cartesian </span>variables  defined  on  the  same  group  of  atoms,  the  corresponding  2D
     histogram can be weighted on a per-atom basis in the definition of <span class="t1-zi4r-0-x-x-109">histogram</span>.  </li></ul>
<!--l. 6670--><p class="indent">   As with any other biasing and analysis method, when a histogram is applied to an extended-system colvar
(<a href="#x1-960004.21">4.21<!--tex4ht:ref: sec:colvar_extended --></a>), it accesses the value of the extended coordinate rather than that of the actual colvar. This can be overridden
by enabling the <a href="#colvarbias|bypassExtendedLagrangian"><span class="t1-zi4r-0-x-x-109">bypassExtendedLagrangian</span></a> option. A <span class="ptmri8t-x-x-109">joint histogram </span>of the actual colvar and the extended
coordinate may be collected by specifying the colvar name twice in a row in the <span class="t1-zi4r-0-x-x-109">colvars </span>parameter (e.g. <span class="t1-zi4r-0-x-x-109">colvars</span>
<span class="t1-zi4r-0-x-x-109">myColvar myColvar</span>): the first instance will be understood as the actual colvar, and the second, as the extended
coordinate.
</p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-135009"></a><span class="colorbox" id="colorbox362"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">bypassExtendedLagrangian:</span></span> see definition of <a href="#colvarbias|bypassExtendedLagrangian"><span class="t1-zi4r-0-x-x-109">bypassExtendedLagrangian</span></a> (biasing
     and analysis methods)</li></ul>
<div class="flushright">
<!--l. 6683--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 6683--><p class="noindent"></p><div class="heading"><a id="sec:colvarbias_histogram_grid">
<h5 class="subsubsectionHead"><span class="titlemark">6.10.1    </span> <a id="x1-1360006.10.1"></a>Grid definition for multidimensional histograms</h5>
<!--l. 6683--><p class="noindent"></p></a> </div> 
<!--l. 6685--><p class="indent">   Like the ABF and metadynamics biases, <span class="t1-zi4r-0-x-x-109">histogram </span>uses the parameters <span class="t1-zi4r-0-x-x-109">lowerBoundary</span>, <span class="t1-zi4r-0-x-x-109">upperBoundary</span>, and
<span class="t1-zi4r-0-x-x-109">width </span>to define its grid. These values can be overridden if a configuration block <span class="t1-zi4r-0-x-x-109">histogramGrid { </span><span class="t1-zi4r-0-x-x-109">…} </span>is
provided inside the configuration of <span class="t1-zi4r-0-x-x-109">histogram</span>. The options supported inside this configuration block are:
</p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-136001"></a><span class="colorbox" id="colorbox363"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">lowerBoundaries</span></span> <!--l. 6697--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Lower
     boundaries of the grid<!--l. 6697--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">histogramGrid</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>list of space-separated decimals <br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>This option defines the lower boundaries of the grid, overriding any values defined
     by the <span class="t1-zi4r-0-x-x-109">lowerBoundary </span>keyword of each colvar. Note that when <span class="t1-zi4r-0-x-x-109">gatherVectorColvars </span>is <span class="t1-zi4r-0-x-x-109">on</span>, each
     vector variable is automatically treated as a scalar, and a single value should be provided for it.
     </li>
<li class="itemize"><a id="dx1-136002"></a><span class="colorbox" id="colorbox364"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">upperBoundaries:</span></span> analogous to <span class="t1-zi4r-0-x-x-109">lowerBoundaries</span>
</li>
<li class="itemize"><a id="dx1-136003"></a><span class="colorbox" id="colorbox365"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">widths:</span></span> analogous to <span class="t1-zi4r-0-x-x-109">lowerBoundaries</span></li></ul>
<div class="flushright">
<!--l. 6759--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 6759--><p class="noindent"></p><div class="heading"><a id="sec:reweighted_fel_from_aMD">
<h4 class="subsectionHead"><span class="titlemark">6.11    </span> <a id="x1-1370006.11"></a>Reweighted free-energy landscapes from accelerated MD</h4>
<!--l. 6759--><p class="noindent"></p></a> </div> 
<!--l. 6759--><p class="indent">   The <span class="t1-zi4r-0-x-x-109">reweightaMD </span>can be used to build reweighted free-energy landscapes of a set of collective variables on-the-fly
from accelerated molecular dynamics (aMD)<span class="cite">[<a href="#Xhamelberg2004">41</a>]</span> and Gaussian aMD (GaMD)<span class="cite">[<a href="#XMiao2015">42</a>]</span>. More specifically, it collects the boost
potential, <!--l. 6759--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>Δ</mi><mi>V</mi> <mrow><mo class="MathClass-open">(</mo><mrow><mi mathvariant="bold">X</mi></mrow><mo class="MathClass-close">)</mo></mrow></math>,
and computes the corresponding free-energies: </p><table class="equation"><tr><td> <a id="x1-137001r42"></a>
<!--l. 6759--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>A</mi><mrow><mo class="MathClass-open">(</mo><mrow><mi>ξ</mi></mrow><mo class="MathClass-close">)</mo></mrow> <mo class="MathClass-rel">=</mo> <mo class="MathClass-bin">−</mo><mfrac><mrow><mn>1</mn></mrow>
<mrow><mi>β</mi></mrow></mfrac><mo class="qopname">ln</mo><!--nolimits--><mrow><mo class="MathClass-open">⟨</mo><mrow><mo class="qopname">exp</mo><!--nolimits--><mrow><mo class="MathClass-open">{</mo><mrow><mi>β</mi><mi>Δ</mi><mi>V</mi> <mrow><mo class="MathClass-open">(</mo><mrow><mi mathvariant="bold">X</mi></mrow><mo class="MathClass-close">)</mo></mrow><mi>δ</mi><mrow><mo class="MathClass-open">[</mo><mrow><mi>ξ</mi><mrow><mo class="MathClass-open">(</mo><mrow><mi mathvariant="bold">X</mi></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-bin">−</mo><mi>ξ</mi></mrow><mo class="MathClass-close">]</mo></mrow></mrow><mo class="MathClass-close">}</mo></mrow></mrow><mo class="MathClass-close">⟩</mo></mrow><mo class="MathClass-punc">,</mo>
</math></td><td class="eq-no">(42)</td></tr></table>
<!--l. 6759--><p class="indent">   where <!--l. 6759--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>β</mi></math> is
<!--l. 6759--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>1</mn><mo class="MathClass-bin">∕</mo><msub><mrow><mi>k</mi></mrow><mrow><mi>B</mi></mrow></msub><mi>T</mi> </math>. If the
distribution of <!--l. 6759--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>Δ</mi><mi>V</mi> <mrow><mo class="MathClass-open">(</mo><mrow><mi mathvariant="bold">X</mi></mrow><mo class="MathClass-close">)</mo></mrow><mi>δ</mi><mrow><mo class="MathClass-open">[</mo><mrow><mi>ξ</mi><mrow><mo class="MathClass-open">(</mo><mrow><mi mathvariant="bold">X</mi></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-bin">−</mo><mi>ξ</mi></mrow><mo class="MathClass-close">]</mo></mrow></math>
is assumed to be near Gaussian, the equation above can be approximated by the cumulant expansion: </p><table class="equation"><tr><td>
<a id="x1-137002r43"></a>
<!--l. 6759--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>A</mi><mrow><mo class="MathClass-open">(</mo><mrow><mi>ξ</mi></mrow><mo class="MathClass-close">)</mo></mrow> <mo class="MathClass-rel">=</mo> <mo class="MathClass-bin">−</mo><msub><mrow><mi>C</mi></mrow><mrow><mn>1</mn></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>ξ</mi></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-bin">−</mo><mfrac><mrow><mn>1</mn></mrow>
<mrow><mn>2</mn></mrow></mfrac><mi>β</mi><msub><mrow><mi>C</mi></mrow><mrow><mn>2</mn></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>ξ</mi></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-punc">,</mo>
</math></td><td class="eq-no">(43)</td></tr></table>
<!--l. 6759--><p class="indent">   where <!--l. 6759--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>C</mi></mrow><mrow><mn>1</mn></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>ξ</mi></mrow><mo class="MathClass-close">)</mo></mrow></math> and
<!--l. 6759--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>C</mi></mrow><mrow><mn>2</mn></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>ξ</mi></mrow><mo class="MathClass-close">)</mo></mrow></math> are the average
and the variance of <!--l. 6759--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>Δ</mi><mi>V</mi> <mrow><mo class="MathClass-open">(</mo><mrow><mi mathvariant="bold">X</mi></mrow><mo class="MathClass-close">)</mo></mrow><mi>δ</mi><mrow><mo class="MathClass-open">[</mo><mrow><mi>ξ</mi><mrow><mo class="MathClass-open">(</mo><mrow><mi mathvariant="bold">X</mi></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-bin">−</mo><mi>ξ</mi></mrow><mo class="MathClass-close">]</mo></mrow></math>,
respectively. It should be noted that this implementation is initially intended for the combination of aMD/GaMD
with other CV-based methods such as eABF<span class="cite">[<a href="#XChen2021">43</a>]</span>, where the term associated with the biasing distribution
<!--l. 6759--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mover accent="true"><mrow><mi>ρ</mi></mrow><mo class="MathClass-op">̃</mo></mover><mrow><mo class="MathClass-open">(</mo><mrow><mi>ξ</mi></mrow><mo class="MathClass-close">)</mo></mrow></math>, namely
<!--l. 6759--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-bin">−</mo><mfrac><mrow><mn>1</mn></mrow> <mrow><mi>β</mi></mrow></mfrac><mo class="qopname">ln</mo><!--nolimits--><mover accent="true"><mrow><mi>ρ</mi></mrow><mo class="qopname">̃</mo></mover><mrow><mo class="MathClass-open">(</mo><mrow><mi>ξ</mi></mrow><mo class="MathClass-close">)</mo></mrow></math>,
                                                                                         
                                                                                         
is included, so users are expected to add the output from Eq. (<a href="#x1-137001r42">42<!--tex4ht:ref: eq:amd-reweighting --></a>) or (<a href="#x1-137002r43">43<!--tex4ht:ref: eq:cumulant-expansion --></a>) with those
from CZAR (see Eq. (<a href="#x1-116001r32">32<!--tex4ht:ref: eq:czar --></a>)) <span class="cite">[<a href="#XLesage2017">25</a>]</span> directly. Consequently, in Eq. (<a href="#x1-137001r42">42<!--tex4ht:ref: eq:amd-reweighting --></a>) and (<a href="#x1-137002r43">43<!--tex4ht:ref: eq:cumulant-expansion --></a>) the term
<!--l. 6759--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-bin">−</mo><mfrac><mrow><mn>1</mn></mrow> <mrow><mi>β</mi></mrow></mfrac><mo class="qopname">ln</mo><!--nolimits--><mover accent="true"><mrow><mi>ρ</mi></mrow><mo class="qopname">̃</mo></mover><mrow><mo class="MathClass-open">(</mo><mrow><mi>ξ</mi></mrow><mo class="MathClass-close">)</mo></mrow></math> is not included.
Therefore, if users want to perform just aMD/GaMD simulations with this on-the-fly reweighting implementation,
the term <!--l. 6759--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-bin">−</mo><mfrac><mrow><mn>1</mn></mrow> <mrow><mi>β</mi></mrow></mfrac><mo class="qopname">ln</mo><!--nolimits--><mover accent="true"><mrow><mi>ρ</mi></mrow><mo class="qopname">̃</mo></mover><mrow><mo class="MathClass-open">(</mo><mrow><mi>ξ</mi></mrow><mo class="MathClass-close">)</mo></mrow></math>
should be manually computed by post-processing.
</p><!--l. 6759--><p class="indent">   Currently, in all MD engines that Colvars supports, only NAMD implements aMD/GaMD, so this component is
basically exclusive to NAMD.
</p><!--l. 6759--><p class="indent">   The list of options is as follows: </p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-137003"></a><span class="colorbox" id="colorbox366"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">name:</span></span> see definition of <a href="#sec:colvarbias"><span class="t1-zi4r-0-x-x-109">name</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="dx1-137004"></a><span class="colorbox" id="colorbox367"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">colvars:</span></span> see definition of <a href="#sec:colvarbias"><span class="t1-zi4r-0-x-x-109">colvars</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="dx1-137005"></a><span class="colorbox" id="colorbox368"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">outputFreq:</span></span> see definition of <a href="#sec:colvarbias"><span class="t1-zi4r-0-x-x-109">outputFreq</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="dx1-137006"></a><span class="colorbox" id="colorbox369"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">outputFile:</span></span> see definition of <a href="#sec:colvarbias"><span class="t1-zi4r-0-x-x-109">outputFile</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="dx1-137007"></a><span class="colorbox" id="colorbox370"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">historyFreq:</span></span> see definition of <a href="#sec:colvarbias_abf"><span class="t1-zi4r-0-x-x-109">historyFreq</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="dx1-137008"></a><span class="colorbox" id="colorbox371"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">CollectAfterSteps</span></span> <!--l. 6759--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Collect
     the histogram after a specified number of simulation steps<!--l. 6759--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">reweightaMD</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive integer <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span>0 <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Start to populate the histogram after a specified number of step. This feature is useful
     to skip the preparation steps in GaMD simulations.
     </li>
<li class="itemize"><a id="dx1-137009"></a><span class="colorbox" id="colorbox372"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">CumulantExpansion</span></span> <!--l. 6759--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Enable
     the cumulant expansion estimator<!--l. 6759--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">reweightaMD</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">yes</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>If this option is set to <span class="t1-zi4r-0-x-x-109">yes</span>, the cumulant expansion in Eq. (<span class="ptmb8t-x-x-109">??</span>) will be used. The output
     file have a suffix of <span class="t1-zi4r-0-x-x-109">.cumulant.*</span>.
     </li>
<li class="itemize"><a id="dx1-137010"></a><span class="colorbox" id="colorbox373"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">writePMFGradients</span></span> <!--l. 6759--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Calculate
     the numerical gradients of Eq. (<span class="ptmb8t-x-x-109">??</span>) and (<span class="ptmb8t-x-x-109">??</span>)<!--l. 6759--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">reweightaMD</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">yes</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>If this option is set to <span class="t1-zi4r-0-x-x-109">yes</span>, the numerical gradients Eq. (<span class="ptmb8t-x-x-109">??</span>) and (<span class="ptmb8t-x-x-109">??</span>) with respect to
     <!--l. 6759--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>
                                                                                         
                                                                                         
     are calculated. This option is generally used when <!--l. 6759--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>
     is stratified into windows, and the corresponding free-energies are required to merge afterwards.
     </li></ul>
<div class="flushright">
<!--l. 6762--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 6762--><p class="noindent"></p><div class="heading"><a id="sec:colvarbias_restraint_histogram">
<h4 class="subsectionHead"><span class="titlemark">6.12    </span> <a id="x1-1380006.12"></a>Probability distribution-restraints</h4>
<!--l. 6762--><p class="noindent"></p></a> </div> 
<!--l. 6764--><p class="indent">   The <span class="t1-zi4r-0-x-x-109">histogramRestraint </span>bias implements a continuous potential of many variables (or of a single high-dimensional
variable) aimed at reproducing a one-dimensional statistical distribution that is provided by the user. The
<!--l. 6765--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>M</mi></math> variables
<!--l. 6765--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-punc">,</mo><mo class="MathClass-op">…</mo><mo class="MathClass-punc">,</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>M</mi></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math> are interpreted as multiple
observations of a random variable <!--l. 6765--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>
with unknown probability distribution. The potential is minimized when the histogram
<!--l. 6766--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>h</mi><mrow><mo class="MathClass-open">(</mo><mrow><mi>ξ</mi></mrow><mo class="MathClass-close">)</mo></mrow></math>, estimated as a sum of Gaussian
functions centered at <!--l. 6766--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-punc">,</mo><mo class="MathClass-op">…</mo><mo class="MathClass-punc">,</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>M</mi></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></math>, is equal
to the reference histogram <!--l. 6766--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>h</mi></mrow><mrow><mn>0</mn></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>ξ</mi></mrow><mo class="MathClass-close">)</mo></mrow></math>:
</p><table class="equation"><tr><td> <a id="x1-138001r44"></a>
<!--l. 6767--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>V</mi> <mrow><mo class="MathClass-open">(</mo><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-punc">,</mo><mo class="MathClass-op">…</mo><mo class="MathClass-punc">,</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>M</mi></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow> <mo class="MathClass-rel">=</mo> <mfrac><mrow><mn>1</mn></mrow>
<mrow><mn>2</mn></mrow></mfrac><mi>k</mi><mo class="MathClass-op">∫
 <!--nolimits--></mo><!--nolimits--><msup><mrow><mfenced close=")" open="(" separators=""><mrow><mi>h</mi><mrow><mo class="MathClass-open">(</mo><mrow><mi>ξ</mi></mrow><mo class="MathClass-close">)</mo></mrow><mo class="MathClass-bin">−</mo><msub><mrow><mi>h</mi></mrow><mrow><mn>0</mn></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>ξ</mi></mrow><mo class="MathClass-close">)</mo></mrow></mrow></mfenced></mrow><mrow><mn>2</mn></mrow></msup><mi mathvariant="normal">d</mi><mi>ξ</mi>
</math></td><td class="eq-no">(44)</td></tr></table>
<table class="equation"><tr><td> <a id="x1-138002r45"></a>
<!--l. 6771--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>h</mi><mrow><mo class="MathClass-open">(</mo><mrow><mi>ξ</mi></mrow><mo class="MathClass-close">)</mo></mrow> <mo class="MathClass-rel">=</mo> <mfrac><mrow><mn>1</mn></mrow>
<mrow><mi>M</mi><msqrt><mrow><mn>2</mn><mi>π</mi><msup><mrow><mi>σ</mi></mrow><mrow><mn>2</mn></mrow></msup></mrow></msqrt></mrow></mfrac><munderover accent="false" accentunder="false"><mrow><mo mathsize="big">∑</mo></mrow><mrow><mi>i</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><mi>M</mi></mrow></munderover><mo class="qopname">exp</mo><!--nolimits--> <mfenced close=")" open="(" separators=""><mrow><mo class="MathClass-bin">−</mo><mfrac><mrow><msup><mrow><mrow><mo class="MathClass-open">(</mo><mrow><mi>ξ</mi> <mo class="MathClass-bin">−</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow><mo class="MathClass-close">)</mo></mrow></mrow><mrow><mn>2</mn></mrow></msup></mrow>
<mrow><mn>2</mn><msup><mrow><mi>σ</mi></mrow><mrow><mn>2</mn></mrow></msup></mrow></mfrac> </mrow></mfenced>
</math></td><td class="eq-no">(45)</td></tr></table>
<!--l. 6775--><p class="indent">   When used in combination with a <span class="t1-zi4r-0-x-x-109">distancePairs </span>multi-dimensional variable, this bias implements the
refinement algorithm against ESR/DEER experiments published by Shen <span class="ptmri8t-x-x-109">et al </span><span class="cite">[<a href="#XShen2015">44</a>]</span>.
</p><!--l. 6777--><p class="indent">   This bias behaves similarly to the <span class="t1-zi4r-0-x-x-109">histogram </span>bias with the <span class="t1-zi4r-0-x-x-109">gatherVectorColvars </span>option, with the important
difference that <span class="ptmri8t-x-x-109">all </span>variables are gathered, resulting in a one-dimensional histogram. Future versions will include
support for multi-dimensional histograms.
</p><!--l. 6780--><p class="indent">   The list of options is as follows: </p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-138003"></a><span class="colorbox" id="colorbox374"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">name:</span></span> see definition of <a href="#sec:colvarbias"><span class="t1-zi4r-0-x-x-109">name</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="dx1-138004"></a><span class="colorbox" id="colorbox375"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">colvars:</span></span> see definition of <a href="#sec:colvarbias"><span class="t1-zi4r-0-x-x-109">colvars</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="dx1-138005"></a><span class="colorbox" id="colorbox376"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">outputEnergy:</span></span> see definition of <a href="#sec:colvarbias"><span class="t1-zi4r-0-x-x-109">outputEnergy</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="dx1-138006"></a><span class="colorbox" id="colorbox377"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">lowerBoundary</span></span> <!--l. 6795--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Lower
     boundary of the colvar grid<!--l. 6795--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">histogramRestraint</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>decimal <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Defines the lowest end of the interval where the reference distribution <!--l. 6795--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>h</mi></mrow><mrow><mn>0</mn></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>ξ</mi></mrow><mo class="MathClass-close">)</mo></mrow></math>
     is defined. Exactly one value must be provided, because only one-dimensional histograms are supported
     by the current version.
     </li>
<li class="itemize"><a id="dx1-138007"></a><span class="colorbox" id="colorbox378"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">upperBoundary:</span></span> analogous to <span class="t1-zi4r-0-x-x-109">lowerBoundary</span>
</li>
<li class="itemize"><a id="dx1-138008"></a><span class="colorbox" id="colorbox379"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">width</span></span> <!--l. 6807--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Width
     of the colvar grid<!--l. 6807--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">histogramRestraint</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive decimal <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Defines the spacing of the grid where the reference distribution <!--l. 6807--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>h</mi></mrow><mrow><mn>0</mn></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>ξ</mi></mrow><mo class="MathClass-close">)</mo></mrow></math>
     is defined.
     </li>
<li class="itemize"><a id="dx1-138009"></a><span class="colorbox" id="colorbox380"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">gaussianSigma</span></span> <!--l. 6817--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Standard
     deviation of the approximating Gaussian<!--l. 6817--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">histogramRestraint</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive decimal <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span>2 <!--l. 6817--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-bin">×</mo></math>
<span class="t1-zi4r-0-x-x-109">width</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Defines the parameter <!--l. 6817--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>σ</mi></math>
     in eq. <a href="#x1-138002r45">45<!--tex4ht:ref: eq:colvarbias_restraint_histogram_gaussian --></a>.
     </li>
<li class="itemize"><a id="dx1-138010"></a><span class="colorbox" id="colorbox381"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">forceConstant</span></span> <!--l. 6827--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Force
     constant (kcal/mol)<!--l. 6827--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">histogramRestraint</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>positive decimal <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">1.0</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Defines the parameter <!--l. 6827--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>k</mi></math>
     in eq. <a href="#x1-138001r44">44<!--tex4ht:ref: eq:colvarbias_restraint_histogram --></a>.
     </li>
<li class="itemize"><a id="dx1-138011"></a><span class="colorbox" id="colorbox382"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refHistogram</span></span> <!--l. 6838--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Reference
     histogram <!--l. 6838--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>h</mi></mrow><mrow><mn>0</mn></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>ξ</mi></mrow><mo class="MathClass-close">)</mo></mrow></math><!--l. 6838--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">histogramRestraint</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>space-separated list of <!--l. 6838--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>M</mi></math>
     positive decimals <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Provides the values of <!--l. 6838--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>h</mi></mrow><mrow><mn>0</mn></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>ξ</mi></mrow><mo class="MathClass-close">)</mo></mrow></math>
     consecutively. The mid-point convention is used, i.e. the first point that should be included is for
     <!--l. 6838--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>
     = <span class="t1-zi4r-0-x-x-109">lowerBoundary</span>+<span class="t1-zi4r-0-x-x-109">width</span>/2. If the integral of <!--l. 6838--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>h</mi></mrow><mrow><mn>0</mn></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>ξ</mi></mrow><mo class="MathClass-close">)</mo></mrow></math>
     is not normalized to 1, <!--l. 6838--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>h</mi></mrow><mrow><mn>0</mn></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>ξ</mi></mrow><mo class="MathClass-close">)</mo></mrow></math>
     is rescaled automatically before use.
     </li>
<li class="itemize"><a id="dx1-138012"></a><span class="colorbox" id="colorbox383"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">refHistogramFile</span></span> <!--l. 6849--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Reference
     histogram <!--l. 6849--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>h</mi></mrow><mrow><mn>0</mn></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>ξ</mi></mrow><mo class="MathClass-close">)</mo></mrow></math><!--l. 6849--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">histogramRestraint</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>UNIX file name <br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>Provides the values of <!--l. 6849--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>h</mi></mrow><mrow><mn>0</mn></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>ξ</mi></mrow><mo class="MathClass-close">)</mo></mrow></math>
     as contents of the corresponding file (mutually exclusive with <span class="t1-zi4r-0-x-x-109">refHistogram</span>). The format is that of a
     text file, with each line containing the space-separated values of <!--l. 6849--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>
     and <!--l. 6849--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>h</mi></mrow><mrow><mn>0</mn></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>ξ</mi></mrow><mo class="MathClass-close">)</mo></mrow></math>.
     The same numerical conventions as <span class="t1-zi4r-0-x-x-109">refHistogram </span>are used.
     </li>
<li class="itemize"><a id="dx1-138013"></a><span class="colorbox" id="colorbox384"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">writeHistogram</span></span> <!--l. 6858--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Periodically
     write the instantaneous histogram <!--l. 6858--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>h</mi><mrow><mo class="MathClass-open">(</mo><mrow><mi>ξ</mi></mrow><mo class="MathClass-close">)</mo></mrow></math><!--l. 6858--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span><span class="t1-zi4r-0-x-x-109">metadynamics</span><br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">off</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>If <span class="t1-zi4r-0-x-x-109">on</span>, the histogram <!--l. 6858--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>h</mi><mrow><mo class="MathClass-open">(</mo><mrow><mi>ξ</mi></mrow><mo class="MathClass-close">)</mo></mrow></math>
     is written every <span class="t1-zi4r-0-x-x-109">colvarsRestartFrequency </span>steps to a file with the name <span class="ptmri8t-x-x-109">outputName</span><span class="t1-zi4r-0-x-x-109">.</span><!--l. 6858--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="t1-zi4r-0-x-x-109">name</span><!--l. 6858--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><span class="t1-zi4r-0-x-x-109">.hist.dat</span>
     This is useful to diagnose the convergence of <!--l. 6859--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>h</mi><mrow><mo class="MathClass-open">(</mo><mrow><mi>ξ</mi></mrow><mo class="MathClass-close">)</mo></mrow></math>
     against <!--l. 6859--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>h</mi></mrow><mrow><mn>0</mn></mrow></msub><mrow><mo class="MathClass-open">(</mo><mrow><mi>ξ</mi></mrow><mo class="MathClass-close">)</mo></mrow></math>.
     </li></ul>
<div class="flushright">
<!--l. 6921--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 6921--><p class="noindent"></p><div class="heading"><a id="sec:scripted_biases">
<h4 class="subsectionHead"><span class="titlemark">6.13    </span> <a id="x1-1390006.13"></a>Defining scripted biases</h4>
<!--l. 6921--><p class="noindent"></p></a> </div> 
<!--l. 6921--><p class="indent">   Rather than using the biasing methods described above, it is possible to apply biases provided at run time as a
Tcl script, in the spirit of <span class="t1-zi4r-0-x-x-109">TclForces</span>.
</p>
<ul class="itemize1">
<li class="itemize"><a id="Colvars-global|scriptedColvarForces"> </a> <a id="dx1-139001"></a><span class="colorbox" id="colorbox385"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">scriptedColvarForces</span></span> <!--l. 6921--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Enable
     custom, scripted forces on colvars <!--l. 6921--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>global <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">off</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:   </span>If this flag is enabled, a Tcl procedure named <span class="t1-zi4r-0-x-x-109">calc_colvar_forces </span>accepting one
     parameter should be defined by the user. It is executed at each timestep, with the current step number
     as parameter, between the calculation of colvars and the application of bias forces. This procedure
     may use the <span class="t1-zi4r-0-x-x-109">cv </span>command to access the values of colvars (e.g. <span class="t1-zi4r-0-x-x-109">cv colvar xi value</span>), apply forces
     on them (<span class="t1-zi4r-0-x-x-109">cv colvar xi addforce $F</span>) or add energy to the simulation system (<span class="t1-zi4r-0-x-x-109">cv addenergy $E</span>),
     effectively defining custom collective variable biases. </li></ul>
<div class="flushright">
<!--l. 6921--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 6921--><p class="noindent"></p><div class="heading"><a id="sec:cv_scripting_opt">
<h4 class="subsectionHead"><span class="titlemark">6.14    </span> <a id="x1-1400006.14"></a>Performance of scripted biases</h4>
<!--l. 6921--><p class="noindent"></p></a> </div> 
<!--l. 6921--><p class="indent">   If concurrent computation over multiple threads is available (this is indicated by the message “SMP parallelism
is available." printed at initialization time), it is useful to take advantage of the scripting interface to combine many
components, all computed in parallel, into a single variable.
</p><!--l. 6921--><p class="indent">   The default SMP schedule is the following:
     </p><ol class="enumerate1">
<li class="enumerate" id="x1-140002x1">distribute the computation of all components across available threads;
     </li>
<li class="enumerate" id="x1-140004x2">on a single thread, collect the results of multi-component variables using polynomial combinations
     (see <a href="#x1-910004.16">4.16<!--tex4ht:ref: sec:cvc_superp --></a>), or custom functions (see <a href="#x1-920004.17">4.17<!--tex4ht:ref: sec:colvar_custom_function --></a>), or scripted functions (see <a href="#x1-930004.18">4.18<!--tex4ht:ref: sec:colvar_scripted --></a>);
     </li>
<li class="enumerate" id="x1-140006x3">distribute the computation of all biases across available threads;
     </li>
<li class="enumerate" id="x1-140008x4">compute     on     a     single     thread     any     scripted     biases     implemented     via     the     keyword
     <a href="#Colvars-global|scriptedColvarForces"><span class="t1-zi4r-0-x-x-109">scriptedColvarForces</span></a>.
                                                                                         
                                                                                         
     </li>
<li class="enumerate" id="x1-140010x5">communicate on a single thread forces to NAMD.</li></ol>
<!--l. 6921--><p class="indent">   The following options allow to fine-tune this schedule: </p>
<ul class="itemize1">
<li class="itemize"><a id="dx1-140011"></a><span class="colorbox" id="colorbox386"><span class="ptmb8t-x-x-109">Keyword </span><span class="t1-zi4r-0-x-x-120">scriptingAfterBiases</span></span> <!--l. 6921--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mrow><mo class="MathClass-open">⟨</mo><mrow><mspace class="thinspace" width="0.3em"></mspace></mrow></mrow></math>Scripted
     colvar forces need updated biases?<!--l. 6921--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.3em"></mspace><mo class="MathClass-close">⟩</mo></math><br class="newline"/><span class="ptmb8t-x-x-109">Context:  </span>global <br class="newline"/><span class="ptmb8t-x-x-109">Acceptable values:  </span>boolean <br class="newline"/><span class="ptmb8t-x-x-109">Default value:  </span><span class="t1-zi4r-0-x-x-109">on</span><br class="newline"/><span class="ptmb8t-x-x-109">Description:  </span>This flag specifies that the <span class="t1-zi4r-0-x-x-109">calc_colvar_forces </span>procedure (last step in the list above)
     is executed only after all biases have been updated (next-to-last step) For example, this allows using
     the energy of a restraint bias, or the force applied on a colvar, to calculate additional scripted forces,
     such as boundary constraints. When this flag is set to <span class="t1-zi4r-0-x-x-109">off</span>, it is assumed that only the values of the
     variables (but not the energy of the biases or applied forces) will be used by <span class="t1-zi4r-0-x-x-109">calc_colvar_forces</span>:
     this can be used to schedule the calculation of scripted forces and biases concurrently to increase
     performance. </li></ul>
<!--l. 6931--><p class="indent">
</p>
<div class="flushright">
<!--l. 6931--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 6931--><p class="noindent"></p><div class="heading"><a id="sec:cvscript_tcl">
<h3 class="sectionHead"><span class="titlemark">7    </span> <a id="x1-1410007"></a>Scripting interface (Tcl): list of commands</h3>
<!--l. 6931--><p class="noindent"></p></a> </div>  This section lists all the commands used in NAMD to control the behavior of the Colvars module from within a
run script.

                                                                                         <div class="flushright">
<!--l. 1--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1--><p class="noindent"></p><div class="heading"><a id="sec:cvscript_tcl_cv">
<h4 class="subsectionHead"><span class="titlemark">7.1    </span> <a id="x1-1420007.1"></a>Commands to manage the Colvars module</h4>
<!--l. 1--><p class="noindent"></p></a> </div> 
<ul class="itemize1">
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv addenergy <e></e></span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Add an energy to the MD engine (no effect in VMD)</span><br class="newline"/><span class="t1-zi4b-0-x-x-109">Parameters</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">E : float - Amount of energy to add</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv config <conf></conf></span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Read configuration from the given string</span><br class="newline"/><span class="t1-zi4b-0-x-x-109">Parameters</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">conf : string - Configuration string</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv configfile <conf_file></conf_file></span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Read configuration from a file</span><br class="newline"/><span class="t1-zi4b-0-x-x-109">Parameters</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">conf_file : string - Path to configuration file</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv delete</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Delete this Colvars module instance (VMD only)</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv featurereport</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Return a summary of Colvars features used so far and their citations</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">report : string - Feature report and citations</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv frame [frame]</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Get or set current frame number (VMD only)</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">frame : integer - Frame number</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv getatomappliedforces</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Get the list of forces applied by Colvars to atoms</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">forces : array of arrays of floats - Atomic forces</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv getatomappliedforcesmax</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Get the maximum norm of forces applied by Colvars to atoms</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">force : float - Maximum atomic force</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv getatomappliedforcesmaxid</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Get the atom ID with the largest applied force</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">id : int - ID of the atom with the maximum atomic force</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv getatomappliedforcesrms</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Get the root-mean-square norm of forces applied by Colvars to atoms</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">force : float - RMS atomic force</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv getatomids</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Get the list of indices of atoms used in Colvars</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">indices : array of ints - Atomic indices</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv getatomcharges</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Get the list of charges of atoms used in Colvars</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">charges : array of floats - Atomic charges</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv getatommasses</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Get the list of masses of atoms used in Colvars</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">masses : array of floats - Atomic masses</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv getatompositions</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Get the list of cached positions of atoms used in Colvars</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">positions : array of arrays of floats - Atomic positions</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv getatomtotalforces</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Get the list of cached total forces of atoms used in Colvars</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">forces : array of arrays of floats - Atomic total foces</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv getconfig</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Get the module's configuration string read so far</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">conf : string - Current configuration string</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv getenergy</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Get the current Colvars energy</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">E : float - Amount of energy (internal units)</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv help [command]</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Get the help string of the Colvars scripting interface</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">help : string - Help string</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv languageversion</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Get the C++ language version number</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">version : string - C++ language version</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv list [param]</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Return a list of all variables or biases</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">list : sequence of strings - List of elements</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv listcommands</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Get the list of script functions, prefixed with "cv_", "colvar_" or "bias_"</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">list : sequence of strings - List of commands</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv listindexfiles</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Get a list of the index files loaded in this session</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv load <prefix></prefix></span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Load data from a state file into all matching colvars and biases</span><br class="newline"/><span class="t1-zi4b-0-x-x-109">Parameters</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">prefix : string - Path to existing state file or input prefix</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv loadfromstring <buffer></buffer></span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Load state data from a string into all matching colvars and biases</span><br class="newline"/><span class="t1-zi4b-0-x-x-109">Parameters</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">buffer : string - String buffer containing the state information</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv molid [molid]</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Get or set the molecule ID on which Colvars is defined (VMD only)</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">molid : integer - Current molecule ID</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv printframe</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Return the values that would be written to colvars.traj</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">values : string - The values</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv printframelabels</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Return the labels that would be written to colvars.traj</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Labels : string - The labels</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv reset</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Delete all internal configuration</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv resetindexgroups</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Clear the index groups loaded so far, allowing to replace them</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv save <prefix></prefix></span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Change the prefix of all output files and save them</span><br class="newline"/><span class="t1-zi4b-0-x-x-109">Parameters</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">prefix : string - Output prefix with trailing ".colvars.state" gets removed)</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv savetostring</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Write the Colvars state to a string and return it</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">state : string - The saved state</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv units [units]</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Get or set the current Colvars unit system</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">units : string - The current unit system</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv update</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Recalculate colvars and biases</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv version</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Get the Colvars Module version number</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">version : integer - Colvars version</span></li></ul>
<div class="flushright">
<!--l. 268--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 268--><p class="noindent"></p><div class="heading"><a id="sec:cvscript_tcl_colvar">
<h4 class="subsectionHead"><span class="titlemark">7.2    </span> <a id="x1-1430007.2"></a>Commands to manage individual collective variables</h4>
<!--l. 268--><p class="noindent"></p></a> </div> 
<ul class="itemize1">
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv colvar name addforce <force></force></span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Apply the given force onto this colvar and return the same</span><br class="newline"/><span class="t1-zi4b-0-x-x-109">Parameters</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">force : float or array - Applied force; must match colvar dimensionality</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">force : float or array - Applied force; matches colvar dimensionality</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv colvar name cvcflags <flags></flags></span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Enable or disable individual components by setting their active flags</span><br class="newline"/><span class="t1-zi4b-0-x-x-109">Parameters</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">flags : integer array - Zero/nonzero value disables/enables the CVC</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv colvar name delete</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Delete this colvar, along with all biases that depend on it</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv colvar name get <feature></feature></span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Get the value of the given feature for this colvar</span><br class="newline"/><span class="t1-zi4b-0-x-x-109">Parameters</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">feature : string - Name of the feature</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">state : 1/0 - State of the given feature</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv colvar name getappliedforce</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Return the total of the forces applied to this colvar</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">force : float - Applied force; matches the colvar dimensionality</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv colvar name getatomgroups</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Return the atom indices used by this colvar as a list of lists</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">groups : array of arrays of ints - Atom indices</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv colvar name getatomids</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Return the list of atom indices used by this colvar</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">indices : array of ints - Atom indices</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv colvar name getconfig</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Return the configuration string of this colvar</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">conf : string - Current configuration string</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv colvar name getgradients</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Return the atomic gradients of this colvar</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">gradients : array of arrays of floats - Atomic gradients</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv colvar name gettotalforce</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Return the sum of internal and external forces to this colvar</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">force : float - Total force; matches the colvar dimensionality</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv colvar name getvolmapids</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Return the list of volumetric map indices used by this colvar</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv colvar name help [command]</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Get a help summary or the help string of one colvar subcommand</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">help : string - Help string</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv colvar name modifycvcs <confs></confs></span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Modify configuration of individual components by passing string arguments</span><br class="newline"/><span class="t1-zi4b-0-x-x-109">Parameters</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">confs : sequence of strings - New configurations; empty strings are skipped</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv colvar name run_ave</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Get the current running average of the value of this colvar</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">value : float or array - Averaged value; matches the colvar dimensionality</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv colvar name set <feature> <value></value></feature></span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Set the given feature of this colvar to a new value</span><br class="newline"/><span class="t1-zi4b-0-x-x-109">Parameters</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">feature : string - Name of the feature</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">value : string - String representation of the new feature value</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv colvar name state</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Print a string representation of the feature state of this colvar</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">state : string - The feature state</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv colvar name type</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Get the type description of this colvar</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">type : string - Type description</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv colvar name update</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Recompute this colvar and return its up-to-date value</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">value : float or array - Current value; matches the colvar dimensionality</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv colvar name value</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Get the current value of this colvar</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">value : float or array - Current value; matches the colvar dimensionality</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv colvar name width</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Get the width of this colvar</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">width : float - Value of the width</span></li></ul>
<div class="flushright">
<!--l. 443--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 443--><p class="noindent"></p><div class="heading"><a id="sec:cvscript_tcl_bias">
<h4 class="subsectionHead"><span class="titlemark">7.3    </span> <a id="x1-1440007.3"></a>Commands to manage individual biases</h4>
<!--l. 443--><p class="noindent"></p></a> </div> 
<ul class="itemize1">
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv bias name bin</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Get the current grid bin index (1D ABF only for now)</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">bin : integer - Bin index</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv bias name bincount [index]</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Get the number of samples at the given grid bin (1D ABF only for now)</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">samples : integer - Number of samples</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv bias name binnum</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Get the total number of grid points of this bias (1D ABF only for now)</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Bins : integer - Number of grid points</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv bias name delete</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Delete this bias</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv bias name energy</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Get the current energy of this bias</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">E : float - Energy value</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv bias name get <feature></feature></span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Get the value of the given feature for this bias</span><br class="newline"/><span class="t1-zi4b-0-x-x-109">Parameters</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">feature : string - Name of the feature</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">state : 1/0 - State of the given feature</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv bias name getconfig</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Return the configuration string of this bias</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">conf : string - Current configuration string</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv bias name help [command]</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Get a help summary or the help string of one bias subcommand</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">help : string - Help string</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv bias name load <prefix></prefix></span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Load data into this bias</span><br class="newline"/><span class="t1-zi4b-0-x-x-109">Parameters</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">prefix : string - Read from a file with this name or prefix</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv bias name loadfromstring <buffer></buffer></span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Load state data into this bias from a string</span><br class="newline"/><span class="t1-zi4b-0-x-x-109">Parameters</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">buffer : string - String buffer containing the state information</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv bias name save <prefix></prefix></span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Save data from this bias into a file with the given prefix</span><br class="newline"/><span class="t1-zi4b-0-x-x-109">Parameters</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">prefix : string - Prefix for the state file of this bias</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv bias name savetostring</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Save data from this bias into a string and return it</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">state : string - The bias state</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv bias name set <feature> <value></value></feature></span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Set the given feature of this bias to a new value</span><br class="newline"/><span class="t1-zi4b-0-x-x-109">Parameters</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">feature : string - Name of the feature</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">value : string - String representation of the new feature value</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv bias name share</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Share bias information with other replicas (multiple-walker scheme)</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv bias name state</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Print a string representation of the feature state of this bias</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">state : string - String representation of the bias features</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv bias name type</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Print the type of this bias object</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">type : string - Type of this bias object (e.g. metadynamics)</span>
</li>
<li class="itemize"><span class="t1-zi4r-0-x-x-109">cv bias name update</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Recompute this bias and return its up-to-date energy</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">Returns</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">-</span><br class="newline"/><span class="t1-zi4r-0-x-x-109">E : float - Energy value</span></li></ul>
<!--l. 6934--><p class="indent">
</p>
<div class="flushright">
<!--l. 6934--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 6934--><p class="noindent"></p><div class="heading"><a id="sec:colvars_config_changes">
<h3 class="sectionHead"><span class="titlemark">8    </span> <a id="x1-1450008"></a>Syntax changes from older versions</h3>
<!--l. 6934--><p class="noindent"></p></a> </div> 
<!--l. 6936--><p class="indent">   The following is a list of syntax changes in Colvars since its first release. Many of the older keywords are still
recognized by the current code, thanks to specific compatibility code. <span class="ptmri8t-x-x-109">This is not a list of new features: </span>its primary
purpose is to make you aware of those improvements that affect the use of old configuration files with new versions
of the code.
</p><!--l. 6940--><p class="noindent"><span class="ptmb8t-x-x-109">Note: </span>if you are using any of the NAMD and VMD tutorials:<br class="newline"/><a class="url" href="https://www.ks.uiuc.edu/Training/Tutorials/"><span class="t1-zi4r-0-x-x-109">https://www.ks.uiuc.edu/Training/Tutorials/</span></a><br class="newline"/>please be aware that <span class="ptmri8t-x-x-109">several of these tutorials are not actively maintained</span>: for those cases, this list will help you
reconcile any inconsistencies.
</p>
<ul class="itemize1">
<li class="itemize"><span class="ptmb8t-x-x-109">Colvars version 2016-06-09 or later (NAMD version 2.12b1 or later).</span><br class="newline"/>The legacy keyword <span class="t1-zi4r-0-x-x-109">refPositionsGroup </span>has been renamed <a href="#atom-group|fittingGroup"><span class="t1-zi4r-0-x-x-109">fittingGroup</span></a> for clarity (the legacy
     version is still supported).
     </li>
<li class="itemize"><span class="ptmb8t-x-x-109">Colvars version 2016-08-10 or later (NAMD version 2.12b1 or later).</span><br class="newline"/>“System forces" have been replaced by “total forces" (see for example <a href="#colvar|outputTotalForce"><span class="t1-zi4r-0-x-x-109">outputTotalForce</span></a>). See the
     following page for more information:<br class="newline"/><a class="url" href="https://colvars.github.io/README-totalforce.html"><span class="t1-zi4r-0-x-x-109">https://colvars.github.io/README-totalforce.html</span></a>
</li>
<li class="itemize"><span class="ptmb8t-x-x-109">Colvars version 2017-01-09 or later (NAMD version 2.13b1 or later).</span><br class="newline"/>A new type of restraint, <span class="t1-zi4r-0-x-x-109">harmonicWalls </span>(see <a href="#x1-1320006.7">6.7<!--tex4ht:ref: sec:colvarbias_harmonic_walls --></a>), replaces and improves upon the legacy keywords
     <span class="t1-zi4r-0-x-x-109">lowerWall </span>and <span class="t1-zi4r-0-x-x-109">upperWall</span>: these are still supported as short-hands.
     </li>
<li class="itemize"><span class="ptmb8t-x-x-109">Colvars version 2018-11-15 or later (NAMD version 2.14b1 or later).</span><br class="newline"/>The global <span class="t1-zi4r-0-x-x-109">analysis </span>keyword has been discontinued: specific analysis tasks are controlled directly
     by the keywords <a href="#colvar|corrFunc"><span class="t1-zi4r-0-x-x-109">corrFunc</span></a> and <a href="#colvar|runAve"><span class="t1-zi4r-0-x-x-109">runAve</span></a>, which continue to remain <span class="t1-zi4r-0-x-x-109">off </span>by default.
     </li>
<li class="itemize"><span class="ptmb8t-x-x-109">Colvars version 2020-02-25 or later (NAMD version 2.14b1 or later).</span><br class="newline"/>The parameter <a href="#metadynamics|hillWidth"><span class="t1-zi4r-0-x-x-109">hillWidth</span></a>, expressing the Gaussian width <!--l. 6961--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>2</mn><mi>σ</mi></math>
     in relative units (number of grid points), does not have a default value any more. A new alternative
     parameter <a href="#metadynamics|gaussianSigmas"><span class="t1-zi4r-0-x-x-109">gaussianSigmas</span></a> allows setting the <!--l. 6962--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>σ</mi></math>
                                                                                         
                                                                                         
     parameters explicitly for each variable if needed.<br class="newline"/>Furthermore, to facilitate the use of other analysis tools such as for example <span class="t1-zi4r-0-x-x-109">sum_hills</span>:<br class="newline"/><a class="url" href="https://www.plumed.org/doc-v2.6/user-doc/html/sum\_hills.html"><span class="t1-zi4r-0-x-x-109">https://www.plumed.org/doc-v2.6/user-doc/html/sum\_hills.html</span></a><br class="newline"/>the format of the file written by <a href="#metadynamics|writeHillsTrajectory"><span class="t1-zi4r-0-x-x-109">writeHillsTrajectory</span></a> has also been changed to use <!--l. 6965--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>σ</mi></math>
     instead of <!--l. 6965--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>2</mn><mi>σ</mi></math>.
     This change does not affect how the biasing potential is written in the state file, or the simulated
     trajectory.
     </li>
<li class="itemize"><span class="ptmb8t-x-x-109">Colvars version 2020-02-25 or later (NAMD version 2.14b1 or later).</span><br class="newline"/>The legacy keywords <span class="t1-zi4r-0-x-x-109">lowerWall </span>and <span class="t1-zi4r-0-x-x-109">upperWall </span>of a <span class="t1-zi4r-0-x-x-109">colvar </span>definition block do not have default
     values any longer, and need to be set explicitly, preferably as part of the <span class="t1-zi4r-0-x-x-109">harmonicWalls </span>restraint.
     When  using  an  ABF  bias,  it  is  recommended  to  set  the  two  walls  equal  to  <a href="#colvar|lowerBoundary"><span class="t1-zi4r-0-x-x-109">lowerBoundary</span></a> and
     <a href="#colvar|upperBoundary"><span class="t1-zi4r-0-x-x-109">upperBoundary</span></a>, respectively. When using a metadynamics bias, it is recommended to set the two
     walls strictly <span class="ptmri8t-x-x-109">within </span><a href="#colvar|lowerBoundary"><span class="t1-zi4r-0-x-x-109">lowerBoundary</span></a> and <a href="#colvar|upperBoundary"><span class="t1-zi4r-0-x-x-109">upperBoundary</span></a>; see <a href="#x1-1200006.4.1">6.4.1<!--tex4ht:ref: sec:colvarbias_meta_boundaries --></a> for details.
     </li>
<li class="itemize"><span class="ptmb8t-x-x-109">Colvars version 2020-11-09 or later.</span><br class="newline"/>The legacy keyword <span class="t1-zi4r-0-x-x-109">disableForces </span>for atom groups is now deprecated and will be discontinued in
     a future release. Atom groups now have an automated way to save computation if forces are not used,
     and enabling this option otherwise would lead to incorrect behavior.
     </li></ul>
<!--l. 6979--><p class="noindent">Up-to-date documentation can always be accessed at:<br class="newline"/><a class="url" href="https://colvars.github.io/colvars-refman-namd/colvars-refman-namd.html"><span class="t1-zi4r-0-x-x-109">https://colvars.github.io/colvars-refman-namd/colvars-refman-namd.html</span></a>
</p><!--l. 7003--><p class="indent">
</p>
<div class="flushright">
<!--l. 7003--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 7003--><p class="noindent"></p><div class="heading"><a id="sec:compilation_notes">
<h3 class="sectionHead"><span class="titlemark">9    </span> <a id="x1-1460009"></a>Compilation notes</h3>
<!--l. 7003--><p class="noindent"></p></a> </div> 
<!--l. 7003--><p class="indent">   The Colvars module is typically built using the recipes of each supported software package: for this reason, no
installation instructions are needed, and the vast majority of the features described in this manual are supported in
the most common builds of each package.
</p><!--l. 7003--><p class="indent">   This section lists the few cases where the choice of compilation settings affects the availability features in the
Colvars module.
</p>
<ul class="itemize1">
<li class="itemize">Scripting  commands  using  the  Tcl  language  (<a class="url" href="https://www.tcl.tk"><span class="t1-zi4r-0-x-x-109">https://www.tcl.tk</span></a>)  are  supported  in  VMD  and
     NAMD. All precompiled builds of each code include Tcl, and it is highly recommended to enable Tcl
     support in any custom build, using precompiled Tcl libraries from the UIUC website.
     </li>
<li class="itemize">The      Lepton      library      (<a class="url" href="https://simtk.org/projects/lepton"><span class="t1-zi4r-0-x-x-109">https://simtk.org/projects/lepton</span></a>)      used      to      implement
     the <span class="t1-zi4r-0-x-x-109">customFunction </span>feature is currently included only in NAMD (always on) and in LAMMPS (on
     by default). For VMD, a <a href="https://github.com/giacomofiorin/vmd-patches">patch</a> that allows to link Lepton is available. There is no mechanism to link
     Lepton to GROMACS at this moment.
     </li>
<li class="itemize">Some  features  require  compilation  using  the  C++11  language  standard,  which  is  either  supported
     (VMD) or required (GROMACS, LAMMPS, NAMD) by the engines. However, for historic reasons
     many of the VMD official builds do not conform to the C++11 standard yet, which results in restricted
     functionality. Detailed and up-to-date information can be found at:<br class="newline"/><a class="url" href="https://colvars.github.io/README-c++11.html"><span class="t1-zi4r-0-x-x-109">https://colvars.github.io/README-c++11.html</span></a></li></ul>
<!--l. 300--><p class="indent">
</p><!--l. 303--><p class="indent">
</p>
<h3 class="likesectionHead"><a id="x1-1470009"></a>References</h3>
<!--l. 1--><p class="noindent">
</p><div class="thebibliography">
<p class="bibitem"><span class="biblabel">
  [1]<span class="bibsp">   </span></span><a id="XFiorin2013"></a>Giacomo  Fiorin,  Michael L.  Klein,  and  Jérôme  Hénin.   Using  collective  variables  to  drive
     molecular dynamics simulations. <span class="ptmri8t-x-x-109">Mol. Phys.</span>, 111(22-23):3345--3362, 2013.
     </p>
<p class="bibitem"><span class="biblabel">
  [2]<span class="bibsp">   </span></span><a id="XPhillips2020"></a>James C. Phillips, David J. Hardy, Julio D. C. Maia, John E. Stone, João V. Ribeiro, Rafael C.
     Bernardi, Ronak Buch, Giacomo Fiorin, Jérôme Hénin, Wei Jiang, Ryan McGreevy, Marcelo C. R.
     Melo,  Brian K.  Radak,  Robert D.  Skeel,  Abhishek  Singharoy,  Yi Wang,  Benoît  Roux,  Aleksei
     Aksimentiev, Zaida Luthey-Schulten, Laxmikant V. Kalé, Klaus Schulten, Christophe Chipot, and
     Emad Tajkhorshid.   Scalable molecular dynamics on CPU and GPU architectures with NAMD.   <span class="ptmri8t-x-x-109">J.</span>
<span class="ptmri8t-x-x-109">Chem. Phys.</span>, 153(4):044130, 2020.
     </p>
<p class="bibitem"><span class="biblabel">
  [3]<span class="bibsp">   </span></span><a id="XIannuzzi2003"></a>M. Iannuzzi,  A. Laio,  and  M. Parrinello.    Efficient  exploration  of  reactive  potential  energy
     surfaces using car-parrinello molecular dynamics. <span class="ptmri8t-x-x-109">Phys. Rev. Lett.</span>, 90(23):238302, 2003.
     </p>
<p class="bibitem"><span class="biblabel">
  [4]<span class="bibsp">   </span></span><a id="XCoutsias2004"></a>E A Coutsias, C Seok, and K A Dill.  Using quaternions to calculate RMSD.  <span class="ptmri8t-x-x-109">J. Comput. Chem.</span>,
     25(15):1849--1857, 2004.
     </p>
<p class="bibitem"><span class="biblabel">
  [5]<span class="bibsp">   </span></span><a id="XFu2017"></a>Haohao Fu, Wensheng Cai, Jérôme Hénin, Benoît Roux, and Christophe Chipot. New coarse
     variables for the accurate determination of standard binding free energies.  <span class="ptmri8t-x-x-109">J. Chem. Theory. Comput.</span>,
     13(11):5173--5178, 2017.
     </p>
<p class="bibitem"><span class="biblabel">
  [6]<span class="bibsp">   </span></span><a id="XMu2005"></a>Yuguang  Mu,  Phuong H.  Nguyen,  and  Gerhard  Stock.    Energy  landscape  of  a  small  peptide
     revealed by dihedral angle principal component analysis. <span class="ptmri8t-x-x-109">Proteins</span>, 58(1):45--52, 2005.
     </p>
<p class="bibitem"><span class="biblabel">
  [7]<span class="bibsp">   </span></span><a id="XAltis2007"></a>Alexandros Altis, Phuong H. Nguyen, Rainer Hegger, and Gerhard Stock. Dihedral angle principal
     component analysis of molecular dynamics simulations. <span class="ptmri8t-x-x-109">J. Chem. Phys.</span>, 126(24):244111, 2007.
     </p>
<p class="bibitem"><span class="biblabel">
  [8]<span class="bibsp">   </span></span><a id="XGlykos2006"></a>Nicholas M  Glykos.    Carma:  a  molecular  dynamics  analysis  program.    <span class="ptmri8t-x-x-109">J.  Comput.  Chem.</span>,
     27(14):1765--1768, 2006.
                                                                                         
                                                                                         
     </p>
<p class="bibitem"><span class="biblabel">
  [9]<span class="bibsp">   </span></span><a id="XLeines2012"></a>G. D. Leines and B. Ensing. Path finding on high-dimensional free energy landscapes. <span class="ptmri8t-x-x-109">Phys. Rev.</span>
<span class="ptmri8t-x-x-109">Lett.</span>, 109:020601, 2012.
     </p>
<p class="bibitem"><span class="biblabel">
 [10]<span class="bibsp">   </span></span><a id="XBranduardi2007"></a>Davide Branduardi, Francesco Luigi Gervasio, and Michele Parrinello. From a to b in free energy
     space. <span class="ptmri8t-x-x-109">J Chem Phys</span>, 126(5):054103, 2007.
     </p>
<p class="bibitem"><span class="biblabel">
 [11]<span class="bibsp">   </span></span><a id="XHovan2019"></a>F. Comitani  L. Hovan  and  F. L.  Gervasio.   Defining  an  optimal  metric  for  the  path  collective
     variables. <span class="ptmri8t-x-x-109">J. Chem. Theory Comput.</span>, 15:25--32, 2019.
     </p>
<p class="bibitem"><span class="biblabel">
 [12]<span class="bibsp">   </span></span><a id="XChen2022"></a>Haochuan  Chen,  Han  Liu,  Heying  Feng,  Haohao  Fu,  Wensheng  Cai,  Xueguang  Shao,  and
     Christophe   Chipot.      Mlcv:   Bridging   Machine-Learning-Based   Dimensionality   Reduction   and
     Free-Energy Calculation. <span class="ptmri8t-x-x-109">J. Chem. Inf. Model.</span>, 62(1):1--8, 2022.
     </p>
<p class="bibitem"><span class="biblabel">
 [13]<span class="bibsp">   </span></span><a id="XFiorin2020"></a>Giacomo  Fiorin,  Fabrizio  Marinelli,  and  José D.  Faraldo-Gómez.   Direct  derivation  of  free
     energies  of  membrane  deformation  and  other  solvent  density  variations  from  enhanced  sampling
     molecular dynamics. <span class="ptmri8t-x-x-109">J. Comp. Chem.</span>, 41(5):449--459, 2020.
     </p>
<p class="bibitem"><span class="biblabel">
 [14]<span class="bibsp">   </span></span><a id="XFerrarotti2015"></a>Marco Jacopo  Ferrarotti,  Sandro  Bottaro,  Andrea  Pérez-Villa,  and  Giovanni  Bussi.   Accurate
     multiple  time  step  in  biased  molecular  simulations.   <span class="ptmri8t-x-x-109">Journal  of  chemical  theory  and  computation</span>,
     11:139--146, 2015.
     </p>
<p class="bibitem"><span class="biblabel">
 [15]<span class="bibsp">   </span></span><a id="XSalari2018"></a>Reza Salari, Thomas Joseph, Ruchi Lohia, Jérôme Hénin, and Grace Brannigan. A streamlined,
     general  approach  for  computing  ligand  binding  free  energies  and  its  application  to  GPCR-bound
     cholesterol. <span class="ptmri8t-x-x-109">Journal of Chemical Theory and Computation</span>, 14(12):6560--6573, 2018.
     </p>
<p class="bibitem"><span class="biblabel">
 [16]<span class="bibsp">   </span></span><a id="XDarve2008"></a>Eric Darve, David Rodríguez-Gómez, and Andrew Pohorille.  Adaptive biasing force method
     for scalar and vector free energy calculations. <span class="ptmri8t-x-x-109">J. Chem. Phys.</span>, 128(14):144120, 2008.
     </p>
<p class="bibitem"><span class="biblabel">
 [17]<span class="bibsp">   </span></span><a id="XHenin2004"></a>J. Hénin  and  C. Chipot.     Overcoming  free  energy  barriers  using  unconstrained  molecular
     dynamics simulations. <span class="ptmri8t-x-x-109">J. Chem. Phys.</span>, 121:2904--2914, 2004.
                                                                                         
                                                                                         
     </p>
<p class="bibitem"><span class="biblabel">
 [18]<span class="bibsp">   </span></span><a id="XHenin2010"></a>Jérôme  Hénin,  Giacomo  Fiorin,  Cristophe  Chipot,  and  Michael L.  Klein.      Exploring
     multidimensional free energy landscapes using time-dependent biases on collective variables. <span class="ptmri8t-x-x-109">J. Chem.</span>
<span class="ptmri8t-x-x-109">Theory Comput.</span>, 6(1):35--47, 2010.
     </p>
<p class="bibitem"><span class="biblabel">
 [19]<span class="bibsp">   </span></span><a id="XCarter1989"></a>A. Carter, E, G. Ciccotti, J. T. Hynes, and R. Kapral. Constrained reaction coordinate dynamics
     for the simulation of rare events. <span class="ptmri8t-x-x-109">Chem. Phys. Lett.</span>, 156:472--477, 1989.
     </p>
<p class="bibitem"><span class="biblabel">
 [20]<span class="bibsp">   </span></span><a id="XRuiz-Montero1997"></a>M. J. Ruiz-Montero, D. Frenkel, and J. J. Brey.  Efficient schemes to compute diffusive barrier
     crossing rates. <span class="ptmri8t-x-x-109">Mol. Phys.</span>, 90:925--941, 1997.
     </p>
<p class="bibitem"><span class="biblabel">
 [21]<span class="bibsp">   </span></span><a id="XdenOtter2000"></a>W. K. den Otter.   Thermodynamic integration of the free energy along a reaction coordinate in
     cartesian coordinates. <span class="ptmri8t-x-x-109">J. Chem. Phys.</span>, 112:7283--7292, 2000.
     </p>
<p class="bibitem"><span class="biblabel">
 [22]<span class="bibsp">   </span></span><a id="XCiccotti2005"></a>Giovanni Ciccotti, Raymond Kapral, and Eric Vanden-Eijnden.   Blue moon sampling, vectorial
     reaction coordinates, and unbiased constrained dynamics. <span class="ptmri8t-x-x-109">ChemPhysChem</span>, 6(9):1809--1814, 2005.
     </p>
<p class="bibitem"><span class="biblabel">
 [23]<span class="bibsp">   </span></span><a id="XMinoukadeh2010"></a>K. Minoukadeh,   C. Chipot,   and   T. Lelièvre.       Potential   of   mean   force   calculations:   A
     multiple-walker adaptive biasing force approach. <span class="ptmri8t-x-x-109">J. Chem. Theor. Comput.</span>, 6:1008--1017, 2010.
     </p>
<p class="bibitem"><span class="biblabel">
 [24]<span class="bibsp">   </span></span><a id="XComer2014c"></a>Jeffrey  Comer,  James C.  Phillips,  Klaus  Schulten,  and  Christophe  Chipot.    Multiple-walker
     strategies for free-energy calculations in NAMD: Shared adaptive biasing force and walker selection
     rules. <span class="ptmri8t-x-x-109">J. Chem. Theor. Comput.</span>, 10(12):5276--5285, 2014.
     </p>
<p class="bibitem"><span class="biblabel">
 [25]<span class="bibsp">   </span></span><a id="XLesage2017"></a>Adrien Lesage, Tony Lelièvre, Gabriel Stoltz, and Jérôme Hénin.  Smoothed biasing forces
     yield unbiased free energies with the extended-system adaptive biasing force method.  <span class="ptmri8t-x-x-109">J. Phys. Chem.</span>
<span class="ptmri8t-x-x-109">B</span>, 121(15):3676--3685, 2017.
     </p>
<p class="bibitem"><span class="biblabel">
 [26]<span class="bibsp">   </span></span><a id="XFu2016"></a>Haohao Fu, Xueguang Shao, Cristophe Chipot, and Wensheng Cai.   Extended adaptive biasing
     force algorithm. An on--the--fly implementation for accurate free--energy calculations. <span class="ptmri8t-x-x-109">J. Chem. Theory</span>
<span class="ptmri8t-x-x-109">Comput.</span>, 12(8):3506----3513, 2016.
                                                                                         
                                                                                         
     </p>
<p class="bibitem"><span class="biblabel">
 [27]<span class="bibsp">   </span></span><a id="XZheng2012"></a>L. Zheng   and   W. Yang.         Practically   efficient   and   robust   free   energy   calculations:
     Double-integration orthogonal space tempering. <span class="ptmri8t-x-x-109">J. Chem. Theor. Compt.</span>, 8:810--823, 2012.
     </p>
<p class="bibitem"><span class="biblabel">
 [28]<span class="bibsp">   </span></span><a id="XKastner2005"></a>J. Kastner  and  W. Thiel.   Bridging  the  gap  between  thermodynamic  integration  and  umbrella
     sampling provides a novel analysis method: “umbrella integration".  <span class="ptmri8t-x-x-109">J. Chem. Phys.</span>, 123(14):144104,
     2005.
     </p>
<p class="bibitem"><span class="biblabel">
 [29]<span class="bibsp">   </span></span><a id="XLaio2002"></a>A. Laio  and  M. Parrinello.     Escaping  free-energy  minima.     <span class="ptmri8t-x-x-109">Proc.  Natl.  Acad.  Sci.  USA</span>,
     99(20):12562--12566, 2002.
     </p>
<p class="bibitem"><span class="biblabel">
 [30]<span class="bibsp">   </span></span><a id="XGrubmuller1995"></a>Helmut   Grubmüller.       Predicting   slow   structural   transitions   in   macromolecular   systems:
     Conformational flooding. <span class="ptmri8t-x-x-109">Phys. Rev. E</span>, 52(3):2893--2906, Sep 1995.
     </p>
<p class="bibitem"><span class="biblabel">
 [31]<span class="bibsp">   </span></span><a id="XHuber1994"></a>T. Huber,  A. E.  Torda,  and  W.F.  van  Gunsteren.    Local  elevation  -  A  method  for  improving
     the searching properties of molecular-dynamics simulation.   <span class="ptmri8t-x-x-109">Journal of Computer-Aided Molecular</span>
<span class="ptmri8t-x-x-109">Design</span>, 8(6):695--708, DEC 1994.
     </p>
<p class="bibitem"><span class="biblabel">
 [32]<span class="bibsp">   </span></span><a id="XBussi2006"></a>G. Bussi,   A. Laio,   and   M. Parrinello.       Equilibrium   free   energies   from   nonequilibrium
     metadynamics. <span class="ptmri8t-x-x-109">Phys. Rev. Lett.</span>, 96(9):090601, 2006.
     </p>
<p class="bibitem"><span class="biblabel">
 [33]<span class="bibsp">   </span></span><a id="XMarinelli2009"></a>Fabrizio  Marinelli,  Fabio  Pietrucci,  Alessandro  Laio,  and  Stefano  Piana.    A  kinetic  model  of
     trp-cage folding from multiple biased molecular dynamics simulations. <span class="ptmri8t-x-x-109">PLOS Computational Biology</span>,
     5(8):1--18, 2009.
     </p>
<p class="bibitem"><span class="biblabel">
 [34]<span class="bibsp">   </span></span><a id="XCrespo2010"></a>Yanier  Crespo,  Fabrizio  Marinelli,  Fabio  Pietrucci,  and  Alessandro  Laio.      Metadynamics
     convergence law in a multidimensional system. <span class="ptmri8t-x-x-109">Phys. Rev. E</span>, 81:055701, May 2010.
     </p>
<p class="bibitem"><span class="biblabel">
 [35]<span class="bibsp">   </span></span><a id="XMarinelli2015"></a>Fabrizio Marinelli and José D. Faraldo-Gómez.  Ensemble-biased metadynamics: A molecular
     simulation method to sample experimental distributions. <span class="ptmri8t-x-x-109">Biophys. J.</span>, 108(12):2779--2782, 2015.
                                                                                         
                                                                                         
     </p>
<p class="bibitem"><span class="biblabel">
 [36]<span class="bibsp">   </span></span><a id="XBarducci2008"></a>Alessandro Barducci, Giovanni Bussi, and Michele Parrinello.  Well-tempered metadynamics: A
     smoothly converging and tunable free-energy method. <span class="ptmri8t-x-x-109">Phys. Rev. Lett.</span>, 100:020603, 2008.
     </p>
<p class="bibitem"><span class="biblabel">
 [37]<span class="bibsp">   </span></span><a id="XRaiteri2006"></a>P. Raiteri, A. Laio, F. L. Gervasio, C. Micheletti, and M. Parrinello.  Efficient reconstruction of
     complex free energy landscapes by multiple walkers metadynamics. <span class="ptmri8t-x-x-109">J. Phys. Chem. B</span>, 110(8):3533--9,
     2006.
     </p>
<p class="bibitem"><span class="biblabel">
 [38]<span class="bibsp">   </span></span><a id="XDeng2009"></a>Yuqing Deng and Benoît Roux.  Computations of standard binding free energies with molecular
     dynamics simulations. <span class="ptmri8t-x-x-109">J. Phys. Chem. B</span>, 113(8):2234--2246, 2009.
     </p>
<p class="bibitem"><span class="biblabel">
 [39]<span class="bibsp">   </span></span><a id="XPitera2012"></a>Jed W. Pitera and John D. Chodera.  On the use of experimental observations to bias simulated
     ensembles. <span class="ptmri8t-x-x-109">J. Chem. Theory Comput.</span>, 8:3445--3451, 2012.
     </p>
<p class="bibitem"><span class="biblabel">
 [40]<span class="bibsp">   </span></span><a id="XWhite2014"></a>Andrew D.  White  and  Gregory A.  Voth.     Efficient  and  minimal  method  to  bias  molecular
     simulations with experimental data. <span class="ptmri8t-x-x-109">J. Chem. Theory Comput.</span>, 10(8):3023----3030, 2014.
     </p>
<p class="bibitem"><span class="biblabel">
 [41]<span class="bibsp">   </span></span><a id="Xhamelberg2004"></a>Donald Hamelberg, John Mongan, and J. Andrew McCammon. Accelerated molecular dynamics:
     A promising and efficient simulation method for biomolecules. <span class="ptmri8t-x-x-109">J. Chem. Phys.</span>, 120(24):11919--11929,
     June 2004.
     </p>
<p class="bibitem"><span class="biblabel">
 [42]<span class="bibsp">   </span></span><a id="XMiao2015"></a>Yinglong Miao, Victoria A. Feher, and J. Andrew McCammon. Gaussian Accelerated Molecular
     Dynamics: Unconstrained Enhanced Sampling and Free Energy Calculation. <span class="ptmri8t-x-x-109">J. Chem. Theory Comput</span>,
     11(8):3584--3595, August 2015.
     </p>
<p class="bibitem"><span class="biblabel">
 [43]<span class="bibsp">   </span></span><a id="XChen2021"></a>Haochuan   Chen,   Haohao   Fu,   Christophe   Chipot,   Xueguang   Shao,   and   Wensheng   Cai.
     Overcoming  free-energy  barriers  with  a  seamless  combination  of  a  biasing  force  and  a  collective
     variable-independent boost potential. <span class="ptmri8t-x-x-109">J. Chem. Theory Comput.</span>, 17(7):3886--3894, 2021.
     </p>
<p class="bibitem"><span class="biblabel">
 [44]<span class="bibsp">   </span></span><a id="XShen2015"></a>Rong Shen, Wei Han, Giacomo Fiorin, Shahidul M Islam, Klaus Schulten, and Benoît Roux.
     Structural refinement of proteins by restrained molecular dynamics simulations with non-interacting
     molecular fragments. <span class="ptmri8t-x-x-109">PLoS Comput. Biol.</span>, 11(10):e1004368, 2015.
                                                                                         
                                                                                         
</p>
</div>
<!--l. 311--><p class="indent"> </p></div> </div>
</body>
</html>
