<!DOCTYPE html>
<html>
<head> <title>Collective Variables Module - Colvars Module - Reference Manual</title>
<meta charset="utf-8"/>
<meta content="width=device-width, initial-scale=1" name="viewport"/>
<meta content="TeX4ht (http://www.cse.ohio-state.edu/~gurari/TeX4ht/)" name="generator"/>
<link href="colvars-refman-namd.css" rel="stylesheet" type="text/css"/>
<script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML" type="text/javascript"></script>
<style type="text/css"> 
.MathJax_MathML {text-indent: 0;} 
</style>
</head><body>
<!--l. 136--><p class="noindent"></p><div id="container"> <div class="inner">
<!--l. 139--><p class="indent">
</p><!--l. 141--><p class="noindent"><span class="colorbox" id="colorbox1"> </span></p><div class="minipage"><!--l. 180--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mspace class="thinspace" width="0.17em"></mspace></math><br class="newline"/>
<!--l. 180--><p class="noindent"><span class="phvb8t-x-x-248">COLLECTIVE VARIABLES MODULE</span><br class="newline"/>
</p><!--l. 180--><p class="noindent"><span class="phvb8t-x-x-248">Reference manual for NAMD</span><br class="newline"/>
</p><!--l. 180--><p class="noindent"><br class="newline"/>Code version: 2025-10-13<br class="newline"/>
</p><!--l. 180--><p class="noindent">Updated versions of this manual: [<a href="https://colvars.github.io/master/colvars-refman-gromacs.html">GROMACS</a>] [<a href="https://colvars.github.io/master/colvars-refman-lammps.html">LAMMPS</a>] [<a href="https://colvars.github.io/master/colvars-refman-namd.html">NAMD</a>] [<a href="https://colvars.github.io/master/colvars-refman-tinkerhp.html">Tinker-HP</a>] [<a href="https://colvars.github.io/master/colvars-refman-vmd.html">VMD</a>]</p></div>
<!--l. 184--><p class="indent"> <img alt="Colvars logo" class="cover_picture" src="cover-512px.jpg"/> <br class="newline"/><span class="phvr8t-x-x-120">Alejandro Bernardin, Haochuan Chen, Jeffrey R. Comer, Giacomo Fiorin, Haohao Fu, J</span><span class="phvr8t-x-x-120">é</span><span class="phvr8t-x-x-120">r</span><span class="phvr8t-x-x-120">ô</span><span class="phvr8t-x-x-120">me</span>
<span class="phvr8t-x-x-120">H</span><span class="phvr8t-x-x-120">é</span><span class="phvr8t-x-x-120">nin, Axel Kohlmeyer, Fabrizio Marinelli, Hubert Santuz, Joshua V. Vermaas, Andrew D.</span>
<span class="phvr8t-x-x-120">White, Wei Zhang</span>
</p><!--l. 192--><p class="noindent">(<a href="https://colvars.github.io/master/colvars-refman-namd.pdf">PDF version</a>)
                                                                                         
                                                                                         
</p><!--l. 203--><p class="indent">
<a id="sec:toc"></a>
</p><!--l. 205--><p class="indent"> 
</p>
<h3 class="likesectionHead"><a id="x1-1000"></a>Contents</h3>
<div class="tableofcontents">
    <span class="sectionToc">1 <a href="#sec:colvars_intro" id="QQ2-1-2">Overview</a></span>
<br/>    <span class="sectionToc">2 <a href="#sec:colvars_crash_course" id="QQ2-1-3">Writing a Colvars configuration: a crash course</a></span>
<br/>    <span class="sectionToc">3 <a href="#sec:colvarmodule" id="QQ2-1-6">Enabling and controlling the Colvars module in NAMD</a></span>
<br/>     <span class="subsectionToc">3.1 <a href="#sec:units" id="QQ2-1-7">Units in the Colvars module</a></span>
<br/>     <span class="subsectionToc">3.2 <a href="#sec:colvars_mdengine_parameters" id="QQ2-1-8">NAMD parameters</a></span>
<br/>     <span class="subsectionToc">3.3 <a href="#sec:cv_scripting" id="QQ2-1-9">Using the scripting interface to control the Colvars module</a></span>
<br/>      <span class="subsubsectionToc">3.3.1 <a href="#sec:cv_command_init" id="QQ2-1-10">Setting up the Colvars module</a></span>
<br/>      <span class="subsubsectionToc">3.3.2 <a href="#sec:cv_command_version" id="QQ2-1-11">Using the Colvars version in scripts</a></span>
<br/>      <span class="subsubsectionToc">3.3.3 <a href="#sec:cv_command_loadsave" id="QQ2-1-12">Loading and saving the Colvars state and other information</a></span>
<br/>      <span class="subsubsectionToc">3.3.4 <a href="#sec:cv_command_atomic_data" id="QQ2-1-13">Accessing atomic data</a></span>
<br/>      <span class="subsubsectionToc">3.3.5 <a href="#sec:cv_command_colvar" id="QQ2-1-14">Managing collective variables</a></span>
<br/>      <span class="subsubsectionToc">3.3.6 <a href="#sec:cv_command_colvar_force" id="QQ2-1-15">Applying and analyzing forces on collective variables</a></span>
<br/>      <span class="subsubsectionToc">3.3.7 <a href="#sec:cv_command_bias" id="QQ2-1-16">Managing collective variable biases</a></span>
<br/>      <span class="subsubsectionToc">3.3.8 <a href="#sec:cv_command_bias_loadsave" id="QQ2-1-17">Loading and saving the state of individual biases</a></span>
<br/>     <span class="subsectionToc">3.4 <a href="#sec:colvars_config_syntax" id="QQ2-1-18">Configuration syntax used by the Colvars module</a></span>
<br/>     <span class="subsectionToc">3.5 <a href="#sec:colvars_global" id="QQ2-1-19">Global keywords</a></span>
<br/>     <span class="subsectionToc">3.6 <a href="#sec:colvars_input" id="QQ2-1-20">Input state file</a></span>
<br/>      <span class="subsubsectionToc">3.6.1 <a href="#" id="QQ2-1-21">Restarting in NAMD.</a></span>
<br/>      <span class="subsubsectionToc">3.6.2 <a href="#sec:change_config_restart" id="QQ2-1-22">Changing configuration upon restarting.</a></span>
<br/>     <span class="subsectionToc">3.7 <a href="#sec:colvars_output" id="QQ2-1-23">Output files</a></span>
<br/>     <span class="subsectionToc">3.8 <a href="#sec:colvars_file_formats" id="QQ2-1-24">File formats</a></span>
<br/>      <span class="subsubsectionToc">3.8.1 <a href="#sec:colvars_text_format" id="QQ2-1-25">Configuration and state files.</a></span>
<br/>      <span class="subsubsectionToc">3.8.2 <a href="#sec:colvars_ndx_format" id="QQ2-1-26">Index (NDX) files</a></span>
<br/>      <span class="subsubsectionToc">3.8.3 <a href="#sec:colvars_xyz_format" id="QQ2-1-27">XYZ coordinate files</a></span>
<br/>      <span class="subsubsectionToc">3.8.4 <a href="#sec:colvars_pdb_format" id="QQ2-1-28">PDB coordinate files</a></span>
<br/>      <span class="subsubsectionToc">3.8.5 <a href="#sec:colvar_multicolumn_grid" id="QQ2-1-29">Grid files: multicolumn text format</a></span>
<br/>      <span class="subsubsectionToc">3.8.6 <a href="#sec:colvars_traj_format" id="QQ2-1-30">Output trajectory files</a></span>
<br/>    <span class="sectionToc">4 <a href="#sec:colvar" id="QQ2-1-31">Defining collective variables</a></span>
<br/>     <span class="subsectionToc">4.1 <a href="#sec:cvc_list" id="QQ2-1-32">Choosing a function</a></span>
<br/>     <span class="subsectionToc">4.2 <a href="#sec:cvc_pbcs" id="QQ2-1-33">Treatment of periodic boundary conditions</a></span>
<br/>     <span class="subsectionToc">4.3 <a href="#sec:cvc_distances" id="QQ2-1-35">Distances</a></span>
<br/>      <span class="subsubsectionToc">4.3.1 <a href="#sec:cvc_distance" id="QQ2-1-36"><span class="ectt-1095">distance</span>: center-of-mass distance between two groups.</a></span>
<br/>      <span class="subsubsectionToc">4.3.2 <a href="#sec:cvc_distanceZ" id="QQ2-1-37"><span class="ectt-1095">distanceZ</span>: projection of a distance vector on an axis.</a></span>
<br/>      <span class="subsubsectionToc">4.3.3 <a href="#sec:cvc_distanceXY" id="QQ2-1-38"><span class="ectt-1095">distanceXY</span>: modulus of the projection of a distance vector on a plane.</a></span>
<br/>      <span class="subsubsectionToc">4.3.4 <a href="#sec:cvc_distanceVec" id="QQ2-1-39"><span class="ectt-1095">distanceVec</span>: distance vector between two groups.</a></span>
<br/>      <span class="subsubsectionToc">4.3.5 <a href="#sec:cvc_distanceDir" id="QQ2-1-40"><span class="ectt-1095">distanceDir</span>: distance unit vector between two groups.</a></span>
<br/>      <span class="subsubsectionToc">4.3.6 <a href="#sec:cvc_distanceInv" id="QQ2-1-41"><span class="ectt-1095">distanceInv</span>: mean distance between two groups of atoms.</a></span>
<br/>     <span class="subsectionToc">4.4 <a href="#sec:cvc_angles" id="QQ2-1-42">Angles</a></span>
<br/>      <span class="subsubsectionToc">4.4.1 <a href="#sec:cvc_angle" id="QQ2-1-43"><span class="ectt-1095">angle</span>: angle between three groups.</a></span>
<br/>      <span class="subsubsectionToc">4.4.2 <a href="#sec:cvc_dipoleAngle" id="QQ2-1-44"><span class="ectt-1095">dipoleAngle</span>: angle between two groups and dipole of a third group.</a></span>
<br/>      <span class="subsubsectionToc">4.4.3 <a href="#sec:cvc_dihedral" id="QQ2-1-45"><span class="ectt-1095">dihedral</span>: torsional angle between four groups.</a></span>
<br/>      <span class="subsubsectionToc">4.4.4 <a href="#sec:cvc_polarTheta" id="QQ2-1-46"><span class="ectt-1095">polarTheta</span>: polar angle in spherical coordinates.</a></span>
<br/>      <span class="subsubsectionToc">4.4.5 <a href="#sec:cvc_polarPhi" id="QQ2-1-47"><span class="ectt-1095">polarPhi</span>: azimuthal angle in spherical coordinates.</a></span>
<br/>     <span class="subsectionToc">4.5 <a href="#sec:cvc_contacts" id="QQ2-1-48">Contacts</a></span>
<br/>      <span class="subsubsectionToc">4.5.1 <a href="#sec:cvc_coordNum" id="QQ2-1-49"><span class="ectt-1095">coordNum</span>: coordination number between two groups.</a></span>
<br/>      <span class="subsubsectionToc">4.5.2 <a href="#sec:cvc_selfCoordNum" id="QQ2-1-50"><span class="ectt-1095">selfCoordNum</span>: coordination number between atoms within a group.</a></span>
<br/>      <span class="subsubsectionToc">4.5.3 <a href="#sec:cvc_hBond" id="QQ2-1-51"><span class="ectt-1095">hBond</span>: hydrogen bond between two atoms.</a></span>
<br/>     <span class="subsectionToc">4.6 <a href="#sec:cvc_collective" id="QQ2-1-52">Collective metrics</a></span>
<br/>      <span class="subsubsectionToc">4.6.1 <a href="#sec:cvc_rmsd" id="QQ2-1-53"><span class="ectt-1095">rmsd</span>: root mean square displacement (RMSD) from reference
positions.</a></span>
<br/>      <span class="subsubsectionToc">4.6.2 <a href="#sec:cvc_rmsd_advanced" id="QQ2-1-54">Advanced usage of the <span class="ectt-1095">rmsd </span>component.</a></span>
<br/>      <span class="subsubsectionToc">4.6.3 <a href="#sec:cvc_eigenvector" id="QQ2-1-55"><span class="ectt-1095">eigenvector</span>: projection of the atomic coordinates on a vector.</a></span>
<br/>      <span class="subsubsectionToc">4.6.4 <a href="#sec:cvc_gyration" id="QQ2-1-56"><span class="ectt-1095">gyration</span>: radius of gyration of a group of atoms.</a></span>
<br/>      <span class="subsubsectionToc">4.6.5 <a href="#sec:cvc_inertia" id="QQ2-1-57"><span class="ectt-1095">inertia</span>: total moment of inertia of a group of atoms.</a></span>
<br/>      <span class="subsubsectionToc">4.6.6 <a href="#sec:cvc_dipoleMagnitude" id="QQ2-1-58"><span class="ectt-1095">dipoleMagnitude</span>: dipole magnitude of a group of atoms.</a></span>
<br/>      <span class="subsubsectionToc">4.6.7 <a href="#sec:cvc_inertiaZ" id="QQ2-1-59"><span class="ectt-1095">inertiaZ</span>: total moment of inertia of a group of atoms around
a chosen axis.</a></span>
<br/>     <span class="subsectionToc">4.7 <a href="#sec:cvc_rotations" id="QQ2-1-60">Rotations</a></span>
<br/>      <span class="subsubsectionToc">4.7.1 <a href="#sec:cvc_orientation" id="QQ2-1-61"><span class="ectt-1095">orientation</span>: orientation from reference coordinates.</a></span>
<br/>      <span class="subsubsectionToc">4.7.2 <a href="#sec:cvc_orientationAngle" id="QQ2-1-63"><span class="ectt-1095">orientationAngle</span>: angle of rotation from reference coordinates.</a></span>
<br/>      <span class="subsubsectionToc">4.7.3 <a href="#sec:cvc_orientationProj" id="QQ2-1-64"><span class="ectt-1095">orientationProj</span>: cosine of the angle of rotation from
reference coordinates.</a></span>
<br/>      <span class="subsubsectionToc">4.7.4 <a href="#sec:cvc_spinAngle" id="QQ2-1-65"><span class="ectt-1095">spinAngle</span>: angle of rotation around a given axis.</a></span>
<br/>      <span class="subsubsectionToc">4.7.5 <a href="#sec:cvc_tilt" id="QQ2-1-66"><span class="ectt-1095">tilt</span>: cosine of the rotation orthogonal to a given axis.</a></span>
<br/>      <span class="subsubsectionToc">4.7.6 <a href="#sec:cvc_eulerPhi" id="QQ2-1-67"><span class="ectt-1095">eulerPhi</span>: Roll angle from references coordinates.</a></span>
<br/>      <span class="subsubsectionToc">4.7.7 <a href="#sec:cvc_eulerTheta" id="QQ2-1-68"><span class="ectt-1095">eulerTheta</span>: Pitch angle from references coordinates.</a></span>
<br/>      <span class="subsubsectionToc">4.7.8 <a href="#sec:cvc_eulerPsi" id="QQ2-1-69"><span class="ectt-1095">eulerPsi</span>: Yaw angle from references coordinates.</a></span>
<br/>     <span class="subsectionToc">4.8 <a href="#sec:cvc_protein" id="QQ2-1-70">Protein structure descriptors</a></span>
<br/>      <span class="subsubsectionToc">4.8.1                                                                              <a href="#sec:cvc_alpha" id="QQ2-1-71"><span class="ectt-1095">alpha</span>:
<!--l. 72--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>α</mi></math>-helix
content of a protein segment.</a></span>
<br/>      <span class="subsubsectionToc">4.8.2 <a href="#sec:cvc_dihedralPC" id="QQ2-1-72"><span class="ectt-1095">dihedralPC</span>: protein dihedral principal component</a></span>
<br/>     <span class="subsectionToc">4.9 <a href="#sec:cvc_alch" id="QQ2-1-73">Alchemical variables for lambda-dynamics</a></span>
<br/>      <span class="subsubsectionToc">4.9.1 <a href="#sec:cvc_alchlambda" id="QQ2-1-74"><span class="ectt-1095">alchLambda</span>: alchemical lambda parameter.</a></span>
<br/>      <span class="paragraphToc"><a href="#x1-750004.9.1" id="QQ2-1-75">Special considerations for running lambda-dynamics in NAMD/Colvars.</a></span>
<br/>      <span class="subsubsectionToc">4.9.2 <a href="#sec:cvc_alchFlambda" id="QQ2-1-76"><span class="ectt-1095">alchFlambda</span>: Force on the alchemical lambda parameter.</a></span>
<br/>     <span class="subsectionToc">4.10 <a href="#sec:cvc_raw" id="QQ2-1-77">Raw data: building blocks for custom functions</a></span>
<br/>      <span class="subsubsectionToc">4.10.1 <a href="#sec:cvc_cartesian" id="QQ2-1-78"><span class="ectt-1095">cartesian</span>: vector of atomic Cartesian coordinates.</a></span>
<br/>      <span class="subsubsectionToc">4.10.2 <a href="#sec:cvc_distancePairs" id="QQ2-1-79"><span class="ectt-1095">distancePairs</span>: set of pairwise distances between two groups.</a></span>
<br/>     <span class="subsectionToc">4.11 <a href="#sec:cvc_gpath" id="QQ2-1-80">Geometric path collective variables</a></span>
<br/>      <span class="subsubsectionToc">4.11.1 <a href="#sec:cvc_gspath" id="QQ2-1-81"><span class="ectt-1095">gspath</span>: progress along a path defined in atomic Cartesian
coordinate space.</a></span>
<br/>      <span class="subsubsectionToc">4.11.2 <a href="#sec:cvc_gzpath" id="QQ2-1-82"><span class="ectt-1095">gzpath</span>: distance from a path defined in atomic Cartesian
coordinate space.</a></span>
<br/>      <span class="subsubsectionToc">4.11.3 <a href="#sec:cvc_linearCombination" id="QQ2-1-83"><span class="ectt-1095">linearCombination</span>: Helper CV to define a linear combination
of other CVs</a></span>
<br/>      <span class="subsubsectionToc">4.11.4 <a href="#sec:cvc_customColvar" id="QQ2-1-84"><span class="ectt-1095">customColvar</span>: Helper CV to define a mathematical expression
as CV from other CVs</a></span>
<br/>      <span class="subsubsectionToc">4.11.5 <a href="#sec:cvc_gspathCV" id="QQ2-1-85"><span class="ectt-1095">gspathCV</span>: progress along a path defined in CV space.</a></span>
<br/>      <span class="subsubsectionToc">4.11.6 <a href="#sec:cvc_gzpathCV" id="QQ2-1-86"><span class="ectt-1095">gzpathCV</span>: distance from a path defined in CV space.</a></span>
<br/>     <span class="subsectionToc">4.12 <a href="#sec:cvc_apathCV" id="QQ2-1-87">Arithmetic path collective variables</a></span>
<br/>      <span class="subsubsectionToc">4.12.1 <a href="#sec:cvc_aspathCV" id="QQ2-1-88"><span class="ectt-1095">aspathCV</span>: progress along a path defined in CV space.</a></span>
<br/>      <span class="subsubsectionToc">4.12.2 <a href="#sec:cvc_azpathCV" id="QQ2-1-89"><span class="ectt-1095">azpathCV</span>: distance from a path defined in CV space.</a></span>
<br/>      <span class="subsubsectionToc">4.12.3 <a href="#sec:apath_scripted" id="QQ2-1-90">Path collective variables in Cartesian coordinates</a></span>
<br/>      <span class="subsubsectionToc">4.12.4 <a href="#sec:cvc_aspath" id="QQ2-1-91"><span class="ectt-1095">aspath</span>: progress along a path defined in atomic Cartesian
coordinate space.</a></span>
<br/>      <span class="subsubsectionToc">4.12.5 <a href="#sec:cvc_azpath" id="QQ2-1-92"><span class="ectt-1095">azpath</span>: distance from a path defined in atomic Cartesian
coordinate space.</a></span>
<br/>     <span class="subsectionToc">4.13 <a href="#sec:cvc_neuralNetwork" id="QQ2-1-93">Dense neural network in CV space (MLCV)</a></span>
<br/>     <span class="subsectionToc">4.14 <a href="#sec:cvc_torchann" id="QQ2-1-95">Torch-based collective variables</a></span>
<br/>     <span class="subsectionToc">4.15 <a href="#sec:cvc_volmaps" id="QQ2-1-96">Volumetric map-based variables</a></span>
<br/>      <span class="subsubsectionToc">4.15.1 <a href="#sec:cvc_map_total" id="QQ2-1-97"><span class="ectt-1095">mapTotal</span>: total value of a volumetric map</a></span>
<br/>      <span class="subsubsectionToc">4.15.2 <a href="#sec:cvc_multimap" id="QQ2-1-98">Multiple volumetric maps collective variables</a></span>
<br/>     <span class="subsectionToc">4.16 <a href="#sec:cvc_common" id="QQ2-1-99">Shared keywords for all components</a></span>
<br/>     <span class="subsectionToc">4.17 <a href="#sec:cvc_periodic" id="QQ2-1-100">Periodic components</a></span>
<br/>     <span class="subsectionToc">4.18 <a href="#sec:cvc_non_scalar" id="QQ2-1-101">Non-scalar components</a></span>
<br/>      <span class="subsubsectionToc">4.18.1 <a href="#sec:cvc_sys_forces" id="QQ2-1-102">Calculating total forces</a></span>
<br/>     <span class="subsectionToc">4.19 <a href="#sec:cvc_superp" id="QQ2-1-103">Linear and polynomial combinations of components</a></span>
<br/>     <span class="subsectionToc">4.20 <a href="#sec:colvar_custom_function" id="QQ2-1-104">Custom functions</a></span>
<br/>     <span class="subsectionToc">4.21 <a href="#sec:colvar_scripted" id="QQ2-1-105">Scripted functions</a></span>
<br/>     <span class="subsectionToc">4.22 <a href="#sec:colvar_grid_params" id="QQ2-1-106">Defining grid parameters for a colvar</a></span>
<br/>     <span class="subsectionToc">4.23 <a href="#sec:colvar_traj_output" id="QQ2-1-107">Trajectory output</a></span>
<br/>     <span class="subsectionToc">4.24 <a href="#sec:colvar_extended" id="QQ2-1-108">Extended Lagrangian</a></span>
<br/>     <span class="subsectionToc">4.25 <a href="#sec:mts_colvar" id="QQ2-1-109">Multiple time-step variables</a></span>
<br/>     <span class="subsectionToc">4.26 <a href="#sec:colvar_compatibility" id="QQ2-1-110">Backward-compatibility</a></span>
<br/>     <span class="subsectionToc">4.27 <a href="#sec:colvar_acf" id="QQ2-1-111">Statistical analysis</a></span>
<br/>    <span class="sectionToc">5 <a href="#sec:colvar_atom_groups" id="QQ2-1-112">Selecting atoms</a></span>
<br/>     <span class="subsectionToc">5.1 <a href="#sec:colvar_atom_groups_sel" id="QQ2-1-113">Atom selection keywords</a></span>
<br/>     <span class="subsectionToc">5.2 <a href="#sec:colvar_atom_groups_ref_frame" id="QQ2-1-114">Moving frame of reference.</a></span>
<br/>     <span class="subsectionToc">5.3 <a href="#sec:colvar_atom_groups_wrapping" id="QQ2-1-117">Treatment of periodic boundary conditions.</a></span>
<br/>     <span class="subsectionToc">5.4 <a href="#sec:colvar_atom_groups_scaling" id="QQ2-1-118">Performance of a Colvars calculation based on group size.</a></span>
<br/>    <span class="sectionToc">6 <a href="#sec:colvarbias" id="QQ2-1-119">Biasing and analysis methods</a></span>
<br/>     <span class="subsectionToc">6.1 <a href="#sec:colvarbias_grid" id="QQ2-1-120">Defining grids for biasing and analysis methods</a></span>
<br/>     <span class="subsectionToc">6.2 <a href="#sec:colvarbias_ti" id="QQ2-1-121">Thermodynamic integration</a></span>
<br/>     <span class="subsectionToc">6.3 <a href="#sec:colvarbias_abf" id="QQ2-1-122">Adaptive Biasing Force</a></span>
<br/>      <span class="subsubsectionToc">6.3.1 <a href="#sec:colvarbias_abf_req" id="QQ2-1-123">ABF requirements on collective variables</a></span>
<br/>      <span class="subsubsectionToc">6.3.2 <a href="#sec:colvarbias_abf_params" id="QQ2-1-124">Parameters for ABF</a></span>
<br/>      <span class="subsubsectionToc">6.3.3 <a href="#sec:colvarbias_abf_shared" id="QQ2-1-125">Multiple-walker ABF</a></span>
<br/>      <span class="paragraphToc"><a href="#x1-1250006.3.3" id="QQ2-1-126">Output files of multiple-walker ABF.</a></span>
<br/>      <span class="subsubsectionToc">6.3.4 <a href="#sec:colvarbias_abf_output" id="QQ2-1-127">Output files</a></span>
<br/>      <span class="subsubsectionToc">6.3.5 <a href="#sec:colvarbias_abf_post" id="QQ2-1-128">Multidimensional free energy surfaces</a></span>
<br/>     <span class="subsectionToc">6.4 <a href="#sec:colvarbias_abf_extended" id="QQ2-1-129">Extended-system Adaptive Biasing Force (eABF)</a></span>
<br/>      <span class="subsubsectionToc">6.4.1 <a href="#sec:colvarbias_ti_ext_czar" id="QQ2-1-130">CZAR estimator of the free energy</a></span>
<br/>      <span class="subsubsectionToc">6.4.2 <a href="#sec:colvarbias_ti_ext_zheng_yang" id="QQ2-1-131">Zheng/Yang estimator of the free energy</a></span>
<br/>     <span class="subsectionToc">6.5 <a href="#sec:colvarbias_abmd" id="QQ2-1-133">Adiabatic Bias Molecular Dynamics (ABMD)</a></span>
<br/>     <span class="subsectionToc">6.6 <a href="#sec:colvarbias_meta" id="QQ2-1-134">Metadynamics</a></span>
<br/>      <span class="subsubsectionToc">6.6.1 <a href="#sec:colvarbias_meta_boundaries" id="QQ2-1-135">Treatment of the PMF boundaries</a></span>
<br/>      <span class="subsubsectionToc">6.6.2 <a href="#sec:colvarbias_meta_basics" id="QQ2-1-136">Required metadynamics keywords</a></span>
<br/>      <span class="subsubsectionToc">6.6.3 <a href="#sec:colvarbias_meta_output" id="QQ2-1-137">Output files</a></span>
<br/>      <span class="subsubsectionToc">6.6.4 <a href="#sec:colvarbias_meta_performance" id="QQ2-1-138">Performance optimization</a></span>
<br/>      <span class="subsubsectionToc">6.6.5 <a href="#sec:colvarbias_ebmeta" id="QQ2-1-139">Ensemble-Biased Metadynamics</a></span>
<br/>      <span class="subsubsectionToc">6.6.6 <a href="#sec:colvarbias_well_tempered" id="QQ2-1-140">Well-tempered metadynamics</a></span>
<br/>      <span class="subsubsectionToc">6.6.7 <a href="#sec:colvarbias_meta_mr" id="QQ2-1-141">Multiple-walker metadynamics</a></span>
<br/>     <span class="subsectionToc">6.7 <a href="#sec:colvarbias_opes" id="QQ2-1-142">On-the-fly probability enhanced sampling (OPES)</a></span>
<br/>      <span class="subsubsectionToc">6.7.1 <a href="#sec:colvarbias_opes_notes" id="QQ2-1-143">Implementation notes</a></span>
<br/>      <span class="subsubsectionToc">6.7.2 <a href="#sec:colvarbias_opes_general_options" id="QQ2-1-144">General options</a></span>
<br/>      <span class="subsubsectionToc">6.7.3 <a href="#sec:colvarbias_opes_multiple_walker_options" id="QQ2-1-145">Multiple-walker options</a></span>
<br/>      <span class="subsubsectionToc">6.7.4 <a href="#sec:colvarbias_opes_output_options" id="QQ2-1-146">Output options</a></span>
<br/>      <span class="subsubsectionToc">6.7.5 <a href="#sec:colvarbias_opes_example" id="QQ2-1-147">Example input</a></span>
<br/>     <span class="subsectionToc">6.8 <a href="#sec:colvarbias_harmonic" id="QQ2-1-148">Harmonic restraints</a></span>
<br/>      <span class="subsubsectionToc">6.8.1 <a href="#sec:colvarbias_smd" id="QQ2-1-149">Moving restraints: steered molecular dynamics</a></span>
<br/>      <span class="subsubsectionToc">6.8.2 <a href="#sec:colvarbias_us" id="QQ2-1-150">Moving restraints: umbrella sampling</a></span>
<br/>      <span class="subsubsectionToc">6.8.3 <a href="#sec:colvarbias_force_k_ti" id="QQ2-1-151">Changing force constant</a></span>
<br/>     <span class="subsectionToc">6.9 <a href="#sec:colvarbias_acc_work" id="QQ2-1-152">Computing the work of a changing restraint</a></span>
<br/>     <span class="subsectionToc">6.10 <a href="#sec:colvarbias_harmonic_walls" id="QQ2-1-153">Harmonic wall restraints</a></span>
<br/>     <span class="subsectionToc">6.11 <a href="#sec:colvarbias_linear" id="QQ2-1-154">Linear restraints</a></span>
<br/>     <span class="subsectionToc">6.12 <a href="#sec:colvarbias_alb" id="QQ2-1-155">Adaptive Linear Bias/Experiment Directed Simulation</a></span>
<br/>     <span class="subsectionToc">6.13 <a href="#sec:colvarbias_histogram" id="QQ2-1-156">Multidimensional histograms</a></span>
<br/>      <span class="subsubsectionToc">6.13.1 <a href="#sec:colvarbias_histogram_output" id="QQ2-1-157">Output options for multi-dimensional histograms</a></span>
<br/>      <span class="subsubsectionToc">6.13.2 <a href="#sec:colvarbias_histogram_vectors" id="QQ2-1-158">Histogramming vector variables</a></span>
<br/>     <span class="subsectionToc">6.14 <a href="#sec:reweighted_fel_from_aMD" id="QQ2-1-159">Reweighted free-energy landscapes from accelerated MD</a></span>
<br/>     <span class="subsectionToc">6.15 <a href="#sec:colvarbias_restraint_histogram" id="QQ2-1-160">Probability distribution-restraints</a></span>
<br/>     <span class="subsectionToc">6.16 <a href="#sec:scripted_biases" id="QQ2-1-161">Defining scripted biases</a></span>
<br/>     <span class="subsectionToc">6.17 <a href="#sec:cv_scripting_opt" id="QQ2-1-162">Performance of scripted biases</a></span>
<br/>    <span class="sectionToc">7 <a href="#sec:cvscript_cmdline" id="QQ2-1-163">Tcl command-line interface: list of commands</a></span>
<br/>     <span class="subsectionToc">7.1 <a href="#sec:cvscript_cmdline_cv" id="QQ2-1-164">Commands to manage the Colvars module</a></span>
<br/>     <span class="subsectionToc">7.2 <a href="#sec:cvscript_cmdline_colvar" id="QQ2-1-165">Commands to manage individual collective variables</a></span>
<br/>     <span class="subsectionToc">7.3 <a href="#sec:cvscript_cmdline_bias" id="QQ2-1-166">Commands to manage individual biases</a></span>
<br/>    <span class="sectionToc">8 <a href="#sec:colvars_config_changes" id="QQ2-1-167">Syntax changes from older versions</a></span>
<br/>    <span class="sectionToc">9 <a href="#sec:compilation_notes" id="QQ2-1-168">Compilation notes</a></span>
</div>
<!--l. 9--><p class="indent">
</p>
<div class="flushright">
<!--l. 9--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 9--><p class="noindent"> </p><div class="heading"> <a id="sec:colvars_intro"></a>
<h3 class="sectionHead"><span class="titlemark">1    </span> <a id="sec:colvars_intro"></a>Overview</h3>
<!--l. 9--><p class="noindent"></p></div>
<!--l. 12--><p class="indent">   In molecular dynamics simulations, it is often useful to reduce the large number of degrees of freedom
of a physical system into few parameters whose statistical distributions can be analyzed individually, or
used to define biasing potentials to alter the dynamics of the system in a controlled manner. These have
been called ‘order parameters', ‘collective variables', ‘(surrogate) reaction coordinates', and many other
terms.
</p><!--l. 15--><p class="indent">   Here we use primarily the term ‘collective variable', often shortened to
<span class="phvro8t-x-x-109">colvar</span>, to indicate any differentiable function of atomic Cartesian coordinates,
<!--l. 15--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>x</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mi>i</mi></mrow></msub></math>, with
<!--l. 15--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math> between
<!--l. 15--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>1</mn></math> and
<!--l. 15--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>N</mi></math>, the
total number of atoms: </p><table class="equation"><tr><td> <a id="x1-2001r1"></a>
<!--l. 17--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>ξ</mi><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo><mspace class="thickpace" width="0.28em"></mspace> <mo class="MathClass-rel">=</mo> <mi>ξ</mi><mo class="MathClass-open">(</mo><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>X</mi></mstyle><mtext></mtext></mstyle><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo><mo class="MathClass-close">)</mo><mspace class="thickpace" width="0.28em"></mspace> <mo class="MathClass-rel">=</mo> <mi>ξ</mi> <mrow><mo fence="true" form="prefix"> (</mo><mrow><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>x</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo><mo class="MathClass-punc">,</mo><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>x</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mi>j</mi></mrow></msub><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo><mo class="MathClass-punc">,</mo><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>x</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mi>k</mi></mrow></msub><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo><mo class="MathClass-punc">,</mo><mi class="MathClass-op">…</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo></mrow><mo fence="true" form="postfix">)</mo></mrow><mspace class="thickpace" width="0.28em"></mspace><mo class="MathClass-punc">,</mo><mspace class="thickpace" width="0.28em"></mspace><mspace class="thickpace" width="0.28em"></mspace><mn>1</mn> <mo class="MathClass-rel">≤</mo> <mi>i</mi><mo class="MathClass-punc">,</mo><mi>j</mi><mo class="MathClass-punc">,</mo><mi>k</mi><mi class="MathClass-op">…</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo> <mo class="MathClass-rel">≤</mo> <mi>N</mi>
</math></td><td class="eq-no">(1)</td></tr></table>
<!--l. 22--><p class="indent">   This manual documents the collective variables module (<span class="phvb8t-x-x-109">Colvars</span>), a software that provides an implementation
for the functions <!--l. 22--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi><mo class="MathClass-open">(</mo><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>X</mi></mstyle><mtext></mtext></mstyle><mo class="MathClass-close">)</mo></math>
with a focus on flexibility, robustness and high performance. The module is designed to perform multiple
tasks concurrently during or after a simulation, the most common of which are: </p>
<ul class="itemize1">
<li class="itemize">apply restraints or biasing potentials to multiple variables, tailored on the system by choosing
     from a wide set of basis functions, without limitations on their number or on the number of
     atoms involved; while this can in principle be done through a TclForces script, using the
     Colvars module is both easier and computationally more efficient;
                                                                                         
                                                                                         
     </li>
<li class="itemize">calculate  potentials  of  mean  force  (PMFs)  along  any  set  of  variables,  using  different
     enhanced sampling methods, such as Adaptive Biasing Force (ABF), metadynamics, steered
     MD and umbrella sampling; variants of these methods that make use of an ensemble of
     replicas are supported as well;
     </li>
<li class="itemize">calculate  statistical  properties  of  the  variables,  such  as  running  averages  and  standard
     deviations, correlation functions of pairs of variables, and multidimensional histograms: this
     can be done either at run-time without the need to save very large trajectory files, or after a
     simulation has been completed (post-processing).
     </li></ul>
<!--l. 36--><p class="indent">   Detailed explanations of the design of the Colvars module are provided in reference <span class="cite">[<a href="#XFiorin2013">1</a>]</span>. Please cite
this reference whenever publishing work that makes use of this module, alongside any other publications
for specific features being, according to the usage summary printed when running a Colvars-enabled MD
simulation or analysis.
</p><!--l. 48--><p class="indent">
</p>
<div class="flushright">
<!--l. 48--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 48--><p class="noindent"> </p><div class="heading"> <a id="sec:colvars_crash_course"></a>
<h3 class="sectionHead"><span class="titlemark">2    </span> <a id="sec:colvars_crash_course"></a>Writing a Colvars configuration: a crash course</h3>
<!--l. 48--><p class="noindent"></p></div>
<!--l. 50--><p class="indent">   The Colvars configuration is a plain text file or string that defines collective variables, biases, and
general parameters of the Colvars module. It is passed to the module using back-end-specific commands
documented in section <a href="#sec:colvarmodule">3<!--tex4ht:ref: sec:colvarmodule --></a>.
</p>
<!--l. 56--><p class="noindent"><span class="likeparagraphHead"><a id="x1-40002"></a>Example: steering two groups of atoms away from each other.</span>
   Now let us look at a complete, non-trivial configuration. Suppose that we want to run a steered MD
experiment where a small molecule is pulled away from a protein binding site. In Colvars terms, this is
done by applying a moving restraint to the distance between the two objects. The configuration will
contain two blocks, one defining the distance variable (see section <a href="#sec:colvar">4<!--tex4ht:ref: sec:colvar --></a> and <a href="#sec:cvc_distance">4.3.1<!--tex4ht:ref: sec:cvc_distance --></a>), and the other the moving
harmonic restraint (<a href="#sec:colvarbias_harmonic">6.8<!--tex4ht:ref: sec:colvarbias_harmonic --></a>).</p>
<div class="mdframed" id="mdframed-1">
<!--l. 64--><p class="noindent"></p><pre>
<!--l. 66--><p class="noindent">colvar {<br class="newline"/>  name dist<br class="newline"/>  distance {<br class="newline"/>    group1 { atomNumbersRange 42-55 }<br class="newline"/>    group2 { indexGroup C-alpha_15-30 }<br class="newline"/>  }<br class="newline"/>}<br class="newline"/><br class="newline"/>harmonic {<br class="newline"/>  colvars dist<br class="newline"/>  forceConstant 20.0<br class="newline"/>  centers 4.0         # initial distance<br class="newline"/>  targetCenters 15.0  # final distance<br class="newline"/>  targetNumSteps 500000<br class="newline"/>} </p></pre><!--l. 81--><p class="noindent"></p>
</div>
<!--l. 83--><p class="indent">   Reading this input in plain English: the variable here named <span class="phvro8t-x-x-109">dist </span>consists in a distance
function between the centers of two groups: the ligand (atoms 42 to 55) and the
<!--l. 83--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>α</mi></math>-carbon atoms of
residues 15 to 30 in the protein (segment name PR). To the “<span class="phvro8t-x-x-109">dist</span>" variable, we apply a <span class="ectt-1095">harmonic </span>potential of force constant
20 kcal/mol/Å<!--l. 84--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow></mrow><mrow><mn>2</mn></mrow></msup></math>,
                                                                                         
                                                                                         
initially centered around a value of 4 Å, which will increase to 15 Å over 500,000 simulation
steps.
</p><!--l. 86--><p class="indent">   The atom selection keywords are detailed in section <a href="#sec:colvar_atom_groups">5<!--tex4ht:ref: sec:colvar_atom_groups --></a>.
</p>
<!--l. 90--><p class="noindent"><span class="likeparagraphHead"><a id="x1-50002"></a>Example: using multiple variables and multiple biasing/analysis methods together.</span>
   A more complex example configuration is included below, showing how a variable
may be constructed by combining multiple existing functions, and how multiple
variables or multiple biases may be used concurrently. The colvar indicated below as
“<!--l. 93--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>d</mi></math>"
is defined as the difference between two distances (see <a href="#sec:cvc_distances">4.3<!--tex4ht:ref: sec:cvc_distances --></a>): the first distance
(<!--l. 93--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>d</mi></mrow><mrow><mn>1</mn></mrow></msub></math>) is
taken between the center of mass of atoms 1 and 2 and that of atoms 3 to 5, the second
(<!--l. 93--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>d</mi></mrow><mrow><mn>2</mn></mrow></msub></math>)
between atom 7 and the center of mass of atoms 8 to 10 (see <a href="#sec:colvar_atom_groups">5<!--tex4ht:ref: sec:colvar_atom_groups --></a>). The difference
<!--l. 94--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>d</mi> <mo class="MathClass-rel">=</mo> <msub><mrow><mi>d</mi></mrow><mrow><mn>1</mn></mrow></msub> <mo class="MathClass-bin">−</mo><msub><mrow><mi>d</mi></mrow><mrow><mn>2</mn></mrow></msub></math> is obtained by multiplying
the two by a coefficient <!--l. 94--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>C</mi> <mo class="MathClass-rel">=</mo> <mo class="MathClass-bin">+</mo><mn>1</mn></math>
or <!--l. 94--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>C</mi> <mo class="MathClass-rel">=</mo> <mo class="MathClass-bin">−</mo><mn>1</mn></math>, respectively (see
<a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a>). The colvar called “<!--l. 95--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>c</mi></math>"
is the coordination number calculated between atoms 1 to 10 and atoms 11 to 20. A harmonic restraint (see <a href="#sec:colvarbias_harmonic">6.8<!--tex4ht:ref: sec:colvarbias_harmonic --></a>) is
applied to both <!--l. 95--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>d</mi></math> and
<!--l. 95--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>c</mi></math>: to allow using the
same force constant <!--l. 95--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>K</mi></math>,
both <!--l. 95--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>d</mi></math> and
<!--l. 95--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>c</mi></math> are scaled by their respective
fluctuation widths <!--l. 95--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>w</mi></mrow><mrow><mi>d</mi></mrow></msub></math> and
<!--l. 95--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>w</mi></mrow><mrow><mi>c</mi></mrow></msub></math>. A third colvar “alpha" is defined
as the <!--l. 96--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>α</mi></math>-helical content of residues
1 to 10 (see <a href="#sec:cvc_alpha">4.8.1<!--tex4ht:ref: sec:cvc_alpha --></a>). The values of “<!--l. 97--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>c</mi></math>"
and “alpha" are also recorded throughout the simulation as a joint 2-dimensional histogram (see
<a href="#sec:colvarbias_histogram">6.13<!--tex4ht:ref: sec:colvarbias_histogram --></a>).</p>
<div class="mdframed" id="mdframed-2">
<!--l. 99--><p class="noindent"></p><pre>colvar {<br class="newline"/>  # difference of two distances<br class="newline"/>  name d <br class="newline"/>  width 0.2  # estimated fluctuation width <br class="newline"/>  distance {<br class="newline"/>    componentCoeff  1.0<br class="newline"/>    group1 { atomNumbers 1 2 }<br class="newline"/>    group2 { atomNumbers 3 4 5 }<br class="newline"/>  }<br class="newline"/>  distance {<br class="newline"/>    componentCoeff -1.0<br class="newline"/>    group1 { atomNumbers 7 }<br class="newline"/>    group2 { atomNumbers 8 9 10 }<br class="newline"/>  }<br class="newline"/>}<br class="newline"/><br class="newline"/>colvar {<br class="newline"/>  name c<br class="newline"/>  coordNum {<br class="newline"/>    cutoff 6.0<br class="newline"/>    group1 { atomNumbersRange  1-10 }<br class="newline"/>    group2 { atomNumbersRange 11-20 }<br class="newline"/>    tolerance 1.0e-6<br class="newline"/>    pairListFrequency 1000<br class="newline"/>  }<br class="newline"/>}<br class="newline"/>colvar {<br class="newline"/>  name alpha<br class="newline"/>  alpha {<br class="newline"/>    psfSegID PROT<br class="newline"/>    residueRange 1-10<br class="newline"/>  }<br class="newline"/>} <br class="newline"/>harmonic {<br class="newline"/>  colvars d c<br class="newline"/>  centers 3.0 4.0<br class="newline"/>  forceConstant 5.0<br class="newline"/>}<br class="newline"/><br class="newline"/>histogram {<br class="newline"/>  colvars c alpha<br class="newline"/>} </pre><!--l. 144--><p class="noindent"></p>
</div>
<!--l. 371--><p class="indent">
</p>
<div class="flushright">
<!--l. 371--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 371--><p class="noindent"> </p><div class="heading"> <a id="sec:colvarmodule"></a>
<h3 class="sectionHead"><span class="titlemark">3    </span> <a id="sec:colvarmodule"></a>Enabling and controlling the Colvars module in NAMD</h3>
<!--l. 371--><p class="noindent"></p></div>
<!--l. 373--><p class="indent">   Here, we document the syntax of the commands and parameters used to set up and use
the Colvars module in NAMD <span class="cite">[<a href="#XPhillips2020">2</a>]</span>. One of these parameters is the configuration file or the
configuration text for the module itself, whose syntax is described in <a href="#sec:colvars_config_syntax">3.4<!--tex4ht:ref: sec:colvars_config_syntax --></a> and in the following
sections.
</p>
<div class="flushright">
<!--l. 376--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 376--><p class="noindent"> </p><div class="heading"> <a id="sec:units"></a>
<h4 class="subsectionHead"><span class="titlemark">3.1    </span> <a id="sec:units"></a>Units in the Colvars module</h4>
<!--l. 376--><p class="noindent"></p></div>
<!--l. 378--><p class="indent">   The “internal units" of the Colvars module are the units in which values are expressed in the
configuration file, and in which collective variable values, energies, etc. are expressed in the output and
colvars trajectory files. Generally <span class="phvb8t-x-x-109">the Colvars module uses internally the same units as its back-end</span>
<span class="phvb8t-x-x-109">MD engine, with the exception of VMD</span>, where different unit sets are supported to allow for
easy setup, visualization and analysis of Colvars simulations performed with any simulation
engine.
</p><!--l. 381--><p class="indent">   Note that <span class="phvb8t-x-x-109">angles </span>are expressed in degrees, and derived quantities such as force constants are based
on degrees as well. Some colvar components have default values, expressed in Ångström
(Å) in this documentation. They are converted to the current length unit, if different from Å.
Atomic coordinates read from <span class="phvb8t-x-x-109">XYZ files </span>(and PDB files where applicable) are expected to be
expressed in Ångström, no matter what unit system is in use by the back-end (NAMD) or the
Colvars Module. They are converted internally to the current length unit as needed. Note that
force constants in <span class="ectt-1095">harmonic </span>and <span class="ectt-1095">harmonicWalls </span>biases (<a href="#sec:colvarbias_harmonic">6.8<!--tex4ht:ref: sec:colvarbias_harmonic --></a>) are rescaled according to the
<a href="#colvar|width"><span class="ectt-1095">width</span></a> parameter of colvars, so that they are formally in energy units, although if <a href="#colvar|width"><span class="ectt-1095">width</span></a> is
given its default value of 1.0, force constants are effectively expressed in kcal/mol/<span class="phvro8t-x-x-109">(colvar</span>
<span class="phvro8t-x-x-109">unit)</span><!--l. 384--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow></mrow><mrow><mn>2</mn></mrow></msup></math>.
</p><!--l. 386--><p class="indent">   To avoid errors due to reading configuration files written in a different unit system, it can be specified
within the input:
</p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox2"><a id="dx1-7001"></a><span class="ectt-1095">units</span></span> — <span class="phvro8t-x-x-109">Unit system to be used</span><br class="newline"/><!--l. 410--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> string,
     context: global <!--l. 410--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>A string defining the units to be used internally by Colvars. In NAMD the only supported
     value is NAMD's native units: <span class="phvro8t-x-x-109">real </span>(Å, kcal/mol). </li></ul>
<div class="flushright">
<!--l. 455--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 455--><p class="noindent"> </p><div class="heading"> <a id="sec:colvars_mdengine_parameters"></a>
<h4 class="subsectionHead"><span class="titlemark">3.2    </span> <a id="sec:colvars_mdengine_parameters"></a>NAMD parameters</h4>
<!--l. 455--><p class="noindent"></p></div>
<!--l. 455--><p class="indent">   To enable a Colvars-based calculation, the <span class="ectt-1095">colvars on </span>command must be added to the
NAMD script. Two optional commands, <span class="ectt-1095">colvarsConfig </span>and <span class="ectt-1095">colvarsInput </span>can be used to
define the module's configuration or continue a previous simulation. Because these are static
parameters, it is typically more convenient to use the <span class="ectt-1095">cv </span>command in the rest of the NAMD
script.
</p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox3"><a id="dx1-8001"></a><span class="ectt-1095">colvars</span></span> — <span class="phvro8t-x-x-109">Enable the Colvars module</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 455--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: NAMD configuration file <!--l. 455--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If this flag is on, the Colvars module within NAMD is enabled.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox4"><a id="dx1-8002"></a><span class="ectt-1095">colvarsConfig</span></span> — <span class="phvro8t-x-x-109">Configuration file for the collective variables</span><br class="newline"/><!--l. 455--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> UNIX
     filename, context: NAMD configuration file <!--l. 455--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Name of the Colvars configuration file (<a href="#sec:colvars_config_syntax">3.4<!--tex4ht:ref: sec:colvars_config_syntax --></a>, <a href="#sec:colvars_global">3.5<!--tex4ht:ref: sec:colvars_global --></a> and following sections). This file can also
     be provided by the Tcl command <span class="ectt-1095">cv configfile</span>. Alternatively, the contents of the file (as
     opposed to the file itself) can be given as a string argument to the command <span class="ectt-1095">cv config</span>.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox5"><a id="dx1-8003"></a><span class="ectt-1095">colvarsInput</span></span> — <span class="phvro8t-x-x-109">Input state file for the collective variables</span><br class="newline"/><!--l. 455--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> UNIX
     filename, context: NAMD configuration file <!--l. 455--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Keyword used to specify the input state file's name (<a href="#sec:colvars_input">3.6<!--tex4ht:ref: sec:colvars_input --></a>). The file name is required to end
     in “<span class="ectt-1095">.colvars.state</span>": if it doesn't, Colvars will strip any further suffix following this string. If
     the input file is meant to be loaded within a Tcl script section, the <span class="ectt-1095">cv load </span>command may be
     used instead.
     </li></ul>
<div class="flushright">
<!--l. 636--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 636--><p class="noindent"> </p><div class="heading"> <a id="sec:cv_scripting"></a>
<h4 class="subsectionHead"><span class="titlemark">3.3    </span> <a id="sec:cv_scripting"></a>Using the scripting interface to control the Colvars module</h4>
<!--l. 636--><p class="noindent"></p></div>
<!--l. 638--><p class="indent">   After the first initialization of the Colvars module, the internal state of Colvars objects may be queried
or modified in a NAMD script:</p>
<div class="mdframed" id="mdframed-3">
<!--l. 639--><p class="noindent"></p><pre>cv <!--l. 639--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="phvro8t-x-x-109">method</span><!--l. 639--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math> arg1 arg2 ...
</pre><!--l. 639--><p class="noindent"></p>
</div>
<!--l. 640--><p class="noindent">where <!--l. 640--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ecit-1095">method</span><!--l. 640--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>
is the name of a specific procedure and <span class="ectt-1095">arg1</span>, <span class="ectt-1095">arg2</span>, …are its required and/or optional arguments.
</p><!--l. 675--><p class="indent">   In the remainder of this section, the most frequently used commands of the Colvars scripting interface
are discussed and exemplified. For a full list of scripting commands available, see section <a href="#sec:cvscript_cmdline">7<!--tex4ht:ref: sec:cvscript_cmdline --></a>.
<br class="newline"/>
</p>
<div class="flushright">
<!--l. 681--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 681--><p class="noindent"> </p><div class="heading"> <a id="sec:cv_command_init"></a>
<h5 class="subsubsectionHead"><span class="titlemark">3.3.1    </span> <a id="sec:cv_command_init"></a>Setting up the Colvars module</h5>
<!--l. 681--><p class="noindent"></p></div>
<!--l. 693--><p class="indent">   If the NAMD configuration parameter <span class="ectt-1095">colvars </span>is on, the <span class="ectt-1095">cv </span>Tcl command can be used
anywhere in the NAMD script, and will be invoked as soon as NAMD begins processing Tcl
commands.
</p><!--l. 695--><p class="indent">   To define new collective variables and/or biases for immediate use in the current session,
configuration can be loaded from an external configuration file:
                                                                                         
                                                                                         
</p>
<div class="mdframed" id="mdframed-4">
<!--l. 696--><p class="noindent"></p><pre>cv configfile "colvars-file.in" </pre><!--l. 696--><p class="noindent"></p>
</div>
<!--l. 697--><p class="indent">   This can in principle be called at any time, if only flags internal to Colvars are being modified. In
practice, when new atoms or any new atomic properties (e.g. total forces) are being requested,
initialization steps will be required that are not carried out during a simulation. Therefore, it is generally
good practice in a simulation to change the Colvars configuration outside the scope between segments of
the same computation.
</p><!--l. 702--><p class="indent">   To load the configuration directly from a string the “<span class="ectt-1095">config</span>" method may be used:</p>
<div class="mdframed" id="mdframed-5">
<!--l. 703--><p class="noindent"></p><pre>cv config "keyword { ... }" </pre><!--l. 703--><p class="noindent"></p>
</div>
<!--l. 704--><p class="noindent">This method is particularly useful to dynamically define the Colvars configuration within a NAMD script.
For example, when running an ensemble of umbrella sampling simulations in NAMD, it may be
convenient to use an identical script and define the specific window through environment
variables.
</p><!--l. 707--><p class="indent">   In the following example, the environment variable “<span class="ectt-1095">SLURM_ARRAY_TASK_ID</span>" is set in a Slurm array job
to increasing values starting at 0. The value of this variable is used to define the window's numeric index,
and the umbrella restraint center beginning at 2.00 for the first window, and increasing in increments of
0.25 for all other windows:</p>
<div class="mdframed" id="mdframed-6">
<!--l. 720--><p class="noindent"></p><pre>cv configfile colvar-definition.in<br class="newline"/>set window $env(SLURM_ARRAY_TASK_ID)<br class="newline"/>cv config "harmonic {<br class="newline"/>  name us_${window}<br class="newline"/>  colvars xi<br class="newline"/>  centers [expr 2.0 + 0.25 * ${window}]<br class="newline"/>  ...<br class="newline"/>}" </pre><!--l. 720--><p class="noindent"></p>
</div>
<div class="flushright">
<!--l. 738--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 738--><p class="noindent"> </p><div class="heading"> <a id="sec:cv_command_version"></a>
<h5 class="subsubsectionHead"><span class="titlemark">3.3.2    </span> <a id="sec:cv_command_version"></a>Using the Colvars version in scripts</h5>
<!--l. 738--><p class="noindent"></p></div>
<!--l. 740--><p class="indent">   The vast majority of the syntax in Colvars is backward-compatible, adding keywords when new
features are introduced. However, when using multiple versions simultaneously it may be useful to test
within the script whether the version is recent enough to support the desired feature. The “<span class="ectt-1095">version</span>" can
be used to get the Colvars version for this use:</p>
<div class="mdframed" id="mdframed-7">
<!--l. 743--><p class="noindent"></p><pre>if { [cv version] &gt;= "2020-02-25" } {<br class="newline"/>  cv config "(use a recent feature)"<br class="newline"/>} </pre><!--l. 747--><p class="noindent"></p>
</div>
<div class="flushright">
<!--l. 750--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 750--><p class="noindent"> </p><div class="heading"> <a id="sec:cv_command_loadsave"></a>
<h5 class="subsubsectionHead"><span class="titlemark">3.3.3    </span> <a id="sec:cv_command_loadsave"></a>Loading and saving the Colvars state and other information</h5>
<!--l. 750--><p class="noindent"></p></div>
<!--l. 752--><p class="indent">   After a configuration is fully defined, the “<span class="ectt-1095">load</span>" method may be used to load a state file from a
previous simulation that contains e.g. data from history-dependent biases), to either continue that
simulation or analyze its results:</p>
<div class="mdframed" id="mdframed-8">
<!--l. 753--><p class="noindent"></p><pre>cv load "<!--l. 753--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>oldjob<!--l. 753--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>.colvars.state"
</pre><!--l. 753--><p class="noindent"></p>
</div>
<!--l. 754--><p class="noindent">or more simply using the prefix of the state file itself:</p>
<div class="mdframed" id="mdframed-9">
<!--l. 755--><p class="noindent"></p><pre>cv load "<!--l. 755--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>oldjob<!--l. 755--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>"
</pre><!--l. 755--><p class="noindent"></p>
</div>
<!--l. 756--><p class="indent">   Anything following the “<span class="ectt-1095">.colvars.state</span>" suffix will be stripped.
</p><!--l. 764--><p class="noindent">The latter version is particularly convenient in combination with the NAMD <span class="ectt-1095">reinitatoms </span>command, for
example:<br class="newline"/></p>
<div class="mdframed" id="mdframed-10">
<!--l. 764--><p class="noindent"></p><pre>reinitatoms <!--l. 764--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>oldjob<!--l. 764--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><br class="newline"/>cv load <!--l. 764--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>oldjob<!--l. 764--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>
</pre><!--l. 764--><p class="noindent"></p>
</div>
<!--l. 764--><p class="indent">   The step number contained by the loaded file will be used internally by Colvars to control
time-dependent biases, unless <span class="ectt-1095">firstTimestep </span>is issued, in which case that value will be
used.
</p><!--l. 774--><p class="indent">   When the system's topology is changed during simulation via the <span class="ectt-1095">structure </span>command (e.g. in
constant-pH simulations), it is generally best to reset and re-initialize the module from scratch before
loading the corresponding snapshot:</p>
<div class="mdframed" id="mdframed-11">
<!--l. 774--><p class="noindent"></p><pre>structure newsystem.psf<br class="newline"/>reinitatoms <!--l. 774--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>snapshot<!--l. 774--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><br class="newline"/>cv reset<br class="newline"/>cv configfile <!--l. 774--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>new_config<!--l. 774--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><br class="newline"/>cv load <!--l. 774--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>snapshot<!--l. 774--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>
</pre><!--l. 774--><p class="noindent"></p>
</div>
<!--l. 776--><p class="indent">   The “<span class="ectt-1095">save</span>" method, analogous to “<span class="ectt-1095">load</span>", allows to save all restart information to a state file. This
is normally not required during a simulation if <a href="#Colvars-global|colvarsRestartFrequency"><span class="ectt-1095">colvarsRestartFrequency</span></a> is defined (either
directly or indirectly by the NAMD restart frequency). Because not only a state file (used to
continue simulations) but also other data files (used to analyze the trajectory) are written, it is
generally recommended to call the <span class="ectt-1095">save </span>method using a prefix, rather than a complete file
name:</p>
<div class="mdframed" id="mdframed-12">
<!--l. 779--><p class="noindent"></p><pre>cv save "<!--l. 779--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>job<!--l. 779--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>"
</pre><!--l. 779--><p class="noindent"></p>
</div>
<div class="flushright">
<!--l. 782--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 782--><p class="noindent"> </p><div class="heading"> <a id="sec:cv_command_atomic_data"></a>
<h5 class="subsubsectionHead"><span class="titlemark">3.3.4    </span> <a id="sec:cv_command_atomic_data"></a>Accessing atomic data</h5>
<!--l. 782--><p class="noindent"></p></div>
<!--l. 784--><p class="indent">   For computational efficiency the Colvars module keeps internal copies of the numeric IDs, masses,
charges, positions, and optionally total forces of the atoms requested for a Colvars computation. At each
simulation step, up-to-date versions of these properties are copied from the central memory of
NAMD into the internal memory of the Colvars module. In a post-processing workflow or
outside a simulation (e.g. when using VMD), this copy can be carried out as part of the <span class="ectt-1095">update</span>
method:</p>
<div class="mdframed" id="mdframed-13">
<!--l. 787--><p class="noindent"></p><pre>cv update   </pre><!--l. 787--><p class="noindent"></p>
</div>
<!--l. 788--><p class="noindent">which also performs the (re-)computation of all variables and biases defined in Colvars.
</p><!--l. 790--><p class="indent">   For example, the current sequence of numeric IDs of the atoms processed by Colvars can be obtained
as:</p>
<div class="mdframed" id="mdframed-14">
<!--l. 791--><p class="noindent"></p><pre>cv getatomids   </pre><!--l. 791--><p class="noindent"></p>
</div>
<!--l. 792--><p class="noindent">and their current positions as:</p>
<div class="mdframed" id="mdframed-15">
<!--l. 793--><p class="noindent"></p><pre>cv getatompositions   </pre><!--l. 793--><p class="noindent"></p>
</div>
<!--l. 794--><p class="noindent">This may prove useful to test the correctness of the coordinates passed to Colvars, particularly in regard
to periodic boundary conditions (<a href="#sec:colvar_atom_groups_wrapping">5.3<!--tex4ht:ref: sec:colvar_atom_groups_wrapping --></a>). There is currently no mechanism to modify the above fields via the
scripting interface, but such capability will be added in the future.
</p><!--l. 798--><p class="indent">   A special case are collective variables that are based on <span class="phvro8t-x-x-109">centers of mass</span>, or <span class="phvro8t-x-x-109">volumetric maps </span>(see
e.g. <a href="#colvar|mapTotal"><span class="ectt-1095">mapTotal</span></a>). These are often computed by NAMD itself, and any atoms that are used only for such
variables will not be included in the lists above.
</p><!--l. 800--><p class="indent">   While running a simulation, or when setting one up in VMD, it is possible to examine all the forces that
                                                                                         
                                                                                         
were last applied by Colvars to the atoms, or are about to be applied:</p>
<div class="mdframed" id="mdframed-16">
<!--l. 801--><p class="noindent"></p><pre>cv getatomappliedforces   </pre><!--l. 801--><p class="noindent"></p>
</div>
<!--l. 802--><p class="noindent">where the length and order of this sequence matches that provided by the <span class="ectt-1095">getatomids </span>method. A simpler
way of testing the stability of a Colvars configuration before or during a simulation makes use of
aggregated data, such as the energy:</p>
<div class="mdframed" id="mdframed-17">
<!--l. 804--><p class="noindent"></p><pre>cv getenergy   </pre><!--l. 804--><p class="noindent"></p>
</div>
<!--l. 805--><p class="noindent">the root-mean-square of the Colvars applied forces:</p>
<div class="mdframed" id="mdframed-18">
<!--l. 806--><p class="noindent"></p><pre>cv getatomappliedforcesrms   </pre><!--l. 806--><p class="noindent"></p>
</div>
<!--l. 807--><p class="noindent">or the maximum norm of the applied forces:</p>
<div class="mdframed" id="mdframed-19">
<!--l. 808--><p class="noindent"></p><pre>cv getatomappliedforcesmax   </pre><!--l. 808--><p class="noindent"></p>
</div>
<!--l. 809--><p class="noindent">which can be matched to a specific atom via its numeric ID obtained as:</p>
<div class="mdframed" id="mdframed-20">
<!--l. 810--><p class="noindent"></p><pre>cv getatomappliedforcesmaxid   </pre><!--l. 810--><p class="noindent"></p>
</div>
<!--l. 812--><p class="indent">   See <a href="#sec:cvscript_cmdline_cv">7.1<!--tex4ht:ref: sec:cvscript_cmdline_cv --></a> for a complete list of scripting commands used to manage atomic data and runtime
parameters of the Colvars module.
</p>
<div class="flushright">
<!--l. 841--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 841--><p class="noindent"> </p><div class="heading"> <a id="sec:cv_command_colvar"></a>
<h5 class="subsubsectionHead"><span class="titlemark">3.3.5    </span> <a id="sec:cv_command_colvar"></a>Managing collective variables</h5>
<!--l. 841--><p class="noindent"></p></div>
<!--l. 843--><p class="indent">   After one or more collective variables are defined, they can be accessed with the following
syntax.</p>
<div class="mdframed" id="mdframed-21">
<!--l. 844--><p class="noindent"></p><pre>cv colvar "<!--l. 844--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>name<!--l. 844--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>" <!--l. 844--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="phvro8t-x-x-109">method</span><!--l. 844--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math> arg1 arg2 ...
</pre><!--l. 844--><p class="noindent"></p>
</div>
<!--l. 845--><p class="noindent">where “<!--l. 845--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">name</span><!--l. 845--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>"
is the name of the variable.
</p><!--l. 847--><p class="indent">   For example, to recompute the collective variable “<span class="ectt-1095">xi</span>" after a change in its parameters, the following
command can be used:
                                                                                         
                                                                                         
</p>
<div class="mdframed" id="mdframed-22">
<!--l. 848--><p class="noindent"></p><pre>cv colvar "xi" update  </pre><!--l. 848--><p class="noindent"></p>
</div>
<!--l. 849--><p class="noindent">This ordinarily is not needed during a simulation run, where all variables are recomputed at every step
(along with biasing forces acting on them). However, when analyzing an existing trajectory, e.g. in VMD, a
call to <span class="ectt-1095">update </span>is generally required.
</p><!--l. 852--><p class="indent">   While in all typical cases all configuration of the variables is done with the “<span class="ectt-1095">config</span>" or “<span class="ectt-1095">configfile</span>"
methods, a limited set of changes can be enacted at runtime using:</p>
<div class="mdframed" id="mdframed-23">
<!--l. 853--><p class="noindent"></p><pre>cv colvar "<!--l. 853--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>name<!--l. 853--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>" modifycvcs arg1 arg2 ...
</pre><!--l. 853--><p class="noindent"></p>
</div>
<!--l. 854--><p class="noindent">where each argument is a string passed to the function or functions that are used to compute the variable,
and are called colvar components, or CVCs (<a href="#sec:cvc_list">4.1<!--tex4ht:ref: sec:cvc_list --></a>). For example, a variable “<span class="ectt-1095">DeltaZ</span>" made of a single
“<span class="ectt-1095">distanceZ</span>" component can be made periodic with a period equal to the unit cell dimension along the
<!--l. 855--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>Z</mi></math>-axis:</p>
<div class="mdframed" id="mdframed-24">
<!--l. 856--><p class="noindent"></p><pre>cv colvar "DeltaZ" modifycvcs "period ${Lz}" </pre><!--l. 856--><p class="noindent"></p>
</div>
<!--l. 857--><p class="noindent">Please note that this option is currently limited to changing the values of the polynomial superposition
parameters <a href="#sec:cvc_superp"><span class="ectt-1095">componentCoeff</span></a>, or of the <a href="#sec:cvc_superp"><span class="ectt-1095">componentExp</span></a> to update on the fly, of <a href="#sec:cvc_periodic"><span class="ectt-1095">period</span></a>, <a href="#sec:cvc_periodic"><span class="ectt-1095">wrapAround</span></a> or
<a href="#colvar|cvc|forceNoPBC"><span class="ectt-1095">forceNoPBC</span></a> options for components where it is relevant.
</p><!--l. 860--><p class="indent">   If the variable is computed using many components, it is possible to selectively turn some of them on
or off:</p>
<div class="mdframed" id="mdframed-25">
<!--l. 861--><p class="noindent"></p><pre>cv colvar "<!--l. 861--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>name<!--l. 861--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>" cvcflags <!--l. 861--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>flags<!--l. 861--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>
</pre><!--l. 861--><p class="noindent"></p>
</div>
<!--l. 862--><p class="noindent">where “<!--l. 862--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">flags</span><!--l. 862--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>"
is a list of 0/1 values, one per component. This is useful for example when script-based path collective
variables in Cartesian coordinates (<a href="#sec:apath_scripted">4.12.3<!--tex4ht:ref: sec:apath_scripted --></a>) are used, to minimize computational cost by disabling the
computation of terms that are very close to zero.
</p><!--l. 865--><p class="indent"> <span class="phvb8t-x-x-109">Important: </span>None of the changes enacted by the “<span class="ectt-1095">modifycvcs</span>" or “<span class="ectt-1095">cvcflags</span>" methods will be saved to
state files, and will be lost when restarting a simulation, deleting the corresponding collective variable, or
resetting the module with the  “<span class="ectt-1095">reset</span>" method.
</p>
<div class="flushright">
<!--l. 868--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 868--><p class="noindent"> </p><div class="heading"> <a id="sec:cv_command_colvar_force"></a>
<h5 class="subsubsectionHead"><span class="titlemark">3.3.6    </span> <a id="sec:cv_command_colvar_force"></a>Applying and analyzing forces on collective variables</h5>
<!--l. 868--><p class="noindent"></p></div>
<!--l. 870--><p class="indent">   As soon as a colvar “<span class="ectt-1095">xi</span>" and its associated biasing potentials are up to date (i.e. during a MD run, or
after the respective “<span class="ectt-1095">update</span>" methods have been called), the force applied onto the colvar is known and
may be accessed through the <span class="ectt-1095">getappliedforce </span>method:</p>
<div class="mdframed" id="mdframed-26">
<!--l. 871--><p class="noindent"></p><pre>cv colvar "xi" getappliedforce  </pre><!--l. 871--><p class="noindent"></p>
</div>
<!--l. 872--><p class="noindent">See also the use of the <a href="#colvar|outputAppliedForce"><span class="ectt-1095">outputAppliedForce</span></a> option to have this force be saved to file during a
simulation.
</p><!--l. 874--><p class="indent">   Aside from the biasing methods already implemented within Colvars (<a href="#sec:colvarbias">6<!--tex4ht:ref: sec:colvarbias --></a>) this force may be incremented
<span class="phvro8t-x-x-109">ad hoc</span>, for example as part of a custom restraint implemented by <a href="#Colvars-global|scriptedColvarForces"><span class="ectt-1095">scriptedColvarForces</span></a>:</p>
<div class="mdframed" id="mdframed-27">
<!--l. 875--><p class="noindent"></p><pre>cv colvar "xi" addforce <!--l. 875--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>force<!--l. 875--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>
</pre><!--l. 875--><p class="noindent"></p>
</div>
<!--l. 876--><p class="noindent">where “<!--l. 876--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">force</span><!--l. 876--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>"
is a scalar or a vector (depending on the type of variable “<span class="ectt-1095">xi</span>"). If <span class="ectt-1095">addforce </span>is used during a simulation
step, the force added by will be communicated to the corresponding atoms as soon as Colvars
communicates all forces to NAMD.
</p><!--l. 881--><p class="indent">   For certain types of variable, the force applied directly on a colvar may be combined with those acting
<span class="phvro8t-x-x-109">indirectly </span>on it via the interatomic force field, making up the <span class="phvro8t-x-x-109">total force</span>. When the <a href="#colvar|outputTotalForce"><span class="ectt-1095">outputTotalForce</span></a>
keyword is enabled, or when a biasing method that makes explicit use of the total force is enabled, the
total force may be obtained as:</p>
<div class="mdframed" id="mdframed-28">
<!--l. 883--><p class="noindent"></p><pre>cv colvar "xi" gettotalforce  </pre><!--l. 883--><p class="noindent"></p>
</div>
<!--l. 884--><p class="noindent">Note that not all types of variable support total-force computation, and the value of the total force may not
be available immediately within the same simulation step: see the documentation of <a href="#colvar|outputTotalForce"><span class="ectt-1095">outputTotalForce</span></a>
for more details.
</p><!--l. 886--><p class="indent">   See <a href="#sec:cvscript_cmdline_colvar">7.2<!--tex4ht:ref: sec:cvscript_cmdline_colvar --></a> for a complete list of scripting commands used to manage collective variables.
</p>
<div class="flushright">
<!--l. 889--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 889--><p class="noindent"> </p><div class="heading"> <a id="sec:cv_command_bias"></a>
<h5 class="subsubsectionHead"><span class="titlemark">3.3.7    </span> <a id="sec:cv_command_bias"></a>Managing collective variable biases</h5>
<!--l. 889--><p class="noindent"></p></div>
<!--l. 891--><p class="indent">   Because biases depend only upon data internal to the Colvars module (i.e. they do not need atomic
coordinates from NAMD), it is generally easy to create them or update their configuration at any time. For
example, given the most current value of the variable “<span class="ectt-1095">xi</span>", an already-defined harmonic restraint on it
named “<span class="ectt-1095">h_xi</span>" can be updated as:</p>
<div class="mdframed" id="mdframed-29">
<!--l. 893--><p class="noindent"></p><pre>cv bias "h_xi" update  </pre><!--l. 893--><p class="noindent"></p>
</div>
<!--l. 894--><p class="noindent">During a running simulation this step is not needed, because an automatic update of each bias is already
carried out.
</p><!--l. 919--><p class="indent">   Another circumstance when “<span class="ectt-1095">update</span>" may be called e.g. as part of the function invoked by
<a href="#Colvars-global|scriptedColvarForces"><span class="ectt-1095">scriptedColvarForces</span></a>, it is executed before any biasing forces are applied to the variables, thus allowing
to modify them. This use of “<span class="ectt-1095">update</span>" is often used e.g. in the definition of custom bias-exchange
algorithms as part of the NAMD script. Because a restraint is a relatively light-weight object,
the easiest way to change the configuration of an existing bias is deleting it and re-creating
it:</p>
<div class="mdframed" id="mdframed-30">
<!--l. 919--><p class="noindent"></p><pre><span class="phvb8t-x-x-109">#</span><span class="phvb8t-x-x-109"> Delete</span><span class="phvb8t-x-x-109"> the</span><span class="phvb8t-x-x-109"> restraint</span><span class="phvb8t-x-x-109"> "harmonic_xi"</span><br class="newline"/>cv bias harmonic_xi delete<br class="newline"/><span class="phvb8t-x-x-109">#</span><span class="phvb8t-x-x-109"> Re-define</span><span class="phvb8t-x-x-109"> it,</span><span class="phvb8t-x-x-109"> but</span><span class="phvb8t-x-x-109"> using</span><span class="phvb8t-x-x-109"> an</span><span class="phvb8t-x-x-109"> updated</span><span class="phvb8t-x-x-109"> restraint</span><span class="phvb8t-x-x-109"> center</span><br class="newline"/>cv config "harmonic {<br class="newline"/>  name harmonic_xi<br class="newline"/>  centers ${new_center}<br class="newline"/>  ...<br class="newline"/>}"<br class="newline"/><span class="phvb8t-x-x-109">#</span><span class="phvb8t-x-x-109"> Now</span><span class="phvb8t-x-x-109"> update</span><span class="phvb8t-x-x-109"> it</span><span class="phvb8t-x-x-109"> (based</span><span class="phvb8t-x-x-109"> on</span><span class="phvb8t-x-x-109"> the</span><span class="phvb8t-x-x-109"> current</span><span class="phvb8t-x-x-109"> value</span><span class="phvb8t-x-x-109"> of</span><span class="phvb8t-x-x-109"> "xi")</span><br class="newline"/>cv bias harmonic_xi update </pre><!--l. 919--><p class="noindent"></p>
</div>
<!--l. 919--><p class="indent">   It is also possible to make the change subject to a condition on the energy of the new
bias:</p>
<div class="mdframed" id="mdframed-31">
<!--l. 919--><p class="noindent"></p><pre>...<br class="newline"/>cv bias harmonic_xi update<br class="newline"/>if { [cv bias harmonic_xi energy] &lt; ${E_accept} } {<br class="newline"/>    ...<br class="newline"/>} </pre><!--l. 919--><p class="noindent"></p>
</div>
<div class="flushright">
<!--l. 922--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 922--><p class="noindent"> </p><div class="heading"> <a id="sec:cv_command_bias_loadsave"></a>
<h5 class="subsubsectionHead"><span class="titlemark">3.3.8    </span> <a id="sec:cv_command_bias_loadsave"></a>Loading and saving the state of individual biases</h5>
<!--l. 922--><p class="noindent"></p></div>
<!--l. 924--><p class="indent">   Some types of bias are history-dependent, and the magnitude of their forces depends not only on the
values of their corresponding variables, but also on previous simulation history. It is thus useful to
load information from a state file that contains information specifically for one bias only, for
example:</p>
<div class="mdframed" id="mdframed-32">
<!--l. 926--><p class="noindent"></p><pre>cv bias "metadynamics1" load "old.colvars.state" </pre><!--l. 926--><p class="noindent"></p>
</div>
<!--l. 927--><p class="noindent">or alternatively, using the prefix of the file instead of its full name:
                                                                                         
                                                                                         
</p>
<div class="mdframed" id="mdframed-33">
<!--l. 928--><p class="noindent"></p><pre>cv bias "metadynamics1" load "old" </pre><!--l. 928--><p class="noindent"></p>
</div>
<!--l. 929--><p class="indent">   A corresponding “<span class="ectt-1095">save</span>" function is also available:</p>
<div class="mdframed" id="mdframed-34">
<!--l. 930--><p class="noindent"></p><pre>cv bias "metadynamics1" save "new" </pre><!--l. 930--><p class="noindent"></p>
</div>
<!--l. 931--><p class="noindent">Please note that the file above must contain only the state information for that particular bias: <span class="phvro8t-x-x-109">loading a</span>
<span class="phvro8t-x-x-109">state file for the whole module is not allowed.</span>
</p><!--l. 933--><p class="indent">   This pair of functions is also used internally by Colvars to implement e.g. multiple-walker
metadynamics (<a href="#sec:colvarbias_meta_mr">6.6.7<!--tex4ht:ref: sec:colvarbias_meta_mr --></a>), but they can be called from a scripted function to implement alternative coupling
schemes.
</p><!--l. 935--><p class="indent">   See <a href="#sec:cvscript_cmdline_bias">7.3<!--tex4ht:ref: sec:cvscript_cmdline_bias --></a> for a complete list of scripting commands used to manage biases.
</p>
<div class="flushright">
<!--l. 940--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 940--><p class="noindent"> </p><div class="heading"> <a id="sec:colvars_config_syntax"></a>
<h4 class="subsectionHead"><span class="titlemark">3.4    </span> <a id="sec:colvars_config_syntax"></a>Configuration syntax used by the Colvars module</h4>
<!--l. 940--><p class="noindent"></p></div>
<!--l. 942--><p class="indent">   Configuration for the Colvars module is passed using an external file, or inlined as a string in a NAMD
script using the Tcl command <span class="ectt-1095">cv config "..."</span>. Configuration lines follow the format “<span class="ectt-1095">keyword value</span>" or
“<span class="ectt-1095">keyword { ... }</span>", where the keyword and its value must be separated by one or more space
characters. The following formatting rules apply:
</p>
<ul class="itemize1">
<li class="itemize"><span class="phvro8t-x-x-109">Keywords are case-insensitive</span>; for example, <span class="ectt-1095">upperBoundary </span>is the same as <span class="ectt-1095">upperboundary</span>
     and <span class="ectt-1095">UPPERBOUNDARY</span>); note that their string values are however still case-sensitive (e.g. names
     of variables, file names).
     </li>
<li class="itemize">A long value, or a list of multiple values, can be distributed across multiple lines by using
     <span class="phvro8t-x-x-109">curly braces</span>, “<span class="ectt-1095">{</span>" and “<span class="ectt-1095">}</span>": the opening brace “<span class="ectt-1095">{</span>" must occur on the same line as the keyword,
     following at least one space character; the closing brace “<span class="ectt-1095">}</span>" may be at any position after that;
     any keywords following the closing brace on the same line are not valid (they should appear
     instead on a different line).
     </li>
<li class="itemize">Many keywords are nested, and are only meaningful within the specific context of another
     keyword; for example, the keyword <span class="ectt-1095">name </span>is available inside the block of the keyword <span class="ectt-1095">colvar</span>
<span class="ectt-1095">{...}</span>,  but  not  outside  of  it;  for  every  keyword  documented  in  the  following,  the  <span class="phvro8t-x-x-109">“parent"</span>
<span class="phvro8t-x-x-109">keyword </span>that defines such context is also indicated.
                                                                                         
                                                                                         
     </li>
<li class="itemize">If a keyword requiring a boolean value (<span class="ectt-1095">yes|on|true </span>or <span class="ectt-1095">no|off|false</span>) is provided without an
     explicit value, it defaults to ‘<span class="ectt-1095">yes|on|true</span>'; for example, ‘<span class="ectt-1095">outputAppliedForce</span>' may be used
     as shorthand for ‘<span class="ectt-1095">outputAppliedForce on</span>'.
     </li>
<li class="itemize">The hash character “<span class="ectt-1095">#</span>" indicates a <span class="phvro8t-x-x-109">comment</span>: all text in the same line following this character
     will be ignored.
     </li>
<li class="itemize">Outside of comments, <span class="phvro8t-x-x-109">only ASCII characters </span>are allowed for defining keywords, and the only
     white-space characters supported are spaces, tabs and newlines: a warning will be printed
     upon detection of non-ASCII characters in a configuration line, which include both characters
     that are visibly “special", as well as those with a very similar appearance to ASCII ones (for
     instance, <a href="https://en.wikipedia.org/wiki/Non-breaking_space">non-breaking spaces</a>); common ways to identify/remove non-ASCII characters are
     using the Emacs text editor, or using <span class="ectt-1095">LC_ALL=C vi</span>.
     </li>
<li class="itemize">Tcl syntax is generally not available inside a Colvars configuration file/string, but it is possible
     to use Tcl variables or bracket expansion of commands when configuration is passed via
     the  command  <span class="ectt-1095">cv config "..."</span>;  for  example,  this  is  particularly  useful  when  combined
     with NAMD's parameter introspection, e.g. <span class="ectt-1095">cv config "colvarsTrajFrequency [DCDFreq]"</span>
     allows synchronizing the output frequencies of NAMD and Colvars trajectory files. .
     </li></ul>
<div class="flushright">
<!--l. 967--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 967--><p class="noindent"> </p><div class="heading"> <a id="sec:colvars_global"></a>
<h4 class="subsectionHead"><span class="titlemark">3.5    </span> <a id="sec:colvars_global"></a>Global keywords</h4>
<!--l. 967--><p class="noindent"></p></div>
<!--l. 969--><p class="indent">   The following keywords are available in the global context of the Colvars configuration, i.e. they are
not nested inside other keywords: </p>
<ul class="itemize1">
<li class="itemize"><a id="Colvars-global|colvarsTrajFrequency"></a> <span class="colorbox" id="colorbox6"><a id="dx1-19001"></a><span class="ectt-1095">colvarsTrajFrequency</span></span> — <span class="phvro8t-x-x-109">Colvar value trajectory frequency</span><br class="newline"/>Default: 100 <!--l. 982--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     integer, context: global <!--l. 982--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>The values of each colvar (and of other related quantities, if requested) are written to the file
     <span class="phvro8t-x-x-109">outputName</span><span class="ectt-1095">.colvars.traj </span>every these many steps throughout the simulation. If the value is
     0, such trajectory file is not written. For optimization the output is buffered, and synchronized
     with the disk only when the restart file is being written.
                                                                                         
                                                                                         
     </li>
<li class="itemize"><a id="Colvars-global|colvarsRestartFrequency"></a> <span class="colorbox" id="colorbox7"><a id="dx1-19002"></a><span class="ectt-1095">colvarsRestartFrequency</span></span> — <span class="phvro8t-x-x-109">Colvar module restart frequency</span><br class="newline"/>Default: NAMD parameter <span class="ectt-1095">restartFreq</span> <!--l. 997--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     integer, context: global <!--l. 997--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>When this value is non-zero, a state file suitable for restarting will be written every these many
     steps. Additionally, any other output files produced by Colvars are written as well (except the
     trajectory file, which is written every <a href="#Colvars-global|colvarsTrajFrequency"><span class="ectt-1095">colvarsTrajFrequency</span></a> steps).   It is generally a good
     idea to leave this parameter at its default value, unless needed for special cases or to disable
     automatic writing of output files altogether.   If neeed, writing of all output files can still be
     triggered at any time via the <span class="ectt-1095">save </span>scripting function.
     </li>
<li class="itemize"><a id="Colvars-global|indexFile"></a> <span class="colorbox" id="colorbox8"><a id="dx1-19003"></a><span class="ectt-1095">indexFile</span></span> — <span class="phvro8t-x-x-109">Index file for atom selection (GROMACS “ndx" format)</span><br class="newline"/><!--l. 1016--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> UNIX
     filename, context: global <!--l. 1016--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option reads an index file (usually with a <span class="ectt-1095">.ndx </span>extension) as produced by the <span class="ectt-1095">make_ndx</span>
     tool  of  GROMACS.  This  keyword  may  be  repeated  to  load  multiple  index  files.  A  group
     with the same name may appear multiple times, as long as it contains the same indices in
     identical order each time: an error is raised otherwise.  The names of index groups contained
     in  this  file  can  then  be  used  to  define  atom  groups  with  the  <span class="ectt-1095">indexGroup </span>keyword.  Other
     supported methods to select atoms are described in <a href="#sec:colvar_atom_groups">5<!--tex4ht:ref: sec:colvar_atom_groups --></a>.
     </li>
<li class="itemize"><a id="Colvars-global|smp"></a> <span class="colorbox" id="colorbox9"><a id="dx1-19004"></a><span class="ectt-1095">smp</span></span> — <span class="phvro8t-x-x-109">Whether SMP parallelism should be used</span><br class="newline"/>Default: <span class="ectt-1095">on</span> <!--l. 1026--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: global <!--l. 1026--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If this flag is enabled (default), SMP parallelism over threads will be used to compute variables
     and biases, provided that this is supported by the NAMD build in use.
     </li>
<li class="itemize"><a id="Colvars-global|sourceTclFile"></a> <span class="colorbox" id="colorbox10"><a id="dx1-19005"></a><span class="ectt-1095">sourceTclFile</span></span> — <span class="phvro8t-x-x-109">Tcl file to be sourced at startup</span><br class="newline"/><!--l. 1036--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> UNIX
     filename, context: global <!--l. 1036--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option causes the provided Tcl script file to be loaded at startup, e.g. to initialize the Tcl
     callback procedures for scripted variables or biases.
     </li></ul>
<div class="flushright">
<!--l. 1042--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1042--><p class="noindent"> </p><div class="heading"> <a id="sec:colvars_input"></a>
<h4 class="subsectionHead"><span class="titlemark">3.6    </span> <a id="sec:colvars_input"></a>Input state file</h4>
<!--l. 1042--><p class="noindent"></p></div>
<!--l. 1044--><p class="indent">   Several of the sampling methods implemented in Colvars are time- or history-dependent, i.e. they
work by accumulating data as a simulation progresses, and use these data to determine their biasing
forces. If the simulation engine uses a checkpoint or restart file (as GROMACS and LAMMPS do), any
data needed by Colvars are embedded into that file. Otherwise, a dedicated <span class="phvro8t-x-x-109">state file </span>can be loaded into
Colvars directly.
</p><!--l. 1046--><p class="indent">   When a dedicated Colvars state file is used, it may be in either one of two formats: </p>
<ul class="itemize1">
<li class="itemize">Formatted, i.e. <span class="phvro8t-x-x-109">“text" </span>format, which takes more space and is slower to to load/save but is also
     portable across different platforms and even different simulation engines (save for changes
     in physical units). This format is used by default, unless explicitly requested otherwise.
     </li>
<li class="itemize">Unformatted,  i.e. <span class="phvro8t-x-x-109">“binary"  </span>format,  which  is  both  space-efficient  and  quick  to  load/save,
     but  requires  that  the  same  NAMD  build  was  used  to  write  the  file  and  that  the  Colvars
     configuration  remains  the  same.  This  format  is  supported  by  Colvars  versions  starting
     2023-09-25. Colvars state files can be written in binary format by setting the environment
     variable “<span class="ectt-1095">COLVARS_BINARY_RESTART</span>" to 1.
     </li></ul>
<!--l. 1058--><p class="indent">   In either format, the state file contains accumulated data as well as the step number at the end of the
run. The step number read from a state file overrides any value that NAMD provides, and will be
incremented if the simulation proceeds. This means that the step number used internally
by Colvars may not always match the step number reported by NAMD. This is particularly
inmportant in NAMD, which represents step numbers as a 32-bit integers that overflows after
<!--l. 1061--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">∼</mo></math> 2 billion
steps, effectively negating the usefulness of the <span class="ectt-1095">firstTimeStep </span>keyword. However, step numbers are
implemented correctly in the Colvars state file.
</p>
<div class="flushright">
<!--l. 1096--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1096--><p class="noindent"> </p><div class="heading"> <a></a>
<h5 class="subsubsectionHead"><span class="titlemark">3.6.1    </span> <a id=""></a>Restarting in NAMD.</h5>
<!--l. 1096--><p class="noindent"></p></div> Before the Colvars module is initialized in NAMD, the <span class="ectt-1095">colvarsInput </span>keyword can be used to give the
name of a state file. After initialization of the Colvars module, a state file may be loaded at any time with
the Tcl command <span class="ectt-1095">cv load</span>. Both versions support loading Colvars state files in either format (binary or
text).

                                                                                         
                                                                                         
                                                                                         <div class="flushright">
<!--l. 1099--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1099--><p class="noindent"> </p><div class="heading"> <a id="sec:change_config_restart"></a>
<h5 class="subsubsectionHead"><span class="titlemark">3.6.2    </span> <a id="sec:change_config_restart"></a>Changing configuration upon restarting.</h5>
<!--l. 1099--><p class="noindent"></p></div>
<!--l. 1101--><p class="indent">   In some cases, it is useful to modify the configuration of variables or biases between consecutive runs,
for example by adding or removing a restraint. Some special provisions will happen in that case. When a
state file is loaded, no information is available about any newly added variable or bias, which will
thus remain uninitialized until the first compute step. Conversely, any information that the
state file may contain about variables or biases that are no longer defined will be silently
ignored. Please note that these checks are performed based only on the <span class="phvro8t-x-x-109">names </span>of variables and
biases: it is your responsibility to ensure that these names have <span class="phvro8t-x-x-109">consistent definitions between</span>
<span class="phvro8t-x-x-109">runs.</span>
</p><!--l. 1107--><p class="indent">   The flexibility just described carries some limitations: namely, it is only supported when reading
<span class="phvro8t-x-x-109">text-format </span>Colvars state files. Instead, restarting from binary files after a configuration change will trigger
an error. It is also important to remind that when switching to a different build of NAMD, the binary format
may change slightly, even if the release version is the same.
</p><!--l. 1111--><p class="indent">   To work around the potential issues just described, a text-format Colvars state file should be loaded.
This is the default in NAMD unless the “<span class="ectt-1095">COLVARS_BINARY_RESTART</span>" is set to 1, and this information is only
provided here for troubleshooting purposes.
</p>
<div class="flushright">
<!--l. 1131--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1131--><p class="noindent"> </p><div class="heading"> <a id="sec:colvars_output"></a>
<h4 class="subsectionHead"><span class="titlemark">3.7    </span> <a id="sec:colvars_output"></a>Output files</h4>
<!--l. 1131--><p class="noindent"></p></div>
<!--l. 1133--><p class="indent">   When the output prefix <span class="phvro8t-x-x-109">outputName </span>is defined, the following output files are written during a simulation
run:
</p>
<ul class="itemize1">
<li class="itemize">A <span class="phvro8t-x-x-109">state file</span>, named <span class="phvro8t-x-x-109">outputName</span><span class="ectt-1095">.colvars.state</span>, which is written at the end of the specified
     run, and can also be written at any time with the scripting command <span class="ectt-1095">save </span>(<a href="#sec:cv_command_loadsave">3.3.3<!--tex4ht:ref: sec:cv_command_loadsave --></a>). This file is in
     plain text format by default, regardless of the value of <span class="ectt-1095">binaryOutput </span>of the NAMD coordinate
     and velocity files, or in binary format if the environment variable <span class="ectt-1095">COLVARS_BINARY_RESTART</span>
     is set to a non-zero integer. The state file can be used to continue a simulation: unless its
                                                                                         
                                                                                         
     contents are embedded in the checkpoint file of the MD engine itself (currently, GROMACS
     and LAMMPS support this), instructions for loading the Colvars state file will be required in
     the simulation script (see <a href="#sec:colvars_input">3.6<!--tex4ht:ref: sec:colvars_input --></a>).
     </li>
<li class="itemize">If the parameter <a href="#Colvars-global|colvarsRestartFrequency"><span class="ectt-1095">colvarsRestartFrequency</span></a> is larger than zero and the <span class="phvro8t-x-x-109">restartName </span>prefix is
     defined, a <span class="phvro8t-x-x-109">restart file </span>is written every that many steps: this file is fully equivalent to the final
     state file. The name of this file is <span class="phvro8t-x-x-109">restartName</span><span class="ectt-1095">.colvars.state</span>.
     </li>
<li class="itemize">If  the  parameter  <a href="#Colvars-global|colvarsTrajFrequency"><span class="ectt-1095">colvarsTrajFrequency</span></a> is  greater  than  0  (default  value:  100  steps),  a
     <span class="phvro8t-x-x-109">trajectory file</span>, named <span class="phvro8t-x-x-109">outputName</span><span class="ectt-1095">.colvars.traj</span>, is written during the simulation. Unlike a
     state file, this file is not needed to restart a simulation, but can be used for post-processing
     and analysis. The format of this file is described in sec. <a href="#sec:colvars_traj_format">3.8.6<!--tex4ht:ref: sec:colvars_traj_format --></a>.
     </li>
<li class="itemize">Additionally, certain features, when enabled, can emit output files with a specific purpose:
     for example, potentials of mean force (PMFs) can be written to file to be analyzed or plotted.
     These  files  are  described  in  the  respective  sections,  but  as  a  general  rule  they  all  use
     names beginning with the <span class="phvro8t-x-x-109">outputName </span>prefix. Like the trajectory file, these additional files
     are needed only for analyzing a simulation's results, but not to continue it.
     </li></ul>
<!--l. 1151--><p class="indent">   Lastly, the total energy of all biases or restraints applied to the colvars appears under the NAMD
standard output, under the MISC column.
</p>
<div class="flushright">
<!--l. 1154--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1154--><p class="noindent"> </p><div class="heading"> <a id="sec:colvars_file_formats"></a>
<h4 class="subsectionHead"><span class="titlemark">3.8    </span> <a id="sec:colvars_file_formats"></a>File formats</h4>
<!--l. 1154--><p class="noindent"></p></div>
<!--l. 1156--><p class="indent">   This section summarizes the file formats of various files that Colvars may be reading or
writing.
</p>
<div class="flushright">
<!--l. 1158--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1158--><p class="noindent"> </p><div class="heading"> <a id="sec:colvars_text_format"></a>
<h5 class="subsubsectionHead"><span class="titlemark">3.8.1    </span> <a id="sec:colvars_text_format"></a>Configuration and state files.</h5>
<!--l. 1158--><p class="noindent"></p></div>
<!--l. 1160--><p class="indent"> <span class="phvro8t-x-x-109">Configuration files </span>are text files that are generally read as input by NAMD, and may be optionally
inlined in a NAMD script (see <a href="#sec:cv_command_init">3.3.1<!--tex4ht:ref: sec:cv_command_init --></a>). Starting from version 2017-02-01, changes in newline encodings are
handled transparently, i.e. it is possible to typeset a configuration file in Windows (CR-LF newlines) and
then use it with Linux or macOS (LF-only newlines).
</p><!--l. 1163--><p class="indent"> <span class="phvro8t-x-x-109">Formatted state files</span>, although not written manually, follow otherwise the same text format as
configuration files. <span class="ectt-1095">Binary state files </span>can only be read by the Colvars code itself.
</p>
<div class="flushright">
<!--l. 1166--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1166--><p class="noindent"> </p><div class="heading"> <a id="sec:colvars_ndx_format"></a>
<h5 class="subsubsectionHead"><span class="titlemark">3.8.2    </span> <a id="sec:colvars_ndx_format"></a>Index (NDX) files</h5>
<!--l. 1166--><p class="noindent"></p></div>
<!--l. 1168--><p class="indent">   For atom selections that cannot be specified only by using internal Colvars keywords, external <span class="phvro8t-x-x-109">index</span>
<span class="phvro8t-x-x-109">files </span>may also be used following the <a href="http://manual.gromacs.org/documentation/current/reference-manual/file-formats.html#ndx">NDX format</a> used in GROMACS:</p>
<div class="mdframed" id="mdframed-35">
<!--l. 1169--><p class="noindent"></p><pre>[ group_1_name ]<br class="newline"/>  i1  i2  i3  i4  ...<br class="newline"/>  ...             ...  iN<br class="newline"/>[ group_2_name ]<br class="newline"/>  ... </pre><!--l. 1175--><p class="noindent"></p>
</div>
<!--l. 1176--><p class="noindent">where <span class="phvro8t-x-x-109">i1 </span>through <span class="phvro8t-x-x-109">iN </span>are 1-based indices. Each group name may not contain spaces or tabs: otherwise, a
parsing error will be raised.
</p><!--l. 1178--><p class="indent">   Multiple index files may be provided to Colvars, each using the keyword <a href="#Colvars-global|indexFile"><span class="ectt-1095">indexFile</span></a>. Two index files
may contain groups with the same names, however these must also represent identical atom selections,
i.e. the same sequence of indices including order.
</p><!--l. 1182--><p class="indent">   Other than with GROMACS, an index group may also be generated from the VMD command-line
interface, using the helper function <span class="ectt-1095">write_index_group </span>provided in the <span class="ectt-1095">colvartools </span>folder:</p>
<div class="mdframed" id="mdframed-36">
<!--l. 1183--><p class="noindent"></p><pre>source colvartools/write_index_group.tcl<br class="newline"/>set sel [atomselect top "resname XXX and not hydrogen"]<br class="newline"/>write_index_group indexfile.ndx $sel "Ligand" </pre><!--l. 1187--><p class="noindent"></p>
</div>
<div class="flushright">
<!--l. 1191--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1191--><p class="noindent"> </p><div class="heading"> <a id="sec:colvars_xyz_format"></a>
<h5 class="subsubsectionHead"><span class="titlemark">3.8.3    </span> <a id="sec:colvars_xyz_format"></a>XYZ coordinate files</h5>
<!--l. 1191--><p class="noindent"></p></div> XYZ coordinate files are text files with the extension “<span class="ectt-1095">.xyz</span>". They are read by the Colvars module using
an internal reader, and expect the following format:
<!--l. 1194--><p class="noindent"><!--tex4ht:inline--></p><div class="tabular"> <table class="tabular" id="TBL-1"><colgroup id="TBL-1-1g"><col id="TBL-1-1"/><col id="TBL-1-2"/><col id="TBL-1-3"/><col id="TBL-1-4"/></colgroup><tr id="TBL-1-1-" style="vertical-align:baseline;"><td class="td11" id="TBL-1-1-1" style="text-align:right; white-space:nowrap;"> <!--l. 1195--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>N</mi></math> </td>
</tr><tr id="TBL-1-2-" style="vertical-align:baseline;"><td class="td11" id="TBL-1-2-1" style="text-align:right; white-space:nowrap;"> </td><td class="td11" id="TBL-1-2-2" style="text-align:right; white-space:nowrap;"> <span class="phvro8t-x-x-109">Comment  </span></td><td class="td11" id="TBL-1-2-3" style="text-align:right; white-space:nowrap;"> <span class="phvro8t-x-x-109">line  </span></td><td class="td11" id="TBL-1-2-4" style="text-align:right; white-space:nowrap;"> </td></tr><tr id="TBL-1-3-" style="vertical-align:baseline;"><td class="td11" id="TBL-1-3-1" style="text-align:right; white-space:nowrap;"> <!--l. 1197--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>E</mi></mrow><mrow>
<mn>1</mn></mrow></msub></math> </td><td class="td11" id="TBL-1-3-2" style="text-align:right; white-space:nowrap;"> <!--l. 1197--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>x</mi></mrow><mrow><mn>1</mn></mrow></msub></math> </td><td class="td11" id="TBL-1-3-3" style="text-align:right; white-space:nowrap;"> <!--l. 1197--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>y</mi></mrow><mrow><mn>1</mn></mrow></msub></math> </td><td class="td11" id="TBL-1-3-4" style="text-align:right; white-space:nowrap;"> <!--l. 1197--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>z</mi></mrow><mrow><mn>1</mn></mrow></msub></math> </td>
</tr><tr id="TBL-1-4-" style="vertical-align:baseline;"><td class="td11" id="TBL-1-4-1" style="text-align:right; white-space:nowrap;"> <!--l. 1198--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>E</mi></mrow><mrow><mn>2</mn></mrow></msub></math> </td><td class="td11" id="TBL-1-4-2" style="text-align:right; white-space:nowrap;"> <!--l. 1198--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>x</mi></mrow><mrow><mn>2</mn></mrow></msub></math> </td><td class="td11" id="TBL-1-4-3" style="text-align:right; white-space:nowrap;"> <!--l. 1198--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>y</mi></mrow><mrow><mn>2</mn></mrow></msub></math> </td><td class="td11" id="TBL-1-4-4" style="text-align:right; white-space:nowrap;"> <!--l. 1198--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>z</mi></mrow><mrow><mn>2</mn></mrow></msub></math> </td>
</tr><tr id="TBL-1-5-" style="vertical-align:baseline;"><td class="td11" id="TBL-1-5-1" style="text-align:right; white-space:nowrap;">                                                                                    …  </td>
</tr><tr id="TBL-1-6-" style="vertical-align:baseline;"><td class="td11" id="TBL-1-6-1" style="text-align:right; white-space:nowrap;"> <!--l. 1200--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>E</mi></mrow><mrow><mi>N</mi></mrow></msub></math> </td><td class="td11" id="TBL-1-6-2" style="text-align:right; white-space:nowrap;"> <!--l. 1200--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>x</mi></mrow><mrow><mi>N</mi></mrow></msub></math> </td><td class="td11" id="TBL-1-6-3" style="text-align:right; white-space:nowrap;"> <!--l. 1200--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>y</mi></mrow><mrow><mi>N</mi></mrow></msub></math> </td><td class="td11" id="TBL-1-6-4" style="text-align:right; white-space:nowrap;"> <!--l. 1200--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>z</mi></mrow><mrow><mi>N</mi></mrow></msub></math> </td>
</tr><tr id="TBL-1-7-" style="vertical-align:baseline;"><td class="td11" id="TBL-1-7-1" style="text-align:right; white-space:nowrap;"> </td></tr></table>
</div>
<!--l. 1203--><p class="noindent">where <!--l. 1203--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>N</mi></math> is the number of atomic
coordinates in the file and <!--l. 1203--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>E</mi></mrow><mrow><mi>i</mi></mrow></msub></math> is
the chemical element of the <!--l. 1203--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math>-th
atom. Because <!--l. 1204--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>E</mi></mrow><mrow><mi>i</mi></mrow></msub></math>
is not used in Colvars, any string that does not contain tabs or spaces is acceptable.
</p><!--l. 1206--><p class="indent"> <span class="phvb8t-x-x-109">Note: </span>all XYZ coordinates are assumed to be expressed in Å units.
</p><!--l. 1208--><p class="indent">   An XYZ file may contain either one of the following scenarios:
     </p><ol class="enumerate1">
<li class="enumerate" id="x1-27001x3.8.3">The  file  contains  as  many  coordinates  as  the  atoms  that  they  are  being  read  for:  all
     coordinates will be read from the file following the same order as the atoms appear in the
     selection generated using the keywords listed in section <a href="#sec:colvar_atom_groups">5<!--tex4ht:ref: sec:colvar_atom_groups --></a>. (Note that the order is guaranteed
     only if a single type of selection keyword is used one or more times, and not guaranteed
     when different types of selection keywords are used.)
     </li>
<li class="enumerate" id="x1-27002x3.8.3">The file contains more coordinates than needed, and it is assumed to contain coordinates
     for the entire system: only coordinates that match the numeric indices of the selected atoms
     are read, in order of increasing number.</li></ol>
<!--l. 1215--><p class="indent">   XYZ-file coordinates are read directly by Colvars and stored internally as double-precision floating
point numbers.
</p>
<div class="flushright">
<!--l. 1244--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1244--><p class="noindent"> </p><div class="heading"> <a id="sec:colvars_pdb_format"></a>
<h5 class="subsubsectionHead"><span class="titlemark">3.8.4    </span> <a id="sec:colvars_pdb_format"></a>PDB coordinate files</h5>
<!--l. 1244--><p class="noindent"></p></div> PDB coordinate files are read by the Colvars module using existing functionality in NAMD, and therefore
follow the same format. The values of the atomic coordinates and other fields, such as occupancy or
temperature factors, are then communicated to Colvars by NAMD.
                                                                                         
                                                                                         
<!--l. 1244--><p class="indent">   PDB files may be used either as one of the available mechanisms to select atoms (see the <a href="#atom-group|atomsFile"><span class="ectt-1095">atomsFile</span></a>
keyword), or more frequently to read reference coordinates for least-squares fit alignment (see the
<a href="#atom-group|refPositionsFile"><span class="ectt-1095">refPositionsFile</span></a> keyword).
</p><!--l. 1244--><p class="indent">   To select atoms via the <a href="#atom-group|atomsFile"><span class="ectt-1095">atomsFile</span></a> keyword, the option <a href="#atom-group|atomsCol"><span class="ectt-1095">atomsFile</span></a> is required, and atoms are selected
based on either one of the following cases.
     </p><ol class="enumerate1">
<li class="enumerate" id="x1-28001x3.8.4">All atoms with a non-zero value of the corresponding column are selected.
     </li>
<li class="enumerate" id="x1-28002x3.8.4">If  and  only  if  <a href="#atom-group|atomsColValue"><span class="ectt-1095">atomsColValue</span></a> is  provided,  only  atoms  with  a  value  of  <span class="ectt-1095">atomsCol </span>equaling
     <span class="ectt-1095">atomsColValue </span>are selected. This can be useful to reuse the same PDB file for generating
     multiple selections.</li></ol>
<!--l. 1244--><p class="indent">   To read coordinates via the <a href="#atom-group|refPositionsFile"><span class="ectt-1095">refPositionsFile</span></a> keyword, there are four possible scenarios.
     </p><ol class="enumerate1">
<li class="enumerate" id="x1-28003x3.8.4">The  file  contains  as  many  coordinates  as  the  atoms  that  they  are  being  read  for:  all
     coordinates will be read from the file following the same order as the atoms appear in the
     selection generated using the keywords listed in section <a href="#sec:colvar_atom_groups">5<!--tex4ht:ref: sec:colvar_atom_groups --></a>. (Note that the order is guaranteed
     only if a single type of selection keyword is used one or more times, and not guaranteed
     when different types of selection keywords are used.)
     </li>
<li class="enumerate" id="x1-28004x3.8.4">The file contains more coordinates than needed, and it is assumed to contain coordinates
     for the entire system: only coordinates that match the numeric indices of the selected atoms
     are read, in order of increasing number.
     </li>
<li class="enumerate" id="x1-28005x3.8.4">If the corresponding <a href="#atom-group|refPositionsCol"><span class="ectt-1095">refPositionsCol</span></a> keyword is specified, only positions with a non-zero
     value  of  the  column  <span class="ectt-1095">refPositionsCol </span>are  read.  This  is  particularly  useful  when  all
     atoms  were  selected  via  <a href="#atom-group|atomsFile"><span class="ectt-1095">atomsFile</span></a> and  <a href="#atom-group|atomsCol"><span class="ectt-1095">atomsCol</span></a>,  which  can  then  be  the  same  as
     <span class="ectt-1095">refPositionsCol </span>for loading coordinates.
     </li>
<li class="enumerate" id="x1-28006x3.8.4">This    is    a    special    case    of    the    previous    point:    if    both    <a href="#atom-group|refPositionsCol"><span class="ectt-1095">refPositionsCol</span></a>  and
     <a href="#atom-group|refPositionsColValue"><span class="ectt-1095">refPositionsColValue</span></a> are specified, only atoms for which the <span class="ectt-1095">refPositionsCol </span>column as
     the value <span class="ectt-1095">refPositionsColValue </span>are read.</li></ol>
<!--l. 1244--><p class="indent">   Due to the fixed-precision PDB format, it is not recommended to use PDB files to read
coordinates when precision is a concern, and the XYZ format (see <a href="#sec:colvars_ndx_format">3.8.2<!--tex4ht:ref: sec:colvars_ndx_format --></a>) is recommended
instead.
</p>
<div class="flushright">
<!--l. 1247--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1247--><p class="noindent"> </p><div class="heading"> <a id="sec:colvar_multicolumn_grid"></a>
<h5 class="subsubsectionHead"><span class="titlemark">3.8.5    </span> <a id="sec:colvar_multicolumn_grid"></a>Grid files: multicolumn text format</h5>
<!--l. 1247--><p class="noindent"></p></div>
<!--l. 1249--><p class="indent">   Many simulation methods and analysis tools write files that contain functions of the collective variables
tabulated on a grid (e.g. potentials of mean force or multidimensional histograms) for the purpose of
analyzing results. Such files are produced by ABF (<a href="#sec:colvarbias_abf">6.3<!--tex4ht:ref: sec:colvarbias_abf --></a>), metadynamics (<a href="#sec:colvarbias_meta">6.6<!--tex4ht:ref: sec:colvarbias_meta --></a>), multidimensional
histograms (<a href="#sec:colvarbias_histogram">6.13<!--tex4ht:ref: sec:colvarbias_histogram --></a>), as well as any restraint with optional thermodynamic integration support
(<a href="#sec:colvarbias_ti">6.2<!--tex4ht:ref: sec:colvarbias_ti --></a>).
</p><!--l. 1252--><p class="indent">   In some cases, these files may also be read as input of a new simulation. Suitable input files for
this purpose are typically generated as output files of previous simulations, or directly by
the user in the specific case of ensemble-biased metadynamics (<a href="#sec:colvarbias_ebmeta">6.6.5<!--tex4ht:ref: sec:colvarbias_ebmeta --></a>). This section
explains the “multicolumn" format used by these files. For a multidimensional function
<!--l. 1255--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>f</mi><mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mn>1</mn></mrow></msub></math>,
<!--l. 1255--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mn>2</mn></mrow></msub></math>,
…<!--l. 1255--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">)</mo></math> the
multicolumn grid format is defined as follows:<br class="newline"/>
</p>
<div class="tabular"> <table class="tabular" id="TBL-2"><colgroup id="TBL-2-1g"><col id="TBL-2-1"/><col id="TBL-2-2"/><col id="TBL-2-3"/><col id="TBL-2-4"/><col id="TBL-2-5"/><col id="TBL-2-6"/><col id="TBL-2-7"/></colgroup><tr id="TBL-2-1-" style="vertical-align:baseline;"><td class="td11" id="TBL-2-1-1" style="text-align:left; white-space:nowrap;"> #  </td><td class="td11" id="TBL-2-1-2" style="text-align:left; white-space:nowrap;"> <!--l. 1258--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>N</mi></mrow><mrow><mstyle mathvariant="normal"><mi>c</mi><mi>v</mi></mstyle></mrow></msub></math> </td><td class="td11" id="TBL-2-1-3" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-2-1-4" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-2-1-5" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-2-1-6" style="text-align:left; white-space:nowrap;"> </td>
</tr><tr id="TBL-2-2-" style="vertical-align:baseline;"><td class="td11" id="TBL-2-2-1" style="text-align:left; white-space:nowrap;"> #   </td><td class="td11" id="TBL-2-2-2" style="text-align:left; white-space:nowrap;"> <!--l. 1259--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="monospace"><mi>m</mi><mi>i</mi><mi>n</mi></mstyle><mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-close">)</mo></math> </td><td class="td11" id="TBL-2-2-3" style="text-align:left; white-space:nowrap;"> <!--l. 1259--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="monospace"><mi>w</mi><mi>i</mi><mi>d</mi><mi>t</mi><mi>h</mi></mstyle><mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-close">)</mo></math> </td><td class="td11" id="TBL-2-2-4" style="text-align:left; white-space:nowrap;"> <!--l. 1259--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="monospace"><mi>n</mi><mi>p</mi><mi>o</mi><mi>i</mi><mi>n</mi><mi>t</mi><mi>s</mi></mstyle><mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-close">)</mo></math> </td><td class="td11" id="TBL-2-2-5" style="text-align:left; white-space:nowrap;"> <!--l. 1259--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="monospace"><mi>p</mi><mi>e</mi><mi>r</mi><mi>i</mi><mi>o</mi><mi>d</mi><mi>i</mi><mi>c</mi></mstyle><mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-close">)</mo></math> </td>
</tr><tr id="TBL-2-3-" style="vertical-align:baseline;"><td class="td11" id="TBL-2-3-1" style="text-align:left; white-space:nowrap;"> #   </td><td class="td11" id="TBL-2-3-2" style="text-align:left; white-space:nowrap;"> <!--l. 1260--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="monospace"><mi>m</mi><mi>i</mi><mi>n</mi></mstyle><mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mn>2</mn></mrow></msub><mo class="MathClass-close">)</mo></math> </td><td class="td11" id="TBL-2-3-3" style="text-align:left; white-space:nowrap;"> <!--l. 1260--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="monospace"><mi>w</mi><mi>i</mi><mi>d</mi><mi>t</mi><mi>h</mi></mstyle><mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mn>2</mn></mrow></msub><mo class="MathClass-close">)</mo></math> </td><td class="td11" id="TBL-2-3-4" style="text-align:left; white-space:nowrap;"> <!--l. 1260--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="monospace"><mi>n</mi><mi>p</mi><mi>o</mi><mi>i</mi><mi>n</mi><mi>t</mi><mi>s</mi></mstyle><mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mn>2</mn></mrow></msub><mo class="MathClass-close">)</mo></math> </td><td class="td11" id="TBL-2-3-5" style="text-align:left; white-space:nowrap;"> <!--l. 1260--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="monospace"><mi>p</mi><mi>e</mi><mi>r</mi><mi>i</mi><mi>o</mi><mi>d</mi><mi>i</mi><mi>c</mi></mstyle><mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mn>2</mn></mrow></msub><mo class="MathClass-close">)</mo></math> </td>
</tr><tr id="TBL-2-4-" style="vertical-align:baseline;"><td class="td11" id="TBL-2-4-1" style="text-align:left; white-space:nowrap;"> #   </td><td class="td11" id="TBL-2-4-2" style="text-align:left; white-space:nowrap;"> …                                                                                                                                                          </td><td class="td11" id="TBL-2-4-3" style="text-align:left; white-space:nowrap;"> …                                                                                                                                                          </td><td class="td11" id="TBL-2-4-4" style="text-align:left; white-space:nowrap;"> …                                                                                                                                                          </td><td class="td11" id="TBL-2-4-5" style="text-align:left; white-space:nowrap;"> …                                                                                                                                                          </td></tr><tr id="TBL-2-5-" style="vertical-align:baseline;"><td class="td 11" id="TBL-2-5-1" style="text-align:left; white-space:nowrap;"> #   </td><td class="td11" id="TBL-2-5-2" style="text-align:left; white-space:nowrap;"> <!--l. 1262--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="monospace"><mi>m</mi><mi>i</mi><mi>n</mi></mstyle><mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><msub><mrow><mi>N</mi></mrow><mrow><mstyle mathvariant="normal"><mi>c</mi><mi>v</mi></mstyle></mrow></msub></mrow></msub><mo class="MathClass-close">)</mo></math> </td><td class="td11" id="TBL-2-5-3" style="text-align:left; white-space:nowrap;"> <!--l. 1262--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="monospace"><mi>w</mi><mi>i</mi><mi>d</mi><mi>t</mi><mi>h</mi></mstyle><mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><msub><mrow><mi>N</mi></mrow><mrow><mstyle mathvariant="normal"><mi>c</mi><mi>v</mi></mstyle></mrow></msub></mrow></msub><mo class="MathClass-close">)</mo></math> </td><td class="td11" id="TBL-2-5-4" style="text-align:left; white-space:nowrap;"> <!--l. 1262--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="monospace"><mi>n</mi><mi>p</mi><mi>o</mi><mi>i</mi><mi>n</mi><mi>t</mi><mi>s</mi></mstyle><mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><msub><mrow><mi>N</mi></mrow><mrow><mstyle mathvariant="normal"><mi>c</mi><mi>v</mi></mstyle></mrow></msub></mrow></msub><mo class="MathClass-close">)</mo></math> </td><td class="td11" id="TBL-2-5-5" style="text-align:left; white-space:nowrap;"> <!--l. 1262--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="monospace"><mi>p</mi><mi>e</mi><mi>r</mi><mi>i</mi><mi>o</mi><mi>d</mi><mi>i</mi><mi>c</mi></mstyle><mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><msub><mrow><mi>N</mi></mrow><mrow><mstyle mathvariant="normal"><mi>c</mi><mi>v</mi></mstyle></mrow></msub></mrow></msub><mo class="MathClass-close">)</mo></math> </td><td class="td11" id="TBL-2-5-6" style="text-align:left; white-space:nowrap;"> </td>
</tr><tr id="TBL-2-6-" style="vertical-align:baseline;"><td class="td11" id="TBL-2-6-1" style="text-align:left; white-space:nowrap;"> </td>
</tr><tr id="TBL-2-7-" style="vertical-align:baseline;"><td class="td11" id="TBL-2-7-1" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-2-7-2" style="text-align:left; white-space:nowrap;"> <!--l. 1264--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msubsup><mrow><mi>ξ</mi></mrow><mrow><mn>1</mn></mrow><mrow><mn>1</mn></mrow></msubsup></math> </td><td class="td11" id="TBL-2-7-3" style="text-align:left; white-space:nowrap;"> <!--l. 1264--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msubsup><mrow><mi>ξ</mi></mrow><mrow><mn>2</mn></mrow><mrow><mn>1</mn></mrow></msubsup></math> </td><td class="td11" id="TBL-2-7-4" style="text-align:left; white-space:nowrap;"> …                                                                                                                                                          </td><td class="td11" id="TBL-2-7-5" style="text-align:left; white-space:nowrap;"> <!--l. 1264--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msubsup><mrow><mi>ξ</mi></mrow><mrow><msub><mrow><mi>N</mi></mrow><mrow><mstyle mathvariant="normal"><mi>c</mi><mi>v</mi></mstyle></mrow></msub></mrow><mrow><mn>1</mn></mrow></msubsup></math> </td><td class="td11" id="TBL-2-7-6" style="text-align:left; white-space:nowrap;"> f(<!--l. 1264--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msubsup><mrow><mi>ξ</mi></mrow><mrow><mn>1</mn></mrow><mrow><mn>1</mn></mrow></msubsup></math>, <!--l. 1264--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msubsup><mrow><mi>ξ</mi></mrow><mrow><mn>2</mn></mrow><mrow><mn>1</mn></mrow></msubsup></math>, …, <!--l. 1264--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msubsup><mrow><mi>ξ</mi></mrow><mrow><msub><mrow><mi>N</mi></mrow><mrow><mstyle mathvariant="normal"><mi>c</mi><mi>v</mi></mstyle></mrow></msub></mrow><mrow><mn>1</mn></mrow></msubsup></math>)  </td><td class="td11" id="TBL-2-7-7" style="text-align:left; white-space:nowrap;"> </td>
</tr><tr id="TBL-2-8-" style="vertical-align:baseline;"><td class="td11" id="TBL-2-8-1" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-2-8-2" style="text-align:left; white-space:nowrap;"> <!--l. 1265--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msubsup><mrow><mi>ξ</mi></mrow><mrow><mn>1</mn></mrow><mrow><mn>1</mn></mrow></msubsup></math> </td><td class="td11" id="TBL-2-8-3" style="text-align:left; white-space:nowrap;"> <!--l. 1265--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msubsup><mrow><mi>ξ</mi></mrow><mrow><mn>2</mn></mrow><mrow><mn>1</mn></mrow></msubsup></math> </td><td class="td11" id="TBL-2-8-4" style="text-align:left; white-space:nowrap;"> …                                                                                                                                                          </td><td class="td11" id="TBL-2-8-5" style="text-align:left; white-space:nowrap;"> <!--l. 1265--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msubsup><mrow><mi>ξ</mi></mrow><mrow><msub><mrow><mi>N</mi></mrow><mrow><mstyle mathvariant="normal"><mi>c</mi><mi>v</mi></mstyle></mrow></msub></mrow><mrow><mn>2</mn></mrow></msubsup></math> </td><td class="td11" id="TBL-2-8-6" style="text-align:left; white-space:nowrap;"> f(<!--l. 1265--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msubsup><mrow><mi>ξ</mi></mrow><mrow><mn>1</mn></mrow><mrow><mn>1</mn></mrow></msubsup></math>, <!--l. 1265--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msubsup><mrow><mi>ξ</mi></mrow><mrow><mn>2</mn></mrow><mrow><mn>1</mn></mrow></msubsup></math>, …, <!--l. 1265--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msubsup><mrow><mi>ξ</mi></mrow><mrow><msub><mrow><mi>N</mi></mrow><mrow><mstyle mathvariant="normal"><mi>c</mi><mi>v</mi></mstyle></mrow></msub></mrow><mrow><mn>2</mn></mrow></msubsup></math>)  </td></tr><tr id="TBL-2-9-" style="vertical-align:baseline;"><td class="td 11" id="TBL-2-9-1" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-2-9-2" style="text-align:left; white-space:nowrap;"> …                                                                                                                                                          </td><td class="td11" id="TBL-2-9-3" style="text-align:left; white-space:nowrap;"> …                                                                                                                                                          </td><td class="td11" id="TBL-2-9-4" style="text-align:left; white-space:nowrap;"> …                                                                                                                                                          </td><td class="td11" id="TBL-2-9-5" style="text-align:left; white-space:nowrap;"> …                                                                                                                                                          </td><td class="td11" id="TBL-2-9-6" style="text-align:left; white-space:nowrap;"> …                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          </td></tr><tr id="TBL-2-10-" style="vertical-align:baseline;"><td class="td11" id="TBL-2-10-1" style="text-align:left; white-space:nowrap;"> </td>
</tr><tr id="TBL-2-11-" style="vertical-align:baseline;"><td class="td11" id="TBL-2-11-1" style="text-align:left; white-space:nowrap;"> </td> </tr></table>
</div>
<!--l. 1270--><p class="noindent">Lines beginning with the character “<span class="ectt-1095">#</span>" are the header of the file.
<!--l. 1271--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>N</mi></mrow><mrow><mstyle mathvariant="normal"><mi>c</mi><mi>v</mi></mstyle></mrow></msub></math>
is the number of collective variables sampled by the grid. For each variable
<!--l. 1272--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></math>,
<!--l. 1272--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="monospace"><mi>m</mi><mi>i</mi><mi>n</mi></mstyle><mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-close">)</mo></math>
is the lowest value sampled by the grid (i.e. the left-most boundary of the grid along
<!--l. 1272--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></math>),
<!--l. 1272--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="monospace"><mi>w</mi><mi>i</mi><mi>d</mi><mi>t</mi><mi>h</mi></mstyle><mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-close">)</mo></math> is the width of each
grid step along <!--l. 1272--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></math>,
<!--l. 1272--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="monospace"><mi>n</mi><mi>p</mi><mi>o</mi><mi>i</mi><mi>n</mi><mi>t</mi><mi>s</mi></mstyle><mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-close">)</mo></math> is the number
of points and <!--l. 1272--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="monospace"><mi>p</mi><mi>e</mi><mi>r</mi><mi>i</mi><mi>o</mi><mi>d</mi><mi>i</mi><mi>c</mi></mstyle><mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-close">)</mo></math>
is a flag whose value is 1 or 0 depending on whether the grid is periodic along
<!--l. 1272--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></math>. In
most situations: </p>
<ul class="itemize1">
<li class="itemize"><!--l. 1275--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="monospace"><mi>m</mi><mi>i</mi><mi>n</mi></mstyle><mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-close">)</mo></math>
     is given by the <a href="#colvar|lowerBoundary"><span class="ectt-1095">lowerBoundary</span></a> keyword of the variable <!--l. 1275--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></math>;
     </li>
<li class="itemize"><!--l. 1276--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="monospace"><mi>w</mi><mi>i</mi><mi>d</mi><mi>t</mi><mi>h</mi></mstyle><mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-close">)</mo></math>
                                                                                         
                                                                                         
     is given by the <a href="#colvar|width"><span class="ectt-1095">width</span></a> keyword;
     </li>
<li class="itemize"><!--l. 1277--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="monospace"><mi>n</mi><mi>p</mi><mi>o</mi><mi>i</mi><mi>n</mi><mi>t</mi><mi>s</mi></mstyle><mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-close">)</mo></math>
     is calculated from the two above numbers and the <a href="#colvar|upperBoundary"><span class="ectt-1095">upperBoundary</span></a> keyword;
     </li>
<li class="itemize"><!--l. 1278--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="monospace"><mi>p</mi><mi>e</mi><mi>r</mi><mi>i</mi><mi>o</mi><mi>d</mi><mi>i</mi><mi>c</mi></mstyle><mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-close">)</mo></math>
     is set to 1 if and only if <!--l. 1278--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></math>
     is periodic and the grids' boundaries cover its period.</li></ul>
<!--l. 1281--><p class="indent">   How the grid's boundaries affect the sequence of points depends on how the contents of the file were
computed. In many cases, such as histograms and PMFs computed by metadynamics (<a href="#sec:colvarbias_ebmeta">6.6.5<!--tex4ht:ref: sec:colvarbias_ebmeta --></a>), the values
of <!--l. 1281--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></math>
in the first few columns correspond to the <span class="phvro8t-x-x-109">midpoints </span>of the corresponding bins,
i.e. <!--l. 1281--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msubsup><mrow><mi>ξ</mi></mrow><mrow><mn>1</mn></mrow><mrow><mn>1</mn></mrow></msubsup> <mo class="MathClass-rel">=</mo> <mstyle mathvariant="monospace"><mi>m</mi><mi>i</mi><mi>n</mi></mstyle><mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-close">)</mo><mo class="MathClass-bin">+</mo><mstyle mathvariant="monospace"><mi>w</mi><mi>i</mi><mi>d</mi><mi>t</mi><mi>h</mi></mstyle><mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-close">)</mo><mo class="MathClass-bin">∕</mo><mn>2</mn></math>.
However, there is a slightly different format in PMF files computed by ABF (<a href="#sec:colvarbias_abf">6.3<!--tex4ht:ref: sec:colvarbias_abf --></a>) or
other biases that use thermodynamic integration (<a href="#sec:colvarbias_ti">6.2<!--tex4ht:ref: sec:colvarbias_ti --></a>). In these cases, it is free-energy
gradients that are accumulated on an <span class="ectt-1095">(npoints)</span>-long grid along each variable
<!--l. 1283--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>: after these
gradients are integrated, the resulting PMF is discretized on a slightly larger grid with <span class="ectt-1095">(npoints+1) </span>points
along <!--l. 1283--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>
(unless the interval is periodic). Therefore, the grid's outer edges extend by
<!--l. 1284--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="monospace"><mi>w</mi><mi>i</mi><mi>d</mi><mi>t</mi><mi>h</mi></mstyle><mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-close">)</mo><mo class="MathClass-bin">∕</mo><mn>2</mn></math>
above and below the specified boundaries, so that for instance
<!--l. 1284--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="monospace"><mi>m</mi><mi>i</mi><mi>n</mi></mstyle><mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-close">)</mo></math> in the header appears
to be shifted back by <!--l. 1284--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="monospace"><mi>w</mi><mi>i</mi><mi>d</mi><mi>t</mi><mi>h</mi></mstyle><mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-close">)</mo><mo class="MathClass-bin">∕</mo><mn>2</mn></math>
compared to what would be expected. <span class="phvro8t-x-x-109">Please keep this difference in mind when comparing PMFs</span>
<span class="phvro8t-x-x-109">computed by different methods.</span>
</p><!--l. 1286--><p class="indent">   After the header, the rest of the file contains values of the tabulated function
<!--l. 1287--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>f</mi><mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mn>1</mn></mrow></msub></math>,
<!--l. 1287--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mn>2</mn></mrow></msub></math>,
…<!--l. 1287--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><msub><mrow><mi>N</mi></mrow><mrow><mstyle mathvariant="normal"><mi>c</mi><mi>v</mi></mstyle></mrow></msub></mrow></msub><mo class="MathClass-close">)</mo></math>, one for each line.
The first <!--l. 1288--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>N</mi></mrow><mrow><mstyle mathvariant="normal"><mi>c</mi><mi>v</mi></mstyle></mrow></msub></math> columns
contain values of <!--l. 1288--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mn>1</mn></mrow></msub></math>,
<!--l. 1288--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mn>2</mn></mrow></msub></math>,
…<!--l. 1288--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><msub><mrow><mi>N</mi></mrow><mrow><mstyle mathvariant="normal"><mi>c</mi><mi>v</mi></mstyle></mrow></msub></mrow></msub></math> and the last column contains
the value of the function <!--l. 1288--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>f</mi></math>.
Points are sorted in ascending order with the fastest-changing values at the right (“C-style" order). Each sweep of the
right-most variable <!--l. 1290--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><msub><mrow><mi>N</mi></mrow><mrow><mstyle mathvariant="normal"><mi>c</mi><mi>v</mi></mstyle></mrow></msub></mrow></msub></math>
is terminated by an empty line. For two dimensional grid files, this allows quick visualization by programs
such as GNUplot. <br class="newline"/>
</p><!--l. 1294--><p class="noindent"><span class="phvb8t-x-x-109">Example 1: </span>multicolumn text file for a one-dimensional histogram with <span class="ectt-1095">lowerBoundary </span>= 15,
<span class="ectt-1095">upperBoundary </span>= 48 and <span class="ectt-1095">width </span>= 0.1.
</p>
<div class="tabular"> <table class="tabular" id="TBL-3"><colgroup id="TBL-3-1g"><col id="TBL-3-1"/><col id="TBL-3-2"/><col id="TBL-3-3"/><col id="TBL-3-4"/><col id="TBL-3-5"/><col id="TBL-3-6"/></colgroup><tr id="TBL-3-1-" style="vertical-align:baseline;"><td class="td11" id="TBL-3-1-1" style="text-align:left; white-space:nowrap;"> </td></tr><tr id="TBL-3-2-" style="vertical-align:baseline;"><td class="td11" id="TBL-3-2-1" style="text-align:left; white-space:nowrap;"> #</td><td class="td11" id="TBL-3-2-2" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">1     </span></td><td class="td11" id="TBL-3-2-3" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-3-2-4" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-3-2-5" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-3-2-6" style="text-align:left; white-space:nowrap;"> </td>
</tr><tr id="TBL-3-3-" style="vertical-align:baseline;"><td class="td11" id="TBL-3-3-1" style="text-align:left; white-space:nowrap;"> #  </td><td class="td11" id="TBL-3-3-2" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">15    </span></td><td class="td11" id="TBL-3-3-3" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">0.1         </span></td><td class="td11" id="TBL-3-3-4" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">330 </span></td><td class="td11" id="TBL-3-3-5" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">0 </span></td></tr><tr id="TBL-3-4-" style="vertical-align:baseline;"><td class="td11" id="TBL-3-4-1" style="text-align:left; white-space:nowrap;"> </td>
</tr><tr id="TBL-3-5-" style="vertical-align:baseline;"><td class="td11" id="TBL-3-5-1" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-3-5-2" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">15.05 </span></td><td class="td11" id="TBL-3-5-3" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">6.14012e-07 </span></td><td class="td11" id="TBL-3-5-4" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-3-5-5" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-3-5-6" style="text-align:left; white-space:nowrap;"> </td></tr><tr id="TBL-3-6-" style="vertical-align:baseline;"><td class="td11" id="TBL-3-6-1" style="text-align:left; white-space:nowrap;"> </td> <td class="td11" id="TBL-3-6-2" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">15.15 </span></td> <td class="td11" id="TBL-3-6-3" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">7.47644e-07 </span></td> <td class="td11" id="TBL-3-6-4" style="text-align:left; white-space:nowrap;"> </td> <td class="td11" id="TBL-3-6-5" style="text-align:left; white-space:nowrap;"> </td> <td class="td11" id="TBL-3-6-6" style="text-align:left; white-space:nowrap;"></td>
</tr><tr id="TBL-3-7-" style="vertical-align:baseline;"><td class="td11" id="TBL-3-7-1" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-3-7-2" style="text-align:left; white-space:nowrap;"> …         </td><td class="td11" id="TBL-3-7-3" style="text-align:left; white-space:nowrap;"> …                     </td><td class="td11" id="TBL-3-7-4" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-3-7-5" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-3-7-6" style="text-align:left; white-space:nowrap;"> </td></tr><tr id="TBL-3-8-" style="vertical-align:baseline;"><td class="td11" id="TBL-3-8-1" style="text-align:left; white-space:nowrap;"> </td> <td class="td11" id="TBL-3-8-2" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">47.85 </span></td> <td class="td11" id="TBL-3-8-3" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">1.65944e-06 </span></td> <td class="td11" id="TBL-3-8-4" style="text-align:left; white-space:nowrap;"> </td> <td class="td11" id="TBL-3-8-5" style="text-align:left; white-space:nowrap;"> </td> <td class="td11" id="TBL-3-8-6" style="text-align:left; white-space:nowrap;"></td>
</tr><tr id="TBL-3-9-" style="vertical-align:baseline;"><td class="td11" id="TBL-3-9-1" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-3-9-2" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">47.95 </span></td><td class="td11" id="TBL-3-9-3" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">1.46712e-06 </span></td><td class="td11" id="TBL-3-9-4" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-3-9-5" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-3-9-6" style="text-align:left; white-space:nowrap;"> </td></tr><tr id="TBL-3-10-" style="vertical-align:baseline;"><td class="td11" id="TBL-3-10-1" style="text-align:left; white-space:nowrap;"> </td>
</tr><tr id="TBL-3-11-" style="vertical-align:baseline;"><td class="td11" id="TBL-3-11-1" style="text-align:left; white-space:nowrap;"> </td> </tr></table></div>
<!--l. 1309--><p class="noindent"><span class="phvb8t-x-x-109">Example 2: </span>multicolumn text file for a two-dimensional histogram of two dihedral angles (periodic interval
with 6<!--l. 1309--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></math>
bins):
</p>
<div class="tabular"> <table class="tabular" id="TBL-4"><colgroup id="TBL-4-1g"><col id="TBL-4-1"/><col id="TBL-4-2"/><col id="TBL-4-3"/><col id="TBL-4-4"/><col id="TBL-4-5"/><col id="TBL-4-6"/></colgroup><tr id="TBL-4-1-" style="vertical-align:baseline;"><td class="td11" id="TBL-4-1-1" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-4-1-2" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-4-1-3" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-4-1-4" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-4-1-5" style="text-align:left; white-space:nowrap;"> </td></tr><tr id="TBL-4-2-" style="vertical-align:baseline;"><td class="td11" id="TBL-4-2-1" style="text-align:left; white-space:nowrap;"> # </td> <td class="td11" id="TBL-4-2-2" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">2 </span></td> <td class="td11" id="TBL-4-2-3" style="text-align:left; white-space:nowrap;"> </td> <td class="td11" id="TBL-4-2-4" style="text-align:left; white-space:nowrap;"> </td> <td class="td11" id="TBL-4-2-5" style="text-align:left; white-space:nowrap;"></td><td class="td11" id="TBL-4-2-6" style="text-align:left; white-space:nowrap;"> </td>
</tr><tr id="TBL-4-3-" style="vertical-align:baseline;"><td class="td11" id="TBL-4-3-1" style="text-align:left; white-space:nowrap;"> #  </td><td class="td11" id="TBL-4-3-2" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">-180.0 </span></td><td class="td11" id="TBL-4-3-3" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">6.0    </span></td><td class="td11" id="TBL-4-3-4" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">30          </span></td><td class="td11" id="TBL-4-3-5" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">1 </span></td></tr><tr id="TBL-4-4-" style="vertical-align:baseline;"><td class="td11" id="TBL-4-4-1" style="text-align:left; white-space:nowrap;"> # </td> <td class="td11" id="TBL-4-4-2" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">-180.0 </span></td> <td class="td11" id="TBL-4-4-3" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">6.0 </span></td> <td class="td11" id="TBL-4-4-4" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">30 </span></td> <td class="td11" id="TBL-4-4-5" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">1</span></td>
</tr><tr id="TBL-4-5-" style="vertical-align:baseline;"><td class="td11" id="TBL-4-5-1" style="text-align:left; white-space:nowrap;"> </td></tr><tr id="TBL-4-6-" style="vertical-align:baseline;"><td class="td11" id="TBL-4-6-1" style="text-align:left; white-space:nowrap;"></td><td class="td11" id="TBL-4-6-2" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">-177.0 </span></td><td class="td11" id="TBL-4-6-3" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">-177.0 </span></td><td class="td11" id="TBL-4-6-4" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">8.97117e-06 </span></td><td class="td11" id="TBL-4-6-5" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-4-6-6" style="text-align:left; white-space:nowrap;"> </td>
</tr><tr id="TBL-4-7-" style="vertical-align:baseline;"><td class="td11" id="TBL-4-7-1" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-4-7-2" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">-177.0 </span></td><td class="td11" id="TBL-4-7-3" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">-171.0 </span></td><td class="td11" id="TBL-4-7-4" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">1.53525e-06 </span></td><td class="td11" id="TBL-4-7-5" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-4-7-6" style="text-align:left; white-space:nowrap;"> </td></tr><tr id="TBL-4-8-" style="vertical-align:baseline;"><td class="td11" id="TBL-4-8-1" style="text-align:left; white-space:nowrap;"> </td> <td class="td11" id="TBL-4-8-2" style="text-align:left; white-space:nowrap;"> … </td> <td class="td11" id="TBL-4-8-3" style="text-align:left; white-space:nowrap;"> … </td> <td class="td11" id="TBL-4-8-4" style="text-align:left; white-space:nowrap;"> … </td> <td class="td11" id="TBL-4-8-5" style="text-align:left; white-space:nowrap;"> </td> <td class="td11" id="TBL-4-8-6" style="text-align:left; white-space:nowrap;"></td>
</tr><tr id="TBL-4-9-" style="vertical-align:baseline;"><td class="td11" id="TBL-4-9-1" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-4-9-2" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">-177.0 </span></td><td class="td11" id="TBL-4-9-3" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">177.0 </span></td><td class="td11" id="TBL-4-9-4" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">2.442956-06 </span></td><td class="td11" id="TBL-4-9-5" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-4-9-6" style="text-align:left; white-space:nowrap;"> </td></tr><tr id="TBL-4-10-" style="vertical-align:baseline;"><td class="td11" id="TBL-4-10-1" style="text-align:left; white-space:nowrap;"> </td>
</tr><tr id="TBL-4-11-" style="vertical-align:baseline;"><td class="td11" id="TBL-4-11-1" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-4-11-2" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">-171.0 </span></td><td class="td11" id="TBL-4-11-3" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">-177.0 </span></td><td class="td11" id="TBL-4-11-4" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">2.04702e-05 </span></td><td class="td11" id="TBL-4-11-5" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-4-11-6" style="text-align:left; white-space:nowrap;"> </td></tr><tr id="TBL-4-12-" style="vertical-align:baseline;"><td class="td11" id="TBL-4-12-1" style="text-align:left; white-space:nowrap;"> </td> <td class="td11" id="TBL-4-12-2" style="text-align:left; white-space:nowrap;"> … </td> <td class="td11" id="TBL-4-12-3" style="text-align:left; white-space:nowrap;"> … </td> <td class="td11" id="TBL-4-12-4" style="text-align:left; white-space:nowrap;"> … </td> <td class="td11" id="TBL-4-12-5" style="text-align:left; white-space:nowrap;"> </td> <td class="td11" id="TBL-4-12-6" style="text-align:left; white-space:nowrap;"></td>
</tr><tr id="TBL-4-13-" style="vertical-align:baseline;"><td class="td11" id="TBL-4-13-1" style="text-align:left; white-space:nowrap;"> </td> </tr></table></div>
<div class="flushright">
<!--l. 1327--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1327--><p class="noindent"> </p><div class="heading"> <a id="sec:colvars_traj_format"></a>
<h5 class="subsubsectionHead"><span class="titlemark">3.8.6    </span> <a id="sec:colvars_traj_format"></a>Output trajectory files</h5>
<!--l. 1327--><p class="noindent"></p></div>
<!--l. 1329--><p class="indent">   The Colvars trajectory file (with a suffix <span class="ectt-1095">.colvars.traj</span>) is a plain text file (scientific
notation with 14-digit precision) whose columns represent quantities such as colvar
values, applied forces, or individual restraints' energies. Under most scenarios, plotting
or analyzing this file is straightforward: for example, the following contains a variable
“<!--l. 1329--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>A</mi></math>" and the energy
of a restraint “<!--l. 1329--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>r</mi><mi>A</mi></math>":</p>
<div class="mdframed" id="mdframed-37">
<!--l. 1330--><p class="noindent"></p><pre>#       step   A                     E_rA<br class="newline"/>           0    1.42467449615693e+01  6.30982865292123e+02<br class="newline"/>         100    1.42282559728026e+01  6.20640585041317e+02<br class="newline"/>…</pre><!--l. 1335--><p class="noindent"></p>
</div>
<!--l. 1337--><p class="indent">   Occasionally, if the Colvars configuration is changed mid-run certain quantities may be
added or removed, changing the column layout. Labels in comment lines can assist in such
cases: for example, consider the trajectory above with the addition of a second variable,
“<!--l. 1337--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>B</mi></math>", after
10,000 steps:</p>
<div class="mdframed" id="mdframed-38">
<!--l. 1338--><p class="noindent"></p><pre>#       step   A                     E_rA<br class="newline"/>           0    1.42467449615693e+01  6.30982865292123e+02<br class="newline"/>         100    1.42282559728026e+01  6.20640585041317e+02<br class="newline"/>…<br class="newline"/>#       step   A                     B                     E_rA              <br class="newline"/>       10000    1.38136915830229e+01  9.99574098859265e-01  4.11184644791030e+02<br class="newline"/>       10100    1.36437184346326e+01  9.99574091957314e-01  3.37726286543895e+02
</pre><!--l. 1346--><p class="noindent"></p>
</div>
<!--l. 1348--><p class="indent">   Analyzing the above file with standard tools is possible, but laborious: a <a href="https://github.com/Colvars/colvars/blob/master/colvartools/plot_colvars_traj.py?raw=true">convenience script</a> is provided
for this and related purposes. It may be used either as a command-line tool or as a Python
module:</p>
<div class="mdframed" id="mdframed-39">
<!--l. 1349--><p class="noindent"></p><pre>&gt;&gt;&gt; from plot_colvars_traj import Colvars_traj<br class="newline"/>&gt;&gt;&gt; traj = Colvars_traj('test.colvars.traj')<br class="newline"/>&gt;&gt;&gt; print(traj['A'].steps, traj['A'].values)<br class="newline"/>[    0   100  ...  10000 10100] [14.246745 14.228256 ... 13.813692 13.643718]<br class="newline"/>&gt;&gt;&gt; print(traj['B'].steps, traj['B'].values)<br class="newline"/>[10000 10100] [0.999574  0.9995741] </pre><!--l. 1356--><p class="noindent"></p>
</div>
<!--l. 1359--><p class="indent">
</p>
<div class="flushright">
<!--l. 1359--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1359--><p class="noindent"> </p><div class="heading"> <a id="sec:colvar"></a>
<h3 class="sectionHead"><span class="titlemark">4    </span> <a id="sec:colvar"></a>Defining collective variables</h3>
<!--l. 1359--><p class="noindent"></p></div>
<!--l. 1361--><p class="indent">   A collective variable is defined by the keyword <span class="ectt-1095">colvar </span>followed by its configuration options contained
within curly braces:</p>
<div class="mdframed" id="mdframed-40">
<!--l. 1363--><p class="noindent"></p><pre>colvar {<br class="newline"/>  name xi<br class="newline"/>  <!--l. 1366--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>other options<!--l. 1366--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><br class="newline"/>  function_name {<br class="newline"/>    <!--l. 1368--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>parameters<!--l. 1368--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><br class="newline"/>    <!--l. 1369--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>atom selection<!--l. 1369--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><br class="newline"/>  }<br class="newline"/>} </pre><!--l. 1372--><p class="noindent"></p>
</div>
<!--l. 1375--><p class="noindent">There are multiple ways of defining a variable: </p>
<ul class="itemize1">
<li class="itemize">The <span class="phvro8t-x-x-109">simplest and most common way  </span>way is using one of the precompiled functions (here
     called “components"), which are listed in section <a href="#sec:cvc_list">4.1<!--tex4ht:ref: sec:cvc_list --></a>. For example, using the keyword <span class="ectt-1095">rmsd</span>
     (section  <a href="#sec:cvc_rmsd">4.6.1<!--tex4ht:ref: sec:cvc_rmsd --></a>)  defines  the  variable  as  the  root  mean  squared  deviation  (RMSD)  of  the
     selected atoms.
     </li>
<li class="itemize">A  new  variable  may  also  be  constructed  as  a  linear  or  polynomial  combination  of  the
     components listed in section <a href="#sec:cvc_list">4.1<!--tex4ht:ref: sec:cvc_list --></a> (see <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for details).
     </li>
<li class="itemize">A user-defined mathematical function of the existing components (see list in section <a href="#sec:cvc_list">4.1<!--tex4ht:ref: sec:cvc_list --></a>),
     or of the atomic coordinates directly (see the <span class="ectt-1095">cartesian </span>keyword in <a href="#sec:cvc_cartesian">4.10.1<!--tex4ht:ref: sec:cvc_cartesian --></a>). The function is
     defined through the keyword <a href="#colvar|customFunction"><span class="ectt-1095">customFunction</span></a> (see <a href="#sec:colvar_custom_function">4.20<!--tex4ht:ref: sec:colvar_custom_function --></a> for details).
     </li>
<li class="itemize">A user-defined Tcl function of the existing components (see list in section <a href="#sec:cvc_list">4.1<!--tex4ht:ref: sec:cvc_list --></a>), or of the
     atomic coordinates directly (see the <span class="ectt-1095">cartesian </span>keyword in <a href="#sec:cvc_cartesian">4.10.1<!--tex4ht:ref: sec:cvc_cartesian --></a>). The function is provided
     by a separate Tcl script, and referenced through the keyword <a href="#colvar|scriptedFunction"><span class="ectt-1095">scriptedFunction</span></a> (see <a href="#sec:colvar_scripted">4.21<!--tex4ht:ref: sec:colvar_scripted --></a>
     for details).</li></ul>
<!--l. 1388--><p class="noindent">Choosing a component (function) is the only parameter strictly required to define a collective variable. It is
also highly recommended to specify a name for the variable: </p>
<ul class="itemize1">
<li class="itemize"><a id="colvar|name"></a> <span class="colorbox" id="colorbox11"><a id="dx1-31001"></a><span class="ectt-1095">name</span></span> — <span class="phvro8t-x-x-109">Name of this colvar</span><br class="newline"/>Default: “<span class="ectt-1095">colvar</span>" + numeric id <!--l. 1403--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> string,
     context: <span class="ectt-1095">colvar</span> <!--l. 1403--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>The name is an unique case-sensitive string which allows the Colvars module to identify this
     colvar unambiguously; it is also used in the trajectory file to label to the columns corresponding
     to this colvar.
     </li></ul>
<div class="flushright">
<!--l. 1408--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1408--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_list"></a>
<h4 class="subsectionHead"><span class="titlemark">4.1    </span> <a id="sec:cvc_list"></a>Choosing a function</h4>
<!--l. 1408--><p class="noindent"></p></div>
<!--l. 1410--><p class="indent">   In this context, the function that computes a colvar is called a <span class="phvro8t-x-x-109">component</span>. A component's choice
and definition consists of including in the variable's configuration a keyword indicating the
type of function (e.g. <span class="ectt-1095">rmsd</span>), followed by a definition block specifying the atoms involved (see
<a href="#sec:colvar_atom_groups">5<!--tex4ht:ref: sec:colvar_atom_groups --></a>) and any additional parameters (cutoffs, “reference" values, …). <span class="phvro8t-x-x-109">At least one component</span>
<span class="phvro8t-x-x-109">must be chosen to define a variable: </span>if none of the keywords listed below is found, an error is
raised.
</p><!--l. 1414--><p class="indent">   The following components implement functions with a scalar value (i.e. a real number):
</p>
<ul class="itemize1">
<li class="itemize"><a href="#colvar|distance"><span class="ectt-1095">distance</span></a>: distance between two groups;
     </li>
<li class="itemize"><a href="#colvar|distanceZ"><span class="ectt-1095">distanceZ</span></a>: projection of a distance vector on an axis;
     </li>
<li class="itemize"><a href="#colvar|distanceXY"><span class="ectt-1095">distanceXY</span></a>: projection of a distance vector on a plane;
     </li>
<li class="itemize"><a href="#colvar|distanceInv"><span class="ectt-1095">distanceInv</span></a>: mean distance between two groups of atoms (e.g. NOE-based distance);
     </li>
<li class="itemize"><a href="#colvar|angle"><span class="ectt-1095">angle</span></a>: angle between three groups;
     </li>
<li class="itemize"><a href="#colvar|dihedral"><span class="ectt-1095">dihedral</span></a>: torsional (dihedral) angle between four groups;
     </li>
<li class="itemize"><a href="#colvar|dipoleAngle"><span class="ectt-1095">dipoleAngle</span></a>: angle between two groups and dipole of a third group;
     </li>
<li class="itemize"><a href="#colvar|dipoleMagnitude"><span class="ectt-1095">dipoleMagnitude</span></a>: magnitude of the dipole of a group of atoms;
                                                                                         
                                                                                         
     </li>
<li class="itemize"><a href="#colvar|polarTheta"><span class="ectt-1095">polarTheta</span></a>: polar angle of a group in spherical coordinates;
     </li>
<li class="itemize"><a href="#colvar|polarPhi"><span class="ectt-1095">polarPhi</span></a>: azimuthal angle of a group in spherical coordinates;
     </li>
<li class="itemize"><a href="#colvar|eulerPhi"><span class="ectt-1095">eulerPhi</span></a>: Roll angle of rotation from references coordinates;
     </li>
<li class="itemize"><a href="#colvar|eulerTheta"><span class="ectt-1095">eulerTheta</span></a>: Pitch angle of rotation from references coordinates;
     </li>
<li class="itemize"><a href="#colvar|eulerPsi"><span class="ectt-1095">eulerPsi</span></a>: Yaw angle of rotation from references coordinates;
     </li>
<li class="itemize"><a href="#colvar|coordNum"><span class="ectt-1095">coordNum</span></a>: coordination number between two groups;
     </li>
<li class="itemize"><a href="#colvar|selfCoordNum"><span class="ectt-1095">selfCoordNum</span></a>: coordination number of atoms within a group;
     </li>
<li class="itemize"><a href="#colvar|hBond"><span class="ectt-1095">hBond</span></a>: hydrogen bond between two atoms;
     </li>
<li class="itemize"><a href="#colvar|rmsd"><span class="ectt-1095">rmsd</span></a>: root mean square deviation (RMSD) from a set of reference coordinates;
     </li>
<li class="itemize"><a href="#colvar|eigenvector"><span class="ectt-1095">eigenvector</span></a>: projection of the atomic coordinates on a vector;
     </li>
<li class="itemize"><a href="#colvar|mapTotal"><span class="ectt-1095">mapTotal</span></a>: total value of a volumetric map;
     </li>
<li class="itemize"><a href="#colvar|orientationAngle"><span class="ectt-1095">orientationAngle</span></a>: angle of the best-fit rotation from a set of reference coordinates;
     </li>
<li class="itemize"><a href="#colvar|orientationProj"><span class="ectt-1095">orientationProj</span></a>: cosine of <a href="#colvar|orientationProj"><span class="ectt-1095">orientationProj</span></a>;
     </li>
<li class="itemize"><a href="#colvar|spinAngle"><span class="ectt-1095">spinAngle</span></a>: projection orthogonal to an axis of the best-fit rotation from a set of reference
     coordinates;
     </li>
<li class="itemize"><a href="#colvar|tilt"><span class="ectt-1095">tilt</span></a>: projection on an axis of the best-fit rotation from a set of reference coordinates;
     </li>
<li class="itemize"><a href="#colvar|gyration"><span class="ectt-1095">gyration</span></a>: radius of gyration of a group of atoms;
     </li>
<li class="itemize"><a href="#colvar|inertia"><span class="ectt-1095">inertia</span></a>: moment of inertia of a group of atoms;
     </li>
<li class="itemize"><a href="#colvar|inertiaZ"><span class="ectt-1095">inertiaZ</span></a>: moment of inertia of a group of atoms around a chosen axis;
                                                                                         
                                                                                         
     </li>
<li class="itemize"><a href="#colvar|alpha"><span class="ectt-1095">alpha</span></a>: <!--l. 1448--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>α</mi></math>-helix
     content of a protein segment.
     </li>
<li class="itemize"><a href="#colvar|dihedralPC"><span class="ectt-1095">dihedralPC</span></a>: projection of protein backbone dihedrals onto a dihedral principal component.
     </li>
<li class="itemize"><a href="#colvar|alchLambda"><span class="ectt-1095">alchLambda</span></a>: alchemical lambda parameter (for controlling the back-end).
     </li>
<li class="itemize"><a href="#colvar|alchFlambda"><span class="ectt-1095">alchFlambda</span></a>: force along alchLambda. </li></ul>
<!--l. 1456--><p class="indent">   Some components do not return scalar, but vector values: </p>
<ul class="itemize1">
<li class="itemize"><a href="#colvar|distanceVec"><span class="ectt-1095">distanceVec</span></a>: distance vector between two groups (length: 3);
     </li>
<li class="itemize"><a href="#colvar|distanceDir"><span class="ectt-1095">distanceDir</span></a>: unit vector parallel to distanceVec (length: 3);
     </li>
<li class="itemize"><a href="#colvar|cartesian"><span class="ectt-1095">cartesian</span></a>: vector of atomic Cartesian coordinates (length: <!--l. 1460--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>N</mi></math>
     times the number of Cartesian components requested, X, Y or Z);
     </li>
<li class="itemize"><a href="#colvar|distancePairs"><span class="ectt-1095">distancePairs</span></a>: vector of mutual distances (length: <!--l. 1461--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>N</mi></mrow><mrow><mstyle mathvariant="normal"><mn>1</mn></mstyle></mrow></msub> <mo class="MathClass-bin">×</mo><msub><mrow><mi>N</mi></mrow><mrow><mstyle mathvariant="normal"><mn>2</mn></mstyle></mrow></msub></math>);
     </li>
<li class="itemize"><a href="#colvar|orientation"><span class="ectt-1095">orientation</span></a>: best-fit rotation, expressed as a unit quaternion (length: 4).</li></ul>
<!--l. 1465--><p class="indent">   The types of components used in a colvar (scalar or not) determine the properties of that colvar, and
particularly which biasing or analysis methods can be applied.
</p><!--l. 1469--><p class="indent"> <span class="phvb8t-x-x-109">What if “X" is not listed? </span>If a function type is not available on this list, it may be possible to define it
as a polynomial superposition of existing ones (see <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a>), a custom function (see <a href="#sec:colvar_custom_function">4.20<!--tex4ht:ref: sec:colvar_custom_function --></a>), or a scripted
function (see <a href="#sec:colvar_scripted">4.21<!--tex4ht:ref: sec:colvar_scripted --></a>).
</p><!--l. 1471--><p class="indent">   In the rest of this section, all available component types are listed, along with their physical units and
their ranges of values, if limited. Such ranges are often used to define automatically default sampling
intervals, for example by setting the parameters <a href="#colvar|lowerBoundary"><span class="ectt-1095">lowerBoundary</span></a> and <a href="#colvar|upperBoundary"><span class="ectt-1095">upperBoundary</span></a> in the parent
colvar.
</p><!--l. 1474--><p class="indent">   For each type of component, the available configurations keywords are listed: when two components
share certain keywords, the second component references to the documentation of the first one that uses
that keyword. The very few keywords that are available for all types of components are listed in a separate
section <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>.
</p>
<div class="flushright">
<!--l. 1480--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1480--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_pbcs"></a>
<h4 class="subsectionHead"><span class="titlemark">4.2    </span> <a id="sec:cvc_pbcs"></a>Treatment of periodic boundary conditions</h4>
<!--l. 1480--><p class="noindent"></p></div>
<!--l. 1482--><p class="indent">   In all colvar components described below, the following rules apply concerning periodic boundary
conditions (PBCs):
     </p><ol class="enumerate1">
<li class="enumerate" id="x1-33002x1">Distance vectors between two coordinates <!--l. 1486--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>d</mi></mstyle></mrow><mrow><mi>i</mi><mo class="MathClass-punc">,</mo><mi>j</mi></mrow></msub> <mo class="MathClass-rel">=</mo> <mrow><mo fence="true" form="prefix"> (</mo><mrow><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mn>1</mn></mrow></msub> <mo class="MathClass-bin">−</mo><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mn>2</mn></mrow></msub></mrow><mo fence="true" form="postfix">)</mo></mrow></math>,
     are  calculated  following  the  minimum-image  convention  by  default,  unless  <a href="#colvar|cvc|forceNoPBC"><span class="ectt-1095">forceNoPBC</span></a> is
     enabled. (<!--l. 1488--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mn>1</mn></mrow></msub></math>
     and <!--l. 1488--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mn>2</mn></mrow></msub></math>
     may be either individual atomic coordinates, or centers of mass of two groups.)
     </li>
<li class="enumerate" id="x1-33004x2">For all other functions of individual atomic coordinates, <!--l. 1491--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>f</mi> <mrow><mo fence="true" form="prefix"> (</mo><mrow><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mn>2</mn></mrow></msub><mo class="MathClass-punc">,</mo><mi class="MathClass-op">…</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo></mrow><mo fence="true" form="postfix">)</mo></mrow></math>,
     it is assumed that all atoms that are part of the same group are in the same periodic unit cell
     (see <a href="#sec:colvar_atom_groups_wrapping">5.3<!--tex4ht:ref: sec:colvar_atom_groups_wrapping --></a>).</li></ol>
<ul class="itemize1">
<li class="itemize"><a id="colvar|cvc|forceNoPBC"></a><span class="colorbox" id="colorbox12"><a id="dx1-33005"></a><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances?</span><br class="newline"/>Default: <span class="ectt-1095">no</span> <!--l. 1507--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: any component <!--l. 1507--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If this keyword is set to <span class="ectt-1095">yes</span>, PBCs will be ignored when calculating this component. This
     is only useful in a limited number of special cases, e.g. to describe the distance between
     remote points of a single macromolecule, which cannot be split across periodic cell boundaries,
     and for which the minimum-image distance might give the wrong result because of a relatively
     small periodic cell.</li></ul>
<!--l. 1510--><p class="noindent"><span class="likeparagraphHead"><a id="x1-340004.2"></a>Example: ignoring PBCs in a distance between two groups.</span>
   Using <span class="ectt-1095">forceNoPBC yes </span>when defining a <span class="ectt-1095">distance { ... } </span>function lets Colvars ignore the
minimum-image convention when computing the distance between the centers of mass of <span class="ectt-1095">group1 </span>and
<span class="ectt-1095">group2</span>. (Note that the centers of mass themselves are always calculated as weighted averages
anyway).</p>
<div class="mdframed" id="mdframed-41">
<!--l. 1513--><p class="noindent"></p><pre>colvar {<br class="newline"/>  name d<br class="newline"/>  distance {<br class="newline"/>    forceNoPBC yes<br class="newline"/>    group1 { … }<br class="newline"/>    group2 { … }<br class="newline"/>  }<br class="newline"/>}<br class="newline"/></pre><!--l. 1522--><p class="noindent"></p>
</div>
<div class="flushright">
<!--l. 1525--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1525--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_distances"></a>
<h4 class="subsectionHead"><span class="titlemark">4.3    </span> <a id="sec:cvc_distances"></a>Distances</h4>
<!--l. 1525--><p class="noindent"></p></div>
<div class="flushright">
<!--l. 1528--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1528--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_distance"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.3.1    </span> <a id="sec:cvc_distance"></a><span class="ectt-1095">distance</span>: center-of-mass distance between two groups.</h5>
<!--l. 1528--><p class="noindent"></p></div> <a id="colvar|distance"></a>
<!--l. 1531--><p class="indent">   The <span class="ectt-1095">distance {...} </span>block defines a distance component between the two atom groups, <span class="ectt-1095">group1 </span>and
<span class="ectt-1095">group2</span>.
</p><!--l. 1533--><p class="noindent"><span class="phvb8t-x-x-109">List of keywords </span>(see also <a href="#sec:cvc_pbcs">4.2<!--tex4ht:ref: sec:cvc_pbcs --></a>, <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>, <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> and <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="colvar|distance|group1"></a> <span class="colorbox" id="colorbox13"><a id="dx1-36001"></a><span class="ectt-1095">group1</span></span> — <span class="phvro8t-x-x-109">First group of atoms</span><br class="newline"/><!--l. 1541--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> <a href="#sec:colvar_atom_groups">Atom
     group</a>, context: <span class="ectt-1095">distance</span> <!--l. 1541--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>First group of atoms.
     </li>
<li class="itemize"><a id="colvar|distance|group2"></a> <span class="colorbox" id="colorbox14"><a id="dx1-36002"></a><span class="ectt-1095">group2</span></span> — analogous to <span class="ectt-1095">group1</span>
</li>
<li class="itemize"><a id="colvar|distance|oneSiteTotalForce"></a> <span class="colorbox" id="colorbox15"><a id="dx1-36003"></a><span class="ectt-1095">oneSiteTotalForce</span></span> — <span class="phvro8t-x-x-109">Measure total force on group 1 only?</span><br class="newline"/>Default: <span class="ectt-1095">no</span> <!--l. 1560--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">angle</span>, <span class="ectt-1095">dipoleAngle</span>, <span class="ectt-1095">dihedral</span> <!--l. 1560--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If this is set to <span class="ectt-1095">yes</span>, the total force is measured along a vector field (see equation (<a href="#eq:gradient_vector">28<!--tex4ht:ref: eq:gradient_vector --></a>) in
     section <a href="#sec:colvarbias_abf">6.3<!--tex4ht:ref: sec:colvarbias_abf --></a>) that only involves atoms of <span class="ectt-1095">group1</span>. This option is only useful for ABF, or custom
     biases that compute total forces. See section <a href="#sec:colvarbias_abf">6.3<!--tex4ht:ref: sec:colvarbias_abf --></a> for details.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox16"><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances? </span>(see <a href="#colvar|cvc|forceNoPBC">description</a>)<br class="newline"/>Default: <span class="ectt-1095">no</span></li></ul>
<!--l. 1564--><p class="indent">   The value returned is a positive number (in Å), ranging from
<!--l. 1564--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>0</mn></math> to the
largest possible interatomic distance within the chosen boundary conditions (with PBCs, the minimum
image convention is used unless the <span class="ectt-1095">forceNoPBC </span>option is set).
</p>
<div class="flushright">
<!--l. 1570--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1570--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_distanceZ"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.3.2    </span> <a id="sec:cvc_distanceZ"></a><span class="ectt-1095">distanceZ</span>: projection of a distance vector on an axis.</h5>
<!--l. 1570--><p class="noindent"></p></div> <a id="colvar|distanceZ"></a>
<!--l. 1573--><p class="indent">   The <span class="ectt-1095">distanceZ</span><span class="ectt-1095"> {...} </span>block defines a distance projection component, which can be seen as
measuring the distance between two groups projected onto an axis, or the position of a group along such
an axis. The axis can be defined using either one reference group and a constant vector, or dynamically
based on two reference groups. One of the groups can be set to a dummy atom to allow the use of an
absolute Cartesian coordinate.
</p><!--l. 1580--><p class="noindent"><span class="phvb8t-x-x-109">List of keywords </span>(see also <a href="#sec:cvc_pbcs">4.2<!--tex4ht:ref: sec:cvc_pbcs --></a>, <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>, <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> and <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="colvar|distanceZ|main"></a> <span class="colorbox" id="colorbox17"><a id="dx1-37001"></a><span class="ectt-1095">main</span></span> — <span class="phvro8t-x-x-109">Main group of atoms</span><br class="newline"/><!--l. 1588--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> <a href="#sec:colvar_atom_groups">Atom
     group</a>, context: <span class="ectt-1095">distanceZ</span> <!--l. 1588--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Group of atoms whose position <!--l. 1588--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>r</mi></mstyle><mtext></mtext></mstyle></math>
     is measured.
     </li>
<li class="itemize"><a id="colvar|distanceZ|ref"></a> <span class="colorbox" id="colorbox18"><a id="dx1-37002"></a><span class="ectt-1095">ref</span></span> — <span class="phvro8t-x-x-109">Reference group of atoms</span><br class="newline"/><!--l. 1599--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> <a href="#sec:colvar_atom_groups">Atom
     group</a>, context: <span class="ectt-1095">distanceZ</span> <!--l. 1599--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Reference group of atoms. The position of its center of mass is noted <!--l. 1599--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>r</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mn>1</mn></mrow></msub></math>
     below.
     </li>
<li class="itemize"><a id="colvar|distanceZ|ref2"></a> <span class="colorbox" id="colorbox19"><a id="dx1-37003"></a><span class="ectt-1095">ref2</span></span> — <span class="phvro8t-x-x-109">Secondary reference group</span><br class="newline"/>Default: none <!--l. 1614--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> <a href="#sec:colvar_atom_groups">Atom
     group</a>, context: <span class="ectt-1095">distanceZ</span> <!--l. 1614--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Optional group of reference atoms, whose position <!--l. 1614--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>r</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mn>2</mn></mrow></msub></math>
     can be used to define a dynamic projection axis: <!--l. 1614--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>e</mi></mstyle><mtext></mtext></mstyle> <mo class="MathClass-rel">=</mo> <msup><mrow><mo class="MathClass-open">(</mo><mo class="MathClass-rel">∥</mo><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>r</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mn>2</mn></mrow></msub> <mo class="MathClass-bin">−</mo><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>r</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-rel">∥</mo><mo class="MathClass-close">)</mo></mrow><mrow><mo class="MathClass-bin">−</mo><mn>1</mn></mrow></msup> <mo class="MathClass-bin">×</mo><mo class="MathClass-open">(</mo><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>r</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mn>2</mn></mrow></msub> <mo class="MathClass-bin">−</mo><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>r</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-close">)</mo></math>.
     In this case, the origin is <!--l. 1614--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>r</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mi>m</mi></mrow></msub> <mo class="MathClass-rel">=</mo> <mn>1</mn><mo class="MathClass-bin">∕</mo><mn>2</mn><mo class="MathClass-open">(</mo><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>r</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mn>1</mn></mrow></msub> <mo class="MathClass-bin">+</mo><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>r</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mn>2</mn></mrow></msub><mo class="MathClass-close">)</mo></math>,
     and the value of the component is <!--l. 1614--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>e</mi></mstyle><mtext></mtext></mstyle><mo class="MathClass-bin">⋅</mo><mo class="MathClass-open">(</mo><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>r</mi></mstyle><mtext></mtext></mstyle><mo class="MathClass-bin">−</mo><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>r</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mi>m</mi></mrow></msub><mo class="MathClass-close">)</mo></math>.
                                                                                         
                                                                                         
     </li>
<li class="itemize"><a id="colvar|distanceZ|axis"></a> <span class="colorbox" id="colorbox20"><a id="dx1-37004"></a><span class="ectt-1095">axis</span></span> — <span class="phvro8t-x-x-109">Projection axis</span><br class="newline"/>Default: <span class="ectt-1095">(0.0, 0.0, 1.0)</span> <!--l. 1629--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> <span class="ectt-1095">(x,</span>
<span class="ectt-1095">y, z) </span>triplet, context: <span class="ectt-1095">distanceZ</span> <!--l. 1629--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This vector will be normalized to define a projection axis <!--l. 1629--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>e</mi></mstyle><mtext></mtext></mstyle></math>
     for the distance vector <!--l. 1629--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>r</mi></mstyle><mtext></mtext></mstyle><mo class="MathClass-bin">−</mo><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>r</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mn>1</mn></mrow></msub></math>
     joining the centers of groups <span class="ectt-1095">ref </span>and <span class="ectt-1095">main</span>. The value of the component is then <!--l. 1629--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>e</mi></mstyle><mtext></mtext></mstyle><mo class="MathClass-bin">⋅</mo><mo class="MathClass-open">(</mo><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>r</mi></mstyle><mtext></mtext></mstyle><mo class="MathClass-bin">−</mo><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>r</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-close">)</mo></math>.
     The vector should be written as three components separated by commas and enclosed in
     parentheses.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox21"><a id="dx1-37005"></a><span class="ectt-1095">oneSiteTotalForce</span></span> — same definition as <a href="#colvar|distance|oneSiteTotalForce"><span class="ectt-1095">oneSiteTotalForce</span></a> (<span class="ectt-1095">distance </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox22"><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances? </span>(see <a href="#colvar|cvc|forceNoPBC">description</a>)<br class="newline"/>Default: <span class="ectt-1095">no</span></li></ul>
<!--l. 1634--><p class="noindent">This component returns a number (in Å) whose range is determined by the chosen boundary conditions. For
instance, if the <!--l. 1635--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math>
axis is used in a simulation with periodic boundaries, the returned value ranges between
<!--l. 1637--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-bin">−</mo><msub><mrow><mi>b</mi></mrow><mrow><mi>z</mi></mrow></msub><mo class="MathClass-bin">∕</mo><mn>2</mn></math> and
<!--l. 1637--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>b</mi></mrow><mrow><mi>z</mi></mrow></msub><mo class="MathClass-bin">∕</mo><mn>2</mn></math>, where
<!--l. 1637--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>b</mi></mrow><mrow><mi>z</mi></mrow></msub></math> is the box
length along <!--l. 1638--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math>
(this behavior is disabled if <span class="ectt-1095">forceNoPBC </span>is set).
</p>
<div class="flushright">
<!--l. 1641--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1641--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_distanceXY"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.3.3    </span> <a id="sec:cvc_distanceXY"></a><span class="ectt-1095">distanceXY</span>: modulus of the projection of a distance vector on a plane.</h5>
<!--l. 1641--><p class="noindent"></p></div> <a id="colvar|distanceXY"></a>
<!--l. 1644--><p class="indent">   The <span class="ectt-1095">distanceXY</span><span class="ectt-1095"> {...} </span>block defines a distance projected on a plane, and accepts the same
keywords as the component <span class="ectt-1095">distanceZ</span>, i.e. <span class="ectt-1095">main</span>, <span class="ectt-1095">ref</span>, either <span class="ectt-1095">ref2 </span>or <span class="ectt-1095">axis</span>, and <span class="ectt-1095">oneSiteTotalForce</span>. It
returns the norm of the projection of the distance vector between <span class="ectt-1095">main </span>and <span class="ectt-1095">ref </span>onto the plane orthogonal
to the axis. The axis is defined using the <span class="ectt-1095">axis </span>parameter or as the vector joining <span class="ectt-1095">ref </span>and <span class="ectt-1095">ref2 </span>(see
<span class="ectt-1095">distanceZ </span>above).
</p><!--l. 1653--><p class="noindent"><span class="phvb8t-x-x-109">List of keywords </span>(see also <a href="#sec:cvc_pbcs">4.2<!--tex4ht:ref: sec:cvc_pbcs --></a>, <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>, <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> and <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox23"><a id="dx1-38001"></a><span class="ectt-1095">main</span></span> — same definition as <a href="#colvar|distanceZ|main"><span class="ectt-1095">main</span></a> (<span class="ectt-1095">distanceZ </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox24"><a id="dx1-38002"></a><span class="ectt-1095">ref</span></span> — same definition as <a href="#colvar|distanceZ|ref"><span class="ectt-1095">ref</span></a> (<span class="ectt-1095">distanceZ </span>component)
                                                                                         
                                                                                         
     </li>
<li class="itemize"><span class="colorbox" id="colorbox25"><a id="dx1-38003"></a><span class="ectt-1095">ref2</span></span> — same definition as <a href="#colvar|distanceZ|ref2"><span class="ectt-1095">ref2</span></a> (<span class="ectt-1095">distanceZ </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox26"><a id="dx1-38004"></a><span class="ectt-1095">axis</span></span> — same definition as <a href="#colvar|distanceZ|axis"><span class="ectt-1095">axis</span></a> (<span class="ectt-1095">distanceZ </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox27"><a id="dx1-38005"></a><span class="ectt-1095">oneSiteTotalForce</span></span> — same definition as <a href="#colvar|distance|oneSiteTotalForce"><span class="ectt-1095">oneSiteTotalForce</span></a> (<span class="ectt-1095">distance </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox28"><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances? </span>(see <a href="#colvar|cvc|forceNoPBC">description</a>)<br class="newline"/>Default: <span class="ectt-1095">no</span></li></ul>
<div class="flushright">
<!--l. 1668--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1668--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_distanceVec"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.3.4    </span> <a id="sec:cvc_distanceVec"></a><span class="ectt-1095">distanceVec</span>: distance vector between two groups.</h5>
<!--l. 1668--><p class="noindent"></p></div> <a id="colvar|distanceVec"></a>
<!--l. 1671--><p class="indent">   The <span class="ectt-1095">distanceVec </span>component computes the 3-dimensional vector joining the centers of
mass of <span class="ectt-1095">group1 </span>and <span class="ectt-1095">group2</span>. Its values are therefore multi-dimensional and are subject to the
restrictions listed in <a href="#sec:cvc_non_scalar">4.18<!--tex4ht:ref: sec:cvc_non_scalar --></a>. Moreover, when computing differences between two different values of
a <span class="ectt-1095">distanceVec </span>variable the minimum-image convention is assumed (unless <a href="#colvar|cvc|forceNoPBC"><span class="ectt-1095">forceNoPBC</span></a> is
enabled).
</p><!--l. 1677--><p class="noindent"><span class="phvb8t-x-x-109">List of keywords </span>(see also <a href="#sec:cvc_pbcs">4.2<!--tex4ht:ref: sec:cvc_pbcs --></a>, <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>, <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> and <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox29"><a id="dx1-39001"></a><span class="ectt-1095">group1</span></span> — same definition as <a href="#colvar|distance|group1"><span class="ectt-1095">group1</span></a> (<span class="ectt-1095">distance </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox30"><a id="dx1-39002"></a><span class="ectt-1095">group2</span></span> — analogous to <span class="ectt-1095">group1</span>
</li>
<li class="itemize"><span class="colorbox" id="colorbox31"><a id="dx1-39003"></a><span class="ectt-1095">oneSiteTotalForce</span></span> — same definition as <a href="#colvar|distance|oneSiteTotalForce"><span class="ectt-1095">oneSiteTotalForce</span></a> (<span class="ectt-1095">distance </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox32"><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances? </span>(see <a href="#colvar|cvc|forceNoPBC">description</a>)<br class="newline"/>Default: <span class="ectt-1095">no</span></li></ul>
<div class="flushright">
<!--l. 1688--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1688--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_distanceDir"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.3.5    </span> <a id="sec:cvc_distanceDir"></a><span class="ectt-1095">distanceDir</span>: distance unit vector between two groups.</h5>
<!--l. 1688--><p class="noindent"></p></div> <a id="colvar|distanceDir"></a>
<!--l. 1691--><p class="indent">   The <span class="ectt-1095">distanceDir</span><span class="ectt-1095"> {...} </span>block defines a distance unit vector component, which accepts the same
keywords as the component <span class="ectt-1095">distance</span>: <span class="ectt-1095">group1</span>, <span class="ectt-1095">group2</span>, and <span class="ectt-1095">forceNoPBC</span>. It returns a 3-dimensional unit
vector <!--l. 1695--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="bold"><mi>d</mi></mstyle> <mo class="MathClass-rel">=</mo> <mo class="MathClass-open">(</mo><msub><mrow><mi>d</mi></mrow><mrow><mi>x</mi></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>d</mi></mrow><mrow><mi>y</mi></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>d</mi></mrow><mrow><mi>z</mi></mrow></msub><mo class="MathClass-close">)</mo></math>,
with <!--l. 1696--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-rel">|</mo><mstyle mathvariant="bold"><mi>d</mi></mstyle><mo class="MathClass-rel">|</mo> <mo class="MathClass-rel">=</mo> <mn>1</mn></math>.
</p><!--l. 1698--><p class="indent">   This multi-dimensional variable has two intrinsic degrees of freedom: however, these cannot be sampled
independently as one-dimensional variables. A decomposition in two dimensions can be done using
<a href="#colvar|polarTheta"><span class="ectt-1095">polarTheta</span></a> and <a href="#colvar|polarPhi"><span class="ectt-1095">polarPhi</span></a> angles, with the caveat that the latter is ill-defined when the former approaches
0<!--l. 1701--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></math> or
180<!--l. 1701--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></math>.
</p><!--l. 1703--><p class="indent">   The distance between two values of <span class="ectt-1095">distanceDir </span>is calculated internally as the angle (in radians)
between the two unit vectors: this definition adapts the standard Euclidean distance to the unit sphere, to
ensure that restraint forces comply with the mathematical constraint.
</p><!--l. 1708--><p class="noindent"><span class="phvb8t-x-x-109">List of keywords </span>(see also <a href="#sec:cvc_pbcs">4.2<!--tex4ht:ref: sec:cvc_pbcs --></a>, <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>, <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> and <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox33"><a id="dx1-40001"></a><span class="ectt-1095">group1</span></span> — same definition as <a href="#colvar|distance|group1"><span class="ectt-1095">group1</span></a> (<span class="ectt-1095">distance </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox34"><a id="dx1-40002"></a><span class="ectt-1095">group2</span></span> — analogous to <span class="ectt-1095">group1</span>
</li>
<li class="itemize"><span class="colorbox" id="colorbox35"><a id="dx1-40003"></a><span class="ectt-1095">oneSiteTotalForce</span></span> — same definition as <a href="#colvar|distance|oneSiteTotalForce"><span class="ectt-1095">oneSiteTotalForce</span></a> (<span class="ectt-1095">distance </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox36"><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances? </span>(see <a href="#colvar|cvc|forceNoPBC">description</a>)<br class="newline"/>Default: <span class="ectt-1095">no</span></li></ul>
<div class="flushright">
<!--l. 1718--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1718--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_distanceInv"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.3.6    </span> <a id="sec:cvc_distanceInv"></a><span class="ectt-1095">distanceInv</span>: mean distance between two groups of atoms.</h5>
<!--l. 1718--><p class="noindent"></p></div> <a id="colvar|distanceInv"></a>
<!--l. 1721--><p class="indent">   The <span class="ectt-1095">distanceInv</span><span class="ectt-1095"> {...} </span>block defines a generalized mean distance between
two groups of atoms 1 and 2, where each distance is taken to the power
<!--l. 1722--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-bin">−</mo><mi>n</mi></math>: </p><table class="equation"><tr><td>
<a id="x1-41001r2"></a>
<!--l. 1723--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<msubsup><mrow><mi>d</mi></mrow><mrow><mstyle mathvariant="normal"><mn>1</mn><mo class="MathClass-punc">,</mo><mn>2</mn></mstyle></mrow><mrow><mo class="MathClass-open">[</mo><mi>n</mi><mo class="MathClass-close">]</mo></mrow></msubsup><mspace class="thickpace" width="0.28em"></mspace> <mo class="MathClass-rel">=</mo> <mspace class="thickpace" width="0.28em"></mspace><msup><mrow> <mrow><mo fence="true" form="prefix"> (</mo><mrow> <mfrac><mrow><mn>1</mn></mrow>
<mrow><msub><mrow><mi>N</mi></mrow><mrow><mstyle mathvariant="normal"><mn>1</mn></mstyle></mrow></msub><msub><mrow><mi>N</mi></mrow><mrow><mstyle mathvariant="normal"><mn>2</mn></mstyle></mrow></msub></mrow></mfrac><munder class="msub"><mrow><mo>∑</mo></mrow><mrow><mi>i</mi><mo class="MathClass-punc">,</mo><mi>j</mi></mrow></munder><msubsup><mrow><mi>d</mi></mrow><mrow><mi>i</mi><mi>j</mi></mrow><mrow><mo class="MathClass-bin">−</mo><mi>n</mi></mrow></msubsup></mrow><mo fence="true" form="postfix">)</mo></mrow></mrow><mrow><mo class="MathClass-bin">−</mo><mn>1</mn><mo class="MathClass-bin">∕</mo><mi>n</mi></mrow></msup>
</math></td><td class="eq-no">(2)</td></tr></table>
<!--l. 1727--><p class="indent">   where <!--l. 1727--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>d</mi></mrow><mrow><mi>i</mi><mi>j</mi></mrow></msub></math> is the
distance between atoms <!--l. 1727--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math>
and <!--l. 1727--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>j</mi></math> in groups 1 and
2 respectively, and <!--l. 1727--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>n</mi></math>
is an even integer.
</p><!--l. 1729--><p class="noindent"><span class="phvb8t-x-x-109">List of keywords </span>(see also <a href="#sec:cvc_pbcs">4.2<!--tex4ht:ref: sec:cvc_pbcs --></a>, <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>, <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> and <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox37"><a id="dx1-41002"></a><span class="ectt-1095">group1</span></span> — same definition as <a href="#colvar|distance|group1"><span class="ectt-1095">group1</span></a> (<span class="ectt-1095">distance </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox38"><a id="dx1-41003"></a><span class="ectt-1095">group2</span></span> — analogous to <span class="ectt-1095">group1</span>
</li>
<li class="itemize"><span class="colorbox" id="colorbox39"><a id="dx1-41004"></a><span class="ectt-1095">oneSiteTotalForce</span></span> — same definition as <a href="#colvar|distance|oneSiteTotalForce"><span class="ectt-1095">oneSiteTotalForce</span></a> (<span class="ectt-1095">distance </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox40"><a id="dx1-41005"></a><span class="ectt-1095">exponent</span></span> — <span class="phvro8t-x-x-109">Exponent </span><!--l. 1742--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>n</mi></math>
<span class="phvro8t-x-x-109">in equation</span><span class="phvro8t-x-x-109"> </span><a href="#x1-41001r2"><span class="phvro8t-x-x-109">2</span><!--tex4ht:ref: eq:distanceInv --></a><br class="newline"/>Default: 6 <!--l. 1742--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     even integer, context: <span class="ectt-1095">distanceInv</span> <!--l. 1742--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Defines the exponent to which the individual distances are elevated before averaging. The
     default  value  of  6  is  useful  for  example  to  applying  restraints  based  on  NOE-measured
     distances.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox41"><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances? </span>(see <a href="#colvar|cvc|forceNoPBC">description</a>)<br class="newline"/>Default: <span class="ectt-1095">no</span></li></ul>
<!--l. 1744--><p class="noindent">This component returns a number ranging from <!--l. 1744--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>0</mn></math>
to the largest possible distance within the chosen boundary conditions.
</p>
<div class="flushright">
<!--l. 1747--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1747--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_angles"></a>
<h4 class="subsectionHead"><span class="titlemark">4.4    </span> <a id="sec:cvc_angles"></a>Angles</h4>
<!--l. 1747--><p class="noindent"></p></div>
<div class="flushright">
<!--l. 1750--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1750--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_angle"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.4.1    </span> <a id="sec:cvc_angle"></a><span class="ectt-1095">angle</span>: angle between three groups.</h5>
<!--l. 1750--><p class="noindent"></p></div> <a id="colvar|angle"></a>
<!--l. 1753--><p class="indent">   The <span class="ectt-1095">angle</span><span class="ectt-1095"> {...} </span>block defines an angle, and contains the three blocks <span class="ectt-1095">group1</span>, <span class="ectt-1095">group2</span>
and <span class="ectt-1095">group3</span>, defining the three groups. It returns an angle (in degrees) within the interval
<!--l. 1756--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo><mn>0</mn> <mo class="MathClass-punc">:</mo> <mn>1</mn><mn>8</mn><mn>0</mn><mo class="MathClass-close">]</mo></math>.
</p><!--l. 1758--><p class="noindent"><span class="phvb8t-x-x-109">List of keywords </span>(see also <a href="#sec:cvc_pbcs">4.2<!--tex4ht:ref: sec:cvc_pbcs --></a>, <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>, <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> and <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox42"><a id="dx1-43001"></a><span class="ectt-1095">group1</span></span> — same definition as <a href="#colvar|distance|group1"><span class="ectt-1095">group1</span></a> (<span class="ectt-1095">distance </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox43"><a id="dx1-43002"></a><span class="ectt-1095">group2</span></span> — analogous to <span class="ectt-1095">group1</span>
</li>
<li class="itemize"><span class="colorbox" id="colorbox44"><a id="dx1-43003"></a><span class="ectt-1095">group3</span></span> — analogous to <span class="ectt-1095">group1</span>
</li>
<li class="itemize"><span class="colorbox" id="colorbox45"><a id="dx1-43004"></a><span class="ectt-1095">oneSiteTotalForce</span></span> — same definition as <a href="#colvar|distance|oneSiteTotalForce"><span class="ectt-1095">oneSiteTotalForce</span></a> (<span class="ectt-1095">distance </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox46"><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances? </span>(see <a href="#colvar|cvc|forceNoPBC">description</a>)<br class="newline"/>Default: <span class="ectt-1095">no</span></li></ul>
<div class="flushright">
<!--l. 1771--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1771--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_dipoleAngle"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.4.2    </span> <a id="sec:cvc_dipoleAngle"></a><span class="ectt-1095">dipoleAngle</span>: angle between two groups and dipole of a third group.</h5>
<!--l. 1771--><p class="noindent"></p></div> <a id="colvar|dipoleAngle"></a>
<!--l. 1774--><p class="indent">   The <span class="ectt-1095">dipoleAngle</span><span class="ectt-1095"> {...} </span>block defines an angle, and contains the three blocks <span class="ectt-1095">group1</span>, <span class="ectt-1095">group2 </span>and <span class="ectt-1095">group3</span>,
defining the three groups, being <span class="ectt-1095">group1 </span>the group where dipole is calculated. It returns an angle (in degrees) within
the interval <!--l. 1777--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo><mn>0</mn> <mo class="MathClass-punc">:</mo> <mn>1</mn><mn>8</mn><mn>0</mn><mo class="MathClass-close">]</mo></math>.
</p><!--l. 1779--><p class="noindent"><span class="phvb8t-x-x-109">List of keywords </span>(see also <a href="#sec:cvc_pbcs">4.2<!--tex4ht:ref: sec:cvc_pbcs --></a>, <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>, <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> and <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox47"><a id="dx1-44001"></a><span class="ectt-1095">group1</span></span> — same definition as <a href="#colvar|distance|group1"><span class="ectt-1095">group1</span></a> (<span class="ectt-1095">distance </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox48"><a id="dx1-44002"></a><span class="ectt-1095">group2</span></span> — analogous to <span class="ectt-1095">group1</span>
</li>
<li class="itemize"><span class="colorbox" id="colorbox49"><a id="dx1-44003"></a><span class="ectt-1095">group3</span></span> — analogous to <span class="ectt-1095">group1</span>
</li>
<li class="itemize"><span class="colorbox" id="colorbox50"><a id="dx1-44004"></a><span class="ectt-1095">oneSiteTotalForce</span></span> — same definition as <a href="#colvar|distance|oneSiteTotalForce"><span class="ectt-1095">oneSiteTotalForce</span></a> (<span class="ectt-1095">distance </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox51"><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances? </span>(see <a href="#colvar|cvc|forceNoPBC">description</a>)<br class="newline"/>Default: <span class="ectt-1095">no</span></li></ul>
<div class="flushright">
<!--l. 1791--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1791--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_dihedral"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.4.3    </span> <a id="sec:cvc_dihedral"></a><span class="ectt-1095">dihedral</span>: torsional angle between four groups.</h5>
<!--l. 1791--><p class="noindent"></p></div> <a id="colvar|dihedral"></a>
<!--l. 1794--><p class="indent">   The <span class="ectt-1095">dihedral</span><span class="ectt-1095"> {...} </span>block defines a torsional angle, and contains the blocks <span class="ectt-1095">group1</span>, <span class="ectt-1095">group2</span>,
<span class="ectt-1095">group3 </span>and <span class="ectt-1095">group4</span>, defining the four groups. It returns an angle (in degrees) within the interval
<!--l. 1797--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo><mo class="MathClass-bin">−</mo><mn>1</mn><mn>8</mn><mn>0</mn> <mo class="MathClass-punc">:</mo> <mn>1</mn><mn>8</mn><mn>0</mn><mo class="MathClass-close">]</mo></math>. The
Colvars module calculates all the distances between two angles taking into account periodicity. For
instance, reference values for restraints or range boundaries can be defined by using any real number of
choice.
</p><!--l. 1802--><p class="noindent"><span class="phvb8t-x-x-109">List of keywords </span>(see also <a href="#sec:cvc_pbcs">4.2<!--tex4ht:ref: sec:cvc_pbcs --></a>, <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>, <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> and <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox52"><a id="dx1-45001"></a><span class="ectt-1095">group1</span></span> — same definition as <a href="#colvar|distance|group1"><span class="ectt-1095">group1</span></a> (<span class="ectt-1095">distance </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox53"><a id="dx1-45002"></a><span class="ectt-1095">group2</span></span> — analogous to <span class="ectt-1095">group1</span>
</li>
<li class="itemize"><span class="colorbox" id="colorbox54"><a id="dx1-45003"></a><span class="ectt-1095">group3</span></span> — analogous to <span class="ectt-1095">group1</span>
</li>
<li class="itemize"><span class="colorbox" id="colorbox55"><a id="dx1-45004"></a><span class="ectt-1095">group4</span></span> — analogous to <span class="ectt-1095">group1</span>
</li>
<li class="itemize"><span class="colorbox" id="colorbox56"><a id="dx1-45005"></a><span class="ectt-1095">oneSiteTotalForce</span></span> — same definition as <a href="#colvar|distance|oneSiteTotalForce"><span class="ectt-1095">oneSiteTotalForce</span></a> (<span class="ectt-1095">distance </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox57"><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances? </span>(see <a href="#colvar|cvc|forceNoPBC">description</a>)<br class="newline"/>Default: <span class="ectt-1095">no</span></li></ul>
<div class="flushright">
<!--l. 1816--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1816--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_polarTheta"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.4.4    </span> <a id="sec:cvc_polarTheta"></a><span class="ectt-1095">polarTheta</span>: polar angle in spherical coordinates.</h5>
<!--l. 1816--><p class="noindent"></p></div> <a id="colvar|polarTheta"></a>
<!--l. 1819--><p class="indent">   The <span class="ectt-1095">polarTheta</span><span class="ectt-1095"> {...} </span>block defines the polar angle in spherical coordinates, for the center of mass
of a group of atoms described by the block <span class="ectt-1095">atoms</span>. It returns an angle (in degrees) within the interval
<!--l. 1822--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo><mn>0</mn> <mo class="MathClass-punc">:</mo> <mn>1</mn><mn>8</mn><mn>0</mn><mo class="MathClass-close">]</mo></math>. To
obtain spherical coordinates in a frame of reference tied to another group of atoms, use the <span class="ectt-1095">fittingGroup</span>
(<a href="#sec:colvar_atom_groups_ref_frame">5.2<!--tex4ht:ref: sec:colvar_atom_groups_ref_frame --></a>) option within the <span class="ectt-1095">atoms </span>block. An example is provided in file <span class="ectt-1095">examples/11_polar_angles.in </span>of the
Colvars public repository.
</p><!--l. 1828--><p class="noindent"><span class="phvb8t-x-x-109">List of keywords </span>(see also <a href="#sec:cvc_pbcs">4.2<!--tex4ht:ref: sec:cvc_pbcs --></a>, <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>, <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> and <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="colvar|polarTheta|atoms"></a> <span class="colorbox" id="colorbox58"><a id="dx1-46001"></a><span class="ectt-1095">atoms</span></span> — <span class="phvro8t-x-x-109">Group of atoms defining this function</span><br class="newline"/><!--l. 1837--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> <a href="#sec:colvar_atom_groups">Atom
     group</a>, context: <span class="ectt-1095">polarPhi</span> <!--l. 1837--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Defines the group of atoms for the COM of which the angle should be calculated.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox59"><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances? </span>(see <a href="#colvar|cvc|forceNoPBC">description</a>)<br class="newline"/>Default: <span class="ectt-1095">no</span></li></ul>
<div class="flushright">
<!--l. 1841--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1841--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_polarPhi"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.4.5    </span> <a id="sec:cvc_polarPhi"></a><span class="ectt-1095">polarPhi</span>: azimuthal angle in spherical coordinates.</h5>
<!--l. 1841--><p class="noindent"></p></div> <a id="colvar|polarPhi"></a>
<!--l. 1844--><p class="indent">   The <span class="ectt-1095">polarPhi</span><span class="ectt-1095"> {...} </span>block defines the azimuthal angle in spherical coordinates, for the center of
mass of a group of atoms described by the block <span class="ectt-1095">atoms</span>. It returns an angle (in degrees) within the interval
<!--l. 1847--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo><mo class="MathClass-bin">−</mo><mn>1</mn><mn>8</mn><mn>0</mn> <mo class="MathClass-punc">:</mo> <mn>1</mn><mn>8</mn><mn>0</mn><mo class="MathClass-close">]</mo></math>. The
Colvars module calculates all the distances between two angles taking into account periodicity. For
instance, reference values for restraints or range boundaries can be defined by using any real number
of choice. To obtain spherical coordinates in a frame of reference tied to another group of
atoms, use the <span class="ectt-1095">fittingGroup </span>(<a href="#sec:colvar_atom_groups_ref_frame">5.2<!--tex4ht:ref: sec:colvar_atom_groups_ref_frame --></a>) option within the <span class="ectt-1095">atoms </span>block. An example is provided in file
<span class="ectt-1095">examples/11_polar_angles.in </span>of the Colvars public repository.
</p><!--l. 1857--><p class="noindent"><span class="phvb8t-x-x-109">List of keywords </span>(see also <a href="#sec:cvc_pbcs">4.2<!--tex4ht:ref: sec:cvc_pbcs --></a>, <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>, <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> and <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="colvar|polarPhi|atoms"></a> <span class="colorbox" id="colorbox60"><a id="dx1-47001"></a><span class="ectt-1095">atoms</span></span> — <span class="phvro8t-x-x-109">Group of atoms defining this function</span><br class="newline"/><!--l. 1866--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> <a href="#sec:colvar_atom_groups">Atom
     group</a>, context: <span class="ectt-1095">polarPhi</span> <!--l. 1866--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Defines the group of atoms for the COM of which the angle should be calculated.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox61"><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances? </span>(see <a href="#colvar|cvc|forceNoPBC">description</a>)<br class="newline"/>Default: <span class="ectt-1095">no</span></li></ul>
<!--l. 1869--><p class="indent"> <span class="phvb8t-x-x-109">Note: </span><span class="ectt-1095">polarPhi </span>is ill-defined when the corresponding <a href="#colvar|polarTheta"><span class="ectt-1095">polarTheta</span></a> component is close to
0<!--l. 1870--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></math> or
180<!--l. 1870--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></math>;
please take measures to avoid sampling these configurations in your simulations.
</p>
<div class="flushright">
<!--l. 1874--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1874--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_contacts"></a>
<h4 class="subsectionHead"><span class="titlemark">4.5    </span> <a id="sec:cvc_contacts"></a>Contacts</h4>
<!--l. 1874--><p class="noindent"></p></div>
<div class="flushright">
<!--l. 1877--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1877--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_coordNum"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.5.1    </span> <a id="sec:cvc_coordNum"></a><span class="ectt-1095">coordNum</span>: coordination number between two groups.</h5>
<!--l. 1877--><p class="noindent"></p></div> <a id="colvar|coordNum"></a>
<!--l. 1880--><p class="indent">   The <span class="ectt-1095">coordNum {...} </span>block defines a coordination number (or number of contacts), which calculates the
function <!--l. 1882--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">(</mo><mn>1</mn><mo class="MathClass-bin">−</mo><msup><mrow><mo class="MathClass-open">(</mo><mi>d</mi><mo class="MathClass-bin">∕</mo><msub><mrow><mi>d</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-close">)</mo></mrow><mrow><mi>n</mi></mrow></msup><mo class="MathClass-close">)</mo><mo class="MathClass-bin">∕</mo><mo class="MathClass-open">(</mo><mn>1</mn><mo class="MathClass-bin">−</mo><msup><mrow><mo class="MathClass-open">(</mo><mi>d</mi><mo class="MathClass-bin">∕</mo><msub><mrow><mi>d</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-close">)</mo></mrow><mrow><mi>m</mi></mrow></msup><mo class="MathClass-close">)</mo></math>, where
<!--l. 1882--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>d</mi></mrow><mrow><mn>0</mn></mrow></msub></math> is the “cutoff"
distance, and <!--l. 1883--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>n</mi></math>
and <!--l. 1883--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>m</mi></math> are
exponents that can control its long range behavior and stiffness <span class="cite">[<a href="#XIannuzzi2003">3</a>]</span>. This function is summed over all pairs
of atoms in <span class="ectt-1095">group1 </span>and <span class="ectt-1095">group2</span>: </p><table class="equation"><tr><td> <a id="eq:cvc_coordNum"></a>
<!--l. 1887--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>C</mi><mo class="MathClass-open">(</mo><mstyle mathvariant="monospace"><mi>g</mi><mi>r</mi><mi>o</mi><mi>u</mi><mi>p</mi><mi>1</mi></mstyle><mo class="MathClass-punc">,</mo><mstyle mathvariant="monospace"><mi>g</mi><mi>r</mi><mi>o</mi><mi>u</mi><mi>p</mi><mi>2</mi></mstyle><mo class="MathClass-close">)</mo><mspace class="thickpace" width="0.28em"></mspace> <mo class="MathClass-rel">=</mo> <mspace class="thickpace" width="0.28em"></mspace><munder class="msub"><mrow><mo>∑</mo></mrow><mrow><mi>i</mi><mo class="MathClass-rel">∈</mo><mstyle mathvariant="monospace"><mi>g</mi><mi>r</mi><mi>o</mi><mi>u</mi><mi>p</mi><mi>1</mi></mstyle></mrow></munder><munder class="msub"><mrow><mo>∑</mo></mrow><mrow><mi>j</mi><mo class="MathClass-rel">∈</mo><mstyle mathvariant="monospace"><mi>g</mi><mi>r</mi><mi>o</mi><mi>u</mi><mi>p</mi><mi>2</mi></mstyle></mrow></munder><mfrac><mrow><mn>1</mn><mo class="MathClass-bin">−</mo><msup><mrow><mo class="MathClass-open">(</mo><mo class="MathClass-rel">|</mo><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-bin">−</mo><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>j</mi></mrow></msub><mo class="MathClass-rel">|</mo><mo class="MathClass-bin">∕</mo><msub><mrow><mi>d</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-close">)</mo></mrow><mrow><mi>n</mi></mrow></msup></mrow>
<mrow><mn>1</mn><mo class="MathClass-bin">−</mo><msup><mrow><mo class="MathClass-open">(</mo><mo class="MathClass-rel">|</mo><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-bin">−</mo><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>j</mi></mrow></msub><mo class="MathClass-rel">|</mo><mo class="MathClass-bin">∕</mo><msub><mrow><mi>d</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-close">)</mo></mrow><mrow><mi>m</mi></mrow></msup></mrow></mfrac>
</math></td><td class="eq-no">(3)</td></tr></table>
<!--l. 1896--><p class="noindent"><span class="phvb8t-x-x-109">List of keywords </span>(see also <a href="#sec:cvc_pbcs">4.2<!--tex4ht:ref: sec:cvc_pbcs --></a>, <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>, <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> and <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="colvar|coordNum|group1"></a> <span class="colorbox" id="colorbox62"><a id="dx1-49002"></a><span class="ectt-1095">group1</span></span> — same definition as <a href="#colvar|distance|group1"><span class="ectt-1095">group1</span></a> (<span class="ectt-1095">distance </span>component)
     </li>
<li class="itemize"><a id="colvar|coordNum|group2"></a> <span class="colorbox" id="colorbox63"><a id="dx1-49003"></a><span class="ectt-1095">group2</span></span> — analogous to <span class="ectt-1095">group1</span>
</li>
<li class="itemize"><a id="colvar|coordNum|cutoff"></a> <span class="colorbox" id="colorbox64"><a id="dx1-49004"></a><span class="ectt-1095">cutoff</span></span> — <span class="phvro8t-x-x-109">“Interaction" distance (Å)</span><br class="newline"/>Default: 4.0 Å <!--l. 1919--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     decimal, context: <span class="ectt-1095">coordNum</span> <!--l. 1919--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This number defines the switching distance to define an interatomic contact: for <!--l. 1919--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>d</mi> <mo class="MathClass-rel">≪</mo> <msub><mrow><mi>d</mi></mrow><mrow><mn>0</mn></mrow></msub></math>,
     the switching function <!--l. 1919--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">(</mo><mn>1</mn><mo class="MathClass-bin">−</mo><msup><mrow><mo class="MathClass-open">(</mo><mi>d</mi><mo class="MathClass-bin">∕</mo><msub><mrow><mi>d</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-close">)</mo></mrow><mrow><mi>n</mi></mrow></msup><mo class="MathClass-close">)</mo><mo class="MathClass-bin">∕</mo><mo class="MathClass-open">(</mo><mn>1</mn><mo class="MathClass-bin">−</mo><msup><mrow><mo class="MathClass-open">(</mo><mi>d</mi><mo class="MathClass-bin">∕</mo><msub><mrow><mi>d</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-close">)</mo></mrow><mrow><mi>m</mi></mrow></msup><mo class="MathClass-close">)</mo></math>
     is close to 1, at <!--l. 1919--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>d</mi> <mo class="MathClass-rel">=</mo> <msub><mrow><mi>d</mi></mrow><mrow><mn>0</mn></mrow></msub></math>
     it has a value of <!--l. 1919--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>n</mi><mo class="MathClass-bin">∕</mo><mi>m</mi></math>
     (<!--l. 1919--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>1</mn><mo class="MathClass-bin">∕</mo><mn>2</mn></math>
     with the default <!--l. 1919--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>n</mi></math>
     and <!--l. 1919--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>m</mi></math>),
     and at <!--l. 1919--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>d</mi> <mo class="MathClass-rel">≫</mo> <msub><mrow><mi>d</mi></mrow><mrow><mn>0</mn></mrow></msub></math>
     it goes to zero approximately like <!--l. 1919--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow><mi>d</mi></mrow><mrow><mi>m</mi><mo class="MathClass-bin">−</mo><mi>n</mi></mrow></msup></math>.
     Hence, for a proper behavior, <!--l. 1919--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>m</mi></math>
     must be larger than <!--l. 1919--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>n</mi></math>.
     </li>
<li class="itemize"><a id="colvar|coordNum|cutoff3"></a> <span class="colorbox" id="colorbox65"><a id="dx1-49005"></a><span class="ectt-1095">cutoff3</span></span> — <span class="phvro8t-x-x-109">Reference distance vector (Å)</span><br class="newline"/>Default: <span class="ectt-1095">(4.0, 4.0, 4.0)</span><span class="ectt-1095"> Å</span> <!--l. 1931--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> “<span class="ectt-1095">(x,</span>
<span class="ectt-1095">y, z)</span>" triplet of positive decimals, context: <span class="ectt-1095">coordNum</span> <!--l. 1931--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>The three components of this vector define three different cutoffs <!--l. 1931--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>d</mi></mrow><mrow><mn>0</mn></mrow></msub></math>
     for each direction. This option is mutually exclusive with <span class="ectt-1095">cutoff</span>.
     </li>
<li class="itemize"><a id="colvar|coordNum|expNumer"></a> <span class="colorbox" id="colorbox66"><a id="dx1-49006"></a><span class="ectt-1095">expNumer</span></span> — <span class="phvro8t-x-x-109">Numerator exponent</span><br class="newline"/>Default: 6 <!--l. 1941--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     even integer, context: <span class="ectt-1095">coordNum</span> <!--l. 1941--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This number defines the <!--l. 1941--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>n</mi></math>
     exponent for the switching function.
     </li>
<li class="itemize"><a id="colvar|coordNum|expDenom"></a> <span class="colorbox" id="colorbox67"><a id="dx1-49007"></a><span class="ectt-1095">expDenom</span></span> — <span class="phvro8t-x-x-109">Denominator exponent</span><br class="newline"/>Default: 12 <!--l. 1951--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     even integer, context: <span class="ectt-1095">coordNum</span> <!--l. 1951--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This number defines the <!--l. 1951--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>m</mi></math>
     exponent for the switching function.
     </li>
<li class="itemize"><a id="colvar|coordNum|group2CenterOnly"></a> <span class="colorbox" id="colorbox68"><a id="dx1-49008"></a><span class="ectt-1095">group2CenterOnly</span></span> — <span class="phvro8t-x-x-109">Use only </span><span class="ecit-1095">group2</span><span class="phvro8t-x-x-109">'s center of mass</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 1964--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">coordNum</span> <!--l. 1964--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If this option is <span class="ectt-1095">on</span>, only contacts between each atoms in <span class="ectt-1095">group1 </span>and the center of mass of
     <span class="ectt-1095">group2 </span>are calculated (by default, the sum extends over all pairs of atoms in <span class="ectt-1095">group1 </span>and
     <span class="ectt-1095">group2</span>). If <span class="ectt-1095">group2 </span>is a <span class="ectt-1095">dummyAtom</span>, this option is set to <span class="ectt-1095">yes </span>by default.
     </li>
<li class="itemize"><a id="colvar|coordNum|tolerance"></a> <span class="colorbox" id="colorbox69"><a id="dx1-49009"></a><span class="ectt-1095">tolerance</span></span> — <span class="phvro8t-x-x-109">Pairlist control</span><br class="newline"/>Default: 0.0 <!--l. 1974--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> decimal,
     context: <span class="ectt-1095">coordNum</span> <!--l. 1974--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This controls the pair list feature, dictating the minimum value for each summation element
     in Eq. <a href="#eq:cvc_coordNum">3<!--tex4ht:ref: eq:cvc_coordNum --></a> such that the pair that contributed the summation element is included in subsequent
     simulation timesteps until the next pai r list recalculation. For most applications, this value
     should  be  small  (eg.  0.001)  to  avoid  missing  important  contributions  to  the  overall  sum.
     Higher values will improve performance by reducing the number of pairs that contribute to
     the sum. Values above 1 will exclude all possible pair interactions. Similarly, values below 0
     will never exclude a pair from consideration. To ensure continuous forces, Eq. <a href="#eq:cvc_coordNum">3<!--tex4ht:ref: eq:cvc_coordNum --></a> is further
     modified by subtracting the tolerance and then rescaling so that each pair covers the range
     <!--l. 1974--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mrow><mo fence="true" form="prefix"> [</mo><mrow><mn>0</mn><mo class="MathClass-punc">,</mo><mn>1</mn></mrow><mo fence="true" form="postfix">]</mo></mrow></math>.
     </li>
<li class="itemize"><a id="colvar|coordNum|pairListFrequency"></a> <span class="colorbox" id="colorbox70"><a id="dx1-49010"></a><span class="ectt-1095">pairListFrequency</span></span> — <span class="phvro8t-x-x-109">Pairlist regeneration frequency</span><br class="newline"/>Default: 100 <!--l. 1984--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     integer, context: <span class="ectt-1095">coordNum</span> <!--l. 1984--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This controls the pairlist feature, dictating how many steps are taken between regenerating
     pair lists if the tolerance is greater than 0.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox71"><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances? </span>(see <a href="#colvar|cvc|forceNoPBC">description</a>)<br class="newline"/>Default: <span class="ectt-1095">no</span></li></ul>
<!--l. 1987--><p class="indent">   This component returns a dimensionless number, which ranges from
approximately 0 (all interatomic distances are much larger than the cutoff) to
<!--l. 1989--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>N</mi></mrow><mrow><mstyle mathvariant="monospace"><mi>g</mi><mi>r</mi><mi>o</mi><mi>u</mi><mi>p</mi><mi>1</mi></mstyle></mrow></msub><mo class="MathClass-bin">×</mo><msub><mrow><mi>N</mi></mrow><mrow><mstyle mathvariant="monospace"><mi>g</mi><mi>r</mi><mi>o</mi><mi>u</mi><mi>p</mi><mi>2</mi></mstyle></mrow></msub></math> (all distances are less
than the cutoff), or <!--l. 1990--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>N</mi></mrow><mrow><mstyle mathvariant="monospace"><mi>g</mi><mi>r</mi><mi>o</mi><mi>u</mi><mi>p</mi><mi>1</mi></mstyle></mrow></msub></math>
if <span class="ectt-1095">group2CenterOnly </span>is used. For performance reasons, at least one of <span class="ectt-1095">group1 </span>and <span class="ectt-1095">group2 </span>should be of
limited size or <span class="ectt-1095">group2CenterOnly </span>should be used: the cost of the loop over all pairs grows as
<!--l. 1992--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>N</mi></mrow><mrow><mstyle mathvariant="monospace"><mi>g</mi><mi>r</mi><mi>o</mi><mi>u</mi><mi>p</mi><mi>1</mi></mstyle></mrow></msub><mo class="MathClass-bin">×</mo><msub><mrow><mi>N</mi></mrow><mrow><mstyle mathvariant="monospace"><mi>g</mi><mi>r</mi><mi>o</mi><mi>u</mi><mi>p</mi><mi>2</mi></mstyle></mrow></msub></math>. Setting
<!--l. 1993--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="monospace"><mi>t</mi><mi>o</mi><mi>l</mi><mi>e</mi><mi>r</mi><mi>a</mi><mi>n</mi><mi>c</mi><mi>e</mi></mstyle> <mo class="MathClass-rel">&gt;</mo> <mn>0</mn></math>
ameliorates this to some degree, although every pair is still checked to regenerate the pair
list.
</p>
<div class="flushright">
<!--l. 1997--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1997--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_selfCoordNum"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.5.2    </span> <a id="sec:cvc_selfCoordNum"></a><span class="ectt-1095">selfCoordNum</span>: coordination number between atoms within a group.</h5>
<!--l. 1997--><p class="noindent"></p></div> <a id="colvar|selfCoordNum"></a>
<!--l. 2000--><p class="indent">   The <span class="ectt-1095">selfCoordNum {...} </span>block defines a coordination number similarly to the component <span class="ectt-1095">coordNum</span>,
but the function is summed over atom pairs within <span class="ectt-1095">group1</span>: </p><table class="equation"><tr><td> <a id="x1-50001r4"></a>
<!--l. 2003--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>C</mi><mo class="MathClass-open">(</mo><mstyle mathvariant="monospace"><mi>g</mi><mi>r</mi><mi>o</mi><mi>u</mi><mi>p</mi><mi>1</mi></mstyle><mo class="MathClass-close">)</mo><mspace class="thickpace" width="0.28em"></mspace> <mo class="MathClass-rel">=</mo> <mspace class="thickpace" width="0.28em"></mspace><munder class="msub"><mrow><mo>∑</mo></mrow><mrow><mi>i</mi><mo class="MathClass-rel">∈</mo><mstyle mathvariant="monospace"><mi>g</mi><mi>r</mi><mi>o</mi><mi>u</mi><mi>p</mi><mi>1</mi></mstyle></mrow></munder><munder class="msub"><mrow><mo>∑</mo></mrow><mrow><mi>j</mi><mo class="MathClass-rel">&gt;</mo><mi>i</mi></mrow></munder> <mfrac><mrow><mn>1</mn><mo class="MathClass-bin">−</mo><msup><mrow><mo class="MathClass-open">(</mo><mo class="MathClass-rel">|</mo><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-bin">−</mo><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>j</mi></mrow></msub><mo class="MathClass-rel">|</mo><mo class="MathClass-bin">∕</mo><msub><mrow><mi>d</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-close">)</mo></mrow><mrow><mi>n</mi></mrow></msup></mrow>
<mrow><mn>1</mn><mo class="MathClass-bin">−</mo><msup><mrow><mo class="MathClass-open">(</mo><mo class="MathClass-rel">|</mo><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-bin">−</mo><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>j</mi></mrow></msub><mo class="MathClass-rel">|</mo><mo class="MathClass-bin">∕</mo><msub><mrow><mi>d</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-close">)</mo></mrow><mrow><mi>m</mi></mrow></msup></mrow></mfrac>
</math></td><td class="eq-no">(4)</td></tr></table>
<!--l. 2011--><p class="indent">   The keywords accepted by <span class="ectt-1095">selfCoordNum </span>are a subset of those accepted by <span class="ectt-1095">coordNum</span>, namely <span class="ectt-1095">group1</span>
(here defining <span class="phvro8t-x-x-109">all </span>of the atoms to be considered), <span class="ectt-1095">cutoff</span>, <span class="ectt-1095">expNumer</span>, and <span class="ectt-1095">expDenom</span>.
</p><!--l. 2016--><p class="noindent"><span class="phvb8t-x-x-109">List of keywords </span>(see also <a href="#sec:cvc_pbcs">4.2<!--tex4ht:ref: sec:cvc_pbcs --></a>, <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>, <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> and <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox72"><a id="dx1-50002"></a><span class="ectt-1095">group1</span></span> — same definition as <a href="#colvar|coordNum|group1"><span class="ectt-1095">group1</span></a> (<span class="ectt-1095">coordNum </span>component)
                                                                                         
                                                                                         
     </li>
<li class="itemize"><span class="colorbox" id="colorbox73"><a id="dx1-50003"></a><span class="ectt-1095">cutoff</span></span> — same definition as <a href="#colvar|coordNum|cutoff"><span class="ectt-1095">cutoff</span></a> (<span class="ectt-1095">coordNum </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox74"><a id="dx1-50004"></a><span class="ectt-1095">cutoff3</span></span> — same definition as <a href="#colvar|coordNum|cutoff3"><span class="ectt-1095">cutoff3</span></a> (<span class="ectt-1095">coordNum </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox75"><a id="dx1-50005"></a><span class="ectt-1095">expNumer</span></span> — same definition as <a href="#colvar|coordNum|expNumer"><span class="ectt-1095">expNumer</span></a> (<span class="ectt-1095">coordNum </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox76"><a id="dx1-50006"></a><span class="ectt-1095">expDenom</span></span> — same definition as <a href="#colvar|coordNum|expDenom"><span class="ectt-1095">expDenom</span></a> (<span class="ectt-1095">coordNum </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox77"><a id="dx1-50007"></a><span class="ectt-1095">tolerance</span></span> — same definition as <a href="#colvar|coordNum|tolerance"><span class="ectt-1095">tolerance</span></a> (<span class="ectt-1095">coordNum </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox78"><a id="dx1-50008"></a><span class="ectt-1095">pairListFrequency</span></span> — same definition as <a href="#colvar|coordNum|pairListFrequency"><span class="ectt-1095">pairListFrequency</span></a> (<span class="ectt-1095">coordNum </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox79"><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances? </span>(see <a href="#colvar|cvc|forceNoPBC">description</a>)<br class="newline"/>Default: <span class="ectt-1095">no</span></li></ul>
<!--l. 2033--><p class="indent">   This component returns a dimensionless number, which ranges from
approximately 0 (all interatomic distances much larger than the cutoff) to
<!--l. 2035--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>N</mi></mrow><mrow><mstyle mathvariant="monospace"><mi>g</mi><mi>r</mi><mi>o</mi><mi>u</mi><mi>p</mi><mi>1</mi></mstyle></mrow></msub><mo class="MathClass-bin">×</mo><mo class="MathClass-open">(</mo><msub><mrow><mi>N</mi></mrow><mrow><mstyle mathvariant="monospace"><mi>g</mi><mi>r</mi><mi>o</mi><mi>u</mi><mi>p</mi><mi>1</mi></mstyle></mrow></msub><mo class="MathClass-bin">−</mo><mn>1</mn><mo class="MathClass-close">)</mo><mo class="MathClass-bin">∕</mo><mn>2</mn></math> (all distances
within the cutoff). For performance reasons, <span class="ectt-1095">group1 </span>should be of limited size, because the cost of the loop over all
pairs grows as <!--l. 2038--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msubsup><mrow><mi>N</mi></mrow><mrow><mstyle mathvariant="monospace"><mi>g</mi><mi>r</mi><mi>o</mi><mi>u</mi><mi>p</mi><mi>1</mi></mstyle></mrow><mrow><mn>2</mn></mrow></msubsup></math>.
</p>
<div class="flushright">
<!--l. 2042--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2042--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_hBond"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.5.3    </span> <a id="sec:cvc_hBond"></a><span class="ectt-1095">hBond</span>: hydrogen bond between two atoms.</h5>
<!--l. 2042--><p class="noindent"></p></div> <a id="colvar|hBond"></a>
<!--l. 2045--><p class="indent">   The <span class="ectt-1095">hBond {...} </span>block defines a hydrogen bond, implemented as a coordination number (eq. <a href="#eq:cvc_coordNum">3<!--tex4ht:ref: eq:cvc_coordNum --></a>)
between the donor and the acceptor atoms. Therefore, it accepts the same options <span class="ectt-1095">cutoff</span>
(with a different default value of 3.3 Å), <span class="ectt-1095">expNumer </span>(with a default value of 6) and <span class="ectt-1095">expDenom</span>
(with a default value of 8). Unlike <span class="ectt-1095">coordNum</span>, it requires two atom numbers, <span class="ectt-1095">acceptor </span>and
<span class="ectt-1095">donor</span>, to be defined. It returns a dimensionless number, with values between 0 (acceptor
and donor far outside the cutoff distance) and 1 (acceptor and donor much closer than the
cutoff).
</p><!--l. 2056--><p class="noindent"><span class="phvb8t-x-x-109">List of keywords </span>(see also <a href="#sec:cvc_pbcs">4.2<!--tex4ht:ref: sec:cvc_pbcs --></a>, <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>, <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> and <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox80"><a id="dx1-51001"></a><span class="ectt-1095">acceptor</span></span> — <span class="phvro8t-x-x-109">Number of the acceptor atom</span><br class="newline"/><!--l. 2063--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     integer, context: <span class="ectt-1095">hBond</span> <!--l. 2063--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Number that uses the same convention as <span class="ectt-1095">atomNumbers</span>.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox81"><a id="dx1-51002"></a><span class="ectt-1095">donor</span></span> — analogous to <span class="ectt-1095">acceptor</span>
</li>
<li class="itemize"><span class="colorbox" id="colorbox82"><a id="dx1-51003"></a><span class="ectt-1095">cutoff</span></span> — same definition as <a href="#colvar|coordNum|cutoff"><span class="ectt-1095">cutoff</span></a> (<span class="ectt-1095">coordNum </span>component)<br class="newline"/><span class="phvb8t-x-x-109">Note: </span>default value is 3.3 Å.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox83"><a id="dx1-51004"></a><span class="ectt-1095">expNumer</span></span> — same definition as <a href="#colvar|coordNum|expNumer"><span class="ectt-1095">expNumer</span></a> (<span class="ectt-1095">coordNum </span>component)<br class="newline"/><span class="phvb8t-x-x-109">Note: </span>default value is 6.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox84"><a id="dx1-51005"></a><span class="ectt-1095">expDenom</span></span> — same definition as <a href="#colvar|coordNum|expDenom"><span class="ectt-1095">expDenom</span></a> (<span class="ectt-1095">coordNum </span>component)<br class="newline"/><span class="phvb8t-x-x-109">Note: </span>default value is 8.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox85"><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances? </span>(see <a href="#colvar|cvc|forceNoPBC">description</a>)<br class="newline"/>Default: <span class="ectt-1095">no</span></li></ul>
<div class="flushright">
<!--l. 2078--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2078--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_collective"></a>
<h4 class="subsectionHead"><span class="titlemark">4.6    </span> <a id="sec:cvc_collective"></a>Collective metrics</h4>
<!--l. 2078--><p class="noindent"></p></div>
<div class="flushright">
<!--l. 2081--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2081--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_rmsd"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.6.1    </span> <a id="sec:cvc_rmsd"></a><span class="ectt-1095">rmsd</span>: root mean square displacement (RMSD) from reference positions.</h5>
<!--l. 2081--><p class="noindent"></p></div> <a id="colvar|rmsd"></a>
<!--l. 2084--><p class="indent">   The block <span class="ectt-1095">rmsd</span><span class="ectt-1095"> {...} </span>defines the root mean square replacement (RMSD) of a
group of atoms with respect to a reference structure. For each set of coordinates
<!--l. 2086--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">{</mo><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo><mo class="MathClass-punc">,</mo><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mn>2</mn></mrow></msub><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo><mo class="MathClass-punc">,</mo><mi class="MathClass-op">…</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>N</mi></mrow></msub><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo><mo class="MathClass-close">}</mo></math>,
the colvar component <span class="ectt-1095">rmsd </span>calculates the optimal rotation
<!--l. 2089--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow><mi>U</mi></mrow><mrow><mo class="MathClass-open">{</mo><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo><mo class="MathClass-close">}</mo><mo class="MathClass-rel">→</mo><mo class="MathClass-open">{</mo><msubsup><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>i</mi></mrow><mrow><mstyle mathvariant="normal"><mo class="MathClass-open">(</mo><mi>r</mi><mi>e</mi><mi>f</mi><mo class="MathClass-close">)</mo></mstyle></mrow></msubsup><mo class="MathClass-close">}</mo></mrow></msup></math> that best superimposes
the coordinates <!--l. 2090--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">{</mo><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo><mo class="MathClass-close">}</mo></math> onto a set
of reference coordinates <!--l. 2091--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">{</mo><msubsup><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>i</mi></mrow><mrow><mstyle mathvariant="normal"><mo class="MathClass-open">(</mo><mi>r</mi><mi>e</mi><mi>f</mi><mo class="MathClass-close">)</mo></mstyle></mrow></msubsup><mo class="MathClass-close">}</mo></math>.
Both the current and the reference coordinates are centered on their centers of geometry,
<!--l. 2093--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mstyle mathvariant="normal"><mi>c</mi><mi>o</mi><mi>g</mi></mstyle></mrow></msub><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo></math> and
<!--l. 2094--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msubsup><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mstyle mathvariant="normal"><mi>c</mi><mi>o</mi><mi>g</mi></mstyle></mrow><mrow><mstyle mathvariant="normal"><mo class="MathClass-open">(</mo><mi>r</mi><mi>e</mi><mi>f</mi><mo class="MathClass-close">)</mo></mstyle></mrow></msubsup></math>. The
root mean square displacement is then defined as: </p><table class="equation"><tr><td> <a id="x1-53001r5"></a>
<!--l. 2096--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mstyle mathvariant="normal"><mi>R</mi><mi>M</mi><mi>S</mi><mi>D</mi></mstyle> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mrow><mo fence="true" form="prefix"> {</mo><mrow><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>i</mi></mrow></msub> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mi>t</mi></mrow><mo fence="true" form="postfix">)</mo></mrow></mrow><mo fence="true" form="postfix">}</mo></mrow><mo class="MathClass-punc">,</mo> <mrow><mo fence="true" form="prefix"> {</mo><mrow><msubsup><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>i</mi></mrow><mrow><mstyle mathvariant="normal"><mo class="MathClass-open">(</mo><mi>r</mi><mi>e</mi><mi>f</mi><mo class="MathClass-close">)</mo></mstyle></mrow></msubsup></mrow><mo fence="true" form="postfix">}</mo></mrow></mrow><mo fence="true" form="postfix">)</mo></mrow><mspace class="thickpace" width="0.28em"></mspace> <mo class="MathClass-rel">=</mo> <mspace class="thickpace" width="0.28em"></mspace><msqrt><mrow> <mfrac> <mrow> <mn>1</mn></mrow>
<mrow><mi>N</mi></mrow></mfrac><munderover accent="false" accentunder="false"><mrow><mo>∑</mo></mrow><mrow><mi>i</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><mi>N</mi></mrow></munderover><msup><mrow> <mrow><mo fence="true" form="prefix"> |</mo><mrow><mi>U</mi> <mrow><mo fence="true" form="prefix"> (</mo><mrow><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo><mo class="MathClass-bin">−</mo><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mstyle mathvariant="normal"><mi>c</mi><mi>o</mi><mi>g</mi></mstyle></mrow></msub><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo></mrow><mo fence="true" form="postfix">)</mo></mrow><mo class="MathClass-bin">−</mo><mrow><mo fence="true" form="prefix"> (</mo><mrow><msubsup><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>i</mi></mrow><mrow><mstyle mathvariant="normal"><mo class="MathClass-open">(</mo><mi>r</mi><mi>e</mi><mi>f</mi><mo class="MathClass-close">)</mo></mstyle></mrow></msubsup> <mo class="MathClass-bin">−</mo><msubsup><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mstyle mathvariant="normal"><mi>c</mi><mi>o</mi><mi>g</mi></mstyle></mrow><mrow><mstyle mathvariant="normal"><mo class="MathClass-open">(</mo><mi>r</mi><mi>e</mi><mi>f</mi><mo class="MathClass-close">)</mo></mstyle></mrow></msubsup></mrow><mo fence="true" form="postfix">)</mo></mrow></mrow><mo fence="true" form="postfix">|</mo></mrow></mrow><mrow><mn>2</mn></mrow></msup></mrow></msqrt>
</math></td><td class="eq-no">(5)</td></tr></table>
<!--l. 2106--><p class="indent">   The optimal rotation <!--l. 2107--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow><mi>U</mi></mrow><mrow><mo class="MathClass-open">{</mo><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo><mo class="MathClass-close">}</mo><mo class="MathClass-rel">→</mo><mo class="MathClass-open">{</mo><msubsup><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>i</mi></mrow><mrow><mstyle mathvariant="normal"><mo class="MathClass-open">(</mo><mi>r</mi><mi>e</mi><mi>f</mi><mo class="MathClass-close">)</mo></mstyle></mrow></msubsup><mo class="MathClass-close">}</mo></mrow></msup></math>
is calculated within the formalism developed in reference <span class="cite">[<a href="#XCoutsias2004">4</a>]</span>, which guarantees a continuous dependence
of <!--l. 2111--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow><mi>U</mi></mrow><mrow><mo class="MathClass-open">{</mo><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo><mo class="MathClass-close">}</mo><mo class="MathClass-rel">→</mo><mo class="MathClass-open">{</mo><msubsup><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>i</mi></mrow><mrow><mstyle mathvariant="normal"><mo class="MathClass-open">(</mo><mi>r</mi><mi>e</mi><mi>f</mi><mo class="MathClass-close">)</mo></mstyle></mrow></msubsup><mo class="MathClass-close">}</mo></mrow></msup></math> with
respect to <!--l. 2112--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">{</mo><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo><mo class="MathClass-close">}</mo></math>.
</p><!--l. 2114--><p class="noindent"><span class="phvb8t-x-x-109">List of keywords </span>(see also <a href="#sec:cvc_pbcs">4.2<!--tex4ht:ref: sec:cvc_pbcs --></a>, <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>, <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> and <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="colvar|rmsd|atoms"></a> <span class="colorbox" id="colorbox86"><a id="dx1-53002"></a><span class="ectt-1095">atoms</span></span> — <span class="phvro8t-x-x-109">Group of atoms defining this function</span><br class="newline"/><!--l. 2128--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> <a href="#sec:colvar_atom_groups">Atom
     group</a>, context: <span class="ectt-1095">rmsd</span> <!--l. 2128--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Defines the group of atoms of which the RMSD should be calculated. Optimal fit options
     (such  as  <span class="ectt-1095">refPositions </span>and  <span class="ectt-1095">rotateToReference</span>)  should  typically  NOT  be  set  within  this
     block. Exceptions to this rule are the special cases discussed in the <span class="phvro8t-x-x-109">Advanced usage </span>paragraph
     below.
     </li>
<li class="itemize"><a id="colvar|rmsd|refPositions"></a> <span class="colorbox" id="colorbox87"><a id="dx1-53003"></a><span class="ectt-1095">refPositions</span></span> — <span class="phvro8t-x-x-109">Reference coordinates</span><br class="newline"/><!--l. 2139--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> space-separated
     list of <span class="ectt-1095">(x, y, z) </span>triplets, context: <span class="ectt-1095">rmsd</span> <!--l. 2139--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option (mutually exclusive with <span class="ectt-1095">refPositionsFile</span>) sets the reference coordinates for
                                                                                         
                                                                                         
     RMSD calculation, and uses these to compute the roto-translational fit. See the equivalent
     option <a href="#atom-group|refPositions"><span class="ectt-1095">refPositions</span></a> within the atom group definition for details on acceptable formats and
     other features.
     </li>
<li class="itemize"><a id="colvar|rmsd|refPositionsFile"></a> <span class="colorbox" id="colorbox88"><a id="dx1-53004"></a><span class="ectt-1095">refPositionsFile</span></span> — <span class="phvro8t-x-x-109">Reference coordinates file</span><br class="newline"/><!--l. 2150--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> UNIX
     filename, context: <span class="ectt-1095">rmsd</span> <!--l. 2150--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option (mutually exclusive with <span class="ectt-1095">refPositions</span>) sets the reference coordinates for RMSD
     calculation, and uses these to compute the roto-translational fit. See the equivalent option
     <a href="#atom-group|refPositionsFile"><span class="ectt-1095">refPositionsFile</span></a> within the atom group definition for details on acceptable file formats and
     other features.
     </li>
<li class="itemize"><a id="colvar|rmsd|refPositionsCol"></a> <span class="colorbox" id="colorbox89"><a id="dx1-53005"></a><span class="ectt-1095">refPositionsCol</span></span> — <span class="phvro8t-x-x-109">PDB column containing atom flags</span><br class="newline"/><!--l. 2175--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> <span class="ectt-1095">O</span>,
     <span class="ectt-1095">B</span>, <span class="ectt-1095">X</span>, <span class="ectt-1095">Y</span>, or <span class="ectt-1095">Z</span>, context: <span class="ectt-1095">rmsd</span> <!--l. 2175--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If <span class="ectt-1095">refPositionsFile </span>is a PDB file that contains all the atoms in the topology, this option may
     be provided to set which PDB field is used to flag the reference coordinates for <span class="ectt-1095">atoms</span>.
     </li>
<li class="itemize"><a id="colvar|rmsd|refPositionsColValue"></a> <span class="colorbox" id="colorbox90"><a id="dx1-53006"></a><span class="ectt-1095">refPositionsColValue</span></span> — <span class="phvro8t-x-x-109">Atom selection flag in the PDB column</span><br class="newline"/><!--l. 2175--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     decimal, context: <span class="ectt-1095">rmsd</span> <!--l. 2175--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If defined, this value identifies in the PDB column <span class="ectt-1095">refPositionsCol </span>of the file <span class="ectt-1095">refPositionsFile</span>
     which atom positions are to be read. Otherwise, all positions with a non-zero value are read.
     </li>
<li class="itemize"><a id="colvar|rmsd|atomPermutation"></a> <span class="colorbox" id="colorbox91"><a id="dx1-53007"></a><span class="ectt-1095">atomPermutation</span></span> — <span class="phvro8t-x-x-109">Alternate ordering of atoms for RMSD computation</span><br class="newline"/><!--l. 2198--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> List
     of atom numbers, context: <span class="ectt-1095">rmsd</span> <!--l. 2198--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If defined, this parameter defines a re-ordering (permutation) of the 1-based atom numbers
     that can be used to compute the RMSD, typically due to molecular symmetry. This parameter
     can be specified multiple times, each one defining a new permutation: the returned RMSD
     value is the minimum over the set of permutations. For example, if the atoms making up
     the group are 6, 7, 8, 9, and atoms 7, 8, and 9 are invariant by circular permutation (as the
     hydrogens in a CH3 group), a symmetry-adapted RMSD would be obtained by adding:<br class="newline"/><span class="ectt-1095">atomPermutation 6 8 9 7</span><br class="newline"/><span class="ectt-1095">atomPermutation 6 9 7 8</span><br class="newline"/>This will result in these 2 atom orders being considered <span class="phvro8t-x-x-109">in addition to the order used when</span>
<span class="phvro8t-x-x-109">defining the atom group</span>. Note that this does not affect the least-squares roto-translational
     fit. Therefore, this feature is mostly useful when using custom fitting parameters within the
     atom group, such as <a href="#atom-group|fittingGroup"><span class="ectt-1095">fittingGroup</span></a>, or when fitting is disabled altogether. For details, see
     reference <span class="cite">[<a href="#XEbrahimi2022">5</a>]</span>.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox92"><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances? </span>(see <a href="#colvar|cvc|forceNoPBC">description</a>)<br class="newline"/>Default: <span class="ectt-1095">no</span></li></ul>
<!--l. 2200--><p class="noindent">This component returns a positive real number (in Å).
</p>
<div class="flushright">
<!--l. 2203--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2203--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_rmsd_advanced"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.6.2    </span> <a id="sec:cvc_rmsd_advanced"></a>Advanced usage of the <span class="ectt-1095">rmsd </span>component.</h5>
<!--l. 2203--><p class="noindent"></p></div> In the standard usage as described above, the <span class="ectt-1095">rmsd </span>component calculates a minimum RMSD, that is,
current coordinates are optimally fitted onto the same reference coordinates that are used to compute the
RMSD value. The fit itself is handled by the atom group object, whose parameters are automatically set
by the <span class="ectt-1095">rmsd </span>component. For very specific applications, however, it may be useful to control
the fitting process separately from the definition of the reference coordinates, to evaluate
various types of non-minimal RMSD values. This can be achieved by setting the related options
(<span class="ectt-1095">refPositions</span>, etc.) explicitly in the atom group block. This allows for the following non-standard
cases:
<!--l. 2217--><p class="indent">
</p><ol class="enumerate1">
<li class="enumerate" id="x1-54002x1">applying the optimal translation, but no rotation (<span class="ectt-1095">rotateToReference off</span>), to bias or restrain
     the shape and orientation, but not the position of the atom group;
     </li>
<li class="enumerate" id="x1-54004x2">applying the optimal rotation, but no translation (<span class="ectt-1095">centerToReference off</span>), to bias or restrain
     the shape and position, but not the orientation of the atom group;
     </li>
<li class="enumerate" id="x1-54006x3">disabling  the  application  of  optimal  roto-translations,  which  lets  the  RMSD  component
     describe the deviation of atoms from fixed positions in the laboratory frame: this allows for
     custom positional restraints within the Colvars module;
     </li>
<li class="enumerate" id="x1-54008x4">fitting the atomic positions to different reference coordinates than those used in the RMSD
     calculation itself (by specifying <a href="#atom-group|refPositions"><span class="ectt-1095">refPositions</span></a> or <a href="#atom-group|refPositionsFile"><span class="ectt-1095">refPositionsFile</span></a> within the atom group as
     well as within the <span class="ectt-1095">rmsd </span>block);
     </li>
<li class="enumerate" id="x1-54010x5">applying the optimal rotation and/or translation from a separate atom group, defined through
     <span class="ectt-1095">fittingGroup</span>:  the  RMSD  then  reflects  the  deviation  from  reference  coordinates  in  a
     separate, moving reference frame (see example in the section on <a href="#atom-group|fittingGroup"><span class="ectt-1095">fittingGroup</span></a>).</li></ol>
<div class="flushright">
<!--l. 2240--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2240--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_eigenvector"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.6.3    </span> <a id="sec:cvc_eigenvector"></a><span class="ectt-1095">eigenvector</span>: projection of the atomic coordinates on a vector.</h5>
<!--l. 2240--><p class="noindent"></p></div> <a id="colvar|eigenvector"></a>
<!--l. 2243--><p class="indent">   The block <span class="ectt-1095">eigenvector</span><span class="ectt-1095"> {...} </span>defines the projection of the coordinates of a group of
atoms (or more precisely, their deviations from the reference coordinates) onto a vector in
<!--l. 2245--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow><mi>ℝ</mi></mrow><mrow><mn>3</mn><mi>n</mi></mrow></msup></math>, where
<!--l. 2245--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>n</mi></math> is the
number of atoms in the group. The computed quantity is the total projection: </p><table class="equation"><tr><td> <a id="x1-55001r6"></a>
<!--l. 2248--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>p</mi> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mrow><mo fence="true" form="prefix"> {</mo><mrow><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>i</mi></mrow></msub> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mi>t</mi></mrow><mo fence="true" form="postfix">)</mo></mrow></mrow><mo fence="true" form="postfix">}</mo></mrow><mo class="MathClass-punc">,</mo> <mrow><mo fence="true" form="prefix"> {</mo><mrow><msubsup><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>i</mi></mrow><mrow><mstyle mathvariant="normal"><mo class="MathClass-open">(</mo><mi>r</mi><mi>e</mi><mi>f</mi><mo class="MathClass-close">)</mo></mstyle></mrow></msubsup></mrow><mo fence="true" form="postfix">}</mo></mrow></mrow><mo fence="true" form="postfix">)</mo></mrow><mspace class="thickpace" width="0.28em"></mspace> <mo class="MathClass-rel">=</mo> <mspace class="thickpace" width="0.28em"></mspace><munderover accent="false" accentunder="false"><mrow><mo>∑</mo></mrow><mrow>
<mi>i</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><mi>n</mi></mrow></munderover><msub><mrow><mstyle mathvariant="bold"><mi>v</mi></mstyle></mrow><mrow>
<mi>i</mi></mrow></msub><mo class="MathClass-bin">⋅</mo><mrow><mo fence="true" form="prefix"> (</mo><mrow><mi>U</mi><mo class="MathClass-open">(</mo><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo><mo class="MathClass-bin">−</mo><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mstyle mathvariant="normal"><mi>c</mi><mi>o</mi><mi>g</mi></mstyle></mrow></msub><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo><mo class="MathClass-close">)</mo><mo class="MathClass-bin">−</mo><mo class="MathClass-open">(</mo><msubsup><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>i</mi></mrow><mrow><mstyle mathvariant="normal"><mo class="MathClass-open">(</mo><mi>r</mi><mi>e</mi><mi>f</mi><mo class="MathClass-close">)</mo></mstyle></mrow></msubsup> <mo class="MathClass-bin">−</mo><msubsup><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mstyle mathvariant="normal">
<mi>c</mi><mi>o</mi><mi>g</mi></mstyle></mrow><mrow><mstyle mathvariant="normal"><mo class="MathClass-open">(</mo><mi>r</mi><mi>e</mi><mi>f</mi><mo class="MathClass-close">)</mo></mstyle></mrow></msubsup><mo class="MathClass-close">)</mo></mrow><mo fence="true" form="postfix">)</mo></mrow><mstyle mathvariant="normal"><mo class="MathClass-punc">,</mo></mstyle>
</math></td><td class="eq-no">(6)</td></tr></table>
<!--l. 2257--><p class="indent">   where, as in the <span class="ectt-1095">rmsd </span>component, <!--l. 2257--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>U</mi></math>
is the optimal rotation matrix, <!--l. 2258--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mstyle mathvariant="normal"><mi>c</mi><mi>o</mi><mi>g</mi></mstyle></mrow></msub><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo></math>
and <!--l. 2259--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msubsup><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mstyle mathvariant="normal"><mi>c</mi><mi>o</mi><mi>g</mi></mstyle></mrow><mrow><mstyle mathvariant="normal"><mo class="MathClass-open">(</mo><mi>r</mi><mi>e</mi><mi>f</mi><mo class="MathClass-close">)</mo></mstyle></mrow></msubsup></math>
are the centers of geometry of the current and reference positions respectively, and
<!--l. 2261--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>v</mi></mstyle></mrow><mrow><mi>i</mi></mrow></msub></math>
are the components of the vector for each atom. Example choices for
<!--l. 2262--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">(</mo><msub><mrow><mstyle mathvariant="bold"><mi>v</mi></mstyle></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-close">)</mo></math> are an
eigenvector of the covariance matrix (essential mode), or a normal mode of the system. It is assumed that
<!--l. 2264--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi class="MathClass-op">∑</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo><!--nolimits--></mrow><mrow><mi>i</mi></mrow></msub><msub><mrow><mstyle mathvariant="bold"><mi>v</mi></mstyle></mrow><mrow><mi>i</mi></mrow></msub> <mo class="MathClass-rel">=</mo> <mn>0</mn></math>: otherwise, the Colvars
module centers the <!--l. 2265--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>v</mi></mstyle></mrow><mrow><mi>i</mi></mrow></msub></math>
automatically when reading them from the configuration.
</p><!--l. 2268--><p class="noindent"><span class="phvb8t-x-x-109">List of keywords </span>(see also <a href="#sec:cvc_pbcs">4.2<!--tex4ht:ref: sec:cvc_pbcs --></a>, <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>, <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> and <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox93"><a id="dx1-55002"></a><span class="ectt-1095">atoms</span></span> — same definition as <a href="#colvar|rmsd|atoms"><span class="ectt-1095">atoms</span></a> (<span class="ectt-1095">rmsd </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox94"><a id="dx1-55003"></a><span class="ectt-1095">refPositions</span></span> — same definition as <a href="#colvar|rmsd|refPositions"><span class="ectt-1095">refPositions</span></a> (<span class="ectt-1095">rmsd </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox95"><a id="dx1-55004"></a><span class="ectt-1095">refPositionsFile</span></span> — same definition as <a href="#colvar|rmsd|refPositionsFile"><span class="ectt-1095">refPositionsFile</span></a> (<span class="ectt-1095">rmsd </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox96"><a id="dx1-55005"></a><span class="ectt-1095">refPositionsCol</span></span> — same definition as <a href="#colvar|rmsd|refPositionsCol"><span class="ectt-1095">refPositionsCol</span></a> (<span class="ectt-1095">rmsd </span>component)
                                                                                         
                                                                                         
     </li>
<li class="itemize"><span class="colorbox" id="colorbox97"><a id="dx1-55006"></a><span class="ectt-1095">refPositionsColValue</span></span> — same definition as <a href="#colvar|rmsd|refPositionsColValue"><span class="ectt-1095">refPositionsColValue</span></a> (<span class="ectt-1095">rmsd </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox98"><a id="dx1-55007"></a><span class="ectt-1095">vector</span></span> — <span class="phvro8t-x-x-109">Vector components</span><br class="newline"/><!--l. 2288--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> space-separated
     list of <span class="ectt-1095">(x, y, z) </span>triplets, context: <span class="ectt-1095">eigenvector</span> <!--l. 2288--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option (mutually exclusive with <span class="ectt-1095">vectorFile</span>) sets the values of the vector components.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox99"><a id="dx1-55008"></a><span class="ectt-1095">vectorFile</span></span> — <span class="phvro8t-x-x-109">file containing vector components</span><br class="newline"/><!--l. 2300--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> UNIX
     filename, context: <span class="ectt-1095">eigenvector</span> <!--l. 2300--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This  option  (mutually  exclusive  with  <span class="ectt-1095">vector</span>)  sets  the  name  of  an  XYZ  (<a href="#sec:colvars_xyz_format">3.8.3<!--tex4ht:ref: sec:colvars_xyz_format --></a>)   or  PDB
     coordinate file containing the vector components. For a PDB file specifically, the components
     are read from the X, Y and Z fields. <span class="phvb8t-x-x-109">Note: </span><span class="phvro8t-x-x-109">The PDB file has limited precision and fixed-point</span>
<span class="phvro8t-x-x-109">numbers: in some cases, the vector components may not be accurately represented; a XYZ</span>
<span class="phvro8t-x-x-109">file should be used instead, containing floating-point numbers. </span><span class="phvb8t-x-x-109">Note: </span><span class="phvro8t-x-x-109">Reading data from a</span>
<span class="phvro8t-x-x-109">coordinate file may entail an automatic unit conversion if the length unit currently used by</span>
<span class="phvro8t-x-x-109">the MD engine is not Å. If this is not the desired behavior, this can be remedied using the</span>
<span class="ecit-1095">normalizeVector </span><span class="phvro8t-x-x-109">option described below.</span>
</li>
<li class="itemize"><span class="colorbox" id="colorbox100"><a id="dx1-55009"></a><span class="ectt-1095">vectorCol</span></span> — <span class="phvro8t-x-x-109">PDB column used to flag participating atoms</span><br class="newline"/><!--l. 2318--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> <span class="ectt-1095">O</span>
     or <span class="ectt-1095">B</span>, context: <span class="ectt-1095">eigenvector</span> <!--l. 2318--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Analogous to <span class="ectt-1095">atomsCol</span>.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox101"><a id="dx1-55010"></a><span class="ectt-1095">vectorColValue</span></span> — <span class="phvro8t-x-x-109">Value used to flag participating atoms in the PDB file</span><br class="newline"/><!--l. 2318--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     decimal, context: <span class="ectt-1095">eigenvector</span> <!--l. 2318--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Analogous to <span class="ectt-1095">atomsColValue</span>.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox102"><a id="dx1-55011"></a><span class="ectt-1095">normalizeVector</span></span> — <span class="phvro8t-x-x-109">Normalize the vector components when reading them?</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 2328--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">eigenvector</span> <!--l. 2328--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If this flag is activated, the norm of the vector <!--l. 2328--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-rel">|</mo><mstyle mathvariant="bold"><mi>v</mi></mstyle><mo class="MathClass-rel">|</mo> <mo class="MathClass-rel">=</mo> <msqrt><mrow><msub><mrow><mi class="MathClass-op">∑</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo> <!--nolimits--></mrow><mrow><mi>i</mi> </mrow> </msub><msup><mrow> <mrow><mo fence="true" form="prefix"> |</mo><mrow><msub><mrow><mstyle mathvariant="bold"><mi>v</mi></mstyle> </mrow><mrow><mi>i</mi> </mrow> </msub> </mrow><mo fence="true" form="postfix">|</mo></mrow></mrow><mrow><mn>2</mn></mrow></msup></mrow></msqrt></math>
     is set equal to 1 by automatically rescaling all the components <!--l. 2328--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>v</mi></mstyle></mrow><mrow><mi>i</mi></mrow></msub></math>;
     alternatively, the value of <!--l. 2328--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-rel">|</mo><mstyle mathvariant="bold"><mi>v</mi></mstyle><mo class="MathClass-rel">|</mo></math>
     is printed.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox103"><a id="dx1-55012"></a><span class="ectt-1095">differenceVector</span></span> — <span class="phvro8t-x-x-109">The </span><!--l. 2340--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>3</mn><mi>n</mi></math><span class="phvro8t-x-x-109">-dimensional</span>
<span class="phvro8t-x-x-109">vector is the difference between </span><span class="ecit-1095">vector </span><span class="phvro8t-x-x-109">and </span><span class="ecit-1095">refPositions</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 2340--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">eigenvector</span> <!--l. 2340--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If this option is <span class="ectt-1095">on</span>, the numbers provided by <span class="ectt-1095">vector </span>or <span class="ectt-1095">vectorFile </span>are interpreted as another
     set of positions, <!--l. 2340--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msubsup><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>i</mi></mrow><mrow><mi>′</mi></mrow></msubsup></math>:
     the vector <!--l. 2340--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>v</mi></mstyle></mrow><mrow><mi>i</mi></mrow></msub></math>
     is then defined as <!--l. 2340--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>v</mi></mstyle></mrow><mrow><mi>i</mi></mrow></msub> <mo class="MathClass-rel">=</mo> <mrow><mo fence="true" form="prefix"> (</mo><mrow><msubsup><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>i</mi></mrow><mrow><mi>′</mi></mrow></msubsup><mo class="MathClass-bin">−</mo><msubsup><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>i</mi></mrow><mrow><mstyle mathvariant="normal"><mo class="MathClass-open">(</mo><mi>r</mi><mi>e</mi><mi>f</mi><mo class="MathClass-close">)</mo></mstyle></mrow></msubsup></mrow><mo fence="true" form="postfix">)</mo></mrow></math>.
     This allows to conveniently define a colvar <!--l. 2340--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>
     as a projection on the linear transformation between two sets of positions, “A" and “B". If this
     flag is <span class="ectt-1095">on</span>, the vector is normalized so that <!--l. 2340--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi> <mo class="MathClass-rel">=</mo> <mn>0</mn></math>
     when the atoms are at the set of positions “A" and <!--l. 2340--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi> <mo class="MathClass-rel">=</mo> <mn>1</mn></math>
     at the set of positions “B". Setting <span class="ectt-1095">normalizeVector on </span>overrides this behavior.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox104"><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances? </span>(see <a href="#colvar|cvc|forceNoPBC">description</a>)<br class="newline"/>Default: <span class="ectt-1095">no</span></li></ul>
<div class="flushright">
<!--l. 2344--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2344--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_gyration"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.6.4    </span> <a id="sec:cvc_gyration"></a><span class="ectt-1095">gyration</span>: radius of gyration of a group of atoms.</h5>
<!--l. 2344--><p class="noindent"></p></div> <a id="colvar|gyration"></a>
<!--l. 2347--><p class="indent">   The block <span class="ectt-1095">gyration</span><span class="ectt-1095"> {...} </span>defines the parameters for calculating the radius of gyration of a group of atomic positions
<!--l. 2349--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">{</mo><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo><mo class="MathClass-punc">,</mo><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mn>2</mn></mrow></msub><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo><mo class="MathClass-punc">,</mo><mi class="MathClass-op">…</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>N</mi></mrow></msub><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo><mo class="MathClass-close">}</mo></math> with respect to their
center of geometry, <!--l. 2351--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mstyle mathvariant="normal"><mi>c</mi><mi>o</mi><mi>g</mi></mstyle></mrow></msub><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo></math>:
</p><table class="equation"><tr><td> <a id="x1-56001r7"></a>
<!--l. 2352--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<msub><mrow><mi>R</mi></mrow><mrow><mstyle mathvariant="normal"><mi>g</mi><mi>y</mi><mi>r</mi></mstyle></mrow></msub><mspace class="thickpace" width="0.28em"></mspace> <mo class="MathClass-rel">=</mo> <mspace class="thickpace" width="0.28em"></mspace><msqrt><mrow> <mfrac> <mrow> <mn>1</mn></mrow>
<mrow><mi>N</mi></mrow></mfrac><munderover accent="false" accentunder="false"><mrow><mo>∑</mo></mrow><mrow><mi>i</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><mi>N</mi></mrow></munderover><msup><mrow> <mrow><mo fence="true" form="prefix"> |</mo><mrow><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo><mo class="MathClass-bin">−</mo><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mstyle mathvariant="normal"><mi>c</mi><mi>o</mi><mi>g</mi></mstyle></mrow></msub><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo></mrow><mo fence="true" form="postfix">|</mo></mrow></mrow><mrow><mn>2</mn></mrow></msup></mrow></msqrt>
</math></td><td class="eq-no">(7)</td></tr></table>
<!--l. 2358--><p class="indent">   This component must contain one <span class="ectt-1095">atoms</span><span class="ectt-1095"> {...} </span>block to define the atom group, and returns a positive
number, expressed in Å.
</p><!--l. 2362--><p class="noindent"><span class="phvb8t-x-x-109">List of keywords </span>(see also <a href="#sec:cvc_pbcs">4.2<!--tex4ht:ref: sec:cvc_pbcs --></a>, <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>, <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> and <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox105"><a id="dx1-56002"></a><span class="ectt-1095">atoms</span></span> — same definition as <a href="#colvar|rmsd|atoms"><span class="ectt-1095">atoms</span></a> (<span class="ectt-1095">rmsd </span>component)
                                                                                         
                                                                                         
     </li>
<li class="itemize"><span class="colorbox" id="colorbox106"><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances? </span>(see <a href="#colvar|cvc|forceNoPBC">description</a>)<br class="newline"/>Default: <span class="ectt-1095">no</span></li></ul>
<div class="flushright">
<!--l. 2368--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2368--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_inertia"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.6.5    </span> <a id="sec:cvc_inertia"></a><span class="ectt-1095">inertia</span>: total moment of inertia of a group of atoms.</h5>
<!--l. 2368--><p class="noindent"></p></div> <a id="colvar|inertia"></a>
<!--l. 2371--><p class="indent">   The block <span class="ectt-1095">inertia</span><span class="ectt-1095"> {...} </span>defines the parameters for calculating the total moment of inertia of a group of atomic positions
<!--l. 2373--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">{</mo><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo><mo class="MathClass-punc">,</mo><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mn>2</mn></mrow></msub><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo><mo class="MathClass-punc">,</mo><mi class="MathClass-op">…</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>N</mi></mrow></msub><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo><mo class="MathClass-close">}</mo></math> with respect to their
center of geometry, <!--l. 2375--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mstyle mathvariant="normal"><mi>c</mi><mi>o</mi><mi>g</mi></mstyle></mrow></msub><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo></math>:
</p><table class="equation"><tr><td> <a id="x1-57001r8"></a>
<!--l. 2376--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>I</mi><mspace class="thickpace" width="0.28em"></mspace> <mo class="MathClass-rel">=</mo> <mspace class="thickpace" width="0.28em"></mspace><munderover accent="false" accentunder="false"><mrow><mo>∑</mo></mrow><mrow><mi>i</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><mi>N</mi></mrow></munderover><msup><mrow> <mrow><mo fence="true" form="prefix"> |</mo><mrow><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow>
<mi>i</mi></mrow></msub><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo><mo class="MathClass-bin">−</mo><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mstyle mathvariant="normal"><mi>c</mi><mi>o</mi><mi>g</mi></mstyle></mrow></msub><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo></mrow><mo fence="true" form="postfix">|</mo></mrow></mrow><mrow><mn>2</mn></mrow></msup>
</math></td><td class="eq-no">(8)</td></tr></table>
<!--l. 2381--><p class="indent"> <span class="phvro8t-x-x-109">Note that all atomic masses are set to 1 for simplicity. </span>This component must contain one
<span class="ectt-1095">atoms</span><span class="ectt-1095"> {...} </span>block to define the atom group, and returns a positive number, expressed in
Å<!--l. 2384--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow></mrow><mrow><mn>2</mn></mrow></msup></math>.
</p><!--l. 2386--><p class="noindent"><span class="phvb8t-x-x-109">List of keywords </span>(see also <a href="#sec:cvc_pbcs">4.2<!--tex4ht:ref: sec:cvc_pbcs --></a>, <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>, <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> and <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox107"><a id="dx1-57002"></a><span class="ectt-1095">atoms</span></span> — same definition as <a href="#colvar|rmsd|atoms"><span class="ectt-1095">atoms</span></a> (<span class="ectt-1095">rmsd </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox108"><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances? </span>(see <a href="#colvar|cvc|forceNoPBC">description</a>)<br class="newline"/>Default: <span class="ectt-1095">no</span></li></ul>
<div class="flushright">
<!--l. 2392--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2392--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_dipoleMagnitude"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.6.6    </span> <a id="sec:cvc_dipoleMagnitude"></a><span class="ectt-1095">dipoleMagnitude</span>: dipole magnitude of a group of atoms.</h5>
<!--l. 2392--><p class="noindent"></p></div> <a id="colvar|dipoleMagnitude"></a> The <span class="ectt-1095">dipoleMagnitude</span><span class="ectt-1095"> {...} </span>block defines the dipole magnitude of a group of atoms (norm of the dipole moment's
vector), being <span class="ectt-1095">atoms </span>the group where dipole magnitude is calculated. It returns the magnitude in elementary
charge <!--l. 2395--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>e</mi></math>
times Å.
<!--l. 2397--><p class="noindent"><span class="phvb8t-x-x-109">List of keywords </span>(see also <a href="#sec:cvc_pbcs">4.2<!--tex4ht:ref: sec:cvc_pbcs --></a>, <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>, <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> and <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox109"><a id="dx1-58001"></a><span class="ectt-1095">atoms</span></span> — same definition as <a href="#colvar|rmsd|atoms"><span class="ectt-1095">atoms</span></a> (<span class="ectt-1095">rmsd </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox110"><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances? </span>(see <a href="#colvar|cvc|forceNoPBC">description</a>)<br class="newline"/>Default: <span class="ectt-1095">no</span></li></ul>
<div class="flushright">
<!--l. 2403--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2403--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_inertiaZ"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.6.7    </span> <a id="sec:cvc_inertiaZ"></a><span class="ectt-1095">inertiaZ</span>: total moment of inertia of a group of atoms around a chosen axis.</h5>
<!--l. 2403--><p class="noindent"></p></div> <a id="colvar|inertiaZ"></a>
<!--l. 2406--><p class="indent">   The block <span class="ectt-1095">inertiaZ</span><span class="ectt-1095"> {...} </span>defines the parameters for calculating the component along the axis
<!--l. 2407--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="bold"><mi>e</mi></mstyle></math> of the moment of inertia of a group
of atomic positions <!--l. 2408--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">{</mo><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo><mo class="MathClass-punc">,</mo><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mn>2</mn></mrow></msub><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo><mo class="MathClass-punc">,</mo><mi class="MathClass-op">…</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>N</mi></mrow></msub><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo><mo class="MathClass-close">}</mo></math> with respect
to their center of geometry, <!--l. 2410--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mstyle mathvariant="normal"><mi>c</mi><mi>o</mi><mi>g</mi></mstyle></mrow></msub><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo></math>:
</p><table class="equation"><tr><td> <a id="x1-59001r9"></a>
<!--l. 2411--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<msub><mrow><mi>I</mi></mrow><mrow><mstyle mathvariant="bold"><mi>e</mi></mstyle></mrow></msub><mspace class="thickpace" width="0.28em"></mspace> <mo class="MathClass-rel">=</mo> <mspace class="thickpace" width="0.28em"></mspace><munderover accent="false" accentunder="false"><mrow><mo>∑</mo></mrow><mrow><mi>i</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><mi>N</mi></mrow></munderover><msup><mrow> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mrow><mo fence="true" form="prefix"> (</mo><mrow><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow>
<mi>i</mi></mrow></msub><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo><mo class="MathClass-bin">−</mo><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mstyle mathvariant="normal"><mi>c</mi><mi>o</mi><mi>g</mi></mstyle></mrow></msub><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo></mrow><mo fence="true" form="postfix">)</mo></mrow><mo class="MathClass-bin">⋅</mo><mstyle mathvariant="bold"><mi>e</mi></mstyle></mrow><mo fence="true" form="postfix">)</mo></mrow></mrow><mrow><mn>2</mn></mrow></msup>
</math></td><td class="eq-no">(9)</td></tr></table>
<!--l. 2416--><p class="indent"> <span class="phvro8t-x-x-109">Note that all atomic masses are set to 1 for simplicity. </span>This component must contain one
<span class="ectt-1095">atoms</span><span class="ectt-1095"> {...} </span>block to define the atom group, and returns a positive number, expressed in
Å<!--l. 2419--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow></mrow><mrow><mn>2</mn></mrow></msup></math>.
</p><!--l. 2422--><p class="noindent"><span class="phvb8t-x-x-109">List of keywords </span>(see also <a href="#sec:cvc_pbcs">4.2<!--tex4ht:ref: sec:cvc_pbcs --></a>, <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>, <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> and <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox111"><a id="dx1-59002"></a><span class="ectt-1095">atoms</span></span> — same definition as <a href="#colvar|rmsd|atoms"><span class="ectt-1095">atoms</span></a> (<span class="ectt-1095">rmsd </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox112"><a id="dx1-59003"></a><span class="ectt-1095">axis</span></span> — <span class="phvro8t-x-x-109">Projection axis</span><br class="newline"/>Default: <span class="ectt-1095">(0.0, 0.0, 1.0)</span> <!--l. 2433--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> <span class="ectt-1095">(x,</span>
<span class="ectt-1095">y, z) </span>triplet, context: <span class="ectt-1095">inertiaZ</span> <!--l. 2433--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>The three components of this vector define (when normalized) the projection axis <!--l. 2433--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="bold"><mi>e</mi></mstyle></math>.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox113"><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances? </span>(see <a href="#colvar|cvc|forceNoPBC">description</a>)<br class="newline"/>Default: <span class="ectt-1095">no</span></li></ul>
<div class="flushright">
<!--l. 2437--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2437--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_rotations"></a>
<h4 class="subsectionHead"><span class="titlemark">4.7    </span> <a id="sec:cvc_rotations"></a>Rotations</h4>
<!--l. 2437--><p class="noindent"></p></div>
<!--l. 2439--><p class="indent">   The variables discussed in this section quantify the rotations of macromolecules (or other quasi-rigid
objects) from a given set of reference coordinates to the current coordinates. Such rotations are
computed following the same method used for best-fit RMSDs (see <a href="#colvar|rmsd"><span class="ectt-1095">rmsd</span></a> and <a href="#atom-group|fittingGroup"><span class="ectt-1095">fittingGroup</span></a>). The
underlying mathematical formalism is described in reference <span class="cite">[<a href="#XCoutsias2004">4</a>]</span>, and the implementation in
reference <span class="cite">[<a href="#XFiorin2013">1</a>]</span>.
</p><!--l. 2443--><p class="indent">   The first of the functions described is the <a href="#colvar|orientation"><span class="ectt-1095">orientation</span></a>, which describes the full rotation as a unit
quaternion <!--l. 2443--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="sans-serif"><mi>q</mi></mstyle> <mo class="MathClass-rel">=</mo> <mo class="MathClass-open">(</mo><msub><mrow><mi>q</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>q</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>q</mi></mrow><mrow><mn>2</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>q</mi></mrow><mrow><mn>3</mn></mrow></msub><mo class="MathClass-close">)</mo></math>,
i.e. 4 numbers with one constraint (3 degrees of freedom). The quaternion
<!--l. 2444--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="sans-serif"><mi>q</mi></mstyle></math>
is <span class="phvro8t-x-x-109">one of only two representations that are both complete and accurate</span>, the other being a
<!--l. 2444--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>3</mn><mo class="MathClass-bin">×</mo><mn>3</mn></math> unit matrix with 3 independent
parameters. Although <!--l. 2445--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="sans-serif"><mi>q</mi></mstyle></math>
is used internally in the Colvars module for features such as the <a href="#colvar|rmsd"><span class="ectt-1095">rmsd</span></a> function and the <a href="#atom-group|fittingGroup"><span class="ectt-1095">fittingGroup</span></a>
option, its direct use as a collective variable is more difficult, and mostly limited to fixed or moving
restraints.
</p><!--l. 2447--><p class="indent">   The two functions <a href="#colvar|orientationAngle"><span class="ectt-1095">orientationAngle</span></a> and <a href="#colvar|orientationProj"><span class="ectt-1095">orientationProj</span></a>, with the latter being
the cosine of the former, represent the amplitude of the full rotation described by
<!--l. 2447--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="sans-serif"><mi>q</mi></mstyle></math>,
regardless of the direction of its axis. As one-dimensional scalar variables, both <a href="#colvar|orientationAngle"><span class="ectt-1095">orientationAngle</span></a> and
<a href="#colvar|orientationProj"><span class="ectt-1095">orientationProj</span></a> are a much reduced simplification of the full rotation. However, they can be used in a
variety of methods including both restraints and PMF computations.
</p><!--l. 2451--><p class="indent">   A slightly more complete parametrization is achieved by decomposing the full rotation into the two parameters,
<a href="#colvar|tilt"><span class="ectt-1095">tilt</span></a> and <a href="#colvar|spinAngle"><span class="ectt-1095">spinAngle</span></a>. These quantify the amplitudes of two independent sub-rotations away from a certain axis
                                                                                         
                                                                                         
<!--l. 2452--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="bold"><mi>e</mi></mstyle></math>, and around the same
axis <!--l. 2452--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="bold"><mi>e</mi></mstyle></math>, respectively.
The axis <!--l. 2453--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="bold"><mi>e</mi></mstyle></math> is chosen
by the user, and is by default the Z axis: under that choice, <a href="#colvar|tilt"><span class="ectt-1095">tilt</span></a> is equivalent to the sine of the Euler “pitch" angle
<!--l. 2454--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>𝜃</mi></math>, and <a href="#colvar|spinAngle"><span class="ectt-1095">spinAngle</span></a> to the sum
of the other two angles, <!--l. 2454--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">(</mo><mi>ϕ</mi> <mo class="MathClass-bin">+</mo><mi>ψ</mi><mo class="MathClass-close">)</mo></math>.
This parameterization is mathematically well defined for almost all full rotations, including small ones when
the current coordinates are almost completely aligned with the reference ones. However, a mathematical
singularity prevents using the <a href="#colvar|spinAngle"><span class="ectt-1095">spinAngle</span></a> function near configurations where the value of tilt <a href="#colvar|tilt"><span class="ectt-1095">tilt</span></a> is -1 (i.e. a
-180<!--l. 2456--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></math> rotation around
an axis orthogonal to <!--l. 2456--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="bold"><mi>e</mi></mstyle></math>).
For these reasons, <a href="#colvar|tilt"><span class="ectt-1095">tilt</span></a> and <a href="#colvar|spinAngle"><span class="ectt-1095">spinAngle</span></a> are useful when the allowed rotations are known to have
approximately the same axis, and differ only in the magnitude of the corresponding angle. In
this use case, <a href="#colvar|spinAngle"><span class="ectt-1095">spinAngle</span></a> measures the angle of the sub-rotation around the chosen axis
<!--l. 2458--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="bold"><mi>e</mi></mstyle></math>, whereas <a href="#colvar|tilt"><span class="ectt-1095">tilt</span></a> measures
the dot product between <!--l. 2458--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="bold"><mi>e</mi></mstyle></math>
and the actual axis of the full rotation.
</p><!--l. 2460--><p class="indent">   Lastly, the traditional Euler angles are also available as the functions <a href="#colvar|eulerPhi"><span class="ectt-1095">eulerPhi</span></a>, <a href="#colvar|eulerTheta"><span class="ectt-1095">eulerTheta</span></a> and
<a href="#colvar|eulerPsi"><span class="ectt-1095">eulerPsi</span></a>. Altogether, these are sufficient to represent all three degrees of freedom of a full rotation.
However, they also suffer from the potential “gimbal lock" problem, which emerges whenever
<!--l. 2462--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>𝜃</mi> <mo class="MathClass-rel">≃</mo><mo class="MathClass-bin">±</mo><mn>9</mn><msup><mrow><mn>0</mn></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></math>, which
includes also the case where the full rotation is small. Under such conditions, the angles
<!--l. 2463--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ϕ</mi></math> and
<!--l. 2463--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ψ</mi></math>
are both ill-defined and cannot be used as collective variables. For these reasons,
it is highly recommended that Euler angles are used only in simulations where their
range of applicability is <span class="phvro8t-x-x-109">known ahead of time</span>, and excludes configurations where
<!--l. 2464--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>𝜃</mi> <mo class="MathClass-rel">≃</mo><mo class="MathClass-bin">±</mo><mn>9</mn><msup><mrow><mn>0</mn></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></math>
altogether.
</p>
<div class="flushright">
<!--l. 2466--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2466--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_orientation"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.7.1    </span> <a id="sec:cvc_orientation"></a><span class="ectt-1095">orientation</span>: orientation from reference coordinates.</h5>
<!--l. 2466--><p class="noindent"></p></div> <a id="colvar|orientation"></a>
<!--l. 2469--><p class="indent">   The block <span class="ectt-1095">orientation</span><span class="ectt-1095"> {...} </span>returns the same optimal rotation used in the <span class="ectt-1095">rmsd </span>component to superimpose the coordinates
<!--l. 2471--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">{</mo><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo><mo class="MathClass-close">}</mo></math> onto a set of reference
coordinates <!--l. 2472--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">{</mo><msubsup><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>i</mi></mrow><mrow><mstyle mathvariant="normal"><mo class="MathClass-open">(</mo><mi>r</mi><mi>e</mi><mi>f</mi><mo class="MathClass-close">)</mo></mstyle></mrow></msubsup><mo class="MathClass-close">}</mo></math>. Such component
returns a four dimensional vector <!--l. 2473--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="sans-serif"><mi>q</mi></mstyle> <mo class="MathClass-rel">=</mo> <mo class="MathClass-open">(</mo><msub><mrow><mi>q</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>q</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>q</mi></mrow><mrow><mn>2</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>q</mi></mrow><mrow><mn>3</mn></mrow></msub><mo class="MathClass-close">)</mo></math>,
with <!--l. 2474--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi class="MathClass-op">∑</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo><!--nolimits--></mrow><mrow><mi>i</mi></mrow></msub><msubsup><mrow><mi>q</mi></mrow><mrow><mi>i</mi></mrow><mrow><mn>2</mn></mrow></msubsup> <mo class="MathClass-rel">=</mo> <mn>1</mn></math>; this <span class="phvro8t-x-x-109">quaternion</span>
expresses the optimal rotation <!--l. 2475--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">{</mo><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo><mo class="MathClass-close">}</mo><mo class="MathClass-rel">→</mo><mo class="MathClass-open">{</mo><msubsup><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>i</mi></mrow><mrow><mstyle mathvariant="normal"><mo class="MathClass-open">(</mo><mi>r</mi><mi>e</mi><mi>f</mi><mo class="MathClass-close">)</mo></mstyle></mrow></msubsup><mo class="MathClass-close">}</mo></math>
                                                                                         
                                                                                         
according to the formalism in reference <span class="cite">[<a href="#XCoutsias2004">4</a>]</span>. The quaternion
<!--l. 2477--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">(</mo><msub><mrow><mi>q</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>q</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>q</mi></mrow><mrow><mn>2</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>q</mi></mrow><mrow><mn>3</mn></mrow></msub><mo class="MathClass-close">)</mo></math> can also be
written as <!--l. 2478--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mi class="qopname">cos</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--><mo class="MathClass-open">(</mo><mi>𝜃</mi><mo class="MathClass-bin">∕</mo><mn>2</mn><mo class="MathClass-close">)</mo><mo class="MathClass-punc">,</mo><mspace class="thinspace" width="0.17em"></mspace><mi class="qopname">sin</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--><mo class="MathClass-open">(</mo><mi>𝜃</mi><mo class="MathClass-bin">∕</mo><mn>2</mn><mo class="MathClass-close">)</mo><mstyle mathvariant="bold"><mi>u</mi></mstyle></mrow><mo fence="true" form="postfix">)</mo></mrow></math>, where
<!--l. 2479--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>𝜃</mi></math> is the angle and
<!--l. 2480--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="bold"><mi>u</mi></mstyle></math> the normalized axis of rotation;
for example, a rotation of 90<!--l. 2481--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></math>
around the <!--l. 2481--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math> axis is
expressed as “<span class="ectt-1095">(0.707, 0.0, 0.0, 0.707)</span>". The script <a href="https://raw.githubusercontent.com/Colvars/colvars/master/colvartools/quaternion2rmatrix.tcl"><span class="ectt-1095">quaternion2rmatrix.tcl</span></a> provides Tcl functions for converting
to and from a <!--l. 2484--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>4</mn><mo class="MathClass-bin">×</mo><mn>4</mn></math>
rotation matrix in a format suitable for usage in VMD.
</p><!--l. 2487--><p class="indent">   As for the component <span class="ectt-1095">rmsd</span>, the available options are <span class="ectt-1095">atoms</span>, <span class="ectt-1095">refPositionsFile</span>, <span class="ectt-1095">refPositionsCol </span>and
<span class="ectt-1095">refPositionsColValue</span>,  and <span class="ectt-1095">refPositions</span>.
</p><!--l. 2489--><p class="indent"> <span class="phvb8t-x-x-109">Note: </span><span class="ectt-1095">refPositions </span>and <span class="ectt-1095">refPositionsFile </span>define the set of positions <span class="phvro8t-x-x-109">from which </span>the optimal rotation
is calculated, but this rotation is not applied to the coordinates of the atoms involved: it is used instead to
define the variable itself.
</p><!--l. 2491--><p class="noindent"><span class="phvb8t-x-x-109">List of keywords </span>(see also <a href="#sec:cvc_pbcs">4.2<!--tex4ht:ref: sec:cvc_pbcs --></a>, <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>, <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> and <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox114"><a id="dx1-61001"></a><span class="ectt-1095">atoms</span></span> — same definition as <a href="#colvar|rmsd|atoms"><span class="ectt-1095">atoms</span></a> (<span class="ectt-1095">rmsd </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox115"><a id="dx1-61002"></a><span class="ectt-1095">refPositions</span></span> — same definition as <a href="#colvar|rmsd|refPositions"><span class="ectt-1095">refPositions</span></a> (<span class="ectt-1095">rmsd </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox116"><a id="dx1-61003"></a><span class="ectt-1095">refPositionsFile</span></span> — same definition as <a href="#colvar|rmsd|refPositionsFile"><span class="ectt-1095">refPositionsFile</span></a> (<span class="ectt-1095">rmsd </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox117"><a id="dx1-61004"></a><span class="ectt-1095">refPositionsCol</span></span> — same definition as <a href="#colvar|rmsd|refPositionsCol"><span class="ectt-1095">refPositionsCol</span></a> (<span class="ectt-1095">rmsd </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox118"><a id="dx1-61005"></a><span class="ectt-1095">refPositionsColValue</span></span> — same definition as <a href="#colvar|rmsd|refPositionsColValue"><span class="ectt-1095">refPositionsColValue</span></a> (<span class="ectt-1095">rmsd </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox119"><a id="dx1-61006"></a><span class="ectt-1095">closestToQuaternion</span></span> — <span class="phvro8t-x-x-109">Reference rotation</span><br class="newline"/>Default: <span class="ectt-1095">(1.0, 0.0, 0.0, 0.0) </span>(“null" rotation) <!--l. 2518--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> “<span class="ectt-1095">(q0,</span>
<span class="ectt-1095">q1, q2, q3)</span>" quadruplet, context: <span class="ectt-1095">orientation</span> <!--l. 2518--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Between the two equivalent quaternions <!--l. 2518--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">(</mo><msub><mrow><mi>q</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>q</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>q</mi></mrow><mrow><mn>2</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>q</mi></mrow><mrow><mn>3</mn></mrow></msub><mo class="MathClass-close">)</mo></math>
     and <!--l. 2518--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">(</mo><mo class="MathClass-bin">−</mo><msub><mrow><mi>q</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-punc">,</mo><mo class="MathClass-bin">−</mo><msub><mrow><mi>q</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-punc">,</mo><mo class="MathClass-bin">−</mo><msub><mrow><mi>q</mi></mrow><mrow><mn>2</mn></mrow></msub><mo class="MathClass-punc">,</mo><mo class="MathClass-bin">−</mo><msub><mrow><mi>q</mi></mrow><mrow><mn>3</mn></mrow></msub><mo class="MathClass-close">)</mo></math>,
     the closer to <span class="ectt-1095">(1.0, 0.0, 0.0, 0.0) </span>is chosen. This simplifies the visualization of the colvar
     trajectory when sampled values are a smaller subset of all possible rotations. <span class="phvb8t-x-x-109">Note: </span><span class="phvro8t-x-x-109">this only</span>
<span class="phvro8t-x-x-109">affects the output, never the dynamics</span>.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox120"><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances? </span>(see <a href="#colvar|cvc|forceNoPBC">description</a>)<br class="newline"/>Default: <span class="ectt-1095">no</span></li></ul>
<!--l. 2523--><p class="noindent"><span class="likeparagraphHead"><a id="x1-620004.7.1"></a>Example: stopping the rotation of a protein.</span>
   To stop the rotation of an elongated macromolecule in solution (and use an anisotropic box to save
water molecules), it is possible to define a colvar with an <span class="ectt-1095">orientation </span>component, and restrain it through
the <span class="ectt-1095">harmonic </span>bias around the identity rotation, <span class="ectt-1095">(1.0, 0.0, 0.0, 0.0)</span>. Only the overall orientation of the
macromolecule is affected, and <span class="phvro8t-x-x-109">not </span>its internal degrees of freedom. In NAMD, the user should
also take care that the macromolecule is composed by a single chain , or disable <span class="ectt-1095">wrapAll</span>
otherwise.</p>
<div class="mdframed" id="mdframed-42">
<!--l. 2533--><p class="noindent"></p><pre>colvar {<br class="newline"/>  name Orient<br class="newline"/>  orientation {<br class="newline"/>    atoms { … }<br class="newline"/>    refPositionsFile reference.pdb<br class="newline"/>  }<br class="newline"/>}<br class="newline"/><br class="newline"/>harmonic {             # Define a harmonic restraint<br class="newline"/>  colvars Orient       # acting on colvar "Orient"<br class="newline"/>  centers  (1.0, 0.0, 0.0, 0.0)  # center the unit quaternion (no rotation)<br class="newline"/>  forceConstant 500.0            # unit is energy: quaternions are dimensionless<br class="newline"/>}<br class="newline"/></pre><!--l. 2547--><p class="noindent"></p>
</div>
<div class="flushright">
<!--l. 2550--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2550--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_orientationAngle"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.7.2    </span> <a id="sec:cvc_orientationAngle"></a><span class="ectt-1095">orientationAngle</span>: angle of rotation from reference coordinates.</h5>
<!--l. 2550--><p class="noindent"></p></div> <a id="colvar|orientationAngle"></a>
<!--l. 2553--><p class="indent">   The block <span class="ectt-1095">orientationAngle</span><span class="ectt-1095"> {...} </span>accepts the same base options as the component <span class="ectt-1095">orientation</span>: <span class="ectt-1095">atoms</span>,
<span class="ectt-1095">refPositions</span>, <span class="ectt-1095">refPositionsFile</span>, <span class="ectt-1095">refPositionsCol </span>and <span class="ectt-1095">refPositionsColValue</span>. The returned value is the angle
of rotation <!--l. 2555--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>𝜃</mi></math>
between the current and the reference positions. This angle is expressed in degrees within the range
[0<!--l. 2556--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></math>:180<!--l. 2556--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></math>].
</p><!--l. 2558--><p class="noindent"><span class="phvb8t-x-x-109">List of keywords </span>(see also <a href="#sec:cvc_pbcs">4.2<!--tex4ht:ref: sec:cvc_pbcs --></a>, <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>, <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> and <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox121"><a id="dx1-63001"></a><span class="ectt-1095">atoms</span></span> — same definition as <a href="#colvar|rmsd|atoms"><span class="ectt-1095">atoms</span></a> (<span class="ectt-1095">rmsd </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox122"><a id="dx1-63002"></a><span class="ectt-1095">refPositions</span></span> — same definition as <a href="#colvar|rmsd|refPositions"><span class="ectt-1095">refPositions</span></a> (<span class="ectt-1095">rmsd </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox123"><a id="dx1-63003"></a><span class="ectt-1095">refPositionsFile</span></span> — same definition as <a href="#colvar|rmsd|refPositionsFile"><span class="ectt-1095">refPositionsFile</span></a> (<span class="ectt-1095">rmsd </span>component)
                                                                                         
                                                                                         
     </li>
<li class="itemize"><span class="colorbox" id="colorbox124"><a id="dx1-63004"></a><span class="ectt-1095">refPositionsCol</span></span> — same definition as <a href="#colvar|rmsd|refPositionsCol"><span class="ectt-1095">refPositionsCol</span></a> (<span class="ectt-1095">rmsd </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox125"><a id="dx1-63005"></a><span class="ectt-1095">refPositionsColValue</span></span> — same definition as <a href="#colvar|rmsd|refPositionsColValue"><span class="ectt-1095">refPositionsColValue</span></a> (<span class="ectt-1095">rmsd </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox126"><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances? </span>(see <a href="#colvar|cvc|forceNoPBC">description</a>)<br class="newline"/>Default: <span class="ectt-1095">no</span></li></ul>
<div class="flushright">
<!--l. 2575--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2575--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_orientationProj"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.7.3    </span> <a id="sec:cvc_orientationProj"></a><span class="ectt-1095">orientationProj</span>: cosine of the angle of rotation from reference coordinates.</h5>
<!--l. 2575--><p class="noindent"></p></div> <a id="colvar|orientationProj"></a>
<!--l. 2578--><p class="indent">   The block <span class="ectt-1095">orientationProj</span><span class="ectt-1095"> {...} </span>accepts the same base options as the component
<span class="ectt-1095">orientation</span>: <span class="ectt-1095">atoms</span>, <span class="ectt-1095">refPositions</span>, <span class="ectt-1095">refPositionsFile</span>, <span class="ectt-1095">refPositionsCol </span>and
<span class="ectt-1095">refPositionsColValue</span>. The returned value is the cosine of the angle of rotation
<!--l. 2580--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>𝜃</mi></math>
between the current and the reference positions. The range of values is [-1:1].
</p><!--l. 2583--><p class="noindent"><span class="phvb8t-x-x-109">List of keywords </span>(see also <a href="#sec:cvc_pbcs">4.2<!--tex4ht:ref: sec:cvc_pbcs --></a>, <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>, <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> and <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox127"><a id="dx1-64001"></a><span class="ectt-1095">atoms</span></span> — same definition as <a href="#colvar|rmsd|atoms"><span class="ectt-1095">atoms</span></a> (<span class="ectt-1095">rmsd </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox128"><a id="dx1-64002"></a><span class="ectt-1095">refPositions</span></span> — same definition as <a href="#colvar|rmsd|refPositions"><span class="ectt-1095">refPositions</span></a> (<span class="ectt-1095">rmsd </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox129"><a id="dx1-64003"></a><span class="ectt-1095">refPositionsFile</span></span> — same definition as <a href="#colvar|rmsd|refPositionsFile"><span class="ectt-1095">refPositionsFile</span></a> (<span class="ectt-1095">rmsd </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox130"><a id="dx1-64004"></a><span class="ectt-1095">refPositionsCol</span></span> — same definition as <a href="#colvar|rmsd|refPositionsCol"><span class="ectt-1095">refPositionsCol</span></a> (<span class="ectt-1095">rmsd </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox131"><a id="dx1-64005"></a><span class="ectt-1095">refPositionsColValue</span></span> — same definition as <a href="#colvar|rmsd|refPositionsColValue"><span class="ectt-1095">refPositionsColValue</span></a> (<span class="ectt-1095">rmsd </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox132"><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances? </span>(see <a href="#colvar|cvc|forceNoPBC">description</a>)<br class="newline"/>Default: <span class="ectt-1095">no</span></li></ul>
<div class="flushright">
<!--l. 2599--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2599--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_spinAngle"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.7.4    </span> <a id="sec:cvc_spinAngle"></a><span class="ectt-1095">spinAngle</span>: angle of rotation around a given axis.</h5>
<!--l. 2599--><p class="noindent"></p></div> <a id="colvar|spinAngle"></a>
<!--l. 2602--><p class="indent">   The complete rotation described by <span class="ectt-1095">orientation </span>can optionally be decomposed into two
sub-rotations: one is a “<span class="phvro8t-x-x-109">spin</span>" rotation around <span class="phvb8t-x-x-109">e</span>, and the other a “<span class="phvro8t-x-x-109">tilt</span>" rotation around an axis
orthogonal to <span class="phvb8t-x-x-109">e</span>. The component <span class="ectt-1095">spinAngle </span>measures the angle of the “spin" sub-rotation around
<span class="phvb8t-x-x-109">e</span>.
</p><!--l. 2605--><p class="noindent"><span class="phvb8t-x-x-109">List of keywords </span>(see also <a href="#sec:cvc_pbcs">4.2<!--tex4ht:ref: sec:cvc_pbcs --></a>, <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>, <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> and <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox133"><a id="dx1-65001"></a><span class="ectt-1095">atoms</span></span> — same definition as <a href="#colvar|rmsd|atoms"><span class="ectt-1095">atoms</span></a> (<span class="ectt-1095">rmsd </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox134"><a id="dx1-65002"></a><span class="ectt-1095">refPositions</span></span> — same definition as <a href="#colvar|rmsd|refPositions"><span class="ectt-1095">refPositions</span></a> (<span class="ectt-1095">rmsd </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox135"><a id="dx1-65003"></a><span class="ectt-1095">refPositionsFile</span></span> — same definition as <a href="#colvar|rmsd|refPositionsFile"><span class="ectt-1095">refPositionsFile</span></a> (<span class="ectt-1095">rmsd </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox136"><a id="dx1-65004"></a><span class="ectt-1095">refPositionsCol</span></span> — same definition as <a href="#colvar|rmsd|refPositionsCol"><span class="ectt-1095">refPositionsCol</span></a> (<span class="ectt-1095">rmsd </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox137"><a id="dx1-65005"></a><span class="ectt-1095">refPositionsColValue</span></span> — same definition as <a href="#colvar|rmsd|refPositionsColValue"><span class="ectt-1095">refPositionsColValue</span></a> (<span class="ectt-1095">rmsd </span>component)
     </li>
<li class="itemize"><a id="colvar|spinAngle|axis"></a> <span class="colorbox" id="colorbox138"><a id="dx1-65006"></a><span class="ectt-1095">axis</span></span> — <span class="phvro8t-x-x-109">Special rotation axis</span><br class="newline"/>Default: <span class="ectt-1095">(0.0, 0.0, 1.0)</span> <!--l. 2626--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> <span class="ectt-1095">(x,</span>
<span class="ectt-1095">y, z) </span>triplet, context: <span class="ectt-1095">tilt</span> <!--l. 2626--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>The three components of this vector define (when normalized) the special rotation axis used
     to calculate the <span class="ectt-1095">tilt </span>and <span class="ectt-1095">spinAngle </span>components.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox139"><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances? </span>(see <a href="#colvar|cvc|forceNoPBC">description</a>)<br class="newline"/>Default: <span class="ectt-1095">no</span></li></ul>
<!--l. 2628--><p class="noindent">The component <span class="ectt-1095">spinAngle </span>returns an angle (in degrees) within the periodic interval
<!--l. 2628--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo><mo class="MathClass-bin">−</mo><mn>1</mn><mn>8</mn><mn>0</mn> <mo class="MathClass-punc">:</mo> <mn>1</mn><mn>8</mn><mn>0</mn><mo class="MathClass-close">]</mo></math>.
</p><!--l. 2630--><p class="indent"> <span class="phvb8t-x-x-109">Note: </span>the value of <span class="ectt-1095">spinAngle </span>is a continuous function almost everywhere,
with the exception of configurations with the corresponding “tilt" angle equal to
180<!--l. 2630--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></math> (i.e. the <span class="ectt-1095">tilt</span>
component is equal to <!--l. 2630--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-bin">−</mo><mn>1</mn></math>):
in those cases, <span class="ectt-1095">spinAngle </span>is undefined. If such configurations are expected, consider
defining a <span class="ectt-1095">tilt </span>colvar using the same axis <span class="phvb8t-x-x-109">e</span>, and restraining it with a lower wall away from
<!--l. 2630--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-bin">−</mo><mn>1</mn></math>.
</p>
<div class="flushright">
<!--l. 2633--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2633--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_tilt"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.7.5    </span> <a id="sec:cvc_tilt"></a><span class="ectt-1095">tilt</span>: cosine of the rotation orthogonal to a given axis.</h5>
<!--l. 2633--><p class="noindent"></p></div> <a id="colvar|tilt"></a>
<!--l. 2636--><p class="indent">   The component <span class="ectt-1095">tilt </span>measures the cosine of the angle of the “tilt" sub-rotation, which
combined with the “spin" sub-rotation provides the complete rotation of a group of atoms.
The cosine of the tilt angle rather than the tilt angle itself is implemented, because the latter
is unevenly distributed even for an isotropic system: consider as an analogy the angle
<!--l. 2637--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>𝜃</mi></math> in the
spherical coordinate system. The component <span class="ectt-1095">tilt </span>relies on the same options as <span class="ectt-1095">spinAngle</span>,
including the definition of the axis <span class="phvb8t-x-x-109">e</span>. The values of <span class="ectt-1095">tilt </span>are real numbers in the interval
<!--l. 2639--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo><mo class="MathClass-bin">−</mo><mn>1</mn> <mo class="MathClass-punc">:</mo> <mn>1</mn><mo class="MathClass-close">]</mo></math>: the value
<!--l. 2639--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>1</mn></math> represents an orientation fully
parallel to <span class="phvb8t-x-x-109">e </span>(tilt angle = 0<!--l. 2639--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></math>),
and the value <!--l. 2639--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-bin">−</mo><mn>1</mn></math>
represents an anti-parallel orientation.
</p><!--l. 2641--><p class="noindent"><span class="phvb8t-x-x-109">List of keywords </span>(see also <a href="#sec:cvc_pbcs">4.2<!--tex4ht:ref: sec:cvc_pbcs --></a>, <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>, <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> and <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox140"><a id="dx1-66001"></a><span class="ectt-1095">atoms</span></span> — same definition as <a href="#colvar|rmsd|atoms"><span class="ectt-1095">atoms</span></a> (<span class="ectt-1095">rmsd </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox141"><a id="dx1-66002"></a><span class="ectt-1095">refPositions</span></span> — same definition as <a href="#colvar|rmsd|refPositions"><span class="ectt-1095">refPositions</span></a> (<span class="ectt-1095">rmsd </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox142"><a id="dx1-66003"></a><span class="ectt-1095">refPositionsFile</span></span> — same definition as <a href="#colvar|rmsd|refPositionsFile"><span class="ectt-1095">refPositionsFile</span></a> (<span class="ectt-1095">rmsd </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox143"><a id="dx1-66004"></a><span class="ectt-1095">refPositionsCol</span></span> — same definition as <a href="#colvar|rmsd|refPositionsCol"><span class="ectt-1095">refPositionsCol</span></a> (<span class="ectt-1095">rmsd </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox144"><a id="dx1-66005"></a><span class="ectt-1095">refPositionsColValue</span></span> — same definition as <a href="#colvar|rmsd|refPositionsColValue"><span class="ectt-1095">refPositionsColValue</span></a> (<span class="ectt-1095">rmsd </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox145"><a id="dx1-66006"></a><span class="ectt-1095">axis</span></span> — same definition as <a href="#colvar|spinAngle|axis"><span class="ectt-1095">axis</span></a> (<span class="ectt-1095">spinAngle </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox146"><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances? </span>(see <a href="#colvar|cvc|forceNoPBC">description</a>)<br class="newline"/>Default: <span class="ectt-1095">no</span></li></ul>
<div class="flushright">
<!--l. 2660--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2660--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_eulerPhi"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.7.6    </span> <a id="sec:cvc_eulerPhi"></a><span class="ectt-1095">eulerPhi</span>: Roll angle from references coordinates.</h5>
<!--l. 2660--><p class="noindent"></p></div> <a id="colvar|eulerPhi"></a>
<!--l. 2663--><p class="indent">   Assuming the axes of the original frame are denoted as <span class="phvro8t-x-x-109">x, y, z </span>and the axes of the rotated frame as <span class="phvro8t-x-x-109">X,</span>
<span class="phvro8t-x-x-109">Y, Z</span>, the line of nodes, <span class="phvro8t-x-x-109">N</span>, can be defined as the intersection of the plane <span class="phvro8t-x-x-109">xy </span>and <span class="phvro8t-x-x-109">XY</span>. The axis
perpendicular to <span class="phvro8t-x-x-109">N </span>and <span class="phvro8t-x-x-109">z </span>is defined as <span class="phvro8t-x-x-109">P</span>. In this case, as illustrated in the figure below, the complete
rotation described by <span class="ectt-1095">orientation </span>can optionally be decomposed into three Euler angles:
</p>
<ul class="itemize1">
<li class="itemize">the “roll" angle <!--l. 2671--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ϕ</mi></math>,
     i.e. the rotation angle from the <span class="phvro8t-x-x-109">x </span>axis to the <span class="phvro8t-x-x-109">N </span>axis;
     </li>
<li class="itemize">the “pitch" angle <!--l. 2672--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>𝜃</mi></math>,
     i.e. the rotation angle from the <span class="phvro8t-x-x-109">P </span>axis to the <span class="phvro8t-x-x-109">Z </span>axis, and
     </li>
<li class="itemize">the “yaw" angle <!--l. 2674--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ψ</mi></math>,
     i.e. the rotation angle from the <span class="phvro8t-x-x-109">N </span>axis to the <span class="phvro8t-x-x-109">X </span>axis.</li></ul>
<div class="center">
<!--l. 2677--><p class="noindent">
</p><!--l. 2679--><p class="noindent"><img alt="PIC" src="eulerangles-512px.png"/></p></div>
<!--l. 2685--><p class="indent">   Although Euler angles are more straightforward to use than quaternions, they are also potentially
subject to the “gimbal lock" problem:<br class="newline"/><a class="url" href="https://en.wikipedia.org/wiki/Gimbal_lock"><span class="ectt-1095">https://en.wikipedia.org/wiki/Gimbal_lock</span></a><br class="newline"/>which arises whenever <!--l. 2687--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>𝜃</mi> <mo class="MathClass-rel">≃</mo><mo class="MathClass-bin">±</mo><mn>9</mn><msup><mrow><mn>0</mn></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></math>,
including the common case when <span class="phvro8t-x-x-109">the simulated coordinates are near the reference coordinates</span>.
Therefore, a safe use of Euler angles as collective variables requires the use of restraints to avoid
such singularities, such as done in reference <span class="cite">[<a href="#XFu2017">6</a>]</span> and in the <a href="http://www.ks.uiuc.edu/Training/Tutorials/namd/PLB/tutorial-protein-ligand.pdf">protein-ligand binding NAMD
tutorial</a>.
</p><!--l. 2690--><p class="indent">   The <span class="ectt-1095">eulerPhi </span>component accepts exactly the same options as <span class="ectt-1095">orientation</span>, and measures the
rotation angle from the <span class="phvro8t-x-x-109">x </span>axis to the <span class="phvro8t-x-x-109">N </span>axis. This angle is expressed in degrees within the periodic range
<!--l. 2693--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo><mo class="MathClass-bin">−</mo><mn>1</mn><mn>8</mn><msup><mrow><mn>0</mn></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup> <mo class="MathClass-punc">:</mo> <mn>1</mn><mn>8</mn><msup><mrow><mn>0</mn></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup><mo class="MathClass-close">]</mo></math>.
</p><!--l. 2695--><p class="noindent"><span class="phvb8t-x-x-109">List of keywords </span>(see also <a href="#sec:cvc_pbcs">4.2<!--tex4ht:ref: sec:cvc_pbcs --></a>, <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>, <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> and <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox147"><a id="dx1-67001"></a><span class="ectt-1095">atoms</span></span> — same definition as <a href="#colvar|rmsd|atoms"><span class="ectt-1095">atoms</span></a> (<span class="ectt-1095">rmsd </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox148"><a id="dx1-67002"></a><span class="ectt-1095">refPositions</span></span> — same definition as <a href="#colvar|rmsd|refPositions"><span class="ectt-1095">refPositions</span></a> (<span class="ectt-1095">rmsd </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox149"><a id="dx1-67003"></a><span class="ectt-1095">refPositionsFile</span></span> — same definition as <a href="#colvar|rmsd|refPositionsFile"><span class="ectt-1095">refPositionsFile</span></a> (<span class="ectt-1095">rmsd </span>component)
                                                                                         
                                                                                         
     </li>
<li class="itemize">
<div class="colorbox" id="colorbox150"><a id="dx1-67004"></a><span class="ectt-1095">refPositionsCol</span></div>
<!--l. 2714--><p class="noindent">— same definition as <a href="#colvar|rmsd|refPositionsCol"><span class="ectt-1095">refPositionsCol</span></a> (<span class="ectt-1095">rmsd </span>component)
     </p></li>
<li class="itemize">
<div class="colorbox" id="colorbox151"><a id="dx1-67005"></a><span class="ectt-1095">refPositionsColValue</span></div>
<!--l. 2714--><p class="noindent">— same definition as <a href="#colvar|rmsd|refPositionsColValue"><span class="ectt-1095">refPositionsColValue</span></a> (<span class="ectt-1095">rmsd </span>component)
     </p></li>
<li class="itemize"><span class="colorbox" id="colorbox152"><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances? </span>(see <a href="#colvar|cvc|forceNoPBC">description</a>)<br class="newline"/>Default: <span class="ectt-1095">no</span></li></ul>
<div class="flushright">
<!--l. 2719--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2719--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_eulerTheta"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.7.7    </span> <a id="sec:cvc_eulerTheta"></a><span class="ectt-1095">eulerTheta</span>: Pitch angle from references coordinates.</h5>
<!--l. 2719--><p class="noindent"></p></div> <a id="colvar|eulerTheta"></a>
<!--l. 2722--><p class="indent">   This component accepts exactly the same options as <span class="ectt-1095">orientation</span>, and measures the
rotation angle from the <span class="phvro8t-x-x-109">P </span>axis to the <span class="phvro8t-x-x-109">Z </span>axis. This angle is expressed in degrees within the range
<!--l. 2724--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo><mo class="MathClass-bin">−</mo><mn>9</mn><msup><mrow><mn>0</mn></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup> <mo class="MathClass-punc">:</mo> <mn>9</mn><msup><mrow><mn>0</mn></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup><mo class="MathClass-close">]</mo></math>.
</p><!--l. 2726--><p class="indent"> <span class="phvb8t-x-x-109">Warning: </span>When this angle reaches <!--l. 2726--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-bin">−</mo><mn>9</mn><msup><mrow><mn>0</mn></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></math>
or <!--l. 2726--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>9</mn><msup><mrow><mn>0</mn></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></math>, the
definition of orientation by euler angles suffers from the gimbal lock issue. You may need to apply a
restraint to keep <span class="ectt-1095">eulerTheta </span>away from the singularities.
</p><!--l. 2731--><p class="noindent"><span class="phvb8t-x-x-109">List of keywords </span>(see also <a href="#sec:cvc_pbcs">4.2<!--tex4ht:ref: sec:cvc_pbcs --></a>, <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>, <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> and <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox153"><a id="dx1-68001"></a><span class="ectt-1095">atoms</span></span> — same definition as <a href="#colvar|rmsd|atoms"><span class="ectt-1095">atoms</span></a> (<span class="ectt-1095">rmsd </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox154"><a id="dx1-68002"></a><span class="ectt-1095">refPositions</span></span> — same definition as <a href="#colvar|rmsd|refPositions"><span class="ectt-1095">refPositions</span></a> (<span class="ectt-1095">rmsd </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox155"><a id="dx1-68003"></a><span class="ectt-1095">refPositionsFile</span></span> — same definition as <a href="#colvar|rmsd|refPositionsFile"><span class="ectt-1095">refPositionsFile</span></a> (<span class="ectt-1095">rmsd </span>component)
     </li>
<li class="itemize">
<div class="colorbox" id="colorbox156"><a id="dx1-68004"></a><span class="ectt-1095">refPositionsCol</span></div>
<!--l. 2750--><p class="noindent">— same definition as <a href="#colvar|rmsd|refPositionsCol"><span class="ectt-1095">refPositionsCol</span></a> (<span class="ectt-1095">rmsd </span>component)
                                                                                         
                                                                                         
     </p></li>
<li class="itemize">
<div class="colorbox" id="colorbox157"><a id="dx1-68005"></a><span class="ectt-1095">refPositionsColValue</span></div>
<!--l. 2750--><p class="noindent">— same definition as <a href="#colvar|rmsd|refPositionsColValue"><span class="ectt-1095">refPositionsColValue</span></a> (<span class="ectt-1095">rmsd </span>component)
     </p></li>
<li class="itemize"><span class="colorbox" id="colorbox158"><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances? </span>(see <a href="#colvar|cvc|forceNoPBC">description</a>)<br class="newline"/>Default: <span class="ectt-1095">no</span></li></ul>
<div class="flushright">
<!--l. 2755--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2755--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_eulerPsi"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.7.8    </span> <a id="sec:cvc_eulerPsi"></a><span class="ectt-1095">eulerPsi</span>: Yaw angle from references coordinates.</h5>
<!--l. 2755--><p class="noindent"></p></div> <a id="colvar|eulerPsi"></a>
<!--l. 2758--><p class="indent">   This component accepts exactly the same options as <span class="ectt-1095">orientation</span>, and measures the rotation angle
from the <span class="phvro8t-x-x-109">N </span>axis to the <span class="phvro8t-x-x-109">X  </span>axis. This angle is expressed in degrees within the periodic range
<!--l. 2761--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo><mo class="MathClass-bin">−</mo><mn>1</mn><mn>8</mn><msup><mrow><mn>0</mn></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup> <mo class="MathClass-punc">:</mo> <mn>1</mn><mn>8</mn><msup><mrow><mn>0</mn></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup><mo class="MathClass-close">]</mo></math>.
</p><!--l. 2763--><p class="noindent"><span class="phvb8t-x-x-109">List of keywords </span>(see also <a href="#sec:cvc_pbcs">4.2<!--tex4ht:ref: sec:cvc_pbcs --></a>, <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>, <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> and <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox159"><a id="dx1-69001"></a><span class="ectt-1095">atoms</span></span> — same definition as <a href="#colvar|rmsd|atoms"><span class="ectt-1095">atoms</span></a> (<span class="ectt-1095">rmsd </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox160"><a id="dx1-69002"></a><span class="ectt-1095">refPositions</span></span> — same definition as <a href="#colvar|rmsd|refPositions"><span class="ectt-1095">refPositions</span></a> (<span class="ectt-1095">rmsd </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox161"><a id="dx1-69003"></a><span class="ectt-1095">refPositionsFile</span></span> — same definition as <a href="#colvar|rmsd|refPositionsFile"><span class="ectt-1095">refPositionsFile</span></a> (<span class="ectt-1095">rmsd </span>component)
     </li>
<li class="itemize">
<div class="colorbox" id="colorbox162"><a id="dx1-69004"></a><span class="ectt-1095">refPositionsCol</span></div>
<!--l. 2782--><p class="noindent">— same definition as <a href="#colvar|rmsd|refPositionsCol"><span class="ectt-1095">refPositionsCol</span></a> (<span class="ectt-1095">rmsd </span>component)
     </p></li>
<li class="itemize">
<div class="colorbox" id="colorbox163"><a id="dx1-69005"></a><span class="ectt-1095">refPositionsColValue</span></div>
<!--l. 2782--><p class="noindent">— same definition as <a href="#colvar|rmsd|refPositionsColValue"><span class="ectt-1095">refPositionsColValue</span></a> (<span class="ectt-1095">rmsd </span>component)
     </p></li>
<li class="itemize"><span class="colorbox" id="colorbox164"><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances? </span>(see <a href="#colvar|cvc|forceNoPBC">description</a>)<br class="newline"/>Default: <span class="ectt-1095">no</span></li></ul>
<div class="flushright">
<!--l. 2788--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2788--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_protein"></a>
<h4 class="subsectionHead"><span class="titlemark">4.8    </span> <a id="sec:cvc_protein"></a>Protein structure descriptors</h4>
<!--l. 2788--><p class="noindent"></p></div>
<div class="flushright">
<!--l. 2790--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2790--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_alpha"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.8.1    </span> <a id="sec:cvc_alpha"></a><span class="ectt-1095">alpha</span>: <!--l. 2790--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>α</mi></math>-helix
content of a protein segment.</h5>
<!--l. 2790--><p class="noindent"></p></div> <a id="colvar|alpha"></a>
<!--l. 2793--><p class="indent">   The block <span class="ectt-1095">alpha</span><span class="ectt-1095"> {...} </span>defines a measure of the helical content of a segment of protein
residues, as a tunable combination of an angle term between alpha carbon atoms, and a 1-4
hydrogen-bond term. To create a colvar that computes the helical content of a protein with several,
non-contiguous helical segments, just define several <span class="ectt-1095">alpha </span>blocks inside a single colvar block,
and give them linear combination coefficients that sum to 1 (see <span class="ectt-1095">componentCoeff </span>in section
<a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a>).
</p><!--l. 2800--><p class="indent">   The <!--l. 2800--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>α</mi></math>-helical
content across the <!--l. 2800--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>N</mi> <mo class="MathClass-bin">+</mo><mn>1</mn></math>
residues <!--l. 2800--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>N</mi></mrow><mrow><mn>0</mn></mrow></msub></math> to
<!--l. 2800--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>N</mi></mrow><mrow><mn>0</mn></mrow></msub> <mo class="MathClass-bin">+</mo><mi>N</mi></math> is
calculated by the formula: </p>
<!--tex4ht:inline--><!--l. 2801--><math display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mtable class="eqnarray" columnalign="right center left">
<mtr><mtd class="eqnarray-1"> <mi>α</mi> <mrow><mo fence="true" form="prefix"> (</mo><mrow><msubsup><mrow><mstyle mathvariant="normal"><mi>C</mi></mstyle></mrow><mrow><mi>α</mi></mrow><mrow><mo class="MathClass-open">(</mo><msub><mrow><mi>N</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-close">)</mo></mrow></msubsup><mo class="MathClass-punc">,</mo><msup><mrow><mstyle mathvariant="normal"><mi>O</mi></mstyle></mrow><mrow><mo class="MathClass-open">(</mo><msub><mrow><mi>N</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-close">)</mo></mrow></msup><mo class="MathClass-punc">,</mo><msubsup><mrow><mstyle mathvariant="normal"><mi>C</mi></mstyle></mrow><mrow>
<mi>α</mi></mrow><mrow><mo class="MathClass-open">(</mo><msub><mrow><mi>N</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-bin">+</mo><mn>1</mn><mo class="MathClass-close">)</mo></mrow></msubsup><mo class="MathClass-punc">,</mo><msup><mrow><mstyle mathvariant="normal"><mi>O</mi></mstyle></mrow><mrow><mo class="MathClass-open">(</mo><msub><mrow><mi>N</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-bin">+</mo><mn>1</mn><mo class="MathClass-close">)</mo></mrow></msup><mo class="MathClass-punc">,</mo><mi class="MathClass-op">…</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo><msup><mrow><mstyle mathvariant="normal"><mi>N</mi></mstyle></mrow><mrow><mo class="MathClass-open">(</mo><msub><mrow><mi>N</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-bin">+</mo><mn>5</mn><mo class="MathClass-close">)</mo></mrow></msup><mo class="MathClass-punc">,</mo><msubsup><mrow><mstyle mathvariant="normal"><mi>C</mi></mstyle></mrow><mrow>
<mi>α</mi></mrow><mrow><mo class="MathClass-open">(</mo><msub><mrow><mi>N</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-bin">+</mo><mn>5</mn><mo class="MathClass-close">)</mo></mrow></msubsup><mo class="MathClass-punc">,</mo><msup><mrow><mstyle mathvariant="normal"><mi>O</mi></mstyle></mrow><mrow><mo class="MathClass-open">(</mo><msub><mrow><mi>N</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-bin">+</mo><mn>5</mn><mo class="MathClass-close">)</mo></mrow></msup><mo class="MathClass-punc">,</mo><mi class="MathClass-op">…</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo><msup><mrow><mstyle mathvariant="normal"><mi>N</mi></mstyle></mrow><mrow><mo class="MathClass-open">(</mo><msub><mrow><mi>N</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-bin">+</mo><mi>N</mi><mo class="MathClass-close">)</mo></mrow></msup><mo class="MathClass-punc">,</mo><msubsup><mrow><mstyle mathvariant="normal"><mi>C</mi></mstyle></mrow><mrow>
<mi>α</mi></mrow><mrow><mo class="MathClass-open">(</mo><msub><mrow><mi>N</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-bin">+</mo><mi>N</mi><mo class="MathClass-close">)</mo></mrow></msubsup></mrow><mo fence="true" form="postfix">)</mo></mrow><mspace class="thickpace" width="0.28em"></mspace> <mo class="MathClass-rel">=</mo> <mspace class="thickpace" width="0.28em"></mspace><mspace class="thickpace" width="0.28em"></mspace><mspace class="thickpace" width="0.28em"></mspace><mspace class="thickpace" width="0.28em"></mspace></mtd><mtd class="eqnarray-2"> </mtd><mtd class="eqnarray-3"> </mtd><mtd class="eqnarray-4"> <mtext class="eqnarray">(10)</mtext><mtext class="label" id="x1-71001r10"></mtext><mtext class="endlabel"></mtext></mtd>
</mtr><mtr><mtd class="eqnarray-1"> <mspace class="thickpace" width="0.28em"></mspace><mspace class="thickpace" width="0.28em"></mspace><mspace class="thickpace" width="0.28em"></mspace><mspace class="thickpace" width="0.28em"></mspace><mfrac><mrow><mo class="MathClass-open">(</mo><mn>1</mn><mo class="MathClass-bin">−</mo><msub><mrow><mi>C</mi></mrow><mrow><mstyle class="text"><mtext>hb</mtext></mstyle></mrow></msub><mo class="MathClass-close">)</mo></mrow>
<mrow><mi>N</mi> <mo class="MathClass-bin">−</mo><mn>1</mn></mrow></mfrac> <munderover accent="false" accentunder="false"><mrow><mo>∑</mo></mrow><mrow><mi>n</mi><mo class="MathClass-rel">=</mo><msub><mrow><mi>N</mi></mrow><mrow><mn>0</mn></mrow></msub></mrow><mrow><msub><mrow><mi>N</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-bin">+</mo><mi>N</mi><mo class="MathClass-bin">−</mo><mn>2</mn></mrow></munderover><mstyle mathvariant="normal"><mi>a</mi><mi>n</mi><mi>g</mi><mi>f</mi></mstyle> <mrow><mo fence="true" form="prefix"> (</mo><mrow><msubsup><mrow><mstyle mathvariant="normal"><mi>C</mi></mstyle></mrow><mrow>
<mi>α</mi></mrow><mrow><mo class="MathClass-open">(</mo><mi>n</mi><mo class="MathClass-close">)</mo></mrow></msubsup><mo class="MathClass-punc">,</mo><msubsup><mrow><mstyle mathvariant="normal"><mi>C</mi></mstyle></mrow><mrow>
<mi>α</mi></mrow><mrow><mo class="MathClass-open">(</mo><mi>n</mi><mo class="MathClass-bin">+</mo><mn>1</mn><mo class="MathClass-close">)</mo></mrow></msubsup><mo class="MathClass-punc">,</mo><msubsup><mrow><mstyle mathvariant="normal"><mi>C</mi></mstyle></mrow><mrow>
<mi>α</mi></mrow><mrow><mo class="MathClass-open">(</mo><mi>n</mi><mo class="MathClass-bin">+</mo><mn>2</mn><mo class="MathClass-close">)</mo></mrow></msubsup></mrow><mo fence="true" form="postfix">)</mo></mrow><mspace class="thickpace" width="0.28em"></mspace> <mo class="MathClass-bin">+</mo><mspace class="thickpace" width="0.28em"></mspace> <mfrac><mrow><msub><mrow><mi>C</mi></mrow><mrow><mstyle class="text"><mtext>hb</mtext></mstyle></mrow></msub></mrow>
<mrow><mi>N</mi> <mo class="MathClass-bin">−</mo><mn>3</mn></mrow></mfrac><munderover accent="false" accentunder="false"><mrow><mo>∑</mo></mrow><mrow><mi>n</mi><mo class="MathClass-rel">=</mo><msub><mrow><mi>N</mi></mrow><mrow><mn>0</mn></mrow></msub></mrow><mrow><msub><mrow><mi>N</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-bin">+</mo><mi>N</mi><mo class="MathClass-bin">−</mo><mn>4</mn></mrow></munderover><mstyle mathvariant="normal"><mi>h</mi><mi>b</mi><mi>f</mi></mstyle> <mrow><mo fence="true" form="prefix"> (</mo><mrow><msup><mrow><mstyle mathvariant="normal"><mi>O</mi></mstyle></mrow><mrow><mo class="MathClass-open">(</mo><mi>n</mi><mo class="MathClass-close">)</mo></mrow></msup><mo class="MathClass-punc">,</mo><msup><mrow><mstyle mathvariant="normal"><mi>N</mi></mstyle></mrow><mrow><mo class="MathClass-open">(</mo><mi>n</mi><mo class="MathClass-bin">+</mo><mn>4</mn><mo class="MathClass-close">)</mo></mrow></msup></mrow><mo fence="true" form="postfix">)</mo></mrow><mstyle mathvariant="normal"><mo class="MathClass-punc">,</mo></mstyle></mtd><mtd class="eqnarray-2"> </mtd><mtd class="eqnarray-3"> </mtd><mtd class="eqnarray-4"> <mtext class="eqnarray"></mtext></mtd>
</mtr><mtr><mtd class="eqnarray-1"> </mtd><mtd class="eqnarray-2"> </mtd><mtd class="eqnarray-3"> </mtd><mtd class="eqnarray-4"> <mtext class="eqnarray">(11)</mtext><mtext class="label" id="x1-71001r11"></mtext><mtext class="endlabel"></mtext></mtd> </mtr></mtable>
</math>
<!--l. 2832--><p class="nopar">
</p>
<!--l. 2833--><p class="indent">   where <!--l. 2833--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>C</mi></mrow><mrow><mstyle class="text"><mtext>hb</mtext></mstyle></mrow></msub></math> is defined by
<a href="#colvar|alpha|hBondCoeff"><span class="ectt-1095">hBondCoeff</span></a>, the score function <!--l. 2834--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="normal"><mi>a</mi><mi>n</mi><mi>g</mi><mi>f</mi></mstyle></math>
for the <!--l. 2834--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="normal"><mi>C</mi></mstyle></mrow><mrow><mi>α</mi></mrow></msub><mo class="MathClass-bin">−</mo><msub><mrow><mstyle mathvariant="normal"><mi>C</mi></mstyle></mrow><mrow><mi>α</mi></mrow></msub><mo class="MathClass-bin">−</mo><msub><mrow><mstyle mathvariant="normal"><mi>C</mi></mstyle></mrow><mrow><mi>α</mi></mrow></msub></math>
angle is defined as: </p><table class="equation"><tr><td> <a id="eq:colvars_alpha_Calpha"></a>
<!--l. 2836--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mstyle mathvariant="normal"><mi>a</mi><mi>n</mi><mi>g</mi><mi>f</mi></mstyle> <mrow><mo fence="true" form="prefix"> (</mo><mrow><msubsup><mrow><mstyle mathvariant="normal"><mi>C</mi></mstyle></mrow><mrow><mi>α</mi></mrow><mrow><mo class="MathClass-open">(</mo><mi>n</mi><mo class="MathClass-close">)</mo></mrow></msubsup><mo class="MathClass-punc">,</mo><msubsup><mrow><mstyle mathvariant="normal"><mi>C</mi></mstyle></mrow><mrow>
<mi>α</mi></mrow><mrow><mo class="MathClass-open">(</mo><mi>n</mi><mo class="MathClass-bin">+</mo><mn>1</mn><mo class="MathClass-close">)</mo></mrow></msubsup><mo class="MathClass-punc">,</mo><msubsup><mrow><mstyle mathvariant="normal"><mi>C</mi></mstyle></mrow><mrow>
<mi>α</mi></mrow><mrow><mo class="MathClass-open">(</mo><mi>n</mi><mo class="MathClass-bin">+</mo><mn>2</mn><mo class="MathClass-close">)</mo></mrow></msubsup></mrow><mo fence="true" form="postfix">)</mo></mrow><mspace class="thickpace" width="0.28em"></mspace> <mo class="MathClass-rel">=</mo> <mspace class="thickpace" width="0.28em"></mspace><mfrac><mrow><mn>1</mn><mo class="MathClass-bin">−</mo><msup><mrow><mrow><mo fence="true" form="prefix"> (</mo><mrow><mi>𝜃</mi><mo class="MathClass-open">(</mo><msubsup><mrow><mstyle mathvariant="normal"><mi>C</mi></mstyle></mrow><mrow><mi>α</mi></mrow><mrow><mo class="MathClass-open">(</mo><mi>n</mi><mo class="MathClass-close">)</mo></mrow></msubsup><mo class="MathClass-punc">,</mo><msubsup><mrow><mstyle mathvariant="normal"><mi>C</mi></mstyle></mrow><mrow><mi>α</mi></mrow><mrow><mo class="MathClass-open">(</mo><mi>n</mi><mo class="MathClass-bin">+</mo><mn>1</mn><mo class="MathClass-close">)</mo></mrow></msubsup><mo class="MathClass-punc">,</mo><msubsup><mrow><mstyle mathvariant="normal"><mi>C</mi></mstyle></mrow><mrow><mi>α</mi></mrow><mrow><mo class="MathClass-open">(</mo><mi>n</mi><mo class="MathClass-bin">+</mo><mn>2</mn><mo class="MathClass-close">)</mo></mrow></msubsup><mo class="MathClass-close">)</mo><mo class="MathClass-bin">−</mo><msub><mrow><mi>𝜃</mi></mrow><mrow><mn>0</mn></mrow></msub></mrow><mo fence="true" form="postfix">)</mo></mrow></mrow><mrow><mn>2</mn></mrow></msup><mo class="MathClass-bin">∕</mo><msup><mrow> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mi mathvariant="normal">Δ</mi><msub><mrow><mi>𝜃</mi></mrow><mrow><mstyle mathvariant="normal"><mi>t</mi><mi>o</mi><mi>l</mi></mstyle></mrow></msub></mrow><mo fence="true" form="postfix">)</mo></mrow></mrow><mrow><mn>2</mn></mrow></msup></mrow>
<mrow><mn>1</mn><mo class="MathClass-bin">−</mo><msup><mrow><mrow><mo fence="true" form="prefix"> (</mo><mrow><mi>𝜃</mi><mo class="MathClass-open">(</mo><msubsup><mrow><mstyle mathvariant="normal"><mi>C</mi></mstyle></mrow><mrow><mi>α</mi></mrow><mrow><mo class="MathClass-open">(</mo><mi>n</mi><mo class="MathClass-close">)</mo></mrow></msubsup><mo class="MathClass-punc">,</mo><msubsup><mrow><mstyle mathvariant="normal"><mi>C</mi></mstyle></mrow><mrow><mi>α</mi></mrow><mrow><mo class="MathClass-open">(</mo><mi>n</mi><mo class="MathClass-bin">+</mo><mn>1</mn><mo class="MathClass-close">)</mo></mrow></msubsup><mo class="MathClass-punc">,</mo><msubsup><mrow><mstyle mathvariant="normal"><mi>C</mi></mstyle></mrow><mrow><mi>α</mi></mrow><mrow><mo class="MathClass-open">(</mo><mi>n</mi><mo class="MathClass-bin">+</mo><mn>2</mn><mo class="MathClass-close">)</mo></mrow></msubsup><mo class="MathClass-close">)</mo><mo class="MathClass-bin">−</mo><msub><mrow><mi>𝜃</mi></mrow><mrow><mn>0</mn></mrow></msub></mrow><mo fence="true" form="postfix">)</mo></mrow></mrow><mrow><mn>4</mn></mrow></msup><mo class="MathClass-bin">∕</mo><msup><mrow> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mi mathvariant="normal">Δ</mi><msub><mrow><mi>𝜃</mi></mrow><mrow><mstyle mathvariant="normal"><mi>t</mi><mi>o</mi><mi>l</mi></mstyle></mrow></msub></mrow><mo fence="true" form="postfix">)</mo></mrow></mrow><mrow><mn>4</mn></mrow></msup></mrow></mfrac><mstyle mathvariant="normal"><mo class="MathClass-punc">,</mo></mstyle>
</math></td><td class="eq-no">(12)</td></tr></table>
<!--l. 2858--><p class="indent">   and the score function <!--l. 2858--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="normal"><mi>h</mi><mi>b</mi><mi>f</mi></mstyle></math>
for the <!--l. 2858--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow><mstyle mathvariant="normal"><mi>O</mi></mstyle></mrow><mrow><mo class="MathClass-open">(</mo><mi>n</mi><mo class="MathClass-close">)</mo></mrow></msup> <mo class="MathClass-rel">↔</mo><msup><mrow><mstyle mathvariant="normal"><mi>N</mi></mstyle></mrow><mrow><mo class="MathClass-open">(</mo><mi>n</mi><mo class="MathClass-bin">+</mo><mn>4</mn><mo class="MathClass-close">)</mo></mrow></msup></math>
hydrogen bond is defined through a <span class="ectt-1095">hBond </span>colvar component on the same atoms.
</p><!--l. 2863--><p class="noindent"><span class="phvb8t-x-x-109">List of keywords </span>(see also <a href="#sec:cvc_pbcs">4.2<!--tex4ht:ref: sec:cvc_pbcs --></a>, <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>, <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> and <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="colvar|alpha|prefix"></a> <span class="colorbox" id="colorbox165"><a id="dx1-71003"></a><span class="ectt-1095">prefix</span></span> — <span class="phvro8t-x-x-109">Prefix of atom groups to be used for computing the </span><!--l. 2879--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>α</mi></math><span class="phvro8t-x-x-109">-helix</span>
<span class="phvro8t-x-x-109">content</span><br class="newline"/>Default: <span class="ectt-1095">alpha_</span> <!--l. 2879--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> string,
     context: <span class="ectt-1095">alpha</span> <!--l. 2879--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Atoms involved in an alpha-helix component are based on groups defined within a GROMACS-style
     index file specified with <a href="#Colvars-global|indexFile"><span class="ectt-1095">indexFile</span></a> in the global Colvars configuration, i.e. outside of a <span class="ectt-1095">colvar</span>
     block. The group names must use a common prefix followed by atom names, e.g. <span class="ectt-1095">alpha_N</span>,
     <span class="ectt-1095">alpha_CA</span>, and <span class="ectt-1095">alpha_O </span>when using the default prefix <span class="ectt-1095">alpha_</span>. The CA group is only required
     when the angle term is enabled, that is when <span class="ectt-1095">hBondCoeff </span>is less than 1. This option lets users
     set a different prefix, for example, when several <!--l. 2879--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>α</mi></math>-helical
                                                                                         
                                                                                         
     segments must be tracked.
     </li>
<li class="itemize"><a id="colvar|alpha|residueRange"></a> <span class="colorbox" id="colorbox166"><a id="dx1-71004"></a><span class="ectt-1095">residueRange</span></span> — <span class="phvro8t-x-x-109">Potential </span><!--l. 2907--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>α</mi></math><span class="phvro8t-x-x-109">-helical</span>
<span class="phvro8t-x-x-109">residues</span><br class="newline"/><!--l. 2907--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> “<!--l. 2907--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>Initial
     residue number<!--l. 2907--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>-<!--l. 2907--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>Final
     residue number<!--l. 2907--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>",
     context: <span class="ectt-1095">alpha</span> <!--l. 2907--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>In some back-ends like NAMD and VMD, the atoms may be defined using their PSF segment
     name (segname) and a range of residue numbers. This option specifies the range of residues
     on which this component should be defined. The Colvars module looks for the atoms within
     these residues named “<span class="ectt-1095">CA</span>", “<span class="ectt-1095">N</span>" and “<span class="ectt-1095">O</span>", and raises an error if any of those atoms is not found.
     </li>
<li class="itemize"><a id="colvar|alpha|psfSegID"></a> <span class="colorbox" id="colorbox167"><a id="dx1-71005"></a><span class="ectt-1095">psfSegID</span></span> — <span class="phvro8t-x-x-109">PSF segment identifier</span><br class="newline"/><!--l. 2907--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> string
     (max 4 characters), context: <span class="ectt-1095">alpha</span> <!--l. 2907--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option sets the PSF segment identifier for the residues specified in <span class="ectt-1095">residueRange</span>. This
     option is only required when PSF topologies are used.
     </li>
<li class="itemize"><a id="colvar|alpha|hBondCoeff"></a> <span class="colorbox" id="colorbox168"><a id="dx1-71006"></a><span class="ectt-1095">hBondCoeff</span></span> — <span class="phvro8t-x-x-109">Coefficient for the hydrogen bond term</span><br class="newline"/>Default: 0.5 <!--l. 2919--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     between 0 and 1, context: <span class="ectt-1095">alpha</span> <!--l. 2919--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This number specifies the contribution to the total value from the hydrogen bond terms. 0
     disables the hydrogen bond terms, 1 disables the angle terms.
     </li>
<li class="itemize"><a id="colvar|alpha|angleRef"></a> <span class="colorbox" id="colorbox169"><a id="dx1-71007"></a><span class="ectt-1095">angleRef</span></span> — <span class="phvro8t-x-x-109">Reference </span><!--l. 2931--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="normal"><mi>C</mi></mstyle></mrow><mrow><mi>α</mi></mrow></msub><mo class="MathClass-bin">−</mo><msub><mrow><mstyle mathvariant="normal"><mi>C</mi></mstyle></mrow><mrow><mi>α</mi></mrow></msub><mo class="MathClass-bin">−</mo><msub><mrow><mstyle mathvariant="normal"><mi>C</mi></mstyle></mrow><mrow><mi>α</mi></mrow></msub></math>
<span class="phvro8t-x-x-109">angle</span><br class="newline"/>Default: 88<!--l. 2931--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></math> <!--l. 2931--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     decimal, context: <span class="ectt-1095">alpha</span> <!--l. 2931--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option sets the reference angle used in the score function (<a href="#eq:colvars_alpha_Calpha">12<!--tex4ht:ref: eq:colvars_alpha_Calpha --></a>).
     </li>
<li class="itemize"><a id="colvar|alpha|angleTol"></a> <span class="colorbox" id="colorbox170"><a id="dx1-71008"></a><span class="ectt-1095">angleTol</span></span> — <span class="phvro8t-x-x-109">Tolerance in the </span><!--l. 2943--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="normal"><mi>C</mi></mstyle></mrow><mrow><mi>α</mi></mrow></msub><mo class="MathClass-bin">−</mo><msub><mrow><mstyle mathvariant="normal"><mi>C</mi></mstyle></mrow><mrow><mi>α</mi></mrow></msub><mo class="MathClass-bin">−</mo><msub><mrow><mstyle mathvariant="normal"><mi>C</mi></mstyle></mrow><mrow><mi>α</mi></mrow></msub></math>
<span class="phvro8t-x-x-109">angle</span><br class="newline"/>Default: 15<!--l. 2943--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></math> <!--l. 2943--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     decimal, context: <span class="ectt-1095">alpha</span> <!--l. 2943--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option sets the angle tolerance used in the score function (<a href="#eq:colvars_alpha_Calpha">12<!--tex4ht:ref: eq:colvars_alpha_Calpha --></a>).
     </li>
<li class="itemize"><a id="colvar|alpha|hBondCutoff"></a> <span class="colorbox" id="colorbox171"><a id="dx1-71009"></a><span class="ectt-1095">hBondCutoff</span></span> — <span class="phvro8t-x-x-109">Hydrogen bond cutoff</span><br class="newline"/>Default: 3.3 Å <!--l. 2954--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     decimal, context: <span class="ectt-1095">alpha</span> <!--l. 2954--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Equivalent to the <span class="ectt-1095">cutoff </span>option in the <span class="ectt-1095">hBond </span>component.
     </li>
<li class="itemize"><a id="colvar|alpha|hBondExpNumer"></a> <span class="colorbox" id="colorbox172"><a id="dx1-71010"></a><span class="ectt-1095">hBondExpNumer</span></span> — <span class="phvro8t-x-x-109">Hydrogen bond numerator exponent</span><br class="newline"/>Default: 6 <!--l. 2965--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     integer, context: <span class="ectt-1095">alpha</span> <!--l. 2965--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Equivalent to the <span class="ectt-1095">expNumer </span>option in the <span class="ectt-1095">hBond </span>component.
     </li>
<li class="itemize"><a id="colvar|alpha|hBondExpDenom"></a> <span class="colorbox" id="colorbox173"><a id="dx1-71011"></a><span class="ectt-1095">hBondExpDenom</span></span> — <span class="phvro8t-x-x-109">Hydrogen bond denominator exponent</span><br class="newline"/>Default: 8 <!--l. 2976--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     integer, context: <span class="ectt-1095">alpha</span> <!--l. 2976--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Equivalent to the <span class="ectt-1095">expDenom </span>option in the <span class="ectt-1095">hBond </span>component.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox174"><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances? </span>(see <a href="#colvar|cvc|forceNoPBC">description</a>)<br class="newline"/>Default: <span class="ectt-1095">no</span></li></ul>
<!--l. 2980--><p class="indent">   This component returns positive values, always comprised between 0 (lowest
<!--l. 2981--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>α</mi></math>-helical score) and
1 (highest <!--l. 2981--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>α</mi></math>-helical
score).
</p>
<div class="flushright">
<!--l. 2985--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 2985--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_dihedralPC"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.8.2    </span> <a id="sec:cvc_dihedralPC"></a><span class="ectt-1095">dihedralPC</span>: protein dihedral principal component</h5>
<!--l. 2985--><p class="noindent"></p></div> <a id="colvar|dihedralPC"></a>
<!--l. 2988--><p class="indent">   The block <span class="ectt-1095">dihedralPC</span><span class="ectt-1095"> {...} </span>defines the parameters of the projection of backbone
dihedral angles within a protein segment onto a <span class="phvro8t-x-x-109">dihedral principal component</span>, following
the formalism of dihedral principal component analysis (dPCA) proposed by Mu et
al.<span class="cite">[<a href="#XMu2005">7</a>]</span> and documented in detail by Altis et al.<span class="cite">[<a href="#XAltis2007">8</a>]</span>. Given a peptide or protein segment of
<!--l. 2994--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>N</mi></math> residues, each with
Ramachandran angles <!--l. 2995--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ϕ</mi></mrow><mrow><mi>i</mi></mrow></msub></math>
and <!--l. 2995--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ψ</mi></mrow><mrow><mi>i</mi></mrow></msub></math>, dPCA rests on a
variance/covariance analysis of the <!--l. 2996--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>4</mn><mo class="MathClass-open">(</mo><mi>N</mi> <mo class="MathClass-bin">−</mo><mn>1</mn><mo class="MathClass-close">)</mo></math>
variables <!--l. 2996--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi class="qopname">cos</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--><mo class="MathClass-open">(</mo><msub><mrow><mi>ψ</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-close">)</mo><mo class="MathClass-punc">,</mo><mi class="qopname">sin</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--><mo class="MathClass-open">(</mo><msub><mrow><mi>ψ</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-close">)</mo><mo class="MathClass-punc">,</mo><mi class="qopname">cos</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--><mo class="MathClass-open">(</mo><msub><mrow><mi>ϕ</mi></mrow><mrow><mn>2</mn></mrow></msub><mo class="MathClass-close">)</mo><mo class="MathClass-punc">,</mo><mi class="qopname">sin</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--><mo class="MathClass-open">(</mo><msub><mrow><mi>ϕ</mi></mrow><mrow><mn>2</mn></mrow></msub><mo class="MathClass-close">)</mo><mo class="MathClass-rel">⋯</mo><mi class="qopname">cos</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--><mo class="MathClass-open">(</mo><msub><mrow><mi>ϕ</mi></mrow><mrow><mi>N</mi></mrow></msub><mo class="MathClass-close">)</mo><mo class="MathClass-punc">,</mo><mi class="qopname">sin</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--><mo class="MathClass-open">(</mo><msub><mrow><mi>ϕ</mi></mrow><mrow><mi>N</mi></mrow></msub><mo class="MathClass-close">)</mo></math>. Note
that angles <!--l. 2997--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ϕ</mi></mrow><mrow><mn>1</mn></mrow></msub></math>
and <!--l. 2997--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ψ</mi></mrow><mrow><mi>N</mi></mrow></msub></math>
have little impact on chain conformation, and are therefore discarded, following the implementation of
dPCA in the analysis software Carma.<span class="cite">[<a href="#XGlykos2006">9</a>]</span>
</p><!--l. 3001--><p class="indent">   For a given principal component (eigenvector) of coefficients
<!--l. 3002--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mo class="MathClass-open">(</mo><msub><mrow><mi>k</mi></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-close">)</mo></mrow><mrow><mn>1</mn><mo class="MathClass-rel">≤</mo><mi>i</mi><mo class="MathClass-rel">≤</mo><mn>4</mn><mo class="MathClass-open">(</mo><mi>N</mi><mo class="MathClass-bin">−</mo><mn>1</mn><mo class="MathClass-close">)</mo></mrow></msub></math>, the
projection of the current backbone conformation is: </p><table class="equation"><tr><td> <a id="x1-72001r13"></a>
<!--l. 3004--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>ξ</mi> <mo class="MathClass-rel">=</mo><munderover accent="false" accentunder="false"><mrow><mo> ∑</mo></mrow><mrow><mi>n</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><mi>N</mi><mo class="MathClass-bin">−</mo><mn>1</mn></mrow></munderover><msub><mrow><mi>k</mi></mrow><mrow>
<mn>4</mn><mi>n</mi><mo class="MathClass-bin">−</mo><mn>3</mn></mrow></msub><mi class="qopname">cos</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--><mo class="MathClass-open">(</mo><msub><mrow><mi>ψ</mi></mrow><mrow><mi>n</mi></mrow></msub><mo class="MathClass-close">)</mo><mo class="MathClass-bin">+</mo><msub><mrow><mi>k</mi></mrow><mrow><mn>4</mn><mi>n</mi><mo class="MathClass-bin">−</mo><mn>2</mn></mrow></msub><mi class="qopname">sin</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--><mo class="MathClass-open">(</mo><msub><mrow><mi>ψ</mi></mrow><mrow><mi>n</mi></mrow></msub><mo class="MathClass-close">)</mo><mo class="MathClass-bin">+</mo><msub><mrow><mi>k</mi></mrow><mrow><mn>4</mn><mi>n</mi><mo class="MathClass-bin">−</mo><mn>1</mn></mrow></msub><mi class="qopname">cos</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--><mo class="MathClass-open">(</mo><msub><mrow><mi>ϕ</mi></mrow><mrow><mi>n</mi><mo class="MathClass-bin">+</mo><mn>1</mn></mrow></msub><mo class="MathClass-close">)</mo><mo class="MathClass-bin">+</mo><msub><mrow><mi>k</mi></mrow><mrow><mn>4</mn><mi>n</mi></mrow></msub><mi class="qopname">sin</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--><mo class="MathClass-open">(</mo><msub><mrow><mi>ϕ</mi></mrow><mrow><mi>n</mi><mo class="MathClass-bin">+</mo><mn>1</mn></mrow></msub><mo class="MathClass-close">)</mo>
</math></td><td class="eq-no">(13)</td></tr></table>
<!--l. 3009--><p class="indent">   The atoms involved in <span class="ectt-1095">dihedralPC </span>are defined by the same parameters as the <span class="ectt-1095">alpha </span>component.
</p><!--l. 3012--><p class="noindent"><span class="phvb8t-x-x-109">List of keywords </span>(see also <a href="#sec:cvc_pbcs">4.2<!--tex4ht:ref: sec:cvc_pbcs --></a>, <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>, <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> and <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="colvar|dihedralPC|prefix"></a> <span class="colorbox" id="colorbox175"><a id="dx1-72002"></a><span class="ectt-1095">prefix</span></span> — <span class="phvro8t-x-x-109">Prefix of atom groups to be used for computing a dihedralPC projection</span><br class="newline"/>Default: <span class="ectt-1095">dihed_</span> <!--l. 3025--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> string,
     context: <span class="ectt-1095">dihedralPC</span> <!--l. 3025--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Backbone atoms involved in a <span class="ectt-1095">dihedralPC </span>component are based on groups defined within a
     GROMACS-style index file specified with <a href="#Colvars-global|indexFile"><span class="ectt-1095">indexFile</span></a> in the global Colvars configuration, i.e.
     outside of a <span class="ectt-1095">colvar </span>block. The group names must use a common prefix followed by atom
     names CA, N, and C, e.g. <span class="ectt-1095">dihed_CA</span>, <span class="ectt-1095">dihed_N</span>, and <span class="ectt-1095">dihed_C </span>when using the default prefix
     <span class="ectt-1095">dihed_</span>.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox176"><a id="dx1-72003"></a><span class="ectt-1095">residueRange</span></span> — same definition as <a href="#colvar|alpha|residueRange"><span class="ectt-1095">residueRange</span></a> (<span class="ectt-1095">alpha </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox177"><a id="dx1-72004"></a><span class="ectt-1095">psfSegID</span></span> — same definition as <a href="#colvar|alpha|psfSegID"><span class="ectt-1095">psfSegID</span></a> (<span class="ectt-1095">alpha </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox178"><a id="dx1-72005"></a><span class="ectt-1095">vectorFile</span></span> — <span class="phvro8t-x-x-109">File containing dihedral PCA eigenvector(s)</span><br class="newline"/><!--l. 3043--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> file
     name, context: <span class="ectt-1095">dihedralPC</span> <!--l. 3043--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>A text file containing the coefficients of dihedral PCA eigenvectors on the cosine and sine
     coordinates. The vectors should be arranged in columns, as in the files output by Carma.<span class="cite">[<a href="#XGlykos2006">9</a>]</span>
</li>
<li class="itemize"><span class="colorbox" id="colorbox179"><a id="dx1-72006"></a><span class="ectt-1095">vectorNumber</span></span> — <span class="phvro8t-x-x-109">File containing dihedralPCA eigenvector(s)</span><br class="newline"/><!--l. 3051--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     integer, context: <span class="ectt-1095">dihedralPC</span> <!--l. 3051--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Number of the eigenvector to be used for this component.
                                                                                         
                                                                                         
     </li>
<li class="itemize"><span class="colorbox" id="colorbox180"><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances? </span>(see <a href="#colvar|cvc|forceNoPBC">description</a>)<br class="newline"/>Default: <span class="ectt-1095">no</span></li></ul>
<div class="flushright">
<!--l. 3134--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3134--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_alch"></a>
<h4 class="subsectionHead"><span class="titlemark">4.9    </span> <a id="sec:cvc_alch"></a>Alchemical variables for lambda-dynamics</h4>
<!--l. 3134--><p class="noindent"></p></div>
<div class="flushright">
<!--l. 3134--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3134--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_alchlambda"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.9.1    </span> <a id="sec:cvc_alchlambda"></a><span class="ectt-1095">alchLambda</span>: alchemical lambda parameter.</h5>
<!--l. 3134--><p class="noindent"></p></div> <a id="colvar|alchLambda"></a>
<!--l. 3134--><p class="indent">   The <span class="ectt-1095">alchLambda </span>block defines a component returning the alchemical lambda parameter of the
back-end simulation in NAMD, provided that it is enabled within that software. Specifically, the flag <span class="ectt-1095">alchOn</span>
should be enabled, and <span class="ectt-1095">alchType </span>should be set to <span class="ectt-1095">TI</span>. The current lambda value is obtained from the
back-end at the beginning of a simulation, and synchronized from Colvars to NAMD at every following
time step. This enables lambda-dynamics simulations as documented in detail in <span class="cite">[<a href="#XLagardere2023">10</a>]</span>. The <span class="ectt-1095">alchLambda</span>
keyword itself takes no parameters.
</p><!--l. 3134--><p class="indent">   To run lambda-dynamics, the enclosing <span class="ectt-1095">colvar </span>block should contain extended-system parameters
(<a href="#sec:colvar_extended">4.24<!--tex4ht:ref: sec:colvar_extended --></a>) as in the following example:</p>
<div class="mdframed" id="mdframed-43">
<!--l. 3134--><p class="noindent"></p><pre>colvar {<br class="newline"/>  name lambda<br class="newline"/>  extendedLagrangian on<br class="newline"/>  extendedMass 150000<br class="newline"/>  extendedLangevinDamping 1000<br class="newline"/><br class="newline"/>  lowerBoundary 0<br class="newline"/>  upperBoundary 1<br class="newline"/>  reflectingLowerBoundary<br class="newline"/>  reflectingUpperBoundary<br class="newline"/><br class="newline"/>  alchLambda<br class="newline"/>} </pre><!--l. 3134--><p class="noindent"></p>
</div>
<!--l. 3134--><p class="indent">   These parameter values have been found to give stable dynamics for a wide range of applications<span class="cite">[<a href="#XLagardere2023">10</a>]</span>.
Reflecting boundary conditions are recommended for lambda-dynamics to ensure that the alchemical
variable remains precisely within its physical range.
</p><!--l. 3134--><p class="indent">   The NAMD configuration file should also contain the parameters for setting up the alchemical
simulation, for example:</p>
<div class="mdframed" id="mdframed-44">
<!--l. 3134--><p class="noindent"></p><pre>computeEnergies         1<br class="newline"/><br class="newline"/>alch                    on<br class="newline"/>alchType                TI<br class="newline"/>alchFile                ${alchFile}.pdb<br class="newline"/>alchCol                 B<br class="newline"/>alchOutFile             ${outName}.fepout<br class="newline"/>alchOutFreq             100<br class="newline"/>alchVdwLambdaEnd        0.5<br class="newline"/>alchElecLambdaStart     0.5<br class="newline"/>alchVdWShiftCoeff       6.0<br class="newline"/>alchlambda              0<br class="newline"/></pre><!--l. 3134--><p class="noindent"></p>
</div>
<!--l. 3134--><p class="indent">   Here, the value of 1 for <span class="ectt-1095">computeEnergies </span>and TI for <span class="ectt-1095">alchType </span>are mandatory. All other parameters can
be chosen freely as documented in the NAMD User's Guide.
</p>
<!--l. 3134--><p class="noindent"><span class="paragraphHead"><a id="x1-750004.9.1"></a><span class="phvb8t-x-x-109">Special considerations for running lambda-dynamics in NAMD/Colvars.</span></span>
   As of NAMD 3.0, lambda-dynamics is not compatible with multiple timestepping. We recommend
applying hydrogen mass repartitioning (HMR) to the topology and increasing the base timestep to 2 fs to
limit the performance impact of disabling multiple timestepping. When running in parallel on CPUs,
synchronization issues prevent lambda-dynamics trajectories from being fully reproducible. This can be
circumvented by running NAMD in GPU-resident mode.
</p>
<div class="flushright">
<!--l. 3134--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3134--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_alchFlambda"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.9.2    </span> <a id="sec:cvc_alchFlambda"></a><span class="ectt-1095">alchFlambda</span>: Force on the alchemical lambda parameter.</h5>
<!--l. 3134--><p class="noindent"></p></div> <a id="colvar|alchFlambda"></a>
<!--l. 3134--><p class="indent">   The <span class="ectt-1095">alchFlambda</span><span class="ectt-1095"> {} </span>block defines a component returning the force exterted on the alchemical
lambda parameter of the back-end simulation in NAMD, provided that it is enabled using
                                                                                         
                                                                                         
the appropriate options of NAMD. This coordinate is obtained from the back-end at each
time step of a simulation. The <span class="ectt-1095">alchFlambda </span>block itself takes no parameters, it should be left
empty.
</p>
<div class="flushright">
<!--l. 3138--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3138--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_raw"></a>
<h4 class="subsectionHead"><span class="titlemark">4.10    </span> <a id="sec:cvc_raw"></a>Raw data: building blocks for custom functions</h4>
<!--l. 3138--><p class="noindent"></p></div>
<div class="flushright">
<!--l. 3140--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3140--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_cartesian"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.10.1    </span> <a id="sec:cvc_cartesian"></a><span class="ectt-1095">cartesian</span>: vector of atomic Cartesian coordinates.</h5>
<!--l. 3140--><p class="noindent"></p></div> <a id="colvar|cartesian"></a>
<!--l. 3143--><p class="indent">   The <span class="ectt-1095">cartesian</span><span class="ectt-1095"> {...} </span>block defines a component returning a flat vector
containing the Cartesian coordinates of all participating atoms, in the order
<!--l. 3145--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">(</mo><msub><mrow><mi>x</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>y</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>z</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-punc">,</mo><mo class="MathClass-rel">⋯</mo><mspace class="thinspace" width="0.17em"></mspace><mo class="MathClass-punc">,</mo><msub><mrow><mi>x</mi></mrow><mrow><mi>n</mi></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>y</mi></mrow><mrow><mi>n</mi></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>z</mi></mrow><mrow><mi>n</mi></mrow></msub><mo class="MathClass-close">)</mo></math>.
</p><!--l. 3147--><p class="noindent"><span class="phvb8t-x-x-109">List of keywords </span>(see also <a href="#sec:cvc_pbcs">4.2<!--tex4ht:ref: sec:cvc_pbcs --></a>, <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>, <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> and <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox181"><a id="dx1-78001"></a><span class="ectt-1095">atoms</span></span> — <span class="phvro8t-x-x-109">Group of atoms</span><br class="newline"/><!--l. 3156--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> <a href="#sec:colvar_atom_groups">Atom
     group</a>, context: <span class="ectt-1095">cartesian</span> <!--l. 3156--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Defines the atoms whose coordinates make up the value of the component. If <span class="ectt-1095">rotateToReference</span>,
     <span class="ectt-1095">centerToReference</span>, or <span class="ectt-1095">centerToOrigin </span>are defined, coordinates are evaluated within the
     moving frame of reference.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox182"><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances? </span>(see <a href="#colvar|cvc|forceNoPBC">description</a>)<br class="newline"/>Default: <span class="ectt-1095">no</span></li></ul>
<div class="flushright">
<!--l. 3160--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3160--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_distancePairs"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.10.2    </span> <a id="sec:cvc_distancePairs"></a><span class="ectt-1095">distancePairs</span>: set of pairwise distances between two groups.</h5>
<!--l. 3160--><p class="noindent"></p></div> <a id="colvar|distancePairs"></a>
<!--l. 3163--><p class="indent">   The  <span class="ectt-1095">distancePairs</span><span class="ectt-1095"> {...} </span>block  defines  a
<!--l. 3163--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>N</mi></mrow><mrow><mstyle mathvariant="normal"><mn>1</mn></mstyle></mrow></msub> <mo class="MathClass-bin">×</mo><msub><mrow><mi>N</mi></mrow><mrow><mstyle mathvariant="normal"><mn>2</mn></mstyle></mrow></msub></math>-dimensional
variable that includes all mutual distances between the atoms of two groups. This can be useful, for
example, to develop a new variable defined over two groups, by using the <span class="ectt-1095">scriptedFunction</span>
feature.
</p><!--l. 3166--><p class="noindent"><span class="phvb8t-x-x-109">List of keywords </span>(see also <a href="#sec:cvc_pbcs">4.2<!--tex4ht:ref: sec:cvc_pbcs --></a>, <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>, <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> and <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox183"><a id="dx1-79001"></a><span class="ectt-1095">group1</span></span> — same definition as <a href="#colvar|distance|group1"><span class="ectt-1095">group1</span></a> (<span class="ectt-1095">distance </span>component)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox184"><a id="dx1-79002"></a><span class="ectt-1095">group2</span></span> — analogous to <span class="ectt-1095">group1</span>
</li>
<li class="itemize"><span class="colorbox" id="colorbox185"><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances? </span>(see <a href="#colvar|cvc|forceNoPBC">description</a>)<br class="newline"/>Default: <span class="ectt-1095">no</span></li></ul>
<!--l. 3172--><p class="noindent">This component returns a <!--l. 3172--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>N</mi></mrow><mrow><mstyle mathvariant="normal"><mn>1</mn></mstyle></mrow></msub> <mo class="MathClass-bin">×</mo><msub><mrow><mi>N</mi></mrow><mrow><mstyle mathvariant="normal"><mn>2</mn></mstyle></mrow></msub></math>-dimensional
vector of numbers, each ranging from <!--l. 3172--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>0</mn></math>
to the largest possible distance within the chosen boundary conditions.
</p>
<div class="flushright">
<!--l. 3175--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3175--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_gpath"></a>
<h4 class="subsectionHead"><span class="titlemark">4.11    </span> <a id="sec:cvc_gpath"></a>Geometric path collective variables</h4>
<!--l. 3175--><p class="noindent"></p></div>
<!--l. 3177--><p class="indent">   The geometric path collective variables define the progress along a path,
<!--l. 3177--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>s</mi></math>, and the distance
from the path, <!--l. 3177--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math>.
These CVs are proposed by Leines and Ensing<span class="cite">[<a href="#XLeines2012">11</a>]</span> , which differ from that<span class="cite">[<a href="#XBranduardi2007">12</a>]</span> proposed
by Branduardi et al., and utilize a set of geometric algorithms. The path is defined
as a series of frames in the atomic Cartesian coordinate space or the CV space.
<!--l. 3177--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>s</mi></math> and
                                                                                         
                                                                                         
<!--l. 3177--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math> are
computed as
</p>
<table class="equation"><tr><td> <a id="x1-80001r14"></a>
<!--l. 3179--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>s</mi> <mo class="MathClass-rel">=</mo> <mfrac><mrow><mi>m</mi></mrow>
<mrow><mi>M</mi></mrow></mfrac> <mo class="MathClass-bin">±</mo> <mfrac><mrow><mn>1</mn></mrow>
<mrow><mn>2</mn><mi>M</mi></mrow></mfrac> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mfrac><mrow><msqrt><mrow><msup><mrow><mo class="MathClass-open">(</mo><msub><mrow><mstyle mathvariant="bold"><mi>v</mi></mstyle> </mrow><mrow><mn>1</mn> </mrow> </msub> <mo class="MathClass-bin">⋅</mo><msub><mrow><mstyle mathvariant="bold"><mi>v</mi></mstyle> </mrow><mrow><mn>3</mn> </mrow> </msub> <mo class="MathClass-close">)</mo></mrow><mrow><mn>2</mn> </mrow> </msup> <mo class="MathClass-bin">−</mo><mo class="MathClass-rel">|</mo><msub><mrow><mstyle mathvariant="bold"><mi>v</mi></mstyle> </mrow><mrow><mn>3</mn> </mrow> </msub> <msup><mrow><mo class="MathClass-rel">|</mo></mrow><mrow><mn>2</mn> </mrow> </msup> <mo class="MathClass-open">(</mo><mo class="MathClass-rel">|</mo><msub><mrow><mstyle mathvariant="bold"><mi>v</mi></mstyle> </mrow><mrow><mn>1</mn> </mrow> </msub> <msup><mrow><mo class="MathClass-rel">|</mo></mrow><mrow><mn>2</mn> </mrow> </msup> <mo class="MathClass-bin">−</mo><mo class="MathClass-rel">|</mo><msub><mrow><mstyle mathvariant="bold"><mi>v</mi></mstyle> </mrow><mrow><mn>2</mn> </mrow> </msub> <msup><mrow><mo class="MathClass-rel">|</mo></mrow><mrow><mn>2</mn> </mrow> </msup> <mo class="MathClass-close">)</mo></mrow></msqrt><mo class="MathClass-bin">−</mo><mo class="MathClass-open">(</mo><msub><mrow><mstyle mathvariant="bold"><mi>v</mi></mstyle></mrow><mrow><mn>1</mn></mrow></msub> <mo class="MathClass-bin">⋅</mo><msub><mrow><mstyle mathvariant="bold"><mi>v</mi></mstyle></mrow><mrow><mn>3</mn></mrow></msub><mo class="MathClass-close">)</mo></mrow>
<mrow><mo class="MathClass-rel">|</mo><msub><mrow><mstyle mathvariant="bold"><mi>v</mi></mstyle></mrow><mrow><mn>3</mn></mrow></msub><msup><mrow><mo class="MathClass-rel">|</mo></mrow><mrow><mn>2</mn></mrow></msup></mrow></mfrac> <mo class="MathClass-bin">−</mo><mn>1</mn></mrow><mo fence="true" form="postfix">)</mo></mrow>
</math></td><td class="eq-no">(14)</td></tr></table>
<table class="equation"><tr><td> <a id="x1-80002r15"></a>
<!--l. 3183--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>z</mi> <mo class="MathClass-rel">=</mo> <msqrt><mrow><msup><mrow><mrow><mo fence="true" form="prefix"> (</mo><mrow><msub><mrow><mstyle mathvariant="bold"><mi>v</mi></mstyle> </mrow><mrow><mn>1</mn> </mrow> </msub> <mo class="MathClass-bin">+</mo> <mfrac> <mrow> <mn>1</mn></mrow>
<mrow><mn>2</mn></mrow></mfrac> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mfrac><mrow><msqrt><mrow><msup><mrow><mo class="MathClass-open">(</mo><msub><mrow><mstyle mathvariant="bold"><mi>v</mi></mstyle> </mrow><mrow><mn>1</mn> </mrow> </msub> <mo class="MathClass-bin">⋅</mo><msub><mrow><mstyle mathvariant="bold"><mi>v</mi></mstyle> </mrow><mrow><mn>3</mn> </mrow> </msub> <mo class="MathClass-close">)</mo></mrow><mrow><mn>2</mn> </mrow> </msup> <mo class="MathClass-bin">−</mo><mo class="MathClass-rel">|</mo><msub><mrow><mstyle mathvariant="bold"><mi>v</mi></mstyle> </mrow><mrow><mn>3</mn> </mrow> </msub> <msup><mrow><mo class="MathClass-rel">|</mo></mrow><mrow><mn>2</mn> </mrow> </msup> <mo class="MathClass-open">(</mo><mo class="MathClass-rel">|</mo><msub><mrow><mstyle mathvariant="bold"><mi>v</mi></mstyle> </mrow><mrow><mn>1</mn> </mrow> </msub> <msup><mrow><mo class="MathClass-rel">|</mo></mrow><mrow><mn>2</mn> </mrow> </msup> <mo class="MathClass-bin">−</mo><mo class="MathClass-rel">|</mo><msub><mrow><mstyle mathvariant="bold"><mi>v</mi></mstyle> </mrow><mrow><mn>2</mn> </mrow> </msub> <msup><mrow><mo class="MathClass-rel">|</mo></mrow><mrow><mn>2</mn> </mrow> </msup> <mo class="MathClass-close">)</mo></mrow></msqrt><mo class="MathClass-bin">−</mo><mo class="MathClass-open">(</mo><msub><mrow><mstyle mathvariant="bold"><mi>v</mi></mstyle></mrow><mrow><mn>1</mn></mrow></msub> <mo class="MathClass-bin">⋅</mo><msub><mrow><mstyle mathvariant="bold"><mi>v</mi></mstyle></mrow><mrow><mn>3</mn></mrow></msub><mo class="MathClass-close">)</mo></mrow>
<mrow><mo class="MathClass-rel">|</mo><msub><mrow><mstyle mathvariant="bold"><mi>v</mi></mstyle></mrow><mrow><mn>3</mn></mrow></msub><msup><mrow><mo class="MathClass-rel">|</mo></mrow><mrow><mn>2</mn></mrow></msup></mrow></mfrac> <mo class="MathClass-bin">−</mo><mn>1</mn></mrow><mo fence="true" form="postfix">)</mo></mrow><msub><mrow><mstyle mathvariant="bold"><mi>v</mi></mstyle></mrow><mrow><mn>4</mn></mrow></msub></mrow><mo fence="true" form="postfix">)</mo></mrow></mrow><mrow><mn>2</mn></mrow></msup></mrow></msqrt>
</math></td><td class="eq-no">(15)</td></tr></table>
<!--l. 3187--><p class="indent">   where <!--l. 3187--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>v</mi></mstyle></mrow><mrow><mn>1</mn></mrow></msub> <mo class="MathClass-rel">=</mo><msub><mrow> <mstyle mathvariant="bold"><mi>s</mi></mstyle></mrow><mrow><mi>m</mi></mrow></msub><mo class="MathClass-bin">−</mo><mstyle mathvariant="bold"><mi>z</mi></mstyle></math>
is the vector connecting the current position to the closest frame,
<!--l. 3187--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>v</mi></mstyle></mrow><mrow><mn>2</mn></mrow></msub> <mo class="MathClass-rel">=</mo> <mstyle mathvariant="bold"><mi>z</mi></mstyle> <mo class="MathClass-bin">−</mo><msub><mrow><mstyle mathvariant="bold"><mi>s</mi></mstyle></mrow><mrow><mi>m</mi><mo class="MathClass-bin">−</mo><mn>1</mn></mrow></msub></math>
is the vector connecting the second closest frame to the current position,
<!--l. 3187--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>v</mi></mstyle></mrow><mrow><mn>3</mn></mrow></msub> <mo class="MathClass-rel">=</mo><msub><mrow> <mstyle mathvariant="bold"><mi>s</mi></mstyle></mrow><mrow><mi>m</mi><mo class="MathClass-bin">+</mo><mn>1</mn></mrow></msub> <mo class="MathClass-bin">−</mo><msub><mrow><mstyle mathvariant="bold"><mi>s</mi></mstyle></mrow><mrow><mi>m</mi></mrow></msub></math>
is the vector connecting the closest frame to the third closest frame, and
<!--l. 3187--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>v</mi></mstyle></mrow><mrow><mn>4</mn></mrow></msub> <mo class="MathClass-rel">=</mo><msub><mrow> <mstyle mathvariant="bold"><mi>s</mi></mstyle></mrow><mrow><mi>m</mi></mrow></msub><mo class="MathClass-bin">−</mo><msub><mrow><mstyle mathvariant="bold"><mi>s</mi></mstyle></mrow><mrow><mi>m</mi><mo class="MathClass-bin">−</mo><mn>1</mn></mrow></msub></math>
is the vector connecting the second closest frame to the closest frame.
<!--l. 3187--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>m</mi></math> and
<!--l. 3187--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>M</mi></math>
are the current index of the closest frame and the total number of frames,
respectively. If the current position is on the left of the closest reference frame, the
<!--l. 3187--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-bin">±</mo></math> in
<!--l. 3187--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>s</mi></math> turns to
the positive sign. Otherwise it turns to the negative sign.
</p><!--l. 3189--><p class="indent">   The equations above assume: (i) the frames are equidistant and (ii) the second and the third closest
frames are neighbouring to the closest frame. When these assumptions are not satisfied, this set of path
CV should be used carefully.
</p>
<div class="flushright">
<!--l. 3191--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3191--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_gspath"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.11.1    </span> <a id="sec:cvc_gspath"></a><span class="ectt-1095">gspath</span>: progress along a path defined in atomic Cartesian coordinate space.</h5>
<!--l. 3191--><p class="noindent"></p></div> <a id="colvar|gspath"></a>
<!--l. 3194--><p class="indent">   In the <span class="ectt-1095">gspath</span><span class="ectt-1095"> {...} </span>and the <span class="ectt-1095">gzpath</span><span class="ectt-1095"> {...} </span>block all vectors, namely
<!--l. 3194--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="bold"><mi>z</mi></mstyle></math> and
<!--l. 3194--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>s</mi></mstyle></mrow><mrow><mi>k</mi></mrow></msub></math>
are defined in atomic Cartesian coordinate space. More specifically,
<!--l. 3194--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="bold"><mi>z</mi></mstyle> <mo class="MathClass-rel">=</mo> <mrow><mo fence="true" form="prefix"> [</mo><mrow><msub><mrow><mstyle mathvariant="bold"><mi>r</mi></mstyle></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-punc">,</mo><mo class="MathClass-rel">⋯</mo><mspace class="thinspace" width="0.17em"></mspace><mo class="MathClass-punc">,</mo><msub><mrow><mstyle mathvariant="bold"><mi>r</mi></mstyle></mrow><mrow><mi>n</mi></mrow></msub></mrow><mo fence="true" form="postfix">]</mo></mrow></math>, where
<!--l. 3194--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>r</mi></mstyle></mrow><mrow><mi>i</mi></mrow></msub></math> is the
<!--l. 3194--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math>-th atom specified
in the <span class="ectt-1095">atoms </span>block. <!--l. 3194--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>s</mi></mstyle></mrow><mrow><mi>k</mi></mrow></msub> <mo class="MathClass-rel">=</mo> <mrow><mo fence="true" form="prefix"> [</mo><mrow><msub><mrow><mstyle mathvariant="bold"><mi>r</mi></mstyle></mrow><mrow><mi>k</mi><mo class="MathClass-punc">,</mo><mn>1</mn></mrow></msub><mo class="MathClass-punc">,</mo><mo class="MathClass-rel">⋯</mo><mspace class="thinspace" width="0.17em"></mspace><mo class="MathClass-punc">,</mo><msub><mrow><mstyle mathvariant="bold"><mi>r</mi></mstyle></mrow><mrow><mi>k</mi><mo class="MathClass-punc">,</mo><mi>n</mi></mrow></msub></mrow><mo fence="true" form="postfix">]</mo></mrow></math>,
where <!--l. 3194--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>r</mi></mstyle></mrow><mrow><mi>k</mi><mo class="MathClass-punc">,</mo><mi>i</mi></mrow></msub></math> means
the <!--l. 3194--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math>-th atom
of the <!--l. 3194--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>k</mi></math>-th
reference frame.
</p><!--l. 3196--><p class="noindent"><span class="phvb8t-x-x-109">List of keywords </span>(see also <a href="#sec:cvc_pbcs">4.2<!--tex4ht:ref: sec:cvc_pbcs --></a>, <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>, <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> and <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox186"><a id="dx1-81001"></a><span class="ectt-1095">atoms</span></span> — <span class="phvro8t-x-x-109">Group of atoms</span><br class="newline"/><!--l. 3203--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> <a href="#sec:colvar_atom_groups">Atom
     group</a>, context: <span class="ectt-1095">gspath </span>and <span class="ectt-1095">gzpath</span> <!--l. 3203--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Defines the atoms whose coordinates make up the value of the component.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox187"><a id="dx1-81002"></a><span class="ectt-1095">refPositionsCol</span></span> — <span class="phvro8t-x-x-109">PDB column containing atom flags</span><br class="newline"/><!--l. 3211--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> <span class="ectt-1095">O</span>,
     <span class="ectt-1095">B</span>, <span class="ectt-1095">X</span>, <span class="ectt-1095">Y</span>, or <span class="ectt-1095">Z</span>, context: <span class="ectt-1095">gspath </span>and <span class="ectt-1095">gzpath</span> <!--l. 3211--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If <span class="ectt-1095">refPositionsFileN </span>is a PDB file that contains all the atoms in the topology, this option
     may be provided to set which PDB field is used to flag the reference coordinates for <span class="ectt-1095">atoms</span>.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox188"><a id="dx1-81003"></a><span class="ectt-1095">refPositionsFileN</span></span> — <span class="phvro8t-x-x-109">File containing the reference positions for fitting</span><br class="newline"/><!--l. 3220--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> UNIX
     filename, context: <span class="ectt-1095">gspath </span>and <span class="ectt-1095">gzpath</span> <!--l. 3220--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>The path is defined by multiple <span class="ectt-1095">refPositionsFile</span>s which are similiar to <span class="ectt-1095">refPositionsFile</span>
     in the <span class="ectt-1095">rmsd </span>CV. If your path consists of <!--l. 3220--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>1</mn><mn>0</mn></math>
     nodes, you can list the coordinate file (in PDB or XYZ format) from <span class="ectt-1095">refPositionsFile1 </span>to
     <span class="ectt-1095">refPositionsFile10</span>.
                                                                                         
                                                                                         
     </li>
<li class="itemize"><span class="colorbox" id="colorbox189"><a id="dx1-81004"></a><span class="ectt-1095">useSecondClosestFrame</span></span> — <span class="phvro8t-x-x-109">Define </span><!--l. 3230--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>s</mi></mstyle></mrow><mrow><mi>m</mi><mo class="MathClass-bin">−</mo><mn>1</mn></mrow></msub></math>
<span class="phvro8t-x-x-109">as the second closest frame?</span><br class="newline"/>Default: <span class="ectt-1095">on</span> <!--l. 3230--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">gspath </span>and <span class="ectt-1095">gzpath</span> <!--l. 3230--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>The definition assumes the second closest frame is neighbouring to the closest frame. This
     is not always true especially when the path is crooked. If this option is set to <span class="ectt-1095">on </span>(default),
     <!--l. 3230--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>s</mi></mstyle></mrow><mrow><mi>m</mi><mo class="MathClass-bin">−</mo><mn>1</mn></mrow></msub></math>
     is defined as the second closest frame. If this option is set to <span class="ectt-1095">off</span>, <!--l. 3230--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>s</mi></mstyle></mrow><mrow><mi>m</mi><mo class="MathClass-bin">−</mo><mn>1</mn></mrow></msub></math>
     is defined as the left or right neighbouring frame of the closest frame.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox190"><a id="dx1-81005"></a><span class="ectt-1095">useThirdClosestFrame</span></span> — <span class="phvro8t-x-x-109">Define </span><!--l. 3240--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>s</mi></mstyle></mrow><mrow><mi>m</mi><mo class="MathClass-bin">+</mo><mn>1</mn></mrow></msub></math>
<span class="phvro8t-x-x-109">as the third closest frame?</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 3240--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">gspath </span>and <span class="ectt-1095">gzpath</span> <!--l. 3240--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>The definition assumes the third closest frame is neighbouring to the closest frame. This is
     not always true especially when the path is crooked. If this option is set to <span class="ectt-1095">on</span>, <!--l. 3240--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>s</mi></mstyle></mrow><mrow><mi>m</mi><mo class="MathClass-bin">+</mo><mn>1</mn></mrow></msub></math>
     is defined as the third closest frame. If this option is set to <span class="ectt-1095">off </span>(default), <!--l. 3240--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>s</mi></mstyle></mrow><mrow><mi>m</mi><mo class="MathClass-bin">+</mo><mn>1</mn></mrow></msub></math>
     is defined as the left or right neighbouring frame of the closest frame.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox191"><a id="dx1-81006"></a><span class="ectt-1095">fittingAtoms</span></span> — <span class="phvro8t-x-x-109">The atoms that are used for alignment</span><br class="newline"/><!--l. 3249--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> Group
     of atoms, context: <span class="ectt-1095">gspath </span>and <span class="ectt-1095">gzpath</span> <!--l. 3249--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Before calculating <!--l. 3249--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>v</mi></mstyle></mrow><mrow><mn>1</mn></mrow></msub></math>,
     <!--l. 3249--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>v</mi></mstyle></mrow><mrow><mn>2</mn></mrow></msub></math>,
     <!--l. 3249--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>v</mi></mstyle></mrow><mrow><mn>3</mn></mrow></msub></math>
     and <!--l. 3249--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>v</mi></mstyle></mrow><mrow><mn>4</mn></mrow></msub></math>,
     the current frame need to be aligned to the corresponding reference frames. This option
     specifies which atoms are used to do alignment.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox192"><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances? </span>(see <a href="#colvar|cvc|forceNoPBC">description</a>)<br class="newline"/>Default: <span class="ectt-1095">no</span></li></ul>
<div class="flushright">
<!--l. 3253--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3253--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_gzpath"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.11.2    </span> <a id="sec:cvc_gzpath"></a><span class="ectt-1095">gzpath</span>: distance from a path defined in atomic Cartesian coordinate space.</h5>
<!--l. 3253--><p class="noindent"></p></div> <a id="colvar|gzpath"></a>
<!--l. 3256--><p class="noindent"><span class="phvb8t-x-x-109">List of keywords </span>(see also <a href="#sec:cvc_pbcs">4.2<!--tex4ht:ref: sec:cvc_pbcs --></a>, <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>, <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> and <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox193"><a id="dx1-82001"></a><span class="ectt-1095">useZsquare</span></span> — <span class="phvro8t-x-x-109">Compute </span><!--l. 3266--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow><mi>z</mi></mrow><mrow><mn>2</mn></mrow></msup></math>
<span class="phvro8t-x-x-109">instead of </span><!--l. 3266--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 3266--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">gzpath</span> <!--l. 3266--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/><!--l. 3266--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math>
     is not differentiable when it is zero. This implementation workarounds it by setting the derivative
     of <!--l. 3266--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math>
     to zero when <!--l. 3266--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi> <mo class="MathClass-rel">=</mo> <mn>0</mn></math>.
     Another workaround is set this option to <span class="ectt-1095">on</span>, which computes <!--l. 3266--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow><mi>z</mi></mrow><mrow><mn>2</mn></mrow></msup></math>
     instead of <!--l. 3266--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math>,
     and then <!--l. 3266--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow><mi>z</mi></mrow><mrow><mn>2</mn></mrow></msup></math>
     is differentiable when it is zero.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox194"><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances? </span>(see <a href="#colvar|cvc|forceNoPBC">description</a>)<br class="newline"/>Default: <span class="ectt-1095">no</span></li></ul>
<!--l. 3270--><p class="indent">   The usage of <span class="ectt-1095">gzpath </span>and <span class="ectt-1095">gspath </span>is illustrated below:</p>
<div class="mdframed" id="mdframed-45">
<!--l. 3272--><p class="noindent"></p><pre>colvar {<br class="newline"/>  # Progress along the path<br class="newline"/>  name gs<br class="newline"/>  # The path is defined by 5 reference frames (from string-00.pdb to string-04.pdb)<br class="newline"/>  # Use atomic coordinate from atoms 1, 2 and 3 to compute the path<br class="newline"/>  gspath {<br class="newline"/>    atoms {atomnumbers { 1 2 3 }}<br class="newline"/>    refPositionsFile1 string-00.pdb<br class="newline"/>    refPositionsFile2 string-01.pdb<br class="newline"/>    refPositionsFile3 string-02.pdb<br class="newline"/>    refPositionsFile4 string-03.pdb<br class="newline"/>    refPositionsFile5 string-04.pdb<br class="newline"/>  }<br class="newline"/>}<br class="newline"/>colvar {<br class="newline"/>  # Distance from the path<br class="newline"/>  name gz<br class="newline"/>  # The path is defined by 5 reference frames (from string-00.pdb to string-04.pdb)<br class="newline"/>  # Use atomic coordinate from atoms 1, 2 and 3 to compute the path<br class="newline"/>  gzpath {<br class="newline"/>    atoms {atomnumbers { 1 2 3 }}<br class="newline"/>    refPositionsFile1 string-00.pdb<br class="newline"/>    refPositionsFile2 string-01.pdb<br class="newline"/>    refPositionsFile3 string-02.pdb<br class="newline"/>    refPositionsFile4 string-03.pdb<br class="newline"/>    refPositionsFile5 string-04.pdb<br class="newline"/>  }<br class="newline"/>} </pre><!--l. 3301--><p class="noindent"></p>
</div>
<div class="flushright">
<!--l. 3304--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3304--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_linearCombination"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.11.3    </span> <a id="sec:cvc_linearCombination"></a><span class="ectt-1095">linearCombination</span>: Helper CV to define a linear combination of other CVs</h5>
<!--l. 3304--><p class="noindent"></p></div> <a id="colvar|linearCombination"></a>
<!--l. 3307--><p class="indent">   This is a helper CV which can be defined as a linear combination of other CVs. It maybe useful when
you want to define the <span class="ectt-1095">gspathCV</span><span class="ectt-1095"> {...} </span>and the <span class="ectt-1095">gzpathCV</span><span class="ectt-1095"> {...} </span>as combinations of other CVs. Total
forces (required by <a href="#sec:colvarbias_abf_req"><span class="ectt-1095">ABF</span></a>) of this CV are not available.
</p>
<div class="flushright">
<!--l. 3311--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3311--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_customColvar"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.11.4    </span> <a id="sec:cvc_customColvar"></a><span class="ectt-1095">customColvar</span>: Helper CV to define a mathematical expression as CV from other
CVs</h5>
<!--l. 3311--><p class="noindent"></p></div> <a id="colvar|customColvar"></a>
<!--l. 3315--><p class="indent">   This is a helper CV which can be defined as a mathematical expression (see <a href="#sec:colvar_custom_function">4.20<!--tex4ht:ref: sec:colvar_custom_function --></a>) of other CVs by
using <a href="#colvar|customFunction"><span class="ectt-1095">customFunction</span></a>. Currently only the scalar type of <a href="#colvar|customFunction"><span class="ectt-1095">customFunction</span></a> is supported. If <a href="#colvar|customFunction"><span class="ectt-1095">customFunction</span></a>
is not provided, this component falls back to <a href="#colvar|linearCombination"><span class="ectt-1095">linearCombination</span></a>. It maybe useful when you want to define
the <span class="ectt-1095">gspathCV</span><span class="ectt-1095"> {...}</span>, the <span class="ectt-1095">gzpathCV</span><span class="ectt-1095"> {...} </span>and <span class="ectt-1095">NeuralNetwork</span><span class="ectt-1095"> {...} </span>as combinations of other CVs. Total
forces (required by <a href="#sec:colvarbias_abf_req"><span class="ectt-1095">ABF</span></a>) of this CV are not available.
</p>
<div class="flushright">
<!--l. 3317--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3317--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_gspathCV"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.11.5    </span> <a id="sec:cvc_gspathCV"></a><span class="ectt-1095">gspathCV</span>: progress along a path defined in CV space.</h5>
<!--l. 3317--><p class="noindent"></p></div> <a id="colvar|gspathCV"></a>
<!--l. 3320--><p class="indent">   In the <span class="ectt-1095">gspathCV</span><span class="ectt-1095"> {...} </span>and the <span class="ectt-1095">gzpathCV</span><span class="ectt-1095"> {...} </span>block all vectors, namely
<!--l. 3320--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="bold"><mi>z</mi></mstyle></math> and
<!--l. 3320--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>s</mi></mstyle></mrow><mrow><mi>k</mi></mrow></msub></math> are defined in CV space.
More specifically, <!--l. 3320--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="bold"><mi>z</mi></mstyle> <mo class="MathClass-rel">=</mo> <mrow><mo fence="true" form="prefix"> [</mo><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-punc">,</mo><mo class="MathClass-rel">⋯</mo><mspace class="thinspace" width="0.17em"></mspace><mo class="MathClass-punc">,</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>n</mi></mrow></msub></mrow><mo fence="true" form="postfix">]</mo></mrow></math>,
where <!--l. 3320--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></math> is
the <!--l. 3320--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math>-th CV.
<!--l. 3320--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>s</mi></mstyle></mrow><mrow><mi>k</mi></mrow></msub> <mo class="MathClass-rel">=</mo> <mrow><mo fence="true" form="prefix"> [</mo><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>k</mi><mo class="MathClass-punc">,</mo><mn>1</mn></mrow></msub><mo class="MathClass-punc">,</mo><mo class="MathClass-rel">⋯</mo><mspace class="thinspace" width="0.17em"></mspace><mo class="MathClass-punc">,</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>k</mi><mo class="MathClass-punc">,</mo><mi>n</mi></mrow></msub></mrow><mo fence="true" form="postfix">]</mo></mrow></math>, where
<!--l. 3320--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>k</mi><mo class="MathClass-punc">,</mo><mi>i</mi></mrow></msub></math> means the
<!--l. 3320--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math>-th CV of
the <!--l. 3320--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>k</mi></math>-th
reference frame. It should be note that these two CVs requires the <span class="ectt-1095">pathFile </span>option, which specifies a
path file. Each line in the path file contains a set of space-seperated CV value of the reference frame. The
sequence of reference frames matches the sequence of the lines.
</p><!--l. 3323--><p class="noindent"><span class="phvb8t-x-x-109">List of keywords </span>(see also <a href="#sec:cvc_pbcs">4.2<!--tex4ht:ref: sec:cvc_pbcs --></a>, <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>, <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> and <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox195"><a id="dx1-85001"></a><span class="ectt-1095">useSecondClosestFrame</span></span> — <span class="phvro8t-x-x-109">Define </span><!--l. 3332--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>s</mi></mstyle></mrow><mrow><mi>m</mi><mo class="MathClass-bin">−</mo><mn>1</mn></mrow></msub></math>
<span class="phvro8t-x-x-109">as the second closest frame?</span><br class="newline"/>Default: <span class="ectt-1095">on</span> <!--l. 3332--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">gspathCV </span>and <span class="ectt-1095">gzpathCV</span> <!--l. 3332--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>The definition assumes the second closest frame is neighbouring to the closest frame. This
     is not always true especially when the path is crooked. If this option is set to <span class="ectt-1095">on </span>(default),
     <!--l. 3332--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>s</mi></mstyle></mrow><mrow><mi>m</mi><mo class="MathClass-bin">−</mo><mn>1</mn></mrow></msub></math>
     is defined as the second closest frame. If this option is set to <span class="ectt-1095">off</span>, <!--l. 3332--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>s</mi></mstyle></mrow><mrow><mi>m</mi><mo class="MathClass-bin">−</mo><mn>1</mn></mrow></msub></math>
     is defined as the left or right neighbouring frame of the closest frame.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox196"><a id="dx1-85002"></a><span class="ectt-1095">useThirdClosestFrame</span></span> — <span class="phvro8t-x-x-109">Define </span><!--l. 3342--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>s</mi></mstyle></mrow><mrow><mi>m</mi><mo class="MathClass-bin">+</mo><mn>1</mn></mrow></msub></math>
<span class="phvro8t-x-x-109">as the third closest frame?</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 3342--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">gspathCV </span>and <span class="ectt-1095">gzpathCV</span> <!--l. 3342--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>The definition assumes the third closest frame is neighbouring to the closest frame. This is
     not always true especially when the path is crooked. If this option is set to <span class="ectt-1095">on</span>, <!--l. 3342--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>s</mi></mstyle></mrow><mrow><mi>m</mi><mo class="MathClass-bin">+</mo><mn>1</mn></mrow></msub></math>
     is defined as the third closest frame. If this option is set to <span class="ectt-1095">off </span>(default), <!--l. 3342--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>s</mi></mstyle></mrow><mrow><mi>m</mi><mo class="MathClass-bin">+</mo><mn>1</mn></mrow></msub></math>
     is defined as the left or right neighbouring frame of the closest frame.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox197"><a id="dx1-85003"></a><span class="ectt-1095">pathFile</span></span> — <span class="phvro8t-x-x-109">The file name of the path file.</span><br class="newline"/><!--l. 3351--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> UNIX
     filename, context: <span class="ectt-1095">gspathCV </span>and <span class="ectt-1095">gzpathCV</span> <!--l. 3351--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Defines the nodes or images that constitutes the path in CV space. The CVs of an image
     are listed in a line of <span class="ectt-1095">pathFile </span>using space-seperated format. Lines from top to button in
     <span class="ectt-1095">pathFile </span>corresponds images from initial to last.
                                                                                         
                                                                                         
     </li>
<li class="itemize"><span class="colorbox" id="colorbox198"><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances? </span>(see <a href="#colvar|cvc|forceNoPBC">description</a>)<br class="newline"/>Default: <span class="ectt-1095">no</span></li></ul>
<div class="flushright">
<!--l. 3355--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3355--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_gzpathCV"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.11.6    </span> <a id="sec:cvc_gzpathCV"></a><span class="ectt-1095">gzpathCV</span>: distance from a path defined in CV space.</h5>
<!--l. 3355--><p class="noindent"></p></div> <a id="colvar|gzpathCV"></a>
<!--l. 3358--><p class="noindent"><span class="phvb8t-x-x-109">List of keywords </span>(see also <a href="#sec:cvc_pbcs">4.2<!--tex4ht:ref: sec:cvc_pbcs --></a>, <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>, <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> and <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox199"><a id="dx1-86001"></a><span class="ectt-1095">useZsquare</span></span> — <span class="phvro8t-x-x-109">Compute </span><!--l. 3368--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow><mi>z</mi></mrow><mrow><mn>2</mn></mrow></msup></math>
<span class="phvro8t-x-x-109">instead of </span><!--l. 3368--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 3368--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">gzpathCV</span> <!--l. 3368--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/><!--l. 3368--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math>
     is not differentiable when it is zero. This implementation workarounds it by setting the derivative
     of <!--l. 3368--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math>
     to zero when <!--l. 3368--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi> <mo class="MathClass-rel">=</mo> <mn>0</mn></math>.
     Another workaround is set this option to <span class="ectt-1095">on</span>, which computes <!--l. 3368--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow><mi>z</mi></mrow><mrow><mn>2</mn></mrow></msup></math>
     instead of <!--l. 3368--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math>,
     and then <!--l. 3368--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow><mi>z</mi></mrow><mrow><mn>2</mn></mrow></msup></math>
     is differentiable when it is zero.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox200"><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances? </span>(see <a href="#colvar|cvc|forceNoPBC">description</a>)<br class="newline"/>Default: <span class="ectt-1095">no</span></li></ul>
<!--l. 3372--><p class="indent">   The usage of <span class="ectt-1095">gzpathCV </span>and <span class="ectt-1095">gspathCV </span>is illustrated below:</p>
<div class="mdframed" id="mdframed-46">
<!--l. 3374--><p class="noindent"></p><pre>colvar {<br class="newline"/>  # Progress along the path<br class="newline"/>  name gs<br class="newline"/>  # Path defined by the CV space of two dihedral angles<br class="newline"/>  gspathCV {<br class="newline"/>    pathFile ./path.txt<br class="newline"/>    dihedral {<br class="newline"/>      name 001<br class="newline"/>      group1 {atomNumbers {5}}<br class="newline"/>      group2 {atomNumbers {7}}<br class="newline"/>      group3 {atomNumbers {9}}<br class="newline"/>      group4 {atomNumbers {15}}<br class="newline"/>    }<br class="newline"/>    dihedral {<br class="newline"/>      name 002<br class="newline"/>      group1 {atomNumbers {7}}<br class="newline"/>      group2 {atomNumbers {9}}<br class="newline"/>      group3 {atomNumbers {15}}<br class="newline"/>      group4 {atomNumbers {17}}<br class="newline"/>    }<br class="newline"/>  }<br class="newline"/>}<br class="newline"/><br class="newline"/>colvar {<br class="newline"/>  # Distance from the path<br class="newline"/>  name gz<br class="newline"/>  gzpathCV {<br class="newline"/>    pathFile ./path.txt<br class="newline"/>    dihedral {<br class="newline"/>      name 001<br class="newline"/>      group1 {atomNumbers {5}}<br class="newline"/>      group2 {atomNumbers {7}}<br class="newline"/>      group3 {atomNumbers {9}}<br class="newline"/>      group4 {atomNumbers {15}}<br class="newline"/>    }<br class="newline"/>    dihedral {<br class="newline"/>      name 002<br class="newline"/>      group1 {atomNumbers {7}}<br class="newline"/>      group2 {atomNumbers {9}}<br class="newline"/>      group3 {atomNumbers {15}}<br class="newline"/>      group4 {atomNumbers {17}}<br class="newline"/>    }<br class="newline"/>  }<br class="newline"/>} </pre><!--l. 3419--><p class="noindent"></p>
</div>
<div class="flushright">
<!--l. 3422--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3422--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_apathCV"></a>
<h4 class="subsectionHead"><span class="titlemark">4.12    </span> <a id="sec:cvc_apathCV"></a>Arithmetic path collective variables</h4>
<!--l. 3422--><p class="noindent"></p></div>
<!--l. 3424--><p class="indent">   The arithmetic path collective variable in CV space uses a similar formula
as the one proposed by Branduardi<span class="cite">[<a href="#XBranduardi2007">12</a>]</span> et al., except that it computes
<!--l. 3426--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>s</mi></math> and
<!--l. 3426--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math> in CV
                                                                                         
                                                                                         
space instead of RMSDs in Cartesian space. Moreover, this implementation allows different
coefficients for each CV components as described in <span class="cite">[<a href="#XHovan2019">13</a>]</span>. Assuming a path is composed of
<!--l. 3428--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>N</mi></math> reference frames and
defined in an <!--l. 3429--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>M</mi></math>-dimensional CV
space, then the equations of <!--l. 3429--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>s</mi></math>
and <!--l. 3430--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math> of
the path are
</p>
<table class="equation"><tr><td> <a id="eq:apath_s"></a>
<!--l. 3432--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>s</mi> <mo class="MathClass-rel">=</mo> <mfrac><mrow><mn>1</mn></mrow>
<mrow><mi>N</mi> <mo class="MathClass-bin">−</mo><mn>1</mn></mrow></mfrac> <mfrac><mrow><munderover accent="false" accentunder="false"><mrow><mo>∑</mo></mrow><mrow><mi>i</mi><mo class="MathClass-rel">=</mo><mn>0</mn></mrow><mrow><mi>N</mi><mo class="MathClass-bin">−</mo><mn>1</mn></mrow></munderover><mi>i</mi><mi class="qopname">exp</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mo class="MathClass-bin">−</mo><mi>λ</mi><munderover accent="false" accentunder="false"><mrow><mo>∑</mo></mrow><mrow><mi>j</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><mi>M</mi></mrow></munderover><msubsup><mrow><mi>c</mi></mrow><mrow><mi>j</mi></mrow><mrow><mn>2</mn></mrow></msubsup><msup><mrow> <mrow><mo fence="true" form="prefix"> (</mo><mrow><msub><mrow><mi>x</mi></mrow><mrow><mi>j</mi></mrow></msub><mo class="MathClass-bin">−</mo><msub><mrow><mi>x</mi></mrow><mrow><mi>i</mi><mo class="MathClass-punc">,</mo><mi>j</mi></mrow></msub></mrow><mo fence="true" form="postfix">)</mo></mrow></mrow><mrow><mn>2</mn></mrow></msup></mrow><mo fence="true" form="postfix">)</mo></mrow></mrow>
<mrow><munderover accent="false" accentunder="false"><mrow><mo>∑</mo></mrow><mrow><mi>i</mi><mo class="MathClass-rel">=</mo><mn>0</mn></mrow><mrow><mi>N</mi><mo class="MathClass-bin">−</mo><mn>1</mn></mrow></munderover><mi class="qopname">exp</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mo class="MathClass-bin">−</mo><mi>λ</mi><munderover accent="false" accentunder="false"><mrow><mo>∑</mo></mrow><mrow><mi>j</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><mi>M</mi></mrow></munderover><msubsup><mrow><mi>c</mi></mrow><mrow><mi>j</mi></mrow><mrow><mn>2</mn></mrow></msubsup><msup><mrow> <mrow><mo fence="true" form="prefix"> (</mo><mrow><msub><mrow><mi>x</mi></mrow><mrow><mi>j</mi></mrow></msub><mo class="MathClass-bin">−</mo><msub><mrow><mi>x</mi></mrow><mrow><mi>i</mi><mo class="MathClass-punc">,</mo><mi>j</mi></mrow></msub></mrow><mo fence="true" form="postfix">)</mo></mrow></mrow><mrow><mn>2</mn></mrow></msup></mrow><mo fence="true" form="postfix">)</mo></mrow></mrow></mfrac>
</math></td><td class="eq-no">(16)</td></tr></table>
<table class="equation"><tr><td> <a id="eq:apath_z"></a>
<!--l. 3439--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>z</mi> <mo class="MathClass-rel">=</mo> <mo class="MathClass-bin">−</mo><mfrac><mrow><mn>1</mn></mrow>
<mrow><mi>λ</mi></mrow></mfrac><mi class="qopname">ln</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--> <mrow><mo fence="true" form="prefix"> (</mo><mrow><munderover accent="false" accentunder="false"><mrow><mo>∑</mo></mrow><mrow><mi>i</mi><mo class="MathClass-rel">=</mo><mn>0</mn></mrow><mrow><mi>N</mi><mo class="MathClass-bin">−</mo><mn>1</mn></mrow></munderover><mi class="qopname">exp</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mo class="MathClass-bin">−</mo><mi>λ</mi><munderover accent="false" accentunder="false"><mrow><mo>∑</mo></mrow><mrow>
<mi>j</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><mi>M</mi></mrow></munderover><msubsup><mrow><mi>c</mi></mrow><mrow>
<mi>j</mi></mrow><mrow><mn>2</mn></mrow></msubsup><msup><mrow> <mrow><mo fence="true" form="prefix"> (</mo><mrow><msub><mrow><mi>x</mi></mrow><mrow>
<mi>j</mi></mrow></msub><mo class="MathClass-bin">−</mo><msub><mrow><mi>x</mi></mrow><mrow><mi>i</mi><mo class="MathClass-punc">,</mo><mi>j</mi></mrow></msub></mrow><mo fence="true" form="postfix">)</mo></mrow></mrow><mrow><mn>2</mn></mrow></msup></mrow><mo fence="true" form="postfix">)</mo></mrow></mrow><mo fence="true" form="postfix">)</mo></mrow>
</math></td><td class="eq-no">(17)</td></tr></table>
<!--l. 3444--><p class="indent">   where <!--l. 3444--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>c</mi></mrow><mrow><mi>j</mi></mrow></msub></math> is the
coefficient(weight) of the <!--l. 3444--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>j</mi></math>-th
CV, <!--l. 3444--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>x</mi></mrow><mrow><mi>i</mi><mo class="MathClass-punc">,</mo><mi>j</mi></mrow></msub></math> is the value
of <!--l. 3445--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>j</mi></math>-th CV of
<!--l. 3445--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math>-th reference frame
and <!--l. 3445--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>x</mi></mrow><mrow><mi>j</mi></mrow></msub></math> is the value of
<!--l. 3445--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>j</mi></math>-th CV of current frame.
<!--l. 3446--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>λ</mi></math> is a parameter to
smooth the variation of <!--l. 3446--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>s</mi></math>
and <!--l. 3446--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math>. It should be
noted that the index <!--l. 3447--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math>
ranges from <!--l. 3447--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>0</mn></math> to
<!--l. 3447--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>N</mi> <mo class="MathClass-bin">−</mo><mn>1</mn></math>, and the definition
of <!--l. 3448--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>s</mi></math> is normalized
by <!--l. 3448--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>1</mn><mo class="MathClass-bin">∕</mo><mo class="MathClass-open">(</mo><mi>N</mi> <mo class="MathClass-bin">−</mo><mn>1</mn><mo class="MathClass-close">)</mo></math>. Consequently,
the scope of <!--l. 3448--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>s</mi></math>
                                                                                         
                                                                                         
is <!--l. 3449--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo><mn>0</mn> <mo class="MathClass-punc">:</mo> <mn>1</mn><mo class="MathClass-close">]</mo></math>.
</p>
<div class="flushright">
<!--l. 3451--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3451--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_aspathCV"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.12.1    </span> <a id="sec:cvc_aspathCV"></a><span class="ectt-1095">aspathCV</span>: progress along a path defined in CV space.</h5>
<!--l. 3451--><p class="noindent"></p></div>
<!--l. 3453--><p class="indent">   This colvar component computes the <!--l. 3453--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>s</mi></math>
variable.
</p><!--l. 3455--><p class="indent"> <a id="colvar|aspathCV"></a>
</p><!--l. 3457--><p class="noindent"><span class="phvb8t-x-x-109">List of keywords </span>(see also <a href="#sec:cvc_pbcs">4.2<!--tex4ht:ref: sec:cvc_pbcs --></a>, <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>, <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> and <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox201"><a id="dx1-88001"></a><span class="ectt-1095">weights</span></span> — <span class="phvro8t-x-x-109">Coefficients of the collective variables</span><br class="newline"/>Default: <span class="ectt-1095">{1.0 ...}</span> <!--l. 3466--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> space-separated
     numbers in a <span class="ectt-1095">{...} </span>block, context: <span class="ectt-1095">aspathCV </span>and <span class="ectt-1095">azpathCV</span> <!--l. 3466--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Define the coefficients. The <!--l. 3466--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>j</mi></math>-th
     value in the <span class="ectt-1095">{...} </span>block corresponds the <!--l. 3466--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>c</mi></mrow><mrow><mi>j</mi></mrow></msub></math>
     in the equations.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox202"><a id="dx1-88002"></a><span class="ectt-1095">lambda</span></span> — <span class="phvro8t-x-x-109">Smoothness of the variation of </span><!--l. 3476--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>s</mi></math>
<span class="phvro8t-x-x-109">and </span><!--l. 3476--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math><br class="newline"/>Default: inverse of the mean square displacements of successive reference frames <!--l. 3476--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> decimal,
     context: <span class="ectt-1095">aspathCV </span>and <span class="ectt-1095">azpathCV</span> <!--l. 3476--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>The value of <!--l. 3476--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>λ</mi></math>
     in the equations.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox203"><a id="dx1-88003"></a><span class="ectt-1095">pathFile</span></span> — <span class="phvro8t-x-x-109">The file name of the path file.</span><br class="newline"/><!--l. 3485--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> UNIX
     filename, context: <span class="ectt-1095">aspathCV </span>and <span class="ectt-1095">azpathCV</span> <!--l. 3485--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Defines the nodes or images that constitutes the path in CV space. The CVs of an image
     are listed in a line of <span class="ectt-1095">pathFile </span>using space-separated format. Lines from top to button in
     <span class="ectt-1095">pathFile </span>corresponds images from initial to last.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox204"><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances? </span>(see <a href="#colvar|cvc|forceNoPBC">description</a>)<br class="newline"/>Default: <span class="ectt-1095">no</span></li></ul>
<div class="flushright">
<!--l. 3489--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3489--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_azpathCV"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.12.2    </span> <a id="sec:cvc_azpathCV"></a><span class="ectt-1095">azpathCV</span>: distance from a path defined in CV space.</h5>
<!--l. 3489--><p class="noindent"></p></div>
<!--l. 3491--><p class="indent">   This colvar component computes the <!--l. 3491--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math>
variable. Options are the same as in <a href="#sec:cvc_aspathCV">4.12.1<!--tex4ht:ref: sec:cvc_aspathCV --></a>.
</p><!--l. 3493--><p class="indent">   The usage of <span class="ectt-1095">azpathCV </span>and <span class="ectt-1095">aspathCV </span>is illustrated below:</p>
<div class="mdframed" id="mdframed-47">
<!--l. 3495--><p class="noindent"></p><pre>colvar {<br class="newline"/>  # Progress along the path<br class="newline"/>  name as<br class="newline"/>  # Path defined by the CV space of two dihedral angles<br class="newline"/>  aspathCV {<br class="newline"/>    pathFile ./path.txt<br class="newline"/>    weights {1.0 1.0}<br class="newline"/>    lambda 0.005<br class="newline"/>    dihedral {<br class="newline"/>      name 001<br class="newline"/>      group1 {atomNumbers {5}}<br class="newline"/>      group2 {atomNumbers {7}}<br class="newline"/>      group3 {atomNumbers {9}}<br class="newline"/>      group4 {atomNumbers {15}}<br class="newline"/>    }<br class="newline"/>    dihedral {<br class="newline"/>      name 002<br class="newline"/>      group1 {atomNumbers {7}}<br class="newline"/>      group2 {atomNumbers {9}}<br class="newline"/>      group3 {atomNumbers {15}}<br class="newline"/>      group4 {atomNumbers {17}}<br class="newline"/>    }<br class="newline"/>  }<br class="newline"/>}<br class="newline"/><br class="newline"/>colvar {<br class="newline"/>  # Distance from the path<br class="newline"/>  name az<br class="newline"/>  azpathCV {<br class="newline"/>    pathFile ./path.txt<br class="newline"/>    weights {1.0 1.0}<br class="newline"/>    lambda 0.005<br class="newline"/>    dihedral {<br class="newline"/>      name 001<br class="newline"/>      group1 {atomNumbers {5}}<br class="newline"/>      group2 {atomNumbers {7}}<br class="newline"/>      group3 {atomNumbers {9}}<br class="newline"/>      group4 {atomNumbers {15}}<br class="newline"/>    }<br class="newline"/>    dihedral {<br class="newline"/>      name 002<br class="newline"/>      group1 {atomNumbers {7}}<br class="newline"/>      group2 {atomNumbers {9}}<br class="newline"/>      group3 {atomNumbers {15}}<br class="newline"/>      group4 {atomNumbers {17}}<br class="newline"/>    }<br class="newline"/>  }<br class="newline"/>} </pre><!--l. 3544--><p class="noindent"></p>
</div>
<div class="flushright">
<!--l. 3548--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3548--><p class="noindent"> </p><div class="heading"> <a id="sec:apath_scripted"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.12.3    </span> <a id="sec:apath_scripted"></a>Path collective variables in Cartesian coordinates</h5>
<!--l. 3548--><p class="noindent"></p></div>
<!--l. 3550--><p class="indent">   The path collective variables defined by Branduardi et al. <span class="cite">[<a href="#XBranduardi2007">12</a>]</span> are based on RMSDs in Cartesian coordinates.
Noting <!--l. 3552--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>d</mi></mrow><mrow><mi>i</mi></mrow></msub></math>
the RMSD between the current set of Cartesian coordinates and those of image number
<!--l. 3552--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math> of the
path:
</p>
<table class="equation"><tr><td> <a id="x1-90001r18"></a>
<!--l. 3554--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>s</mi> <mo class="MathClass-rel">=</mo> <mfrac><mrow><mn>1</mn></mrow>
<mrow><mi>N</mi> <mo class="MathClass-bin">−</mo><mn>1</mn></mrow></mfrac> <mfrac><mrow><munderover accent="false" accentunder="false"><mrow><mo>∑</mo></mrow><mrow><mi>i</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><mi>N</mi></mrow></munderover><mo class="MathClass-open">(</mo><mi>i</mi><mo class="MathClass-bin">−</mo><mn>1</mn><mo class="MathClass-close">)</mo><mi class="qopname">exp</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mo class="MathClass-bin">−</mo><mi>λ</mi><msubsup><mrow><mi>d</mi></mrow><mrow><mi>i</mi></mrow><mrow><mn>2</mn></mrow></msubsup></mrow><mo fence="true" form="postfix">)</mo></mrow></mrow>
<mrow><munderover accent="false" accentunder="false"><mrow><mo>∑</mo></mrow><mrow><mi>i</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><mi>N</mi></mrow></munderover><mi class="qopname">exp</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mo class="MathClass-bin">−</mo><mi>λ</mi><msubsup><mrow><mi>d</mi></mrow><mrow><mi>i</mi></mrow><mrow><mn>2</mn></mrow></msubsup></mrow><mo fence="true" form="postfix">)</mo></mrow></mrow></mfrac>
</math></td><td class="eq-no">(18)</td></tr></table>
<table class="equation"><tr><td> <a id="x1-90002r19"></a>
<!--l. 3559--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>z</mi> <mo class="MathClass-rel">=</mo> <mo class="MathClass-bin">−</mo><mfrac><mrow><mn>1</mn></mrow>
<mrow><mi>λ</mi></mrow></mfrac><mi class="qopname">ln</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--> <mrow><mo fence="true" form="prefix"> (</mo><mrow><munderover accent="false" accentunder="false"><mrow><mo>∑</mo></mrow><mrow><mi>i</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><mi>N</mi></mrow></munderover><mi class="qopname">exp</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--><mo class="MathClass-open">(</mo><mo class="MathClass-bin">−</mo><mi>λ</mi><msubsup><mrow><mi>d</mi></mrow><mrow>
<mi>i</mi></mrow><mrow><mn>2</mn></mrow></msubsup><mo class="MathClass-close">)</mo></mrow><mo fence="true" form="postfix">)</mo></mrow>
</math></td><td class="eq-no">(19)</td></tr></table>
<!--l. 3563--><p class="indent">   where <!--l. 3563--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>λ</mi></math>
is the smoothing parameter.
</p><!--l. 3566--><p class="indent">   These coordinates are implemented as Tcl-scripted combinations of <span class="ectt-1095">rmsd </span>components. The
implementation is available as file <span class="ectt-1095">colvartools/pathCV.tcl</span>, and an example is provided in file
<span class="ectt-1095">examples/10_pathCV.namd </span>of the Colvars public repository. It implements an optimization procedure,
whereby the distance to a given image is only calculated if its contribution to the sum is larger than a
user-defined <span class="ectt-1095">tolerance </span>parameter. All distances are calculated every <span class="ectt-1095">freq </span>timesteps to update the list of
nearby images.
</p>
<div class="flushright">
<!--l. 3574--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3574--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_aspath"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.12.4    </span> <a id="sec:cvc_aspath"></a><span class="ectt-1095">aspath</span>: progress along a path defined in atomic Cartesian coordinate space.</h5>
<!--l. 3574--><p class="noindent"></p></div> <a id="colvar|aspath"></a>
<!--l. 3577--><p class="indent">   This CV computes a special case of Eq. <a href="#eq:apath_s">16<!--tex4ht:ref: eq:apath_s --></a>, where
<!--l. 3577--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>x</mi></mrow><mrow><mi>j</mi></mrow></msub></math> is the
<!--l. 3578--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>j</mi></math>-th atomic
position, <!--l. 3578--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>x</mi></mrow><mrow><mi>i</mi><mo class="MathClass-punc">,</mo><mi>j</mi></mrow></msub></math> is the
<!--l. 3578--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>j</mi></math>-th atomic position of
the <!--l. 3578--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math>-th reference frame.
The subtraction <!--l. 3579--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>x</mi></mrow><mrow><mi>j</mi></mrow></msub><mo class="MathClass-bin">−</mo><msub><mrow><mi>x</mi></mrow><mrow><mi>i</mi><mo class="MathClass-punc">,</mo><mi>j</mi></mrow></msub></math> is
actually calculated as <!--l. 3580--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>j</mi></mrow></msub><mo class="MathClass-bin">−</mo><msub><mrow><mi>R</mi></mrow><mrow><mi>i</mi></mrow></msub><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>i</mi><mo class="MathClass-punc">,</mo><mi>j</mi></mrow></msub></math>,
where <!--l. 3580--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="normal"><mi>R</mi></mstyle></mrow><mrow><mi>i</mi></mrow></msub></math> is a
3x3 rotation matrix that minimizes the RMSD between the current atomic positions of simulation and the
<!--l. 3582--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math>-th reference
frame. Bold <!--l. 3582--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>j</mi></mrow></msub></math>
is used since an atomic position is a vector.
</p><!--l. 3586--><p class="indent">   For NAMD and VMD users, this component can be regarded as an improved C++ implementation of
the PCV progress component of the Tcl-scripted version (see <a href="#sec:apath_scripted">4.12.3<!--tex4ht:ref: sec:apath_scripted --></a>).
</p><!--l. 3591--><p class="noindent"><span class="phvb8t-x-x-109">List of keywords </span>(see also <a href="#sec:cvc_pbcs">4.2<!--tex4ht:ref: sec:cvc_pbcs --></a>, <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>, <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> and <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox205"><a id="dx1-91001"></a><span class="ectt-1095">atoms</span></span> — <span class="phvro8t-x-x-109">Group of atoms</span><br class="newline"/><!--l. 3598--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> <a href="#sec:colvar_atom_groups">Atom
     group</a>, context: <span class="ectt-1095">aspath </span>and <span class="ectt-1095">azpath</span> <!--l. 3598--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Defines the atoms whose coordinates make up the value of the component.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox206"><a id="dx1-91002"></a><span class="ectt-1095">refPositionsCol</span></span> — <span class="phvro8t-x-x-109">PDB column containing atom flags</span><br class="newline"/><!--l. 3606--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> <span class="ectt-1095">O</span>,
     <span class="ectt-1095">B</span>, <span class="ectt-1095">X</span>, <span class="ectt-1095">Y</span>, or <span class="ectt-1095">Z</span>, context: <span class="ectt-1095">aspath </span>and <span class="ectt-1095">azpath</span> <!--l. 3606--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If <span class="ectt-1095">refPositionsFileN </span>is a PDB file that contains all the atoms in the topology, this option
     may be provided to set which PDB field is used to flag the reference coordinates for <span class="ectt-1095">atoms</span>.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox207"><a id="dx1-91003"></a><span class="ectt-1095">refPositionsFileN</span></span> — <span class="phvro8t-x-x-109">File containing the reference positions for fitting</span><br class="newline"/><!--l. 3615--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> UNIX
     filename, context: <span class="ectt-1095">aspath </span>and <span class="ectt-1095">azpath</span> <!--l. 3615--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>The path is defined by multiple <span class="ectt-1095">refPositionsFile</span>s which are similiar to <span class="ectt-1095">refPositionsFile</span>
     in the <span class="ectt-1095">rmsd </span>CV. If your path consists of <!--l. 3615--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>1</mn><mn>0</mn></math>
     nodes, you can list the coordinate file (in PDB or XYZ format) from <span class="ectt-1095">refPositionsFile1 </span>to
     <span class="ectt-1095">refPositionsFile10</span>.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox208"><a id="dx1-91004"></a><span class="ectt-1095">fittingAtoms</span></span> — <span class="phvro8t-x-x-109">The atoms that are used for alignment</span><br class="newline"/><!--l. 3624--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> Group
     of atoms, context: <span class="ectt-1095">aspath </span>and <span class="ectt-1095">azpath</span> <!--l. 3624--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Before calculating <!--l. 3624--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>v</mi></mstyle></mrow><mrow><mn>1</mn></mrow></msub></math>,
     <!--l. 3624--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>v</mi></mstyle></mrow><mrow><mn>2</mn></mrow></msub></math>,
     <!--l. 3624--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>v</mi></mstyle></mrow><mrow><mn>3</mn></mrow></msub></math>
     and <!--l. 3624--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>v</mi></mstyle></mrow><mrow><mn>4</mn></mrow></msub></math>,
     the current frame need to be aligned to the corresponding reference frames. This option
     specifies which atoms are used to do alignment.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox209"><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances? </span>(see <a href="#colvar|cvc|forceNoPBC">description</a>)<br class="newline"/>Default: <span class="ectt-1095">no</span></li></ul>
<div class="flushright">
<!--l. 3628--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3628--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_azpath"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.12.5    </span> <a id="sec:cvc_azpath"></a><span class="ectt-1095">azpath</span>: distance from a path defined in atomic Cartesian coordinate space.</h5>
<!--l. 3628--><p class="noindent"></p></div> <a id="colvar|azpath"></a>
<!--l. 3631--><p class="indent">   Similar to <a href="#colvar|aspath"><span class="ectt-1095">aspath</span></a>, this CV computes a special case of Eq. <a href="#eq:apath_z">17<!--tex4ht:ref: eq:apath_z --></a>, and shares the same options as
<a href="#colvar|aspath"><span class="ectt-1095">aspath</span></a>.
</p><!--l. 3634--><p class="indent">   The usage of <span class="ectt-1095">azpath </span>and <span class="ectt-1095">aspath </span>is illustrated below:</p>
<div class="mdframed" id="mdframed-48">
<!--l. 3636--><p class="noindent"></p><pre>colvar {<br class="newline"/>  # Progress along the path<br class="newline"/>  name as<br class="newline"/>  # The path is defined by 5 reference frames (from string-00.pdb to string-04.pdb)<br class="newline"/>  # Use atomic coordinate from atoms 1, 2 and 3 to compute the path<br class="newline"/>  aspath {<br class="newline"/>    atoms {atomnumbers { 1 2 3 }}<br class="newline"/>    refPositionsFile1 string-00.pdb<br class="newline"/>    refPositionsFile2 string-01.pdb<br class="newline"/>    refPositionsFile3 string-02.pdb<br class="newline"/>    refPositionsFile4 string-03.pdb<br class="newline"/>    refPositionsFile5 string-04.pdb<br class="newline"/>  }<br class="newline"/>}<br class="newline"/><br class="newline"/>colvar {<br class="newline"/>  # Distance from the path<br class="newline"/>  name az<br class="newline"/>  # The path is defined by 5 reference frames (from string-00.pdb to string-04.pdb)<br class="newline"/>  # Use atomic coordinate from atoms 1, 2 and 3 to compute the path<br class="newline"/>  azpath {<br class="newline"/>    atoms {atomnumbers { 1 2 3 }}<br class="newline"/>    refPositionsFile1 string-00.pdb<br class="newline"/>    refPositionsFile2 string-01.pdb<br class="newline"/>    refPositionsFile3 string-02.pdb<br class="newline"/>    refPositionsFile4 string-03.pdb<br class="newline"/>    refPositionsFile5 string-04.pdb<br class="newline"/>  }<br class="newline"/>} </pre><!--l. 3666--><p class="noindent"></p>
</div>
<div class="flushright">
<!--l. 3668--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3668--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_neuralNetwork"></a>
<h4 class="subsectionHead"><span class="titlemark">4.13    </span> <a id="sec:cvc_neuralNetwork"></a>Dense neural network in CV space (MLCV)</h4>
<!--l. 3668--><p class="noindent"></p></div> <a id="colvar|NeuralNetwork"></a>
<!--l. 3671--><p class="indent">   This colvar component computes a non-linear combination of other scalar colvar components, where
the transformation is defined by a dense neural network.<span class="cite">[<a href="#XChen2022">14</a>]</span> The network can be optimized using any
framework, and its parameters are provided to Colvars in plain text files, as detailed below.
An example Python script to export the parameters of a TensorFlow model is provided in
                                                                                         
                                                                                         
<span class="ectt-1095">colvartools/extract_weights_biases.py </span>in the Colvars source tree.
</p>
<hr class="figure"/><div class="figure">
<a id="x1-930011"></a>
<!--l. 3677--><p class="noindent"><img alt="Dense neural network" class="neural_network" src="dense_nn.png"/>
<br/> </p><div class="caption"><span class="id">Figure 1: </span><span class="content">Graphical representation of an example dense neural network with two hidden layers
<!--l. 3679--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>y</mi></mrow><mrow><mn>1</mn></mrow></msub></math>
and <!--l. 3679--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>y</mi></mrow><mrow><mn>2</mn></mrow></msub></math>.
Both the input layer <!--l. 3679--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>x</mi></math>
and the output layer <!--l. 3679--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math>
have two nodes. The input nodes can be any existing scalar CVs. </span></div><!--tex4ht:label?: x1-930011 -->
</div><hr class="endfigure"/>
<!--l. 3683--><p class="indent">   The output of the <!--l. 3683--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>j</mi></math>-th
node of a <!--l. 3683--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>k</mi></math>-th
layer that has <!--l. 3683--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>N</mi></mrow><mrow><mi>k</mi></mrow></msub></math>
nodes is computed by
</p>
<table class="equation"><tr><td> <a id="x1-93002r20"></a>
<!--l. 3685--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<msub><mrow><mi>y</mi></mrow><mrow><mi>k</mi><mo class="MathClass-punc">,</mo><mi>j</mi></mrow></msub> <mo class="MathClass-rel">=</mo> <msub><mrow><mi>f</mi></mrow><mrow><mi>k</mi></mrow></msub> <mrow><mo fence="true" form="prefix"> (</mo><mrow><munderover accent="false" accentunder="false"><mrow><mo>∑</mo></mrow><mrow><mi>i</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><msub><mrow><mi>N</mi></mrow><mrow><mi>k</mi><mo class="MathClass-bin">−</mo><mn>1</mn></mrow></msub></mrow></munderover><msub><mrow><mi>w</mi></mrow><mrow>
<mo class="MathClass-open">(</mo><mi>k</mi><mo class="MathClass-punc">,</mo><mi>j</mi><mo class="MathClass-close">)</mo><mo class="MathClass-punc">,</mo><mo class="MathClass-open">(</mo><mi>k</mi><mo class="MathClass-bin">−</mo><mn>1</mn><mo class="MathClass-punc">,</mo><mi>i</mi><mo class="MathClass-close">)</mo></mrow></msub><msub><mrow><mi>y</mi></mrow><mrow><mi>k</mi><mo class="MathClass-bin">−</mo><mn>1</mn><mo class="MathClass-punc">,</mo><mi>i</mi></mrow></msub><mo class="MathClass-bin">+</mo><msub><mrow><mi>b</mi></mrow><mrow><mi>k</mi><mo class="MathClass-punc">,</mo><mi>j</mi></mrow></msub></mrow><mo fence="true" form="postfix">)</mo></mrow><mo class="MathClass-punc">,</mo>
</math></td><td class="eq-no">(20)</td></tr></table>
<!--l. 3689--><p class="indent">   where <!--l. 3689--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>f</mi></mrow><mrow><mi>k</mi></mrow></msub></math> is the activation
function of the <!--l. 3689--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>k</mi></math>-th
layer, <!--l. 3689--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>w</mi></mrow><mrow><mo class="MathClass-open">(</mo><mi>k</mi><mo class="MathClass-punc">,</mo><mi>j</mi><mo class="MathClass-close">)</mo><mo class="MathClass-punc">,</mo><mo class="MathClass-open">(</mo><mi>k</mi><mo class="MathClass-bin">−</mo><mn>1</mn><mo class="MathClass-punc">,</mo><mi>i</mi><mo class="MathClass-close">)</mo></mrow></msub></math> is the weight of
<!--l. 3689--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>j</mi></math>-th node with respect
to the <!--l. 3689--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math>-th output of
previous layer, and <!--l. 3689--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>b</mi></mrow><mrow><mi>k</mi><mo class="MathClass-punc">,</mo><mi>j</mi></mrow></msub></math>
is the bias of <!--l. 3689--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>j</mi></math>-th
node of <!--l. 3689--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>k</mi></math>-th
layer.
</p><!--l. 3692--><p class="noindent"><span class="phvb8t-x-x-109">List of keywords </span>(see also <a href="#sec:cvc_pbcs">4.2<!--tex4ht:ref: sec:cvc_pbcs --></a>, <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>, <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> and <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="colvar|NeuralNetwork|output_component"></a> <span class="colorbox" id="colorbox210"><a id="dx1-93003"></a><span class="ectt-1095">output_component</span></span> — <span class="phvro8t-x-x-109">The </span><!--l. 3702--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>j</mi></math><span class="phvro8t-x-x-109">-th</span>
<span class="phvro8t-x-x-109">node of the output or the last layer</span><br class="newline"/><!--l. 3702--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> integer
     starting from 0, context: <span class="ectt-1095">NeuralNetwork</span> <!--l. 3702--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>The value of this option specifies the output node to be used as the value of this CV.
     </li>
<li class="itemize"><a id="colvar|NeuralNetwork|layeri_WeightsFile"></a> <span class="colorbox" id="colorbox211"><a id="dx1-93004"></a><span class="ectt-1095">layer</span><!--l. 3716--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math><span class="ectt-1095">_WeightsFile</span></span>
     — <span class="phvro8t-x-x-109">The weights from layer </span><!--l. 3716--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi><mo class="MathClass-bin">−</mo><mn>1</mn></math>
<span class="phvro8t-x-x-109">to layer </span><!--l. 3716--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math><br class="newline"/><!--l. 3716--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> UNIX
     filename, context: <span class="ectt-1095">NeuralNetwork</span> <!--l. 3716--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>The letter <!--l. 3716--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math>
     in this option needs to be replaced with the indexing number starting from 1, for example,
     <span class="ectt-1095">layer1_WeightsFile </span>and <span class="ectt-1095">layer2_WeightsFile</span>. The value of this option specifies a plain text
     file containing the weights from layer <!--l. 3716--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi><mo class="MathClass-bin">−</mo><mn>1</mn></math>
     to layer <!--l. 3716--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math>.
                                                                                         
                                                                                         
     In the file, the number at <!--l. 3716--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>k</mi></math>-th
     column and <!--l. 3716--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>l</mi></math>-th
     row represents the weight from node <!--l. 3716--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>k</mi></math>
     at layer <!--l. 3716--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi><mo class="MathClass-bin">−</mo><mn>1</mn></math>
     to node <!--l. 3716--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>l</mi></math>
     at layer <!--l. 3716--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math>.
     </li>
<li class="itemize"><a id="colvar|NeuralNetwork|layeri_BiasesFile"></a> <span class="colorbox" id="colorbox212"><a id="dx1-93005"></a><span class="ectt-1095">layer</span><!--l. 3730--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math><span class="ectt-1095">_BiasesFile</span></span>
     — <span class="phvro8t-x-x-109">The biases from layer </span><!--l. 3730--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi><mo class="MathClass-bin">−</mo><mn>1</mn></math>
<span class="phvro8t-x-x-109">to layer </span><!--l. 3730--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math><br class="newline"/><!--l. 3730--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> UNIX
     filename, context: <span class="ectt-1095">NeuralNetwork</span> <!--l. 3730--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>The letter <!--l. 3730--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math>
     in this option needs to be replaced with the indexing number starting from 1, for example,
     <span class="ectt-1095">layer1_BiasesFile </span>and <span class="ectt-1095">layer2_BiasesFile</span>. The value of this option specifies a plain text
     file containing the weights of layer <!--l. 3730--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math>.
     The file should have only one column, where the number at <!--l. 3730--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>l</mi></math>-th
     row represents the bias of node <!--l. 3730--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>l</mi></math>
     from layer <!--l. 3730--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi><mo class="MathClass-bin">−</mo><mn>1</mn></math>
     to layer <!--l. 3730--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math>.
     </li>
<li class="itemize"><a id="colvar|NeuralNetwork|layeri_activation"></a> <span class="colorbox" id="colorbox213"><a id="dx1-93006"></a><span class="ectt-1095">layer</span><!--l. 3744--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math><span class="ectt-1095">_activation</span></span>
     — <span class="phvro8t-x-x-109">The activation function from layer </span><!--l. 3744--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi><mo class="MathClass-bin">−</mo><mn>1</mn></math>
<span class="phvro8t-x-x-109">to layer </span><!--l. 3744--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math><br class="newline"/><!--l. 3744--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> <span class="ectt-1095">tanh</span>,
     <span class="ectt-1095">sigmoid</span>, <span class="ectt-1095">linear</span>, <span class="ectt-1095">relu</span>, <span class="ectt-1095">lrelu100</span>, <span class="ectt-1095">elu </span>, context: <span class="ectt-1095">NeuralNetwork</span> <!--l. 3744--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>The letter <!--l. 3744--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math>
     in this option needs to be replaced with the indexing number starting from 1, for example,
     <span class="ectt-1095">layer1_activation </span>and <span class="ectt-1095">layer2_activation</span>. The activation function from layer <!--l. 3744--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi><mo class="MathClass-bin">−</mo><mn>1</mn></math>
     to layer <!--l. 3744--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math>.
     Available  choices  are  <span class="ectt-1095">tanh</span>,  <span class="ectt-1095">sigmoid</span>,  <span class="ectt-1095">linear </span>(identity),  <span class="ectt-1095">relu</span>,  <span class="ectt-1095">lrelu100 </span>(a  leaky  rely  with
     coefficients <!--l. 3744--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>1</mn><msup><mrow><mn>0</mn></mrow><mrow><mo class="MathClass-bin">−</mo><mn>2</mn></mrow></msup></math>
     and 1), and <span class="ectt-1095">elu </span>(with coefficient 1).
     </li>
<li class="itemize"><a id="colvar|NeuralNetwork|layeri_custom_activation"></a> <span class="colorbox" id="colorbox214"><a id="dx1-93007"></a><span class="ectt-1095">layer</span><!--l. 3760--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math><span class="ectt-1095">_custom_activation</span></span>
     — <span class="phvro8t-x-x-109">An alternative custom expression as the activation function from layer </span><!--l. 3760--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi><mo class="MathClass-bin">−</mo><mn>1</mn></math>
<span class="phvro8t-x-x-109">to layer </span><!--l. 3760--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math><br class="newline"/><!--l. 3760--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> string,
     context: <span class="ectt-1095">NeuralNetwork</span> <!--l. 3760--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Mathematical expression to define the activation function from layer <!--l. 3760--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi><mo class="MathClass-bin">−</mo><mn>1</mn></math>
     to layer <!--l. 3760--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math>.
     The  input  value  must  be  written  as  <span class="ectt-1095">x</span>.  For  example,  the  ELU  activation  function  can  be
     expressed as <span class="ectt-1095">select(step(x), alpha*(exp(x)-1), x)</span>. For details of the expression syntax,
                                                                                         
                                                                                         
     see <a href="#colvar|customFunction"><span class="ectt-1095">customFunction</span></a>. This option is mutually exclusive with <a href="#."><span class="ectt-1095">colvar|NeuralNetwork|layeri_activation</span></a>
</li>
<li class="itemize"><span class="colorbox" id="colorbox215"><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances? </span>(see <a href="#colvar|cvc|forceNoPBC">description</a>)<br class="newline"/>Default: <span class="ectt-1095">no</span></li></ul>
<!--l. 3764--><p class="indent">   An example of configuration using <span class="ectt-1095">NeuralNetwork </span>is shown below:</p>
<div class="mdframed" id="mdframed-49">
<!--l. 3766--><p class="noindent"></p><pre>colvar {<br class="newline"/>  # Define a neural network with 2 layers<br class="newline"/>  # The inputs are two torsion angles<br class="newline"/>  # and the first node at the output layer is used as the final CV<br class="newline"/>  name nn_output_1<br class="newline"/>  NeuralNetwork {<br class="newline"/>    output_component 0<br class="newline"/>    layer1_WeightsFile      dense_1_weights.txt<br class="newline"/>    layer1_BiasesFile       dense_1_biases.txt<br class="newline"/>    layer1_activation       tanh<br class="newline"/>    layer2_WeightsFile      dense_2_weights.txt<br class="newline"/>    layer2_BiasesFile       dense_2_biases.txt<br class="newline"/>    layer2_activation       tanh<br class="newline"/>    # The component coefficient is used for normalization<br class="newline"/>    componentCoeff 180.0<br class="newline"/>    dihedral {<br class="newline"/>      name 001<br class="newline"/>      # normalization factor 1.0/180.0<br class="newline"/>      componentCoeff 0.00555555555555555556<br class="newline"/>      group1 {atomNumbers {5}}<br class="newline"/>      group2 {atomNumbers {7}}<br class="newline"/>      group3 {atomNumbers {9}}<br class="newline"/>      group4 {atomNumbers {15}}<br class="newline"/>    }<br class="newline"/>    dihedral {<br class="newline"/>      name 002<br class="newline"/>      # normalization factor 1.0/180.0<br class="newline"/>      componentCoeff 0.00555555555555555556<br class="newline"/>      group1 {atomNumbers {7}}<br class="newline"/>      group2 {atomNumbers {9}}<br class="newline"/>      group3 {atomNumbers {15}}<br class="newline"/>      group4 {atomNumbers {17}}<br class="newline"/>    }<br class="newline"/>  }<br class="newline"/>}<br class="newline"/></pre><!--l. 3802--><p class="noindent"></p>
</div>
<div class="flushright">
<!--l. 3805--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3805--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_torchann"></a>
<h4 class="subsectionHead"><span class="titlemark">4.14    </span> <a id="sec:cvc_torchann"></a>Torch-based collective variables</h4>
<!--l. 3805--><p class="noindent"></p></div>
<!--l. 3807--><p class="indent">   The <span class="ectt-1095">torchANN </span>keyword defines a scalar component based on an artificial neural network (ANN) model
previously trained using PyTorch. Similarly to the <span class="ectt-1095">linearCombination </span>and <span class="ectt-1095">neuralNetwork </span>components,
the ANN model is expressed as a function of collective variable components. The model is loaded from a
PyTorch-formatted file, defined by the keyword <span class="ectt-1095">modelFile</span>.
</p><!--l. 3809--><p class="indent">   This component requires that NAMD is linked against the libTorch library; see <a href="#sec:compilation_notes">9<!--tex4ht:ref: sec:compilation_notes --></a> for more
details.
</p><!--l. 3811--><p class="noindent"><span class="phvb8t-x-x-109">List of keywords </span>(see also <a href="#sec:cvc_pbcs">4.2<!--tex4ht:ref: sec:cvc_pbcs --></a>, <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>, <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> and <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="colvar|torchANN|modelFile"></a> <span class="colorbox" id="colorbox216"><a id="dx1-94001"></a><span class="ectt-1095">modelFile</span></span> — <span class="phvro8t-x-x-109">Name of the file containing the model (PyTorch format)</span><br class="newline"/><!--l. 3822--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> filename,
     context: <span class="ectt-1095">torchANN</span> <!--l. 3822--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This file contains the PyTorch model, as saved for example by the <span class="ectt-1095">torch.jit.save() </span>function.
     The model is assumed to take a one-dimensional tensor (excluding batch dimension) as
     input.
     </li>
<li class="itemize"><a id="colvar|torchANN|output_component"></a> <span class="colorbox" id="colorbox217"><a id="dx1-94002"></a><span class="ectt-1095">output_component</span></span> — <span class="phvro8t-x-x-109">Index of the output node to use</span><br class="newline"/>Default: 0 <!--l. 3833--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> 0-based
     integer, context: <span class="ectt-1095">torchANN</span> <!--l. 3833--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option specifies the index of the output node to be used as the value of the collective
     variable (see also the equivalent keyword <a href="#colvar|NeuralNetwork|output_component"><span class="ectt-1095">output_component</span></a> in the built-in neural network
     implementation).
     </li>
<li class="itemize"><a id="colvar|torchANN|doubleInputTensor"></a> <span class="colorbox" id="colorbox218"><a id="dx1-94003"></a><span class="ectt-1095">doubleInputTensor</span></span> — <span class="phvro8t-x-x-109">Specify that the input model is in double precision</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 3844--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">torchANN</span> <!--l. 3844--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This flags controls the precision of the PyTorch model contained in <span class="ectt-1095">modelFile</span>.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox219"><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances? </span>(see <a href="#colvar|cvc|forceNoPBC">description</a>)<br class="newline"/>Default: <span class="ectt-1095">no</span></li></ul>
<!--l. 3848--><p class="indent">   An example of creating a model file and defining a collective variable component using <span class="ectt-1095">torchANN </span>is
shown below:</p>
<div class="mdframed" id="mdframed-50">
<!--l. 3850--><p class="noindent"></p><pre>import torch<br class="newline"/>import torch.nn as nn<br class="newline"/><br class="newline"/># The neural network model is a function: <br class="newline"/>#     <!--l. 3855--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>f</mi> <mo class="MathClass-rel">=</mo> <mo class="MathClass-open">(</mo><msub><mrow><mi>f</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>f</mi></mrow><mrow><mn>2</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>f</mi></mrow><mrow><mn>3</mn></mrow></msub><mo class="MathClass-close">)</mo> <mo class="MathClass-punc">:</mo> <msup><mrow><mi>ℝ</mi></mrow><mrow><mn>8</mn></mrow></msup> <mo class="MathClass-rel">→</mo> <msup><mrow><mi>ℝ</mi></mrow><mrow><mn>3</mn></mrow></msup></math>.<br class="newline"/>model = nn.Sequential(nn.Linear(8, 100), <br class="newline"/>                      nn.ReLU(), <br class="newline"/>                      nn.Linear(100, 3))<br class="newline"/># Train the model<br class="newline"/>...<br class="newline"/># Save the trained model<br class="newline"/>torch.jit.script(model).save("./model.pt") </pre><!--l. 3863--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-51">
<!--l. 3865--><p class="noindent"></p><pre>colvar {<br class="newline"/>  # Define a scalar CV as <!--l. 3867--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi> <mo class="MathClass-rel">=</mo> <msub><mrow><mi>f</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-open">(</mo><msub><mrow><mi>x</mi></mrow><mrow><mn>2</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>y</mi></mrow><mrow><mn>2</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>z</mi></mrow><mrow><mn>2</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>x</mi></mrow><mrow><mn>4</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>y</mi></mrow><mrow><mn>4</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>z</mi></mrow><mrow><mn>4</mn></mrow></msub><mo class="MathClass-punc">,</mo><mi>ϕ</mi><mo class="MathClass-punc">,</mo><mi>ψ</mi><mo class="MathClass-close">)</mo></math>,
<br class="newline"/>  # i.e. the function represented by the first component of the Torch model, <br class="newline"/>  # evaluated at the coordinates of the two atoms with atomNumbers 2 and 4,<br class="newline"/>  # and the two torsion angles <!--l. 3870--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ϕ</mi><mo class="MathClass-punc">,</mo><mi>ψ</mi></math>.<br class="newline"/>  name nn_output_1<br class="newline"/>  torchann {<br class="newline"/>    modelFile model.pt<br class="newline"/>    output_component 0<br class="newline"/>    cartesian {<br class="newline"/>      atoms {<br class="newline"/>        atomNumbers {{2 4}}<br class="newline"/>      }<br class="newline"/>    }<br class="newline"/>    dihedral {<br class="newline"/>      name 001<br class="newline"/>      group1 {atomnumbers {5}}<br class="newline"/>      group2 {atomnumbers {7}}<br class="newline"/>      group3 {atomnumbers {9}}<br class="newline"/>      group4 {atomnumbers {15}}<br class="newline"/>    }<br class="newline"/>    dihedral {<br class="newline"/>      name 002<br class="newline"/>      group1 {atomnumbers {7}}<br class="newline"/>      group2 {atomnumbers {9}}<br class="newline"/>      group3 {atomnumbers {15}}<br class="newline"/>      group4 {atomnumbers {17}}<br class="newline"/>    }<br class="newline"/>  }<br class="newline"/>} </pre><!--l. 3896--><p class="noindent"></p>
</div>
<div class="flushright">
<!--l. 3900--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3900--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_volmaps"></a>
<h4 class="subsectionHead"><span class="titlemark">4.15    </span> <a id="sec:cvc_volmaps"></a>Volumetric map-based variables</h4>
<!--l. 3900--><p class="noindent"></p></div>
<!--l. 3902--><p class="indent">   Volumetric maps of the Cartesian coordinates, typically defined as mesh grid along the three
Cartesian axes, may be used to define collective variables. In NAMD this feature implemented as an
interface between Colvars and <a href="https://www.ks.uiuc.edu/Research/namd/2.14/ug/node43.html">GridForces</a>. Please cite <span class="cite">[<a href="#XFiorin2020">15</a>]</span> when using this implementation of collective
variables based on volumetric maps.
</p>
<div class="flushright">
<!--l. 3906--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 3906--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_map_total"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.15.1    </span> <a id="sec:cvc_map_total"></a><span class="ectt-1095">mapTotal</span>: total value of a volumetric map</h5>
<!--l. 3906--><p class="noindent"></p></div> <a id="colvar|mapTotal"></a>
<!--l. 3909--><p class="indent">   Given  a  function  of  the  Cartesian  coordinates
<!--l. 3909--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ϕ</mi><mo class="MathClass-open">(</mo><mstyle mathvariant="bold"><mi>x</mi></mstyle><mo class="MathClass-close">)</mo> <mo class="MathClass-rel">=</mo> <mi>ϕ</mi><mo class="MathClass-open">(</mo><mi>x</mi><mo class="MathClass-punc">,</mo><mi>y</mi><mo class="MathClass-punc">,</mo><mi>z</mi><mo class="MathClass-close">)</mo></math>, a <span class="ectt-1095">mapTotal </span>collective variable
component <!--l. 3909--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>Φ</mi><mo class="MathClass-open">(</mo><mstyle mathvariant="bold"><mi>X</mi></mstyle><mo class="MathClass-close">)</mo></math> is defined as the sum of
the values of the function <!--l. 3909--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ϕ</mi><mo class="MathClass-open">(</mo><mstyle mathvariant="bold"><mi>x</mi></mstyle><mo class="MathClass-close">)</mo></math> evaluated
at the coordinates of each atom, <!--l. 3909--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>i</mi></mrow></msub></math>:
</p><table class="equation"><tr><td> <a id="x1-96001r21"></a>
<!--l. 3910--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>Φ</mi> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mstyle mathvariant="bold"><mi>X</mi></mstyle></mrow><mo fence="true" form="postfix">)</mo></mrow> <mo class="MathClass-rel">=</mo><munderover accent="false" accentunder="false"><mrow><mo> ∑</mo></mrow><mrow><mi>i</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><mi>N</mi></mrow></munderover><msub><mrow><mi>w</mi></mrow><mrow>
<mi>i</mi></mrow></msub><mi>ϕ</mi> <mrow><mo fence="true" form="prefix"> (</mo><mrow><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>i</mi></mrow></msub></mrow><mo fence="true" form="postfix">)</mo></mrow>
</math></td><td class="eq-no">(21)</td></tr></table>
<!--l. 3914--><p class="noindent">where <!--l. 3914--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>w</mi></mrow><mrow><mi>i</mi></mrow></msub></math>
are weights assigned to each variable (1 by default). This formulation allows, for example, to
“count" the number of atoms within a region of space by using a positive-valued function
<!--l. 3915--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ϕ</mi> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mo fence="true" form="postfix">)</mo></mrow></math>, such
as for example the number of water molecules in a hydrophobic cavity <span class="cite">[<a href="#XFiorin2020">15</a>]</span>.<br class="newline"/>
</p><!--l. 3917--><p class="indent">   Because the volumetric map itself and the atoms affected by it are defined externally to Colvars, this
component has a very limited number of keywords.
</p><!--l. 3919--><p class="noindent"><span class="phvb8t-x-x-109">List of keywords </span>(see also <a href="#sec:cvc_pbcs">4.2<!--tex4ht:ref: sec:cvc_pbcs --></a>, <a href="#sec:cvc_common">4.16<!--tex4ht:ref: sec:cvc_common --></a>, <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> and <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
<ul class="itemize1">
<li class="itemize"><a id="colvar|mapTotal|mapName"></a> <span class="colorbox" id="colorbox220"><a id="dx1-96002"></a><span class="ectt-1095">mapName</span></span> — <span class="phvro8t-x-x-109">Specify the name of the volumetric map to use as a colvar</span><br class="newline"/><!--l. 3931--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> string,
     context: <span class="ectt-1095">mapTotal</span> <!--l. 3931--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>The value of this option specifies the label of the volumetric map to use for this collective
     variable component. This label must identify a map already loaded in NAMD via <span class="ectt-1095">mGridForcePotFile</span>,
     and the corresponding value of <span class="ectt-1095">mGridForceScale </span>must be set to (0, 0, 0), so that its collective
     force is computed dynamically.
     </li>
<li class="itemize"><a id="colvar|mapTotal|mapID"></a> <span class="colorbox" id="colorbox221"><a id="dx1-96003"></a><span class="ectt-1095">mapID</span></span> — <span class="phvro8t-x-x-109">Specify the numeric index of the volumetric map to use as a colvar</span><br class="newline"/><!--l. 3943--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> non-negative
     integer, context: <span class="ectt-1095">mapTotal</span> <!--l. 3943--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>The value of this option specifies the numeric index of the volumetric map to use for this
     collective variable component. The map referred to must be already loaded before using this
     option. This option is mutually exclusive with <a href="#colvar|mapTotal|mapName"><span class="ectt-1095">mapName</span></a>; if the latter is given, the numeric ID is
     obtained internally from NAMD.
     </li>
<li class="itemize"><a id="colvar|mapTotal|atoms"></a> <span class="colorbox" id="colorbox222"><a id="dx1-96004"></a><span class="ectt-1095">atoms</span></span> — <span class="phvro8t-x-x-109">Group of atoms defining this function</span><br class="newline"/><!--l. 3956--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> <a href="#sec:colvar_atom_groups">Atom
     group</a>, context: <span class="ectt-1095">mapTotal</span> <!--l. 3956--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If defined, this option allows defining the set of atoms to be aligned onto the volumetric map
     following the syntax described in section <a href="#sec:colvar_atom_groups">5<!--tex4ht:ref: sec:colvar_atom_groups --></a>. This also allows, for instance, to express the
     volumetric map in a rotated frame of reference (see <a href="#sec:colvar_atom_groups_ref_frame">5.2<!--tex4ht:ref: sec:colvar_atom_groups_ref_frame --></a>).  In NAMD, this keyword is optional;
     unless a frame of reference other than the laboratory is needed, it is computationally much
     more efficient to select atoms via <a href="https://www.ks.uiuc.edu/Research/namd/2.14/ug/node43.html">GridForces</a> keywords.
     </li>
<li class="itemize"><a id="colvar|mapTotal|atomWeights"></a> <span class="colorbox" id="colorbox223"><a id="dx1-96005"></a><span class="ectt-1095">atomWeights</span></span> — <span class="phvro8t-x-x-109">Weights to assign to each atom</span><br class="newline"/>Default: all weights equal to 1 <!--l. 3968--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> list
     of space-separated decimals, context: <span class="ectt-1095">mapTotal</span> <!--l. 3968--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If defined, the value of this option assigns weights <!--l. 3968--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>w</mi></mrow><mrow><mi>i</mi></mrow></msub></math>
     to the individual atoms in a <span class="ectt-1095">mapTotal </span>variable. This option requires defining <span class="ectt-1095">atoms </span>explicitly,
     and the number of weights must match the number of atoms.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox224"><span class="ectt-1095">forceNoPBC</span></span> — <span class="phvro8t-x-x-109">Use absolute rather than minimum-image distances? </span>(see <a href="#colvar|cvc|forceNoPBC">description</a>)<br class="newline"/>Default: <span class="ectt-1095">no</span></li></ul>
<!--l. 4006--><p class="noindent"><span class="phvb8t-x-x-109">Example: biasing the number of molecules inside a cavity using a volumetric map.</span>
</p><!--l. 4006--><p class="indent">   Firstly, a volumetric map that has a value of 1 inside the cavity and 0 outside should be prepared. A
reasonable starting point may be obtained, for example, with VMD: using an existing trajectory where the
cavity is occupied by solvent and a spatial selection that identifies all the molecules within the cavity,
<span class="ectt-1095">volmap occupancy -allframes -combine max </span>computes the occupancy map as a step function
(values 1 or 0), and <span class="ectt-1095">volutil -smooth </span><span class="ectt-1095">… </span>makes it a continuous map, suitable for use as a MD
simulation bias. A PDB file defining the selection (for example, where all water oxygens and ions
have an occupancy of 1 and other atoms 0) is also prepared using VMD. Both the map file
and the atom selection file are then loaded via the <span class="ectt-1095">mGridForcePotFile </span>and related NAMD
                                                                                         
                                                                                         
commands:</p>
<div class="mdframed" id="mdframed-52">
<!--l. 4006--><p class="noindent"></p><pre>mGridForce yes<br class="newline"/>mGridForcePotFile Cavity cavity.dx   # OpenDX map file<br class="newline"/>mGridForceFile Cavity water-sel.pdb  # PDB file used for atom selection<br class="newline"/>mGridForceCol Cavity O               # Use the occupancy column of the PDB file<br class="newline"/>mGridForceChargeCol Cavity B         # Use beta as “charge" (default: electric charge)<br class="newline"/>mGridForceScale Cavity 0.0 0.0 0.0   # Do not use GridForces for this map<br class="newline"/></pre><!--l. 4006--><p class="noindent"></p>
</div>
<!--l. 4006--><p class="indent">   The value of <span class="ectt-1095">mGridForceScale </span>is particularly important, because it determines the GridForces force
constant for the “Cavity" map. A non-zero value enables a conventional GridForces calculation, where the
force constant remains fixed within each <span class="ectt-1095">run </span>command and the forces on the atoms depend only on their
positions in space. However, setting <span class="ectt-1095">mGridForceScale </span>to zero signals to NAMD that the force acting
through the volumetric map may be computed dynamically, as part of a collective-variable
biasing scheme. To do so, the map labeled “Cavity" needs to be referred to in the Colvars
configuration:<br class="newline"/></p>
<div class="mdframed" id="mdframed-53">
<!--l. 4006--><p class="noindent"></p><pre>cv config "<br class="newline"/>colvar {<br class="newline"/>  name n_waters<br class="newline"/>  mapTotal {<br class="newline"/>    mapName Cavity # Same label as the GridForce map<br class="newline"/>  }<br class="newline"/>}" </pre><!--l. 4006--><p class="noindent"></p>
</div>
<!--l. 4006--><p class="indent">   The variable “<span class="ectt-1095">n_waters</span>" may then be used with any of the enhanced sampling methods available (<a href="#sec:colvarbias">6<!--tex4ht:ref: sec:colvarbias --></a>):
new forces applied to it at each simulation step will be transmitted to the corresponding atoms within the
same step.
</p>
<div class="flushright">
<!--l. 4009--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 4009--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_multimap"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.15.2    </span> <a id="sec:cvc_multimap"></a>Multiple volumetric maps collective variables</h5>
<!--l. 4009--><p class="noindent"></p></div>
<!--l. 4011--><p class="indent">   To study processes that involve changes in shape of a macromolecular aggregate (for example,
deformations of lipid membranes) it is useful to define collective variables based on more than one
volumetric map at a time, measuring the relative similarity with each map while still achieving correct
thermodynamic sampling of each state.
</p><!--l. 4013--><p class="indent">   This is achieved by combining multiple <span class="ectt-1095">mapTotal </span>components, each
based on a differently-shaped volumetric map, into a single collective variable
<!--l. 4013--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>.
                                                                                         
                                                                                         
To track transitions between states, the contribution of each map to
<!--l. 4014--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math> should
be discriminated from the others, for example by assigning to it a different weight. The “Multi-Map"
progress variable <span class="cite">[<a href="#XFiorin2020">15</a>]</span> uses a weight sum of these components, using linearly-increasing weights: </p><table class="equation"><tr><td>
<a id="x1-97001r22"></a>
<!--l. 4016--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>ξ</mi> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mstyle mathvariant="bold"><mi>X</mi></mstyle></mrow><mo fence="true" form="postfix">)</mo></mrow> <mo class="MathClass-rel">=</mo><munderover accent="false" accentunder="false"><mrow><mo> ∑</mo></mrow><mrow><mi>k</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><mi>K</mi></mrow></munderover><mi>k</mi><msub><mrow><mi>Φ</mi></mrow><mrow>
<mi>k</mi></mrow></msub> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mstyle mathvariant="bold"><mi>X</mi></mstyle></mrow><mo fence="true" form="postfix">)</mo></mrow> <mo class="MathClass-rel">=</mo><munderover accent="false" accentunder="false"><mrow><mo> ∑</mo></mrow><mrow><mi>k</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><mi>K</mi></mrow></munderover><mi>k</mi><munderover accent="false" accentunder="false"><mrow><mo>∑</mo></mrow><mrow>
<mi>i</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><mi>N</mi></mrow></munderover><msub><mrow><mi>ϕ</mi></mrow><mrow>
<mi>k</mi></mrow></msub> <mrow><mo fence="true" form="prefix"> (</mo><mrow><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>i</mi></mrow></msub></mrow><mo fence="true" form="postfix">)</mo></mrow>
</math></td><td class="eq-no">(22)</td></tr></table>
<!--l. 4020--><p class="noindent">where <!--l. 4020--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>K</mi></math> is the number of
maps employed and each <!--l. 4020--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>Φ</mi></mrow><mrow><mi>k</mi></mrow></msub></math>
is a <span class="ectt-1095">mapTotal </span>component.
</p><!--l. 4022--><p class="indent">   Here is a link to the <span class="phvb8t-x-x-109">Multi-Map tutorial </span>page: <a class="url" href="https://colvars.github.io/multi-map/multi-map.html"><span class="ectt-1095">https://colvars.github.io/multi-map/multi-map.html</span></a><br class="newline"/>An example configuration for illustration purposes is also included below.
</p><!--l. 4025--><p class="noindent"><span class="phvb8t-x-x-109">Example: transitions between macromolecular shapes using volumetric maps.</span><br class="newline"/>A series of map files, each representing a different shape, is loaded into NAMD:<br class="newline"/></p>
<div class="mdframed" id="mdframed-54">
<!--l. 4027--><p class="noindent"></p><pre>mGridForce yes<br class="newline"/>for { set k 1 } { $k &lt;= $K } { incr k } {<br class="newline"/>  mGridForcePotFile Shape_$k map_$k.dx   # Density map of the k-th state<br class="newline"/>  mGridForceFile Shape_$k atoms.pdb   # PDB file used for atom selection<br class="newline"/>  mGridForceCol Shape_$k O   # Use the occupancy column of the PDB file atoms.pdb<br class="newline"/>  mGridForceChargeCol Shape_$k B   # Use beta as “charge" (default: electric charge)<br class="newline"/>  mGridForceScale Shape_$k 0.0 0.0 0.0  # Do not use GridForces for this map<br class="newline"/>} </pre><!--l. 4036--><p class="noindent"></p>
</div>
<!--l. 4038--><p class="indent">   The GridForces maps thus loaded are then used to define the Multi-Map collective variable, with
coefficients <!--l. 4038--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>k</mi></mrow></msub> <mo class="MathClass-rel">=</mo> <mi>k</mi></math>
<span class="cite">[<a href="#XFiorin2020">15</a>]</span>:<br class="newline"/></p>
<div class="mdframed" id="mdframed-55">
<!--l. 4039--><p class="noindent"></p><pre># Collect the definition of all components into one string<br class="newline"/>set components "<br class="newline"/>for { set k 1 } { $k &lt;= $K } { incr k } {<br class="newline"/>  set components "${components}<br class="newline"/>  mapTotal {<br class="newline"/>    mapName Shape_$k<br class="newline"/>    componentCoeff $k<br class="newline"/>  }<br class="newline"/>}<br class="newline"/>"<br class="newline"/># Use this string to define the variable<br class="newline"/>cv config "<br class="newline"/>colvar {<br class="newline"/>  name shapes<br class="newline"/>  ${components}<br class="newline"/>}" </pre><!--l. 4056--><p class="noindent"></p>
</div>
<!--l. 4058--><p class="indent">   The above example illustrates a use case where a weighted sum (i.e. a linear combination) is used to
define a single variable from multiple components. Depending on the problem under study, non-linear
functions may be more appropriate. These may be defined a custom functions if implemented (see <a href="#sec:colvar_custom_function">4.20<!--tex4ht:ref: sec:colvar_custom_function --></a>),
or scripted functions (see <a href="#sec:colvar_scripted">4.21<!--tex4ht:ref: sec:colvar_scripted --></a>).
</p>
<div class="flushright">
<!--l. 4065--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 4065--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_common"></a>
<h4 class="subsectionHead"><span class="titlemark">4.16    </span> <a id="sec:cvc_common"></a>Shared keywords for all components</h4>
<!--l. 4065--><p class="noindent"></p></div>
<!--l. 4067--><p class="indent">   The following options can be used for any of the above colvar components in order to
obtain a polynomial combination or any user-supplied function provided by <a href="#sec:cvc_superp"><span class="ectt-1095">scriptedFunction</span></a>.
</p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox225"><a id="dx1-98001"></a><span class="ectt-1095">name</span></span> — <span class="phvro8t-x-x-109">Name of this component</span><br class="newline"/>Default: type of component + numeric id <!--l. 4079--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> string,
     context: any component <!--l. 4079--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>The name is an unique case-sensitive string which allows the Colvars module to identify
     this  component.  This  is  useful,  for  example,  when  combining  multiple  components  via  a
     <span class="ectt-1095">scriptedFunction</span>.It also defines the variable name representing the component's value in
     a <a href="#colvar|customFunction"><span class="ectt-1095">customFunction</span></a> expression.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox226"><a id="dx1-98002"></a><span class="ectt-1095">scalable</span></span> — <span class="phvro8t-x-x-109">Attempt to calculate this component in parallel?</span><br class="newline"/>Default: <span class="ectt-1095">on</span>, if available <!--l. 4091--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: any component <!--l. 4091--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If set to <span class="ectt-1095">on </span>(default), the Colvars module will attempt to calculate this component in parallel
     to reduce overhead. Whether this option is available depends on the type of component:
     currently supported are <span class="ectt-1095">distance</span>, <span class="ectt-1095">distanceZ</span>, <span class="ectt-1095">distanceXY</span>, <span class="ectt-1095">distanceVec</span>, <span class="ectt-1095">distanceDir</span>, <span class="ectt-1095">angle</span>
     and <span class="ectt-1095">dihedral</span>. This flag influences computational cost, but does not affect numerical results:
     therefore, it should only be turned off for debugging or testing purposes. </li></ul>
<div class="flushright">
<!--l. 4095--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 4095--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_periodic"></a>
<h4 class="subsectionHead"><span class="titlemark">4.17    </span> <a id="sec:cvc_periodic"></a>Periodic components</h4>
<!--l. 4095--><p class="noindent"></p></div>
<!--l. 4097--><p class="indent">   Certain components, such as <a href="#colvar|dihedral"><span class="ectt-1095">dihedral</span></a> or <a href="#colvar|spinAngle"><span class="ectt-1095">dihedral</span></a>, compute angles that lie in a periodic interval
between <!--l. 4099--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-bin">−</mo><mn>1</mn><mn>8</mn><msup><mrow><mn>0</mn></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></math>
and <!--l. 4099--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>1</mn><mn>8</mn><msup><mrow><mn>0</mn></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></math>.
When computing pairwise distances between values of those angles (e.g. for the sake of computing
restraint potentials, or sampling PMFs), periodicity is taken into account by following the minimum-image
convention.
</p><!--l. 4103--><p class="indent">   Additionally, several other components, such as <a href="#colvar|distanceZ"><span class="ectt-1095">distanceZ</span></a>, support optional periodicity if this is
provided in the configuration.
</p><!--l. 4106--><p class="indent">   The following keywords can be used within periodic components, or within custom variables (<a href="#sec:colvar_custom_function">4.20<!--tex4ht:ref: sec:colvar_custom_function --></a>), or
wthin scripted variables <a href="#sec:colvar_scripted">4.21<!--tex4ht:ref: sec:colvar_scripted --></a>).
</p>
<ul class="itemize1">
<li class="itemize"><a id="colvar|cvc|period"></a> <span class="colorbox" id="colorbox227"><a id="dx1-99001"></a><span class="ectt-1095">period</span></span> — <span class="phvro8t-x-x-109">Period of the component</span><br class="newline"/>Default: 0.0, unless otherwise stated <!--l. 4123--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     decimal, context: components with optional periodicity <!--l. 4123--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Setting this number enables the treatment of as a periodic component: by default, <span class="ectt-1095">distanceZ</span>
     is  not  considered  periodic.  The  keyword  is  supported,  but  irrelevant  within  <span class="ectt-1095">dihedral </span>or
     <span class="ectt-1095">spinAngle</span>, because their period is always 360 degrees.
     </li>
<li class="itemize"><a id="colvar|cvc|wrapAround"></a> <span class="colorbox" id="colorbox228"><a id="dx1-99002"></a><span class="ectt-1095">wrapAround</span></span> — <span class="phvro8t-x-x-109">Center of the wrapping interval for periodic variables</span><br class="newline"/>Default: 0.0 <!--l. 4135--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> decimal,
     context: periodic components <!--l. 4135--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>By default, values of the periodic components are centered around zero, ranging from <!--l. 4135--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-bin">−</mo><mi>P</mi><mo class="MathClass-bin">∕</mo><mn>2</mn></math>
     to <!--l. 4135--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>P</mi><mo class="MathClass-bin">∕</mo><mn>2</mn></math>,
     where <!--l. 4135--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>P</mi></math>
     is the period. Setting this number centers the interval around this value. This can be useful
     for convenience of output, or to set the walls for a <span class="ectt-1095">harmonicWalls </span>in an order that would not
     otherwise be allowed.</li></ul>
<!--l. 4138--><p class="indent"> <span class="phvro8t-x-x-109">Note: using linear/polynomial combinations of periodic components (see </span><a href="#sec:cvc_superp"><span class="phvro8t-x-x-109">4.19</span><!--tex4ht:ref: sec:cvc_superp --></a><span class="phvro8t-x-x-109">), or other custom or</span>
<span class="phvro8t-x-x-109">scripted function may invalidate the periodicity. Use such combinations carefully: estimate the range of</span>
<span class="phvro8t-x-x-109">possible values of each component in a given simulation, and make use of </span><span class="ecit-1095">wrapAround </span><span class="phvro8t-x-x-109">to limit this</span>
<span class="phvro8t-x-x-109">problem whenever possible.</span>
</p>
<div class="flushright">
<!--l. 4141--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 4141--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_non_scalar"></a>
<h4 class="subsectionHead"><span class="titlemark">4.18    </span> <a id="sec:cvc_non_scalar"></a>Non-scalar components</h4>
<!--l. 4141--><p class="noindent"></p></div>
<!--l. 4143--><p class="indent">   When one of the following components are used, the defined colvar returns a value that is not a scalar
number: </p>
<ul class="itemize1">
<li class="itemize"><span class="ectt-1095">distanceVec</span>: 3-dimensional vector of the distance between two groups;
     </li>
<li class="itemize"><span class="ectt-1095">distanceDir</span>: 3-dimensional unit vector of the distance between two groups;
     </li>
<li class="itemize"><span class="ectt-1095">orientation</span>: 4-dimensional unit quaternion representing the best-fit rotation from a set of
     reference coordinates.</li></ul>
<!--l. 4152--><p class="noindent">The distance between two 3-dimensional unit vectors is computed as the angle
between them. The distance between two quaternions is computed as the angle
between the two 4-dimensional unit vectors: because the orientation represented by
<!--l. 4155--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="sans-serif"><mi>q</mi></mstyle></math> is the same as the
one represented by <!--l. 4156--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-bin">−</mo><mstyle mathvariant="sans-serif"><mi>q</mi></mstyle></math>,
distances between two quaternions are computed considering the closest of the two symmetric
images.
</p><!--l. 4159--><p class="indent">   Non-scalar components carry the following restrictions: </p>
<ul class="itemize1">
<li class="itemize">Calculation of total forces (<span class="ectt-1095">outputTotalForce </span>option) is currently not implemented.
     </li>
<li class="itemize">Each colvar can only contain one non-scalar component.
     </li>
<li class="itemize">Binning on a grid (<span class="ectt-1095">abf</span>, <span class="ectt-1095">histogram </span>and <span class="ectt-1095">metadynamics </span>with <span class="ectt-1095">useGrids </span>enabled) is currently not
     implemented for colvars based on such components.</li></ul>
<!--l. 4172--><p class="indent"> <span class="phvro8t-x-x-109">Note: while these restrictions apply to individual colvars based on non-scalar components, no limit is</span>
<span class="phvro8t-x-x-109">set to the number of scalar colvars. To compute multi-dimensional histograms and PMFs, use sets of</span>
<span class="phvro8t-x-x-109">scalar colvars of arbitrary size.</span>
</p>
<div class="flushright">
<!--l. 4175--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 4175--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_sys_forces"></a>
<h5 class="subsubsectionHead"><span class="titlemark">4.18.1    </span> <a id="sec:cvc_sys_forces"></a>Calculating total forces</h5>
<!--l. 4175--><p class="noindent"></p></div> In addition to the restrictions due to the type of value computed (scalar or non-scalar), a final restriction
can arise when calculating total force (<span class="ectt-1095">outputTotalForce </span>option or application of a <span class="ectt-1095">abf </span>bias). total forces
are available currently only for the following components: <span class="ectt-1095">distance</span>, <span class="ectt-1095">distanceZ</span>, <span class="ectt-1095">distanceXY</span>, <span class="ectt-1095">angle</span>,
<span class="ectt-1095">dihedral</span>, <span class="ectt-1095">rmsd</span>, <span class="ectt-1095">eigenvector </span>and <span class="ectt-1095">gyration</span>.

                                                                                         <div class="flushright">
<!--l. 4186--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 4186--><p class="noindent"> </p><div class="heading"> <a id="sec:cvc_superp"></a>
<h4 class="subsectionHead"><span class="titlemark">4.19    </span> <a id="sec:cvc_superp"></a>Linear and polynomial combinations of components</h4>
<!--l. 4186--><p class="noindent"></p></div>
<!--l. 4188--><p class="indent">   To  extend  the  set  of  possible  definitions  of  colvars
<!--l. 4188--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi><mo class="MathClass-open">(</mo><mstyle mathvariant="bold"><mi>r</mi></mstyle><mo class="MathClass-close">)</mo></math>, multiple
components <!--l. 4189--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>q</mi></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-open">(</mo><mstyle mathvariant="bold"><mi>r</mi></mstyle><mo class="MathClass-close">)</mo></math>
can be summed with the formula: </p><table class="equation"><tr><td> <a id="eq:colvar_combination"></a>
<!--l. 4190--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>ξ</mi> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mstyle mathvariant="bold"><mi>r</mi></mstyle></mrow><mo fence="true" form="postfix">)</mo></mrow> <mo class="MathClass-rel">=</mo><munder class="msub"><mrow><mo> ∑</mo></mrow><mrow><mi>i</mi></mrow></munder><msub><mrow><mi>c</mi></mrow><mrow><mi>i</mi></mrow></msub><msup><mrow><mo class="MathClass-open">[</mo><msub><mrow><mi>q</mi></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-open">(</mo><mstyle mathvariant="bold"><mi>r</mi></mstyle><mo class="MathClass-close">)</mo><mo class="MathClass-close">]</mo></mrow><mrow><msub><mrow><mi>n</mi></mrow><mrow><mi>i</mi></mrow></msub>
</mrow></msup>
</math></td><td class="eq-no">(23)</td></tr></table>
<!--l. 4194--><p class="indent">   where each component appears with a unique coefficient
<!--l. 4194--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>c</mi></mrow><mrow><mi>i</mi></mrow></msub></math> (1.0 by default) the
positive integer exponent <!--l. 4195--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>n</mi></mrow><mrow><mi>i</mi></mrow></msub></math>
(1 by default).
</p><!--l. 4197--><p class="indent">   Any set of components can be combined within a colvar, provided that they return the same type of
values (scalar, unit vector, vector, or quaternion). By default, the colvar is the sum of its components.
Linear or polynomial combinations (following equation (<a href="#eq:colvar_combination">23<!--tex4ht:ref: eq:colvar_combination --></a>)) can be obtained by setting the following
parameters, which are common to all components: </p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox229"><a id="dx1-102002"></a><span class="ectt-1095">componentCoeff</span></span> — <span class="phvro8t-x-x-109">Coefficient of this component in the colvar</span><br class="newline"/>Default: <span class="ectt-1095">1.0</span> <!--l. 4213--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> decimal,
     context: any component <!--l. 4213--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Defines the coefficient by which this component is multiplied (after being raised to <span class="ectt-1095">componentExp</span>)
     before being added to the sum.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox230"><a id="dx1-102003"></a><span class="ectt-1095">componentExp</span></span> — <span class="phvro8t-x-x-109">Exponent of this component in the colvar</span><br class="newline"/>Default: <span class="ectt-1095">1</span> <!--l. 4226--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> integer,
     context: any component <!--l. 4226--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Defines the power at which the value of this component is raised before being added to the
     sum. When this exponent is different than 1 (non-linear sum), total forces and the Jacobian
     force are not available, making the colvar unsuitable for ABF calculations (eABF remains
     possible).</li></ul>
<!--l. 4229--><p class="indent"> <span class="phvb8t-x-x-109">Example: </span>To define the <span class="phvro8t-x-x-109">average </span>of a colvar across different parts of the system,
simply define within the same colvar block a series of components of the same type
(applied to different atom groups), and assign to each component a <span class="ectt-1095">componentCoeff </span>of
<!--l. 4233--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>1</mn><mo class="MathClass-bin">∕</mo><mi>N</mi></math>.
</p>
<div class="flushright">
<!--l. 4349--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 4349--><p class="noindent"> </p><div class="heading"> <a id="sec:colvar_custom_function"></a>
<h4 class="subsectionHead"><span class="titlemark">4.20    </span> <a id="sec:colvar_custom_function"></a>Custom functions</h4>
<!--l. 4349--><p class="noindent"></p></div>
<!--l. 4349--><p class="indent">   Collective variables may be defined by specifying a custom function of multiple components, i.e. an
analytical expression that is more general than the linear combinations described in <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a>. Such
expression is parsed and calculated by Lepton, the lightweight expression parser written by
Peter Eastman (<a class="url" href="https://simtk.org/projects/lepton"><span class="ectt-1095">https://simtk.org/projects/lepton</span></a>) that produces efficient evaluation
routines for both the expression and its derivatives. Although Lepton is generally available
in most applications and builds where Colvars is included, it is best to check section <a href="#sec:compilation_notes">9<!--tex4ht:ref: sec:compilation_notes --></a> to
confirm.
</p>
<ul class="itemize1">
<li class="itemize"><a id="colvar|customFunction"></a> <span class="colorbox" id="colorbox231"><a id="dx1-103001"></a><span class="ectt-1095">customFunction</span></span> — <span class="phvro8t-x-x-109">Compute colvar as a custom function of its components</span><br class="newline"/><!--l. 4349--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> string,
     context: <span class="ectt-1095">colvar</span> <!--l. 4349--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Mathematical expression to define the colvar as a closed-form function of its colvar components.
     See below for the detailed syntax of Lepton expressions. Multiple mentions of this keyword
     can be used to define a vector variable (as in the example below).
                                                                                         
                                                                                         
     </li>
<li class="itemize"><span class="colorbox" id="colorbox232"><a id="dx1-103002"></a><span class="ectt-1095">customFunctionType</span></span> — <span class="phvro8t-x-x-109">Type of value returned by the custom function</span><br class="newline"/>Default: <span class="ectt-1095">scalar</span> <!--l. 4349--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> string,
     context: <span class="ectt-1095">colvar</span> <!--l. 4349--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>With this flag, the user may specify whether the colvar is a scalar or one of the following
     vector types: <span class="ectt-1095">vector3 </span>(a 3D vector), <span class="ectt-1095">unit_vector3 </span>(a normalized 3D vector), or <span class="ectt-1095">unit_quaternion</span>
     (a normalized quaternion), or <span class="ectt-1095">vector</span>. Note that the scalar and vector cases are not necessary,
     as they are detected automatically.</li></ul>
<!--l. 4349--><p class="indent">   The expression may use the collective variable components as variables, referred to by their
user-defined <span class="ectt-1095">name</span>. Scalar elements of vector components may be accessed by appending a 1-based
index to their <span class="ectt-1095">name</span>, as in the example below. When implementing generic functions of Cartesian
coordinates rather than functions of existing components, the <span class="ectt-1095">cartesian </span>component may be particularly
useful. A scalar-valued custom variable may be manually defined as periodic by providing the keyword
<span class="ectt-1095">period</span>, and the optional keyword <span class="ectt-1095">wrapAround</span>, with the same meaning as in periodic components (see
<a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> for details). A vector variable may be defined by specifying the <span class="ectt-1095">customFunction </span>parameter
several times: each expression defines one scalar element of the vector colvar, as in this
example:</p>
<div class="mdframed" id="mdframed-56">
<!--l. 4349--><p class="noindent"></p><pre>colvar {<br class="newline"/>  name custom<br class="newline"/><br class="newline"/>  # A 2-dimensional vector function of a scalar x and a 3-vector r<br class="newline"/>  customFunction cos(x) * (r1 + r2 + r3)<br class="newline"/>  customFunction sqrt(r1 * r2)<br class="newline"/><br class="newline"/>  distance {<br class="newline"/>    name x<br class="newline"/>    group1 { atomNumbers 1 }<br class="newline"/>    group2 { atomNumbers 50 }<br class="newline"/>  }<br class="newline"/>  distanceVec {<br class="newline"/>    name r<br class="newline"/>    group1 { atomNumbers 10 11 12 }<br class="newline"/>    group2 { atomNumbers  20 21 22 }<br class="newline"/>  }<br class="newline"/>} </pre><!--l. 4349--><p class="noindent"></p>
</div>
<!--l. 4349--><p class="indent">   Numeric constants may be given in either decimal or exponential form (e.g. 3.12e-2). An expression
may be followed by definitions for intermediate values that appear in the expression, separated by
semicolons. For example, the expression:<br class="newline"/><span class="ectt-1095">a</span><span class="ectt-1095">^2 + a*b + b</span><span class="ectt-1095">^2; a = a1 + a2; b = b1 + b2</span><br class="newline"/>is exactly equivalent to:<br class="newline"/><span class="ectt-1095">(a1 + a2)</span><span class="ectt-1095">^2 + (a1 + a2) * (b1 + b2) + (b1 + b2)</span><span class="ectt-1095">^2</span>.<br class="newline"/>The definition of an intermediate value may itself involve other intermediate values. All uses of a value
must appear <span class="phvro8t-x-x-109">before </span>that value's definition.
</p><!--l. 4349--><p class="indent">   Lepton supports the usual arithmetic operators +, -, *, /, and ^ (power), as well as the following
                                                                                         
                                                                                         
functions:
</p>
<div class="center">
<!--l. 4349--><p class="noindent">
</p>
<div class="tabular"> <table class="tabular" id="TBL-5" rules="groups"><colgroup id="TBL-5-1g"><col id="TBL-5-1"/></colgroup><colgroup id="TBL-5-2g"><col id="TBL-5-2"/></colgroup><tr class="hline"><td><hr/></td><td><hr/></td></tr><tr id="TBL-5-1-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-1-1" style="text-align:left; white-space:nowrap;"> sqrt    </td><td class="td11" id="TBL-5-1-2" style="text-align:left; white-space:nowrap;"> Square root                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    </td>
</tr><tr id="TBL-5-2-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-2-1" style="text-align:left; white-space:nowrap;"> exp      </td><td class="td11" id="TBL-5-2-2" style="text-align:left; white-space:nowrap;"> Exponential                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    </td>
</tr><tr id="TBL-5-3-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-3-1" style="text-align:left; white-space:nowrap;"> log       </td><td class="td11" id="TBL-5-3-2" style="text-align:left; white-space:nowrap;"> Natural logarithm                                                                                                                                                                                                                                                                                                                                                                                                                                                                            </td>
</tr><tr id="TBL-5-4-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-4-1" style="text-align:left; white-space:nowrap;"> erf       </td><td class="td11" id="TBL-5-4-2" style="text-align:left; white-space:nowrap;"> Error function                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  </td></tr><tr id="TBL-5-5-" style="vertical-align:baseline;"><td class="td 11" id="TBL-5-5-1" style="text-align:left; white-space:nowrap;"> erfc     </td><td class="td11" id="TBL-5-5-2" style="text-align:left; white-space:nowrap;"> Complementary error function                                                                                                                                                                                                                                                                                                                                                                                                                                                       </td>
</tr><tr class="hline"><td><hr/></td><td><hr/></td></tr><tr id="TBL-5-6-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-6-1" style="text-align:left; white-space:nowrap;"> sin       </td><td class="td11" id="TBL-5-6-2" style="text-align:left; white-space:nowrap;"> Sine (angle in radians)                                                                                                                                                                                                                                                                                                                                                                                                                                                                   </td>
</tr><tr id="TBL-5-7-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-7-1" style="text-align:left; white-space:nowrap;"> cos      </td><td class="td11" id="TBL-5-7-2" style="text-align:left; white-space:nowrap;"> Cosine (angle in radians)                                                                                                                                                                                                                                                                                                                                                                                                                                                               </td>
</tr><tr id="TBL-5-8-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-8-1" style="text-align:left; white-space:nowrap;"> sec      </td><td class="td11" id="TBL-5-8-2" style="text-align:left; white-space:nowrap;"> Secant (angle in radians)                                                                                                                                                                                                                                                                                                                                                                                                                                                               </td>
</tr><tr id="TBL-5-9-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-9-1" style="text-align:left; white-space:nowrap;"> csc      </td><td class="td11" id="TBL-5-9-2" style="text-align:left; white-space:nowrap;"> Cosecant (angle in radians)                                                                                                                                                                                                                                                                                                                                                                                                                                                           </td>
</tr><tr id="TBL-5-10-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-10-1" style="text-align:left; white-space:nowrap;"> tan      </td><td class="td11" id="TBL-5-10-2" style="text-align:left; white-space:nowrap;"> Tangent (angle in radians)                                                                                                                                                                                                                                                                                                                                                                                                                                                              </td>
</tr><tr id="TBL-5-11-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-11-1" style="text-align:left; white-space:nowrap;"> cot       </td><td class="td11" id="TBL-5-11-2" style="text-align:left; white-space:nowrap;"> Cotangent (angle in radians)                                                                                                                                                                                                                                                                                                                                                                                                                                                          </td>
</tr><tr id="TBL-5-12-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-12-1" style="text-align:left; white-space:nowrap;"> asin     </td><td class="td11" id="TBL-5-12-2" style="text-align:left; white-space:nowrap;"> Inverse sine (in radians)                                                                                                                                                                                                                                                                                                                                                                                                                                                                 </td>
</tr><tr id="TBL-5-13-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-13-1" style="text-align:left; white-space:nowrap;"> acos    </td><td class="td11" id="TBL-5-13-2" style="text-align:left; white-space:nowrap;"> Inverse cosine (in radians)                                                                                                                                                                                                                                                                                                                                                                                                                                                             </td>
</tr><tr id="TBL-5-14-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-14-1" style="text-align:left; white-space:nowrap;"> atan    </td><td class="td11" id="TBL-5-14-2" style="text-align:left; white-space:nowrap;"> Inverse tangent (in radians)                                                                                                                                                                                                                                                                                                                                                                                                                                                            </td>
</tr><tr id="TBL-5-15-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-15-1" style="text-align:left; white-space:nowrap;"> atan2  </td><td class="td11" id="TBL-5-15-2" style="text-align:left; white-space:nowrap;"> Two-argument inverse tangent (in radians)                                                                                                                                                                                                                                                                                                                                                                                                                                    </td>
</tr><tr class="hline"><td><hr/></td><td><hr/></td></tr><tr id="TBL-5-16-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-16-1" style="text-align:left; white-space:nowrap;"> sinh     </td><td class="td11" id="TBL-5-16-2" style="text-align:left; white-space:nowrap;"> Hyperbolic sine                                                                                                                                                                                                                                                                                                                                                                                                                                                                               </td>
</tr><tr id="TBL-5-17-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-17-1" style="text-align:left; white-space:nowrap;"> cosh    </td><td class="td11" id="TBL-5-17-2" style="text-align:left; white-space:nowrap;"> Hyperbolic cosine                                                                                                                                                                                                                                                                                                                                                                                                                                                                           </td>
</tr><tr id="TBL-5-18-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-18-1" style="text-align:left; white-space:nowrap;"> tanh    </td><td class="td11" id="TBL-5-18-2" style="text-align:left; white-space:nowrap;"> Hyperbolic tangent                                                                                                                                                                                                                                                                                                                                                                                                                                                                         </td>
</tr><tr class="hline"><td><hr/></td><td><hr/></td></tr><tr id="TBL-5-19-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-19-1" style="text-align:left; white-space:nowrap;"> abs      </td><td class="td11" id="TBL-5-19-2" style="text-align:left; white-space:nowrap;"> Absolute value                                                                                                                                                                                                                                                                                                                                                                                                                                                                                </td>
</tr><tr id="TBL-5-20-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-20-1" style="text-align:left; white-space:nowrap;"> floor    </td><td class="td11" id="TBL-5-20-2" style="text-align:left; white-space:nowrap;"> Floor                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               </td>
</tr><tr id="TBL-5-21-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-21-1" style="text-align:left; white-space:nowrap;"> ceil      </td><td class="td11" id="TBL-5-21-2" style="text-align:left; white-space:nowrap;"> Ceiling                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            </td>
</tr><tr id="TBL-5-22-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-22-1" style="text-align:left; white-space:nowrap;"> min      </td><td class="td11" id="TBL-5-22-2" style="text-align:left; white-space:nowrap;"> Minimum of two values                                                                                                                                                                                                                                                                                                                                                                                                                                                                   </td>
</tr><tr id="TBL-5-23-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-23-1" style="text-align:left; white-space:nowrap;"> max     </td><td class="td11" id="TBL-5-23-2" style="text-align:left; white-space:nowrap;"> Maximum of two values                                                                                                                                                                                                                                                                                                                                                                                                                                                                  </td>
</tr><tr id="TBL-5-24-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-24-1" style="text-align:left; white-space:nowrap;"> delta    </td><td class="td11" id="TBL-5-24-2" style="text-align:left; white-space:nowrap;"> <!--l. 4349--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="normal"><mi>d</mi><mi>e</mi><mi>l</mi><mi>t</mi><mi>a</mi></mstyle><mo class="MathClass-open">(</mo><mi>x</mi><mo class="MathClass-close">)</mo> <mo class="MathClass-rel">=</mo> <mn>1</mn></math> if <!--l. 4349--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>x</mi> <mo class="MathClass-rel">=</mo> <mn>0</mn></math>, 0 otherwise                                                                                                                                                          </td>
</tr><tr id="TBL-5-25-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-25-1" style="text-align:left; white-space:nowrap;"> step     </td><td class="td11" id="TBL-5-25-2" style="text-align:left; white-space:nowrap;"> <!--l. 4349--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="normal"><mi>s</mi><mi>t</mi><mi>e</mi><mi>p</mi></mstyle><mo class="MathClass-open">(</mo><mi>x</mi><mo class="MathClass-close">)</mo> <mo class="MathClass-rel">=</mo> <mn>0</mn></math> if <!--l. 4349--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>x</mi> <mo class="MathClass-rel">&lt;</mo> <mn>0</mn></math>, 1 if <!--l. 4349--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>x</mi> <mo class="MathClass-rel">&gt;</mo><mo class="MathClass-rel">=</mo> <mn>0</mn></math> </td>
</tr><tr id="TBL-5-26-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-26-1" style="text-align:left; white-space:nowrap;"> select  </td><td class="td11" id="TBL-5-26-2" style="text-align:left; white-space:nowrap;"> <!--l. 4349--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="normal"><mi>s</mi><mi>e</mi><mi>l</mi><mi>e</mi><mi>c</mi><mi>t</mi></mstyle><mo class="MathClass-open">(</mo><mi>x</mi><mo class="MathClass-punc">,</mo><mi>y</mi><mo class="MathClass-punc">,</mo><mi>z</mi><mo class="MathClass-close">)</mo> <mo class="MathClass-rel">=</mo> <mi>z</mi></math> if <!--l. 4349--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>x</mi> <mo class="MathClass-rel">=</mo> <mn>0</mn></math>, <!--l. 4349--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>y</mi></math> otherwise  </td>
</tr><tr class="hline"><td><hr/></td><td><hr/></td></tr><tr id="TBL-5-27-" style="vertical-align:baseline;"><td class="td11" id="TBL-5-27-1" style="text-align:left; white-space:nowrap;"> </td></tr></table>
</div></div>
<div class="flushright">
<!--l. 4353--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 4353--><p class="noindent"> </p><div class="heading"> <a id="sec:colvar_scripted"></a>
<h4 class="subsectionHead"><span class="titlemark">4.21    </span> <a id="sec:colvar_scripted"></a>Scripted functions</h4>
<!--l. 4353--><p class="noindent"></p></div> When scripting is supported (default in NAMD), a colvar may be defined as a scripted function of its
components, rather than a linear or polynomial combination. When implementing generic functions of
Cartesian coordinates rather than functions of existing components, the <span class="ectt-1095">cartesian </span>component may be
particularly useful. A scalar-valued scripted variable may be manually defined as periodic by providing the
keyword <span class="ectt-1095">period</span>, and the optional keyword <span class="ectt-1095">wrapAround</span>, with the same meaning as in periodic
components (see <a href="#sec:cvc_periodic">4.17<!--tex4ht:ref: sec:cvc_periodic --></a> for details).
<!--l. 4364--><p class="indent">   An example of elaborate scripted colvar is given in example 10, in the form of path-based collective
variables as defined by Branduardi et al<span class="cite">[<a href="#XBranduardi2007">12</a>]</span> (Section <a href="#sec:apath_scripted">4.12.3<!--tex4ht:ref: sec:apath_scripted --></a>).
</p>
<ul class="itemize1">
<li class="itemize"><a id="colvar|scriptedFunction"></a> <span class="colorbox" id="colorbox233"><a id="dx1-104001"></a><span class="ectt-1095">scriptedFunction</span></span> — <span class="phvro8t-x-x-109">Compute colvar as a scripted function of its components</span><br class="newline"/><!--l. 4394--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> string,
     context: <span class="ectt-1095">colvar</span> <!--l. 4394--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If this option is specified, the colvar will be computed as a scripted function of the values of its
     components. To that effect, the user should define two Tcl procedures: <span class="ectt-1095">calc_</span><!--l. 4394--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">scriptedFunction</span><!--l. 4394--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>
     and <span class="ectt-1095">calc_</span><!--l. 4394--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">scriptedFunction</span><!--l. 4394--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><span class="ectt-1095">_gradient</span>,
     both accepting as many parameters as the colvar has components. Values of the components
     will be passed to those procedures in the order defined by their sorted <span class="ectt-1095">name </span>strings. Note that
     if all components are of the same type, their default names are sorted in the order in which
     they are defined, so that names need only be specified for combinations of components of
     different types. <span class="ectt-1095">calc_</span><!--l. 4394--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">scriptedFunction</span><!--l. 4394--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>
     should return one value of type <!--l. 4394--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>scriptedFunctionType<!--l. 4394--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>,
     corresponding to the colvar value. <span class="ectt-1095">calc_</span><!--l. 4394--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">scriptedFunction</span><!--l. 4394--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><span class="ectt-1095">_gradient</span>
     should return a Tcl list containing the derivatives of the function with respect to each component.
     If both the function and some of the components are vectors, the gradient is really a Jacobian
     matrix that should be passed as a linear vector in row-major order, i.e. for a function <!--l. 4394--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>f</mi></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-open">(</mo><msub><mrow><mi>x</mi></mrow><mrow><mi>j</mi></mrow></msub><mo class="MathClass-close">)</mo></math>:
     <!--l. 4394--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi class="MathClass-op">∇</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo></mrow><mrow><mi>x</mi></mrow></msub><msub><mrow><mi>f</mi></mrow><mrow><mn>1</mn></mrow></msub><msub><mrow><mi class="MathClass-op">∇</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo></mrow><mrow><mi>x</mi></mrow></msub><msub><mrow><mi>f</mi></mrow><mrow><mn>2</mn></mrow></msub><mo class="MathClass-rel">⋯</mo><mspace class="thinspace" width="0.17em"></mspace></math>.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox234"><a id="dx1-104002"></a><span class="ectt-1095">scriptedFunctionType</span></span> — <span class="phvro8t-x-x-109">Type of value returned by the scripted colvar</span><br class="newline"/>Default: <span class="ectt-1095">scalar</span> <!--l. 4409--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> string,
     context: <span class="ectt-1095">colvar</span> <!--l. 4409--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If a colvar is defined as a scripted function, its type is not constrained by the types of its
     components. With this flag, the user may specify whether the colvar is a scalar or one of
     the following vector types: <span class="ectt-1095">vector3 </span>(a 3D vector), <span class="ectt-1095">unit_vector3 </span>(a normalized 3D vector),
     or <span class="ectt-1095">unit_quaternion </span>(a normalized quaternion), or <span class="ectt-1095">vector </span>(a vector whose size is specified
     by <span class="ectt-1095">scriptedFunctionVectorSize</span>). Non-scalar values should be passed as space-separated
     lists.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox235"><a id="dx1-104003"></a><span class="ectt-1095">scriptedFunctionVectorSize</span></span> — <span class="phvro8t-x-x-109">Dimension of the vector value of a scripted colvar</span><br class="newline"/><!--l. 4419--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     integer, context: <span class="ectt-1095">colvar</span> <!--l. 4419--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This parameter is only valid when <span class="ectt-1095">scriptedFunctionType </span>is set to <span class="ectt-1095">vector</span>. It defines the
     vector length of the colvar value returned by the function.</li></ul>
<div class="flushright">
<!--l. 4425--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 4425--><p class="noindent"> </p><div class="heading"> <a id="sec:colvar_grid_params"></a>
<h4 class="subsectionHead"><span class="titlemark">4.22    </span> <a id="sec:colvar_grid_params"></a>Defining grid parameters for a colvar</h4>
<!--l. 4425--><p class="noindent"></p></div>
<!--l. 4427--><p class="indent">   Many algorithms require the definition of two boundaries and a bin width for each colvar, which are
necessary to compute discrete “states" for a collective variable's otherwise continuous values. The
following keywords define these parameters for a specific variable, and will be used by all bias that refer to
that variable unless otherwise specified. Some of these parameters can be set directly within a bias
configuration by inserting a <a href="#sec:colvarbias_grid"><span class="ectt-1095">grid</span></a> block (<a href="#sec:colvarbias_grid">6.1<!--tex4ht:ref: sec:colvarbias_grid --></a>).
</p>
<ul class="itemize1">
<li class="itemize"><a id="colvar|lowerBoundary"></a> <span class="colorbox" id="colorbox236"><a id="dx1-105001"></a><span class="ectt-1095">lowerBoundary</span></span> — <span class="phvro8t-x-x-109">Lower boundary of the colvar</span><br class="newline"/>Default: natural boundary of the function <!--l. 4445--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> decimal,
     context: <span class="ectt-1095">colvar</span> <!--l. 4445--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Defines  the  lowest  end  of  the  interval  of  “relevant"  values  for  the  variable.  This  number
     can be, for example, a true physical boundary imposed by the choice of function (e.g. the
     <span class="ectt-1095">distance </span>function is always larger than zero): if this is the case, and only one function is
     used to define the variable, the default value of this number is set to the lowest end of the
     range of values of that function, if available (see Section <a href="#sec:cvc_list">4.1<!--tex4ht:ref: sec:cvc_list --></a>). Alternatively, this value may be
     provided by the user, to represent for example the left-most point of a PMF calculation along
     this variable. In the latter case, it is the user's responsibility to either <span class="phvro8t-x-x-109">(a) </span>ensure the variable
     does  not  go  significantly  beyond  the  boundary  (for  example  by  adding  a  <span class="ectt-1095">harmonicWalls</span>
     restraint,  <a href="#sec:colvarbias_harmonic_walls">6.10<!--tex4ht:ref: sec:colvarbias_harmonic_walls --></a>),  or  <span class="phvro8t-x-x-109">(b)  </span>instruct  the  code  that  this  is  a  true  physical  boundary  by  setting
     <a href="#colvar|hardLowerBoundary"><span class="ectt-1095">hardLowerBoundary</span></a>.
     </li>
<li class="itemize"><a id="colvar|upperBoundary"></a> <span class="colorbox" id="colorbox237"><a id="dx1-105002"></a><span class="ectt-1095">upperBoundary</span></span> — <span class="phvro8t-x-x-109">Upper boundary of the colvar</span><br class="newline"/>Default: natural boundary of the function <!--l. 4455--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> decimal,
     context: <span class="ectt-1095">colvar</span> <!--l. 4455--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Similarly to <span class="ectt-1095">lowerBoundary</span>, defines the highest of the “relevant" values of the variable.
     </li>
<li class="itemize"><a id="colvar|width"></a> <span class="colorbox" id="colorbox238"><a id="dx1-105003"></a><span class="ectt-1095">width</span></span> — <span class="phvro8t-x-x-109">grid spacing, or unit of the variable</span><br class="newline"/>Default: 1.0 <!--l. 4471--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     decimal, context: <span class="ectt-1095">colvar</span> <!--l. 4471--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This number defines the width of a discrete “state" for a collective variable, and is used by
     the many biasing methods to achieve different purposes. Histograms (<a href="#sec:colvarbias_histogram">6.13<!--tex4ht:ref: sec:colvarbias_histogram --></a>), ABF (<a href="#sec:colvarbias_abf">6.3<!--tex4ht:ref: sec:colvarbias_abf --></a>) and
     metadynamics (<a href="#sec:colvarbias_meta">6.6<!--tex4ht:ref: sec:colvarbias_meta --></a>) all use this number as the initial choice for the grid spacing along this
     variable. As a typical rule of thumb, <span class="ectt-1095">width </span>should be no larger than the standard deviation
     of the colvar in an unbiased simulation (to characterize a local free-energy minimum with at
     least two points).
     <!--l. 4471--><p class="noindent">Further, many restraints such as harmonic potentials (<a href="#sec:colvarbias_harmonic">6.8<!--tex4ht:ref: sec:colvarbias_harmonic --></a>), harmonic walls (<a href="#sec:colvarbias_harmonic_walls">6.10<!--tex4ht:ref: sec:colvarbias_harmonic_walls --></a>) and linear
     restraints (<a href="#sec:colvarbias_linear">6.11<!--tex4ht:ref: sec:colvarbias_linear --></a>) also use this parameter to define the <span class="phvro8t-x-x-109">expected fluctuations </span>of the colvar,
     allowing to express the force constant in terms of this unit. This is most useful with multi-dimensional
                                                                                         
                                                                                         
     restraints acting on variables that have very different units (for examples, working with Å and
     degrees <!--l. 4471--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></math>
     simultaneously): a single force constant can be used for all, which is converted to the respective
     unit of each variable when forces are applied (the are printed at initialization time.
     </p></li>
<li class="itemize"><a id="colvar|hardLowerBoundary"></a> <span class="colorbox" id="colorbox239"><a id="dx1-105004"></a><span class="ectt-1095">hardLowerBoundary</span></span> — <span class="phvro8t-x-x-109">Whether the lower boundary is the physical lower limit</span><br class="newline"/>Default: provided by the component <!--l. 4483--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">colvar</span> <!--l. 4483--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>When the colvar has a “natural" boundary (for example, a <span class="ectt-1095">distance </span>colvar cannot go below 0)
     this flag is automatically enabled. For more complex variable definitions, or when <a href="#colvar|lowerBoundary"><span class="ectt-1095">lowerBoundary</span></a>
     is provided directly by the user, it may be useful to set this flag explicitly. This option does not
     affect simulation results, but enables some internal optimizations by letting the code know
     that the variable is unable to cross the lower boundary, regardless of whether restraints are
     applied to it.
     </li>
<li class="itemize"><a id="colvar|hardUpperBoundary"></a> <span class="colorbox" id="colorbox240"><a id="dx1-105005"></a><span class="ectt-1095">hardUpperBoundary</span></span> — <span class="phvro8t-x-x-109">Whether the upper boundary is the physical upper limit of the colvar's</span>
<span class="phvro8t-x-x-109">values</span><br class="newline"/>Default: provided by the component <!--l. 4493--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">colvar</span> <!--l. 4493--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Analogous to <span class="ectt-1095">hardLowerBoundary</span>.
     </li>
<li class="itemize"><a id="colvar|expandBoundaries"></a> <span class="colorbox" id="colorbox241"><a id="dx1-105006"></a><span class="ectt-1095">expandBoundaries</span></span> — <span class="phvro8t-x-x-109">Allow to expand the two boundaries if needed</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 4505--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">colvar</span> <!--l. 4505--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If defined, <span class="ectt-1095">lowerBoundary </span>and <span class="ectt-1095">upperBoundary </span>may be automatically expanded to accommodate
     colvar values that do not fit in the initial range. Currently, this option is used by the metadynamics
     bias (<a href="#sec:colvarbias_meta">6.6<!--tex4ht:ref: sec:colvarbias_meta --></a>) to keep all of its hills fully within the grid. Enabling this option does not affect any
     boundaries that are defined as “hard" (see above), or any boundaries that span the full period
     of a periodic colvar.
     </li></ul>
<div class="flushright">
<!--l. 4510--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 4510--><p class="noindent"> </p><div class="heading"> <a id="sec:colvar_traj_output"></a>
<h4 class="subsectionHead"><span class="titlemark">4.23    </span> <a id="sec:colvar_traj_output"></a>Trajectory output</h4>
<!--l. 4510--><p class="noindent"></p></div>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox242"><a id="dx1-106001"></a><span class="ectt-1095">outputValue</span></span> — <span class="phvro8t-x-x-109">Output a trajectory for this colvar</span><br class="newline"/>Default: <span class="ectt-1095">on</span> <!--l. 4523--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">colvar</span> <!--l. 4523--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If <span class="ectt-1095">colvarsTrajFrequency </span>is non-zero, the value of this colvar is written to the trajectory file
     every <span class="ectt-1095">colvarsTrajFrequency </span>steps in the column labeled “<!--l. 4523--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">name</span><!--l. 4523--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>".
     </li>
<li class="itemize"><span class="colorbox" id="colorbox243"><a id="dx1-106002"></a><span class="ectt-1095">outputVelocity</span></span> — <span class="phvro8t-x-x-109">Output a velocity trajectory for this colvar</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 4535--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">colvar</span> <!--l. 4535--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If <span class="ectt-1095">colvarsTrajFrequency </span>is defined, the finite-difference calculated velocity of this colvar are
     written to the trajectory file under the label “<span class="ectt-1095">v_</span><!--l. 4535--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">name</span><!--l. 4535--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>".
     </li>
<li class="itemize"><span class="colorbox" id="colorbox244"><a id="dx1-106003"></a><span class="ectt-1095">outputEnergy</span></span> — <span class="phvro8t-x-x-109">Output an energy trajectory for this colvar</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 4549--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">colvar</span> <!--l. 4549--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option applies only to extended Lagrangian colvars. If <span class="ectt-1095">colvarsTrajFrequency </span>is defined,
     the  kinetic  energy  of  the  extended  degree  and  freedom  and  the  potential  energy  of  the
     restraining spring are are written to the trajectory file under the labels “<span class="ectt-1095">Ek_</span><!--l. 4549--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">name</span><!--l. 4549--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>"
     and “<span class="ectt-1095">Ep_</span><!--l. 4549--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">name</span><!--l. 4549--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>".
     </li>
<li class="itemize"><a id="colvar|outputTotalForce"></a> <span class="colorbox" id="colorbox245"><a id="dx1-106004"></a><span class="ectt-1095">outputTotalForce</span></span> — <span class="phvro8t-x-x-109">Output a total force trajectory for this colvar</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 4571--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">colvar</span> <!--l. 4571--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If <span class="ectt-1095">colvarsTrajFrequency </span>is defined, the total force on this colvar (i.e. the projection of all
     atomic total forces onto this colvar — see equation (<a href="#eq:gradient_vector">28<!--tex4ht:ref: eq:gradient_vector --></a>) in section <a href="#sec:colvarbias_abf">6.3<!--tex4ht:ref: sec:colvarbias_abf --></a>) are written to the
     trajectory file under the label “<span class="ectt-1095">fs_</span><!--l. 4571--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">name</span><!--l. 4571--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>".
     For extended Lagrangian colvars, the “total force" felt by the extended degree of freedom
     is  simply  the  force  from  the  harmonic  spring.  Due  to  design  constraints,  the  total  force
     reported by NAMD to Colvars was computed at the previous simulation step.  <span class="phvb8t-x-x-109">Note: </span>not all
     components support this option. The physical unit for this force is kcal/mol/<span class="phvro8t-x-x-109">(colvar unit)</span>.
     </li>
<li class="itemize"><a id="colvar|outputAppliedForce"></a> <span class="colorbox" id="colorbox246"><a id="dx1-106005"></a><span class="ectt-1095">outputAppliedForce</span></span> — <span class="phvro8t-x-x-109">Output an applied force trajectory for this colvar</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 4589--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">colvar</span> <!--l. 4589--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If <span class="ectt-1095">colvarsTrajFrequency </span>is defined, the total force applied on this colvar by Colvars biases
     are written to the trajectory under the label “<span class="ectt-1095">fa_</span><!--l. 4589--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">name</span><!--l. 4589--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>".
     For extended Lagrangian colvars, this force is actually applied to the extended degree of
     freedom rather than the geometric colvar itself. The physical unit for this force is kcal/mol/<span class="phvro8t-x-x-109">(colvar</span>
<span class="phvro8t-x-x-109">unit)</span>.
     </li></ul>
<div class="flushright">
<!--l. 4594--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 4594--><p class="noindent"> </p><div class="heading"> <a id="sec:colvar_extended"></a>
<h4 class="subsectionHead"><span class="titlemark">4.24    </span> <a id="sec:colvar_extended"></a>Extended Lagrangian</h4>
<!--l. 4594--><p class="noindent"></p></div>
<!--l. 4596--><p class="indent">   The following options enable extended-system dynamics, where a colvar is coupled to an
additional degree of freedom (fictitious particle) by a harmonic spring. This extended coordinate
masks the colvar and replaces it transparently from the perspective of biasing and analysis
methods. Biasing forces are then applied to the extended degree of freedom, and the actual
geometric colvar (function of Cartesian coordinates) only feels the force from the harmonic
spring. This is particularly useful when combined with an <a href="#sec:colvarbias_abf"><span class="ectt-1095">abf</span></a> bias to perform eABF simulations
(<a href="#sec:colvarbias_abf_extended">6.4<!--tex4ht:ref: sec:colvarbias_abf_extended --></a>).
</p><!--l. 4607--><p class="indent">   Note that for some biases (<a href="#sec:colvarbias_harmonic_walls"><span class="ectt-1095">harmonicWalls</span></a>, <a href="#sec:colvarbias_histogram"><span class="ectt-1095">histogram</span></a>), this masking behavior is controlled by the
keyword <a href="#colvarbias|bypassExtendedLagrangian"><span class="ectt-1095">bypassExtendedLagrangian</span></a>. Specifically for <span class="ectt-1095">harmonicWalls</span>, the default behavior is to bypass
extended Lagrangian coordinates and act directly on the actual colvars.
</p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox247"><a id="dx1-107001"></a><span class="ectt-1095">extendedLagrangian</span></span> — <span class="phvro8t-x-x-109">Add extended degree of freedom</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 4630--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">colvar</span> <!--l. 4630--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Adds  a  fictitious  particle  to  be  coupled  to  the  colvar  by  a  harmonic  spring.  The  fictitious
     mass  and  the  force  constant  of  the  coupling  potential  are  derived  from  the  parameters
     <span class="ectt-1095">extendedTimeConstant </span>and <span class="ectt-1095">extendedFluctuation</span>, described below. Biasing forces on the
     colvar are applied to this fictitious particle, rather than to the atoms directly. This implements
     the extended Lagrangian formalism used in some metadynamics simulations <span class="cite">[<a href="#XIannuzzi2003">3</a>]</span>. The energy
     associated with the extended degree of freedom is reported along with bias energies under
     the MISC title in NAMD's energy output.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox248"><a id="dx1-107002"></a><span class="ectt-1095">extendedFluctuation</span></span> — <span class="phvro8t-x-x-109">Standard deviation between the colvar and the fictitious particle</span>
<span class="phvro8t-x-x-109">(colvar unit)</span><br class="newline"/><!--l. 4643--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     decimal, context: <span class="ectt-1095">colvar</span> <!--l. 4643--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Defines the spring stiffness for the <span class="ectt-1095">extendedLagrangian </span>mode, by setting the typical deviation
     between  the  colvar  and  the  extended  degree  of  freedom  due  to  thermal  fluctuation.  The
     spring force constant is calculated internally as <!--l. 4643--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>k</mi></mrow><mrow><mi>B</mi></mrow></msub><mi>T</mi> <mo class="MathClass-bin">∕</mo><msup><mrow><mi>σ</mi></mrow><mrow><mn>2</mn></mrow></msup></math>,
     where <!--l. 4643--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>σ</mi></math>
     is the value of <span class="ectt-1095">extendedFluctuation</span>.
                                                                                         
                                                                                         
     </li>
<li class="itemize"><span class="colorbox" id="colorbox249"><a id="dx1-107003"></a><span class="ectt-1095">extendedTimeConstant</span></span> — <span class="phvro8t-x-x-109">Oscillation period of the fictitious particle (fs)</span><br class="newline"/>Default: <span class="ectt-1095">200</span> <!--l. 4658--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     decimal, context: <span class="ectt-1095">colvar</span> <!--l. 4658--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Defines  the  inertial  mass  of  the  fictitious  particle,  by  setting  the  oscillation  period  of  the
     harmonic oscillator formed by the fictitious particle and the spring. The period should be
     much larger than the MD time step to ensure accurate integration of the extended particle's
     equation of motion. The fictitious mass is calculated internally as <!--l. 4658--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>k</mi></mrow><mrow><mi>B</mi></mrow></msub><mi>T</mi> <msup><mrow><mo class="MathClass-open">(</mo><mi>τ</mi><mo class="MathClass-bin">∕</mo><mn>2</mn><mi>π</mi><mi>σ</mi><mo class="MathClass-close">)</mo></mrow><mrow><mn>2</mn></mrow></msup></math>,
     where <!--l. 4658--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>τ</mi></math>
     is the period and <!--l. 4658--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>σ</mi></math>
     is the typical fluctuation (see above).
     </li>
<li class="itemize"><a id="colvar|extendedMass"></a> <span class="colorbox" id="colorbox250"><a id="dx1-107004"></a><span class="ectt-1095">extendedMass</span></span> — <span class="phvro8t-x-x-109">Fictitious mass of the alchemical particle ([E]</span><!--l. 4673--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-bin">×</mo></math><span class="phvro8t-x-x-109">fs</span><!--l. 4673--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow></mrow><mrow><mn>2</mn></mrow></msup></math><span class="phvro8t-x-x-109">)</span><br class="newline"/><!--l. 4673--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     decimal, context: <span class="ectt-1095">colvar</span> <!--l. 4673--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This parameter specifies the fictitious mass in the case of an alchemical variable used in
     lambda-dynamics (<a href="#colvar|alchLambda"><span class="ectt-1095">alchLambda</span></a>). Note that the dimension is not that of a usual mass: its unit
     is the energy unit (<a href="#sec:units">3.1<!--tex4ht:ref: sec:units --></a>) times fs<!--l. 4673--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow></mrow><mrow><mn>2</mn></mrow></msup></math>.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox251"><a id="dx1-107005"></a><span class="ectt-1095">extendedTemp</span></span> — <span class="phvro8t-x-x-109">Temperature for the extended degree of freedom (K)</span><br class="newline"/>Default: thermostat temperature <!--l. 4686--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     decimal, context: <span class="ectt-1095">colvar</span> <!--l. 4686--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Temperature used for calculating the coupling force constant of the extended variable (see
     <span class="ectt-1095">extendedFluctuation</span>) and, if needed, as a target temperature for extended Langevin dynamics
     (see <span class="ectt-1095">extendedLangevinDamping</span>). This should normally be left at its default value.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox252"><a id="dx1-107006"></a><span class="ectt-1095">extendedLangevinDamping</span></span> — <span class="phvro8t-x-x-109">Damping factor for extended Langevin dynamics (ps</span><!--l. 4703--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow></mrow><mrow><mo class="MathClass-bin">−</mo><mn>1</mn></mrow></msup></math><span class="phvro8t-x-x-109">)</span><br class="newline"/>Default: <span class="ectt-1095">1.0</span> <!--l. 4703--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     decimal, context: <span class="ectt-1095">colvar</span> <!--l. 4703--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If this is non-zero, the extended degree of freedom undergoes Langevin dynamics at temperature
     <span class="ectt-1095">extendedTemp</span>. The friction force is minus <span class="ectt-1095">extendedLangevinDamping </span>times the velocity. This
     is useful because the extended dynamics coordinate may heat up in the transient non-equilibrium
     regime of ABF. Use moderate damping values, to limit viscous friction (potentially slowing
     down diffusive sampling) and stochastic noise (increasing the variance of statistical measurements).
     In doubt, use the default value.
     </li>
<li class="itemize"><a id="colvar|reflectingLowerBoundary"></a> <span class="colorbox" id="colorbox253"><a id="dx1-107007"></a><span class="ectt-1095">reflectingLowerBoundary</span></span> — <span class="phvro8t-x-x-109">Whether the lower boundary reflects the extended Lagrangian</span>
<span class="phvro8t-x-x-109">particle</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 4714--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">colvar</span> <!--l. 4714--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This turns the specified <a href="#colvar|lowerBoundary"><span class="ectt-1095">lowerBoundary</span></a> into a reflecting wall for the extended particle: upon
     collision, the particle is reflected with opposite momentum.
     </li>
<li class="itemize"><a id="colvar|reflectingUpperBoundary"></a> <span class="colorbox" id="colorbox254"><a id="dx1-107008"></a><span class="ectt-1095">reflectingUpperBoundary</span></span> — <span class="phvro8t-x-x-109">Whether the upper boundary reflects the extended Lagrangian</span>
<span class="phvro8t-x-x-109">particle</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 4725--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">colvar</span> <!--l. 4725--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This turns the specified <a href="#colvar|upperBoundary"><span class="ectt-1095">upperBoundary</span></a> into a reflecting wall for the extended particle: upon
     collision, the particle is reflected with opposite momentum.</li></ul>
<div class="flushright">
<!--l. 4729--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 4729--><p class="noindent"> </p><div class="heading"> <a id="sec:mts_colvar"></a>
<h4 class="subsectionHead"><span class="titlemark">4.25    </span> <a id="sec:mts_colvar"></a>Multiple time-step variables</h4>
<!--l. 4729--><p class="noindent"></p></div>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox255"><a id="dx1-108001"></a><span class="ectt-1095">timeStepFactor</span></span> — <span class="phvro8t-x-x-109">Compute this colvar once in a certain number of timesteps</span><br class="newline"/>Default: <span class="ectt-1095">1</span> <!--l. 4746--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     integer, context: <span class="ectt-1095">colvar</span> <!--l. 4746--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Instructs  this  colvar  to  activate  at  a  time  interval  equal  to  the  base  (MD)  timestep  times
     <span class="ectt-1095">timeStepFactor</span>.<span class="cite">[<a href="#XFerrarotti2015">16</a>]</span> At other time steps, the value of the variable is not updated, and no
     biasing forces are applied. Any forces exerted by biases are accumulated over the given time
     interval, then applied as an impulse at the next update. </li></ul>
<div class="flushright">
<!--l. 4749--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 4749--><p class="noindent"> </p><div class="heading"> <a id="sec:colvar_compatibility"></a>
<h4 class="subsectionHead"><span class="titlemark">4.26    </span> <a id="sec:colvar_compatibility"></a>Backward-compatibility</h4>
<!--l. 4749--><p class="noindent"></p></div>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox256"><a id="dx1-109001"></a><span class="ectt-1095">subtractAppliedForce</span></span> — <span class="phvro8t-x-x-109">Do not include biasing forces in the total force for this colvar</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 4765--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">colvar</span> <!--l. 4765--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If  the  colvar  supports  total  force  calculation  (see  <a href="#sec:cvc_sys_forces">4.18.1<!--tex4ht:ref: sec:cvc_sys_forces --></a>),  all  forces  applied  to  this  colvar
     by  biases  will  be  removed  from  the  total  force.  This  keyword  allows  to  recover  some  of
     the “system force" calculation available in the Colvars module before version 2016-08-10.
     Please note that removal of <span class="phvro8t-x-x-109">all  </span>other external forces (including biasing forces applied to a
     different colvar) is <span class="phvro8t-x-x-109">no longer supported</span>, due to changes in the underlying simulation engines
     (primarily NAMD). This option may be useful when continuing a previous simulation where
     the removal of external/applied forces is essential. <span class="phvro8t-x-x-109">For all new simulations, the use of this</span>
<span class="phvro8t-x-x-109">option is not recommended.</span>
</li></ul>
<div class="flushright">
<!--l. 4770--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 4770--><p class="noindent"> </p><div class="heading"> <a id="sec:colvar_acf"></a>
<h4 class="subsectionHead"><span class="titlemark">4.27    </span> <a id="sec:colvar_acf"></a>Statistical analysis</h4>
<!--l. 4770--><p class="noindent"></p></div>
<!--l. 4772--><p class="indent">   Run-time calculations of statistical properties that depend explicitly on time can be performed for
individual collective variables. Currently, several types of time correlation functions, running averages and
running standard deviations are implemented. For run-time computation of histograms, please see the
histogram bias (<a href="#sec:colvarbias_histogram">6.13<!--tex4ht:ref: sec:colvarbias_histogram --></a>).
</p>
<ul class="itemize1">
<li class="itemize"><a id="colvar|corrFunc"></a> <span class="colorbox" id="colorbox257"><a id="dx1-110001"></a><span class="ectt-1095">corrFunc</span></span> — <span class="phvro8t-x-x-109">Calculate a time correlation function?</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 4787--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">colvar</span> <!--l. 4787--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Whether or not a time correlaction function should be calculated for this colvar.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox258"><a id="dx1-110002"></a><span class="ectt-1095">corrFuncWithColvar</span></span> — <span class="phvro8t-x-x-109">Colvar name for the correlation function</span><br class="newline"/><!--l. 4799--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> string,
     context: <span class="ectt-1095">colvar</span> <!--l. 4799--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>By default, the auto-correlation function (ACF) of this colvar, <!--l. 4799--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></math>,
     is calculated. When this option is specified, the correlation function is calculated instead with
     another colvar, <!--l. 4799--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>j</mi></mrow></msub></math>,
     which must be of the same type (scalar, vector, or quaternion) as <!--l. 4799--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></math>.
                                                                                         
                                                                                         
     </li>
<li class="itemize"><span class="colorbox" id="colorbox259"><a id="dx1-110003"></a><span class="ectt-1095">corrFuncType</span></span> — <span class="phvro8t-x-x-109">Type of the correlation function</span><br class="newline"/>Default: <span class="ectt-1095">velocity</span> <!--l. 4817--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> <span class="ectt-1095">velocity</span>,
     <span class="ectt-1095">coordinate </span>or <span class="ectt-1095">coordinate_p2</span>, context: <span class="ectt-1095">colvar</span> <!--l. 4817--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>With <span class="ectt-1095">coordinate </span>or <span class="ectt-1095">velocity</span>, the correlation function <!--l. 4817--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>C</mi></mrow><mrow><mi>i</mi><mo class="MathClass-punc">,</mo><mi>j</mi></mrow></msub><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo></math> =
     <!--l. 4817--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mrow><mo fence="true" form="prefix"> ⟨</mo><mrow><mi mathvariant="normal">Π</mi> <mrow><mo fence="true" form="prefix"> (</mo><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-open">(</mo><msub><mrow><mi>t</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-close">)</mo><mo class="MathClass-punc">,</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>j</mi></mrow></msub><mo class="MathClass-open">(</mo><msub><mrow><mi>t</mi></mrow><mrow><mn>0</mn></mrow></msub> <mo class="MathClass-bin">+</mo><mi>t</mi><mo class="MathClass-close">)</mo></mrow><mo fence="true" form="postfix">)</mo></mrow></mrow><mo fence="true" form="postfix">⟩</mo></mrow></math>
     is calculated between the variables <!--l. 4817--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></math>
     and <!--l. 4817--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>j</mi></mrow></msub></math>,
     or their velocities. <!--l. 4817--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="normal">Π</mi><mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>j</mi></mrow></msub><mo class="MathClass-close">)</mo></math>
     is the scalar product when calculated between scalar or vector values, whereas for quaternions
     it  is  the  cosine  between  the  two  corresponding  rotation  axes.  With  <span class="ectt-1095">coordinate_p2</span>,  the
     second order Legendre polynomial, <!--l. 4817--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">(</mo><mn>3</mn><mi class="qopname">cos</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--><msup><mrow><mo class="MathClass-open">(</mo><mi>𝜃</mi><mo class="MathClass-close">)</mo></mrow><mrow><mn>2</mn></mrow></msup> <mo class="MathClass-bin">−</mo><mn>1</mn><mo class="MathClass-close">)</mo><mo class="MathClass-bin">∕</mo><mn>2</mn></math>,
     is used instead of the cosine.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox260"><a id="dx1-110004"></a><span class="ectt-1095">corrFuncNormalize</span></span> — <span class="phvro8t-x-x-109">Normalize the time correlation function?</span><br class="newline"/>Default: <span class="ectt-1095">on</span> <!--l. 4828--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">colvar</span> <!--l. 4828--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If enabled, the value of the correlation function at <!--l. 4828--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>t</mi></math> =
     0 is normalized to 1; otherwise, it equals to <!--l. 4828--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mrow><mo fence="true" form="prefix"> ⟨</mo><mrow><mi>O</mi> <mrow><mo fence="true" form="prefix"> (</mo><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>j</mi></mrow></msub></mrow><mo fence="true" form="postfix">)</mo></mrow></mrow><mo fence="true" form="postfix">⟩</mo></mrow></math>.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox261"><a id="dx1-110005"></a><span class="ectt-1095">corrFuncLength</span></span> — <span class="phvro8t-x-x-109">Length of the time correlation function</span><br class="newline"/>Default: <span class="ectt-1095">1000</span> <!--l. 4837--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     integer, context: <span class="ectt-1095">colvar</span> <!--l. 4837--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Length (in number of points) of the time correlation function.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox262"><a id="dx1-110006"></a><span class="ectt-1095">corrFuncStride</span></span> — <span class="phvro8t-x-x-109">Stride of the time correlation function</span><br class="newline"/>Default: <span class="ectt-1095">1</span> <!--l. 4846--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     integer, context: <span class="ectt-1095">colvar</span> <!--l. 4846--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Number of steps between two values of the time correlation function.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox263"><a id="dx1-110007"></a><span class="ectt-1095">corrFuncOffset</span></span> — <span class="phvro8t-x-x-109">Offset of the time correlation function</span><br class="newline"/>Default: <span class="ectt-1095">0</span> <!--l. 4857--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     integer, context: <span class="ectt-1095">colvar</span> <!--l. 4857--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>The starting time (in number of steps) of the time correlation function (default: <!--l. 4857--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>t</mi></math> =
     0). <span class="phvb8t-x-x-109">Note: </span><span class="phvro8t-x-x-109">the value at </span><!--l. 4857--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>t</mi></math><span class="phvro8t-x-x-109"> =</span>
<span class="phvro8t-x-x-109">0 is always used for the normalization</span>.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox264"><a id="dx1-110008"></a><span class="ectt-1095">corrFuncOutputFile</span></span> — <span class="phvro8t-x-x-109">Output file for the time correlation function</span><br class="newline"/>Default: <span class="phvro8t-x-x-109">outputName</span><span class="ectt-1095">.</span><!--l. 4866--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">name</span><!--l. 4866--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><span class="ectt-1095">.corrfunc.dat</span> <!--l. 4866--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> UNIX
     filename, context: <span class="ectt-1095">colvar</span> <!--l. 4866--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>The time correlation function is saved in this file.
     </li>
<li class="itemize"><a id="colvar|runAve"></a> <span class="colorbox" id="colorbox265"><a id="dx1-110009"></a><span class="ectt-1095">runAve</span></span> — <span class="phvro8t-x-x-109">Calculate the running average and standard deviation</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 4877--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">colvar</span> <!--l. 4877--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Whether or not the running average and standard deviation should be calculated for this
     colvar.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox266"><a id="dx1-110010"></a><span class="ectt-1095">runAveLength</span></span> — <span class="phvro8t-x-x-109">Length of the running average window</span><br class="newline"/>Default: <span class="ectt-1095">1000</span> <!--l. 4886--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     integer, context: <span class="ectt-1095">colvar</span> <!--l. 4886--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Length (in number of points) of the running average window.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox267"><a id="dx1-110011"></a><span class="ectt-1095">runAveStride</span></span> — <span class="phvro8t-x-x-109">Stride of the running average window values</span><br class="newline"/>Default: <span class="ectt-1095">1</span> <!--l. 4895--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     integer, context: <span class="ectt-1095">colvar</span> <!--l. 4895--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Number of steps between two values within the running average window.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox268"><a id="dx1-110012"></a><span class="ectt-1095">runAveOutputFile</span></span> — <span class="phvro8t-x-x-109">Output file for the running average and standard deviation</span><br class="newline"/>Default: <span class="phvro8t-x-x-109">outputName</span><span class="ectt-1095">.</span><!--l. 4904--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">name</span><!--l. 4904--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><span class="ectt-1095">.runave.traj</span> <!--l. 4904--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> UNIX
     filename, context: <span class="ectt-1095">colvar</span> <!--l. 4904--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>The running average and standard deviation are saved in this file.
     </li></ul>
<!--l. 4909--><p class="indent">
</p>
<div class="flushright">
<!--l. 4909--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 4909--><p class="noindent"> </p><div class="heading"> <a id="sec:colvar_atom_groups"></a>
<h3 class="sectionHead"><span class="titlemark">5    </span> <a id="sec:colvar_atom_groups"></a>Selecting atoms</h3>
<!--l. 4909--><p class="noindent"></p></div>
<!--l. 4911--><p class="indent">   To define collective variables, atoms are usually selected as groups. Each group is defined using an
identifying keyword that is unique in the context of the specific colvar component (e.g. for a distance
component, the two groups are identified by the <span class="ectt-1095">group1 </span>and <span class="ectt-1095">group2 </span>keywords).
</p><!--l. 4913--><p class="indent">   The group's identifying keyword is followed by a brace-delimited block containing selection keywords
and other parameters, one of which is <span class="ectt-1095">name</span>: </p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox269"><a id="dx1-111001"></a><span class="ectt-1095">name</span></span> — <span class="phvro8t-x-x-109">Unique name for the atom group</span><br class="newline"/><!--l. 4925--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> string,
     context: atom group <!--l. 4925--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This parameter defines a globally unique name for this atom group, which can be referred to
     in the definition of other atom groups (including in other colvars) by invoking <span class="ectt-1095">atomsOfGroup</span>
     as  a  selection  keyword.  This  has  two  benefits:  (1)  it  can  make  the  configuration  shorter
     and more legible, and (2) several related atom groups can be defined using a common set
     of atoms, so that this selection can be changed in one place, keeping all related groups
     synchronized with the same atom set.</li></ul>
<!--l. 4927--><p class="noindent">Other keywords are documented in the following sections.
</p><!--l. 4929--><p class="indent">   In the example below, the <span class="ectt-1095">gyration </span>component uses the identifying keyword <span class="ectt-1095">atoms </span>to define its
associated group, which is defined based on the index group named “<span class="ectt-1095">Protein-H</span>". Optionally, the group is
also given the unique name “<span class="ectt-1095">my_protein</span>", so that atom groups defined later in the Colvars configuration
may refer to it.</p>
<div class="mdframed" id="mdframed-57">
<!--l. 4930--><p class="noindent"></p><pre>colvar {<br class="newline"/>  name rgyr<br class="newline"/>  gyration {<br class="newline"/>    atoms {<br class="newline"/>      name my_protein<br class="newline"/>      indexGroup Protein-H<br class="newline"/>    }<br class="newline"/>  }<br class="newline"/>} </pre><!--l. 4940--><p class="noindent"></p>
</div>
<div class="flushright">
<!--l. 4942--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 4942--><p class="noindent"> </p><div class="heading"> <a id="sec:colvar_atom_groups_sel"></a>
<h4 class="subsectionHead"><span class="titlemark">5.1    </span> <a id="sec:colvar_atom_groups_sel"></a>Atom selection keywords</h4>
<!--l. 4942--><p class="noindent"></p></div>
<!--l. 4944--><p class="indent">   Selection keywords may be used individually or in combination with each other, and each can be
repeated any number of times. Selection is incremental: each keyword adds the corresponding
atoms to the selection, so that different sets of atoms can be combined. However, atoms
included by multiple keywords are only counted once. Below is an example configuration
for an atom group called “<span class="ectt-1095">atoms</span>". <span class="phvb8t-x-x-109">Note: </span><span class="phvro8t-x-x-109">this is an unusually varied combination of selection</span>
<span class="phvro8t-x-x-109">keywords, demonstrating how they can be combined together: most simulations only use one of</span>
<span class="phvro8t-x-x-109">them.</span><br class="newline"/></p>
<div class="mdframed" id="mdframed-58">
<!--l. 4950--><p class="noindent"></p><pre>atoms {<br class="newline"/><br class="newline"/>  # add atoms 1 and 3 to this group (note: first atom in the system is 1)<br class="newline"/>  atomNumbers { <br class="newline"/>    1 3<br class="newline"/>  }<br class="newline"/><br class="newline"/>  # add atoms starting from 20 up to and including 50<br class="newline"/>  atomNumbersRange  20-50<br class="newline"/>  # add all the atoms with occupancy 2 in the file atoms.pdb<br class="newline"/>  atomsFile             atoms.pdb<br class="newline"/>  atomsCol              O<br class="newline"/>  atomsColValue         2.0<br class="newline"/><br class="newline"/>  # add all the C-alphas within residues 11 to 20 of segments "PR1" and "PR2"<br class="newline"/>  psfSegID              PR1 PR2<br class="newline"/>  atomNameResidueRange  CA 11-20<br class="newline"/>  atomNameResidueRange  CA 11-20<br class="newline"/>  # add index group (requires a .ndx file to be provided globally)<br class="newline"/>  indexGroup Water<br class="newline"/>} </pre><!--l. 4974--><p class="noindent"></p>
</div>
<!--l. 4977--><p class="indent">   The resulting selection includes atoms 1 and 3, those between 20 and 50, the
<!--l. 4977--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="normal"><mi>C</mi></mstyle></mrow><mrow><mi>α</mi></mrow></msub></math> atoms
between residues 11 and 20 of the two segments <span class="ectt-1095">PR1 </span>and <span class="ectt-1095">PR2</span>, and those in the index group
called “Water". The indices of this group are read from the file provided by the global keyword
<a href="#Colvars-global|indexFile"><span class="ectt-1095">indexFile</span></a>.
</p><!--l. 4981--><p class="indent">   The complete list of selection keywords available in NAMD is:
</p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox270"><a id="dx1-112001"></a><span class="ectt-1095">atomNumbers</span></span> — <span class="phvro8t-x-x-109">List of atom numbers</span><br class="newline"/><!--l. 4993--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> space-separated
     list of positive integers, context: atom group <!--l. 4993--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option adds to the group all the atoms whose numbers are in the list. <span class="phvro8t-x-x-109">The number of the</span>
<span class="phvro8t-x-x-109">first atom in the system is 1: to convert from a VMD selection, use “atomselect get serial".</span>
</li>
<li class="itemize"><span class="colorbox" id="colorbox271"><a id="dx1-112002"></a><span class="ectt-1095">indexGroup</span></span> — <span class="phvro8t-x-x-109">Name of index group to be used (GROMACS format)</span><br class="newline"/><!--l. 5001--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> string,
     context: atom group <!--l. 5001--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If the name of an index file has been provided by <span class="ectt-1095">indexFile</span>, this option allows to select one
     index group from that file: the atoms from that index group will be used to define the current
     group.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox272"><a id="dx1-112003"></a><span class="ectt-1095">atomsOfGroup</span></span> — <span class="phvro8t-x-x-109">Name of group defined previously</span><br class="newline"/><!--l. 5010--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> string,
     context: atom group <!--l. 5010--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Refers to a group defined previously using its user-defined <span class="ectt-1095">name</span>. This adds all atoms of that
     named group to the current group.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox273"><a id="dx1-112004"></a><span class="ectt-1095">atomNumbersRange</span></span> — <span class="phvro8t-x-x-109">Atoms within a number range</span><br class="newline"/><!--l. 5019--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> <!--l. 5019--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>Starting
     number<!--l. 5019--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>-<!--l. 5019--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>Ending
     number<!--l. 5019--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>,
     context: atom group <!--l. 5019--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option includes in the group all atoms whose numbers are within the range specified.
     <span class="phvro8t-x-x-109">The number of the first atom in the system is 1.</span>
</li>
<li class="itemize"><span class="colorbox" id="colorbox274"><a id="dx1-112005"></a><span class="ectt-1095">atomNameResidueRange</span></span> — <span class="phvro8t-x-x-109">Named atoms within a range of residue numbers</span><br class="newline"/><!--l. 5074--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> <!--l. 5074--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>Atom
     name<!--l. 5074--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>
<!--l. 5074--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>Starting
     residue<!--l. 5074--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>-<!--l. 5074--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>Ending
     residue<!--l. 5074--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>,
     context: atom group <!--l. 5074--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option adds to the group all the atoms with the provided name, within residues in the
     given range.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox275"><a id="dx1-112006"></a><span class="ectt-1095">psfSegID</span></span> — <span class="phvro8t-x-x-109">PSF segment identifier</span><br class="newline"/><!--l. 5074--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> space-separated
     list of strings (max 4 characters), context: atom group <!--l. 5074--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option sets the PSF segment identifier for <span class="ectt-1095">atomNameResidueRange</span>. Multiple values may
     be provided, which correspond to multiple instances of <span class="ectt-1095">atomNameResidueRange</span>, in order of
     their occurrence. This option is only necessary if a PSF topology file is used.
     </li>
<li class="itemize"><a id="atom-group|atomsFile"></a><span class="colorbox" id="colorbox276"><a id="dx1-112007"></a><span class="ectt-1095">atomsFile</span></span> — <span class="phvro8t-x-x-109">PDB file name for atom selection</span><br class="newline"/><!--l. 5074--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> UNIX
     filename, context: atom group <!--l. 5074--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option selects atoms from the PDB file provided and adds them to the group according
     to numerical flags in the column <span class="ectt-1095">atomsCol</span>. <span class="phvb8t-x-x-109">Note: </span><span class="phvro8t-x-x-109">the sequence of atoms in the PDB file</span>
<span class="phvro8t-x-x-109">provided must match that in the system's topology</span>.
     </li>
<li class="itemize"><a id="atom-group|atomsCol"></a><span class="colorbox" id="colorbox277"><a id="dx1-112008"></a><span class="ectt-1095">atomsCol</span></span> — <span class="phvro8t-x-x-109">PDB column to use for atom selection flags</span><br class="newline"/><!--l. 5074--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> <span class="ectt-1095">O</span>,
     <span class="ectt-1095">B</span>, <span class="ectt-1095">X</span>, <span class="ectt-1095">Y</span>, or <span class="ectt-1095">Z</span>, context: atom group <!--l. 5074--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option specifies which PDB column in <span class="ectt-1095">atomsFile </span>is used to determine which atoms are
     to be included in the group.
     </li>
<li class="itemize"><a id="atom-group|atomsColValue"></a><span class="colorbox" id="colorbox278"><a id="dx1-112009"></a><span class="ectt-1095">atomsColValue</span></span> — <span class="phvro8t-x-x-109">Atom selection flag in the PDB column</span><br class="newline"/><!--l. 5074--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     decimal, context: atom group <!--l. 5074--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If defined, this value in <span class="ectt-1095">atomsCol </span>identifies atoms in <span class="ectt-1095">atomsFile </span>that are included in the group.
     If undefined, all atoms with a non-zero value in <span class="ectt-1095">atomsCol </span>are included.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox279"><a id="dx1-112010"></a><span class="ectt-1095">dummyAtom</span></span> — <span class="phvro8t-x-x-109">Dummy atom position (Å)</span><br class="newline"/><!--l. 5082--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> <span class="ectt-1095">(x,</span>
<span class="ectt-1095">y, z) </span>triplet, context: atom group <!--l. 5082--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Instead of selecting any atom, this option makes the group a virtual particle at a fixed position
     in space. This is useful e.g. to replace a group's center of geometry with a user-defined
     position.
     </li></ul>
<div class="flushright">
<!--l. 5086--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 5086--><p class="noindent"> </p><div class="heading"> <a id="sec:colvar_atom_groups_ref_frame"></a>
<h4 class="subsectionHead"><span class="titlemark">5.2    </span> <a id="sec:colvar_atom_groups_ref_frame"></a>Moving frame of reference.</h4>
<!--l. 5086--><p class="noindent"></p></div>
<!--l. 5088--><p class="indent">   The following options define an automatic calculation of an optimal translation (<span class="ectt-1095">centerToReference</span>) or
optimal rotation (<span class="ectt-1095">rotateToReference</span>), that superimposes the positions of this group to a provided set of
reference coordinates. Alternately, <span class="ectt-1095">centerToOrigin </span>applies a translation to place the geometric
center of the group at (0, 0, 0). This can allow, for example, to effectively remove from certain
colvars the effects of molecular tumbling and of diffusion. Given the set of atomic positions
<!--l. 5091--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>i</mi></mrow></msub></math>, the colvar
<!--l. 5091--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math> can be defined on a set of
roto-translated positions <!--l. 5091--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msubsup><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>i</mi></mrow><mrow><mi>′</mi></mrow></msubsup> <mo class="MathClass-rel">=</mo> <mi>R</mi><mo class="MathClass-open">(</mo><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-bin">−</mo><msup><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mstyle mathvariant="normal"><mi>C</mi></mstyle></mrow></msup><mo class="MathClass-close">)</mo><mo class="MathClass-bin">+</mo><msup><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mstyle mathvariant="normal"><mi>r</mi><mi>e</mi><mi>f</mi></mstyle></mrow></msup></math>.
<!--l. 5092--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mstyle mathvariant="normal"><mi>C</mi></mstyle></mrow></msup></math> is the geometric
center of the <!--l. 5092--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>i</mi></mrow></msub></math>,
<!--l. 5092--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>R</mi></math> is the optimal rotation matrix
to the reference positions and <!--l. 5092--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mstyle mathvariant="normal"><mi>r</mi><mi>e</mi><mi>f</mi></mstyle></mrow></msup></math>
is the geometric center of the reference positions.
</p><!--l. 5094--><p class="indent">   Components that are defined based on pairwise distances are naturally invariant under global
roto-translations. Other components are instead affected by global rotations or translations: however, they
can be made invariant if they are expressed in the frame of reference of a chosen group of atoms, using
the <span class="ectt-1095">centerToReference </span>and <span class="ectt-1095">rotateToReference </span>options. Finally, a few components are defined by
convention using a roto-translated frame (e.g. the minimal RMSD): for these components,
<span class="ectt-1095">centerToReference </span>and <span class="ectt-1095">rotateToReference </span>are enabled by default. In typical applications, the default
settings result in the expected behavior.
</p>
<!--l. 5099--><p class="noindent"><span class="likeparagraphHead"><a id="x1-1140005.2"></a>Warning on rotating frames of reference and periodic boundary conditions.</span>
<span class="ectt-1095">rotateToReference </span>affects coordinates that depend on minimum-image distances in periodic
boundary conditions (PBC). After rotation of the coordinates, the periodic cell vectors become irrelevant:
the rotated system is effectively non-periodic. A safe way to handle this is to ensure that the relevant
inter-group distance vectors remain smaller than the half-size of the periodic cell. If this is not desirable,
one should avoid the rotating frame of reference, and apply orientational restraints to the reference group
instead, in order to keep the orientation of the reference group consistent with the orientation of the
periodic cell.
</p>
<!--l. 5105--><p class="noindent"><span class="likeparagraphHead"><a id="x1-1150005.2"></a>Warning on rotating frames of reference and ABF.</span>
   Note that <span class="ectt-1095">centerToReference </span>and <span class="ectt-1095">rotateToReference </span>may affect the Jacobian derivative of colvar
components in a way that is not taken into account by default. Be careful when using these options in ABF
simulations or when using total force values.
</p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox280"><a id="dx1-115001"></a><span class="ectt-1095">centerToReference</span></span> — <span class="phvro8t-x-x-109">Implicitly remove translations for this group</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 5121--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: atom group <!--l. 5121--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If  this  option  is  <span class="ectt-1095">on</span>,  the  center  of  geometry  of  the  group  will  be  aligned  with  that  of  the
     reference positions provided by either <span class="ectt-1095">refPositions </span>or <span class="ectt-1095">refPositionsFile</span>. Colvar components
                                                                                         
                                                                                         
     will only have access to the aligned positions. <span class="phvb8t-x-x-109">Note</span>: unless otherwise specified, <span class="ectt-1095">rmsd </span>and
     <span class="ectt-1095">eigenvector </span>set this option to <span class="ectt-1095">on </span><span class="phvro8t-x-x-109">by default</span>.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox281"><a id="dx1-115002"></a><span class="ectt-1095">centerToOrigin</span></span> — <span class="phvro8t-x-x-109">Implicitly remove translations for this group by keeping its center at the</span>
<span class="phvro8t-x-x-109">origin</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 5134--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: atom group <!--l. 5134--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option implies <span class="ectt-1095">centerToReference</span>. If this option is <span class="ectt-1095">on</span>, coordinates from the group will
     be  translated  so  that  the  center  of  geometry  of  the  group  remains  at  (0,  0,  0),  except  if
     <span class="ectt-1095">fittingGroup </span>is enabled. In that case, the translation applied is the translation that brings
     the center of geometry of the fitting group to (0, 0, 0).
     </li>
<li class="itemize"><span class="colorbox" id="colorbox282"><a id="dx1-115003"></a><span class="ectt-1095">rotateToReference</span></span> — <span class="phvro8t-x-x-109">Implicitly remove rotations for this group</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 5148--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: atom group <!--l. 5148--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If  this  option  is  <span class="ectt-1095">on</span>,  the  coordinates  of  this  group  will  be  optimally  superimposed  to  the
     reference  positions  provided  by  either  <span class="ectt-1095">refPositions </span>or  <span class="ectt-1095">refPositionsFile</span>.  The  rotation
     will  be  performed  around  the  center  of  geometry  if  <span class="ectt-1095">centerToReference </span>is  <span class="ectt-1095">on</span>,  or  around
     the origin otherwise. The algorithm used is the same employed by the <span class="ectt-1095">orientation </span>colvar
     component <span class="cite">[<a href="#XCoutsias2004">4</a>]</span>. Forces applied to the atoms of this group will also be implicitly rotated back
     to the original frame. <span class="phvb8t-x-x-109">Note</span>: unless otherwise specified, <span class="ectt-1095">rmsd </span>and <span class="ectt-1095">eigenvector </span>set this option
     to <span class="ectt-1095">on </span><span class="phvro8t-x-x-109">by default</span>.
     </li>
<li class="itemize"><a id="atom-group|refPositions"></a> <span class="colorbox" id="colorbox283"><a id="dx1-115004"></a><span class="ectt-1095">refPositions</span></span> — <span class="phvro8t-x-x-109">Reference positions for fitting (Å)</span><br class="newline"/><!--l. 5160--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> space-separated
     list of <span class="ectt-1095">(x, y, z) </span>triplets, context: atom group <!--l. 5160--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option provides a list of reference coordinates for <span class="ectt-1095">centerToReference </span>and/or <span class="ectt-1095">rotateToReference</span>,
     and is mutually exclusive with <span class="ectt-1095">refPositionsFile</span>. If only <span class="ectt-1095">centerToReference </span>is <span class="ectt-1095">on</span>, the list
     may contain a single (x, y, z) triplet; if also <span class="ectt-1095">rotateToReference </span>is <span class="ectt-1095">on</span>, the list should be as
     long as the atom group, and <span class="phvro8t-x-x-109">its order must match the order in which atoms were defined</span>.
     </li>
<li class="itemize"><a id="atom-group|refPositionsFile"></a> <span class="colorbox" id="colorbox284"><a id="dx1-115005"></a><span class="ectt-1095">refPositionsFile</span></span> — <span class="phvro8t-x-x-109">File containing the reference positions for fitting</span><br class="newline"/><!--l. 5173--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> UNIX
     filename, context: atom group <!--l. 5173--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option provides a list of reference coordinates for <span class="ectt-1095">centerToReference </span>and/or <span class="ectt-1095">rotateToReference</span>,
     and is mutually exclusive with <span class="ectt-1095">refPositions</span>. The acceptable file format is XYZ (<a href="#sec:colvars_xyz_format">3.8.3<!--tex4ht:ref: sec:colvars_xyz_format --></a>), which
     is read in double precision. Alternatively, a PDB file may be read (see <a href="#sec:colvars_pdb_format">3.8.4<!--tex4ht:ref: sec:colvars_pdb_format --></a>) using NAMD's
     reader; however, due to the constraints of the PDB format <span class="phvro8t-x-x-109">PDB files are discouraged if the</span>
<span class="phvro8t-x-x-109">precision of the reference coordinates is a concern</span>.
     </li>
<li class="itemize"><a id="atom-group|refPositionsCol"></a> <span class="colorbox" id="colorbox285"><a id="dx1-115006"></a><span class="ectt-1095">refPositionsCol</span></span> — <span class="phvro8t-x-x-109">PDB column containing atom flags</span><br class="newline"/><!--l. 5194--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> <span class="ectt-1095">O</span>,
     <span class="ectt-1095">B</span>, <span class="ectt-1095">X</span>, <span class="ectt-1095">Y</span>, or <span class="ectt-1095">Z</span>, context: atom group <!--l. 5194--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Like <span class="ectt-1095">atomsCol </span>for <span class="ectt-1095">atomsFile</span>, indicates which column to use to identify the atoms in <span class="ectt-1095">refPositionsFile</span>
     (if this is a PDB file).
     </li>
<li class="itemize"><a id="atom-group|refPositionsColValue"></a> <span class="colorbox" id="colorbox286"><a id="dx1-115007"></a><span class="ectt-1095">refPositionsColValue</span></span> — <span class="phvro8t-x-x-109">Atom selection flag in the PDB column</span><br class="newline"/><!--l. 5194--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     decimal, context: atom group <!--l. 5194--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Analogous to <span class="ectt-1095">atomsColValue</span>, but applied to <span class="ectt-1095">refPositionsCol</span>.
     </li>
<li class="itemize"><a id="atom-group|fittingGroup"></a> <span class="colorbox" id="colorbox287"><a id="dx1-115008"></a><span class="ectt-1095">fittingGroup</span></span> — <span class="phvro8t-x-x-109">Use an alternate set of atoms to define the roto-translation</span><br class="newline"/>Default: This atom group itself <!--l. 5207--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> <a href="#sec:colvar_atom_groups">Atom
     group</a>, context: atom group <!--l. 5207--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If either <span class="ectt-1095">centerToReference </span>or <span class="ectt-1095">rotateToReference </span>is defined, this keyword defines an alternate
     atom group to calculate the optimal roto-translation. Use this option to define a continuous
     rotation if the structure of the group involved changes significantly (a typical symptom would
     be  the  message  “Warning:  discontinuous  rotation!").  <span class="phvb8t-x-x-109">Performance  considerations:  </span>note
     that enabling this option will result in projecting each of the atomic gradients of the colvar (e.g.
     the RMSD) onto each the gradients of the roto-translation, which may be a computationally
     expensive operation: see the closely related <a href="#atom-group|enableFitGradients"><span class="ectt-1095">enableFitGradients</span></a> for details.
     </li></ul>
<!--l. 5238--><p class="indent">   The following example illustrates the use of <span class="ectt-1095">fittingGroup </span>as part of a Distance to Bound
Configuration (DBC) coordinate for use in ligand restraints for binding affinity calculations.<span class="cite">[<a href="#XSalari2018">17</a>]</span> The group
called “<span class="ectt-1095">atoms</span>" describes coordinates of a ligand's atoms, expressed in a moving frame of reference tied to
a binding site (here within a protein). An optimal roto-translation is calculated automatically by fitting the
C<!--l. 5238--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow></mrow><mrow><mi>α</mi></mrow></msub></math> trace
of the rest of the protein onto the coordinates provided by a PDB file. To define a DBC coordinate, this
atom group would be used within an <a href="#sec:cvc_rmsd"><span class="ectt-1095">rmsd</span></a> function.</p>
<div class="mdframed" id="mdframed-59">
<!--l. 5238--><p class="noindent"></p><pre># Example: defining a group "atoms" (the ligand) whose coordinates are expressed <br class="newline"/># in a roto-translated frame of reference defined by a second group (the receptor)<br class="newline"/>atoms {<br class="newline"/><br class="newline"/>  atomNumbers 1 2 3 4 5 6 7 # atoms of the ligand (1-based)<br class="newline"/><br class="newline"/>  centerToReference yes<br class="newline"/>  rotateToReference yes<br class="newline"/>  fittingGroup {<br class="newline"/>    # define the frame by fitting alpha carbon atoms<br class="newline"/>    # in 2 protein segments close to the site<br class="newline"/>    psfSegID              PROT PROT<br class="newline"/>    atomNameResidueRange  CA 1-40<br class="newline"/>    atomNameResidueRange  CA 59-100<br class="newline"/>  } <br class="newline"/>  refPositionsFile all.pdb  # can be the entire system<br class="newline"/>} </pre><!--l. 5238--><p class="noindent"></p>
</div>
<!--l. 5240--><p class="indent">   The following options have default values appropriate for the vast majority of applications, and are only
provided to support rare, special cases. </p>
<ul class="itemize1">
<li class="itemize"><a id="atom-group|enableFitGradients"></a> <span class="colorbox" id="colorbox288"><a id="dx1-115009"></a><span class="ectt-1095">enableFitGradients</span></span> — <span class="phvro8t-x-x-109">Include the roto-translational contribution to colvar gradients</span><br class="newline"/>Default: <span class="ectt-1095">on</span> <!--l. 5265--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: atom group <!--l. 5265--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>When either <span class="ectt-1095">centerToReference </span>or <span class="ectt-1095">rotateToReference </span>is on, the gradients of some colvars
     include terms proportional to <!--l. 5265--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>∂</mi><mi>R</mi><mo class="MathClass-bin">∕</mo><mi>∂</mi><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>i</mi></mrow></msub></math>
     (rotational gradients) and <!--l. 5265--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>∂</mi><msup><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mstyle mathvariant="normal"><mi>C</mi></mstyle></mrow></msup><mo class="MathClass-bin">∕</mo><mi>∂</mi><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>i</mi></mrow></msub></math>
     (translational gradients). In practice, these terms are calculated as follows,<br class="newline"/><!--l. 5265--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mtable class="aligned"><mtr><mtd columnalign="right"><msub><mrow><mi class="MathClass-op">∇</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo></mrow><mrow><mi>k</mi></mrow></msub><mi>ζ</mi><mo class="MathClass-open">(</mo><mstyle mathvariant="bold"><mi>X</mi></mstyle><mo class="MathClass-close">)</mo></mtd><mtd columnalign="left"> <mo class="MathClass-rel">=</mo><msub><mrow><mi class="MathClass-op"> ∑</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo><!--nolimits--></mrow><mrow><mi>l</mi><mo class="MathClass-punc">,</mo><mi>m</mi><mi>a</mi><mi>i</mi><mi>n</mi></mrow></msub> <mrow><mo fence="true" form="prefix"> [</mo><mrow><msub><mrow><mi class="MathClass-op">∇</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo></mrow><mrow><mi>l</mi></mrow></msub><mi>f</mi> <mo class="MathClass-bin">⋅</mo><msub><mrow><mi class="MathClass-op">∇</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo></mrow><mrow><mi>k</mi></mrow></msub><mi>R</mi> <mrow><mo fence="true" form="prefix"> (</mo><mrow><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>l</mi></mrow></msub><mo class="MathClass-bin">−</mo><msup><mrow><mstyle mathvariant="bold"><mover accent="true"><mrow><mi>x</mi></mrow><mo accent="true">¯</mo></mover></mstyle></mrow><mrow><mo class="MathClass-open">(</mo><mi>f</mi><mi>i</mi><mi>t</mi><mo class="MathClass-close">)</mo></mrow></msup></mrow><mo fence="true" form="postfix">)</mo></mrow> <mo class="MathClass-bin">−</mo> <mfrac><mrow><mn>1</mn></mrow>
<mrow><msup><mrow><mi>n</mi></mrow><mrow><mo class="MathClass-open">(</mo><mi>f</mi><mi>i</mi><mi>t</mi><mo class="MathClass-close">)</mo></mrow></msup></mrow></mfrac><msub><mrow><mi class="MathClass-op">∇</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo></mrow><mrow><mi>l</mi></mrow></msub><mi>f</mi> <mo class="MathClass-bin">⋅</mo><mi>R</mi><mi>1</mi></mrow><mo fence="true" form="postfix">]</mo></mrow></mtd> <mtd columnalign="right"></mtd>
</mtr><mtr><mtd columnalign="right"></mtd> <mtd columnalign="left"> <mo class="MathClass-rel">=</mo> <mrow><mo fence="true" form="prefix"> (</mo><mrow><msub><mrow><mi class="MathClass-op">∑</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo><!--nolimits--></mrow><mrow><mi>l</mi><mo class="MathClass-punc">,</mo><mi>m</mi><mi>a</mi><mi>i</mi><mi>n</mi></mrow></msub><msub><mrow><mi class="MathClass-op">∇</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo></mrow><mrow><mi>l</mi></mrow></msub><mi>f</mi> <mo class="MathClass-bin">⋅</mo> <mfrac><mrow><mi>∂</mi><mi>R</mi></mrow>
<mrow><mi>∂</mi><msub><mrow><mi>q</mi></mrow><mrow><mn>0</mn></mrow></msub></mrow></mfrac> <mrow><mo fence="true" form="prefix"> (</mo><mrow><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>l</mi></mrow></msub><mo class="MathClass-bin">−</mo><msup><mrow><mstyle mathvariant="bold"><mover accent="true"><mrow><mi>x</mi></mrow><mo accent="true">¯</mo></mover></mstyle></mrow><mrow><mo class="MathClass-open">(</mo><mi>f</mi><mi>i</mi><mi>t</mi><mo class="MathClass-close">)</mo></mrow></msup></mrow><mo fence="true" form="postfix">)</mo></mrow><mo class="MathClass-punc">,</mo><msub><mrow><mi class="MathClass-op">∑</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo><!--nolimits--></mrow><mrow>
<mi>l</mi><mo class="MathClass-punc">,</mo><mi>m</mi><mi>a</mi><mi>i</mi><mi>n</mi></mrow></msub><msub><mrow><mi class="MathClass-op">∇</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo></mrow><mrow><mi>l</mi></mrow></msub><mi>f</mi> <mo class="MathClass-bin">⋅</mo> <mfrac><mrow><mi>∂</mi><mi>R</mi></mrow>
<mrow><mi>∂</mi><msub><mrow><mi>q</mi></mrow><mrow><mn>1</mn></mrow></msub></mrow></mfrac> <mrow><mo fence="true" form="prefix"> (</mo><mrow><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>l</mi></mrow></msub><mo class="MathClass-bin">−</mo><msup><mrow><mstyle mathvariant="bold"><mover accent="true"><mrow><mi>x</mi></mrow><mo accent="true">¯</mo></mover></mstyle></mrow><mrow><mo class="MathClass-open">(</mo><mi>f</mi><mi>i</mi><mi>t</mi><mo class="MathClass-close">)</mo></mrow></msup></mrow><mo fence="true" form="postfix">)</mo></mrow><mo class="MathClass-punc">,</mo><msub><mrow><mi class="MathClass-op">∑</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo><!--nolimits--></mrow><mrow>
<mi>l</mi><mo class="MathClass-punc">,</mo><mi>m</mi><mi>a</mi><mi>i</mi><mi>n</mi></mrow></msub><msub><mrow><mi class="MathClass-op">∇</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo></mrow><mrow><mi>l</mi></mrow></msub><mi>f</mi> <mo class="MathClass-bin">⋅</mo> <mfrac><mrow><mi>∂</mi><mi>R</mi></mrow>
<mrow><mi>∂</mi><msub><mrow><mi>q</mi></mrow><mrow><mn>2</mn></mrow></msub></mrow></mfrac> <mrow><mo fence="true" form="prefix"> (</mo><mrow><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>l</mi></mrow></msub><mo class="MathClass-bin">−</mo><msup><mrow><mstyle mathvariant="bold"><mover accent="true"><mrow><mi>x</mi></mrow><mo accent="true">¯</mo></mover></mstyle></mrow><mrow><mo class="MathClass-open">(</mo><mi>f</mi><mi>i</mi><mi>t</mi><mo class="MathClass-close">)</mo></mrow></msup></mrow><mo fence="true" form="postfix">)</mo></mrow><mo class="MathClass-punc">,</mo></mrow><mo fence="true" form="postfix"></mo></mrow></mtd>
</mtr><mtr><mtd columnalign="right"></mtd> <mtd columnalign="left"> <mrow><mo fence="true" form="prefix"> </mo><mrow><msub><mrow><mi class="MathClass-op">∑</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo><!--nolimits--></mrow><mrow><mi>l</mi><mo class="MathClass-punc">,</mo><mi>m</mi><mi>a</mi><mi>i</mi><mi>n</mi></mrow></msub><msub><mrow><mi class="MathClass-op">∇</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo></mrow><mrow><mi>l</mi></mrow></msub><mi>f</mi> <mo class="MathClass-bin">⋅</mo> <mfrac><mrow><mi>∂</mi><mi>R</mi></mrow>
<mrow><mi>∂</mi><msub><mrow><mi>q</mi></mrow><mrow><mn>3</mn></mrow></msub></mrow></mfrac> <mrow><mo fence="true" form="prefix"> (</mo><mrow><msub><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mrow><mi>l</mi></mrow></msub><mo class="MathClass-bin">−</mo><msup><mrow><mstyle mathvariant="bold"><mover accent="true"><mrow><mi>x</mi></mrow><mo accent="true">¯</mo></mover></mstyle></mrow><mrow><mo class="MathClass-open">(</mo><mi>f</mi><mi>i</mi><mi>t</mi><mo class="MathClass-close">)</mo></mrow></msup></mrow><mo fence="true" form="postfix">)</mo></mrow></mrow><mo fence="true" form="postfix">)</mo></mrow> <mo class="MathClass-bin">⋅</mo><mrow><mo fence="true" form="prefix"> (</mo><mrow><msub><mrow><mi class="MathClass-op">∇</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo></mrow><mrow>
<mi>k</mi></mrow></msub><msub><mrow><mi>q</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi class="MathClass-op">∇</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo></mrow><mrow><mi>k</mi></mrow></msub><msub><mrow><mi>q</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi class="MathClass-op">∇</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo></mrow><mrow><mi>k</mi></mrow></msub><msub><mrow><mi>q</mi></mrow><mrow><mn>2</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi class="MathClass-op">∇</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo></mrow><mrow><mi>k</mi></mrow></msub><msub><mrow><mi>q</mi></mrow><mrow><mn>3</mn></mrow></msub></mrow><mo fence="true" form="postfix">)</mo></mrow> <mo class="MathClass-bin">−</mo> <mfrac><mrow><mn>1</mn></mrow>
<mrow><msup><mrow><mi>n</mi></mrow><mrow><mo class="MathClass-open">(</mo><mi>f</mi><mi>i</mi><mi>t</mi><mo class="MathClass-close">)</mo></mrow></msup></mrow></mfrac><msup><mrow><mi>R</mi></mrow><mrow><mo class="MathClass-bin">⊺</mo></mrow></msup><msub><mrow><mi class="MathClass-op">∑</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo><!--nolimits--></mrow><mrow>
<mi>l</mi><mo class="MathClass-punc">,</mo><mi>m</mi><mi>a</mi><mi>i</mi><mi>n</mi></mrow></msub><msub><mrow><mi class="MathClass-op">∇</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo></mrow><mrow><mi>l</mi></mrow></msub><mi>f</mi></mtd>
</mtr> </mtable></math><br class="newline"/>By default, these terms are calculated and included in the total gradients; if this option is
     set to <span class="ectt-1095">off</span>, they are neglected. In the case of a minimum RMSD component, this flag is
     automatically disabled because the contributions of those derivatives to the gradients cancel
     out; other types of variable will require projecting each of the gradients of the variable onto
     each of the gradients of the roto-translation. When <a href="#atom-group|fittingGroup"><span class="ectt-1095">fittingGroup</span></a> is enabled, the computation
     is a <!--l. 5265--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>O</mi><mo class="MathClass-open">(</mo><msup><mrow><mi>n</mi></mrow><mrow><mo class="MathClass-open">(</mo><mi>f</mi><mi>i</mi><mi>t</mi><mo class="MathClass-close">)</mo></mrow></msup> <mo class="MathClass-bin">+</mo><msup><mrow><mi>n</mi></mrow><mrow><mo class="MathClass-open">(</mo><mi>m</mi><mi>a</mi><mi>i</mi><mi>n</mi><mo class="MathClass-close">)</mo></mrow></msup><mo class="MathClass-close">)</mo></math>
     loop for all variables, including RMSDs.
     </li></ul>
<div class="flushright">
<!--l. 5270--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 5270--><p class="noindent"> </p><div class="heading"> <a id="sec:colvar_atom_groups_wrapping"></a>
<h4 class="subsectionHead"><span class="titlemark">5.3    </span> <a id="sec:colvar_atom_groups_wrapping"></a>Treatment of periodic boundary conditions.</h4>
<!--l. 5270--><p class="noindent"></p></div>
<!--l. 5272--><p class="indent">   In simulations with periodic boundary conditions (PBCs), Colvars computes all distances between two
points following the nearest-image convention, using PBC parameters provided by NAMD. However,
many common variables rely on a consistent definition of the center of mass or geometry of a group of
atoms. This requires the use of <span class="phvro8t-x-x-109">unwrapped </span>coordinates, which are not subject to “jumps" when they
diffuse across periodic boundaries.
</p><!--l. 5288--><p class="indent">   When PBCs are defined, NAMD maintains the coordinates of all the atoms within a molecule
contiguous to each other (i.e. there are no spurious “jumps" in the molecular bonds). However, this
condition may fail if the group spans different molecules. In that case, writing the NAMD output and restart
                                                                                         
                                                                                         
files using <span class="ectt-1095">wrapAll </span>or <span class="ectt-1095">wrapWater </span>could produce wrong results when a simulation run is continued from a
previous one. The user should then determine, according to which type of colvars are being calculated,
whether <span class="ectt-1095">wrapAll </span>or <span class="ectt-1095">wrapWater </span>can be enabled while ensuring that the input coordinates for each run
contain all atom groups in one piece.
</p><!--l. 5304--><p class="indent">   In general, internal coordinate wrapping by NAMD does not affect the calculation of colvars if each
atom group satisfies one or more of the following:
     </p><ol class="enumerate1">
<li class="enumerate" id="x1-116001x5.3">it is composed by only one atom;
     </li>
<li class="enumerate" id="x1-116002x5.3">it is used by a colvar component which does not make use of its center of geometry, but only
     of pairwise distances (<span class="ectt-1095">distanceInv</span>, <span class="ectt-1095">coordNum</span>, <span class="ectt-1095">hBond</span>, <span class="ectt-1095">alpha</span>, <span class="ectt-1095">dihedralPC</span>);
     </li>
<li class="enumerate" id="x1-116003x5.3">it is used by a colvar component that ignores the ill-defined Cartesian components of its
     center of mass (such as the <!--l. 5308--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>x</mi></math>
     and <!--l. 5308--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>y</mi></math>
     components of a membrane's center of mass modeled with <span class="ectt-1095">distanceZ</span>);
     </li>
<li class="enumerate" id="x1-116004x5.3">it has all of its atoms within the same molecular fragment.</li></ol>
<div class="flushright">
<!--l. 5315--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 5315--><p class="noindent"> </p><div class="heading"> <a id="sec:colvar_atom_groups_scaling"></a>
<h4 class="subsectionHead"><span class="titlemark">5.4    </span> <a id="sec:colvar_atom_groups_scaling"></a>Performance of a Colvars calculation based on group size.</h4>
<!--l. 5315--><p class="noindent"></p></div>
<!--l. 5317--><p class="indent">   In simulations performed with MD simulation engines such as GROMACS, LAMMPS or
NAMD, the computation of energy and forces is distributed (i.e., parallelized) over multiple
nodes, as well as over the CPU/GPU cores of each node. When Colvars is enabled, atomic
coordinates are collected on a single CPU core, where collective variables and their biases
are computed. This means that in the case of simulations that are already being run over
large numbers of nodes, or inside a GPU, a Colvars calculation may produce a significant
overhead. This overhead comes from the combined cost of two operation: transmitting atomic
coordinates, and computing functions of the same. The latency-tolerant design and dynamic load
balancing of NAMD may alleviate both factors, but a noticeable performance impact may be
observed.
</p><!--l. 5323--><p class="indent">   Performance can be improved in multiple ways:
</p>
<ul class="itemize1">
<li class="itemize">As  a  general  rule,  the  size  of  atom  groups  should  be  kept  relatively  small  (up  to  a  few
     thousands of atoms, depending on the size of the entire system in comparison). For example,
     restraining a protein through a RMSD colvar defined over all of its atoms is only marginally
     different than one defined over only the <!--l. 5328--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>α</mi></math>
     carbon atoms, but the difference in computational cost is much higher. To gain an estimate
     of the computational cost of a specific Colvars configuration, one may use a test calculation
     of the same colvar in VMD (hint: use the <span class="ectt-1095">time </span>Tcl command to measure the cost of running
     <span class="ectt-1095">cv update</span>).
     </li>
<li class="itemize">The calculation of variables, components and biases can be distributed over the processor
     cores  of  the  node  where  the  Colvars  module  is  executed.  Currently,  an  equal  weight  is
     assigned  to  each  colvar,  or  to  each  component  of  those  colvars  that  include  more  than
     one component. The performance of simulations that use many colvars or components is
     improved automatically. For simulations that use a single large colvar, it may be advisable to
     partition it in multiple components, which will be then distributed across the available cores.
     In NAMD, this feature is enabled in all binaries compiled using SMP builds of Charm++ with
     the CkLoop extension (including “multicore" builds).    The messages “SMP parallelism is
     available" or “SMP parallelism is enabled", printed by Colvars at initialization time, indicate
     the availability or status of this feature. If available, the option is turned on by default, but may
     be disabled using the keyword <a href="#Colvars-global|smp"><span class="ectt-1095">smp</span></a> if required for debugging or troubleshooting.
     </li>
<li class="itemize">NAMD also offers a parallelized calculation of the centers of mass of groups of atoms. This
     option is on by default for all components that are simple functions of centers of mass, and
     is controlled by the keyword <a href="#sec:cvc_common"><span class="ectt-1095">scalable</span></a>. When supported, the message “Will enable scalable
     calculation for group …" is printed for each group.
     </li></ul>
<!--l. 5352--><p class="indent">
</p>
<div class="flushright">
<!--l. 5352--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 5352--><p class="noindent"> </p><div class="heading"> <a id="sec:colvarbias"></a>
<h3 class="sectionHead"><span class="titlemark">6    </span> <a id="sec:colvarbias"></a>Biasing and analysis methods</h3>
<!--l. 5352--><p class="noindent"></p></div>
<!--l. 5354--><p class="indent">   A biasing or analysis method can be applied to existing collective variables by using the following
configuration:</p>
<div class="mdframed" id="mdframed-60">
<!--l. 5356--><p class="noindent"></p><pre><!--l. 5357--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>biastype<!--l. 5357--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math> {<br class="newline"/>  name <!--l. 5358--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>name<!--l. 5358--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>
<br class="newline"/>  colvars <!--l. 5359--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>xi1<!--l. 5359--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math> <!--l. 5359--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>xi2<!--l. 5359--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math> ...<br class="newline"/>  <!--l. 5360--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math>parameters<!--l. 5360--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><br class="newline"/>} </pre><!--l. 5362--><p class="noindent"></p>
</div>
<!--l. 5364--><p class="noindent">The keyword <!--l. 5364--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">biastype</span><!--l. 5364--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>
indicates the method of choice. There can be multiple instances of the same method, e.g. using multiple
<span class="ectt-1095">harmonic </span>blocks allows defining multiple restraints.
</p><!--l. 5367--><p class="indent">   All biasing and analysis methods implemented recognize the following options: </p>
<ul class="itemize1">
<li class="itemize"><a id="colvarbias|name"></a> <span class="colorbox" id="colorbox289"><a id="dx1-118001"></a><span class="ectt-1095">name</span></span> — <span class="phvro8t-x-x-109">Identifier for the bias</span><br class="newline"/>Default: <!--l. 5380--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">type</span>
<span class="ectt-1095">of bias</span><!--l. 5380--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo><mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">bias</span>
<span class="ectt-1095">index</span><!--l. 5380--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math> <!--l. 5380--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> string,
     context: colvar bias <!--l. 5380--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This string is used to identify the bias or analysis method in the output, and to name some
     output files. <span class="phvb8t-x-x-109">Tip: </span>although a unique name is assigned automatically, <span class="phvro8t-x-x-109">you are strongly encouraged</span>
<span class="phvro8t-x-x-109">to give a name to a bias that you may want to analyze later</span>. For example, processing a
     trajectory  file  (sec. <a href="#sec:colvars_traj_format">3.8.6<!--tex4ht:ref: sec:colvars_traj_format --></a>)  containing  a  harmonic  restraint  is  much  simpler  with  a  clearly
     identifiable name, like “<span class="ectt-1095">smd</span>" for a moving harmonic restraint or “<span class="ectt-1095">us</span>" for a static one.
     </li>
<li class="itemize"><a id="colvarbias|colvars"></a> <span class="colorbox" id="colorbox290"><a id="dx1-118002"></a><span class="ectt-1095">colvars</span></span> — <span class="phvro8t-x-x-109">Collective variables involved</span><br class="newline"/><!--l. 5389--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> space-separated
     list of colvar names, context: colvar bias <!--l. 5389--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option selects by name all the variables to which this bias or analysis will be applied.
     </li>
<li class="itemize"><a id="colvarbias|outputEnergy"></a> <span class="colorbox" id="colorbox291"><a id="dx1-118003"></a><span class="ectt-1095">outputEnergy</span></span> — <span class="phvro8t-x-x-109">Write the current bias energy to the trajectory file</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 5401--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: colvar bias <!--l. 5401--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If  this  option  is  chosen  and  <span class="ectt-1095">colvarsTrajFrequency </span>is  not  zero,  the  current  value  of  the
     biasing energy will be written to the trajectory file during the simulation. The total energy of
     all Colvars biases is also reported by NAMD, as part of the MISC title.
     </li>
<li class="itemize"><a id="colvarbias|outputFreq"></a> <span class="colorbox" id="colorbox292"><a id="dx1-118004"></a><span class="ectt-1095">outputFreq</span></span> — <span class="phvro8t-x-x-109">Frequency (number of steps) at which output files are written</span><br class="newline"/>Default: <a href="#Colvars-global|colvarsRestartFrequency"><span class="ectt-1095">colvarsRestartFrequency</span></a> <!--l. 5413--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     integer, context: colvar bias <!--l. 5413--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If this bias produces aggregated data that needs to be written to disk (for example, a PMF),
     this number specifies the number of steps after which these data are written to files. A value
     of zero disables writing files for this bias during the simulation (except for <a href="#colvarbias|outputEnergy"><span class="ectt-1095">outputEnergy</span></a>,
     which is controlled by <a href="#Colvars-global|colvarsTrajFrequency"><span class="ectt-1095">colvarsTrajFrequency</span></a>). All output files are also written at the end of
     a simulation run, regardless of the value of this number.
     </li>
<li class="itemize"><a id="colvarbias|bypassExtendedLagrangian"></a> <span class="colorbox" id="colorbox293"><a id="dx1-118005"></a><span class="ectt-1095">bypassExtendedLagrangian</span></span> — <span class="phvro8t-x-x-109">Apply bias to actual colvars, bypassing extended coordinates</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 5428--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: colvar bias <!--l. 5428--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option is implemented by the <a href="#sec:colvarbias_harmonic_walls"><span class="ectt-1095">harmonicWalls</span></a> and <a href="#sec:colvarbias_histogram"><span class="ectt-1095">histogram</span></a> biases. It is only relevant if
     the bias is applied to one or several extended-Lagrangian colvars (<a href="#sec:colvar_extended">4.24<!--tex4ht:ref: sec:colvar_extended --></a>), for example within
     an eABF (<a href="#sec:colvarbias_abf_extended">6.4<!--tex4ht:ref: sec:colvarbias_abf_extended --></a>) simulation. Usually, biases use the value of the extended coordinate as a proxy
     for the actual colvar, and their biasing forces are applied to the extended coordinates as well.
     If <span class="ectt-1095">bypassExtendedLagrangian </span>is enabled, the bias behaves as if there were no extended
     coordinates, and accesses the value of the underlying colvars, applying any biasing forces
     along the gradients of those variables.
     </li>
<li class="itemize"><a id="colvarbias|stepZeroData"></a> <span class="colorbox" id="colorbox294"><a id="dx1-118006"></a><span class="ectt-1095">stepZeroData</span></span> — <span class="phvro8t-x-x-109">Accumulate data starting at step 0 of a simulation run</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 5443--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: colvar bias <!--l. 5443--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option is meaningful for biases that record and accumulate data during a simulation,
     such  as  ABF  (<a href="#sec:colvarbias_abf">6.3<!--tex4ht:ref: sec:colvarbias_abf --></a>),  metadynamics  (<a href="#sec:colvarbias_meta">6.6<!--tex4ht:ref: sec:colvarbias_meta --></a>),  histograms  (<a href="#sec:colvarbias_histogram">6.13<!--tex4ht:ref: sec:colvarbias_histogram --></a>)  and  in  general  any  bias  that
     accumulates free-energy samples with thermodynamic integration, or TI (<a href="#sec:colvarbias_ti">6.2<!--tex4ht:ref: sec:colvarbias_ti --></a>). When this
     option is disabled (default), data will only be recorded into the bias after the first coordinate
     update: this is generally the correct choice in simulation runs. Biasing energy and forces
     will always be computed for all active biases, regardless of this option. Note that in some
     cases the bias may require data from previous simulation steps: for example, TI requires
     total atomic forces (see <a href="#colvar|outputTotalForce"><span class="ectt-1095">outputTotalForce</span></a>) which are only available at the following step in
     NAMD; turning on this flag in those cases will raise an error.
     </li>
<li class="itemize"><a id="colvarbias|scaledBiasingForce"></a> <span class="colorbox" id="colorbox295"><a id="dx1-118007"></a><span class="ectt-1095">scaledBiasingForce</span></span> — <span class="phvro8t-x-x-109">Scale biasing force by a factor in an external histogram</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 5449--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: colvar bias <!--l. 5449--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If this option is set to <span class="ectt-1095">on</span>, the biasing force at each step will be scaled by a factor provided in
     the grid of an external histogram. The histogram file is provided by <span class="ectt-1095">scaledBiasingForceFactorsGrid</span>.
     It ought to be noted that if the variables are not in any grids of the histogram, the scaling
     factor is 1.0.
     </li>
<li class="itemize"><a id="colvarbias|scaledBiasingForceFactorsGrid"></a> <span class="colorbox" id="colorbox296"><a id="dx1-118008"></a><span class="ectt-1095">scaledBiasingForceFactorsGrid</span></span> — <span class="phvro8t-x-x-109">A histogram file with the scaling factor of biasing force</span>
<span class="phvro8t-x-x-109">in each bin</span><br class="newline"/><!--l. 5456--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> string,
     context: colvar bias <!--l. 5456--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If <span class="ectt-1095">scaledBiasingForce </span>is set to <span class="ectt-1095">on</span>, this option accepts the filename of the histogram file
     that contains the scaling factors. The histogram file is expected to be in “multicolumn" format
     (<a href="#sec:colvar_multicolumn_grid">3.8.5<!--tex4ht:ref: sec:colvar_multicolumn_grid --></a>), similar to the low-dimensional the PMF files written by <span class="ectt-1095">metadynamics </span>and <span class="ectt-1095">ABF</span>.
     </li></ul>
<div class="flushright">
<!--l. 5462--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 5462--><p class="noindent"> </p><div class="heading"> <a id="sec:colvarbias_grid"></a>
<h4 class="subsectionHead"><span class="titlemark">6.1    </span> <a id="sec:colvarbias_grid"></a>Defining grids for biasing and analysis methods</h4>
<!--l. 5462--><p class="noindent"></p></div>
<!--l. 5464--><p class="indent">   Several methods (<a href="#sec:colvarbias_histogram">histograms</a>, <a href="#sec:colvarbias_abf">ABF</a> and <a href="#sec:colvarbias_abf_extended">eABF</a>, optionally <a href="#sec:colvarbias_metadynamics">metadynamics</a>) rely on discretizing
colvars on regular grids to represent functions such as probability distributions or free energy
surfaces.
</p><!--l. 5466--><p class="indent">   Grid parameters are specific to each variable, and some of them may even be
defined automatically: for instance, periodic angles have default boundaries of
-180<!--l. 5466--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></math> and
180<!--l. 5466--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></math>.
When not given by default, boundaries may be also specified when defining the collective variable itself
(<a href="#sec:colvar_grid_params">4.22<!--tex4ht:ref: sec:colvar_grid_params --></a>).
</p><!--l. 5469--><p class="indent">   Each bias will use by default any colvar boundaries that are automatically defined or explicitly given.
Optionally, the bias may also contain a dedicated configuration block <span class="ectt-1095">grid { </span><span class="ectt-1095">…} </span>inside its configuration.
The options supported within the <span class="ectt-1095">grid { </span><span class="ectt-1095">…} </span>block are:
</p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox297"><a id="dx1-119001"></a><span class="ectt-1095">lowerBoundary</span></span> — <span class="phvro8t-x-x-109">Lower boundaries of the grid</span><br class="newline"/><!--l. 5482--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> list
     of space-separated decimals, context: <span class="ectt-1095">grid</span> <!--l. 5482--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option is specified once, and takes as many parameters as there are colvars associated
     with this bias. It defines the lower boundaries of the grid, overriding any values defined by
     the <a href="#colvar|lowerBoundary"><span class="ectt-1095">lowerBoundary</span></a> keyword of each colvar.
                                                                                         
                                                                                         
     </li>
<li class="itemize"><span class="colorbox" id="colorbox298"><a id="dx1-119002"></a><span class="ectt-1095">upperBoundary</span></span> — analogous to <span class="ectt-1095">lowerBoundary</span>
</li>
<li class="itemize"><span class="colorbox" id="colorbox299"><a id="dx1-119003"></a><span class="ectt-1095">width</span></span> — <span class="phvro8t-x-x-109">Bin widths of the grid</span><br class="newline"/><!--l. 5493--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> list
     of space-separated decimals, context: <span class="ectt-1095">grid</span> <!--l. 5493--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option is specified once, and takes as many parameters as there are colvars associated
     with  this  bias.  It  defines  the  bin  widths  of  the  grid  along  each  dimension,  overriding  any
     values defined by the <a href="#colvar|width"><span class="ectt-1095">width</span></a> keyword of each colvar. </li></ul>
<!--l. 5496--><p class="noindent"><span class="phvb8t-x-x-109">Example: </span>Definition of custom grid parameters for a histogram. Note how some of the boundaries of the
variables are automatically defined, but this histogram's grid is custom-defined to also specify a finer
width and narrower boundaries.</p>
<div class="mdframed" id="mdframed-61">
<!--l. 5497--><p class="noindent"></p><pre>colvar {<br class="newline"/>  name r<br class="newline"/>  # lowerBoundary 0.0 is automatically defined<br class="newline"/>  upperBoundary 20.0<br class="newline"/>  distance { ... }<br class="newline"/>}<br class="newline"/><br class="newline"/>colvar {<br class="newline"/>  name theta<br class="newline"/>  # lowerBoundary -180.0 is automatically defined<br class="newline"/>  # upperBoundary 180.0 is automatically defined<br class="newline"/>  dihedral { ... }<br class="newline"/>}<br class="newline"/><br class="newline"/>histogram {<br class="newline"/>  name hist2d<br class="newline"/>  colvars r theta<br class="newline"/>  grid {<br class="newline"/>    width  0.1 1.0<br class="newline"/>    lowerBoundary   2.0 30.0<br class="newline"/>    upperBoundary  10.0 90.0<br class="newline"/>  }<br class="newline"/>}<br class="newline"/></pre><!--l. 5521--><p class="noindent"></p>
</div>
<div class="flushright">
<!--l. 5525--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 5525--><p class="noindent"> </p><div class="heading"> <a id="sec:colvarbias_ti"></a>
<h4 class="subsectionHead"><span class="titlemark">6.2    </span> <a id="sec:colvarbias_ti"></a>Thermodynamic integration</h4>
<!--l. 5525--><p class="noindent"></p></div>
<!--l. 5527--><p class="indent">   The methods implemented here provide a variety of estimators of conformational free-energies. These
are carried out at run-time, or with the use of post-processing tools over the generated output files. The
specifics of each estimator are discussed in the documentation of each biasing or analysis
method.
</p><!--l. 5531--><p class="indent">   A special case is the traditional thermodynamic integration (TI) method, used for example to compute
potentials of mean force (PMFs). Most types of restraints (<a href="#sec:colvarbias_harmonic">6.8<!--tex4ht:ref: sec:colvarbias_harmonic --></a>, <a href="#sec:colvarbias_harmonic_walls">6.10<!--tex4ht:ref: sec:colvarbias_harmonic_walls --></a>, <a href="#sec:colvarbias_linear">6.11<!--tex4ht:ref: sec:colvarbias_linear --></a>, ...) as well as metadynamics
(<a href="#sec:colvarbias_meta">6.6<!--tex4ht:ref: sec:colvarbias_meta --></a>) can optionally use TI alongside their own estimator, based on the keywords documented
below.
</p>
<ul class="itemize1">
<li class="itemize"><a id="colvarbias|writeTIPMF"></a> <span class="colorbox" id="colorbox300"><a id="dx1-120001"></a><span class="ectt-1095">writeTIPMF</span></span> — <span class="phvro8t-x-x-109">Write the PMF computed by thermodynamic integration</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 5547--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: colvar bias <!--l. 5547--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If the bias is applied to a variable that supports the calculation of total forces (see <a href="#sec:colvar"><span class="ectt-1095">outputTotalForce</span></a>
     and <a href="#sec:cvc_sys_forces">4.18.1<!--tex4ht:ref: sec:cvc_sys_forces --></a>), this option allows calculating the corresponding PMF by thermodynamic integration,
     and writing it to the file <span class="phvro8t-x-x-109">outputName</span><span class="ectt-1095">.</span><!--l. 5547--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">name</span><!--l. 5547--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><span class="ectt-1095">.ti.pmf</span>,
     where <!--l. 5547--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">name</span><!--l. 5547--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>
     is the name of the bias and the contents of the file are in multicolumn text format (<a href="#sec:colvar_multicolumn_grid">3.8.5<!--tex4ht:ref: sec:colvar_multicolumn_grid --></a>). The
     total force includes the forces applied to the variable by all bias, except those from this bias
     itself. If any bias applies time-dependent forces besides the one using this option, an error is
     raised.
     </li>
<li class="itemize"><a id="colvarbias|writeTISamples"></a> <span class="colorbox" id="colorbox301"><a id="dx1-120002"></a><span class="ectt-1095">writeTISamples</span></span> — <span class="phvro8t-x-x-109">Write the free-energy gradient samples</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 5563--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: colvar bias <!--l. 5563--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option allows to compute total forces for use with thermodynamic integration as done
     by the keyword <a href="#sec:colvarbias"><span class="ectt-1095">writeTIPMF</span></a>. The names of the files containing the variables' histogram and
     mean thermodynamic forces are <span class="phvro8t-x-x-109">outputName</span><span class="ectt-1095">.</span><!--l. 5563--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">name</span><!--l. 5563--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><span class="ectt-1095">.ti.count</span>
     and <span class="phvro8t-x-x-109">outputName</span><span class="ectt-1095">.</span><!--l. 5563--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">name</span><!--l. 5563--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><span class="ectt-1095">.ti.force</span>,
     respectively: these can be used by <span class="ectt-1095">abf_integrate </span>(see <a href="#sec:colvarbias_abf_post">6.3.5<!--tex4ht:ref: sec:colvarbias_abf_post --></a>) or similar utility. Note that
     because the <span class="ectt-1095">.force </span>file contains mean forces instead of free-energy gradients, <span class="ectt-1095">abf_integrate</span>
<!--l. 5563--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">filename</span><!--l. 5563--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>
<span class="ectt-1095">-s -1.0 </span>should be used. This option is on by default when <span class="ectt-1095">writeTIPMF </span>is on, but can be
     enabled separately if the bias is applied to more than one variable, making not possible the
     direct integration of the PMF at runtime. If any bias applies time-dependent forces besides
     the one using this option, an error is raised.
     </li></ul>
<!--l. 5567--><p class="indent">   In adaptive biasing force (ABF) (<a href="#sec:colvarbias_abf">6.3<!--tex4ht:ref: sec:colvarbias_abf --></a>) the above keywords are not recognized, because their
functionality is either included already (conventional ABF) or not available (extended-system
ABF).
</p>
<div class="flushright">
<!--l. 5571--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 5571--><p class="noindent"> </p><div class="heading"> <a id="sec:colvarbias_abf"></a>
<h4 class="subsectionHead"><span class="titlemark">6.3    </span> <a id="sec:colvarbias_abf"></a>Adaptive Biasing Force</h4>
<!--l. 5571--><p class="noindent"></p></div>
<!--l. 5573--><p class="indent">   For a full description of the Adaptive Biasing Force method, see reference <span class="cite">[<a href="#XDarve2008">18</a>]</span>. For details about this
implementation, see references <span class="cite">[<a href="#XHenin2004">19</a>]</span> and <span class="cite">[<a href="#XHenin2010">20</a>]</span>. <span class="phvb8t-x-x-109">When publishing research that makes use of this</span>
<span class="phvb8t-x-x-109">functionality, please cite references</span><span class="phvb8t-x-x-109"> </span><span class="cite"><span class="phvb8t-x-x-109">[</span><a href="#XDarve2008"><span class="phvb8t-x-x-109">18</span></a><span class="phvb8t-x-x-109">]</span></span> <span class="phvb8t-x-x-109">and </span><span class="cite"><span class="phvb8t-x-x-109">[</span><a href="#XHenin2010"><span class="phvb8t-x-x-109">20</span></a><span class="phvb8t-x-x-109">]</span></span><span class="phvb8t-x-x-109">.</span>
</p><!--l. 5579--><p class="indent">   An alternate usage of this feature is the application of custom tabulated biasing potentials to one or
more colvars. See <span class="ectt-1095">inputPrefix </span>and <span class="ectt-1095">updateBias </span>below.
</p><!--l. 5583--><p class="indent">   Combining ABF with the extended Lagrangian feature (<a href="#sec:colvar_extended">4.24<!--tex4ht:ref: sec:colvar_extended --></a>) of the variables produces the
extended-system ABF variant of the method (<a href="#sec:colvarbias_abf_extended">6.4<!--tex4ht:ref: sec:colvarbias_abf_extended --></a>).
</p><!--l. 5587--><p class="indent">   ABF is based on the thermodynamic integration (TI) scheme for computing
free energy profiles. The free energy as a function of a set of collective variables
<!--l. 5589--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle> <mo class="MathClass-rel">=</mo> <msub><mrow><mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-close">)</mo></mrow><mrow><mi>i</mi><mo class="MathClass-rel">∈</mo><mo class="MathClass-open">[</mo><mn>1</mn><mo class="MathClass-punc">,</mo><mi>n</mi><mo class="MathClass-close">]</mo></mrow></msub></math> is defined from the
canonical distribution of <!--l. 5590--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></math>,
<!--l. 5590--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="bold-script">𝒫</mi><mo class="MathClass-open">(</mo><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle><mo class="MathClass-close">)</mo></math>:
</p>
<table class="equation"><tr><td> <a id="x1-121001r24"></a>
<!--l. 5592--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>A</mi> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></mrow><mo fence="true" form="postfix">)</mo></mrow> <mo class="MathClass-rel">=</mo> <mo class="MathClass-bin">−</mo><mfrac><mrow><mn>1</mn></mrow>
<mrow><mi>β</mi></mrow></mfrac><mi class="qopname">ln</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--><mi mathvariant="bold-script">𝒫</mi><mo class="MathClass-open">(</mo><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle><mo class="MathClass-close">)</mo><mo class="MathClass-bin">+</mo><msub><mrow><mi>A</mi></mrow><mrow><mn>0</mn></mrow></msub>
</math></td><td class="eq-no">(24)</td></tr></table>
<!--l. 5597--><p class="indent">   In the TI formalism, the free energy is obtained from its gradient, which is generally calculated in the form of the
average of a force <!--l. 5599--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>F</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mi>ξ</mi></mrow></msub></math>
exerted on <!--l. 5599--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></math>, taken
over an iso-<!--l. 5599--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></math>
surface:
</p>
<table class="equation"><tr><td> <a id="eq:gradient"></a>
<!--l. 5601--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow>
<mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi class="MathClass-op">∇</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo></mstyle><mtext></mtext></mstyle></mrow><mrow><mi>ξ</mi></mrow></msub><mi>A</mi> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></mrow><mo fence="true" form="postfix">)</mo></mrow> <mo class="MathClass-rel">=</mo><msub><mrow> <mrow><mo fence="true" form="prefix"> ⟨</mo><mrow><mo class="MathClass-bin">−</mo><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>F</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mi>ξ</mi></mrow></msub></mrow><mo fence="true" form="postfix">⟩</mo></mrow></mrow><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></mrow></msub>
</math></td><td class="eq-no">(25)</td></tr></table>
<!--l. 5606--><p class="indent">   Several formulae that take the form of (<a href="#eq:gradient">25<!--tex4ht:ref: eq:gradient --></a>) have been proposed. This implementation
relies partly on the classic formulation <span class="cite">[<a href="#XCarter1989">21</a>]</span>, and partly on a more versatile scheme originating
in a work by Ruiz-Montero et al. <span class="cite">[<a href="#XRuiz-Montero1997">22</a>]</span>, generalized by den Otter <span class="cite">[<a href="#XdenOtter2000">23</a>]</span> and extended to
multiple variables by Ciccotti et al. <span class="cite">[<a href="#XCiccotti2005">24</a>]</span>. Consider a system subject to constraints of the form
<!--l. 5612--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>σ</mi></mrow><mrow><mi>k</mi></mrow></msub><mo class="MathClass-open">(</mo><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>x</mi></mstyle><mtext></mtext></mstyle><mo class="MathClass-close">)</mo> <mo class="MathClass-rel">=</mo> <mn>0</mn></math>. Let
<!--l. 5613--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mo class="MathClass-open">(</mo><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>v</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-close">)</mo></mrow><mrow><mi>i</mi><mo class="MathClass-rel">∈</mo><mo class="MathClass-open">[</mo><mn>1</mn><mo class="MathClass-punc">,</mo><mi>n</mi><mo class="MathClass-close">]</mo></mrow></msub></math> be arbitrarily chosen
vector fields (<!--l. 5614--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow><mi>ℝ</mi></mrow><mrow><mn>3</mn><mi>N</mi></mrow></msup> <mo class="MathClass-rel">→</mo> <msup><mrow><mi>ℝ</mi></mrow><mrow><mn>3</mn><mi>N</mi></mrow></msup></math>)
verifying, for all <!--l. 5614--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi></math>,
<!--l. 5615--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>j</mi></math>, and
<!--l. 5615--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>k</mi></math>:
</p>
<!--tex4ht:inline--><!--l. 5617--><math display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mtable class="eqnarray" columnalign="right center left">
<mtr><mtd class="eqnarray-1"><msub><mrow> <mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>v</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-bin">⋅</mo><mstyle class="mbox"><mtext></mtext><mstyle class="math"><msub><mrow><mi class="MathClass-op">∇</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo></mrow><mrow><mspace class="negthinspace" width="-0.17em"></mspace><mspace class="negthinspace" width="-0.17em"></mspace><mi>x</mi></mrow></msub><mspace class="thinspace" width="0.17em"></mspace></mstyle><mtext></mtext></mstyle><msub><mrow><mi>ξ</mi></mrow><mrow><mi>j</mi></mrow></msub></mtd><mtd class="eqnarray-2"> <mo class="MathClass-rel">=</mo></mtd><mtd class="eqnarray-3"> <msub><mrow><mi>δ</mi></mrow><mrow><mi>i</mi><mi>j</mi></mrow></msub></mtd><mtd class="eqnarray-4"> <mtext class="eqnarray">(26)</mtext><mtext class="label" id="eq:ortho_gradient"></mtext><mtext class="endlabel"></mtext></mtd>
</mtr><mtr><mtd class="eqnarray-1"><msub><mrow> <mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>v</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-bin">⋅</mo><mstyle class="mbox"><mtext></mtext><mstyle class="math"><msub><mrow><mi class="MathClass-op">∇</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo></mrow><mrow><mspace class="negthinspace" width="-0.17em"></mspace><mspace class="negthinspace" width="-0.17em"></mspace><mi>x</mi></mrow></msub><mspace class="thinspace" width="0.17em"></mspace></mstyle><mtext></mtext></mstyle><msub><mrow><mi>σ</mi></mrow><mrow><mi>k</mi></mrow></msub></mtd><mtd class="eqnarray-2"> <mo class="MathClass-rel">=</mo></mtd><mtd class="eqnarray-3"> <mn>0</mn> </mtd><mtd class="eqnarray-4"> <mtext class="eqnarray">(27)</mtext><mtext class="label" id="eq:ortho_constraints"></mtext><mtext class="endlabel"></mtext></mtd> </mtr></mtable>
</math>
<!--l. 5622--><p class="nopar">
</p>
<!--l. 5624--><p class="indent">   then the following holds <span class="cite">[<a href="#XCiccotti2005">24</a>]</span>:
</p>
<table class="equation"><tr><td> <a id="eq:gradient_vector"></a>
<!--l. 5626--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mfrac><mrow><mi>∂</mi><mi>A</mi></mrow>
<mrow><mi>∂</mi><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow></mfrac> <mo class="MathClass-rel">=</mo><msub><mrow> <mrow><mo fence="true" form="prefix"> ⟨</mo><mrow><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>v</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-bin">⋅</mo><mstyle class="mbox"><mtext></mtext><mstyle class="math"><msub><mrow><mi class="MathClass-op">∇</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo></mrow><mrow><mspace class="negthinspace" width="-0.17em"></mspace><mspace class="negthinspace" width="-0.17em"></mspace><mi>x</mi></mrow></msub><mspace class="thinspace" width="0.17em"></mspace></mstyle><mtext></mtext></mstyle><mi>V</mi> <mo class="MathClass-bin">−</mo><msub><mrow><mi>k</mi></mrow><mrow><mi>B</mi></mrow></msub><mi>T</mi> <mstyle class="mbox"><mtext></mtext><mstyle class="math"><msub><mrow><mi class="MathClass-op">∇</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo></mrow><mrow><mspace class="negthinspace" width="-0.17em"></mspace><mspace class="negthinspace" width="-0.17em"></mspace><mi>x</mi></mrow></msub><mspace class="thinspace" width="0.17em"></mspace></mstyle><mtext></mtext></mstyle><mo class="MathClass-bin">⋅</mo><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>v</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mi>i</mi></mrow></msub></mrow><mo fence="true" form="postfix">⟩</mo></mrow></mrow><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></mrow></msub>
</math></td><td class="eq-no">(28)</td></tr></table>
<!--l. 5632--><p class="indent">   where <!--l. 5632--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>V</mi></math> is the potential
energy function. <!--l. 5633--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>v</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mi>i</mi></mrow></msub></math>
can be interpreted as the direction along which the force acting on variable
<!--l. 5634--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></math> is measured,
whereas the second term in the average corresponds to the geometric entropy contribution that appears as a Jacobian
correction in the classic formalism <span class="cite">[<a href="#XCarter1989">21</a>]</span>. Condition (<a href="#eq:ortho_gradient">26<!--tex4ht:ref: eq:ortho_gradient --></a>) states that the direction along which the total force on
<!--l. 5638--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></math> is measured is orthogonal to
the gradient of <!--l. 5639--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>j</mi></mrow></msub></math>, which means
that the force measured on <!--l. 5639--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></math>
does not act on <!--l. 5640--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>j</mi></mrow></msub></math>.
</p><!--l. 5642--><p class="indent">   Equation (<a href="#eq:ortho_constraints">27<!--tex4ht:ref: eq:ortho_constraints --></a>) implies that constraint forces are orthogonal to the directions along which the free
energy gradient is measured, so that the measurement is effectively performed on unconstrained degrees
of freedom. In NAMD, constraints are typically applied to the lengths of bonds involving hydrogen atoms,
for example in TIP3P water molecules (parameter <span class="ectt-1095">rigidBonds</span>).
</p><!--l. 5649--><p class="indent">   In the framework of ABF, <!--l. 5650--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>F</mi></mrow><mrow><mi>ξ</mi></mrow></msub></math> is
accumulated in bins of finite size <!--l. 5650--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>δ</mi><mi>ξ</mi></math>,
thereby providing an estimate of the free energy gradient according to equation (<a href="#eq:gradient">25<!--tex4ht:ref: eq:gradient --></a>). The biasing
force applied along the collective variables to overcome free energy barriers is calculated
as:
</p>
<table class="equation"><tr><td> <a id="x1-121005r29"></a>
<!--l. 5656--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<msup><mrow><mi>F</mi></mrow><mrow><mi>A</mi><mi>B</mi><mi>F</mi> </mrow></msup> <mo class="MathClass-rel">=</mo> <mi>α</mi><mo class="MathClass-open">(</mo><msub><mrow><mi>N</mi></mrow><mrow>
<mi>ξ</mi></mrow></msub><mo class="MathClass-close">)</mo><mo class="MathClass-bin">×</mo><mstyle class="mbox"><mtext></mtext><mstyle class="math"><msub><mrow><mi class="MathClass-op">∇</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo></mrow><mrow><mspace class="negthinspace" width="-0.17em"></mspace><mspace class="negthinspace" width="-0.17em"></mspace><mi>x</mi></mrow></msub><mspace class="thinspace" width="0.17em"></mspace></mstyle><mtext></mtext></mstyle><mover accent="false"><mrow><mi>A</mi></mrow><mo accent="true">~</mo></mover><mo class="MathClass-open">(</mo><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle><mo class="MathClass-close">)</mo>
</math></td><td class="eq-no">(29)</td></tr></table>
<!--l. 5661--><p class="indent">   where <!--l. 5661--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle class="mbox"><mtext></mtext><mstyle class="math"><msub><mrow><mi class="MathClass-op">∇</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo></mrow><mrow><mspace class="negthinspace" width="-0.17em"></mspace><mspace class="negthinspace" width="-0.17em"></mspace><mi>x</mi></mrow></msub><mspace class="thinspace" width="0.17em"></mspace></mstyle><mtext></mtext></mstyle><mover accent="false"><mrow><mi>A</mi></mrow><mo accent="true">~</mo></mover></math>
denotes the current estimate of the free energy gradient at the current point
<!--l. 5662--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></math> in the collective
variable subspace, and <!--l. 5663--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>α</mi><mo class="MathClass-open">(</mo><msub><mrow><mi>N</mi></mrow><mrow><mi>ξ</mi></mrow></msub><mo class="MathClass-close">)</mo></math>
is a scaling factor that is ramped from 0 to 1 as the local number of samples
<!--l. 5664--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>N</mi></mrow><mrow><mi>ξ</mi></mrow></msub></math>
increases to prevent non-equilibrium effects in the early phase of the simulation, when the gradient
                                                                                         
                                                                                         
estimate has a large variance. See the <a href="#abf|fullSamples"><span class="ectt-1095">fullSamples</span></a> parameter below for details.
</p><!--l. 5669--><p class="indent">   As sampling of the phase space proceeds, the estimate
<!--l. 5670--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle class="mbox"><mtext></mtext><mstyle class="math"><msub><mrow><mi class="MathClass-op">∇</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo></mrow><mrow><mspace class="negthinspace" width="-0.17em"></mspace><mspace class="negthinspace" width="-0.17em"></mspace><mi>x</mi></mrow></msub><mspace class="thinspace" width="0.17em"></mspace></mstyle><mtext></mtext></mstyle><mover accent="false"><mrow><mi>A</mi></mrow><mo accent="true">~</mo></mover></math> is
progressively refined. The biasing force introduced in the equations of motion guarantees that in the bin centered
around <!--l. 5672--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></math>,
the forces acting along the selected collective variables average to zero over time. Eventually, as the
underlying free energy surface is canceled by the adaptive bias, evolution of the system along
<!--l. 5675--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></math> is
governed mainly by diffusion. Although this implementation of ABF can in principle be used in
arbitrary dimension, a higher-dimension collective variable space is likely to be difficult to
sample and visualize. Most commonly, the number of variables is one or two, sometimes
three.
</p>
<div class="flushright">
<!--l. 5683--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 5683--><p class="noindent"> </p><div class="heading"> <a id="sec:colvarbias_abf_req"></a>
<h5 class="subsubsectionHead"><span class="titlemark">6.3.1    </span> <a id="sec:colvarbias_abf_req"></a>ABF requirements on collective variables</h5>
<!--l. 5683--><p class="noindent"></p></div>
<!--l. 5685--><p class="indent">   The following conditions must be met for an ABF simulation to be possible and to produce an accurate
estimate of the free energy profile. Note that these requirements do not apply when using the
extended-system ABF method (<a href="#sec:colvarbias_abf_extended">6.4<!--tex4ht:ref: sec:colvarbias_abf_extended --></a>).
</p><!--l. 5690--><p class="indent">
</p><ol class="enumerate1">
<li class="enumerate" id="x1-122002x1"><span class="phvro8t-x-x-109">Only linear combinations </span>of colvar components can be used in ABF calculations.
     </li>
<li class="enumerate" id="x1-122004x2"><span class="phvro8t-x-x-109">Availability of total forces </span>is necessary. The following colvar components can be used in ABF
     calculations:  <span class="ectt-1095">distance</span>,  <span class="ectt-1095">distance_xy</span>,  <span class="ectt-1095">distance_z</span>,  <span class="ectt-1095">angle</span>,  <span class="ectt-1095">dihedral</span>,  <span class="ectt-1095">gyration</span>,  <span class="ectt-1095">rmsd </span>and
     <span class="ectt-1095">eigenvector</span>. Atom groups may not be replaced by dummy atoms, unless they are excluded
     from the force measurement by specifying <span class="ectt-1095">oneSiteTotalForce</span>, if available.
     </li>
<li class="enumerate" id="x1-122006x3"><span class="phvro8t-x-x-109">Mutual orthogonality of colvars</span>. In a multidimensional ABF calculation, equation (<a href="#eq:ortho_gradient">26<!--tex4ht:ref: eq:ortho_gradient --></a>) must be satisfied for any
     two colvars <!--l. 5699--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></math>
     and <!--l. 5699--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>j</mi></mrow></msub></math>.
     Various cases fulfill this orthogonality condition:
          <ul class="itemize1">
<li class="itemize"><!--l. 5702--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></math>
          and <!--l. 5702--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>j</mi></mrow></msub></math>
          are based on non-overlapping sets of atoms.
                                                                                         
                                                                                         
          </li>
<li class="itemize">atoms involved in the force measurement on <!--l. 5703--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></math>
          do not participate in the definition of <!--l. 5704--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>j</mi></mrow></msub></math>.
          This can be obtained using the option <span class="ectt-1095">oneSiteTotalForce </span>of the <span class="ectt-1095">distance</span>, <span class="ectt-1095">angle</span>, and
          <span class="ectt-1095">dihedral </span>components (example: Ramachandran angles <!--l. 5706--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ϕ</mi></math>,
          <!--l. 5706--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ψ</mi></math>).
          </li>
<li class="itemize"><!--l. 5707--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></math>
          and <!--l. 5707--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>j</mi></mrow></msub></math>
          are orthogonal by construction. Useful cases are the sum and difference of two components,
          or <span class="ectt-1095">distance_z </span>and <span class="ectt-1095">distance_xy </span>using the same axis.</li></ul>
</li>
<li class="enumerate" id="x1-122008x4"><span class="phvro8t-x-x-109">Mutual orthogonality of components</span>: when several components are combined into a colvar, it is assumed that
     their vectors <!--l. 5711--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>v</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mi>i</mi></mrow></msub></math>
     (equation (<a href="#eq:gradient_vector">28<!--tex4ht:ref: eq:gradient_vector --></a>)) are mutually orthogonal. The cases described for colvars in the previous paragraph
     apply.
     </li>
<li class="enumerate" id="x1-122010x5"><span class="phvro8t-x-x-109">Orthogonality of colvars and constraints</span>: equation <a href="#eq:ortho_constraints">27<!--tex4ht:ref: eq:ortho_constraints --></a> can be satisfied in two simple ways, if either
     no constrained atoms are involved in the force measurement (see point 3 above) or pairs of atoms
     joined by a constrained bond are part of an <span class="phvro8t-x-x-109">atom group </span>which only intervenes through its center
     (center of mass or geometric center) in the force measurement. In the latter case, the contributions
     of the two atoms to the left-hand side of equation <a href="#eq:ortho_constraints">27<!--tex4ht:ref: eq:ortho_constraints --></a> cancel out. For example, all atoms of a rigid
     TIP3P water molecule can safely be included in an atom group used in a <span class="ectt-1095">distance</span>
     component.</li></ol>
<div class="flushright">
<!--l. 5724--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 5724--><p class="noindent"> </p><div class="heading"> <a id="sec:colvarbias_abf_params"></a>
<h5 class="subsubsectionHead"><span class="titlemark">6.3.2    </span> <a id="sec:colvarbias_abf_params"></a>Parameters for ABF</h5>
<!--l. 5724--><p class="noindent"></p></div>
<!--l. 5726--><p class="indent">   ABF depends on parameters from each collective variable to define the grid on which free energy
gradients are computed: see <a href="#sec:colvar_grid_params">4.22<!--tex4ht:ref: sec:colvar_grid_params --></a> for details. Alternately, grid parameters can be set directly inside the
<span class="ectt-1095">abf </span>block by inserting a <span class="ectt-1095">grid </span>sub-block, as explained in <a href="#sec:colvarbias_grid">6.1<!--tex4ht:ref: sec:colvarbias_grid --></a>. Other parameters to control the ABF runtime
can be set in the ABF configuration block:
</p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox302"><a id="dx1-123001"></a><span class="ectt-1095">name</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">name</span></a> (biasing and analysis methods)
                                                                                         
                                                                                         
     </li>
<li class="itemize"><span class="colorbox" id="colorbox303"><a id="dx1-123002"></a><span class="ectt-1095">colvars</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">colvars</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox304"><a id="dx1-123003"></a><span class="ectt-1095">outputEnergy</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">outputEnergy</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox305"><a id="dx1-123004"></a><span class="ectt-1095">outputFreq</span></span> — same definition as <a href="#colvarbias|outputFreq"><span class="ectt-1095">outputFreq</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox306"><a id="dx1-123005"></a><span class="ectt-1095">stepZeroData</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">stepZeroData</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="abf|fullSamples"></a> <span class="colorbox" id="colorbox307"><a id="dx1-123006"></a><span class="ectt-1095">fullSamples</span></span> — <span class="phvro8t-x-x-109">Number of samples in a bin prior to full application of the ABF</span><br class="newline"/>Default: 200 <!--l. 5752--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     integer, context: <span class="ectt-1095">abf</span> <!--l. 5752--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>To  avoid  non-equilibrium  effects  due  to  large  fluctuations  of  the  force  exerted  along  the
     colvars, it is recommended to apply a biasing force only after a the estimate has started
     converging. If <span class="ectt-1095">fullSamples </span>is non-zero, the applied biasing force is scaled by a factor <!--l. 5752--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>α</mi><mo class="MathClass-open">(</mo><msub><mrow><mi>N</mi></mrow><mrow><mi>ξ</mi></mrow></msub><mo class="MathClass-close">)</mo></math>
     between 0 and 1. If the number of samples <!--l. 5752--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>N</mi></mrow><mrow><mi>ξ</mi></mrow></msub></math>
     in the current bin is higher than <span class="ectt-1095">fullSamples</span>, the factor is one. If it is less than <span class="ectt-1095">minSamples</span>
     (by default, half of <span class="ectt-1095">fullSamples</span>), the factor is zero and no bias is applied. Between those two
     thresholds, the factor follows a linear ramp from 0 to 1: <!--l. 5752--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>α</mi><mo class="MathClass-open">(</mo><msub><mrow><mi>N</mi></mrow><mrow><mi>ξ</mi></mrow></msub><mo class="MathClass-close">)</mo> <mo class="MathClass-rel">=</mo> <mo class="MathClass-open">(</mo><msub><mrow><mi>N</mi></mrow><mrow><mi>ξ</mi></mrow></msub><mo class="MathClass-bin">−</mo><mstyle mathvariant="normal"><mi>m</mi><mi>i</mi><mi>n</mi><mi>S</mi><mi>a</mi><mi>m</mi><mi>p</mi><mi>l</mi><mi>e</mi><mi>s</mi></mstyle><mo class="MathClass-close">)</mo><mo class="MathClass-bin">∕</mo><mo class="MathClass-open">(</mo><mstyle mathvariant="normal"><mi>f</mi><mi>u</mi><mi>l</mi><mi>l</mi><mi>S</mi><mi>a</mi><mi>m</mi><mi>p</mi><mi>l</mi><mi>e</mi><mi>s</mi></mstyle><mo class="MathClass-bin">−</mo><mstyle mathvariant="normal"><mi>m</mi><mi>i</mi><mi>n</mi><mi>S</mi><mi>a</mi><mi>m</mi><mi>p</mi><mi>l</mi><mi>e</mi><mi>s</mi></mstyle><mo class="MathClass-close">)</mo></math>.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox308"><a id="dx1-123007"></a><span class="ectt-1095">minSamples</span></span> — <span class="phvro8t-x-x-109">Number of samples in a bin prior to application of the ABF</span><br class="newline"/>Default: fullSamples / 2 <!--l. 5759--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     integer, context: <span class="ectt-1095">abf</span> <!--l. 5759--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>See <a href="#abf|fullSamples"><span class="ectt-1095">fullSamples</span></a>.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox309"><a id="dx1-123008"></a><span class="ectt-1095">maxForce</span></span> — <span class="phvro8t-x-x-109">Maximum magnitude of the ABF force</span><br class="newline"/>Default: disabled <!--l. 5771--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     decimals (one per colvar), context: <span class="ectt-1095">abf</span> <!--l. 5771--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option enforces a cap on the magnitude of the biasing force effectively applied by this
     ABF bias on each colvar. This can be useful in the presence of singularities in the PMF
     such as hard walls, where the discretization of the average force becomes very inaccurate,
     causing the colvar's diffusion to get “stuck" at the singularity. To enable this cap, provide one
     non-negative value for each colvar. The unit of force is kcal/mol/<span class="phvro8t-x-x-109">(colvar unit)</span>.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox310"><a id="dx1-123009"></a><span class="ectt-1095">hideJacobian</span></span> — <span class="phvro8t-x-x-109">Remove geometric entropy term from calculated free energy gradient?</span><br class="newline"/>Default: <span class="ectt-1095">no</span> <!--l. 5794--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">abf</span> <!--l. 5794--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>In a few special cases, most notably distance-based variables, an alternate definition of the
     potential of mean force is traditionally used, which excludes the Jacobian term describing
     the effect of geometric entropy on the distribution of the variable. This results, for example, in
                                                                                         
                                                                                         
     particle-particle potentials of mean force being flat at large separations. The Jacobian term is
     exactly represented in equation (<a href="#eq:gradient_vector">28<!--tex4ht:ref: eq:gradient_vector --></a>) by the second term of the average, <!--l. 5794--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-bin">−</mo><msub><mrow><mi>k</mi></mrow><mrow><mi>B</mi></mrow></msub><mi>T</mi> <mstyle class="mbox"><mtext></mtext><mstyle class="math"><msub><mrow><mi class="MathClass-op">∇</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo></mrow><mrow><mspace class="negthinspace" width="-0.17em"></mspace><mspace class="negthinspace" width="-0.17em"></mspace><mi>x</mi></mrow></msub><mspace class="thinspace" width="0.17em"></mspace></mstyle><mtext></mtext></mstyle><mo class="MathClass-bin">⋅</mo><msub><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>v</mi></mstyle><mtext></mtext></mstyle></mrow><mrow><mi>i</mi></mrow></msub></math>.
     Enabling the <span class="ectt-1095">hideJacobian </span>option causes the output data to follow the traditional potential of
     mean force convention, by omitting this contribution from the measured free energy gradients.
     To ensure uniform sampling despite the incomplete description of the free energy, an additional
     biasing force that counteracts the Jacobian force is applied internally by the colvar. <span class="phvb8t-x-x-109">Warning:</span>
     using this option without detailed knowledge of the terms of equation (<a href="#eq:gradient_vector">28<!--tex4ht:ref: eq:gradient_vector --></a>) is <span class="phvro8t-x-x-109">not recommended</span>.
     <span class="phvb8t-x-x-109">Warning: </span>a bug affecting Colvars version until October 2020 leads to this term to be subtracted
     twice  in  the  PMFs  reported  by  conventional  ABF.  The  bug  does  not  affect  the  simulated
     trajectory, and the correct PMF can be recovered by adding back the term manually. Other
     than conventional ABF, the PMFs computed by other sampling methods were never affected.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox311"><a id="dx1-123010"></a><span class="ectt-1095">historyFreq</span></span> — <span class="phvro8t-x-x-109">Frequency (in timesteps) at which ABF history files are accumulated</span><br class="newline"/>Default: 0 <!--l. 5805--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     integer, context: <span class="ectt-1095">abf</span> <!--l. 5805--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If this number is non-zero, the free energy gradient estimate and sampling histogram (and
     the PMF in one-dimensional calculations) are written to files on disk at the given time interval.
     History file names use the same prefix as output files, with “<span class="ectt-1095">.hist</span>" appended (<span class="phvro8t-x-x-109">outputName</span><span class="ectt-1095">.hist.pmf</span>).
     <span class="ectt-1095">historyFreq </span>must be a multiple of <a href="#colvarbias|outputFreq"><span class="ectt-1095">outputFreq</span></a>.
     </li>
<li class="itemize"><a id="abf|inputPrefix"></a> <span class="colorbox" id="colorbox312"><a id="dx1-123011"></a><span class="ectt-1095">inputPrefix</span></span> — <span class="phvro8t-x-x-109">Filename prefix for reading ABF data</span><br class="newline"/><!--l. 5823--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> list
     of strings, context: <span class="ectt-1095">abf</span> <!--l. 5823--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If this parameter is set, for each item in the list, ABF tries to read a gradient and a sampling
     files named <!--l. 5823--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">inputPrefix</span><!--l. 5823--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><span class="ectt-1095">.grad</span>
     and <!--l. 5823--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">inputPrefix</span><!--l. 5823--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><span class="ectt-1095">.count</span>.
     This is done at startup and sets the initial state of the ABF algorithm. The data from all
     provided files is combined appropriately. Also, the grid definition (min and max values, width)
     need not be the same that for the current run. This command is useful to piece together
     data from simulations in different regions of collective variable space, or change the colvar
     boundary values and widths. Note that it is not recommended to use it to switch to a smaller
     width, as that will leave some bins empty in the finer data grid. This option is NOT compatible
     with reading the data from a restart file (<span class="ectt-1095">colvarsInput </span>option of the NAMD config file).
     </li>
<li class="itemize"><span class="colorbox" id="colorbox313"><a id="dx1-123012"></a><span class="ectt-1095">applyBias</span></span> — <span class="phvro8t-x-x-109">Apply the ABF bias?</span><br class="newline"/>Default: <span class="ectt-1095">yes</span> <!--l. 5833--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">abf</span> <!--l. 5833--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If this is set to no, the calculation proceeds normally but the adaptive biasing force is not
     applied. Data is still collected to compute the free energy gradient. This is mostly intended
     for testing purposes, and should not be used in routine simulations.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox314"><a id="dx1-123013"></a><span class="ectt-1095">updateBias</span></span> — <span class="phvro8t-x-x-109">Update the ABF bias?</span><br class="newline"/>Default: <span class="ectt-1095">yes</span> <!--l. 5846--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">abf</span> <!--l. 5846--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If this is set to no, the initial biasing force (e.g. read from a restart file or through <span class="ectt-1095">inputPrefix</span>)
     is not updated during the simulation. As a result, a constant bias is applied. This can be
     used to apply a custom, tabulated biasing potential to any combination of colvars. To that
     effect, one should prepare a gradient file containing the gradient of the potential to be applied
     (negative of the bias force), and a count file containing only values greater than <a href="#abf|fullSamples"><span class="ectt-1095">fullSamples</span></a>.
     These files must match the grid parameters of the colvars. </li></ul>
<div class="flushright">
<!--l. 5849--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 5849--><p class="noindent"> </p><div class="heading"> <a id="sec:colvarbias_abf_shared"></a>
<h5 class="subsubsectionHead"><span class="titlemark">6.3.3    </span> <a id="sec:colvarbias_abf_shared"></a>Multiple-walker ABF</h5>
<!--l. 5849--><p class="noindent"></p></div>
<!--l. 5852--><p class="indent">   This implements the multiple-walker ABF scheme described in <span class="cite">[<a href="#XMinoukadeh2010">25</a>]</span>. The references for this
implementation are <span class="cite">[<a href="#XComer2014c">26</a>]</span> and <span class="cite">[<a href="#XFiorin2024">27</a>]</span>. This feature requires that NAMD be compiled and executed with
multiple-replica support.
</p><!--l. 5857--><p class="indent">   If <a href="#abf|shared"><span class="ectt-1095">shared</span></a> is enabled, the total force samples will be synchronized among all replicas at intervals
defined by <a href="#abf|sharedFreq"><span class="ectt-1095">sharedFreq</span></a>. Each replica maintains a separate buffer of total force samples that determine the
biasing force. Every <a href="#abf|sharedFreq"><span class="ectt-1095">sharedFreq</span></a> steps, the replicas communicate the samples that have been gathered
since the last synchronization time, ensuring all replicas apply a similar biasing force. Thus, it is as if total
force samples among all replicas are gathered in a single shared buffer. Shared ABF allows all replicas to
benefit from the sampling done by other replicas and can lead to faster convergence of the biasing
force.
</p><!--l. 5869--><p class="indent">   An implementation of the selection mechanism described in <span class="cite">[<a href="#XComer2014c">26</a>]</span> is provided as a set of Tcl scripts in
<span class="ectt-1095">colvartools/mwABF_selection.tcl </span>in the Colvars repository. Compared with the initial implementation,
the current implementation supports selection on sets of up to 3 colvars. To reduce noise,
samples can be counted in a hybercube around the current bin (see Tcl script for details). A set
of example NAMD inputs can be found in the <span class="ectt-1095">namd/tests/library/multiple_walker_abf</span>
directory.
</p>
<!--l. 5871--><p class="noindent"><span class="paragraphHead"><a id="x1-1250006.3.3"></a><span class="phvb8t-x-x-109">Output files of multiple-walker ABF.</span></span>
   In multiple-walker ABF runs, since Colvars version 2024-01-06, each walker outputs gradient and
count files containing only data collected locally. In addition, the first walker outputs the collected data
using the common prefix, with an additional <span class="ectt-1095">".all" </span>string (i.e. file names ending with <span class="ectt-1095">".all.count"</span>,
<span class="ectt-1095">".all.grad" </span>etc.).
</p>
<ul class="itemize1">
<li class="itemize"><a id="abf|shared"></a> <span class="colorbox" id="colorbox315"><a id="dx1-125001"></a><span class="ectt-1095">shared</span></span> — <span class="phvro8t-x-x-109">Apply multiple-walker ABF, sharing force samples among the replicas?</span><br class="newline"/>Default: <span class="ectt-1095">no</span> <!--l. 5885--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">abf</span> <!--l. 5885--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Enable sharing of ABF data between replicas, in a multi-replica simulation.
     </li>
<li class="itemize"><a id="abf|sharedFreq"></a> <span class="colorbox" id="colorbox316"><a id="dx1-125002"></a><span class="ectt-1095">sharedFreq</span></span> — <span class="phvro8t-x-x-109">Frequency (in timesteps) at which force samples are synchronized among</span>
<span class="phvro8t-x-x-109">the replicas</span><br class="newline"/>Default: <a href="#colvarbias|outputFreq"><span class="ectt-1095">outputFreq</span></a> <!--l. 5896--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     integer or zero, context: <span class="ectt-1095">abf</span> <!--l. 5896--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Every <span class="ectt-1095">sharedFreq </span>steps, the replicas communicate the samples that have been gathered
     since the last synchronization time. Because this forces the replicas to synchronize, small
     values of this parameter may slightly reduce the overall throughput, especially on systems
     with unequal performance. </li></ul>
<div class="flushright">
<!--l. 5899--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 5899--><p class="noindent"> </p><div class="heading"> <a id="sec:colvarbias_abf_output"></a>
<h5 class="subsubsectionHead"><span class="titlemark">6.3.4    </span> <a id="sec:colvarbias_abf_output"></a>Output files</h5>
<!--l. 5899--><p class="noindent"></p></div>
<!--l. 5901--><p class="indent">   The ABF bias produces the following files, all in multicolumn text format (<a href="#sec:colvar_multicolumn_grid">3.8.5<!--tex4ht:ref: sec:colvar_multicolumn_grid --></a>): </p>
<ul class="itemize1">
<li class="itemize"><span class="phvro8t-x-x-109">outputName</span><span class="ectt-1095">.grad</span>: current estimate of the free energy gradient (grid), in multicolumn;
     </li>
<li class="itemize"><span class="phvro8t-x-x-109">outputName</span><span class="ectt-1095">.count</span>: histogram of samples collected, on the same grid;
     </li>
<li class="itemize"><span class="phvro8t-x-x-109">outputName</span><span class="ectt-1095">.pmf</span>: integrated free energy profile or PMF (for dimension 1, and dimension 2
     or 3).</li></ul>
<!--l. 5909--><p class="indent">   Also in the case of one-dimensional calculations, the ABF bias can report its current energy via
<span class="ectt-1095">outputEnergy</span>; in higher dimensions, such computation is not implemented and the energy reported is
zero.
</p><!--l. 5911--><p class="indent">   If several ABF biases are defined concurrently, their name is inserted to produce unique filenames for
output, as in <span class="phvro8t-x-x-109">outputName</span><span class="ectt-1095">.abf1.grad</span>. This should not be done routinely and could lead to meaningless
results: only do it if you know what you are doing!
</p><!--l. 5916--><p class="indent">   If the colvar space has been partitioned into sections (<span class="phvro8t-x-x-109">windows</span>) in which independent ABF
simulations have been run, the resulting data can be merged using the <span class="ectt-1095">inputPrefix </span>option described
above (a run of 0 steps is enough).
                                                                                         
                                                                                         
</p>
<div class="flushright">
<!--l. 5922--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 5922--><p class="noindent"> </p><div class="heading"> <a id="sec:colvarbias_abf_post"></a>
<h5 class="subsubsectionHead"><span class="titlemark">6.3.5    </span> <a id="sec:colvarbias_abf_post"></a>Multidimensional free energy surfaces</h5>
<!--l. 5922--><p class="noindent"></p></div>
<!--l. 5924--><p class="indent">   The ABF method only produces an estimate of the free energy <span class="phvro8t-x-x-109">gradient</span>. The free energy surface itself
can be computed depending on the value of <span class="ectt-1095">integrate </span>and related options.
</p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox317"><a id="dx1-127001"></a><span class="ectt-1095">integrate</span></span> — <span class="phvro8t-x-x-109">Integrate free energy surface from ABF gradients (dim &lt; 3)</span><br class="newline"/>Default: <span class="ectt-1095">yes</span> <!--l. 5947--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">abf</span> <!--l. 5947--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option, active by default when the dimension of the colvar space is 3 or less, enables
     the calculation of an integrated free energy surface every time ABF output files are
     written. In dimension 2 or 3, integration is performed by solving a Poisson equation: <span class="cite">[<a href="#XHenin2021">28</a>]</span> <table class="equation"><tr><td>
<a id="x1-127002r30"></a>
<!--l. 5947--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<msup><mrow><mi class="MathClass-op">∇</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo></mrow><mrow><mn>2</mn></mrow></msup><msub><mrow><mi>A</mi></mrow><mrow>
<mi>t</mi></mrow></msub> <mo class="MathClass-rel">=</mo> <mi class="MathClass-op">∇</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo><mo class="MathClass-bin">⋅</mo><msub><mrow><mi>G</mi></mrow><mrow><mi>t</mi></mrow></msub>
</math></td><td class="eq-no">(30)</td></tr></table>
<!--l. 5947--><p class="noindent">wehere <!--l. 5947--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>G</mi></mrow><mrow><mi>t</mi></mrow></msub></math> is the estimated
     gradient at time <!--l. 5947--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>t</mi></math>,
     and <!--l. 5947--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>A</mi></mrow><mrow><mi>t</mi></mrow></msub></math>
     is corresponding free energy surface. The free energy surface is written under the file name
     <span class="ectt-1095">&lt;outputName&gt;.pmf</span>, in a plain text format (see <a href="#sec:colvar_multicolumn_grid">3.8.5<!--tex4ht:ref: sec:colvar_multicolumn_grid --></a>) that can be read by most data plotting and
     analysis programs (e.g. Gnuplot). Periodic boundary conditions are applied to periodic coordinates,
     and Neumann boundary conditions otherwise (imposed free energy gradient at the boundary
     of the domain). The grid used for free energy discretization is extended by one point
     along non-periodic coordinates, but not along periodic coordinates. See ref. <span class="cite">[<a href="#XHenin2021">28</a>]</span> for
     details.
                                                                                         
                                                                                         
     </p></li>
<li class="itemize"><span class="colorbox" id="colorbox318"><a id="dx1-127003"></a><span class="ectt-1095">integrateTol</span></span> — <span class="phvro8t-x-x-109">Tolerance for free energy integration</span><br class="newline"/>Default: 1e-6 <!--l. 5956--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     decimal, context: <span class="ectt-1095">abf</span> <!--l. 5956--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>The conjugate gradients algorithm used to integrate the free energy surface is stopped when the
     RMS error reaches <span class="ectt-1095">integrateTol</span>.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox319"><a id="dx1-127004"></a><span class="ectt-1095">integrateMaxIterations</span></span> — <span class="phvro8t-x-x-109">Maximum iterations for free energy integration</span><br class="newline"/>Default: 10000 <!--l. 5967--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> integer,
     context: <span class="ectt-1095">abf</span> <!--l. 5967--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>The conjugate gradients algorithm used to integrate the free energy surface is stopped when the
     number of iterations reaches <span class="ectt-1095">integrateMaxIterations</span>, unless the RMS error has reached
     <span class="ectt-1095">integrateTol </span>before. </li></ul>
<!--l. 5971--><p class="indent">   In dimension 4 or greater, integrating the discretized gradient becomes non-trivial. The
standalone utility <span class="ectt-1095">abf_integrate </span>is provided to perform that task. Because 4D ABF calculations
are uncommon, <span class="phvb8t-x-x-109">this tool is practically deprecated </span>by the Poisson integration described
above.
</p><!--l. 5976--><p class="indent"> <span class="ectt-1095">abf_integrate </span>reads the gradient data and uses it to perform a Monte-Carlo (M-C) simulation in
discretized collective variable space (specifically, on the same grid used by ABF to discretize the free
energy gradient). By default, a history-dependent bias (similar in spirit to metadynamics) is used: at each
M-C step, the bias at the current position is incremented by a preset amount (the <span class="phvro8t-x-x-109">hill height</span>). Upon
convergence, this bias counteracts optimally the underlying gradient; it is negated to obtain the estimate
of the free energy surface.
</p><!--l. 5985--><p class="indent"> <span class="ectt-1095">abf_integrate </span>is invoked using the command-line:<br class="newline"/><span class="ectt-1000">abf_integrate &lt;gradient_file&gt; [-n &lt;nsteps&gt;] [-t &lt;temp&gt;] [-m (0|1)] [-h &lt;hill_height&gt;] [-f</span>
<span class="ectt-1000">&lt;factor&gt;]</span>
</p><!--l. 5990--><p class="indent">   The gradient file name is provided first, followed by other parameters in any order. They are described
below, with their default value in square brackets: </p>
<ul class="itemize1">
<li class="itemize"><span class="ectt-1095">-n</span>: number of M-C steps to be performed; by default, a minimal number of steps is chosen
     based on the size of the grid, and the integration runs until a convergence criterion is satisfied
     (based on the RMSD between the target gradient and the real PMF gradient)
     </li>
<li class="itemize"><span class="ectt-1095">-t</span>:  temperature  for  M-C  sampling;  expects  kcal/mol  as  energy  unit  (unrelated  to  the
     simulation temperature) [500 K]
     </li>
<li class="itemize"><span class="ectt-1095">-s</span>: scaling factor for the gradients; when using a histogram of total forces obtained from
     <a href="#colvar|outputTotalForce"><span class="ectt-1095">outputTotalForce</span></a> or the <span class="ectt-1095">.force </span>file written by <a href="#colvarbias|writeTISamples"><span class="ectt-1095">writeTISamples</span></a>, a scaling factor of -1 should
     be used [1.0]
     </li>
<li class="itemize"><span class="ectt-1095">-m</span>: use metadynamics-like biased sampling? (0 = false) [1]
     </li>
<li class="itemize"><span class="ectt-1095">-h</span>: increment for the history-dependent bias (“hill height") [0.01 kcal/mol]
                                                                                         
                                                                                         
     </li>
<li class="itemize"><span class="ectt-1095">-f</span>: if non-zero, this factor is used to scale the increment stepwise in the second half of the
     M-C sampling to refine the free energy estimate [0.5]</li></ul>
<!--l. 6006--><p class="indent">   Using the default values of all parameters should give reasonable results in most cases.
</p><!--l. 6009--><p class="indent"> <span class="ectt-1095">abf_integrate </span>produces the following output files: </p>
<ul class="itemize1">
<li class="itemize"><span class="ectt-1095">&lt;gradient_file&gt;.pmf</span>: computed free energy surface
     </li>
<li class="itemize"><span class="ectt-1095">&lt;gradient_file&gt;.histo</span>: histogram of M-C sampling (not usable in a straightforward way if
     the history-dependent bias has been applied)
     </li>
<li class="itemize"><span class="ectt-1095">&lt;gradient_file&gt;.est</span>: estimated gradient of the calculated free energy surface (from finite
     differences)
     </li>
<li class="itemize"><span class="ectt-1095">&lt;gradient_file&gt;.dev</span>:  deviation  between  the  user-provided  numerical  gradient  and  the
     actual gradient of the calculated free energy surface. The RMS norm of this vector field is
     used as a convergence criterion and output periodically during integration.</li></ul>
<!--l. 6022--><p class="indent"> <span class="phvb8t-x-x-109">Note: </span>Typically, the “deviation" vector field does not vanish as the integration converges. This happens
because the numerical estimate of the gradient does not exactly derive from a potential, due to numerical
approximations used to obtain it (finite sampling and discretization on a grid). See Ref.<span class="cite">[<a href="#XHenin2021">28</a>]</span> for
details.
</p>
<div class="flushright">
<!--l. 6030--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 6030--><p class="noindent"> </p><div class="heading"> <a id="sec:colvarbias_abf_extended"></a>
<h4 class="subsectionHead"><span class="titlemark">6.4    </span> <a id="sec:colvarbias_abf_extended"></a>Extended-system Adaptive Biasing Force (eABF)</h4>
<!--l. 6030--><p class="noindent"></p></div>
<!--l. 6032--><p class="indent">   Extended-system ABF (eABF) is a variant of ABF (<a href="#sec:colvarbias_abf">6.3<!--tex4ht:ref: sec:colvarbias_abf --></a>) where the bias is not applied
directly to the collective variable, but to an extended coordinate (“fictitious variable")
<!--l. 6035--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>λ</mi></math> that
evolves dynamically according to Newtonian or Langevin dynamics. Such an extended coordinate is
enabled for a given colvar using the <span class="ectt-1095">extendedLagrangian </span>and associated keywords (<a href="#sec:colvar_extended">4.24<!--tex4ht:ref: sec:colvar_extended --></a>). The theory of
eABF and the present implementation are documented in detail in reference <span class="cite">[<a href="#XLesage2017">29</a>]</span>.
</p><!--l. 6041--><p class="indent">   Defining an ABF bias on a colvar wherein the <span class="ectt-1095">extendedLagrangian </span>option is active will perform eABF
automatically; there is no dedicated option.
                                                                                         
                                                                                         
</p><!--l. 6044--><p class="indent">   The extended variable <!--l. 6044--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>λ</mi></math> is
coupled to the colvar <!--l. 6044--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi> <mo class="MathClass-rel">=</mo> <mi>ξ</mi><mo class="MathClass-open">(</mo><mi>q</mi><mo class="MathClass-close">)</mo></math> by the
harmonic potential <!--l. 6045--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">(</mo><mi>k</mi><mo class="MathClass-bin">∕</mo><mn>2</mn><mo class="MathClass-close">)</mo><msup><mrow><mo class="MathClass-open">(</mo><mi>z</mi><mo class="MathClass-bin">−</mo><mi>λ</mi><mo class="MathClass-close">)</mo></mrow><mrow><mn>2</mn></mrow></msup></math>. Under eABF
dynamics, the adaptive bias on <!--l. 6046--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>λ</mi></math>
is the running estimate of the average spring force: </p><table class="equation"><tr><td> <a id="x1-128001r31"></a>
<!--l. 6048--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<msup><mrow><mi>F</mi></mrow><mrow><mstyle mathvariant="normal"><mi>b</mi><mi>i</mi><mi>a</mi><mi>s</mi></mstyle></mrow></msup><mo class="MathClass-open">(</mo><msup><mrow><mi>λ</mi></mrow><mrow><mo class="MathClass-bin">∗</mo></mrow></msup><mo class="MathClass-close">)</mo> <mo class="MathClass-rel">=</mo><msub><mrow> <mrow><mo fence="true" form="prefix"> ⟨</mo><mrow><mi>k</mi><mo class="MathClass-open">(</mo><mi>λ</mi> <mo class="MathClass-bin">−</mo><mi>z</mi><mo class="MathClass-close">)</mo></mrow><mo fence="true" form="postfix">⟩</mo></mrow></mrow><mrow><msup><mrow>
<mi>λ</mi></mrow><mrow><mo class="MathClass-bin">∗</mo></mrow></msup></mrow></msub>
</math></td><td class="eq-no">(31)</td></tr></table>
<!--l. 6051--><p class="indent">   where the angle brackets indicate a canonical average conditioned by
<!--l. 6051--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>λ</mi> <mo class="MathClass-rel">=</mo> <msup><mrow><mi>λ</mi></mrow><mrow><mo class="MathClass-bin">∗</mo></mrow></msup></math>.
At long simulation times, eABF produces a flat histogram of the extended variable
<!--l. 6052--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>λ</mi></math>, and a flattened
histogram of <!--l. 6053--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>,
whose exact shape depends on the strength of the coupling as defined by <span class="ectt-1095">extendedFluctuation </span>in the
colvar.
</p><!--l. 6056--><p class="indent">   Coupling should be strong enough that the bias helps overcome barriers along the colvar.<span class="cite">[<a href="#XLesage2017">29</a>]</span> A value
of <span class="ectt-1095">extendedFluctuation </span>equal to the bin width is often reasonable. In the limit of strong coupling (small
<span class="ectt-1095">extendedFluctuation</span>), the eABF free energy surface (<span class="ectt-1095">.pmf </span>file) becomes close to the true one, making
the asymptotically more accurate but noisier CZAR estimator (<span class="ectt-1095">.czar.pmf </span>file) unnecessary. In
practice, we recommend comparing these two free energy surfaces to decide which one to
use.
</p><!--l. 6063--><p class="indent">   Distribution of the colvar may be assessed by plotting its histogram, which is written to
the <span class="phvro8t-x-x-109">outputName</span><span class="ectt-1095">.zcount </span>file in every eABF simulation. Note that a <span class="ectt-1095">histogram </span>bias (<a href="#sec:colvarbias_histogram">6.13<!--tex4ht:ref: sec:colvarbias_histogram --></a>)
applied to an extended-Lagrangian colvar will access the extended degree of freedom
<!--l. 6067--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>λ</mi></math>, not the original
colvar <!--l. 6067--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>;
however, the joint histogram may be explicitly requested by listing the name of the colvar twice in a row
within the <span class="ectt-1095">colvars </span>parameter of the <span class="ectt-1095">histogram </span>block.
</p><!--l. 6071--><p class="indent">   The eABF PMF is that of the coordinate <!--l. 6071--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>λ</mi></math>,
it is not exactly the free energy profile of <!--l. 6071--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>.
That quantity can be calculated based on either the CZAR estimator or the Zheng/Yang estimator.
</p>
<div class="flushright">
<!--l. 6076--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 6076--><p class="noindent"> </p><div class="heading"> <a id="sec:colvarbias_ti_ext_czar"></a>
<h5 class="subsubsectionHead"><span class="titlemark">6.4.1    </span> <a id="sec:colvarbias_ti_ext_czar"></a>CZAR estimator of the free energy</h5>
<!--l. 6076--><p class="noindent"></p></div>
<!--l. 6078--><p class="indent">   The <span class="phvro8t-x-x-109">corrected z-averaged restraint </span>(CZAR) estimator is described in detail in reference <span class="cite">[<a href="#XLesage2017">29</a>]</span>. It is
computed automatically in eABF simulations, regardless of the number of colvars involved. Note that ABF
may also be applied on a combination of extended and non-extended colvars; in that case, CZAR still
provides an unbiased estimate of the free energy gradient.
</p><!--l. 6085--><p class="indent">   CZAR estimates the free energy gradient as: </p><table class="equation"><tr><td> <a id="eq:czar"></a>
<!--l. 6086--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<msup><mrow><mi>A</mi></mrow><mrow><mi>′</mi></mrow></msup><mo class="MathClass-open">(</mo><mi>z</mi><mo class="MathClass-close">)</mo> <mo class="MathClass-rel">=</mo> <mo class="MathClass-bin">−</mo><mfrac><mrow><mn>1</mn></mrow>
<mrow><mi>β</mi></mrow></mfrac> <mfrac><mrow><mi>d</mi><mi class="qopname">ln</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--><mover accent="true"><mrow><mi>ρ</mi></mrow><mo accent="true">~</mo></mover><mo class="MathClass-open">(</mo><mi>z</mi><mo class="MathClass-close">)</mo></mrow>
<mrow><mi>d</mi><mi>z</mi></mrow></mfrac> <mo class="MathClass-bin">+</mo><mi>k</mi><mo class="MathClass-open">(</mo><msub><mrow><mo class="MathClass-open">⟨</mo><mi>λ</mi><mo class="MathClass-close">⟩</mo></mrow><mrow><mi>z</mi></mrow></msub><mo class="MathClass-bin">−</mo><mi>z</mi><mo class="MathClass-close">)</mo><mo class="MathClass-punc">.</mo>
</math></td><td class="eq-no">(32)</td></tr></table>
<!--l. 6090--><p class="indent">   where <!--l. 6090--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi> <mo class="MathClass-rel">=</mo> <mi>ξ</mi><mo class="MathClass-open">(</mo><mi>q</mi><mo class="MathClass-close">)</mo></math> is the colvar,
<!--l. 6090--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>λ</mi></math> is the extended variable
harmonically coupled to <!--l. 6091--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math>
with a force constant <!--l. 6091--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>k</mi></math>,
and <!--l. 6091--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mover accent="true"><mrow><mi>ρ</mi></mrow><mo accent="true">~</mo></mover><mo class="MathClass-open">(</mo><mi>z</mi><mo class="MathClass-close">)</mo></math> is the observed
distribution (histogram) of <!--l. 6092--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math>,
affected by the eABF bias.
</p><!--l. 6094--><p class="indent">   Parameters for the CZAR estimator are: </p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox320"><a id="dx1-129002"></a><span class="ectt-1095">CZARestimator</span></span> — <span class="phvro8t-x-x-109">Calculate CZAR estimator of the free energy?</span><br class="newline"/>Default: <span class="ectt-1095">yes</span> <!--l. 6101--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">abf</span> <!--l. 6101--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option is only available when ABF is performed on extended-Lagrangian colvars. When
     enabled, it triggers calculation of the free energy following the CZAR estimator.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox321"><a id="dx1-129003"></a><span class="ectt-1095">writeCZARwindowFile</span></span> — <span class="phvro8t-x-x-109">Write internal data from CZAR to a separate file?</span><br class="newline"/>Default: <span class="ectt-1095">no</span> <!--l. 6122--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">abf</span> <!--l. 6122--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>When this option is enabled, eABF simulations will write a file containing the <!--l. 6122--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math>-averaged
     restraint force under the name <span class="phvro8t-x-x-109">outputName</span><span class="ectt-1095">.zgrad</span>. The same information is always included
     in the colvars state file, which is sufficient for restarting an eABF simulation. These separate
     file is only useful when joining adjacent windows from a stratified eABF simulation, either to
     continue the simulation in a broader window or to compute a CZAR estimate of the PMF over
     the full range of the coordinate(s). <span class="phvb8t-x-x-109">Important warning. </span>Unbiased free-energy estimators
     from  eABF  dynamics  rely  on  some  form  of  sampling  histogram.  When  running  stratified
                                                                                         
                                                                                         
     (windowed)  calculations  this  histogram  becomes  discontinuous,  and  as  a  result  the  free
     energy gradient estimated by CZAR is inaccurate at the window boundary, resulting in visible
     "blips" in the PMF. As a workaround, we recommend manually replacing the two free energy
     gradient values at the boundary, either with the ABF values from .grad files (accurate in the
     limit of tight coupling), or with values interpolated for the neighboring values of the CZAR
     gradient. </li></ul>
<!--l. 6125--><p class="indent">   Similar to ABF, the CZAR estimator produces two output files in multicolumn text format (<a href="#sec:colvar_multicolumn_grid">3.8.5<!--tex4ht:ref: sec:colvar_multicolumn_grid --></a>):
</p>
<ul class="itemize1">
<li class="itemize"><span class="phvro8t-x-x-109">outputName</span><span class="ectt-1095">.czar.grad</span>: current estimate of the free energy gradient (grid), in multicolumn;
     </li>
<li class="itemize"><span class="phvro8t-x-x-109">outputName</span><span class="ectt-1095">.czar.pmf</span>: only for one-dimensional calculations, integrated free energy profile
     or PMF.</li></ul>
<!--l. 6132--><p class="noindent">The sampling histogram associated with the CZAR estimator is the
<!--l. 6132--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math>-histogram,
which is written in the file <span class="phvro8t-x-x-109">outputName</span><span class="ectt-1095">.zcount</span>.
</p>
<div class="flushright">
<!--l. 6203--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 6203--><p class="noindent"> </p><div class="heading"> <a id="sec:colvarbias_ti_ext_zheng_yang"></a>
<h5 class="subsubsectionHead"><span class="titlemark">6.4.2    </span> <a id="sec:colvarbias_ti_ext_zheng_yang"></a>Zheng/Yang estimator of the free energy</h5>
<!--l. 6203--><p class="noindent"></p></div> This feature has been contributed to NAMD by the following authors:
<!--l. 6203--><p class="indent">
</p><blockquote class="quote">
<!--l. 6203--><p class="noindent">Haohao Fu and Christophe Chipot<br class="newline"/>Laboratoire International Associé Centre National de la Recherche Scientifique et
     University of Illinois at Urbana--Champaign, <br class="newline"/>Unité Mixte de Recherche No. 7565, Université de Lorraine, <br class="newline"/>B.P. 70239, 54506 Vandœuvre-lès-Nancy cedex, France</p></blockquote>
<!--l. 6203--><p class="indent"> <span class="phvr8c-x-x-109">©</span> 2016, <span class="phvrc8t-x-x-109">C<span class="small-caps">E</span><span class="small-caps">N</span><span class="small-caps">T</span><span class="small-caps">R</span><span class="small-caps">E</span> N<span class="small-caps">A</span><span class="small-caps">T</span><span class="small-caps">I</span><span class="small-caps">O</span><span class="small-caps">N</span><span class="small-caps">A</span><span class="small-caps">L</span> <span class="small-caps">D</span><span class="small-caps">E</span> <span class="small-caps">L</span><span class="small-caps">A</span> R<span class="small-caps">E</span><span class="small-caps">C</span><span class="small-caps">H</span><span class="small-caps">E</span><span class="small-caps">R</span><span class="small-caps">C</span><span class="small-caps">H</span><span class="small-caps">E</span> S<span class="small-caps">C</span><span class="small-caps">I</span><span class="small-caps">E</span><span class="small-caps">N</span><span class="small-caps">T</span><span class="small-caps">I</span><span class="small-caps">fi</span><span class="small-caps">Q</span><span class="small-caps">U</span><span class="small-caps">E</span></span>
</p><!--l. 6203--><p class="indent">   This implementation is fully documented in <span class="cite">[<a href="#XFu2016">30</a>]</span>. The Zheng and Yang estimator <span class="cite">[<a href="#XZheng2012">31</a>]</span> is based on
Umbrella Integration <span class="cite">[<a href="#XKastner2005">32</a>]</span>. The free energy gradient is estimated as :
</p>
<table class="equation"><tr><td> <a id="x1-130001r33"></a>
<!--l. 6203--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<msup><mrow><mi>A</mi></mrow><mrow><mi>′</mi></mrow></msup><mo class="MathClass-open">(</mo><msup><mrow><mi>ξ</mi></mrow><mrow><mo class="MathClass-bin">∗</mo></mrow></msup><mo class="MathClass-close">)</mo> <mo class="MathClass-rel">=</mo> <mfrac><mrow><munder class="msub"><mrow><mo>∑</mo></mrow><mrow><mi>λ</mi></mrow></munder><mi>N</mi><mo class="MathClass-open">(</mo><msup><mrow><mi>ξ</mi></mrow><mrow><mo class="MathClass-bin">∗</mo></mrow></msup><mo class="MathClass-punc">,</mo><mi>λ</mi><mo class="MathClass-close">)</mo> <mrow><mo fence="true" form="prefix"> [</mo><mrow><mfrac><mrow><mo class="MathClass-open">(</mo><msup><mrow><mi>ξ</mi></mrow><mrow><mo class="MathClass-bin">∗</mo></mrow></msup><mo class="MathClass-bin">−</mo><msub><mrow><mo class="MathClass-open">⟨</mo><mi>ξ</mi><mo class="MathClass-close">⟩</mo></mrow><mrow>
<mi>λ</mi></mrow></msub><mo class="MathClass-close">)</mo></mrow>
<mrow><mi>β</mi><msubsup><mrow><mi>σ</mi></mrow><mrow><mi>λ</mi></mrow><mrow><mn>2</mn></mrow></msubsup></mrow></mfrac> <mo class="MathClass-bin">−</mo><mi>k</mi><mo class="MathClass-open">(</mo><msup><mrow><mi>ξ</mi></mrow><mrow><mo class="MathClass-bin">∗</mo></mrow></msup><mo class="MathClass-bin">−</mo><mi>λ</mi><mo class="MathClass-close">)</mo></mrow><mo fence="true" form="postfix">]</mo></mrow></mrow>
<mrow><munder class="msub"><mrow><mo>∑</mo></mrow><mrow><mi>λ</mi></mrow></munder><mi>N</mi><mo class="MathClass-open">(</mo><msup><mrow><mi>ξ</mi></mrow><mrow><mo class="MathClass-bin">∗</mo></mrow></msup><mo class="MathClass-punc">,</mo><mi>λ</mi><mo class="MathClass-close">)</mo></mrow></mfrac>
</math></td><td class="eq-no">(33)</td></tr></table>
<!--l. 6203--><p class="indent">   where <!--l. 6203--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math> is the colvar,
<!--l. 6203--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>λ</mi></math> is the extended variable
harmonically coupled to <!--l. 6203--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>
with a force constant <!--l. 6203--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>k</mi></math>,
<!--l. 6203--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>N</mi><mo class="MathClass-open">(</mo><mi>ξ</mi><mo class="MathClass-punc">,</mo><mi>λ</mi><mo class="MathClass-close">)</mo></math> is the number of samples collected
in a <!--l. 6203--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">(</mo><mi>ξ</mi><mo class="MathClass-punc">,</mo><mi>λ</mi><mo class="MathClass-close">)</mo></math> bin, which is assumed
to be a Gaussian function of <!--l. 6203--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>
with mean <!--l. 6203--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mo class="MathClass-open">⟨</mo><mi>ξ</mi><mo class="MathClass-close">⟩</mo></mrow><mrow><mi>λ</mi></mrow></msub></math> and
standard deviation <!--l. 6203--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>σ</mi></mrow><mrow><mi>λ</mi></mrow></msub></math>.
</p><!--l. 6203--><p class="indent">   The estimator is enabled through the following option: </p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox322"><a id="dx1-130002"></a><span class="ectt-1095">UIestimator</span></span> — <span class="phvro8t-x-x-109">Calculate UI estimator of the free energy?</span><br class="newline"/>Default: <span class="ectt-1095">no</span> <!--l. 6203--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">abf</span> <!--l. 6203--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option is only available when ABF is performed on extended-Lagrangian colvars. When
     enabled, it triggers calculation of the free energy following the UI estimator.</li></ul>
<!--l. 6203--><p class="noindent"><span class="likeparagraphHead"><a id="x1-1310006.4.2"></a>Usage for multiple--replica eABF.</span>
   The eABF algorithm can be associated with a multiple--walker strategy <span class="cite">[<a href="#XMinoukadeh2010">25</a>, <a href="#XComer2014c">26</a>]</span> (<a href="#sec:colvarbias_abf_shared">6.3.3<!--tex4ht:ref: sec:mw-ABF --></a>). To run a
multiple--replica eABF simulation, start a multiple-replica NAMD run (option <span class="ectt-1095">+replicas</span>) and set <span class="ectt-1095">shared</span>
<span class="ectt-1095">on </span>in the Colvars config file to enable the multiple--walker ABF algorithm. It should be noted that in
contrast with classical MW--ABF simulations, the output files of an MW--eABF simulation only show the
free energy estimate of the corresponding replica.
</p><!--l. 6203--><p class="indent">   One can merge the results, using <span class="ectt-1095">./eabf.tcl -mergemwabf [merged_filename] [eabf_output1]</span>
<span class="ectt-1095">[eabf_output2] ...</span>, e.g., <span class="ectt-1095">./eabf.tcl -mergemwabf merge.eabf eabf.0.UI eabf.1.UI eabf.2.UI</span>
<span class="ectt-1095">eabf.3.UI</span>.
</p><!--l. 6203--><p class="indent">   If one runs an ABF--based calculation, breaking the reaction pathway into several non--overlapping
windows, one can use <span class="ectt-1095">./eabf.tcl -mergesplitwindow [merged_fileprefix] [eabf_output]</span>
<span class="ectt-1095">[eabf_output2] ... </span>to merge the data accrued in these non--overlapping windows. This option can be
utilized in both eABF and classical ABF simulations, e.g., <span class="ectt-1095">./eabf.tcl -mergesplitwindow merge</span>
<span class="ectt-1095">window0.czar window1.czar window2.czar window3.czar</span>, <span class="ectt-1095">./eabf.tcl -mergesplitwindow merge</span>
<span class="ectt-1095">window0.UI window1.UI window2.UI window3.UI </span>or <span class="ectt-1095">./eabf.tcl -mergesplitwindow merge abf0 abf1</span>
<span class="ectt-1095">abf2 abf3</span>.
</p>
<div class="flushright">
<!--l. 6207--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 6207--><p class="noindent"> </p><div class="heading"> <a id="sec:colvarbias_abmd"></a>
<h4 class="subsectionHead"><span class="titlemark">6.5    </span> <a id="sec:colvarbias_abmd"></a>Adiabatic Bias Molecular Dynamics (ABMD)</h4>
<!--l. 6207--><p class="noindent"></p></div>
<!--l. 6209--><p class="indent">   This implements the Adiabatic Bias Molecular Dynamics (ABMD) method of Marchi and Ballone <span class="cite">[<a href="#XMarchi1999">33</a>]</span>,
sometimes referred to as <span class="phvro8t-x-x-109">ratchet-and-pawl </span>or <span class="phvro8t-x-x-109">ratcheted MD</span>. ABMD is a non-equilibrium process that
enhances the motion of a scalar colvar in a given direction. For simplicity, the case of an increasing value
is described below, but enhancing downward motion of the variable is also supported via the <a href="#abmd|decreasing"><span class="ectt-1095">decreasing</span></a>
flag.
</p><!--l. 6213--><p class="indent">   ABMD does not directly push the variable forward, but prevents it from backtracking by applying a time-dependent
half-harmonic potential <!--l. 6213--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>V</mi></mrow><mrow><mi>t</mi></mrow></msub></math>,
the center of which is the highest value attained by the variable so far (its high-water mark). This design
implies that the bias is conservative at all times and therefore exerts zero net work, hence the “adiabatic"
qualifier:
</p>
<table class="equation"><tr><td> <a id="eq:colvarbias_abmd"></a>
<!--l. 6216--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow>
<mi>V</mi></mrow><mrow><mi>t</mi></mrow></msub><mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>t</mi></mrow></msub><mo class="MathClass-close">)</mo> <mo class="MathClass-rel">=</mo> <mrow><mo fence="true" form="prefix"> {</mo><mrow> <mtable align="axis" class="array" columnlines="none" equalcolumns="false" equalrows="false" style=""> <mtr><mtd class="array" columnalign="left"> <mfrac><mrow><mn>1</mn></mrow>
<mrow><mn>2</mn></mrow></mfrac><mi>k</mi><msup><mrow> <mrow><mo fence="true" form="prefix"> (</mo><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>t</mi></mrow></msub><mo class="MathClass-bin">−</mo><msubsup><mrow><mi>ξ</mi></mrow><mrow><mi>t</mi></mrow><mrow><mstyle mathvariant="normal"><mi>r</mi><mi>e</mi><mi>f</mi></mstyle></mrow></msubsup></mrow><mo fence="true" form="postfix">)</mo></mrow></mrow><mrow><mn>2</mn></mrow></msup> </mtd><mtd class="array" columnalign="left"> <mstyle class="text"><mtext>if </mtext></mstyle><mspace class="nbsp" width="0.33em"></mspace><msub><mrow><mi>ξ</mi></mrow><mrow><mi>t</mi></mrow></msub> <mo class="MathClass-rel">&lt;</mo> <msubsup><mrow><mi>ξ</mi></mrow><mrow><mi>t</mi></mrow><mrow><mstyle mathvariant="normal"><mi>r</mi><mi>e</mi><mi>f</mi></mstyle></mrow></msubsup> </mtd>
</mtr> <mtr><mtd class="array" columnalign="left"> <mn>0</mn> </mtd><mtd class="array" columnalign="left"> <mstyle class="text"><mtext>otherwise</mtext></mstyle> </mtd></mtr> <!--l l--></mtable> </mrow><mo fence="true" form="postfix"></mo></mrow>
</math></td><td class="eq-no">(34)</td></tr></table>
<!--l. 6225--><p class="noindent">where <!--l. 6225--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msubsup><mrow><mi>ξ</mi></mrow><mrow><mi>t</mi></mrow><mrow><mstyle mathvariant="normal"><mi>r</mi><mi>e</mi><mi>f</mi></mstyle></mrow></msubsup></math> is the high-water
mark at time <!--l. 6225--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>t</mi></math>, bounded by a
user-defined stopping value <!--l. 6225--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow><mi>ξ</mi></mrow><mrow><mstyle mathvariant="normal"><mi>s</mi><mi>t</mi><mi>o</mi><mi>p</mi></mstyle></mrow></msup></math>:
</p>
<table class="equation"><tr><td> <a id="x1-132002r35"></a>
<!--l. 6227--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<msubsup><mrow><mi>ξ</mi></mrow><mrow><mi>t</mi></mrow><mrow><mstyle mathvariant="normal"><mi>r</mi><mi>e</mi><mi>f</mi></mstyle></mrow></msubsup> <mo class="MathClass-rel">=</mo><mi class="qopname"> min</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo> <mrow><mo fence="true" form="prefix"> (</mo><mrow><munderover accent="false" accentunder="false"><mrow><mi class="qopname">max</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo></mrow><mrow>
<mi>s</mi><mo class="MathClass-rel">=</mo><mn>0</mn></mrow><mrow><mi>t</mi></mrow></munderover><msub><mrow><mi>ξ</mi></mrow><mrow>
<mi>s</mi></mrow></msub><mo class="MathClass-punc">,</mo><msup><mrow><mi>ξ</mi></mrow><mrow><mstyle mathvariant="normal"><mi>s</mi><mi>t</mi><mi>o</mi><mi>p</mi></mstyle></mrow></msup></mrow><mo fence="true" form="postfix">)</mo></mrow><mo class="MathClass-punc">.</mo>
</math></td><td class="eq-no">(35)</td></tr></table>
<!--l. 6231--><p class="indent"> <span class="phvro8t-x-x-109">Note: </span>because the ABMD potential in eq. <a href="#eq:colvarbias_abmd">34<!--tex4ht:ref: eq:colvarbias_abmd --></a> is never defined for more than one variable, no internal unit conversion
is applied to <!--l. 6231--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>k</mi></math>:
this behavior is different from other restraints available in Colvars, such as the harmonic wall restraints in
<a href="#sec:colvarbias_harmonic_walls">6.10<!--tex4ht:ref: sec:colvarbias_harmonic_walls --></a>.
</p><!--l. 6233--><p class="indent">   Besides the name of the biased variable specified by the <span class="ectt-1095">colvars </span>keyword, the tunable parameters of ABMD are the
force constant <!--l. 6233--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>k</mi></math> and
the stopping value <!--l. 6233--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow><mi>ξ</mi></mrow><mrow><mstyle mathvariant="normal"><mi>s</mi><mi>t</mi><mi>o</mi><mi>p</mi></mstyle></mrow></msup></math>,
set by the following user keywords:
</p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox323"><a id="dx1-132003"></a><span class="ectt-1095">forceConstant</span></span> — <span class="phvro8t-x-x-109">Force constant for the half-harmonic bias</span><br class="newline"/><!--l. 6240--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     decimal, context: <span class="ectt-1095">abmd</span> <!--l. 6240--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>The value of <!--l. 6240--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>k</mi></math>.
     Stronger values result if faster motion of the coordinate, along more biased pathways.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox324"><a id="dx1-132004"></a><span class="ectt-1095">stoppingValue</span></span> — <span class="phvro8t-x-x-109">Value beyong which the reference value is not updated</span><br class="newline"/><!--l. 6246--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> decimal,
     context: <span class="ectt-1095">abmd</span> <!--l. 6246--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This sets the value of <!--l. 6246--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow><mi>ξ</mi></mrow><mrow><mstyle mathvariant="normal"><mi>s</mi><mi>t</mi><mi>o</mi><mi>p</mi></mstyle></mrow></msup></math>.
     It is useful to prevent the variable from reaching unrealistic values after the relevant range
     has been covered.
     </li>
<li class="itemize"><a id="abmd|decreasing"></a> <span class="colorbox" id="colorbox325"><a id="dx1-132005"></a><span class="ectt-1095">decreasing</span></span> — <span class="phvro8t-x-x-109">Should the variable decrease over time?</span><br class="newline"/>Default: no <!--l. 6255--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">abmd</span> <!--l. 6255--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This is used to make ABMD enhance motion toward lower colvar values. When enabled, it
     reverses the shape of the half-harmonic potential, and makes the reference value follow the
     minimum of the variable.
     </li></ul>
<!--l. 6259--><p class="indent">   ABMD also supports the following common bias parameters: </p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox326"><a id="dx1-132006"></a><span class="ectt-1095">name</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">name</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox327"><a id="dx1-132007"></a><span class="ectt-1095">colvars</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">colvars</span></a> (biasing and analysis methods)
                                                                                         
                                                                                         
     </li>
<li class="itemize"><span class="colorbox" id="colorbox328"><a id="dx1-132008"></a><span class="ectt-1095">outputEnergy</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">outputEnergy</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox329"><a id="dx1-132009"></a><span class="ectt-1095">outputFreq</span></span> — same definition as <a href="#colvarbias|outputFreq"><span class="ectt-1095">outputFreq</span></a> (biasing and analysis methods)</li></ul>
<div class="flushright">
<!--l. 6269--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 6269--><p class="noindent"> </p><div class="heading"> <a id="sec:colvarbias_meta"></a>
<h4 class="subsectionHead"><span class="titlemark">6.6    </span> <a id="sec:colvarbias_meta"></a>Metadynamics</h4>
<!--l. 6269--><p class="noindent"></p></div>
<!--l. 6271--><p class="indent">   The metadynamics method uses a history-dependent potential <span class="cite">[<a href="#XLaio2002">34</a>]</span> that generalizes to
any type of colvars the conformational flooding <span class="cite">[<a href="#XGrubmuller1995">35</a>]</span> and local elevation <span class="cite">[<a href="#XHuber1994">36</a>]</span> methods,
originally formulated to use as colvars the principal components of a covariance matrix
or a set of dihedral angles, respectively. The metadynamics potential on the colvars
<!--l. 6272--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle> <mo class="MathClass-rel">=</mo> <mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mn>2</mn></mrow></msub><mo class="MathClass-punc">,</mo><mi class="MathClass-op">…</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo><mo class="MathClass-punc">,</mo><msub><mrow><mi>ξ</mi></mrow><mrow><msub><mrow><mi>N</mi></mrow><mrow><mstyle mathvariant="normal"><mi>c</mi><mi>v</mi></mstyle></mrow></msub></mrow></msub><mo class="MathClass-close">)</mo></math> is
defined as: </p><table class="equation"><tr><td> <a id="eq:colvars_meta_pot"></a>
<!--l. 6273--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<msub><mrow><mi>V</mi></mrow><mrow><mstyle mathvariant="normal"><mi>m</mi><mi>e</mi><mi>t</mi><mi>a</mi></mstyle></mrow></msub> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mi>t</mi></mrow><mo fence="true" form="postfix">)</mo></mrow></mrow><mo fence="true" form="postfix">)</mo></mrow><mspace class="thickpace" width="0.28em"></mspace> <mo class="MathClass-rel">=</mo> <mspace class="thickpace" width="0.28em"></mspace><munderover accent="false" accentunder="false"><mrow><mo>∑</mo></mrow><mrow><msup><mrow><mi>t</mi></mrow><mrow><mi>′</mi></mrow></msup><mo class="MathClass-rel">=</mo><mi>δ</mi><mi>t</mi><mo class="MathClass-punc">,</mo><mn>2</mn><mi>δ</mi><mi>t</mi><mo class="MathClass-punc">,</mo><mo>…</mo></mrow><mrow><msup><mrow><mi>t</mi></mrow><mrow><mi>′</mi></mrow></msup><mo class="MathClass-rel">&lt;</mo><mi>t</mi></mrow></munderover><mi>W</mi><mspace class="thinspace" width="0.22em"></mspace><munderover accent="false" accentunder="false"><mrow><mo>∏</mo></mrow><mrow>
<mi>i</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><msub><mrow><mi>N</mi></mrow><mrow><mstyle mathvariant="normal"><mi>c</mi><mi>v</mi></mstyle></mrow></msub></mrow></munderover><mi class="qopname"> exp</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mo class="MathClass-bin">−</mo><mfrac><mrow><msup><mrow><mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo><mo class="MathClass-bin">−</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-open">(</mo><msup><mrow><mi>t</mi></mrow><mrow><mi>′</mi></mrow></msup><mo class="MathClass-close">)</mo><mo class="MathClass-close">)</mo></mrow><mrow><mn>2</mn></mrow></msup></mrow>
<mrow><mn>2</mn><msubsup><mrow><mi>σ</mi></mrow><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow><mrow><mn>2</mn></mrow></msubsup></mrow></mfrac> </mrow><mo fence="true" form="postfix">)</mo></mrow><mstyle mathvariant="normal"><mo class="MathClass-punc">,</mo></mstyle>
</math></td><td class="eq-no">(36)</td></tr></table>
<!--l. 6282--><p class="indent">   where <!--l. 6282--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>V</mi></mrow><mrow><mstyle mathvariant="normal"><mi>m</mi><mi>e</mi><mi>t</mi><mi>a</mi></mstyle></mrow></msub></math>
is the history-dependent potential acting on the <span class="phvro8t-x-x-109">current  </span>values of the colvars
<!--l. 6282--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></math>,
and depends only parametrically on the <span class="phvro8t-x-x-109">previous </span>values of the colvars.
<!--l. 6283--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>V</mi></mrow><mrow><mstyle mathvariant="normal"><mi>m</mi><mi>e</mi><mi>t</mi><mi>a</mi></mstyle></mrow></msub></math> is constructed as a
sum of <!--l. 6283--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>N</mi></mrow><mrow><mstyle mathvariant="normal"><mi>c</mi><mi>v</mi></mstyle></mrow></msub></math>-dimensional
repulsive Gaussian “hills", whose height is a chosen energy constant
<!--l. 6283--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>W</mi></math>,
and whose centers are the previously explored configurations
<!--l. 6283--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle><mo class="MathClass-open">(</mo><mi>δ</mi><mi>t</mi><mo class="MathClass-close">)</mo><mo class="MathClass-punc">,</mo><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle><mo class="MathClass-open">(</mo><mn>2</mn><mi>δ</mi><mi>t</mi><mo class="MathClass-close">)</mo><mo class="MathClass-punc">,</mo><mi class="MathClass-op">…</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo></mrow><mo fence="true" form="postfix">)</mo></mrow></math>.
</p><!--l. 6285--><p class="indent">   During the simulation, the system evolves towards the nearest minimum of the “effective" potential of mean
force <!--l. 6285--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>Ã</mi><mo class="MathClass-open">(</mo><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle><mo class="MathClass-close">)</mo></math>,
which is the sum of the “real" underlying potential of mean force
<!--l. 6285--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>A</mi><mo class="MathClass-open">(</mo><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle><mo class="MathClass-close">)</mo></math> and the the
                                                                                         
                                                                                         
metadynamics potential, <!--l. 6285--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>V</mi></mrow><mrow><mstyle mathvariant="normal"><mi>m</mi><mi>e</mi><mi>t</mi><mi>a</mi></mstyle></mrow></msub><mo class="MathClass-open">(</mo><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle><mo class="MathClass-close">)</mo></math>.
Therefore, at any given time the probability of observing the configuration
<!--l. 6286--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle class="mbox"><mtext></mtext><mstyle class="math"><msup><mrow><mi>ξ</mi></mrow><mrow><mo class="MathClass-bin">∗</mo></mrow></msup></mstyle><mtext></mtext></mstyle></math> is proportional
to <!--l. 6286--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi class="qopname">exp</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mo class="MathClass-bin">−</mo><mi>Ã</mi><mo class="MathClass-open">(</mo><mstyle class="mbox"><mtext></mtext><mstyle class="math"><msup><mrow><mi>ξ</mi></mrow><mrow><mo class="MathClass-bin">∗</mo></mrow></msup></mstyle><mtext></mtext></mstyle><mo class="MathClass-close">)</mo><mo class="MathClass-bin">∕</mo><msub><mrow><mi>κ</mi></mrow><mrow><mstyle mathvariant="normal"><mi>B</mi></mstyle></mrow></msub><mi>T</mi> </mrow><mo fence="true" form="postfix">)</mo></mrow></math>:
this is also the probability that a new Gaussian “hill" is added at that configuration.
If the simulation is run for a sufficiently long time, each local minimum is canceled out
by the sum of the Gaussian “hills". At that stage the “effective" potential of mean force
<!--l. 6288--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>Ã</mi><mo class="MathClass-open">(</mo><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle><mo class="MathClass-close">)</mo></math> is constant, and
<!--l. 6288--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-bin">−</mo><msub><mrow><mi>V</mi></mrow><mrow><mstyle mathvariant="normal"><mi>m</mi><mi>e</mi><mi>t</mi><mi>a</mi></mstyle></mrow></msub><mo class="MathClass-open">(</mo><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle><mo class="MathClass-close">)</mo></math> is an estimator of the “real"
potential of mean force <!--l. 6288--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>A</mi><mo class="MathClass-open">(</mo><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle><mo class="MathClass-close">)</mo></math>,
save for an additive constant: </p><table class="equation"><tr><td> <a id="eq:colvars_meta_fes"></a>
<!--l. 6289--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>A</mi><mo class="MathClass-open">(</mo><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle><mo class="MathClass-close">)</mo><mspace class="thickpace" width="0.28em"></mspace> <mo class="MathClass-rel">≃</mo><mspace class="thickpace" width="0.28em"></mspace><mo class="MathClass-bin">−</mo><msub><mrow><mi>V</mi></mrow><mrow><mstyle mathvariant="normal"><mi>m</mi><mi>e</mi><mi>t</mi><mi>a</mi></mstyle></mrow></msub><mo class="MathClass-open">(</mo><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle><mo class="MathClass-close">)</mo><mo class="MathClass-bin">+</mo><mi>K</mi>
</math></td><td class="eq-no">(37)</td></tr></table>
<!--l. 6296--><p class="indent">   Such estimate of the free energy can be provided by enabling <span class="ectt-1095">writeFreeEnergyFile</span>.
Assuming that the set of collective variables includes all relevant degrees of freedom, the
predicted error of the estimate is a simple function of the correlation times of the colvars
<!--l. 6297--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>τ</mi></mrow><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow></msub></math>, and of the user-defined
parameters <!--l. 6297--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>W</mi></math>,
<!--l. 6297--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>σ</mi></mrow><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow></msub></math> and
<!--l. 6297--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>δ</mi><mi>t</mi></math>
<span class="cite">[<a href="#XBussi2006">37</a>]</span>. In typical applications, a good rule of thumb can be to choose the ratio
<!--l. 6298--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>W</mi><mo class="MathClass-bin">∕</mo><mi>δ</mi><mi>t</mi></math> much smaller
than <!--l. 6298--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>κ</mi></mrow><mrow><mstyle mathvariant="normal"><mi>B</mi></mstyle></mrow></msub><mi>T</mi> <mo class="MathClass-bin">∕</mo><msub><mrow><mi>τ</mi></mrow><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></mrow></msub></math>, where
<!--l. 6298--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>τ</mi></mrow><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></mrow></msub></math> is the longest
among <!--l. 6298--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></math>'s
correlation times: <!--l. 6298--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>σ</mi></mrow><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow></msub></math>
then dictates the resolution of the calculated PMF.
</p><!--l. 6300--><p class="indent">   If the metadynamics parameters are chosen correctly, after an equilibration time,
<!--l. 6300--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>t</mi></mrow><mrow><mi>e</mi></mrow></msub></math>,
the estimator provided by eq. <a href="#eq:colvars_meta_fes">37<!--tex4ht:ref: eq:colvars_meta_fes --></a> oscillates on time around the “real" free energy, thereby a
better estimate of the latter can be obtained as the time average of the bias potential after
<!--l. 6301--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>t</mi></mrow><mrow><mi>e</mi></mrow></msub></math>
<span class="cite">[<a href="#XMarinelli2009">38</a>, <a href="#XCrespo2010">39</a>]</span>: </p><table class="equation"><tr><td> <a id="eq:colvars_meta_fes_av"></a>
<!--l. 6302--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>A</mi> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></mrow><mo fence="true" form="postfix">)</mo></mrow><mspace class="thickpace" width="0.28em"></mspace> <mo class="MathClass-rel">=</mo> <mspace class="thickpace" width="0.28em"></mspace><mo class="MathClass-bin">−</mo> <mfrac><mrow><mn>1</mn></mrow>
<mrow><msub><mrow><mi>t</mi></mrow><mrow><mi>t</mi><mi>o</mi><mi>t</mi></mrow></msub><mo class="MathClass-bin">−</mo><msub><mrow><mi>t</mi></mrow><mrow><mi>e</mi></mrow></msub></mrow></mfrac><msubsup><mrow><mo>∫
 <!--nolimits--></mo><!--nolimits--></mrow><mrow><msub><mrow><mi>t</mi></mrow><mrow><mi>e</mi></mrow></msub></mrow><mrow><msub><mrow><mi>t</mi></mrow><mrow><mi>t</mi><mi>o</mi><mi>t</mi></mrow></msub>
</mrow></msubsup><msub><mrow><mi>V</mi></mrow><mrow><mstyle mathvariant="normal"><mi>m</mi><mi>e</mi><mi>t</mi><mi>a</mi></mstyle></mrow></msub><mo class="MathClass-open">(</mo><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle><mo class="MathClass-punc">,</mo><mi>t</mi><mo class="MathClass-close">)</mo><mi>d</mi><mi>t</mi>
</math></td><td class="eq-no">(38)</td></tr></table>
<!--l. 6308--><p class="indent">   where <!--l. 6308--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>t</mi></mrow><mrow><mi>e</mi></mrow></msub></math>
is the time after which the bias potential grows (approximately) evenly during the simulation and
<!--l. 6308--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>t</mi></mrow><mrow><mi>t</mi><mi>o</mi><mi>t</mi></mrow></msub></math> is the
total simulation time. The free energy calculated according to eq. <a href="#eq:colvars_meta_fes_av">38<!--tex4ht:ref: eq:colvars_meta_fes_av --></a> can thus be obtained averaging on
time multiple time-dependent free energy estimates, that can be printed out through the keyword
<span class="ectt-1095">keepFreeEnergyFiles</span>. An alternative is to obtain the free energy profiles by summing the hills
added during the simulation; the hills trajectory can be printed out by enabling the option
<span class="ectt-1095">writeHillsTrajectory</span>.
</p>
<div class="flushright">
<!--l. 6311--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 6311--><p class="noindent"> </p><div class="heading"> <a id="sec:colvarbias_meta_boundaries"></a>
<h5 class="subsubsectionHead"><span class="titlemark">6.6.1    </span> <a id="sec:colvarbias_meta_boundaries"></a>Treatment of the PMF boundaries</h5>
<!--l. 6311--><p class="noindent"></p></div>
<!--l. 6313--><p class="indent">   In typical scenarios the Gaussian hills of a metadynamics potential are interpolated and summed
together onto a grid, which is much more efficient than computing each hill independently at every
step (the keyword <a href="#metadynamics|useGrids"><span class="ectt-1095">useGrids</span></a> is <span class="ectt-1095">on </span>by default). This numerical approximation typically yields
negligible errors in the resulting PMF <span class="cite">[<a href="#XFiorin2013">1</a>]</span>. However, due to the finite thickness of the Gaussian
function, the metadynamics potential would suddenly vanish each time a variable exceeds its grid
boundaries.
</p><!--l. 6317--><p class="indent">   To avoid such discontinuity the Colvars metadynamics code will keep an explicit copy
of each hill that straddles a grid's boundary, and will use it to compute metadynamics
forces outside the grid. This measure is taken to protect the accuracy and stability of
a metadynamics simulation, except in cases of “natural" boundaries (for example, the
<!--l. 6318--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo><mn>0</mn> <mo class="MathClass-punc">:</mo> <mn>1</mn><mn>8</mn><mn>0</mn><mo class="MathClass-close">]</mo></math> interval
of an <span class="ectt-1095">angle </span>colvar) or when the flags <a href="#colvar|hardLowerBoundary"><span class="ectt-1095">hardLowerBoundary</span></a> and <a href="#colvar|hardUpperBoundary"><span class="ectt-1095">hardUpperBoundary</span></a> are explicitly set
by the user. Unfortunately, processing explicit hills alongside the potential and force grids
could easily become inefficient, slowing down the simulation and increasing the state file's
size.
</p><!--l. 6321--><p class="indent">   In general, it is a good idea to <span class="phvro8t-x-x-109">define a repulsive potential to avoid hills from coming too close to the</span>
<span class="phvro8t-x-x-109">grid's boundaries</span>, for example as a <span class="ectt-1095">harmonicWalls </span>restraint (see <a href="#sec:colvarbias_harmonic_walls">6.10<!--tex4ht:ref: sec:colvarbias_harmonic_walls --></a>).<br class="newline"/>
</p><!--l. 6323--><p class="noindent"><span class="phvb8t-x-x-109">Example: </span>Using harmonic walls to protect the grid's boundaries.</p>
<div class="mdframed" id="mdframed-62">
<!--l. 6324--><p class="noindent"></p><pre>colvar {<br class="newline"/>  name r<br class="newline"/>  distance { ... }<br class="newline"/>  upperBoundary 15.0<br class="newline"/>  width 0.2<br class="newline"/>}<br class="newline"/><br class="newline"/>metadynamics {<br class="newline"/>  name meta_r<br class="newline"/>  colvars r<br class="newline"/>  hillWeight 0.001<br class="newline"/>  hillWidth 2.0<br class="newline"/>}<br class="newline"/><br class="newline"/>harmonicWalls {<br class="newline"/>  name wall_r<br class="newline"/>  colvars r<br class="newline"/>  upperWalls 13.0<br class="newline"/>  upperWallConstant 2.0<br class="newline"/>} </pre><!--l. 6345--><p class="noindent"></p>
</div>
<!--l. 6347--><p class="indent">   In the colvar <span class="ectt-1095">r</span>, the <span class="ectt-1095">distance </span>function used has a <span class="ectt-1095">lowerBoundary </span>automatically set to 0 by
default, thus the keyword <span class="ectt-1095">lowerBoundary </span>itself is not mandatory and <span class="ectt-1095">hardLowerBoundary</span>
is set to <span class="ectt-1095">yes </span>internally. However, <span class="ectt-1095">upperBoundary </span>does not have such a “natural" choice of
value. The metadynamics potential <span class="ectt-1095">meta_r </span>will individually process any hill whose center is too
close to the <span class="ectt-1095">upperBoundary</span>, more precisely within fewer grid points than 6 times the Gaussian
<!--l. 6349--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>σ</mi></math>
parameter plus one. It goes without saying that if the colvar <span class="ectt-1095">r </span>represents a distance between two
freely-moving molecules, it will cross this “threshold" rather frequently.
</p><!--l. 6352--><p class="indent">   In  this  example,  where  the  value  of  <span class="ectt-1095">hillWidth</span>
(<!--l. 6352--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>2</mn><mi>σ</mi></math>) amounts
to 2 grid points, the threshold is 6+1 = 7 grid points away from <span class="ectt-1095">upperBoundary</span>. In explicit units, the <span class="ectt-1095">width</span>
of <!--l. 6353--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>r</mi></math> is
<!--l. 6353--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>w</mi></mrow><mrow><mi>r</mi></mrow></msub> <mo class="MathClass-rel">=</mo></math> 0.2 Å, and the
threshold is 15.0 - 7<!--l. 6353--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-bin">×</mo></math>0.2
= 13.6 Å.
</p><!--l. 6355--><p class="indent">   The <span class="ectt-1095">wall_r </span>restraint included in the example prevents this: the position of its <span class="ectt-1095">upperWall </span>is 13 Å, i.e. 3 grid
points below the buffer's threshold (13.6 Å). For the chosen value of <span class="ectt-1095">upperWallConstant</span>, the energy of the <span class="ectt-1095">wall_r</span>
bias at <span class="ectt-1095">r </span>= <!--l. 6356--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>r</mi></mrow><mrow><mstyle mathvariant="normal"><mi>u</mi><mi>p</mi><mi>p</mi><mi>e</mi><mi>r</mi></mstyle></mrow></msub></math>
= 13.6 Å is: </p><table class="equation-star"><tr><td>
<!--l. 6357--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<msup><mrow><mi>E</mi></mrow><mrow><mo class="MathClass-bin">∗</mo></mrow></msup> <mo class="MathClass-rel">=</mo> <mfrac><mrow><mn>1</mn></mrow>
<mrow><mn>2</mn></mrow></mfrac><mi>k</mi><msup><mrow> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mfrac><mrow><mi>r</mi><mo class="MathClass-bin">−</mo><msub><mrow><mi>r</mi></mrow><mrow><mstyle mathvariant="normal"><mi>u</mi><mi>p</mi><mi>p</mi><mi>e</mi><mi>r</mi></mstyle></mrow></msub></mrow>
<mrow><msub><mrow><mi>w</mi></mrow><mrow><mi>r</mi></mrow></msub></mrow></mfrac> </mrow><mo fence="true" form="postfix">)</mo></mrow></mrow><mrow><mn>2</mn></mrow></msup> <mo class="MathClass-rel">=</mo> <mfrac><mrow><mn>1</mn></mrow>
<mrow><mn>2</mn></mrow></mfrac><mn>2</mn><mo class="MathClass-punc">.</mo><mn>0</mn><msup><mrow> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mo class="MathClass-bin">−</mo><mn>3</mn></mrow><mo fence="true" form="postfix">)</mo></mrow></mrow><mrow><mn>2</mn></mrow></msup> <mo class="MathClass-rel">=</mo> <mn>9</mn><mspace class="nbsp" width="0.33em"></mspace><mstyle mathvariant="normal"><mi>k</mi><mi>c</mi><mi>a</mi><mi>l</mi><mo class="MathClass-bin">∕</mo><mi>m</mi><mi>o</mi><mi>l</mi></mstyle>
</math></td></tr></table>
<!--l. 6360--><p class="indent">   which results in a relative probability <!--l. 6360--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi class="qopname">exp</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--><mo class="MathClass-open">(</mo><mo class="MathClass-bin">−</mo><msup><mrow><mi>E</mi></mrow><mrow><mo class="MathClass-bin">∗</mo></mrow></msup><mo class="MathClass-bin">∕</mo><msub><mrow><mi>κ</mi></mrow><mrow><mstyle mathvariant="normal"><mi>B</mi></mstyle></mrow></msub><mi>T</mi> <mo class="MathClass-close">)</mo> <mo class="MathClass-rel">≃</mo></math>
<!--l. 6360--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>3</mn><mo class="MathClass-bin">×</mo><mn>1</mn><msup><mrow><mn>0</mn></mrow><mrow><mo class="MathClass-bin">−</mo><mn>7</mn></mrow></msup></math> that <span class="ectt-1095">r</span>
crosses the threshold. The probability that <span class="ectt-1095">r </span>exceeds <span class="ectt-1095">upperBoundary</span>, which is further away, has
also become vanishingly small. At that point, you may want to set <span class="ectt-1095">hardUpperBoundary </span>to
<span class="ectt-1095">yes </span>for <span class="ectt-1095">r</span>, and let <span class="ectt-1095">meta_r </span>know that no special treatment near the grid's boundaries will be
needed.
</p><!--l. 6364--><p class="indent"> <span class="phvro8t-x-x-109">What is the impact of the wall restraint onto the PMF? </span>Not a very complicated one: the PMF reconstructed
by metadynamics will simply show a sharp increase in free-energy where the wall potential kicks in
(<span class="ectt-1095">r</span> <!--l. 6364--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>
13 Å). You may then choose between using the PMF only up until that point and discard the rest, or
subtracting the energy of the <span class="ectt-1095">harmonicWalls </span>restraint from the PMF itself. Keep in mind, however, that the
statistical convergence of metadynamics may be less accurate where the wall potential is
strong.
</p><!--l. 6368--><p class="indent">   In summary, although it would be simpler to set the wall's position <span class="ectt-1095">upperWall </span>and the grid's boundary
<span class="ectt-1095">upperBoundary </span>to the same number, the finite width of the Gaussian hills calls for setting the former
strictly within the latter.
</p>
<div class="flushright">
<!--l. 6371--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 6371--><p class="noindent"> </p><div class="heading"> <a id="sec:colvarbias_meta_basics"></a>
<h5 class="subsubsectionHead"><span class="titlemark">6.6.2    </span> <a id="sec:colvarbias_meta_basics"></a>Required metadynamics keywords</h5>
<!--l. 6371--><p class="noindent"></p></div>
<!--l. 6373--><p class="indent">   To enable a metadynamics-based calculation, a <span class="ectt-1095">metadynamics {...} </span>block must be included in the
Colvars configuration file.
</p><!--l. 6375--><p class="indent">   By default, metadynamics bias energy and forces will be recorded onto a grid, the parameters of
which can be defined within the definition of each <span class="ectt-1095">colvar</span>, as described in <a href="#sec:colvar_grid_params">4.22<!--tex4ht:ref: sec:colvar_grid_params --></a>. Alternately, grid
parameters can be set directly inside the <span class="ectt-1095">metadynamics </span>block by inserting a <span class="ectt-1095">grid </span>sub-block, as explained
in <a href="#sec:colvarbias_grid">6.1<!--tex4ht:ref: sec:colvarbias_grid --></a>.
</p><!--l. 6379--><p class="indent">   Other required keywords will be specified within the <span class="ectt-1095">metadynamics </span>block: these
are <a href="#colvarbias|colvars"><span class="ectt-1095">colvars</span></a> (the names of the variables involved), <a href="#metadynamics|hillWeight"><span class="ectt-1095">hillWeight</span></a> (the weight parameter
<!--l. 6379--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>W</mi></math>), and the
widths <!--l. 6379--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>2</mn><mi>σ</mi></math>
of the Gaussian hills in each dimension that can be given either as the single dimensionless parameter
<a href="#metadynamics|hillWidth"><span class="ectt-1095">hillWidth</span></a>, or explicitly for each colvar with <a href="#metadynamics|gaussianSigmas"><span class="ectt-1095">gaussianSigmas</span></a>.
</p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox330"><a id="dx1-135001"></a><span class="ectt-1095">name</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">name</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox331"><a id="dx1-135002"></a><span class="ectt-1095">colvars</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">colvars</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox332"><a id="dx1-135003"></a><span class="ectt-1095">outputEnergy</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">outputEnergy</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox333"><a id="dx1-135004"></a><span class="ectt-1095">outputFreq</span></span> — same definition as <a href="#colvarbias|outputFreq"><span class="ectt-1095">outputFreq</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox334"><a id="dx1-135005"></a><span class="ectt-1095">writeTIPMF</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">writeTIPMF</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox335"><a id="dx1-135006"></a><span class="ectt-1095">writeTISamples</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">writeTISamples</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox336"><a id="dx1-135007"></a><span class="ectt-1095">stepZeroData</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">stepZeroData</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="metadynamics|hillWeight"></a> <span class="colorbox" id="colorbox337"><a id="dx1-135008"></a><span class="ectt-1095">hillWeight</span></span> — <span class="phvro8t-x-x-109">Height of each hill (kcal/mol)</span><br class="newline"/><!--l. 6399--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     decimal, context: <span class="ectt-1095">metadynamics</span> <!--l. 6399--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option sets the height <!--l. 6399--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>W</mi></math>
     of the Gaussian hills that are added during this run. Lower values provide more accurate
     sampling  of  the  system's  degrees  of  freedom  at  the  price  of  longer  simulation  times  to
     complete a PMF calculation based on metadynamics.
     </li>
<li class="itemize"><a id="metadynamics|hillWidth"></a> <span class="colorbox" id="colorbox338"><a id="dx1-135009"></a><span class="ectt-1095">hillWidth</span></span> — <span class="phvro8t-x-x-109">Width </span><!--l. 6411--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>2</mn><mi>σ</mi></math>
<span class="phvro8t-x-x-109">of a Gaussian hill, measured in number of grid points</span><br class="newline"/><!--l. 6411--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     decimal, context: <span class="ectt-1095">metadynamics</span> <!--l. 6411--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This keyword sets the Gaussian width <!--l. 6411--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>2</mn><msub><mrow><mi>σ</mi></mrow><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow></msub></math>
     for all colvars, expressed in <span class="phvro8t-x-x-109">number of grid points</span>, with the grid spacing along each colvar
     <!--l. 6411--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>
     determined by the respective value of <a href="#colvar|width"><span class="ectt-1095">width</span></a>. Values between 1 and 3 are recommended for
     this option: smaller numbers will fail to adequately interpolate each Gaussian function <span class="cite">[<a href="#XFiorin2013">1</a>]</span>,
     while larger values may be unable to account for steep free-energy gradients. The values of
     each half-width <!--l. 6411--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>σ</mi></mrow><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow></msub></math>
     in the physical units of <!--l. 6411--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></math>
     are also printed by NAMD at initialization time; alternatively, they may be set explicitly via
     <a href="#metadynamics|gaussianSigmas"><span class="ectt-1095">gaussianSigmas</span></a>.
     </li>
<li class="itemize"><a id="metadynamics|gaussianSigmas"></a> <span class="colorbox" id="colorbox339"><a id="dx1-135010"></a><span class="ectt-1095">gaussianSigmas</span></span> — <span class="phvro8t-x-x-109">Half-widths </span><!--l. 6423--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>σ</mi></math>
<span class="phvro8t-x-x-109">of the Gaussian hill (one for each colvar)</span><br class="newline"/><!--l. 6423--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> space-separated
     list of decimals, context: <span class="ectt-1095">metadynamics</span> <!--l. 6423--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option sets the parameters <!--l. 6423--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>σ</mi></mrow><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow></msub></math>
     of the Gaussian hills along each colvar <!--l. 6423--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></math>,
     expressed in <span class="phvro8t-x-x-109">the same unit of </span><!--l. 6423--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></math>.
     No restrictions are placed on each value, but a warning will be printed if <a href="#metadynamics|useGrids"><span class="ectt-1095">useGrids</span></a> is <span class="ectt-1095">on </span>and
     the Gaussian width <!--l. 6423--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>2</mn><msub><mrow><mi>σ</mi></mrow><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow></msub></math>
     is smaller than the corresponding grid spacing, <!--l. 6423--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="monospace"><mi>w</mi><mi>i</mi><mi>d</mi><mi>t</mi><mi>h</mi></mstyle><mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-close">)</mo></math>.
     If not given, default values will be computed from the dimensionless number <a href="#metadynamics|hillWidth"><span class="ectt-1095">hillWidth</span></a>.
     </li>
<li class="itemize"><a id="metadynamics|newHillFrequency"></a> <span class="colorbox" id="colorbox340"><a id="dx1-135011"></a><span class="ectt-1095">newHillFrequency</span></span> — <span class="phvro8t-x-x-109">Frequency of hill creation</span><br class="newline"/>Default: <span class="ectt-1095">1000</span> <!--l. 6436--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     integer, context: <span class="ectt-1095">metadynamics</span> <!--l. 6436--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option sets the number of steps after which a new Gaussian hill is added to the metadynamics
     potential. The product of this number and the integration time-step defines the parameter
     <!--l. 6436--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>δ</mi><mi>t</mi></math>
     in eq. <a href="#eq:colvars_meta_pot">36<!--tex4ht:ref: eq:colvars_meta_pot --></a>. Higher values provide more accurate statistical sampling, at the price of longer
     simulation times to complete a PMF calculation.
     </li></ul>
<div class="flushright">
<!--l. 6441--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 6441--><p class="noindent"> </p><div class="heading"> <a id="sec:colvarbias_meta_output"></a>
<h5 class="subsubsectionHead"><span class="titlemark">6.6.3    </span> <a id="sec:colvarbias_meta_output"></a>Output files</h5>
<!--l. 6441--><p class="noindent"></p></div>
<!--l. 6443--><p class="indent">   When interpolating grids are enabled (default behavior), the PMF is written by default
every <span class="ectt-1095">colvarsRestartFrequency </span>steps to the file <span class="phvro8t-x-x-109">outputName</span><span class="ectt-1095">.pmf </span>in multicolumn text format
(<a href="#sec:colvar_multicolumn_grid">3.8.5<!--tex4ht:ref: sec:colvar_multicolumn_grid --></a>). The following two options allow to disable or control this behavior and to track statistical
convergence:
</p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox341"><a id="dx1-136001"></a><span class="ectt-1095">writeFreeEnergyFile</span></span> — <span class="phvro8t-x-x-109">Periodically write the PMF for visualization</span><br class="newline"/>Default: <span class="ectt-1095">on</span> <!--l. 6455--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">metadynamics</span> <!--l. 6455--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>When <span class="ectt-1095">useGrids </span>and this option are <span class="ectt-1095">on</span>, the PMF is written every <a href="#colvarbias|outputFreq"><span class="ectt-1095">outputFreq</span></a> steps.
                                                                                         
                                                                                         
     </li>
<li class="itemize"><span class="colorbox" id="colorbox342"><a id="dx1-136002"></a><span class="ectt-1095">keepFreeEnergyFiles</span></span> — <span class="phvro8t-x-x-109">Keep all the PMF files</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 6465--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">metadynamics</span> <!--l. 6465--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>When <span class="ectt-1095">writeFreeEnergyFile </span>and this option are <span class="ectt-1095">on</span>, the step number is included in the file
     name, thus generating a series of PMF files. Activating this option can be useful to follow
     more closely the convergence of the simulation, by comparing PMFs separated by short
     times.
     </li>
<li class="itemize"><a id="metadynamics|writeHillsTrajectory"></a> <span class="colorbox" id="colorbox343"><a id="dx1-136003"></a><span class="ectt-1095">writeHillsTrajectory</span></span> — <span class="phvro8t-x-x-109">Write a log of new hills</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 6483--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">metadynamics</span> <!--l. 6483--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If this option is <span class="ectt-1095">on</span>, a file containing the Gaussian hills written by the <span class="ectt-1095">metadynamics </span>bias, with
     the name:<br class="newline"/>“<span class="phvro8t-x-x-109">outputName</span><span class="ectt-1095">.colvars.</span><!--l. 6483--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">name</span><!--l. 6483--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><span class="ectt-1095">.hills.traj</span>",<br class="newline"/>which can be useful to post-process the time series of the Gassian hills. Each line is written
     every <span class="ectt-1095">newHillFrequency</span>, regardless of the value of <a href="#colvarbias|outputFreq"><span class="ectt-1095">outputFreq</span></a>. When <span class="ectt-1095">multipleReplicas</span>
     is <span class="ectt-1095">on</span>, its name is changed to:<br class="newline"/>“<span class="phvro8t-x-x-109">outputName</span><span class="ectt-1095">.colvars.</span><!--l. 6483--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">name</span><!--l. 6483--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><span class="ectt-1095">.</span><!--l. 6483--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">replicaID</span><!--l. 6483--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><span class="ectt-1095">.hills.traj</span>".<br class="newline"/>The columns of this file are the centers of the hills, <!--l. 6483--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-open">(</mo><msup><mrow><mi>t</mi></mrow><mrow><mi>′</mi></mrow></msup><mo class="MathClass-close">)</mo></math>,
     followed by the half-widths, <!--l. 6483--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>σ</mi></mrow><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow></msub></math>,
     and the weight, <!--l. 6483--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>W</mi></math>.
     <span class="phvb8t-x-x-109">Note: </span>prior to version 2020-02-24, the full-width <!--l. 6483--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>2</mn><mi>σ</mi></math>
     of the Gaussian was reported in lieu of <!--l. 6483--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>σ</mi></math>.
     </li></ul>
<div class="flushright">
<!--l. 6488--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 6488--><p class="noindent"> </p><div class="heading"> <a id="sec:colvarbias_meta_performance"></a>
<h5 class="subsubsectionHead"><span class="titlemark">6.6.4    </span> <a id="sec:colvarbias_meta_performance"></a>Performance optimization</h5>
<!--l. 6488--><p class="noindent"></p></div> The following options control the computational cost of metadynamics calculations, but do not affect
results. Default values are chosen to minimize such cost with no loss of accuracy.

     <ul class="itemize1">
<li class="itemize"><a id="metadynamics|useGrids"></a> <span class="colorbox" id="colorbox344"><a id="dx1-137001"></a><span class="ectt-1095">useGrids</span></span> — <span class="phvro8t-x-x-109">Interpolate the hills with grids</span><br class="newline"/>Default: <span class="ectt-1095">on</span> <!--l. 6510--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">metadynamics</span> <!--l. 6510--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This  option  discretizes  all  hills  for  improved  performance,  accumulating  their  energy  and
     their gradients on two separate grids of equal spacing. Grids are defined by the values of
     <span class="ectt-1095">lowerBoundary</span>, <span class="ectt-1095">upperBoundary </span>and <span class="ectt-1095">width </span>for each colvar. Currently, this option is implemented
     for all types of variables except the non-scalar types (<span class="ectt-1095">distanceDir </span>or <span class="ectt-1095">orientation</span>). If <span class="ectt-1095">expandBoundaries</span>
     is defined in one of the colvars, grids are automatically expanded along the direction of that
     colvar.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox345"><a id="dx1-137002"></a><span class="ectt-1095">rebinGrids</span></span> — <span class="phvro8t-x-x-109">Recompute the grids when reading a state file</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 6523--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">metadynamics</span> <!--l. 6523--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>When restarting from a state file, the grid's parameters (boundaries and widths) saved in the
     state file override those in the configuration file. Enabling this option forces the grids to match
     those in the current configuration file.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox346"><a id="dx1-137003"></a><span class="ectt-1095">keepHills</span></span> — <span class="phvro8t-x-x-109">Write each individual hill to the state file</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 6537--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">metadynamics</span> <!--l. 6537--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>When <span class="ectt-1095">useGrids </span>and this option are <span class="ectt-1095">on</span>, all hills are saved to the state file in their analytic
     form, alongside their grids. This makes it possible to later use exact analytic Gaussians for
     <span class="ectt-1095">rebinGrids</span>. To only keep track of the history of the added hills, <span class="ectt-1095">writeHillsTrajectory </span>is
     preferable.
     </li></ul>
<div class="flushright">
<!--l. 6542--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 6542--><p class="noindent"> </p><div class="heading"> <a id="sec:colvarbias_ebmeta"></a>
<h5 class="subsubsectionHead"><span class="titlemark">6.6.5    </span> <a id="sec:colvarbias_ebmeta"></a>Ensemble-Biased Metadynamics</h5>
<!--l. 6542--><p class="noindent"></p></div>
<!--l. 6544--><p class="indent">   The ensemble-biased metadynamics (EBMetaD) approach <span class="cite">[<a href="#XMarinelli2015">40</a>]</span> is designed to reproduce a target
probability distribution along selected collective variables. Standard metadynamics can be seen as a
special case of EBMetaD with a flat distribution as target. This is achieved by weighing the Gaussian
functions used in the metadynamics approach by the inverse of the target probability distribution: </p><table class="equation"><tr><td>
<a id="eq:colvars_ebmeta_pot"></a>
<!--l. 6547--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<msub><mrow><mi>V</mi></mrow><mrow><mstyle mathvariant="normal"><mi>E</mi><mi>B</mi><mi>m</mi><mi>e</mi><mi>t</mi><mi>a</mi><mi>D</mi></mstyle></mrow></msub> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mi>t</mi></mrow><mo fence="true" form="postfix">)</mo></mrow></mrow><mo fence="true" form="postfix">)</mo></mrow><mspace class="thickpace" width="0.28em"></mspace> <mo class="MathClass-rel">=</mo> <mspace class="thickpace" width="0.28em"></mspace><munderover accent="false" accentunder="false"><mrow><mo>∑</mo></mrow><mrow><msup><mrow><mi>t</mi></mrow><mrow><mi>′</mi></mrow></msup><mo class="MathClass-rel">=</mo><mi>δ</mi><mi>t</mi><mo class="MathClass-punc">,</mo><mn>2</mn><mi>δ</mi><mi>t</mi><mo class="MathClass-punc">,</mo><mo>…</mo></mrow><mrow><msup><mrow><mi>t</mi></mrow><mrow><mi>′</mi></mrow></msup><mo class="MathClass-rel">&lt;</mo><mi>t</mi></mrow></munderover> <mfrac><mrow><mi>W</mi></mrow>
<mrow><mi class="qopname">exp</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--> <mrow><mo fence="true" form="prefix"> (</mo><mrow><msub><mrow><mi>S</mi></mrow><mrow><mi>ρ</mi></mrow></msub></mrow><mo fence="true" form="postfix">)</mo></mrow><msub><mrow><mi>ρ</mi></mrow><mrow><mi>e</mi><mi>x</mi><mi>p</mi></mrow></msub><mo class="MathClass-open">(</mo><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle><mo class="MathClass-open">(</mo><msup><mrow><mi>t</mi></mrow><mrow><mi>′</mi></mrow></msup><mo class="MathClass-close">)</mo><mo class="MathClass-close">)</mo></mrow></mfrac><mspace class="thinspace" width="0.22em"></mspace><munderover accent="false" accentunder="false"><mrow><mo>∏</mo></mrow><mrow><mi>i</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><msub><mrow><mi>N</mi></mrow><mrow><mstyle mathvariant="normal"><mi>c</mi><mi>v</mi></mstyle></mrow></msub></mrow></munderover><mi class="qopname"> exp</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mo class="MathClass-bin">−</mo><mfrac><mrow><msup><mrow><mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo><mo class="MathClass-bin">−</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-open">(</mo><msup><mrow><mi>t</mi></mrow><mrow><mi>′</mi></mrow></msup><mo class="MathClass-close">)</mo><mo class="MathClass-close">)</mo></mrow><mrow><mn>2</mn></mrow></msup></mrow>
<mrow><mn>2</mn><msubsup><mrow><mi>σ</mi></mrow><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub></mrow><mrow><mn>2</mn></mrow></msubsup></mrow></mfrac> </mrow><mo fence="true" form="postfix">)</mo></mrow><mstyle mathvariant="normal"><mo class="MathClass-punc">,</mo></mstyle>
</math></td><td class="eq-no">(39)</td></tr></table>
<!--l. 6556--><p class="indent">   where <!--l. 6556--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ρ</mi></mrow><mrow><mi>e</mi><mi>x</mi><mi>p</mi></mrow></msub><mo class="MathClass-open">(</mo><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle><mo class="MathClass-close">)</mo></math> is the target
probability distribution and <!--l. 6556--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>S</mi></mrow><mrow><mi>ρ</mi></mrow></msub> <mo class="MathClass-rel">=</mo> <mo class="MathClass-bin">−</mo><mi class="MathClass-op">∫
 <!--nolimits--></mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo><!--nolimits--><msub><mrow><mi>ρ</mi></mrow><mrow><mi>e</mi><mi>x</mi><mi>p</mi></mrow></msub><mo class="MathClass-open">(</mo><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle><mo class="MathClass-close">)</mo><mi class="qopname">log</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--><msub><mrow><mi>ρ</mi></mrow><mrow><mi>e</mi><mi>x</mi><mi>p</mi></mrow></msub><mo class="MathClass-open">(</mo><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle><mo class="MathClass-close">)</mo><mspace class="thinspace" width="0.17em"></mspace><mstyle mathvariant="normal"><mi>d</mi></mstyle><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></math>
its corresponding differential entropy. The method is designed so that during the simulation the resulting distribution of
the collective variable <!--l. 6557--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle></math>
converges to <!--l. 6557--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ρ</mi></mrow><mrow><mi>e</mi><mi>x</mi><mi>p</mi></mrow></msub><mo class="MathClass-open">(</mo><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle><mo class="MathClass-close">)</mo></math>.
A practical application of EBMetaD is to reproduce an “experimental" probability distribution,
for example the distance distribution between spectroscopic labels inferred from Förster
resonance energy transfer (FRET) or double electron-electron resonance (DEER) experiments
<span class="cite">[<a href="#XMarinelli2015">40</a>]</span>.
</p><!--l. 6560--><p class="indent">   The PMF along <!--l. 6560--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>
can be estimated from the bias potential and the target ditribution <span class="cite">[<a href="#XMarinelli2015">40</a>]</span>: </p><table class="equation"><tr><td> <a id="x1-138002r40"></a>
<!--l. 6561--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>A</mi><mo class="MathClass-open">(</mo><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle><mo class="MathClass-close">)</mo><mspace class="thickpace" width="0.28em"></mspace> <mo class="MathClass-rel">≃</mo><mspace class="thickpace" width="0.28em"></mspace><mo class="MathClass-bin">−</mo><msub><mrow><mi>V</mi></mrow><mrow><mstyle mathvariant="normal"><mi>E</mi><mi>B</mi><mi>m</mi><mi>e</mi><mi>t</mi><mi>a</mi><mi>D</mi></mstyle></mrow></msub><mo class="MathClass-open">(</mo><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle><mo class="MathClass-close">)</mo><mo class="MathClass-bin">−</mo><msub><mrow><mi>κ</mi></mrow><mrow><mstyle mathvariant="normal"><mi>B</mi></mstyle></mrow></msub><mi>T</mi> <mi class="qopname">log</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--><msub><mrow><mi>ρ</mi></mrow><mrow><mi>e</mi><mi>x</mi><mi>p</mi></mrow></msub><mo class="MathClass-open">(</mo><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle><mo class="MathClass-close">)</mo>
</math></td><td class="eq-no">(40)</td></tr></table>
<!--l. 6567--><p class="indent">   and obtained by enabling <span class="ectt-1095">writeFreeEnergyFile</span>. Similarly to eq. <a href="#eq:colvars_meta_fes_av">38<!--tex4ht:ref: eq:colvars_meta_fes_av --></a>, a more accurate estimate of the
free energy can be obtained by averaging (after an equilibration time) multiple time-dependent free
energy estimates (see <span class="ectt-1095">keepFreeEnergyFiles</span>).
</p><!--l. 6570--><p class="indent">   The following additional options define the configuration for the ensemble-biased metadynamics
approach: </p>
<ul class="itemize1">
<li class="itemize"><a id="metadynamics|ebMeta"></a> <span class="colorbox" id="colorbox347"><a id="dx1-138003"></a><span class="ectt-1095">ebMeta</span></span> — <span class="phvro8t-x-x-109">Perform ensemble-biased metadynamics</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 6583--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">metadynamics</span> <!--l. 6583--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If enabled, this flag activates the ensemble-biased metadynamics as described by Marinelli
     et al.<span class="cite">[<a href="#XMarinelli2015">40</a>]</span>. The target distribution file, <span class="ectt-1095">targetdistfile</span>, is then required. The keywords <span class="ectt-1095">lowerBoundary</span>,
     <span class="ectt-1095">upperBoundary </span>and <span class="ectt-1095">width </span>for the respective variables are also needed to set the binning
     (grid) of the target distribution file.
     </li>
<li class="itemize"><a id="metadynamics|targetDistFile"></a> <span class="colorbox" id="colorbox348"><a id="dx1-138004"></a><span class="ectt-1095">targetDistFile</span></span> — <span class="phvro8t-x-x-109">Target probability distribution file for ensemble-biased metadynamics</span><br class="newline"/><!--l. 6594--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> multicolumn
     text file, context: <span class="ectt-1095">metadynamics</span> <!--l. 6594--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This file provides the target probability distribution, <!--l. 6594--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ρ</mi></mrow><mrow><mi>e</mi><mi>x</mi><mi>p</mi></mrow></msub><mo class="MathClass-open">(</mo><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle><mo class="MathClass-close">)</mo></math>,
     reported in eq. <a href="#eq:colvars_ebmeta_pot">39<!--tex4ht:ref: eq:colvars_ebmeta_pot --></a>. The latter distribution must be a tabulated function provided in a multicolumn
     text format (see <a href="#sec:colvar_multicolumn_grid">3.8.5<!--tex4ht:ref: sec:colvar_multicolumn_grid --></a>). The provided distribution is then normalized.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox349"><a id="dx1-138005"></a><span class="ectt-1095">ebMetaEquilSteps</span></span> — <span class="phvro8t-x-x-109">Number of equilibration steps for ensemble-biased metadynamics</span><br class="newline"/><!--l. 6605--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     integer, context: <span class="ectt-1095">metadynamics</span> <!--l. 6605--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>The EBMetaD approach may introduce large hills in regions with small values of the target
     probability  distribution  (eq.  <a href="#eq:colvars_ebmeta_pot">39<!--tex4ht:ref: eq:colvars_ebmeta_pot --></a>).  This  happens,  for  example,  if  the  probability  distribution
     sampled by a conventional molecular dynamics simulation is significantly different from the
     target distribution. This may lead to instabilities at the beginning of the simulation related to
     large biasing forces. In this case, it is useful to introduce an equilibration stage in which the
     bias potential gradually switches from standard metadynamics (eq. <a href="#eq:colvars_meta_pot">36<!--tex4ht:ref: eq:colvars_meta_pot --></a>) to EBmetaD (eq. <a href="#eq:colvars_ebmeta_pot">39<!--tex4ht:ref: eq:colvars_ebmeta_pot --></a>)
     as <!--l. 6605--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>λ</mi><msub><mrow><mi>V</mi></mrow><mrow><mstyle mathvariant="normal"><mi>m</mi><mi>e</mi><mi>t</mi><mi>a</mi></mstyle></mrow></msub><mo class="MathClass-open">(</mo><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle><mo class="MathClass-close">)</mo><mo class="MathClass-bin">+</mo><mo class="MathClass-open">(</mo><mn>1</mn><mo class="MathClass-bin">−</mo><mi>λ</mi><mo class="MathClass-close">)</mo><msub><mrow><mi>V</mi></mrow><mrow><mstyle mathvariant="normal"><mi>E</mi><mi>B</mi><mi>m</mi><mi>e</mi><mi>t</mi><mi>a</mi><mi>D</mi></mstyle></mrow></msub><mo class="MathClass-open">(</mo><mstyle class="mbox"><mtext></mtext><mstyle class="math"><mi>ξ</mi></mstyle><mtext></mtext></mstyle><mo class="MathClass-close">)</mo></math>,
     where <!--l. 6605--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>λ</mi> <mo class="MathClass-rel">=</mo> <mo class="MathClass-open">(</mo><mstyle mathvariant="monospace"><mi>e</mi><mi>b</mi><mi>M</mi><mi>e</mi><mi>t</mi><mi>a</mi><mi>E</mi><mi>q</mi><mi>u</mi><mi>i</mi><mi>l</mi><mi>S</mi><mi>t</mi><mi>e</mi><mi>p</mi><mi>s</mi></mstyle><mo class="MathClass-bin">−</mo><mstyle mathvariant="monospace"><mi>s</mi><mi>t</mi><mi>e</mi><mi>p</mi></mstyle><mo class="MathClass-close">)</mo><mo class="MathClass-bin">∕</mo><mstyle mathvariant="monospace"><mi>e</mi><mi>b</mi><mi>M</mi><mi>e</mi><mi>t</mi><mi>a</mi><mi>E</mi><mi>q</mi><mi>u</mi><mi>i</mi><mi>l</mi><mi>S</mi><mi>t</mi><mi>e</mi><mi>p</mi><mi>s</mi></mstyle></math>
     and <span class="ectt-1095">step </span>is the current simulation step number.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox350"><a id="dx1-138006"></a><span class="ectt-1095">targetDistMinVal</span></span> — <span class="phvro8t-x-x-109">Minimum value of the target distribution in reference to its </span><span class="ecit-1095">maximum</span>
<span class="ecit-1095">value</span><br class="newline"/><!--l. 6618--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     decimal, context: <span class="ectt-1095">metadynamics</span> <!--l. 6618--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>It is useful to set a minimum value of the target probability distribution to avoid values of
     the latter that are nearly zero, leading to very large hills. This parameter sets the minimum
     value of the target probability distribution that is expressed as a fraction of its <span class="ectt-1095">maximum value</span>:
     <span class="ectt-1095">minimum value = maximum value X targetDistMinVal</span>. This implies that 0 <span class="ectt-1095">&lt;targetDistMinVal</span>
<span class="ectt-1095">&lt; </span>1  and  its  default  value  is  set  to  1/1000000.  To  avoid  divisions  by  zero  (see  eq.  <a href="#eq:colvars_ebmeta_pot">39<!--tex4ht:ref: eq:colvars_ebmeta_pot --></a>),  if
     <span class="ectt-1095">targetDistMinVal </span>is set as zero, values of <!--l. 6618--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ρ</mi></mrow><mrow><mi>e</mi><mi>x</mi><mi>p</mi></mrow></msub></math>
     equal to zero are replaced by the smallest positive value read in the same file.
     </li></ul>
<!--l. 6622--><p class="indent">   As with standard metadynamics, multidimensional probability distributions can be targeted using a
single <span class="ectt-1095">metadynamics </span>block using multiple <span class="ectt-1095">colvars </span>and a multidimensional target distribution file (see
<a href="#sec:colvar_multicolumn_grid">3.8.5<!--tex4ht:ref: sec:colvar_multicolumn_grid --></a>). Instead, multiple probability distributions on different variables can be targeted separately in
the same simulation by introducing multiple <span class="ectt-1095">metadynamics </span>blocks with the <span class="ectt-1095">ebMeta </span>option.
<br class="newline"/>
</p><!--l. 6626--><p class="noindent"><span class="phvb8t-x-x-109">Example: </span>EBmetaD configuration for a single variable.</p>
<div class="mdframed" id="mdframed-63">
<!--l. 6627--><p class="noindent"></p><pre>colvar {<br class="newline"/>  name r <br class="newline"/>  distance {<br class="newline"/>    group1 { atomNumbers 991 992 }<br class="newline"/>    group2 { atomNumbers 1762 1763 }<br class="newline"/>  }<br class="newline"/>  upperBoundary  100.0 <br class="newline"/>  width            0.1 <br class="newline"/>}<br class="newline"/><br class="newline"/>metadynamics {<br class="newline"/>  name              ebmeta<br class="newline"/>  colvars           r<br class="newline"/>  hillWeight        0.01<br class="newline"/>  hillWidth         3.0<br class="newline"/>  ebMeta            on<br class="newline"/>  targetDistFile    targetdist1.dat<br class="newline"/>  ebMetaEquilSteps  500000<br class="newline"/>} </pre><!--l. 6647--><p class="noindent"></p>
</div>
<!--l. 6649--><p class="noindent">where <span class="ectt-1095">targetdist1.dat </span>is a text file in “multicolumn" format (<a href="#sec:colvar_multicolumn_grid">3.8.5<!--tex4ht:ref: sec:colvar_multicolumn_grid --></a>) with the same <span class="ectt-1095">width </span>as the variable <span class="ectt-1095">r</span>
(0.1 in this case):<br class="newline"/><!--tex4ht:inline--></p><div class="tabular"> <table class="tabular" id="TBL-7"><colgroup id="TBL-7-1g"><col id="TBL-7-1"/><col id="TBL-7-2"/><col id="TBL-7-3"/><col id="TBL-7-4"/><col id="TBL-7-5"/><col id="TBL-7-6"/></colgroup><tr id="TBL-7-1-" style="vertical-align:baseline;"><td class="td11" id="TBL-7-1-1" style="text-align:left; white-space:nowrap;"> </td></tr><tr id="TBL-7-2-" style="vertical-align:baseline;"><td class="td11" id="TBL-7-2-1" style="text-align:left; white-space:nowrap;"> #</td><td class="td11" id="TBL-7-2-2" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">1     </span></td><td class="td11" id="TBL-7-2-3" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-7-2-4" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-7-2-5" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-7-2-6" style="text-align:left; white-space:nowrap;"> </td>
</tr><tr id="TBL-7-3-" style="vertical-align:baseline;"><td class="td11" id="TBL-7-3-1" style="text-align:left; white-space:nowrap;"> #  </td><td class="td11" id="TBL-7-3-2" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">0.0   </span></td><td class="td11" id="TBL-7-3-3" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">0.1    </span></td><td class="td11" id="TBL-7-3-4" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">1000 </span></td><td class="td11" id="TBL-7-3-5" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">0 </span></td></tr><tr id="TBL-7-4-" style="vertical-align:baseline;"><td class="td11" id="TBL-7-4-1" style="text-align:left; white-space:nowrap;"> </td>
</tr><tr id="TBL-7-5-" style="vertical-align:baseline;"><td class="td11" id="TBL-7-5-1" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-7-5-2" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">0.05 </span></td><td class="td11" id="TBL-7-5-3" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">0.0012 </span></td>
</tr><tr id="TBL-7-6-" style="vertical-align:baseline;"><td class="td11" id="TBL-7-6-1" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-7-6-2" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">0.15 </span></td><td class="td11" id="TBL-7-6-3" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">0.0014 </span></td>
</tr><tr id="TBL-7-7-" style="vertical-align:baseline;"><td class="td11" id="TBL-7-7-1" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-7-7-2" style="text-align:left; white-space:nowrap;"> …         </td><td class="td11" id="TBL-7-7-3" style="text-align:left; white-space:nowrap;"> …           </td>
</tr><tr id="TBL-7-8-" style="vertical-align:baseline;"><td class="td11" id="TBL-7-8-1" style="text-align:left; white-space:nowrap;"> </td><td class="td11" id="TBL-7-8-2" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">99.95 </span></td><td class="td11" id="TBL-7-8-3" style="text-align:left; white-space:nowrap;"> <span class="ectt-1095">0.0010 </span></td>
</tr><tr id="TBL-7-9-" style="vertical-align:baseline;"><td class="td11" id="TBL-7-9-1" style="text-align:left; white-space:nowrap;"> </td>
</tr><tr id="TBL-7-10-" style="vertical-align:baseline;"><td class="td11" id="TBL-7-10-1" style="text-align:left; white-space:nowrap;"> </td></tr></table></div>
<!--l. 6662--><p class="indent"> <span class="phvb8t-x-x-109">Tip: </span>Besides setting a meaningful value for <span class="ectt-1095">targetDistMinVal</span>, the exploration of unphysically low
values of the target distribution (which would lead to very large hills and possibly numerical instabilities)
can be also prevented by restricting sampling to a given interval, using e.g. <span class="ectt-1095">harmonicWalls </span>restraint
(<a href="#sec:colvarbias_harmonic_walls">6.10<!--tex4ht:ref: sec:colvarbias_harmonic_walls --></a>).
</p>
<div class="flushright">
<!--l. 6667--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 6667--><p class="noindent"> </p><div class="heading"> <a id="sec:colvarbias_well_tempered"></a>
<h5 class="subsubsectionHead"><span class="titlemark">6.6.6    </span> <a id="sec:colvarbias_well_tempered"></a>Well-tempered metadynamics</h5>
<!--l. 6667--><p class="noindent"></p></div> The following options define the configuration for the “well-tempered" metadynamics approach
<span class="cite">[<a href="#XBarducci2008">41</a>]</span>:

     <ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox351"><a id="dx1-139001"></a><span class="ectt-1095">wellTempered</span></span> — <span class="phvro8t-x-x-109">Perform well-tempered metadynamics</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 6682--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">metadynamics</span> <!--l. 6682--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If enabled, this flag causes well-tempered metadynamics as described by Barducci et al.<span class="cite">[<a href="#XBarducci2008">41</a>]</span>
                                                                                         
                                                                                         
     to be performed, rather than standard metadynamics. The parameter <span class="ectt-1095">biasTemperature </span>is
     then required. This feature was contributed by Li Li (Luthey-Schulten group, Department of
     Chemistry, UIUC).
     </li>
<li class="itemize"><span class="colorbox" id="colorbox352"><a id="dx1-139002"></a><span class="ectt-1095">biasTemperature</span></span> — <span class="phvro8t-x-x-109">Temperature bias for well-tempered metadynamics</span><br class="newline"/><!--l. 6696--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     decimal, context: <span class="ectt-1095">metadynamics</span> <!--l. 6696--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>When running metadynamics in the long time limit, collective variable space is sampled to a
     modified temperature <!--l. 6696--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>T</mi> <mo class="MathClass-bin">+</mo><mi mathvariant="normal">Δ</mi><mi>T</mi> </math>.
     In conventional metadynamics, the temperature “boost" <!--l. 6696--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="normal">Δ</mi><mi>T</mi> </math>
     would constantly increases with time. Instead, in well-tempered metadynamics <!--l. 6696--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="normal">Δ</mi><mi>T</mi> </math>
     must be defined by the user via <span class="ectt-1095">biasTemperature</span>. The written PMF includes the scaling
     factor <!--l. 6696--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">(</mo><mi>T</mi> <mo class="MathClass-bin">+</mo><mi mathvariant="normal">Δ</mi><mi>T</mi> <mo class="MathClass-close">)</mo><mo class="MathClass-bin">∕</mo><mi mathvariant="normal">Δ</mi><mi>T</mi> </math>
<span class="cite">[<a href="#XBarducci2008">41</a>]</span>. A careful choice of <!--l. 6696--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="normal">Δ</mi><mi>T</mi> </math>
     determines the sampling and convergence rate, and is hence crucial to the success of a
     well-tempered metadynamics simulation.</li></ul>
<!--l. 6699--><p class="indent"> <span class="phvb8t-x-x-109">Note: </span>usage of well-tempered metadynamics in combination with multiple walkers (<a href="#sec:colvarbias_meta_mr">6.6.7<!--tex4ht:ref: sec:colvarbias_meta_mr --></a>) is
discouraged. Should you want to use the two features, keep in mind that in Colvars, the well-tempered
scaling factor is applied independently of the communication between replicas, i.e. each walker has its
Gaussian hills scaled independently of the other walkers.
</p>
<div class="flushright">
<!--l. 6702--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 6702--><p class="noindent"> </p><div class="heading"> <a id="sec:colvarbias_meta_mr"></a>
<h5 class="subsubsectionHead"><span class="titlemark">6.6.7    </span> <a id="sec:colvarbias_meta_mr"></a>Multiple-walker metadynamics</h5>
<!--l. 6702--><p class="noindent"></p></div> Metadynamics calculations can be performed concurrently by multiple replicas that share a common
history. This variant of the method is called multiple-walker metadynamics <span class="cite">[<a href="#XRaiteri2006">42</a>]</span>: the Gaussian hills of all
replicas are periodically combined into a single biasing potential, intended to converge to a single
PMF.
<!--l. 6706--><p class="indent">   In the implementation here described <span class="cite">[<a href="#XFiorin2013">1</a>]</span>, replicas communicate through files. This arrangement
allows launching the replicas either (1) as a bundle (i.e. a single job in a cluster's queueing system) or (2)
as fully independent runs (i.e. as separate jobs for the queueing system). One advantage of the use case
(1) is that an identical Colvars configuration can be used for all replicas (otherwise, <span class="ectt-1095">replicaID </span>needs to
be manually set to a different string for each replica). However, the use case (2) is less demanding in
terms of high-performance computing resources: a typical scenario would be a computer
cluster (including virtual servers from a cloud provider) where not all nodes are connected to
each other at high speed, and thus each replica runs on a small group of nodes or a single
node.
                                                                                         
                                                                                         
</p><!--l. 6711--><p class="indent">   Whichever way the replicas are started (coupled or not), a shared filesystem is needed so that each
replica can read the files created by the others: paths to these files are stored in the shared file
<span class="ectt-1095">replicasRegistry</span>. This file, and those listed in it, are read every <span class="ectt-1095">replicaUpdateFrequency </span>steps. Each
time the Colvars state file is written (for example, <span class="ectt-1095">colvarsRestartFrequency </span>steps), the file
named:<br class="newline"/><span class="phvro8t-x-x-109">outputName</span><span class="ectt-1095">.colvars.</span><span class="phvro8t-x-x-109">name</span><span class="ectt-1095">.</span><span class="phvro8t-x-x-109">replicaID</span><span class="ectt-1095">.state</span><br class="newline"/>is written as well; this file contains only the state of the metadynamics bias, which the other replicas will
read in turn. In between the times when this file is modified/replaced, new hills are also temporarily written
to the file named:<br class="newline"/><span class="phvro8t-x-x-109">outputName</span><span class="ectt-1095">.colvars.</span><span class="phvro8t-x-x-109">name</span><span class="ectt-1095">.</span><span class="phvro8t-x-x-109">replicaID</span><span class="ectt-1095">.hills</span><br class="newline"/>Both files are only used for communication, and may be deleted after the replica begins writing files with a
new <span class="phvro8t-x-x-109">outputName</span>.<br class="newline"/>
</p><!--l. 6720--><p class="noindent"><span class="phvb8t-x-x-109">Example: </span>Multiple-walker metadynamics with file-based communication.<br class="newline"/></p>
<div class="mdframed" id="mdframed-64">
<!--l. 6721--><p class="noindent"></p><pre>metadynamics {<br class="newline"/>  name mymtd<br class="newline"/>  colvars x<br class="newline"/>  hillWeight 0.001<br class="newline"/>  newHillFrequency 1000<br class="newline"/>  hillWidth 3.0<br class="newline"/>  <br class="newline"/>  multipleReplicas       on<br class="newline"/>  replicasRegistry       /shared-folder/mymtd-replicas.txt<br class="newline"/>  replicaUpdateFrequency 50000  # Best if larger than newHillFrequency<br class="newline"/>} </pre><!--l. 6733--><p class="noindent"></p>
</div>
<!--l. 6735--><p class="indent">   The following are the multiple-walkers related options:
</p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox353"><a id="dx1-140001"></a><span class="ectt-1095">multipleReplicas</span></span> — <span class="phvro8t-x-x-109">Enable multiple-walker metadynamics</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 6746--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">metadynamics</span> <!--l. 6746--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option turns on multiple-walker communication between replicas.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox354"><a id="dx1-140002"></a><span class="ectt-1095">replicasRegistry</span></span> — <span class="phvro8t-x-x-109">Multiple replicas database file</span><br class="newline"/><!--l. 6756--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> UNIX
     filename, context: <span class="ectt-1095">metadynamics</span> <!--l. 6756--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If <span class="ectt-1095">multipleReplicas </span>is <span class="ectt-1095">on</span>, this option sets the path to the replicas' shared database file.
     It is best to use an absolute path (especially when running individual replicas in separate
     folders).
     </li>
<li class="itemize"><span class="colorbox" id="colorbox355"><a id="dx1-140003"></a><span class="ectt-1095">replicaUpdateFrequency</span></span> — <span class="phvro8t-x-x-109">How often hills are shared between replicas</span><br class="newline"/><!--l. 6766--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     integer, context: <span class="ectt-1095">metadynamics</span> <!--l. 6766--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If <span class="ectt-1095">multipleReplicas </span>is <span class="ectt-1095">on</span>, this option sets the number of steps after which each replica tries
     to read the other replicas' files. On a networked file system, it is best to use a number of
     steps that corresponds to at least a minute of wall time.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox356"><a id="dx1-140004"></a><span class="ectt-1095">replicaID</span></span> — <span class="phvro8t-x-x-109">Set the identifier for this replica (required only for independent jobs)</span><br class="newline"/>Default: replica index (only if MPIor Charm++ is used) <!--l. 6778--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> string,
     context: <span class="ectt-1095">metadynamics</span> <!--l. 6778--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If <span class="ectt-1095">multipleReplicas </span>is <span class="ectt-1095">on</span>, this option sets a unique identifier for this replicas. Specifying this
     option is thus only required when the replicas are launched as independent computations:
     when  the  replicas  share  a  common  parallel  communication  framework  (i.e. they  are  all
     launched together as a single message-passing computation via MPIor Charm++) the default
     value of this keyword is the replica's numeric index (zero-based).
     </li>
<li class="itemize"><span class="colorbox" id="colorbox357"><a id="dx1-140005"></a><span class="ectt-1095">writePartialFreeEnergyFile</span></span> —  <span class="phvro8t-x-x-109">Periodically  write  the  contribution  to  the  PMF  from  this</span>
<span class="phvro8t-x-x-109">replica</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 6790--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">metadynamics</span> <!--l. 6790--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If <span class="ectt-1095">multipleReplicas </span>is <span class="ectt-1095">on</span>, enabling this option produces an additional file <span class="phvro8t-x-x-109">outputName</span><span class="ectt-1095">.partial.pmf</span>,
     which can be useful to monitor the contribution of each replica to the total PMF (which is
     written to the file <span class="phvro8t-x-x-109">outputName</span><span class="ectt-1095">.pmf</span>). <span class="phvb8t-x-x-109">Note: </span>the name of this file is chosen for consistency and
     convenience, <span class="phvro8t-x-x-109">but its content is not a PMF </span>and it is not expected to converge, even if the total
     PMF does.
     </li></ul>
<div class="flushright">
<!--l. 6795--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 6795--><p class="noindent"> </p><div class="heading"> <a id="sec:colvarbias_opes"></a>
<h4 class="subsectionHead"><span class="titlemark">6.7    </span> <a id="sec:colvarbias_opes"></a>On-the-fly probability enhanced sampling (OPES)</h4>
<!--l. 6795--><p class="noindent"></p></div>
<!--l. 6797--><p class="indent">   This biasing method implements the on-the-fly probability enhanced sampling (OPES) with metadynamics-like
target distribution.<span class="cite">[<a href="#XInvernizzi2020">43</a>]</span> The bias samples target distributions defined via their marginal distribution
<!--l. 6799--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow><mi>p</mi></mrow><mrow><mstyle mathvariant="normal"><mi>t</mi><mi>g</mi></mstyle></mrow></msup><mo class="MathClass-open">(</mo><mstyle mathvariant="bold-italic"><mi>ξ</mi></mstyle><mo class="MathClass-close">)</mo></math> over some CVs,
<!--l. 6799--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="bold-italic"><mi>ξ</mi></mstyle> <mo class="MathClass-rel">=</mo> <mstyle mathvariant="bold-italic"><mi>ξ</mi></mstyle><mo class="MathClass-open">(</mo><mi>x</mi><mo class="MathClass-close">)</mo></math>. By default <span class="ectt-1095">opes_metad </span>targets
the well-tempered distribution, <!--l. 6800--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow><mi>p</mi></mrow><mrow><mstyle mathvariant="normal"><mi>W</mi><mi>T</mi></mstyle></mrow></msup><mo class="MathClass-open">(</mo><mstyle mathvariant="bold-italic"><mi>ξ</mi></mstyle><mo class="MathClass-close">)</mo> <mo class="MathClass-rel">=</mo> <msup><mrow><mo class="MathClass-open">[</mo><mi>P</mi><mo class="MathClass-open">(</mo><mstyle mathvariant="bold-italic"><mi>ξ</mi></mstyle><mo class="MathClass-close">)</mo><mo class="MathClass-close">]</mo></mrow><mrow><mn>1</mn><mo class="MathClass-bin">∕</mo><mi>γ</mi></mrow></msup></math>,
where <!--l. 6800--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>γ</mi></math>
is known as the <span class="ectt-1095">biasfactor</span>. Similarly to <span class="ectt-1095">metadynamics</span>, <span class="ectt-1095">opes_metad </span>optimizes the bias on-the-fly, with a given
<a href="#opes_metad|newHillFrequency"><span class="ectt-1095">newHillfrequency</span></a>. It does so by reweighting via kernel density estimation of the unbiased distribution in the CV
space, <!--l. 6802--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>P</mi><mo class="MathClass-open">(</mo><mstyle mathvariant="bold-italic"><mi>ξ</mi></mstyle><mo class="MathClass-close">)</mo></math>. A
                                                                                         
                                                                                         
compression algorithm is used to prevent the number of kernels from growing linearly with the simulation time. The
bias at step <!--l. 6804--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>n</mi></math>
is </p><table class="equation"><tr><td> <a id="eq:opes_potential"></a>
<!--l. 6805--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<msub><mrow><mi>V</mi></mrow><mrow><mi>n</mi></mrow></msub> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mstyle mathvariant="bold-italic"><mi>ξ</mi></mstyle></mrow><mo fence="true" form="postfix">)</mo></mrow> <mo class="MathClass-rel">=</mo> <mo class="MathClass-bin">−</mo><msub><mrow><mi>k</mi></mrow><mrow><mstyle mathvariant="normal"><mi>B</mi></mstyle></mrow></msub><mi>T</mi> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mn>1</mn><mo class="MathClass-bin">−</mo><mn>1</mn><mo class="MathClass-bin">∕</mo><mi>γ</mi></mrow><mo fence="true" form="postfix">)</mo></mrow><mi class="qopname">ln</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mfrac><mrow><msub><mrow><mi>P</mi></mrow><mrow><mi>n</mi></mrow></msub><mo class="MathClass-open">(</mo><mstyle mathvariant="bold-italic"><mi>ξ</mi></mstyle><mo class="MathClass-close">)</mo></mrow>
<mrow><msub><mrow><mi>Z</mi></mrow><mrow><mi>n</mi></mrow></msub></mrow></mfrac> <mo class="MathClass-bin">+</mo><mi>𝜖</mi></mrow><mo fence="true" form="postfix">)</mo></mrow><mo class="MathClass-punc">,</mo>
</math></td><td class="eq-no">(41)</td></tr></table>
<!--l. 6808--><p class="indent">   where the probability <!--l. 6808--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>P</mi></mrow><mrow><mi>n</mi></mrow></msub> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mstyle mathvariant="bold-italic"><mi>ξ</mi></mstyle></mrow><mo fence="true" form="postfix">)</mo></mrow></math>
and the normalization factor <!--l. 6808--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>Z</mi></mrow><mrow><mi>n</mi></mrow></msub></math>
are computed as </p><table class="equation"><tr><td> <a id="x1-141002r42"></a>
<!--l. 6809--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<msub><mrow><mi>P</mi></mrow><mrow><mi>n</mi></mrow></msub><mo class="MathClass-open">(</mo><mstyle mathvariant="bold-italic"><mi>ξ</mi></mstyle><mo class="MathClass-close">)</mo> <mo class="MathClass-rel">=</mo> <mfrac><mrow><munderover accent="false" accentunder="false"><mrow><mo>∑</mo></mrow><mrow><mi>k</mi></mrow><mrow><mi>n</mi></mrow></munderover><msub><mrow><mi>w</mi></mrow><mrow><mi>k</mi></mrow></msub><mi>G</mi><mo class="MathClass-open">(</mo><mstyle mathvariant="bold-italic"><mi>ξ</mi></mstyle><mo class="MathClass-punc">,</mo><msub><mrow><mstyle mathvariant="bold-italic"><mi>ξ</mi></mstyle></mrow><mrow><mi>k</mi></mrow></msub><mo class="MathClass-close">)</mo></mrow>
<mrow><munderover accent="false" accentunder="false"><mrow><mo>∑</mo></mrow><mrow><mi>k</mi></mrow><mrow><mi>n</mi></mrow></munderover><msub><mrow><mi>w</mi></mrow><mrow><mi>k</mi></mrow></msub></mrow></mfrac>
</math></td><td class="eq-no">(42)</td></tr></table>
<table class="equation"><tr><td> <a id="x1-141003r43"></a>
<!--l. 6812--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<msub><mrow><mi>Z</mi></mrow><mrow><mi>n</mi></mrow></msub> <mo class="MathClass-rel">=</mo> <mfrac><mrow><mn>1</mn></mrow>
<mrow><mo class="MathClass-rel">|</mo><msub><mrow><mi mathvariant="normal">Ω</mi></mrow><mrow><mi>n</mi></mrow></msub><mo class="MathClass-rel">|</mo></mrow></mfrac><msub><mrow><mo>∫
 <!--nolimits--></mo><!--nolimits--></mrow><mrow><msub><mrow><mi mathvariant="normal">Ω</mi></mrow><mrow><mi>n</mi></mrow></msub></mrow></msub><msub><mrow><mi>P</mi></mrow><mrow><mi>n</mi></mrow></msub> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mstyle mathvariant="bold-italic"><mi>ξ</mi></mstyle></mrow><mo fence="true" form="postfix">)</mo></mrow><mstyle mathvariant="normal"><mi>d</mi></mstyle><mstyle mathvariant="bold-italic"><mi>ξ</mi></mstyle><mo class="MathClass-punc">,</mo>
</math></td><td class="eq-no">(43)</td></tr></table>
<!--l. 6815--><p class="indent">   where the weights <!--l. 6815--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>w</mi></mrow><mrow><mi>k</mi></mrow></msub></math>
are given by <!--l. 6815--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>w</mi></mrow><mrow><mi>k</mi></mrow></msub> <mo class="MathClass-rel">=</mo><mi class="qopname"> exp</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--><mo class="MathClass-open">(</mo><mi>β</mi><msub><mrow><mi>V</mi></mrow><mrow><mi>k</mi><mo class="MathClass-bin">−</mo><mn>1</mn></mrow></msub><mo class="MathClass-open">(</mo><msub><mrow><mstyle mathvariant="bold-italic"><mi>ξ</mi></mstyle></mrow><mrow><mi>k</mi></mrow></msub><mo class="MathClass-close">)</mo><mo class="MathClass-close">)</mo></math>, and the
Gaussian kernels <!--l. 6816--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>G</mi> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mstyle mathvariant="bold-italic"><mi>ξ</mi></mstyle><mo class="MathClass-punc">,</mo><msup><mrow><mstyle mathvariant="bold-italic"><mi>ξ</mi></mstyle></mrow><mrow><mi>′</mi></mrow></msup></mrow><mo fence="true" form="postfix">)</mo></mrow> <mo class="MathClass-rel">=</mo> <mi>h</mi><mi class="qopname">exp</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--> <mrow><mo fence="true" form="prefix"> [</mo><mrow><mo class="MathClass-bin">−</mo><mn>1</mn><mo class="MathClass-bin">∕</mo><mn>2</mn><msup><mrow> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mstyle mathvariant="bold-italic"><mi>ξ</mi></mstyle><mo class="MathClass-bin">−</mo><msup><mrow><mstyle mathvariant="bold-italic"><mi>ξ</mi></mstyle></mrow><mrow><mi>′</mi></mrow></msup></mrow><mo fence="true" form="postfix">)</mo></mrow></mrow><mrow><mi>T</mi> </mrow></msup><msup><mrow><mi mathvariant="normal">Σ</mi></mrow><mrow><mo class="MathClass-bin">−</mo><mn>1</mn></mrow></msup> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mstyle mathvariant="bold-italic"><mi>ξ</mi></mstyle><mo class="MathClass-bin">−</mo><msup><mrow><mstyle mathvariant="bold-italic"><mi>ξ</mi></mstyle></mrow><mrow><mi>′</mi></mrow></msup></mrow><mo fence="true" form="postfix">)</mo></mrow></mrow><mo fence="true" form="postfix">]</mo></mrow></math> have a
diagonal covariance matrix <!--l. 6816--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi mathvariant="normal">Σ</mi></mrow><mrow><mi>i</mi><mi>j</mi></mrow></msub> <mo class="MathClass-rel">=</mo> <msup><mrow><mi>σ</mi></mrow><mrow><mn>2</mn></mrow></msup><msub><mrow><mi>δ</mi></mrow><mrow><mi>i</mi><mi>j</mi></mrow></msub></math>
and fixed height <!--l. 6816--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>h</mi> <mo class="MathClass-rel">=</mo><msub><mrow> <mi mathvariant="normal">Π</mi></mrow><mrow><mi>i</mi></mrow></msub><msup><mrow><mo class="MathClass-open">(</mo><msub><mrow><mi>σ</mi></mrow><mrow><mi>i</mi></mrow></msub><msqrt><mrow><mn>2</mn><mi>π</mi></mrow></msqrt><mo class="MathClass-close">)</mo></mrow><mrow><mo class="MathClass-bin">−</mo><mn>1</mn></mrow></msup></math>
(see Ref.<span class="cite">[<a href="#XInvernizzi2020">43</a>]</span> for a complete description of the method).
                                                                                         
                                                                                         
</p><!--l. 6818--><p class="indent">   If the exploration mode (keyword <a href="#opes_metad|explore"><span class="ectt-1095">explore</span></a>) is on, then the on-the-fly target probability distribution
<!--l. 6818--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow><mi>p</mi></mrow><mrow><mstyle mathvariant="normal"><mi>W</mi><mi>T</mi></mstyle></mrow></msup><mo class="MathClass-open">(</mo><mstyle mathvariant="bold-italic"><mi>ξ</mi></mstyle><mo class="MathClass-close">)</mo></math> is used
to define the biasing energy: </p><table class="equation"><tr><td> <a id="x1-141004r44"></a>
<!--l. 6819--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<msub><mrow><mi>V</mi></mrow><mrow><mi>n</mi></mrow></msub> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mstyle mathvariant="bold-italic"><mi>ξ</mi></mstyle></mrow><mo fence="true" form="postfix">)</mo></mrow> <mo class="MathClass-rel">=</mo> <mo class="MathClass-bin">−</mo><msub><mrow><mi>k</mi></mrow><mrow><mstyle mathvariant="normal"><mi>B</mi></mstyle></mrow></msub><mi>T</mi> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mi>γ</mi> <mo class="MathClass-bin">−</mo><mn>1</mn></mrow><mo fence="true" form="postfix">)</mo></mrow><mi class="qopname">ln</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mfrac><mrow><msubsup><mrow><mi>p</mi></mrow><mrow><mi>n</mi></mrow><mrow><mstyle mathvariant="normal"><mi>W</mi><mi>T</mi></mstyle></mrow></msubsup><mo class="MathClass-open">(</mo><mstyle mathvariant="bold-italic"><mi>ξ</mi></mstyle><mo class="MathClass-close">)</mo></mrow>
<mrow><msub><mrow><mi>Z</mi></mrow><mrow><mi>n</mi></mrow></msub></mrow></mfrac> <mo class="MathClass-bin">+</mo><mi>𝜖</mi></mrow><mo fence="true" form="postfix">)</mo></mrow><mo class="MathClass-punc">,</mo>
</math></td><td class="eq-no">(44)</td></tr></table>
<!--l. 6822--><p class="indent">   (See Ref.<span class="cite">[<a href="#XInvernizzi2022">44</a>]</span> for a complete description of the exploration mode.)
</p><!--l. 6824--><p class="indent">   The implementation of <span class="ectt-1095">opes_metad</span> and its documentation are largely based on the <a href="https://www.plumed.org/doc-master/user-doc/html/_o_p_e_s__m_e_t_a_d__e_x_p_l_o_r_e.html">OPES module</a> of
the <a href="https://www.plumed.org/">PLUMED</a> package.
</p>
<div class="flushright">
<!--l. 6827--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 6827--><p class="noindent"> </p><div class="heading"> <a id="sec:colvarbias_opes_notes"></a>
<h5 class="subsubsectionHead"><span class="titlemark">6.7.1    </span> <a id="sec:colvarbias_opes_notes"></a>Implementation notes</h5>
<!--l. 6827--><p class="noindent"></p></div> Compared to the the OPES module in PLUMED <span class="cite">[<a href="#Xtribello_plumed_2014">45</a>]</span>, this implementation currently has the following
limitations: 
     <ul class="itemize1">
<li class="itemize">Colvars  does  not  support  chaining  bias  actions  in  PLUMED,  so  the  Colvars  OPES
     implementation  cannot  take  the  biasing  energies  of  other  biases,  such  as  harmonic
     restraints, into account for the internal reweighting, which means that there is no way to
     achieve the same functionality as the <span class="ectt-1095">EXTRA_BIAS </span>option in the OPES module in PLUMED.
     This means that it is not possible to sample a custom target distribution in the Colvars OPES
     implementation.
     </li>
<li class="itemize">Due  to  the  same  limitation  of  Colvars  as  described  above,  you  cannot  collect  samples
     on-the-fly by chaining <span class="ectt-1095">opes_metad </span>with a weighted histogram for PMF calculations. Although
     the PMF calculation via reweighting is implemented into the <a href="#opes_metad|pmf"><span class="ectt-1095">pmf</span></a> option, the functionality is
     quite limited. The CVs in <a href="#opes_metad|pmf_colvars"><span class="ectt-1095">pmfColvars</span></a> must be among the biased ones. In other words, you
     cannot apply bias on <!--l. 6831--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mn>2</mn></mrow></msub><mo class="MathClass-close">)</mo></math>
     and compute the PMF along <!--l. 6831--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mn>3</mn></mrow></msub></math>
     by the on-the-fly reweighting using <a href="#opes_metad|pmf"><span class="ectt-1095">pmf</span></a> (you can only compute the PMF along <!--l. 6831--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mn>1</mn></mrow></msub></math>,
     <!--l. 6831--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mn>2</mn></mrow></msub></math>
                                                                                         
                                                                                         
     or <!--l. 6831--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mn>2</mn></mrow></msub><mo class="MathClass-close">)</mo></math>).
     In addition, you can only specify one set of CVs for estimating the PMF (you cannot apply the
     OPES bias on <!--l. 6831--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mn>2</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mn>3</mn></mrow></msub><mo class="MathClass-close">)</mo></math>
     and compute two 2D PMFs along <!--l. 6831--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mn>2</mn></mrow></msub><mo class="MathClass-close">)</mo></math>
     and <!--l. 6831--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mn>2</mn></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mn>3</mn></mrow></msub><mo class="MathClass-close">)</mo></math>
     simultaneously  using  the  <a href="#opes_metad|pmf"><span class="ectt-1095">pmf</span></a> option).  However,  you  can  always  bypass  this  limitation  by
     post-hoc reweighting with the trajectories.
     </li>
<li class="itemize">Colvars computes the biases in parallel with SMP, so it cannot use multiple threads again
     for computing the probability and the normalization factor, searching the kernels that are
     able to compress and updating the neighbor list. As a result, the CPU performance of this
     implementation may be slower than the PLUMED OPES module.</li></ul>
<!--l. 6835--><p class="indent">   The following table summarizes the differences of the option names in Colvars and the corresponding
option names in the PLUMED OPES module: </p><div class="table">
<!--l. 6836--><p class="indent"> </p><hr class="float"/><div class="float">
<div class="tabular"> <table class="tabular" id="TBL-8"><colgroup id="TBL-8-1g"><col id="TBL-8-1"/><col id="TBL-8-2"/></colgroup><tr id="TBL-8-1-" style="vertical-align:baseline;"><td class="td11" id="TBL-8-1-1" style="text-align:left; white-space:nowrap;"> <span class="phvb8t-x-x-109">Colvars keyword              </span></td><td class="td11" id="TBL-8-1-2" style="text-align:left; white-space:nowrap;"> <span class="phvb8t-x-x-109">PLUMED keyword                                            </span></td>
</tr><tr class="hline"><td><hr/></td><td><hr/></td></tr><tr id="TBL-8-2-" style="vertical-align:baseline;"><td class="td11" id="TBL-8-2-1" style="text-align:left; white-space:nowrap;"> barrier                                 </td><td class="td11" id="TBL-8-2-2" style="text-align:left; white-space:nowrap;"> BARRIER                                                          </td></tr><tr id="TBL-8-3-" style="vertical-align:baseline;"><td class="td 11" id="TBL-8-3-1" style="text-align:left; white-space:nowrap;"> newHillFrequency               </td><td class="td11" id="TBL-8-3-2" style="text-align:left; white-space:nowrap;"> PACE                                                                 </td>
</tr><tr id="TBL-8-4-" style="vertical-align:baseline;"><td class="td11" id="TBL-8-4-1" style="text-align:left; white-space:nowrap;"> gaussianSigma                  </td><td class="td11" id="TBL-8-4-2" style="text-align:left; white-space:nowrap;"> SIGMA                                                               </td>
</tr><tr id="TBL-8-5-" style="vertical-align:baseline;"><td class="td11" id="TBL-8-5-1" style="text-align:left; white-space:nowrap;"> gaussianSigmaMin             </td><td class="td11" id="TBL-8-5-2" style="text-align:left; white-space:nowrap;"> SIGMA_MIN                                                      </td>
</tr><tr id="TBL-8-6-" style="vertical-align:baseline;"><td class="td11" id="TBL-8-6-1" style="text-align:left; white-space:nowrap;"> kernelCutoff                        </td><td class="td11" id="TBL-8-6-2" style="text-align:left; white-space:nowrap;"> KERNEL_CUTOFF                                            </td></tr><tr id="TBL-8-7-" style="vertical-align:baseline;"><td class="td 11" id="TBL-8-7-1" style="text-align:left; white-space:nowrap;"> compressionThreshold       </td><td class="td11" id="TBL-8-7-2" style="text-align:left; white-space:nowrap;"> COMPRESSION_THRESHOLD                       </td>
</tr><tr id="TBL-8-8-" style="vertical-align:baseline;"><td class="td11" id="TBL-8-8-1" style="text-align:left; white-space:nowrap;"> adaptiveSigma                   </td><td class="td11" id="TBL-8-8-2" style="text-align:left; white-space:nowrap;"> (use SIGMA=ADAPTIVE)                                  </td>
</tr><tr id="TBL-8-9-" style="vertical-align:baseline;"><td class="td11" id="TBL-8-9-1" style="text-align:left; white-space:nowrap;"> adaptiveSigmaStride          </td><td class="td11" id="TBL-8-9-2" style="text-align:left; white-space:nowrap;"> ADAPTIVE_SIGMA_STRIDE                            </td>
</tr><tr id="TBL-8-10-" style="vertical-align:baseline;"><td class="td11" id="TBL-8-10-1" style="text-align:left; white-space:nowrap;"> neighborList                       </td><td class="td11" id="TBL-8-10-2" style="text-align:left; white-space:nowrap;"> NLIST                                                                </td>
</tr><tr id="TBL-8-11-" style="vertical-align:baseline;"><td class="td11" id="TBL-8-11-1" style="text-align:left; white-space:nowrap;"> neighborListNewHillReset  </td><td class="td11" id="TBL-8-11-2" style="text-align:left; white-space:nowrap;"> NLIST_PACE_RESET                                       </td>
</tr><tr id="TBL-8-12-" style="vertical-align:baseline;"><td class="td11" id="TBL-8-12-1" style="text-align:left; white-space:nowrap;"> neighborListParameters     </td><td class="td11" id="TBL-8-12-2" style="text-align:left; white-space:nowrap;"> NLIST_PARAMETERS                                      </td>
</tr><tr id="TBL-8-13-" style="vertical-align:baseline;"><td class="td11" id="TBL-8-13-1" style="text-align:left; white-space:nowrap;"> noZed                                 </td><td class="td11" id="TBL-8-13-2" style="text-align:left; white-space:nowrap;"> NO_ZED                                                            </td>
</tr><tr id="TBL-8-14-" style="vertical-align:baseline;"><td class="td11" id="TBL-8-14-1" style="text-align:left; white-space:nowrap;"> fixedGaussianSigma          </td><td class="td11" id="TBL-8-14-2" style="text-align:left; white-space:nowrap;"> FIXED_SIGMA                                                  </td>
</tr><tr id="TBL-8-15-" style="vertical-align:baseline;"><td class="td11" id="TBL-8-15-1" style="text-align:left; white-space:nowrap;"> recursiveMerge                  </td><td class="td11" id="TBL-8-15-2" style="text-align:left; white-space:nowrap;"> (the opposite of RECURSIVE_MERGE_OFF)  </td>
</tr><tr id="TBL-8-16-" style="vertical-align:baseline;"><td class="td11" id="TBL-8-16-1" style="text-align:left; white-space:nowrap;"> calcWork                            </td><td class="td11" id="TBL-8-16-2" style="text-align:left; white-space:nowrap;"> CALC_WORK                                                    </td>
</tr><tr id="TBL-8-17-" style="vertical-align:baseline;"><td class="td11" id="TBL-8-17-1" style="text-align:left; white-space:nowrap;"> multipleReplicas                 </td><td class="td11" id="TBL-8-17-2" style="text-align:left; white-space:nowrap;"> WALKERS_MPI                                                 </td></tr></table></div>
</div><hr class="endfloat"/>
</div>
<!--l. 6860--><p class="indent">   The PLUMED options for restarting (<span class="ectt-1095">STATE_RFILE</span>, <span class="ectt-1095">STATE_WFILE </span>and <span class="ectt-1095">STATE_WSTRIDE</span>) are not
necessary in Colvars, since Colvars has a unified mechanism to save the information required for
restarting in the <span class="ectt-1095">.colvars.state </span>file.
</p>
<div class="flushright">
<!--l. 6862--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 6862--><p class="noindent"> </p><div class="heading"> <a id="sec:colvarbias_opes_general_options"></a>
<h5 class="subsubsectionHead"><span class="titlemark">6.7.2    </span> <a id="sec:colvarbias_opes_general_options"></a>General options</h5>
<!--l. 6862--><p class="noindent"></p></div> To enable a OPES-based calculation, a <span class="ectt-1095">opes_metad {...} </span>block must be included in the Colvars
configuration file. The <span class="ectt-1095">opes_metad </span>block supports the following options: 
     <ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox358"><a id="dx1-143001"></a><span class="ectt-1095">name</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">name</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox359"><a id="dx1-143002"></a><span class="ectt-1095">colvars</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">colvars</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox360"><a id="dx1-143003"></a><span class="ectt-1095">outputEnergy</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">outputEnergy</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="opes_metad|barrier"></a> <span class="colorbox" id="colorbox361"><a id="dx1-143004"></a><span class="ectt-1095">barrier</span></span> — <span class="phvro8t-x-x-109">The highest free-energy barrier you wish to overcome (kcal/mol)</span><br class="newline"/><!--l. 6875--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     decimal, context: <span class="ectt-1095">opes_metad</span> <!--l. 6875--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This parameter should be set to be at least equal to the highest free energy barrier you wish
     to overcome. If it is much lower than that, you will not cross the barrier, if it is much higher,
     convergence might take a little longer. If the system has a basin that is clearly more stable
     than the others, it is better to start the simulation from there.
     </li>
<li class="itemize"><a id="opes_metad|newHillFrequency"></a> <span class="colorbox" id="colorbox362"><a id="dx1-143005"></a><span class="ectt-1095">newHillFrequency</span></span> — <span class="phvro8t-x-x-109">Frequency of hill creation</span><br class="newline"/><!--l. 6883--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     integer, context: <span class="ectt-1095">opes_metad</span> <!--l. 6883--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>The frequency for kernel deposition.
     </li>
<li class="itemize"><a id="opes_metad|biasfactor"></a> <span class="colorbox" id="colorbox363"><a id="dx1-143006"></a><span class="ectt-1095">biasfactor</span></span> — <span class="phvro8t-x-x-109">The </span><!--l. 6892--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>γ</mi></math>
<span class="phvro8t-x-x-109">biasfactor used for the well-tempered target distribution</span><br class="newline"/>Default: <!--l. 6892--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="normal"><mi>b</mi><mi>a</mi><mi>r</mi><mi>r</mi><mi>i</mi><mi>e</mi><mi>r</mi></mstyle><mo class="MathClass-bin">∕</mo><msub><mrow><mi>k</mi></mrow><mrow><mstyle mathvariant="normal"><mi>B</mi></mstyle></mrow></msub><mi>T</mi> </math> <!--l. 6892--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     decimal or “inf", context: <span class="ectt-1095">opes_metad</span> <!--l. 6892--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option sets the <!--l. 6892--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>γ</mi></math>
     in Eq.(<a href="#eq:opes_potential">41<!--tex4ht:ref: eq:opes_potential --></a>). If setting to “inf", the target distribution is uniform. It cannot be “inf" in the exploration
     mode.
     </li>
<li class="itemize"><a id="opes_metad|sigma"></a> <span class="colorbox" id="colorbox364"><a id="dx1-143007"></a><span class="ectt-1095">gaussianSigma</span></span> — <span class="phvro8t-x-x-109">The initial widths of the kernels (divided by the square root of gamma in</span>
<span class="phvro8t-x-x-109">exploration mode)</span><br class="newline"/><!--l. 6900--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     decimal, context: <span class="ectt-1095">opes_metad</span> <!--l. 6900--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option sets the initial <!--l. 6900--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>σ</mi></math>
     values  of  the  Gaussian  kernels.  The  number  of  <span class="ectt-1095">gaussianSigma </span>values  should  match  the
     number of CVs used by <span class="ectt-1095">colvars</span>. If <a href="#opes_metad|adaptive_sigma"><span class="ectt-1095">adaptiveSigma</span></a> is set, then this option can be ignored.
     </li>
<li class="itemize"><a id="opes_metad|sigma_min"></a> <span class="colorbox" id="colorbox365"><a id="dx1-143008"></a><span class="ectt-1095">gaussianSigmaMin</span></span> — <span class="phvro8t-x-x-109">The minimum widths of the kernels (divided by the square root of</span>
<span class="phvro8t-x-x-109">gamma in exploration mode)</span><br class="newline"/><!--l. 6908--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     decimal (optional), context: <span class="ectt-1095">opes_metad</span> <!--l. 6908--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option sets the minimum <!--l. 6908--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>σ</mi></math>
     values of the Gaussian kernels. The number of <span class="ectt-1095">gaussianSigmaMin </span>values should match the
     number of CVs used by <span class="ectt-1095">colvars</span>. The OPES method scales down the <!--l. 6908--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>σ</mi></math>
     during the simulation, and will never shrink <!--l. 6908--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>σ</mi></math>
     under <span class="ectt-1095">gaussianSigmaMin </span>if this option is set.
     </li>
<li class="itemize"><a id="opes_metad|explore"></a> <span class="colorbox" id="colorbox366"><a id="dx1-143009"></a><span class="ectt-1095">explore</span></span> — <span class="phvro8t-x-x-109">Exploration mode</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 6917--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">opes_metad</span> <!--l. 6917--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If enabled, the OPES exploration mode will be used (see Ref.<span class="cite">[<a href="#XInvernizzi2022">44</a>]</span> for more information about
     the exploration mode).
     </li>
<li class="itemize"><a id="opes_metad|epsilon"></a> <span class="colorbox" id="colorbox367"><a id="dx1-143010"></a><span class="ectt-1095">epsilon</span></span> — <span class="phvro8t-x-x-109">The value of the regularization constant for the probability</span><br class="newline"/>Default: <!--l. 6926--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi class="qopname">exp</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--><mo class="MathClass-open">(</mo><mo class="MathClass-bin">−</mo><mstyle mathvariant="normal"><mi>b</mi><mi>a</mi><mi>r</mi><mi>r</mi><mi>i</mi><mi>e</mi><mi>r</mi></mstyle><mo class="MathClass-bin">∕</mo><msub><mrow><mi>k</mi></mrow><mrow><mstyle mathvariant="normal"><mi>B</mi></mstyle></mrow></msub><mi>T</mi> <mo class="MathClass-bin">∕</mo><mo class="MathClass-open">(</mo><mn>1</mn><mo class="MathClass-bin">−</mo><mn>1</mn><mo class="MathClass-bin">∕</mo><mi>γ</mi><mo class="MathClass-close">)</mo><mo class="MathClass-close">)</mo></math> <!--l. 6926--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     decimal, context: <span class="ectt-1095">opes_metad</span> <!--l. 6926--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option sets the <!--l. 6926--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>𝜖</mi></math>
     in Eq.(<a href="#eq:opes_potential">41<!--tex4ht:ref: eq:opes_potential --></a>).
     </li>
<li class="itemize"><a id="opes_metad|kernel_cutoff"></a> <span class="colorbox" id="colorbox368"><a id="dx1-143011"></a><span class="ectt-1095">kernelCutoff</span></span> — <span class="phvro8t-x-x-109">Truncate kernels at this distance, in units of </span><!--l. 6935--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>σ</mi></math><br class="newline"/>Default: <!--l. 6935--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msqrt><mrow><mn>2</mn><mo class="MathClass-bin">∗</mo><mstyle mathvariant="normal"><mi>b</mi><mi>a</mi><mi>r</mi><mi>r</mi><mi>i</mi><mi>e</mi><mi>r</mi> </mstyle> <mo class="MathClass-bin">∕</mo><msub><mrow><mi>k</mi></mrow><mrow><mstyle mathvariant="normal"><mi>B</mi></mstyle> </mrow> </msub> <mi>T</mi> <mo class="MathClass-bin">∕</mo><mo class="MathClass-open">(</mo><mn>1</mn><mo class="MathClass-bin">−</mo><mn>1</mn><mo class="MathClass-bin">∕</mo><mi>γ</mi><mo class="MathClass-close">)</mo></mrow></msqrt></math> <!--l. 6935--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     decimal, context: <span class="ectt-1095">opes_metad</span> <!--l. 6935--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If the value of <!--l. 6935--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow><mo class="MathClass-open">(</mo><mstyle mathvariant="bold-italic"><mi>ξ</mi></mstyle><mo class="MathClass-bin">−</mo><msup><mrow><mstyle mathvariant="bold-italic"><mi>ξ</mi></mstyle></mrow><mrow><mi>′</mi></mrow></msup><mo class="MathClass-close">)</mo></mrow><mrow><mi>T</mi> </mrow></msup><msup><mrow><mi mathvariant="normal">Σ</mi></mrow><mrow><mo class="MathClass-bin">−</mo><mn>1</mn></mrow></msup><mo class="MathClass-open">(</mo><mstyle mathvariant="bold-italic"><mi>ξ</mi></mstyle><mo class="MathClass-bin">−</mo><msup><mrow><mstyle mathvariant="bold-italic"><mi>ξ</mi></mstyle></mrow><mrow><mi>′</mi></mrow></msup><mo class="MathClass-close">)</mo></math>
     is larger than the square of this value, then <!--l. 6935--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>G</mi><mo class="MathClass-open">(</mo><mstyle mathvariant="bold-italic"><mi>ξ</mi></mstyle><mo class="MathClass-punc">,</mo><msup><mrow><mstyle mathvariant="bold-italic"><mi>ξ</mi></mstyle></mrow><mrow><mi>′</mi></mrow></msup><mo class="MathClass-close">)</mo></math>
     is evaluated as zero.
                                                                                         
                                                                                         
     </li>
<li class="itemize"><a id="opes_metad|compression_threshold"></a> <span class="colorbox" id="colorbox369"><a id="dx1-143012"></a><span class="ectt-1095">compressionThreshold</span></span> — <span class="phvro8t-x-x-109">Merge kernels if closer than this threshold, in units of </span><!--l. 6944--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>σ</mi></math><br class="newline"/>Default: 1.0 <!--l. 6944--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     decimal, context: <span class="ectt-1095">opes_metad</span> <!--l. 6944--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Kernels will be merged the distances of their centers (scaled by the square of <!--l. 6944--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>σ</mi></math>)
     are closer than this threshold.
     </li>
<li class="itemize"><a id="opes_metad|adaptive_sigma"></a> <span class="colorbox" id="colorbox370"><a id="dx1-143013"></a><span class="ectt-1095">adaptiveSigma</span></span> — <span class="phvro8t-x-x-109">Adaptively compute the bandwidths</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 6953--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">opes_metad</span> <!--l. 6953--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If enabled, the values of <!--l. 6953--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>σ</mi></math>
     of CVs are computed adaptively based on their variances. See the Supporting Information
     of Ref.<span class="cite">[<a href="#XInvernizzi2022">44</a>]</span> for more details.
     </li>
<li class="itemize"><a id="opes_metad|adaptive_sigma_stride"></a> <span class="colorbox" id="colorbox371"><a id="dx1-143014"></a><span class="ectt-1095">adaptiveSigmaStride</span></span> — <span class="phvro8t-x-x-109">Number of steps for measuring adaptive sigma</span><br class="newline"/>Default: <!--l. 6962--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>1</mn><mn>0</mn><mo class="MathClass-bin">×</mo><mstyle mathvariant="normal"><mi>n</mi><mi>e</mi><mi>w</mi><mi>H</mi><mi>i</mi><mi>l</mi><mi>l</mi><mi>F</mi><mi>r</mi><mi>e</mi><mi>q</mi><mi>u</mi><mi>e</mi><mi>n</mi><mi>c</mi><mi>y</mi></mstyle></math> <!--l. 6962--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     integer, context: <span class="ectt-1095">opes_metad</span> <!--l. 6962--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option sets the number of steps for measuring the rolling variances of CVs that are used
     by the adaptive kernel bandwidth algorithm. See the Supporting Information of Ref.<span class="cite">[<a href="#XInvernizzi2022">44</a>]</span> for
     more details.
     </li>
<li class="itemize"><a id="opes_metad|nlist"></a> <span class="colorbox" id="colorbox372"><a id="dx1-143015"></a><span class="ectt-1095">neighborList</span></span> — <span class="phvro8t-x-x-109">Use neighbor list for kernels summation (experimental)</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 6971--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">opes_metad</span> <!--l. 6971--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If enabled, a neighbor list will be used for kernels summation.
     </li>
<li class="itemize"><a id="opes_metad|nlist_pace_reset"></a> <span class="colorbox" id="colorbox373"><a id="dx1-143016"></a><span class="ectt-1095">neighborListNewHillReset</span></span> —  <span class="phvro8t-x-x-109">Force  the  reset  of  the  neighbor  list  at  each  time  when</span>
<span class="phvro8t-x-x-109">depositing a new hill</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 6980--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">opes_metad</span> <!--l. 6980--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If enabled, the neighbor list will be rebuilt at each time when depositing a new hill.
     </li>
<li class="itemize"><a id="opes_metad|nlist_param"></a> <span class="colorbox" id="colorbox374"><a id="dx1-143017"></a><span class="ectt-1095">neighborListParameters</span></span> — <span class="phvro8t-x-x-109">The two cutoff parameters of the neighbor list</span><br class="newline"/>Default: 3.0 0.5 <!--l. 6989--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     decimals, context: <span class="ectt-1095">opes_metad</span> <!--l. 6989--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>The first parameter is the maximum distance of neighboring kernels. The second parameters
     is the threshold for rebuilding.
     </li>
<li class="itemize"><a id="opes_metad|no_zed"></a> <span class="colorbox" id="colorbox375"><a id="dx1-143018"></a><span class="ectt-1095">noZed</span></span> — <span class="phvro8t-x-x-109">Do not normalize the probability over the explored CV space</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 6998--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">opes_metad</span> <!--l. 6998--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If enabled, <!--l. 6998--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>Z</mi></mrow><mrow><mi>n</mi></mrow></msub></math>
     is always 1.
     </li>
<li class="itemize"><a id="opes_metad|fixed_sigma"></a> <span class="colorbox" id="colorbox376"><a id="dx1-143019"></a><span class="ectt-1095">fixedSigma</span></span> — <span class="phvro8t-x-x-109">Do not scale down </span><!--l. 7007--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="bold-italic"><mi>σ</mi></mstyle></math><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 7007--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">opes_metad</span> <!--l. 7007--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If enabled, <!--l. 7007--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="bold-italic"><mi>σ</mi></mstyle></math>
     will not be decreased as the simulation proceeds.
     </li>
<li class="itemize"><a id="opes_metad|recursive_merge"></a> <span class="colorbox" id="colorbox377"><a id="dx1-143020"></a><span class="ectt-1095">recursiveMerge</span></span> — <span class="phvro8t-x-x-109">Merge the kernels recursively</span><br class="newline"/>Default: <span class="ectt-1095">on</span> <!--l. 7016--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">opes_metad</span> <!--l. 7016--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If enabled, the kernels will be compressed recursively when a new kernel is added.
     </li>
<li class="itemize"><a id="opes_metad|calc_work"></a> <span class="colorbox" id="colorbox378"><a id="dx1-143021"></a><span class="ectt-1095">calcWork</span></span> — <span class="phvro8t-x-x-109">Calculate the work done by the bias</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 7025--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">opes_metad</span> <!--l. 7025--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If enabled, the total accumulated work will be computed.</li></ul>
<div class="flushright">
<!--l. 7028--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 7028--><p class="noindent"> </p><div class="heading"> <a id="sec:colvarbias_opes_multiple_walker_options"></a>
<h5 class="subsubsectionHead"><span class="titlemark">6.7.3    </span> <a id="sec:colvarbias_opes_multiple_walker_options"></a>Multiple-walker options</h5>
<!--l. 7028--><p class="noindent"></p></div>
<!--l. 7030--><p class="indent">   The following options can be used for the multiple-walker OPES, and require the direct
communication between replicas through MPI or a Charm++ platform that supports partitions:
</p>
<ul class="itemize1">
<li class="itemize"><a id="opes_metad|multipleReplicas"></a> <span class="colorbox" id="colorbox379"><a id="dx1-144001"></a><span class="ectt-1095">multipleReplicas</span></span> — <span class="phvro8t-x-x-109">Enable multiple-walker OPES</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 7040--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">opes_metad</span> <!--l. 7040--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option turns on multiple-walker communication between replicas.
                                                                                         
                                                                                         
     </li>
<li class="itemize"><a id="opes_metad|replicaID"></a> <span class="colorbox" id="colorbox380"><a id="dx1-144002"></a><span class="ectt-1095">replicaID</span></span> — <span class="phvro8t-x-x-109">Set the identifier for this replica (required only for independent jobs)</span><br class="newline"/>Default: replica index) <!--l. 7052--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> string,
     context: <span class="ectt-1095">opes_metad</span> <!--l. 7052--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If <a href="#opes_metad|multipleReplicas"><span class="ectt-1095">multipleReplicas</span></a> is <span class="ectt-1095">on</span>, this option sets a unique identifier for this replicas. Specifying this
     option is thus only required when the replicas are launched as independent computations:
     when  the  replicas  share  a  common  parallel  communication  framework  (i.e. they  are  all
     launched together as a single message-passing computation via MPIor Charm++) the default
     value of this keyword is the replica's numeric index (zero-based).
     </li>
<li class="itemize"><a id="opes_metad|sharedFreq"></a> <span class="colorbox" id="colorbox381"><a id="dx1-144003"></a><span class="ectt-1095">sharedFreq</span></span> — <span class="phvro8t-x-x-109">Frequency (in number of steps) at which force samples are synchronized</span>
<span class="phvro8t-x-x-109">among the replicas</span><br class="newline"/>Default: <a href="#colvarbias|outputFreq"><span class="ectt-1095">outputFreq</span></a> <!--l. 7062--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     integer or zero, context: <span class="ectt-1095">opes_metad</span> <!--l. 7062--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Every <span class="ectt-1095">sharedFreq </span>steps, the replicas communicate the samples that have been gathered
     since the last synchronization time. Because this forces the replicas to synchronize, small
     values of this parameter may slightly reduce the overall throughput, especially on systems
     with unequal performance. </li></ul>
<div class="flushright">
<!--l. 7065--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 7065--><p class="noindent"> </p><div class="heading"> <a id="sec:colvarbias_opes_output_options"></a>
<h5 class="subsubsectionHead"><span class="titlemark">6.7.4    </span> <a id="sec:colvarbias_opes_output_options"></a>Output options</h5>
<!--l. 7065--><p class="noindent"></p></div>
<!--l. 7067--><p class="indent">   The following options are used for setting the PMF and trajectory output. The PMF is calculated based
on reweighting the CVs specified by <a href="#opes_metad|pmf_colvars"><span class="ectt-1095">pmfColvars</span></a>, which collects on-the-fly the biasing energy
<!--l. 7067--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>V</mi> <mo class="MathClass-open">(</mo><mstyle mathvariant="bold-italic"><mi>ξ</mi></mstyle><mo class="MathClass-close">)</mo></math> and the
values of CVs of every step, and builds a weighted histogram to calculate the unbiased probability of
<!--l. 7067--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mstyle mathvariant="bold-italic"><mi>ξ</mi></mstyle></math> </p><table class="equation"><tr><td>
<a id="x1-145001r45"></a>
<!--l. 7068--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mo class="MathClass-open">⟨</mo><mi mathvariant="bold-script">𝒫</mi><mrow><mo fence="true" form="prefix"> (</mo><mrow><msup><mrow><mstyle mathvariant="bold-italic"><mi>ξ</mi></mstyle></mrow><mrow><mi>′</mi></mrow></msup></mrow><mo fence="true" form="postfix">)</mo></mrow><mo class="MathClass-close">⟩</mo> <mo class="MathClass-rel">=</mo><mfrac><mrow> <munderover accent="false" accentunder="false"><mrow><mo>∑</mo></mrow><mrow><mi>t</mi><mo class="MathClass-rel">=</mo><mn>0</mn></mrow><mrow><mi>N</mi></mrow></munderover><mi>δ</mi> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mstyle mathvariant="bold-italic"><mi>ξ</mi></mstyle> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mstyle mathvariant="bold"><mi>x</mi></mstyle></mrow><mo fence="true" form="postfix">)</mo></mrow> <mo class="MathClass-bin">−</mo><msup><mrow><mstyle mathvariant="bold-italic"><mi>ξ</mi></mstyle></mrow><mrow><mi>′</mi></mrow></msup></mrow><mo fence="true" form="postfix">)</mo></mrow><mi class="qopname">exp</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mfrac><mrow><mi>V</mi> <mo class="MathClass-open">(</mo><mstyle mathvariant="bold-italic"><mi>ξ</mi></mstyle><mo class="MathClass-close">)</mo></mrow>
<mrow><msub><mrow><mi>k</mi></mrow><mrow><mstyle mathvariant="normal"><mi>B</mi></mstyle></mrow></msub><mi>T</mi> </mrow></mfrac> </mrow><mo fence="true" form="postfix">)</mo></mrow></mrow>
<mrow><munderover accent="false" accentunder="false"><mrow><mo>∑</mo></mrow><mrow><mi>t</mi><mo class="MathClass-rel">=</mo><mn>0</mn></mrow><mrow><mi>N</mi></mrow></munderover><mi class="qopname">exp</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mfrac><mrow><mi>V</mi> <mo class="MathClass-open">(</mo><mstyle mathvariant="bold-italic"><mi>ξ</mi></mstyle><mo class="MathClass-close">)</mo></mrow>
<mrow><msub><mrow><mi>k</mi></mrow><mrow><mstyle mathvariant="normal"><mi>B</mi></mstyle></mrow></msub><mi>T</mi> </mrow></mfrac> </mrow><mo fence="true" form="postfix">)</mo></mrow></mrow></mfrac>
</math></td><td class="eq-no">(45)</td></tr></table>
<!--l. 7071--><p class="indent">   where <!--l. 7071--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>N</mi></math> is the total number
of simulation steps, <!--l. 7071--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow><mstyle mathvariant="bold-italic"><mi>ξ</mi></mstyle></mrow><mrow><mi>′</mi></mrow></msup></math> is the grid
center in the histogram, and <!--l. 7071--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>δ</mi></math>
is the dirac delta function. The PMF is then obtained by </p><table class="equation"><tr><td> <a id="x1-145002r46"></a>
<!--l. 7072--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>A</mi><mo class="MathClass-open">(</mo><msup><mrow><mstyle mathvariant="bold-italic"><mi>ξ</mi></mstyle></mrow><mrow><mi>′</mi></mrow></msup><mo class="MathClass-close">)</mo> <mo class="MathClass-rel">=</mo> <mo class="MathClass-bin">−</mo><msub><mrow><mi>k</mi></mrow><mrow><mstyle mathvariant="normal">
<mi>B</mi></mstyle></mrow></msub><mi>T</mi> <mi class="qopname">ln</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--><mo class="MathClass-open">⟨</mo><mi mathvariant="bold-script">𝒫</mi><mo class="MathClass-open">(</mo><msup><mrow><mstyle mathvariant="bold-italic"><mi>ξ</mi></mstyle></mrow><mrow><mi>′</mi></mrow></msup><mo class="MathClass-close">)</mo><mo class="MathClass-close">⟩</mo><mo class="MathClass-bin">+</mo><msub><mrow><mi>A</mi></mrow><mrow>
<mn>0</mn></mrow></msub>
</math></td><td class="eq-no">(46)</td></tr></table>
<!--l. 7075--><p class="indent">   where <!--l. 7075--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>A</mi></mrow><mrow><mn>0</mn></mrow></msub></math> is a
constant to ensure <!--l. 7075--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>A</mi><mo class="MathClass-open">(</mo><msup><mrow><mstyle mathvariant="bold-italic"><mi>ξ</mi></mstyle></mrow><mrow><mi>′</mi></mrow></msup><mo class="MathClass-close">)</mo></math>
on every grid of the histogram is not negative.
</p><!--l. 7077--><p class="indent">   The grid used for discretizing the free energy surface (PMF) computated by OPES is defined by the
relevant options of the colvars. Alternately, grid parameters can be set directly inside the <span class="ectt-1095">opes_metad</span>
block by inserting a <span class="ectt-1095">grid </span>sub-block, as explained in <a href="#sec:colvarbias_grid">6.1<!--tex4ht:ref: sec:colvarbias_grid --></a>.
</p>
<ul class="itemize1">
<li class="itemize"><a id="opes_metad|pmf"></a> <span class="colorbox" id="colorbox382"><a id="dx1-145003"></a><span class="ectt-1095">pmf</span></span> — <span class="phvro8t-x-x-109">Output the PMF</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 7089--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">opes_metad</span> <!--l. 7089--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option turns on the PMF output based on reweighting.
     </li>
<li class="itemize"><a id="opes_metad|pmf_colvars"></a> <span class="colorbox" id="colorbox383"><a id="dx1-145004"></a><span class="ectt-1095">pmfColvars</span></span> — <span class="phvro8t-x-x-109">The CVs used for estimating the PMF</span><br class="newline"/><!--l. 7097--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     decimal, context: <span class="ectt-1095">opes_metad</span> <!--l. 7097--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This  option  sets  the  CVs  used  for  the  PMF.  The  CVs  specified  in  this  option  must  have
     appeared in <a href="#colvarbias|colvars"><span class="ectt-1095">colvars</span></a> and had the boundaries and widths.
     </li>
<li class="itemize"><a id="opes_metad|pmf_hist_freq"></a> <span class="colorbox" id="colorbox384"><a id="dx1-145005"></a><span class="ectt-1095">pmfHistoryFrequency</span></span> — <span class="phvro8t-x-x-109">The frequency of saving the PMF history file</span><br class="newline"/>Default: 0 <!--l. 7106--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     integer, context: <span class="ectt-1095">opes_metad</span> <!--l. 7106--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If  this  option  is  larger  than  0  and  <a href="#opes_metad|pmf"><span class="ectt-1095">pmf</span></a> is  on,  then  every  <span class="ectt-1095">pmfHistoryFrequency </span>steps  the
     current PMF will also be saved into “<span class="phvro8t-x-x-109">outputName</span><span class="ectt-1095">.colvars.</span><!--l. 7106--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">name</span><!--l. 7106--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><span class="ectt-1095">.</span><!--l. 7106--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">replicaID</span><!--l. 7106--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><span class="ectt-1095">.hist.pmf</span>".
     </li>
<li class="itemize"><a id="opes_metad|pmf_shared"></a> <span class="colorbox" id="colorbox385"><a id="dx1-145006"></a><span class="ectt-1095">pmfShared</span></span> — <span class="phvro8t-x-x-109">Output a global PMF for multiple-walker OPES</span><br class="newline"/>Default: <span class="ectt-1095">on</span> <!--l. 7115--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">opes_metad</span> <!--l. 7115--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If <a href="#opes_metad|multipleReplicas"><span class="ectt-1095">multipleReplicas</span></a> is on and this option is also on, then a global reweighting histogram will
     be built to collect samples from all replicas, and the corresponding global PMF is written into
     “<span class="phvro8t-x-x-109">outputName</span><span class="ectt-1095">.colvars.</span><!--l. 7115--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">name</span><!--l. 7115--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><span class="ectt-1095">.</span><!--l. 7115--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">replicaID</span><!--l. 7115--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><span class="ectt-1095">.global.pmf</span>"
     by replica 0. The history global PMF is also written to “<span class="phvro8t-x-x-109">outputName</span><span class="ectt-1095">.colvars.</span><!--l. 7115--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">name</span><!--l. 7115--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><span class="ectt-1095">.</span><!--l. 7115--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">replicaID</span><!--l. 7115--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><span class="ectt-1095">.global.hist.pmf</span>"
     by replica 0.
     </li>
<li class="itemize"><a id="opes_metad|print_trajectory_frequency"></a> <span class="colorbox" id="colorbox386"><a id="dx1-145007"></a><span class="ectt-1095">printTrajectoryFrequency</span></span> — <span class="phvro8t-x-x-109">The frequency of saving the trajectories</span><br class="newline"/>Default: <a href="#Colvars-global|colvarsTrajFrequency"><span class="ectt-1095">colvarsTrajFrequency</span></a> <!--l. 7126--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     integer, context: <span class="ectt-1095">opes_metad</span> <!--l. 7126--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If this option is larger than 0, then every <span class="ectt-1095">printTrajectoryFrequency</span>, the simulation time in
     ps, the values of CVs, the biasing energy, <!--l. 7126--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>c</mi><mo class="MathClass-open">(</mo><mi>t</mi><mo class="MathClass-close">)</mo></math>,
     <!--l. 7126--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>Z</mi></mrow><mrow><mi>n</mi></mrow></msub></math>
     (if <a href="#opes_metad|no_zed"><span class="ectt-1095">noZed</span></a> is off), the effective sample size (<!--l. 7126--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>N</mi></mrow><mrow><mstyle mathvariant="normal"><mi>e</mi><mi>ff</mi></mstyle></mrow></msub></math>
     in Ref.<span class="cite">[<a href="#XInvernizzi2020">43</a>]</span>), the work (if <a href="#opes_metad|calc_work"><span class="ectt-1095">calcWork</span></a> is on), the total number of compressed kernels, the number
     of kernels in the neighbor list (if <a href="#opes_metad|nlist"><span class="ectt-1095">neighborList</span></a> is on), and the number of steps from the last
     neighbor list update (if <a href="#opes_metad|nlist"><span class="ectt-1095">neighborList</span></a> is on) will be written to <br class="newline"/>“<span class="phvro8t-x-x-109">outputName</span><span class="ectt-1095">.colvars.</span><!--l. 7126--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">name</span><!--l. 7126--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><span class="ectt-1095">.</span><!--l. 7126--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">replicaID</span><!--l. 7126--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><span class="ectt-1095">.misc.traj</span>".
     </li></ul>
<!--l. 7129--><p class="indent">   Similar to the output file specified by the <span class="ectt-1095">FILE </span>option in the PLUMED implementation, the output file
“<span class="phvro8t-x-x-109">outputName</span><span class="ectt-1095">.colvars.</span><!--l. 7129--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">name</span><!--l. 7129--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><span class="ectt-1095">.</span><!--l. 7129--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">replicaID</span><!--l. 7129--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><span class="ectt-1095">.kernels.dat</span>"
includes all deposited uncompressed kernels (the step number of deposition, the kernel center, the Gaussian kernel
<!--l. 7129--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>σ</mi></math>, the height of the kernel,
and the biasing energy in <!--l. 7129--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>k</mi></mrow><mrow><mstyle mathvariant="normal"><mi>B</mi></mstyle></mrow></msub><mi>T</mi> </math>).
The format of <span class="ectt-1095">.kernels.dat </span>files manages to be compatible with the post-processing tools provided in
<a href="https://www.plumed.org/doc-v2.7/user-doc/html/opes-metad.html">the PLUMED OPES tutorial</a>, so that it is possible to run <span class="ectt-1095">State_from_Kernels.py </span>and then
<span class="ectt-1095">FES_from_State.py </span>to get the PMF along the biased CVs. Besides the <a href="#opes_metad|pmf"><span class="ectt-1095">pmf</span></a> option, there are
two ways to manually estimate the PMF, namely (i) reweighting the trajectories using the
biasing energy either in the <span class="ectt-1095">.colvars.traj </span>file (with <span class="ectt-1095">outputEnergy</span>) or the <a href="#opes_metad|print_trajectory_frequency"><span class="ectt-1095">.misc.traj</span></a> file (the
<!--l. 7129--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">name</span><!--l. 7129--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><span class="ectt-1095">.bias</span>
column), and (ii) summing up all the kernels (see Ref.<span class="cite">[<a href="#XInvernizzi2020">43</a>]</span> for more information).
</p>
<div class="flushright">
<!--l. 7131--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 7131--><p class="noindent"> </p><div class="heading"> <a id="sec:colvarbias_opes_example"></a>
<h5 class="subsubsectionHead"><span class="titlemark">6.7.5    </span> <a id="sec:colvarbias_opes_example"></a>Example input</h5>
<!--l. 7131--><p class="noindent"></p></div>
<!--l. 7133--><p class="noindent"><span class="phvb8t-x-x-109">Example: </span>OPES with adaptive kernel bandwidth, neighbor list and PMF based on reweighting.<br class="newline"/></p>
<div class="mdframed" id="mdframed-65">
<!--l. 7134--><p class="noindent"></p><pre>colvarsTrajFrequency    500 <br class="newline"/><br class="newline"/>colvar { <br class="newline"/>  name phi <br class="newline"/>  lowerBoundary -180 <br class="newline"/>  upperBoundary 180 <br class="newline"/>  width 5.0 <br class="newline"/>  dihedral { <br class="newline"/>    group1 {atomNumbers { 5 }}<br class="newline"/>    group2 {atomNumbers { 7 }}<br class="newline"/>    group3 {atomNumbers { 9 }}<br class="newline"/>    group4 {atomNumbers { 15}}<br class="newline"/>  } <br class="newline"/>} <br class="newline"/><br class="newline"/>colvar { <br class="newline"/>  name psi <br class="newline"/>  lowerBoundary -180 <br class="newline"/>  upperBoundary 180 <br class="newline"/>  width 5.0 <br class="newline"/>  dihedral { <br class="newline"/>    group1 {atomNumbers { 7 }} <br class="newline"/>    group2 {atomNumbers { 9 }} <br class="newline"/>    group3 {atomNumbers { 15 }} <br class="newline"/>    group4 {atomNumbers { 17 }} <br class="newline"/>  } <br class="newline"/>} <br class="newline"/><br class="newline"/>opes_metad { <br class="newline"/>  colvars phi psi <br class="newline"/>  newHillFrequency 500 <br class="newline"/>  barrier 11.950286806883364 <br class="newline"/>  adaptiveSigma on <br class="newline"/>  neighborList on <br class="newline"/>  printTrajectoryFrequency 500 <br class="newline"/>  pmf on <br class="newline"/>  pmfColvars phi psi <br class="newline"/>  pmfHistoryFrequency 1000 <br class="newline"/>  outputEnergy on <br class="newline"/>} </pre><!--l. 7175--><p class="noindent"></p>
</div>
<div class="flushright">
<!--l. 7178--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 7178--><p class="noindent"> </p><div class="heading"> <a id="sec:colvarbias_harmonic"></a>
<h4 class="subsectionHead"><span class="titlemark">6.8    </span> <a id="sec:colvarbias_harmonic"></a>Harmonic restraints</h4>
<!--l. 7178--><p class="noindent"></p></div>
<!--l. 7180--><p class="indent">   The harmonic biasing method may be used to enforce fixed or moving restraints, including variants of
Steered and Targeted MD. Within energy minimization runs, it allows for restrained minimization, e.g. to
calculate relaxed potential energy surfaces. In the context of the Colvars module, harmonic potentials are
meant according to their textbook definition: </p><table class="equation"><tr><td> <a id="x1-147001r47"></a>
<!--l. 7185--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>V</mi> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mi>ξ</mi> </mrow><mo fence="true" form="postfix">)</mo></mrow> <mo class="MathClass-rel">=</mo> <mfrac><mrow><mn>1</mn></mrow>
<mrow><mn>2</mn></mrow></mfrac><mi>k</mi><msup><mrow> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mfrac><mrow><mi>ξ</mi> <mo class="MathClass-bin">−</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mn>0</mn></mrow></msub></mrow>
<mrow><msub><mrow><mi>w</mi></mrow><mrow><mi>ξ</mi></mrow></msub></mrow></mfrac> </mrow><mo fence="true" form="postfix">)</mo></mrow></mrow><mrow><mn>2</mn></mrow></msup>
</math></td><td class="eq-no">(47)</td></tr></table>
<!--l. 7189--><p class="indent">   There are two noteworthy aspects of this expression:
     </p><ol class="enumerate1">
<li class="enumerate" id="x1-147003x1">Because the <span class="phvro8t-x-x-109">standard coefficient of </span><!--l. 7191--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>1</mn><mo class="MathClass-bin">∕</mo><mn>2</mn></math>
     of the harmonic potential is included, this expression differs from harmonic bond and angle
     potentials historically used in common force fields, where the factor was typically omitted
     resulting in a non-standard definition of the force constant.
     </li>
<li class="enumerate" id="x1-147005x2">The variable <!--l. 7192--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>
     is not only centered at <!--l. 7192--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mn>0</mn></mrow></msub></math>,
     but is also <span class="phvro8t-x-x-109">scaled by its characteristic length scale </span><!--l. 7192--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>w</mi></mrow><mrow><mi>ξ</mi></mrow></msub></math>
     (keyword <a href="#colvar|width"><span class="ectt-1095">width</span></a>). The resulting dimensionless variable <!--l. 7193--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi> <mo class="MathClass-rel">=</mo> <mo class="MathClass-open">(</mo><mi>ξ</mi> <mo class="MathClass-bin">−</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-close">)</mo><mo class="MathClass-bin">∕</mo><msub><mrow><mi>w</mi></mrow><mrow><mi>ξ</mi></mrow></msub></math>
     is typically easier to treat numerically: for example, when the forces typically experienced by
     <!--l. 7193--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>
     are much smaller than <!--l. 7193--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>k</mi><mo class="MathClass-bin">∕</mo><msub><mrow><mi>w</mi></mrow><mrow><mi>ξ</mi></mrow></msub></math>
     and <!--l. 7193--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>k</mi></math>
     is chosen equal to <!--l. 7193--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>κ</mi></mrow><mrow><mstyle mathvariant="normal"><mi>B</mi></mstyle></mrow></msub><mi>T</mi> </math>
     (thermal energy), the resulting probability distribution of <!--l. 7193--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>z</mi></math>
     is approximately a Gaussian with mean equal to 0 and standard deviation equal to 1.
     <!--l. 7195--><p class="noindent">This property can be used for setting the force constant in umbrella-sampling ensemble runs:
     if the restraint centers are chosen in increments of <!--l. 7195--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>w</mi></mrow><mrow><mi>ξ</mi></mrow></msub></math>,
                                                                                         
                                                                                         
     the resulting distributions of <!--l. 7195--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>
     are most often optimally overlapped. In regions where the underlying free-energy landscape
     induces highly skewed distributions of <!--l. 7196--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>,
     additional windows may be added as needed, with spacings finer than <!--l. 7196--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>w</mi></mrow><mrow><mi>ξ</mi></mrow></msub></math>.</p></li></ol>
<!--l. 7199--><p class="indent">   Beyond one dimension, the use of a scaled harmonic potential also allows a standard definition of a
multi-dimensional restraint with a unified force constant: </p><table class="equation"><tr><td> <a id="eq:colvarbias_harmonic_multi"></a>
<!--l. 7200--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>V</mi> <mrow><mo fence="true" form="prefix"> (</mo><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-punc">,</mo><mi class="MathClass-op">…</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo><mo class="MathClass-punc">,</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>M</mi></mrow></msub></mrow><mo fence="true" form="postfix">)</mo></mrow> <mo class="MathClass-rel">=</mo> <mfrac><mrow><mn>1</mn></mrow>
<mrow><mn>2</mn></mrow></mfrac><mi>k</mi><munderover accent="false" accentunder="false"><mrow><mo>∑</mo></mrow><mrow><mi>i</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><mi>M</mi></mrow></munderover><msup><mrow> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mfrac><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-bin">−</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mn>0</mn></mrow></msub></mrow>
<mrow><msub><mrow><mi>w</mi></mrow><mrow><mi>ξ</mi></mrow></msub></mrow></mfrac> </mrow><mo fence="true" form="postfix">)</mo></mrow></mrow><mrow><mn>2</mn></mrow></msup>
</math></td><td class="eq-no">(48)</td></tr></table>
<!--l. 7205--><p class="indent">   If one-dimensional or homogeneous multi-dimensional restraints are defined, and there are no other uses for the
parameter <!--l. 7205--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>w</mi></mrow><mrow><mi>ξ</mi></mrow></msub></math>, <span class="ectt-1095">width </span><span class="phvro8t-x-x-109">can be</span>
<span class="phvro8t-x-x-109">left at its default value of </span><!--l. 7205--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>1</mn></math>.
</p><!--l. 7207--><p class="indent">   A harmonic restraint is defined by a <span class="ectt-1095">harmonic</span><span class="ectt-1095"> {...} </span>block, which may contain the following keywords:
</p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox387"><a id="dx1-147007"></a><span class="ectt-1095">name</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">name</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox388"><a id="dx1-147008"></a><span class="ectt-1095">colvars</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">colvars</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox389"><a id="dx1-147009"></a><span class="ectt-1095">outputEnergy</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">outputEnergy</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox390"><a id="dx1-147010"></a><span class="ectt-1095">writeTIPMF</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">writeTIPMF</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox391"><a id="dx1-147011"></a><span class="ectt-1095">writeTISamples</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">writeTISamples</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox392"><a id="dx1-147012"></a><span class="ectt-1095">stepZeroData</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">stepZeroData</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><a id="harmonic|forceConstant"></a> <span class="colorbox" id="colorbox393"><a id="dx1-147013"></a><span class="ectt-1095">forceConstant</span></span> — <span class="phvro8t-x-x-109">Scaled force constant (kcal/mol)</span><br class="newline"/>Default: <span class="ectt-1095">1.0</span> <!--l. 7231--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     decimal, context: <span class="ectt-1095">harmonic</span> <!--l. 7231--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option defines a <span class="phvro8t-x-x-109">scaled </span>force constant <!--l. 7231--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>k</mi></math>
     for the harmonic potential (eq. <a href="#eq:colvarbias_harmonic_multi">48<!--tex4ht:ref: eq:colvarbias_harmonic_multi --></a>). To ensure consistency for multidimensional restraints,
     it is divided internally by the square of the specific <span class="ectt-1095">width </span>of each variable (which is 1 by
     default).  This  makes  all  values  effectively  dimensionless  and  of  commensurate  size.  For
                                                                                         
                                                                                         
     instance, if this force constant is set to the thermal energy <!--l. 7231--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>κ</mi></mrow><mrow><mstyle mathvariant="normal"><mi>B</mi></mstyle></mrow></msub><mi>T</mi> </math>
     (equal to <!--l. 7231--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>R</mi><mi>T</mi> </math>
     if  molar  units  are  used),  then  the  amplitude  of  the  thermal  fluctuations  of  each  variable
     <!--l. 7231--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>
     will be on the order of its <span class="ectt-1095">width</span>, <!--l. 7231--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>w</mi></mrow><mrow><mi>ξ</mi></mrow></msub></math>.
     This can be used to estimate the optimal spacing of umbrella-sampling windows (under the
     assumption that the force constant is larger than the curvature of the underlying free energy).
     <span class="phvro8t-x-x-109">The values of the actual force constants </span><!--l. 7231--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>k</mi><mo class="MathClass-bin">∕</mo><msubsup><mrow><mi>w</mi></mrow><mrow><mi>ξ</mi></mrow><mrow><mn>2</mn></mrow></msubsup></math>
<span class="phvro8t-x-x-109">are always printed when the restraint is defined.</span>
</li>
<li class="itemize"><a id="harmonic|centers"></a> <span class="colorbox" id="colorbox394"><a id="dx1-147014"></a><span class="ectt-1095">centers</span></span> — <span class="phvro8t-x-x-109">Initial harmonic restraint centers</span><br class="newline"/><!--l. 7247--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> space-separated
     list of colvar values, context: <span class="ectt-1095">harmonic</span> <!--l. 7247--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>The centers (equilibrium values) of the restraint, <!--l. 7247--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mn>0</mn></mrow></msub></math>,
     are entered here. The number of values must be the number of requested colvars. Each
     value is a decimal number if the corresponding colvar returns a scalar, a “<span class="ectt-1095">(x, y, z)</span>" triplet
     if it returns a unit vector or a vector, and a “<span class="ectt-1095">(q0, q1, q2, q3)</span>" quadruplet if it returns a
     rotational quaternion. If a colvar has periodicities or symmetries, its closest image to the
     restraint center is considered when calculating the harmonic potential.</li></ul>
<!--l. 7250--><p class="indent"> <span class="phvb8t-x-x-109">Tip: </span>A complex set of restraints can be applied to a system, by defining several colvars, and applying
one or more harmonic restraints to different groups of colvars. In some cases, dozens of colvars can be
defined, but their value may not be relevant: to limit the size of the colvars trajectory file, it may be wise to
disable <span class="ectt-1095">outputValue </span>for such “ancillary" variables, and leave it enabled only for “relevant"
ones.
</p>
<div class="flushright">
<!--l. 7259--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 7259--><p class="noindent"> </p><div class="heading"> <a id="sec:colvarbias_smd"></a>
<h5 class="subsubsectionHead"><span class="titlemark">6.8.1    </span> <a id="sec:colvarbias_smd"></a>Moving restraints: steered molecular dynamics</h5>
<!--l. 7259--><p class="noindent"></p></div>
<!--l. 7261--><p class="indent">   The following options allow to change gradually the centers of the harmonic restraints during a
simulations. When the centers are changed continuously, a steered MD in a collective variable space is
carried out.
</p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox395"><a id="dx1-148001"></a><span class="ectt-1095">targetCenters</span></span> — <span class="phvro8t-x-x-109">Steer the restraint centers towards these targets</span><br class="newline"/><!--l. 7289--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> space-separated
     list of colvar values, context: <span class="ectt-1095">harmonic</span> <!--l. 7289--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>When defined, the current <span class="ectt-1095">centers </span>will be moved towards these values during the simulation.
     By default, the centers are moved over a total of <span class="ectt-1095">targetNumSteps </span>steps by a linear interpolation,
     in the spirit of Steered MD. If <span class="ectt-1095">targetNumStages </span>is set to a nonzero value, the change is
     performed in discrete stages, lasting <span class="ectt-1095">targetNumSteps </span>steps <span class="phvro8t-x-x-109">each</span>. This second mode may
     be used to sample successive windows in the context of an Umbrella Sampling simulation.
     When continuing a simulation run, the <span class="ectt-1095">centers </span>specified in the configuration file <!--l. 7289--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">colvarsConfig</span><!--l. 7289--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>
     are overridden by those saved in the restart file <!--l. 7289--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">colvarsInput</span><!--l. 7289--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math>.
     To perform Steered MD in an arbitrary space of colvars, it is sufficient to use this option
     and enable <span class="ectt-1095">outputAccumulatedWork </span>and/or <span class="ectt-1095">outputAppliedForce </span>within each of the colvars
     involved.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox396"><a id="dx1-148002"></a><span class="ectt-1095">targetNumSteps</span></span> — <span class="phvro8t-x-x-109">Number of steps for steering</span><br class="newline"/><!--l. 7303--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     integer, context: <span class="ectt-1095">harmonic</span> <!--l. 7303--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>In single-stage (continuous) transformations, defines the number of MD steps required to
     move the restraint centers (or force constant) towards the values specified with <span class="ectt-1095">targetCenters</span>
     or <span class="ectt-1095">targetForceConstant</span>. After the target values have been reached, the centers (resp. force
     constant) are kept fixed. In multi-stage transformations, this sets the number of MD steps <span class="phvro8t-x-x-109">per</span>
<span class="phvro8t-x-x-109">stage</span>.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox397"><a id="dx1-148003"></a><span class="ectt-1095">outputCenters</span></span> — <span class="phvro8t-x-x-109">Write the current centers to the trajectory file</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 7314--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">harmonic</span> <!--l. 7314--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If this option is chosen and <span class="ectt-1095">colvarsTrajFrequency </span>is not zero, the positions of the restraint
     centers will be written to the trajectory file during the simulation. This option allows to conveniently
     extract the PMF from the Colvars trajectory files in a steered MD calculation.
     </li></ul>
<!--l. 7318--><p class="indent"> <span class="phvb8t-x-x-109">Note on restarting moving restraint simulations: </span>Information about the current step and stage of a
simulation with moving restraints is stored in the restart file (state file). Thus, such simulations can be run
in several chunks, and restarted directly using the same colvars configuration file. In case of a restart, the
values of parameters such as <span class="ectt-1095">targetCenters</span>, <span class="ectt-1095">targetNumSteps</span>, etc. should not be changed
manually.
</p>
<div class="flushright">
<!--l. 7327--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 7327--><p class="noindent"> </p><div class="heading"> <a id="sec:colvarbias_us"></a>
<h5 class="subsubsectionHead"><span class="titlemark">6.8.2    </span> <a id="sec:colvarbias_us"></a>Moving restraints: umbrella sampling</h5>
<!--l. 7327--><p class="noindent"></p></div>
<!--l. 7329--><p class="indent">   The centers of the harmonic restraints can also be changed in discrete stages: in this cases a
one-dimensional umbrella sampling simulation is performed. The sampling windows in simulation are
calculated in sequence. The colvars trajectory file may then be used both to evaluate the correlation times
between consecutive windows, and to calculate the frequency distribution of the colvar of interest in each
window. Furthermore, frequency distributions on a predefined grid can be automatically obtained by using
the <span class="ectt-1095">histogram </span>bias (see <a href="#sec:colvarbias_histogram">6.13<!--tex4ht:ref: sec:colvarbias_histogram --></a>).
</p><!--l. 7334--><p class="indent">   To activate an umbrella sampling simulation, the same keywords as in the previous section can be
used, with the addition of the following: </p>
<ul class="itemize1">
<li class="itemize"><a id="harmonic|targetNumStages"></a> <span class="colorbox" id="colorbox398"><a id="dx1-149001"></a><span class="ectt-1095">targetNumStages</span></span> — <span class="phvro8t-x-x-109">Number of stages for steering</span><br class="newline"/>Default: <span class="ectt-1095">0</span> <!--l. 7349--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> non-negative
     integer, context: <span class="ectt-1095">harmonic</span> <!--l. 7349--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If  non-zero,  sets  the  number  of  stages  in  which  the  restraint  centers  or  force  constant
     are  changed  to  their  target  values.  If  zero,  the  change  is  continuous.  Each  stage  lasts
     <span class="ectt-1095">targetNumSteps </span>MD steps. To sample both ends of the transformation, the simulation should
     be run for <span class="ectt-1095">targetNumSteps</span><!--l. 7349--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-bin">×</mo></math>
     (<span class="ectt-1095">targetNumStages </span>+ 1).
     </li></ul>
<div class="flushright">
<!--l. 7354--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 7354--><p class="noindent"> </p><div class="heading"> <a id="sec:colvarbias_force_k_ti"></a>
<h5 class="subsubsectionHead"><span class="titlemark">6.8.3    </span> <a id="sec:colvarbias_force_k_ti"></a>Changing force constant</h5>
<!--l. 7354--><p class="noindent"></p></div>
<!--l. 7356--><p class="indent">   The force constant of the harmonic restraint may also be changed over time, either to introduce or
remove a restraint smoothly, or to estimate a restraint free energy<span class="cite">[<a href="#XDeng2009">46</a>]</span>.
</p>
<ul class="itemize1">
<li class="itemize"><a id="harmonic|targetForceConstant"></a> <span class="colorbox" id="colorbox399"><a id="dx1-150001"></a><span class="ectt-1095">targetForceConstant</span></span> — <span class="phvro8t-x-x-109">Change the force constant towards this value</span><br class="newline"/><!--l. 7387--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     decimal, context: <span class="ectt-1095">harmonic</span> <!--l. 7387--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>When defined, the current <span class="ectt-1095">forceConstant </span>will be moved towards this value during the simulation.
     Time  evolution  of  the  force  constant  is  dictated  by  the  <span class="ectt-1095">lambdaExponent </span>parameter  (see
     below). By default, the force constant is changed smoothly over a total of <span class="ectt-1095">targetNumSteps</span>
     steps. This is useful to introduce or remove restraints in a progressive manner. If <span class="ectt-1095">targetNumStages</span>
     is set to a nonzero value, the change is performed in discrete stages, lasting <span class="ectt-1095">targetNumSteps</span>
     steps <span class="phvro8t-x-x-109">each</span>. This second mode may be used to compute the conformational free energy
                                                                                         
                                                                                         
     change associated with the restraint, within the FEP or TI formalisms. For convenience, the
     code provides an estimate of the free energy derivative for use in TI, with the format:<br class="newline"/><span class="ectt-1095">colvars: Lambda= ***.** dA/dLambda= ***.**</span><br class="newline"/>A more complete free energy calculation (particularly with regard to convergence analysis),
     while not handled by the Colvars module, can be performed by post-processing the colvars
     trajectory,  if  <span class="ectt-1095">colvarsTrajFrequency </span>is  set  to  a  suitably  small  value.  It  should  be  noted,
     however, that restraint free energy calculations may be handled more efficiently by an indirect
     route, through the determination of a free energy profile for the restrained coordinate.<span class="cite">[<a href="#XDeng2009">46</a>]</span>
</li>
<li class="itemize"><span class="colorbox" id="colorbox400"><a id="dx1-150002"></a><span class="ectt-1095">lambdaExponent</span></span> — <span class="phvro8t-x-x-109">Exponent in the time-dependence of the force constant</span><br class="newline"/>Default: <span class="ectt-1095">1.0</span> <!--l. 7412--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> decimal
     equal to or greater than 1.0, context: <span class="ectt-1095">harmonic</span> <!--l. 7412--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Sets the exponent, <!--l. 7412--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>α</mi></math>,
     in the function used to vary the force constant as a function of time. The force is varied
     according to a coupling parameter <!--l. 7412--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>λ</mi></math>,
     raised to the power <!--l. 7412--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>α</mi></math>:
     <!--l. 7412--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>k</mi></mrow><mrow><mi>λ</mi></mrow></msub> <mo class="MathClass-rel">=</mo> <msub><mrow><mi>k</mi></mrow><mrow><mn>0</mn></mrow></msub> <mo class="MathClass-bin">+</mo><msup><mrow><mi>λ</mi></mrow><mrow><mi>α</mi></mrow></msup><mo class="MathClass-open">(</mo><msub><mrow><mi>k</mi></mrow><mrow><mn>1</mn></mrow></msub> <mo class="MathClass-bin">−</mo><msub><mrow><mi>k</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-close">)</mo></math>,
     where <!--l. 7412--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>k</mi></mrow><mrow><mn>0</mn></mrow></msub></math>,
     <!--l. 7412--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>k</mi></mrow><mrow><mi>λ</mi></mrow></msub></math>,
     and <!--l. 7412--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>k</mi></mrow><mrow><mn>1</mn></mrow></msub></math>
     are the initial, current, and final values of the force constant. The parameter <!--l. 7412--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>λ</mi></math>
     evolves linearly from 0 to 1, either smoothly, or in <span class="ectt-1095">targetNumStages </span>equally spaced discrete
     stages,  or  according  to  an  arbitrary  schedule  set  with  <span class="ectt-1095">lambdaSchedule</span>.  When  the  initial
     value of the force constant is zero, an exponent greater than 1.0 distributes the effects of
     introducing the restraint more smoothly over time than a linear dependence, and ensures
     that there is no singularity in the derivative of the restraint free energy with respect to lambda.
     A value of 4 has been found to give good results in some tests. To remove the restraint over
     time rather than introduce it, use the <span class="ectt-1095">decoupling </span>option.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox401"><a id="dx1-150003"></a><span class="ectt-1095">decoupling</span></span> — <span class="phvro8t-x-x-109">Perform a restraint decoupling transformation</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 7428--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">harmonic</span> <!--l. 7428--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>When this option is enabled, the restraint is decoupled by changing the force constant from
     <span class="ectt-1095">forceConstant </span>to 0.0 over the course of the simulation, as specified by <span class="ectt-1095">targetNumSteps </span>and,
     optionally, <span class="ectt-1095">targetNumStages</span>. The force constant is then varied as: <!--l. 7428--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>k</mi></mrow><mrow><mi>λ</mi></mrow></msub> <mo class="MathClass-rel">=</mo> <msup><mrow><mo class="MathClass-open">(</mo><mn>1</mn><mo class="MathClass-bin">−</mo><mi>λ</mi><mo class="MathClass-close">)</mo></mrow><mrow><mi>α</mi></mrow></msup><mi>k</mi></math>,
     where <!--l. 7428--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>α</mi></math>
     is the value of <span class="ectt-1095">lambdaExponent</span>. Setting <span class="ectt-1095">lambdaExponent </span>to a value greater than 1 (e.g. 4)
     ensures a smoother dependence of the energy on <!--l. 7428--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>λ</mi></math>
     around <!--l. 7428--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>k</mi></mrow><mrow><mi>λ</mi></mrow></msub> <mo class="MathClass-rel">=</mo> <mn>0</mn></math>.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox402"><a id="dx1-150004"></a><span class="ectt-1095">targetEquilSteps</span></span> — <span class="phvro8t-x-x-109">Number of steps discarded from TI estimate</span><br class="newline"/>Default: 0 <!--l. 7439--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     integer, context: <span class="ectt-1095">harmonic</span> <!--l. 7439--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Defines the number of steps within each stage that are considered equilibration and discarded
     from the restraint free energy derivative estimate reported reported in the output.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox403"><a id="dx1-150005"></a><span class="ectt-1095">lambdaSchedule</span></span> — <span class="phvro8t-x-x-109">Schedule of lambda-points for changing force constant</span><br class="newline"/><!--l. 7449--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> list
     of real numbers between 0 and 1, context: <span class="ectt-1095">harmonic</span> <!--l. 7449--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If specified together with targetForceConstant, sets the sequence of discrete <!--l. 7449--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>λ</mi></math>
     values that will be used for different stages.
     </li></ul>
<div class="flushright">
<!--l. 7454--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 7454--><p class="noindent"> </p><div class="heading"> <a id="sec:colvarbias_acc_work"></a>
<h4 class="subsectionHead"><span class="titlemark">6.9    </span> <a id="sec:colvarbias_acc_work"></a>Computing the work of a changing restraint</h4>
<!--l. 7454--><p class="noindent"></p></div>
<!--l. 7456--><p class="indent">   If the restraint centers or force constant are changed continuosly (<span class="ectt-1095">targetNumStages </span>undefined) it is
possible to record the net work performed by the changing restraint: </p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox404"><a id="dx1-151001"></a><span class="ectt-1095">outputAccumulatedWork</span></span> —  <span class="phvro8t-x-x-109">Write  the  accumulated  work  of  the  changing  restraint  to  the</span>
<span class="phvro8t-x-x-109">Colvars trajectory file</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 7468--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">harmonic</span> <!--l. 7468--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If <span class="ectt-1095">targetCenters </span>or <span class="ectt-1095">targetForceConstant </span>are defined and this option is enabled, the accumulated
     work from the beginning of the simulation will be written to the trajectory file (<span class="ectt-1095">colvarsTrajFrequency</span>
     must be non-zero). When the simulation is continued from a state file, the previously accumulated
     work is included in the integral. This option allows to conveniently extract the estimated PMF
     of a steered MD calculation (when <span class="ectt-1095">targetCenters </span>is used), or of other simulation protocols.
     </li></ul>
<div class="flushright">
<!--l. 7472--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 7472--><p class="noindent"> </p><div class="heading"> <a id="sec:colvarbias_harmonic_walls"></a>
<h4 class="subsectionHead"><span class="titlemark">6.10    </span> <a id="sec:colvarbias_harmonic_walls"></a>Harmonic wall restraints</h4>
<!--l. 7472--><p class="noindent"></p></div>
<!--l. 7474--><p class="indent">   The <span class="ectt-1095">harmonicWalls</span><span class="ectt-1095"> {...} </span>bias is closely related to the harmonic bias (see <a href="#sec:colvarbias_harmonic">6.8<!--tex4ht:ref: sec:colvarbias_harmonic --></a>), with the following two
differences: <span class="phvro8t-x-x-109">(i) </span>instead of a center a <span class="phvro8t-x-x-109">lower wall </span>and/or an <span class="phvro8t-x-x-109">upper wall </span>are defined, outside of which the bias
implements a half-harmonic potential; </p><table class="equation"><tr><td> <a id="x1-152001r49"></a>
<!--l. 7475--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>V</mi> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mi>ξ</mi> </mrow><mo fence="true" form="postfix">)</mo></mrow> <mo class="MathClass-rel">=</mo> <mrow><mo fence="true" form="prefix"> {</mo><mrow> <mtable align="axis" class="array" columnlines="none" equalcolumns="false" equalrows="false" style=""> <mtr><mtd class="array" columnalign="left"> <mfrac><mrow><mn>1</mn></mrow> <mrow><mn>2</mn></mrow></mfrac><mi>k</mi><msup><mrow> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mfrac><mrow><mi>ξ</mi><mo class="MathClass-bin">−</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mstyle mathvariant="normal"><mi>u</mi><mi>p</mi><mi>p</mi><mi>e</mi><mi>r</mi></mstyle></mrow></msub></mrow>
<mrow><msub><mrow><mi>w</mi></mrow><mrow><mi>ξ</mi> </mrow></msub></mrow></mfrac> </mrow><mo fence="true" form="postfix">)</mo></mrow> </mrow><mrow><mn>2</mn></mrow></msup> </mtd><mtd class="array" columnalign="left"> <mstyle mathvariant="normal"><mi>i</mi><mi>f</mi></mstyle><mspace class="nbsp" width="0.33em"></mspace><mi>ξ</mi> <mo class="MathClass-rel">&gt;</mo> <msub><mrow><mi>ξ</mi></mrow><mrow><mstyle mathvariant="normal"><mi>u</mi><mi>p</mi><mi>p</mi><mi>e</mi><mi>r</mi></mstyle></mrow></msub> </mtd>
</mtr> <mtr><mtd class="array" columnalign="left"> <mn>0</mn> </mtd><mtd class="array" columnalign="left"> <mstyle mathvariant="normal"><mi>i</mi><mi>f</mi></mstyle><mspace class="nbsp" width="0.33em"></mspace><msub><mrow><mi>ξ</mi></mrow><mrow><mstyle mathvariant="normal"><mi>l</mi><mi>o</mi><mi>w</mi><mi>e</mi><mi>r</mi></mstyle></mrow></msub> <mo class="MathClass-rel">≤</mo> <mi>ξ</mi> <mo class="MathClass-rel">≤</mo> <msub><mrow><mi>ξ</mi></mrow><mrow><mstyle mathvariant="normal"><mi>u</mi><mi>p</mi><mi>p</mi><mi>e</mi><mi>r</mi></mstyle></mrow></msub> </mtd>
</mtr> <mtr><mtd class="array" columnalign="left"> <mfrac><mrow><mn>1</mn></mrow> <mrow><mn>2</mn></mrow></mfrac><mi>k</mi><msup><mrow> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mfrac><mrow><mi>ξ</mi><mo class="MathClass-bin">−</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mstyle mathvariant="normal"><mi>l</mi><mi>o</mi><mi>w</mi><mi>e</mi><mi>r</mi></mstyle></mrow></msub></mrow>
<mrow><msub><mrow><mi>w</mi></mrow><mrow><mi>ξ</mi> </mrow></msub></mrow></mfrac> </mrow><mo fence="true" form="postfix">)</mo></mrow> </mrow><mrow><mn>2</mn></mrow></msup> </mtd><mtd class="array" columnalign="left"> <mstyle mathvariant="normal"><mi>i</mi><mi>f</mi></mstyle><mspace class="nbsp" width="0.33em"></mspace><mi>ξ</mi> <mo class="MathClass-rel">&lt;</mo> <msub><mrow><mi>ξ</mi></mrow><mrow><mstyle mathvariant="normal"><mi>l</mi><mi>o</mi><mi>w</mi><mi>e</mi><mi>r</mi></mstyle></mrow></msub> </mtd>
</mtr> <!--l l--></mtable> </mrow><mo fence="true" form="postfix"></mo></mrow>
</math></td><td class="eq-no">(49)</td></tr></table>
<!--l. 7485--><p class="indent">   where <!--l. 7485--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mstyle mathvariant="normal"><mi>l</mi><mi>o</mi><mi>w</mi><mi>e</mi><mi>r</mi></mstyle></mrow></msub></math>
and <!--l. 7485--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mstyle mathvariant="normal"><mi>u</mi><mi>p</mi><mi>p</mi><mi>e</mi><mi>r</mi></mstyle></mrow></msub></math> are
the lower and upper wall thresholds, respectively; <span class="phvro8t-x-x-109">(ii) </span>because an interval between two walls is defined,
only scalar variables can be used (but any number of variables can be defined, and the wall bias is
intrinsically multi-dimensional).<br class="newline"/>
</p><!--l. 7487--><p class="indent"> <span class="phvb8t-x-x-109">Note: </span>this bias replaces the keywords <span class="ectt-1095">lowerWall</span>, <span class="ectt-1095">lowerWallConstant</span>, <span class="ectt-1095">upperWall </span>and
<span class="ectt-1095">upperWallConstant </span>defined in the <span class="ectt-1095">colvar </span>context. Those keywords are deprecated.
</p><!--l. 7490--><p class="indent">   The <span class="ectt-1095">harmonicWalls </span>bias implements the following options: </p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox405"><a id="dx1-152002"></a><span class="ectt-1095">name</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">name</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox406"><a id="dx1-152003"></a><span class="ectt-1095">colvars</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">colvars</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox407"><a id="dx1-152004"></a><span class="ectt-1095">outputEnergy</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">outputEnergy</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox408"><a id="dx1-152005"></a><span class="ectt-1095">writeTIPMF</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">writeTIPMF</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox409"><a id="dx1-152006"></a><span class="ectt-1095">writeTISamples</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">writeTISamples</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox410"><a id="dx1-152007"></a><span class="ectt-1095">stepZeroData</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">stepZeroData</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox411"><a id="dx1-152008"></a><span class="ectt-1095">lowerWalls</span></span> — <span class="phvro8t-x-x-109">Position of the lower wall(s)</span><br class="newline"/><!--l. 7505--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> Space-separated
     list of decimals, context: <span class="ectt-1095">colvar</span> <!--l. 7505--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Defines the values <!--l. 7505--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mstyle mathvariant="normal"><mi>l</mi><mi>o</mi><mi>w</mi><mi>e</mi><mi>r</mi></mstyle></mrow></msub></math>
     below which a confining restraint on the colvar is applied to each colvar <!--l. 7505--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox412"><a id="dx1-152009"></a><span class="ectt-1095">upperWalls</span></span> — <span class="phvro8t-x-x-109">Position of the upper wall(s)</span><br class="newline"/><!--l. 7513--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> Space-separated
     list of decimals, context: <span class="ectt-1095">colvar</span> <!--l. 7513--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Defines the values <!--l. 7513--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>ξ</mi></mrow><mrow><mstyle mathvariant="normal"><mi>u</mi><mi>p</mi><mi>p</mi><mi>e</mi><mi>r</mi></mstyle></mrow></msub></math>
     above which a confining restraint on the colvar is applied to each colvar <!--l. 7513--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox413"><a id="dx1-152010"></a><span class="ectt-1095">forceConstant</span></span> — same definition as <a href="#sec:colvarbias_harmonic"><span class="ectt-1095">forceConstant</span></a> (Harmonic restraints)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox414"><a id="dx1-152011"></a><span class="ectt-1095">lowerWallConstant</span></span> — <span class="phvro8t-x-x-109">Force constant for the lower wall</span><br class="newline"/>Default: <span class="ectt-1095">forceConstant</span> <!--l. 7527--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     decimal, context: <span class="ectt-1095">harmonicWalls</span> <!--l. 7527--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>When both sets of walls are defined (lower and upper), this keyword allows setting different
     force constants for them. As with <span class="ectt-1095">forceConstant</span>, the specified constant is divided internally
     by the square of the specific <span class="ectt-1095">width </span>of each variable (see also the equivalent keyword for the
     harmonic restraint, <a href="#harmonic|forceConstant"><span class="ectt-1095">forceConstant</span></a>). The force constant reported in the output as “<!--l. 7527--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>k</mi></math>",
     and used in the change of force constant scheme, is the geometric mean of <span class="ectt-1095">upperWallConstant</span>
     and <span class="ectt-1095">upperWallConstant</span>.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox415"><a id="dx1-152012"></a><span class="ectt-1095">upperWallConstant</span></span> — analogous to <span class="ectt-1095">lowerWallConstant</span>
</li>
<li class="itemize"><span class="colorbox" id="colorbox416"><a id="dx1-152013"></a><span class="ectt-1095">targetForceConstant</span></span> — same definition as <a href="#sec:colvarbias_harmonic"><span class="ectt-1095">targetForceConstant</span></a> (harmonic restraints)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox417"><a id="dx1-152014"></a><span class="ectt-1095">targetForceConstant</span></span> — <span class="phvro8t-x-x-109">Change the force constant(s) towards this value</span><br class="newline"/><!--l. 7544--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     decimal, context: <span class="ectt-1095">harmonicWalls</span> <!--l. 7544--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This keyword allows changing either one or both of the wall force constants over time. In the
     case that <span class="ectt-1095">lowerWallConstant </span>and <span class="ectt-1095">upperWallConstant </span>have the same value, the behavior of
     this keyword is identical to the corresponding keyword in the <span class="ectt-1095">harmonic </span>restraint; otherwise,
     the  change  schedule  is  applied  to  the  geometric  mean  of  the  two  constant.  When  only
     one set of walls is defined (<span class="ectt-1095">lowerWall </span>or <span class="ectt-1095">upperWalls</span>), only the respective force constant
     is changed. <span class="phvb8t-x-x-109">Note: </span>if only one of the two force constants is meant to change over time, it is
     possible to use two instances of <span class="ectt-1095">harmonicWalls</span>, and apply the changing schedule only to
     one of them.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox418"><a id="dx1-152015"></a><span class="ectt-1095">targetLowerWalls</span></span> — <span class="phvro8t-x-x-109">Change the lower wall value(s) towards these values</span><br class="newline"/><!--l. 7555--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     decimal, context: <span class="ectt-1095">harmonicWalls</span> <!--l. 7555--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This keyword allows changing the lower wall position over time. By default, this change is
     continuous over <span class="ectt-1095">targetEquilSteps</span>. In that case, the non-equilibrium work performed can be
     obtained by enabling <span class="ectt-1095">outputAccumulatedWork</span>. If <span class="ectt-1095">targetNumStages </span>is provided, the change
     happens in the given number of increments. See <a href="#harmonic|targetNumStages"><span class="ectt-1095">targetNumStages</span></a> for details.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox419"><a id="dx1-152016"></a><span class="ectt-1095">targetUpperWalls</span></span> — <span class="phvro8t-x-x-109">Change the upper wall value(s) towards these values</span><br class="newline"/><!--l. 7564--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     decimal, context: <span class="ectt-1095">harmonicWalls</span> <!--l. 7564--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Analogous to <span class="ectt-1095">targetLowerWalls</span>.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox420"><a id="dx1-152017"></a><span class="ectt-1095">targetNumSteps</span></span> — same definition as <a href="#sec:colvarbias_harmonic"><span class="ectt-1095">targetNumSteps</span></a> (harmonic restraints)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox421"><a id="dx1-152018"></a><span class="ectt-1095">lambdaExponent</span></span> — same definition as <a href="#sec:colvarbias_harmonic"><span class="ectt-1095">lambdaExponent</span></a> (harmonic restraints)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox422"><a id="dx1-152019"></a><span class="ectt-1095">targetEquilSteps</span></span> — same definition as <a href="#sec:colvarbias_harmonic"><span class="ectt-1095">targetEquilSteps</span></a> (harmonic restraints)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox423"><a id="dx1-152020"></a><span class="ectt-1095">targetNumStages</span></span> — same definition as <a href="#sec:colvarbias_harmonic"><span class="ectt-1095">targetNumStages</span></a> (harmonic restraints)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox424"><a id="dx1-152021"></a><span class="ectt-1095">lambdaSchedule</span></span> — same definition as <a href="#sec:colvarbias_harmonic"><span class="ectt-1095">lambdaSchedule</span></a> (harmonic restraints)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox425"><a id="dx1-152022"></a><span class="ectt-1095">outputAccumulatedWork</span></span>  —   same   definition   as   <a href="#sec:colvarbias_harmonic"><span class="ectt-1095">outputAccumulatedWork</span></a>  (harmonic
     restraints)
     </li>
<li class="itemize"><a id="harmonicWalls|bypassExtendedLagrangian"></a> <span class="colorbox" id="colorbox426"><a id="dx1-152023"></a><span class="ectt-1095">bypassExtendedLagrangian</span></span> — <span class="phvro8t-x-x-109">Apply bias to actual colvars, bypassing extended coordinates</span><br class="newline"/>Default: <span class="ectt-1095">on</span> <!--l. 7584--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">harmonicWalls</span> <!--l. 7584--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option behaves as <a href="#colvarbias|bypassExtendedLagrangian"><span class="ectt-1095">bypassExtendedLagrangian</span></a> for other biases, but <span class="phvb8t-x-x-109">it defaults to </span><span class="ectt-1095">on</span><span class="phvb8t-x-x-109">,</span>
<span class="phvb8t-x-x-109">unlike in the general case</span>. Thus, by default, the <span class="ectt-1095">harmonicWalls </span>bias applies to the actual
     colvars, so that the distribution of the colvar between the walls is unaffected by the bias,
     which then applies a flat-bottom potential <span class="phvro8t-x-x-109">as a function of the colvar value</span>. This bias will
     affect  the  extended  coordinate  distribution  near  the  walls.  If  <span class="ectt-1095">bypassExtendedLagrangian</span>
     is  disabled,  <span class="ectt-1095">harmonicWalls </span>applies  a  flat-bottom  potential  <span class="phvro8t-x-x-109">as  a  function  of  the  extended</span>
<span class="phvro8t-x-x-109">coordinate</span>. Conversely, this bias will then modify the distribution of the actual colvar value
     near the walls.
     </li></ul>
<!--l. 7588--><p class="noindent"><span class="phvb8t-x-x-109">Example 1: </span>harmonic walls for one variable with two different force constants.
                                                                                         
                                                                                         
</p>
<div class="mdframed" id="mdframed-66">
<!--l. 7589--><p class="noindent"></p><pre>harmonicWalls {<br class="newline"/>  name  mywalls<br class="newline"/>  colvars    dist<br class="newline"/>  lowerWalls  22.0 <br class="newline"/>  upperWalls  38.0 <br class="newline"/>  lowerWallConstant  2.0 <br class="newline"/>  upperWallConstant 10.0 <br class="newline"/>} </pre><!--l. 7598--><p class="noindent"></p>
</div>
<!--l. 7600--><p class="noindent"><span class="phvb8t-x-x-109">Example 2: </span>harmonic walls for two variables with a single force constant.</p>
<div class="mdframed" id="mdframed-67">
<!--l. 7601--><p class="noindent"></p><pre>harmonicWalls {<br class="newline"/>  name  mywalls<br class="newline"/>  colvars       phi    psi<br class="newline"/>  lowerWalls -180.0    0.0<br class="newline"/>  upperWalls    0.0  180.0<br class="newline"/>  forceConstant 5.0 <br class="newline"/>} </pre><!--l. 7609--><p class="noindent"></p>
</div>
<div class="flushright">
<!--l. 7612--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 7612--><p class="noindent"> </p><div class="heading"> <a id="sec:colvarbias_linear"></a>
<h4 class="subsectionHead"><span class="titlemark">6.11    </span> <a id="sec:colvarbias_linear"></a>Linear restraints</h4>
<!--l. 7612--><p class="noindent"></p></div>
<!--l. 7614--><p class="indent">   The <span class="ectt-1095">linear </span>keyword defines a linear potential: </p><table class="equation"><tr><td> <a id="x1-153001r50"></a>
<!--l. 7615--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>V</mi> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mi>ξ</mi> </mrow><mo fence="true" form="postfix">)</mo></mrow> <mo class="MathClass-rel">=</mo> <mi>k</mi> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mfrac><mrow><mi>ξ</mi> <mo class="MathClass-bin">−</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mn>0</mn></mrow></msub></mrow>
<mrow><msub><mrow><mi>w</mi></mrow><mrow><mi>ξ</mi></mrow></msub></mrow></mfrac> </mrow><mo fence="true" form="postfix">)</mo></mrow>
</math></td><td class="eq-no">(50)</td></tr></table>
<!--l. 7619--><p class="noindent">whose force is simply given by the constant <!--l. 7619--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>k</mi><mo class="MathClass-bin">∕</mo><msub><mrow><mi>w</mi></mrow><mrow><mi>ξ</mi></mrow></msub></math>
itself: </p><table class="equation"><tr><td> <a id="x1-153002r51"></a>
<!--l. 7620--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>f</mi><mo class="MathClass-open">(</mo><mi>ξ</mi><mo class="MathClass-close">)</mo> <mo class="MathClass-rel">=</mo> <mi>k</mi><mo class="MathClass-bin">∕</mo><msub><mrow><mi>w</mi></mrow><mrow><mi>ξ</mi></mrow></msub>
</math></td><td class="eq-no">(51)</td></tr></table>
<!--l. 7624--><p class="noindent">This type of bias is therefore most useful in situations where a <span class="phvro8t-x-x-109">constant force </span>is desired. As all other
restraints, it can be defined on one or more CVs, with each contribution added to the total potential and the
parameters <!--l. 7624--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>w</mi></mrow><mrow><mi>ξ</mi></mrow></msub></math>
determining the relative magnitude for each.
</p><!--l. 7626--><p class="noindent"><span class="phvb8t-x-x-109">Example: </span>A possible use case of the <span class="ectt-1095">linear </span>bias is mimicking a constant electric field acting on a
specific particle, or the center of mass of many particles. In the following example, a linear restraint is
applied on a <span class="ectt-1095">distanceZ </span>variable (<a href="#sec:cvc_distanceZ">4.3.2<!--tex4ht:ref: sec:cvc_distanceZ --></a>), generating a constant force parallel to the Z axis of magnitude
5 kcal/mol/Å:</p>
<div class="mdframed" id="mdframed-68">
<!--l. 7627--><p class="noindent"></p><pre>colvar {<br class="newline"/>  name z<br class="newline"/>  distanceZ {<br class="newline"/>    ...<br class="newline"/>  }<br class="newline"/>}<br class="newline"/><br class="newline"/>linear {<br class="newline"/>  colvars z<br class="newline"/>  forceConstant 5.0<br class="newline"/>  centers 0.0<br class="newline"/>} </pre><!--l. 7640--><p class="noindent"></p>
</div>
<!--l. 7658--><p class="noindent"><span class="phvb8t-x-x-109">Example 2: </span>Another possible use case of a linear restraint is when the variable itself is a function with
well-defined minima or maxima, i.e. the CV itself is a restraint function, whose strength is given by the chosen
<!--l. 7658--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>k</mi></math>. For
example, consider a <span class="ectt-1095">mapTotal </span>variable (<a href="#sec:cvc_map_total">4.15.1<!--tex4ht:ref: sec:cvc_map_total --></a>), constructed from a 3D density map whose minima and
maxima are 0 and 1, and loaded into <a href="https://www.ks.uiuc.edu/Research/namd/2.14/ug/node43.html">GridForces</a>. The following configuration produces an energy
minimum of 10 kcal/mol/Å for each atom, located at the maxima of the map. This type of restraint is
similar to the MDFF protocol, but it also allows to evaluate the biasing energy introduced in the
simulation:</p>
<div class="mdframed" id="mdframed-69">
<!--l. 7658--><p class="noindent"></p><pre>colvar {<br class="newline"/>  name densitymap<br class="newline"/>  mapTotal {<br class="newline"/>    ...<br class="newline"/>  }<br class="newline"/>}<br class="newline"/><br class="newline"/>linear {<br class="newline"/>  colvars z<br class="newline"/>  forceConstant -10.0<br class="newline"/>  centers 0.0<br class="newline"/>} </pre><!--l. 7658--><p class="noindent"></p>
</div>
<!--l. 7660--><p class="indent">   Another useful application of a linear restraint is to enforce experimental constraints in a simulation,
with a lower non-equilibrium work than e.g. harmonic restraints <span class="cite">[<a href="#XPitera2012">47</a>]</span>. There is generally a unique strength
of bias for each CV center, which means you must know the bias force constant specifically for the center
of the CV. This force constant may be found by using experiment directed simulation described in section
<a href="#sec:colvarbias_alb">6.12<!--tex4ht:ref: sec:colvarbias_alb --></a>.
</p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox427"><a id="dx1-153003"></a><span class="ectt-1095">name</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">name</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox428"><a id="dx1-153004"></a><span class="ectt-1095">colvars</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">colvars</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox429"><a id="dx1-153005"></a><span class="ectt-1095">outputEnergy</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">outputEnergy</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox430"><a id="dx1-153006"></a><span class="ectt-1095">forceConstant</span></span> — <span class="phvro8t-x-x-109">Scaled force constant (kcal/mol)</span><br class="newline"/>Default: <span class="ectt-1095">1.0</span> <!--l. 7680--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> Decimal,
     context: <span class="ectt-1095">linear</span> <!--l. 7680--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This option defines a <span class="phvro8t-x-x-109">scaled </span>force constant <!--l. 7680--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>k</mi></math>
     for the linear bias. To ensure consistency for multidimensional restraints, it is divided internally
     by  the  specific  <span class="ectt-1095">width </span>of  each  variable  (which  is  1  by  default),  so  that  all  variables  are
     effectively dimensionless and of commensurate size. <span class="phvro8t-x-x-109">The values of </span><!--l. 7680--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>k</mi><mo class="MathClass-bin">∕</mo><msub><mrow><mi>w</mi></mrow><mrow><mi>ξ</mi></mrow></msub></math>
<span class="phvro8t-x-x-109">are always printed when the restraint is defined.</span>
</li>
<li class="itemize"><span class="colorbox" id="colorbox431"><a id="dx1-153007"></a><span class="ectt-1095">centers</span></span> — <span class="phvro8t-x-x-109">Initial linear restraint centers</span><br class="newline"/><!--l. 7689--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> space-separated
     list of colvar values, context: <span class="ectt-1095">linear</span> <!--l. 7689--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>These are analogous to the <a href="#harmonic|centers"><span class="ectt-1095">centers</span></a> keyword of the harmonic restraint. Although they do not
     affect dynamics, they are here necessary to ensure a well-defined energy for the linear bias.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox432"><a id="dx1-153008"></a><span class="ectt-1095">writeTIPMF</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">writeTIPMF</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox433"><a id="dx1-153009"></a><span class="ectt-1095">writeTISamples</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">writeTISamples</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox434"><a id="dx1-153010"></a><span class="ectt-1095">targetForceConstant</span></span> — same definition as <a href="#sec:colvarbias_harmonic"><span class="ectt-1095">targetForceConstant</span></a> (Harmonic restraints)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox435"><a id="dx1-153011"></a><span class="ectt-1095">targetNumSteps</span></span> — same definition as <a href="#sec:colvarbias_harmonic"><span class="ectt-1095">targetNumSteps</span></a> (Harmonic restraints)
                                                                                         
                                                                                         
     </li>
<li class="itemize"><span class="colorbox" id="colorbox436"><a id="dx1-153012"></a><span class="ectt-1095">lambdaExponent</span></span> — same definition as <a href="#sec:colvarbias_harmonic"><span class="ectt-1095">lambdaExponent</span></a> (Harmonic restraints)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox437"><a id="dx1-153013"></a><span class="ectt-1095">targetEquilSteps</span></span> — same definition as <a href="#sec:colvarbias_harmonic"><span class="ectt-1095">targetEquilSteps</span></a> (Harmonic restraints)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox438"><a id="dx1-153014"></a><span class="ectt-1095">targetNumStages</span></span> — same definition as <a href="#sec:colvarbias_harmonic"><span class="ectt-1095">targetNumStages</span></a> (Harmonic restraints)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox439"><a id="dx1-153015"></a><span class="ectt-1095">lambdaSchedule</span></span> — same definition as <a href="#sec:colvarbias_harmonic"><span class="ectt-1095">lambdaSchedule</span></a> (Harmonic restraints)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox440"><a id="dx1-153016"></a><span class="ectt-1095">outputAccumulatedWork</span></span>  —   same   definition   as   <a href="#sec:colvarbias_harmonic"><span class="ectt-1095">outputAccumulatedWork</span></a>  (Harmonic
     restraints)
     </li></ul>
<div class="flushright">
<!--l. 7704--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 7704--><p class="noindent"> </p><div class="heading"> <a id="sec:colvarbias_alb"></a>
<h4 class="subsectionHead"><span class="titlemark">6.12    </span> <a id="sec:colvarbias_alb"></a>Adaptive Linear Bias/Experiment Directed Simulation</h4>
<!--l. 7704--><p class="noindent"></p></div>
<!--l. 7706--><p class="indent">   Experiment directed simulation applies a linear bias with a changing force constant. Please cite White
and Voth <span class="cite">[<a href="#XWhite2014">48</a>]</span> when using this feature. As opposed to that reference, the force constant here is scaled by
the <span class="ectt-1095">width </span>corresponding to the biased colvar. In White and Voth, each force constant is scaled by the
colvars set center. The bias converges to a linear bias, after which it will be the minimal possible bias. You
may also stop the simulation, take the median of the force constants (ForceConst) found in the
colvars trajectory file, and then apply a linear bias with that constant. All the notes about units
described in sections <a href="#sec:colvarbias_linear">6.11<!--tex4ht:ref: sec:colvarbias_linear --></a> and <a href="#sec:colvarbias_harmonic">6.8<!--tex4ht:ref: sec:colvarbias_harmonic --></a> apply here as well. <span class="phvb8t-x-x-109">This is not a valid simulation of any</span>
<span class="phvb8t-x-x-109">particular statistical ensemble and is only an optimization algorithm until the bias has</span>
<span class="phvb8t-x-x-109">converged</span>.
</p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox441"><a id="dx1-154001"></a><span class="ectt-1095">name</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">name</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox442"><a id="dx1-154002"></a><span class="ectt-1095">colvars</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">colvars</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox443"><a id="dx1-154003"></a><span class="ectt-1095">centers</span></span> — <span class="phvro8t-x-x-109">Collective variable centers</span><br class="newline"/><!--l. 7738--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> space-separated
     list of colvar values, context: <span class="ectt-1095">alb</span> <!--l. 7738--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>The  desired  center  (equilibrium  values)  which  will  be  sought  during  the  adaptive  linear
     biasing.  The  number  of  values  must  be  the  number  of  requested  colvars.  Each  value  is
     a  decimal  number  if  the  corresponding  colvar  returns  a  scalar,  a  “<span class="ectt-1095">(x, y, z)</span>"  triplet  if  it
     returns a unit vector or a vector, and a “<span class="ectt-1095">q0, q1, q2, q3)</span>" quadruplet if it returns a rotational
     quaternion. If a colvar has periodicities or symmetries, its closest image to the restraint center
     is considered when calculating the linear potential.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox444"><a id="dx1-154004"></a><span class="ectt-1095">updateFrequency</span></span> — <span class="phvro8t-x-x-109">The duration of updates</span><br class="newline"/><!--l. 7756--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> An
     integer, context: <span class="ectt-1095">alb</span> <!--l. 7756--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This is, <!--l. 7756--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>N</mi></math>,
     the number of simulation steps to use for each update to the bias. This determines how long
     the system requires to equilibrate after a change in force constant (<!--l. 7756--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>N</mi><mo class="MathClass-bin">∕</mo><mn>2</mn></math>),
     how long statistics are collected for an iteration (<!--l. 7756--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>N</mi><mo class="MathClass-bin">∕</mo><mn>2</mn></math>),
     and how quickly energy is added to the system (at most, <!--l. 7756--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>A</mi><mo class="MathClass-bin">∕</mo><mn>2</mn><mi>N</mi></math>,
     where <!--l. 7756--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>A</mi></math>
     is  the  <span class="ectt-1095">forceRange</span>).  Until  the  force  constant  has  converged,  the  method  as  described  is
     an optimization procedure and not an integration of a particular statistical ensemble. It is
     important that each step should be uncorrelated from the last so that iterations are independent.
     Therefore, <!--l. 7756--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>N</mi></math>
     should be at least twice the autocorrelation time of the collective variable. The system should
     also be able to dissipate energy as fast as <!--l. 7756--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>N</mi><mo class="MathClass-bin">∕</mo><mn>2</mn></math>,
     which can be done by adjusting thermostat parameters. Practically, <!--l. 7756--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>N</mi></math>
     has  been  tested  successfully  at  significantly  shorter  than  the  autocorrelation  time  of  the
     collective variables being biased and still converge correctly.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox445"><a id="dx1-154005"></a><span class="ectt-1095">forceRange</span></span> — <span class="phvro8t-x-x-109">The expected range of the force constant in units of energy</span><br class="newline"/>Default: 3 <!--l. 7771--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>k</mi></mrow><mrow><mi>b</mi></mrow></msub><mi>T</mi> </math> <!--l. 7771--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> A
     space-separated list of decimal numbers, context: <span class="ectt-1095">alb</span> <!--l. 7771--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This is largest magnitude of the force constant which one expects. If this parameter is too low,
     the simulation will not converge. If it is too high the simulation will waste time exploring values
     that are too large. A value of 3 <!--l. 7771--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>k</mi></mrow><mrow><mi>b</mi></mrow></msub><mi>T</mi> </math>
     has worked well in the systems presented as a first choice. This parameter is dynamically
     adjusted over the course of a simulation. The benefit is that a bad guess for the forceRange
     can be corrected. However, this can lead to large amounts of energy being added over time
     to the system. To prevent this dynamic update, add <span class="ectt-1095">hardForceRange yes </span>as a parameter
     </li>
<li class="itemize"><span class="colorbox" id="colorbox446"><a id="dx1-154006"></a><span class="ectt-1095">rateMax</span></span> — <span class="phvro8t-x-x-109">The maximum rate of change of force constant</span><br class="newline"/><!--l. 7781--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> A
     list of space-separated real numbers, context: <span class="ectt-1095">alb</span> <!--l. 7781--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This optional parameter controls how much energy is added to the system from this bias.
     Tuning this separately from the <span class="ectt-1095">updateFrequency </span>and <span class="ectt-1095">forceRange </span>can allow for large bias
     changes but with a low <span class="ectt-1095">rateMax </span>prevents large energy changes that can lead to instability in
                                                                                         
                                                                                         
     the simulation.
     </li></ul>
<div class="flushright">
<!--l. 7786--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 7786--><p class="noindent"> </p><div class="heading"> <a id="sec:colvarbias_histogram"></a>
<h4 class="subsectionHead"><span class="titlemark">6.13    </span> <a id="sec:colvarbias_histogram"></a>Multidimensional histograms</h4>
<!--l. 7786--><p class="noindent"></p></div>
<!--l. 7788--><p class="indent">   The <span class="ectt-1095">histogram </span>feature is used to record the distribution of a set of collective variables in the form of a
N-dimensional histogram. Defining such a histogram is generally useful for analysis purposes, but it has
no effect on the simulation.
</p><!--l. 7790--><p class="noindent"><span class="phvb8t-x-x-109">Example: </span>the two-dimensional histogram of a distance and an angle can be generated
using the configuration below. The histogram code requires that each variable is a scalar
number that is confined within a pre-defined interval. The interval's boundaries may be
specified by hand (e.g. through <a href="#colvar|lowerBoundary"><span class="ectt-1095">lowerBoundary</span></a> and <a href="#colvar|upperBoundary"><span class="ectt-1095">upperBoundary</span></a> in the variable definition),
or auto-detected based on the type of function. In this example, the lower boundary for the
distance variable “<span class="ectt-1095">r</span>" is automatically set to zero, and interval for the three-body angle “<span class="ectt-1095">theta</span>" is
<!--l. 7790--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msup><mrow><mn>0</mn></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></math> and
<!--l. 7790--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>1</mn><mn>8</mn><msup><mrow><mn>0</mn></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></math>: however, that
an upper boundary for the distance “<span class="ectt-1095">r</span>" still needs to be specified manually. The grid spacings for the two variables
are <!--l. 7790--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>0</mn><mo class="MathClass-punc">.</mo><mn>2</mn></math> Åand
<!--l. 7790--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>3</mn><mo class="MathClass-punc">.</mo><msup><mrow><mn>0</mn></mrow><mrow><mo class="MathClass-bin">∘</mo></mrow></msup></math>,
respectively.</p>
<div class="mdframed" id="mdframed-70">
<!--l. 7793--><p class="noindent"></p><pre>colvar {<br class="newline"/>  name r<br class="newline"/>  width 0.2<br class="newline"/>  upperBoundary 20.0<br class="newline"/>  distance { ... }<br class="newline"/>}<br class="newline"/><br class="newline"/>colvar {<br class="newline"/>  name theta<br class="newline"/>  width 3.0<br class="newline"/>  dihedral { ... }<br class="newline"/>}<br class="newline"/><br class="newline"/>histogram {<br class="newline"/>  name hist2d<br class="newline"/>  colvars r theta<br class="newline"/>}<br class="newline"/></pre><!--l. 7811--><p class="noindent"></p>
</div>
<!--l. 7814--><p class="indent">   The standard keywords below are used to control the histogram's computation and to select the
variables that are sampled. See also <a href="#sec:colvarbias_grid">6.1<!--tex4ht:ref: sec:colvarbias_grid --></a> for keywords used to define the grid, <a href="#sec:colvarbias_histogram_output">6.13.1<!--tex4ht:ref: sec:colvarbias_histogram_output --></a> for output
parameters and <a href="#sec:colvarbias_histogram_vectors">6.13.2<!--tex4ht:ref: sec:colvarbias_histogram_vectors --></a> for more advanced keywords.
</p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox447"><a id="dx1-155001"></a><span class="ectt-1095">name</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">name</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox448"><a id="dx1-155002"></a><span class="ectt-1095">colvars</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">colvars</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox449"><a id="dx1-155003"></a><span class="ectt-1095">stepZeroData</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">stepZeroData</span></a> (biasing and analysis methods)
     </li></ul>
<!--l. 7825--><p class="indent">   Grid parameters for the histogram may be provided at the level of the individual variables, or via a
dedicated configuration block <span class="ectt-1095">grid { </span><span class="ectt-1095">…} </span>inside the configuration of this <span class="ectt-1095">histogram</span>. The options
supported <span class="phvro8t-x-x-109">inside this block </span>are described in <a href="#sec:colvarbias_grid">6.1<!--tex4ht:ref: sec:colvarbias_grid --></a>.
</p><!--l. 7828--><p class="indent">   Note that when <span class="ectt-1095">gatherVectorColvars </span>is <span class="ectt-1095">on</span>, each vector variable is automatically treated as a scalar,
and a single value of each grid parameter should be provided for it.
</p>
<div class="flushright">
<!--l. 7831--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 7831--><p class="noindent"> </p><div class="heading"> <a id="sec:colvarbias_histogram_output"></a>
<h5 class="subsubsectionHead"><span class="titlemark">6.13.1    </span> <a id="sec:colvarbias_histogram_output"></a>Output options for multi-dimensional histograms</h5>
<!--l. 7831--><p class="noindent"></p></div>
<!--l. 7833--><p class="indent">   The accumulated histogram is written in the Colvars state file, allowing for its accumulation
across continued runs. Additionally, the following files are written depending on the histogram's
dimensionality:
</p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox450"><a id="dx1-156001"></a><span class="ectt-1095">outputFreq</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">outputFreq</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox451"><a id="dx1-156002"></a><span class="ectt-1095">outputFile</span></span> — <span class="phvro8t-x-x-109">Write the histogram to a file</span><br class="newline"/>Default: <span class="phvro8t-x-x-109">outputName</span><span class="ectt-1095">.</span><!--l. 7848--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">name</span><!--l. 7848--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><span class="ectt-1095">.dat</span> <!--l. 7848--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> UNIX
     filename, context: <span class="ectt-1095">histogram</span> <!--l. 7848--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Name of the file containing histogram data (multicolumn format, see <a href="#sec:colvar_multicolumn_grid">3.8.5<!--tex4ht:ref: sec:colvar_multicolumn_grid --></a>), which is written
     every <a href="#colvarbias|outputFreq"><span class="ectt-1095">outputFreq</span></a> steps. If <span class="ectt-1095">outputFile </span>is set to <span class="ectt-1095">none</span>, the file is not written.
                                                                                         
                                                                                         
     </li>
<li class="itemize"><span class="colorbox" id="colorbox452"><a id="dx1-156003"></a><span class="ectt-1095">outputFileDX</span></span> — <span class="phvro8t-x-x-109">Write the histogram to a file</span><br class="newline"/>Default: <span class="phvro8t-x-x-109">outputName</span><span class="ectt-1095">.</span><!--l. 7861--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">name</span><!--l. 7861--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><span class="ectt-1095">.dx</span> <!--l. 7861--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> UNIX
     filename, context: <span class="ectt-1095">histogram</span> <!--l. 7861--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Name of the file containing histogram data (OpenDX format), which is written every <a href="#colvarbias|outputFreq"><span class="ectt-1095">outputFreq</span></a>
     steps.  For  the  special  case  of  3  variables,  VMD  may  be  used  to  visualize  this  file.  This
     file  is  written  by  default  if  the  dimension  is  3  or  more  (you  cannot  visualize  it  easily  for
     dimensions  4  and  above,  but  the  DX  format  is  still  more  compact  than  the  multicolumn
     format). If <span class="ectt-1095">outputFileDX </span>is set to <span class="ectt-1095">none</span>, the file is not written.
                                                                                         <div class="flushright">
<!--l. 7863--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 7863--><p class="noindent"> </p><div class="heading"> <a id="sec:colvarbias_histogram_vectors"></a>
<h5 class="subsubsectionHead"><span class="titlemark">6.13.2    </span> <a id="sec:colvarbias_histogram_vectors"></a>Histogramming vector variables</h5>
<!--l. 7863--><p class="noindent"></p></div>
</li>
<li class="itemize"><span class="colorbox" id="colorbox453"><a id="dx1-157001"></a><span class="ectt-1095">gatherVectorColvars</span></span> — <span class="phvro8t-x-x-109">Treat vector variables as multiple observations of a scalar</span>
<span class="phvro8t-x-x-109">variable?</span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 7876--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> UNIX filename,
     context: <span class="ectt-1095">histogram</span> <!--l. 7876--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>When this is set to <span class="ectt-1095">on</span>, the components of a multi-dimensional colvar (e.g. one based on <span class="ectt-1095">cartesian</span>,
     <span class="ectt-1095">distancePairs</span>, or a vector of scalar numbers given by <span class="ectt-1095">scriptedFunction</span>) are treated as multiple
     observations of a scalar variable. This results in the histogram being accumulated multiple times for
     each simulation step). When multiple vector variables are included in <span class="ectt-1095">histogram</span>, these must have
     the same length because their components are accumulated together. For example, if
     <!--l. 7876--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>,
     <!--l. 7876--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>λ</mi></math> and
     <!--l. 7876--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>τ</mi></math>
     are three variables of dimensions 5, 5 and 1, respectively, for each iteration 5 triplets
     <!--l. 7876--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-punc">,</mo><msub><mrow><mi>λ</mi></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-punc">,</mo><mi>τ</mi><mo class="MathClass-close">)</mo></math>
     (<!--l. 7876--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>i</mi> <mo class="MathClass-rel">=</mo> <mn>1</mn><mo class="MathClass-punc">,</mo><mi class="MathClass-op">…</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo><mn>5</mn></math>)
     are accumulated into a 3-dimensional histogram.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox454"><a id="dx1-157002"></a><span class="ectt-1095">weights</span></span> — <span class="phvro8t-x-x-109">Relative contributions of each vector component to the histogram</span><br class="newline"/>Default: all weights equal to 1 <!--l. 7887--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> list
     of  space-separated  decimals,  context:
     <span class="ectt-1095">histogram</span> <!--l. 7887--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>When <span class="ectt-1095">gatherVectorColvars </span>is <span class="ectt-1095">on</span>, the components of each multi-dimensional
     colvar are accumulated with a different weight. For example, if
     <!--l. 7887--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>x</mi></math> and
     <!--l. 7887--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>y</mi></math>
     are two distinct <span class="ectt-1095">cartesian </span>variables defined on the same group of atoms, the corresponding 2D
     histogram can be weighted on a per-atom basis in the definition of <span class="ectt-1095">histogram</span>. </li></ul>
<!--l. 7891--><p class="indent">   As with any other biasing and analysis method, when a histogram is applied to an extended-system
colvar (<a href="#sec:colvar_extended">4.24<!--tex4ht:ref: sec:colvar_extended --></a>), it accesses the value of the extended coordinate rather than that of the actual
colvar. This can be overridden by enabling the <a href="#colvarbias|bypassExtendedLagrangian"><span class="ectt-1095">bypassExtendedLagrangian</span></a> option. A <span class="phvro8t-x-x-109">joint</span>
<span class="phvro8t-x-x-109">histogram </span>of the actual colvar and the extended coordinate may be collected by specifying the
colvar name twice in a row in the <span class="ectt-1095">colvars </span>parameter (e.g. <span class="ectt-1095">colvars myColvar myColvar</span>):
the first instance will be understood as the actual colvar, and the second, as the extended
coordinate.
</p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox455"><a id="dx1-157003"></a><span class="ectt-1095">bypassExtendedLagrangian</span></span> — same definition as <a href="#colvarbias|bypassExtendedLagrangian"><span class="ectt-1095">bypassExtendedLagrangian</span></a> (biasing and
     analysis methods)</li></ul>
<div class="flushright">
<!--l. 7964--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 7964--><p class="noindent"> </p><div class="heading"> <a id="sec:reweighted_fel_from_aMD"></a>
<h4 class="subsectionHead"><span class="titlemark">6.14    </span> <a id="sec:reweighted_fel_from_aMD"></a>Reweighted free-energy landscapes from accelerated MD</h4>
<!--l. 7964--><p class="noindent"></p></div>
<!--l. 7964--><p class="indent">   The <span class="ectt-1095">reweightaMD </span>can be used to build reweighted free-energy landscapes of a
set of collective variables on-the-fly from accelerated molecular dynamics (aMD)<span class="cite">[<a href="#Xhamelberg2004">49</a>]</span>
and Gaussian aMD (GaMD)<span class="cite">[<a href="#XMiao2015">50</a>]</span>. More specifically, it collects the boost potential,
<!--l. 7964--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="normal">Δ</mi><mi>V</mi> <mo class="MathClass-open">(</mo><mstyle mathvariant="bold"><mi>X</mi></mstyle><mo class="MathClass-close">)</mo></math>, and
computes the corresponding free-energies: </p><table class="equation"><tr><td> <a id="eq:amd-reweighting"></a>
<!--l. 7964--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>A</mi> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mi>ξ</mi> </mrow><mo fence="true" form="postfix">)</mo></mrow> <mo class="MathClass-rel">=</mo> <mo class="MathClass-bin">−</mo><mfrac><mrow><mn>1</mn></mrow>
<mrow><mi>β</mi></mrow></mfrac><mi class="qopname">ln</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--><mo class="MathClass-open">⟨</mo><mi class="qopname">exp</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--><mo class="MathClass-open">{</mo><mi>β</mi><mi mathvariant="normal">Δ</mi><mi>V</mi> <mo class="MathClass-open">(</mo><mstyle mathvariant="bold"><mi>X</mi></mstyle><mo class="MathClass-close">)</mo><mi>δ</mi><mo class="MathClass-open">[</mo><mi>ξ</mi><mo class="MathClass-open">(</mo><mstyle mathvariant="bold"><mi>X</mi></mstyle><mo class="MathClass-close">)</mo><mo class="MathClass-bin">−</mo><mi>ξ</mi><mo class="MathClass-close">]</mo><mo class="MathClass-close">}</mo><mo class="MathClass-close">⟩</mo><mo class="MathClass-punc">,</mo>
</math></td><td class="eq-no">(52)</td></tr></table>
<!--l. 7964--><p class="indent">   where <!--l. 7964--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>β</mi></math>
is <!--l. 7964--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>1</mn><mo class="MathClass-bin">∕</mo><msub><mrow><mi>k</mi></mrow><mrow><mi>B</mi></mrow></msub><mi>T</mi> </math>. If the
distribution of <!--l. 7964--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="normal">Δ</mi><mi>V</mi> <mo class="MathClass-open">(</mo><mstyle mathvariant="bold"><mi>X</mi></mstyle><mo class="MathClass-close">)</mo><mi>δ</mi><mo class="MathClass-open">[</mo><mi>ξ</mi><mo class="MathClass-open">(</mo><mstyle mathvariant="bold"><mi>X</mi></mstyle><mo class="MathClass-close">)</mo><mo class="MathClass-bin">−</mo><mi>ξ</mi><mo class="MathClass-close">]</mo></math>
is assumed to be near Gaussian, the equation above can be approximated by the cumulant expansion: </p><table class="equation"><tr><td>
<a id="eq:cumulant-expansion"></a>
<!--l. 7964--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>A</mi> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mi>ξ</mi> </mrow><mo fence="true" form="postfix">)</mo></mrow> <mo class="MathClass-rel">=</mo> <mo class="MathClass-bin">−</mo><msub><mrow><mi>C</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-open">(</mo><mi>ξ</mi><mo class="MathClass-close">)</mo><mo class="MathClass-bin">−</mo><mfrac><mrow><mn>1</mn></mrow>
<mrow><mn>2</mn></mrow></mfrac><mi>β</mi><msub><mrow><mi>C</mi></mrow><mrow><mn>2</mn></mrow></msub><mo class="MathClass-open">(</mo><mi>ξ</mi><mo class="MathClass-close">)</mo><mo class="MathClass-punc">,</mo>
</math></td><td class="eq-no">(53)</td></tr></table>
<!--l. 7964--><p class="indent">   where <!--l. 7964--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>C</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-open">(</mo><mi>ξ</mi><mo class="MathClass-close">)</mo></math> and
<!--l. 7964--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>C</mi></mrow><mrow><mn>2</mn></mrow></msub><mo class="MathClass-open">(</mo><mi>ξ</mi><mo class="MathClass-close">)</mo></math> are the average and
the variance of <!--l. 7964--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi mathvariant="normal">Δ</mi><mi>V</mi> <mo class="MathClass-open">(</mo><mstyle mathvariant="bold"><mi>X</mi></mstyle><mo class="MathClass-close">)</mo><mi>δ</mi><mo class="MathClass-open">[</mo><mi>ξ</mi><mo class="MathClass-open">(</mo><mstyle mathvariant="bold"><mi>X</mi></mstyle><mo class="MathClass-close">)</mo><mo class="MathClass-bin">−</mo><mi>ξ</mi><mo class="MathClass-close">]</mo></math>,
respectively. It should be noted that this implementation is initially intended for the combination of aMD/GaMD
with other CV-based methods such as eABF<span class="cite">[<a href="#XChen2021">51</a>]</span>, where the term associated with the biasing distribution
<!--l. 7964--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mover accent="true"><mrow><mi>ρ</mi></mrow><mo accent="true">~</mo></mover><mo class="MathClass-open">(</mo><mi>ξ</mi><mo class="MathClass-close">)</mo></math>, namely
<!--l. 7964--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-bin">−</mo><mfrac><mrow><mn>1</mn></mrow> <mrow><mi>β</mi></mrow></mfrac><mi class="qopname">ln</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--><mover accent="true"><mrow><mi>ρ</mi></mrow><mo accent="true">~</mo></mover><mo class="MathClass-open">(</mo><mi>ξ</mi><mo class="MathClass-close">)</mo></math>,
is included, so users are expected to add the output from Eq. (<a href="#eq:amd-reweighting">52<!--tex4ht:ref: eq:amd-reweighting --></a>) or (<a href="#eq:cumulant-expansion">53<!--tex4ht:ref: eq:cumulant-expansion --></a>) with those
from CZAR (see Eq. (<a href="#eq:czar">32<!--tex4ht:ref: eq:czar --></a>)) <span class="cite">[<a href="#XLesage2017">29</a>]</span> directly. Consequently, in Eq. (<a href="#eq:amd-reweighting">52<!--tex4ht:ref: eq:amd-reweighting --></a>) and (<a href="#eq:cumulant-expansion">53<!--tex4ht:ref: eq:cumulant-expansion --></a>) the term
<!--l. 7964--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-bin">−</mo><mfrac><mrow><mn>1</mn></mrow> <mrow><mi>β</mi></mrow></mfrac><mi class="qopname">ln</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--><mover accent="true"><mrow><mi>ρ</mi></mrow><mo accent="true">~</mo></mover><mo class="MathClass-open">(</mo><mi>ξ</mi><mo class="MathClass-close">)</mo></math> is not included.
Therefore, if users want to perform just aMD/GaMD simulations with this on-the-fly reweighting implementation,
the term <!--l. 7964--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-bin">−</mo><mfrac><mrow><mn>1</mn></mrow> <mrow><mi>β</mi></mrow></mfrac><mi class="qopname">ln</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--><mover accent="true"><mrow><mi>ρ</mi></mrow><mo accent="true">~</mo></mover><mo class="MathClass-open">(</mo><mi>ξ</mi><mo class="MathClass-close">)</mo></math>
should be manually computed by post-processing.
</p><!--l. 7964--><p class="indent">   Currently, in all MD engines that Colvars supports, only NAMD implements aMD/GaMD, so this
component is basically exclusive to NAMD.
</p><!--l. 7964--><p class="indent">   The list of options is as follows: </p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox456"><a id="dx1-158003"></a><span class="ectt-1095">name</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">name</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox457"><a id="dx1-158004"></a><span class="ectt-1095">colvars</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">colvars</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox458"><a id="dx1-158005"></a><span class="ectt-1095">outputFreq</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">outputFreq</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox459"><a id="dx1-158006"></a><span class="ectt-1095">outputFile</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">outputFile</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox460"><a id="dx1-158007"></a><span class="ectt-1095">historyFreq</span></span> — same definition as <a href="#sec:colvarbias_abf"><span class="ectt-1095">historyFreq</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox461"><a id="dx1-158008"></a><span class="ectt-1095">CollectAfterSteps</span></span> — <span class="phvro8t-x-x-109">Collect the histogram after a specified number of simulation steps</span><br class="newline"/>Default: 0 <!--l. 7964--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     integer, context: <span class="ectt-1095">reweightaMD</span> <!--l. 7964--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Start to populate the histogram after a specified number of step. This feature is useful to skip
     the preparation steps in GaMD simulations.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox462"><a id="dx1-158009"></a><span class="ectt-1095">CumulantExpansion</span></span> — <span class="phvro8t-x-x-109">Enable the cumulant expansion estimator</span><br class="newline"/>Default: <span class="ectt-1095">yes</span> <!--l. 7964--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">reweightaMD</span> <!--l. 7964--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If this option is set to <span class="ectt-1095">yes</span>, the cumulant expansion in Eq. (<a href="#eq:cumulant-expansion">53<!--tex4ht:ref: eq:cumulant-expansion --></a>) will be used. The output file
     have a suffix of <span class="ectt-1095">.cumulant.*</span>.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox463"><a id="dx1-158010"></a><span class="ectt-1095">writePMFGradients</span></span> — <span class="phvro8t-x-x-109">Calculate the numerical gradients of Eq.</span><span class="phvro8t-x-x-109"> (</span><a href="#eq:amd-reweighting"><span class="phvro8t-x-x-109">52</span><!--tex4ht:ref: eq:amd-reweighting --></a><span class="phvro8t-x-x-109">) and (</span><a href="#eq:cumulant-expansion"><span class="phvro8t-x-x-109">53</span><!--tex4ht:ref: eq:cumulant-expansion --></a><span class="phvro8t-x-x-109">)</span><br class="newline"/>Default: <span class="ectt-1095">yes</span> <!--l. 7964--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">reweightaMD</span> <!--l. 7964--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If this option is set to <span class="ectt-1095">yes</span>, the numerical gradients Eq. (<a href="#eq:amd-reweighting">52<!--tex4ht:ref: eq:amd-reweighting --></a>) and (<a href="#eq:cumulant-expansion">53<!--tex4ht:ref: eq:cumulant-expansion --></a>) with respect to <!--l. 7964--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>
     are calculated. This option is generally used when <!--l. 7964--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>
     is stratified into windows, and the corresponding free-energies are required to merge afterwards.
     <!--l. 7964--><p class="noindent">The grid used for discretizing the free energy computated by <span class="ectt-1095">reweightaMD </span>is defined by the
     relevant options of the colvars. Alternately, grid parameters can be set directly inside the
     <span class="ectt-1095">reweightaMD </span>block by inserting a <span class="ectt-1095">grid </span>sub-block, as explained in <a href="#sec:colvarbias_grid">6.1<!--tex4ht:ref: sec:colvarbias_grid --></a>.
</p>
</li></ul>
<div class="flushright">
<!--l. 7967--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 7967--><p class="noindent"> </p><div class="heading"> <a id="sec:colvarbias_restraint_histogram"></a>
<h4 class="subsectionHead"><span class="titlemark">6.15    </span> <a id="sec:colvarbias_restraint_histogram"></a>Probability distribution-restraints</h4>
<!--l. 7967--><p class="noindent"></p></div>
<!--l. 7969--><p class="indent">   The <span class="ectt-1095">histogramRestraint </span>bias implements a continuous potential of many variables (or of a single
high-dimensional variable) aimed at reproducing a one-dimensional statistical distribution that is provided by the
user. The <!--l. 7970--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>M</mi></math>
variables <!--l. 7970--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-punc">,</mo><mi class="MathClass-op">…</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo><mo class="MathClass-punc">,</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>M</mi></mrow></msub><mo class="MathClass-close">)</mo></math>
are interpreted as multiple observations of a random variable
<!--l. 7970--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>
with unknown probability distribution. The potential is minimized when the histogram
<!--l. 7971--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>h</mi><mo class="MathClass-open">(</mo><mi>ξ</mi><mo class="MathClass-close">)</mo></math>, estimated as a sum of Gaussian
                                                                                         
                                                                                         
functions centered at <!--l. 7971--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">(</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-punc">,</mo><mi class="MathClass-op">…</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo><mo class="MathClass-punc">,</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>M</mi></mrow></msub><mo class="MathClass-close">)</mo></math>, is equal
to the reference histogram <!--l. 7971--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>h</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-open">(</mo><mi>ξ</mi><mo class="MathClass-close">)</mo></math>:
</p><table class="equation"><tr><td> <a id="eq:colvarbias_restraint_histogram"></a>
<!--l. 7972--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>V</mi> <mrow><mo fence="true" form="prefix"> (</mo><mrow><msub><mrow><mi>ξ</mi></mrow><mrow><mn>1</mn></mrow></msub><mo class="MathClass-punc">,</mo><mi class="MathClass-op">…</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--></mo><mo class="MathClass-punc">,</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>M</mi></mrow></msub></mrow><mo fence="true" form="postfix">)</mo></mrow> <mo class="MathClass-rel">=</mo> <mfrac><mrow><mn>1</mn></mrow>
<mrow><mn>2</mn></mrow></mfrac><mi>k</mi><mo>∫
 <!--nolimits--></mo><!--nolimits--><msup><mrow><mrow><mo fence="true" form="prefix"> (</mo><mrow><mi>h</mi><mo class="MathClass-open">(</mo><mi>ξ</mi><mo class="MathClass-close">)</mo><mo class="MathClass-bin">−</mo><msub><mrow><mi>h</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-open">(</mo><mi>ξ</mi><mo class="MathClass-close">)</mo></mrow><mo fence="true" form="postfix">)</mo></mrow></mrow><mrow><mn>2</mn></mrow></msup><mstyle mathvariant="normal"><mi>d</mi></mstyle><mi>ξ</mi>
</math></td><td class="eq-no">(54)</td></tr></table>
<table class="equation"><tr><td> <a id="eq:colvarbias_restraint_histogram_gaussian"></a>
<!--l. 7976--><math class="equation" display="block" xmlns="http://www.w3.org/1998/Math/MathML">
<mi>h</mi> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mi>ξ</mi> </mrow><mo fence="true" form="postfix">)</mo></mrow> <mo class="MathClass-rel">=</mo> <mfrac><mrow><mn>1</mn></mrow>
<mrow><mi>M</mi><msqrt><mrow><mn>2</mn><mi>π</mi><msup><mrow><mi>σ</mi></mrow><mrow><mn>2</mn></mrow></msup></mrow></msqrt></mrow></mfrac><munderover accent="false" accentunder="false"><mrow><mo>∑</mo></mrow><mrow><mi>i</mi><mo class="MathClass-rel">=</mo><mn>1</mn></mrow><mrow><mi>M</mi></mrow></munderover><mi class="qopname">exp</mi><mo> &amp;ApplyFunction;<!--FUNCTION APPLICATION--> </mo><!--nolimits--> <mrow><mo fence="true" form="prefix"> (</mo><mrow><mo class="MathClass-bin">−</mo><mfrac><mrow><msup><mrow><mo class="MathClass-open">(</mo><mi>ξ</mi> <mo class="MathClass-bin">−</mo><msub><mrow><mi>ξ</mi></mrow><mrow><mi>i</mi></mrow></msub><mo class="MathClass-close">)</mo></mrow><mrow><mn>2</mn></mrow></msup></mrow>
<mrow><mn>2</mn><msup><mrow><mi>σ</mi></mrow><mrow><mn>2</mn></mrow></msup></mrow></mfrac> </mrow><mo fence="true" form="postfix">)</mo></mrow>
</math></td><td class="eq-no">(55)</td></tr></table>
<!--l. 7980--><p class="indent">   When used in combination with a <span class="ectt-1095">distancePairs </span>multi-dimensional variable, this bias
implements the refinement algorithm against ESR/DEER experiments published by Shen <span class="phvro8t-x-x-109">et al</span>
<span class="cite">[<a href="#XShen2015">52</a>]</span>.
</p><!--l. 7982--><p class="indent">   This bias behaves similarly to the <span class="ectt-1095">histogram </span>bias with the <span class="ectt-1095">gatherVectorColvars </span>option, with the
important difference that <span class="phvro8t-x-x-109">all </span>variables are gathered, resulting in a one-dimensional histogram. Future
versions will include support for multi-dimensional histograms.
</p><!--l. 7985--><p class="indent">   The list of options is as follows: </p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox464"><a id="dx1-159003"></a><span class="ectt-1095">name</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">name</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox465"><a id="dx1-159004"></a><span class="ectt-1095">colvars</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">colvars</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox466"><a id="dx1-159005"></a><span class="ectt-1095">outputEnergy</span></span> — same definition as <a href="#sec:colvarbias"><span class="ectt-1095">outputEnergy</span></a> (biasing and analysis methods)
     </li>
<li class="itemize"><span class="colorbox" id="colorbox467"><a id="dx1-159006"></a><span class="ectt-1095">lowerBoundary</span></span> — <span class="phvro8t-x-x-109">Lower boundary of the colvar grid</span><br class="newline"/><!--l. 8000--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> decimal,
     context: <span class="ectt-1095">histogramRestraint</span> <!--l. 8000--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Defines the lowest end of the interval where the reference distribution <!--l. 8000--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>h</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-open">(</mo><mi>ξ</mi><mo class="MathClass-close">)</mo></math>
     is defined. Exactly one value must be provided, because only one-dimensional histograms
     are supported by the current version.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox468"><a id="dx1-159007"></a><span class="ectt-1095">upperBoundary</span></span> — analogous to <span class="ectt-1095">lowerBoundary</span>
</li>
<li class="itemize"><span class="colorbox" id="colorbox469"><a id="dx1-159008"></a><span class="ectt-1095">width</span></span> — <span class="phvro8t-x-x-109">Width of the colvar grid</span><br class="newline"/><!--l. 8012--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     decimal, context: <span class="ectt-1095">histogramRestraint</span> <!--l. 8012--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Defines the spacing of the grid where the reference distribution <!--l. 8012--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>h</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-open">(</mo><mi>ξ</mi><mo class="MathClass-close">)</mo></math>
     is defined.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox470"><a id="dx1-159009"></a><span class="ectt-1095">gaussianSigma</span></span> — <span class="phvro8t-x-x-109">Standard deviation of the approximating Gaussian</span><br class="newline"/>Default: 2 <!--l. 8022--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-bin">×</mo></math>
<span class="ectt-1095">width</span> <!--l. 8022--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     decimal, context: <span class="ectt-1095">histogramRestraint</span> <!--l. 8022--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Defines the parameter <!--l. 8022--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>σ</mi></math>
     in eq. <a href="#eq:colvarbias_restraint_histogram_gaussian">55<!--tex4ht:ref: eq:colvarbias_restraint_histogram_gaussian --></a>.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox471"><a id="dx1-159010"></a><span class="ectt-1095">forceConstant</span></span> — <span class="phvro8t-x-x-109">Force constant (kcal/mol)</span><br class="newline"/>Default: <span class="ectt-1095">1.0</span> <!--l. 8032--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> positive
     decimal, context: <span class="ectt-1095">histogramRestraint</span> <!--l. 8032--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Defines the parameter <!--l. 8032--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>k</mi></math>
     in eq. <a href="#eq:colvarbias_restraint_histogram">54<!--tex4ht:ref: eq:colvarbias_restraint_histogram --></a>.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox472"><a id="dx1-159011"></a><span class="ectt-1095">refHistogram</span></span> — <span class="phvro8t-x-x-109">Reference histogram </span><!--l. 8043--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>h</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-open">(</mo><mi>ξ</mi><mo class="MathClass-close">)</mo></math><br class="newline"/><!--l. 8043--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> space-separated
     list of <!--l. 8043--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>M</mi></math>
     positive decimals, context: <span class="ectt-1095">histogramRestraint</span> <!--l. 8043--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Provides the values of <!--l. 8043--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>h</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-open">(</mo><mi>ξ</mi><mo class="MathClass-close">)</mo></math>
     consecutively. The mid-point convention is used, i.e. the first point that should be included is
     for <!--l. 8043--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>
     = <span class="ectt-1095">lowerBoundary</span>+<span class="ectt-1095">width</span>/2. If the integral of <!--l. 8043--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>h</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-open">(</mo><mi>ξ</mi><mo class="MathClass-close">)</mo></math>
     is not normalized to 1, <!--l. 8043--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>h</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-open">(</mo><mi>ξ</mi><mo class="MathClass-close">)</mo></math>
     is rescaled automatically before use.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox473"><a id="dx1-159012"></a><span class="ectt-1095">refHistogramFile</span></span> — <span class="phvro8t-x-x-109">Reference histogram </span><!--l. 8054--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>h</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-open">(</mo><mi>ξ</mi><mo class="MathClass-close">)</mo></math><br class="newline"/><!--l. 8054--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> UNIX
     file name, context: <span class="ectt-1095">histogramRestraint</span> <!--l. 8054--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>Provides the values of <!--l. 8054--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>h</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-open">(</mo><mi>ξ</mi><mo class="MathClass-close">)</mo></math>
     as contents of the corresponding file (mutually exclusive with <span class="ectt-1095">refHistogram</span>). The format is
     that of a text file, with each line containing the space-separated values of <!--l. 8054--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>ξ</mi></math>
     and <!--l. 8054--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>h</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-open">(</mo><mi>ξ</mi><mo class="MathClass-close">)</mo></math>.
     The same numerical conventions as <span class="ectt-1095">refHistogram </span>are used.
     </li>
<li class="itemize"><span class="colorbox" id="colorbox474"><a id="dx1-159013"></a><span class="ectt-1095">writeHistogram</span></span> — <span class="phvro8t-x-x-109">Periodically write the instantaneous histogram </span><!--l. 8063--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>h</mi><mo class="MathClass-open">(</mo><mi>ξ</mi><mo class="MathClass-close">)</mo></math><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 8063--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: <span class="ectt-1095">metadynamics</span> <!--l. 8063--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If <span class="ectt-1095">on</span>, the histogram <!--l. 8063--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>h</mi><mo class="MathClass-open">(</mo><mi>ξ</mi><mo class="MathClass-close">)</mo></math>
     is written every <span class="ectt-1095">colvarsRestartFrequency </span>steps to a file with the name <span class="phvro8t-x-x-109">outputName</span><span class="ectt-1095">.</span><!--l. 8063--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&lt;</mo></math><span class="ectt-1095">name</span><!--l. 8063--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"> <mo class="MathClass-rel">&gt;</mo></math><span class="ectt-1095">.hist.dat</span>This
     is useful to diagnose the convergence of <!--l. 8064--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>h</mi><mo class="MathClass-open">(</mo><mi>ξ</mi><mo class="MathClass-close">)</mo></math>
     against <!--l. 8064--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><msub><mrow><mi>h</mi></mrow><mrow><mn>0</mn></mrow></msub><mo class="MathClass-open">(</mo><mi>ξ</mi><mo class="MathClass-close">)</mo></math>.
     </li></ul>
<div class="flushright">
<!--l. 8071--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 8071--><p class="noindent"> </p><div class="heading"> <a id="sec:scripted_biases"></a>
<h4 class="subsectionHead"><span class="titlemark">6.16    </span> <a id="sec:scripted_biases"></a>Defining scripted biases</h4>
<!--l. 8071--><p class="noindent"></p></div>
<!--l. 8073--><p class="indent">   Rather than using the biasing methods described above, it is possible to apply biases provided at run
time as a Tcl script, in the spirit of <span class="ectt-1095">TclForces</span>.
</p>
<ul class="itemize1">
<li class="itemize"><a id="Colvars-global|scriptedColvarForces"></a> <span class="colorbox" id="colorbox475"><a id="dx1-160001"></a><span class="ectt-1095">scriptedColvarForces</span></span> — <span class="phvro8t-x-x-109">Enable custom, scripted forces on colvars </span><br class="newline"/>Default: <span class="ectt-1095">off</span> <!--l. 8091--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: global <!--l. 8091--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>If this flag is enabled, a Tcl procedure named <span class="ectt-1095">calc_colvar_forces </span>accepting one parameter
     should be defined by the user. It is executed at each timestep, with the current step number
     as parameter, between the calculation of colvars and the application of bias forces. This
     procedure may use the <span class="ectt-1095">cv </span>command to access the values of colvars (e.g. <span class="ectt-1095">cv colvar xi</span>
<span class="ectt-1095">value</span>), apply forces on them (<span class="ectt-1095">cv colvar xi addforce $F</span>) or add energy to the simulation
     system (<span class="ectt-1095">cv addenergy $E</span>), effectively defining custom collective variable biases.</li></ul>
<div class="flushright">
<!--l. 8095--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 8095--><p class="noindent"> </p><div class="heading"> <a id="sec:cv_scripting_opt"></a>
<h4 class="subsectionHead"><span class="titlemark">6.17    </span> <a id="sec:cv_scripting_opt"></a>Performance of scripted biases</h4>
<!--l. 8095--><p class="noindent"></p></div>
<!--l. 8097--><p class="indent">   If concurrent computation over multiple threads is available (this is indicated by the message
“SMP parallelism is available." printed at initialization time), it is useful to take advantage of
the scripting interface to combine many components, all computed in parallel, into a single
variable.
</p><!--l. 8099--><p class="indent">   The default SMP schedule is the following:
     </p><ol class="enumerate1">
<li class="enumerate" id="x1-161002x1">distribute the computation of all components across available threads;
     </li>
<li class="enumerate" id="x1-161004x2">on  a  single  thread,  collect  the  results  of  multi-component  variables  using  polynomial
     combinations (see <a href="#sec:cvc_superp">4.19<!--tex4ht:ref: sec:cvc_superp --></a>), or custom functions (see <a href="#sec:colvar_custom_function">4.20<!--tex4ht:ref: sec:colvar_custom_function --></a>), or scripted functions (see <a href="#sec:colvar_scripted">4.21<!--tex4ht:ref: sec:colvar_scripted --></a>);
     </li>
<li class="enumerate" id="x1-161006x3">distribute the computation of all biases across available threads;
     </li>
<li class="enumerate" id="x1-161008x4">compute   on   a   single   thread   any   scripted   biases   implemented   via   the   keyword
     <a href="#Colvars-global|scriptedColvarForces"><span class="ectt-1095">scriptedColvarForces</span></a>.
     </li>
<li class="enumerate" id="x1-161010x5">communicate on a single thread forces to NAMD.</li></ol>
<!--l. 8108--><p class="indent">   The following options allow to fine-tune this schedule: </p>
<ul class="itemize1">
<li class="itemize"><span class="colorbox" id="colorbox476"><a id="dx1-161011"></a><span class="ectt-1095">scriptingAfterBiases</span></span> — <span class="phvro8t-x-x-109">Scripted colvar forces need updated biases?</span><br class="newline"/>Default: <span class="ectt-1095">on</span> <!--l. 8123--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-open">[</mo></math> boolean,
     context: global <!--l. 8123--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mo class="MathClass-close">]</mo></math><br class="newline"/>This flag specifies that the <span class="ectt-1095">calc_colvar_forces </span>procedure (last step in the list above) is
     executed only after all biases have been updated (next-to-last step) For example, this allows
     using the energy of a restraint bias, or the force applied on a colvar, to calculate additional
     scripted forces, such as boundary constraints. When this flag is set to <span class="ectt-1095">off</span>, it is assumed that
     only the values of the variables (but not the energy of the biases or applied forces) will be
     used by <span class="ectt-1095">calc_colvar_forces</span>: this can be used to schedule the calculation of scripted forces
     and biases concurrently to increase performance.</li></ul>
<!--l. 8130--><p class="indent">
</p>
<div class="flushright">
<!--l. 8130--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 8130--><p class="noindent"> </p><div class="heading"> <a id="sec:cvscript_cmdline"></a>
<h3 class="sectionHead"><span class="titlemark">7    </span> <a id="sec:cvscript_cmdline"></a>Tcl command-line interface: list of commands</h3>
<!--l. 8130--><p class="noindent"></p></div> This section lists all the commands used in NAMD to control the behavior of the Colvars module from
within a run script.

                                                                                         <div class="flushright">
<!--l. 1--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 1--><p class="noindent"> </p><div class="heading"> <a id="sec:cvscript_cmdline_cv"></a>
<h4 class="subsectionHead"><span class="titlemark">7.1    </span> <a id="sec:cvscript_cmdline_cv"></a>Commands to manage the Colvars module</h4>
<!--l. 1--><p class="noindent"></p></div>
<div class="mdframed" id="mdframed-71">
<!--l. 2--><p class="noindent"></p><pre><span class="ectt-1095">cv addenergy &lt;E&gt; </span><br class="newline"/>    <span class="ectt-1095">Add an energy to the MD engine (no effect in VMD) </span><br class="newline"/>    <span class="ectt-1095">Parameters </span><br class="newline"/>    <span class="ectt-1095">E : float - Amount of energy to add </span></pre><!--l. 10--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-72">
<!--l. 11--><p class="noindent"></p><pre><span class="ectt-1095">cv config &lt;conf&gt; </span><br class="newline"/>    <span class="ectt-1095">Read configuration from the given string </span><br class="newline"/>    <span class="ectt-1095">Parameters </span><br class="newline"/>    <span class="ectt-1095">conf : string - Configuration string </span></pre><!--l. 19--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-73">
<!--l. 20--><p class="noindent"></p><pre><span class="ectt-1095">cv configfile &lt;conf_file&gt; </span><br class="newline"/>    <span class="ectt-1095">Read configuration from a file </span><br class="newline"/>    <span class="ectt-1095">Parameters </span><br class="newline"/>    <span class="ectt-1095">conf_file : string - Path to configuration file </span></pre><!--l. 28--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-74">
<!--l. 29--><p class="noindent"></p><pre><span class="ectt-1095">cv delete </span><br class="newline"/>    <span class="ectt-1095">Delete this Colvars module instance (VMD only) </span></pre><!--l. 33--><p class="noindent"></p>
</div>
<div class="mdframed" id="mdframed-75">
<!--l. 34--><p class="noindent"></p><pre><span class="ectt-1095">cv featurereport </span><br class="newline"/>    <span class="ectt-1095">Return a summary of Colvars features used so far and their citations </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">report : string - Feature report and citations </span></pre><!--l. 42--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-76">
<!--l. 43--><p class="noindent"></p><pre><span class="ectt-1095">cv frame [frame] </span><br class="newline"/>    <span class="ectt-1095">Get or set current frame number (VMD only) </span><br class="newline"/>    <span class="ectt-1095">Parameters </span><br class="newline"/>    <span class="ectt-1095">frame : integer - Frame number (optional) </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">frame : integer - Frame number </span></pre><!--l. 55--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-77">
<!--l. 56--><p class="noindent"></p><pre><span class="ectt-1095">cv getatomappliedforces </span><br class="newline"/>    <span class="ectt-1095">Get the list of forces applied by Colvars to atoms </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">forces : array of arrays of floats - Atomic forces </span></pre><!--l. 64--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-78">
<!--l. 65--><p class="noindent"></p><pre><span class="ectt-1095">cv getatomappliedforcesmax </span><br class="newline"/>    <span class="ectt-1095">Get the maximum norm of forces applied by Colvars to atoms </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">force : float - Maximum atomic force </span></pre><!--l. 73--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-79">
<!--l. 74--><p class="noindent"></p><pre><span class="ectt-1095">cv getatomappliedforcesmaxid </span><br class="newline"/>    <span class="ectt-1095">Get the atom ID with the largest applied force </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">id : int - ID of the atom with the maximum atomic force </span></pre><!--l. 82--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-80">
<!--l. 83--><p class="noindent"></p><pre><span class="ectt-1095">cv getatomappliedforcesrms </span><br class="newline"/>    <span class="ectt-1095">Get the root-mean-square norm of forces applied by Colvars to atoms </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">force : float - RMS atomic force </span></pre><!--l. 91--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-81">
<!--l. 92--><p class="noindent"></p><pre><span class="ectt-1095">cv resetatomappliedforces </span><br class="newline"/>    <span class="ectt-1095">Reset forces applied by Colvars to atoms </span></pre><!--l. 96--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-82">
<!--l. 97--><p class="noindent"></p><pre><span class="ectt-1095">cv getatomids </span><br class="newline"/>    <span class="ectt-1095">Get the list of indices of atoms used in Colvars </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">indices : array of ints - Atom indices </span></pre><!--l. 105--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-83">
<!--l. 106--><p class="noindent"></p><pre><span class="ectt-1095">cv getatomcharges </span><br class="newline"/>    <span class="ectt-1095">Get the list of charges of atoms used in Colvars </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">charges : array of floats - Atomic charges </span></pre><!--l. 114--><p class="noindent"></p>
</div>
<div class="mdframed" id="mdframed-84">
<!--l. 115--><p class="noindent"></p><pre><span class="ectt-1095">cv getatommasses </span><br class="newline"/>    <span class="ectt-1095">Get the list of masses of atoms used in Colvars </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">masses : array of floats - Atomic masses </span></pre><!--l. 123--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-85">
<!--l. 124--><p class="noindent"></p><pre><span class="ectt-1095">cv getatompositions </span><br class="newline"/>    <span class="ectt-1095">Get the list of cached positions of atoms used in Colvars </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">positions : array of arrays of floats - Atomic positions </span></pre><!--l. 132--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-86">
<!--l. 133--><p class="noindent"></p><pre><span class="ectt-1095">cv getatomtotalforces </span><br class="newline"/>    <span class="ectt-1095">Get the list of cached total forces of atoms used in Colvars </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">forces : array of arrays of floats - Atomic total foces </span></pre><!--l. 141--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-87">
<!--l. 142--><p class="noindent"></p><pre><span class="ectt-1095">cv getconfig </span><br class="newline"/>    <span class="ectt-1095">Get the module's configuration string read so far </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">conf : string - Current configuration string </span></pre><!--l. 150--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-88">
<!--l. 151--><p class="noindent"></p><pre><span class="ectt-1095">cv getenergy </span><br class="newline"/>    <span class="ectt-1095">Get the current Colvars energy </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">E : float - Amount of energy (internal units) </span></pre><!--l. 159--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-89">
<!--l. 160--><p class="noindent"></p><pre><span class="ectt-1095">cv getnumactiveatomgroups </span><br class="newline"/>    <span class="ectt-1095">Get the number of atom groups that currently have positive ref counts </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">count : integer - Total number of atom groups </span></pre><!--l. 168--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-90">
<!--l. 169--><p class="noindent"></p><pre><span class="ectt-1095">cv getnumactiveatoms </span><br class="newline"/>    <span class="ectt-1095">Get the number of atoms that currently have positive ref counts </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">count : integer - Total number of atoms </span></pre><!--l. 177--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-91">
<!--l. 178--><p class="noindent"></p><pre><span class="ectt-1095">cv getnumatoms </span><br class="newline"/>    <span class="ectt-1095">Get the number of requested atoms, including those not in use now </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">count : integer - Total number of atoms </span></pre><!--l. 186--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-92">
<!--l. 187--><p class="noindent"></p><pre><span class="ectt-1095">cv getstepabsolute </span><br class="newline"/>    <span class="ectt-1095">Get the current step number of the simulation (including restarts) </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">step : int - Absolute step number </span></pre><!--l. 195--><p class="noindent"></p>
</div>
<div class="mdframed" id="mdframed-93">
<!--l. 196--><p class="noindent"></p><pre><span class="ectt-1095">cv getsteprelative </span><br class="newline"/>    <span class="ectt-1095">Get the current step number from the start of this job </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">step : int - Relative step number </span></pre><!--l. 204--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-94">
<!--l. 205--><p class="noindent"></p><pre><span class="ectt-1095">cv help [command] </span><br class="newline"/>    <span class="ectt-1095">Get the help string of the Colvars scripting interface </span><br class="newline"/>    <span class="ectt-1095">Parameters </span><br class="newline"/>    <span class="ectt-1095">command : string - Get the help string of this specific command (optional)</span>
<br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">help : string - Help string </span></pre><!--l. 217--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-95">
<!--l. 218--><p class="noindent"></p><pre><span class="ectt-1095">cv languageversion </span><br class="newline"/>    <span class="ectt-1095">Get the C++ language version number </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">version : string - C++ language version </span></pre><!--l. 226--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-96">
<!--l. 227--><p class="noindent"></p><pre><span class="ectt-1095">cv list [param] </span><br class="newline"/>    <span class="ectt-1095">Return a list of all variables or biases </span><br class="newline"/>    <span class="ectt-1095">Parameters </span><br class="newline"/>    <span class="ectt-1095">param : string - "colvars" or "biases"; default is "colvars" (optional) </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">list : sequence of strings - List of elements </span></pre><!--l. 239--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-97">
<!--l. 240--><p class="noindent"></p><pre><span class="ectt-1095">cv listcommands </span><br class="newline"/>    <span class="ectt-1095">Get the list of script functions, prefixed with "cv_", "colvar_" or "bias_"</span>
<br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">list : sequence of strings - List of commands </span></pre><!--l. 248--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-98">
<!--l. 249--><p class="noindent"></p><pre><span class="ectt-1095">cv listindexfiles </span><br class="newline"/>    <span class="ectt-1095">Get a list of the index files loaded in this session </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">list : sequence of strings - List of index file names </span></pre><!--l. 257--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-99">
<!--l. 258--><p class="noindent"></p><pre><span class="ectt-1095">cv listinputfiles </span><br class="newline"/>    <span class="ectt-1095">Get a list of all input/configuration files loaded in this session </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">list : sequence of strings - List of file names </span></pre><!--l. 266--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-100">
<!--l. 267--><p class="noindent"></p><pre><span class="ectt-1095">cv load &lt;prefix&gt; </span><br class="newline"/>    <span class="ectt-1095">Load data from a state file into all matching colvars and biases </span><br class="newline"/>    <span class="ectt-1095">Parameters </span><br class="newline"/>    <span class="ectt-1095">prefix : string - Path to existing state file or input prefix </span></pre><!--l. 275--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-101">
<!--l. 276--><p class="noindent"></p><pre><span class="ectt-1095">cv loadfromstring &lt;buffer&gt; </span><br class="newline"/>    <span class="ectt-1095">Load state data from a string into all matching colvars and biases </span><br class="newline"/>    <span class="ectt-1095">Parameters </span><br class="newline"/>    <span class="ectt-1095">buffer : string - String buffer containing the state information </span></pre><!--l. 284--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-102">
<!--l. 285--><p class="noindent"></p><pre><span class="ectt-1095">cv molid [molid] </span><br class="newline"/>    <span class="ectt-1095">Get or set the molecule ID on which Colvars is defined (VMD only) </span><br class="newline"/>    <span class="ectt-1095">Parameters </span><br class="newline"/>    <span class="ectt-1095">molid : integer - New molecule ID; -1 means undefined (optional) </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">molid : integer - Current molecule ID </span></pre><!--l. 297--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-103">
<!--l. 298--><p class="noindent"></p><pre><span class="ectt-1095">cv printframe </span><br class="newline"/>    <span class="ectt-1095">Return the values that would be written to colvars.traj </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">values : string - The values </span></pre><!--l. 306--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-104">
<!--l. 307--><p class="noindent"></p><pre><span class="ectt-1095">cv patchversion </span><br class="newline"/>    <span class="ectt-1095">Get the Colvars patch version number (used for bugfixes only) </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">version : string - Colvars version </span></pre><!--l. 315--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-105">
<!--l. 316--><p class="noindent"></p><pre><span class="ectt-1095">cv printframelabels </span><br class="newline"/>    <span class="ectt-1095">Return the labels that would be written to colvars.traj </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">Labels : string - The labels </span></pre><!--l. 324--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-106">
<!--l. 325--><p class="noindent"></p><pre><span class="ectt-1095">cv reset </span><br class="newline"/>    <span class="ectt-1095">Delete all internal configuration </span></pre><!--l. 329--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-107">
<!--l. 330--><p class="noindent"></p><pre><span class="ectt-1095">cv resetindexgroups </span><br class="newline"/>    <span class="ectt-1095">Clear the index groups loaded so far, allowing to replace them </span></pre><!--l. 334--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-108">
<!--l. 335--><p class="noindent"></p><pre><span class="ectt-1095">cv save &lt;prefix&gt; </span><br class="newline"/>    <span class="ectt-1095">Change the prefix of all output files and save them </span><br class="newline"/>    <span class="ectt-1095">Parameters </span><br class="newline"/>    <span class="ectt-1095">prefix : string - Output prefix with trailing ".colvars.state" gets removed)</span>
</pre><!--l. 343--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-109">
<!--l. 344--><p class="noindent"></p><pre><span class="ectt-1095">cv savetostring </span><br class="newline"/>    <span class="ectt-1095">Write the Colvars state to a string and return it </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">state : string - The saved state </span></pre><!--l. 352--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-110">
<!--l. 353--><p class="noindent"></p><pre><span class="ectt-1095">cv targettemperature [T] </span><br class="newline"/>    <span class="ectt-1095">Get/set target temperature, overriding internally what the MD engine reports</span>
<br class="newline"/>    <span class="ectt-1095">Parameters </span><br class="newline"/>    <span class="ectt-1095">T : float - New target temperature in K (internal use) (optional) </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">T : float - Current target temperature in K </span></pre><!--l. 365--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-111">
<!--l. 366--><p class="noindent"></p><pre><span class="ectt-1095">cv timestep [dt] </span><br class="newline"/>    <span class="ectt-1095">Get/set integration timestep, overriding internally what the MD engine reports</span>
<br class="newline"/>    <span class="ectt-1095">Parameters </span><br class="newline"/>    <span class="ectt-1095">dt : float - New integration timestep in MD engine units (optional) </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">dt : float - Current integration timestep in MD engine units </span></pre><!--l. 378--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-112">
<!--l. 379--><p class="noindent"></p><pre><span class="ectt-1095">cv units [units] </span><br class="newline"/>    <span class="ectt-1095">Get or set the current Colvars unit system </span><br class="newline"/>    <span class="ectt-1095">Parameters </span><br class="newline"/>    <span class="ectt-1095">units : string - The new unit system (optional) </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">units : string - The current unit system </span></pre><!--l. 391--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-113">
<!--l. 392--><p class="noindent"></p><pre><span class="ectt-1095">cv update </span><br class="newline"/>    <span class="ectt-1095">Recalculate colvars and biases </span></pre><!--l. 396--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-114">
<!--l. 397--><p class="noindent"></p><pre><span class="ectt-1095">cv version </span><br class="newline"/>    <span class="ectt-1095">Get the Colvars version string </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">version : string - Colvars version </span></pre><!--l. 405--><p class="noindent"></p>
</div>
<div class="flushright">
<!--l. 406--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 406--><p class="noindent"> </p><div class="heading"> <a id="sec:cvscript_cmdline_colvar"></a>
<h4 class="subsectionHead"><span class="titlemark">7.2    </span> <a id="sec:cvscript_cmdline_colvar"></a>Commands to manage individual collective variables</h4>
<!--l. 406--><p class="noindent"></p></div>
<div class="mdframed" id="mdframed-115">
<!--l. 407--><p class="noindent"></p><pre><span class="ectt-1095">cv colvar name addforce &lt;force&gt; </span><br class="newline"/>    <span class="ectt-1095">Apply the given force onto this colvar (no effects outside run) </span><br class="newline"/>    <span class="ectt-1095">Parameters </span><br class="newline"/>    <span class="ectt-1095">force : float or array - Applied force; must match colvar dimensionality </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">force : float or array - Applied force; matches colvar dimensionality </span></pre><!--l. 419--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-116">
<!--l. 420--><p class="noindent"></p><pre><span class="ectt-1095">cv colvar name communicateforces </span><br class="newline"/>    <span class="ectt-1095">Communicate bias forces from this colvar to atoms </span></pre><!--l. 424--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-117">
<!--l. 425--><p class="noindent"></p><pre><span class="ectt-1095">cv colvar name cvcflags &lt;flags&gt; </span><br class="newline"/>    <span class="ectt-1095">Enable or disable individual components by setting their active flags </span><br class="newline"/>    <span class="ectt-1095">Parameters </span><br class="newline"/>    <span class="ectt-1095">flags : integer array - Zero/nonzero value disables/enables the CVC </span></pre><!--l. 433--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-118">
<!--l. 434--><p class="noindent"></p><pre><span class="ectt-1095">cv colvar name delete </span><br class="newline"/>    <span class="ectt-1095">Delete this colvar, along with all biases that depend on it </span></pre><!--l. 438--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-119">
<!--l. 439--><p class="noindent"></p><pre><span class="ectt-1095">cv colvar name get &lt;feature&gt; </span><br class="newline"/>    <span class="ectt-1095">Get the value of the given feature for this colvar </span><br class="newline"/>    <span class="ectt-1095">Parameters </span><br class="newline"/>    <span class="ectt-1095">feature : string - Name of the feature </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">state : 1/0 - State of the given feature </span></pre><!--l. 451--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-120">
<!--l. 452--><p class="noindent"></p><pre><span class="ectt-1095">cv colvar name getappliedforce </span><br class="newline"/>    <span class="ectt-1095">Return the total of the forces applied to this colvar </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">force : float - Applied force; matches the colvar dimensionality </span></pre><!--l. 460--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-121">
<!--l. 461--><p class="noindent"></p><pre><span class="ectt-1095">cv colvar name resetbiasforce </span><br class="newline"/>    <span class="ectt-1095">Return the total of the forces applied to this colvar </span></pre><!--l. 465--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-122">
<!--l. 466--><p class="noindent"></p><pre><span class="ectt-1095">cv colvar name getatomgroups </span><br class="newline"/>    <span class="ectt-1095">Return the atom indices used by this colvar as a list of lists </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">groups : array of arrays of ints - Atom indices </span></pre><!--l. 474--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-123">
<!--l. 475--><p class="noindent"></p><pre><span class="ectt-1095">cv colvar name getatomids </span><br class="newline"/>    <span class="ectt-1095">Return the list of atom indices used by this colvar </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">indices : array of ints - Atom indices </span></pre><!--l. 483--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-124">
<!--l. 484--><p class="noindent"></p><pre><span class="ectt-1095">cv colvar name getconfig </span><br class="newline"/>    <span class="ectt-1095">Return the configuration string of this colvar </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">conf : string - Current configuration string </span></pre><!--l. 492--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-125">
<!--l. 493--><p class="noindent"></p><pre><span class="ectt-1095">cv colvar name getgradients </span><br class="newline"/>    <span class="ectt-1095">Return the atomic gradients of this colvar </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">gradients : array of arrays of floats - Atomic gradients </span></pre><!--l. 501--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-126">
<!--l. 502--><p class="noindent"></p><pre><span class="ectt-1095">cv colvar name gettotalforce </span><br class="newline"/>    <span class="ectt-1095">Return the sum of internal and external forces to this colvar </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">force : float - Total force; matches the colvar dimensionality </span></pre><!--l. 510--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-127">
<!--l. 511--><p class="noindent"></p><pre><span class="ectt-1095">cv colvar name getvolmapids </span><br class="newline"/>    <span class="ectt-1095">Return the list of volumetric map indices used by this colvar </span></pre><!--l. 515--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-128">
<!--l. 516--><p class="noindent"></p><pre><span class="ectt-1095">cv colvar name help [command] </span><br class="newline"/>    <span class="ectt-1095">Get a help summary or the help string of one colvar subcommand </span><br class="newline"/>    <span class="ectt-1095">Parameters </span><br class="newline"/>    <span class="ectt-1095">command : string - Get the help string of this specific command (optional)</span>
<br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">help : string - Help string </span></pre><!--l. 528--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-129">
<!--l. 529--><p class="noindent"></p><pre><span class="ectt-1095">cv colvar name modifycvcs &lt;confs&gt; </span><br class="newline"/>    <span class="ectt-1095">Modify configuration of individual components by passing string arguments</span>
<br class="newline"/>    <span class="ectt-1095">Parameters </span><br class="newline"/>    <span class="ectt-1095">confs : sequence of strings - New configurations; empty strings are skipped</span>
</pre><!--l. 537--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-130">
<!--l. 538--><p class="noindent"></p><pre><span class="ectt-1095">cv colvar name run_ave </span><br class="newline"/>    <span class="ectt-1095">Get the current running average of the value of this colvar </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">value : float or array - Averaged value; matches the colvar dimensionality</span>
</pre><!--l. 546--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-131">
<!--l. 547--><p class="noindent"></p><pre><span class="ectt-1095">cv colvar name set &lt;feature&gt; &lt;value&gt; </span><br class="newline"/>    <span class="ectt-1095">Set the given feature of this colvar to a new value </span><br class="newline"/>    <span class="ectt-1095">Parameters </span><br class="newline"/>    <span class="ectt-1095">feature : string - Name of the feature </span><br class="newline"/>    <span class="ectt-1095">value : string - String representation of the new feature value </span></pre><!--l. 557--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-132">
<!--l. 558--><p class="noindent"></p><pre><span class="ectt-1095">cv colvar name state </span><br class="newline"/>    <span class="ectt-1095">Print a string representation of the feature state of this colvar </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">state : string - The feature state </span></pre><!--l. 566--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-133">
<!--l. 567--><p class="noindent"></p><pre><span class="ectt-1095">cv colvar name type </span><br class="newline"/>    <span class="ectt-1095">Get the type description of this colvar </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">type : string - Type description </span></pre><!--l. 575--><p class="noindent"></p>
</div>
<div class="mdframed" id="mdframed-134">
<!--l. 576--><p class="noindent"></p><pre><span class="ectt-1095">cv colvar name update </span><br class="newline"/>    <span class="ectt-1095">Recompute this colvar and return its up-to-date value </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">value : float or array - Current value; matches the colvar dimensionality</span>
</pre><!--l. 584--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-135">
<!--l. 585--><p class="noindent"></p><pre><span class="ectt-1095">cv colvar name value </span><br class="newline"/>    <span class="ectt-1095">Get the current value of this colvar </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">value : float or array - Current value; matches the colvar dimensionality</span>
</pre><!--l. 593--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-136">
<!--l. 594--><p class="noindent"></p><pre><span class="ectt-1095">cv colvar name width </span><br class="newline"/>    <span class="ectt-1095">Get the width of this colvar </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">width : float - Value of the width </span></pre><!--l. 602--><p class="noindent"></p>
</div>
<div class="flushright">
<!--l. 603--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 603--><p class="noindent"> </p><div class="heading"> <a id="sec:cvscript_cmdline_bias"></a>
<h4 class="subsectionHead"><span class="titlemark">7.3    </span> <a id="sec:cvscript_cmdline_bias"></a>Commands to manage individual biases</h4>
<!--l. 603--><p class="noindent"></p></div>
<div class="mdframed" id="mdframed-137">
<!--l. 604--><p class="noindent"></p><pre><span class="ectt-1095">cv bias name bin </span><br class="newline"/>    <span class="ectt-1095">Get the current grid bin index (flattened if more than 1d) </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">bin : integer - Bin index </span></pre><!--l. 612--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-138">
<!--l. 613--><p class="noindent"></p><pre><span class="ectt-1095">cv bias name bincount [index] </span><br class="newline"/>    <span class="ectt-1095">Get the number of samples at the given grid bin (1D ABF only for now) </span><br class="newline"/>    <span class="ectt-1095">Parameters </span><br class="newline"/>    <span class="ectt-1095">index : integer - Grid index; defaults to current bin (optional) </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">samples : integer - Number of samples </span></pre><!--l. 625--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-139">
<!--l. 626--><p class="noindent"></p><pre><span class="ectt-1095">cv bias name local_sample_count [radius] </span><br class="newline"/>    <span class="ectt-1095">Get the number of samples around the current binsamples : integer - Number of samples</span>
<br class="newline"/>    <span class="ectt-1095">Parameters </span><br class="newline"/>    <span class="ectt-1095">radius : integer - Sum over radius bins around current bin (optional) </span></pre><!--l. 634--><p class="noindent"></p>
</div>
<div class="mdframed" id="mdframed-140">
<!--l. 635--><p class="noindent"></p><pre><span class="ectt-1095">cv bias name binnum </span><br class="newline"/>    <span class="ectt-1095">Get the total number of grid points of this bias (1D ABF only for now) </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">Bins : integer - Number of grid points </span></pre><!--l. 643--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-141">
<!--l. 644--><p class="noindent"></p><pre><span class="ectt-1095">cv bias name delete </span><br class="newline"/>    <span class="ectt-1095">Delete this bias </span></pre><!--l. 648--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-142">
<!--l. 649--><p class="noindent"></p><pre><span class="ectt-1095">cv bias name energy </span><br class="newline"/>    <span class="ectt-1095">Get the current energy of this bias </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">E : float - Energy value </span></pre><!--l. 657--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-143">
<!--l. 658--><p class="noindent"></p><pre><span class="ectt-1095">cv bias name get &lt;feature&gt; </span><br class="newline"/>    <span class="ectt-1095">Get the value of the given feature for this bias </span><br class="newline"/>    <span class="ectt-1095">Parameters </span><br class="newline"/>    <span class="ectt-1095">feature : string - Name of the feature </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">state : 1/0 - State of the given feature </span></pre><!--l. 670--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-144">
<!--l. 671--><p class="noindent"></p><pre><span class="ectt-1095">cv bias name getconfig </span><br class="newline"/>    <span class="ectt-1095">Return the configuration string of this bias </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">conf : string - Current configuration string </span></pre><!--l. 679--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-145">
<!--l. 680--><p class="noindent"></p><pre><span class="ectt-1095">cv bias name help [command] </span><br class="newline"/>    <span class="ectt-1095">Get a help summary or the help string of one bias subcommand </span><br class="newline"/>    <span class="ectt-1095">Parameters </span><br class="newline"/>    <span class="ectt-1095">command : string - Get the help string of this specific command (optional)</span>
<br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">help : string - Help string </span></pre><!--l. 692--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-146">
<!--l. 693--><p class="noindent"></p><pre><span class="ectt-1095">cv bias name load &lt;prefix&gt; </span><br class="newline"/>    <span class="ectt-1095">Load data into this bias </span><br class="newline"/>    <span class="ectt-1095">Parameters </span><br class="newline"/>    <span class="ectt-1095">prefix : string - Read from a file with this name or prefix </span></pre><!--l. 701--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-147">
<!--l. 702--><p class="noindent"></p><pre><span class="ectt-1095">cv bias name loadfromstring &lt;buffer&gt; </span><br class="newline"/>    <span class="ectt-1095">Load state data into this bias from a string </span><br class="newline"/>    <span class="ectt-1095">Parameters </span><br class="newline"/>    <span class="ectt-1095">buffer : string - String buffer containing the state information </span></pre><!--l. 710--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-148">
<!--l. 711--><p class="noindent"></p><pre><span class="ectt-1095">cv bias name save &lt;prefix&gt; </span><br class="newline"/>    <span class="ectt-1095">Save data from this bias into a file with the given prefix </span><br class="newline"/>    <span class="ectt-1095">Parameters </span><br class="newline"/>    <span class="ectt-1095">prefix : string - Prefix for the state file of this bias </span></pre><!--l. 719--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-149">
<!--l. 720--><p class="noindent"></p><pre><span class="ectt-1095">cv bias name savetostring </span><br class="newline"/>    <span class="ectt-1095">Save data from this bias into a string and return it </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">state : string - The bias state </span></pre><!--l. 728--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-150">
<!--l. 729--><p class="noindent"></p><pre><span class="ectt-1095">cv bias name set &lt;feature&gt; &lt;value&gt; </span><br class="newline"/>    <span class="ectt-1095">Set the given feature of this bias to a new value </span><br class="newline"/>    <span class="ectt-1095">Parameters </span><br class="newline"/>    <span class="ectt-1095">feature : string - Name of the feature </span><br class="newline"/>    <span class="ectt-1095">value : string - String representation of the new feature value </span></pre><!--l. 739--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-151">
<!--l. 740--><p class="noindent"></p><pre><span class="ectt-1095">cv bias name share </span><br class="newline"/>    <span class="ectt-1095">Share bias information with other replicas (multiple-walker scheme) </span></pre><!--l. 744--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-152">
<!--l. 745--><p class="noindent"></p><pre><span class="ectt-1095">cv bias name state </span><br class="newline"/>    <span class="ectt-1095">Print a string representation of the feature state of this bias </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">state : string - String representation of the bias features </span></pre><!--l. 753--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-153">
<!--l. 754--><p class="noindent"></p><pre><span class="ectt-1095">cv bias name type </span><br class="newline"/>    <span class="ectt-1095">Print the type of this bias object </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">type : string - Type of this bias object (e.g. metadynamics) </span></pre><!--l. 762--><p class="noindent"></p>
</div><div class="mdframed" id="mdframed-154">
<!--l. 763--><p class="noindent"></p><pre><span class="ectt-1095">cv bias name update </span><br class="newline"/>    <span class="ectt-1095">Recompute this bias and return its up-to-date energy </span><br class="newline"/>    <span class="ectt-1095">Returns </span><br class="newline"/>    <span class="ectt-1095">E : float - Energy value </span></pre><!--l. 771--><p class="noindent"></p>
</div>
<!--l. 8145--><p class="indent">
</p>
<div class="flushright">
<!--l. 8145--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 8145--><p class="noindent"> </p><div class="heading"> <a id="sec:colvars_config_changes"></a>
<h3 class="sectionHead"><span class="titlemark">8    </span> <a id="sec:colvars_config_changes"></a>Syntax changes from older versions</h3>
<!--l. 8145--><p class="noindent"></p></div>
<!--l. 8147--><p class="indent">   The following is a list of syntax changes in Colvars since its first release. Many of the older keywords
are still recognized by the current code, thanks to specific compatibility code. <span class="phvro8t-x-x-109">This is not a list of new</span>
<span class="phvro8t-x-x-109">features: </span>its primary purpose is to make you aware of those improvements that affect the use of old
configuration files with new versions of the code.
</p><!--l. 8151--><p class="noindent"><span class="phvb8t-x-x-109">Note: </span>if you are using any of the NAMD and VMD tutorials:<br class="newline"/><a class="url" href="https://www.ks.uiuc.edu/Training/Tutorials/"><span class="ectt-1095">https://www.ks.uiuc.edu/Training/Tutorials/</span></a><br class="newline"/>please be aware that <span class="phvro8t-x-x-109">several of these tutorials are not actively maintained</span>: for those cases, this list will
help you reconcile any inconsistencies.
</p>
<ul class="itemize1">
<li class="itemize"><span class="phvb8t-x-x-109">Colvars version 2016-06-09 or later (NAMD version 2.12b1 or later).</span><br class="newline"/>The legacy keyword <span class="ectt-1095">refPositionsGroup </span>has been renamed <a href="#atom-group|fittingGroup"><span class="ectt-1095">fittingGroup</span></a> for clarity (the
     legacy version is still supported).
     </li>
<li class="itemize"><span class="phvb8t-x-x-109">Colvars version 2016-08-10 or later (NAMD version 2.12b1 or later).</span><br class="newline"/>“System forces" have been replaced by “total forces" (see for example <a href="#colvar|outputTotalForce"><span class="ectt-1095">outputTotalForce</span></a>).
     See the following page for more information:<br class="newline"/><a class="url" href="https://colvars.github.io/README-totalforce.html"><span class="ectt-1095">https://colvars.github.io/README-totalforce.html</span></a>
</li>
<li class="itemize"><span class="phvb8t-x-x-109">Colvars version 2017-01-09 or later (NAMD version 2.13b1 or later).</span><br class="newline"/>A new type of restraint, <span class="ectt-1095">harmonicWalls </span>(see <a href="#sec:colvarbias_harmonic_walls">6.10<!--tex4ht:ref: sec:colvarbias_harmonic_walls --></a>), replaces and improves upon the legacy
     keywords <span class="ectt-1095">lowerWall </span>and <span class="ectt-1095">upperWall</span>: these are still supported as short-hands.
     </li>
<li class="itemize"><span class="phvb8t-x-x-109">Colvars version 2018-11-15 or later (NAMD version 2.14b1 or later).</span><br class="newline"/>The global <span class="ectt-1095">analysis </span>keyword has been discontinued: specific analysis tasks are controlled
     directly by the keywords <a href="#colvar|corrFunc"><span class="ectt-1095">corrFunc</span></a> and <a href="#colvar|runAve"><span class="ectt-1095">runAve</span></a>, which continue to remain <span class="ectt-1095">off </span>by default.
     </li>
<li class="itemize"><span class="phvb8t-x-x-109">Colvars version 2020-02-25 or later (NAMD version 2.14b1 or later).</span><br class="newline"/>The parameter <a href="#metadynamics|hillWidth"><span class="ectt-1095">hillWidth</span></a>, expressing the Gaussian width <!--l. 8172--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>2</mn><mi>σ</mi></math>
     in relative units (number of grid points), does not have a default value any more. A new
     alternative parameter <a href="#metadynamics|gaussianSigmas"><span class="ectt-1095">gaussianSigmas</span></a> allows setting the <!--l. 8173--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>σ</mi></math>
                                                                                         
                                                                                         
     parameters explicitly for each variable if needed.<br class="newline"/>Furthermore, to facilitate the use of other analysis tools such as for example <span class="ectt-1095">sum_hills</span>:<br class="newline"/><a class="url" href="https://www.plumed.org/doc-v2.6/user-doc/html/sum_hills.html"><span class="ectt-1095">https://www.plumed.org/doc-v2.6/user-doc/html/sum_hills.html</span></a><br class="newline"/>the format of the file written by <a href="#metadynamics|writeHillsTrajectory"><span class="ectt-1095">writeHillsTrajectory</span></a> has also been changed to use <!--l. 8176--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mi>σ</mi></math>
     instead of <!--l. 8176--><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><mn>2</mn><mi>σ</mi></math>.
     This  change  does  not  affect  how  the  biasing  potential  is  written  in  the  state  file,  or  the
     simulated trajectory.
     </li>
<li class="itemize"><span class="phvb8t-x-x-109">Colvars version 2020-02-25 or later (NAMD version 2.14b1 or later).</span><br class="newline"/>The  legacy  keywords  <span class="ectt-1095">lowerWall </span>and  <span class="ectt-1095">upperWall </span>of  a  <span class="ectt-1095">colvar </span>definition  block  do  not
     have  default  values  any  longer,  and  need  to  be  set  explicitly,  preferably  as  part  of  the
     <span class="ectt-1095">harmonicWalls </span>restraint. When using an ABF bias, it is recommended to set the two walls
     equal to <a href="#colvar|lowerBoundary"><span class="ectt-1095">lowerBoundary</span></a> and <a href="#colvar|upperBoundary"><span class="ectt-1095">upperBoundary</span></a>, respectively. When using a metadynamics bias,
     it is recommended to set the two walls strictly <span class="phvro8t-x-x-109">within </span><a href="#colvar|lowerBoundary"><span class="ectt-1095">lowerBoundary</span></a> and <a href="#colvar|upperBoundary"><span class="ectt-1095">upperBoundary</span></a>; see
     <a href="#sec:colvarbias_meta_boundaries">6.6.1<!--tex4ht:ref: sec:colvarbias_meta_boundaries --></a> for details.
     </li>
<li class="itemize"><span class="phvb8t-x-x-109">Colvars version 2020-11-09 or later.</span><br class="newline"/>The  legacy  keyword  <span class="ectt-1095">disableForces </span>for  atom  groups  is  now  deprecated  and  will  be
     discontinued  in  a  future  release.  Atom  groups  now  have  an  automated  way  to  save
     computation if forces are not used, and enabling this option otherwise would lead to incorrect
     behavior.
     </li></ul>
<!--l. 8191--><p class="indent">
</p>
<div class="flushright">
<!--l. 8191--><p class="noindent">
<a href="#sec:toc" style="text-decoration: none; background-color: white; color: gray;">(Go to Contents)</a><br/></p></div>
<!--l. 8191--><p class="noindent"> </p><div class="heading"> <a id="sec:compilation_notes"></a>
<h3 class="sectionHead"><span class="titlemark">9    </span> <a id="sec:compilation_notes"></a>Compilation notes</h3>
<!--l. 8191--><p class="noindent"></p></div>
<!--l. 8193--><p class="indent">   The Colvars module is typically built using the recipes of each supported software package: for this
reason, no installation instructions are needed, and the vast majority of the features described in this
manual are supported in the most common builds of each package.
</p><!--l. 8195--><p class="indent">   This section lists the few cases where the choice of compilation settings affects the availability
features in the Colvars module.
</p>
<ul class="itemize1">
<li class="itemize">Scripting commands using the Tcl language (<a class="url" href="https://www.tcl.tk"><span class="ectt-1095">https://www.tcl.tk</span></a>) are supported in VMD,
     NAMD, and Tinker-HP. All precompiled builds of NAMD and VMD include Tcl, and it is highly
     recommended to enable Tcl support in any custom build, using precompiled Tcl libraries
     from the UIUC website.
     </li>
<li class="itemize">The   Lepton   library   (<a class="url" href="https://simtk.org/projects/lepton"><span class="ectt-1095">https://simtk.org/projects/lepton</span></a>)   used   to   implement   the
     <span class="ectt-1095">customFunction </span>feature is currently included only in NAMD (always on), in LAMMPS (on by
     default) and in the Colvars-patched GROMACS releases (but currently, not in the standard
     releases). For VMD, a <a href="https://github.com/giacomofiorin/vmd-patches">patch</a> that allows to link Lepton is available.
     </li>
<li class="itemize">The Torch library is required to support the <span class="ectt-1095">torchANN </span>variable type, which is conditionally
     compiled if a Torch library is present.  In most engines, patching will be required to support
     this.
     </li>
<li class="itemize">Colvars requires the C++11 language standard or higher, which is either supported (VMD)
     or required (GROMACS, LAMMPS, NAMD) by all the engines. However, many of the VMD
     official builds are produced on very old architectures, where C++11 features are disabled at
     build time, thus limiting functionality. For details please see:<br class="newline"/><a class="url" href="https://colvars.github.io/README-c++11.html"><span class="ectt-1095">https://colvars.github.io/README-c++11.html</span></a></li></ul>
<!--l. 310--><p class="indent">
</p>
<h3 class="likesectionHead"><a id="x1-1680009"></a>References</h3>
<!--l. 1--><p class="noindent">
</p><div class="thebibliography">
<p class="bibitem"><span class="biblabel">
  [1]<span class="bibsp">   </span></span><a id="XFiorin2013"></a>Giacomo Fiorin, Michael L. Klein, and Jérôme Hénin.   Using collective variables to
     drive molecular dynamics simulations. <span class="phvro8t-x-x-109">Mol. Phys.</span>, 111(22-23):3345--3362, 2013.
     </p>
<p class="bibitem"><span class="biblabel">
  [2]<span class="bibsp">   </span></span><a id="XPhillips2020"></a>James C. Phillips, David J. Hardy, Julio D. C. Maia, John E. Stone, João V. Ribeiro,
     Rafael C.  Bernardi,  Ronak  Buch,  Giacomo  Fiorin,  Jérôme  Hénin,  Wei  Jiang,  Ryan
     McGreevy,  Marcelo  C. R.  Melo,  Brian K.  Radak,  Robert D.  Skeel,  Abhishek  Singharoy,
     Yi Wang, Benoît Roux, Aleksei Aksimentiev, Zaida Luthey-Schulten, Laxmikant V. Kalé,
     Klaus Schulten, Christophe Chipot, and Emad Tajkhorshid. Scalable molecular dynamics on
     CPU and GPU architectures with NAMD. <span class="phvro8t-x-x-109">J. Chem. Phys.</span>, 153(4):044130, 2020.
     </p>
<p class="bibitem"><span class="biblabel">
  [3]<span class="bibsp">   </span></span><a id="XIannuzzi2003"></a>M. Iannuzzi, A. Laio, and M. Parrinello.  Efficient exploration of reactive potential energy
     surfaces using car-parrinello molecular dynamics. <span class="phvro8t-x-x-109">Phys. Rev. Lett.</span>, 90(23):238302, 2003.
     </p>
<p class="bibitem"><span class="biblabel">
  [4]<span class="bibsp">   </span></span><a id="XCoutsias2004"></a>E A Coutsias, C Seok, and K A Dill.  Using quaternions to calculate RMSD.  <span class="phvro8t-x-x-109">J. Comput.</span>
<span class="phvro8t-x-x-109">Chem.</span>, 25(15):1849--1857, 2004.
     </p>
<p class="bibitem"><span class="biblabel">
  [5]<span class="bibsp">   </span></span><a id="XEbrahimi2022"></a>Mina Ebrahimi and Jérôme Hénin. Symmetry-adapted restraints for binding free energy
     calculations. <span class="phvro8t-x-x-109">Journal of Chemical Theory and Computation</span>, 18(4):2494--2502, 2022.
     </p>
<p class="bibitem"><span class="biblabel">
  [6]<span class="bibsp">   </span></span><a id="XFu2017"></a>Haohao  Fu,  Wensheng  Cai,  Jérôme  Hénin,  Benoît  Roux,  and  Christophe  Chipot.
     New coarse variables for the accurate determination of standard binding free energies.  <span class="phvro8t-x-x-109">J.</span>
<span class="phvro8t-x-x-109">Chem. Theory. Comput.</span>, 13(11):5173--5178, 2017.
     </p>
<p class="bibitem"><span class="biblabel">
  [7]<span class="bibsp">   </span></span><a id="XMu2005"></a>Yuguang  Mu,  Phuong H.  Nguyen,  and  Gerhard  Stock.   Energy  landscape  of  a  small
     peptide  revealed  by  dihedral  angle  principal  component  analysis.   <span class="phvro8t-x-x-109">Proteins</span>,  58(1):45--52,
     2005.
     </p>
<p class="bibitem"><span class="biblabel">
  [8]<span class="bibsp">   </span></span><a id="XAltis2007"></a>Alexandros  Altis,  Phuong H.  Nguyen,  Rainer  Hegger,  and  Gerhard  Stock.    Dihedral
     angle  principal  component  analysis  of  molecular  dynamics  simulations.   <span class="phvro8t-x-x-109">J.  Chem.  Phys.</span>,
     126(24):244111, 2007.
     </p>
<p class="bibitem"><span class="biblabel">
  [9]<span class="bibsp">   </span></span><a id="XGlykos2006"></a>Nicholas M Glykos. Carma: a molecular dynamics analysis program. <span class="phvro8t-x-x-109">J. Comput. Chem.</span>,
     27(14):1765--1768, 2006.
     </p>
<p class="bibitem"><span class="biblabel">
 [10]<span class="bibsp">   </span></span><a id="XLagardere2023"></a>Louis  Lagardère,  Lise  Maurin,  Olivier  Adjoua,  Krystel El  Hage,  Pierre  Monmarché,
     Jean-Philip  Piquemal,  and  Jérôme  Hénin.      Lambda-abf:  Simplified,  accurate  and
     cost-effective alchemical free energy computations, 2023.
     </p>
<p class="bibitem"><span class="biblabel">
 [11]<span class="bibsp">   </span></span><a id="XLeines2012"></a>G. D. Leines and B. Ensing.  Path finding on high-dimensional free energy landscapes.
     <span class="phvro8t-x-x-109">Phys. Rev. Lett.</span>, 109:020601, 2012.
     </p>
<p class="bibitem"><span class="biblabel">
 [12]<span class="bibsp">   </span></span><a id="XBranduardi2007"></a>Davide Branduardi, Francesco Luigi Gervasio, and Michele Parrinello. From a to b in free
     energy space. <span class="phvro8t-x-x-109">J Chem Phys</span>, 126(5):054103, 2007.
     </p>
<p class="bibitem"><span class="biblabel">
 [13]<span class="bibsp">   </span></span><a id="XHovan2019"></a>F. Comitani  L. Hovan  and  F. L.  Gervasio.    Defining  an  optimal  metric  for  the  path
     collective variables. <span class="phvro8t-x-x-109">J. Chem. Theory Comput.</span>, 15:25--32, 2019.
     </p>
<p class="bibitem"><span class="biblabel">
 [14]<span class="bibsp">   </span></span><a id="XChen2022"></a>Haochuan Chen, Han Liu, Heying Feng, Haohao Fu, Wensheng Cai, Xueguang Shao,
     and Christophe Chipot.  Mlcv: Bridging Machine-Learning-Based Dimensionality Reduction
     and Free-Energy Calculation. <span class="phvro8t-x-x-109">J. Chem. Inf. Model.</span>, 62(1):1--8, 2022.
     </p>
<p class="bibitem"><span class="biblabel">
 [15]<span class="bibsp">   </span></span><a id="XFiorin2020"></a>Giacomo Fiorin, Fabrizio Marinelli, and José D. Faraldo-Gómez.   Direct derivation of
     free energies of membrane deformation and other solvent density variations from enhanced
     sampling molecular dynamics. <span class="phvro8t-x-x-109">J. Comp. Chem.</span>, 41(5):449--459, 2020.
     </p>
<p class="bibitem"><span class="biblabel">
 [16]<span class="bibsp">   </span></span><a id="XFerrarotti2015"></a>Marco Jacopo  Ferrarotti,  Sandro  Bottaro,  Andrea  Pérez-Villa,  and  Giovanni  Bussi.
     Accurate multiple time step in biased molecular simulations.  <span class="phvro8t-x-x-109">Journal of chemical theory and</span>
<span class="phvro8t-x-x-109">computation</span>, 11:139--146, 2015.
                                                                                         
                                                                                         
     </p>
<p class="bibitem"><span class="biblabel">
 [17]<span class="bibsp">   </span></span><a id="XSalari2018"></a>Reza  Salari,  Thomas  Joseph,  Ruchi  Lohia,  Jérôme  Hénin,  and  Grace  Brannigan.
     A  streamlined,  general  approach  for  computing  ligand  binding  free  energies  and  its
     application  to  GPCR-bound  cholesterol.    <span class="phvro8t-x-x-109">Journal  of  Chemical  Theory  and  Computation</span>,
     14(12):6560--6573, 2018.
     </p>
<p class="bibitem"><span class="biblabel">
 [18]<span class="bibsp">   </span></span><a id="XDarve2008"></a>Eric Darve, David Rodríguez-Gómez, and Andrew Pohorille.   Adaptive biasing force
     method  for  scalar  and  vector  free  energy  calculations.   <span class="phvro8t-x-x-109">J.  Chem.  Phys.</span>,  128(14):144120,
     2008.
     </p>
<p class="bibitem"><span class="biblabel">
 [19]<span class="bibsp">   </span></span><a id="XHenin2004"></a>J. Hénin and C. Chipot. Overcoming free energy barriers using unconstrained molecular
     dynamics simulations. <span class="phvro8t-x-x-109">J. Chem. Phys.</span>, 121:2904--2914, 2004.
     </p>
<p class="bibitem"><span class="biblabel">
 [20]<span class="bibsp">   </span></span><a id="XHenin2010"></a>Jérôme Hénin, Giacomo Fiorin, Christophe Chipot, and Michael L. Klein.  Exploring
     multidimensional   free   energy   landscapes   using   time-dependent   biases   on   collective
     variables. <span class="phvro8t-x-x-109">J. Chem. Theory Comput.</span>, 6(1):35--47, 2010.
     </p>
<p class="bibitem"><span class="biblabel">
 [21]<span class="bibsp">   </span></span><a id="XCarter1989"></a>A. Carter, E, G. Ciccotti, J. T. Hynes, and R. Kapral.  Constrained reaction coordinate
     dynamics for the simulation of rare events. <span class="phvro8t-x-x-109">Chem. Phys. Lett.</span>, 156:472--477, 1989.
     </p>
<p class="bibitem"><span class="biblabel">
 [22]<span class="bibsp">   </span></span><a id="XRuiz-Montero1997"></a>M. J. Ruiz-Montero, D. Frenkel, and J. J. Brey.  Efficient schemes to compute diffusive
     barrier crossing rates. <span class="phvro8t-x-x-109">Mol. Phys.</span>, 90:925--941, 1997.
     </p>
<p class="bibitem"><span class="biblabel">
 [23]<span class="bibsp">   </span></span><a id="XdenOtter2000"></a>W. K.  den  Otter.     Thermodynamic  integration  of  the  free  energy  along  a  reaction
     coordinate in cartesian coordinates. <span class="phvro8t-x-x-109">J. Chem. Phys.</span>, 112:7283--7292, 2000.
     </p>
<p class="bibitem"><span class="biblabel">
 [24]<span class="bibsp">   </span></span><a id="XCiccotti2005"></a>Giovanni  Ciccotti,  Raymond  Kapral,  and  Eric  Vanden-Eijnden.    Blue  moon  sampling,
     vectorial  reaction  coordinates,  and  unbiased  constrained  dynamics.     <span class="phvro8t-x-x-109">ChemPhysChem</span>,
     6(9):1809--1814, 2005.
     </p>
<p class="bibitem"><span class="biblabel">
 [25]<span class="bibsp">   </span></span><a id="XMinoukadeh2010"></a>K. Minoukadeh,  C. Chipot,  and  T. Lelièvre.   Potential  of  mean  force  calculations:  A
     multiple-walker adaptive biasing force approach.   <span class="phvro8t-x-x-109">J. Chem. Theor. Comput.</span>, 6:1008--1017,
     2010.
     </p>
<p class="bibitem"><span class="biblabel">
 [26]<span class="bibsp">   </span></span><a id="XComer2014c"></a>Jeffrey Comer, James C. Phillips, Klaus Schulten, and Christophe Chipot. Multiple-walker
     strategies for free-energy calculations in NAMD: Shared adaptive biasing force and walker
     selection rules. <span class="phvro8t-x-x-109">J. Chem. Theor. Comput.</span>, 10(12):5276--5285, 2014.
     </p>
<p class="bibitem"><span class="biblabel">
 [27]<span class="bibsp">   </span></span><a id="XFiorin2024"></a>Giacomo Fiorin, Fabrizio Marinelli, Lucy R. Forrest, Haochuan Chen, Christophe Chipot,
     Axel Kohlmeyer, Hubert Santuz, and Jérôme Hénin. Expanded functionality and portability
     for the colvars library. <span class="phvro8t-x-x-109">J. Phys. Chem. B</span>, 128(45):11108--11123, 2024.
     </p>
<p class="bibitem"><span class="biblabel">
 [28]<span class="bibsp">   </span></span><a id="XHenin2021"></a>J. Hénin.  Fast and accurate multidimensional free energy integration.  <span class="phvro8t-x-x-109">J. Chem. Theory</span>
<span class="phvro8t-x-x-109">Comput.</span>, 2021.
     </p>
<p class="bibitem"><span class="biblabel">
 [29]<span class="bibsp">   </span></span><a id="XLesage2017"></a>Adrien Lesage, Tony Lelièvre, Gabriel Stoltz, and Jérôme Hénin.  Smoothed biasing
     forces yield unbiased free energies with the extended-system adaptive biasing force method.
     <span class="phvro8t-x-x-109">J. Phys. Chem. B</span>, 121(15):3676--3685, 2017.
     </p>
<p class="bibitem"><span class="biblabel">
 [30]<span class="bibsp">   </span></span><a id="XFu2016"></a>Haohao Fu, Xueguang Shao, Christophe Chipot, and Wensheng Cai. Extended adaptive
     biasing force algorithm. An on--the--fly implementation for accurate free--energy calculations.
     <span class="phvro8t-x-x-109">J. Chem. Theory Comput.</span>, 12(8):3506----3513, 2016.
     </p>
<p class="bibitem"><span class="biblabel">
 [31]<span class="bibsp">   </span></span><a id="XZheng2012"></a>L. Zheng  and  W. Yang.     Practically  efficient  and  robust  free  energy  calculations:
     Double-integration orthogonal space tempering. <span class="phvro8t-x-x-109">J. Chem. Theor. Compt.</span>, 8:810--823, 2012.
     </p>
<p class="bibitem"><span class="biblabel">
 [32]<span class="bibsp">   </span></span><a id="XKastner2005"></a>J. Kastner  and  W. Thiel.    Bridging  the  gap  between  thermodynamic  integration  and
     umbrella sampling provides a novel analysis method: “umbrella integration". <span class="phvro8t-x-x-109">J. Chem. Phys.</span>,
     123(14):144104, 2005.
     </p>
<p class="bibitem"><span class="biblabel">
 [33]<span class="bibsp">   </span></span><a id="XMarchi1999"></a>Massimo  Marchi  and  Pietro  Ballone.    Adiabatic  bias  molecular  dynamics:  A  method
     to  navigate  the  conformational  space  of  complex  molecular  systems.    <span class="phvro8t-x-x-109">J.  Chem.  Phys.</span>,
     110(8):3697--3702, 1999.
     </p>
<p class="bibitem"><span class="biblabel">
 [34]<span class="bibsp">   </span></span><a id="XLaio2002"></a>A. Laio and M. Parrinello.   Escaping free-energy minima.   <span class="phvro8t-x-x-109">Proc. Natl. Acad. Sci. USA</span>,
     99(20):12562--12566, 2002.
     </p>
<p class="bibitem"><span class="biblabel">
 [35]<span class="bibsp">   </span></span><a id="XGrubmuller1995"></a>Helmut Grubmüller.   Predicting slow structural transitions in macromolecular systems:
     Conformational flooding. <span class="phvro8t-x-x-109">Phys. Rev. E</span>, 52(3):2893--2906, Sep 1995.
     </p>
<p class="bibitem"><span class="biblabel">
 [36]<span class="bibsp">   </span></span><a id="XHuber1994"></a>T. Huber, A. E. Torda, and W.F. van Gunsteren. Local elevation - A method for improving
     the  searching  properties  of  molecular-dynamics  simulation.    <span class="phvro8t-x-x-109">Journal  of  Computer-Aided</span>
<span class="phvro8t-x-x-109">Molecular Design</span>, 8(6):695--708, DEC 1994.
     </p>
<p class="bibitem"><span class="biblabel">
 [37]<span class="bibsp">   </span></span><a id="XBussi2006"></a>G. Bussi,  A. Laio,  and  M. Parrinello.    Equilibrium  free  energies  from  nonequilibrium
     metadynamics. <span class="phvro8t-x-x-109">Phys. Rev. Lett.</span>, 96(9):090601, 2006.
     </p>
<p class="bibitem"><span class="biblabel">
 [38]<span class="bibsp">   </span></span><a id="XMarinelli2009"></a>Fabrizio Marinelli, Fabio Pietrucci, Alessandro Laio, and Stefano Piana. A kinetic model of
     trp-cage folding from multiple biased molecular dynamics simulations.  <span class="phvro8t-x-x-109">PLOS Computational</span>
<span class="phvro8t-x-x-109">Biology</span>, 5(8):1--18, 2009.
     </p>
<p class="bibitem"><span class="biblabel">
 [39]<span class="bibsp">   </span></span><a id="XCrespo2010"></a>Yanier Crespo, Fabrizio Marinelli, Fabio Pietrucci, and Alessandro Laio.  Metadynamics
     convergence law in a multidimensional system. <span class="phvro8t-x-x-109">Phys. Rev. E</span>, 81:055701, May 2010.
     </p>
<p class="bibitem"><span class="biblabel">
 [40]<span class="bibsp">   </span></span><a id="XMarinelli2015"></a>Fabrizio  Marinelli  and  José D.  Faraldo-Gómez.     Ensemble-biased  metadynamics:
     A   molecular   simulation   method   to   sample   experimental   distributions.      <span class="phvro8t-x-x-109">Biophys.   J.</span>,
     108(12):2779--2782, 2015.
     </p>
<p class="bibitem"><span class="biblabel">
 [41]<span class="bibsp">   </span></span><a id="XBarducci2008"></a>Alessandro   Barducci,   Giovanni   Bussi,   and   Michele   Parrinello.        Well-tempered
     metadynamics: A smoothly converging and tunable free-energy method.   <span class="phvro8t-x-x-109">Phys. Rev. Lett.</span>,
     100:020603, 2008.
                                                                                         
                                                                                         
     </p>
<p class="bibitem"><span class="biblabel">
 [42]<span class="bibsp">   </span></span><a id="XRaiteri2006"></a>P. Raiteri,   A. Laio,   F. L.   Gervasio,   C. Micheletti,   and   M. Parrinello.        Efficient
     reconstruction  of  complex  free  energy  landscapes  by  multiple  walkers  metadynamics.   <span class="phvro8t-x-x-109">J.</span>
<span class="phvro8t-x-x-109">Phys. Chem. B</span>, 110(8):3533--9, 2006.
     </p>
<p class="bibitem"><span class="biblabel">
 [43]<span class="bibsp">   </span></span><a id="XInvernizzi2020"></a>Michele Invernizzi and Michele Parrinello. Rethinking Metadynamics: From Bias Potentials
     to Probability Distributions. <span class="phvro8t-x-x-109">J. Phys. Chem. Lett.</span>, 11(7):2731--2736, April 2020.
     </p>
<p class="bibitem"><span class="biblabel">
 [44]<span class="bibsp">   </span></span><a id="XInvernizzi2022"></a>Michele  Invernizzi  and  Michele  Parrinello.     Exploration  vs  Convergence  Speed  in
     Adaptive-Bias Enhanced Sampling. <span class="phvro8t-x-x-109">J. Chem. Theory Comput.</span>, 18(6):3988--3996, June 2022.
     </p>
<p class="bibitem"><span class="biblabel">
 [45]<span class="bibsp">   </span></span><a id="Xtribello_plumed_2014"></a>Gareth A.  Tribello,  Massimiliano  Bonomi,  Davide  Branduardi,  Carlo  Camilloni,  and
     Giovanni  Bussi.    PLUMED  2:  New  feathers  for  an  old  bird.    <span class="phvro8t-x-x-109">Comput.  Phys.  Commun.</span>,
     185(2):604--613, February 2014.
     </p>
<p class="bibitem"><span class="biblabel">
 [46]<span class="bibsp">   </span></span><a id="XDeng2009"></a>Yuqing Deng and Benoît Roux.  Computations of standard binding free energies with
     molecular dynamics simulations. <span class="phvro8t-x-x-109">J. Phys. Chem. B</span>, 113(8):2234--2246, 2009.
     </p>
<p class="bibitem"><span class="biblabel">
 [47]<span class="bibsp">   </span></span><a id="XPitera2012"></a>Jed W. Pitera and John D. Chodera.  On the use of experimental observations to bias
     simulated ensembles. <span class="phvro8t-x-x-109">J. Chem. Theory Comput.</span>, 8:3445--3451, 2012.
     </p>
<p class="bibitem"><span class="biblabel">
 [48]<span class="bibsp">   </span></span><a id="XWhite2014"></a>Andrew D. White and Gregory A. Voth.  Efficient and minimal method to bias molecular
     simulations with experimental data. <span class="phvro8t-x-x-109">J. Chem. Theory Comput.</span>, 10(8):3023----3030, 2014.
     </p>
<p class="bibitem"><span class="biblabel">
 [49]<span class="bibsp">   </span></span><a id="Xhamelberg2004"></a>Donald Hamelberg, John Mongan, and J. Andrew McCammon.  Accelerated molecular
     dynamics: A promising and efficient simulation method for biomolecules.   <span class="phvro8t-x-x-109">J. Chem. Phys.</span>,
     120(24):11919--11929, June 2004.
     </p>
<p class="bibitem"><span class="biblabel">
 [50]<span class="bibsp">   </span></span><a id="XMiao2015"></a>Yinglong  Miao,  Victoria A.  Feher,  and  J. Andrew  McCammon.   Gaussian  Accelerated
     Molecular Dynamics: Unconstrained Enhanced Sampling and Free Energy Calculation.   <span class="phvro8t-x-x-109">J.</span>
<span class="phvro8t-x-x-109">Chem. Theory Comput</span>, 11(8):3584--3595, August 2015.
                                                                                         
                                                                                         
     </p>
<p class="bibitem"><span class="biblabel">
 [51]<span class="bibsp">   </span></span><a id="XChen2021"></a>Haochuan Chen, Haohao Fu, Christophe Chipot, Xueguang Shao, and Wensheng Cai.
     Overcoming  free-energy  barriers  with  a  seamless  combination  of  a  biasing  force  and  a
     collective variable-independent boost potential. <span class="phvro8t-x-x-109">J. Chem. Theory Comput.</span>, 17(7):3886--3894,
     2021.
     </p>
<p class="bibitem"><span class="biblabel">
 [52]<span class="bibsp">   </span></span><a id="XShen2015"></a>Rong Shen, Wei Han, Giacomo Fiorin, Shahidul M Islam, Klaus Schulten, and Benoît
     Roux.  Structural refinement of proteins by restrained molecular dynamics simulations with
     non-interacting molecular fragments. <span class="phvro8t-x-x-109">PLoS Comput. Biol.</span>, 11(10):e1004368, 2015.
</p>
</div>
<!--l. 318--><p class="indent"> </p></div> </div>
</body>
</html>
