<?xml version="1.0" encoding="utf-8" ?> 
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" 
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">  
<!--http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd-->  
<html xmlns="http://www.w3.org/1999/xhtml"  
> 
<head><title></title> 
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" /> 
<meta name="generator" content="TeX4ht (http://www.cse.ohio-state.edu/~gurari/TeX4ht/)" /> 
<meta name="originator" content="TeX4ht (http://www.cse.ohio-state.edu/~gurari/TeX4ht/)" /> 
<!-- xhtml,charset=utf-8,html --> 
<meta name="src" content="colvars-refman-namd.tex" /> 
<meta name="date" content="2017-09-19 16:50:00" /> 
<link rel="stylesheet" type="text/css" href="colvars-refman-namd.css" /> 
</head><body 
>
<!--l. 39--><p class="noindent" >
                                                                                         
                                                                                         
</p><!--l. 41--><p class="noindent" ><span 
class="colorbox" id="colorbox1"> </p><div class="minipage"><span style="margin-left:0.3em" class="thinspace"></span><br 
class="newline" />
<!--l. 52--><p class="noindent" ><span 
class="phvb7t-x-x-248">COLLECTIVE VARIABLES MODULE</span><br 
class="newline" />
</p><!--l. 52--><p class="noindent" ><span 
class="phvb7t-x-x-248">Reference manual for NAMD</span>
</p><!--l. 52--><p class="noindent" ><br 
class="newline" /><span 
class="phvr7t-x-x-109">Code version: 2017-09-18</span><br 
class="newline" /></p></div></span>
<!--l. 55--><p class="indent" >   <img 
src="cover.jpg" alt="PIC"  
 /> <span 
class="phvr7t-x-x-120">Giacomo Fiorin, J</span><span 
class="phvr7t-x-x-120">ér</span><span 
class="phvr7t-x-x-120">ôme H</span><span 
class="phvr7t-x-x-120">énin</span>
                                                                                         
                                                                                         
</p><!--l. 60--><p class="noindent" >
                                                                                         
                                                                                         
</p>
   <h3 class="likesectionHead"><a 
 id="x1-1000"></a>Contents</h3>
   <div class="tableofcontents">
   <span class="sectionToc" >1 <a 
href="#x1-20001" id="QQ2-1-2">Introduction</a></span>
<br />   <span class="sectionToc" >2 <a 
href="#x1-30002" id="QQ2-1-3">A crash course</a></span>
<br />   <span class="sectionToc" >3 <a 
href="#x1-40003" id="QQ2-1-4">General parameters and input/output ﬁles</a></span>
<br />    <span class="subsectionToc" >3.1 <a 
href="#x1-50003.1" id="QQ2-1-5">NAMD parameters</a></span>
<br />    <span class="subsectionToc" >3.2 <a 
href="#sec:colvars_config" id="QQ2-1-6">Conﬁguration syntax for the Colvars module</a></span>
<br />    <span class="subsectionToc" >3.3 <a 
href="#x1-70003.3" id="QQ2-1-8">Input state ﬁle (optional)</a></span>
<br />    <span class="subsectionToc" >3.4 <a 
href="#x1-80003.4" id="QQ2-1-9">Output ﬁles</a></span>
<br />   <span class="sectionToc" >4 <a 
href="#sec:colvar" id="QQ2-1-10">Deﬁning collective variables and their properties</a></span>
<br />    <span class="subsectionToc" >4.1 <a 
href="#sec:colvar_general" id="QQ2-1-11">General options for a collective variable</a></span>
<br />    <span class="subsectionToc" >4.2 <a 
href="#x1-110004.2" id="QQ2-1-12">Trajectory output</a></span>
<br />    <span class="subsectionToc" >4.3 <a 
href="#sec:colvar_extended" id="QQ2-1-13">Extended Lagrangian.</a></span>
<br />    <span class="subsectionToc" >4.4 <a 
href="#sec:colvar_acf" id="QQ2-1-14">Statistical analysis of collective variables</a></span>
<br />   <span class="sectionToc" >5 <a 
href="#sec:colvar_atom_groups" id="QQ2-1-15">Selecting atoms for colvars: deﬁning atom groups</a></span>
<br />    <span class="subsectionToc" >5.1 <a 
href="#x1-150005.1" id="QQ2-1-16">Selection keywords</a></span>
<br />    <span class="subsectionToc" >5.2 <a 
href="#key:colvars:atom_group:refPositions" id="QQ2-1-17">Moving frame of reference.</a></span>
<br />    <span class="subsectionToc" >5.3 <a 
href="#x1-170005.3" id="QQ2-1-18">Treatment of periodic boundary conditions.</a></span>
<br />    <span class="subsectionToc" >5.4 <a 
href="#x1-180005.4" id="QQ2-1-19">Performance a Colvars calculation based on group size.</a></span>
<br />   <span class="sectionToc" >6 <a 
href="#sec:cvc" id="QQ2-1-20">Collective variable components (basis functions)</a></span>
<br />    <span class="subsectionToc" >6.1 <a 
href="#x1-200006.1" id="QQ2-1-21">List of available colvar components</a></span>
<br />     <span class="subsubsectionToc" >6.1.1 <a 
href="#colvar|distance|oneSiteTotalForce" id="QQ2-1-22"><span 
class="ot1-zi4r-0-x-x-109">distance</span>: center-of-mass distance between two groups.</a></span>
<br />     <span class="subsubsectionToc" >6.1.2 <a 
href="#colvar|distanceZ|ref" id="QQ2-1-23"><span 
class="ot1-zi4r-0-x-x-109">distanceZ</span>: projection of a distance vector on an axis.</a></span>
<br />     <span class="subsubsectionToc" >6.1.3 <a 
href="#x1-230006.1.3" id="QQ2-1-24"><span 
class="ot1-zi4r-0-x-x-109">distanceXY</span>: modulus of the projection of a distance vector on a plane.</a></span>
<br />     <span class="subsubsectionToc" >6.1.4 <a 
href="#x1-240006.1.4" id="QQ2-1-25"><span 
class="ot1-zi4r-0-x-x-109">distanceVec</span>: distance vector between two groups.</a></span>
<br />     <span class="subsubsectionToc" >6.1.5 <a 
href="#x1-250006.1.5" id="QQ2-1-26"><span 
class="ot1-zi4r-0-x-x-109">distanceDir</span>: distance unit vector between two groups.</a></span>
<br />     <span class="subsubsectionToc" >6.1.6 <a 
href="#x1-260006.1.6" id="QQ2-1-27"><span 
class="ot1-zi4r-0-x-x-109">distanceInv</span>: mean distance between two groups of atoms.</a></span>
<br />     <span class="subsubsectionToc" >6.1.7 <a 
href="#x1-270006.1.7" id="QQ2-1-28"><span 
class="ot1-zi4r-0-x-x-109">distancePairs</span>: set of pairwise distances between two groups.</a></span>
<br />     <span class="subsubsectionToc" >6.1.8 <a 
href="#x1-280006.1.8" id="QQ2-1-29"><span 
class="ot1-zi4r-0-x-x-109">cartesian</span>: vector of atomic Cartesian coordinates.</a></span>
<br />     <span class="subsubsectionToc" >6.1.9 <a 
href="#x1-290006.1.9" id="QQ2-1-30"><span 
class="ot1-zi4r-0-x-x-109">angle</span>: angle between three groups.</a></span>
<br />     <span class="subsubsectionToc" >6.1.10 <a 
href="#x1-300006.1.10" id="QQ2-1-31"><span 
class="ot1-zi4r-0-x-x-109">dipoleAngle</span>: angle between two groups and dipole of a third group.</a></span>
<br />     <span class="subsubsectionToc" >6.1.11 <a 
href="#x1-310006.1.11" id="QQ2-1-32"><span 
class="ot1-zi4r-0-x-x-109">dihedral</span>: torsional angle between four groups.</a></span>
<br />     <span class="subsubsectionToc" >6.1.12 <a 
href="#x1-320006.1.12" id="QQ2-1-33"><span 
class="ot1-zi4r-0-x-x-109">polarTheta</span>: polar angle in spherical coordinates.</a></span>
<br />     <span class="subsubsectionToc" >6.1.13 <a 
href="#x1-330006.1.13" id="QQ2-1-34"><span 
class="ot1-zi4r-0-x-x-109">polarPhi</span>: azimuthal angle in spherical coordinates.</a></span>
<br />     <span class="subsubsectionToc" >6.1.14 <a 
href="#colvar|coordNum|cutoff" id="QQ2-1-35"><span 
class="ot1-zi4r-0-x-x-109">coordNum</span>: coordination number between two groups.</a></span>
<br />     <span class="subsubsectionToc" >6.1.15 <a 
href="#x1-350006.1.15" id="QQ2-1-36"><span 
class="ot1-zi4r-0-x-x-109">selfCoordNum</span>: coordination number between atoms within a group.</a></span>
<br />     <span class="subsubsectionToc" >6.1.16 <a 
href="#x1-360006.1.16" id="QQ2-1-37"><span 
class="ot1-zi4r-0-x-x-109">hBond</span>: hydrogen bond between two atoms.</a></span>
<br />     <span class="subsubsectionToc" >6.1.17 <a 
href="#colvar|rmsd|refPositions" id="QQ2-1-38"><span 
class="ot1-zi4r-0-x-x-109">rmsd</span>: root mean square displacement (RMSD) from reference positions.</a></span>
<br />     <span class="subsubsectionToc" >6.1.18 <a 
href="#x1-380006.1.18" id="QQ2-1-39">Advanced usage of the <span 
class="ot1-zi4r-0-x-x-109">rmsd </span>component.</a></span>
<br />     <span class="subsubsectionToc" >6.1.19 <a 
href="#x1-390006.1.19" id="QQ2-1-40"><span 
class="ot1-zi4r-0-x-x-109">eigenvector</span>: projection of the atomic coordinates on a vector.</a></span>
<br />     <span class="subsubsectionToc" >6.1.20 <a 
href="#x1-400006.1.20" id="QQ2-1-41"><span 
class="ot1-zi4r-0-x-x-109">gyration</span>: radius of gyration of a group of atoms.</a></span>
<br />     <span class="subsubsectionToc" >6.1.21 <a 
href="#x1-410006.1.21" id="QQ2-1-42"><span 
class="ot1-zi4r-0-x-x-109">inertia</span>: total moment of inertia of a group of atoms.</a></span>
<br />     <span class="subsubsectionToc" >6.1.22 <a 
href="#x1-420006.1.22" id="QQ2-1-43"><span 
class="ot1-zi4r-0-x-x-109">inertiaZ</span>: total moment of inertia of a group of atoms around a chosen axis.</a></span>
<br />     <span class="subsubsectionToc" >6.1.23 <a 
href="#x1-430006.1.23" id="QQ2-1-44"><span 
class="ot1-zi4r-0-x-x-109">orientation</span>: orientation from reference coordinates.</a></span>
<br />     <span class="subsubsectionToc" >6.1.24 <a 
href="#x1-440006.1.24" id="QQ2-1-45"><span 
class="ot1-zi4r-0-x-x-109">orientationAngle</span>: angle of rotation from reference coordinates.</a></span>
<br />     <span class="subsubsectionToc" >6.1.25 <a 
href="#x1-450006.1.25" id="QQ2-1-46"><span 
class="ot1-zi4r-0-x-x-109">orientationProj</span>: cosine of the angle of rotation from reference coordinates.</a></span>
                                                                                         
                                                                                         
<br />     <span class="subsubsectionToc" >6.1.26 <a 
href="#colvar|spinAngle|axis" id="QQ2-1-47"><span 
class="ot1-zi4r-0-x-x-109">spinAngle</span>: angle of rotation around a given axis.</a></span>
<br />     <span class="subsubsectionToc" >6.1.27 <a 
href="#x1-470006.1.27" id="QQ2-1-48"><span 
class="ot1-zi4r-0-x-x-109">tilt</span>: cosine of the rotation orthogonal to a given axis.</a></span>
<br />     <span class="subsubsectionToc" >6.1.28 <a 
href="#colvar|alpha|psfSegID" id="QQ2-1-49"><span 
class="ot1-zi4r-0-x-x-109">alpha</span>: <span 
class="zptmcm7m-x-x-109">α</span>-helix content of a protein segment.</a></span>
<br />     <span class="subsubsectionToc" >6.1.29 <a 
href="#x1-490006.1.29" id="QQ2-1-50"><span 
class="ot1-zi4r-0-x-x-109">dihedralPC</span>: protein dihedral pricipal component</a></span>
<br />    <span class="subsectionToc" >6.2 <a 
href="#x1-500006.2" id="QQ2-1-51">Conﬁguration keywords shared by all components</a></span>
<br />    <span class="subsectionToc" >6.3 <a 
href="#x1-510006.3" id="QQ2-1-52">Advanced usage and special considerations</a></span>
<br />     <span class="subsubsectionToc" >6.3.1 <a 
href="#x1-520006.3.1" id="QQ2-1-53">Periodic components.</a></span>
<br />     <span class="subsubsectionToc" >6.3.2 <a 
href="#x1-530006.3.2" id="QQ2-1-54">Non-scalar components.</a></span>
<br />     <span class="subsubsectionToc" >6.3.3 <a 
href="#sec:cvc_sys_forces" id="QQ2-1-55">Calculating total forces.</a></span>
<br />    <span class="subsectionToc" >6.4 <a 
href="#sec:cvc_superp" id="QQ2-1-56">Linear and polynomial combinations of components</a></span>
<br />    <span class="subsectionToc" >6.5 <a 
href="#x1-560006.5" id="QQ2-1-57">Colvars as custom functions of components</a></span>
<br />    <span class="subsectionToc" >6.6 <a 
href="#x1-570006.6" id="QQ2-1-58">Colvars as scripted functions of components</a></span>
<br />   <span class="sectionToc" >7 <a 
href="#sec:colvarbias" id="QQ2-1-59">Biasing and analysis methods</a></span>
<br />    <span class="subsectionToc" >7.1 <a 
href="#sec:colvarbias_abf" id="QQ2-1-60">Adaptive Biasing Force</a></span>
<br />     <span class="subsubsectionToc" >7.1.1 <a 
href="#x1-600007.1.1" id="QQ2-1-61">ABF requirements on collective variables</a></span>
<br />     <span class="subsubsectionToc" >7.1.2 <a 
href="#x1-610007.1.2" id="QQ2-1-62">Parameters for ABF</a></span>
<br />     <span class="subsubsectionToc" >7.1.3 <a 
href="#sec:mw-ABF" id="QQ2-1-63">Multiple-replica ABF</a></span>
<br />     <span class="subsubsectionToc" >7.1.4 <a 
href="#x1-630007.1.4" id="QQ2-1-64">Output ﬁles</a></span>
<br />     <span class="subsubsectionToc" >7.1.5 <a 
href="#x1-640007.1.5" id="QQ2-1-65">Post-processing: reconstructing a multidimensional free energy surface</a></span>
<br />    <span class="subsectionToc" >7.2 <a 
href="#sec:eABF" id="QQ2-1-66">Extended-system Adaptive Biasing Force (eABF)</a></span>
<br />     <span class="subsubsectionToc" >7.2.1 <a 
href="#x1-660007.2.1" id="QQ2-1-67">CZAR estimator of the free energy</a></span>
<br />     <span class="subsubsectionToc" >7.2.2 <a 
href="#x1-670007.2.2" id="QQ2-1-68">Zheng/Yang estimator of the free energy</a></span>
<br />     <span class="paragraphToc" ><a 
href="#x1-680007.2.2" id="QQ2-1-69">Usage for multiple–replica eABF.</a></span>
<br />    <span class="subsectionToc" >7.3 <a 
href="#sec:colvarbias_meta" id="QQ2-1-70">Metadynamics</a></span>
<br />     <span class="subsubsectionToc" >7.3.1 <a 
href="#x1-700007.3.1" id="QQ2-1-71">Output ﬁles</a></span>
<br />     <span class="subsubsectionToc" >7.3.2 <a 
href="#x1-710007.3.2" id="QQ2-1-72">Performance tuning</a></span>
<br />     <span class="subsubsectionToc" >7.3.3 <a 
href="#x1-720007.3.3" id="QQ2-1-73">Well-tempered metadynamics</a></span>
<br />     <span class="subsubsectionToc" >7.3.4 <a 
href="#x1-730007.3.4" id="QQ2-1-74">Multiple-replicas metadynamics</a></span>
<br />     <span class="subsubsectionToc" >7.3.5 <a 
href="#x1-740007.3.5" id="QQ2-1-75">Compatibility and post-processing</a></span>
<br />    <span class="subsectionToc" >7.4 <a 
href="#sec:colvarbias_harmonic" id="QQ2-1-76">Harmonic restraints</a></span>
<br />     <span class="subsubsectionToc" >7.4.1 <a 
href="#x1-760007.4.1" id="QQ2-1-77">Moving restraints: steered molecular dynamics</a></span>
<br />     <span class="subsubsectionToc" >7.4.2 <a 
href="#x1-770007.4.2" id="QQ2-1-78">Moving restraints: umbrella sampling</a></span>
<br />     <span class="subsubsectionToc" >7.4.3 <a 
href="#x1-780007.4.3" id="QQ2-1-79">Changing force constant</a></span>
<br />    <span class="subsectionToc" >7.5 <a 
href="#sec:colvarbias_harmonic_walls" id="QQ2-1-80">Harmonic wall restraints</a></span>
<br />    <span class="subsectionToc" >7.6 <a 
href="#sec:colvarbias_linear" id="QQ2-1-81">Linear restraints</a></span>
<br />    <span class="subsectionToc" >7.7 <a 
href="#sec:colvarbias_alb" id="QQ2-1-82">Adaptive Linear Bias/Experiment Directed Simulation</a></span>
<br />    <span class="subsectionToc" >7.8 <a 
href="#sec:colvarbias_histogram" id="QQ2-1-83">Multidimensional histograms</a></span>
<br />     <span class="subsubsectionToc" >7.8.1 <a 
href="#x1-830007.8.1" id="QQ2-1-84">Grid deﬁnition for multidimensional histograms</a></span>
<br />    <span class="subsectionToc" >7.9 <a 
href="#x1-840007.9" id="QQ2-1-85">Probability distribution-restraints</a></span>
<br />    <span class="subsectionToc" >7.10 <a 
href="#sec:scripted_biases" id="QQ2-1-86">Scripted biases</a></span>
<br />   <span class="sectionToc" >8 <a 
href="#sec:scripting" id="QQ2-1-87">Colvars scripting</a></span>
<br />    <span class="subsectionToc" >8.1 <a 
href="#x1-870008.1" id="QQ2-1-88">Managing the colvars module</a></span>
<br />    <span class="subsectionToc" >8.2 <a 
href="#x1-880008.2" id="QQ2-1-89">Input and output</a></span>
<br />    <span class="subsectionToc" >8.3 <a 
href="#x1-890008.3" id="QQ2-1-90">Accessing collective variables</a></span>
<br />    <span class="subsectionToc" >8.4 <a 
href="#x1-900008.4" id="QQ2-1-91">Accessing biases</a></span>
   </div>
                                                                                         
                                                                                         
   <h3 class="sectionHead"><span class="titlemark">1    </span> <a 
 id="x1-20001"></a>Introduction</h3>
<!--l. 15--><p class="noindent" >In molecular dynamics simulations, it is often useful to reduce the large number of degrees of freedom of a physical
system into few parameters whose statistical distributions can be analyzed individually, or used to
deﬁne biasing potentials to alter the dynamics of the system in a controlled manner. These have been
called ‘order parameters&#x2019;, ‘collective variables&#x2019;, ‘(surrogate) reaction coordinates&#x2019;, and many other
terms.
</p><!--l. 18--><p class="indent" >   Here we use primarily the term ‘collective variable&#x2019; (shortened to <span 
class="ptmri7t-x-x-109">colvar</span>), which indicates any
diﬀerentiable function of atomic Cartesian coordinates, <span 
class="zptmcm7m-x-x-109">x</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub>, with <span 
class="zptmcm7m-x-x-109">i </span>between <span 
class="zptmcm7t-x-x-109">1 </span>and <span 
class="zptmcm7m-x-x-109">N</span>, the total number of
atoms:
</p>
   <table 
class="equation"><tr><td><a 
 id="x1-2001r1"></a>
   <center class="math-display" >
<img 
src="colvars-refman-namd0x.png" alt="ξ(t)= ξ (xi(t),xj(t),xk(t),...),1≤ i,j,k...≤ N
" class="math-display"  /></center></td><td class="equation-label">(1)</td></tr></table>
<!--l. 24--><p class="nopar" >
This manual documents the collective variables module (<span 
class="ptmb7t-x-x-109">Colvars</span>), a portable software that interfaces multiple MD
simulation programs, with a focus on ﬂexibility, robustness and high performance. The module is designed
to perform multiple tasks concurrently during or after a simulation, the most common of which are:
</p>
     <ul class="itemize1">
     <li class="itemize">apply restraints or biasing potentials to multiple colvars, tailored on the system by choosing from a
     wide set of basis functions, without limitations on their number or on the number of atoms involved;
     while this can in principle be done through a TclForces script, using the Colvars module is both easier
     and computationally more eﬃcient;
     </li>
     <li class="itemize">calculate potentials of mean force (PMFs) along any set of colvars, using diﬀerent enhanced sampling
     methods, such as Adaptive Biasing Force (ABF), metadynamics, steered MD and umbrella sampling;
     variants of these methods that make use of an ensemble of replicas are supported as well;
     </li>
     <li class="itemize">calculate  statistical  properties  of  the  colvars,  such  as  running  averages  and  standard  deviations,
     correlation functions of pairs of colvars, and multidimensional histograms: this can be done either at
     run-time without the need to save very large trajectory ﬁles, or after a simulation has been completed
     using VMD and the <span 
class="ot1-zi4r-0-x-x-109">cv </span>command.
     </li></ul>
<!--l. 45--><p class="indent" >   Detailed explanations of the design of the Colvars module are provided in reference <span class="cite">[<a 
href="#XFiorin2013">1</a>]</span>. Please cite this
reference whenever publishing work that makes use of this module.
                                                                                         
                                                                                         
</p><!--l. 48--><p class="noindent" >
</p>
   <h3 class="sectionHead"><span class="titlemark">2    </span> <a 
 id="x1-30002"></a>A crash course</h3>
<!--l. 50--><p class="noindent" >Suppose that we want to run a steered MD experiment where a small molecule is pulled away from a binding site. In
Colvars terms, this is done by applying a moving restraint to the distance between the two objects. The conﬁguration
will contain two blocks, one deﬁning the distance variable (see <a 
href="#sec:colvar">4<!--tex4ht:ref: sec:colvar --></a>, <a 
href="#sec:cvc">6<!--tex4ht:ref: sec:cvc --></a>), and the other the moving harmonic restraint
(<a 
href="#sec:colvarbias_harmonic">7.4<!--tex4ht:ref: sec:colvarbias_harmonic --></a>).
</p><!--l. 56--><p class="noindent" ><span 
class="ot1-zi4r-0-x-x-109">colvar </span><span 
class="cmsy-10x-x-109">{</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> name dist</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> distance </span><span 
class="cmsy-10x-x-109">{</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> group1 </span><span 
class="cmsy-10x-x-109">{ </span><span 
class="ot1-zi4r-0-x-x-109">atomNumbersRange 42-55 </span><span 
class="cmsy-10x-x-109">}</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> group2 </span><span 
class="cmsy-10x-x-109">{</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> psfSegID PR</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> atomNameResidueRange CA 15-30 </span><span 
class="cmsy-10x-x-109">}</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">}</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">}</span><br 
class="newline" /><span 
class="cmsy-10x-x-109">}</span><br 
class="newline" /><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109">harmonic </span><span 
class="cmsy-10x-x-109">{</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> colvars dist</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> forceConstant 20.0</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> centers 4.</span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> # initial distance</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> targetCenters 15.</span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> # final distance</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> targetNumSteps 500000</span><br 
class="newline" /><span 
class="cmsy-10x-x-109">}</span><br 
class="newline" />
</p><!--l. 77--><p class="indent" >   Reading this input in plain English: the variable here named <span 
class="ptmri7t-x-x-109">dist </span>consists in a distance function between the
centers of two groups: the ligand (atoms 42 to 55) and the alpha carbon atoms (CA) of residues 15 to 30 in the
protein (segment name PR). The atom selection syntax is detailed in <a 
href="#sec:colvar_atom_groups">5<!--tex4ht:ref: sec:colvar_atom_groups --></a>.
</p><!--l. 80--><p class="indent" >   To the “<span 
class="ptmri7t-x-x-109">dist</span>” variable, we apply a <span 
class="ot1-zi4r-0-x-x-109">harmonic </span>potential of force constant 20 kcal/mol/Å<sup><span 
class="zptmcm7t-x-x-80">2</span></sup>, initially centered
around a value of 4 Å, which will increase to 15 Å over 500,000 simulation steps.
</p><!--l. 82--><p class="noindent" >
</p>
   <h3 class="sectionHead"><span class="titlemark">3    </span> <a 
 id="x1-40003"></a>General parameters and input/output ﬁles</h3>
<!--l. 85--><p class="noindent" >Here, we document the syntax of the commands and parameters used to set up and use the Colvars module in
NAMD. One of these parameters is the conﬁguration ﬁle or the conﬁguration text for the module itself, whose
syntax is described in <a 
href="#sec:colvars_config">3.2<!--tex4ht:ref: sec:colvars_config --></a> and in the following sections.
</p><!--l. 166--><p class="noindent" >
</p>
   <h4 class="subsectionHead"><span class="titlemark">3.1    </span> <a 
 id="x1-50003.1"></a>NAMD parameters</h4>
<!--l. 166--><p class="noindent" >To enable a Colvars-based calculation, two parameters must be added to the NAMD conﬁguration ﬁle, <span 
class="ot1-zi4r-0-x-x-109">colvars</span>
and <span 
class="ot1-zi4r-0-x-x-109">colvarsConfig</span>. An optional third parameter, <span 
class="ot1-zi4r-0-x-x-109">colvarsInput</span>, can be used to continue a previous
                                                                                         
                                                                                         
simulation.
</p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-5001"></a> <span 
class="ot1-zi4r-0-x-x-120">colvars </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Enable the Colvars module<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span>NAMD conﬁguration ﬁle <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">off</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>If this ﬂag is on, the Colvars module within NAMD is enabled; the module requires a
     separate conﬁguration ﬁle, to be provided with <span 
class="ot1-zi4r-0-x-x-109">colvarsConfig</span>.
     </li>
     <li class="itemize"><a 
 id="dx1-5002"></a> <span 
class="ot1-zi4r-0-x-x-120">colvarsConfig </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Conﬁguration ﬁle for the collective variables<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span>NAMD conﬁguration ﬁle <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>UNIX ﬁlename <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>This ﬁle contains the deﬁnition of all collective variables and their biasing or analysis
     methods.  This  ﬁle  can  also  be  provided  by  the  Tcl  command  <span 
class="ot1-zi4r-0-x-x-109">cv  configfile</span>;  alternatively,  the
     contents of the ﬁle itself can be given as an argument to the command <span 
class="ot1-zi4r-0-x-x-109">cv config</span>.
     </li>
     <li class="itemize"><a 
 id="dx1-5003"></a> <span 
class="ot1-zi4r-0-x-x-120">colvarsInput </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Input state ﬁle for the collective variables<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span>NAMD conﬁguration ﬁle <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>UNIX ﬁlename <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>When continuing a previous simulation run, this ﬁle contains the current state of all
     collective variables and of their associated algorithms. It is written automatically at the end of any
     simulation with collective variables. This ﬁle can also be provided by the Tcl command <span 
class="ot1-zi4r-0-x-x-109">cv load</span>.
     </li></ul>
<!--l. 231--><p class="noindent" >
</p>
   <h4 class="subsectionHead"><span class="titlemark">3.2    </span> <a 
 id="sec:colvars_config"></a>Conﬁguration syntax for the Colvars module</h4>
<!--l. 236--><p class="noindent" >All the parameters deﬁning the colvars and their biasing or analysis algorithms are read from the ﬁle speciﬁed by the
conﬁguration option <span 
class="ot1-zi4r-0-x-x-109">colvarsConfig</span>, or by the Tcl commands <span 
class="ot1-zi4r-0-x-x-109">cv config </span>and <span 
class="ot1-zi4r-0-x-x-109">cv configfile</span>. Hence, none of the
keywords described in this section and the following ones are available as keywords for the NAMD conﬁguration
ﬁle.  The syntax of the Colvars conﬁguration is “<span 
class="ot1-zi4r-0-x-x-109">keyword value</span>”, where the keyword and its value are separated by
any white space. The following rules apply:
</p>
     <ul class="itemize1">
     <li class="itemize">keywords are case-insensitive (<span 
class="ot1-zi4r-0-x-x-109">upperBoundary </span>is the same as <span 
class="ot1-zi4r-0-x-x-109">upperboundary </span>and <span 
class="ot1-zi4r-0-x-x-109">UPPERBOUNDARY</span>):
     their string values are however case-sensitive (e.g. ﬁle names);
     </li>
     <li class="itemize">a long value or a list of multiple values can be distributed across multiple lines by using curly braces,
     “<span 
class="cmsy-10x-x-109">{</span>” and “<span 
class="cmsy-10x-x-109">}</span>”: the opening brace “<span 
class="cmsy-10x-x-109">{</span>” must occur on the same line as the keyword, following a space
     character or other white space; the closing brace “<span 
class="cmsy-10x-x-109">}</span>” can be at any position after that;
                                                                                         
                                                                                         
     </li>
     <li class="itemize">many keywords are nested, and are only meaningful within a speciﬁc context: for every keyword
     documented in the following, the “parent” keyword that deﬁnes such context is also indicated;
     </li>
     <li class="itemize">the ‘<span 
class="ot1-zi4r-0-x-x-109">=</span>&#x2019; sign between a keyword and its value, deprecated in the NAMD main conﬁguration ﬁle, is not
     allowed;
     </li>
     <li class="itemize">Tcl  syntax  is  generally  not  available,  but  it  is  possible  to  use  Tcl  variables  or  bracket  expansion
     of  commands  within  a  conﬁguration  string,  when  this  is  passed  via  the  command  <span 
class="ot1-zi4r-0-x-x-109">cv  config</span>
     <span 
class="ot1-zi4r-0-x-x-109">…</span>;  this  is  particularly  useful  when  combined  with  parameter  introspection,  e.g.  <span 
class="ot1-zi4r-0-x-x-109">cv  config</span>
     <span 
class="ot1-zi4r-0-x-x-109">"colvarsTrajFrequency [DCDFreq]"</span>;
     </li>
     <li class="itemize">if  a  keyword  requiring  a  boolean  value  (<span 
class="ot1-zi4r-0-x-x-109">yes|on|true </span>or  <span 
class="ot1-zi4r-0-x-x-109">no|off|false</span>)  is  provided  without  an
     explicit value, it defaults to ‘<span 
class="ot1-zi4r-0-x-x-109">yes|on|true</span>&#x2019;; for example, ‘<span 
class="ot1-zi4r-0-x-x-109">outputAppliedForce</span>&#x2019; may be used as
     shorthand for ‘<span 
class="ot1-zi4r-0-x-x-109">outputAppliedForce on</span>&#x2019;;
     </li>
     <li class="itemize">the hash character <span 
class="ot1-zi4r-0-x-x-109"># </span>indicates a comment: all text in the same line following this character will be
     ignored.
     </li></ul>
<!--l. 264--><p class="indent" >   The following keywords are available in the global context of the colvars conﬁguration, i.e. they are not nested
inside other keywords: </p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-6001"></a> <span 
class="ot1-zi4r-0-x-x-120">colvarsTrajFrequency </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Colvar value trajectory frequency<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span>global <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive integer <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">100</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>The values of each colvar (and of other related quantities, if requested) are written to
     the ﬁle <span 
class="ptmri7t-x-x-109">outputName</span><span 
class="ot1-zi4r-0-x-x-109">.colvars.traj </span>every these many steps throughout the simulation. If the value is
     <span 
class="ot1-zi4r-0-x-x-109">0</span>, such trajectory ﬁle is not written. For optimization the output is buﬀered, and synchronized with
     the disk only when the restart ﬁle is being written.
     </li>
     <li class="itemize"><a 
 id="dx1-6002"></a> <span 
class="ot1-zi4r-0-x-x-120">colvarsTrajAppend </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Append to trajectory ﬁle?<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span>global <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">off</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>If this ﬂag is enabled, and a ﬁle with the same name as the trajectory ﬁle is already
     present, new data is appended to that ﬁle. Otherwise, a new ﬁle is created with the same name that
     overwrites the previous ﬁle. <span 
class="ptmb7t-x-x-109">Note: </span><span 
class="ptmri7t-x-x-109">when running consecutive simulations with the same </span>outputName
     <span 
class="ptmri7t-x-x-109">(e.g.</span><span 
class="ptmri7t-x-x-109"> in FEP calculations), you should enable this option to preserve the previous contents of the</span>
     <span 
class="ptmri7t-x-x-109">trajectory ﬁle.</span>
     </li>
     <li class="itemize"><a 
 id="dx1-6003"></a> <span 
class="ot1-zi4r-0-x-x-120">colvarsRestartFrequency </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Colvar module restart frequency<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span>global <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive integer <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">restartFreq</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>Allows to choose a diﬀerent restart frequency for the Colvars module. Redeﬁning it
     may be useful to trace the time evolution of those few properties which are not written to the trajectory
     ﬁle for reasons of disk space.
     </li>
     <li class="itemize"><a 
 id="dx1-6004"></a> <span 
class="ot1-zi4r-0-x-x-120">indexFile </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Index ﬁle for atom selection (GROMACS “ndx” format)<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span>global <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>UNIX ﬁlename <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>This option reads an index ﬁle (usually with a <span 
class="ot1-zi4r-0-x-x-109">.ndx </span>extension) as produced by the
     <span 
class="ot1-zi4r-0-x-x-109">make</span><span 
class="ot1-zi4r-0-x-x-109">_ndx </span>tool of GROMACS. This keyword may be repeated to load multiple index ﬁles: the same
     group name cannot appear in multiple index ﬁles.  The names of index groups contained in this ﬁle
     can then be used to deﬁne atom groups with the <span 
class="ot1-zi4r-0-x-x-109">indexGroup </span>keyword. Other supported methods to
     select atoms are described in <a 
href="#sec:colvar_atom_groups">5<!--tex4ht:ref: sec:colvar_atom_groups --></a>.
     </li>
     <li class="itemize"><a 
 id="dx1-6005"></a> <span 
class="ot1-zi4r-0-x-x-120">smp </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Whether SMP parallelism should be used<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span>global <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">on</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>If this ﬂag is enabled (default), SMP parallelism over threads will be used to compute
     variables and biases, provided that this is supported by NAMD.
     </li>
     <li class="itemize"><a 
 id="dx1-6006"></a> <span 
class="ot1-zi4r-0-x-x-120">analysis </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Turn on run-time statistical analysis<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span>global <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">off</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>If this ﬂag is enabled, each colvar is instructed to perform whatever run-time statistical
     analysis it is conﬁgured to, such as correlation functions, or running averages and standard deviations.
     See section <a 
href="#sec:colvar_acf">4.4<!--tex4ht:ref: sec:colvar_acf --></a> for details.
     </li></ul>
<!--l. 342--><p class="indent" >   To illustrate the ﬂexibility of the Colvars module, a non-trivial setup is represented in Figure <a 
href="#fig:colvars_diagram">1<!--tex4ht:ref: fig:colvars_diagram --></a>. The
corresponding conﬁguration is given below. The options within the <span 
class="ot1-zi4r-0-x-x-109">colvar </span>blocks are described in <a 
href="#sec:colvar">4<!--tex4ht:ref: sec:colvar --></a> and <a 
href="#sec:cvc">6<!--tex4ht:ref: sec:cvc --></a>, those
within the <span 
class="ot1-zi4r-0-x-x-109">harmonic </span>and <span 
class="ot1-zi4r-0-x-x-109">histogram </span>blocks in <a 
href="#sec:colvarbias">7<!--tex4ht:ref: sec:colvarbias --></a>. <span 
class="ptmb7t-x-x-109">Note: </span><span 
class="ptmri7t-x-x-109">except </span><span 
class="ot1-zi4r-0-x-x-109">colvar</span><span 
class="ptmri7t-x-x-109">, none of the keywords shown is</span>
<span 
class="ptmri7t-x-x-109">mandatory</span>.<br 
class="newline" />
</p>
   <hr class="figure" /><div class="figure" 
>
                                                                                         
                                                                                         
<a 
 id="fig:colvars_diagram"></a>
                                                                                         
                                                                                         
<!--l. 350--><p class="noindent" ><img 
src="colvars-refman-namd1x.png" alt="PIC" class="graphics" width="341.43306pt" height="263.83302pt"  /><!--tex4ht:graphics  
name="colvars-refman-namd1x.png" src="colvars_diagram.eps"  
-->
<br /> </p><div class="caption" 
><span class="id">Figure 1:  </span><span  
class="content">Graphical  representation  of  a  Colvars  conﬁguration.  The  colvar  called  “<span 
class="zptmcm7m-x-x-109">d</span>”  is  deﬁned  as  the
diﬀerence between two distances: the ﬁrst distance (<span 
class="zptmcm7m-x-x-109">d</span><sub><span 
class="zptmcm7t-x-x-80">1</span></sub>) is taken between the center of mass of atoms 1 and
2 and that of atoms 3 to 5, the second (<span 
class="zptmcm7m-x-x-109">d</span><sub><span 
class="zptmcm7t-x-x-80">2</span></sub>) between atom 7 and the center of mass of atoms 8 to 10. The
diﬀerence <span 
class="zptmcm7m-x-x-109">d </span><span 
class="zptmcm7t-x-x-109">= </span><span 
class="zptmcm7m-x-x-109">d</span><sub><span 
class="zptmcm7t-x-x-80">1</span></sub> <span 
class="zptmcm7y-x-x-109">−</span><span 
class="zptmcm7m-x-x-109">d</span><sub><span 
class="zptmcm7t-x-x-80">2</span></sub> is obtained by multiplying the two by a coeﬃcient <span 
class="zptmcm7m-x-x-109">C </span><span 
class="zptmcm7t-x-x-109">= +1 </span>or <span 
class="zptmcm7m-x-x-109">C </span><span 
class="zptmcm7t-x-x-109">= </span><span 
class="zptmcm7y-x-x-109">−</span><span 
class="zptmcm7t-x-x-109">1</span>, respectively.
The colvar called “<span 
class="zptmcm7m-x-x-109">c</span>” is the coordination number calculated between atoms 1 to 10 and atoms 11 to 20. A
harmonic restraint is applied to both <span 
class="zptmcm7m-x-x-109">d </span>and <span 
class="zptmcm7m-x-x-109">c</span>: to allow using the same force constant <span 
class="zptmcm7m-x-x-109">K</span>, both <span 
class="zptmcm7m-x-x-109">d </span>and <span 
class="zptmcm7m-x-x-109">c </span>are
scaled by their respective ﬂuctuation widths <span 
class="zptmcm7m-x-x-109">w</span><sub><span 
class="zptmcm7m-x-x-80">d</span></sub> and <span 
class="zptmcm7m-x-x-109">w</span><sub><span 
class="zptmcm7m-x-x-80">c</span></sub>. A third colvar “alpha” is deﬁned as the <span 
class="zptmcm7m-x-x-109">α</span>-helical
content of residues 1 to 10. The values of “<span 
class="zptmcm7m-x-x-109">c</span>” and “alpha” are also recorded throughout the simulation as a
joint 2-dimensional histogram. </span></div><!--tex4ht:label?: fig:colvars_diagram -->
                                                                                         
                                                                                         
   </div><hr class="endfigure" />
<!--l. 363--><p class="noindent" ><span 
class="ot1-zi4r-0-x-x-109">colvar </span><span 
class="cmsy-10x-x-109">{</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> # difference of two distances</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> name d </span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> width 0.2 # 0.2 </span><span 
class="ot1-zi4r-0-x-x-109">Å of estimated fluctuation width </span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> distance </span><span 
class="cmsy-10x-x-109">{</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> componentCoeff 1.0</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> group1 </span><span 
class="cmsy-10x-x-109">{ </span><span 
class="ot1-zi4r-0-x-x-109">atomNumbers 1 2 </span><span 
class="cmsy-10x-x-109">}</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> group2 </span><span 
class="cmsy-10x-x-109">{ </span><span 
class="ot1-zi4r-0-x-x-109">atomNumbers 3 4 5 </span><span 
class="cmsy-10x-x-109">}</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">}</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> distance </span><span 
class="cmsy-10x-x-109">{</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> componentCoeff -1.0</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> group1 </span><span 
class="cmsy-10x-x-109">{ </span><span 
class="ot1-zi4r-0-x-x-109">atomNumbers 7 </span><span 
class="cmsy-10x-x-109">}</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> group2 </span><span 
class="cmsy-10x-x-109">{ </span><span 
class="ot1-zi4r-0-x-x-109">atomNumbers 8 9 10 </span><span 
class="cmsy-10x-x-109">}</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">}</span><br 
class="newline" /><span 
class="cmsy-10x-x-109">}</span><br 
class="newline" /><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109">colvar </span><span 
class="cmsy-10x-x-109">{</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> name c</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> coordNum </span><span 
class="cmsy-10x-x-109">{</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> cutoff 6.0</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> group1 </span><span 
class="cmsy-10x-x-109">{ </span><span 
class="ot1-zi4r-0-x-x-109">atomNumbersRange 1-10 </span><span 
class="cmsy-10x-x-109">}</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> group2 </span><span 
class="cmsy-10x-x-109">{ </span><span 
class="ot1-zi4r-0-x-x-109">atomNumbersRange 11-20 </span><span 
class="cmsy-10x-x-109">}</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">}</span><br 
class="newline" /><span 
class="cmsy-10x-x-109">}</span><br 
class="newline" /> <br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109">colvar </span><span 
class="cmsy-10x-x-109">{</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> name alpha</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> alpha </span><span 
class="cmsy-10x-x-109">{</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> psfSegID PROT</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> residueRange 1-10</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">}</span><br 
class="newline" /><span 
class="cmsy-10x-x-109">}</span><br 
class="newline" /><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109">harmonic </span><span 
class="cmsy-10x-x-109">{</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> colvars d c</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> centers 3.0 4.0</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> forceConstant 5.0</span><br 
class="newline" /><span 
class="cmsy-10x-x-109">}</span><br 
class="newline" />
</p><!--l. 406--><p class="noindent" ><span 
class="ot1-zi4r-0-x-x-109">histogram </span><span 
class="cmsy-10x-x-109">{</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> colvars c alpha</span><br 
class="newline" /><span 
class="cmsy-10x-x-109">}</span><br 
class="newline" />
</p><!--l. 412--><p class="indent" >   Section <a 
href="#sec:colvar">4<!--tex4ht:ref: sec:colvar --></a> explains how to deﬁne a colvar and its behavior, regardless of its speciﬁc functional form. To deﬁne
colvars that are appropriate to a speciﬁc physical system, Section <a 
href="#sec:colvar_atom_groups">5<!--tex4ht:ref: sec:colvar_atom_groups --></a> documents how to select atoms, and section <a 
href="#sec:cvc">6<!--tex4ht:ref: sec:cvc --></a>
lists all of the available functional forms, which we call “colvar components”. Finally, section <a 
href="#sec:colvarbias">7<!--tex4ht:ref: sec:colvarbias --></a> lists the
available methods and algorithms to perform biased simulations and multidimensional analysis of
                                                                                         
                                                                                         
colvars.
</p>
   <h4 class="subsectionHead"><span class="titlemark">3.3    </span> <a 
 id="x1-70003.3"></a>Input state ﬁle (optional)</h4>
<!--l. 420--><p class="noindent" >Aside from the colvars conﬁguration, an optional input state ﬁle may be provided to load the relevant data from a
previous simulation. The name of this ﬁle is provided as a value to the keyword <span 
class="ot1-zi4r-0-x-x-109">colvarsInput</span>.
</p><!--l. 427--><p class="noindent" >
</p>
   <h4 class="subsectionHead"><span class="titlemark">3.4    </span> <a 
 id="x1-80003.4"></a>Output ﬁles</h4>
<!--l. 430--><p class="noindent" >During a simulation with collective variables deﬁned, the following three output ﬁles are written:
</p>
     <ul class="itemize1">
     <li class="itemize">a <span 
class="ptmri7t-x-x-109">state ﬁle</span>, named <span 
class="ptmri7t-x-x-109">outputName</span><span 
class="ot1-zi4r-0-x-x-109">.colvars.state</span>; this ﬁle is in ASCII format , regardless of the value
     of <span 
class="ot1-zi4r-0-x-x-109">binaryOutput </span>in the NAMD conﬁguration; to continue the simulation, the name of this ﬁle must
     be included in the conﬁguration of the next run using <span 
class="ot1-zi4r-0-x-x-109">colvarsInput</span>, together with the other NAMD
     output ﬁles;
     </li>
     <li class="itemize">if the NAMD parameter <span 
class="ot1-zi4r-0-x-x-109">restartFreq </span>or   the parameter <span 
class="ot1-zi4r-0-x-x-109">colvarsRestartFrequency </span>is larger than
     zero, a <span 
class="ptmri7t-x-x-109">restart ﬁle </span>named <span 
class="ptmri7t-x-x-109">restartName</span><span 
class="ot1-zi4r-0-x-x-109">.colvars.state </span>is written every that many steps: this ﬁle is
     equivalent to the ﬁnal state ﬁle;
     </li>
     <li class="itemize">if the parameter <span 
class="ot1-zi4r-0-x-x-109">colvarsTrajFrequency </span>is greater than 0 (default: 100), a <span 
class="ptmri7t-x-x-109">trajectory ﬁle </span>is written
     during the simulation: its name is <span 
class="ptmri7t-x-x-109">outputName</span><span 
class="ot1-zi4r-0-x-x-109">.colvars.traj</span>; unlike the state ﬁle, it is not needed to
     restart a simulation, but can be used later for post-processing and analysis.
     </li></ul>
<!--l. 443--><p class="indent" >   Other output ﬁles may be written by speciﬁc methods applied to the colvars (e.g. by the ABF method, see <a 
href="#sec:colvarbias_abf">7.1<!--tex4ht:ref: sec:colvarbias_abf --></a>,
or the metadynamics method, see <a 
href="#sec:colvarbias_meta">7.3<!--tex4ht:ref: sec:colvarbias_meta --></a>). Like the colvar trajectory ﬁle, they are needed only for analyzing, not
continuing a simulation. All such ﬁles&#x2019; names also begin with the preﬁx <span 
class="ptmri7t-x-x-109">outputName</span>.
</p><!--l. 447--><p class="indent" >   Finally, the total energy of all biases or restraints applied to the colvars appears under the NAMD standard
output, under the MISC column.
</p><!--l. 450--><p class="noindent" >
</p>
   <h3 class="sectionHead"><span class="titlemark">4    </span> <a 
 id="sec:colvar"></a>Deﬁning collective variables and their properties</h3>
<!--l. 453--><p class="noindent" >In the conﬁguration ﬁle each colvar is deﬁned by the keyword <span 
class="ot1-zi4r-0-x-x-109">colvar</span>, followed by its conﬁguration options within
curly braces: <span 
class="ot1-zi4r-0-x-x-109">colvar</span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109"> ...</span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">}</span>. One of these options is the name of a colvar component: for example, including
<span 
class="ot1-zi4r-0-x-x-109">rmsd </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109"> ...</span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">} </span>deﬁnes the colvar as a RMSD function. <span 
class="ptmri7t-x-x-109">In most applications, only one component is used, and the</span>
<span 
class="ptmri7t-x-x-109">component is equal to the colvar.</span>
                                                                                         
                                                                                         
</p><!--l. 456--><p class="indent" >   The full list of colvar components can be found in Section <a 
href="#sec:cvc">6<!--tex4ht:ref: sec:cvc --></a>, with the syntax to select atoms in Section <a 
href="#sec:colvar_atom_groups">5<!--tex4ht:ref: sec:colvar_atom_groups --></a>.
The following section lists several options to control the behavior of a single colvar, regardless of its
type.
</p><!--l. 459--><p class="noindent" >
</p>
   <h4 class="subsectionHead"><span class="titlemark">4.1    </span> <a 
 id="sec:colvar_general"></a>General options for a collective variable</h4>
<!--l. 461--><p class="noindent" >The following options are not required by default; however, the ﬁrst four are very frequently used:
</p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-10001"></a> <span 
class="ot1-zi4r-0-x-x-120">name </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Name of this colvar<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">colvar</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>string <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span>“<span 
class="ot1-zi4r-0-x-x-109">colvar</span>” + numeric id <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>The name is an unique case-sensitive string which allows the Colvars module to identify
     this colvar unambiguously; it is also used in the trajectory ﬁle to label to the columns corresponding
     to this colvar.
     </li>
     <li class="itemize"><a 
 id="dx1-10002"></a> <span 
class="ot1-zi4r-0-x-x-120">width </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Colvar ﬂuctuation scale, or resolution for grid-based methods<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">colvar</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive decimal <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span>1.0 <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>This number has the same physical unit as the colvar value and deﬁnes an eﬀective
     colvar unit. Biasing algorithms use it for diﬀerent purposes. Harmonic restraints (<a 
href="#sec:colvarbias_harmonic">7.4<!--tex4ht:ref: sec:colvarbias_harmonic --></a>) use it to set
     the  physical  unit  of  the  force  constant,  which  is  useful  for  multidimensional  restraints  involving
     colvars with diﬀerent units or scale which may then be deﬁned by a single, scaled force constant.
     Histogram (<a 
href="#sec:colvarbias_histogram">7.8<!--tex4ht:ref: sec:colvarbias_histogram --></a>) and ABF biases (<a 
href="#sec:colvarbias_abf">7.1<!--tex4ht:ref: sec:colvarbias_abf --></a>) interpret it as the grid spacing in the direction of this variable.
     Metadynamics (<a 
href="#sec:colvarbias_meta">7.3<!--tex4ht:ref: sec:colvarbias_meta --></a>) uses it to set the width of newly added hills. In other cases, it is simplest to keep
     the default value of 1, so that harmonic force constants are provided in their usual physical unit. When
     a non-unity width is required by the application, the optimal value is application-dependent, but can
     often be thought of as a user-provided estimate of the ﬂuctuation amplitude for the colvar. In those
     cases, it should generally be set smaller than or equal to the standard deviation of the colvar during a
     very short simulation run.
     </li>
     <li class="itemize"><a 
 id="dx1-10003"></a> <span 
class="ot1-zi4r-0-x-x-120">lowerBoundary </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Lower boundary of the colvar<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">colvar</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>decimal <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>Deﬁnes the lowest end of the interval of “relevant” values for the colvar. This number
     can  be  either  a  true  physical  boundary,  or  a  user-deﬁned  number.  Together  with  <span 
class="ot1-zi4r-0-x-x-109">upperBoundary</span>
     and <span 
class="ot1-zi4r-0-x-x-109">width</span>, it is used to deﬁne a grid of values along the colvar (not available for colvars based on
     <span 
class="ot1-zi4r-0-x-x-109">distanceDir</span>, <span 
class="ot1-zi4r-0-x-x-109">distanceVec</span>, and <span 
class="ot1-zi4r-0-x-x-109">orientation</span>). This option does not aﬀect dynamics: to conﬁne a
     colvar within a certain interval, use a <span 
class="ot1-zi4r-0-x-x-109">harmonicWalls </span>bias.
     </li>
     <li class="itemize"><a 
 id="dx1-10004"></a> <span 
class="ot1-zi4r-0-x-x-120">upperBoundary </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Upper boundary of the colvar<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">colvar</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>decimal <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>Similarly to <span 
class="ot1-zi4r-0-x-x-109">lowerBoundary</span>, deﬁnes the highest possible or allowed value.
     </li>
     <li class="itemize"><a 
 id="dx1-10005"></a> <span 
class="ot1-zi4r-0-x-x-120">hardLowerBoundary </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Whether the lower boundary is the physical lower limit<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">colvar</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">off</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>This option does not aﬀect simulation results, but enables some internal optimizations.
     Depending on its mathematical deﬁnition, a colvar may have “natural” boundaries: for example, a
     <span 
class="ot1-zi4r-0-x-x-109">distance </span>colvar has a “natural” lower boundary at 0. Setting this option instructs the Colvars module
     that the user-deﬁned lower boundary is “natural”. See Section <a 
href="#sec:cvc">6<!--tex4ht:ref: sec:cvc --></a> for the physical ranges of values of
     each component.
     </li>
     <li class="itemize"><a 
 id="dx1-10006"></a> <span 
class="ot1-zi4r-0-x-x-120">hardUpperBoundary  </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Whether the upper boundary is the physical upper limit of the colvar&#x2019;s
     values<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">colvar</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">off</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>Analogous to <span 
class="ot1-zi4r-0-x-x-109">hardLowerBoundary</span>.
     </li>
     <li class="itemize"><a 
 id="dx1-10007"></a> <span 
class="ot1-zi4r-0-x-x-120">expandBoundaries </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Allow to expand the two boundaries if needed<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">colvar</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">off</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>If deﬁned, biasing and analysis methods may keep their own copies of <span 
class="ot1-zi4r-0-x-x-109">lowerBoundary</span>
     and <span 
class="ot1-zi4r-0-x-x-109">upperBoundary</span>, and expand them to accommodate values that do not ﬁt in the initial range.
     Currently, this option is used by the metadynamics bias (<a 
href="#sec:colvarbias_meta">7.3<!--tex4ht:ref: sec:colvarbias_meta --></a>) to keep all of its hills fully within the
     grid. This option cannot be used when the initial boundaries already span the full period of a periodic
     colvar.
     </li>
     <li class="itemize"><a 
 id="dx1-10008"></a> <span 
class="ot1-zi4r-0-x-x-120">subtractAppliedForce </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Do not include biasing forces in the total force for this colvar<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">colvar</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">off</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>If  the  colvar  supports  total  force  calculation  (see  <a 
href="#sec:cvc_sys_forces">6.3.3<!--tex4ht:ref: sec:cvc_sys_forces --></a>),  all  forces  applied  to  this
     colvar by biases will be removed from the total force. This keyword allows to recover some of the
     “system force” calculation available in the Colvars module before version 2016-08-10. Please note
     that removal of <span 
class="ptmri7t-x-x-109">all </span>other external forces (including biasing forces applied to a diﬀerent colvar) is
     <span 
class="ptmri7t-x-x-109">no longer supported</span>, due to changes in the underlying simulation engines (primarily NAMD). This
     option may be useful when continuing a previous simulation where the removal of external/applied
     forces is essential. <span 
class="ptmri7t-x-x-109">For all new simulations, the use of this option is not recommended.</span>
     </li></ul>
                                                                                         
                                                                                         
<!--l. 565--><p class="noindent" >
</p>
   <h4 class="subsectionHead"><span class="titlemark">4.2    </span> <a 
 id="x1-110004.2"></a>Trajectory output</h4>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-11001"></a> <span 
class="ot1-zi4r-0-x-x-120">outputValue </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Output a trajectory for this colvar<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">colvar</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">on</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:    </span>If  <span 
class="ot1-zi4r-0-x-x-109">colvarsTrajFrequency </span>is  non-zero,  the  value  of  this  colvar  is  written  to  the
     trajectory ﬁle every <span 
class="ot1-zi4r-0-x-x-109">colvarsTrajFrequency </span>steps in the column labeled “<span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ot1-zi4r-0-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span>”.
     </li>
     <li class="itemize"><a 
 id="dx1-11002"></a> <span 
class="ot1-zi4r-0-x-x-120">outputVelocity </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Output a velocity trajectory for this colvar<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">colvar</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">off</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>If <span 
class="ot1-zi4r-0-x-x-109">colvarsTrajFrequency </span>is deﬁned, the ﬁnite-diﬀerence calculated velocity of this
     colvar are written to the trajectory ﬁle under the label “<span 
class="ot1-zi4r-0-x-x-109">v</span><span 
class="ot1-zi4r-0-x-x-109">_</span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ot1-zi4r-0-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span>”.
     </li>
     <li class="itemize"><a 
 id="dx1-11003"></a> <span 
class="ot1-zi4r-0-x-x-120">outputEnergy </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Output an energy trajectory for this colvar<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">colvar</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">off</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>This option applies only to extended Lagrangian colvars. If <span 
class="ot1-zi4r-0-x-x-109">colvarsTrajFrequency</span>
     is  deﬁned,  the  kinetic  energy  of  the  extended  degree  and  freedom  and  the  potential  energy  of
     the  restraining  spring  are  are  written  to  the  trajectory  ﬁle  under  the  labels  “<span 
class="ot1-zi4r-0-x-x-109">Ek</span><span 
class="ot1-zi4r-0-x-x-109">_</span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ot1-zi4r-0-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span>”  and
     “<span 
class="ot1-zi4r-0-x-x-109">Ep</span><span 
class="ot1-zi4r-0-x-x-109">_</span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ot1-zi4r-0-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span>”.
     </li>
     <li class="itemize"><a 
 id="dx1-11004"></a> <span 
class="ot1-zi4r-0-x-x-120">outputTotalForce </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Output a total force trajectory for this colvar<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">colvar</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">off</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>If <span 
class="ot1-zi4r-0-x-x-109">colvarsTrajFrequency </span>is deﬁned, the total force on this colvar (i.e. the projection
     of  all  atomic  total  forces  onto  this  colvar  —  see  equation (<a 
href="#eq:gradient_vector">19<!--tex4ht:ref: eq:gradient_vector --></a>)  in  section <a 
href="#sec:colvarbias_abf">7.1<!--tex4ht:ref: sec:colvarbias_abf --></a>)  are  written  to  the
     trajectory  ﬁle  under  the  label  “<span 
class="ot1-zi4r-0-x-x-109">fs</span><span 
class="ot1-zi4r-0-x-x-109">_</span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ot1-zi4r-0-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span>”.  For  extended  Lagrangian  colvars,  the  “total  force”
     felt by the extended degree of freedom is simply the force from the harmonic spring. <span 
class="ptmb7t-x-x-109">Note: </span>not all
     components support this option. The physical unit for this force is kcal/mol, divided by the colvar unit
     U.
     </li>
     <li class="itemize"><a 
 id="dx1-11005"></a> <span 
class="ot1-zi4r-0-x-x-120">outputAppliedForce </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Output an applied force trajectory for this colvar<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">colvar</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">off</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>If <span 
class="ot1-zi4r-0-x-x-109">colvarsTrajFrequency </span>is deﬁned, the total force applied on this colvar by Colvars
     biases are written to the trajectory under the label “<span 
class="ot1-zi4r-0-x-x-109">fa</span><span 
class="ot1-zi4r-0-x-x-109">_</span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ot1-zi4r-0-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span>”. For extended Lagrangian colvars,
                                                                                         
                                                                                         
     this force is actually applied to the extended degree of freedom rather than the geometric colvar itself.
     The physical unit for this force is kcal/mol divided by the colvar unit.
     </li></ul>
<!--l. 645--><p class="noindent" >
</p>
   <h4 class="subsectionHead"><span class="titlemark">4.3    </span> <a 
 id="sec:colvar_extended"></a>Extended Lagrangian.</h4>
<!--l. 648--><p class="noindent" >The following options enable extended-system dynamics, where a colvar is coupled to an additional degree of
freedom (ﬁctitious particle) by a harmonic spring. All biasing and conﬁning forces are then applied to the extended
degree of freedom. The “actual” geometric colvar (function of Cartesian coordinates) only feels the force from the
harmonic spring. This is particularly useful when combined with an ABF bias (<a 
href="#sec:colvarbias_abf">7.1<!--tex4ht:ref: sec:colvarbias_abf --></a>) to perform eABF simulations
(<a 
href="#sec:eABF">7.2<!--tex4ht:ref: sec:eABF --></a>).
</p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-12001"></a> <span 
class="ot1-zi4r-0-x-x-120">extendedLagrangian </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Add extended degree of freedom<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">colvar</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">off</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:    </span>Adds  a  ﬁctitious  particle  to  be  coupled  to  the  colvar  by  a  harmonic  spring.  The
     ﬁctitious  mass  and  the  force  constant  of  the  coupling  potential  are  derived  from  the  parameters
     <span 
class="ot1-zi4r-0-x-x-109">extendedTimeConstant </span>and <span 
class="ot1-zi4r-0-x-x-109">extendedFluctuation</span>, described below. Biasing forces on the colvar
     are applied to this ﬁctitious particle, rather than to the atoms directly. This implements the extended
     Lagrangian formalism used in some metadynamics simulations <span class="cite">[<a 
href="#XIannuzzi2003">2</a>]</span>. The energy associated with the
     extended degree of freedom is reported under the MISC title in NAMD&#x2019;s energy output.
     </li>
     <li class="itemize"><a 
 id="dx1-12002"></a> <span 
class="ot1-zi4r-0-x-x-120">extendedFluctuation </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Standard deviation between the colvar and the ﬁctitious particle (colvar
     unit)<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">colvar</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive decimal <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:    </span>Deﬁnes  the  spring  stiﬀness  for  the  <span 
class="ot1-zi4r-0-x-x-109">extendedLagrangian </span>mode,  by  setting  the
     typical   deviation   between   the   colvar   and   the   extended   degree   of   freedom   due   to   thermal
     ﬂuctuation. The spring force constant is calculated internally as <span 
class="zptmcm7m-x-x-109">k</span><sub><span 
class="zptmcm7m-x-x-80">B</span></sub><span 
class="zptmcm7m-x-x-109">T ∕σ</span><sup><span 
class="zptmcm7t-x-x-80">2</span></sup>, where <span 
class="zptmcm7m-x-x-109">σ </span>is the value of
     <span 
class="ot1-zi4r-0-x-x-109">extendedFluctuation</span>.
     </li>
     <li class="itemize"><a 
 id="dx1-12003"></a> <span 
class="ot1-zi4r-0-x-x-120">extendedTimeConstant </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Oscillation period of the ﬁctitious particle (fs)<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">colvar</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive decimal <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">200</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>Deﬁnes the inertial mass of the ﬁctitious particle, by setting the oscillation period of the
     harmonic oscillator formed by the ﬁctitious particle and the spring. The period should be much larger
     than the MD time step to ensure accurate integration of the extended particle&#x2019;s equation of motion.
     The ﬁctitious mass is calculated internally as <span 
class="zptmcm7m-x-x-109">k</span><sub><span 
class="zptmcm7m-x-x-80">B</span></sub><span 
class="zptmcm7m-x-x-109">T </span><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">τ∕</span><span 
class="zptmcm7t-x-x-109">2</span><span 
class="zptmcm7m-x-x-109">πσ</span><span 
class="zptmcm7t-x-x-109">)</span><sup><span 
class="zptmcm7t-x-x-80">2</span></sup>, where <span 
class="zptmcm7m-x-x-109">τ </span>is the period and <span 
class="zptmcm7m-x-x-109">σ </span>is the typical
     ﬂuctuation (see above).
                                                                                         
                                                                                         
     </li>
     <li class="itemize"><a 
 id="dx1-12004"></a> <span 
class="ot1-zi4r-0-x-x-120">extendedTemp </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Temperature for the extended degree of freedom (K)<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">colvar</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive decimal <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span>thermostat temperature <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>Temperature used for calculating the coupling force constant of the extended variable
     (see <span 
class="ot1-zi4r-0-x-x-109">extendedFluctuation</span>) and, if needed, as a target temperature for extended Langevin dynamics
     (see <span 
class="ot1-zi4r-0-x-x-109">extendedLangevinDamping</span>). This should normally be left at its default value.
     </li>
     <li class="itemize"><a 
 id="dx1-12005"></a> <span 
class="ot1-zi4r-0-x-x-120">extendedLangevinDamping </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Damping factor for extended Langevin dynamics (ps<sup><span 
class="zptmcm7y-x-x-80">−</span><span 
class="zptmcm7t-x-x-80">1</span></sup>)<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">colvar</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive decimal <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">1.0</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>If  this  is  non-zero,  the  extended  degree  of  freedom  undergoes  Langevin  dynamics
     at  temperature  <span 
class="ot1-zi4r-0-x-x-109">extendedTemp</span>.  The  friction  force  is  minus  <span 
class="ot1-zi4r-0-x-x-109">extendedLangevinDamping </span>times  the
     velocity.  This  is  useful  because  the  extended  dynamics  coordinate  may  heat  up  in  the  transient
     non-equilibrium regime of ABF. Use moderate damping values, to limit viscous friction (potentially
     slowing  down  diﬀusive  sampling)  and  stochastic  noise  (increasing  the  variance  of  statistical
     measurements). In doubt, use the default value. </li></ul>
<!--l. 736--><p class="noindent" >
</p>
   <h4 class="subsectionHead"><span class="titlemark">4.4    </span> <a 
 id="sec:colvar_acf"></a>Statistical analysis of collective variables</h4>
<!--l. 739--><p class="noindent" >When the global keyword <span 
class="ot1-zi4r-0-x-x-109">analysis </span>is deﬁned in the conﬁguration ﬁle, run-time calculations of statistical properties
for individual colvars can be performed. At the moment, several types of time correlation functions, running
averages and running standard deviations are available.
</p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-13001"></a> <span 
class="ot1-zi4r-0-x-x-120">corrFunc </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Calculate a time correlation function?<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">colvar</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">off</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>Whether or not a time correlaction function should be calculated for this colvar.
     </li>
     <li class="itemize"><a 
 id="dx1-13002"></a> <span 
class="ot1-zi4r-0-x-x-120">corrFuncWithColvar </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Colvar name for the correlation function<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">colvar</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>string <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>By default, the auto-correlation function (ACF) of this colvar, <span 
class="zptmcm7m-x-x-109">ξ</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub>, is calculated. When
     this option is speciﬁed, the correlation function is calculated instead with another colvar, <span 
class="zptmcm7m-x-x-109">ξ</span><sub><span 
class="zptmcm7m-x-x-80">j</span></sub>, which
     must be of the same type (scalar, vector, or quaternion) as <span 
class="zptmcm7m-x-x-109">ξ</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub>.
     </li>
     <li class="itemize"><a 
 id="dx1-13003"></a> <span 
class="ot1-zi4r-0-x-x-120">corrFuncType </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Type of the correlation function<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">colvar</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span><span 
class="ot1-zi4r-0-x-x-109">velocity</span>, <span 
class="ot1-zi4r-0-x-x-109">coordinate </span>or <span 
class="ot1-zi4r-0-x-x-109">coordinate</span><span 
class="ot1-zi4r-0-x-x-109">_p2</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">velocity</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:         </span>With     <span 
class="ot1-zi4r-0-x-x-109">coordinate  </span>or     <span 
class="ot1-zi4r-0-x-x-109">velocity</span>,     the     correlation     function     <span 
class="zptmcm7m-x-x-109">C</span><sub><span 
class="zptmcm7m-x-x-80">i,j</span></sub><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">t</span><span 
class="zptmcm7t-x-x-109">)</span> =
     <img 
src="colvars-refman-namd2x.png" alt="⟨Π (ξ (t),ξ(t + t))⟩
    i 0   j 0"  class="left" align="middle" /> is calculated between the variables <span 
class="zptmcm7m-x-x-109">ξ</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub> and <span 
class="zptmcm7m-x-x-109">ξ</span><sub><span 
class="zptmcm7m-x-x-80">j</span></sub>, or their velocities. <span 
class="zptmcm7t-x-x-109">Π(</span><span 
class="zptmcm7m-x-x-109">ξ</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub><span 
class="zptmcm7m-x-x-109">,ξ</span><sub><span 
class="zptmcm7m-x-x-80">j</span></sub><span 
class="zptmcm7t-x-x-109">) </span>is
     the scalar product when calculated between scalar or vector values, whereas for quaternions it is the
     cosine between the two corresponding rotation axes. With <span 
class="ot1-zi4r-0-x-x-109">coordinate</span><span 
class="ot1-zi4r-0-x-x-109">_p2</span>, the second order Legendre
     polynomial, <span 
class="zptmcm7t-x-x-109">(3</span><span 
class="zptmcm7t-x-x-109">cos</span><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">𝜃</span><span 
class="zptmcm7t-x-x-109">)</span><sup><span 
class="zptmcm7t-x-x-80">2</span></sup> <span 
class="zptmcm7y-x-x-109">−</span><span 
class="zptmcm7t-x-x-109">1)</span><span 
class="zptmcm7m-x-x-109">∕</span><span 
class="zptmcm7t-x-x-109">2</span>, is used instead of the cosine.
     </li>
     <li class="itemize"><a 
 id="dx1-13004"></a> <span 
class="ot1-zi4r-0-x-x-120">corrFuncNormalize </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Normalize the time correlation function?<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">colvar</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">on</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>If enabled, the value of the correlation function at <span 
class="zptmcm7m-x-x-109">t</span> = 0 is normalized to 1; otherwise,
     it equals to <img 
src="colvars-refman-namd3x.png" alt="⟨        ⟩
 O (ξi,ξj)"  class="left" align="middle" />.
     </li>
     <li class="itemize"><a 
 id="dx1-13005"></a> <span 
class="ot1-zi4r-0-x-x-120">corrFuncLength </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Length of the time correlation function<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">colvar</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive integer <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">1000</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>Length (in number of points) of the time correlation function.
     </li>
     <li class="itemize"><a 
 id="dx1-13006"></a> <span 
class="ot1-zi4r-0-x-x-120">corrFuncStride </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Stride of the time correlation function<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">colvar</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive integer <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">1</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>Number of steps between two values of the time correlation function.
     </li>
     <li class="itemize"><a 
 id="dx1-13007"></a> <span 
class="ot1-zi4r-0-x-x-120">corrFuncOffset </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Oﬀset of the time correlation function<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">colvar</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive integer <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">0</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>The starting time (in number of steps) of the time correlation function (default: <span 
class="zptmcm7m-x-x-109">t</span> = 0).
     <span 
class="ptmb7t-x-x-109">Note: </span><span 
class="ptmri7t-x-x-109">the value at </span><span 
class="zptmcm7m-x-x-109">t</span><span 
class="ptmri7t-x-x-109"> = 0 is always used for the normalization</span>.
     </li>
     <li class="itemize"><a 
 id="dx1-13008"></a> <span 
class="ot1-zi4r-0-x-x-120">corrFuncOutputFile </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Output ﬁle for the time correlation function<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">colvar</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>UNIX ﬁlename <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ot1-zi4r-0-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span><span 
class="ot1-zi4r-0-x-x-109">.corrfunc.dat</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>The time correlation function is saved in this ﬁle.
     </li>
     <li class="itemize"><a 
 id="dx1-13009"></a> <span 
class="ot1-zi4r-0-x-x-120">runAve </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Calculate the running average and standard deviation<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">colvar</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">off</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>Whether or not the running average and standard deviation should be calculated for this
     colvar.
     </li>
     <li class="itemize"><a 
 id="dx1-13010"></a> <span 
class="ot1-zi4r-0-x-x-120">runAveLength </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Length of the running average window<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">colvar</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive integer <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">1000</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>Length (in number of points) of the running average window.
     </li>
     <li class="itemize"><a 
 id="dx1-13011"></a> <span 
class="ot1-zi4r-0-x-x-120">runAveStride </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Stride of the running average window values<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">colvar</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive integer <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">1</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>Number of steps between two values within the running average window.
     </li>
     <li class="itemize"><a 
 id="dx1-13012"></a> <span 
class="ot1-zi4r-0-x-x-120">runAveOutputFile </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Output ﬁle for the running average and standard deviation<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">colvar</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>UNIX ﬁlename <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ot1-zi4r-0-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span><span 
class="ot1-zi4r-0-x-x-109">.runave.dat</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>The running average and standard deviation are saved in this ﬁle.
     </li></ul>
<!--l. 876--><p class="noindent" >
</p>
   <h3 class="sectionHead"><span class="titlemark">5    </span> <a 
 id="sec:colvar_atom_groups"></a>Selecting atoms for colvars: deﬁning atom groups</h3>
<!--l. 879--><p class="noindent" >To deﬁne collective variables, atoms are usually selected as groups. Each group is deﬁned using an identiﬁer that is
unique in the context of the speciﬁc colvar component (e.g. for a distance component, the two groups are <span 
class="ot1-zi4r-0-x-x-109">group1</span>
and <span 
class="ot1-zi4r-0-x-x-109">group2</span>). The identiﬁer is followed by a brace-delimited block containing selection keywords and other
parameters, including an optional <span 
class="ot1-zi4r-0-x-x-109">name</span>:
</p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-14001"></a> <span 
class="ot1-zi4r-0-x-x-120">name </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Unique name for the atom group<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span>atom group <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>string <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>This parameter deﬁnes a unique name for this atom group, which can be referred to
     in the deﬁnition of other atom groups (including in other colvars) by invoking <span 
class="ot1-zi4r-0-x-x-109">atomsOfGroup </span>as a
     selection keyword.</li></ul>
                                                                                         
                                                                                         
<!--l. 894--><p class="noindent" >
</p>
   <h4 class="subsectionHead"><span class="titlemark">5.1    </span> <a 
 id="x1-150005.1"></a>Selection keywords</h4>
<!--l. 897--><p class="noindent" >Selection keywords may be used individually or in combination with each other, and each can be repeated any
number of times. Selection is incremental: each keyword adds the corresponding atoms to the selection, so that
diﬀerent sets of atoms can be combined. However, atoms included by multiple keywords are only counted once.
Below is an example conﬁguration for an atom group with identiﬁer “<span 
class="ot1-zi4r-0-x-x-109">atoms</span>”, which uses an unusually varied
combination of selection keywords:<br 
class="newline" />
</p><!--l. 904--><p class="noindent" ><span 
class="ot1-zi4r-0-x-x-109">atoms </span><span 
class="cmsy-10x-x-109">{</span><br 
class="newline" /><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> # add atoms 1 and 3 to this group (note: the first atom in the system is 1)</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> atomNumbers </span><span 
class="cmsy-10x-x-109">{ </span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> 1 3</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">}</span><br 
class="newline" /><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> # add atoms starting from 20 up to and including 50</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> atomNumbersRange 20-50</span><br 
class="newline" /><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> # add index group (requires a .ndx file to be provided globally)</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> indexGroup Water</span><br 
class="newline" /> <br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> # add all the atoms with occupancy 2 in the file atoms.pdb</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> atomsFile atoms.pdb</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> atomsCol O</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> atomsColValue 2.0</span><br 
class="newline" /><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> # add all the C-alphas within residues 11 to 20 of segments "PR1" and "PR2"</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> psfSegID PR1 PR2</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> atomNameResidueRange CA 11-20</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> atomNameResidueRange CA 11-20</span><br 
class="newline" /> <span 
class="cmsy-10x-x-109">}</span><br 
class="newline" />
</p><!--l. 931--><p class="indent" >   The resulting selection includes atoms 1 and 3, those between 20 and 50, and those in the index group called
“Water”; the indices of this group are read from the ﬁle provided by <span 
class="ot1-zi4r-0-x-x-109">indexFile</span>, in the global section of the
conﬁguration ﬁle.
</p><!--l. 934--><p class="indent" >   The complete list of selection keywords available in NAMD is:
</p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-15001"></a> <span 
class="ot1-zi4r-0-x-x-120">atomNumbers </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>List of atom numbers<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span>atom group <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>space-separated list of positive integers <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>This option adds to the group all the atoms whose numbers are in the list. <span 
class="ptmri7t-x-x-109">The number</span>
     <span 
class="ptmri7t-x-x-109">of the ﬁrst atom in the system is 1: to convert from a VMD selection, use “atomselect get serial”.</span>
     </li>
     <li class="itemize"><a 
 id="dx1-15002"></a> <span 
class="ot1-zi4r-0-x-x-120">indexGroup </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Name of index group to be used (GROMACS format)<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span>atom group <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>string <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>If the name of an index ﬁle has been provided by <span 
class="ot1-zi4r-0-x-x-109">indexFile</span>, this option allows to
     select one index group from that ﬁle: the atoms from that index group will be used to deﬁne the current
     group.
     </li>
     <li class="itemize"><a 
 id="dx1-15003"></a> <span 
class="ot1-zi4r-0-x-x-120">atomsOfGroup </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Name of group deﬁned previously<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span>atom group <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>string <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>Refers to a group deﬁned previously using its user-deﬁned <span 
class="ot1-zi4r-0-x-x-109">name</span>. This adds all atoms of
     that named group to the current group.
     </li>
     <li class="itemize"><a 
 id="dx1-15004"></a> <span 
class="ot1-zi4r-0-x-x-120">atomNumbersRange </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Atoms within a number range<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span>atom group <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span><span 
class="zptmcm7m-x-x-109">&#x003C;</span>Starting number<span 
class="zptmcm7m-x-x-109">&#x003E;</span>-<span 
class="zptmcm7m-x-x-109">&#x003C;</span>Ending number<span 
class="zptmcm7m-x-x-109">&#x003E; </span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>This  option  includes  in  the  group  all  atoms  whose  numbers  are  within  the  range
     speciﬁed. <span 
class="ptmri7t-x-x-109">The number of the ﬁrst atom in the system is 1.</span>
     </li>
     <li class="itemize"><a 
 id="dx1-15005"></a> <span 
class="ot1-zi4r-0-x-x-120">atomNameResidueRange </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Named atoms within a range of residue numbers<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span>atom group <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span><span 
class="zptmcm7m-x-x-109">&#x003C;</span>Atom name<span 
class="zptmcm7m-x-x-109">&#x003E; &#x003C;</span>Starting residue<span 
class="zptmcm7m-x-x-109">&#x003E;</span>-<span 
class="zptmcm7m-x-x-109">&#x003C;</span>Ending residue<span 
class="zptmcm7m-x-x-109">&#x003E; </span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>This option adds to the group all the atoms with the provided name, within residues in
     the given range.
     </li>
     <li class="itemize"><a 
 id="dx1-15006"></a> <span 
class="ot1-zi4r-0-x-x-120">psfSegID </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>PSF segment identiﬁer<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span>atom group <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>space-separated list of strings (max 4 characters) <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>This  option  sets  the  PSF  segment  identiﬁer  for  <span 
class="ot1-zi4r-0-x-x-109">atomNameResidueRange</span>.  Multiple
     values may be provided, which correspond to multiple instances of <span 
class="ot1-zi4r-0-x-x-109">atomNameResidueRange</span>, in the
     order of their occurrence. This option is only necessary if a PSF topology ﬁle is used.
     </li>
     <li class="itemize"><a 
 id="dx1-15007"></a> <span 
class="ot1-zi4r-0-x-x-120">atomsFile </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>PDB ﬁle name for atom selection<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span>atom group <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>UNIX ﬁlename <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>This  option  selects  atoms  from  the  PDB  ﬁle  provided  and  adds  them  to  the  group
     according to numerical ﬂags in the column <span 
class="ot1-zi4r-0-x-x-109">atomsCol</span>. <span 
class="ptmb7t-x-x-109">Note: </span><span 
class="ptmri7t-x-x-109">the sequence of atoms in the PDB ﬁle</span>
     <span 
class="ptmri7t-x-x-109">provided must match that in the system&#x2019;s topology</span>.
     </li>
     <li class="itemize"><a 
 id="dx1-15008"></a> <span 
class="ot1-zi4r-0-x-x-120">atomsCol </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>PDB column to use for atom selection ﬂags<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span>atom group <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span><span 
class="ot1-zi4r-0-x-x-109">O</span>, <span 
class="ot1-zi4r-0-x-x-109">B</span>, <span 
class="ot1-zi4r-0-x-x-109">X</span>, <span 
class="ot1-zi4r-0-x-x-109">Y</span>, or <span 
class="ot1-zi4r-0-x-x-109">Z</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>This option speciﬁes which PDB column in <span 
class="ot1-zi4r-0-x-x-109">atomsFile </span>is used to determine which
     atoms are to be included in the group.
                                                                                         
                                                                                         
     </li>
     <li class="itemize"><a 
 id="dx1-15009"></a> <span 
class="ot1-zi4r-0-x-x-120">atomsColValue </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Atom selection ﬂag in the PDB column<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span>atom group <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive decimal <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>If deﬁned, this value in <span 
class="ot1-zi4r-0-x-x-109">atomsCol </span>identiﬁes atoms in <span 
class="ot1-zi4r-0-x-x-109">atomsFile </span>that are included in
     the group. If undeﬁned, all atoms with a non-zero value in <span 
class="ot1-zi4r-0-x-x-109">atomsCol </span>are included.
     </li>
     <li class="itemize"><a 
 id="dx1-15010"></a> <span 
class="ot1-zi4r-0-x-x-120">dummyAtom </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Dummy atom position (Å)<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span>atom group <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span><span 
class="ot1-zi4r-0-x-x-109">(x, y, z) </span>triplet <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>Instead of selecting any atom, this option makes the group a virtual particle at a ﬁxed
     position  in  space.  This  is  useful  e.g. to  replace  a  group&#x2019;s  center  of  geometry  with  a  user-deﬁned
     position.
     </li></ul>
<!--l. 1036--><p class="noindent" >
</p>
   <h4 class="subsectionHead"><span class="titlemark">5.2    </span> <a 
 id="key:colvars:atom_group:refPositions"></a>Moving frame of reference.</h4>
<!--l. 1039--><p class="noindent" >The following options deﬁne an automatic calculation of an optimal translation (<span 
class="ot1-zi4r-0-x-x-109">centerReference</span>) or optimal
rotation (<span 
class="ot1-zi4r-0-x-x-109">rotateReference</span>), that superimposes the positions of this group to a provided set of reference
coordinates. This can allow, for example, to eﬀectively remove from certain colvars the eﬀects of
molecular tumbling and of diﬀusion. Given the set of atomic positions <span 
class="ptmb7t-x-x-109">x</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub>, the colvar <span 
class="zptmcm7m-x-x-109">ξ </span>can be deﬁned
on a set of roto-translated positions <span 
class="ptmb7t-x-x-109">x</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub><span 
class="zptmcm7y-x-x-109">′ </span><span 
class="zptmcm7t-x-x-109">= </span><span 
class="zptmcm7m-x-x-109">R</span><span 
class="zptmcm7t-x-x-109">(</span><span 
class="ptmb7t-x-x-109">x</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub> <span 
class="zptmcm7y-x-x-109">− </span><span 
class="ptmb7t-x-x-109">x</span><sup><span 
class="zptmcm7t-x-x-80">C</span></sup><span 
class="zptmcm7t-x-x-109">)+ </span><span 
class="ptmb7t-x-x-109">x</span><sup><span 
class="zptmcm7t-x-x-80">ref</span></sup>. <span 
class="ptmb7t-x-x-109">x</span><sup><span 
class="zptmcm7t-x-x-80">C</span></sup> is the geometric center of the <span 
class="ptmb7t-x-x-109">x</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub>, <span 
class="zptmcm7m-x-x-109">R </span>is
the optimal rotation matrix to the reference positions and <span 
class="ptmb7t-x-x-109">x</span><sup><span 
class="zptmcm7t-x-x-80">ref</span></sup> is the geometric center of the reference
positions.
</p><!--l. 1044--><p class="indent" >   Components that are deﬁned based on pairwise distances are naturally invariant under global roto-translations.
Other components are instead aﬀected by global rotations or translations: however, they can be made invariant if
they are expressed in the frame of reference of a chosen group of atoms, using the <span 
class="ot1-zi4r-0-x-x-109">centerReference </span>and
<span 
class="ot1-zi4r-0-x-x-109">rotateReference </span>options. Finally, a few components are deﬁned by convention using a roto-translated frame (e.g.
the minimal RMSD): for these components, <span 
class="ot1-zi4r-0-x-x-109">centerReference </span>and <span 
class="ot1-zi4r-0-x-x-109">rotateReference </span>are enabled by default. In
typical applications, the default settings result in the expected behavior.
</p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-16001"></a> <span 
class="ot1-zi4r-0-x-x-120">centerReference </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Implicitly remove translations for this group<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span>atom group <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">off</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>If this option is <span 
class="ot1-zi4r-0-x-x-109">on</span>, the center of geometry of the group will be aligned with that of the
     reference positions provided by either <span 
class="ot1-zi4r-0-x-x-109">refPositions </span>or <span 
class="ot1-zi4r-0-x-x-109">refPositionsFile</span>. Colvar components will
     only have access to the aligned positions. <span 
class="ptmb7t-x-x-109">Note</span>: unless otherwise speciﬁed, <span 
class="ot1-zi4r-0-x-x-109">rmsd </span>and <span 
class="ot1-zi4r-0-x-x-109">eigenvector</span>
     set this option to <span 
class="ot1-zi4r-0-x-x-109">on </span><span 
class="ptmri7t-x-x-109">by default</span>.
     </li>
     <li class="itemize"><a 
 id="dx1-16002"></a> <span 
class="ot1-zi4r-0-x-x-120">rotateReference </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Implicitly remove rotations for this group<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span>atom group <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">off</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>If this option is <span 
class="ot1-zi4r-0-x-x-109">on</span>, the coordinates of this group will be optimally superimposed to
     the reference positions provided by either <span 
class="ot1-zi4r-0-x-x-109">refPositions </span>or <span 
class="ot1-zi4r-0-x-x-109">refPositionsFile</span>. The rotation will
     be performed around the center of geometry if <span 
class="ot1-zi4r-0-x-x-109">centerReference </span>is <span 
class="ot1-zi4r-0-x-x-109">on</span>, around the origin otherwise.
     The algorithm used is the same employed by the <span 
class="ot1-zi4r-0-x-x-109">orientation </span>colvar component <span class="cite">[<a 
href="#XCoutsias2004">3</a>]</span>. Forces applied
     to the atoms of this group will also be implicitly rotated back to the original frame. <span 
class="ptmb7t-x-x-109">Note</span>: unless
     otherwise speciﬁed, <span 
class="ot1-zi4r-0-x-x-109">rmsd </span>and <span 
class="ot1-zi4r-0-x-x-109">eigenvector </span>set this option to <span 
class="ot1-zi4r-0-x-x-109">on </span><span 
class="ptmri7t-x-x-109">by default</span>.
     </li>
     <li class="itemize"><a 
 id="dx1-16003"></a> <span 
class="ot1-zi4r-0-x-x-120">refPositions </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Reference positions for ﬁtting (Å)<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span>atom group <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>space-separated list of <span 
class="ot1-zi4r-0-x-x-109">(x, y, z) </span>triplets <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:      </span>This  option  provides  a  list  of  reference  coordinates  for  <span 
class="ot1-zi4r-0-x-x-109">centerReference </span>or
     <span 
class="ot1-zi4r-0-x-x-109">rotateReference</span>. If only <span 
class="ot1-zi4r-0-x-x-109">centerReference </span>is <span 
class="ot1-zi4r-0-x-x-109">on</span>, the list may contain a single (x, y, z) triplet; if
     also <span 
class="ot1-zi4r-0-x-x-109">rotateReference </span>is <span 
class="ot1-zi4r-0-x-x-109">on</span>, the list should be as long as the atom group.
     </li>
     <li class="itemize"><a 
 id="dx1-16004"></a> <span 
class="ot1-zi4r-0-x-x-120">refPositionsFile </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>File containing the reference positions for ﬁtting<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span>atom group <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>UNIX ﬁlename <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>This keyword provides the reference coordinates for ﬁtting from the given ﬁle, and is
     mutually exclusive with <span 
class="ot1-zi4r-0-x-x-109">refPositions</span>. The acceptable ﬁle format is XYZor PDB. Atomic positions
     are read diﬀerently depending on the three following scenarios: <span 
class="ptmri7t-x-x-109">i) </span>the ﬁle contains exactly as many
     records as the atoms in the group: all positions are read in sequence; <span 
class="ptmri7t-x-x-109">ii) </span>the ﬁle contains coordinates
     for the entire system: only the positions corresponding to the numeric indices of the atom group are
     read; <span 
class="ptmri7t-x-x-109">iii) </span>if the ﬁle is a PDB ﬁle and <span 
class="ot1-zi4r-0-x-x-109">refPositionsCol </span>is speciﬁed, positions are read according to
     the value of the column <span 
class="ot1-zi4r-0-x-x-109">refPositionsCol </span>(which may be the same as <span 
class="ot1-zi4r-0-x-x-109">atomsCol</span>).
     </li>
     <li class="itemize"><a 
 id="dx1-16005"></a> <span 
class="ot1-zi4r-0-x-x-120">refPositionsCol </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>PDB column containing atom ﬂags<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span>atom group <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span><span 
class="ot1-zi4r-0-x-x-109">O</span>, <span 
class="ot1-zi4r-0-x-x-109">B</span>, <span 
class="ot1-zi4r-0-x-x-109">X</span>, <span 
class="ot1-zi4r-0-x-x-109">Y</span>, or <span 
class="ot1-zi4r-0-x-x-109">Z</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>Like <span 
class="ot1-zi4r-0-x-x-109">atomsCol </span>for <span 
class="ot1-zi4r-0-x-x-109">atomsFile</span>, indicates which column to use to identify the atoms in
     <span 
class="ot1-zi4r-0-x-x-109">refPositionsFile </span>(if this is a PDB ﬁle).
     </li>
     <li class="itemize"><a 
 id="dx1-16006"></a> <span 
class="ot1-zi4r-0-x-x-120">refPositionsColValue </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Atom selection ﬂag in the PDB column<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span>atom group <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive decimal <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>Analogous to <span 
class="ot1-zi4r-0-x-x-109">atomsColValue</span>, but applied to <span 
class="ot1-zi4r-0-x-x-109">refPositionsCol</span>.
     </li>
     <li class="itemize"><a 
 id="dx1-16007"></a> <span 
class="ot1-zi4r-0-x-x-120">fittingGroup </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Use an alternate set of atoms to deﬁne the roto-translation<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span>atom group <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>Block <span 
class="ot1-zi4r-0-x-x-109">fittingGroup </span><span 
class="cmsy-10x-x-109">{ </span><span 
class="ot1-zi4r-0-x-x-109">... </span><span 
class="cmsy-10x-x-109">}</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span>This group itself <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>If either <span 
class="ot1-zi4r-0-x-x-109">centerReference </span>or <span 
class="ot1-zi4r-0-x-x-109">rotateReference </span>is deﬁned, this keyword deﬁnes an
     alternate atom group to calculate the optimal roto-translation. Use this option to deﬁne a continuous
                                                                                         
                                                                                         
     rotation if the structure of the group involved changes signiﬁcantly (a typical symptom would be the
     message “Warning: discontinuous rotation!”).
     <!--l. 1155--><p class="noindent" >The following example illustrates the syntax of <span 
class="ot1-zi4r-0-x-x-109">fittingGroup</span>: a group called “<span 
class="ot1-zi4r-0-x-x-109">atoms</span>” is deﬁned,
     including  8  C<sub><span 
class="zptmcm7m-x-x-80">α</span></sub>  atoms  of  a  protein  of  100  residues.  An  optimal  roto-translation  is  calculated
     automatically by ﬁtting the C<sub><span 
class="zptmcm7m-x-x-80">α</span></sub> trace of the rest of the protein onto the coordinates provided by a PDB
     ﬁle.
     </p><!--l. 1155--><p class="noindent" ><span 
class="ot1-zi4r-0-x-x-109"># Example: defining a group "atoms", with its coordinates expressed </span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"># on a roto-translated frame of reference defined by a second group</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109">atoms </span><span 
class="cmsy-10x-x-109">{</span><br 
class="newline" /><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> psfSegID PROT</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> atomNameResidueRange CA 41-48</span><br 
class="newline" /><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> centerReference yes</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> rotateReference yes</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> fittingGroup </span><span 
class="cmsy-10x-x-109">{</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> # define the frame by fitting the rest of the protein</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> psfSegID PROT PROT</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> atomNameResidueRange CA 1-40</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> atomNameResidueRange CA 49-100</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">} </span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> refPositionsFile all.pdb # can be the entire system</span><br 
class="newline" /><span 
class="cmsy-10x-x-109">}</span><br 
class="newline" /> </p></li></ul>
<!--l. 1158--><p class="indent" >   The following two options have default values appropriate for the vast majority of applications, and are only
provided to support rare, special cases. </p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-16008"></a> <span 
class="ot1-zi4r-0-x-x-120">enableFitGradients </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Include the roto-translational contribution to colvar gradients<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span>atom group <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">on</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>When  either  <span 
class="ot1-zi4r-0-x-x-109">centerReference </span>or  <span 
class="ot1-zi4r-0-x-x-109">rotateReference </span>is  on,  the  gradients  of  some
     colvars  include  terms  proportional  to  <span 
class="zptmcm7m-x-x-109">∂R∕∂</span><span 
class="ptmb7t-x-x-109">x</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub>  (rotational  gradients)  and  <span 
class="zptmcm7m-x-x-109">∂</span><span 
class="ptmb7t-x-x-109">x</span><sup><span 
class="zptmcm7t-x-x-80">C</span></sup><span 
class="zptmcm7m-x-x-109">∕∂</span><span 
class="ptmb7t-x-x-109">x</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub>  (translational
     gradients). By default, these terms are calculated and included in the total gradients; if this option is
     set to <span 
class="ot1-zi4r-0-x-x-109">off</span>, they are neglected. In the case of a minimum RMSD component, this ﬂag is automatically
     disabled because the contributions of those derivatives to the gradients cancel out.
     </li>
     <li class="itemize"><a 
 id="dx1-16009"></a> <span 
class="ot1-zi4r-0-x-x-120">enableForces </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Apply forces from this colvar to this group<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span>atom group <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">on</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>If this option is <span 
class="ot1-zi4r-0-x-x-109">off</span>, no forces are applied from this colvar to this group. Other forces
     are not aﬀected (i.e. those from the MD engine, from other colvars, and other external forces). For
     dummy atoms, this option is <span 
class="ot1-zi4r-0-x-x-109">off </span>by default.
     </li></ul>
                                                                                         
                                                                                         
<!--l. 1194--><p class="noindent" >
</p>
   <h4 class="subsectionHead"><span class="titlemark">5.3    </span> <a 
 id="x1-170005.3"></a>Treatment of periodic boundary conditions.</h4>
<!--l. 1210--><p class="noindent" >In simulations with periodic boundary conditions, NAMD maintains the coordinates of all the atoms within a
molecule contiguous to each other (i.e. there are no spurious “jumps” in the molecular bonds). The Colvars module
relies on this when calculating a group&#x2019;s center of geometry, but the condition may fail if the group spans diﬀerent
molecules: in that case, writing the NAMD output ﬁles <span 
class="ot1-zi4r-0-x-x-109">wrapAll </span>or <span 
class="ot1-zi4r-0-x-x-109">wrapWater </span>could produce wrong results when a
simulation run is continued from a previous one. The user should then determine, according to which type of colvars
are being calculated, whether <span 
class="ot1-zi4r-0-x-x-109">wrapAll </span>or <span 
class="ot1-zi4r-0-x-x-109">wrapWater </span>can be enabled. In general, internal coordinate wrapping
by NAMD does not aﬀect the calculation of colvars if each atom group satisﬁes one or more of the
following:
</p><!--l. 1225--><p class="indent" >
     </p><ol  class="enumerate1" >
     <li 
  class="enumerate" id="x1-17001x5.3">it is composed by only one atom;
     </li>
     <li 
  class="enumerate" id="x1-17002x5.3">it is used by a colvar component which does not make use of its center of geometry, but only of
     pairwise distances (<span 
class="ot1-zi4r-0-x-x-109">distanceInv</span>, <span 
class="ot1-zi4r-0-x-x-109">coordNum</span>, <span 
class="ot1-zi4r-0-x-x-109">hBond</span>, <span 
class="ot1-zi4r-0-x-x-109">alpha</span>, <span 
class="ot1-zi4r-0-x-x-109">dihedralPC</span>);
     </li>
     <li 
  class="enumerate" id="x1-17003x5.3">it is used by a colvar component that ignores the ill-deﬁned Cartesian components of its center of
     mass (such as the <span 
class="zptmcm7m-x-x-109">x </span>and <span 
class="zptmcm7m-x-x-109">y </span>components of a membrane&#x2019;s center of mass modeled with <span 
class="ot1-zi4r-0-x-x-109">distanceZ</span>);
     </li>
     <li 
  class="enumerate" id="x1-17004x5.3">it has all of its atoms within the same molecular fragment.</li></ol>
<!--l. 1234--><p class="noindent" >
</p>
   <h4 class="subsectionHead"><span class="titlemark">5.4    </span> <a 
 id="x1-180005.4"></a>Performance a Colvars calculation based on group size.</h4>
<!--l. 1237--><p class="noindent" >In simulations performed with message-passing programs (such as NAMD or LAMMPS), the calculation of energy
and forces is distributed (i.e., parallelized) across multiple nodes, as well as over the processor cores of each node.
Atomic coordinates are typically collected on one node, where the calculation of collective variables and of their
biases is executed. This means that for simulations over large numbers of nodes, a Colvars calculation may produce
a signiﬁcant overhead, coming from the costs of transmitting atomic coordinates to one node and of processing
them. The latency-tolerant design and dynamic load balancing of NAMD may alleviate both factors, but a noticeable
performance impact may be observed.
</p><!--l. 1242--><p class="indent" >   Performance can be improved in multiple ways: </p>
     <ul class="itemize1">
     <li class="itemize">The calculation of variables, components and biases can be distributed over the processor cores of the
     node where the Colvars module is executed. Currently, an equal weight is assigned to each colvar,
     or to each component of those colvars that include more than one component. The performance of
     simulations that use many colvars or components is improved automatically. For simulations that use
     a single large colvar, it may be advisable to partition it in multiple components, which will be then
     distributed across the available cores. In NAMD, this feature is enabled in all binaries compiled using
                                                                                         
                                                                                         
     SMP builds of Charm++ with the CkLoop extension.   If printed, the message “SMP parallelism is
     available.” indicates the availability of the option.
     </li>
     <li class="itemize">NAMD also oﬀers a parallelized calculation of the centers of mass of groups of atoms. This option is
     on by default for all components that are simple functions of centers of mass, and is controlled by the
     keyword <a 
href="#sec:cvc_common"><span 
class="ot1-zi4r-0-x-x-109">scalable</span></a>. When supported, the message “Will enable scalable calculation for group …” is
     printed for each group.
     </li>
     <li class="itemize">As a general rule, the size of atom groups should be kept relatively small (up to a few thousands
     of  atoms,  depending  on  the  size  of  the  entire  system  in  comparison).  To  gain  an  estimate  of  the
     computational cost of a large colvar, one can use a test calculation of the same colvar in VMD (hint:
     use the <span 
class="ot1-zi4r-0-x-x-109">time </span>Tcl command to measure the cost of running <span 
class="ot1-zi4r-0-x-x-109">cv update</span>).</li></ul>
<!--l. 1265--><p class="noindent" >
</p>
   <h3 class="sectionHead"><span class="titlemark">6    </span> <a 
 id="sec:cvc"></a>Collective variable components (basis functions)</h3>
<!--l. 1268--><p class="noindent" >Each colvar is deﬁned by one or more <span 
class="ptmri7t-x-x-109">components </span>(typically only one). Each component consists of a keyword
identifying a functional form, and a deﬁnition block following that keyword, specifying the atoms involved and any
additional parameters (cutoﬀs, “reference” values, …).
</p><!--l. 1274--><p class="indent" >   The types of the components used in a colvar determine the properties of that colvar, and which biasing or
analysis methods can be applied. In most cases, the colvar returns a real number, which is computed by one or more
instances of the following components: </p>
     <ul class="itemize1">
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">distance</span>: distance between two groups;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">distanceZ</span>: projection of a distance vector on an axis;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">distanceXY</span>: projection of a distance vector on a plane;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">distanceInv</span>: mean distance between two groups of atoms (e.g. NOE-based distance);
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">angle</span>: angle between three groups;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">dihedral</span>: torsional (dihedral) angle between four groups;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">dipoleAngle</span>: angle between two groups and dipole of a third group;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">polarTheta</span>: polar angle of a group in spherical coordinates;
                                                                                         
                                                                                         
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">polarPhi</span>: azimuthal angle of a group in spherical coordinates;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">coordNum</span>: coordination number between two groups;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">selfCoordNum</span>: coordination number of atoms within a group;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">hBond</span>: hydrogen bond between two atoms;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">rmsd</span>: root mean square deviation (RMSD) from a set of reference coordinates;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">eigenvector</span>: projection of the atomic coordinates on a vector;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">orientationAngle</span>: angle of the best-ﬁt rotation from a set of reference coordinates;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">orientationProj</span>: cosine of <span 
class="ot1-zi4r-0-x-x-109">orientationProj</span>;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">spinAngle</span>: projection orthogonal to an axis of the best-ﬁt rotation from a set of reference coordinates;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">tilt</span>: projection on an axis of the best-ﬁt rotation from a set of reference coordinates;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">gyration</span>: radius of gyration of a group of atoms;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">inertia</span>: moment of inertia of a group of atoms;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">inertiaZ</span>: moment of inertia of a group of atoms around a chosen axis;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">alpha</span>: <span 
class="zptmcm7m-x-x-109">α</span>-helix content of a protein segment.
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">dihedralPC</span>: projection of protein backbone dihedrals onto a dihedral principal component. </li></ul>
<!--l. 1312--><p class="indent" >   Some components do not return scalar, but vector values. They can only be combined with vector values of the
same type, except within a scripted collective variable. </p>
     <ul class="itemize1">
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">distanceVec</span>: distance vector between two groups;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">distanceDir</span>: unit vector parallel to distanceVec;
                                                                                         
                                                                                         
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">cartesian</span>: vector of atomic Cartesian coordinates;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">orientation</span>: best-ﬁt rotation, expressed as a unit quaternion.</li></ul>
<!--l. 1323--><p class="indent" >   In the following, all the available component types are listed, along with their physical units and the limiting
values, if any. Such limiting values can be used to deﬁne <span 
class="ot1-zi4r-0-x-x-109">lowerBoundary </span>and <span 
class="ot1-zi4r-0-x-x-109">upperBoundary </span>in the parent
colvar.
</p><!--l. 1328--><p class="indent" >   For each type of component, the available conﬁgurations keywords are listed: when two components share
certain keywords, the second component simply references to the documentation of the ﬁrst regarding
that keyword. The keywords that are available for all types of components are listed at the end (see
 
</p><!--l. 1334--><p class="noindent" >
</p>
   <h4 class="subsectionHead"><span class="titlemark">6.1    </span> <a 
 id="x1-200006.1"></a>List of available colvar components</h4>
<!--l. 1343--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">6.1.1    </span> <a 
 id="colvar|distance|oneSiteTotalForce"></a><span 
class="ot1-zi4r-0-x-x-109">distance</span>: center-of-mass distance between two groups.</h5>
<!--l. 1345--><p class="noindent" >The <span 
class="ot1-zi4r-0-x-x-109">distance </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109">...</span><span 
class="cmsy-10x-x-109">} </span>block deﬁnes a distance component between the two atom groups, <span 
class="ot1-zi4r-0-x-x-109">group1 </span>and
<span 
class="ot1-zi4r-0-x-x-109">group2</span>.
</p><!--l. 1347--><p class="noindent" ><span 
class="ptmb7t-x-x-109">List of keywords </span>(see also <a 
href="#sec:cvc_superp">6.4<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="colvar|distance|group1"></a> <a 
 id="dx1-21001"></a><span 
class="ot1-zi4r-0-x-x-120">group1 </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>First group of atoms<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">distance</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>Block <span 
class="ot1-zi4r-0-x-x-109">group1 </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109">...</span><span 
class="cmsy-10x-x-109">}</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>First group of atoms.
     </li>
     <li class="itemize"><a 
 id="colvar|distance|group2"></a> <a 
 id="dx1-21002"></a><span 
class="ot1-zi4r-0-x-x-120">group2: </span>analogous to <span 
class="ot1-zi4r-0-x-x-109">group1</span>
     </li>
     <li class="itemize"><a 
 id="colvar|distance|oneSiteTotalForce"></a><a 
 id="dx1-21003"></a><span 
class="ot1-zi4r-0-x-x-120">forceNoPBC </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Calculate absolute rather than minimum-image distance?<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">distance</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">no</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>By default, in calculations with periodic boundary conditions, the <span 
class="ot1-zi4r-0-x-x-109">distance </span>component
     returns the distance according to the minimum-image convention. If this parameter is set to <span 
class="ot1-zi4r-0-x-x-109">yes</span>, PBC
     will be ignored and the distance between the coordinates as maintained internally will be used. This is
     only useful in a limited number of special cases, e.g. to describe the distance between remote points
     of a single macromolecule, which cannot be split across periodic cell boundaries, and for which the
     minimum-image distance might give the wrong result because of a relatively small periodic cell.
                                                                                         
                                                                                         
     </li>
     <li class="itemize"><a 
 id="colvar|distance|oneSiteTotalForce"></a> <a 
 id="dx1-21004"></a><span 
class="ot1-zi4r-0-x-x-120">oneSiteTotalForce </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Measure total force on group 1 only?<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">angle</span>, <span 
class="ot1-zi4r-0-x-x-109">dipoleAngle</span>, <span 
class="ot1-zi4r-0-x-x-109">dihedral</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">no</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>If this is set to <span 
class="ot1-zi4r-0-x-x-109">yes</span>, the total force is measured along a vector ﬁeld (see equation (<a 
href="#eq:gradient_vector">19<!--tex4ht:ref: eq:gradient_vector --></a>) in
     section <a 
href="#sec:colvarbias_abf">7.1<!--tex4ht:ref: sec:colvarbias_abf --></a>) that only involves atoms of <span 
class="ot1-zi4r-0-x-x-109">group1</span>. This option is only useful for ABF, or custom biases
     that compute total forces. See section <a 
href="#sec:colvarbias_abf">7.1<!--tex4ht:ref: sec:colvarbias_abf --></a> for details.
     </li></ul>
<!--l. 1396--><p class="indent" >   The value returned is a positive number (in Å), ranging from <span 
class="zptmcm7t-x-x-109">0 </span>to the largest possible interatomic distance within
the chosen boundary conditions (with PBCs, the minimum image convention is used unless the <span 
class="ot1-zi4r-0-x-x-109">forceNoPBC </span>option
is set).
</p><!--l. 1402--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">6.1.2    </span> <a 
 id="colvar|distanceZ|ref"></a><span 
class="ot1-zi4r-0-x-x-109">distanceZ</span>: projection of a distance vector on an axis.</h5>
<!--l. 1403--><p class="noindent" >The <span 
class="ot1-zi4r-0-x-x-109">distanceZ</span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109">...</span><span 
class="cmsy-10x-x-109">} </span>block deﬁnes a distance projection component, which can be seen as measuring the distance
between two groups projected onto an axis, or the position of a group along such an axis. The axis can be
deﬁned using either one reference group and a constant vector, or dynamically based on two reference
groups.
</p><!--l. 1409--><p class="noindent" ><span 
class="ptmb7t-x-x-109">List of keywords </span>(see also <a 
href="#sec:cvc_superp">6.4<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="colvar|distanceZ|ref"></a> <a 
 id="dx1-22001"></a><span 
class="ot1-zi4r-0-x-x-120">main </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Main group of atoms<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">distanceZ</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>Block <span 
class="ot1-zi4r-0-x-x-109">main </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109">...</span><span 
class="cmsy-10x-x-109">}</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>Group of atoms whose position <span 
class="zptmcm7m-x-x-109">r</span> is measured.
     </li>
     <li class="itemize"><a 
 id="colvar|distanceZ|ref"></a> <a 
 id="dx1-22002"></a><span 
class="ot1-zi4r-0-x-x-120">ref </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Reference group of atoms<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">distanceZ</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>Block <span 
class="ot1-zi4r-0-x-x-109">ref </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109">...</span><span 
class="cmsy-10x-x-109">}</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>Reference group of atoms. The position of its center of mass is noted <span 
class="zptmcm7m-x-x-109">r</span><sub><span 
class="zptmcm7t-x-x-80">1</span></sub> below.
     </li>
     <li class="itemize"><a 
 id="colvar|distanceZ|ref2"></a> <a 
 id="dx1-22003"></a><span 
class="ot1-zi4r-0-x-x-120">ref2 </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Secondary reference group<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">distanceZ</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>Block <span 
class="ot1-zi4r-0-x-x-109">ref2 </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109">...</span><span 
class="cmsy-10x-x-109">}</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span>none <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>Optional group of reference atoms, whose position <span 
class="zptmcm7m-x-x-109">r</span><sub><span 
class="zptmcm7t-x-x-80">2</span></sub> can be used to deﬁne a dynamic
     projection axis: <span 
class="zptmcm7m-x-x-109">e</span> <span 
class="zptmcm7t-x-x-109">= (</span><span 
class="zptmcm7y-x-x-109">∥</span><span 
class="zptmcm7m-x-x-109">r</span><sub><span 
class="zptmcm7t-x-x-80">2</span></sub> <span 
class="zptmcm7y-x-x-109">−</span><span 
class="zptmcm7m-x-x-109">r</span><sub><span 
class="zptmcm7t-x-x-80">1</span></sub><span 
class="zptmcm7y-x-x-109">∥</span><span 
class="zptmcm7t-x-x-109">)</span><sup><span 
class="zptmcm7y-x-x-80">−</span><span 
class="zptmcm7t-x-x-80">1</span></sup> <span 
class="zptmcm7y-x-x-109">×</span><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">r</span><sub><span 
class="zptmcm7t-x-x-80">2</span></sub> <span 
class="zptmcm7y-x-x-109">−</span><span 
class="zptmcm7m-x-x-109">r</span><sub><span 
class="zptmcm7t-x-x-80">1</span></sub><span 
class="zptmcm7t-x-x-109">)</span>. In this case, the origin is <span 
class="zptmcm7m-x-x-109">r</span><sub><span 
class="zptmcm7m-x-x-80">m</span></sub> <span 
class="zptmcm7t-x-x-109">= 1</span><span 
class="zptmcm7m-x-x-109">∕</span><span 
class="zptmcm7t-x-x-109">2(</span><span 
class="zptmcm7m-x-x-109">r</span><sub><span 
class="zptmcm7t-x-x-80">1</span></sub> <span 
class="zptmcm7t-x-x-109">+</span><span 
class="zptmcm7m-x-x-109">r</span><sub><span 
class="zptmcm7t-x-x-80">2</span></sub><span 
class="zptmcm7t-x-x-109">)</span>, and the
     value of the component is <span 
class="zptmcm7m-x-x-109">e</span><span 
class="zptmcm7y-x-x-109">⋅</span><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">r</span><span 
class="zptmcm7y-x-x-109">−</span><span 
class="zptmcm7m-x-x-109">r</span><sub><span 
class="zptmcm7m-x-x-80">m</span></sub><span 
class="zptmcm7t-x-x-109">)</span>.
     </li>
     <li class="itemize"><a 
 id="colvar|distanceZ|axis"></a> <a 
 id="dx1-22004"></a><span 
class="ot1-zi4r-0-x-x-120">axis </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Projection axis (Å)<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">distanceZ</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span><span 
class="ot1-zi4r-0-x-x-109">(x, y, z) </span>triplet <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">(0.0, 0.0, 1.0)</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>The three components of this vector deﬁne a projection axis <span 
class="zptmcm7m-x-x-109">e</span> for the distance vector
     <span 
class="zptmcm7m-x-x-109">r</span><span 
class="zptmcm7y-x-x-109">−</span><span 
class="zptmcm7m-x-x-109">r</span><sub><span 
class="zptmcm7t-x-x-80">1</span></sub> joining the centers of groups <span 
class="ot1-zi4r-0-x-x-109">ref </span>and <span 
class="ot1-zi4r-0-x-x-109">main</span>. The value of the component is then <span 
class="zptmcm7m-x-x-109">e</span><span 
class="zptmcm7y-x-x-109">⋅</span><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">r</span><span 
class="zptmcm7y-x-x-109">−</span><span 
class="zptmcm7m-x-x-109">r</span><sub><span 
class="zptmcm7t-x-x-80">1</span></sub><span 
class="zptmcm7t-x-x-109">)</span>. The
     vector should be written as three components separated by commas and enclosed in parentheses.
     </li>
     <li class="itemize"><a 
 id="dx1-22005"></a> <span 
class="ot1-zi4r-0-x-x-120">forceNoPBC: </span>see deﬁnition of <a 
href="#colvar|distance|oneSiteTotalForce"><span 
class="ot1-zi4r-0-x-x-109">forceNoPBC</span></a> in sec. <a 
href="#colvar|distance|oneSiteTotalForce">6.1.1<!--tex4ht:ref: colvar|distance|oneSiteTotalForce --></a> (<span 
class="ot1-zi4r-0-x-x-109">distance </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-22006"></a> <span 
class="ot1-zi4r-0-x-x-120">oneSiteTotalForce: </span>see deﬁnition of <a 
href="#colvar|distance|oneSiteTotalForce"><span 
class="ot1-zi4r-0-x-x-109">oneSiteTotalForce</span></a> in sec. <a 
href="#colvar|distance|oneSiteTotalForce">6.1.1<!--tex4ht:ref: colvar|distance|oneSiteTotalForce --></a> (<span 
class="ot1-zi4r-0-x-x-109">distance </span>component)
     </li></ul>
<!--l. 1466--><p class="noindent" >This component returns a number (in Å) whose range is determined by the chosen boundary conditions.
For instance, if the <span 
class="zptmcm7m-x-x-109">z </span>axis is used in a simulation with periodic boundaries, the returned value ranges
between <span 
class="zptmcm7y-x-x-109">−</span><span 
class="zptmcm7m-x-x-109">b</span><sub><span 
class="zptmcm7m-x-x-80">z</span></sub><span 
class="zptmcm7m-x-x-109">∕</span><span 
class="zptmcm7t-x-x-109">2 </span>and <span 
class="zptmcm7m-x-x-109">b</span><sub><span 
class="zptmcm7m-x-x-80">z</span></sub><span 
class="zptmcm7m-x-x-109">∕</span><span 
class="zptmcm7t-x-x-109">2</span>, where <span 
class="zptmcm7m-x-x-109">b</span><sub><span 
class="zptmcm7m-x-x-80">z</span></sub> is the box length along <span 
class="zptmcm7m-x-x-109">z </span>(this behavior is disabled if <span 
class="ot1-zi4r-0-x-x-109">forceNoPBC </span>is
set).
</p><!--l. 1473--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">6.1.3    </span> <a 
 id="x1-230006.1.3"></a><span 
class="ot1-zi4r-0-x-x-109">distanceXY</span>: modulus of the projection of a distance vector on a plane.</h5>
<!--l. 1474--><p class="noindent" >The <span 
class="ot1-zi4r-0-x-x-109">distanceXY</span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109">...</span><span 
class="cmsy-10x-x-109">} </span>block deﬁnes a distance projected on a plane, and accepts the same keywords as the
component <span 
class="ot1-zi4r-0-x-x-109">distanceZ</span>, i.e. <span 
class="ot1-zi4r-0-x-x-109">main</span>, <span 
class="ot1-zi4r-0-x-x-109">ref</span>, either <span 
class="ot1-zi4r-0-x-x-109">ref2 </span>or <span 
class="ot1-zi4r-0-x-x-109">axis</span>, and <span 
class="ot1-zi4r-0-x-x-109">oneSiteTotalForce</span>. It returns the norm
of the projection of the distance vector between <span 
class="ot1-zi4r-0-x-x-109">main </span>and <span 
class="ot1-zi4r-0-x-x-109">ref </span>onto the plane orthogonal to the axis.
The axis is deﬁned using the <span 
class="ot1-zi4r-0-x-x-109">axis </span>parameter or as the vector joining <span 
class="ot1-zi4r-0-x-x-109">ref </span>and <span 
class="ot1-zi4r-0-x-x-109">ref2 </span>(see <span 
class="ot1-zi4r-0-x-x-109">distanceZ</span>
above).
</p><!--l. 1483--><p class="noindent" ><span 
class="ptmb7t-x-x-109">List of keywords </span>(see also <a 
href="#sec:cvc_superp">6.4<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-23001"></a> <span 
class="ot1-zi4r-0-x-x-120">main: </span>see deﬁnition of <a 
href="#colvar|distanceZ|ref"><span 
class="ot1-zi4r-0-x-x-109">main</span></a> in sec. <a 
href="#colvar|distanceZ|ref">6.1.2<!--tex4ht:ref: colvar|distanceZ|ref2 --></a> (<span 
class="ot1-zi4r-0-x-x-109">distanceZ </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-23002"></a> <span 
class="ot1-zi4r-0-x-x-120">ref: </span>see deﬁnition of <a 
href="#colvar|distanceZ|ref"><span 
class="ot1-zi4r-0-x-x-109">ref</span></a> in sec. <a 
href="#colvar|distanceZ|ref">6.1.2<!--tex4ht:ref: colvar|distanceZ|ref --></a> (<span 
class="ot1-zi4r-0-x-x-109">distanceZ </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-23003"></a> <span 
class="ot1-zi4r-0-x-x-120">ref2: </span>see deﬁnition of <a 
href="#colvar|distanceZ|ref2"><span 
class="ot1-zi4r-0-x-x-109">ref2</span></a> in sec. <a 
href="#colvar|distanceZ|ref">6.1.2<!--tex4ht:ref: colvar|distanceZ|ref2 --></a> (<span 
class="ot1-zi4r-0-x-x-109">distanceZ </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-23004"></a> <span 
class="ot1-zi4r-0-x-x-120">axis: </span>see deﬁnition of <a 
href="#colvar|distanceZ|axis"><span 
class="ot1-zi4r-0-x-x-109">axis</span></a> in sec. <a 
href="#colvar|distanceZ|ref">6.1.2<!--tex4ht:ref: colvar|distanceZ|axis --></a> (<span 
class="ot1-zi4r-0-x-x-109">distanceZ </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-23005"></a> <span 
class="ot1-zi4r-0-x-x-120">forceNoPBC: </span>see deﬁnition of <a 
href="#colvar|distance|oneSiteTotalForce"><span 
class="ot1-zi4r-0-x-x-109">forceNoPBC</span></a> in sec. <a 
href="#colvar|distance|oneSiteTotalForce">6.1.1<!--tex4ht:ref: colvar|distance|oneSiteTotalForce --></a> (<span 
class="ot1-zi4r-0-x-x-109">distance </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-23006"></a> <span 
class="ot1-zi4r-0-x-x-120">oneSiteTotalForce: </span>see deﬁnition of <a 
href="#colvar|distance|oneSiteTotalForce"><span 
class="ot1-zi4r-0-x-x-109">oneSiteTotalForce</span></a> in sec. <a 
href="#colvar|distance|oneSiteTotalForce">6.1.1<!--tex4ht:ref: colvar|distance|oneSiteTotalForce --></a> (<span 
class="ot1-zi4r-0-x-x-109">distance </span>component)
     </li></ul>
                                                                                         
                                                                                         
<!--l. 1500--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">6.1.4    </span> <a 
 id="x1-240006.1.4"></a><span 
class="ot1-zi4r-0-x-x-109">distanceVec</span>: distance vector between two groups.</h5>
<!--l. 1501--><p class="noindent" >The <span 
class="ot1-zi4r-0-x-x-109">distanceVec</span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109">...</span><span 
class="cmsy-10x-x-109">} </span>block deﬁnes a distance vector component, which accepts the same keywords as the
component <span 
class="ot1-zi4r-0-x-x-109">distance</span>: <span 
class="ot1-zi4r-0-x-x-109">group1</span>, <span 
class="ot1-zi4r-0-x-x-109">group2</span>, and <span 
class="ot1-zi4r-0-x-x-109">forceNoPBC</span>. Its value is the 3-vector joining the centers of mass of
<span 
class="ot1-zi4r-0-x-x-109">group1 </span>and <span 
class="ot1-zi4r-0-x-x-109">group2</span>.
</p><!--l. 1507--><p class="noindent" ><span 
class="ptmb7t-x-x-109">List of keywords </span>(see also <a 
href="#sec:cvc_superp">6.4<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-24001"></a> <span 
class="ot1-zi4r-0-x-x-120">group1: </span>see deﬁnition of <a 
href="#colvar|distance|group1"><span 
class="ot1-zi4r-0-x-x-109">group1</span></a> in sec. <a 
href="#colvar|distance|oneSiteTotalForce">6.1.1<!--tex4ht:ref: colvar|distance|group1 --></a> (<span 
class="ot1-zi4r-0-x-x-109">distance </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-24002"></a> <span 
class="ot1-zi4r-0-x-x-120">group2: </span>analogous to <span 
class="ot1-zi4r-0-x-x-109">group1</span>
     </li>
     <li class="itemize"><a 
 id="dx1-24003"></a><span 
class="ot1-zi4r-0-x-x-120">forceNoPBC: </span>see deﬁnition of <a 
href="#colvar|distance|oneSiteTotalForce"><span 
class="ot1-zi4r-0-x-x-109">forceNoPBC</span></a> in sec. <a 
href="#colvar|distance|oneSiteTotalForce">6.1.1<!--tex4ht:ref: colvar|distance|oneSiteTotalForce --></a> (<span 
class="ot1-zi4r-0-x-x-109">distance </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-24004"></a> <span 
class="ot1-zi4r-0-x-x-120">oneSiteTotalForce: </span>see deﬁnition of <a 
href="#colvar|distance|oneSiteTotalForce"><span 
class="ot1-zi4r-0-x-x-109">oneSiteTotalForce</span></a> in sec. <a 
href="#colvar|distance|oneSiteTotalForce">6.1.1<!--tex4ht:ref: colvar|distance|oneSiteTotalForce --></a> (<span 
class="ot1-zi4r-0-x-x-109">distance </span>component)
     </li></ul>
<!--l. 1520--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">6.1.5    </span> <a 
 id="x1-250006.1.5"></a><span 
class="ot1-zi4r-0-x-x-109">distanceDir</span>: distance unit vector between two groups.</h5>
<!--l. 1521--><p class="noindent" >The <span 
class="ot1-zi4r-0-x-x-109">distanceDir</span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109">...</span><span 
class="cmsy-10x-x-109">} </span>block deﬁnes a distance unit vector component, which accepts the same keywords as the
component <span 
class="ot1-zi4r-0-x-x-109">distance</span>: <span 
class="ot1-zi4r-0-x-x-109">group1</span>, <span 
class="ot1-zi4r-0-x-x-109">group2</span>, and <span 
class="ot1-zi4r-0-x-x-109">forceNoPBC</span>. It returns a 3-dimensional unit vector <span 
class="ptmb7t-x-x-109">d </span><span 
class="zptmcm7t-x-x-109">= (</span><span 
class="zptmcm7m-x-x-109">d</span><sub><span 
class="zptmcm7m-x-x-80">x</span></sub><span 
class="zptmcm7m-x-x-109">,d</span><sub><span 
class="zptmcm7m-x-x-80">y</span></sub><span 
class="zptmcm7m-x-x-109">,d</span><sub><span 
class="zptmcm7m-x-x-80">z</span></sub><span 
class="zptmcm7t-x-x-109">)</span>,
with <span 
class="zptmcm7y-x-x-109">|</span><span 
class="ptmb7t-x-x-109">d</span><span 
class="zptmcm7y-x-x-109">| </span><span 
class="zptmcm7t-x-x-109">= 1</span>.
</p><!--l. 1528--><p class="noindent" ><span 
class="ptmb7t-x-x-109">List of keywords </span>(see also <a 
href="#sec:cvc_superp">6.4<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-25001"></a> <span 
class="ot1-zi4r-0-x-x-120">group1: </span>see deﬁnition of <a 
href="#colvar|distance|group1"><span 
class="ot1-zi4r-0-x-x-109">group1</span></a> in sec. <a 
href="#colvar|distance|oneSiteTotalForce">6.1.1<!--tex4ht:ref: colvar|distance|group1 --></a> (<span 
class="ot1-zi4r-0-x-x-109">distance </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-25002"></a> <span 
class="ot1-zi4r-0-x-x-120">group2: </span>analogous to <span 
class="ot1-zi4r-0-x-x-109">group1</span>
     </li>
     <li class="itemize"><a 
 id="dx1-25003"></a><span 
class="ot1-zi4r-0-x-x-120">forceNoPBC: </span>see deﬁnition of <a 
href="#colvar|distance|oneSiteTotalForce"><span 
class="ot1-zi4r-0-x-x-109">forceNoPBC</span></a> in sec. <a 
href="#colvar|distance|oneSiteTotalForce">6.1.1<!--tex4ht:ref: colvar|distance|oneSiteTotalForce --></a> (<span 
class="ot1-zi4r-0-x-x-109">distance </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-25004"></a> <span 
class="ot1-zi4r-0-x-x-120">oneSiteTotalForce: </span>see deﬁnition of <a 
href="#colvar|distance|oneSiteTotalForce"><span 
class="ot1-zi4r-0-x-x-109">oneSiteTotalForce</span></a> in sec. <a 
href="#colvar|distance|oneSiteTotalForce">6.1.1<!--tex4ht:ref: colvar|distance|oneSiteTotalForce --></a> (<span 
class="ot1-zi4r-0-x-x-109">distance </span>component)
     </li></ul>
<!--l. 1540--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">6.1.6    </span> <a 
 id="x1-260006.1.6"></a><span 
class="ot1-zi4r-0-x-x-109">distanceInv</span>: mean distance between two groups of atoms.</h5>
<!--l. 1541--><p class="noindent" >The <span 
class="ot1-zi4r-0-x-x-109">distanceInv</span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109">...</span><span 
class="cmsy-10x-x-109">} </span>block deﬁnes a generalized mean distance between two groups of atoms 1 and 2, weighted
with exponent <span 
class="zptmcm7t-x-x-109">1</span><span 
class="zptmcm7m-x-x-109">∕n</span>:
                                                                                         
                                                                                         
</p>
   <table 
class="equation"><tr><td><a 
 id="eq:distanceInv"></a>
   <center class="math-display" >
<img 
src="colvars-refman-namd4x.png" alt="      (        (     )n )− 1∕n
 [n]     -1---    -1---
d1,2 =   N1N2 ∑i,j ∥dij∥
" class="math-display"  /></center></td><td class="equation-label">(2)</td></tr></table>
<!--l. 1545--><p class="nopar" >
where <span 
class="zptmcm7y-x-x-109">∥</span><span 
class="ptmb7t-x-x-109">d</span><sup><span 
class="zptmcm7m-x-x-80">ij</span></sup><span 
class="zptmcm7y-x-x-109">∥ </span>is the distance between atoms <span 
class="zptmcm7m-x-x-109">i </span>and <span 
class="zptmcm7m-x-x-109">j </span>in groups 1 and 2 respectively, and <span 
class="zptmcm7m-x-x-109">n </span>is an even
integer.
</p><!--l. 1548--><p class="noindent" ><span 
class="ptmb7t-x-x-109">List of keywords </span>(see also <a 
href="#sec:cvc_superp">6.4<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-26002"></a> <span 
class="ot1-zi4r-0-x-x-120">group1: </span>see deﬁnition of <a 
href="#colvar|distance|group1"><span 
class="ot1-zi4r-0-x-x-109">group1</span></a> in sec. <a 
href="#colvar|distance|oneSiteTotalForce">6.1.1<!--tex4ht:ref: colvar|distance|group1 --></a> (<span 
class="ot1-zi4r-0-x-x-109">distance </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-26003"></a> <span 
class="ot1-zi4r-0-x-x-120">group2: </span>analogous to <span 
class="ot1-zi4r-0-x-x-109">group1</span>
     </li>
     <li class="itemize"><a 
 id="dx1-26004"></a><span 
class="ot1-zi4r-0-x-x-120">oneSiteTotalForce: </span>see deﬁnition of <a 
href="#colvar|distance|oneSiteTotalForce"><span 
class="ot1-zi4r-0-x-x-109">oneSiteTotalForce</span></a> in sec. <a 
href="#colvar|distance|oneSiteTotalForce">6.1.1<!--tex4ht:ref: colvar|distance|oneSiteTotalForce --></a> (<span 
class="ot1-zi4r-0-x-x-109">distance </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-26005"></a> <span 
class="ot1-zi4r-0-x-x-120">exponent </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Exponent <span 
class="zptmcm7m-x-x-109">n </span>in equation <a 
href="#eq:distanceInv">2<!--tex4ht:ref: eq:distanceInv --></a><span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">distanceInv</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive even integer <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span>6 <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>Deﬁnes the exponent to which the individual distances are elevated before averaging.
     The default value of 6 is useful for example to applying restraints based on NOE-measured distances.
     </li></ul>
<!--l. 1563--><p class="noindent" >This component returns a number in Å, ranging from <span 
class="zptmcm7t-x-x-109">0 </span>to the largest possible distance within the chosen boundary
conditions.
</p><!--l. 1566--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">6.1.7    </span> <a 
 id="x1-270006.1.7"></a><span 
class="ot1-zi4r-0-x-x-109">distancePairs</span>: set of pairwise distances between two groups.</h5>
<!--l. 1567--><p class="noindent" >The <span 
class="ot1-zi4r-0-x-x-109">distancePairs</span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109">...</span><span 
class="cmsy-10x-x-109">} </span>block deﬁnes a <span 
class="zptmcm7m-x-x-109">N</span><sub><span 
class="zptmcm7t-x-x-80">1</span></sub> <span 
class="zptmcm7y-x-x-109">×</span><span 
class="zptmcm7m-x-x-109">N</span><sub><span 
class="zptmcm7t-x-x-80">2</span></sub>-dimensional variable that includes all mutual distances
between the atoms of two groups. This can be useful, for example, to develop a new variable deﬁned over two
groups, by using the <span 
class="ot1-zi4r-0-x-x-109">scriptedFunction </span>feature.
</p><!--l. 1570--><p class="noindent" ><span 
class="ptmb7t-x-x-109">List of keywords </span>(see also <a 
href="#sec:cvc_superp">6.4<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
                                                                                         
                                                                                         
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-27001"></a> <span 
class="ot1-zi4r-0-x-x-120">group1: </span>see deﬁnition of <a 
href="#colvar|distance|group1"><span 
class="ot1-zi4r-0-x-x-109">group1</span></a> in sec. <a 
href="#colvar|distance|oneSiteTotalForce">6.1.1<!--tex4ht:ref: colvar|distance|group1 --></a> (<span 
class="ot1-zi4r-0-x-x-109">distance </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-27002"></a> <span 
class="ot1-zi4r-0-x-x-120">group2: </span>analogous to <span 
class="ot1-zi4r-0-x-x-109">group1</span>
     </li>
     <li class="itemize"><a 
 id="dx1-27003"></a><span 
class="ot1-zi4r-0-x-x-120">forceNoPBC: </span>see deﬁnition of <a 
href="#colvar|distance|oneSiteTotalForce"><span 
class="ot1-zi4r-0-x-x-109">forceNoPBC</span></a> in sec. <a 
href="#colvar|distance|oneSiteTotalForce">6.1.1<!--tex4ht:ref: colvar|distance|oneSiteTotalForce --></a> (<span 
class="ot1-zi4r-0-x-x-109">distance </span>component) </li></ul>
<!--l. 1578--><p class="noindent" >This component returns a <span 
class="zptmcm7m-x-x-109">N</span><sub><span 
class="zptmcm7t-x-x-80">1</span></sub> <span 
class="zptmcm7y-x-x-109">×</span><span 
class="zptmcm7m-x-x-109">N</span><sub><span 
class="zptmcm7t-x-x-80">2</span></sub>-dimensional vector of numbers, each ranging from <span 
class="zptmcm7t-x-x-109">0 </span>to the largest possible distance
within the chosen boundary conditions.
</p><!--l. 1581--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">6.1.8    </span> <a 
 id="x1-280006.1.8"></a><span 
class="ot1-zi4r-0-x-x-109">cartesian</span>: vector of atomic Cartesian coordinates.</h5>
<!--l. 1582--><p class="noindent" >The <span 
class="ot1-zi4r-0-x-x-109">cartesian</span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109">...</span><span 
class="cmsy-10x-x-109">} </span>block deﬁnes a component returning a ﬂat vector containing the Cartesian coordinates of all
participating atoms, in the order <span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">x</span><sub><span 
class="zptmcm7t-x-x-80">1</span></sub><span 
class="zptmcm7m-x-x-109">,y</span><sub><span 
class="zptmcm7t-x-x-80">1</span></sub><span 
class="zptmcm7m-x-x-109">,z</span><sub><span 
class="zptmcm7t-x-x-80">1</span></sub><span 
class="zptmcm7m-x-x-109">,</span><img 
src="colvars-refman-namd5x.png" alt="⋅⋅⋅"  class="@cdots"  /><span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7m-x-x-109">,x</span><sub><span 
class="zptmcm7m-x-x-80">n</span></sub><span 
class="zptmcm7m-x-x-109">,y</span><sub><span 
class="zptmcm7m-x-x-80">n</span></sub><span 
class="zptmcm7m-x-x-109">,z</span><sub><span 
class="zptmcm7m-x-x-80">n</span></sub><span 
class="zptmcm7t-x-x-109">)</span>.
</p><!--l. 1586--><p class="noindent" ><span 
class="ptmb7t-x-x-109">List of keywords </span>(see also <a 
href="#sec:cvc_superp">6.4<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-28001"></a> <span 
class="ot1-zi4r-0-x-x-120">atoms </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Group of atoms<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">cartesian</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>Block <span 
class="ot1-zi4r-0-x-x-109">atoms </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109">...</span><span 
class="cmsy-10x-x-109">}</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:    </span>Deﬁnes  the  atoms  whose  coordinates  make  up  the  value  of  the  component.  If
     <span 
class="ot1-zi4r-0-x-x-109">rotateReference </span>or <span 
class="ot1-zi4r-0-x-x-109">centerReference </span>are deﬁned, coordinates are evaluated within the moving
     frame of reference. </li></ul>
<!--l. 1599--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">6.1.9    </span> <a 
 id="x1-290006.1.9"></a><span 
class="ot1-zi4r-0-x-x-109">angle</span>: angle between three groups.</h5>
<!--l. 1600--><p class="noindent" >The <span 
class="ot1-zi4r-0-x-x-109">angle</span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109">...</span><span 
class="cmsy-10x-x-109">} </span>block deﬁnes an angle, and contains the three blocks <span 
class="ot1-zi4r-0-x-x-109">group1</span>, <span 
class="ot1-zi4r-0-x-x-109">group2 </span>and <span 
class="ot1-zi4r-0-x-x-109">group3</span>, deﬁning the
three groups. It returns an angle (in degrees) within the interval <span 
class="zptmcm7t-x-x-109">[0 : 180]</span>.
</p><!--l. 1605--><p class="noindent" ><span 
class="ptmb7t-x-x-109">List of keywords </span>(see also <a 
href="#sec:cvc_superp">6.4<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-29001"></a> <span 
class="ot1-zi4r-0-x-x-120">group1: </span>see deﬁnition of <a 
href="#colvar|distance|group1"><span 
class="ot1-zi4r-0-x-x-109">group1</span></a> in sec. <a 
href="#colvar|distance|oneSiteTotalForce">6.1.1<!--tex4ht:ref: colvar|distance|group1 --></a> (<span 
class="ot1-zi4r-0-x-x-109">distance </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-29002"></a> <span 
class="ot1-zi4r-0-x-x-120">group2: </span>analogous to <span 
class="ot1-zi4r-0-x-x-109">group1</span>
     </li>
     <li class="itemize"><a 
 id="dx1-29003"></a><span 
class="ot1-zi4r-0-x-x-120">group3: </span>analogous to <span 
class="ot1-zi4r-0-x-x-109">group1</span>
     </li>
     <li class="itemize"><a 
 id="dx1-29004"></a><span 
class="ot1-zi4r-0-x-x-120">forceNoPBC: </span>see deﬁnition of <a 
href="#colvar|distance|oneSiteTotalForce"><span 
class="ot1-zi4r-0-x-x-109">forceNoPBC</span></a> in sec. <a 
href="#colvar|distance|oneSiteTotalForce">6.1.1<!--tex4ht:ref: colvar|distance|oneSiteTotalForce --></a> (<span 
class="ot1-zi4r-0-x-x-109">distance </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-29005"></a> <span 
class="ot1-zi4r-0-x-x-120">oneSiteTotalForce: </span>see deﬁnition of <a 
href="#colvar|distance|oneSiteTotalForce"><span 
class="ot1-zi4r-0-x-x-109">oneSiteTotalForce</span></a> in sec. <a 
href="#colvar|distance|oneSiteTotalForce">6.1.1<!--tex4ht:ref: colvar|distance|oneSiteTotalForce --></a> (<span 
class="ot1-zi4r-0-x-x-109">distance </span>component)
     </li></ul>
                                                                                         
                                                                                         
<!--l. 1620--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">6.1.10    </span> <a 
 id="x1-300006.1.10"></a><span 
class="ot1-zi4r-0-x-x-109">dipoleAngle</span>: angle between two groups and dipole of a third group.</h5>
<!--l. 1621--><p class="noindent" >The <span 
class="ot1-zi4r-0-x-x-109">dipoleAngle</span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109">...</span><span 
class="cmsy-10x-x-109">} </span>block deﬁnes an angle, and contains the three blocks <span 
class="ot1-zi4r-0-x-x-109">group1</span>, <span 
class="ot1-zi4r-0-x-x-109">group2 </span>and <span 
class="ot1-zi4r-0-x-x-109">group3</span>,
deﬁning the three groups, being <span 
class="ot1-zi4r-0-x-x-109">group1 </span>the group where dipole is calculated. It returns an angle (in degrees) within
the interval <span 
class="zptmcm7t-x-x-109">[0 : 180]</span>.
</p><!--l. 1626--><p class="noindent" ><span 
class="ptmb7t-x-x-109">List of keywords </span>(see also <a 
href="#sec:cvc_superp">6.4<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-30001"></a> <span 
class="ot1-zi4r-0-x-x-120">group1: </span>see deﬁnition of <a 
href="#colvar|distance|group1"><span 
class="ot1-zi4r-0-x-x-109">group1</span></a> in sec. <a 
href="#colvar|distance|oneSiteTotalForce">6.1.1<!--tex4ht:ref: colvar|distance|group1 --></a> (<span 
class="ot1-zi4r-0-x-x-109">distance </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-30002"></a> <span 
class="ot1-zi4r-0-x-x-120">group2: </span>analogous to <span 
class="ot1-zi4r-0-x-x-109">group1</span>
     </li>
     <li class="itemize"><a 
 id="dx1-30003"></a><span 
class="ot1-zi4r-0-x-x-120">group3: </span>analogous to <span 
class="ot1-zi4r-0-x-x-109">group1</span>
     </li>
     <li class="itemize"><a 
 id="dx1-30004"></a><span 
class="ot1-zi4r-0-x-x-120">forceNoPBC: </span>see deﬁnition of <a 
href="#colvar|distance|oneSiteTotalForce"><span 
class="ot1-zi4r-0-x-x-109">forceNoPBC</span></a> in sec. <a 
href="#colvar|distance|oneSiteTotalForce">6.1.1<!--tex4ht:ref: colvar|distance|oneSiteTotalForce --></a> (<span 
class="ot1-zi4r-0-x-x-109">distance </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-30005"></a> <span 
class="ot1-zi4r-0-x-x-120">oneSiteTotalForce: </span>see deﬁnition of <a 
href="#colvar|distance|oneSiteTotalForce"><span 
class="ot1-zi4r-0-x-x-109">oneSiteTotalForce</span></a> in sec. <a 
href="#colvar|distance|oneSiteTotalForce">6.1.1<!--tex4ht:ref: colvar|distance|oneSiteTotalForce --></a> (<span 
class="ot1-zi4r-0-x-x-109">distance </span>component)
     </li></ul>
<!--l. 1640--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">6.1.11    </span> <a 
 id="x1-310006.1.11"></a><span 
class="ot1-zi4r-0-x-x-109">dihedral</span>: torsional angle between four groups.</h5>
<!--l. 1641--><p class="noindent" >The <span 
class="ot1-zi4r-0-x-x-109">dihedral</span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109">...</span><span 
class="cmsy-10x-x-109">} </span>block deﬁnes a torsional angle, and contains the blocks <span 
class="ot1-zi4r-0-x-x-109">group1</span>, <span 
class="ot1-zi4r-0-x-x-109">group2</span>, <span 
class="ot1-zi4r-0-x-x-109">group3 </span>and
<span 
class="ot1-zi4r-0-x-x-109">group4</span>, deﬁning the four groups. It returns an angle (in degrees) within the interval <span 
class="zptmcm7t-x-x-109">[</span><span 
class="zptmcm7y-x-x-109">−</span><span 
class="zptmcm7t-x-x-109">180 : 180]</span>. The
Colvars module calculates all the distances between two angles taking into account periodicity. For
instance, reference values for restraints or range boundaries can be deﬁned by using any real number of
choice.
</p><!--l. 1649--><p class="noindent" ><span 
class="ptmb7t-x-x-109">List of keywords </span>(see also <a 
href="#sec:cvc_superp">6.4<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-31001"></a> <span 
class="ot1-zi4r-0-x-x-120">group1: </span>see deﬁnition of <a 
href="#colvar|distance|group1"><span 
class="ot1-zi4r-0-x-x-109">group1</span></a> in sec. <a 
href="#colvar|distance|oneSiteTotalForce">6.1.1<!--tex4ht:ref: colvar|distance|group1 --></a> (<span 
class="ot1-zi4r-0-x-x-109">distance </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-31002"></a> <span 
class="ot1-zi4r-0-x-x-120">group2: </span>analogous to <span 
class="ot1-zi4r-0-x-x-109">group1</span>
     </li>
     <li class="itemize"><a 
 id="dx1-31003"></a><span 
class="ot1-zi4r-0-x-x-120">group3: </span>analogous to <span 
class="ot1-zi4r-0-x-x-109">group1</span>
     </li>
     <li class="itemize"><a 
 id="dx1-31004"></a><span 
class="ot1-zi4r-0-x-x-120">group4: </span>analogous to <span 
class="ot1-zi4r-0-x-x-109">group1</span>
     </li>
     <li class="itemize"><a 
 id="dx1-31005"></a><span 
class="ot1-zi4r-0-x-x-120">forceNoPBC: </span>see deﬁnition of <a 
href="#colvar|distance|oneSiteTotalForce"><span 
class="ot1-zi4r-0-x-x-109">forceNoPBC</span></a> in sec. <a 
href="#colvar|distance|oneSiteTotalForce">6.1.1<!--tex4ht:ref: colvar|distance|oneSiteTotalForce --></a> (<span 
class="ot1-zi4r-0-x-x-109">distance </span>component)
                                                                                         
                                                                                         
     </li>
     <li class="itemize"><a 
 id="dx1-31006"></a> <span 
class="ot1-zi4r-0-x-x-120">oneSiteTotalForce: </span>see deﬁnition of <a 
href="#colvar|distance|oneSiteTotalForce"><span 
class="ot1-zi4r-0-x-x-109">oneSiteTotalForce</span></a> in sec. <a 
href="#colvar|distance|oneSiteTotalForce">6.1.1<!--tex4ht:ref: colvar|distance|oneSiteTotalForce --></a> (<span 
class="ot1-zi4r-0-x-x-109">distance </span>component)
     </li></ul>
<!--l. 1665--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">6.1.12    </span> <a 
 id="x1-320006.1.12"></a><span 
class="ot1-zi4r-0-x-x-109">polarTheta</span>: polar angle in spherical coordinates.</h5>
<!--l. 1666--><p class="noindent" >The <span 
class="ot1-zi4r-0-x-x-109">polarTheta</span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109">...</span><span 
class="cmsy-10x-x-109">} </span>block deﬁnes the polar angle in spherical coordinates, for the center of mass of a group of
atoms described by the block <span 
class="ot1-zi4r-0-x-x-109">atoms</span>. It returns an angle (in degrees) within the interval <span 
class="zptmcm7t-x-x-109">[0 : 180]</span>. To obtain spherical
coordinates in a frame of reference tied to another group of atoms, use the <span 
class="ot1-zi4r-0-x-x-109">fittingGroup</span><a 
href="#key:colvars:atom_group:refPositions">5.2<!--tex4ht:ref: key:colvars:atom_group:refPositionsFile --></a> option within the
<span 
class="ot1-zi4r-0-x-x-109">atoms </span>block.
</p><!--l. 1674--><p class="noindent" ><span 
class="ptmb7t-x-x-109">List of keywords </span>(see also <a 
href="#sec:cvc_superp">6.4<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="colvar|polarTheta|atoms"></a> <a 
 id="dx1-32001"></a><span 
class="ot1-zi4r-0-x-x-120">atoms </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Atom group<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">polarPhi</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span><span 
class="ot1-zi4r-0-x-x-109">atoms</span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109">...</span><span 
class="cmsy-10x-x-109">} </span>block <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>Deﬁnes the group of atoms for the COM of which the angle should be calculated.  </li></ul>
<!--l. 1688--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">6.1.13    </span> <a 
 id="x1-330006.1.13"></a><span 
class="ot1-zi4r-0-x-x-109">polarPhi</span>: azimuthal angle in spherical coordinates.</h5>
<!--l. 1689--><p class="noindent" >The <span 
class="ot1-zi4r-0-x-x-109">polarPhi</span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109">...</span><span 
class="cmsy-10x-x-109">} </span>block deﬁnes the azimuthal angle in spherical coordinates, for the center of mass of a group
of atoms described by the block <span 
class="ot1-zi4r-0-x-x-109">atoms</span>. It returns an angle (in degrees) within the interval <span 
class="zptmcm7t-x-x-109">[</span><span 
class="zptmcm7y-x-x-109">−</span><span 
class="zptmcm7t-x-x-109">180 : 180]</span>. The
Colvars module calculates all the distances between two angles taking into account periodicity. For
instance, reference values for restraints or range boundaries can be deﬁned by using any real number of
choice.
</p><!--l. 1697--><p class="noindent" ><span 
class="ptmb7t-x-x-109">List of keywords </span>(see also <a 
href="#sec:cvc_superp">6.4<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="colvar|polarPhi|atoms"></a> <a 
 id="dx1-33001"></a><span 
class="ot1-zi4r-0-x-x-120">atoms </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Atom group<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">polarPhi</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span><span 
class="ot1-zi4r-0-x-x-109">atoms</span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109">...</span><span 
class="cmsy-10x-x-109">} </span>block <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>Deﬁnes the group of atoms for the COM of which the angle should be calculated.  </li></ul>
<!--l. 1709--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">6.1.14    </span> <a 
 id="colvar|coordNum|cutoff"></a><span 
class="ot1-zi4r-0-x-x-109">coordNum</span>: coordination number between two groups.</h5>
<!--l. 1710--><p class="noindent" >The <span 
class="ot1-zi4r-0-x-x-109">coordNum </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109">...</span><span 
class="cmsy-10x-x-109">} </span>block deﬁnes a coordination number (or number of contacts), which calculates the function
<span 
class="zptmcm7t-x-x-109">(1</span><span 
class="zptmcm7y-x-x-109">−</span><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">d∕d</span><sub><span 
class="zptmcm7t-x-x-80">0</span></sub><span 
class="zptmcm7t-x-x-109">)</span><sup><span 
class="zptmcm7m-x-x-80">n</span></sup><span 
class="zptmcm7t-x-x-109">)</span><span 
class="zptmcm7m-x-x-109">∕</span><span 
class="zptmcm7t-x-x-109">(1</span><span 
class="zptmcm7y-x-x-109">−</span><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">d∕d</span><sub><span 
class="zptmcm7t-x-x-80">0</span></sub><span 
class="zptmcm7t-x-x-109">)</span><sup><span 
class="zptmcm7m-x-x-80">m</span></sup><span 
class="zptmcm7t-x-x-109">)</span>, where <span 
class="zptmcm7m-x-x-109">d</span><sub><span 
class="zptmcm7t-x-x-80">0</span></sub> is the “cutoﬀ” distance, and <span 
class="zptmcm7m-x-x-109">n </span>and <span 
class="zptmcm7m-x-x-109">m </span>are exponents that can control its
long range behavior and stiﬀness <span class="cite">[<a 
href="#XIannuzzi2003">2</a>]</span>. This function is summed over all pairs of atoms in <span 
class="ot1-zi4r-0-x-x-109">group1 </span>and
<span 
class="ot1-zi4r-0-x-x-109">group2</span>:
</p>
   <table 
class="equation"><tr><td><a 
 id="eq:cvc_coordNum"></a>
                                                                                         
                                                                                         
   <center class="math-display" >
<img 
src="colvars-refman-namd6x.png" alt="                                                n
C(group1,group2 )=    ∑     ∑    1-−-(|xi−-xj|∕d0)-
                    i∈group1j∈group21− (|xi− xj|∕d0)m
" class="math-display"  /></center></td><td class="equation-label">(3)</td></tr></table>
<!--l. 1724--><p class="nopar" >
</p><!--l. 1726--><p class="noindent" ><span 
class="ptmb7t-x-x-109">List of keywords </span>(see also <a 
href="#sec:cvc_superp">6.4<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="colvar|coordNum|cutoff"></a> <a 
 id="dx1-34002"></a><span 
class="ot1-zi4r-0-x-x-120">group1: </span>see deﬁnition of <a 
href="#colvar|distance|group1"><span 
class="ot1-zi4r-0-x-x-109">group1</span></a> in sec. <a 
href="#colvar|distance|oneSiteTotalForce">6.1.1<!--tex4ht:ref: colvar|distance|group1 --></a> (<span 
class="ot1-zi4r-0-x-x-109">distance </span>component)
     </li>
     <li class="itemize"><a 
 id="colvar|coordNum|group2"></a> <a 
 id="dx1-34003"></a><span 
class="ot1-zi4r-0-x-x-120">group2: </span>analogous to <span 
class="ot1-zi4r-0-x-x-109">group1</span>
     </li>
     <li class="itemize"><a 
 id="colvar|coordNum|cutoff"></a><a 
 id="dx1-34004"></a><span 
class="ot1-zi4r-0-x-x-120">cutoff </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>“Interaction” distance (Å)<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">coordNum</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive decimal <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span>4.0 <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>This number deﬁnes the switching distance to deﬁne an interatomic contact: for <span 
class="zptmcm7m-x-x-109">d </span><span 
class="zptmcm7y-x-x-109">≪</span><span 
class="zptmcm7m-x-x-109">d</span><sub><span 
class="zptmcm7t-x-x-80">0</span></sub>,
     the switching function <span 
class="zptmcm7t-x-x-109">(1</span><span 
class="zptmcm7y-x-x-109">−</span><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">d∕d</span><sub><span 
class="zptmcm7t-x-x-80">0</span></sub><span 
class="zptmcm7t-x-x-109">)</span><sup><span 
class="zptmcm7m-x-x-80">n</span></sup><span 
class="zptmcm7t-x-x-109">)</span><span 
class="zptmcm7m-x-x-109">∕</span><span 
class="zptmcm7t-x-x-109">(1</span><span 
class="zptmcm7y-x-x-109">−</span><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">d∕d</span><sub><span 
class="zptmcm7t-x-x-80">0</span></sub><span 
class="zptmcm7t-x-x-109">)</span><sup><span 
class="zptmcm7m-x-x-80">m</span></sup><span 
class="zptmcm7t-x-x-109">) </span>is close to 1, at <span 
class="zptmcm7m-x-x-109">d </span><span 
class="zptmcm7t-x-x-109">= </span><span 
class="zptmcm7m-x-x-109">d</span><sub><span 
class="zptmcm7t-x-x-80">0</span></sub> it has a value of <span 
class="zptmcm7m-x-x-109">n∕m </span>(<span 
class="zptmcm7t-x-x-109">1</span><span 
class="zptmcm7m-x-x-109">∕</span><span 
class="zptmcm7t-x-x-109">2</span>
     with the default <span 
class="zptmcm7m-x-x-109">n </span>and <span 
class="zptmcm7m-x-x-109">m</span>), and at <span 
class="zptmcm7m-x-x-109">d </span><span 
class="zptmcm7y-x-x-109">≫ </span><span 
class="zptmcm7m-x-x-109">d</span><sub><span 
class="zptmcm7t-x-x-80">0</span></sub> it goes to zero approximately like <span 
class="zptmcm7m-x-x-109">d</span><sup><span 
class="zptmcm7m-x-x-80">m</span><span 
class="zptmcm7y-x-x-80">−</span><span 
class="zptmcm7m-x-x-80">n</span></sup>. Hence, for a proper
     behavior, <span 
class="zptmcm7m-x-x-109">m </span>must be larger than <span 
class="zptmcm7m-x-x-109">n</span>.
     </li>
     <li class="itemize"><a 
 id="colvar|coordNum|cutoff3"></a> <a 
 id="dx1-34005"></a><span 
class="ot1-zi4r-0-x-x-120">cutoff3 </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Reference distance vector (Å)<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">coordNum</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>“<span 
class="ot1-zi4r-0-x-x-109">(x, y, z)</span>” triplet of positive decimals <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">(4.0, 4.0, 4.0)</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>The three components of this vector deﬁne three diﬀerent cutoﬀs <span 
class="zptmcm7m-x-x-109">d</span><sub><span 
class="zptmcm7t-x-x-80">0</span></sub> for each direction.
     This option is mutually exclusive with <span 
class="ot1-zi4r-0-x-x-109">cutoff</span>.
     </li>
     <li class="itemize"><a 
 id="colvar|coordNum|expNumer"></a> <a 
 id="dx1-34006"></a><span 
class="ot1-zi4r-0-x-x-120">expNumer </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Numerator exponent<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">coordNum</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive even integer <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span>6 <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>This number deﬁnes the <span 
class="zptmcm7m-x-x-109">n </span>exponent for the switching function.
     </li>
     <li class="itemize"><a 
 id="colvar|coordNum|expDenom"></a> <a 
 id="dx1-34007"></a><span 
class="ot1-zi4r-0-x-x-120">expDenom </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Denominator exponent<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">coordNum</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive even integer <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span>12 <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>This number deﬁnes the <span 
class="zptmcm7m-x-x-109">m </span>exponent for the switching function.
                                                                                         
                                                                                         
     </li>
     <li class="itemize"><a 
 id="colvar|coordNum|group2CenterOnly"></a> <a 
 id="dx1-34008"></a><span 
class="ot1-zi4r-0-x-x-120">group2CenterOnly </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Use only <span 
class="ot1-zi4r-0-x-x-109">group2</span>&#x2019;s center of mass<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">coordNum</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">off</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>If this option is <span 
class="ot1-zi4r-0-x-x-109">on</span>, only contacts between each atoms in <span 
class="ot1-zi4r-0-x-x-109">group1 </span>and the center of mass
     of <span 
class="ot1-zi4r-0-x-x-109">group2 </span>are calculated (by default, the sum extends over all pairs of atoms in <span 
class="ot1-zi4r-0-x-x-109">group1 </span>and <span 
class="ot1-zi4r-0-x-x-109">group2</span>).
     If <span 
class="ot1-zi4r-0-x-x-109">group2 </span>is a <span 
class="ot1-zi4r-0-x-x-109">dummyAtom</span>, this option is set to <span 
class="ot1-zi4r-0-x-x-109">yes </span>by default.   </li></ul>
<!--l. 1797--><p class="indent" >   This component returns a dimensionless number, which ranges from approximately 0 (all interatomic distances
are much larger than the cutoﬀ) to <span 
class="zptmcm7m-x-x-109">N</span><sub><span 
class="cmtt-8">group1</span></sub><span 
class="zptmcm7y-x-x-109">×</span><span 
class="zptmcm7m-x-x-109">N</span><sub><span 
class="cmtt-8">group2</span></sub> (all distances are less than the cutoﬀ), or <span 
class="zptmcm7m-x-x-109">N</span><sub><span 
class="cmtt-8">group1</span></sub>
if <span 
class="ot1-zi4r-0-x-x-109">group2CenterOnly </span>is used. For performance reasons, at least one of <span 
class="ot1-zi4r-0-x-x-109">group1 </span>and <span 
class="ot1-zi4r-0-x-x-109">group2 </span>should
be of limited size or <span 
class="ot1-zi4r-0-x-x-109">group2CenterOnly </span>should be used: the cost of the loop over all pairs grows as
<span 
class="zptmcm7m-x-x-109">N</span><sub><span 
class="cmtt-8">group1</span></sub><span 
class="zptmcm7y-x-x-109">×</span><span 
class="zptmcm7m-x-x-109">N</span><sub><span 
class="cmtt-8">group2</span></sub>.
</p><!--l. 1806--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">6.1.15    </span> <a 
 id="x1-350006.1.15"></a><span 
class="ot1-zi4r-0-x-x-109">selfCoordNum</span>: coordination number between atoms within a group.</h5>
<!--l. 1807--><p class="noindent" >The <span 
class="ot1-zi4r-0-x-x-109">selfCoordNum </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109">...</span><span 
class="cmsy-10x-x-109">} </span>block deﬁnes a coordination number similarly to the component <span 
class="ot1-zi4r-0-x-x-109">coordNum</span>, but the
function is summed over atom pairs within <span 
class="ot1-zi4r-0-x-x-109">group1</span>:
</p>
   <table 
class="equation"><tr><td><a 
 id="x1-35001r4"></a>
   <center class="math-display" >
<img 
src="colvars-refman-namd7x.png" alt="                     1-−-(|xi−-xj|∕d0)n
C(group1) =   ∑    ∑ 1 − (|xi− xj|∕d0)m
            i∈group1j&#x003E;i
" class="math-display"  /></center></td><td class="equation-label">(4)</td></tr></table>
<!--l. 1817--><p class="nopar" >
The keywords accepted by <span 
class="ot1-zi4r-0-x-x-109">selfCoordNum </span>are a subset of those accepted by <span 
class="ot1-zi4r-0-x-x-109">coordNum</span>, namely <span 
class="ot1-zi4r-0-x-x-109">group1 </span>(here
deﬁning <span 
class="ptmri7t-x-x-109">all </span>of the atoms to be considered), <span 
class="ot1-zi4r-0-x-x-109">cutoff</span>, <span 
class="ot1-zi4r-0-x-x-109">expNumer</span>, and <span 
class="ot1-zi4r-0-x-x-109">expDenom</span>.
</p><!--l. 1823--><p class="noindent" ><span 
class="ptmb7t-x-x-109">List of keywords </span>(see also <a 
href="#sec:cvc_superp">6.4<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-35002"></a> <span 
class="ot1-zi4r-0-x-x-120">group1: </span>see deﬁnition of <a 
href="#colvar|coordNum|cutoff"><span 
class="ot1-zi4r-0-x-x-109">group1</span></a> in sec. <a 
href="#colvar|coordNum|cutoff">6.1.14<!--tex4ht:ref: colvar|coordNum|cutoff3 --></a> (<span 
class="ot1-zi4r-0-x-x-109">coordNum </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-35003"></a> <span 
class="ot1-zi4r-0-x-x-120">cutoff: </span>see deﬁnition of <a 
href="#colvar|coordNum|cutoff"><span 
class="ot1-zi4r-0-x-x-109">cutoff</span></a> in sec. <a 
href="#colvar|coordNum|cutoff">6.1.14<!--tex4ht:ref: colvar|coordNum|cutoff --></a> (<span 
class="ot1-zi4r-0-x-x-109">coordNum </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-35004"></a> <span 
class="ot1-zi4r-0-x-x-120">cutoff3: </span>see deﬁnition of <a 
href="#colvar|coordNum|cutoff3"><span 
class="ot1-zi4r-0-x-x-109">cutoff3</span></a> in sec. <a 
href="#colvar|coordNum|cutoff">6.1.14<!--tex4ht:ref: colvar|coordNum|cutoff3 --></a> (<span 
class="ot1-zi4r-0-x-x-109">coordNum </span>component)
                                                                                         
                                                                                         
     </li>
     <li class="itemize"><a 
 id="dx1-35005"></a> <span 
class="ot1-zi4r-0-x-x-120">expNumer: </span>see deﬁnition of <a 
href="#colvar|coordNum|expNumer"><span 
class="ot1-zi4r-0-x-x-109">expNumer</span></a> in sec. <a 
href="#colvar|coordNum|cutoff">6.1.14<!--tex4ht:ref: colvar|coordNum|expNumer --></a> (<span 
class="ot1-zi4r-0-x-x-109">coordNum </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-35006"></a> <span 
class="ot1-zi4r-0-x-x-120">expDenom: </span>see deﬁnition of <a 
href="#colvar|coordNum|expDenom"><span 
class="ot1-zi4r-0-x-x-109">expDenom</span></a> in sec. <a 
href="#colvar|coordNum|cutoff">6.1.14<!--tex4ht:ref: colvar|coordNum|expDenom --></a> (<span 
class="ot1-zi4r-0-x-x-109">coordNum </span>component) </li></ul>
<!--l. 1836--><p class="indent" >   This component returns a dimensionless number, which ranges from approximately 0 (all interatomic
distances much larger than the cutoﬀ) to <span 
class="zptmcm7m-x-x-109">N</span><sub><span 
class="cmtt-8">group1</span></sub><span 
class="zptmcm7y-x-x-109">×</span><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">N</span><sub><span 
class="cmtt-8">group1</span></sub><span 
class="zptmcm7y-x-x-109">−</span><span 
class="zptmcm7t-x-x-109">1)</span><span 
class="zptmcm7m-x-x-109">∕</span><span 
class="zptmcm7t-x-x-109">2 </span>(all distances within the cutoﬀ). For
performance reasons, <span 
class="ot1-zi4r-0-x-x-109">group1 </span>should be of limited size, because the cost of the loop over all pairs grows as
<span 
class="zptmcm7m-x-x-109">N</span><sub><span 
class="cmtt-8">group1</span></sub><sup><span 
class="zptmcm7t-x-x-80">2</span></sup>.
</p><!--l. 1846--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">6.1.16    </span> <a 
 id="x1-360006.1.16"></a><span 
class="ot1-zi4r-0-x-x-109">hBond</span>: hydrogen bond between two atoms.</h5>
<!--l. 1846--><p class="noindent" >The <span 
class="ot1-zi4r-0-x-x-109">hBond </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109">...</span><span 
class="cmsy-10x-x-109">} </span>block deﬁnes a hydrogen bond, implemented as a coordination number (eq. <a 
href="#eq:cvc_coordNum">3<!--tex4ht:ref: eq:cvc_coordNum --></a>) between the donor
and the acceptor atoms. Therefore, it accepts the same options <span 
class="ot1-zi4r-0-x-x-109">cutoff </span>(with a diﬀerent default value of 3.3 Å),
<span 
class="ot1-zi4r-0-x-x-109">expNumer </span>(with a default value of 6) and <span 
class="ot1-zi4r-0-x-x-109">expDenom </span>(with a default value of 8). Unlike <span 
class="ot1-zi4r-0-x-x-109">coordNum</span>, it requires two
atom numbers, <span 
class="ot1-zi4r-0-x-x-109">acceptor </span>and <span 
class="ot1-zi4r-0-x-x-109">donor</span>, to be deﬁned. It returns an adimensional number, with values between 0
(acceptor and donor far outside the cutoﬀ distance) and 1 (acceptor and donor much closer than the
cutoﬀ).
</p><!--l. 1857--><p class="noindent" ><span 
class="ptmb7t-x-x-109">List of keywords </span>(see also <a 
href="#sec:cvc_superp">6.4<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-36001"></a> <span 
class="ot1-zi4r-0-x-x-120">acceptor </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Number of the acceptor atom<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">hBond</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive integer <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>Number that uses the same convention as <span 
class="ot1-zi4r-0-x-x-109">atomNumbers</span>.
     </li>
     <li class="itemize"><a 
 id="dx1-36002"></a> <span 
class="ot1-zi4r-0-x-x-120">donor: </span>analogous to <span 
class="ot1-zi4r-0-x-x-109">acceptor</span>
     </li>
     <li class="itemize"><a 
 id="dx1-36003"></a><span 
class="ot1-zi4r-0-x-x-120">cutoff: </span>see deﬁnition of <a 
href="#colvar|coordNum|cutoff"><span 
class="ot1-zi4r-0-x-x-109">cutoff</span></a> in sec. <a 
href="#colvar|coordNum|cutoff">6.1.14<!--tex4ht:ref: colvar|coordNum|cutoff --></a> (<span 
class="ot1-zi4r-0-x-x-109">coordNum </span>component)<br 
class="newline" /><span 
class="ptmb7t-x-x-109">Note: </span>default value is 3.3 Å.
     </li>
     <li class="itemize"><a 
 id="dx1-36004"></a> <span 
class="ot1-zi4r-0-x-x-120">expNumer: </span>see deﬁnition of <a 
href="#colvar|coordNum|expNumer"><span 
class="ot1-zi4r-0-x-x-109">expNumer</span></a> in sec. <a 
href="#colvar|coordNum|cutoff">6.1.14<!--tex4ht:ref: colvar|coordNum|expNumer --></a> (<span 
class="ot1-zi4r-0-x-x-109">coordNum </span>component)<br 
class="newline" /><span 
class="ptmb7t-x-x-109">Note: </span>default value is 6.
     </li>
     <li class="itemize"><a 
 id="dx1-36005"></a> <span 
class="ot1-zi4r-0-x-x-120">expDenom: </span>see deﬁnition of <a 
href="#colvar|coordNum|expDenom"><span 
class="ot1-zi4r-0-x-x-109">expDenom</span></a> in sec. <a 
href="#colvar|coordNum|cutoff">6.1.14<!--tex4ht:ref: colvar|coordNum|expDenom --></a> (<span 
class="ot1-zi4r-0-x-x-109">coordNum </span>component)<br 
class="newline" /><span 
class="ptmb7t-x-x-109">Note: </span>default value is 8. </li></ul>
<!--l. 1878--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">6.1.17    </span> <a 
 id="colvar|rmsd|refPositions"></a><span 
class="ot1-zi4r-0-x-x-109">rmsd</span>: root mean square displacement (RMSD) from reference positions.</h5>
                                                                                         
                                                                                         
<!--l. 1879--><p class="noindent" >The block <span 
class="ot1-zi4r-0-x-x-109">rmsd</span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109">...</span><span 
class="cmsy-10x-x-109">} </span>deﬁnes the root mean square replacement (RMSD) of a group of atoms with respect to
a reference structure. For each set of coordinates <span 
class="zptmcm7y-x-x-109">{</span><span 
class="ptmb7t-x-x-109">x</span><sub><span 
class="zptmcm7t-x-x-80">1</span></sub><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">t</span><span 
class="zptmcm7t-x-x-109">)</span><span 
class="zptmcm7m-x-x-109">,</span><span 
class="ptmb7t-x-x-109">x</span><sub><span 
class="zptmcm7t-x-x-80">2</span></sub><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">t</span><span 
class="zptmcm7t-x-x-109">)</span><span 
class="zptmcm7m-x-x-109">,</span><span 
class="zptmcm7m-x-x-109">…</span><span 
class="ptmb7t-x-x-109">x</span><sub><span 
class="zptmcm7m-x-x-80">N</span></sub><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">t</span><span 
class="zptmcm7t-x-x-109">)</span><span 
class="zptmcm7y-x-x-109">}</span>, the colvar component <span 
class="ot1-zi4r-0-x-x-109">rmsd</span>
calculates the optimal rotation <span 
class="zptmcm7m-x-x-109">U</span><sup><span 
class="zptmcm7y-x-x-80">{</span><span 
class="ptmb7t-x-x-80">x</span><sub><span 
class="zptmcm7m-x-x-60">i</span></sub><span 
class="zptmcm7t-x-x-80">(</span><span 
class="zptmcm7m-x-x-80">t</span><span 
class="zptmcm7t-x-x-80">)</span><span 
class="zptmcm7y-x-x-80">}→{</span><span 
class="ptmb7t-x-x-80">x</span><sub><span 
class="zptmcm7m-x-x-60">i</span></sub><sup><span 
class="zptmcm7t-x-x-60">(ref)</span></sup><span 
class="zptmcm7y-x-x-80">}</span></sup> that best superimposes the coordinates <span 
class="zptmcm7y-x-x-109">{</span><span 
class="ptmb7t-x-x-109">x</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">t</span><span 
class="zptmcm7t-x-x-109">)</span><span 
class="zptmcm7y-x-x-109">} </span>onto a
set of reference coordinates <span 
class="zptmcm7y-x-x-109">{</span><span 
class="ptmb7t-x-x-109">x</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub><sup><span 
class="zptmcm7t-x-x-80">(ref)</span></sup><span 
class="zptmcm7y-x-x-109">}</span>. Both the current and the reference coordinates are centered on
their centers of geometry, <span 
class="ptmb7t-x-x-109">x</span><sub><span 
class="zptmcm7t-x-x-80">cog</span></sub><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">t</span><span 
class="zptmcm7t-x-x-109">) </span>and <span 
class="ptmb7t-x-x-109">x</span><sub><span 
class="zptmcm7t-x-x-80">cog</span></sub><sup><span 
class="zptmcm7t-x-x-80">(ref)</span></sup>. The root mean square displacement is then deﬁned
as:
</p>
   <table 
class="equation"><tr><td><a 
 id="x1-37001r5"></a>
   <center class="math-display" >
<img 
src="colvars-refman-namd8x.png" alt="                       ∘ ---N-|-----------------(-----------)|-
RMSD  ({xi(t)},{x(ref)})=    1-  ||U (xi(t)− xcog(t)) −  x(ref)− x(creogf) ||2
                i         N ∑i=1                     i
" class="math-display"  /></center></td><td class="equation-label">(5)</td></tr></table>
<!--l. 1900--><p class="nopar" >
The optimal rotation <span 
class="zptmcm7m-x-x-109">U</span><sup><span 
class="zptmcm7y-x-x-80">{</span><span 
class="ptmb7t-x-x-80">x</span><sub><span 
class="zptmcm7m-x-x-60">i</span></sub><span 
class="zptmcm7t-x-x-80">(</span><span 
class="zptmcm7m-x-x-80">t</span><span 
class="zptmcm7t-x-x-80">)</span><span 
class="zptmcm7y-x-x-80">}→{</span><span 
class="ptmb7t-x-x-80">x</span><sub><span 
class="zptmcm7m-x-x-60">i</span></sub><sup><span 
class="zptmcm7t-x-x-60">(ref)</span></sup><span 
class="zptmcm7y-x-x-80">}</span></sup> is calculated within the formalism developed in reference <span class="cite">[<a 
href="#XCoutsias2004">3</a>]</span>, which
guarantees a continuous dependence of <span 
class="zptmcm7m-x-x-109">U</span><sup><span 
class="zptmcm7y-x-x-80">{</span><span 
class="ptmb7t-x-x-80">x</span><sub><span 
class="zptmcm7m-x-x-60">i</span></sub><span 
class="zptmcm7t-x-x-80">(</span><span 
class="zptmcm7m-x-x-80">t</span><span 
class="zptmcm7t-x-x-80">)</span><span 
class="zptmcm7y-x-x-80">}→{</span><span 
class="ptmb7t-x-x-80">x</span><sub><span 
class="zptmcm7m-x-x-60">i</span></sub><sup><span 
class="zptmcm7t-x-x-60">(ref)</span></sup><span 
class="zptmcm7y-x-x-80">}</span></sup> with respect to <span 
class="zptmcm7y-x-x-109">{</span><span 
class="ptmb7t-x-x-109">x</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">t</span><span 
class="zptmcm7t-x-x-109">)</span><span 
class="zptmcm7y-x-x-109">}</span>.
</p><!--l. 1909--><p class="noindent" ><span 
class="ptmb7t-x-x-109">List of keywords </span>(see also <a 
href="#sec:cvc_superp">6.4<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="colvar|rmsd|refPositions"></a> <a 
 id="dx1-37002"></a><span 
class="ot1-zi4r-0-x-x-120">atoms </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Atom group<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">rmsd</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span><span 
class="ot1-zi4r-0-x-x-109">atoms</span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109">...</span><span 
class="cmsy-10x-x-109">} </span>block <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>Deﬁnes  the  group  of  atoms  of  which  the  RMSD  should  be  calculated.  Optimal  ﬁt
     options  (such  as  <span 
class="ot1-zi4r-0-x-x-109">refPositions </span>and  <span 
class="ot1-zi4r-0-x-x-109">rotateReference</span>)  should  typically  NOT  be  set  within  this
     block. Exceptions to this rule are the special cases discussed in the <span 
class="ptmri7t-x-x-109">Advanced usage </span>paragraph below.
     </li>
     <li class="itemize"><a 
 id="colvar|rmsd|refPositions"></a> <a 
 id="dx1-37003"></a><span 
class="ot1-zi4r-0-x-x-120">refPositions </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Reference coordinates<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">rmsd</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>space-separated list of <span 
class="ot1-zi4r-0-x-x-109">(x, y, z) </span>triplets <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:     </span>This   option   (mutually   exclusive   with   <span 
class="ot1-zi4r-0-x-x-109">refPositionsFile</span>)   sets   the   reference
     coordinates.  If  only  <span 
class="ot1-zi4r-0-x-x-109">centerReference </span>is  <span 
class="ot1-zi4r-0-x-x-109">on</span>,  the  list  can  be  a  single  (x,  y,  z)  triplet;  if  also
     <span 
class="ot1-zi4r-0-x-x-109">rotateReference </span>is <span 
class="ot1-zi4r-0-x-x-109">on</span>, the list should be as long as the atom group. This option is independent from
     that with the same keyword within the <span 
class="ot1-zi4r-0-x-x-109">atoms</span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109">...</span><span 
class="cmsy-10x-x-109">} </span>block (see <a 
href="#key:colvars:atom_group:refPositions">5.2<!--tex4ht:ref: key:colvars:atom_group:refPositions --></a>). The latter (and related ﬁtting
     options for the atom group) are normally not needed, and should be omitted altogether except for
     advanced usage cases.
     </li>
     <li class="itemize"><a 
 id="colvar|rmsd|refPositionsFile"></a> <a 
 id="dx1-37004"></a><span 
class="ot1-zi4r-0-x-x-120">refPositionsFile </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Reference coordinates ﬁle<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">rmsd</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>UNIX ﬁlename <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>This option (mutually exclusive with <span 
class="ot1-zi4r-0-x-x-109">refPositions</span>) sets the ﬁle name for the reference
                                                                                         
                                                                                         
     coordinates to be compared with. The format is the same as that provided by <span 
class="ot1-zi4r-0-x-x-109">refPositionsFile</span>
     within an atom group&#x2019;s deﬁnition (see <a 
href="#key:colvars:atom_group:refPositions">5.2<!--tex4ht:ref: key:colvars:atom_group:refPositionsFile --></a>).
     </li>
     <li class="itemize"><a 
 id="colvar|rmsd|refPositionsCol"></a> <a 
 id="dx1-37005"></a><span 
class="ot1-zi4r-0-x-x-120">refPositionsCol </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>PDB column containing atom ﬂags<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">rmsd</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span><span 
class="ot1-zi4r-0-x-x-109">O</span>, <span 
class="ot1-zi4r-0-x-x-109">B</span>, <span 
class="ot1-zi4r-0-x-x-109">X</span>, <span 
class="ot1-zi4r-0-x-x-109">Y</span>, or <span 
class="ot1-zi4r-0-x-x-109">Z</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>If <span 
class="ot1-zi4r-0-x-x-109">refPositionsFile </span>is a PDB ﬁle that contains all the atoms in the topology, this
     option may be provided to set which PDB ﬁeld is used to ﬂag the reference coordinates for <span 
class="ot1-zi4r-0-x-x-109">atoms</span>.
     </li>
     <li class="itemize"><a 
 id="colvar|rmsd|refPositionsColValue"></a> <a 
 id="dx1-37006"></a><span 
class="ot1-zi4r-0-x-x-120">refPositionsColValue </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Atom selection ﬂag in the PDB column<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">rmsd</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive decimal <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:    </span>If  deﬁned,  this  value  identiﬁes  in  the  PDB  column  <span 
class="ot1-zi4r-0-x-x-109">refPositionsCol </span>of  the  ﬁle
     <span 
class="ot1-zi4r-0-x-x-109">refPositionsFile </span>which atom positions are to be read. Otherwise, all positions with a non-zero
     value are read.   </li></ul>
<!--l. 1974--><p class="noindent" >This component returns a positive real number (in Å).
</p><!--l. 1976--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">6.1.18    </span> <a 
 id="x1-380006.1.18"></a>Advanced usage of the <span 
class="ot1-zi4r-0-x-x-109">rmsd </span>component.</h5>
<!--l. 1977--><p class="noindent" >In the standard usage as described above, the <span 
class="ot1-zi4r-0-x-x-109">rmsd </span>component calculates a minimum RMSD, that is, current
coordinates are optimally ﬁtted onto the same reference coordinates that are used to compute the RMSD value. The
ﬁt itself is handled by the atom group object, whose parameters are automatically set by the <span 
class="ot1-zi4r-0-x-x-109">rmsd </span>component. For
very speciﬁc applications, however, it may be useful to control the ﬁtting process separately from the deﬁnition of
the reference coordinates, to evaluate various types of non-minimal RMSD values. This can be achieved by setting
the related options (<span 
class="ot1-zi4r-0-x-x-109">refPositions</span>, etc.) explicitly in the atom group block. This allows for the following
non-standard cases:
</p><!--l. 1990--><p class="indent" >
     </p><ol  class="enumerate1" >
     <li 
  class="enumerate" id="x1-38002x1">applying the optimal translation, but no rotation (<span 
class="ot1-zi4r-0-x-x-109">rotateReference off</span>), to bias or restrain the shape
     and orientation, but not the position of the atom group;
     </li>
     <li 
  class="enumerate" id="x1-38004x2">applying the optimal rotation, but no translation (<span 
class="ot1-zi4r-0-x-x-109">translateReference off</span>), to bias or restrain the
     shape and position, but not the orientation of the atom group;
     </li>
     <li 
  class="enumerate" id="x1-38006x3">disabling the application of optimal roto-translations, which lets the RMSD component decribe the
     deviation of atoms from ﬁxed positions in the laboratory frame: this allows for custom positional
     restraints within the Colvars module;
     </li>
     <li 
  class="enumerate" id="x1-38008x4">ﬁtting the atomic positions to diﬀerent reference coordinates than those used in the RMSD calculation
     itself;
                                                                                         
                                                                                         
     </li>
     <li 
  class="enumerate" id="x1-38010x5">applying  the  optimal  rotation  and/or  translation  from  a  separate  atom  group,  deﬁned  through
     <span 
class="ot1-zi4r-0-x-x-109">fittingGroup</span>: the RMSD then reﬂects the deviation from reference coordinates in a separate, moving
     reference frame.</li></ol>
<!--l. 2010--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">6.1.19    </span> <a 
 id="x1-390006.1.19"></a><span 
class="ot1-zi4r-0-x-x-109">eigenvector</span>: projection of the atomic coordinates on a vector.</h5>
<!--l. 2011--><p class="noindent" >The block <span 
class="ot1-zi4r-0-x-x-109">eigenvector</span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109">...</span><span 
class="cmsy-10x-x-109">} </span>deﬁnes the projection of the coordinates of a group of atoms (or more precisely,
their deviations from the reference coordinates) onto a vector in <span 
class="msbm-10x-x-109">ℝ</span><sup><span 
class="zptmcm7t-x-x-80">3</span><span 
class="zptmcm7m-x-x-80">n</span></sup>, where <span 
class="zptmcm7m-x-x-109">n </span>is the number of atoms in the group.
The computed quantity is the total projection:
</p>
   <table 
class="equation"><tr><td><a 
 id="eq:cvc_eigenvector"></a>
   <center class="math-display" >
<img 
src="colvars-refman-namd9x.png" alt="          (ref)     n   (                   (ref)   (ref))
p({xi(t)},{xi  }) = ∑ vi⋅ U (xi(t)− xcog(t)) − (xi  − xcog ) ,
                  i=1
" class="math-display"  /></center></td><td class="equation-label">(6)</td></tr></table>
<!--l. 2024--><p class="nopar" >
where, as in the <span 
class="ot1-zi4r-0-x-x-109">rmsd </span>component, <span 
class="zptmcm7m-x-x-109">U </span>is the optimal rotation matrix, <span 
class="ptmb7t-x-x-109">x</span><sub><span 
class="zptmcm7t-x-x-80">cog</span></sub><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">t</span><span 
class="zptmcm7t-x-x-109">) </span>and <span 
class="ptmb7t-x-x-109">x</span><sub><span 
class="zptmcm7t-x-x-80">cog</span></sub><sup><span 
class="zptmcm7t-x-x-80">(ref)</span></sup> are the centers of geometry of
the current and reference positions respectively, and <span 
class="ptmb7t-x-x-109">v</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub> are the components of the vector for each atom. Example
choices for <span 
class="zptmcm7t-x-x-109">(</span><span 
class="ptmb7t-x-x-109">v</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub><span 
class="zptmcm7t-x-x-109">) </span>are an eigenvector of the covariance matrix (essential mode), or a normal mode of the system. It is
assumed that <span 
class="zptmcm7v-x-x-109">∑</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub><span 
class="ptmb7t-x-x-109">v</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub> <span 
class="zptmcm7t-x-x-109">= 0</span>: otherwise, the Colvars module centers the <span 
class="ptmb7t-x-x-109">v</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub> automatically when reading them from the
conﬁguration.
</p><!--l. 2036--><p class="noindent" ><span 
class="ptmb7t-x-x-109">List of keywords </span>(see also <a 
href="#sec:cvc_superp">6.4<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-39002"></a> <span 
class="ot1-zi4r-0-x-x-120">atoms: </span>see deﬁnition of <a 
href="#colvar|rmsd|refPositions"><span 
class="ot1-zi4r-0-x-x-109">atoms</span></a> in sec. <a 
href="#colvar|rmsd|refPositions">6.1.17<!--tex4ht:ref: colvar|rmsd|refPositionsFile --></a> (<span 
class="ot1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-39003"></a> <span 
class="ot1-zi4r-0-x-x-120">refPositions: </span>see deﬁnition of <a 
href="#colvar|rmsd|refPositions"><span 
class="ot1-zi4r-0-x-x-109">refPositions</span></a> in sec. <a 
href="#colvar|rmsd|refPositions">6.1.17<!--tex4ht:ref: colvar|rmsd|refPositions --></a> (<span 
class="ot1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-39004"></a> <span 
class="ot1-zi4r-0-x-x-120">refPositionsFile: </span>see deﬁnition of <a 
href="#colvar|rmsd|refPositionsFile"><span 
class="ot1-zi4r-0-x-x-109">refPositionsFile</span></a> in sec. <a 
href="#colvar|rmsd|refPositions">6.1.17<!--tex4ht:ref: colvar|rmsd|refPositionsFile --></a> (<span 
class="ot1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-39005"></a> <span 
class="ot1-zi4r-0-x-x-120">refPositionsCol: </span>see deﬁnition of <a 
href="#colvar|rmsd|refPositionsCol"><span 
class="ot1-zi4r-0-x-x-109">refPositionsCol</span></a> in sec. <a 
href="#colvar|rmsd|refPositions">6.1.17<!--tex4ht:ref: colvar|rmsd|refPositionsCol --></a> (<span 
class="ot1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-39006"></a>   <span 
class="ot1-zi4r-0-x-x-120">refPositionsColValue: </span>see   deﬁnition   of   <a 
href="#colvar|rmsd|refPositionsColValue"><span 
class="ot1-zi4r-0-x-x-109">refPositionsColValue</span></a> in   sec. <a 
href="#colvar|rmsd|refPositions">6.1.17<!--tex4ht:ref: colvar|rmsd|refPositionsColValue --></a>   (<span 
class="ot1-zi4r-0-x-x-109">rmsd</span>
     component)
                                                                                         
                                                                                         
     </li>
     <li class="itemize"><a 
 id="dx1-39007"></a> <span 
class="ot1-zi4r-0-x-x-120">vector </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Vector components<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">eigenvector</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>space-separated list of <span 
class="ot1-zi4r-0-x-x-109">(x, y, z) </span>triplets <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:    </span>This  option  (mutually  exclusive  with  <span 
class="ot1-zi4r-0-x-x-109">vectorFile</span>)  sets  the  values  of  the  vector
     components.
     </li>
     <li class="itemize"><a 
 id="dx1-39008"></a> <span 
class="ot1-zi4r-0-x-x-120">vectorFile </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>PDB ﬁle containing vector components<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">eigenvector</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>UNIX ﬁlename <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>This option (mutually exclusive with <span 
class="ot1-zi4r-0-x-x-109">vector</span>) sets the name of a PDB ﬁle where the
     vector components will be read from the X, Y, and Z ﬁelds. <span 
class="ptmb7t-x-x-109">Note: </span><span 
class="ptmri7t-x-x-109">The PDB ﬁle has limited precision</span>
     <span 
class="ptmri7t-x-x-109">and ﬁxed point numbers: in some cases, the vector may not be accurately represented, and </span><span 
class="ot1-zi4r-0-x-x-109">vector</span>
     <span 
class="ptmri7t-x-x-109">should be used instead.</span>
     </li>
     <li class="itemize"><a 
 id="dx1-39009"></a> <span 
class="ot1-zi4r-0-x-x-120">vectorCol </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>PDB column used to ﬂag participating atoms<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">eigenvector</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span><span 
class="ot1-zi4r-0-x-x-109">O </span>or <span 
class="ot1-zi4r-0-x-x-109">B</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>Analogous to <span 
class="ot1-zi4r-0-x-x-109">atomsCol</span>.
     </li>
     <li class="itemize"><a 
 id="dx1-39010"></a> <span 
class="ot1-zi4r-0-x-x-120">vectorColValue </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Value used to ﬂag participating atoms in the PDB ﬁle<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">eigenvector</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive decimal <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>Analogous to <span 
class="ot1-zi4r-0-x-x-109">atomsColValue</span>.
     </li>
     <li class="itemize"><a 
 id="dx1-39011"></a>  <span 
class="ot1-zi4r-0-x-x-120">differenceVector   </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>The  <span 
class="zptmcm7t-x-x-109">3</span><span 
class="zptmcm7m-x-x-109">n</span>-dimensional  vector  is  the  diﬀerence  between  <span 
class="ot1-zi4r-0-x-x-109">vector </span>and
     <span 
class="ot1-zi4r-0-x-x-109">refPositions</span><span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">eigenvector</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">off</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>If this option is <span 
class="ot1-zi4r-0-x-x-109">on</span>, the numbers provided by <span 
class="ot1-zi4r-0-x-x-109">vector </span>or <span 
class="ot1-zi4r-0-x-x-109">vectorFile </span>are interpreted
     as  another  set  of  positions,  <span 
class="ptmb7t-x-x-109">x</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub><span 
class="zptmcm7y-x-x-109">′</span>:  the  vector  <span 
class="ptmb7t-x-x-109">v</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub> is  then  deﬁned  as  <span 
class="ptmb7t-x-x-109">v</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub> <span 
class="zptmcm7t-x-x-109">= </span><img 
src="colvars-refman-namd10x.png" alt="(        )
 x′i− x(ref)
      i"  class="left" align="middle" />.  This  allows  to
     conveniently  deﬁne  a  colvar  <span 
class="zptmcm7m-x-x-109">ξ </span>as  a  projection  on  the  linear  transformation  between  two  sets  of
     positions, “A” and “B”. For convenience, the vector is also normalized so that <span 
class="zptmcm7m-x-x-109">ξ </span><span 
class="zptmcm7t-x-x-109">= 0 </span>when the atoms
     are at the set of positions “A” and <span 
class="zptmcm7m-x-x-109">ξ </span><span 
class="zptmcm7t-x-x-109">= 1 </span>at the set of positions “B”.   </li></ul>
<!--l. 2102--><p class="noindent" >This component returns a number (in Å), whose value ranges between the smallest and largest absolute positions in the
unit cell during the simulations (see also <span 
class="ot1-zi4r-0-x-x-109">distanceZ</span>). Due to the normalization in eq. <a 
href="#eq:cvc_eigenvector">6<!--tex4ht:ref: eq:cvc_eigenvector --></a>, this range does not depend
on the number of atoms involved.
</p><!--l. 2109--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">6.1.20    </span> <a 
 id="x1-400006.1.20"></a><span 
class="ot1-zi4r-0-x-x-109">gyration</span>: radius of gyration of a group of atoms.</h5>
                                                                                         
                                                                                         
<!--l. 2110--><p class="noindent" >The block <span 
class="ot1-zi4r-0-x-x-109">gyration</span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109">...</span><span 
class="cmsy-10x-x-109">} </span>deﬁnes the parameters for calculating the radius of gyration of a group of atomic
positions <span 
class="zptmcm7y-x-x-109">{</span><span 
class="ptmb7t-x-x-109">x</span><sub><span 
class="zptmcm7t-x-x-80">1</span></sub><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">t</span><span 
class="zptmcm7t-x-x-109">)</span><span 
class="zptmcm7m-x-x-109">,</span><span 
class="ptmb7t-x-x-109">x</span><sub><span 
class="zptmcm7t-x-x-80">2</span></sub><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">t</span><span 
class="zptmcm7t-x-x-109">)</span><span 
class="zptmcm7m-x-x-109">,</span><span 
class="zptmcm7m-x-x-109">…</span><span 
class="ptmb7t-x-x-109">x</span><sub><span 
class="zptmcm7m-x-x-80">N</span></sub><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">t</span><span 
class="zptmcm7t-x-x-109">)</span><span 
class="zptmcm7y-x-x-109">} </span>with respect to their center of geometry, <span 
class="ptmb7t-x-x-109">x</span><sub><span 
class="zptmcm7t-x-x-80">cog</span></sub><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">t</span><span 
class="zptmcm7t-x-x-109">)</span>:
</p>
   <table 
class="equation"><tr><td><a 
 id="x1-40001r7"></a>
   <center class="math-display" >
<img 
src="colvars-refman-namd11x.png" alt="      ∘ -------------------
        -1 N ||           ||2
Rgyr =  N ∑   xi(t)− xcog(t)
          i=1
" class="math-display"  /></center></td><td class="equation-label">(7)</td></tr></table>
<!--l. 2120--><p class="nopar" >
This component must contain one <span 
class="ot1-zi4r-0-x-x-109">atoms</span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109">...</span><span 
class="cmsy-10x-x-109">} </span>block to deﬁne the atom group, and returns a positive number,
expressed in Å.
</p><!--l. 2125--><p class="noindent" ><span 
class="ptmb7t-x-x-109">List of keywords </span>(see also <a 
href="#sec:cvc_superp">6.4<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-40002"></a> <span 
class="ot1-zi4r-0-x-x-120">atoms: </span>see deﬁnition of <a 
href="#colvar|rmsd|refPositions"><span 
class="ot1-zi4r-0-x-x-109">atoms</span></a> in sec. <a 
href="#colvar|rmsd|refPositions">6.1.17<!--tex4ht:ref: colvar|rmsd|refPositionsFile --></a> (<span 
class="ot1-zi4r-0-x-x-109">rmsd </span>component) </li></ul>
<!--l. 2130--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">6.1.21    </span> <a 
 id="x1-410006.1.21"></a><span 
class="ot1-zi4r-0-x-x-109">inertia</span>: total moment of inertia of a group of atoms.</h5>
<!--l. 2131--><p class="noindent" >The block <span 
class="ot1-zi4r-0-x-x-109">inertia</span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109">...</span><span 
class="cmsy-10x-x-109">} </span>deﬁnes the parameters for calculating the total moment of inertia of a group of atomic
positions <span 
class="zptmcm7y-x-x-109">{</span><span 
class="ptmb7t-x-x-109">x</span><sub><span 
class="zptmcm7t-x-x-80">1</span></sub><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">t</span><span 
class="zptmcm7t-x-x-109">)</span><span 
class="zptmcm7m-x-x-109">,</span><span 
class="ptmb7t-x-x-109">x</span><sub><span 
class="zptmcm7t-x-x-80">2</span></sub><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">t</span><span 
class="zptmcm7t-x-x-109">)</span><span 
class="zptmcm7m-x-x-109">,</span><span 
class="zptmcm7m-x-x-109">…</span><span 
class="ptmb7t-x-x-109">x</span><sub><span 
class="zptmcm7m-x-x-80">N</span></sub><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">t</span><span 
class="zptmcm7t-x-x-109">)</span><span 
class="zptmcm7y-x-x-109">} </span>with respect to their center of geometry, <span 
class="ptmb7t-x-x-109">x</span><sub><span 
class="zptmcm7t-x-x-80">cog</span></sub><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">t</span><span 
class="zptmcm7t-x-x-109">)</span>:
</p>
   <table 
class="equation"><tr><td><a 
 id="x1-41001r8"></a>
   <center class="math-display" >
<img 
src="colvars-refman-namd12x.png" alt="     N
I =    ||xi(t)− xcog(t)||2
    ∑i=1
" class="math-display"  /></center></td><td class="equation-label">(8)</td></tr></table>
<!--l. 2140--><p class="nopar" >
<span 
class="ptmri7t-x-x-109">Note that all atomic masses are set to 1 for simplicity. </span>This component must contain one <span 
class="ot1-zi4r-0-x-x-109">atoms</span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109">...</span><span 
class="cmsy-10x-x-109">} </span>block to
deﬁne the atom group, and returns a positive number, expressed in Å<sup><span 
class="zptmcm7t-x-x-80">2</span></sup>.
</p><!--l. 2146--><p class="noindent" ><span 
class="ptmb7t-x-x-109">List of keywords </span>(see also <a 
href="#sec:cvc_superp">6.4<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
                                                                                         
                                                                                         
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-41002"></a> <span 
class="ot1-zi4r-0-x-x-120">atoms: </span>see deﬁnition of <a 
href="#colvar|rmsd|refPositions"><span 
class="ot1-zi4r-0-x-x-109">atoms</span></a> in sec. <a 
href="#colvar|rmsd|refPositions">6.1.17<!--tex4ht:ref: colvar|rmsd|refPositionsFile --></a> (<span 
class="ot1-zi4r-0-x-x-109">rmsd </span>component) </li></ul>
<!--l. 2151--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">6.1.22    </span> <a 
 id="x1-420006.1.22"></a><span 
class="ot1-zi4r-0-x-x-109">inertiaZ</span>: total moment of inertia of a group of atoms around a chosen axis.</h5>
<!--l. 2152--><p class="noindent" >The block <span 
class="ot1-zi4r-0-x-x-109">inertiaZ</span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109">...</span><span 
class="cmsy-10x-x-109">} </span>deﬁnes the parameters for calculating the component along the axis <span 
class="ptmb7t-x-x-109">e </span>of the moment of
inertia of a group of atomic positions <span 
class="zptmcm7y-x-x-109">{</span><span 
class="ptmb7t-x-x-109">x</span><sub><span 
class="zptmcm7t-x-x-80">1</span></sub><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">t</span><span 
class="zptmcm7t-x-x-109">)</span><span 
class="zptmcm7m-x-x-109">,</span><span 
class="ptmb7t-x-x-109">x</span><sub><span 
class="zptmcm7t-x-x-80">2</span></sub><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">t</span><span 
class="zptmcm7t-x-x-109">)</span><span 
class="zptmcm7m-x-x-109">,</span><span 
class="zptmcm7m-x-x-109">…</span><span 
class="ptmb7t-x-x-109">x</span><sub><span 
class="zptmcm7m-x-x-80">N</span></sub><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">t</span><span 
class="zptmcm7t-x-x-109">)</span><span 
class="zptmcm7y-x-x-109">} </span>with respect to their center of geometry,
<span 
class="ptmb7t-x-x-109">x</span><sub><span 
class="zptmcm7t-x-x-80">cog</span></sub><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">t</span><span 
class="zptmcm7t-x-x-109">)</span>:
</p>
   <table 
class="equation"><tr><td><a 
 id="x1-42001r9"></a>
   <center class="math-display" >
<img 
src="colvars-refman-namd13x.png" alt="     N
I =    ((x (t)− x  (t))⋅e)2
e   ∑i=1    i     cog
" class="math-display"  /></center></td><td class="equation-label">(9)</td></tr></table>
<!--l. 2161--><p class="nopar" >
<span 
class="ptmri7t-x-x-109">Note that all atomic masses are set to 1 for simplicity. </span>This component must contain one <span 
class="ot1-zi4r-0-x-x-109">atoms</span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109">...</span><span 
class="cmsy-10x-x-109">} </span>block to
deﬁne the atom group, and returns a positive number, expressed in Å<sup><span 
class="zptmcm7t-x-x-80">2</span></sup>.
</p><!--l. 2167--><p class="noindent" ><span 
class="ptmb7t-x-x-109">List of keywords </span>(see also <a 
href="#sec:cvc_superp">6.4<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-42002"></a> <span 
class="ot1-zi4r-0-x-x-120">atoms: </span>see deﬁnition of <a 
href="#colvar|rmsd|refPositions"><span 
class="ot1-zi4r-0-x-x-109">atoms</span></a> in sec. <a 
href="#colvar|rmsd|refPositions">6.1.17<!--tex4ht:ref: colvar|rmsd|refPositionsFile --></a> (<span 
class="ot1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-42003"></a> <span 
class="ot1-zi4r-0-x-x-120">axis </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Projection axis (Å)<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">inertiaZ</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span><span 
class="ot1-zi4r-0-x-x-109">(x, y, z) </span>triplet <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">(0.0, 0.0, 1.0)</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>The three components of this vector deﬁne (when normalized) the projection axis <span 
class="ptmb7t-x-x-109">e</span>.  </li></ul>
<!--l. 2182--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">6.1.23    </span> <a 
 id="x1-430006.1.23"></a><span 
class="ot1-zi4r-0-x-x-109">orientation</span>: orientation from reference coordinates.</h5>
<!--l. 2183--><p class="noindent" >The block <span 
class="ot1-zi4r-0-x-x-109">orientation</span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109">...</span><span 
class="cmsy-10x-x-109">} </span>returns the same optimal rotation used in the <span 
class="ot1-zi4r-0-x-x-109">rmsd </span>component to superimpose
the coordinates <span 
class="zptmcm7y-x-x-109">{</span><span 
class="ptmb7t-x-x-109">x</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">t</span><span 
class="zptmcm7t-x-x-109">)</span><span 
class="zptmcm7y-x-x-109">} </span>onto a set of reference coordinates <span 
class="zptmcm7y-x-x-109">{</span><span 
class="ptmb7t-x-x-109">x</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub><sup><span 
class="zptmcm7t-x-x-80">(ref)</span></sup><span 
class="zptmcm7y-x-x-109">}</span>. Such component returns a four
dimensional vector <span 
class="cmss-10x-x-109">q</span> <span 
class="zptmcm7t-x-x-109">= (</span><span 
class="zptmcm7m-x-x-109">q</span><sub><span 
class="zptmcm7t-x-x-80">0</span></sub><span 
class="zptmcm7m-x-x-109">,q</span><sub><span 
class="zptmcm7t-x-x-80">1</span></sub><span 
class="zptmcm7m-x-x-109">,q</span><sub><span 
class="zptmcm7t-x-x-80">2</span></sub><span 
class="zptmcm7m-x-x-109">,q</span><sub><span 
class="zptmcm7t-x-x-80">3</span></sub><span 
class="zptmcm7t-x-x-109">)</span>, with <span 
class="zptmcm7v-x-x-109">∑</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub><span 
class="zptmcm7m-x-x-109">q</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub><sup><span 
class="zptmcm7t-x-x-80">2</span></sup> <span 
class="zptmcm7t-x-x-109">= 1</span>; this <span 
class="ptmri7t-x-x-109">quaternion </span>expresses the optimal rotation
<span 
class="zptmcm7y-x-x-109">{</span><span 
class="ptmb7t-x-x-109">x</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">t</span><span 
class="zptmcm7t-x-x-109">)</span><span 
class="zptmcm7y-x-x-109">}→{</span><span 
class="ptmb7t-x-x-109">x</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub><sup><span 
class="zptmcm7t-x-x-80">(ref)</span></sup><span 
class="zptmcm7y-x-x-109">} </span>according to the formalism in reference <span class="cite">[<a 
href="#XCoutsias2004">3</a>]</span>. The quaternion <span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">q</span><sub><span 
class="zptmcm7t-x-x-80">0</span></sub><span 
class="zptmcm7m-x-x-109">,q</span><sub><span 
class="zptmcm7t-x-x-80">1</span></sub><span 
class="zptmcm7m-x-x-109">,q</span><sub><span 
class="zptmcm7t-x-x-80">2</span></sub><span 
class="zptmcm7m-x-x-109">,q</span><sub><span 
class="zptmcm7t-x-x-80">3</span></sub><span 
class="zptmcm7t-x-x-109">) </span>can also be written as
                                                                                         
                                                                                         
<img 
src="colvars-refman-namd14x.png" alt="(cos(𝜃∕2),sin(𝜃 ∕2)u)"  class="left" align="middle" />, where <span 
class="zptmcm7m-x-x-109">𝜃 </span>is the angle and <span 
class="ptmb7t-x-x-109">u </span>the normalized axis of rotation; for example, a rotation of 90<sup><span 
class="zptmcm7y-x-x-80">∘</span></sup>
around the <span 
class="zptmcm7m-x-x-109">z </span>axis is expressed as “<span 
class="ot1-zi4r-0-x-x-109">(0.707, 0.0, 0.0, 0.707)</span>”. The script <span 
class="ot1-zi4r-0-x-x-109">quaternion2rmatrix.tcl</span>
provides Tcl functions for converting to and from a <span 
class="zptmcm7t-x-x-109">4</span><span 
class="zptmcm7y-x-x-109">×</span><span 
class="zptmcm7t-x-x-109">4 </span>rotation matrix in a format suitable for usage in
VMD.
</p><!--l. 2201--><p class="indent" >   As for the component <span 
class="ot1-zi4r-0-x-x-109">rmsd</span>, the available options are <span 
class="ot1-zi4r-0-x-x-109">atoms</span>, <span 
class="ot1-zi4r-0-x-x-109">refPositionsFile</span>, <span 
class="ot1-zi4r-0-x-x-109">refPositionsCol </span>and
<span 
class="ot1-zi4r-0-x-x-109">refPositionsColValue</span>,  and <span 
class="ot1-zi4r-0-x-x-109">refPositions</span>.
</p><!--l. 2203--><p class="indent" >   <span 
class="ptmb7t-x-x-109">Note: </span><span 
class="ot1-zi4r-0-x-x-109">refPositions</span>and <span 
class="ot1-zi4r-0-x-x-109">refPositionsFile </span>deﬁne the set of positions <span 
class="ptmri7t-x-x-109">from which </span>the optimal rotation is
calculated, but this rotation is not applied to the coordinates of the atoms involved: it is used instead to deﬁne the
variable itself.
</p><!--l. 2205--><p class="noindent" ><span 
class="ptmb7t-x-x-109">List of keywords </span>(see also <a 
href="#sec:cvc_superp">6.4<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-43001"></a> <span 
class="ot1-zi4r-0-x-x-120">atoms: </span>see deﬁnition of <a 
href="#colvar|rmsd|refPositions"><span 
class="ot1-zi4r-0-x-x-109">atoms</span></a> in sec. <a 
href="#colvar|rmsd|refPositions">6.1.17<!--tex4ht:ref: colvar|rmsd|refPositionsFile --></a> (<span 
class="ot1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-43002"></a> <span 
class="ot1-zi4r-0-x-x-120">refPositions: </span>see deﬁnition of <a 
href="#colvar|rmsd|refPositions"><span 
class="ot1-zi4r-0-x-x-109">refPositions</span></a> in sec. <a 
href="#colvar|rmsd|refPositions">6.1.17<!--tex4ht:ref: colvar|rmsd|refPositions --></a> (<span 
class="ot1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-43003"></a> <span 
class="ot1-zi4r-0-x-x-120">refPositionsFile: </span>see deﬁnition of <a 
href="#colvar|rmsd|refPositionsFile"><span 
class="ot1-zi4r-0-x-x-109">refPositionsFile</span></a> in sec. <a 
href="#colvar|rmsd|refPositions">6.1.17<!--tex4ht:ref: colvar|rmsd|refPositionsFile --></a> (<span 
class="ot1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-43004"></a> <span 
class="ot1-zi4r-0-x-x-120">refPositionsCol: </span>see deﬁnition of <a 
href="#colvar|rmsd|refPositionsCol"><span 
class="ot1-zi4r-0-x-x-109">refPositionsCol</span></a> in sec. <a 
href="#colvar|rmsd|refPositions">6.1.17<!--tex4ht:ref: colvar|rmsd|refPositionsCol --></a> (<span 
class="ot1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-43005"></a>   <span 
class="ot1-zi4r-0-x-x-120">refPositionsColValue: </span>see   deﬁnition   of   <a 
href="#colvar|rmsd|refPositionsColValue"><span 
class="ot1-zi4r-0-x-x-109">refPositionsColValue</span></a> in   sec. <a 
href="#colvar|rmsd|refPositions">6.1.17<!--tex4ht:ref: colvar|rmsd|refPositionsColValue --></a>   (<span 
class="ot1-zi4r-0-x-x-109">rmsd</span>
     component)
     </li>
     <li class="itemize"><a 
 id="dx1-43006"></a> <span 
class="ot1-zi4r-0-x-x-120">closestToQuaternion </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Reference rotation<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">orientation</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>“<span 
class="ot1-zi4r-0-x-x-109">(q0, q1, q2, q3)</span>” quadruplet <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">(1.0, 0.0, 0.0, 0.0) </span>(“null” rotation) <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>Between the two equivalent quaternions <span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">q</span><sub><span 
class="zptmcm7t-x-x-80">0</span></sub><span 
class="zptmcm7m-x-x-109">,q</span><sub><span 
class="zptmcm7t-x-x-80">1</span></sub><span 
class="zptmcm7m-x-x-109">,q</span><sub><span 
class="zptmcm7t-x-x-80">2</span></sub><span 
class="zptmcm7m-x-x-109">,q</span><sub><span 
class="zptmcm7t-x-x-80">3</span></sub><span 
class="zptmcm7t-x-x-109">) </span>and <span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7y-x-x-109">−</span><span 
class="zptmcm7m-x-x-109">q</span><sub><span 
class="zptmcm7t-x-x-80">0</span></sub><span 
class="zptmcm7m-x-x-109">,</span><span 
class="zptmcm7y-x-x-109">−</span><span 
class="zptmcm7m-x-x-109">q</span><sub><span 
class="zptmcm7t-x-x-80">1</span></sub><span 
class="zptmcm7m-x-x-109">,</span><span 
class="zptmcm7y-x-x-109">−</span><span 
class="zptmcm7m-x-x-109">q</span><sub><span 
class="zptmcm7t-x-x-80">2</span></sub><span 
class="zptmcm7m-x-x-109">,</span><span 
class="zptmcm7y-x-x-109">−</span><span 
class="zptmcm7m-x-x-109">q</span><sub><span 
class="zptmcm7t-x-x-80">3</span></sub><span 
class="zptmcm7t-x-x-109">)</span>, the
     closer to <span 
class="ot1-zi4r-0-x-x-109">(1.0, 0.0, 0.0, 0.0) </span>is chosen. This simpliﬁes the visualization of the colvar trajectory
     when samples values are a smaller subset of all possible rotations. <span 
class="ptmb7t-x-x-109">Note: </span><span 
class="ptmri7t-x-x-109">this only aﬀects the output,</span>
     <span 
class="ptmri7t-x-x-109">never the dynamics</span>.
     </li></ul>
<!--l. 2236--><p class="indent" >   <span 
class="ptmb7t-x-x-109">Tip: stopping the rotation of a protein. </span>To stop the rotation of an elongated macromolecule in solution (and
use an anisotropic box to save water molecules), it is possible to deﬁne a colvar with an <span 
class="ot1-zi4r-0-x-x-109">orientation </span>component,
and restrain it throuh the <span 
class="ot1-zi4r-0-x-x-109">harmonic </span>bias around the identity rotation, <span 
class="ot1-zi4r-0-x-x-109">(1.0, 0.0, 0.0, 0.0)</span>. Only
the overall orientation of the macromolecule is aﬀected, and <span 
class="ptmri7t-x-x-109">not </span>its internal degrees of freedom. The
user should also take care that the macromolecule is composed by a single chain, or disable <span 
class="ot1-zi4r-0-x-x-109">wrapAll</span>
otherwise.
</p><!--l. 2248--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">6.1.24    </span> <a 
 id="x1-440006.1.24"></a><span 
class="ot1-zi4r-0-x-x-109">orientationAngle</span>: angle of rotation from reference coordinates.</h5>
                                                                                         
                                                                                         
<!--l. 2249--><p class="noindent" >The block <span 
class="ot1-zi4r-0-x-x-109">orientationAngle</span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109">...</span><span 
class="cmsy-10x-x-109">} </span>accepts the same base options as the component <span 
class="ot1-zi4r-0-x-x-109">orientation</span>: <span 
class="ot1-zi4r-0-x-x-109">atoms</span>,
<span 
class="ot1-zi4r-0-x-x-109">refPositions</span>, <span 
class="ot1-zi4r-0-x-x-109">refPositionsFile</span>, <span 
class="ot1-zi4r-0-x-x-109">refPositionsCol </span>and <span 
class="ot1-zi4r-0-x-x-109">refPositionsColValue</span>. The returned value is the
angle of rotation <span 
class="zptmcm7m-x-x-109">𝜃 </span>between the current and the reference positions. This angle is expressed in degrees within the
range [0<sup><span 
class="zptmcm7y-x-x-80">∘</span></sup>:180<sup><span 
class="zptmcm7y-x-x-80">∘</span></sup>].
</p><!--l. 2254--><p class="noindent" ><span 
class="ptmb7t-x-x-109">List of keywords </span>(see also <a 
href="#sec:cvc_superp">6.4<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-44001"></a> <span 
class="ot1-zi4r-0-x-x-120">atoms: </span>see deﬁnition of <a 
href="#colvar|rmsd|refPositions"><span 
class="ot1-zi4r-0-x-x-109">atoms</span></a> in sec. <a 
href="#colvar|rmsd|refPositions">6.1.17<!--tex4ht:ref: colvar|rmsd|refPositionsFile --></a> (<span 
class="ot1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-44002"></a> <span 
class="ot1-zi4r-0-x-x-120">refPositions: </span>see deﬁnition of <a 
href="#colvar|rmsd|refPositions"><span 
class="ot1-zi4r-0-x-x-109">refPositions</span></a> in sec. <a 
href="#colvar|rmsd|refPositions">6.1.17<!--tex4ht:ref: colvar|rmsd|refPositions --></a> (<span 
class="ot1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-44003"></a> <span 
class="ot1-zi4r-0-x-x-120">refPositionsFile: </span>see deﬁnition of <a 
href="#colvar|rmsd|refPositionsFile"><span 
class="ot1-zi4r-0-x-x-109">refPositionsFile</span></a> in sec. <a 
href="#colvar|rmsd|refPositions">6.1.17<!--tex4ht:ref: colvar|rmsd|refPositionsFile --></a> (<span 
class="ot1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-44004"></a> <span 
class="ot1-zi4r-0-x-x-120">refPositionsCol: </span>see deﬁnition of <a 
href="#colvar|rmsd|refPositionsCol"><span 
class="ot1-zi4r-0-x-x-109">refPositionsCol</span></a> in sec. <a 
href="#colvar|rmsd|refPositions">6.1.17<!--tex4ht:ref: colvar|rmsd|refPositionsCol --></a> (<span 
class="ot1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-44005"></a>   <span 
class="ot1-zi4r-0-x-x-120">refPositionsColValue: </span>see   deﬁnition   of   <a 
href="#colvar|rmsd|refPositionsColValue"><span 
class="ot1-zi4r-0-x-x-109">refPositionsColValue</span></a> in   sec. <a 
href="#colvar|rmsd|refPositions">6.1.17<!--tex4ht:ref: colvar|rmsd|refPositionsColValue --></a>   (<span 
class="ot1-zi4r-0-x-x-109">rmsd</span>
     component)
     </li></ul>
<!--l. 2271--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">6.1.25    </span> <a 
 id="x1-450006.1.25"></a><span 
class="ot1-zi4r-0-x-x-109">orientationProj</span>: cosine of the angle of rotation from reference coordinates.</h5>
<!--l. 2272--><p class="noindent" >The block <span 
class="ot1-zi4r-0-x-x-109">orientationProj</span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109">...</span><span 
class="cmsy-10x-x-109">} </span>accepts the same base options as the component <span 
class="ot1-zi4r-0-x-x-109">orientation</span>: <span 
class="ot1-zi4r-0-x-x-109">atoms</span>,
<span 
class="ot1-zi4r-0-x-x-109">refPositions</span>, <span 
class="ot1-zi4r-0-x-x-109">refPositionsFile</span>, <span 
class="ot1-zi4r-0-x-x-109">refPositionsCol </span>and <span 
class="ot1-zi4r-0-x-x-109">refPositionsColValue</span>. The returned value is the
cosine of the angle of rotation <span 
class="zptmcm7m-x-x-109">𝜃 </span>between the current and the reference positions. The range of values is
[-1:1].
</p><!--l. 2277--><p class="noindent" ><span 
class="ptmb7t-x-x-109">List of keywords </span>(see also <a 
href="#sec:cvc_superp">6.4<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-45001"></a> <span 
class="ot1-zi4r-0-x-x-120">atoms: </span>see deﬁnition of <a 
href="#colvar|rmsd|refPositions"><span 
class="ot1-zi4r-0-x-x-109">atoms</span></a> in sec. <a 
href="#colvar|rmsd|refPositions">6.1.17<!--tex4ht:ref: colvar|rmsd|refPositionsFile --></a> (<span 
class="ot1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-45002"></a> <span 
class="ot1-zi4r-0-x-x-120">refPositions: </span>see deﬁnition of <a 
href="#colvar|rmsd|refPositions"><span 
class="ot1-zi4r-0-x-x-109">refPositions</span></a> in sec. <a 
href="#colvar|rmsd|refPositions">6.1.17<!--tex4ht:ref: colvar|rmsd|refPositions --></a> (<span 
class="ot1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-45003"></a> <span 
class="ot1-zi4r-0-x-x-120">refPositionsFile: </span>see deﬁnition of <a 
href="#colvar|rmsd|refPositionsFile"><span 
class="ot1-zi4r-0-x-x-109">refPositionsFile</span></a> in sec. <a 
href="#colvar|rmsd|refPositions">6.1.17<!--tex4ht:ref: colvar|rmsd|refPositionsFile --></a> (<span 
class="ot1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-45004"></a> <span 
class="ot1-zi4r-0-x-x-120">refPositionsCol: </span>see deﬁnition of <a 
href="#colvar|rmsd|refPositionsCol"><span 
class="ot1-zi4r-0-x-x-109">refPositionsCol</span></a> in sec. <a 
href="#colvar|rmsd|refPositions">6.1.17<!--tex4ht:ref: colvar|rmsd|refPositionsCol --></a> (<span 
class="ot1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-45005"></a>   <span 
class="ot1-zi4r-0-x-x-120">refPositionsColValue: </span>see   deﬁnition   of   <a 
href="#colvar|rmsd|refPositionsColValue"><span 
class="ot1-zi4r-0-x-x-109">refPositionsColValue</span></a> in   sec. <a 
href="#colvar|rmsd|refPositions">6.1.17<!--tex4ht:ref: colvar|rmsd|refPositionsColValue --></a>   (<span 
class="ot1-zi4r-0-x-x-109">rmsd</span>
     component)  </li></ul>
                                                                                         
                                                                                         
<!--l. 2293--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">6.1.26    </span> <a 
 id="colvar|spinAngle|axis"></a><span 
class="ot1-zi4r-0-x-x-109">spinAngle</span>: angle of rotation around a given axis.</h5>
<!--l. 2294--><p class="noindent" >The complete rotation described by <span 
class="ot1-zi4r-0-x-x-109">orientation </span>can optionally be decomposed into two sub-rotations: one is a
“<span 
class="ptmri7t-x-x-109">spin</span>” rotation around <span 
class="ptmb7t-x-x-109">e</span>, and the other a “<span 
class="ptmri7t-x-x-109">tilt</span>” rotation around an axis orthogonal to <span 
class="ptmb7t-x-x-109">e</span>. The component <span 
class="ot1-zi4r-0-x-x-109">spinAngle</span>
measures the angle of the “spin” sub-rotation around <span 
class="ptmb7t-x-x-109">e</span>.
</p><!--l. 2297--><p class="noindent" ><span 
class="ptmb7t-x-x-109">List of keywords </span>(see also <a 
href="#sec:cvc_superp">6.4<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-46001"></a> <span 
class="ot1-zi4r-0-x-x-120">atoms: </span>see deﬁnition of <a 
href="#colvar|rmsd|refPositions"><span 
class="ot1-zi4r-0-x-x-109">atoms</span></a> in sec. <a 
href="#colvar|rmsd|refPositions">6.1.17<!--tex4ht:ref: colvar|rmsd|refPositionsFile --></a> (<span 
class="ot1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-46002"></a> <span 
class="ot1-zi4r-0-x-x-120">refPositions: </span>see deﬁnition of <a 
href="#colvar|rmsd|refPositions"><span 
class="ot1-zi4r-0-x-x-109">refPositions</span></a> in sec. <a 
href="#colvar|rmsd|refPositions">6.1.17<!--tex4ht:ref: colvar|rmsd|refPositions --></a> (<span 
class="ot1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-46003"></a> <span 
class="ot1-zi4r-0-x-x-120">refPositionsFile: </span>see deﬁnition of <a 
href="#colvar|rmsd|refPositionsFile"><span 
class="ot1-zi4r-0-x-x-109">refPositionsFile</span></a> in sec. <a 
href="#colvar|rmsd|refPositions">6.1.17<!--tex4ht:ref: colvar|rmsd|refPositionsFile --></a> (<span 
class="ot1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-46004"></a> <span 
class="ot1-zi4r-0-x-x-120">refPositionsCol: </span>see deﬁnition of <a 
href="#colvar|rmsd|refPositionsCol"><span 
class="ot1-zi4r-0-x-x-109">refPositionsCol</span></a> in sec. <a 
href="#colvar|rmsd|refPositions">6.1.17<!--tex4ht:ref: colvar|rmsd|refPositionsCol --></a> (<span 
class="ot1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-46005"></a>   <span 
class="ot1-zi4r-0-x-x-120">refPositionsColValue: </span>see   deﬁnition   of   <a 
href="#colvar|rmsd|refPositionsColValue"><span 
class="ot1-zi4r-0-x-x-109">refPositionsColValue</span></a> in   sec. <a 
href="#colvar|rmsd|refPositions">6.1.17<!--tex4ht:ref: colvar|rmsd|refPositionsColValue --></a>   (<span 
class="ot1-zi4r-0-x-x-109">rmsd</span>
     component)
     </li>
     <li class="itemize"><a 
 id="colvar|spinAngle|axis"></a> <a 
 id="dx1-46006"></a><span 
class="ot1-zi4r-0-x-x-120">axis </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Special rotation axis (Å)<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">tilt</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span><span 
class="ot1-zi4r-0-x-x-109">(x, y, z) </span>triplet <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">(0.0, 0.0, 1.0)</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>The three components of this vector deﬁne (when normalized) the special rotation axis
     used to calculate the <span 
class="ot1-zi4r-0-x-x-109">tilt </span>and <span 
class="ot1-zi4r-0-x-x-109">spinAngle </span>components.  </li></ul>
<!--l. 2320--><p class="noindent" >The component <span 
class="ot1-zi4r-0-x-x-109">spinAngle </span>returns an angle (in degrees) within the periodic interval <span 
class="zptmcm7t-x-x-109">[</span><span 
class="zptmcm7y-x-x-109">−</span><span 
class="zptmcm7t-x-x-109">180 : 180]</span>.
</p><!--l. 2322--><p class="indent" >   <span 
class="ptmb7t-x-x-109">Note: </span>the value of <span 
class="ot1-zi4r-0-x-x-109">spinAngle </span>is a continuous function almost everywhere, with the exception of conﬁgurations
with the corresponding “tilt” angle equal to 180<sup><span 
class="zptmcm7y-x-x-80">∘</span></sup> (i.e. the <span 
class="ot1-zi4r-0-x-x-109">tilt </span>component is equal to <span 
class="zptmcm7y-x-x-109">−</span><span 
class="zptmcm7t-x-x-109">1</span>): in those cases,
<span 
class="ot1-zi4r-0-x-x-109">spinAngle </span>is undeﬁned. If such conﬁgurations are expected, consider deﬁning a <span 
class="ot1-zi4r-0-x-x-109">tilt </span>colvar using the same axis <span 
class="ptmb7t-x-x-109">e</span>,
and restraining it with a lower wall away from <span 
class="zptmcm7y-x-x-109">−</span><span 
class="zptmcm7t-x-x-109">1</span>.
</p><!--l. 2325--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">6.1.27    </span> <a 
 id="x1-470006.1.27"></a><span 
class="ot1-zi4r-0-x-x-109">tilt</span>: cosine of the rotation orthogonal to a given axis.</h5>
<!--l. 2326--><p class="noindent" >The component <span 
class="ot1-zi4r-0-x-x-109">tilt </span>measures the cosine of the angle of the “tilt” sub-rotation, which combined with the “spin”
sub-rotation provides the complete rotation of a group of atoms. The cosine of the tilt angle rather than the tilt angle
itself is implemented, because the latter is unevenly distributed even for an isotropic system: consider as an analogy
the angle <span 
class="zptmcm7m-x-x-109">𝜃 </span>in the spherical coordinate system. The component <span 
class="ot1-zi4r-0-x-x-109">tilt </span>relies on the same options as <span 
class="ot1-zi4r-0-x-x-109">spinAngle</span>,
including the deﬁnition of the axis <span 
class="ptmb7t-x-x-109">e</span>. The values of <span 
class="ot1-zi4r-0-x-x-109">tilt </span>are real numbers in the interval <span 
class="zptmcm7t-x-x-109">[</span><span 
class="zptmcm7y-x-x-109">−</span><span 
class="zptmcm7t-x-x-109">1 : 1]</span>: the value <span 
class="zptmcm7t-x-x-109">1</span>
represents an orientation fully parallel to <span 
class="ptmb7t-x-x-109">e </span>(tilt angle = 0<sup><span 
class="zptmcm7y-x-x-80">∘</span></sup>), and the value <span 
class="zptmcm7y-x-x-109">−</span><span 
class="zptmcm7t-x-x-109">1 </span>represents an anti-parallel
orientation.
</p><!--l. 2331--><p class="noindent" ><span 
class="ptmb7t-x-x-109">List of keywords </span>(see also <a 
href="#sec:cvc_superp">6.4<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
                                                                                         
                                                                                         
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-47001"></a> <span 
class="ot1-zi4r-0-x-x-120">atoms: </span>see deﬁnition of <a 
href="#colvar|rmsd|refPositions"><span 
class="ot1-zi4r-0-x-x-109">atoms</span></a> in sec. <a 
href="#colvar|rmsd|refPositions">6.1.17<!--tex4ht:ref: colvar|rmsd|refPositionsFile --></a> (<span 
class="ot1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-47002"></a> <span 
class="ot1-zi4r-0-x-x-120">refPositions: </span>see deﬁnition of <a 
href="#colvar|rmsd|refPositions"><span 
class="ot1-zi4r-0-x-x-109">refPositions</span></a> in sec. <a 
href="#colvar|rmsd|refPositions">6.1.17<!--tex4ht:ref: colvar|rmsd|refPositions --></a> (<span 
class="ot1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-47003"></a> <span 
class="ot1-zi4r-0-x-x-120">refPositionsFile: </span>see deﬁnition of <a 
href="#colvar|rmsd|refPositionsFile"><span 
class="ot1-zi4r-0-x-x-109">refPositionsFile</span></a> in sec. <a 
href="#colvar|rmsd|refPositions">6.1.17<!--tex4ht:ref: colvar|rmsd|refPositionsFile --></a> (<span 
class="ot1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-47004"></a> <span 
class="ot1-zi4r-0-x-x-120">refPositionsCol: </span>see deﬁnition of <a 
href="#colvar|rmsd|refPositionsCol"><span 
class="ot1-zi4r-0-x-x-109">refPositionsCol</span></a> in sec. <a 
href="#colvar|rmsd|refPositions">6.1.17<!--tex4ht:ref: colvar|rmsd|refPositionsCol --></a> (<span 
class="ot1-zi4r-0-x-x-109">rmsd </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-47005"></a>   <span 
class="ot1-zi4r-0-x-x-120">refPositionsColValue: </span>see   deﬁnition   of   <a 
href="#colvar|rmsd|refPositionsColValue"><span 
class="ot1-zi4r-0-x-x-109">refPositionsColValue</span></a> in   sec. <a 
href="#colvar|rmsd|refPositions">6.1.17<!--tex4ht:ref: colvar|rmsd|refPositionsColValue --></a>   (<span 
class="ot1-zi4r-0-x-x-109">rmsd</span>
     component)
     </li>
     <li class="itemize"><a 
 id="dx1-47006"></a> <span 
class="ot1-zi4r-0-x-x-120">axis: </span>see deﬁnition of <a 
href="#colvar|spinAngle|axis"><span 
class="ot1-zi4r-0-x-x-109">axis</span></a> in sec. <a 
href="#colvar|spinAngle|axis">6.1.26<!--tex4ht:ref: colvar|spinAngle|axis --></a> (<span 
class="ot1-zi4r-0-x-x-109">spinAngle </span>component) </li></ul>
<!--l. 2589--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">6.1.28    </span> <a 
 id="colvar|alpha|psfSegID"></a><span 
class="ot1-zi4r-0-x-x-109">alpha</span>: <span 
class="zptmcm7m-x-x-109">α</span>-helix content of a protein segment.</h5>
<!--l. 2589--><p class="noindent" >The block <span 
class="ot1-zi4r-0-x-x-109">alpha</span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109">...</span><span 
class="cmsy-10x-x-109">} </span>deﬁnes the parameters to calculate the helical content of a segment of protein
residues. The <span 
class="zptmcm7m-x-x-109">α</span>-helical content across the <span 
class="zptmcm7m-x-x-109">N </span><span 
class="zptmcm7t-x-x-109">+1 </span>residues <span 
class="zptmcm7m-x-x-109">N</span><sub><span 
class="zptmcm7t-x-x-80">0</span></sub> to <span 
class="zptmcm7m-x-x-109">N</span><sub><span 
class="zptmcm7t-x-x-80">0</span></sub> <span 
class="zptmcm7t-x-x-109">+</span><span 
class="zptmcm7m-x-x-109">N </span>is calculated by the formula:
</p><div class="eqnarray">
   <center class="math-display" >
<img 
src="colvars-refman-namd15x.png" alt=" (  (N)       (N +1)                  (N +5)                   (N +N))
α  Cα 0,O(N0),Cα 0  ,O (N0+1),...N(N0+5),Cα 0  ,O(N0+5),...N(N0+N ),Cα 0    =         (1 0)
           N+N −2    (                )          N +N−4   (          )
   ---1---- 0    angf C(n),C (n+1),C (n+2) + ----1--- 0    hbf  O(n),N(n+4) ,
   2(N− 2)  ∑n=N0        α    α     α      2(N − 4) n∑=N0

                                                                               (1 1)
" class="math-display"  /><a 
 id="x1-48001r10"></a></center>
</div>where the score function for the <span 
class="zptmcm7t-x-x-109">C</span><sub><span 
class="zptmcm7m-x-x-80">α</span></sub><span 
class="zptmcm7y-x-x-109">−</span><span 
class="zptmcm7t-x-x-109">C</span><sub><span 
class="zptmcm7m-x-x-80">α</span></sub><span 
class="zptmcm7y-x-x-109">−</span><span 
class="zptmcm7t-x-x-109">C</span><sub><span 
class="zptmcm7m-x-x-80">α</span></sub> angle is deﬁned as:
   <table 
class="equation"><tr><td><a 
 id="eq:colvars_alpha_Calpha"></a>
   <center class="math-display" >
                                                                                         
                                                                                         
<img 
src="colvars-refman-namd16x.png" alt="                            (                       )
    (                )   1−  𝜃 (C (n),C(n+1),C(n+2))− 𝜃  2 ∕(Δ𝜃  )2
angf C(αn),C (nα+1),C (nα+2)  = ---(----α---α-----α-------0)------tol--,
                         1−  𝜃 (C (n),C(n+1),C(n+2))− 𝜃  4 ∕(Δ𝜃  )4
                                 α   α     α       0       tol
" class="math-display"  /></center></td><td class="equation-label">(12)</td></tr></table>
<!--l. 2589--><p class="nopar" >
and the score function for the <span 
class="zptmcm7t-x-x-109">O</span><sup><span 
class="zptmcm7t-x-x-80">(</span><span 
class="zptmcm7m-x-x-80">n</span><span 
class="zptmcm7t-x-x-80">)</span></sup> <span 
class="zptmcm7y-x-x-109">↔ </span><span 
class="zptmcm7t-x-x-109">N</span><sup><span 
class="zptmcm7t-x-x-80">(</span><span 
class="zptmcm7m-x-x-80">n</span><span 
class="zptmcm7t-x-x-80">+4)</span></sup> hydrogen bond is deﬁned through a <span 
class="ot1-zi4r-0-x-x-109">hBond </span>colvar component on the
same atoms.
</p><!--l. 2589--><p class="noindent" ><span 
class="ptmb7t-x-x-109">List of keywords </span>(see also <a 
href="#sec:cvc_superp">6.4<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="colvar|alpha|psfSegID"></a> <a 
 id="dx1-48003"></a><span 
class="ot1-zi4r-0-x-x-120">residueRange </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Potential <span 
class="zptmcm7m-x-x-109">α</span>-helical residues<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">alpha</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>“<span 
class="zptmcm7m-x-x-109">&#x003C;</span>Initial residue number<span 
class="zptmcm7m-x-x-109">&#x003E;</span>-<span 
class="zptmcm7m-x-x-109">&#x003C;</span>Final residue number<span 
class="zptmcm7m-x-x-109">&#x003E;</span>” <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>This option speciﬁes the range of residues on which this component should be deﬁned.
     The Colvars module looks for the atoms within these residues named “<span 
class="ot1-zi4r-0-x-x-109">CA</span>”, “<span 
class="ot1-zi4r-0-x-x-109">N</span>” and “<span 
class="ot1-zi4r-0-x-x-109">O</span>”, and raises an
     error if any of those atoms is not found.
     </li>
     <li class="itemize"><a 
 id="colvar|alpha|psfSegID"></a> <a 
 id="dx1-48004"></a><span 
class="ot1-zi4r-0-x-x-120">psfSegID </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>PSF segment identiﬁer<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">alpha</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>string (max 4 characters) <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>This option sets the PSF segment identiﬁer for the residues speciﬁed in <span 
class="ot1-zi4r-0-x-x-109">residueRange</span>.
     This option is only required when PSF topologies are used.
     </li>
     <li class="itemize"><a 
 id="colvar|alpha|hBondCoeff"></a> <a 
 id="dx1-48005"></a><span 
class="ot1-zi4r-0-x-x-120">hBondCoeff </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Coeﬃcient for the hydrogen bond term<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">alpha</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive between 0 and 1 <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span>0.5 <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>This number speciﬁes the contribution to the total value from the hydrogen bond terms.
     0 disables the hydrogen bond terms, 1 disables the angle terms.
     </li>
     <li class="itemize"><a 
 id="colvar|alpha|angleRef"></a> <a 
 id="dx1-48006"></a><span 
class="ot1-zi4r-0-x-x-120">angleRef </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Reference <span 
class="zptmcm7t-x-x-109">C</span><sub><span 
class="zptmcm7m-x-x-80">α</span></sub><span 
class="zptmcm7y-x-x-109">−</span><span 
class="zptmcm7t-x-x-109">C</span><sub><span 
class="zptmcm7m-x-x-80">α</span></sub><span 
class="zptmcm7y-x-x-109">−</span><span 
class="zptmcm7t-x-x-109">C</span><sub><span 
class="zptmcm7m-x-x-80">α</span></sub> angle<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">alpha</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive decimal <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span>88<sup><span 
class="zptmcm7y-x-x-80">∘</span></sup><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>This option sets the reference angle used in the score function (<a 
href="#eq:colvars_alpha_Calpha">12<!--tex4ht:ref: eq:colvars_alpha_Calpha --></a>).
     </li>
     <li class="itemize"><a 
 id="colvar|alpha|angleTol"></a> <a 
 id="dx1-48007"></a><span 
class="ot1-zi4r-0-x-x-120">angleTol </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Tolerance in the <span 
class="zptmcm7t-x-x-109">C</span><sub><span 
class="zptmcm7m-x-x-80">α</span></sub><span 
class="zptmcm7y-x-x-109">−</span><span 
class="zptmcm7t-x-x-109">C</span><sub><span 
class="zptmcm7m-x-x-80">α</span></sub><span 
class="zptmcm7y-x-x-109">−</span><span 
class="zptmcm7t-x-x-109">C</span><sub><span 
class="zptmcm7m-x-x-80">α</span></sub> angle<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">alpha</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive decimal <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span>15<sup><span 
class="zptmcm7y-x-x-80">∘</span></sup><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>This option sets the angle tolerance used in the score function (<a 
href="#eq:colvars_alpha_Calpha">12<!--tex4ht:ref: eq:colvars_alpha_Calpha --></a>).
     </li>
     <li class="itemize"><a 
 id="colvar|alpha|hBondCutoff"></a> <a 
 id="dx1-48008"></a><span 
class="ot1-zi4r-0-x-x-120">hBondCutoff </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Hydrogen bond cutoﬀ<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">alpha</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive decimal <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span>3.3 Å <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>Equivalent to the <span 
class="ot1-zi4r-0-x-x-109">cutoff </span>option in the <span 
class="ot1-zi4r-0-x-x-109">hBond </span>component.
     </li>
     <li class="itemize"><a 
 id="colvar|alpha|hBondExpNumer"></a> <a 
 id="dx1-48009"></a><span 
class="ot1-zi4r-0-x-x-120">hBondExpNumer </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Hydrogen bond numerator exponent<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">alpha</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive integer <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span>6 <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>Equivalent to the <span 
class="ot1-zi4r-0-x-x-109">expNumer </span>option in the <span 
class="ot1-zi4r-0-x-x-109">hBond </span>component.
     </li>
     <li class="itemize"><a 
 id="colvar|alpha|hBondExpDenom"></a> <a 
 id="dx1-48010"></a><span 
class="ot1-zi4r-0-x-x-120">hBondExpDenom </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Hydrogen bond denominator exponent<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">alpha</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive integer <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span>8 <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>Equivalent to the <span 
class="ot1-zi4r-0-x-x-109">expDenom </span>option in the <span 
class="ot1-zi4r-0-x-x-109">hBond </span>component.
     </li></ul>
<!--l. 2589--><p class="indent" >   This component returns positive values, always comprised between 0 (lowest <span 
class="zptmcm7m-x-x-109">α</span>-helical score) and 1 (highest
<span 
class="zptmcm7m-x-x-109">α</span>-helical score).
</p><!--l. 2589--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">6.1.29    </span> <a 
 id="x1-490006.1.29"></a><span 
class="ot1-zi4r-0-x-x-109">dihedralPC</span>: protein dihedral pricipal component</h5>
<!--l. 2589--><p class="noindent" >The block <span 
class="ot1-zi4r-0-x-x-109">dihedralPC</span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109">...</span><span 
class="cmsy-10x-x-109">} </span>deﬁnes the parameters to calculate the projection of backbone dihedral angles within
a protein segment onto a <span 
class="ptmri7t-x-x-109">dihedral principal component</span>, following the formalism of dihedral principal component
analysis (dPCA) proposed by Mu et al.<span class="cite">[<a 
href="#XMu2005">4</a>]</span> and documented in detail by Altis et al.<span class="cite">[<a 
href="#XAltis2007">5</a>]</span>. Given a peptide or protein
segment of <span 
class="zptmcm7m-x-x-109">N </span>residues, each with Ramachandran angles <span 
class="zptmcm7m-x-x-109">ϕ</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub> and <span 
class="zptmcm7m-x-x-109">ψ</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub>, dPCA rests on a variance/covariance analysis of
the <span 
class="zptmcm7t-x-x-109">4(</span><span 
class="zptmcm7m-x-x-109">N </span><span 
class="zptmcm7y-x-x-109">−</span><span 
class="zptmcm7t-x-x-109">1) </span>variables <span 
class="zptmcm7t-x-x-109">cos</span><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">ψ</span><sub><span 
class="zptmcm7t-x-x-80">1</span></sub><span 
class="zptmcm7t-x-x-109">)</span><span 
class="zptmcm7m-x-x-109">,</span><span 
class="zptmcm7t-x-x-109">sin</span><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">ψ</span><sub><span 
class="zptmcm7t-x-x-80">1</span></sub><span 
class="zptmcm7t-x-x-109">)</span><span 
class="zptmcm7m-x-x-109">,</span><span 
class="zptmcm7t-x-x-109">cos</span><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">ϕ</span><sub><span 
class="zptmcm7t-x-x-80">2</span></sub><span 
class="zptmcm7t-x-x-109">)</span><span 
class="zptmcm7m-x-x-109">,</span><span 
class="zptmcm7t-x-x-109">sin</span><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">ϕ</span><sub><span 
class="zptmcm7t-x-x-80">2</span></sub><span 
class="zptmcm7t-x-x-109">)</span><img 
src="colvars-refman-namd17x.png" alt="⋅⋅⋅"  class="@cdots"  /><span 
class="zptmcm7t-x-x-109">cos</span><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">ϕ</span><sub><span 
class="zptmcm7m-x-x-80">N</span></sub><span 
class="zptmcm7t-x-x-109">)</span><span 
class="zptmcm7m-x-x-109">,</span><span 
class="zptmcm7t-x-x-109">sin</span><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">ϕ</span><sub><span 
class="zptmcm7m-x-x-80">N</span></sub><span 
class="zptmcm7t-x-x-109">)</span>. Note that angles <span 
class="zptmcm7m-x-x-109">ϕ</span><sub><span 
class="zptmcm7t-x-x-80">1</span></sub> and <span 
class="zptmcm7m-x-x-109">ψ</span><sub><span 
class="zptmcm7m-x-x-80">N</span></sub> have
little impact on chain conformation, and are therefore discarded, following the implementation of dPCA in the
analysis software Carma.<span class="cite">[<a 
href="#XGlykos2006">6</a>]</span>
</p><!--l. 2589--><p class="indent" >   For a given principal component (eigenvector) of coeﬃcients <span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">k</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub><span 
class="zptmcm7t-x-x-109">)</span><sub><span 
class="zptmcm7t-x-x-80">1</span><span 
class="zptmcm7y-x-x-80">≤</span><span 
class="zptmcm7m-x-x-80">i</span><span 
class="zptmcm7y-x-x-80">≤</span><span 
class="zptmcm7t-x-x-80">4(</span><span 
class="zptmcm7m-x-x-80">N</span><span 
class="zptmcm7y-x-x-80">−</span><span 
class="zptmcm7t-x-x-80">1)</span></sub>, the projection of the current
backbone conformation is:
</p>
   <table 
class="equation"><tr><td><a 
 id="x1-49001r13"></a>
   <center class="math-display" >
<img 
src="colvars-refman-namd18x.png" alt="   N− 1
ξ =    k   cos(ψ )+ k    sin(ψ  )+ k   cos(ϕ   )+ k  sin(ϕ   )
    ∑n=1  4n−3     n    4n−2     n   4n−1     n+1    4n     n+1
                                                                                         
                                                                                         
" class="math-display"  /></center></td><td class="equation-label">(13)</td></tr></table>
<!--l. 2589--><p class="nopar" >
</p><!--l. 2589--><p class="indent" >   <span 
class="ot1-zi4r-0-x-x-109">dihedralPC </span>expects the same parameters as the <span 
class="ot1-zi4r-0-x-x-109">alpha </span>component for deﬁning the relevant residues
(<span 
class="ot1-zi4r-0-x-x-109">residueRange </span>and <span 
class="ot1-zi4r-0-x-x-109">psfSegID</span>) in addition to the following:
</p><!--l. 2589--><p class="noindent" ><span 
class="ptmb7t-x-x-109">List of keywords </span>(see also <a 
href="#sec:cvc_superp">6.4<!--tex4ht:ref: sec:cvc_superp --></a> for additional options): </p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-49002"></a> <span 
class="ot1-zi4r-0-x-x-120">residueRange: </span>see deﬁnition of <a 
href="#colvar|alpha|psfSegID"><span 
class="ot1-zi4r-0-x-x-109">residueRange</span></a> in sec. <a 
href="#colvar|alpha|psfSegID">6.1.28<!--tex4ht:ref: colvar|alpha|hBondCoeff --></a> (<span 
class="ot1-zi4r-0-x-x-109">alpha </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-49003"></a> <span 
class="ot1-zi4r-0-x-x-120">psfSegID: </span>see deﬁnition of <a 
href="#colvar|alpha|psfSegID"><span 
class="ot1-zi4r-0-x-x-109">psfSegID</span></a> in sec. <a 
href="#colvar|alpha|psfSegID">6.1.28<!--tex4ht:ref: colvar|alpha|psfSegID --></a> (<span 
class="ot1-zi4r-0-x-x-109">alpha </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-49004"></a> <span 
class="ot1-zi4r-0-x-x-120">hBondCoeff: </span>see deﬁnition of <a 
href="#colvar|alpha|hBondCoeff"><span 
class="ot1-zi4r-0-x-x-109">hBondCoeff</span></a> in sec. <a 
href="#colvar|alpha|psfSegID">6.1.28<!--tex4ht:ref: colvar|alpha|hBondCoeff --></a> (<span 
class="ot1-zi4r-0-x-x-109">alpha </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-49005"></a> <span 
class="ot1-zi4r-0-x-x-120">angleRef: </span>see deﬁnition of <a 
href="#colvar|alpha|angleRef"><span 
class="ot1-zi4r-0-x-x-109">angleRef</span></a> in sec. <a 
href="#colvar|alpha|psfSegID">6.1.28<!--tex4ht:ref: colvar|alpha|angleRef --></a> (<span 
class="ot1-zi4r-0-x-x-109">alpha </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-49006"></a> <span 
class="ot1-zi4r-0-x-x-120">angleTol: </span>see deﬁnition of <a 
href="#colvar|alpha|angleTol"><span 
class="ot1-zi4r-0-x-x-109">angleTol</span></a> in sec. <a 
href="#colvar|alpha|psfSegID">6.1.28<!--tex4ht:ref: colvar|alpha|angleTol --></a> (<span 
class="ot1-zi4r-0-x-x-109">alpha </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-49007"></a> <span 
class="ot1-zi4r-0-x-x-120">hBondCutoff: </span>see deﬁnition of <a 
href="#colvar|alpha|hBondCutoff"><span 
class="ot1-zi4r-0-x-x-109">hBondCutoff</span></a> in sec. <a 
href="#colvar|alpha|psfSegID">6.1.28<!--tex4ht:ref: colvar|alpha|hBondCutoff --></a> (<span 
class="ot1-zi4r-0-x-x-109">alpha </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-49008"></a> <span 
class="ot1-zi4r-0-x-x-120">hBondExpNumer: </span>see deﬁnition of <a 
href="#colvar|alpha|hBondExpNumer"><span 
class="ot1-zi4r-0-x-x-109">hBondExpNumer</span></a> in sec. <a 
href="#colvar|alpha|psfSegID">6.1.28<!--tex4ht:ref: colvar|alpha|hBondExpNumer --></a> (<span 
class="ot1-zi4r-0-x-x-109">alpha </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-49009"></a> <span 
class="ot1-zi4r-0-x-x-120">hBondExpDenom: </span>see deﬁnition of <a 
href="#colvar|alpha|hBondExpDenom"><span 
class="ot1-zi4r-0-x-x-109">hBondExpDenom</span></a> in sec. <a 
href="#colvar|alpha|psfSegID">6.1.28<!--tex4ht:ref: colvar|alpha|hBondExpDenom --></a> (<span 
class="ot1-zi4r-0-x-x-109">alpha </span>component)
     </li>
     <li class="itemize"><a 
 id="dx1-49010"></a> <span 
class="ot1-zi4r-0-x-x-120">vectorFile </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>File containing dihedral PCA eigenvector(s)<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">dihedralPC</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>ﬁle name <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>A text ﬁle containing the coeﬃcients of dihedral PCA eigenvectors on the cosine and
     sine coordinates. The vectors should be arranged in columns, as in the ﬁles output by Carma.<span class="cite">[<a 
href="#XGlykos2006">6</a>]</span>
     </li>
     <li class="itemize"><a 
 id="dx1-49011"></a> <span 
class="ot1-zi4r-0-x-x-120">vectorNumber </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>File containing dihedralPCA eigenvector(s)<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">dihedralPC</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive integer <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>Number of the eigenvector to be used for this component. </li></ul>
<!--l. 2592--><p class="noindent" >
</p>
   <h4 class="subsectionHead"><span class="titlemark">6.2    </span> <a 
 id="x1-500006.2"></a>Conﬁguration keywords shared by all components</h4>
<!--l. 2595--><p class="noindent" >The following options can be used for any of the above colvar components in order to obtain a polynomial
combination or any user-supplied function provided by <a 
href="#sec:cvc_superp"><span 
class="ot1-zi4r-0-x-x-109">scriptedFunction</span></a>. </p>
                                                                                         
                                                                                         
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-50001"></a> <span 
class="ot1-zi4r-0-x-x-120">name </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Name of this component<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span>any component <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>string <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span>type of component + numeric id <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:    </span>The  name  is  an  unique  case-sensitive  string  which  allows  the  Colvars  module  to
     identify  this  component.  This  is  useful,  for  example,  when  combining  multiple  components  via
     a  <span 
class="ot1-zi4r-0-x-x-109">scriptedFunction</span>.  It  also  deﬁnes  the  variable  name  representing  the  component&#x2019;s  value  in  a
     <span 
class="ot1-zi4r-0-x-x-109">customFunction </span>expression.
     </li>
     <li class="itemize"><a 
 id="dx1-50002"></a> <span 
class="ot1-zi4r-0-x-x-120">scalable </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Attempt to calculate this component in parallel?<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span>any component <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">on</span>, if available <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>If  set  to  <span 
class="ot1-zi4r-0-x-x-109">on </span>(default),  the  Colvars  module  will  attempt  to  calculate  this  component
     in parallel to reduce overhead. Whether this option is available depends on the type of component:
     currently supported are <span 
class="ot1-zi4r-0-x-x-109">distance</span>, <span 
class="ot1-zi4r-0-x-x-109">distanceZ</span>, <span 
class="ot1-zi4r-0-x-x-109">distanceXY</span>, <span 
class="ot1-zi4r-0-x-x-109">distanceVec</span>, <span 
class="ot1-zi4r-0-x-x-109">distanceDir</span>, <span 
class="ot1-zi4r-0-x-x-109">angle </span>and
     <span 
class="ot1-zi4r-0-x-x-109">dihedral</span>. This ﬂag inﬂuences computational cost, but does not aﬀect numerical results: therefore, it
     should only be turned oﬀ for debugging or testing purposes.  </li></ul>
<!--l. 2623--><p class="noindent" >
</p>
   <h4 class="subsectionHead"><span class="titlemark">6.3    </span> <a 
 id="x1-510006.3"></a>Advanced usage and special considerations</h4>
<!--l. 2626--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">6.3.1    </span> <a 
 id="x1-520006.3.1"></a>Periodic components.</h5>
<!--l. 2628--><p class="noindent" >The following components returns real numbers that lie in a periodic interval: </p>
     <ul class="itemize1">
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">dihedral</span>: torsional angle between four groups;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">spinAngle</span>:  angle  of  rotation  around  a  predeﬁned  axis  in  the  best-ﬁt  from  a  set  of  reference
     coordinates.</li></ul>
<!--l. 2635--><p class="noindent" >In certain conditions, <span 
class="ot1-zi4r-0-x-x-109">distanceZ </span>can also be periodic, namely when periodic boundary conditions (PBCs) are deﬁned in
the simulation and <span 
class="ot1-zi4r-0-x-x-109">distanceZ</span>&#x2019;s axis is parallel to a unit cell vector.
</p><!--l. 2639--><p class="indent" >   The following keywords can be used within periodic components (and are illegal elsewhere):
</p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-52001"></a> <span 
class="ot1-zi4r-0-x-x-120">period </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Period of the component<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">distanceZ</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive decimal <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span>0.0 <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>Setting  this  number  enables  the  treatment  of  <span 
class="ot1-zi4r-0-x-x-109">distanceZ </span>as  a  periodic  component:
     by default, <span 
class="ot1-zi4r-0-x-x-109">distanceZ </span>is not considered periodic. The keyword is supported, but irrelevant within
     <span 
class="ot1-zi4r-0-x-x-109">dihedral </span>or <span 
class="ot1-zi4r-0-x-x-109">spinAngle</span>, because their period is always 360 degrees.
     </li>
     <li class="itemize"><a 
 id="dx1-52002"></a> <span 
class="ot1-zi4r-0-x-x-120">wrapAround </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Center of the wrapping interval for periodic variables<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">distanceZ</span>, <span 
class="ot1-zi4r-0-x-x-109">dihedral </span>or <span 
class="ot1-zi4r-0-x-x-109">spinAngle</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>decimal <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span>0.0 <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>By default, values of the periodic components are centered around zero, ranging from
     <span 
class="zptmcm7y-x-x-109">−</span><span 
class="zptmcm7m-x-x-109">P∕</span><span 
class="zptmcm7t-x-x-109">2 </span>to <span 
class="zptmcm7m-x-x-109">P∕</span><span 
class="zptmcm7t-x-x-109">2</span>, where <span 
class="zptmcm7m-x-x-109">P </span>is the period. Setting this number centers the interval around this value. This
     can be useful for convenience of output, or to set the walls for a <span 
class="ot1-zi4r-0-x-x-109">harmonicWalls </span>in an order that
     would not otherwise be allowed. </li></ul>
<!--l. 2668--><p class="noindent" >Internally, all diﬀerences between two values of a periodic colvar follow the minimum image convention: they are
calculated based on the two periodic images that are closest to each other.
</p><!--l. 2676--><p class="indent" >   <span 
class="ptmri7t-x-x-109">Note: linear or polynomial combinations of periodic components may become meaningless when components</span>
<span 
class="ptmri7t-x-x-109">cross the periodic boundary. Use such combinations carefully: estimate the range of possible values of</span>
<span 
class="ptmri7t-x-x-109">each component in a given simulation, and make use of  </span><span 
class="ot1-zi4r-0-x-x-109">wrapAround </span><span 
class="ptmri7t-x-x-109">to limit this problem whenever</span>
<span 
class="ptmri7t-x-x-109">possible.</span>
</p><!--l. 2679--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">6.3.2    </span> <a 
 id="x1-530006.3.2"></a>Non-scalar components.</h5>
<!--l. 2681--><p class="noindent" >When one of the following components are used, the deﬁned colvar returns a value that is not a scalar number:
</p>
     <ul class="itemize1">
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">distanceVec</span>: 3-dimensional vector of the distance between two groups;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">distanceDir</span>: 3-dimensional unit vector of the distance between two groups;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">orientation</span>: 4-dimensional unit quaternion representing the best-ﬁt rotation from a set of reference
     coordinates.</li></ul>
<!--l. 2690--><p class="noindent" >The distance between two 3-dimensional unit vectors is computed as the angle between them. The distance between two
quaternions is computed as the angle between the two 4-dimensional unit vectors: because the orientation
represented by <span 
class="cmss-10x-x-109">q </span>is the same as the one represented by <span 
class="zptmcm7y-x-x-109">−</span><span 
class="cmss-10x-x-109">q</span>, distances between two quaternions are computed
considering the closest of the two symmetric images.
</p><!--l. 2697--><p class="indent" >   Non-scalar components carry the following restrictions: </p>
     <ul class="itemize1">
     <li class="itemize">Calculation of total forces (<span 
class="ot1-zi4r-0-x-x-109">outputTotalForce </span>option) is currently not implemented.
     </li>
     <li class="itemize">Each colvar can only contain one non-scalar component.
                                                                                         
                                                                                         
     </li>
     <li class="itemize">Binning  on  a  grid  (<span 
class="ot1-zi4r-0-x-x-109">abf</span>,  <span 
class="ot1-zi4r-0-x-x-109">histogram </span>and  <span 
class="ot1-zi4r-0-x-x-109">metadynamics </span>with  <span 
class="ot1-zi4r-0-x-x-109">useGrids </span>enabled)  is  currently  not
     implemented for colvars based on such components.</li></ul>
<!--l. 2710--><p class="indent" >   <span 
class="ptmri7t-x-x-109">Note: while these restrictions apply to individual colvars based on non-scalar components, no limit is set to the</span>
<span 
class="ptmri7t-x-x-109">number of scalar colvars. To compute multi-dimensional histograms and PMFs, use sets of scalar colvars of</span>
<span 
class="ptmri7t-x-x-109">arbitrary size.</span>
</p><!--l. 2713--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">6.3.3    </span> <a 
 id="sec:cvc_sys_forces"></a>Calculating total forces.</h5>
<!--l. 2715--><p class="noindent" >In addition to the restrictions due to the type of value computed (scalar or non-scalar), a ﬁnal restriction can arise
when calculating total force (<span 
class="ot1-zi4r-0-x-x-109">outputTotalForce </span>option or application of a <span 
class="ot1-zi4r-0-x-x-109">abf </span>bias). total forces are available
currently only for the following components: <span 
class="ot1-zi4r-0-x-x-109">distance</span>, <span 
class="ot1-zi4r-0-x-x-109">distanceZ</span>, <span 
class="ot1-zi4r-0-x-x-109">distanceXY</span>, <span 
class="ot1-zi4r-0-x-x-109">angle</span>, <span 
class="ot1-zi4r-0-x-x-109">dihedral</span>, <span 
class="ot1-zi4r-0-x-x-109">rmsd</span>,
<span 
class="ot1-zi4r-0-x-x-109">eigenvector </span>and <span 
class="ot1-zi4r-0-x-x-109">gyration</span>.
</p><!--l. 2725--><p class="noindent" >
</p>
   <h4 class="subsectionHead"><span class="titlemark">6.4    </span> <a 
 id="sec:cvc_superp"></a>Linear and polynomial combinations of components</h4>
<!--l. 2728--><p class="noindent" >To extend the set of possible deﬁnitions of colvars <span 
class="zptmcm7m-x-x-109">ξ</span><span 
class="zptmcm7t-x-x-109">(</span><span 
class="ptmb7t-x-x-109">r</span><span 
class="zptmcm7t-x-x-109">)</span>, multiple components <span 
class="zptmcm7m-x-x-109">q</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub><span 
class="zptmcm7t-x-x-109">(</span><span 
class="ptmb7t-x-x-109">r</span><span 
class="zptmcm7t-x-x-109">) </span>can be summed with the
formula:
</p>
   <table 
class="equation"><tr><td><a 
 id="eq:colvar_combination"></a>
   <center class="math-display" >
<img 
src="colvars-refman-namd19x.png" alt="                n
ξ(r)=  ∑ ci[qi(r)]i
       i
" class="math-display"  /></center></td><td class="equation-label">(14)</td></tr></table>
<!--l. 2733--><p class="nopar" >
where each component appears with a unique coeﬃcient <span 
class="zptmcm7m-x-x-109">c</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub> (1.0 by default) the positive integer exponent <span 
class="zptmcm7m-x-x-109">n</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub> (1 by
default).
</p><!--l. 2737--><p class="indent" >   Any set of components can be combined within a colvar, provided that they return the same type of values
(scalar, unit vector, vector, or quaternion). By default, the colvar is the sum of its components. Linear or polynomial
combinations (following equation (<a 
href="#eq:colvar_combination">14<!--tex4ht:ref: eq:colvar_combination --></a>)) can be obtained by setting the following parameters, which are common to
all components: </p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-55002"></a> <span 
class="ot1-zi4r-0-x-x-120">componentCoeff </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Coeﬃcient of this component in the colvar<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span>any component <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>decimal <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">1.0</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>Deﬁnes the coeﬃcient by which this component is multiplied (after being raised to
     <span 
class="ot1-zi4r-0-x-x-109">componentExp</span>) before being added to the sum.
     </li>
     <li class="itemize"><a 
 id="dx1-55003"></a> <span 
class="ot1-zi4r-0-x-x-120">componentExp </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Exponent of this component in the colvar<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span>any component <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>integer <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">1</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>Deﬁnes the power at which the value of this component is raised before being added to
     the sum. When this exponent is diﬀerent than 1 (non-linear sum), total forces and the Jacobian force
     are not available, making the colvar unsuitable for ABF calculations. </li></ul>
<!--l. 2768--><p class="indent" >   <span 
class="ptmb7t-x-x-109">Example: </span>To deﬁne the <span 
class="ptmri7t-x-x-109">average </span>of a colvar across diﬀerent parts of the system, simply deﬁne within the same
colvar block a series of components of the same type (applied to diﬀerent atom groups), and assign to each
component a <span 
class="ot1-zi4r-0-x-x-109">componentCoeff </span>of <span 
class="zptmcm7t-x-x-109">1</span><span 
class="zptmcm7m-x-x-109">∕N</span>.
</p><!--l. 2835--><p class="noindent" >
</p>
   <h4 class="subsectionHead"><span class="titlemark">6.5    </span> <a 
 id="x1-560006.5"></a>Colvars as custom functions of components</h4>
<!--l. 2835--><p class="noindent" >Collective variables may be deﬁned by specifying a custom function as an analytical expression such as <span 
class="ot1-zi4r-0-x-x-109">cos(x) +</span>
<span 
class="ot1-zi4r-0-x-x-109">y^2</span>. The expression is parsed by the Lepton expression parser (written by Peter Eastman), which produces eﬃcient
evaluation routines for the function itself as well as its derivatives. The expression may use the collective variable
components as variables, refered to as their <span 
class="ot1-zi4r-0-x-x-109">name </span>string. Scalar elements of vector components may
be accessed by appending a 1-based index to their <span 
class="ot1-zi4r-0-x-x-109">name</span>. When implementing generic functions of
Cartesian coordinates rather than functions of existing components, the <span 
class="ot1-zi4r-0-x-x-109">cartesian </span>component may be
particularly useful. A vector variable may be deﬁned by specifying the <span 
class="ot1-zi4r-0-x-x-109">customFunction </span>parameter several
times: each expression deﬁnes one scalar element of the vector colvar. This is illustrated in the example
below.
</p><!--l. 2835--><p class="noindent" ><span 
class="ot1-zi4r-0-x-x-109">colvar </span><span 
class="cmsy-10x-x-109">{</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> name custom</span><br 
class="newline" /><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> # A 2-dimensional vector function of a scalar x and a 3-vector r</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> customFunction cos(x) * (r1 + r2 + r3)</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> customFunction sqrt(r1 * r2)</span><br 
class="newline" /><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> distance </span><span 
class="cmsy-10x-x-109">{</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> name x</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> group1 </span><span 
class="cmsy-10x-x-109">{ </span><span 
class="ot1-zi4r-0-x-x-109">atomNumbers 1 </span><span 
class="cmsy-10x-x-109">}</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> group2 </span><span 
class="cmsy-10x-x-109">{ </span><span 
class="ot1-zi4r-0-x-x-109">atomNumbers 50 </span><span 
class="cmsy-10x-x-109">}</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">}</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> distanceVec </span><span 
class="cmsy-10x-x-109">{</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> name r</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> group1 </span><span 
class="cmsy-10x-x-109">{ </span><span 
class="ot1-zi4r-0-x-x-109">atomNumbers 10 11 12 </span><span 
class="cmsy-10x-x-109">}</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> group2 </span><span 
class="cmsy-10x-x-109">{ </span><span 
class="ot1-zi4r-0-x-x-109">atomNumbers 20 21 22 </span><span 
class="cmsy-10x-x-109">}</span><br 
class="newline" /><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">}</span><br 
class="newline" /><span 
class="cmsy-10x-x-109">}</span>
                                                                                         
                                                                                         
</p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="colvar|customFunction"></a><a 
 id="dx1-56001"></a><span 
class="ot1-zi4r-0-x-x-120">customFunction </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Compute colvar as a custom function of its components<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">colvar</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>string <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>Deﬁnes the colvar as a scalar expression of its colvar components. Multiple mentions
     can be used to deﬁne a vector variable (as in the example above).
     </li>
     <li class="itemize"><a 
 id="dx1-56002"></a> <span 
class="ot1-zi4r-0-x-x-120">customFunctionType </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Type of value returned by the scripted colvar<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">colvar</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>string <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">scalar</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>With this ﬂag, the user may specify whether the colvar is a scalar or one of the following
     vector types: <span 
class="ot1-zi4r-0-x-x-109">vector3 </span>(a 3D vector), <span 
class="ot1-zi4r-0-x-x-109">unit</span><span 
class="ot1-zi4r-0-x-x-109">_vector3 </span>(a normalized 3D vector), or <span 
class="ot1-zi4r-0-x-x-109">unit</span><span 
class="ot1-zi4r-0-x-x-109">_quaternion</span>
     (a normalized quaternion), or <span 
class="ot1-zi4r-0-x-x-109">vector</span>. Note that the scalar and vector cases are not necessary, as they
     are detected automatically. </li></ul>
<!--l. 2904--><p class="noindent" >
</p>
   <h4 class="subsectionHead"><span class="titlemark">6.6    </span> <a 
 id="x1-570006.6"></a>Colvars as scripted functions of components</h4>
<!--l. 2904--><p class="noindent" >When scripting is supported (default in NAMD), a colvar may be deﬁned as a scripted function of its components,
rather than a linear or polynomial combination. When implementing generic functions of Cartesian
coordinates rather than functions of existing components, the <span 
class="ot1-zi4r-0-x-x-109">cartesian </span>component may be particularly
useful.
</p><!--l. 2904--><p class="indent" >   An example of elaborate scripted colvar is given in example 10, in the form of path-based collective
variables as deﬁned by Branduardi et al<span class="cite">[<a 
href="#XBranduardi2007">7</a>]</span>. The required Tcl procedures are provided in the <span 
class="ot1-zi4r-0-x-x-109">colvartools</span>
directory.
</p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="colvar|scriptedFunction"></a> <a 
 id="dx1-57001"></a><span 
class="ot1-zi4r-0-x-x-120">scriptedFunction </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Compute colvar as a scripted function of its components<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">colvar</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>string <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:    </span>If  this  option  is  speciﬁed,  the  colvar  will  be  computed  as  a  scripted  function
     of  the  values  of  its  components.  To  that  eﬀect,  the  user  should  deﬁne  two  Tcl  procedures:
     <span 
class="ot1-zi4r-0-x-x-109">calc</span><span 
class="ot1-zi4r-0-x-x-109">_</span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ot1-zi4r-0-x-x-109">scriptedFunction</span><span 
class="zptmcm7m-x-x-109">&#x003E; </span>and <span 
class="ot1-zi4r-0-x-x-109">calc</span><span 
class="ot1-zi4r-0-x-x-109">_</span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ot1-zi4r-0-x-x-109">scriptedFunction</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span><span 
class="ot1-zi4r-0-x-x-109">_gradient</span>, both accepting as many
     parameters as the colvar has components. Values of the components will be passed to those procedures
     in  the  order  deﬁned  by  their  sorted  <span 
class="ot1-zi4r-0-x-x-109">name </span>strings.  Note  that  if  all  components  are  of  the  same
     type,  their  default  names  are  sorted  in  the  order  in  which  they  are  deﬁned,  so  that  names  need
     only be speciﬁed for combinations of components of diﬀerent types. <span 
class="ot1-zi4r-0-x-x-109">calc</span><span 
class="ot1-zi4r-0-x-x-109">_</span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ot1-zi4r-0-x-x-109">scriptedFunction</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span>
     should  return  one  value  of  type  <span 
class="zptmcm7m-x-x-109">&#x003C;</span>scriptedFunctionType<span 
class="zptmcm7m-x-x-109">&#x003E;</span>,  corresponding  to  the  colvar  value.
     <span 
class="ot1-zi4r-0-x-x-109">calc</span><span 
class="ot1-zi4r-0-x-x-109">_</span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ot1-zi4r-0-x-x-109">scriptedFunction</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span><span 
class="ot1-zi4r-0-x-x-109">_gradient </span>should  return  a  Tcl  list  containing  the  derivatives  of  the
     function with respect to each component. If both the function and some of the components are vectors,
                                                                                         
                                                                                         
     the gradient is really a Jacobian matrix that should be passed as a linear vector in row-major order, i.e.
     for a function <span 
class="zptmcm7m-x-x-109">f</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">x</span><sub><span 
class="zptmcm7m-x-x-80">j</span></sub><span 
class="zptmcm7t-x-x-109">)</span>: <span 
class="zptmcm7y-x-x-109">∇</span><sub><span 
class="zptmcm7m-x-x-80">x</span></sub><span 
class="zptmcm7m-x-x-109">f</span><sub><span 
class="zptmcm7t-x-x-80">1</span></sub><span 
class="zptmcm7y-x-x-109">∇</span><sub><span 
class="zptmcm7m-x-x-80">x</span></sub><span 
class="zptmcm7m-x-x-109">f</span><sub><span 
class="zptmcm7t-x-x-80">2</span></sub><img 
src="colvars-refman-namd20x.png" alt="⋅⋅⋅"  class="@cdots"  /><span style="margin-left:0.3em" class="thinspace"></span>.
     </li>
     <li class="itemize"><a 
 id="dx1-57002"></a> <span 
class="ot1-zi4r-0-x-x-120">scriptedFunctionType </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Type of value returned by the scripted colvar<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">colvar</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>string <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">scalar</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>If a colvar is deﬁned as a scripted function, its type is not constrained by the types
     of  its  components.  With  this  ﬂag,  the  user  may  specify  whether  the  colvar  is  a  scalar  or  one
     of  the  following  vector  types:  <span 
class="ot1-zi4r-0-x-x-109">vector3 </span>(a  3D  vector),  <span 
class="ot1-zi4r-0-x-x-109">unit</span><span 
class="ot1-zi4r-0-x-x-109">_vector3 </span>(a  normalized  3D  vector),
     or  <span 
class="ot1-zi4r-0-x-x-109">unit</span><span 
class="ot1-zi4r-0-x-x-109">_quaternion </span>(a  normalized  quaternion),  or  <span 
class="ot1-zi4r-0-x-x-109">vector </span>(a  vector  whose  size  is  speciﬁed  by
     <span 
class="ot1-zi4r-0-x-x-109">scriptedFunctionVectorSize</span>). Non-scalar values should be passed as space-separated lists.
     </li>
     <li class="itemize"><a 
 id="dx1-57003"></a> <span 
class="ot1-zi4r-0-x-x-120">scriptedFunctionVectorSize </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Dimension of the vector value of a scripted colvar<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">colvar</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive integer <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>This parameter is only valid when <span 
class="ot1-zi4r-0-x-x-109">scriptedFunctionType </span>is set to <span 
class="ot1-zi4r-0-x-x-109">vector</span>. It deﬁnes
     the vector length of the colvar value returned by the function.</li></ul>
<!--l. 2907--><p class="noindent" >
</p>
   <h3 class="sectionHead"><span class="titlemark">7    </span> <a 
 id="sec:colvarbias"></a>Biasing and analysis methods</h3>
<!--l. 2910--><p class="noindent" >All of the biasing and analysis methods implemented (<span 
class="ot1-zi4r-0-x-x-109">abf</span>, <span 
class="ot1-zi4r-0-x-x-109">harmonic</span>, <span 
class="ot1-zi4r-0-x-x-109">histogram </span>and <span 
class="ot1-zi4r-0-x-x-109">metadynamics</span>) recognize the
following options: </p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-58001"></a> <span 
class="ot1-zi4r-0-x-x-120">name </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Identiﬁer for the bias<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span>colvar bias <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>string <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ot1-zi4r-0-x-x-109">type of bias</span><span 
class="zptmcm7m-x-x-109">&#x003E;&#x003C;</span><span 
class="ot1-zi4r-0-x-x-109">bias index</span><span 
class="zptmcm7m-x-x-109">&#x003E; </span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>This string is used to identify the bias or analysis method in output messages and to
     name some output ﬁles.
     </li>
     <li class="itemize"><a 
 id="dx1-58002"></a> <span 
class="ot1-zi4r-0-x-x-120">colvars </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Collective variables involved<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span>colvar bias <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>space-separated list of colvar names <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>This option selects by name all the colvars to which this bias or analysis will be applied.
     </li>
     <li class="itemize"><a 
 id="dx1-58003"></a> <span 
class="ot1-zi4r-0-x-x-120">outputEnergy </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Write the current bias energy to the trajectory ﬁle<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span>colvar bias <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">off</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>If this option is chosen and <span 
class="ot1-zi4r-0-x-x-109">colvarsTrajFrequency </span>is not zero, the current value of
     the biasing energy will be written to the trajectory ﬁle during the simulation.
                                                                                         
                                                                                         
     </li></ul>
<!--l. 2946--><p class="indent" >   In addition, restraint biases (<a 
href="#sec:colvarbias_harmonic">7.4<!--tex4ht:ref: sec:colvarbias_harmonic --></a>, <a 
href="#sec:colvarbias_harmonic_walls">7.5<!--tex4ht:ref: sec:colvarbias_harmonic_walls --></a>, <a 
href="#sec:colvarbias_linear">7.6<!--tex4ht:ref: sec:colvarbias_linear --></a>, ...) and metadynamics biases (<a 
href="#sec:colvarbias_meta">7.3<!--tex4ht:ref: sec:colvarbias_meta --></a>) oﬀer the following optional
keywords, which allow the use of thermodynamic integration (TI) to compute potentials of mean force (PMFs). In
adaptive biasing force (ABF) biases (<a 
href="#sec:colvarbias_abf">7.1<!--tex4ht:ref: sec:colvarbias_abf --></a>) the same keywords are not recognized because their functionality is
always included.
</p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-58004"></a> <span 
class="ot1-zi4r-0-x-x-120">writeTIPMF </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Write the PMF computed by thermodynamic integration<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span>colvar bias <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">off</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:    </span>If  the  bias  is  applied  to  a  variable  that  supports  the  calculation  of  total  forces
     (see  <a 
href="#sec:colvar"><span 
class="ot1-zi4r-0-x-x-109">outputTotalForce</span></a> and  <a 
href="#sec:cvc_sys_forces">6.3.3<!--tex4ht:ref: sec:cvc_sys_forces --></a>),  this  option  allows  calculating  the  corresponding  PMF  by
     thermodynanic integration, and writing it to the ﬁle <span 
class="ptmri7t-x-x-109">outputName</span><span 
class="ot1-zi4r-0-x-x-109">.</span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ot1-zi4r-0-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span><span 
class="ot1-zi4r-0-x-x-109">.ti.pmf</span>, where <span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ot1-zi4r-0-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span>
     is the name of the bias. The total force includes the forces applied to the variable by all bias, except
     those from this bias itself. If any bias applies time-dependent forces besides the one using this option,
     an error is raised.
     </li>
     <li class="itemize"><a 
 id="dx1-58005"></a> <span 
class="ot1-zi4r-0-x-x-120">writeTISamples </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Write the free-energy gradient samples<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span>colvar bias <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">off</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:      </span>This   option   allows   to   compute   total   forces   for   use   with   thermodynamic
     integration   as   done   by   the   keyword   <a 
href="#sec:colvarbias"><span 
class="ot1-zi4r-0-x-x-109">writeTIPMF</span></a>.   The   names   of   the   ﬁles   containing   the
     variables&#x2019;  histogram  and  mean  thermodyunamic  forces  are  <span 
class="ptmri7t-x-x-109">outputName</span><span 
class="ot1-zi4r-0-x-x-109">.</span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ot1-zi4r-0-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span><span 
class="ot1-zi4r-0-x-x-109">.ti.count </span>and
     <span 
class="ptmri7t-x-x-109">outputName</span><span 
class="ot1-zi4r-0-x-x-109">.</span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ot1-zi4r-0-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span><span 
class="ot1-zi4r-0-x-x-109">.ti.grad</span>, respectively: these can be used by <span 
class="ot1-zi4r-0-x-x-109">abf</span><span 
class="ot1-zi4r-0-x-x-109">_integrate </span>or similar utility.
     This option on by default when <span 
class="ot1-zi4r-0-x-x-109">writeTIPMF </span>is on, but can be enabled separately if the bias is applied
     to more than one variable, making not possible the direct integration of the PMF at runtime. If any
     bias applies time-dependent forces besides the one using this option, an error is raised.
     </li></ul>
<!--l. 2979--><p class="noindent" >
</p>
   <h4 class="subsectionHead"><span class="titlemark">7.1    </span> <a 
 id="sec:colvarbias_abf"></a>Adaptive Biasing Force</h4>
<!--l. 2983--><p class="noindent" >For a full description of the Adaptive Biasing Force method, see reference <span class="cite">[<a 
href="#XDarve2008">8</a>]</span>. For details about this
implementation, see references <span class="cite">[<a 
href="#XHenin2004">9</a>]</span> and <span class="cite">[<a 
href="#XHenin2010">10</a>]</span>. <span 
class="ptmb7t-x-x-109">When publishing research that makes use of this functionality,</span>
<span 
class="ptmb7t-x-x-109">please cite references</span><span 
class="ptmb7t-x-x-109"> </span><span class="cite"><span 
class="ptmb7t-x-x-109">[</span><a 
href="#XDarve2008"><span 
class="ptmb7t-x-x-109">8</span></a><span 
class="ptmb7t-x-x-109">]</span></span> <span 
class="ptmb7t-x-x-109">and </span><span class="cite"><span 
class="ptmb7t-x-x-109">[</span><a 
href="#XHenin2010"><span 
class="ptmb7t-x-x-109">10</span></a><span 
class="ptmb7t-x-x-109">]</span></span><span 
class="ptmb7t-x-x-109">.</span>
</p><!--l. 2989--><p class="indent" >   An alternate usage of this feature is the application of custom tabulated biasing potentials to one or more colvars.
See <span 
class="ot1-zi4r-0-x-x-109">inputPrefix </span>and <span 
class="ot1-zi4r-0-x-x-109">updateBias </span>below.
</p><!--l. 2993--><p class="indent" >   Combining ABF with the extended Lagrangian feature (<a 
href="#sec:colvar_extended">4.3<!--tex4ht:ref: sec:colvar_extended --></a>) of the variables produces the extended-system ABF
variant of the method (<a 
href="#sec:eABF">7.2<!--tex4ht:ref: sec:eABF --></a>).
</p><!--l. 2997--><p class="indent" >   ABF is based on the thermodynamic integration (TI) scheme for computing free energy proﬁles. The free energy
as a function of a set of collective variables <span 
class="zptmcm7m-x-x-109">ξ</span> <span 
class="zptmcm7t-x-x-109">= (</span><span 
class="zptmcm7m-x-x-109">ξ</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub><span 
class="zptmcm7t-x-x-109">)</span><sub><span 
class="zptmcm7m-x-x-80">i</span><span 
class="zptmcm7y-x-x-80">∈</span><span 
class="zptmcm7t-x-x-80">[1</span><span 
class="zptmcm7m-x-x-80">,n</span><span 
class="zptmcm7t-x-x-80">]</span></sub> is deﬁned from the canonical distribution of <span 
class="zptmcm7m-x-x-109">ξ</span> ,
                                                                                         
                                                                                         
<span 
class="zptmcm7y-x-x-109">𝒫</span><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">ξ</span><span 
class="zptmcm7t-x-x-109">)</span>:
</p>
   <table 
class="equation"><tr><td><a 
 id="x1-59001r15"></a>
   <center class="math-display" >
<img 
src="colvars-refman-namd21x.png" alt="        1
A(ξ)= − --ln𝒫 (ξ)+ A0
        β
" class="math-display"  /></center></td><td class="equation-label">(15)</td></tr></table>
<!--l. 3005--><p class="nopar" >
</p><!--l. 3007--><p class="indent" >   In the TI formalism, the free energy is obtained from its gradient, which is generally calculated in the form of
the average of a force <span 
class="zptmcm7m-x-x-109">F</span><sub><span 
class="zptmcm7m-x-x-80">ξ</span></sub> exerted on <span 
class="zptmcm7m-x-x-109">ξ</span> , taken over an iso-<span 
class="zptmcm7m-x-x-109">ξ</span> surface:
</p>
   <table 
class="equation"><tr><td><a 
 id="eq:gradient"></a>
   <center class="math-display" >
<img 
src="colvars-refman-namd22x.png" alt="∇  A(ξ)= ⟨− F ⟩
  ξ          ξ ξ
" class="math-display"  /></center></td><td class="equation-label">(16)</td></tr></table>
<!--l. 3014--><p class="nopar" >
</p><!--l. 3016--><p class="indent" >   Several formulae that take the form of (<a 
href="#eq:gradient">16<!--tex4ht:ref: eq:gradient --></a>) have been proposed. This implementation relies partly on the classic
formulation <span class="cite">[<a 
href="#XCarter1989">11</a>]</span>, and partly on a more versatile scheme originating in a work by Ruiz-Montero et al. <span class="cite">[<a 
href="#XRuiz-Montero1997">12</a>]</span>,
generalized by den Otter <span class="cite">[<a 
href="#XdenOtter2000">13</a>]</span> and extended to multiple variables by Ciccotti et al. <span class="cite">[<a 
href="#XCiccotti2005">14</a>]</span>. Consider a system subject
to constraints of the form <span 
class="zptmcm7m-x-x-109">σ</span><sub><span 
class="zptmcm7m-x-x-80">k</span></sub><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">x</span><span 
class="zptmcm7t-x-x-109">) = 0</span>. Let (<span 
class="zptmcm7m-x-x-109">v</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub><span 
class="zptmcm7t-x-x-109">)</span><sub><span 
class="zptmcm7m-x-x-80">i</span><span 
class="zptmcm7y-x-x-80">∈</span><span 
class="zptmcm7t-x-x-80">[1</span><span 
class="zptmcm7m-x-x-80">,n</span><span 
class="zptmcm7t-x-x-80">]</span></sub> be arbitrarily chosen vector ﬁelds (<span 
class="msbm-10x-x-109">ℝ</span><sup><span 
class="zptmcm7t-x-x-80">3</span><span 
class="zptmcm7m-x-x-80">N</span></sup> <span 
class="zptmcm7y-x-x-109">→ </span><span 
class="msbm-10x-x-109">ℝ</span><sup><span 
class="zptmcm7t-x-x-80">3</span><span 
class="zptmcm7m-x-x-80">N</span></sup>) verifying, for all
<span 
class="zptmcm7m-x-x-109">i</span>, <span 
class="zptmcm7m-x-x-109">j</span>, and <span 
class="zptmcm7m-x-x-109">k</span>:
</p>
   <div class="eqnarray">
   <center class="math-display" >
<img 
src="colvars-refman-namd23x.png" alt="vi⋅∇xξj  =   δij                                  (1 7)
vi⋅∇xσk  =   0                                   (1 8)
" class="math-display"  /><a 
 id="eq:ortho_gradient"></a><a 
 id="eq:ortho_constraints"></a></center>
</div>
<!--l. 3034--><p class="indent" >   then the following holds <span class="cite">[<a 
href="#XCiccotti2005">14</a>]</span>:
</p>
   <table 
class="equation"><tr><td><a 
 id="eq:gradient_vector"></a>
   <center class="math-display" >
<img 
src="colvars-refman-namd24x.png" alt="∂A
---= ⟨vi⋅∇xV − kBT∇x ⋅vi⟩ξ
∂ξi
" class="math-display"  /></center></td><td class="equation-label">(19)</td></tr></table>
<!--l. 3040--><p class="nopar" >
</p><!--l. 3042--><p class="indent" >   where <span 
class="zptmcm7m-x-x-109">V </span>is the potential energy function. <span 
class="zptmcm7m-x-x-109">v</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub> can be interpreted as the direction along which the force acting on
variable <span 
class="zptmcm7m-x-x-109">ξ</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub> is measured, whereas the second term in the average corresponds to the geometric entropy contribution
that appears as a Jacobian correction in the classic formalism <span class="cite">[<a 
href="#XCarter1989">11</a>]</span>. Condition (<a 
href="#eq:ortho_gradient">17<!--tex4ht:ref: eq:ortho_gradient --></a>) states that the direction along
which the total force on <span 
class="zptmcm7m-x-x-109">ξ</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub> is measured is orthogonal to the gradient of <span 
class="zptmcm7m-x-x-109">ξ</span><sub><span 
class="zptmcm7m-x-x-80">j</span></sub>, which means that the force measured on
<span 
class="zptmcm7m-x-x-109">ξ</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub> does not act on <span 
class="zptmcm7m-x-x-109">ξ</span><sub><span 
class="zptmcm7m-x-x-80">j</span></sub>.
</p><!--l. 3052--><p class="indent" >   Equation (<a 
href="#eq:ortho_constraints">18<!--tex4ht:ref: eq:ortho_constraints --></a>) implies that constraint forces are orthogonal to the directions along which the free energy
gradient is measured, so that the measurement is eﬀectively performed on unconstrained degrees of freedom. In
NAMD, constraints are typically applied to the lengths of bonds involving hydrogen atoms, for example in TIP3P
water molecules (parameter <span 
class="ot1-zi4r-0-x-x-109">rigidBonds</span>).
</p><!--l. 3059--><p class="indent" >   In the framework of ABF, <span 
class="zptmcm7m-x-x-109">F</span><sub><span 
class="zptmcm7m-x-x-80">ξ</span></sub> is accumulated in bins of ﬁnite size <span 
class="zptmcm7m-x-x-109">δξ </span>, thereby providing an estimate of the free
energy gradient according to equation (<a 
href="#eq:gradient">16<!--tex4ht:ref: eq:gradient --></a>). The biasing force applied along the collective variables to overcome
free energy barriers is calculated as:
</p>
   <table 
class="equation"><tr><td><a 
 id="x1-59005r20"></a>
   <center class="math-display" >
<img 
src="colvars-refman-namd25x.png" alt=" ABF             ^
F    = α(N ξ)× ∇xA(ξ)
" class="math-display"  /></center></td><td class="equation-label">(20)</td></tr></table>
                                                                                         
                                                                                         
<!--l. 3069--><p class="nopar" >
</p><!--l. 3071--><p class="indent" >   where <span 
class="zptmcm7y-x-x-109">∇</span><sub><span style="margin-left:0.3em" class="thinspace"></span><span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7m-x-x-80">x</span></sub><span style="margin-left:0.3em" class="thinspace"></span><img 
src="colvars-refman-namd26x.png" alt="A^"  class="widetilde"  /> denotes the current estimate of the free energy gradient at the current point <span 
class="zptmcm7m-x-x-109">ξ</span> in the collective
variable subspace, and <span 
class="zptmcm7m-x-x-109">α</span><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">N</span><sub><span 
class="zptmcm7m-x-x-80">ξ</span></sub><span 
class="zptmcm7t-x-x-109">) </span>is a scaling factor that is ramped from 0 to 1 as the local number of samples <span 
class="zptmcm7m-x-x-109">N</span><sub><span 
class="zptmcm7m-x-x-80">ξ</span></sub>
increases to prevent nonequilibrium eﬀects in the early phase of the simulation, when the gradient estimate has a
large variance. See the <span 
class="ot1-zi4r-0-x-x-109">fullSamples </span>parameter below for details.
</p><!--l. 3079--><p class="indent" >   As sampling of the phase space proceeds, the estimate <span 
class="zptmcm7y-x-x-109">∇</span><sub><span style="margin-left:0.3em" class="thinspace"></span><span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7m-x-x-80">x</span></sub><span style="margin-left:0.3em" class="thinspace"></span><img 
src="colvars-refman-namd27x.png" alt="^
A"  class="widetilde"  /> is progressively reﬁned. The biasing force
introduced in the equations of motion guarantees that in the bin centered around <span 
class="zptmcm7m-x-x-109">ξ</span> , the forces acting along the
selected collective variables average to zero over time. Eventually, as the undelying free energy surface is canceled
by the adaptive bias, evolution of the system along <span 
class="zptmcm7m-x-x-109">ξ</span> is governed mainly by diﬀusion. Although this
implementation of ABF can in principle be used in arbitrary dimension, a higher-dimension collective variable
space is likely to result in sampling diﬃculties. Most commonly, the number of variables is one or
two.
</p><!--l. 3093--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">7.1.1    </span> <a 
 id="x1-600007.1.1"></a>ABF requirements on collective variables</h5>
<!--l. 3096--><p class="noindent" >The following conditions must be met for an ABF simulation to be possible and to produce an accurate estimate of
the free energy proﬁle. Note that these requirements do not apply when using the extended-system ABF method
(<a 
href="#sec:eABF">7.2<!--tex4ht:ref: sec:eABF --></a>).
</p><!--l. 3101--><p class="indent" >
     </p><ol  class="enumerate1" >
     <li 
  class="enumerate" id="x1-60002x1"><span 
class="ptmri7t-x-x-109">Only linear combinations </span>of colvar components can be used in ABF calculations.
     </li>
     <li 
  class="enumerate" id="x1-60004x2"><span 
class="ptmri7t-x-x-109">Availability   of   total   forces   </span>is   necessary.   The   following   colvar   components   can   be   used   in
     ABF  calculations:  <span 
class="ot1-zi4r-0-x-x-109">distance</span>,  <span 
class="ot1-zi4r-0-x-x-109">distance</span><span 
class="ot1-zi4r-0-x-x-109">_xy</span>,  <span 
class="ot1-zi4r-0-x-x-109">distance</span><span 
class="ot1-zi4r-0-x-x-109">_z</span>,  <span 
class="ot1-zi4r-0-x-x-109">angle</span>,  <span 
class="ot1-zi4r-0-x-x-109">dihedral</span>,  <span 
class="ot1-zi4r-0-x-x-109">gyration</span>,  <span 
class="ot1-zi4r-0-x-x-109">rmsd </span>and
     <span 
class="ot1-zi4r-0-x-x-109">eigenvector</span>. Atom groups may not be replaced by dummy atoms, unless they are excluded from the
     force measurement by specifying <span 
class="ot1-zi4r-0-x-x-109">oneSiteTotalForce</span>, if available.
     </li>
     <li 
  class="enumerate" id="x1-60006x3"><span 
class="ptmri7t-x-x-109">Mutual orthogonality of colvars</span>. In a multidimensional ABF calculation, equation (<a 
href="#eq:ortho_gradient">17<!--tex4ht:ref: eq:ortho_gradient --></a>) must be satisﬁed for
     any two colvars <span 
class="zptmcm7m-x-x-109">ξ</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub> and <span 
class="zptmcm7m-x-x-109">ξ</span><sub><span 
class="zptmcm7m-x-x-80">j</span></sub>. Various cases fulﬁll this orthogonality condition:
          <ul class="itemize1">
          <li class="itemize"><span 
class="zptmcm7m-x-x-109">ξ</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub> and <span 
class="zptmcm7m-x-x-109">ξ</span><sub><span 
class="zptmcm7m-x-x-80">j</span></sub> are based on non-overlapping sets of atoms.
          </li>
          <li class="itemize">atoms involved in the force measurement on <span 
class="zptmcm7m-x-x-109">ξ</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub> do not participate in the deﬁnition of <span 
class="zptmcm7m-x-x-109">ξ</span><sub><span 
class="zptmcm7m-x-x-80">j</span></sub>. This
          can be obtained using the option <span 
class="ot1-zi4r-0-x-x-109">oneSiteTotalForce </span>of the <span 
class="ot1-zi4r-0-x-x-109">distance</span>, <span 
class="ot1-zi4r-0-x-x-109">angle</span>, and <span 
class="ot1-zi4r-0-x-x-109">dihedral</span>
          components (example: Ramachandran angles <span 
class="zptmcm7m-x-x-109">ϕ</span>, <span 
class="zptmcm7m-x-x-109">ψ</span>).
          </li>
          <li class="itemize"><span 
class="zptmcm7m-x-x-109">ξ</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub> and  <span 
class="zptmcm7m-x-x-109">ξ</span><sub><span 
class="zptmcm7m-x-x-80">j</span></sub> are  orthogonal  by  construction.  Useful  cases  are  the  sum  and  diﬀerence  of  two
          components, or <span 
class="ot1-zi4r-0-x-x-109">distance</span><span 
class="ot1-zi4r-0-x-x-109">_z </span>and <span 
class="ot1-zi4r-0-x-x-109">distance</span><span 
class="ot1-zi4r-0-x-x-109">_xy </span>using the same axis.</li></ul>
     </li>
     <li 
  class="enumerate" id="x1-60008x4"><span 
class="ptmri7t-x-x-109">Mutual orthogonality of components</span>: when several components are combined into a colvar, it is assumed that
     their vectors <span 
class="zptmcm7m-x-x-109">v</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub> (equation (<a 
href="#eq:gradient_vector">19<!--tex4ht:ref: eq:gradient_vector --></a>)) are mutually orthogonal. The cases described for colvars in the previous
     paragraph apply.
                                                                                         
                                                                                         
     </li>
     <li 
  class="enumerate" id="x1-60010x5"><span 
class="ptmri7t-x-x-109">Orthogonality of colvars and constraints</span>: equation <a 
href="#eq:ortho_constraints">18<!--tex4ht:ref: eq:ortho_constraints --></a> can be satisﬁed in two simple ways, if
     either no constrained atoms are involved in the force measurement (see point 3 above) or pairs of
     atoms joined by a constrained bond are part of an <span 
class="ptmri7t-x-x-109">atom group </span>which only intervenes through its
     center (center of mass or geometric center) in the force measurement. In the latter case, the
     contributions of the two atoms to the left-hand side of equation <a 
href="#eq:ortho_constraints">18<!--tex4ht:ref: eq:ortho_constraints --></a> cancel out. For example, all atoms
     of a rigid TIP3P water molecule can safely be included in an atom group used in a <span 
class="ot1-zi4r-0-x-x-109">distance</span>
     component.</li></ol>
<!--l. 3135--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">7.1.2    </span> <a 
 id="x1-610007.1.2"></a>Parameters for ABF</h5>
<!--l. 3137--><p class="noindent" >ABF depends on parameters from collective variables to deﬁne the grid on which free energy gradients are
computed. In the direction of each colvar, the grid ranges from <span 
class="ot1-zi4r-0-x-x-109">lowerBoundary </span>to <span 
class="ot1-zi4r-0-x-x-109">upperBoundary</span>, and the bin
width (grid spacing) is set by the <span 
class="ot1-zi4r-0-x-x-109">width </span>parameter (see <a 
href="#sec:colvar_general">4.1<!--tex4ht:ref: sec:colvar_general --></a>). The following speciﬁc parameters can be set in the
ABF conﬁguration block:
</p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-61001"></a> <span 
class="ot1-zi4r-0-x-x-120">name: </span>see deﬁnition of <a 
href="#sec:colvarbias"><span 
class="ot1-zi4r-0-x-x-109">name</span></a> in sec. <a 
href="#sec:colvarbias">7<!--tex4ht:ref: sec:colvarbias --></a> (biasing and analysis methods)
     </li>
     <li class="itemize"><a 
 id="dx1-61002"></a> <span 
class="ot1-zi4r-0-x-x-120">colvars: </span>see deﬁnition of <a 
href="#sec:colvarbias"><span 
class="ot1-zi4r-0-x-x-109">colvars</span></a> in sec. <a 
href="#sec:colvarbias">7<!--tex4ht:ref: sec:colvarbias --></a> (biasing and analysis methods)
     </li>
     <li class="itemize"><a 
 id="dx1-61003"></a> <span 
class="ot1-zi4r-0-x-x-120">fullSamples </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Number of samples in a bin prior to application of the ABF<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">abf</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive integer <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span>200 <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>To avoid nonequilibrium eﬀects due to large ﬂuctuations of the force exerted along the
     colvars, it is recommended to apply a biasing force only after a the estimate has started converging. If
     <span 
class="ot1-zi4r-0-x-x-109">fullSamples </span>is non-zero, the applied biasing force is scaled by a factor <span 
class="zptmcm7m-x-x-109">α</span><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">N</span><sub><span 
class="zptmcm7m-x-x-80">ξ</span></sub><span 
class="zptmcm7t-x-x-109">) </span>between 0 and 1. If
     the number of samples <span 
class="zptmcm7m-x-x-109">N</span><sub><span 
class="zptmcm7m-x-x-80">ξ</span></sub> in the current bin is higher than <span 
class="ot1-zi4r-0-x-x-109">fullSamples</span>, the factor is one. If it is less
     than half of <span 
class="ot1-zi4r-0-x-x-109">fullSamples</span>, the factor is zero and no bias is applied. Between those two thresholds, the
     factor follows a linear ramp from 0 to 1: <span 
class="zptmcm7m-x-x-109">α</span><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">N</span><sub><span 
class="zptmcm7m-x-x-80">ξ</span></sub><span 
class="zptmcm7t-x-x-109">) = (2</span><span 
class="zptmcm7m-x-x-109">N</span><sub><span 
class="zptmcm7m-x-x-80">ξ</span></sub><span 
class="zptmcm7m-x-x-109">∕</span><span 
class="zptmcm7t-x-x-109">fullSamples</span><span 
class="zptmcm7t-x-x-109">)</span><span 
class="zptmcm7y-x-x-109">−</span><span 
class="zptmcm7t-x-x-109">1 </span>.
     </li>
     <li class="itemize"><a 
 id="dx1-61004"></a> <span 
class="ot1-zi4r-0-x-x-120">maxForce </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Maximum magnitude of the ABF force<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">abf</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive decimals (one per colvar) <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span>disabled <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>This option enforces a cap on the magnitude of the biasing force eﬀectively applied by
     this ABF bias on each colvar. This can be useful in the presence of singularities in the PMF such as
     hard walls, where the discretization of the average force becomes very inaccurate, causing the colvar&#x2019;s
     diﬀusion to get “stuck” at the singularity. To enable this cap, provide one non-negative value for each
     colvar. The unit of force is kcal/mol divided by the colvar unit.
                                                                                         
                                                                                         
     </li>
     <li class="itemize"><a 
 id="dx1-61005"></a> <span 
class="ot1-zi4r-0-x-x-120">hideJacobian </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Remove geometric entropy term from calculated free energy gradient?<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">abf</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">no</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>In a few special cases, most notably distance-based variables, an alternate deﬁnition of
     the potential of mean force is traditionally used, which excludes the Jacobian term describing the eﬀect
     of geometric entropy on the distribution of the variable. This results, for example, in particle-particle
     potentials of mean force being ﬂat at large separations. Setting this parameter to <span 
class="ot1-zi4r-0-x-x-109">yes </span>causes the output
     data to follow that convention, by removing this contribution from the output gradients while applying
     internally the corresponding correction to ensure uniform sampling. It is not allowed for colvars with
     multiple components.
     </li>
     <li class="itemize"><a 
 id="dx1-61006"></a> <span 
class="ot1-zi4r-0-x-x-120">outputFreq </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Frequency (in timesteps) at which ABF data ﬁles are refreshed<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">abf</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive integer <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span>Colvars module restart frequency <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>The ﬁles containing the free energy gradient estimate and sampling histogram (and the
     PMF in one-dimensional calculations) are written on disk at the given time interval.
     </li>
     <li class="itemize"><a 
 id="dx1-61007"></a> <span 
class="ot1-zi4r-0-x-x-120">historyFreq </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Frequency (in timesteps) at which ABF history ﬁles are accumulated<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">abf</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive integer <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span>0 <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>If this number is non-zero, the free energy gradient estimate and sampling histogram
     (and the PMF in one-dimensional calculations) are appended to ﬁles on disk at the given time interval.
     History ﬁle names use the same preﬁx as output ﬁles, with “<span 
class="ot1-zi4r-0-x-x-109">.hist</span>” appended.
     </li>
     <li class="itemize"><a 
 id="dx1-61008"></a> <span 
class="ot1-zi4r-0-x-x-120">inputPrefix </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Filename preﬁx for reading ABF data<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">abf</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>list of strings <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:    </span>If  this  parameter  is  set,  for  each  item  in  the  list,  ABF  tries  to  read  a  gradient
     and a sampling ﬁles named <span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ot1-zi4r-0-x-x-109">inputPrefix</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span><span 
class="ot1-zi4r-0-x-x-109">.grad </span>and <span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ot1-zi4r-0-x-x-109">inputPrefix</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span><span 
class="ot1-zi4r-0-x-x-109">.count</span>. This is done at
     startup and sets the initial state of the ABF algorithm. The data from all provided ﬁles is combined
     appropriately. Also, the grid deﬁnition (min and max values, width) need not be the same that for
     the current run. This command is useful to piece together data from simulations in diﬀerent regions
     of  collective  variable  space,  or  change  the  colvar  boundary  values  and  widths.  Note  that  it  is  not
     recommended to use it to switch to a smaller width, as that will leave some bins empty in the ﬁner data
     grid. This option is NOT compatible with reading the data from a restart ﬁle (<span 
class="ot1-zi4r-0-x-x-109">colvarsInput </span>option
     of the NAMD conﬁg ﬁle).
     </li>
     <li class="itemize"><a 
 id="dx1-61009"></a> <span 
class="ot1-zi4r-0-x-x-120">applyBias </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Apply the ABF bias?<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">abf</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">yes</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>If this is set to no, the calculation proceeds normally but the adaptive biasing force is
     not applied. Data is still collected to compute the free energy gradient. This is mostly intended for
     testing purposes, and should not be used in routine simulations.
     </li>
     <li class="itemize"><a 
 id="dx1-61010"></a> <span 
class="ot1-zi4r-0-x-x-120">updateBias </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Update the ABF bias?<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">abf</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">yes</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:    </span>If this is set to no, the initial biasing force (e.g. read from a restart ﬁle or through
     <span 
class="ot1-zi4r-0-x-x-109">inputPrefix</span>) is not updated during the simulation. As a result, a constant bias is applied. This can
     be used to apply a custom, tabulated biasing potential to any combination of colvars. To that eﬀect,
     one should prepare a gradient ﬁle containing the gradient of the potential to be applied (negative of
     the bias force), and a count ﬁle containing only values greater than <span 
class="ot1-zi4r-0-x-x-109">fullSamples</span>. These ﬁles must
     match the grid parameters of the colvars.  </li></ul>
<!--l. 3279--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">7.1.3    </span> <a 
 id="sec:mw-ABF"></a>Multiple-replica ABF</h5>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-62001"></a> <span 
class="ot1-zi4r-0-x-x-120">shared </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Apply multiple-replica ABF, sharing force samples among the replicas?<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">abf</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">no</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>This is command requires that NAMD be compiled and executed with multiple-replica
     support. If <span 
class="ot1-zi4r-0-x-x-109">shared </span>is set to yes, the total force samples will be synchronized among all replicas at
     intervals  deﬁned  by  <span 
class="ot1-zi4r-0-x-x-109">sharedFreq</span>.  This  implements  the  multiple-walker  ABF  scheme  described  in
     <span class="cite">[<a 
href="#XMinoukadeh2010">15</a>]</span>; this implementation is documented in <span class="cite">[<a 
href="#XComer2014c">16</a>]</span>. Thus, it is as if total force samples among all replicas
     are gathered in a single shared buﬀer, which why the algorithm is referred to as shared ABF. Shared
     ABF allows all replicas to beneﬁt from the sampling done by other replicas and can lead to faster
     convergence of the biasing force.
     </li>
     <li class="itemize"><a 
 id="dx1-62002"></a>  <span 
class="ot1-zi4r-0-x-x-120">sharedFreq   </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Frequency  (in  timesteps)  at  which  force  samples  are  synchronized  among  the
     replicas<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">abf</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive integer <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">outputFreq</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:     </span>In  the  current  implementation  of  shared  ABF,  each  replica  maintains  a  separate
     buﬀer of total force samples that determine the biasing force. Every <span 
class="ot1-zi4r-0-x-x-109">sharedFreq </span>steps, the replicas
     communicate the samples that have been gathered since the last synchronization time, ensuring all
     replicas apply a similar biasing force.  </li></ul>
                                                                                         
                                                                                         
<!--l. 3281--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">7.1.4    </span> <a 
 id="x1-630007.1.4"></a>Output ﬁles</h5>
<!--l. 3283--><p class="noindent" >The ABF bias produces the following ﬁles, all in multicolumn text format: </p>
     <ul class="itemize1">
     <li class="itemize"><span 
class="ptmri7t-x-x-109">outputName</span><span 
class="ot1-zi4r-0-x-x-109">.grad</span>: current estimate of the free energy gradient (grid), in multicolumn;
     </li>
     <li class="itemize"><span 
class="ptmri7t-x-x-109">outputName</span><span 
class="ot1-zi4r-0-x-x-109">.count</span>: histogram of samples collected, on the same grid;
     </li>
     <li class="itemize"><span 
class="ptmri7t-x-x-109">outputName</span><span 
class="ot1-zi4r-0-x-x-109">.pmf</span>: only for one-dimensional calculations, integrated free energy proﬁle or PMF.</li></ul>
<!--l. 3292--><p class="indent" >   If several ABF biases are deﬁned concurrently, their name is inserted to produce unique ﬁlenames for output, as
in <span 
class="ptmri7t-x-x-109">outputName</span><span 
class="ot1-zi4r-0-x-x-109">.abf1.grad</span>. This should not be done routinely and could lead to meaningless results: only do it if
you know what you are doing!
</p><!--l. 3297--><p class="indent" >   If the colvar space has been partitioned into sections (<span 
class="ptmri7t-x-x-109">windows</span>) in which independent ABF simulations have
been run, the resulting data can be merged using the <span 
class="ot1-zi4r-0-x-x-109">inputPrefix </span>option described above (a run of 0 steps is
enough).
</p><!--l. 3302--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">7.1.5    </span> <a 
 id="x1-640007.1.5"></a>Post-processing: reconstructing a multidimensional free energy surface</h5>
<!--l. 3304--><p class="noindent" >If a one-dimensional calculation is performed, the estimated free energy gradient is automatically integrated and a
potential of mean force is written under the ﬁle name <span 
class="ot1-zi4r-0-x-x-109"><outputName>.pmf</span>, in a plain text format that can be read by
most data plotting and analysis programs (e.g. gnuplot).
</p><!--l. 3309--><p class="indent" >   In dimension 2 or greater, integrating the discretized gradient becomes non-trivial. The standalone utility
<span 
class="ot1-zi4r-0-x-x-109">abf</span><span 
class="ot1-zi4r-0-x-x-109">_integrate </span>is provided to perform that task. <span 
class="ot1-zi4r-0-x-x-109">abf</span><span 
class="ot1-zi4r-0-x-x-109">_integrate </span>reads the gradient data and uses it to perform a
Monte-Carlo (M-C) simulation in discretized collective variable space (speciﬁcally, on the same grid used by ABF
to discretize the free energy gradient). By default, a history-dependent bias (similar in spirit to metadynamics) is
used: at each M-C step, the bias at the current position is incremented by a preset amount (the <span 
class="ptmri7t-x-x-109">hill height</span>). Upon
convergence, this bias counteracts optimally the underlying gradient; it is negated to obtain the estimate of the free
energy surface.
</p><!--l. 3320--><p class="indent" >   <span 
class="ot1-zi4r-0-x-x-109">abf</span><span 
class="ot1-zi4r-0-x-x-109">_integrate </span>is invoked using the command-line:<br 
class="newline" /><span 
class="ot1-zi4r-0-">abf</span><span 
class="ot1-zi4r-0-">_integrate <gradient</span><span 
class="ot1-zi4r-0-">_file> [-n <nsteps>] [-t <temp>] [-m (0|1)] [-h <hill</span><span 
class="ot1-zi4r-0-">_height>] [-f</span>
<span 
class="ot1-zi4r-0-"><factor>]</span>
</p><!--l. 3325--><p class="indent" >   The gradient ﬁle name is provided ﬁrst, followed by other parameters in any order. They are described below,
with their default value in square brackets: </p>
     <ul class="itemize1">
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">-n</span>: number of M-C steps to be performed; by default, a minimal number of steps is chosen based on
     the size of the grid, and the integration runs until a convergence criterion is satisﬁed (based on the
     RMSD between the target gradient and the real PMF gradient)
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">-t</span>: temperature for M-C sampling (unrelated to the simulation temperature) [500 K]
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">-m</span>: use metadynamics-like biased sampling? (0 = false) [1]
                                                                                         
                                                                                         
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">-h</span>: increment for the history-dependent bias (“hill height”) [0.01 kcal/mol]
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">-f</span>: if non-zero, this factor is used to scale the increment stepwise in the second half of the M-C
     sampling to reﬁne the free energy estimate [0.5]</li></ul>
<!--l. 3340--><p class="indent" >   Using the default values of all parameters should give reasonable results in most cases.
</p><!--l. 3343--><p class="indent" >   <span 
class="ot1-zi4r-0-x-x-109">abf</span><span 
class="ot1-zi4r-0-x-x-109">_integrate </span>produces the following output ﬁles: </p>
     <ul class="itemize1">
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109"><gradient</span><span 
class="ot1-zi4r-0-x-x-109">_file>.pmf</span>: computed free energy surface
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109"><gradient</span><span 
class="ot1-zi4r-0-x-x-109">_file>.histo</span>: histogram of M-C sampling (not usable in a straightforward way if the
     history-dependent bias has been applied)
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109"><gradient</span><span 
class="ot1-zi4r-0-x-x-109">_file>.est</span>:  estimated  gradient  of  the  calculated  free  energy  surface  (from  ﬁnite
     diﬀerences)
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109"><gradient</span><span 
class="ot1-zi4r-0-x-x-109">_file>.dev</span>:  deviation  between  the  user-provided  numerical  gradient  and  the  actual
     gradient  of  the  calculated  free  energy  surface.  The  RMS  norm  of  this  vector  ﬁeld  is  used  as  a
     convergence criteria and displayed periodically during the integration.</li></ul>
<!--l. 3356--><p class="indent" >   <span 
class="ptmb7t-x-x-109">Note: </span>Typically, the “deviation” vector ﬁeld does not vanish as the integration converges. This happens because
the numerical estimate of the gradient does not exactly derive from a potential, due to numerical approximations
used to obtain it (ﬁnite sampling and discretization on a grid).
</p><!--l. 3364--><p class="noindent" >
</p>
   <h4 class="subsectionHead"><span class="titlemark">7.2    </span> <a 
 id="sec:eABF"></a>Extended-system Adaptive Biasing Force (eABF)</h4>
<!--l. 3367--><p class="noindent" >Extended-system ABF (eABF) is a variant of ABF (<a 
href="#sec:colvarbias_abf">7.1<!--tex4ht:ref: sec:colvarbias_abf --></a>) where the bias is not applied directly to the collective
variable, but to an extended coordinate (“ﬁctitious variable”) <span 
class="zptmcm7m-x-x-109">λ </span>that evolves dynamically according to Newtonian or
Langevin dynamics. Such an extended coordinate is enabled for a given colvar using the <span 
class="ot1-zi4r-0-x-x-109">extendedLagrangian </span>and
associated keywords (<a 
href="#sec:colvar_extended">4.3<!--tex4ht:ref: sec:colvar_extended --></a>). The theory of eABF and the present implementation are documented in detail in
reference <span class="cite">[<a 
href="#XLesage2017">17</a>]</span>.
</p><!--l. 3376--><p class="indent" >   Deﬁning an ABF bias on a colvar wherein the <span 
class="ot1-zi4r-0-x-x-109">extendedLagrangian </span>option is active will perform eABF; there
is no dedicated option.
</p><!--l. 3379--><p class="indent" >   The extended variable <span 
class="zptmcm7m-x-x-109">λ </span>is coupled to the colvar <span 
class="zptmcm7m-x-x-109">z </span><span 
class="zptmcm7t-x-x-109">= </span><span 
class="zptmcm7m-x-x-109">ξ</span><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">q</span><span 
class="zptmcm7t-x-x-109">) </span>by the harmonic potential <span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">k∕</span><span 
class="zptmcm7t-x-x-109">2)(</span><span 
class="zptmcm7m-x-x-109">z</span><span 
class="zptmcm7y-x-x-109">−</span><span 
class="zptmcm7m-x-x-109">λ</span><span 
class="zptmcm7t-x-x-109">)</span><sup><span 
class="zptmcm7t-x-x-80">2</span></sup>. Under eABF
dynamics, the adaptive bias on <span 
class="zptmcm7m-x-x-109">λ </span>is the running estimate of the average spring force:
</p>
   <table 
class="equation"><tr><td><a 
 id="x1-65001r21"></a>
   <center class="math-display" >
                                                                                         
                                                                                         
<img 
src="colvars-refman-namd28x.png" alt="Fbias(λ∗)= ⟨k(λ − z)⟩λ∗
" class="math-display"  /></center></td><td class="equation-label">(21)</td></tr></table>
<!--l. 3385--><p class="nopar" >
where the angle brackets indicate a canonical average conditioned by <span 
class="zptmcm7m-x-x-109">λ </span><span 
class="zptmcm7t-x-x-109">= </span><span 
class="zptmcm7m-x-x-109">λ</span><sup><span 
class="zptmcm7y-x-x-80">∗</span></sup>. At long simulation
times, eABF produces a ﬂat histogram of the extended variable <span 
class="zptmcm7m-x-x-109">λ</span>, and a ﬂattened histogram of <span 
class="zptmcm7m-x-x-109">ξ </span>,
whose exact shape depends on the strength of the coupling as deﬁned by <span 
class="ot1-zi4r-0-x-x-109">extendedFluctuation </span>in the
colvar. Coupling should be somewhat loose for faster exploration and convergence, but strong enough
that the bias does help overcome barriers along the colvar <span 
class="zptmcm7m-x-x-109">ξ </span>.<span class="cite">[<a 
href="#XLesage2017">17</a>]</span> Distribution of the colvar may be
assessed by plotting its histogram, which is written to the <span 
class="ptmri7t-x-x-109">outputName</span><span 
class="ot1-zi4r-0-x-x-109">.zcount </span>ﬁle in every eABF
simulation. Note that a <span 
class="ot1-zi4r-0-x-x-109">histogram </span>bias (<a 
href="#sec:colvarbias_histogram">7.8<!--tex4ht:ref: sec:colvarbias_histogram --></a>) applied to an extended-Lagrangian colvar will access the
extended degree of freedom <span 
class="zptmcm7m-x-x-109">λ</span>, not the original colvar <span 
class="zptmcm7m-x-x-109">ξ </span>; however, the joint histogram may be explicitly
requested by listing the name of the colvar twice in a row within the <span 
class="ot1-zi4r-0-x-x-109">colvars </span>parameter of the <span 
class="ot1-zi4r-0-x-x-109">histogram</span>
block.
</p><!--l. 3400--><p class="indent" >   The eABF PMF is that of the coordinate <span 
class="zptmcm7m-x-x-109">λ</span>, it is not exactly the free energy proﬁle of <span 
class="zptmcm7m-x-x-109">ξ </span>. That quantity can be
calculated based on either the CZAR estimator or the Zheng/Yang estimator.
</p><!--l. 3404--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">7.2.1    </span> <a 
 id="x1-660007.2.1"></a>CZAR estimator of the free energy</h5>
<!--l. 3406--><p class="noindent" >The <span 
class="ptmri7t-x-x-109">corrected z-averaged restraint </span>(CZAR) estimator is described in detail in reference <span class="cite">[<a 
href="#XLesage2017">17</a>]</span>. It is computed
automatically in eABF simulations, regardless of the number of colvars involved. Note that ABF may also be
applied on a combination of extended and non-extended colvars; in that case, CZAR still provides an unbiased
estimate of the free energy gradient.
</p><!--l. 3413--><p class="indent" >   CZAR estimates the free energy gradient as:
</p>
   <table 
class="equation"><tr><td><a 
 id="x1-66001r22"></a>
   <center class="math-display" >
<img 
src="colvars-refman-namd29x.png" alt="A′(z)= − 1-dlnρ~(z) + k(⟨λ⟩ − z).
        β    dz         z
" class="math-display"  /></center></td><td class="equation-label">(22)</td></tr></table>
<!--l. 3417--><p class="nopar" >
where <span 
class="zptmcm7m-x-x-109">z </span><span 
class="zptmcm7t-x-x-109">= </span><span 
class="zptmcm7m-x-x-109">ξ</span><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">q</span><span 
class="zptmcm7t-x-x-109">) </span>is the colvar, <span 
class="zptmcm7m-x-x-109">λ </span>is the extended variable harmonically coupled to <span 
class="zptmcm7m-x-x-109">z </span>with a force constant <span 
class="zptmcm7m-x-x-109">k</span>, and <img 
src="colvars-refman-namd30x.png" alt="ρ~"  class="tilde"  /><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">z</span><span 
class="zptmcm7t-x-x-109">)</span>
is the observed distribution (histogram) of <span 
class="zptmcm7m-x-x-109">z</span>, aﬀected by the eABF bias.
</p><!--l. 3422--><p class="indent" >   Parameters for the CZAR estimator are: </p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-66002"></a> <span 
class="ot1-zi4r-0-x-x-120">CZARestimator </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Calculate CZAR estimator of the free energy?<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">abf</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">yes</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>This option is only available when ABF is performed on extended-Lagrangian colvars.
     When enabled, it triggers calculation of the free energy following the CZAR estimator.
     </li>
     <li class="itemize"><a 
 id="dx1-66003"></a> <span 
class="ot1-zi4r-0-x-x-120">writeCZARwindowFile </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Write internal data from CZAR to a separate ﬁle?<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">abf</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">no</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:    </span>When  this  option  is  enabled,  eABF  simulations  will  write  a  ﬁle  containing  the
     <span 
class="zptmcm7m-x-x-109">z</span>-averaged  restraint  force  under  the  name  <span 
class="ptmri7t-x-x-109">outputName</span><span 
class="ot1-zi4r-0-x-x-109">.zgrad</span>.  The  same  information  is  always
     included in the colvars state ﬁle, which is suﬃcient for restarting an eABF simulation. These separate
     ﬁle is only useful when joining adjacent windows from a stratiﬁed eABF simulation, either to continue
     the simulation in a broader window or to compute a CZAR estimate of the PMF over the full range of
     the coordinate(s). </li></ul>
<!--l. 3444--><p class="indent" >   Similar to ABF, the CZAR estimator produces two output ﬁles in multicolumn text format: </p>
     <ul class="itemize1">
     <li class="itemize"><span 
class="ptmri7t-x-x-109">outputName</span><span 
class="ot1-zi4r-0-x-x-109">.czar.grad</span>: current estimate of the free energy gradient (grid), in multicolumn;
     </li>
     <li class="itemize"><span 
class="ptmri7t-x-x-109">outputName</span><span 
class="ot1-zi4r-0-x-x-109">.czar.pmf</span>: only for one-dimensional calculations, integrated free energy proﬁle or PMF.</li></ul>
<!--l. 3451--><p class="noindent" >The sampling histogram associated with the CZAR estimator is the <span 
class="zptmcm7m-x-x-109">z</span>-histogram, which is written in the ﬁle
<span 
class="ptmri7t-x-x-109">outputName</span><span 
class="ot1-zi4r-0-x-x-109">.zcount</span>.
</p><!--l. 3522--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">7.2.2    </span> <a 
 id="x1-670007.2.2"></a>Zheng/Yang estimator of the free energy</h5>
<!--l. 3522--><p class="noindent" >This feature has been contributed to NAMD by the following authors:
</p><!--l. 3522--><p class="indent" >
     </p><div class="quote">
     <!--l. 3522--><p class="noindent" >Haohao Fu and Christophe Chipot<br 
class="newline" /> Laboratoire International Associé Centre National de la Recherche Scientiﬁque et University
     of Illinois at Urbana–Champaign, <br 
class="newline" />Unité Mixte de Recherche No. 7565, Université de Lorraine, <br 
class="newline" />B.P. 70239, 54506 Vandœuvre-ls-Nancy cedex, France</p></div>
<!--l. 3522--><p class="indent" >   <span 
class="ptmr8c-x-x-109">©</span> 2016, <span 
class="ptmrc7t-x-x-109">C<span 
class="small-caps">e</span><span 
class="small-caps">n</span><span 
class="small-caps">t</span><span 
class="small-caps">r</span><span 
class="small-caps">e</span> N<span 
class="small-caps">a</span><span 
class="small-caps">t</span><span 
class="small-caps">i</span><span 
class="small-caps">o</span><span 
class="small-caps">n</span><span 
class="small-caps">a</span><span 
class="small-caps">l</span> <span 
class="small-caps">d</span><span 
class="small-caps">e</span> <span 
class="small-caps">l</span><span 
class="small-caps">a</span> R<span 
class="small-caps">e</span><span 
class="small-caps">c</span><span 
class="small-caps">h</span><span 
class="small-caps">e</span><span 
class="small-caps">r</span><span 
class="small-caps">c</span><span 
class="small-caps">h</span><span 
class="small-caps">e</span> S<span 
class="small-caps">c</span><span 
class="small-caps">i</span><span 
class="small-caps">e</span><span 
class="small-caps">n</span><span 
class="small-caps">t</span><span 
class="small-caps">i</span><span 
class="small-caps">ﬁ</span><span 
class="small-caps">q</span><span 
class="small-caps">u</span><span 
class="small-caps">e</span></span>
</p><!--l. 3522--><p class="indent" >   This implementation is fully documented in <span class="cite">[<a 
href="#XFu2016">18</a>]</span>. The Zheng and Yang estimator <span class="cite">[<a 
href="#XZheng2012">19</a>]</span> is based on Umbrella
Integration <span class="cite">[<a 
href="#XKastner2005">20</a>]</span>. The free energy gradient is estimated as :
</p>
                                                                                         
                                                                                         
   <table 
class="equation"><tr><td><a 
 id="x1-67001r23"></a>
   <center class="math-display" >
<img 
src="colvars-refman-namd31x.png" alt="                   [  ∗                  ]
         ∑ N (ξ∗,λ ) (ξ--−-⟨ξ2-⟩λ-)− k(ξ∗− λ)
  ′ ∗    λ-------------β-σλ----------------
A (ξ )=                N (ξ ∗,λ )
                     ∑λ
" class="math-display"  /></center></td><td class="equation-label">(23)</td></tr></table>
<!--l. 3522--><p class="nopar" >
where <span 
class="zptmcm7m-x-x-109">ξ </span>is the colvar, <span 
class="zptmcm7m-x-x-109">λ </span>is the extended variable harmonically coupled to <span 
class="zptmcm7m-x-x-109">ξ </span>with a force constant <span 
class="zptmcm7m-x-x-109">k</span>, <span 
class="zptmcm7m-x-x-109">N</span><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">ξ,λ</span><span 
class="zptmcm7t-x-x-109">) </span>is the
number of samples collected in a <span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">ξ,λ</span><span 
class="zptmcm7t-x-x-109">) </span>bin, which is assumed to be a Gaussian function of <span 
class="zptmcm7m-x-x-109">ξ </span>with mean <span 
class="zptmcm7y-x-x-109">⟨</span><span 
class="zptmcm7m-x-x-109">ξ</span><span 
class="zptmcm7y-x-x-109">⟩</span><sub><span 
class="zptmcm7m-x-x-80">λ</span></sub> and
standard deviation <span 
class="zptmcm7m-x-x-109">σ</span><sub><span 
class="zptmcm7m-x-x-80">λ</span></sub>.
</p><!--l. 3522--><p class="indent" >   The estimator is enabled through the following option: </p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-67002"></a> <span 
class="ot1-zi4r-0-x-x-120">UIestimator </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Calculate UI estimator of the free energy?<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">abf</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">no</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>This option is only available when ABF is performed on extended-Lagrangian colvars.
     When enabled, it triggers calculation of the free energy following the UI estimator. </li></ul>
<!--l. 3522--><p class="noindent" ><span class="paragraphHead"><a 
 id="x1-680007.2.2"></a><span 
class="ptmb7t-x-x-109">Usage for multiple–replica eABF.</span></span>
   The eABF algorithm can be associated with a multiple–walker strategy <span class="cite">[<a 
href="#XMinoukadeh2010">15</a>, <a 
href="#XComer2014c">16</a>]</span> (<a 
href="#sec:mw-ABF">7.1.3<!--tex4ht:ref: sec:mw-ABF --></a>). To run a
multiple–replica eABF simulation, start a multiple-replica NAMD run (option <span 
class="ot1-zi4r-0-x-x-109">+replicas</span>) and set <span 
class="ot1-zi4r-0-x-x-109">shared on </span>in the
Colvars conﬁg ﬁle to enable the multiple–walker ABF algorithm. It should be noted that in contrast with classical
MW–ABF simulations, the output ﬁles of an MW–eABF simulation only show the free energy estimate of the
corresponding replica.
</p><!--l. 3522--><p class="indent" >   One can merge the results, using <span 
class="ot1-zi4r-0-x-x-109">./eabf.tcl -mergemwabf [merged</span><span 
class="ot1-zi4r-0-x-x-109">_filename] [eabf</span><span 
class="ot1-zi4r-0-x-x-109">_output1]</span>
<span 
class="ot1-zi4r-0-x-x-109">[eabf</span><span 
class="ot1-zi4r-0-x-x-109">_output2] ...</span>, e.g., <span 
class="ot1-zi4r-0-x-x-109">./eabf.tcl -mergemwabf merge.eabf eabf.0.UI eabf.1.UI eabf.2.UI</span>
<span 
class="ot1-zi4r-0-x-x-109">eabf.3.UI</span>.
</p><!--l. 3522--><p class="indent" >   If one runs an ABF–based calculation, breaking the reaction pathway into several non–overlapping windows,
one can use <span 
class="ot1-zi4r-0-x-x-109">./eabf.tcl -mergesplitwindow [merged</span><span 
class="ot1-zi4r-0-x-x-109">_fileprefix] [eabf</span><span 
class="ot1-zi4r-0-x-x-109">_output] [eabf</span><span 
class="ot1-zi4r-0-x-x-109">_output2]</span>
<span 
class="ot1-zi4r-0-x-x-109">... </span>to merge the data accrued in these non–overlapping windows. This option can be utilized in both
eABF and classical ABF simulations, e.g., <span 
class="ot1-zi4r-0-x-x-109">./eabf.tcl -mergesplitwindow merge window0.czar</span>
<span 
class="ot1-zi4r-0-x-x-109">window1.czar window2.czar window3.czar</span>, <span 
class="ot1-zi4r-0-x-x-109">./eabf.tcl -mergesplitwindow merge window0.UI</span>
<span 
class="ot1-zi4r-0-x-x-109">window1.UI window2.UI window3.UI </span>or <span 
class="ot1-zi4r-0-x-x-109">./eabf.tcl -mergesplitwindow merge abf0 abf1 abf2</span>
<span 
class="ot1-zi4r-0-x-x-109">abf3</span>.
                                                                                         
                                                                                         
</p><!--l. 3524--><p class="noindent" >
</p>
   <h4 class="subsectionHead"><span class="titlemark">7.3    </span> <a 
 id="sec:colvarbias_meta"></a>Metadynamics</h4>
<!--l. 3527--><p class="noindent" >The metadynamics method uses a history-dependent potential <span class="cite">[<a 
href="#XLaio2002">21</a>]</span> that generalizes to any type of colvars the
conformational ﬂooding <span class="cite">[<a 
href="#XGrubmuller1995">22</a>]</span> and local elevation <span class="cite">[<a 
href="#XHuber1994">23</a>]</span> methods, originally formulated to use as colvars the principal
components of a covariance matrix or a set of dihedral angles, respectively. The metadynamics potential on the
colvars <span 
class="zptmcm7m-x-x-109">ξ</span> <span 
class="zptmcm7t-x-x-109">= (</span><span 
class="zptmcm7m-x-x-109">ξ</span><sub><span 
class="zptmcm7t-x-x-80">1</span></sub><span 
class="zptmcm7m-x-x-109">,ξ</span><sub><span 
class="zptmcm7t-x-x-80">2</span></sub><span 
class="zptmcm7m-x-x-109">,</span><span 
class="zptmcm7m-x-x-109">…</span><span 
class="zptmcm7m-x-x-109">,ξ</span><sub><span 
class="zptmcm7m-x-x-80">N</span><sub><span 
class="zptmcm7t-x-x-60">cv</span></sub></sub><span 
class="zptmcm7t-x-x-109">) </span>is deﬁned as:
</p>
   <table 
class="equation"><tr><td><a 
 id="eq:colvars_meta_pot"></a>
   <center class="math-display" >
<img 
src="colvars-refman-namd32x.png" alt="               t′&#x003C;t    Ncv   (            ′ 2 )
Vmeta(ξ(t))=    ∑    W ∏  exp  − (ξi(t)−-ξi(t)) ,
            t′=δt,2δt,... i=1           2σξ2i
" class="math-display"  /></center></td><td class="equation-label">(24)</td></tr></table>
<!--l. 3537--><p class="nopar" >
where <span 
class="zptmcm7m-x-x-109">V</span><sub><span 
class="zptmcm7t-x-x-80">meta</span></sub> is the history-dependent potential acting on the <span 
class="ptmri7t-x-x-109">current </span>values of the colvars <span 
class="zptmcm7m-x-x-109">ξ</span> , and depends only
parametrically on the <span 
class="ptmri7t-x-x-109">previous </span>values of the colvars. <span 
class="zptmcm7m-x-x-109">V</span><sub><span 
class="zptmcm7t-x-x-80">meta</span></sub> is constructed as a sum of <span 
class="zptmcm7m-x-x-109">N</span><sub><span 
class="zptmcm7t-x-x-80">cv</span></sub>-dimensional repulsive
Gaussian “hills”, whose height is a chosen energy constant <span 
class="zptmcm7m-x-x-109">W </span>, and whose centers are the previously explored
conﬁgurations <img 
src="colvars-refman-namd33x.png" alt="(ξ(δt),ξ (2δt),...)"  class="left" align="middle" />.
</p><!--l. 3541--><p class="indent" >   During the simulation, the system evolves towards the nearest minimum of the “eﬀective” potential of mean
force <span 
class="zptmcm7m-x-x-109">Ã</span><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">ξ</span><span 
class="zptmcm7t-x-x-109">)</span>, which is the sum of the “real” underlying potential of mean force <span 
class="zptmcm7m-x-x-109">A</span><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">ξ</span><span 
class="zptmcm7t-x-x-109">) </span>and the the metadynamics
potential, <span 
class="zptmcm7m-x-x-109">V</span><sub><span 
class="zptmcm7t-x-x-80">meta</span></sub><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">ξ</span><span 
class="zptmcm7t-x-x-109">)</span>. Therefore, at any given time the probability of observing the conﬁguration <span 
class="zptmcm7m-x-x-109">ξ</span><sup><span 
class="zptmcm7y-x-x-80">∗</span></sup> is
proportional to <span 
class="zptmcm7t-x-x-109">exp</span><img 
src="colvars-refman-namd34x.png" alt="(−A~(ξ ∗)∕κ  T)
         B"  class="left" align="middle" />: this is also the probability that a new Gaussian “hill” is added at that
conﬁguration. If the simulation is run for a suﬃciently long time, each local minimum is canceled out by the
sum of the Gaussian “hills”. At that stage the “eﬀective” potential of mean force <span 
class="zptmcm7m-x-x-109">Ã</span><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">ξ</span><span 
class="zptmcm7t-x-x-109">) </span>is constant, and
<span 
class="zptmcm7y-x-x-109">−</span><span 
class="zptmcm7m-x-x-109">V</span><sub><span 
class="zptmcm7t-x-x-80">meta</span></sub><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">ξ</span><span 
class="zptmcm7t-x-x-109">) </span>is an accurate estimator of the “real” potential of mean force <span 
class="zptmcm7m-x-x-109">A</span><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">ξ</span><span 
class="zptmcm7t-x-x-109">)</span>, save for an additive
constant:
</p>
   <table 
class="equation"><tr><td><a 
 id="x1-69002r25"></a>
   <center class="math-display" >
<img 
src="colvars-refman-namd35x.png" alt="A(ξ) ≃ − Vmeta(ξ )+ K
" class="math-display"  /></center></td><td class="equation-label">(25)</td></tr></table>
                                                                                         
                                                                                         
<!--l. 3550--><p class="nopar" >
</p><!--l. 3552--><p class="indent" >   Assuming that the set of collective variables includes all relevant degrees of freedom, the predicted error of the
estimate is a simple function of the correlation times of the colvars <span 
class="zptmcm7m-x-x-109">τ</span><sub><span 
class="zptmcm7m-x-x-80">ξ</span><sub><span 
class="zptmcm7m-x-x-60">i</span></sub></sub>, and of the user-deﬁned parameters <span 
class="zptmcm7m-x-x-109">W </span>, <span 
class="zptmcm7m-x-x-109">σ</span><sub><span 
class="zptmcm7m-x-x-80">ξ</span><sub><span 
class="zptmcm7m-x-x-60">i</span></sub></sub>
and <span 
class="zptmcm7m-x-x-109">δt </span><span class="cite">[<a 
href="#XBussi2006">24</a>]</span>. In typical applications, a good rule of thumb can be to choose the ratio <span 
class="zptmcm7m-x-x-109">W∕δt </span>much smaller than
<span 
class="zptmcm7m-x-x-109">κ</span><sub><span 
class="zptmcm7t-x-x-80">B</span></sub><span 
class="zptmcm7m-x-x-109">T ∕τ</span><sub><span 
class="zptmcm7m-x-x-109">ξ</span></sub>, where <span 
class="zptmcm7m-x-x-109">τ</span><sub><span 
class="zptmcm7m-x-x-109">ξ</span></sub> is the longest among <span 
class="zptmcm7m-x-x-109">ξ</span> &#x2019;s correlation times: <span 
class="zptmcm7m-x-x-109">σ</span><sub><span 
class="zptmcm7m-x-x-80">ξ</span><sub><span 
class="zptmcm7m-x-x-60">i</span></sub></sub> then dictates the resolution of the calculated
PMF.
</p><!--l. 3559--><p class="indent" >   To enable a metadynamics calculation, a <span 
class="ot1-zi4r-0-x-x-109">metadynamics </span>block must be deﬁned in the colvars conﬁguration ﬁle.
Its mandatory keywords are <span 
class="ot1-zi4r-0-x-x-109">colvars</span>, which lists all the variables involved, and <span 
class="ot1-zi4r-0-x-x-109">hillWeight</span>, which speciﬁes the
weight parameter <span 
class="zptmcm7m-x-x-109">W </span>. The parameters <span 
class="zptmcm7m-x-x-109">δt </span>and <span 
class="zptmcm7m-x-x-109">σ</span><sub><span 
class="zptmcm7m-x-x-80">ξ</span></sub> speciﬁed by the optional keywords <span 
class="ot1-zi4r-0-x-x-109">newHillFrequency </span>and
<span 
class="ot1-zi4r-0-x-x-109">hillWidth</span>: </p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-69003"></a> <span 
class="ot1-zi4r-0-x-x-120">name: </span>see deﬁnition of <a 
href="#sec:colvarbias"><span 
class="ot1-zi4r-0-x-x-109">name</span></a> in sec. <a 
href="#sec:colvarbias">7<!--tex4ht:ref: sec:colvarbias --></a> (biasing and analysis methods)
     </li>
     <li class="itemize"><a 
 id="dx1-69004"></a> <span 
class="ot1-zi4r-0-x-x-120">colvars: </span>see deﬁnition of <a 
href="#sec:colvarbias"><span 
class="ot1-zi4r-0-x-x-109">colvars</span></a> in sec. <a 
href="#sec:colvarbias">7<!--tex4ht:ref: sec:colvarbias --></a> (biasing and analysis methods)
     </li>
     <li class="itemize"><a 
 id="dx1-69005"></a> <span 
class="ot1-zi4r-0-x-x-120">outputEnergy: </span>see deﬁnition of <a 
href="#sec:colvarbias"><span 
class="ot1-zi4r-0-x-x-109">outputEnergy</span></a> in sec. <a 
href="#sec:colvarbias">7<!--tex4ht:ref: sec:colvarbias --></a> (biasing and analysis methods)
     </li>
     <li class="itemize"><a 
 id="dx1-69006"></a> <span 
class="ot1-zi4r-0-x-x-120">hillWeight </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Height of each hill (kcal/mol)<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">metadynamics</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive decimal <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>This option sets the height <span 
class="zptmcm7m-x-x-109">W </span>of the Gaussian hills that are added during this run. Lower
     values  provide  more  accurate  sampling  of  the  system&#x2019;s  degrees  of  freedom  at  the  price  of  longer
     simulation times to complete a PMF calculation based on metadynamics.
     </li>
     <li class="itemize"><a 
 id="dx1-69007"></a> <span 
class="ot1-zi4r-0-x-x-120">newHillFrequency </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Frequency of hill creation<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">metadynamics</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive integer <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">1000</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>This option sets the number of integration steps after which a new hill is added to the
     metadynamics potential. Its value determines the parameter <span 
class="zptmcm7m-x-x-109">δt </span>in eq. <a 
href="#eq:colvars_meta_pot">24<!--tex4ht:ref: eq:colvars_meta_pot --></a>. Higher values provide more
     accurate sampling at the price of longer simulation times to complete a PMF calculation.
     </li>
     <li class="itemize"><a 
 id="dx1-69008"></a> <span 
class="ot1-zi4r-0-x-x-120">hillWidth </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Relative width of a Gaussian hill with respect to the colvar&#x2019;s width<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">metadynamics</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive decimal <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:</span>  <img 
src="colvars-refman-namd36x.png" alt="√ ---
  2π"  class="sqrt"  /><span 
class="zptmcm7m-x-x-109">∕</span><span 
class="zptmcm7t-x-x-109">2 </span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>The Gaussian width along each colvar, <span 
class="zptmcm7t-x-x-109">2</span><span 
class="zptmcm7m-x-x-109">σ</span><sub><span 
class="zptmcm7m-x-x-80">ξ</span><sub><span 
class="zptmcm7m-x-x-60">i</span></sub></sub>, is set as the product between this number
     and the colvar&#x2019;s parameter <span 
class="zptmcm7m-x-x-109">w</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub> given by <span 
class="ot1-zi4r-0-x-x-109">width </span>(see <a 
href="#sec:colvar_general">4.1<!--tex4ht:ref: sec:colvar_general --></a>); such product is printed in the standard output
     of NAMD. The default value of this number gives a Gaussian hill function whose volume is equal
     to the product of <span 
class="zptmcm7v-x-x-109">∏</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub><span 
class="zptmcm7m-x-x-109">w</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub>, the volume of one <span 
class="ot1-zi4r-0-x-x-109">histogram </span>bin (see <a 
href="#sec:colvarbias_histogram">7.8<!--tex4ht:ref: sec:colvarbias_histogram --></a>), and <span 
class="zptmcm7m-x-x-109">W </span>. <span 
class="ptmb7t-x-x-109">Tip: </span><span 
class="ptmri7t-x-x-109">use this property</span>
     <span 
class="ptmri7t-x-x-109">to estimate the fraction of colvar space covered by the Gaussian bias within a given simulation time.</span>
     When <span 
class="ot1-zi4r-0-x-x-109">useGrids </span>is on, the default value also gives acceptable discretization errors <span class="cite">[<a 
href="#XFiorin2013">1</a>]</span>: for smoother
     visualization, this parameter may be increased and the <span 
class="ot1-zi4r-0-x-x-109">width </span><span 
class="zptmcm7m-x-x-109">w</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub> decreased in the same proportion.
     <span 
class="ptmb7t-x-x-109">Note: </span><span 
class="ptmri7t-x-x-109">values smaller than 1 are not recommended</span>. </li></ul>
                                                                                         
                                                                                         
<!--l. 3605--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">7.3.1    </span> <a 
 id="x1-700007.3.1"></a>Output ﬁles</h5>
<!--l. 3607--><p class="noindent" >When interpolating grids are enabled (default behavior), the PMF is written every <span 
class="ot1-zi4r-0-x-x-109">colvarsRestartFrequency </span>steps
to the ﬁle <span 
class="ptmri7t-x-x-109">outputName</span><span 
class="ot1-zi4r-0-x-x-109">.pmf</span>. The following two options allow to control this behavior and to visually track statistical
convergence:
</p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-70001"></a> <span 
class="ot1-zi4r-0-x-x-120">writeFreeEnergyFile </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Periodically write the PMF for visualization<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">metadynamics</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">on</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:       </span>When    <span 
class="ot1-zi4r-0-x-x-109">useGrids  </span>and    this    option    are    <span 
class="ot1-zi4r-0-x-x-109">on</span>,    the    PMF    is    written    every
     <span 
class="ot1-zi4r-0-x-x-109">colvarsRestartFrequency </span>steps.
     </li>
     <li class="itemize"><a 
 id="dx1-70002"></a> <span 
class="ot1-zi4r-0-x-x-120">keepFreeEnergyFiles </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Keep all the PMF ﬁles<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">metadynamics</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">off</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>When <span 
class="ot1-zi4r-0-x-x-109">writeFreeEnergyFile </span>and this option are <span 
class="ot1-zi4r-0-x-x-109">on</span>, the step number is included in the
     ﬁle name, thus generating a series of PMF ﬁles. Activating this option can be useful to follow more
     closely the convergence of the simulation, by comparing PMFs separated by short times.
     </li></ul>
<!--l. 3634--><p class="indent" >   <span 
class="ptmb7t-x-x-109">Note: </span>when Gaussian hills are deposited near <span 
class="ot1-zi4r-0-x-x-109">lowerBoundary </span>or <span 
class="ot1-zi4r-0-x-x-109">upperBoundary </span>(see <a 
href="#sec:colvar_general">4.1<!--tex4ht:ref: sec:colvar_general --></a>) and interpolating
grids are used (default behavior), their truncation can give rise to accumulating errors. In these cases, as a measure
of fault-tolerance all Gaussian hills near the boundaries are included in the output state ﬁle, and are recalculated
analytically whenever the colvar falls outside the grid&#x2019;s boundaries. (Such measure protects the accuracy of the
calculation, and can only be disabled by <span 
class="ot1-zi4r-0-x-x-109">hardLowerBoundary </span>or <span 
class="ot1-zi4r-0-x-x-109">hardUpperBoundary</span>.) To avoid gradual loss
of performance and growth of the state ﬁle, either one of the following solutions is recommended:
</p>
     <ul class="itemize1">
     <li class="itemize">enabling the option <span 
class="ot1-zi4r-0-x-x-109">expandBoundaries</span>, so that the grid&#x2019;s boundaries are automatically recalculated
     whenever necessary; the resulting <span 
class="ot1-zi4r-0-x-x-109">.pmf </span>will have its abscissas expanded accordingly;
     </li>
     <li class="itemize">applying  a  <span 
class="ot1-zi4r-0-x-x-109">harmonicWalls </span>bias  with  the  wall  locations  well  within  the  interval  delimited  by
     <span 
class="ot1-zi4r-0-x-x-109">lowerBoundary </span>and <span 
class="ot1-zi4r-0-x-x-109">upperBoundary</span>.</li></ul>
                                                                                         
                                                                                         
<!--l. 3644--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">7.3.2    </span> <a 
 id="x1-710007.3.2"></a>Performance tuning</h5>
<!--l. 3646--><p class="noindent" >The following options control the computational cost of metadynamics calculations, but do not aﬀect results.
Default values are chosen to minimize such cost with no loss of accuracy.
</p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-71001"></a> <span 
class="ot1-zi4r-0-x-x-120">useGrids </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Interpolate the hills with grids<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">metadynamics</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">on</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:     </span>This  option  discretizes  all  hills  for  improved  performance,  accumulating  their
     energy  and  their  gradients  on  two  separate  grids  of  equal  spacing.  Grids  are  deﬁned  by  the
     values  of  <span 
class="ot1-zi4r-0-x-x-109">lowerBoundary</span>,  <span 
class="ot1-zi4r-0-x-x-109">upperBoundary </span>and  <span 
class="ot1-zi4r-0-x-x-109">width </span>for  each  colvar.  Currently,  this  option  is
     implemented for all types of variables except the non-scalar types (<span 
class="ot1-zi4r-0-x-x-109">distanceDir </span>or <span 
class="ot1-zi4r-0-x-x-109">orientation</span>).
     If <span 
class="ot1-zi4r-0-x-x-109">expandBoundaries </span>is deﬁned in one of the colvars, grids are automatically expanded along the
     direction of that colvar.
     </li>
     <li class="itemize"><a 
 id="dx1-71002"></a> <span 
class="ot1-zi4r-0-x-x-120">rebinGrids </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Recompute the grids when reading a state ﬁle<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">metadynamics</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">off</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>When restarting from a state ﬁle, the grid&#x2019;s parameters (boundaries and widths) saved
     in the state ﬁle override those in the conﬁguration ﬁle. Enabling this option forces the grids to match
     those in the current conﬁguration ﬁle.
     </li></ul>
<!--l. 3684--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">7.3.3    </span> <a 
 id="x1-720007.3.3"></a>Well-tempered metadynamics</h5>
<!--l. 3686--><p class="noindent" >The following options deﬁne the conﬁguration for the “well-tempered” metadynamics approach <span class="cite">[<a 
href="#XBarducci2008">25</a>]</span>:
</p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-72001"></a> <span 
class="ot1-zi4r-0-x-x-120">wellTempered </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Perform well-tempered metadynamics<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">metadynamics</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">off</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>If enabled, this ﬂag causes well-tempered metadynamics as described by Barducci et
     al.<span class="cite">[<a 
href="#XBarducci2008">25</a>]</span> to be performed, rather than standard metadynamics. The parameter <span 
class="ot1-zi4r-0-x-x-109">biasTemperature </span>is then
     required.This feature was contributed by Li Li (Luthey-Schulten group, Departement of Chemistry,
     UIUC).
                                                                                         
                                                                                         
     </li>
     <li class="itemize"><a 
 id="dx1-72002"></a> <span 
class="ot1-zi4r-0-x-x-120">biasTemperature </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Temperature bias for well-tempered metadynamics<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">metadynamics</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive decimal <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>When running metadynamics in the long time limit, collective variable space is sampled
     to a modiﬁed temperature <span 
class="zptmcm7m-x-x-109">T </span><span 
class="zptmcm7t-x-x-109">+Δ</span><span 
class="zptmcm7m-x-x-109">T </span>. In conventional metadynamics, the temperature “boost” <span 
class="zptmcm7t-x-x-109">Δ</span><span 
class="zptmcm7m-x-x-109">T </span>would
     constantly increases with time. Instead, in well-tempered metadynamics <span 
class="zptmcm7t-x-x-109">Δ</span><span 
class="zptmcm7m-x-x-109">T </span>must be deﬁned by the
     user via <span 
class="ot1-zi4r-0-x-x-109">biasTemperature</span>. The written PMF includes the scaling factor <span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">T </span><span 
class="zptmcm7t-x-x-109">+Δ</span><span 
class="zptmcm7m-x-x-109">T </span><span 
class="zptmcm7t-x-x-109">)</span><span 
class="zptmcm7m-x-x-109">∕</span><span 
class="zptmcm7t-x-x-109">Δ</span><span 
class="zptmcm7m-x-x-109">T </span><span class="cite">[<a 
href="#XBarducci2008">25</a>]</span>. A careful
     choice of <span 
class="zptmcm7t-x-x-109">Δ</span><span 
class="zptmcm7m-x-x-109">T </span>determines the sampling and convergence rate, and is hence crucial to the success of a
     well-tempered metadynamics simulation.</li></ul>
<!--l. 3719--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">7.3.4    </span> <a 
 id="x1-730007.3.4"></a>Multiple-replicas metadynamics</h5>
<!--l. 3721--><p class="noindent" >The following options deﬁne metadynamics calculations with more than one replica:
</p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-73001"></a> <span 
class="ot1-zi4r-0-x-x-120">multipleReplicas </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Multiple replicas metadynamics<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">metadynamics</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">off</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>If this option is <span 
class="ot1-zi4r-0-x-x-109">on</span>, multiple (independent) replica of the same system can be run at the
     same time, and their hills will be combined to obtain a single PMF <span class="cite">[<a 
href="#XRaiteri2005">26</a>]</span>. Replicas are identiﬁed by
     the value of <span 
class="ot1-zi4r-0-x-x-109">replicaID</span>. Communication is done by ﬁles: each replica must be able to read the ﬁles
     created by the others, whose paths are communicated through the ﬁle <span 
class="ot1-zi4r-0-x-x-109">replicasRegistry</span>. This ﬁle,
     and the ﬁles listed in it, are read every <span 
class="ot1-zi4r-0-x-x-109">replicaUpdateFrequency </span>steps. Every time the colvars state
     ﬁle is written (<span 
class="ot1-zi4r-0-x-x-109">colvarsRestartFrequency</span>), the ﬁle:<br 
class="newline" />“<span 
class="ptmri7t-x-x-109">outputName</span><span 
class="ot1-zi4r-0-x-x-109">.colvars.</span><span 
class="ptmri7t-x-x-109">name</span><span 
class="ot1-zi4r-0-x-x-109">.</span><span 
class="ptmri7t-x-x-109">replicaID</span><span 
class="ot1-zi4r-0-x-x-109">.state</span>”   is   also   written,   containing   the   state   of   the
     metadynamics bias for <span 
class="ot1-zi4r-0-x-x-109">replicaID</span>. In the time steps between <span 
class="ot1-zi4r-0-x-x-109">colvarsRestartFrequency</span>, new hills
     are temporarily written to the ﬁle:<br 
class="newline" />“<span 
class="ptmri7t-x-x-109">outputName</span><span 
class="ot1-zi4r-0-x-x-109">.colvars.</span><span 
class="ptmri7t-x-x-109">name</span><span 
class="ot1-zi4r-0-x-x-109">.</span><span 
class="ptmri7t-x-x-109">replicaID</span><span 
class="ot1-zi4r-0-x-x-109">.hills</span>”, which serves as communication buﬀer. These ﬁles
     are only required for communication, and may be deleted after a new MD run is started with a diﬀerent
     <span 
class="ot1-zi4r-0-x-x-109">outputName</span>.
     </li>
     <li class="itemize"><a 
 id="dx1-73002"></a> <span 
class="ot1-zi4r-0-x-x-120">replicaID </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Set the identiﬁer for this replica<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">metadynamics</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>string <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>If <span 
class="ot1-zi4r-0-x-x-109">multipleReplicas </span>is <span 
class="ot1-zi4r-0-x-x-109">on</span>, this option sets a unique identiﬁer for this replica. All
     replicas should use identical collective variable conﬁgurations, except for the value of this option.
     </li>
     <li class="itemize"><a 
 id="dx1-73003"></a> <span 
class="ot1-zi4r-0-x-x-120">replicasRegistry </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Multiple replicas database ﬁle<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">metadynamics</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>UNIX ﬁlename <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span>“<span 
class="ptmri7t-x-x-109">name</span><span 
class="ot1-zi4r-0-x-x-109">.replica</span><span 
class="ot1-zi4r-0-x-x-109">_files.txt</span>” <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>If <span 
class="ot1-zi4r-0-x-x-109">multipleReplicas </span>is <span 
class="ot1-zi4r-0-x-x-109">on</span>, this option sets the path to the replicas&#x2019; database ﬁle.
     </li>
     <li class="itemize"><a 
 id="dx1-73004"></a> <span 
class="ot1-zi4r-0-x-x-120">replicaUpdateFrequency </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>How often hills are communicated between replicas<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">metadynamics</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive integer <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">newHillFrequency</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>If <span 
class="ot1-zi4r-0-x-x-109">multipleReplicas </span>is <span 
class="ot1-zi4r-0-x-x-109">on</span>, this option sets the number of steps after which each replica
     (re)reads the other replicas&#x2019; ﬁles. The lowest meaningful value of this number is <span 
class="ot1-zi4r-0-x-x-109">newHillFrequency</span>.
     If  access  to  the  ﬁle  system  is  signiﬁcantly  aﬀecting  the  simulation  performance,  this  number
     can  be  increased,  at  the  price  of  reduced  synchronization  between  replicas.  Values  higher  than
     <span 
class="ot1-zi4r-0-x-x-109">colvarsRestartFrequency </span>may not improve performance signiﬁcantly.
     </li>
     <li class="itemize"><a 
 id="dx1-73005"></a> <span 
class="ot1-zi4r-0-x-x-120">dumpPartialFreeEnergyFile </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Periodically write the contribution to the PMF from this replica<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span>
     <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">metadynamics</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">on</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>When <span 
class="ot1-zi4r-0-x-x-109">multipleReplicas </span>is <span 
class="ot1-zi4r-0-x-x-109">on</span>, the ﬁle <span 
class="ptmri7t-x-x-109">outputName</span><span 
class="ot1-zi4r-0-x-x-109">.pmf </span>contains the combined PMF
     from all replicas, provided that <span 
class="ot1-zi4r-0-x-x-109">useGrids </span>is <span 
class="ot1-zi4r-0-x-x-109">on </span>(default). Enabling this option produces an additional
     ﬁle  <span 
class="ptmri7t-x-x-109">outputName</span><span 
class="ot1-zi4r-0-x-x-109">.partial.pmf</span>,  which  can  be  useful  to  quickly  monitor  the  contribution  of  each
     replica to the PMF.
     </li></ul>
<!--l. 3819--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">7.3.5    </span> <a 
 id="x1-740007.3.5"></a>Compatibility and post-processing</h5>
<!--l. 3820--><p class="noindent" >The following options may be useful only for applications that go beyond the calculation of a PMF by
metadynamics: </p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-74001"></a> <span 
class="ot1-zi4r-0-x-x-120">name </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Name of this metadynamics instance<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">metadynamics</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>string <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span>“<span 
class="ot1-zi4r-0-x-x-109">meta</span>” + rank number <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>This option sets the name for this metadynamics instance. While it is not advisable to
     use more than one metadynamics instance within the same simulation, this allows to distinguish each
     instance from the others. If there is more than one metadynamics instance, the name of this bias is
     included in the metadynamics output ﬁle names, such as e.g. the <span 
class="ot1-zi4r-0-x-x-109">.pmf </span>ﬁle.
     </li>
     <li class="itemize"><a 
 id="dx1-74002"></a> <span 
class="ot1-zi4r-0-x-x-120">keepHills </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Write each individual hill to the state ﬁle<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">metadynamics</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">off</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>When  <span 
class="ot1-zi4r-0-x-x-109">useGrids </span>and  this  option  are  <span 
class="ot1-zi4r-0-x-x-109">on</span>,  all  hills  are  saved  to  the  state  ﬁle  in  their
     analytic  form,  alongside  their  grids.  This  makes  it  possible  to  later  use  exact  analytic  Gaussians
     for  <span 
class="ot1-zi4r-0-x-x-109">rebinGrids</span>.  To  only  keep  track  of  the  history  of  the  added  hills,  <span 
class="ot1-zi4r-0-x-x-109">writeHillsTrajectory </span>is
     preferable.
     </li>
     <li class="itemize"><a 
 id="dx1-74003"></a> <span 
class="ot1-zi4r-0-x-x-120">writeHillsTrajectory </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Write a log of new hills<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">metadynamics</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">on</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>If  this  option  is  <span 
class="ot1-zi4r-0-x-x-109">on</span>,  a  logﬁle  is  written  by  the  <span 
class="ot1-zi4r-0-x-x-109">metadynamics </span>bias,  with  the  name
     “<span 
class="ptmri7t-x-x-109">outputName</span><span 
class="ot1-zi4r-0-x-x-109">.colvars.</span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ot1-zi4r-0-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span><span 
class="ot1-zi4r-0-x-x-109">.hills.traj</span>”, which can be useful to follow the time series of the
     hills. When <span 
class="ot1-zi4r-0-x-x-109">multipleReplicas </span>is <span 
class="ot1-zi4r-0-x-x-109">on</span>, its name changes to<br 
class="newline" />“<span 
class="ptmri7t-x-x-109">outputName</span><span 
class="ot1-zi4r-0-x-x-109">.colvars.</span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ot1-zi4r-0-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span><span 
class="ot1-zi4r-0-x-x-109">.</span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ot1-zi4r-0-x-x-109">replicaID</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span><span 
class="ot1-zi4r-0-x-x-109">.hills.traj</span>”.  This  ﬁle  can  be  used  to  quickly
     visualize  the  positions  of  all  added  hills,  in  case  <span 
class="ot1-zi4r-0-x-x-109">newHillFrequency </span>does  not  coincide  with
     <span 
class="ot1-zi4r-0-x-x-109">colvarsRestartFrequency</span>.
     </li></ul>
<!--l. 3882--><p class="noindent" >
</p>
   <h4 class="subsectionHead"><span class="titlemark">7.4    </span> <a 
 id="sec:colvarbias_harmonic"></a>Harmonic restraints</h4>
<!--l. 3885--><p class="noindent" >The harmonic biasing method may be used to enforce ﬁxed or moving restraints, including variants of Steered and
Targeted MD. Within energy minimization runs, it allows for restrained minimization, e.g. to calculate relaxed
potential energy surfaces. In the context of the Colvars module, harmonic potentials are meant according to their
textbook deﬁnition:
</p>
   <table 
class="equation"><tr><td><a 
 id="x1-75001r26"></a>
   <center class="math-display" >
<img 
src="colvars-refman-namd37x.png" alt="         (       )
       1-  ξ-−-ξ0 2
V(ξ) = 2k    wξ
" class="math-display"  /></center></td><td class="equation-label">(26)</td></tr></table>
<!--l. 3893--><p class="nopar" >
Note that this diﬀers from harmonic bond and angle potentials in common force ﬁelds, where the factor of one half
is typically omitted, resulting in a non-standard deﬁnition of the force constant.
</p><!--l. 3898--><p class="indent" >   The formula above includes the characteristic length scale <span 
class="zptmcm7m-x-x-109">w</span><sub><span 
class="zptmcm7m-x-x-80">ξ</span></sub> of the colvar <span 
class="zptmcm7m-x-x-109">ξ </span>(keyword <span 
class="ot1-zi4r-0-x-x-109">width</span>, see <a 
href="#sec:colvar_general">4.1<!--tex4ht:ref: sec:colvar_general --></a>) to allow
the deﬁnition of a multi-dimensional restraint with a uniﬁed force constant:
</p>
   <table 
class="equation"><tr><td><a 
 id="eq:colvarbias_harmonic_multi"></a>
                                                                                         
                                                                                         
   <center class="math-display" >
<img 
src="colvars-refman-namd38x.png" alt="                 M (       )2
V(ξ ,...,ξ ) = 1k     ξi−-ξ0
   1     M    2 i∑=1   w ξ
" class="math-display"  /></center></td><td class="equation-label">(27)</td></tr></table>
<!--l. 3902--><p class="nopar" >
</p><!--l. 3904--><p class="indent" >   If one-dimensional or homogeneous multi-dimensional restraints are deﬁned, and there are no other uses for the
parameter <span 
class="zptmcm7m-x-x-109">w</span><sub><span 
class="zptmcm7m-x-x-80">ξ</span></sub>, <span 
class="ptmri7t-x-x-109">the parameter </span><span 
class="ot1-zi4r-0-x-x-109">width </span><span 
class="ptmri7t-x-x-109">can be left at its default value of </span><span 
class="zptmcm7t-x-x-109">1</span>.
</p><!--l. 3907--><p class="indent" >   The restraint energy is reported by NAMD under the MISC title. A harmonic restraint is set up by a
<span 
class="ot1-zi4r-0-x-x-109">harmonic</span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109">...</span><span 
class="cmsy-10x-x-109">} </span>block, which may contain (in addition to the standard option <span 
class="ot1-zi4r-0-x-x-109">colvars</span>) the following keywords:
</p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-75003"></a> <span 
class="ot1-zi4r-0-x-x-120">name: </span>see deﬁnition of <a 
href="#sec:colvarbias"><span 
class="ot1-zi4r-0-x-x-109">name</span></a> in sec. <a 
href="#sec:colvarbias">7<!--tex4ht:ref: sec:colvarbias --></a> (biasing and analysis methods)
     </li>
     <li class="itemize"><a 
 id="dx1-75004"></a> <span 
class="ot1-zi4r-0-x-x-120">colvars: </span>see deﬁnition of <a 
href="#sec:colvarbias"><span 
class="ot1-zi4r-0-x-x-109">colvars</span></a> in sec. <a 
href="#sec:colvarbias">7<!--tex4ht:ref: sec:colvarbias --></a> (biasing and analysis methods)
     </li>
     <li class="itemize"><a 
 id="dx1-75005"></a> <span 
class="ot1-zi4r-0-x-x-120">outputEnergy: </span>see deﬁnition of <a 
href="#sec:colvarbias"><span 
class="ot1-zi4r-0-x-x-109">outputEnergy</span></a> in sec. <a 
href="#sec:colvarbias">7<!--tex4ht:ref: sec:colvarbias --></a> (biasing and analysis methods)
     </li>
     <li class="itemize"><a 
 id="dx1-75006"></a> <span 
class="ot1-zi4r-0-x-x-120">forceConstant </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Scaled force constant (kcal/mol)<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">harmonic</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive decimal <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">1.0</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>This deﬁnes a scaled force constant <span 
class="zptmcm7m-x-x-109">k </span>for the harmonic potential (eq. <a 
href="#eq:colvarbias_harmonic_multi">27<!--tex4ht:ref: eq:colvarbias_harmonic_multi --></a>). To ensure
     consistency for multidimensional restraints, it is divided internally by the square of the speciﬁc <span 
class="ot1-zi4r-0-x-x-109">width</span>
     for each colvar involved (which is 1 by default), so that all colvars are eﬀectively dimensionless and of
     commensurate size. For instance, setting a scaled force constant of 10 kcal/mol acting on two colvars,
     an angle with a <span 
class="ot1-zi4r-0-x-x-109">width </span>of 5 degrees and a distance with a width of 0.5 Å, will apply actual force
     constants of 0.4 kcal/mol<span 
class="zptmcm7y-x-x-109">×</span>degree<sup><span 
class="zptmcm7y-x-x-80">−</span><span 
class="zptmcm7t-x-x-80">2</span></sup> for the angle and 40 kcal/mol/Å<sup><span 
class="zptmcm7t-x-x-80">2</span></sup> for the distance.
     </li>
     <li class="itemize"><a 
 id="dx1-75007"></a> <span 
class="ot1-zi4r-0-x-x-120">centers </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Initial harmonic restraint centers<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">harmonic</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>space-separated list of colvar values <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>The centers (equilibrium values) of the restraint, <span 
class="zptmcm7m-x-x-109">ξ</span><sub><span 
class="zptmcm7t-x-x-80">0</span></sub>, are entered here. The number of
     values must be the number of requested colvars. Each value is a decimal number if the corresponding
     colvar returns a scalar, a “<span 
class="ot1-zi4r-0-x-x-109">(x, y, z)</span>” triplet if it returns a unit vector or a vector, and a “<span 
class="ot1-zi4r-0-x-x-109">(q0, q1,</span>
     <span 
class="ot1-zi4r-0-x-x-109">q2, q3)</span>” quadruplet if it returns a rotational quaternion. If a colvar has periodicities or symmetries,
     its closest image to the restraint center is considered when calculating the harmonic potential.</li></ul>
                                                                                         
                                                                                         
<!--l. 3951--><p class="indent" >   <span 
class="ptmb7t-x-x-109">Tip: </span>A complex set of restraints can be applied to a system, by deﬁning several colvars, and applying one
or more harmonic restraints to diﬀerent groups of colvars. In some cases, dozens of colvars can be
deﬁned, but their value may not be relevant: to limit the size of the colvars trajectory ﬁle, it may be
wise to disable <span 
class="ot1-zi4r-0-x-x-109">outputValue </span>for such “ancillary” variables, and leave it enabled only for “relevant”
ones.
</p><!--l. 3959--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">7.4.1    </span> <a 
 id="x1-760007.4.1"></a>Moving restraints: steered molecular dynamics</h5>
<!--l. 3962--><p class="noindent" >The following options allow to change gradually the centers of the harmonic restraints during a simulations.
When the centers are changed continuously, a steered MD in a collective variable space is carried
out.
</p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-76001"></a> <span 
class="ot1-zi4r-0-x-x-120">targetCenters </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Steer the restraint centers towards these targets<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">harmonic</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>space-separated list of colvar values <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>When deﬁned, the current <span 
class="ot1-zi4r-0-x-x-109">centers </span>will be moved towards these values during the
     simulation.  By  default,  the  centers  are  moved  over  a  total  of  <span 
class="ot1-zi4r-0-x-x-109">targetNumSteps </span>steps  by  a  linear
     interpolation, in the spirit of Steered MD. If <span 
class="ot1-zi4r-0-x-x-109">targetNumStages </span>is set to a nonzero value, the change
     is performed in discrete stages, lasting <span 
class="ot1-zi4r-0-x-x-109">targetNumSteps </span>steps <span 
class="ptmri7t-x-x-109">each</span>. This second mode may be used
     to sample successive windows in the context of an Umbrella Sampling simulation. When continuing
     a simulation run, the <span 
class="ot1-zi4r-0-x-x-109">centers </span>speciﬁed in the conﬁguration ﬁle <span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ot1-zi4r-0-x-x-109">colvarsConfig</span><span 
class="zptmcm7m-x-x-109">&#x003E; </span>are overridden
     by those saved in the restart ﬁle <span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ot1-zi4r-0-x-x-109">colvarsInput</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span>. To perform Steered MD in an arbitrary space of
     colvars, it is suﬃcient to use this option and enable <span 
class="ot1-zi4r-0-x-x-109">outputAppliedForce </span>within each of the colvars
     involved.
     </li>
     <li class="itemize"><a 
 id="dx1-76002"></a> <span 
class="ot1-zi4r-0-x-x-120">targetNumSteps </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Number of steps for steering<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">harmonic</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive integer <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>In single-stage (continuous) transformations, deﬁnes the number of MD steps required
     to move the restraint centers (or force constant) towards the values speciﬁed with <span 
class="ot1-zi4r-0-x-x-109">targetCenters </span>or
     <span 
class="ot1-zi4r-0-x-x-109">targetForceConstant</span>. After the target values have been reached, the centers (resp. force constant)
     are kept ﬁxed. In multi-stage transformations, this sets the number of MD steps <span 
class="ptmri7t-x-x-109">per stage</span>.
     </li>
     <li class="itemize"><a 
 id="dx1-76003"></a> <span 
class="ot1-zi4r-0-x-x-120">outputCenters </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Write the current centers to the trajectory ﬁle<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">harmonic</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">off</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>If this option is chosen and <span 
class="ot1-zi4r-0-x-x-109">colvarsTrajFrequency </span>is not zero, the positions of the
     restraint  centers  will  be  written  to  the  trajectory  ﬁle  during  the  simulation.  This  option  allows  to
     conveniently extract the PMF from the colvars trajectory ﬁles in a steered MD calculation.
     </li>
                                                                                         
                                                                                         
     <li class="itemize"><a 
 id="dx1-76004"></a> <span 
class="ot1-zi4r-0-x-x-120">outputAccumulatedWork  </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Write the accumulated work of the moving restraint to the trajectory
     ﬁle<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">harmonic</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">off</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>If this option is chosen, <span 
class="ot1-zi4r-0-x-x-109">targetCenters </span>is deﬁned, and <span 
class="ot1-zi4r-0-x-x-109">colvarsTrajFrequency </span>is not
     zero, the accumulated work from the beginning of the simulation will be written to the trajectory ﬁle.
     If the simulation has been continued from a previous state ﬁle, the previously accumulated work is
     included in the integral. This option allows to conveniently extract the PMF from the colvars trajectory
     ﬁles in a steered MD calculation.
     </li></ul>
<!--l. 4031--><p class="indent" >   <span 
class="ptmb7t-x-x-109">Note on restarting moving restraint simulations: </span>Information about the current step and stage of a
simulation with moving restraints is stored in the restart ﬁle (state ﬁle). Thus, such simulations can be
run in several chunks, and restarted directly using the same colvars conﬁguration ﬁle. In case of a
restart, the values of parameters such as <span 
class="ot1-zi4r-0-x-x-109">targetCenters</span>, <span 
class="ot1-zi4r-0-x-x-109">targetNumSteps</span>, etc. should not be changed
manually.
</p><!--l. 4040--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">7.4.2    </span> <a 
 id="x1-770007.4.2"></a>Moving restraints: umbrella sampling</h5>
<!--l. 4043--><p class="noindent" >The centers of the harmonic restraints can also be changed in discrete stages: in this cases a one-dimensional
umbrella sampling simulation is performed. The sampling windows in simulation are calculated in sequence. The
colvars trajectory ﬁle may then be used both to evaluate the correlation times between consecutive windows, and to
calculate the frequency distribution of the colvar of interest in each window. Furthermore, frequency
distributions on a predeﬁned grid can be automatically obtained by using the <span 
class="ot1-zi4r-0-x-x-109">histogram </span>bias (see
<a 
href="#sec:colvarbias_histogram">7.8<!--tex4ht:ref: sec:colvarbias_histogram --></a>).
</p><!--l. 4048--><p class="indent" >   To activate an umbrella sampling simulation, the same keywords as in the previous section can be used, with the
addition of the following: </p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-77001"></a> <span 
class="ot1-zi4r-0-x-x-120">targetNumStages </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Number of stages for steering<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">harmonic</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>non-negative integer <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">0</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>If non-zero, sets the number of stages in which the restraint centers or force constant are
     changed to their target values. If zero, the change is continuous. Each stage lasts <span 
class="ot1-zi4r-0-x-x-109">targetNumSteps </span>MD
     steps. To sample both ends of the transformation, the simulation should be run for <span 
class="ot1-zi4r-0-x-x-109">targetNumSteps</span>
     <span 
class="zptmcm7y-x-x-109">× </span>(<span 
class="ot1-zi4r-0-x-x-109">targetNumStages </span>+ 1).
     </li></ul>
<!--l. 4067--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">7.4.3    </span> <a 
 id="x1-780007.4.3"></a>Changing force constant</h5>
<!--l. 4070--><p class="noindent" >The force constant of the harmonic restraint may also be changed to equilibrate <span class="cite">[<a 
href="#XDeng2009">27</a>]</span>.
                                                                                         
                                                                                         
</p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-78001"></a> <span 
class="ot1-zi4r-0-x-x-120">targetForceConstant </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Change the force constant towards this value<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">harmonic</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive decimal <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>When deﬁned, the current <span 
class="ot1-zi4r-0-x-x-109">forceConstant </span>will be moved towards this value during the
     simulation. Time evolution of the force constant is dictated by the <span 
class="ot1-zi4r-0-x-x-109">targetForceExponent </span>parameter
     (see below). By default, the force constant is changed smoothly over a total of <span 
class="ot1-zi4r-0-x-x-109">targetNumSteps </span>steps.
     This is useful to introduce or remove restraints in a progressive manner. If <span 
class="ot1-zi4r-0-x-x-109">targetNumStages </span>is set
     to a nonzero value, the change is performed in discrete stages, lasting <span 
class="ot1-zi4r-0-x-x-109">targetNumSteps </span>steps <span 
class="ptmri7t-x-x-109">each</span>.
     This second mode may be used to compute the conformational free energy change associated with the
     restraint, within the FEP or TI formalisms. For convenience, the code provides an estimate of the free
     energy derivative for use in TI. A more complete free energy calculation (particularly with regard to
     convergence analysis), while not handled by the Colvars module, can be performed by post-processing
     the colvars trajectory, if <span 
class="ot1-zi4r-0-x-x-109">colvarsTrajFrequency </span>is set to a suitably small value. It should be noted,
     however, that restraint free energy calculations may be handled more eﬃciently by an indirect route,
     through the determination of a PMF for the restrained coordinate.<span class="cite">[<a 
href="#XDeng2009">27</a>]</span>
     </li>
     <li class="itemize"><a 
 id="dx1-78002"></a> <span 
class="ot1-zi4r-0-x-x-120">targetForceExponent </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Exponent in the time-dependence of the force constant<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">harmonic</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>decimal equal to or greater than 1.0 <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">1.0</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>Sets  the  exponent,  <span 
class="zptmcm7m-x-x-109">α</span>,  in  the  function  used  to  vary  the  force  constant  as  a  function
     of  time.  The  force  is  varied  according  to  a  coupling  parameter  <span 
class="zptmcm7m-x-x-109">λ</span>,  raised  to  the  power  <span 
class="zptmcm7m-x-x-109">α</span>:  <span 
class="zptmcm7m-x-x-109">k</span><sub><span 
class="zptmcm7m-x-x-80">λ</span></sub> <span 
class="zptmcm7t-x-x-109">=</span>
     <span 
class="zptmcm7m-x-x-109">k</span><sub><span 
class="zptmcm7t-x-x-80">0</span></sub> <span 
class="zptmcm7t-x-x-109">+</span><span 
class="zptmcm7m-x-x-109">λ</span><sup><span 
class="zptmcm7m-x-x-80">α</span></sup><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">k</span><sub><span 
class="zptmcm7t-x-x-80">1</span></sub> <span 
class="zptmcm7y-x-x-109">−</span><span 
class="zptmcm7m-x-x-109">k</span><sub><span 
class="zptmcm7t-x-x-80">0</span></sub><span 
class="zptmcm7t-x-x-109">)</span>, where <span 
class="zptmcm7m-x-x-109">k</span><sub><span 
class="zptmcm7t-x-x-80">0</span></sub>, <span 
class="zptmcm7m-x-x-109">k</span><sub><span 
class="zptmcm7m-x-x-80">λ</span></sub>, and <span 
class="zptmcm7m-x-x-109">k</span><sub><span 
class="zptmcm7t-x-x-80">1</span></sub> are the initial, current, and ﬁnal values of the force constant.
     The parameter <span 
class="zptmcm7m-x-x-109">λ </span>evolves linearly from 0 to 1, either smoothly, or in <span 
class="ot1-zi4r-0-x-x-109">targetNumStages </span>equally spaced
     discrete stages, or according to an arbitrary schedule set with <span 
class="ot1-zi4r-0-x-x-109">lambdaSchedule</span>. When the initial value
     of the force constant is zero, an exponent greater than 1.0 distributes the eﬀects of introducing the
     restraint more smoothly over time than a linear dependence, and ensures that there is no singularity in
     the derivative of the restraint free energy with respect to lambda. A value of 4 has been found to give
     good results in some tests.
     </li>
     <li class="itemize"><a 
 id="dx1-78003"></a> <span 
class="ot1-zi4r-0-x-x-120">targetEquilSteps </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Number of steps discarded from TI estimate<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">harmonic</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive integer <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>Deﬁnes the number of steps within each stage that are considered equilibration and
     discarded from the restraint free energy derivative estimate reported reported in the output.
     </li>
     <li class="itemize"><a 
 id="dx1-78004"></a> <span 
class="ot1-zi4r-0-x-x-120">lambdaSchedule </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Schedule of lambda-points for changing force constant<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">harmonic</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>list of real numbers between 0 and 1 <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>If speciﬁed together with targetForceConstant, sets the sequence of discrete <span 
class="zptmcm7m-x-x-109">λ </span>values
     that will be used for diﬀerent stages.
     </li></ul>
                                                                                         
                                                                                         
<!--l. 4146--><p class="noindent" >
</p>
   <h4 class="subsectionHead"><span class="titlemark">7.5    </span> <a 
 id="sec:colvarbias_harmonic_walls"></a>Harmonic wall restraints</h4>
<!--l. 4149--><p class="noindent" >The <span 
class="ot1-zi4r-0-x-x-109">harmonicWalls</span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="cmsy-10x-x-109">{</span><span 
class="ot1-zi4r-0-x-x-109">...</span><span 
class="cmsy-10x-x-109">} </span>bias is closely related to the harmonic bias (see <a 
href="#sec:colvarbias_harmonic">7.4<!--tex4ht:ref: sec:colvarbias_harmonic --></a>), with the following two
diﬀerences: <span 
class="ptmri7t-x-x-109">(i) </span>instead of a center a <span 
class="ptmri7t-x-x-109">lower wall </span>and/or an <span 
class="ptmri7t-x-x-109">upper wall </span>are deﬁned, outside of which the bias
implements a half-harmonic potential;
</p>
   <table 
class="equation"><tr><td><a 
 id="x1-79001r28"></a>
   <center class="math-display" >
<img 
src="colvars-refman-namd39x.png" alt="       (     (      )2
       ||  1k  ξ−ξupper    if ξ &#x003E; ξupper
       {  2     wξ
V (ξ)= ||  0  (      )2  if ξlower ≤ ξ ≥ ξupper
       (  12k  ξ−ξwlower    if ξ &#x003C; ξlower
                 ξ
" class="math-display"  /></center></td><td class="equation-label">(28)</td></tr></table>
<!--l. 4159--><p class="nopar" >
where <span 
class="zptmcm7m-x-x-109">ξ</span><sub><span 
class="zptmcm7t-x-x-80">lower</span></sub> and <span 
class="zptmcm7m-x-x-109">ξ</span><sub><span 
class="zptmcm7t-x-x-80">upper</span></sub> are the lower and upper wall thresholds, respectively; <span 
class="ptmri7t-x-x-109">(ii) </span>because an interval between two
walls is deﬁned, only scalar variables can be used (but any number of variables can be deﬁned, and the wall bias is
intrinsically multi-dimensional).
</p><!--l. 4162--><p class="indent" >   <span 
class="ptmb7t-x-x-109">Note: </span>this bias replaces the keywords <span 
class="ot1-zi4r-0-x-x-109">lowerWall</span>, <span 
class="ot1-zi4r-0-x-x-109">lowerWallConstant</span>, <span 
class="ot1-zi4r-0-x-x-109">upperWall </span>and <span 
class="ot1-zi4r-0-x-x-109">upperWallConstant</span>
deﬁned in the <span 
class="ot1-zi4r-0-x-x-109">colvar </span>context (see <a 
href="#sec:colvar">4<!--tex4ht:ref: sec:colvar --></a>). These keywords are still supported, but will be deprecated in the
future.
</p><!--l. 4165--><p class="indent" >   The <span 
class="ot1-zi4r-0-x-x-109">harmonicWalls </span>bias implements the following options: </p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-79002"></a> <span 
class="ot1-zi4r-0-x-x-120">name: </span>see deﬁnition of <a 
href="#sec:colvarbias"><span 
class="ot1-zi4r-0-x-x-109">name</span></a> in sec. <a 
href="#sec:colvarbias">7<!--tex4ht:ref: sec:colvarbias --></a> (biasing and analysis methods)
     </li>
     <li class="itemize"><a 
 id="dx1-79003"></a> <span 
class="ot1-zi4r-0-x-x-120">colvars: </span>see deﬁnition of <a 
href="#sec:colvarbias"><span 
class="ot1-zi4r-0-x-x-109">colvars</span></a> in sec. <a 
href="#sec:colvarbias">7<!--tex4ht:ref: sec:colvarbias --></a> (biasing and analysis methods)
     </li>
     <li class="itemize"><a 
 id="dx1-79004"></a> <span 
class="ot1-zi4r-0-x-x-120">outputEnergy: </span>see deﬁnition of <a 
href="#sec:colvarbias"><span 
class="ot1-zi4r-0-x-x-109">outputEnergy</span></a> in sec. <a 
href="#sec:colvarbias">7<!--tex4ht:ref: sec:colvarbias --></a> (biasing and analysis methods)
     </li>
     <li class="itemize"><a 
 id="dx1-79005"></a> <span 
class="ot1-zi4r-0-x-x-120">lowerWalls </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Position of the lower wall<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">colvar</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>Space-separated list of decimals <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>Deﬁnes the values <span 
class="zptmcm7m-x-x-109">ξ</span><sub><span 
class="zptmcm7t-x-x-80">lower</span></sub> below which a conﬁning restraint on the colvar is applied to
     each colvar <span 
class="zptmcm7m-x-x-109">ξ </span>.
                                                                                         
                                                                                         
     </li>
     <li class="itemize"><a 
 id="dx1-79006"></a> <span 
class="ot1-zi4r-0-x-x-120">upperWalls </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Position of the lower wall<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">colvar</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>Space-separated list of decimals <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>Deﬁnes the values <span 
class="zptmcm7m-x-x-109">ξ</span><sub><span 
class="zptmcm7t-x-x-80">upper</span></sub> above which a conﬁning restraint on the colvar is applied to
     each colvar <span 
class="zptmcm7m-x-x-109">ξ </span>.
     </li>
     <li class="itemize"><a 
 id="dx1-79007"></a> <span 
class="ot1-zi4r-0-x-x-120">forceConstant: </span>see deﬁnition of <a 
href="#sec:colvarbias_harmonic"><span 
class="ot1-zi4r-0-x-x-109">forceConstant</span></a> in sec. <a 
href="#sec:colvarbias_harmonic">7.4<!--tex4ht:ref: sec:colvarbias_harmonic --></a> (Harmonic restraints)
     </li>
     <li class="itemize"><a 
 id="dx1-79008"></a> <span 
class="ot1-zi4r-0-x-x-120">lowerWallConstant </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Force constant for the lower wall<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">harmonicWalls</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive decimal <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">forceConstant</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>When both sets of walls are deﬁned (lower and upper), this keyword allows setting
     diﬀerent force constants for them. The force constant reported in the output as “<span 
class="zptmcm7m-x-x-109">k</span>”, and used in the
     change of force constant scheme, is the geometric mean of the two.
     </li>
     <li class="itemize"><a 
 id="dx1-79009"></a> <span 
class="ot1-zi4r-0-x-x-120">upperWallConstant: </span>analogous to <span 
class="ot1-zi4r-0-x-x-109">lowerWallConstant</span>
     </li>
     <li class="itemize"><a 
 id="dx1-79010"></a> <span 
class="ot1-zi4r-0-x-x-120">targetForceConstant: </span>see   deﬁnition   of   <a 
href="#sec:colvarbias_harmonic"><span 
class="ot1-zi4r-0-x-x-109">targetForceConstant</span></a> in   sec. <a 
href="#sec:colvarbias_harmonic">7.4<!--tex4ht:ref: sec:colvarbias_harmonic --></a>   (Harmonic
     restraints)
     </li>
     <li class="itemize"><a 
 id="dx1-79011"></a> <span 
class="ot1-zi4r-0-x-x-120">targetNumSteps: </span>see deﬁnition of <a 
href="#sec:colvarbias_harmonic"><span 
class="ot1-zi4r-0-x-x-109">targetNumSteps</span></a> in sec. <a 
href="#sec:colvarbias_harmonic">7.4<!--tex4ht:ref: sec:colvarbias_harmonic --></a> (Harmonic restraints)
     </li>
     <li class="itemize"><a 
 id="dx1-79012"></a>   <span 
class="ot1-zi4r-0-x-x-120">targetForceExponent: </span>see   deﬁnition   of   <a 
href="#sec:colvarbias_harmonic"><span 
class="ot1-zi4r-0-x-x-109">targetForceExponent</span></a> in   sec. <a 
href="#sec:colvarbias_harmonic">7.4<!--tex4ht:ref: sec:colvarbias_harmonic --></a>   (Harmonic
     restraints)
     </li>
     <li class="itemize"><a 
 id="dx1-79013"></a> <span 
class="ot1-zi4r-0-x-x-120">targetEquilSteps: </span>see deﬁnition of <a 
href="#sec:colvarbias_harmonic"><span 
class="ot1-zi4r-0-x-x-109">targetEquilSteps</span></a> in sec. <a 
href="#sec:colvarbias_harmonic">7.4<!--tex4ht:ref: sec:colvarbias_harmonic --></a> (Harmonic restraints)
     </li>
     <li class="itemize"><a 
 id="dx1-79014"></a> <span 
class="ot1-zi4r-0-x-x-120">targetNumStages: </span>see deﬁnition of <a 
href="#sec:colvarbias_harmonic"><span 
class="ot1-zi4r-0-x-x-109">targetNumStages</span></a> in sec. <a 
href="#sec:colvarbias_harmonic">7.4<!--tex4ht:ref: sec:colvarbias_harmonic --></a> (Harmonic restraints)
     </li>
     <li class="itemize"><a 
 id="dx1-79015"></a> <span 
class="ot1-zi4r-0-x-x-120">lambdaSchedule: </span>see deﬁnition of <a 
href="#sec:colvarbias_harmonic"><span 
class="ot1-zi4r-0-x-x-109">lambdaSchedule</span></a> in sec. <a 
href="#sec:colvarbias_harmonic">7.4<!--tex4ht:ref: sec:colvarbias_harmonic --></a> (Harmonic restraints)
     </li></ul>
<!--l. 4215--><p class="noindent" >
</p>
   <h4 class="subsectionHead"><span class="titlemark">7.6    </span> <a 
 id="sec:colvarbias_linear"></a>Linear restraints</h4>
<!--l. 4218--><p class="noindent" >The linear restraint biasing method is used to minimally bias a simulation. There is generally a unique strength of
bias for each CV center, which means you must know the bias force constant speciﬁcally for the center of the CV.
This force constant may be found by using experiment directed simulation described in section <a 
href="#sec:colvarbias_alb">7.7<!--tex4ht:ref: sec:colvarbias_alb --></a>. Please cite
Pitera and Chodera when using <span class="cite">[<a 
href="#XPitera2012">28</a>]</span>.
                                                                                         
                                                                                         
</p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-80001"></a> <span 
class="ot1-zi4r-0-x-x-120">name: </span>see deﬁnition of <a 
href="#sec:colvarbias"><span 
class="ot1-zi4r-0-x-x-109">name</span></a> in sec. <a 
href="#sec:colvarbias">7<!--tex4ht:ref: sec:colvarbias --></a> (biasing and analysis methods)
     </li>
     <li class="itemize"><a 
 id="dx1-80002"></a> <span 
class="ot1-zi4r-0-x-x-120">colvars: </span>see deﬁnition of <a 
href="#sec:colvarbias"><span 
class="ot1-zi4r-0-x-x-109">colvars</span></a> in sec. <a 
href="#sec:colvarbias">7<!--tex4ht:ref: sec:colvarbias --></a> (biasing and analysis methods)
     </li>
     <li class="itemize"><a 
 id="dx1-80003"></a> <span 
class="ot1-zi4r-0-x-x-120">forceConstant </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Scaled force constant (kcal/mol)<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">linear</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive decimal <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">1.0</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>This  deﬁnes  a  scaled  force  constant  for  the  linear  bias.  To  ensure  consistency  for
     multidimensional restraints, it is divided internally by the speciﬁc <span 
class="ot1-zi4r-0-x-x-109">width </span>for each colvar involved
     (which is 1 by default), so that all colvars are eﬀectively dimensionless and of commensurate size.
     </li>
     <li class="itemize"><a 
 id="dx1-80004"></a> <span 
class="ot1-zi4r-0-x-x-120">centers </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Initial linear restraint centers<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">linear</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>space-separated list of colvar values <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>The centers (equilibrium values) of the restraint are entered here. The number of values
     must be the number of requested colvars. Each value is a decimal number if the corresponding colvar
     returns a scalar, a “<span 
class="ot1-zi4r-0-x-x-109">(x, y, z)</span>” triplet if it returns a unit vector or a vector, and a “<span 
class="ot1-zi4r-0-x-x-109">q0, q1, q2, q3)</span>”
     quadruplet if it returns a rotational quaternion. If a colvar has periodicities or symmetries, its closest
     image to the restraint center is considered when calculating the linear potential.</li></ul>
<!--l. 4261--><p class="noindent" >
</p>
   <h4 class="subsectionHead"><span class="titlemark">7.7    </span> <a 
 id="sec:colvarbias_alb"></a>Adaptive Linear Bias/Experiment Directed Simulation</h4>
<!--l. 4264--><p class="noindent" >Experiment directed simulation applies a linear bias with a changing force constant. Please cite White and Voth <span class="cite">[<a 
href="#XWhite2014">29</a>]</span>
when using this feature. As opposed to that reference, the force constant here is scaled by the <span 
class="ot1-zi4r-0-x-x-109">width </span>corresponding
to the biased colvar. In White and Voth, each force constant is scaled by the colvars set center. The bias converges to
a linear bias, after which it will be the minimal possible bias. You may also stop the simulation, take the median of
the force constants (ForceConst) found in the colvars trajectory ﬁle, and then apply a linear bias with that constant.
All the notes about units described in sections <a 
href="#sec:colvarbias_linear">7.6<!--tex4ht:ref: sec:colvarbias_linear --></a> and <a 
href="#sec:colvarbias_harmonic">7.4<!--tex4ht:ref: sec:colvarbias_harmonic --></a> apply here as well. <span 
class="ptmb7t-x-x-109">This is not a valid simulation</span>
<span 
class="ptmb7t-x-x-109">of any particular statistical ensemble and is only an optimization algorithm until the bias has</span>
<span 
class="ptmb7t-x-x-109">converged</span>.
</p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-81001"></a> <span 
class="ot1-zi4r-0-x-x-120">name: </span>see deﬁnition of <a 
href="#sec:colvarbias"><span 
class="ot1-zi4r-0-x-x-109">name</span></a> in sec. <a 
href="#sec:colvarbias">7<!--tex4ht:ref: sec:colvarbias --></a> (biasing and analysis methods)
     </li>
     <li class="itemize"><a 
 id="dx1-81002"></a> <span 
class="ot1-zi4r-0-x-x-120">colvars: </span>see deﬁnition of <a 
href="#sec:colvarbias"><span 
class="ot1-zi4r-0-x-x-109">colvars</span></a> in sec. <a 
href="#sec:colvarbias">7<!--tex4ht:ref: sec:colvarbias --></a> (biasing and analysis methods)
                                                                                         
                                                                                         
     </li>
     <li class="itemize"><a 
 id="dx1-81003"></a> <span 
class="ot1-zi4r-0-x-x-120">centers </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Collective variable centers<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">alb</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>space-separated list of colvar values <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>The desired center (equilibrium values) which will be sought during the adaptive linear
     biasing. The number of values must be the number of requested colvars. Each value is a decimal
     number if the corresponding colvar returns a scalar, a “<span 
class="ot1-zi4r-0-x-x-109">(x, y, z)</span>” triplet if it returns a unit vector
     or a vector, and a “<span 
class="ot1-zi4r-0-x-x-109">q0, q1, q2, q3)</span>” quadruplet if it returns a rotational quaternion. If a colvar has
     periodicities or symmetries, its closest image to the restraint center is considered when calculating the
     linear potential.
     </li>
     <li class="itemize"><a 
 id="dx1-81004"></a> <span 
class="ot1-zi4r-0-x-x-120">updateFrequency </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>The duration of updates<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">alb</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>An integer <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>This is, <span 
class="zptmcm7m-x-x-109">N</span>, the number of simulation steps to use for each update to the bias. This
     determines how long the system requires to equilibrate after a change in force constant (<span 
class="zptmcm7m-x-x-109">N∕</span><span 
class="zptmcm7t-x-x-109">2</span>), how
     long statistics are collected for an iteration (<span 
class="zptmcm7m-x-x-109">N∕</span><span 
class="zptmcm7t-x-x-109">2</span>), and how quickly energy is added to the system
     (at most, <span 
class="zptmcm7m-x-x-109">A∕</span><span 
class="zptmcm7t-x-x-109">2</span><span 
class="zptmcm7m-x-x-109">N</span>, where <span 
class="zptmcm7m-x-x-109">A </span>is the <span 
class="ot1-zi4r-0-x-x-109">forceRange</span>). Until the force constant has converged, the method as
     described is an optimization procedure and not an integration of a particular statistical ensemble. It
     is important that each step should be uncorrelated from the last so that iterations are independent.
     Therefore, <span 
class="zptmcm7m-x-x-109">N </span>should be at least twice the autocorrelation time of the collective variable. The system
     should also be able to dissipate energy as fast as <span 
class="zptmcm7m-x-x-109">N∕</span><span 
class="zptmcm7t-x-x-109">2</span>, which can be done by adjusting thermostat
     parameters. Practically, <span 
class="zptmcm7m-x-x-109">N </span>has been tested successfully at signiﬁcantly shorter than the autocorrelation
     time of the collective variables being biased and still converge correctly.
     </li>
     <li class="itemize"><a 
 id="dx1-81005"></a> <span 
class="ot1-zi4r-0-x-x-120">forceRange </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>The expected range of the force constant in units of energy<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">alb</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>A space-separated list of decimal numbers <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span>3 <span 
class="zptmcm7m-x-x-109">k</span><sub><span 
class="zptmcm7m-x-x-80">b</span></sub><span 
class="zptmcm7m-x-x-109">T </span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>This is largest magnitude of the force constant which one expects. If this parameter is
     too low, the simulation will not converge. If it is too high the simulation will waste time exploring
     values that are too large. A value of 3 <span 
class="zptmcm7m-x-x-109">k</span><sub><span 
class="zptmcm7m-x-x-80">b</span></sub><span 
class="zptmcm7m-x-x-109">T </span>has worked well in the systems presented as a ﬁrst choice.
     This parameter is dynamically adjusted over the course of a simulation. The beneﬁt is that a bad guess
     for the forceRange can be corrected. However, this can lead to large amounts of energy being added
     over time to the system. To prevent this dynamic update, add <span 
class="ot1-zi4r-0-x-x-109">hardForceRange yes </span>as a parameter
     </li>
     <li class="itemize"><a 
 id="dx1-81006"></a> <span 
class="ot1-zi4r-0-x-x-120">rateMax </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>The maximum rate of change of force constant<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">alb</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>A list of space-separated real numbers <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>This optional parameter controls how much energy is added to the system from this bias.
     Tuning this separately from the <span 
class="ot1-zi4r-0-x-x-109">updateFrequency </span>and <span 
class="ot1-zi4r-0-x-x-109">forceRange </span>can allow for large bias changes
     but with a low <span 
class="ot1-zi4r-0-x-x-109">rateMax </span>prevents large energy changes that can lead to instability in the simulation.
     </li></ul>
                                                                                         
                                                                                         
<!--l. 4343--><p class="noindent" >
</p>
   <h4 class="subsectionHead"><span class="titlemark">7.8    </span> <a 
 id="sec:colvarbias_histogram"></a>Multidimensional histograms</h4>
<!--l. 4346--><p class="noindent" >The <span 
class="ot1-zi4r-0-x-x-109">histogram </span>feature is used to record the distribution of a set of collective variables in the form of a
N-dimensional histogram. It functions as a “collective variable bias”, and is invoked by adding a <span 
class="ot1-zi4r-0-x-x-109">histogram </span>block
to the Colvars conﬁguration ﬁle.
</p><!--l. 4351--><p class="indent" >   As with any other biasing and analysis method, when a histogram is applied to an extended-system colvar (<a 
href="#sec:colvar_extended">4.3<!--tex4ht:ref: sec:colvar_extended --></a>),
it accesses the value of the ﬁctitious coordinate rather than that of the “true” colvar. A joint histogram of the “true”
colvar and the ﬁctitious coordinate may be obtained by specifying the colvar name twice in a row in the <span 
class="ot1-zi4r-0-x-x-109">colvars</span>
parameter: the ﬁrst instance will be understood as the “true” colvar, and the second, as the ﬁctitious
coordinate.
</p><!--l. 4359--><p class="indent" >   In addition to the common parameters <span 
class="ot1-zi4r-0-x-x-109">name </span>and <span 
class="ot1-zi4r-0-x-x-109">colvars </span>described above, a <span 
class="ot1-zi4r-0-x-x-109">histogram </span>block may deﬁne the
following parameter:
</p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-82001"></a> <span 
class="ot1-zi4r-0-x-x-120">name: </span>see deﬁnition of <a 
href="#sec:colvarbias"><span 
class="ot1-zi4r-0-x-x-109">name</span></a> in sec. <a 
href="#sec:colvarbias">7<!--tex4ht:ref: sec:colvarbias --></a> (biasing and analysis methods)
     </li>
     <li class="itemize"><a 
 id="dx1-82002"></a> <span 
class="ot1-zi4r-0-x-x-120">colvars: </span>see deﬁnition of <a 
href="#sec:colvarbias"><span 
class="ot1-zi4r-0-x-x-109">colvars</span></a> in sec. <a 
href="#sec:colvarbias">7<!--tex4ht:ref: sec:colvarbias --></a> (biasing and analysis methods)
     </li>
     <li class="itemize"><a 
 id="dx1-82003"></a> <span 
class="ot1-zi4r-0-x-x-120">outputFreq </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Frequency (in timesteps) at which the histogram ﬁles are refreshed<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">histogram</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive integer <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">colvarsRestartFrequency</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>The histogram data are written to ﬁles at the given time interval. A value of 0 disables
     the creation of these ﬁles (<span 
class="ptmb7t-x-x-109">note: </span>all data to continue a simulation are still included in the state ﬁle).
     </li>
     <li class="itemize"><a 
 id="dx1-82004"></a> <span 
class="ot1-zi4r-0-x-x-120">outputFile </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Write the histogram to a ﬁle<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">histogram</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>UNIX ﬁlename <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ptmri7t-x-x-109">outputName</span><span 
class="ot1-zi4r-0-x-x-109">.</span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ot1-zi4r-0-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span><span 
class="ot1-zi4r-0-x-x-109">.dat</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>Name  of  the  ﬁle  containing  histogram  data  (multicolumn  format),  which  is  written
     every <span 
class="ot1-zi4r-0-x-x-109">outputFreq </span>steps. For the special case of 2 variables, Gnuplot may be used to visualize this
     ﬁle.
     </li>
     <li class="itemize"><a 
 id="dx1-82005"></a> <span 
class="ot1-zi4r-0-x-x-120">outputFileDX </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Write the histogram to a ﬁle<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">histogram</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>UNIX ﬁlename <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ptmri7t-x-x-109">outputName</span><span 
class="ot1-zi4r-0-x-x-109">.</span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ot1-zi4r-0-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span><span 
class="ot1-zi4r-0-x-x-109">.dat</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>Name of the ﬁle containing histogram data (OpenDX format), which is written every
     <span 
class="ot1-zi4r-0-x-x-109">outputFreq </span>steps. For the special case of 3 variables, VMD may be used to visualize this ﬁle.
     </li>
     <li class="itemize"><a 
 id="dx1-82006"></a> <span 
class="ot1-zi4r-0-x-x-120">gatherVectorColvars </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Treat vector variables as multiple observations of a scalar variable?<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">histogram</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>UNIX ﬁlename <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">off</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>When this is set to <span 
class="ot1-zi4r-0-x-x-109">on</span>, the components of a multi-dimensional colvar (e.g. one based on
     <span 
class="ot1-zi4r-0-x-x-109">cartesian</span>, <span 
class="ot1-zi4r-0-x-x-109">distancePairs</span>, or a vector of scalar numbers given by <span 
class="ot1-zi4r-0-x-x-109">scriptedFunction</span>) are treated
     as multiple observations of a scalar variable. This results in the histogram being accumulated multiple
     times for each simulation step). When multiple vector variables are included in <span 
class="ot1-zi4r-0-x-x-109">histogram</span>, these
     must have the same length because their components are accumulated together. For example, if <span 
class="zptmcm7m-x-x-109">ξ </span>, <span 
class="zptmcm7m-x-x-109">λ</span>
     and <span 
class="zptmcm7m-x-x-109">τ </span>are three variables of dimensions 5, 5 and 1, respectively, for each iteration 5 triplets <span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">ξ</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub><span 
class="zptmcm7m-x-x-109">,λ</span><sub><span 
class="zptmcm7m-x-x-80">i</span></sub><span 
class="zptmcm7m-x-x-109">,τ</span><span 
class="zptmcm7t-x-x-109">)</span>
     (<span 
class="zptmcm7m-x-x-109">i </span><span 
class="zptmcm7t-x-x-109">= 1</span><span 
class="zptmcm7m-x-x-109">,</span><span 
class="zptmcm7m-x-x-109">…</span><span 
class="zptmcm7t-x-x-109">5</span>) are accumulated into a 3-dimensional histogram.
     </li>
     <li class="itemize"><a 
 id="dx1-82007"></a> <span 
class="ot1-zi4r-0-x-x-120">weights </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Treat vector variables as multiple observations of a scalar variable?<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">histogram</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>list of space-separated decimals <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span>all weights equal to 1 <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>When <span 
class="ot1-zi4r-0-x-x-109">gatherVectorColvars </span>is <span 
class="ot1-zi4r-0-x-x-109">on</span>, the components of each multi-dimensional colvar
     are accumulated with a diﬀerent weight. For example, if <span 
class="zptmcm7m-x-x-109">x </span>and <span 
class="zptmcm7m-x-x-109">y </span>are two distinct <span 
class="ot1-zi4r-0-x-x-109">cartesian </span>variables
     deﬁned on the same group of atoms, the corresponding 2D histogram can be weighted on a per-atom
     basis in the deﬁnition of <span 
class="ot1-zi4r-0-x-x-109">histogram</span>.
     </li></ul>
<!--l. 4427--><p class="noindent" >
</p>
   <h5 class="subsubsectionHead"><span class="titlemark">7.8.1    </span> <a 
 id="x1-830007.8.1"></a>Grid deﬁnition for multidimensional histograms</h5>
<!--l. 4430--><p class="noindent" >Like the ABF and metadynamics biases, <span 
class="ot1-zi4r-0-x-x-109">histogram </span>uses the parameters <span 
class="ot1-zi4r-0-x-x-109">lowerBoundary</span>, <span 
class="ot1-zi4r-0-x-x-109">upperBoundary</span>, and
<span 
class="ot1-zi4r-0-x-x-109">width </span>to deﬁne its grid. These values can be overridden if a conﬁguration block <span 
class="ot1-zi4r-0-x-x-109">histogramGrid </span><span 
class="cmsy-10x-x-109">{ </span><span 
class="ot1-zi4r-0-x-x-109">…</span><span 
class="cmsy-10x-x-109">} </span>is
provided inside the conﬁguration of <span 
class="ot1-zi4r-0-x-x-109">histogram</span>. The options supported inside this conﬁguration block are:
</p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-83001"></a> <span 
class="ot1-zi4r-0-x-x-120">lowerBoundaries </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Lower boundaries of the grid<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">histogramGrid</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>list of space-separated decimals <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>This option deﬁnes the lower boundaries of the grid, overriding any values deﬁned
     by the <span 
class="ot1-zi4r-0-x-x-109">lowerBoundary </span>keyword of each colvar. Note that when <span 
class="ot1-zi4r-0-x-x-109">gatherVectorColvars </span>is <span 
class="ot1-zi4r-0-x-x-109">on</span>, each
     vector variable is automatically treated as a scalar, and a single value should be provided for it.
     </li>
     <li class="itemize"><a 
 id="dx1-83002"></a> <span 
class="ot1-zi4r-0-x-x-120">upperBoundaries: </span>analogous to <span 
class="ot1-zi4r-0-x-x-109">lowerBoundaries</span>
     </li>
     <li class="itemize"><a 
 id="dx1-83003"></a><span 
class="ot1-zi4r-0-x-x-120">widths: </span>analogous to <span 
class="ot1-zi4r-0-x-x-109">lowerBoundaries</span></li></ul>
                                                                                         
                                                                                         
<!--l. 4450--><p class="noindent" >
</p>
   <h4 class="subsectionHead"><span class="titlemark">7.9    </span> <a 
 id="x1-840007.9"></a>Probability distribution-restraints</h4>
<!--l. 4453--><p class="noindent" >The <span 
class="ot1-zi4r-0-x-x-109">histogramRestraint </span>bias implements a continuous potential of many variables (or of a single
high-dimensional variable) aimed at reproducing a one-dimensional statistical distribution that is provided by the
user. The <span 
class="zptmcm7m-x-x-109">M </span>variables <span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">ξ</span><sub><span 
class="zptmcm7t-x-x-80">1</span></sub><span 
class="zptmcm7m-x-x-109">,</span><span 
class="zptmcm7m-x-x-109">…</span><span 
class="zptmcm7m-x-x-109">,ξ</span><sub><span 
class="zptmcm7m-x-x-80">M</span></sub><span 
class="zptmcm7t-x-x-109">) </span>are interpreted as multiple observations of a random variable <span 
class="zptmcm7m-x-x-109">ξ </span>with unknown
probability distribution. The potential is minimized when the histogram <span 
class="zptmcm7m-x-x-109">h</span><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">ξ</span><span 
class="zptmcm7t-x-x-109">)</span>, estimated as a sum of Gaussian
functions centered at <span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">ξ</span><sub><span 
class="zptmcm7t-x-x-80">1</span></sub><span 
class="zptmcm7m-x-x-109">,</span><span 
class="zptmcm7m-x-x-109">…</span><span 
class="zptmcm7m-x-x-109">,ξ</span><sub><span 
class="zptmcm7m-x-x-80">M</span></sub><span 
class="zptmcm7t-x-x-109">)</span>, is equal to the reference histogram <span 
class="zptmcm7m-x-x-109">h</span><sub><span 
class="zptmcm7t-x-x-80">0</span></sub><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">ξ</span><span 
class="zptmcm7t-x-x-109">)</span>:
</p>
   <table 
class="equation"><tr><td><a 
 id="eq:colvarbias_restraint_histogram"></a>
   <center class="math-display" >
<img 
src="colvars-refman-namd40x.png" alt="              1  ∫
V (ξ1,...,ξM )= --k  (h(ξ)− h0(ξ))2 dξ
              2
" class="math-display"  /></center></td><td class="equation-label">(29)</td></tr></table>
<!--l. 4459--><p class="nopar" >
</p>
   <table 
class="equation"><tr><td><a 
 id="eq:colvarbias_restraint_histogram_gaussian"></a>
   <center class="math-display" >
<img 
src="colvars-refman-namd41x.png" alt="                M     (          )
h(ξ )= --√-1----  exp  − (ξ −-ξi)2
       M   2πσ 2∑i=1         2σ 2
" class="math-display"  /></center></td><td class="equation-label">(30)</td></tr></table>
<!--l. 4463--><p class="nopar" >
When used in combination with a <span 
class="ot1-zi4r-0-x-x-109">distancePairs </span>multi-dimensional variable, this bias implements the reﬁnement
algorithm against ESR/DEER experiments published by Shen <span 
class="ptmri7t-x-x-109">et al </span><span class="cite">[<a 
href="#XShen2015">30</a>]</span>.
</p><!--l. 4466--><p class="indent" >   This bias behaves similarly to the <span 
class="ot1-zi4r-0-x-x-109">histogram </span>bias with the <span 
class="ot1-zi4r-0-x-x-109">gatherVectorColvars </span>option, with the important
diﬀerence that <span 
class="ptmri7t-x-x-109">all </span>variables are gathered, resulting in a one-dimensional histogram. Future versions will include
support for multi-dimensional histograms.
</p><!--l. 4469--><p class="indent" >   The list of options is as follows: </p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-84003"></a> <span 
class="ot1-zi4r-0-x-x-120">name: </span>see deﬁnition of <a 
href="#sec:colvarbias"><span 
class="ot1-zi4r-0-x-x-109">name</span></a> in sec. <a 
href="#sec:colvarbias">7<!--tex4ht:ref: sec:colvarbias --></a> (biasing and analysis methods)
                                                                                         
                                                                                         
     </li>
     <li class="itemize"><a 
 id="dx1-84004"></a> <span 
class="ot1-zi4r-0-x-x-120">colvars: </span>see deﬁnition of <a 
href="#sec:colvarbias"><span 
class="ot1-zi4r-0-x-x-109">colvars</span></a> in sec. <a 
href="#sec:colvarbias">7<!--tex4ht:ref: sec:colvarbias --></a> (biasing and analysis methods)
     </li>
     <li class="itemize"><a 
 id="dx1-84005"></a> <span 
class="ot1-zi4r-0-x-x-120">outputEnergy: </span>see deﬁnition of <a 
href="#sec:colvarbias"><span 
class="ot1-zi4r-0-x-x-109">outputEnergy</span></a> in sec. <a 
href="#sec:colvarbias">7<!--tex4ht:ref: sec:colvarbias --></a> (biasing and analysis methods)
     </li>
     <li class="itemize"><a 
 id="dx1-84006"></a> <span 
class="ot1-zi4r-0-x-x-120">lowerBoundary </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Lower boundary of the colvar grid<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">histogramRestraint</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>decimal <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>Deﬁnes the lowest end of the interval where the reference distribution <span 
class="zptmcm7m-x-x-109">h</span><sub><span 
class="zptmcm7t-x-x-80">0</span></sub><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">ξ</span><span 
class="zptmcm7t-x-x-109">) </span>is deﬁned.
     Exactly one value must be provided, because only one-dimensional histograms are supported by the
     current version.
     </li>
     <li class="itemize"><a 
 id="dx1-84007"></a> <span 
class="ot1-zi4r-0-x-x-120">upperBoundary: </span>analogous to <span 
class="ot1-zi4r-0-x-x-109">lowerBoundary</span>
     </li>
     <li class="itemize"><a 
 id="dx1-84008"></a><span 
class="ot1-zi4r-0-x-x-120">width </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Width of the colvar grid<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">histogramRestraint</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive decimal <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>Deﬁnes the spacing of the grid where the reference distribution <span 
class="zptmcm7m-x-x-109">h</span><sub><span 
class="zptmcm7t-x-x-80">0</span></sub><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">ξ</span><span 
class="zptmcm7t-x-x-109">) </span>is deﬁned.
     </li>
     <li class="itemize"><a 
 id="dx1-84009"></a> <span 
class="ot1-zi4r-0-x-x-120">gaussianSigma </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Standard deviation of the approximating Gaussian<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">histogramRestraint</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive decimal <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span>2 <span 
class="zptmcm7y-x-x-109">×</span><span 
class="ot1-zi4r-0-x-x-109">width</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>Deﬁnes the parameter <span 
class="zptmcm7m-x-x-109">σ </span>in eq. <a 
href="#eq:colvarbias_restraint_histogram_gaussian">30<!--tex4ht:ref: eq:colvarbias_restraint_histogram_gaussian --></a>.
     </li>
     <li class="itemize"><a 
 id="dx1-84010"></a> <span 
class="ot1-zi4r-0-x-x-120">forceConstant </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Force constant (kcal/mol)<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">histogramRestraint</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>positive decimal <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">1.0</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>Deﬁnes the parameter <span 
class="zptmcm7m-x-x-109">k </span>in eq. <a 
href="#eq:colvarbias_restraint_histogram">29<!--tex4ht:ref: eq:colvarbias_restraint_histogram --></a>.
     </li>
     <li class="itemize"><a 
 id="dx1-84011"></a> <span 
class="ot1-zi4r-0-x-x-120">refHistogram </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Reference histogram <span 
class="zptmcm7m-x-x-109">h</span><sub><span 
class="zptmcm7t-x-x-80">0</span></sub><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">ξ</span><span 
class="zptmcm7t-x-x-109">)</span><span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">histogramRestraint</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>space-separated list of <span 
class="zptmcm7m-x-x-109">M </span>positive decimals <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>Provides the values of <span 
class="zptmcm7m-x-x-109">h</span><sub><span 
class="zptmcm7t-x-x-80">0</span></sub><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">ξ</span><span 
class="zptmcm7t-x-x-109">) </span>consecutively. The mid-point convention is used, i.e. the
     ﬁrst point that should be included is for <span 
class="zptmcm7m-x-x-109">ξ </span>= <span 
class="ot1-zi4r-0-x-x-109">lowerBoundary</span>+<span 
class="ot1-zi4r-0-x-x-109">width</span>/2. If the integral of <span 
class="zptmcm7m-x-x-109">h</span><sub><span 
class="zptmcm7t-x-x-80">0</span></sub><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">ξ</span><span 
class="zptmcm7t-x-x-109">) </span>is not
     normalized to 1, <span 
class="zptmcm7m-x-x-109">h</span><sub><span 
class="zptmcm7t-x-x-80">0</span></sub><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">ξ</span><span 
class="zptmcm7t-x-x-109">) </span>is rescaled automatically before use.
     </li>
     <li class="itemize"><a 
 id="dx1-84012"></a> <span 
class="ot1-zi4r-0-x-x-120">refHistogramFile </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Reference histogram <span 
class="zptmcm7m-x-x-109">h</span><sub><span 
class="zptmcm7t-x-x-80">0</span></sub><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">ξ</span><span 
class="zptmcm7t-x-x-109">)</span><span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">histogramRestraint</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>UNIX ﬁle name <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>Provides the values of <span 
class="zptmcm7m-x-x-109">h</span><sub><span 
class="zptmcm7t-x-x-80">0</span></sub><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">ξ</span><span 
class="zptmcm7t-x-x-109">) </span>as contents of the corresponding ﬁle (mutually exclusive
     with <span 
class="ot1-zi4r-0-x-x-109">refHistogram</span>). The format is that of a text ﬁle, with each line containing the space-separated
                                                                                         
                                                                                         
     values of <span 
class="zptmcm7m-x-x-109">ξ </span>and <span 
class="zptmcm7m-x-x-109">h</span><sub><span 
class="zptmcm7t-x-x-80">0</span></sub><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">ξ</span><span 
class="zptmcm7t-x-x-109">)</span>. The same numerical conventions as <span 
class="ot1-zi4r-0-x-x-109">refHistogram </span>are used.
     </li>
     <li class="itemize"><a 
 id="dx1-84013"></a> <span 
class="ot1-zi4r-0-x-x-120">writeHistogram </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Periodically write the instantaneous histogram <span 
class="zptmcm7m-x-x-109">h</span><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">ξ</span><span 
class="zptmcm7t-x-x-109">)</span><span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span><span 
class="ot1-zi4r-0-x-x-109">metadynamics</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">off</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>If <span 
class="ot1-zi4r-0-x-x-109">on</span>, the histogram <span 
class="zptmcm7m-x-x-109">h</span><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">ξ</span><span 
class="zptmcm7t-x-x-109">) </span>is written every <span 
class="ot1-zi4r-0-x-x-109">colvarsRestartFrequency </span>steps to a ﬁle
     with the name <span 
class="ptmri7t-x-x-109">outputName</span><span 
class="ot1-zi4r-0-x-x-109">.</span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ot1-zi4r-0-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span><span 
class="ot1-zi4r-0-x-x-109">.hist.dat </span>This is useful to diagnose the convergence of <span 
class="zptmcm7m-x-x-109">h</span><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">ξ</span><span 
class="zptmcm7t-x-x-109">)</span>
     against <span 
class="zptmcm7m-x-x-109">h</span><sub><span 
class="zptmcm7t-x-x-80">0</span></sub><span 
class="zptmcm7t-x-x-109">(</span><span 
class="zptmcm7m-x-x-109">ξ</span><span 
class="zptmcm7t-x-x-109">)</span>.
     </li></ul>
<!--l. 4578--><p class="noindent" >
</p>
   <h4 class="subsectionHead"><span class="titlemark">7.10    </span> <a 
 id="sec:scripted_biases"></a>Scripted biases</h4>
<!--l. 4578--><p class="noindent" >Rather than using the biasing methods described above, it is possible to apply biases provided at run time as a Tcl
script, in the spirit of <span 
class="ot1-zi4r-0-x-x-109">TclForces</span>.
</p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-85001"></a> <span 
class="ot1-zi4r-0-x-x-120">scriptedColvarForces </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Enable custom, scripted forces on colvars <span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span>global <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">off</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:   </span>If this ﬂag is enabled, a Tcl procedure named <span 
class="ot1-zi4r-0-x-x-109">calc</span><span 
class="ot1-zi4r-0-x-x-109">_colvar</span><span 
class="ot1-zi4r-0-x-x-109">_forces </span>accepting one
     parameter should be deﬁned by the user. It is executed at each timestep, with the current step number
     as parameter, between the calculation of colvars and the application of bias forces. This procedure may
     use the scripting interface (see <a 
href="#sec:scripting">8<!--tex4ht:ref: sec:scripting --></a>) to access the values of colvars and apply forces on them, eﬀectively
     deﬁning custom collective variable biases. </li></ul>
<!--l. 4614--><p class="noindent" >
</p>
   <h3 class="sectionHead"><span class="titlemark">8    </span> <a 
 id="sec:scripting"></a>Colvars scripting</h3>
<!--l. 4614--><p class="noindent" >This interface is particularly useful to implement custom biases as scripted colvar forces. See the
<span 
class="ot1-zi4r-0-x-x-109">scriptedColvarForces </span>option in <a 
href="#sec:scripted_biases">7.10<!--tex4ht:ref: sec:scripted_biases --></a>. Note that scripting commands may not be used directly in the
NAMD conﬁguration ﬁle before the ﬁrst <span 
class="ot1-zi4r-0-x-x-109">run </span>or <span 
class="ot1-zi4r-0-x-x-109">minimize </span>statement. They may be used either within the
callback procedures (e.g. <span 
class="ot1-zi4r-0-x-x-109">calc</span><span 
class="ot1-zi4r-0-x-x-109">_colvar</span><span 
class="ot1-zi4r-0-x-x-109">_forces</span>) or in the NAMD conﬁg ﬁle after a <span 
class="ot1-zi4r-0-x-x-109">run </span>or <span 
class="ot1-zi4r-0-x-x-109">minimize</span>
statement.
</p><!--l. 1--><p class="indent" >   Collective variables and biases can be added, queried and deleted through the scripting command <span 
class="ot1-zi4r-0-x-x-109">cv</span>, with the
following syntax: <span 
class="ot1-zi4r-0-x-x-109">cv</span><span 
class="ot1-zi4r-0-x-x-109"> </span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ot1-zi4r-0-x-x-109">subcommand</span><span 
class="zptmcm7m-x-x-109">&#x003E; </span><span 
class="ot1-zi4r-0-x-x-109">[args...]</span>. For example, to query the value of a collective variable named
<span 
class="ot1-zi4r-0-x-x-109">myVar</span>, use the following syntax: <span 
class="ot1-zi4r-0-x-x-109">set value [cv colvar myVar value]</span>. All subcommands of <span 
class="ot1-zi4r-0-x-x-109">cv </span>are documented
below.
                                                                                         
                                                                                         
</p><!--l. 12--><p class="noindent" >
</p>
   <h4 class="subsectionHead"><span class="titlemark">8.1    </span> <a 
 id="x1-870008.1"></a>Managing the colvars module</h4>
     <ul class="itemize1">
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">configfile </span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ptmri7t-x-x-109">ﬁle name</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span>: read conﬁguration from a ﬁle;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">config </span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ptmri7t-x-x-109">string</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span>:   read   conﬁguration   from   the   given   string;   both   <span 
class="ot1-zi4r-0-x-x-109">config </span>and   <span 
class="ot1-zi4r-0-x-x-109">configfile</span>
     subcommands may be invoked multiple times;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">reset</span>: delete all internal conﬁguration of the colvars module;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">version</span>: return the version of the colvars code.</li></ul>
<!--l. 26--><p class="noindent" >
</p>
   <h4 class="subsectionHead"><span class="titlemark">8.2    </span> <a 
 id="x1-880008.2"></a>Input and output</h4>
     <ul class="itemize1">
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">list</span>: return a list of all currently deﬁned variables;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">list biases</span>: return a list of all currently deﬁned biases (i.e. sampling and analysis algorithms);
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">load </span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ptmri7t-x-x-109">ﬁle  name</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span>:  load  a  collective  variables  state  ﬁle,  typically  produced  during  a  previous
     simulation;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">load </span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ptmri7t-x-x-109">preﬁx</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span>: same as above, but without the <span 
class="ot1-zi4r-0-x-x-109">.colvars.state </span>suﬃx;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">save </span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ptmri7t-x-x-109">preﬁx</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span>: save the current state in a ﬁle whose name begins with the given argument; if any of
     the biases have additional output ﬁles deﬁned, those are saved as well using the same preﬁx;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">update</span>: recalculate all colvars and biases based on the current atomic coordinates;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">addenergy</span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ptmri7t-x-x-109">E</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span>: add value <span 
class="ptmri7t-x-x-109">E </span>to the total bias energy; this must be used within <span 
class="ot1-zi4r-0-x-x-109">calc</span><span 
class="ot1-zi4r-0-x-x-109">_colvar</span><span 
class="ot1-zi4r-0-x-x-109">_forces</span>;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">printframe</span>: return a summary of the current frame, in a format equivalent to a line of the collective
     variables trajectory ﬁle;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">printframelabels</span>: return text labels for the columns of <span 
class="ot1-zi4r-0-x-x-109">printframe</span>&#x2019;s output; </li></ul>
                                                                                         
                                                                                         
<!--l. 45--><p class="noindent" >
</p>
   <h4 class="subsectionHead"><span class="titlemark">8.3    </span> <a 
 id="x1-890008.3"></a>Accessing collective variables</h4>
     <ul class="itemize1">
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">colvar </span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ptmri7t-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E; </span><span 
class="ot1-zi4r-0-x-x-109">value</span>: return the current value of colvar <span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ptmri7t-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span>;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">colvar </span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ptmri7t-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E; </span><span 
class="ot1-zi4r-0-x-x-109">update</span>: recalculate colvar <span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ptmri7t-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span>;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">colvar </span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ptmri7t-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E; </span><span 
class="ot1-zi4r-0-x-x-109">type</span>: return the type of colvar <span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ptmri7t-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span>;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">colvar </span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ptmri7t-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E; </span><span 
class="ot1-zi4r-0-x-x-109">delete</span>: delete colvar <span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ptmri7t-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span>;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">colvar </span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ptmri7t-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E; </span><span 
class="ot1-zi4r-0-x-x-109">addforce </span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ptmri7t-x-x-109">F</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span>: apply given force on colvar <span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ptmri7t-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span>;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">colvar </span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ptmri7t-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E; </span><span 
class="ot1-zi4r-0-x-x-109">getappliedforce</span>: get the force currently being applied on colvar <span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ptmri7t-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span>;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">colvar </span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ptmri7t-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E; </span><span 
class="ot1-zi4r-0-x-x-109">gettotalforce</span>: get the total force acting on colvar <span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ptmri7t-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E; </span>at the previous step
     (see <a 
href="#sec:cvc_sys_forces">6.3.3<!--tex4ht:ref: sec:cvc_sys_forces --></a>);
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">colvar </span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ptmri7t-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E; </span><span 
class="ot1-zi4r-0-x-x-109">getconfig</span>: return conﬁg string of colvar <span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ptmri7t-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span>.
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">colvar </span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ptmri7t-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E; </span><span 
class="ot1-zi4r-0-x-x-109">cvcflags </span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ptmri7t-x-x-109">ﬂags</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span>: for a colvar with several cvcs (numbered according to their
     name string order), set which cvcs are enabled or disabled in subsequent evaluations according to a
     list of 0/1 ﬂags (one per cvc).</li></ul>
<!--l. 62--><p class="noindent" >
</p>
   <h4 class="subsectionHead"><span class="titlemark">8.4    </span> <a 
 id="x1-900008.4"></a>Accessing biases</h4>
     <ul class="itemize1">
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">bias </span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ptmri7t-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E; </span><span 
class="ot1-zi4r-0-x-x-109">energy</span>: return the current energy of the bias <span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ptmri7t-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span>;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">bias </span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ptmri7t-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E; </span><span 
class="ot1-zi4r-0-x-x-109">update</span>: recalculate the bias <span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ptmri7t-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span>;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">bias </span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ptmri7t-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E; </span><span 
class="ot1-zi4r-0-x-x-109">delete</span>: delete the bias <span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ptmri7t-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span>;
     </li>
     <li class="itemize"><span 
class="ot1-zi4r-0-x-x-109">bias </span><span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ptmri7t-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E; </span><span 
class="ot1-zi4r-0-x-x-109">getconfig</span>: return conﬁg string of bias <span 
class="zptmcm7m-x-x-109">&#x003C;</span><span 
class="ptmri7t-x-x-109">name</span><span 
class="zptmcm7m-x-x-109">&#x003E;</span>.</li></ul>
                                                                                         
                                                                                         
<!--l. 4614--><p class="noindent" >The following conﬁguration options can modify the behavior of the scripting interface for optimization purposes:
</p>
     <ul class="itemize1">
     <li class="itemize"><a 
 id="dx1-90001"></a> <span 
class="ot1-zi4r-0-x-x-120">scriptingAfterBiases </span><span 
class="zptmcm7y-x-x-109">⟨</span><span style="margin-left:0.3em" class="thinspace"></span>Scripted colvar forces need updated biases<span style="margin-left:0.3em" class="thinspace"></span><span 
class="zptmcm7y-x-x-109">⟩</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Context:  </span>global <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br 
class="newline" /><span 
class="ptmb7t-x-x-109">Default value:  </span><span 
class="ot1-zi4r-0-x-x-109">on</span><br 
class="newline" /><span 
class="ptmb7t-x-x-109">Description:  </span>This ﬂag speciﬁes that the <span 
class="ot1-zi4r-0-x-x-109">calc</span><span 
class="ot1-zi4r-0-x-x-109">_colvar</span><span 
class="ot1-zi4r-0-x-x-109">_forces </span>procedure, when deﬁned, is executed
     only after all biases have been updated. For example, this allows using the energy of a restraint bias,
     or the force applied on a colvar, to calculate additional scripted forces, such as boundary constraints.
     When this ﬂag is set to <span 
class="ot1-zi4r-0-x-x-109">off</span>, it is assumed that only the values of their colvars (but not their energy or
     forces) will be used by <span 
class="ot1-zi4r-0-x-x-109">calc</span><span 
class="ot1-zi4r-0-x-x-109">_colvar</span><span 
class="ot1-zi4r-0-x-x-109">_forces</span>: this can be used to schedule the calculation of scripted
     forces and biases concurrently to increase performance. </li></ul>
<!--l. 121--><p class="indent" >
                                                                                         
                                                                                         
</p><!--l. 124--><p class="indent" >
                                                                                         
                                                                                         
</p>
   <h3 class="likesectionHead"><a 
 id="x1-910008.4"></a>References</h3>
<!--l. 1--><p class="noindent" >
     </p><div class="thebibliography">
     <p class="bibitem" ><span class="biblabel">
  [1]<span class="bibsp">   </span></span><a 
 id="XFiorin2013"></a>G. Fiorin, M. L. Klein, and J. Hénin.   Using collective variables to drive molecular dynamics
     simulations. <span 
class="ptmri7t-x-x-109">Mol. Phys.</span>, 111(22-23):3345–3362, 2013.
     </p>
     <p class="bibitem" ><span class="biblabel">
  [2]<span class="bibsp">   </span></span><a 
 id="XIannuzzi2003"></a>M. Iannuzzi,  A. Laio,  and  M. Parrinello.    Eﬃcient  exploration  of  reactive  potential  energy
     surfaces using car-parrinello molecular dynamics. <span 
class="ptmri7t-x-x-109">Phys. Rev. Lett.</span>, 90(23):238302, 2003.
     </p>
     <p class="bibitem" ><span class="biblabel">
  [3]<span class="bibsp">   </span></span><a 
 id="XCoutsias2004"></a>E A Coutsias, C Seok, and K A Dill.  Using quaternions to calculate RMSD.  <span 
class="ptmri7t-x-x-109">J. Comput. Chem.</span>,
     25(15):1849–1857, 2004.
     </p>
     <p class="bibitem" ><span class="biblabel">
  [4]<span class="bibsp">   </span></span><a 
 id="XMu2005"></a>Yuguang  Mu,  Phuong H.  Nguyen,  and  Gerhard  Stock.    Energy  landscape  of  a  small  peptide
     revealed by dihedral angle principal component analysis. <span 
class="ptmri7t-x-x-109">Proteins</span>, 58(1):45–52, 2005.
     </p>
     <p class="bibitem" ><span class="biblabel">
  [5]<span class="bibsp">   </span></span><a 
 id="XAltis2007"></a>Alexandros Altis, Phuong H. Nguyen, Rainer Hegger, and Gerhard Stock. Dihedral angle principal
     component analysis of molecular dynamics simulations. <span 
class="ptmri7t-x-x-109">J. Chem. Phys.</span>, 126(24):244111, 2007.
     </p>
     <p class="bibitem" ><span class="biblabel">
  [6]<span class="bibsp">   </span></span><a 
 id="XGlykos2006"></a>Nicholas M  Glykos.    Carma:  a  molecular  dynamics  analysis  program.    <span 
class="ptmri7t-x-x-109">J.  Comput.  Chem.</span>,
     27(14):1765–1768, 2006.
     </p>
     <p class="bibitem" ><span class="biblabel">
  [7]<span class="bibsp">   </span></span><a 
 id="XBranduardi2007"></a>Davide Branduardi, Francesco Luigi Gervasio, and Michele Parrinello. From a to b in free energy
     space. <span 
class="ptmri7t-x-x-109">J Chem Phys</span>, 126(5):054103, 2007.
     </p>
     <p class="bibitem" ><span class="biblabel">
  [8]<span class="bibsp">   </span></span><a 
 id="XDarve2008"></a>Eric Darve, David Rodríguez-Gómez, and Andrew Pohorille.  Adaptive biasing force method for
     scalar and vector free energy calculations. <span 
class="ptmri7t-x-x-109">J. Chem. Phys.</span>, 128(14):144120, 2008.
     </p>
     <p class="bibitem" ><span class="biblabel">
  [9]<span class="bibsp">   </span></span><a 
 id="XHenin2004"></a>J. Hénin  and  C. Chipot.     Overcoming  free  energy  barriers  using  unconstrained  molecular
     dynamics simulations. <span 
class="ptmri7t-x-x-109">J. Chem. Phys.</span>, 121:2904–2914, 2004.
                                                                                         
                                                                                         
     </p>
     <p class="bibitem" ><span class="biblabel">
 [10]<span class="bibsp">   </span></span><a 
 id="XHenin2010"></a>J. Hénin,  G. Fiorin,  C. Chipot,  and  M. L.  Klein.     Exploring  multidimensional  free  energy
     landscapes using time-dependent biases on collective variables. <span 
class="ptmri7t-x-x-109">J. Chem. Theory Comput.</span>, 6(1):35–47,
     2010.
     </p>
     <p class="bibitem" ><span class="biblabel">
 [11]<span class="bibsp">   </span></span><a 
 id="XCarter1989"></a>A. Carter, E, G. Ciccotti, J. T. Hynes, and R. Kapral. Constrained reaction coordinate dynamics
     for the simulation of rare events. <span 
class="ptmri7t-x-x-109">Chem. Phys. Lett.</span>, 156:472–477, 1989.
     </p>
     <p class="bibitem" ><span class="biblabel">
 [12]<span class="bibsp">   </span></span><a 
 id="XRuiz-Montero1997"></a>M. J. Ruiz-Montero, D. Frenkel, and J. J. Brey.  Eﬃcient schemes to compute diﬀusive barrier
     crossing rates. <span 
class="ptmri7t-x-x-109">Mol. Phys.</span>, 90:925–941, 1997.
     </p>
     <p class="bibitem" ><span class="biblabel">
 [13]<span class="bibsp">   </span></span><a 
 id="XdenOtter2000"></a>W. K. den Otter.   Thermodynamic integration of the free energy along a reaction coordinate in
     cartesian coordinates. <span 
class="ptmri7t-x-x-109">J. Chem. Phys.</span>, 112:7283–7292, 2000.
     </p>
     <p class="bibitem" ><span class="biblabel">
 [14]<span class="bibsp">   </span></span><a 
 id="XCiccotti2005"></a>Giovanni Ciccotti, Raymond Kapral, and Eric Vanden-Eijnden.   Blue moon sampling, vectorial
     reaction coordinates, and unbiased constrained dynamics. <span 
class="ptmri7t-x-x-109">ChemPhysChem</span>, 6(9):1809–1814, 2005.
     </p>
     <p class="bibitem" ><span class="biblabel">
 [15]<span class="bibsp">   </span></span><a 
 id="XMinoukadeh2010"></a>K. Minoukadeh,   C. Chipot,   and   T. Lelièvre.       Potential   of   mean   force   calculations:   A
     multiple-walker adaptive biasing force approach. <span 
class="ptmri7t-x-x-109">J. Chem. Theor. Comput.</span>, 6:1008–1017, 2010.
     </p>
     <p class="bibitem" ><span class="biblabel">
 [16]<span class="bibsp">   </span></span><a 
 id="XComer2014c"></a>J. Comer,  J. Phillips,  K. Schulten,  and  C. Chipot.   Multiple-walker  strategies  for  free-energy
     calculations  in  namd:  Shared  adaptive  biasing  force  and  walker  selection  rules.    <span 
class="ptmri7t-x-x-109">J.  Chem.  Theor.</span>
     <span 
class="ptmri7t-x-x-109">Comput.</span>, 10:5276–5285, 2014.
     </p>
     <p class="bibitem" ><span class="biblabel">
 [17]<span class="bibsp">   </span></span><a 
 id="XLesage2017"></a>Adrien Lesage, Tony Lelièvre, Gabriel Stoltz, and Jérôme Hénin.  Smoothed biasing forces yield
     unbiased free energies with the extended-system adaptive biasing force method.   <span 
class="ptmri7t-x-x-109">J. Phys. Chem. B</span>,
     121(15):3676–3685, 2017.
     </p>
     <p class="bibitem" ><span class="biblabel">
 [18]<span class="bibsp">   </span></span><a 
 id="XFu2016"></a>H. Fu, X. Shao, C. Chipot, and W. Cai. Extended adaptive biasing force algorithm. an on–the–ﬂy
     implementation for accurate free–energy calculations. <span 
class="ptmri7t-x-x-109">J. Chem. Theory Comput.</span>, 2016.
                                                                                         
                                                                                         
     </p>
     <p class="bibitem" ><span class="biblabel">
 [19]<span class="bibsp">   </span></span><a 
 id="XZheng2012"></a>L. Zheng   and   W. Yang.         Practically   eﬃcient   and   robust   free   energy   calculations:
     Double-integration orthogonal space tempering. <span 
class="ptmri7t-x-x-109">J. Chem. Theor. Compt.</span>, 8:810–823, 2012.
     </p>
     <p class="bibitem" ><span class="biblabel">
 [20]<span class="bibsp">   </span></span><a 
 id="XKastner2005"></a>J. Kastner  and  W. Thiel.   Bridging  the  gap  between  thermodynamic  integration  and  umbrella
     sampling provides a novel analysis method: “umbrella integration”.  <span 
class="ptmri7t-x-x-109">J. Chem. Phys.</span>, 123(14):144104,
     2005.
     </p>
     <p class="bibitem" ><span class="biblabel">
 [21]<span class="bibsp">   </span></span><a 
 id="XLaio2002"></a>A. Laio  and  M. Parrinello.     Escaping  free-energy  minima.     <span 
class="ptmri7t-x-x-109">Proc.  Natl.  Acad.  Sci.  USA</span>,
     99(20):12562–12566, 2002.
     </p>
     <p class="bibitem" ><span class="biblabel">
 [22]<span class="bibsp">   </span></span><a 
 id="XGrubmuller1995"></a>Helmut   Grubmüller.       Predicting   slow   structural   transitions   in   macromolecular   systems:
     Conformational ﬂooding. <span 
class="ptmri7t-x-x-109">Phys. Rev. E</span>, 52(3):2893–2906, Sep 1995.
     </p>
     <p class="bibitem" ><span class="biblabel">
 [23]<span class="bibsp">   </span></span><a 
 id="XHuber1994"></a>T. Huber,  A. E.  Torda,  and  W.F.  van  Gunsteren.    Local  elevation  -  A  method  for  improving
     the searching properties of molecular-dynamics simulation.   <span 
class="ptmri7t-x-x-109">Journal of Computer-Aided Molecular</span>
     <span 
class="ptmri7t-x-x-109">Design</span>, 8(6):695–708, DEC 1994.
     </p>
     <p class="bibitem" ><span class="biblabel">
 [24]<span class="bibsp">   </span></span><a 
 id="XBussi2006"></a>G. Bussi,   A. Laio,   and   M. Parrinello.       Equilibrium   free   energies   from   nonequilibrium
     metadynamics. <span 
class="ptmri7t-x-x-109">Phys. Rev. Lett.</span>, 96(9):090601, 2006.
     </p>
     <p class="bibitem" ><span class="biblabel">
 [25]<span class="bibsp">   </span></span><a 
 id="XBarducci2008"></a>Alessandro Barducci, Giovanni Bussi, and Michele Parrinello.  Well-tempered metadynamics: A
     smoothly converging and tunable free-energy method. <span 
class="ptmri7t-x-x-109">Phys. Rev. Lett.</span>, 100:020603, 2008.
     </p>
     <p class="bibitem" ><span class="biblabel">
 [26]<span class="bibsp">   </span></span><a 
 id="XRaiteri2005"></a>P. Raiteri, A. Laio, F. L. Gervasio, C. Micheletti, and M. Parrinello.  Eﬃcient reconstruction of
     complex free energy landscapes by multiple walkers metadynamics. <span 
class="ptmri7t-x-x-109">J. Phys. Chem. B</span>, 110(8):3533–9,
     2005.
     </p>
     <p class="bibitem" ><span class="biblabel">
 [27]<span class="bibsp">   </span></span><a 
 id="XDeng2009"></a>Yuqing Deng and Benoît Roux.  Computations of standard binding free energies with molecular
     dynamics simulations. <span 
class="ptmri7t-x-x-109">J. Phys. Chem. B</span>, 113(8):2234–2246, 2009.
                                                                                         
                                                                                         
     </p>
     <p class="bibitem" ><span class="biblabel">
 [28]<span class="bibsp">   </span></span><a 
 id="XPitera2012"></a>Jed W. Pitera and John D. Chodera.  On the use of experimental observations to bias simulated
     ensembles. <span 
class="ptmri7t-x-x-109">J. Chem. Theory Comput.</span>, 8:3445–3451, 2012.
     </p>
     <p class="bibitem" ><span class="biblabel">
 [29]<span class="bibsp">   </span></span><a 
 id="XWhite2014"></a>A. D. White and G. A. Voth.  Eﬃcient and minimal method to bias molecular simulations with
     experimental data. <span 
class="ptmri7t-x-x-109">J. Chem. Theory Comput.</span>, ASAP, 2014.
     </p>
     <p class="bibitem" ><span class="biblabel">
 [30]<span class="bibsp">   </span></span><a 
 id="XShen2015"></a>Rong  Shen,  Wei  Han,  Giacomo  Fiorin,  Shahidul M  Islam,  Klaus  Schulten,  and  Benoît  Roux.
     Structural reﬁnement of proteins by restrained molecular dynamics simulations with non-interacting
     molecular fragments. <span 
class="ptmri7t-x-x-109">PLoS Comput. Biol.</span>, 11(10):e1004368, 2015.
</p>
     </div>
    
</body></html> 

                                                                                         


