<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
  "http://www.w3.org/TR/html4/loose.dtd">
<html >
<head><title></title>
<meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1">
<meta name="generator" content="TeX4ht (http://www.cse.ohio-state.edu/~gurari/TeX4ht/)">
<meta name="originator" content="TeX4ht (http://www.cse.ohio-state.edu/~gurari/TeX4ht/)">
<!-- html -->
<meta name="src" content="colvars-refman-lammps.tex">
<meta name="date" content="2015-09-03 20:17:00">
<link rel="stylesheet" type="text/css" href="colvars-refman-lammps.css">
</head><body
>
<!--l. 37--><p class="noindent" >


<!--l. 39--><p class="noindent" ><span
class="colorbox" id="colorbox1"> <div class="minipage"><span style="margin-left:0.3em" class="thinspace"></span><br
class="newline" />
<!--l. 50--><p class="noindent" ><span
class="phvb7t-x-x-248">COLLECTIVE VARIABLES MODULE</span><br
class="newline" />
<!--l. 50--><p class="noindent" ><span
class="phvb7t-x-x-248">Reference manual for LAMMPS</span><!--l. 50--><p class="noindent" >
<span
class="phvr7t-x-x-144">September 3, 2015</span></div></span>
<!--l. 53--><p class="indent" >   <img
src="cover.jpg" alt="PIC"
> <span
class="phvr7t-x-x-120">Giacomo Fiorin, J</span><span
class="phvr7t-x-x-120">ér</span><span
class="phvr7t-x-x-120">ôme H</span><span
class="phvr7t-x-x-120">énin, Axel Kohlmeyer</span>


<!--l. 58--><p class="noindent" >


   <h3 class="likesectionHead"><a
 id="x1-1000"></a>Contents</h3>
   <div class="tableofcontents">
   <span class="sectionToc" >1 <a
href="#x1-20001" id="QQ2-1-2">Introduction</a></span>
<br />   <span class="sectionToc" >2 <a
href="#x1-30002" id="QQ2-1-4">General parameters and input/output files</a></span>
<br />   &#x00A0;<span class="subsectionToc" >2.1 <a
href="#x1-40002.1" id="QQ2-1-5">LAMMPS keywords</a></span>
<br />   &#x00A0;<span class="subsectionToc" >2.2 <a
href="#x1-50002.2" id="QQ2-1-6">Configuration syntax for the collective variables module</a></span>
<br />   &#x00A0;<span class="subsectionToc" >2.3 <a
href="#x1-60002.3" id="QQ2-1-7">Input state file (optional)</a></span>
<br />   &#x00A0;<span class="subsectionToc" >2.4 <a
href="#x1-70002.4" id="QQ2-1-8">Output files</a></span>
<br />   <span class="sectionToc" >3 <a
href="#x1-80003" id="QQ2-1-9">Defining collective variables and their properties</a></span>
<br />   &#x00A0;<span class="subsectionToc" >3.1 <a
href="#x1-90003.1" id="QQ2-1-10">General options for a collective variable</a></span>
<br />   &#x00A0;<span class="subsectionToc" >3.2 <a
href="#x1-100003.2" id="QQ2-1-11">Artificial boundary potentials (walls)</a></span>
<br />   &#x00A0;<span class="subsectionToc" >3.3 <a
href="#x1-110003.3" id="QQ2-1-12">Trajectory output</a></span>
<br />   &#x00A0;<span class="subsectionToc" >3.4 <a
href="#x1-120003.4" id="QQ2-1-13">Extended Lagrangian.</a></span>
<br />   &#x00A0;<span class="subsectionToc" >3.5 <a
href="#x1-130003.5" id="QQ2-1-14">Statistical analysis of collective variables</a></span>
<br />   <span class="sectionToc" >4 <a
href="#x1-140004" id="QQ2-1-15">Selecting atoms for colvars: defining atom groups</a></span>
<br />   &#x00A0;<span class="subsectionToc" >4.1 <a
href="#x1-150004.1" id="QQ2-1-16">Selection keywords</a></span>
<br />   &#x00A0;<span class="subsectionToc" >4.2 <a
href="#x1-160004.2" id="QQ2-1-17">Moving frame of reference.</a></span>
<br />   &#x00A0;<span class="subsectionToc" >4.3 <a
href="#x1-170004.3" id="QQ2-1-18">Treatment of periodic boundary conditions.</a></span>
<br />   &#x00A0;<span class="subsectionToc" >4.4 <a
href="#x1-180004.4" id="QQ2-1-19">Computational cost of colvars based on group size.</a></span>
<br />   <span class="sectionToc" >5 <a
href="#x1-190005" id="QQ2-1-20">Collective variable components (basis functions)</a></span>
<br />   &#x00A0;<span class="subsectionToc" >5.1 <a
href="#x1-200005.1" id="QQ2-1-21">List of available colvar components</a></span>
<br />   &#x00A0;&#x00A0;<span class="subsubsectionToc" >5.1.1 <a
href="#x1-210005.1.1" id="QQ2-1-22"><span
class="ot1-zi4r-0-x-x-109">distance</span>: center-of-mass distance between two groups.</a></span>
<br />   &#x00A0;&#x00A0;<span class="subsubsectionToc" >5.1.2 <a
href="#x1-220005.1.2" id="QQ2-1-23"><span
class="ot1-zi4r-0-x-x-109">distanceZ</span>: projection of a distance vector on an axis.</a></span>
<br />   &#x00A0;&#x00A0;<span class="subsubsectionToc" >5.1.3 <a
href="#x1-230005.1.3" id="QQ2-1-24"><span
class="ot1-zi4r-0-x-x-109">distanceXY</span>: modulus of the projection of a distance vector on a plane.</a></span>
<br />   &#x00A0;&#x00A0;<span class="subsubsectionToc" >5.1.4 <a
href="#x1-240005.1.4" id="QQ2-1-25"><span
class="ot1-zi4r-0-x-x-109">distanceVec</span>: distance vector between two groups.</a></span>
<br />   &#x00A0;&#x00A0;<span class="subsubsectionToc" >5.1.5 <a
href="#x1-250005.1.5" id="QQ2-1-26"><span
class="ot1-zi4r-0-x-x-109">distanceDir</span>: distance unit vector between two groups.</a></span>
<br />   &#x00A0;&#x00A0;<span class="subsubsectionToc" >5.1.6 <a
href="#x1-260005.1.6" id="QQ2-1-27"><span
class="ot1-zi4r-0-x-x-109">distanceInv</span>: mean distance between two groups of atoms.</a></span>
<br />   &#x00A0;&#x00A0;<span class="subsubsectionToc" >5.1.7 <a
href="#x1-270005.1.7" id="QQ2-1-28"><span
class="ot1-zi4r-0-x-x-109">cartesian</span>: vector of atomic Cartesian coordinates.</a></span>
<br />   &#x00A0;&#x00A0;<span class="subsubsectionToc" >5.1.8 <a
href="#x1-280005.1.8" id="QQ2-1-29"><span
class="ot1-zi4r-0-x-x-109">angle</span>: angle between three groups.</a></span>
<br />   &#x00A0;&#x00A0;<span class="subsubsectionToc" >5.1.9 <a
href="#x1-290005.1.9" id="QQ2-1-30"><span
class="ot1-zi4r-0-x-x-109">dihedral</span>: torsional angle between four groups.</a></span>
<br />   &#x00A0;&#x00A0;<span class="subsubsectionToc" >5.1.10 <a
href="#x1-300005.1.10" id="QQ2-1-31"><span
class="ot1-zi4r-0-x-x-109">coordNum</span>: coordination number between two groups.</a></span>
<br />   &#x00A0;&#x00A0;<span class="subsubsectionToc" >5.1.11 <a
href="#x1-310005.1.11" id="QQ2-1-32"><span
class="ot1-zi4r-0-x-x-109">selfCoordNum</span>: coordination number between atoms within a group.</a></span>
<br />   &#x00A0;&#x00A0;<span class="subsubsectionToc" >5.1.12 <a
href="#x1-320005.1.12" id="QQ2-1-33"><span
class="ot1-zi4r-0-x-x-109">hBond</span>: hydrogen bond between two atoms.</a></span>
<br />   &#x00A0;&#x00A0;<span class="subsubsectionToc" >5.1.13 <a
href="#x1-330005.1.13" id="QQ2-1-34"><span
class="ot1-zi4r-0-x-x-109">rmsd</span>: root mean square displacement (RMSD) from reference positions.</a></span>
<br />   &#x00A0;&#x00A0;<span class="subsubsectionToc" >5.1.14 <a
href="#x1-340005.1.14" id="QQ2-1-35">Advanced usage of the <span
class="ot1-zi4r-0-x-x-109">rmsd </span>component.</a></span>
<br />   &#x00A0;&#x00A0;<span class="subsubsectionToc" >5.1.15 <a
href="#x1-350005.1.15" id="QQ2-1-36"><span
class="ot1-zi4r-0-x-x-109">eigenvector</span>: projection of the atomic coordinates on a vector.</a></span>
<br />   &#x00A0;&#x00A0;<span class="subsubsectionToc" >5.1.16 <a
href="#x1-360005.1.16" id="QQ2-1-37"><span
class="ot1-zi4r-0-x-x-109">gyration</span>: radius of gyration of a group of atoms.</a></span>
<br />   &#x00A0;&#x00A0;<span class="subsubsectionToc" >5.1.17 <a
href="#x1-370005.1.17" id="QQ2-1-38"><span
class="ot1-zi4r-0-x-x-109">inertia</span>: total moment of inertia of a group of atoms.</a></span>
<br />   &#x00A0;&#x00A0;<span class="subsubsectionToc" >5.1.18 <a
href="#x1-380005.1.18" id="QQ2-1-39"><span
class="ot1-zi4r-0-x-x-109">inertiaZ</span>: total moment of inertia of a group of atoms around a chosen axis.</a></span>
<br />   &#x00A0;&#x00A0;<span class="subsubsectionToc" >5.1.19 <a
href="#x1-390005.1.19" id="QQ2-1-40"><span
class="ot1-zi4r-0-x-x-109">orientation</span>: orientation from reference coordinates.</a></span>
<br />   &#x00A0;&#x00A0;<span class="subsubsectionToc" >5.1.20 <a
href="#x1-400005.1.20" id="QQ2-1-41"><span
class="ot1-zi4r-0-x-x-109">orientationAngle</span>: angle of rotation from reference coordinates.</a></span>
<br />   &#x00A0;&#x00A0;<span class="subsubsectionToc" >5.1.21 <a
href="#x1-410005.1.21" id="QQ2-1-42"><span
class="ot1-zi4r-0-x-x-109">orientationProj</span>: cosine of the angle of rotation from reference coordinates.</a></span>
<br />   &#x00A0;&#x00A0;<span class="subsubsectionToc" >5.1.22 <a
href="#x1-420005.1.22" id="QQ2-1-43"><span
class="ot1-zi4r-0-x-x-109">spinAngle</span>: angle of rotation around a given axis.</a></span>
<br />   &#x00A0;&#x00A0;<span class="subsubsectionToc" >5.1.23 <a
href="#x1-430005.1.23" id="QQ2-1-44"><span
class="ot1-zi4r-0-x-x-109">tilt</span>: cosine of the rotation orthogonal to a given axis.</a></span>
<br />   &#x00A0;<span class="subsectionToc" >5.2 <a
href="#x1-440005.2" id="QQ2-1-45">Advanced usage and special considerations</a></span>
<br />   &#x00A0;&#x00A0;<span class="subsubsectionToc" >5.2.1 <a
href="#x1-450005.2.1" id="QQ2-1-46">Periodic components.</a></span>
<br />   &#x00A0;&#x00A0;<span class="subsubsectionToc" >5.2.2 <a
href="#x1-460005.2.2" id="QQ2-1-47">Non-scalar components.</a></span>


<br />   &#x00A0;&#x00A0;<span class="subsubsectionToc" >5.2.3 <a
href="#x1-470005.2.3" id="QQ2-1-48">Calculating system forces.</a></span>
<br />   &#x00A0;<span class="subsectionToc" >5.3 <a
href="#x1-480005.3" id="QQ2-1-49">Linear and polynomial combinations of components</a></span>
<br />   <span class="sectionToc" >6 <a
href="#x1-490006" id="QQ2-1-50">Biasing and analysis methods</a></span>
<br />   &#x00A0;<span class="subsectionToc" >6.1 <a
href="#x1-500006.1" id="QQ2-1-51">Adaptive Biasing Force</a></span>
<br />   &#x00A0;&#x00A0;<span class="subsubsectionToc" >6.1.1 <a
href="#x1-510006.1.1" id="QQ2-1-52">ABF requirements on collective variables</a></span>
<br />   &#x00A0;&#x00A0;<span class="subsubsectionToc" >6.1.2 <a
href="#x1-520006.1.2" id="QQ2-1-53">Parameters for ABF</a></span>
<br />   &#x00A0;&#x00A0;<span class="subsubsectionToc" >6.1.3 <a
href="#x1-530006.1.3" id="QQ2-1-54">Output files</a></span>
<br />   &#x00A0;&#x00A0;<span class="subsubsectionToc" >6.1.4 <a
href="#x1-540006.1.4" id="QQ2-1-55">Post-processing: reconstructing a multidimensional free energy surface</a></span>
<br />   &#x00A0;<span class="subsectionToc" >6.2 <a
href="#x1-550006.2" id="QQ2-1-56">Metadynamics</a></span>
<br />   &#x00A0;&#x00A0;<span class="subsubsectionToc" >6.2.1 <a
href="#x1-560006.2.1" id="QQ2-1-57">Output files</a></span>
<br />   &#x00A0;&#x00A0;<span class="subsubsectionToc" >6.2.2 <a
href="#x1-570006.2.2" id="QQ2-1-58">Performance tuning</a></span>
<br />   &#x00A0;&#x00A0;<span class="subsubsectionToc" >6.2.3 <a
href="#x1-580006.2.3" id="QQ2-1-59">Well-tempered metadynamics</a></span>
<br />   &#x00A0;&#x00A0;<span class="subsubsectionToc" >6.2.4 <a
href="#x1-590006.2.4" id="QQ2-1-60">Multiple-replicas metadynamics</a></span>
<br />   &#x00A0;&#x00A0;<span class="subsubsectionToc" >6.2.5 <a
href="#x1-600006.2.5" id="QQ2-1-61">Compatibility and post-processing</a></span>
<br />   &#x00A0;<span class="subsectionToc" >6.3 <a
href="#x1-610006.3" id="QQ2-1-62">Harmonic restraints</a></span>
<br />   &#x00A0;&#x00A0;<span class="subsubsectionToc" >6.3.1 <a
href="#x1-620006.3.1" id="QQ2-1-63">Moving restraints: steered molecular dynamics</a></span>
<br />   &#x00A0;&#x00A0;<span class="subsubsectionToc" >6.3.2 <a
href="#x1-630006.3.2" id="QQ2-1-64">Moving restraints: umbrella sampling</a></span>
<br />   &#x00A0;&#x00A0;<span class="subsubsectionToc" >6.3.3 <a
href="#x1-640006.3.3" id="QQ2-1-65">Changing force constant</a></span>
<br />   &#x00A0;<span class="subsectionToc" >6.4 <a
href="#x1-650006.4" id="QQ2-1-66">Linear restraints</a></span>
<br />   &#x00A0;<span class="subsectionToc" >6.5 <a
href="#x1-660006.5" id="QQ2-1-67">Adaptive Linear Bias/Experiment Directed Simulation</a></span>
<br />   &#x00A0;<span class="subsectionToc" >6.6 <a
href="#x1-670006.6" id="QQ2-1-68">Multidimensional histograms</a></span>
   </div>


   <h3 class="sectionHead"><span class="titlemark">1    </span> <a
 id="x1-20001"></a>Introduction</h3>
<!--l. 12--><p class="noindent" >In today&#8217;s molecular dynamics simulations, it is often useful to reduce the large number of degrees of freedom of a
physical system into few parameters whose statistical distributions can be analyzed individually, or used to
define biasing potentials to alter the dynamics of the system in a controlled manner. These have been
called &#8216;order parameters&#8217;, &#8216;collective variables&#8217;, &#8216;(surrogate) reaction coordinates&#8217;, and many other
terms. Here we use primarily the term &#8216;collective variable&#8217; (shortened to <span
class="ptmri7t-x-x-109">colvar</span>), which indicates any
differentiable function of atomic Cartesian coordinates, <span
class="zptmcm7m-x-x-109">x</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub>, with <span
class="zptmcm7m-x-x-109">i </span>between <span
class="zptmcm7t-x-x-109">1 </span>and <span
class="zptmcm7m-x-x-109">N</span>, the total number of
atoms:
   <table
class="equation"><tr><td><a
 id="x1-2001r1"></a>
   <center class="math-display" >
<img
src="colvars-refman-lammps0x.png" alt="&#x03BE;(t)= &#x03BE; (xi(t),xj(t),xk(t),...),1&#x2264; i,j,k...&#x2264; N
" class="math-display" ></center></td><td class="equation-label">(1)</td></tr></table>
<!--l. 20--><p class="nopar" >
This manual documents the collective variables module (colvars), a portable software that interfaces multiple MD
simulation programs, with a focus on flexibility, robustness and high performance. The module is
designed to perform multiple tasks concurrently during or after a simulation, the most common of which
are:
     <ul class="itemize1">
     <li class="itemize">apply restraints or biasing potentials to multiple colvars, tailored on the system by choosing from a
     wide set of basis functions, without limitations on their number or on the number of atoms involved;
     </li>
     <li class="itemize">calculate potentials of mean force (PMFs) along any set of colvars, using different enhanced sampling
     methods, such as Adaptive Biasing Force (ABF), metadynamics, steered MD and umbrella sampling;
     variants of these methods that make use of an ensemble of replicas are supported as well;
     </li>
     <li class="itemize">calculate  statistical  properties  of  the  colvars,  such  as  running  averages  and  standard  deviations,
     correlation functions of pairs of colvars, and multidimensional histograms: this can be done either at
     run-time without the need to save very large trajectory files, or after a simulation has been completed
     using VMD and the <span
class="ot1-zi4r-0-x-x-109">cv </span>command.
     </li></ul>
<!--l. 41--><p class="indent" >   To briefly illustrate the flexibility of the colvars module, Figure&#x00A0;<a
href="#x1-20021">1<!--tex4ht:ref: fig:colvars_diagram --></a> shows an example of a non-trivial
configuration (the corresponding input can be found in <a
href="#x1-50002.2">2.2<!--tex4ht:ref: sec:colvars_config --></a>).
<!--l. 48--><p class="indent" >   <hr class="figure"><div class="figure"
>


<a
 id="x1-20021"></a>


<!--l. 52--><p class="noindent" ><img
src="colvars-refman-lammps1x.png" alt="PIC" class="graphics" width="341.43306pt" height="263.83302pt" ><!--tex4ht:graphics
name="colvars-refman-lammps1x.png" src="colvars_diagram.eps"
-->
<br /> <div class="caption"
><span class="id">Figure&#x00A0;1: </span><span
class="content">Graphical representation of a collective variables configuration <span
class="ptmb7t-x-x-109">(note: </span><span
class="ptmri7t-x-x-109">currently, the </span><span
class="zptmcm7m-x-x-109">&#x03B1;</span><span
class="ptmri7t-x-x-109">-helical</span>
<span
class="ptmri7t-x-x-109">content colvar is unavailable in LAMMPS)</span>. The colvar called &#8220;<span
class="zptmcm7m-x-x-109">d</span>&#8221; is defined as the difference between two
distances: the first distance (<span
class="zptmcm7m-x-x-109">d</span><sub><span
class="zptmcm7t-x-x-80">1</span></sub>) is taken between the center of mass of atoms 1 and 2 and that of atoms 3
to 5, the second (<span
class="zptmcm7m-x-x-109">d</span><sub><span
class="zptmcm7t-x-x-80">2</span></sub>) between atom 7 and the center of mass of atoms 8 to 10. The difference <span
class="zptmcm7m-x-x-109">d </span><span
class="zptmcm7t-x-x-109">= </span><span
class="zptmcm7m-x-x-109">d</span><sub><span
class="zptmcm7t-x-x-80">1</span></sub> <span
class="zptmcm7y-x-x-109">-</span><span
class="zptmcm7m-x-x-109">d</span><sub><span
class="zptmcm7t-x-x-80">2</span></sub>
is obtained by multiplying the two by a coefficient <span
class="zptmcm7m-x-x-109">C </span><span
class="zptmcm7t-x-x-109">= +1 </span>or <span
class="zptmcm7m-x-x-109">C </span><span
class="zptmcm7t-x-x-109">= </span><span
class="zptmcm7y-x-x-109">-</span><span
class="zptmcm7t-x-x-109">1</span>, respectively. The colvar called &#8220;<span
class="zptmcm7m-x-x-109">c</span>&#8221;
is the coordination number calculated between atoms 1 to 10 and atoms 11 to 20. A harmonic restraint is
applied to both <span
class="zptmcm7m-x-x-109">d </span>and <span
class="zptmcm7m-x-x-109">c</span>: to allow using the same force constant <span
class="zptmcm7m-x-x-109">K</span>, both <span
class="zptmcm7m-x-x-109">d </span>and <span
class="zptmcm7m-x-x-109">c </span>are scaled by their respective
fluctuation widths <span
class="zptmcm7m-x-x-109">w</span><sub><span
class="zptmcm7m-x-x-80">d</span></sub> and <span
class="zptmcm7m-x-x-109">w</span><sub><span
class="zptmcm7m-x-x-80">c</span></sub>.   The values of &#8220;<span
class="zptmcm7m-x-x-109">c</span>&#8221; are also recorded throughout the simulation as a joint
2-dimensional histogram. </span></div><!--tex4ht:label?: x1-20021 -->


<!--l. 61--><p class="indent" >   </div><hr class="endfigure">
<!--l. 63--><p class="indent" >   Detailed explanations of the design of the colvars module are provided in reference&#x00A0;<span class="cite">[<a
href="#XFiorin2013">1</a>]</span>. Please cite this
reference whenever publishing work that makes use of this module.
   <h3 class="sectionHead"><span class="titlemark">2    </span> <a
 id="x1-30002"></a>General parameters and input/output files</h3>
<!--l. 69--><p class="noindent" >Here, we document the syntax of the commands and parameters used to set up and use the collective variables
module in LAMMPS. One of these parameters is the configuration file or the configuration text for the module itself,
whose syntax is described in <a
href="#x1-50002.2">2.2<!--tex4ht:ref: sec:colvars_config --></a> and in the following sections.
<!--l. 207--><p class="noindent" >
   <h4 class="subsectionHead"><span class="titlemark">2.1    </span> <a
 id="x1-40002.1"></a>LAMMPS keywords</h4>
<!--l. 207--><p class="noindent" >To enable a collective variables-based calculation, the following line must be added to the LAMMPS configuration
file:<br
class="newline" /><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">fix </span><span
class="ptmri7t-x-x-109">ID  </span><span
class="ot1-zi4r-0-x-x-109">all colvars </span><span
class="ptmri7t-x-x-109">configfile  keyword value pairs ...</span><br
class="newline" /><br
class="newline" />where <span
class="ptmri7t-x-x-109">ID </span>is a string that uniquely identifies this fix command inside a LAMMPS script, <span
class="ptmri7t-x-x-109">configfile </span>is the name of the
configuration file for the collective variables module, followed by one or more of the following optional keywords
with their corresponding arguments:
     <ul class="itemize1">
     <li class="itemize"><a
 id="dx1-4001"></a> <span
class="ot1-zi4r-0-x-x-120">input </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Name or prefix of the input state file<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span>Keyword of the <span
class="ot1-zi4r-0-x-x-109">fix colvars </span>command <br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>string <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:   </span>If a value is provided, it is interpreted as either the name of the input state file, or
     as the prefix of the file named <span
class="ptmri7t-x-x-109">input</span><span
class="ot1-zi4r-0-x-x-109">.colvars.state</span>. This allows to continue a previous collective
     variables-based calculation when a regular binary LAMMPS restart file is not available (see <a
href="#x1-60002.3">2.3<!--tex4ht:ref: sec:colvars_input --></a>).
     </li>
     <li class="itemize"><a
 id="dx1-4002"></a> <span
class="ot1-zi4r-0-x-x-120">output </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Prefix of the output state file<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span>Keyword of the <span
class="ot1-zi4r-0-x-x-109">fix colvars </span>command <br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>string <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span>&#8220;out&#8221; <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>If a value is provided, it is interpreted as the prefix to all output files that will be written
     by the collective variables module (see <a
href="#x1-70002.4">2.4<!--tex4ht:ref: sec:colvars_output --></a>).
     </li>
     <li class="itemize"><a
 id="dx1-4003"></a> <span
class="ot1-zi4r-0-x-x-120">unwrap </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Whether to unwrap coordinates passed to the colvars module<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span>keyword of the <span
class="ot1-zi4r-0-x-x-109">fix colvars </span>command <br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>&#8220;yes&#8221; or &#8220;no&#8221; <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span>&#8220;yes&#8221; <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:   </span>This keyword controls whether wrapped or unwrapped coordinates are passed to the
     colvars module for calculation of the collective variables and of the resulting forces. The default is
     to use the image flags to reconstruct the absolute atom positions: under this convention, centers of
     mass and centers of geometry are calculated as a weighted vector sum (see <a
href="#x1-170004.3">4.3<!--tex4ht:ref: sec:colvar_atom_groups_wrapping --></a>). Setting this to <span
class="ptmri7t-x-x-109">no </span>will


     use the current local coordinates that are wrapped back into the simulation cell at each re-neighboring
     instead.
     </li>
     <li class="itemize"><a
 id="dx1-4004"></a> <span
class="ot1-zi4r-0-x-x-120">seed </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Seed for the random number generator<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span>Keyword of the <span
class="ot1-zi4r-0-x-x-109">fix colvars </span>command <br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>positive integer <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span>1966 <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>If defined, the value of this keyword is provided as seed to the random number generator.
     This is only meaningful when the <span
class="ot1-zi4r-0-x-x-109">extendedLangevinDamping </span>keyword is used (see <a
href="#x1-120003.4">3.4<!--tex4ht:ref: sec:colvar_extended --></a>).
     </li>
     <li class="itemize"><a
 id="dx1-4005"></a> <span
class="ot1-zi4r-0-x-x-120">tstat </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Thermostating fix<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span>Keyword of the <span
class="ot1-zi4r-0-x-x-109">fix colvars </span>command <br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>string <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span>NULL <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>This keyword provides the <span
class="ptmri7t-x-x-109">ID </span>of an applicable thermostating fix command. This will
     be used to provide the colvars module with the current thermostat target temperature when using a
     method that needs this information.
     </li></ul>
<!--l. 210--><p class="noindent" >
   <h4 class="subsectionHead"><span class="titlemark">2.2    </span> <a
 id="x1-50002.2"></a>Configuration syntax for the collective variables module</h4>
<!--l. 217--><p class="noindent" >The syntax of the colvars configuration is &#8220;<span
class="ot1-zi4r-0-x-x-109">keyword value</span>&#8221;, where the keyword and its value are separated by any
white space. The following rules apply:
     <ul class="itemize1">
     <li class="itemize">keywords are case-insensitive (<span
class="ot1-zi4r-0-x-x-109">upperBoundary </span>is the same as <span
class="ot1-zi4r-0-x-x-109">upperboundary </span>and <span
class="ot1-zi4r-0-x-x-109">UPPERBOUNDARY</span>):
     their string values are however case-sensitive (e.g.&#x00A0;file names);
     </li>
     <li class="itemize">a long value or a list of multiple values can be distributed across multiple lines by using curly braces,
     &#8220;<span
class="cmsy-10x-x-109">{</span>&#8221; and &#8220;<span
class="cmsy-10x-x-109">}</span>&#8221;: the opening brace &#8220;<span
class="cmsy-10x-x-109">{</span>&#8221; must occur on the same line as the keyword, following a space
     character or other white space; the closing brace &#8220;<span
class="cmsy-10x-x-109">}</span>&#8221; can be at any position after that;
     </li>
     <li class="itemize">many keywords are nested, and are only meaningful within a specific context: for every keyword
     documented in the following, the &#8220;parent&#8221; keyword that defines such context is also indicated;
     </li>
     <li class="itemize">if  a  keyword  requiring  a  boolean  value  (<span
class="ot1-zi4r-0-x-x-109">yes|on|true </span>or  <span
class="ot1-zi4r-0-x-x-109">no|off|false</span>)  is  provided  without  an
     explicit value, it defaults to &#8216;<span
class="ot1-zi4r-0-x-x-109">yes|on|true</span>&#8217;; for example, &#8216;<span
class="ot1-zi4r-0-x-x-109">outputAppliedForce</span>&#8217; may be used as
     shorthand for &#8216;<span
class="ot1-zi4r-0-x-x-109">outputAppliedForce on</span>&#8217;;
     </li>
     <li class="itemize">the hash character <span
class="ot1-zi4r-0-x-x-109"># </span>indicates a comment: all text in the same line following this character will be
     ignored.
     </li></ul>


<!--l. 244--><p class="indent" >   The following keywords are available in the global context of the colvars configuration, i.e.&#x00A0;they are not nested
inside other keywords:
     <ul class="itemize1">
     <li class="itemize"><a
 id="dx1-5001"></a> <span
class="ot1-zi4r-0-x-x-120">colvarsTrajFrequency </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Colvar value trajectory frequency<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span>global <br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>positive integer <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">100</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>The values of each colvar (and of other related quantities, if requested) are written to
     the file <span
class="ptmri7t-x-x-109">output</span><span
class="ot1-zi4r-0-x-x-109">.colvars.traj </span>every these many steps throughout the simulation. If the value is <span
class="ot1-zi4r-0-x-x-109">0</span>,
     such trajectory file is not written. For optimization the output is buffered, and synchronized with the
     disk only when the restart file is being written.
     </li>
     <li class="itemize"><a
 id="dx1-5002"></a> <span
class="ot1-zi4r-0-x-x-120">colvarsTrajAppend </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Append to trajectory file?<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span>global <br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">off</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:   </span>If this flag is enabled, and a file with the same name as the trajectory file is already
     present, new data is appended to that file. Otherwise, a new file is created with the same name that
     overwrites the previous file.
     </li>
     <li class="itemize"><a
 id="dx1-5003"></a> <span
class="ot1-zi4r-0-x-x-120">colvarsRestartFrequency </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Colvar module restart frequency<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span>global <br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>positive integer <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">restartFreq</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:   </span>Allows  to  choose  a  different  restart  frequency  for  the  collective  variables  module.
     Redefining it may be useful to trace the time evolution of those few properties which are not written
     to the trajectory file for reasons of disk space.
     </li>
     <li class="itemize"><a
 id="dx1-5004"></a> <span
class="ot1-zi4r-0-x-x-120">indexFile </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Index file for atom selection (GROMACS &#8220;ndx&#8221; format)<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span>global <br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>UNIX filename <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:   </span>This option reads an index file (usually with a <span
class="ot1-zi4r-0-x-x-109">.ndx </span>extension) as produced by the
     <span
class="ot1-zi4r-0-x-x-109">make</span><span
class="ot1-zi4r-0-x-x-109">_ndx </span>tool of GROMACS. This keyword may be repeated to load multiple index files: the same
     group name cannot appear in multiple index files. In LAMMPS, the <span
class="ot1-zi4r-0-x-x-109">group2ndx </span>command can be used
     to generate such file from existing groups. Note that the collective variables module reads the indices
     of atoms from the index file: therefore, the LAMMPS groups do not need to remain active during the
     simulation, and could be deleted right after issuing <span
class="ot1-zi4r-0-x-x-109">group2ndx</span>.  The names of index groups contained
     in this file can then be used to define atom groups with the <span
class="ot1-zi4r-0-x-x-109">indexGroup </span>keyword. Other supported
     methods to select atoms are described in <a
href="#x1-140004">4<!--tex4ht:ref: sec:colvar_atom_groups --></a>.
     </li>
     <li class="itemize"><a
 id="dx1-5005"></a> <span
class="ot1-zi4r-0-x-x-120">analysis </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Turn on run-time statistical analysis<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span>global <br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">off</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>If this flag is enabled, each colvar is instructed to perform whatever run-time statistical


     analysis it is configured to, such as correlation functions, or running averages and standard deviations.
     See section&#x00A0;<a
href="#x1-130003.5">3.5<!--tex4ht:ref: sec:colvar_acf --></a> for details.
     </li></ul>
<!--l. 312--><p class="indent" >   The example below defines the same configuration shown in Fig.&#x00A0;<a
href="#x1-20021">1<!--tex4ht:ref: fig:colvars_diagram --></a>. The options within the <span
class="ot1-zi4r-0-x-x-109">colvar </span>blocks are
described in <a
href="#x1-80003">3<!--tex4ht:ref: sec:colvar --></a> and <a
href="#x1-190005">5<!--tex4ht:ref: sec:cvc --></a>, the ones within the <span
class="ot1-zi4r-0-x-x-109">harmonic </span>and <span
class="ot1-zi4r-0-x-x-109">histogram </span>blocks in <a
href="#x1-490006">6<!--tex4ht:ref: sec:colvarbias --></a>. <span
class="ptmb7t-x-x-109">Note: </span><span
class="ptmri7t-x-x-109">except </span><span
class="ot1-zi4r-0-x-x-109">colvar</span><span
class="ptmri7t-x-x-109">, none of the</span>
<span
class="ptmri7t-x-x-109">keywords shown is mandatory</span>.<br
class="newline" />
<!--l. 317--><p class="noindent" ><span
class="ot1-zi4r-0-x-x-109">colvar </span><span
class="cmsy-10x-x-109">{</span><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;# difference of two distances</span><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;name d </span><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;width 0.2 # 0.2 </span><span
class="ot1-zi4r-0-x-x-109">Å of estimated fluctuation width </span><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;distance </span><span
class="cmsy-10x-x-109">{</span><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;componentCoeff 1.0</span><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;group1 </span><span
class="cmsy-10x-x-109">{ </span><span
class="ot1-zi4r-0-x-x-109">atomNumbers 1 2 </span><span
class="cmsy-10x-x-109">}</span><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;group2 </span><span
class="cmsy-10x-x-109">{ </span><span
class="ot1-zi4r-0-x-x-109">atomNumbers 3 4 5 </span><span
class="cmsy-10x-x-109">}</span><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="cmsy-10x-x-109">}</span><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;distance </span><span
class="cmsy-10x-x-109">{</span><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;componentCoeff -1.0</span><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;group1 </span><span
class="cmsy-10x-x-109">{ </span><span
class="ot1-zi4r-0-x-x-109">atomNumbers 7 </span><span
class="cmsy-10x-x-109">}</span><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;group2 </span><span
class="cmsy-10x-x-109">{ </span><span
class="ot1-zi4r-0-x-x-109">atomNumbers 8 9 10 </span><span
class="cmsy-10x-x-109">}</span><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="cmsy-10x-x-109">}</span><br
class="newline" /><span
class="cmsy-10x-x-109">}</span><br
class="newline" /><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">colvar </span><span
class="cmsy-10x-x-109">{</span><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;name c</span><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;coordNum </span><span
class="cmsy-10x-x-109">{</span><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;cutoff 6.0</span><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;group1 </span><span
class="cmsy-10x-x-109">{ </span><span
class="ot1-zi4r-0-x-x-109">atomNumbersRange 1-10 </span><span
class="cmsy-10x-x-109">}</span><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;group2 </span><span
class="cmsy-10x-x-109">{ </span><span
class="ot1-zi4r-0-x-x-109">atomNumbersRange 11-20 </span><span
class="cmsy-10x-x-109">}</span><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="cmsy-10x-x-109">}</span><br
class="newline" /><span
class="cmsy-10x-x-109">}</span><br
class="newline" /> <br
class="newline" /><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">harmonic </span><span
class="cmsy-10x-x-109">{</span><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;colvars d c</span><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;centers 3.0 4.0</span><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;forceConstant 5.0</span><br
class="newline" /><span
class="cmsy-10x-x-109">}</span><br
class="newline" />
<!--l. 360--><p class="noindent" ><span
class="ot1-zi4r-0-x-x-109">histogram </span><span
class="cmsy-10x-x-109">{</span><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;colvars c</span><br
class="newline" /><span
class="cmsy-10x-x-109">}</span><br
class="newline" />
<!--l. 366--><p class="indent" >   Section <a
href="#x1-80003">3<!--tex4ht:ref: sec:colvar --></a> explains how to define a colvar and its behavior, regardless of its specific functional form. To define
colvars that are appropriate to a specific physical system, Section <a
href="#x1-140004">4<!--tex4ht:ref: sec:colvar_atom_groups --></a> documents how to select atoms, and section <a
href="#x1-190005">5<!--tex4ht:ref: sec:cvc --></a>
lists all of the available functional forms, which we call &#8220;colvar components&#8221;. Finally, section <a
href="#x1-490006">6<!--tex4ht:ref: sec:colvarbias --></a> lists the
available methods and algorithms to perform biased simulations and multidimensional analysis of


colvars.
<!--l. 371--><p class="noindent" >
   <h4 class="subsectionHead"><span class="titlemark">2.3    </span> <a
 id="x1-60002.3"></a>Input state file (optional)</h4>
<!--l. 374--><p class="noindent" >Aside from the colvars configuration, an optional input state file may be provided to load the relevant data from a
previous simulation.   The name of this file is provided as the argument to the <span
class="ot1-zi4r-0-x-x-109">input </span>keyword of the <span
class="ot1-zi4r-0-x-x-109">fix ID all</span>
<span
class="ot1-zi4r-0-x-x-109">colvars </span>command. The same information is stored in the binary restart files of LAMMPS, so it not needed when
continuing a calculation from such a restart.
<!--l. 381--><p class="noindent" >
   <h4 class="subsectionHead"><span class="titlemark">2.4    </span> <a
 id="x1-70002.4"></a>Output files</h4>
<!--l. 384--><p class="noindent" >During a simulation with collective variables defined, the following three output files are written:
     <ul class="itemize1">
     <li class="itemize">a <span
class="ptmri7t-x-x-109">state file</span>, named <span
class="ptmri7t-x-x-109">output</span><span
class="ot1-zi4r-0-x-x-109">.colvars.state</span>; this file is in ASCII format;
     </li>
     <li class="itemize">if      the   parameter   <span
class="ot1-zi4r-0-x-x-109">colvarsRestartFrequency  </span>is   larger   than   zero,   a   <span
class="ptmri7t-x-x-109">restart   file   </span>named
     <span
class="ptmri7t-x-x-109">restart</span><span
class="ot1-zi4r-0-x-x-109">.colvars.state </span>is written every that many steps: this file is equivalent to the final state file;
     </li>
     <li class="itemize">if the parameter <span
class="ot1-zi4r-0-x-x-109">colvarsTrajFrequency </span>is greater than 0 (default: 100), a <span
class="ptmri7t-x-x-109">trajectory file </span>is written
     during the simulation: its name is <span
class="ptmri7t-x-x-109">output</span><span
class="ot1-zi4r-0-x-x-109">.colvars.traj</span>; unlike the state file, it is not needed to restart
     a simulation, but can be used later for post-processing and analysis.
     </li></ul>
<!--l. 397--><p class="indent" >   Other output files may be written by specific methods applied to the colvars (e.g.&#x00A0;by the ABF method, see <a
href="#x1-500006.1">6.1<!--tex4ht:ref: sec:colvarbias_abf --></a>,
or the metadynamics method, see <a
href="#x1-550006.2">6.2<!--tex4ht:ref: sec:colvarbias_meta --></a>). Like the colvar trajectory file, they are needed only for analyzing, not
continuing a simulation. All such files&#8217; names also begin with the prefix <span
class="ptmri7t-x-x-109">output</span>.
<!--l. 404--><p class="noindent" >
   <h3 class="sectionHead"><span class="titlemark">3    </span> <a
 id="x1-80003"></a>Defining collective variables and their properties</h3>
<!--l. 407--><p class="noindent" >In the configuration file each colvar is defined by the keyword <span
class="ot1-zi4r-0-x-x-109">colvar</span>, followed by its configuration options within
curly braces: <span
class="ot1-zi4r-0-x-x-109">colvar</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="cmsy-10x-x-109">{</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;...</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="cmsy-10x-x-109">}</span>. One of these options is the name of a colvar component: for example, including
<span
class="ot1-zi4r-0-x-x-109">rmsd </span><span
class="cmsy-10x-x-109">{</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;...</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="cmsy-10x-x-109">} </span>defines the colvar as a RMSD function. <span
class="ptmri7t-x-x-109">In most applications, only one component is used, and the</span>
<span
class="ptmri7t-x-x-109">component is equal to the colvar.</span>
<!--l. 410--><p class="indent" >   The full list of colvar components can be found in Section&#x00A0;<a
href="#x1-190005">5<!--tex4ht:ref: sec:cvc --></a>, with the syntax to select atoms in Section&#x00A0;<a
href="#x1-140004">4<!--tex4ht:ref: sec:colvar_atom_groups --></a>.
The following section lists several options to control the behavior of a single colvar, regardless of its
type.


<!--l. 413--><p class="noindent" >
   <h4 class="subsectionHead"><span class="titlemark">3.1    </span> <a
 id="x1-90003.1"></a>General options for a collective variable</h4>
<!--l. 415--><p class="noindent" >The following options are not required by default; however, the first four are very frequently used:
     <ul class="itemize1">
     <li class="itemize"><a
 id="dx1-9001"></a> <span
class="ot1-zi4r-0-x-x-120">name </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Name of this colvar<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">colvar</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>string <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span>&#8220;<span
class="ot1-zi4r-0-x-x-109">colvar</span>&#8221; + numeric id <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>The name is an unique case-sensitive string which allows the colvar module to identify
     this colvar unambiguously; it is also used in the trajectory file to label to the columns corresponding
     to this colvar.
     </li>
     <li class="itemize"><a
 id="dx1-9002"></a> <span
class="ot1-zi4r-0-x-x-120">width </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Colvar fluctuation scale, or resolution for grid-based methods<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">colvar</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>positive decimal <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span>1.0 <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:   </span>This number has the same physical unit as the colvar value and defines an effective
     colvar unit. Biasing algorithms use it for different purposes. Harmonic restraints (<a
href="#x1-610006.3">6.3<!--tex4ht:ref: sec:colvarbias_harmonic --></a>) use it to set
     the  physical  unit  of  the  force  constant,  which  is  useful  for  multidimensional  restraints  involving
     colvars with different units or scale which may then be defined by a single, scaled force constant.
     Histogram (<a
href="#x1-670006.6">6.6<!--tex4ht:ref: sec:colvarbias_histogram --></a>) and ABF biases (<a
href="#x1-500006.1">6.1<!--tex4ht:ref: sec:colvarbias_abf --></a>) interpret it as the grid spacing in the direction of this variable.
     Metadynamics (<a
href="#x1-550006.2">6.2<!--tex4ht:ref: sec:colvarbias_meta --></a>) uses it to set the width of newly added hills. In other cases, it is simplest to keep
     the default value of 1, so that harmonic force constants are provided in their usual physical unit. When
     a non-unity width is required by the application, the optimal value is application-dependent, but can
     often be thought of as a user-provided estimate of the fluctuation amplitude for the colvar. In those
     cases, it should generally be set smaller than or equal to the standard deviation of the colvar during a
     very short simulation run.
     </li>
     <li class="itemize"><a
 id="dx1-9003"></a> <span
class="ot1-zi4r-0-x-x-120">lowerBoundary </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Lower boundary of the colvar<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">colvar</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>decimal <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>Defines the lowest end of the interval of &#8220;relevant&#8221; values for the colvar. This number
     can  be  either  a  true  physical  boundary,  or  a  user-defined  number.  Together  with  <span
class="ot1-zi4r-0-x-x-109">upperBoundary</span>
     and <span
class="ot1-zi4r-0-x-x-109">width</span>, it is used to define a grid of values along the colvar (not available for colvars based on
     <span
class="ot1-zi4r-0-x-x-109">distanceDir</span>, <span
class="ot1-zi4r-0-x-x-109">distanceVec</span>, and <span
class="ot1-zi4r-0-x-x-109">orientation</span>). This option does not affect dynamics: to confine a
     colvar within a certain interval, the options <span
class="ot1-zi4r-0-x-x-109">lowerWall </span>and <span
class="ot1-zi4r-0-x-x-109">lowerWallConstant </span>should be used.
     </li>
     <li class="itemize"><a
 id="dx1-9004"></a> <span
class="ot1-zi4r-0-x-x-120">upperBoundary </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Upper boundary of the colvar<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">colvar</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>decimal <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>Similarly to <span
class="ot1-zi4r-0-x-x-109">lowerBoundary</span>, defines the highest possible or allowed value.
     </li>
     <li class="itemize"><a
 id="dx1-9005"></a> <span
class="ot1-zi4r-0-x-x-120">hardLowerBoundary </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Whether the lower boundary is the physical lower limit<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">colvar</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">off</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>This option does not affect simulation results, but enables some internal optimizations.
     Depending on its mathematical definition, a colvar may have &#8220;natural&#8221; boundaries: for example, a
     <span
class="ot1-zi4r-0-x-x-109">distance </span>colvar has a &#8220;natural&#8221; lower boundary at 0. Setting this option instructs the colvars module
     that the user-defined lower boundary is &#8220;natural&#8221;. See Section&#x00A0;<a
href="#x1-190005">5<!--tex4ht:ref: sec:cvc --></a> for the physical ranges of values of
     each component.
     </li>
     <li class="itemize"><a
 id="dx1-9006"></a> <span
class="ot1-zi4r-0-x-x-120">hardUpperBoundary  </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Whether the upper boundary is the physical upper limit of the colvar&#8217;s
     values<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">colvar</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">off</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>Analogous to <span
class="ot1-zi4r-0-x-x-109">hardLowerBoundary</span>.
     </li>
     <li class="itemize"><a
 id="dx1-9007"></a> <span
class="ot1-zi4r-0-x-x-120">expandBoundaries </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Allow to expand the two boundaries if needed<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">colvar</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">off</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>If defined, biasing and analysis methods may keep their own copies of <span
class="ot1-zi4r-0-x-x-109">lowerBoundary</span>
     and <span
class="ot1-zi4r-0-x-x-109">upperBoundary</span>, and expand them to accommodate values that do not fit in the initial range.
     Currently, this option is used by the metadynamics bias (<a
href="#x1-550006.2">6.2<!--tex4ht:ref: sec:colvarbias_meta --></a>) to keep all of its hills fully within the
     grid. This option cannot be used when the initial boundaries already span the full period of a periodic
     colvar.</li></ul>
<!--l. 504--><p class="noindent" >
   <h4 class="subsectionHead"><span class="titlemark">3.2    </span> <a
 id="x1-100003.2"></a>Artificial boundary potentials (walls)</h4>
<!--l. 506--><p class="noindent" >The following options are useful to define restraints (confining potentials) for this colvar. To apply moving restraints,
or restraints to more than one colvar simultaneously, a more convenient option is to use the <span
class="ot1-zi4r-0-x-x-109">harmonic </span>bias
(<a
href="#x1-610006.3">6.3<!--tex4ht:ref: sec:colvarbias_harmonic --></a>).
     <ul class="itemize1">
     <li class="itemize"><a
 id="dx1-10001"></a> <span
class="ot1-zi4r-0-x-x-120">lowerWallConstant </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Lower wall force constant (unit of energy specified by <span
class="ot1-zi4r-0-x-x-109">units</span>)<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">colvar</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>positive decimal <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:    </span>Defines  the  force  constant  for  a  confining  restraint  on  the  colvar,  in  the  form  of
     a  &#8220;half-harmonic&#8221;  potential.  The  potential  starts  at  <span
class="ot1-zi4r-0-x-x-109">lowerWall </span>if  it  is  defined,  or  <span
class="ot1-zi4r-0-x-x-109">lowerBoundary</span>
     otherwise. The energy unit of the constant is the unit of energy specified by <span
class="ot1-zi4r-0-x-x-109">units</span>, while the spatial
     unit U is that of the colvar.
     </li>
     <li class="itemize"><a
 id="dx1-10002"></a> <span
class="ot1-zi4r-0-x-x-120">lowerWall </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Position of the lower wall<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">colvar</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>decimal <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">lowerBoundary</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>Defines the value below which a confining restraint on the colvar is applied, in the form
     of a &#8220;half-harmonic&#8221; potential. Allows to use a different position of the wall than <span
class="ot1-zi4r-0-x-x-109">lowerBoundary</span>.
     </li>
     <li class="itemize"><a
 id="dx1-10003"></a> <span
class="ot1-zi4r-0-x-x-120">upperWallConstant </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Upper wall force constant (unit of energy specified by <span
class="ot1-zi4r-0-x-x-109">units</span>)<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">colvar</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>positive decimal <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>Analogous to <span
class="ot1-zi4r-0-x-x-109">lowerWallConstant</span>.
     </li>
     <li class="itemize"><a
 id="dx1-10004"></a> <span
class="ot1-zi4r-0-x-x-120">upperWall </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Position of the upper wall<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">colvar</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>decimal <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">upperBoundary</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>Analogous to <span
class="ot1-zi4r-0-x-x-109">lowerWall</span>.</li></ul>
<!--l. 550--><p class="noindent" >
   <h4 class="subsectionHead"><span class="titlemark">3.3    </span> <a
 id="x1-110003.3"></a>Trajectory output</h4>
     <ul class="itemize1">
     <li class="itemize"><a
 id="dx1-11001"></a> <span
class="ot1-zi4r-0-x-x-120">outputValue </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Output a trajectory for this colvar<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">colvar</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">on</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:    </span>If  <span
class="ot1-zi4r-0-x-x-109">colvarsTrajFrequency </span>is  non-zero,  the  value  of  this  colvar  is  written  to  the
     trajectory file every <span
class="ot1-zi4r-0-x-x-109">colvarsTrajFrequency </span>steps in the column labeled &#8220;<span
class="zptmcm7m-x-x-109">&#x003C;</span><span
class="ot1-zi4r-0-x-x-109">name</span><span
class="zptmcm7m-x-x-109">&#x003E;</span>&#8221;.
     </li>
     <li class="itemize"><a
 id="dx1-11002"></a> <span
class="ot1-zi4r-0-x-x-120">outputVelocity </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Output a velocity trajectory for this colvar<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">colvar</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">off</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>If <span
class="ot1-zi4r-0-x-x-109">colvarsTrajFrequency </span>is defined, the finite-difference calculated velocity of this
     colvar are written to the trajectory file under the label &#8220;<span
class="ot1-zi4r-0-x-x-109">v</span><span
class="ot1-zi4r-0-x-x-109">_</span><span
class="zptmcm7m-x-x-109">&#x003C;</span><span
class="ot1-zi4r-0-x-x-109">name</span><span
class="zptmcm7m-x-x-109">&#x003E;</span>&#8221;.
     </li>
     <li class="itemize"><a
 id="dx1-11003"></a> <span
class="ot1-zi4r-0-x-x-120">outputEnergy </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Output an energy trajectory for this colvar<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">colvar</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">off</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:   </span>This option applies only to extended Lagrangian colvars. If <span
class="ot1-zi4r-0-x-x-109">colvarsTrajFrequency</span>
     is  defined,  the  kinetic  energy  of  the  extended  degree  and  freedom  and  the  potential  energy  of
     the  restraining  spring  are  are  written  to  the  trajectory  file  under  the  labels  &#8220;<span
class="ot1-zi4r-0-x-x-109">Ek</span><span
class="ot1-zi4r-0-x-x-109">_</span><span
class="zptmcm7m-x-x-109">&#x003C;</span><span
class="ot1-zi4r-0-x-x-109">name</span><span
class="zptmcm7m-x-x-109">&#x003E;</span>&#8221;  and
     &#8220;<span
class="ot1-zi4r-0-x-x-109">Ep</span><span
class="ot1-zi4r-0-x-x-109">_</span><span
class="zptmcm7m-x-x-109">&#x003C;</span><span
class="ot1-zi4r-0-x-x-109">name</span><span
class="zptmcm7m-x-x-109">&#x003E;</span>&#8221;.
     </li>
     <li class="itemize"><a
 id="dx1-11004"></a> <span
class="ot1-zi4r-0-x-x-120">outputSystemForce </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Output a system force trajectory for this colvar<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">colvar</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">off</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:   </span>If <span
class="ot1-zi4r-0-x-x-109">colvarsTrajFrequency </span>is defined, the total system force on this colvar (i.e.&#x00A0;the
     projection of all interatomic forces except constraint forces on this colvar &#8212; see equation&#x00A0;(<a
href="#x1-50004r15">15<!--tex4ht:ref: eq:gradient_vector --></a>) in
     section&#x00A0;<a
href="#x1-500006.1">6.1<!--tex4ht:ref: sec:colvarbias_abf --></a>) are written to the trajectory file under the label &#8220;<span
class="ot1-zi4r-0-x-x-109">fs</span><span
class="ot1-zi4r-0-x-x-109">_</span><span
class="zptmcm7m-x-x-109">&#x003C;</span><span
class="ot1-zi4r-0-x-x-109">name</span><span
class="zptmcm7m-x-x-109">&#x003E;</span>&#8221;. For extended Lagrangian
     colvars,  the  &#8221;system  force&#8221;  felt  by  the  extended  degree  of  freedom  is  simply  the  force  from  the
     harmonic spring. <span
class="ptmb7t-x-x-109">Note: </span>not all components support this option. The physical unit for this force is the
     unit of energy specified by <span
class="ot1-zi4r-0-x-x-109">units</span>, divided by the colvar unit U.
     </li>
     <li class="itemize"><a
 id="dx1-11005"></a> <span
class="ot1-zi4r-0-x-x-120">outputAppliedForce </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Output an applied force trajectory for this colvar<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">colvar</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">off</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>If <span
class="ot1-zi4r-0-x-x-109">colvarsTrajFrequency </span>is defined, the total force applied on this colvar by biases
     and confining potentials (walls) within the colvar module are written to the trajectory under the label
     &#8220;<span
class="ot1-zi4r-0-x-x-109">fa</span><span
class="ot1-zi4r-0-x-x-109">_</span><span
class="zptmcm7m-x-x-109">&#x003C;</span><span
class="ot1-zi4r-0-x-x-109">name</span><span
class="zptmcm7m-x-x-109">&#x003E;</span>&#8221;. For extended Lagrangian colvars, this force is actually applied to the extended degree
     of freedom rather than the geometric colvar itself. The physical unit for this force is the unit of energy
     specified by <span
class="ot1-zi4r-0-x-x-109">units </span>divided by the colvar unit.</li></ul>
<!--l. 630--><p class="noindent" >
   <h4 class="subsectionHead"><span class="titlemark">3.4    </span> <a
 id="x1-120003.4"></a>Extended Lagrangian.</h4>
<!--l. 633--><p class="noindent" >The following options enable extended-system dynamics, where a colvar is coupled to an additional degree of
freedom (fictitious particle) by a harmonic spring. All biasing and confining forces are then applied to the extended
degree of freedom, and the actual, geometric colvar (function of Cartesian coordinates) only feels the force from the
harmonic spring.
     <ul class="itemize1">
     <li class="itemize"><a
 id="dx1-12001"></a> <span
class="ot1-zi4r-0-x-x-120">extendedLagrangian </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Add extended degree of freedom<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">colvar</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">off</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:    </span>Adds  a  fictitious  particle  to  be  coupled  to  the  colvar  by  a  harmonic  spring.  The
     fictitious  mass  and  the  force  constant  of  the  coupling  potential  are  derived  from  the  parameters
     <span
class="ot1-zi4r-0-x-x-109">extendedTimeConstant </span>and <span
class="ot1-zi4r-0-x-x-109">extendedFluctuation</span>, described below. Biasing forces on the colvar
     are applied to this fictitious particle, rather than to the atoms directly. This implements the extended
     Lagrangian formalism used in some metadynamics simulations&#x00A0;<span class="cite">[<a
href="#XIannuzzi2003">2</a>]</span>.
     </li>
     <li class="itemize"><a
 id="dx1-12002"></a> <span
class="ot1-zi4r-0-x-x-120">extendedFluctuation </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Standard deviation between the colvar and the fictitious particle (colvar
     unit)<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">colvar</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>positive decimal <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:    </span>Defines  the  spring  stiffness  for  the  <span
class="ot1-zi4r-0-x-x-109">extendedLagrangian </span>mode,  by  setting  the
     typical   deviation   between   the   colvar   and   the   extended   degree   of   freedom   due   to   thermal


     fluctuation. The spring force constant is calculated internally as <span
class="zptmcm7m-x-x-109">k</span><sub><span
class="zptmcm7m-x-x-80">B</span></sub><span
class="zptmcm7m-x-x-109">T &#x2215;&#x03C3;</span><sup><span
class="zptmcm7t-x-x-80">2</span></sup>, where <span
class="zptmcm7m-x-x-109">&#x03C3; </span>is the value of
     <span
class="ot1-zi4r-0-x-x-109">extendedFluctuation</span>.
     </li>
     <li class="itemize"><a
 id="dx1-12003"></a> <span
class="ot1-zi4r-0-x-x-120">extendedTimeConstant </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Oscillation period of the fictitious particle (fs)<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">colvar</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>positive decimal <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">200</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>Defines the inertial mass of the fictitious particle, by setting the oscillation period of the
     harmonic oscillator formed by the fictitious particle and the spring. The period should be much larger
     than the MD time step to ensure accurate integration of the extended particle&#8217;s equation of motion.
     The fictitious mass is calculated internally as <span
class="zptmcm7m-x-x-109">k</span><sub><span
class="zptmcm7m-x-x-80">B</span></sub><span
class="zptmcm7m-x-x-109">T </span><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">&#x03C4;&#x2215;</span><span
class="zptmcm7t-x-x-109">2</span><span
class="zptmcm7m-x-x-109">&#x03C0;&#x03C3;</span><span
class="zptmcm7t-x-x-109">)</span><sup><span
class="zptmcm7t-x-x-80">2</span></sup>, where <span
class="zptmcm7m-x-x-109">&#x03C4; </span>is the period and <span
class="zptmcm7m-x-x-109">&#x03C3; </span>is the typical
     fluctuation (see above).
     </li>
     <li class="itemize"><a
 id="dx1-12004"></a> <span
class="ot1-zi4r-0-x-x-120">extendedTemp </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Temperature for the extended degree of freedom (K)<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">colvar</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>positive decimal <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span>thermostat temperature <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>Temperature used for calculating the coupling force constant of the extended coordinate
     (see <span
class="ot1-zi4r-0-x-x-109">extendedFluctuation</span>) and, if needed, as a target temperature for extended Langevin dynamics
     (see <span
class="ot1-zi4r-0-x-x-109">extendedLangevinDamping</span>). This should normally be left at its default value.
     </li>
     <li class="itemize"><a
 id="dx1-12005"></a> <span
class="ot1-zi4r-0-x-x-120">extendedLangevinDamping </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Damping factor for extended Langevin dynamics (ps<sup><span
class="zptmcm7y-x-x-80">-</span><span
class="zptmcm7t-x-x-80">1</span></sup>)<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">colvar</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>positive decimal <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">1.0</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:   </span>If  this  is  non-zero,  the  extended  degree  of  freedom  undergoes  Langevin  dynamics
     at  temperature  <span
class="ot1-zi4r-0-x-x-109">extendedTemp</span>.  The  friction  force  is  minus  <span
class="ot1-zi4r-0-x-x-109">extendedLangevinDamping </span>times  the
     velocity.  This  is  useful  because  the  extended  dynamics  coordinate  may  heat  up  in  the  transient
     non-equilibrium regime of ABF. Use moderate damping values, to limit viscous friction (potentially
     slowing  down  diffusive  sampling)  and  stochastic  noise  (increasing  the  variance  of  statistical
     measurements). In doubt, use the default value.</li></ul>
<!--l. 719--><p class="noindent" >
   <h4 class="subsectionHead"><span class="titlemark">3.5    </span> <a
 id="x1-130003.5"></a>Statistical analysis of collective variables</h4>
<!--l. 722--><p class="noindent" >When the global keyword <span
class="ot1-zi4r-0-x-x-109">analysis </span>is defined in the configuration file, run-time calculations of statistical properties
for individual colvars can be performed. At the moment, several types of time correlation functions, running
averages and running standard deviations are available.
     <ul class="itemize1">
     <li class="itemize"><a
 id="dx1-13001"></a> <span
class="ot1-zi4r-0-x-x-120">corrFunc </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Calculate a time correlation function?<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">colvar</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">off</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>Whether or not a time correlaction function should be calculated for this colvar.


     </li>
     <li class="itemize"><a
 id="dx1-13002"></a> <span
class="ot1-zi4r-0-x-x-120">corrFuncWithColvar </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Colvar name for the correlation function<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">colvar</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>string <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>By default, the auto-correlation function (ACF) of this colvar, <span
class="zptmcm7m-x-x-109">&#x03BE;</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub>, is calculated. When
     this option is specified, the correlation function is calculated instead with another colvar, <span
class="zptmcm7m-x-x-109">&#x03BE;</span><sub><span
class="zptmcm7m-x-x-80">j</span></sub>, which
     must be of the same type (scalar, vector, or quaternion) as <span
class="zptmcm7m-x-x-109">&#x03BE;</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub>.
     </li>
     <li class="itemize"><a
 id="dx1-13003"></a> <span
class="ot1-zi4r-0-x-x-120">corrFuncType </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Type of the correlation function<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">colvar</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span><span
class="ot1-zi4r-0-x-x-109">velocity</span>, <span
class="ot1-zi4r-0-x-x-109">coordinate </span>or <span
class="ot1-zi4r-0-x-x-109">coordinate</span><span
class="ot1-zi4r-0-x-x-109">_p2</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">velocity</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:         </span>With     <span
class="ot1-zi4r-0-x-x-109">coordinate  </span>or     <span
class="ot1-zi4r-0-x-x-109">velocity</span>,     the     correlation     function     <span
class="zptmcm7m-x-x-109">C</span><sub><span
class="zptmcm7m-x-x-80">i,j</span></sub><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">t</span><span
class="zptmcm7t-x-x-109">)</span>&#x00A0;=
     <img
src="colvars-refman-lammps2x.png" alt="&#x27E8;&#x03A0; (&#x03BE; (t),&#x03BE;(t + t))&#x27E9;
    i 0   j 0"  class="left" align="middle"> is calculated between the variables <span
class="zptmcm7m-x-x-109">&#x03BE;</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub> and <span
class="zptmcm7m-x-x-109">&#x03BE;</span><sub><span
class="zptmcm7m-x-x-80">j</span></sub>, or their velocities. <span
class="zptmcm7t-x-x-109">&#x03A0;(</span><span
class="zptmcm7m-x-x-109">&#x03BE;</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub><span
class="zptmcm7m-x-x-109">,&#x03BE;</span><sub><span
class="zptmcm7m-x-x-80">j</span></sub><span
class="zptmcm7t-x-x-109">) </span>is
     the scalar product when calculated between scalar or vector values, whereas for quaternions it is the
     cosine between the two corresponding rotation axes. With <span
class="ot1-zi4r-0-x-x-109">coordinate</span><span
class="ot1-zi4r-0-x-x-109">_p2</span>, the second order Legendre
     polynomial, <span
class="zptmcm7t-x-x-109">(3</span><span
class="zptmcm7t-x-x-109">cos</span><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">&#x03B8;</span><span
class="zptmcm7t-x-x-109">)</span><sup><span
class="zptmcm7t-x-x-80">2</span></sup> <span
class="zptmcm7y-x-x-109">-</span><span
class="zptmcm7t-x-x-109">1)</span><span
class="zptmcm7m-x-x-109">&#x2215;</span><span
class="zptmcm7t-x-x-109">2</span>, is used instead of the cosine.
     </li>
     <li class="itemize"><a
 id="dx1-13004"></a> <span
class="ot1-zi4r-0-x-x-120">corrFuncNormalize </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Normalize the time correlation function?<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">colvar</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">on</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>If enabled, the value of the correlation function at <span
class="zptmcm7m-x-x-109">t</span>&#x00A0;= 0 is normalized to 1; otherwise,
     it equals to <img
src="colvars-refman-lammps3x.png" alt="&#x27E8;O (&#x03BE;,&#x03BE; )&#x27E9;
     i j"  class="left" align="middle">.
     </li>
     <li class="itemize"><a
 id="dx1-13005"></a> <span
class="ot1-zi4r-0-x-x-120">corrFuncLength </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Length of the time correlation function<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">colvar</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>positive integer <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">1000</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>Length (in number of points) of the time correlation function.
     </li>
     <li class="itemize"><a
 id="dx1-13006"></a> <span
class="ot1-zi4r-0-x-x-120">corrFuncStride </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Stride of the time correlation function<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">colvar</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>positive integer <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">1</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>Number of steps between two values of the time correlation function.
     </li>
     <li class="itemize"><a
 id="dx1-13007"></a> <span
class="ot1-zi4r-0-x-x-120">corrFuncOffset </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Offset of the time correlation function<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">colvar</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>positive integer <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">0</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>The starting time (in number of steps) of the time correlation function (default: <span
class="zptmcm7m-x-x-109">t</span>&#x00A0;= 0).
     <span
class="ptmb7t-x-x-109">Note: </span><span
class="ptmri7t-x-x-109">the value at </span><span
class="zptmcm7m-x-x-109">t</span><span
class="ptmri7t-x-x-109">&#x00A0;= 0 is always used for the normalization</span>.
     </li>


     <li class="itemize"><a
 id="dx1-13008"></a> <span
class="ot1-zi4r-0-x-x-120">corrFuncOutputFile </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Output file for the time correlation function<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">colvar</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>UNIX filename <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="zptmcm7m-x-x-109">&#x003C;</span><span
class="ot1-zi4r-0-x-x-109">name</span><span
class="zptmcm7m-x-x-109">&#x003E;</span><span
class="ot1-zi4r-0-x-x-109">.corrfunc.dat</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>The time correlation function is saved in this file.
     </li>
     <li class="itemize"><a
 id="dx1-13009"></a> <span
class="ot1-zi4r-0-x-x-120">runAve </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Calculate the running average and standard deviation<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">colvar</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">off</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>Whether or not the running average and standard deviation should be calculated for this
     colvar.
     </li>
     <li class="itemize"><a
 id="dx1-13010"></a> <span
class="ot1-zi4r-0-x-x-120">runAveLength </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Length of the running average window<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">colvar</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>positive integer <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">1000</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>Length (in number of points) of the running average window.
     </li>
     <li class="itemize"><a
 id="dx1-13011"></a> <span
class="ot1-zi4r-0-x-x-120">runAveStride </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Stride of the running average window values<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">colvar</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>positive integer <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">1</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>Number of steps between two values within the running average window.
     </li>
     <li class="itemize"><a
 id="dx1-13012"></a> <span
class="ot1-zi4r-0-x-x-120">runAveOutputFile </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Output file for the running average and standard deviation<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">colvar</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>UNIX filename <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="zptmcm7m-x-x-109">&#x003C;</span><span
class="ot1-zi4r-0-x-x-109">name</span><span
class="zptmcm7m-x-x-109">&#x003E;</span><span
class="ot1-zi4r-0-x-x-109">.runave.dat</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>The running average and standard deviation are saved in this file.
     </li></ul>
<!--l. 859--><p class="noindent" >
   <h3 class="sectionHead"><span class="titlemark">4    </span> <a
 id="x1-140004"></a>Selecting atoms for colvars: defining atom groups</h3>
<!--l. 862--><p class="noindent" >
   <h4 class="subsectionHead"><span class="titlemark">4.1    </span> <a
 id="x1-150004.1"></a>Selection keywords</h4>
<!--l. 865--><p class="noindent" >To define collective variables, atoms are usually selected by group. Each group is identified by a name that is unique
in the context of the specific colvar component (e.g.&#x00A0;for a distance component, the names of the two groups are
<span
class="ot1-zi4r-0-x-x-109">group1 </span>and <span
class="ot1-zi4r-0-x-x-109">group2</span>). The name is followed by a brace-delimited block of selection keywords: these may be used
individually or in combination with each other, and each can be repeated any number of times. Selection is


incremental: each keyword adds the corresponding atoms to the selection, so that different sets of atoms can be
combined. However, atoms included by multiple keywords are only counted once. Below is an example
configuration for an atom group named &#8220;<span
class="ot1-zi4r-0-x-x-109">atoms</span>&#8221;, which uses an unusually varied combination of selection
keywords:<br
class="newline" />
<!--l. 873--><p class="noindent" ><span
class="ot1-zi4r-0-x-x-109">atoms </span><span
class="cmsy-10x-x-109">{</span><br
class="newline" /><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;# add atoms 1 and 3 to this group (note: the first atom in the system is 1)</span><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;atomNumbers </span><span
class="cmsy-10x-x-109">{ </span><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;1 3</span><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="cmsy-10x-x-109">}</span><br
class="newline" /><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;# add atoms starting from 20 up to and including 50</span><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;atomNumbersRange 20-50</span><br
class="newline" /><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;# add index group (requires a .ndx file to be provided globally)</span><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;indexGroup Water</span><br
class="newline" /> <span
class="cmsy-10x-x-109">}</span><br
class="newline" />
<!--l. 900--><p class="indent" >   The resulting selection includes atoms 1 and 3, those between 20 and 50, and those in the index group called
&#8220;Water&#8221;; the indices of this group are read from the file provided by <span
class="ot1-zi4r-0-x-x-109">indexFile</span>, in the global section of the
configuration file.
<!--l. 903--><p class="indent" >   The complete list of selection keywords available in LAMMPS is:
     <ul class="itemize1">
     <li class="itemize"><a
 id="dx1-15001"></a> <span
class="ot1-zi4r-0-x-x-120">atomNumbers </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>List of atom numbers<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span>atom group <br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>space-separated list of positive integers <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>This option adds to the group all the atoms whose numbers are in the list. <span
class="ptmri7t-x-x-109">The number</span>
     <span
class="ptmri7t-x-x-109">of the first atom in the system is 1: to convert from a VMD selection, use &#8220;atomselect get serial&#8221;.</span>
     </li>
     <li class="itemize"><a
 id="dx1-15002"></a> <span
class="ot1-zi4r-0-x-x-120">indexGroup </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Name of index group to be used (GROMACS format)<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span>atom group <br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>string <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:   </span>If the name of an index file has been provided by <span
class="ot1-zi4r-0-x-x-109">indexFile</span>, this option allows to
     select one index group from that file: the atoms from that index group will be used to define the current
     group.
     </li>
     <li class="itemize"><a
 id="dx1-15003"></a> <span
class="ot1-zi4r-0-x-x-120">atomNumbersRange </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Atoms within a number range<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span>atom group <br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span><span
class="zptmcm7m-x-x-109">&#x003C;</span>Starting number<span
class="zptmcm7m-x-x-109">&#x003E;</span>-<span
class="zptmcm7m-x-x-109">&#x003C;</span>Ending number<span
class="zptmcm7m-x-x-109">&#x003E; </span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:   </span>This  option  includes  in  the  group  all  atoms  whose  numbers  are  within  the  range
     specified. <span
class="ptmri7t-x-x-109">The number of the first atom in the system is 1.</span>
     </li>
     <li class="itemize"><a
 id="dx1-15004"></a> <span
class="ot1-zi4r-0-x-x-120">dummyAtom </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Dummy atom position (Å)<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span>atom group <br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span><span
class="ot1-zi4r-0-x-x-109">(x, y, z) </span>triplet <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>Instead of selecting any atom, this option makes the group a virtual particle at a fixed
     position  in  space.  This  is  useful  e.g.&#x00A0;to  replace  a  group&#8217;s  center  of  geometry  with  a  user-defined
     position.
     </li></ul>
<!--l. 997--><p class="noindent" >
   <h4 class="subsectionHead"><span class="titlemark">4.2    </span> <a
 id="x1-160004.2"></a>Moving frame of reference.</h4>
<!--l. 1000--><p class="noindent" >The following options define an automatic calculation of an optimal translation (<span
class="ot1-zi4r-0-x-x-109">centerReference</span>) or optimal
rotation (<span
class="ot1-zi4r-0-x-x-109">rotateReference</span>), that superimposes the positions of this group to a provided set of reference
coordinates. This can allow, for example, to effectively remove from certain colvars the effects of
molecular tumbling and of diffusion. Given the set of atomic positions <span
class="ptmb7t-x-x-109">x</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub>, the colvar <span
class="zptmcm7m-x-x-109">&#x03BE; </span>can be defined
on a set of roto-translated positions <span
class="ptmb7t-x-x-109">x</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub><span
class="zptmcm7y-x-x-109">&#x2032; </span><span
class="zptmcm7t-x-x-109">= </span><span
class="zptmcm7m-x-x-109">R</span><span
class="zptmcm7t-x-x-109">(</span><span
class="ptmb7t-x-x-109">x</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub> <span
class="zptmcm7y-x-x-109">- </span><span
class="ptmb7t-x-x-109">x</span><sup><span
class="zptmcm7t-x-x-80">C</span></sup><span
class="zptmcm7t-x-x-109">)+ </span><span
class="ptmb7t-x-x-109">x</span><sup><span
class="zptmcm7t-x-x-80">ref</span></sup>. <span
class="ptmb7t-x-x-109">x</span><sup><span
class="zptmcm7t-x-x-80">C</span></sup> is the geometric center of the <span
class="ptmb7t-x-x-109">x</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub>, <span
class="zptmcm7m-x-x-109">R </span>is
the optimal rotation matrix to the reference positions and <span
class="ptmb7t-x-x-109">x</span><sup><span
class="zptmcm7t-x-x-80">ref</span></sup> is the geometric center of the reference
positions.
<!--l. 1005--><p class="indent" >   Components that are defined based on pairwise distances are naturally invariant under global roto-translations.
Other components are instead affected by global rotations or translations: however, they can be made invariant if
they are expressed in the frame of reference of a chosen group of atoms, using the <span
class="ot1-zi4r-0-x-x-109">centerReference </span>and
<span
class="ot1-zi4r-0-x-x-109">rotateReference </span>options. Finally, a few components are defined by convention using a roto-translated frame (e.g.
the minimal RMSD): for these components, <span
class="ot1-zi4r-0-x-x-109">centerReference </span>and <span
class="ot1-zi4r-0-x-x-109">rotateReference </span>are enabled by default. In
typical applications, the default settings result in the expected behavior.
     <ul class="itemize1">
     <li class="itemize"><a
 id="dx1-16001"></a> <span
class="ot1-zi4r-0-x-x-120">centerReference </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Implicitly remove translations for this group<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span>atom group <br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">off</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:   </span>If this option is <span
class="ot1-zi4r-0-x-x-109">on</span>, the center of geometry of the group will be aligned with that of
     the reference positions provided by  <span
class="ot1-zi4r-0-x-x-109">refPositions</span>. Colvar components will only have access to the
     aligned positions. <span
class="ptmb7t-x-x-109">Note</span>: unless otherwise specified, <span
class="ot1-zi4r-0-x-x-109">rmsd </span>and <span
class="ot1-zi4r-0-x-x-109">eigenvector </span>set this option to <span
class="ot1-zi4r-0-x-x-109">on </span><span
class="ptmri7t-x-x-109">by</span>
     <span
class="ptmri7t-x-x-109">default</span>.
     </li>
     <li class="itemize"><a
 id="dx1-16002"></a> <span
class="ot1-zi4r-0-x-x-120">rotateReference </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Implicitly remove rotations for this group<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span>atom group <br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">off</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:   </span>If this option is <span
class="ot1-zi4r-0-x-x-109">on</span>, the coordinates of this group will be optimally superimposed to
     the reference positions provided by  <span
class="ot1-zi4r-0-x-x-109">refPositions</span>. The rotation will be performed around the center
     of geometry if <span
class="ot1-zi4r-0-x-x-109">centerReference </span>is <span
class="ot1-zi4r-0-x-x-109">on</span>, around the origin otherwise. The algorithm used is the same
     employed by the <span
class="ot1-zi4r-0-x-x-109">orientation </span>colvar component&#x00A0;<span class="cite">[<a
href="#XCoutsias2004">3</a>]</span>. Forces applied to the atoms of this group will
     also  be  implicitly  rotated  back  to  the  original  frame.  <span
class="ptmb7t-x-x-109">Note</span>:  unless  otherwise  specified,  <span
class="ot1-zi4r-0-x-x-109">rmsd </span>and
     <span
class="ot1-zi4r-0-x-x-109">eigenvector </span>set this option to <span
class="ot1-zi4r-0-x-x-109">on </span><span
class="ptmri7t-x-x-109">by default</span>.


     </li>
     <li class="itemize"><a
 id="dx1-16003"></a> <span
class="ot1-zi4r-0-x-x-120">refPositions </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Reference positions for fitting (Å)<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span>atom group <br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>space-separated list of <span
class="ot1-zi4r-0-x-x-109">(x, y, z) </span>triplets <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:     </span>This  option  provides  a  list  of  reference  coordinates  for  <span
class="ot1-zi4r-0-x-x-109">centerReference </span>or
     <span
class="ot1-zi4r-0-x-x-109">rotateReference</span>. If only <span
class="ot1-zi4r-0-x-x-109">centerReference </span>is <span
class="ot1-zi4r-0-x-x-109">on</span>, the list may contain a single (x, y, z) triplet; if
     also <span
class="ot1-zi4r-0-x-x-109">rotateReference </span>is <span
class="ot1-zi4r-0-x-x-109">on</span>, the list should be as long as the atom group.
     </li>
     <li class="itemize"><a
 id="dx1-16004"></a> <span
class="ot1-zi4r-0-x-x-120">refPositionsGroup </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Use an alternate set of atoms to define the roto-translation<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span>atom group <br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>Block <span
class="ot1-zi4r-0-x-x-109">refPositionsGroup </span><span
class="cmsy-10x-x-109">{ </span><span
class="ot1-zi4r-0-x-x-109">... </span><span
class="cmsy-10x-x-109">}</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span>This group itself <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:   </span>If either <span
class="ot1-zi4r-0-x-x-109">centerReference </span>or <span
class="ot1-zi4r-0-x-x-109">rotateReference </span>is defined, this keyword defines an
     alternate atom group to calculate the optimal roto-translation. Use this option to define a continuous
     rotation if the structure of the group involved changes significantly (a typical symptom would be the
     message &#8220;Warning: discontinuous rotation!&#8221;).
     <!--l. 1113--><p class="noindent" ><span
class="ot1-zi4r-0-x-x-109"># Example: defining a group "atoms", with its coordinates expressed </span><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109"># on a roto-translated frame of reference defined by a second group</span><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">atoms </span><span
class="cmsy-10x-x-109">{</span><br
class="newline" /><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;psfSegID PROT</span><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;atomNameResidueRange CA 41-48</span><br
class="newline" /><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;centerReference yes</span><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;rotateReference yes</span><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;refPositionsGroup </span><span
class="cmsy-10x-x-109">{</span><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;# define the frame by fitting the rest of the protein</span><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;psfSegID PROT PROT</span><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;atomNameResidueRange CA 1-40</span><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;atomNameResidueRange CA 49-100</span><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="cmsy-10x-x-109">} </span><br
class="newline" /><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;refPositionsFile all.pdb # can be the entire system</span><br
class="newline" /><span
class="cmsy-10x-x-109">}</span><br
class="newline" /></li></ul>
<!--l. 1116--><p class="indent" >   The following two options have default values appropriate for the vast majority of applications, and are only
provided to support rare, special cases.
     <ul class="itemize1">
     <li class="itemize"><a
 id="dx1-16005"></a> <span
class="ot1-zi4r-0-x-x-120">enableFitGradients </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Include the roto-translational contribution to colvar gradients<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span>atom group <br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">on</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:   </span>When  either  <span
class="ot1-zi4r-0-x-x-109">centerReference </span>or  <span
class="ot1-zi4r-0-x-x-109">rotateReference </span>is  on,  the  gradients  of  some
     colvars  include  terms  proportional  to  <span
class="zptmcm7m-x-x-109">&#x2202;R&#x2215;&#x2202;</span><span
class="ptmb7t-x-x-109">x</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub>  (rotational  gradients)  and  <span
class="zptmcm7m-x-x-109">&#x2202;</span><span
class="ptmb7t-x-x-109">x</span><sup><span
class="zptmcm7t-x-x-80">C</span></sup><span
class="zptmcm7m-x-x-109">&#x2215;&#x2202;</span><span
class="ptmb7t-x-x-109">x</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub>  (translational
     gradients). By default, these terms are calculated and included in the total gradients; if this option is
     set to <span
class="ot1-zi4r-0-x-x-109">off</span>, they are neglected. In the case of a minimum RMSD component, this flag is automatically
     disabled because the contributions of those derivatives to the gradients cancel out.


     </li>
     <li class="itemize"><a
 id="dx1-16006"></a> <span
class="ot1-zi4r-0-x-x-120">enableForces </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Apply forces from this colvar to this group<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span>atom group <br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">on</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>If this option is <span
class="ot1-zi4r-0-x-x-109">off</span>, no forces are applied from this colvar to this group. Other forces
     are not affected (i.e. those from the MD engine, from other colvars, and other external forces). For
     dummy atoms, this option is <span
class="ot1-zi4r-0-x-x-109">off </span>by default.
     </li></ul>
<!--l. 1152--><p class="noindent" >
   <h4 class="subsectionHead"><span class="titlemark">4.3    </span> <a
 id="x1-170004.3"></a>Treatment of periodic boundary conditions.</h4>
<!--l. 1175--><p class="noindent" >In simulations with periodic boundary conditions, many of the implemented colvar components rely on the fact that
each position within a group of atoms is at the nearest periodic image from the center of geometry of the group
itself. However, due to the internal wrapping of individual atomic positions done by LAMMPS, this
assumption is inaccurate if groups lies at one of the unit cell&#8217;s boundaries. For this reason, within
the colvars module coordinates are unwrapped by default to avoid discontinuities due to coordinate
wrapping (see <span
class="ot1-zi4r-0-x-x-109">unwrap </span>keyword in <span
class="ptmb7t-x-x-109">??</span>). The user should determine whether maintaining the default value
of <span
class="ot1-zi4r-0-x-x-109">unwrap</span>, depending on the specifics of each system. In general, internal coordinate wrapping by
LAMMPS does not affect the calculation of colvars if each atom group satisfies one or more of the
following:
<!--l. 1182--><p class="indent" >
     <ol  class="enumerate1" >
     <li
  class="enumerate" id="x1-17001x4.3">it is composed by only one atom;
     </li>
     <li
  class="enumerate" id="x1-17002x4.3">it is used by a colvar component which does not make use of its center of geometry, but only of
     pairwise distances (<span
class="ot1-zi4r-0-x-x-109">distanceInv</span>, <span
class="ot1-zi4r-0-x-x-109">coordNum</span>, <span
class="ot1-zi4r-0-x-x-109">hBond</span>, <span
class="ot1-zi4r-0-x-x-109">alpha</span>, <span
class="ot1-zi4r-0-x-x-109">dihedralPC</span>);
     </li>
     <li
  class="enumerate" id="x1-17003x4.3">it is used by a colvar component that ignores the ill-defined Cartesian components of its center of
     mass (such as the <span
class="zptmcm7m-x-x-109">x </span>and <span
class="zptmcm7m-x-x-109">y </span>components of a membrane&#8217;s center of mass modeled with <span
class="ot1-zi4r-0-x-x-109">distanceZ</span>).</li></ol>
<!--l. 1191--><p class="noindent" >
   <h4 class="subsectionHead"><span class="titlemark">4.4    </span> <a
 id="x1-180004.4"></a>Computational cost of colvars based on group size.</h4>
<!--l. 1194--><p class="noindent" >In parallel MD simulations, the calculation of most interaction terms are spread over many computational nodes, but
the calculation of colvars is not parallelized. Therefore, additional calculations are executed by the node calculating
the colvars, and most importantly, additional communication is added between the first node and the other nodes.  To
mitigate that, atom groups should be kept relatively small (up to a few thousands, depending on the computational
cost to simulate the system by itself).


<!--l. 1199--><p class="noindent" >
   <h3 class="sectionHead"><span class="titlemark">5    </span> <a
 id="x1-190005"></a>Collective variable components (basis functions)</h3>
<!--l. 1202--><p class="noindent" >Each colvar is defined by one or more <span
class="ptmri7t-x-x-109">components </span>(typically only one). Each component consists of a keyword
identifying a functional form, and a definition block following that keyword, specifying the atoms involved and any
additional parameters (cutoffs, &#8220;reference&#8221; values, &#x2026;).
<!--l. 1208--><p class="indent" >   The types of the components used in a colvar determine the properties of that colvar, and which biasing or
analysis methods can be applied. In most cases, the colvar returns a real number, which is computed by one or more
instances of the following components:
     <ul class="itemize1">
     <li class="itemize"><span
class="ot1-zi4r-0-x-x-109">distance</span>: distance between two groups;
     </li>
     <li class="itemize"><span
class="ot1-zi4r-0-x-x-109">distanceZ</span>: projection of a distance vector on an axis;
     </li>
     <li class="itemize"><span
class="ot1-zi4r-0-x-x-109">distanceXY</span>: projection of a distance vector on a plane;
     </li>
     <li class="itemize"><span
class="ot1-zi4r-0-x-x-109">distanceInv</span>: mean distance between two groups of atoms (e.g.&#x00A0;NOE-based distance);
     </li>
     <li class="itemize"><span
class="ot1-zi4r-0-x-x-109">angle</span>: angle between three groups;
     </li>
     <li class="itemize"><span
class="ot1-zi4r-0-x-x-109">coordNum</span>: coordination number between two groups;
     </li>
     <li class="itemize"><span
class="ot1-zi4r-0-x-x-109">selfCoordNum</span>: coordination number of atoms within a group;
     </li>
     <li class="itemize"><span
class="ot1-zi4r-0-x-x-109">hBond</span>: hydrogen bond between two atoms;
     </li>
     <li class="itemize"><span
class="ot1-zi4r-0-x-x-109">rmsd</span>: root mean square deviation (RMSD) from a set of reference coordinates;
     </li>
     <li class="itemize"><span
class="ot1-zi4r-0-x-x-109">eigenvector</span>: projection of the atomic coordinates on a vector;
     </li>
     <li class="itemize"><span
class="ot1-zi4r-0-x-x-109">orientationAngle</span>: angle of the best-fit rotation from a set of reference coordinates;
     </li>
     <li class="itemize"><span
class="ot1-zi4r-0-x-x-109">orientationProj</span>: cosine of <span
class="ot1-zi4r-0-x-x-109">orientationProj</span>;
     </li>
     <li class="itemize"><span
class="ot1-zi4r-0-x-x-109">spinAngle</span>: projection orthogonal to an axis of the best-fit rotation from a set of reference coordinates;
     </li>
     <li class="itemize"><span
class="ot1-zi4r-0-x-x-109">tilt</span>: projection on an axis of the best-fit rotation from a set of reference coordinates;
     </li>
     <li class="itemize"><span
class="ot1-zi4r-0-x-x-109">gyration</span>: radius of gyration of a group of atoms;


     </li>
     <li class="itemize"><span
class="ot1-zi4r-0-x-x-109">inertia</span>: moment of inertia of a group of atoms;
     </li>
     <li class="itemize"><span
class="ot1-zi4r-0-x-x-109">inertiaZ</span>: moment of inertia of a group of atoms around a chosen axis; </li></ul>
<!--l. 1242--><p class="indent" >   Some components do not return scalar, but vector values. They can only be combined with vector values of the
same type.
     <ul class="itemize1">
     <li class="itemize"><span
class="ot1-zi4r-0-x-x-109">distanceVec</span>: distance vector between two groups;
     </li>
     <li class="itemize"><span
class="ot1-zi4r-0-x-x-109">distanceDir</span>: unit vector parallel to distanceVec;
     </li>
     <li class="itemize"><span
class="ot1-zi4r-0-x-x-109">cartesian</span>: vector of atomic Cartesian coordinates;
     </li>
     <li class="itemize"><span
class="ot1-zi4r-0-x-x-109">orientation</span>: best-fit rotation, expressed as a unit quaternion.</li></ul>
<!--l. 1252--><p class="indent" >   In the following, all the available component types are listed, along with their physical units and the limiting
values, if any. Such limiting values can be used to define <span
class="ot1-zi4r-0-x-x-109">lowerBoundary </span>and <span
class="ot1-zi4r-0-x-x-109">upperBoundary </span>in the parent
colvar.
<!--l. 1258--><p class="noindent" >
   <h4 class="subsectionHead"><span class="titlemark">5.1    </span> <a
 id="x1-200005.1"></a>List of available colvar components</h4>
<!--l. 1260--><p class="noindent" >
   <h5 class="subsubsectionHead"><span class="titlemark">5.1.1    </span> <a
 id="x1-210005.1.1"></a><span
class="ot1-zi4r-0-x-x-109">distance</span>: center-of-mass distance between two groups.</h5>
<!--l. 1261--><p class="noindent" >The <span
class="ot1-zi4r-0-x-x-109">distance </span><span
class="cmsy-10x-x-109">{</span><span
class="ot1-zi4r-0-x-x-109">...</span><span
class="cmsy-10x-x-109">} </span>block defines a distance component, between two atom groups, <span
class="ot1-zi4r-0-x-x-109">group1 </span>and
<span
class="ot1-zi4r-0-x-x-109">group2</span>.
     <ul class="itemize1">
     <li class="itemize"><a
 id="dx1-21001"></a> <span
class="ot1-zi4r-0-x-x-120">group1 </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>First group of atoms<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">distance</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>Block <span
class="ot1-zi4r-0-x-x-109">group1 </span><span
class="cmsy-10x-x-109">{</span><span
class="ot1-zi4r-0-x-x-109">...</span><span
class="cmsy-10x-x-109">}</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>First group of atoms.
     </li>
     <li class="itemize"><a
 id="dx1-21002"></a> <span
class="ot1-zi4r-0-x-x-120">group2 </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Second group of atoms<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">distance</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>Block <span
class="ot1-zi4r-0-x-x-109">group2 </span><span
class="cmsy-10x-x-109">{</span><span
class="ot1-zi4r-0-x-x-109">...</span><span
class="cmsy-10x-x-109">}</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>Second group of atoms.
     </li>
     <li class="itemize"><a
 id="dx1-21003"></a> <span
class="ot1-zi4r-0-x-x-120">forceNoPBC </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Calculate absolute rather than minimum-image distance?<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">distance</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">no</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>By default, in calculations with periodic boundary conditions, the <span
class="ot1-zi4r-0-x-x-109">distance </span>component
     returns the distance according to the minimum-image convention. If this parameter is set to <span
class="ot1-zi4r-0-x-x-109">yes</span>, PBC
     will be ignored and the distance between the coordinates as maintained internally will be used. This is
     only useful in a limited number of special cases, e.g. to describe the distance between remote points
     of a single macromolecule, which cannot be split across periodic cell boundaries, and for which the
     minimum-image distance might give the wrong result because of a relatively small periodic cell.
     </li>
     <li class="itemize"><a
 id="dx1-21004"></a> <span
class="ot1-zi4r-0-x-x-120">oneSiteSystemForce </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Measure system force on group 1 only?<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">distance</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">no</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>If this is set to <span
class="ot1-zi4r-0-x-x-109">yes</span>, the system force is measured along a vector field (see equation&#x00A0;(<a
href="#x1-50004r15">15<!--tex4ht:ref: eq:gradient_vector --></a>)
     in section&#x00A0;<a
href="#x1-500006.1">6.1<!--tex4ht:ref: sec:colvarbias_abf --></a>) that only involves atoms of <span
class="ot1-zi4r-0-x-x-109">group1</span>. This option is only useful for ABF, or custom
     biases that compute system forces. See section&#x00A0;<a
href="#x1-500006.1">6.1<!--tex4ht:ref: sec:colvarbias_abf --></a> for details.</li></ul>
<!--l. 1309--><p class="indent" >   The value returned is a positive number (in Å), ranging from <span
class="zptmcm7t-x-x-109">0 </span>to the largest possible interatomic distance within
the chosen boundary conditions (with PBCs, the minimum image convention is used unless the <span
class="ot1-zi4r-0-x-x-109">forceNoPBC </span>option
is set).
<!--l. 1315--><p class="noindent" >
   <h5 class="subsubsectionHead"><span class="titlemark">5.1.2    </span> <a
 id="x1-220005.1.2"></a><span
class="ot1-zi4r-0-x-x-109">distanceZ</span>: projection of a distance vector on an axis.</h5>
<!--l. 1316--><p class="noindent" >The <span
class="ot1-zi4r-0-x-x-109">distanceZ</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="cmsy-10x-x-109">{</span><span
class="ot1-zi4r-0-x-x-109">...</span><span
class="cmsy-10x-x-109">} </span>block defines a distance projection component, which can be seen as measuring the distance
between two groups projected onto an axis, or the position of a group along such an axis. The axis can be
defined using either one reference group and a constant vector, or dynamically based on two reference
groups.
     <ul class="itemize1">
     <li class="itemize"><a
 id="dx1-22001"></a> <span
class="ot1-zi4r-0-x-x-120">main </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Main group of atoms<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">distanceZ, distanceXY</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>Block <span
class="ot1-zi4r-0-x-x-109">main </span><span
class="cmsy-10x-x-109">{</span><span
class="ot1-zi4r-0-x-x-109">...</span><span
class="cmsy-10x-x-109">}</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>Group of atoms whose position <span
class="zptmcm7m-x-x-109">r</span> is measured.
     </li>
     <li class="itemize"><a
 id="dx1-22002"></a> <span
class="ot1-zi4r-0-x-x-120">ref </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Reference group of atoms<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">distanceZ, distanceXY</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>Block <span
class="ot1-zi4r-0-x-x-109">ref </span><span
class="cmsy-10x-x-109">{</span><span
class="ot1-zi4r-0-x-x-109">...</span><span
class="cmsy-10x-x-109">}</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>Reference group of atoms. The position of its center of mass is noted <span
class="zptmcm7m-x-x-109">r</span><sub><span
class="zptmcm7t-x-x-80">1</span></sub> below.
     </li>
     <li class="itemize"><a
 id="dx1-22003"></a> <span
class="ot1-zi4r-0-x-x-120">ref2 </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Secondary reference group<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">distanceZ, distanceXY</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>Block <span
class="ot1-zi4r-0-x-x-109">ref2 </span><span
class="cmsy-10x-x-109">{</span><span
class="ot1-zi4r-0-x-x-109">...</span><span
class="cmsy-10x-x-109">}</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span>none <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>Optional group of reference atoms, whose position <span
class="zptmcm7m-x-x-109">r</span><sub><span
class="zptmcm7t-x-x-80">2</span></sub> can be used to define a dynamic
     projection axis: <span
class="zptmcm7m-x-x-109">e</span> <span
class="zptmcm7t-x-x-109">= (</span><span
class="zptmcm7y-x-x-109">&#x2225;</span><span
class="zptmcm7m-x-x-109">r</span><sub><span
class="zptmcm7t-x-x-80">2</span></sub> <span
class="zptmcm7y-x-x-109">-</span><span
class="zptmcm7m-x-x-109">r</span><sub><span
class="zptmcm7t-x-x-80">1</span></sub><span
class="zptmcm7y-x-x-109">&#x2225;</span><span
class="zptmcm7t-x-x-109">)</span><sup><span
class="zptmcm7y-x-x-80">-</span><span
class="zptmcm7t-x-x-80">1</span></sup> <span
class="zptmcm7y-x-x-109">&#x00D7;</span><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">r</span><sub><span
class="zptmcm7t-x-x-80">2</span></sub> <span
class="zptmcm7y-x-x-109">-</span><span
class="zptmcm7m-x-x-109">r</span><sub><span
class="zptmcm7t-x-x-80">1</span></sub><span
class="zptmcm7t-x-x-109">)</span>. In this case, the origin is <span
class="zptmcm7m-x-x-109">r</span><sub><span
class="zptmcm7m-x-x-80">m</span></sub> <span
class="zptmcm7t-x-x-109">= 1</span><span
class="zptmcm7m-x-x-109">&#x2215;</span><span
class="zptmcm7t-x-x-109">2(</span><span
class="zptmcm7m-x-x-109">r</span><sub><span
class="zptmcm7t-x-x-80">1</span></sub> <span
class="zptmcm7t-x-x-109">+</span><span
class="zptmcm7m-x-x-109">r</span><sub><span
class="zptmcm7t-x-x-80">2</span></sub><span
class="zptmcm7t-x-x-109">)</span>, and the
     value of the component is <span
class="zptmcm7m-x-x-109">e</span><span
class="zptmcm7y-x-x-109">&#x22C5;</span><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">r</span><span
class="zptmcm7y-x-x-109">-</span><span
class="zptmcm7m-x-x-109">r</span><sub><span
class="zptmcm7m-x-x-80">m</span></sub><span
class="zptmcm7t-x-x-109">)</span>.


     </li>
     <li class="itemize"><a
 id="dx1-22004"></a> <span
class="ot1-zi4r-0-x-x-120">axis </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Projection axis (Å)<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">distanceZ</span>, <span
class="ot1-zi4r-0-x-x-109">distanceXY</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span><span
class="ot1-zi4r-0-x-x-109">(x, y, z) </span>triplet <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">(0.0, 0.0, 1.0)</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>The three components of this vector define (when normalized) a projection axis <span
class="zptmcm7m-x-x-109">e</span> for
     the distance vector <span
class="zptmcm7m-x-x-109">r</span><span
class="zptmcm7y-x-x-109">-</span><span
class="zptmcm7m-x-x-109">r</span><sub><span
class="zptmcm7t-x-x-80">1</span></sub> joining the centers of groups <span
class="ot1-zi4r-0-x-x-109">ref </span>and <span
class="ot1-zi4r-0-x-x-109">main</span>. The value of the component is
     then <span
class="zptmcm7m-x-x-109">e</span><span
class="zptmcm7y-x-x-109">&#x22C5;</span><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">r</span><span
class="zptmcm7y-x-x-109">-</span><span
class="zptmcm7m-x-x-109">r</span><sub><span
class="zptmcm7t-x-x-80">1</span></sub><span
class="zptmcm7t-x-x-109">)</span>. The vector should be written as three components separated by commas and enclosed
     in parentheses.
     </li>
     <li class="itemize"><a
 id="dx1-22005"></a> <span
class="ot1-zi4r-0-x-x-120">forceNoPBC </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Calculate absolute rather than minimum-image distance?<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">distanceZ, distanceXY</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">no</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:    </span>This  parameter  has  the  same  meaning  as  that  described  above  for  the  <span
class="ot1-zi4r-0-x-x-109">distance</span>
     component.
     </li>
     <li class="itemize"><a
 id="dx1-22006"></a> <span
class="ot1-zi4r-0-x-x-120">oneSiteSystemForce </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Measure system force on group <span
class="ot1-zi4r-0-x-x-109">main </span>only?<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">distanceZ, distanceXY</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">no</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>If this is set to <span
class="ot1-zi4r-0-x-x-109">yes</span>, the system force is measured along a vector field (see equation&#x00A0;(<a
href="#x1-50004r15">15<!--tex4ht:ref: eq:gradient_vector --></a>)
     in section&#x00A0;<a
href="#x1-500006.1">6.1<!--tex4ht:ref: sec:colvarbias_abf --></a>) that only involves atoms of <span
class="ot1-zi4r-0-x-x-109">main</span>. This option is only useful for ABF, or custom biases
     that compute system forces. See section&#x00A0;<a
href="#x1-500006.1">6.1<!--tex4ht:ref: sec:colvarbias_abf --></a> for details.</li></ul>
<!--l. 1386--><p class="noindent" >This component returns a number (in Å) whose range is determined by the chosen boundary conditions.
For instance, if the <span
class="zptmcm7m-x-x-109">z </span>axis is used in a simulation with periodic boundaries, the returned value ranges
between <span
class="zptmcm7y-x-x-109">-</span><span
class="zptmcm7m-x-x-109">b</span><sub><span
class="zptmcm7m-x-x-80">z</span></sub><span
class="zptmcm7m-x-x-109">&#x2215;</span><span
class="zptmcm7t-x-x-109">2 </span>and <span
class="zptmcm7m-x-x-109">b</span><sub><span
class="zptmcm7m-x-x-80">z</span></sub><span
class="zptmcm7m-x-x-109">&#x2215;</span><span
class="zptmcm7t-x-x-109">2</span>, where <span
class="zptmcm7m-x-x-109">b</span><sub><span
class="zptmcm7m-x-x-80">z</span></sub> is the box length along <span
class="zptmcm7m-x-x-109">z </span>(this behavior is disabled if <span
class="ot1-zi4r-0-x-x-109">forceNoPBC </span>is
set).
<!--l. 1393--><p class="noindent" >
   <h5 class="subsubsectionHead"><span class="titlemark">5.1.3    </span> <a
 id="x1-230005.1.3"></a><span
class="ot1-zi4r-0-x-x-109">distanceXY</span>: modulus of the projection of a distance vector on a plane.</h5>
<!--l. 1394--><p class="noindent" >The <span
class="ot1-zi4r-0-x-x-109">distanceXY</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="cmsy-10x-x-109">{</span><span
class="ot1-zi4r-0-x-x-109">...</span><span
class="cmsy-10x-x-109">} </span>block defines a distance projected on a plane, and accepts the same keywords as the
component <span
class="ot1-zi4r-0-x-x-109">distanceZ</span>, i.e. <span
class="ot1-zi4r-0-x-x-109">main</span>, <span
class="ot1-zi4r-0-x-x-109">ref</span>, either <span
class="ot1-zi4r-0-x-x-109">ref2 </span>or <span
class="ot1-zi4r-0-x-x-109">axis</span>, and <span
class="ot1-zi4r-0-x-x-109">oneSiteSystemForce</span>. It returns the norm
of the projection of the distance vector between <span
class="ot1-zi4r-0-x-x-109">main </span>and <span
class="ot1-zi4r-0-x-x-109">ref </span>onto the plane orthogonal to the axis.
The axis is defined using the <span
class="ot1-zi4r-0-x-x-109">axis </span>parameter or as the vector joining <span
class="ot1-zi4r-0-x-x-109">ref </span>and <span
class="ot1-zi4r-0-x-x-109">ref2 </span>(see <span
class="ot1-zi4r-0-x-x-109">distanceZ</span>
above).
<!--l. 1404--><p class="noindent" >
   <h5 class="subsubsectionHead"><span class="titlemark">5.1.4    </span> <a
 id="x1-240005.1.4"></a><span
class="ot1-zi4r-0-x-x-109">distanceVec</span>: distance vector between two groups.</h5>
<!--l. 1405--><p class="noindent" >The <span
class="ot1-zi4r-0-x-x-109">distanceVec</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="cmsy-10x-x-109">{</span><span
class="ot1-zi4r-0-x-x-109">...</span><span
class="cmsy-10x-x-109">} </span>block defines a distance vector component, which accepts the same keywords as the
component <span
class="ot1-zi4r-0-x-x-109">distance</span>: <span
class="ot1-zi4r-0-x-x-109">group1</span>, <span
class="ot1-zi4r-0-x-x-109">group2</span>, and <span
class="ot1-zi4r-0-x-x-109">forceNoPBC</span>. Its value is the 3-vector joining the centers of mass of
<span
class="ot1-zi4r-0-x-x-109">group1 </span>and <span
class="ot1-zi4r-0-x-x-109">group2</span>.


<!--l. 1412--><p class="noindent" >
   <h5 class="subsubsectionHead"><span class="titlemark">5.1.5    </span> <a
 id="x1-250005.1.5"></a><span
class="ot1-zi4r-0-x-x-109">distanceDir</span>: distance unit vector between two groups.</h5>
<!--l. 1413--><p class="noindent" >The <span
class="ot1-zi4r-0-x-x-109">distanceDir</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="cmsy-10x-x-109">{</span><span
class="ot1-zi4r-0-x-x-109">...</span><span
class="cmsy-10x-x-109">} </span>block defines a distance unit vector component, which accepts the same keywords as the
component <span
class="ot1-zi4r-0-x-x-109">distance</span>: <span
class="ot1-zi4r-0-x-x-109">group1</span>, <span
class="ot1-zi4r-0-x-x-109">group2</span>, and <span
class="ot1-zi4r-0-x-x-109">forceNoPBC</span>. It returns a 3-dimensional unit vector <span
class="ptmb7t-x-x-109">d </span><span
class="zptmcm7t-x-x-109">= (</span><span
class="zptmcm7m-x-x-109">d</span><sub><span
class="zptmcm7m-x-x-80">x</span></sub><span
class="zptmcm7m-x-x-109">,d</span><sub><span
class="zptmcm7m-x-x-80">y</span></sub><span
class="zptmcm7m-x-x-109">,d</span><sub><span
class="zptmcm7m-x-x-80">z</span></sub><span
class="zptmcm7t-x-x-109">)</span>,
with <span
class="zptmcm7y-x-x-109">|</span><span
class="ptmb7t-x-x-109">d</span><span
class="zptmcm7y-x-x-109">| </span><span
class="zptmcm7t-x-x-109">= 1</span>.
<!--l. 1421--><p class="noindent" >
   <h5 class="subsubsectionHead"><span class="titlemark">5.1.6    </span> <a
 id="x1-260005.1.6"></a><span
class="ot1-zi4r-0-x-x-109">distanceInv</span>: mean distance between two groups of atoms.</h5>
<!--l. 1422--><p class="noindent" >The <span
class="ot1-zi4r-0-x-x-109">distanceInv</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="cmsy-10x-x-109">{</span><span
class="ot1-zi4r-0-x-x-109">...</span><span
class="cmsy-10x-x-109">} </span>block defines a generalized mean distance between two groups of atoms 1 and 2, weighted
with exponent <span
class="zptmcm7t-x-x-109">1</span><span
class="zptmcm7m-x-x-109">&#x2215;n</span>:
   <table
class="equation"><tr><td><a
 id="x1-26001r2"></a>
   <center class="math-display" >
<img
src="colvars-refman-lammps4x.png" alt="      (                 )
         1     (  1  )n  - 1&#x2215;n
d[n1],2 =   ----&#x2211;    --ij--
        N1N2 i,j  &#x2225;d &#x2225;
" class="math-display" ></center></td><td class="equation-label">(2)</td></tr></table>
<!--l. 1426--><p class="nopar" >
where <span
class="zptmcm7y-x-x-109">&#x2225;</span><span
class="ptmb7t-x-x-109">d</span><sup><span
class="zptmcm7m-x-x-80">ij</span></sup><span
class="zptmcm7y-x-x-109">&#x2225; </span>is the distance between atoms <span
class="zptmcm7m-x-x-109">i </span>and <span
class="zptmcm7m-x-x-109">j </span>in groups 1 and 2 respectively, and <span
class="zptmcm7m-x-x-109">n </span>is an even integer. This
component accepts the same keywords as the component <span
class="ot1-zi4r-0-x-x-109">distance</span>: <span
class="ot1-zi4r-0-x-x-109">group1</span>, <span
class="ot1-zi4r-0-x-x-109">group2</span>, and <span
class="ot1-zi4r-0-x-x-109">forceNoPBC</span>. In addition,
the following option may be provided:
     <ul class="itemize1">
     <li class="itemize"><a
 id="dx1-26002"></a> <span
class="ot1-zi4r-0-x-x-120">exponent </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Exponent <span
class="zptmcm7m-x-x-109">n </span>in equation&#x00A0;<a
href="#x1-26001r2">2<!--tex4ht:ref: eq:distanceInv --></a><span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">distanceInv</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>positive even integer <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span>6 <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>Defines the exponent to which the individual distances are elevated before averaging.
     The default value of 6 is useful for example to applying restraints based on NOE-measured distances.</li></ul>
<!--l. 1438--><p class="noindent" >This component returns a number in Å, ranging from <span
class="zptmcm7t-x-x-109">0 </span>to the largest possible distance within the chosen boundary
conditions.
<!--l. 1441--><p class="noindent" >
   <h5 class="subsubsectionHead"><span class="titlemark">5.1.7    </span> <a
 id="x1-270005.1.7"></a><span
class="ot1-zi4r-0-x-x-109">cartesian</span>: vector of atomic Cartesian coordinates.</h5>
<!--l. 1442--><p class="noindent" >The <span
class="ot1-zi4r-0-x-x-109">cartesian</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="cmsy-10x-x-109">{</span><span
class="ot1-zi4r-0-x-x-109">...</span><span
class="cmsy-10x-x-109">} </span>block defines a component returning a flat vector containing the Cartesian coordinates of
all participating atoms, in the order <span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">x</span><sub><span
class="zptmcm7t-x-x-80">1</span></sub><span
class="zptmcm7m-x-x-109">,y</span><sub><span
class="zptmcm7t-x-x-80">1</span></sub><span
class="zptmcm7m-x-x-109">,z</span><sub><span
class="zptmcm7t-x-x-80">1</span></sub><span
class="zptmcm7m-x-x-109">,</span><img
src="colvars-refman-lammps5x.png" alt="&#x22C5;&#x22C5;&#x22C5;"  class="@cdots" ><span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7m-x-x-109">,x</span><sub><span
class="zptmcm7m-x-x-80">n</span></sub><span
class="zptmcm7m-x-x-109">,y</span><sub><span
class="zptmcm7m-x-x-80">n</span></sub><span
class="zptmcm7m-x-x-109">,z</span><sub><span
class="zptmcm7m-x-x-80">n</span></sub><span
class="zptmcm7t-x-x-109">)</span>. This component accepts the following
keyword:


     <ul class="itemize1">
     <li class="itemize"><a
 id="dx1-27001"></a> <span
class="ot1-zi4r-0-x-x-120">atoms </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Group of atoms<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">cartesian</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>Block <span
class="ot1-zi4r-0-x-x-109">atoms </span><span
class="cmsy-10x-x-109">{</span><span
class="ot1-zi4r-0-x-x-109">...</span><span
class="cmsy-10x-x-109">}</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:    </span>Defines  the  atoms  whose  coordinates  make  up  the  value  of  the  component.  If
     <span
class="ot1-zi4r-0-x-x-109">rotateReference </span>or <span
class="ot1-zi4r-0-x-x-109">centerReference </span>are defined, coordinates are evaluated within the moving
     frame of reference.</li></ul>
<!--l. 1459--><p class="noindent" >
   <h5 class="subsubsectionHead"><span class="titlemark">5.1.8    </span> <a
 id="x1-280005.1.8"></a><span
class="ot1-zi4r-0-x-x-109">angle</span>: angle between three groups.</h5>
<!--l. 1460--><p class="noindent" >The <span
class="ot1-zi4r-0-x-x-109">angle</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="cmsy-10x-x-109">{</span><span
class="ot1-zi4r-0-x-x-109">...</span><span
class="cmsy-10x-x-109">} </span>block defines an angle, and contains the three blocks <span
class="ot1-zi4r-0-x-x-109">group1</span>, <span
class="ot1-zi4r-0-x-x-109">group2 </span>and <span
class="ot1-zi4r-0-x-x-109">group3</span>, defining the
three groups. It returns an angle (in degrees) within the interval <span
class="zptmcm7t-x-x-109">[0 : 180]</span>.
<!--l. 1466--><p class="noindent" >
   <h5 class="subsubsectionHead"><span class="titlemark">5.1.9    </span> <a
 id="x1-290005.1.9"></a><span
class="ot1-zi4r-0-x-x-109">dihedral</span>: torsional angle between four groups.</h5>
<!--l. 1467--><p class="noindent" >The <span
class="ot1-zi4r-0-x-x-109">dihedral</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="cmsy-10x-x-109">{</span><span
class="ot1-zi4r-0-x-x-109">...</span><span
class="cmsy-10x-x-109">} </span>block defines a torsional angle, and contains the blocks <span
class="ot1-zi4r-0-x-x-109">group1</span>, <span
class="ot1-zi4r-0-x-x-109">group2</span>, <span
class="ot1-zi4r-0-x-x-109">group3 </span>and
<span
class="ot1-zi4r-0-x-x-109">group4</span>, defining the four groups. It returns an angle (in degrees) within the interval <span
class="zptmcm7t-x-x-109">[</span><span
class="zptmcm7y-x-x-109">-</span><span
class="zptmcm7t-x-x-109">180 : 180]</span>. The
colvar module calculates all the distances between two angles taking into account periodicity. For
instance, reference values for restraints or range boundaries can be defined by using any real number of
choice.
     <ul class="itemize1">
     <li class="itemize"><a
 id="dx1-29001"></a> <span
class="ot1-zi4r-0-x-x-120">oneSiteSystemForce </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Measure system force on group 1 only?<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">angle</span>, <span
class="ot1-zi4r-0-x-x-109">dihedral</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">no</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>If this is set to <span
class="ot1-zi4r-0-x-x-109">yes</span>, the system force is measured along a vector field (see equation&#x00A0;(<a
href="#x1-50004r15">15<!--tex4ht:ref: eq:gradient_vector --></a>)
     in section&#x00A0;<a
href="#x1-500006.1">6.1<!--tex4ht:ref: sec:colvarbias_abf --></a>) that only involves atoms of <span
class="ot1-zi4r-0-x-x-109">group1</span>. See section&#x00A0;<a
href="#x1-500006.1">6.1<!--tex4ht:ref: sec:colvarbias_abf --></a> for an example.</li></ul>
<!--l. 1489--><p class="noindent" >
   <h5 class="subsubsectionHead"><span class="titlemark">5.1.10    </span> <a
 id="x1-300005.1.10"></a><span
class="ot1-zi4r-0-x-x-109">coordNum</span>: coordination number between two groups.</h5>
<!--l. 1490--><p class="noindent" >The <span
class="ot1-zi4r-0-x-x-109">coordNum </span><span
class="cmsy-10x-x-109">{</span><span
class="ot1-zi4r-0-x-x-109">...</span><span
class="cmsy-10x-x-109">} </span>block defines a coordination number (or number of contacts), which calculates the function
<span
class="zptmcm7t-x-x-109">(1</span><span
class="zptmcm7y-x-x-109">-</span><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">d&#x2215;d</span><sub><span
class="zptmcm7t-x-x-80">0</span></sub><span
class="zptmcm7t-x-x-109">)</span><sup><span
class="zptmcm7m-x-x-80">n</span></sup><span
class="zptmcm7t-x-x-109">)</span><span
class="zptmcm7m-x-x-109">&#x2215;</span><span
class="zptmcm7t-x-x-109">(1</span><span
class="zptmcm7y-x-x-109">-</span><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">d&#x2215;d</span><sub><span
class="zptmcm7t-x-x-80">0</span></sub><span
class="zptmcm7t-x-x-109">)</span><sup><span
class="zptmcm7m-x-x-80">m</span></sup><span
class="zptmcm7t-x-x-109">)</span>, where <span
class="zptmcm7m-x-x-109">d</span><sub><span
class="zptmcm7t-x-x-80">0</span></sub> is the &#8220;cutoff&#8221; distance, and <span
class="zptmcm7m-x-x-109">n </span>and <span
class="zptmcm7m-x-x-109">m </span>are exponents that can control its
long range behavior and stiffness <span class="cite">[<a
href="#XIannuzzi2003">2</a>]</span>. This function is summed over all pairs of atoms in <span
class="ot1-zi4r-0-x-x-109">group1 </span>and
<span
class="ot1-zi4r-0-x-x-109">group2</span>:
   <table
class="equation"><tr><td><a
 id="x1-30001r3"></a>
   <center class="math-display" >
<img
src="colvars-refman-lammps6x.png" alt="                                                n
C(group1,group2 )=               1---(|xi--xj|&#x2215;d0)-
                    i&#x2208;g&#x2211;roup1j&#x2208;&#x2211;group21- (|xi- xj|&#x2215;d0)m
" class="math-display" ></center></td><td class="equation-label">(3)</td></tr></table>
<!--l. 1504--><p class="nopar" >
This colvar component accepts the same keywords as the component <span
class="ot1-zi4r-0-x-x-109">distance</span>, <span
class="ot1-zi4r-0-x-x-109">group1 </span>and <span
class="ot1-zi4r-0-x-x-109">group2</span>. In addition to
them, it recognizes the following keywords:
     <ul class="itemize1">
     <li class="itemize"><a
 id="dx1-30002"></a> <span
class="ot1-zi4r-0-x-x-120">cutoff </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>&#8220;Interaction&#8221; distance (Å)<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">coordNum</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>positive decimal <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span>4.0 <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>This number defines the switching distance to define an interatomic contact: for <span
class="zptmcm7m-x-x-109">d </span><span
class="zptmcm7y-x-x-109">&#x226A;</span><span
class="zptmcm7m-x-x-109">d</span><sub><span
class="zptmcm7t-x-x-80">0</span></sub>,
     the switching function <span
class="zptmcm7t-x-x-109">(1</span><span
class="zptmcm7y-x-x-109">-</span><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">d&#x2215;d</span><sub><span
class="zptmcm7t-x-x-80">0</span></sub><span
class="zptmcm7t-x-x-109">)</span><sup><span
class="zptmcm7m-x-x-80">n</span></sup><span
class="zptmcm7t-x-x-109">)</span><span
class="zptmcm7m-x-x-109">&#x2215;</span><span
class="zptmcm7t-x-x-109">(1</span><span
class="zptmcm7y-x-x-109">-</span><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">d&#x2215;d</span><sub><span
class="zptmcm7t-x-x-80">0</span></sub><span
class="zptmcm7t-x-x-109">)</span><sup><span
class="zptmcm7m-x-x-80">m</span></sup><span
class="zptmcm7t-x-x-109">) </span>is close to 1, at <span
class="zptmcm7m-x-x-109">d </span><span
class="zptmcm7t-x-x-109">= </span><span
class="zptmcm7m-x-x-109">d</span><sub><span
class="zptmcm7t-x-x-80">0</span></sub> it has a value of <span
class="zptmcm7m-x-x-109">n&#x2215;m </span>(<span
class="zptmcm7t-x-x-109">1</span><span
class="zptmcm7m-x-x-109">&#x2215;</span><span
class="zptmcm7t-x-x-109">2</span>
     with the default <span
class="zptmcm7m-x-x-109">n </span>and <span
class="zptmcm7m-x-x-109">m</span>), and at <span
class="zptmcm7m-x-x-109">d </span><span
class="zptmcm7y-x-x-109">&#x226B; </span><span
class="zptmcm7m-x-x-109">d</span><sub><span
class="zptmcm7t-x-x-80">0</span></sub> it goes to zero approximately like <span
class="zptmcm7m-x-x-109">d</span><sup><span
class="zptmcm7m-x-x-80">m</span><span
class="zptmcm7y-x-x-80">-</span><span
class="zptmcm7m-x-x-80">n</span></sup>. Hence, for a proper
     behavior, <span
class="zptmcm7m-x-x-109">m </span>must be larger than <span
class="zptmcm7m-x-x-109">n</span>.
     </li>
     <li class="itemize"><a
 id="dx1-30003"></a> <span
class="ot1-zi4r-0-x-x-120">cutoff3 </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Reference distance vector (Å)<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">coordNum</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>&#8220;<span
class="ot1-zi4r-0-x-x-109">(x, y, z)</span>&#8221; triplet of positive decimals <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">(4.0, 4.0, 4.0)</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>The three components of this vector define three different cutoffs <span
class="zptmcm7m-x-x-109">d</span><sub><span
class="zptmcm7t-x-x-80">0</span></sub> for each direction.
     This option is mutually exclusive with <span
class="ot1-zi4r-0-x-x-109">cutoff</span>.
     </li>
     <li class="itemize"><a
 id="dx1-30004"></a> <span
class="ot1-zi4r-0-x-x-120">expNumer </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Numerator exponent<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">coordNum</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>positive even integer <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span>6 <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>This number defines the <span
class="zptmcm7m-x-x-109">n </span>exponent for the switching function.
     </li>
     <li class="itemize"><a
 id="dx1-30005"></a> <span
class="ot1-zi4r-0-x-x-120">expDenom </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Denominator exponent<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">coordNum</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>positive even integer <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span>12 <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>This number defines the <span
class="zptmcm7m-x-x-109">m </span>exponent for the switching function.
     </li>
     <li class="itemize"><a
 id="dx1-30006"></a> <span
class="ot1-zi4r-0-x-x-120">group2CenterOnly </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Use only <span
class="ot1-zi4r-0-x-x-109">group2</span>&#8217;s center of mass<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">coordNum</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">off</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>If this option is <span
class="ot1-zi4r-0-x-x-109">on</span>, only contacts between each atoms in <span
class="ot1-zi4r-0-x-x-109">group1 </span>and the center of mass
     of <span
class="ot1-zi4r-0-x-x-109">group2 </span>are calculated (by default, the sum extends over all pairs of atoms in <span
class="ot1-zi4r-0-x-x-109">group1 </span>and <span
class="ot1-zi4r-0-x-x-109">group2</span>).
     If <span
class="ot1-zi4r-0-x-x-109">group2 </span>is a <span
class="ot1-zi4r-0-x-x-109">dummyAtom</span>, this option is set to <span
class="ot1-zi4r-0-x-x-109">yes </span>by default.
     </li></ul>


<!--l. 1568--><p class="indent" >   This component returns a dimensionless number, which ranges from approximately 0 (all interatomic distances
are much larger than the cutoff) to <span
class="zptmcm7m-x-x-109">N</span><sub><span
class="cmtt-8">group1</span></sub><span
class="zptmcm7y-x-x-109">&#x00D7;</span><span
class="zptmcm7m-x-x-109">N</span><sub><span
class="cmtt-8">group2</span></sub> (all distances are less than the cutoff), or <span
class="zptmcm7m-x-x-109">N</span><sub><span
class="cmtt-8">group1</span></sub>
if <span
class="ot1-zi4r-0-x-x-109">group2CenterOnly </span>is used. For performance reasons, at least one of <span
class="ot1-zi4r-0-x-x-109">group1 </span>and <span
class="ot1-zi4r-0-x-x-109">group2 </span>should
be of limited size or <span
class="ot1-zi4r-0-x-x-109">group2CenterOnly </span>should be used: the cost of the loop over all pairs grows as
<span
class="zptmcm7m-x-x-109">N</span><sub><span
class="cmtt-8">group1</span></sub><span
class="zptmcm7y-x-x-109">&#x00D7;</span><span
class="zptmcm7m-x-x-109">N</span><sub><span
class="cmtt-8">group2</span></sub>.
<!--l. 1577--><p class="noindent" >
   <h5 class="subsubsectionHead"><span class="titlemark">5.1.11    </span> <a
 id="x1-310005.1.11"></a><span
class="ot1-zi4r-0-x-x-109">selfCoordNum</span>: coordination number between atoms within a group.</h5>
<!--l. 1578--><p class="noindent" >The <span
class="ot1-zi4r-0-x-x-109">selfCoordNum </span><span
class="cmsy-10x-x-109">{</span><span
class="ot1-zi4r-0-x-x-109">...</span><span
class="cmsy-10x-x-109">} </span>block defines a coordination number similarly to the component <span
class="ot1-zi4r-0-x-x-109">coordNum</span>, but the
function is summed over atom pairs within <span
class="ot1-zi4r-0-x-x-109">group1</span>:
   <table
class="equation"><tr><td><a
 id="x1-31001r4"></a>
   <center class="math-display" >
<img
src="colvars-refman-lammps7x.png" alt="                     1---(|xi--xj|&#x2215;d0)n
C(group1) = i&#x2208;g&#x2211;roup1&#x2211;j&#x003E;i1 - (|xi- xj|&#x2215;d0)m
" class="math-display" ></center></td><td class="equation-label">(4)</td></tr></table>
<!--l. 1588--><p class="nopar" >
The keywords accepted by <span
class="ot1-zi4r-0-x-x-109">selfCoordNum </span>are a subset of those accepted by <span
class="ot1-zi4r-0-x-x-109">coordNum</span>, namely <span
class="ot1-zi4r-0-x-x-109">group1 </span>(here
defining <span
class="ptmri7t-x-x-109">all </span>of the atoms to be considered), <span
class="ot1-zi4r-0-x-x-109">cutoff</span>, <span
class="ot1-zi4r-0-x-x-109">expNumer</span>, and <span
class="ot1-zi4r-0-x-x-109">expDenom</span>.
<!--l. 1594--><p class="indent" >   This component returns a dimensionless number, which ranges from approximately 0 (all interatomic
distances much larger than the cutoff) to <span
class="zptmcm7m-x-x-109">N</span><sub><span
class="cmtt-8">group1</span></sub><span
class="zptmcm7y-x-x-109">&#x00D7;</span><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">N</span><sub><span
class="cmtt-8">group1</span></sub><span
class="zptmcm7y-x-x-109">-</span><span
class="zptmcm7t-x-x-109">1)</span><span
class="zptmcm7m-x-x-109">&#x2215;</span><span
class="zptmcm7t-x-x-109">2 </span>(all distances within the cutoff). For
performance reasons, <span
class="ot1-zi4r-0-x-x-109">group1 </span>should be of limited size, because the cost of the loop over all pairs grows as
<span
class="zptmcm7m-x-x-109">N</span><sub><span
class="cmtt-8">group1</span></sub><sup><span
class="zptmcm7t-x-x-80">2</span></sup>.
<!--l. 1604--><p class="noindent" >
   <h5 class="subsubsectionHead"><span class="titlemark">5.1.12    </span> <a
 id="x1-320005.1.12"></a><span
class="ot1-zi4r-0-x-x-109">hBond</span>: hydrogen bond between two atoms.</h5>
<!--l. 1604--><p class="noindent" >The <span
class="ot1-zi4r-0-x-x-109">hBond </span><span
class="cmsy-10x-x-109">{</span><span
class="ot1-zi4r-0-x-x-109">...</span><span
class="cmsy-10x-x-109">} </span>block defines a hydrogen bond, implemented as a coordination number (eq.&#x00A0;<a
href="#x1-30001r3">3<!--tex4ht:ref: eq:cvc_coordNum --></a>) between the donor
and the acceptor atoms. Therefore, it accepts the same options <span
class="ot1-zi4r-0-x-x-109">cutoff </span>(with a different default value of 3.3&#x00A0;Å),
<span
class="ot1-zi4r-0-x-x-109">expNumer </span>(with a default value of 6) and <span
class="ot1-zi4r-0-x-x-109">expDenom </span>(with a default value of 8). Unlike <span
class="ot1-zi4r-0-x-x-109">coordNum</span>, it requires two
atom numbers, <span
class="ot1-zi4r-0-x-x-109">acceptor </span>and <span
class="ot1-zi4r-0-x-x-109">donor</span>, to be defined. It returns an adimensional number, with values between 0
(acceptor and donor far outside the cutoff distance) and 1 (acceptor and donor much closer than the
cutoff).
<!--l. 1616--><p class="noindent" >
   <h5 class="subsubsectionHead"><span class="titlemark">5.1.13    </span> <a
 id="x1-330005.1.13"></a><span
class="ot1-zi4r-0-x-x-109">rmsd</span>: root mean square displacement (RMSD) from reference positions.</h5>
<!--l. 1617--><p class="noindent" >The block <span
class="ot1-zi4r-0-x-x-109">rmsd</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="cmsy-10x-x-109">{</span><span
class="ot1-zi4r-0-x-x-109">...</span><span
class="cmsy-10x-x-109">} </span>defines the root mean square replacement (RMSD) of a group of atoms with respect to
a reference structure. For each set of coordinates <span
class="zptmcm7y-x-x-109">{</span><span
class="ptmb7t-x-x-109">x</span><sub><span
class="zptmcm7t-x-x-80">1</span></sub><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">t</span><span
class="zptmcm7t-x-x-109">)</span><span
class="zptmcm7m-x-x-109">,</span><span
class="ptmb7t-x-x-109">x</span><sub><span
class="zptmcm7t-x-x-80">2</span></sub><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">t</span><span
class="zptmcm7t-x-x-109">)</span><span
class="zptmcm7m-x-x-109">,</span><span
class="zptmcm7m-x-x-109">&#x2026;</span><span
class="ptmb7t-x-x-109">x</span><sub><span
class="zptmcm7m-x-x-80">N</span></sub><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">t</span><span
class="zptmcm7t-x-x-109">)</span><span
class="zptmcm7y-x-x-109">}</span>, the colvar component <span
class="ot1-zi4r-0-x-x-109">rmsd</span>
calculates the optimal rotation <span
class="zptmcm7m-x-x-109">U</span><sup><span
class="zptmcm7y-x-x-80">{</span><span
class="ptmb7t-x-x-80">x</span><sub><span
class="zptmcm7m-x-x-60">i</span></sub><span
class="zptmcm7t-x-x-80">(</span><span
class="zptmcm7m-x-x-80">t</span><span
class="zptmcm7t-x-x-80">)</span><span
class="zptmcm7y-x-x-80">}&#x2192;{</span><span
class="ptmb7t-x-x-80">x</span><sub><span
class="zptmcm7m-x-x-60">i</span></sub><sup><span
class="zptmcm7t-x-x-60">(ref)</span></sup><span
class="zptmcm7y-x-x-80">}</span></sup> that best superimposes the coordinates <span
class="zptmcm7y-x-x-109">{</span><span
class="ptmb7t-x-x-109">x</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">t</span><span
class="zptmcm7t-x-x-109">)</span><span
class="zptmcm7y-x-x-109">} </span>onto a


set of reference coordinates <span
class="zptmcm7y-x-x-109">{</span><span
class="ptmb7t-x-x-109">x</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub><sup><span
class="zptmcm7t-x-x-80">(ref)</span></sup><span
class="zptmcm7y-x-x-109">}</span>. Both the current and the reference coordinates are centered on
their centers of geometry, <span
class="ptmb7t-x-x-109">x</span><sub><span
class="zptmcm7t-x-x-80">cog</span></sub><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">t</span><span
class="zptmcm7t-x-x-109">) </span>and <span
class="ptmb7t-x-x-109">x</span><sub><span
class="zptmcm7t-x-x-80">cog</span></sub><sup><span
class="zptmcm7t-x-x-80">(ref)</span></sup>. The root mean square displacement is then defined
as:
   <table
class="equation"><tr><td><a
 id="x1-33001r5"></a>
   <center class="math-display" >
<img
src="colvars-refman-lammps8x.png" alt="                       &#x2218; --------------------------------------
                (ref)       1-N ||  (           )  (  (ref)   (ref))||2
RMSD  ({xi(t)},{xi  })=    N &#x2211; |U  xi(t)- xcog(t)  -  xi  - xcog  |
                            i=1
" class="math-display" ></center></td><td class="equation-label">(5)</td></tr></table>
<!--l. 1638--><p class="nopar" >
The optimal rotation <span
class="zptmcm7m-x-x-109">U</span><sup><span
class="zptmcm7y-x-x-80">{</span><span
class="ptmb7t-x-x-80">x</span><sub><span
class="zptmcm7m-x-x-60">i</span></sub><span
class="zptmcm7t-x-x-80">(</span><span
class="zptmcm7m-x-x-80">t</span><span
class="zptmcm7t-x-x-80">)</span><span
class="zptmcm7y-x-x-80">}&#x2192;{</span><span
class="ptmb7t-x-x-80">x</span><sub><span
class="zptmcm7m-x-x-60">i</span></sub><sup><span
class="zptmcm7t-x-x-60">(ref)</span></sup><span
class="zptmcm7y-x-x-80">}</span></sup> is calculated within the formalism developed in reference&#x00A0;<span class="cite">[<a
href="#XCoutsias2004">3</a>]</span>, which
guarantees a continuous dependence of <span
class="zptmcm7m-x-x-109">U</span><sup><span
class="zptmcm7y-x-x-80">{</span><span
class="ptmb7t-x-x-80">x</span><sub><span
class="zptmcm7m-x-x-60">i</span></sub><span
class="zptmcm7t-x-x-80">(</span><span
class="zptmcm7m-x-x-80">t</span><span
class="zptmcm7t-x-x-80">)</span><span
class="zptmcm7y-x-x-80">}&#x2192;{</span><span
class="ptmb7t-x-x-80">x</span><sub><span
class="zptmcm7m-x-x-60">i</span></sub><sup><span
class="zptmcm7t-x-x-60">(ref)</span></sup><span
class="zptmcm7y-x-x-80">}</span></sup> with respect to <span
class="zptmcm7y-x-x-109">{</span><span
class="ptmb7t-x-x-109">x</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">t</span><span
class="zptmcm7t-x-x-109">)</span><span
class="zptmcm7y-x-x-109">}</span>. The options for <span
class="ot1-zi4r-0-x-x-109">rmsd</span>
are:
     <ul class="itemize1">
     <li class="itemize"><a
 id="dx1-33002"></a> <span
class="ot1-zi4r-0-x-x-120">atoms </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Atom group<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">rmsd</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span><span
class="ot1-zi4r-0-x-x-109">atoms</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="cmsy-10x-x-109">{</span><span
class="ot1-zi4r-0-x-x-109">...</span><span
class="cmsy-10x-x-109">} </span>block <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:   </span>Defines  the  group  of  atoms  of  which  the  RMSD  should  be  calculated.  Optimal  fit
     options  (such  as  <span
class="ot1-zi4r-0-x-x-109">refPositions </span>and  <span
class="ot1-zi4r-0-x-x-109">rotateReference</span>)  should  typically  NOT  be  set  within  this
     block. Exceptions to this rule are the special cases discussed in the <span
class="ptmri7t-x-x-109">Advanced usage </span>paragraph below.
     </li>
     <li class="itemize"><a
 id="dx1-33003"></a> <span
class="ot1-zi4r-0-x-x-120">refPositions </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Reference coordinates<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">rmsd</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>space-separated list of <span
class="ot1-zi4r-0-x-x-109">(x, y, z) </span>triplets <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>This option sets the reference coordinates. If only <span
class="ot1-zi4r-0-x-x-109">centerReference </span>is <span
class="ot1-zi4r-0-x-x-109">on</span>, the list can
     be a single (x, y, z) triplet; if also <span
class="ot1-zi4r-0-x-x-109">rotateReference </span>is <span
class="ot1-zi4r-0-x-x-109">on</span>, the list should be as long as the atom
     group. This option is independent from that with the same keyword within the <span
class="ot1-zi4r-0-x-x-109">atoms</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="cmsy-10x-x-109">{</span><span
class="ot1-zi4r-0-x-x-109">...</span><span
class="cmsy-10x-x-109">} </span>block
     (see <a
href="#x1-140004">4<!--tex4ht:ref: sec:colvar_atom_groups --></a>). The latter (and related fitting options for the atom group) are normally not needed, and should
     be omitted altogether except for advanced usage cases.
     </li></ul>
<!--l. 1715--><p class="noindent" >This component returns a positive real number (in Å).
<!--l. 1717--><p class="noindent" >
   <h5 class="subsubsectionHead"><span class="titlemark">5.1.14    </span> <a
 id="x1-340005.1.14"></a>Advanced usage of the <span
class="ot1-zi4r-0-x-x-109">rmsd </span>component.</h5>
<!--l. 1718--><p class="noindent" >In the standard usage as described above, the <span
class="ot1-zi4r-0-x-x-109">rmsd </span>component calculates a minimum RMSD, that is, current
coordinates are optimally fitted onto the same reference coordinates that are used to compute the RMSD value. The
fit itself is handled by the atom group object, whose parameters are automatically set by the <span
class="ot1-zi4r-0-x-x-109">rmsd </span>component. For
very specific applications, however, it may be useful to control the fitting process separately from the definition of


the reference coordinates, to evaluate various types of non-minimal RMSD values. This can be achieved by setting
the related options (<span
class="ot1-zi4r-0-x-x-109">refPositions</span>, etc.) explicitly in the atom group block. This allows for the following
non-standard cases:
<!--l. 1731--><p class="indent" >
     <ol  class="enumerate1" >
     <li
  class="enumerate" id="x1-34002x1">applying the optimal translation, but no rotation (<span
class="ot1-zi4r-0-x-x-109">rotateReference off</span>), to bias or restrain the shape
     and orientation, but not the position of the atom group;
     </li>
     <li
  class="enumerate" id="x1-34004x2">applying the optimal rotation, but no translation (<span
class="ot1-zi4r-0-x-x-109">translateReference off</span>), to bias or restrain the
     shape and position, but not the orientation of the atom group;
     </li>
     <li
  class="enumerate" id="x1-34006x3">disabling the application of optimal roto-translations, which lets the RMSD component decribe the
     deviation of atoms from fixed positions in the laboratory frame: this allows for custom positional
     restraints within the colvars module;
     </li>
     <li
  class="enumerate" id="x1-34008x4">fitting the atomic positions to different reference coordinates than those used in the RMSD calculation
     itself;
     </li>
     <li
  class="enumerate" id="x1-34010x5">applying  the  optimal  rotation  and/or  translation  from  a  separate  atom  group,  defined  through
     <span
class="ot1-zi4r-0-x-x-109">refPositionsGroup</span>: the RMSD then reflects the deviation from reference coordinates in a separate,
     moving reference frame.</li></ol>
<!--l. 1751--><p class="noindent" >
   <h5 class="subsubsectionHead"><span class="titlemark">5.1.15    </span> <a
 id="x1-350005.1.15"></a><span
class="ot1-zi4r-0-x-x-109">eigenvector</span>: projection of the atomic coordinates on a vector.</h5>
<!--l. 1752--><p class="noindent" >The block <span
class="ot1-zi4r-0-x-x-109">eigenvector</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="cmsy-10x-x-109">{</span><span
class="ot1-zi4r-0-x-x-109">...</span><span
class="cmsy-10x-x-109">} </span>defines the projection of the coordinates of a group of atoms (or more precisely,
their deviations from the reference coordinates) onto a vector in <span
class="msbm-10x-x-109">&#x211D;</span><sup><span
class="zptmcm7t-x-x-80">3</span><span
class="zptmcm7m-x-x-80">n</span></sup>, where <span
class="zptmcm7m-x-x-109">n </span>is the number of atoms in the group.
The computed quantity is the total projection:
   <table
class="equation"><tr><td><a
 id="x1-35001r6"></a>
   <center class="math-display" >
<img
src="colvars-refman-lammps9x.png" alt="                   n   (                             )
p({xi(t)},{x(ref)}) =   vi&#x22C5; U (xi(t)- xcog(t)) - (x(ref)- x(rcoefg)) ,
          i       &#x2211;i=1                       i
" class="math-display" ></center></td><td class="equation-label">(6)</td></tr></table>
<!--l. 1765--><p class="nopar" >
where, as in the <span
class="ot1-zi4r-0-x-x-109">rmsd </span>component, <span
class="zptmcm7m-x-x-109">U </span>is the optimal rotation matrix, <span
class="ptmb7t-x-x-109">x</span><sub><span
class="zptmcm7t-x-x-80">cog</span></sub><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">t</span><span
class="zptmcm7t-x-x-109">) </span>and <span
class="ptmb7t-x-x-109">x</span><sub><span
class="zptmcm7t-x-x-80">cog</span></sub><sup><span
class="zptmcm7t-x-x-80">(ref)</span></sup> are the centers of geometry of
the current and reference positions respectively, and <span
class="ptmb7t-x-x-109">v</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub> are the components of the vector for each atom. Example


choices for <span
class="zptmcm7t-x-x-109">(</span><span
class="ptmb7t-x-x-109">v</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub><span
class="zptmcm7t-x-x-109">) </span>are an eigenvector of the covariance matrix (essential mode), or a normal mode of the system. It is
assumed that <span
class="zptmcm7v-x-x-109">&#x2211;</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub><span
class="ptmb7t-x-x-109">v</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub> <span
class="zptmcm7t-x-x-109">= 0</span>: otherwise, the colvars module centers the <span
class="ptmb7t-x-x-109">v</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub> automatically when reading them from the
configuration.
<!--l. 1777--><p class="indent" >   As for the component <span
class="ot1-zi4r-0-x-x-109">rmsd</span>, the available options are <span
class="ot1-zi4r-0-x-x-109">atoms </span>and <span
class="ot1-zi4r-0-x-x-109">refPositions</span>. In addition, the following are
recognized:
     <ul class="itemize1">
     <li class="itemize"><a
 id="dx1-35002"></a> <span
class="ot1-zi4r-0-x-x-120">vector </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Vector components<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">eigenvector</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>space-separated list of <span
class="ot1-zi4r-0-x-x-109">(x, y, z) </span>triplets <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>This option  sets the values of the vector components.
     </li>
     <li class="itemize"><a
 id="dx1-35003"></a>  <span
class="ot1-zi4r-0-x-x-120">differenceVector   </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>The  <span
class="zptmcm7t-x-x-109">3</span><span
class="zptmcm7m-x-x-109">n</span>-dimensional  vector  is  the  difference  between  <span
class="ot1-zi4r-0-x-x-109">vector </span>and
     <span
class="ot1-zi4r-0-x-x-109">refPositions</span><span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">eigenvector</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">off</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>If this option is <span
class="ot1-zi4r-0-x-x-109">on</span>, the numbers provided by <span
class="ot1-zi4r-0-x-x-109">vector </span>are interpreted as another set of
     positions, <span
class="ptmb7t-x-x-109">x</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub><span
class="zptmcm7y-x-x-109">&#x2032;</span>: the vector <span
class="ptmb7t-x-x-109">v</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub> is then defined as <span
class="ptmb7t-x-x-109">v</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub> <span
class="zptmcm7t-x-x-109">= </span><img
src="colvars-refman-lammps10x.png" alt="(     (ref))
 x&#x2032;i- xi"  class="left" align="middle">. This allows to conveniently define a
     colvar <span
class="zptmcm7m-x-x-109">&#x03BE; </span>as a projection on the linear transformation between two sets of positions, &#8220;A&#8221; and &#8220;B&#8221;. For
     convenience, the vector is also normalized so that <span
class="zptmcm7m-x-x-109">&#x03BE; </span><span
class="zptmcm7t-x-x-109">= 0 </span>when the atoms are at the set of positions &#8220;A&#8221;
     and <span
class="zptmcm7m-x-x-109">&#x03BE; </span><span
class="zptmcm7t-x-x-109">= 1 </span>at the set of positions &#8220;B&#8221;.
     </li></ul>
<!--l. 1834--><p class="noindent" >This component returns a number (in Å), whose value ranges between the smallest and largest absolute positions in the
unit cell during the simulations (see also <span
class="ot1-zi4r-0-x-x-109">distanceZ</span>). Due to the normalization in eq.&#x00A0;<a
href="#x1-35001r6">6<!--tex4ht:ref: eq:cvc_eigenvector --></a>, this range does not depend
on the number of atoms involved.
<!--l. 1841--><p class="noindent" >
   <h5 class="subsubsectionHead"><span class="titlemark">5.1.16    </span> <a
 id="x1-360005.1.16"></a><span
class="ot1-zi4r-0-x-x-109">gyration</span>: radius of gyration of a group of atoms.</h5>
<!--l. 1842--><p class="noindent" >The block <span
class="ot1-zi4r-0-x-x-109">gyration</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="cmsy-10x-x-109">{</span><span
class="ot1-zi4r-0-x-x-109">...</span><span
class="cmsy-10x-x-109">} </span>defines the parameters for calculating the radius of gyration of a group of atomic
positions <span
class="zptmcm7y-x-x-109">{</span><span
class="ptmb7t-x-x-109">x</span><sub><span
class="zptmcm7t-x-x-80">1</span></sub><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">t</span><span
class="zptmcm7t-x-x-109">)</span><span
class="zptmcm7m-x-x-109">,</span><span
class="ptmb7t-x-x-109">x</span><sub><span
class="zptmcm7t-x-x-80">2</span></sub><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">t</span><span
class="zptmcm7t-x-x-109">)</span><span
class="zptmcm7m-x-x-109">,</span><span
class="zptmcm7m-x-x-109">&#x2026;</span><span
class="ptmb7t-x-x-109">x</span><sub><span
class="zptmcm7m-x-x-80">N</span></sub><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">t</span><span
class="zptmcm7t-x-x-109">)</span><span
class="zptmcm7y-x-x-109">} </span>with respect to their center of geometry, <span
class="ptmb7t-x-x-109">x</span><sub><span
class="zptmcm7t-x-x-80">cog</span></sub><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">t</span><span
class="zptmcm7t-x-x-109">)</span>:
   <table
class="equation"><tr><td><a
 id="x1-36001r7"></a>
   <center class="math-display" >
<img
src="colvars-refman-lammps11x.png" alt="      &#x2218; -------------------
        -1 N ||           ||2
Rgyr =  N &#x2211;   xi(t)- xcog(t)
          i=1
" class="math-display" ></center></td><td class="equation-label">(7)</td></tr></table>


<!--l. 1852--><p class="nopar" >
This component must contain one <span
class="ot1-zi4r-0-x-x-109">atoms</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="cmsy-10x-x-109">{</span><span
class="ot1-zi4r-0-x-x-109">...</span><span
class="cmsy-10x-x-109">} </span>block to define the atom group, and returns a positive number,
expressed in Å.
<!--l. 1858--><p class="noindent" >
   <h5 class="subsubsectionHead"><span class="titlemark">5.1.17    </span> <a
 id="x1-370005.1.17"></a><span
class="ot1-zi4r-0-x-x-109">inertia</span>: total moment of inertia of a group of atoms.</h5>
<!--l. 1859--><p class="noindent" >The block <span
class="ot1-zi4r-0-x-x-109">inertia</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="cmsy-10x-x-109">{</span><span
class="ot1-zi4r-0-x-x-109">...</span><span
class="cmsy-10x-x-109">} </span>defines the parameters for calculating the total moment of inertia of a group of atomic
positions <span
class="zptmcm7y-x-x-109">{</span><span
class="ptmb7t-x-x-109">x</span><sub><span
class="zptmcm7t-x-x-80">1</span></sub><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">t</span><span
class="zptmcm7t-x-x-109">)</span><span
class="zptmcm7m-x-x-109">,</span><span
class="ptmb7t-x-x-109">x</span><sub><span
class="zptmcm7t-x-x-80">2</span></sub><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">t</span><span
class="zptmcm7t-x-x-109">)</span><span
class="zptmcm7m-x-x-109">,</span><span
class="zptmcm7m-x-x-109">&#x2026;</span><span
class="ptmb7t-x-x-109">x</span><sub><span
class="zptmcm7m-x-x-80">N</span></sub><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">t</span><span
class="zptmcm7t-x-x-109">)</span><span
class="zptmcm7y-x-x-109">} </span>with respect to their center of geometry, <span
class="ptmb7t-x-x-109">x</span><sub><span
class="zptmcm7t-x-x-80">cog</span></sub><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">t</span><span
class="zptmcm7t-x-x-109">)</span>:
   <table
class="equation"><tr><td><a
 id="x1-37001r8"></a>
   <center class="math-display" >
<img
src="colvars-refman-lammps12x.png" alt="     N
I =    ||x (t)- x   (t)||2
    &#x2211;i=1  i     cog
" class="math-display" ></center></td><td class="equation-label">(8)</td></tr></table>
<!--l. 1868--><p class="nopar" >
<span
class="ptmri7t-x-x-109">Note that all atomic masses are set to 1 for simplicity. </span>This component must contain one <span
class="ot1-zi4r-0-x-x-109">atoms</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="cmsy-10x-x-109">{</span><span
class="ot1-zi4r-0-x-x-109">...</span><span
class="cmsy-10x-x-109">} </span>block to
define the atom group, and returns a positive number, expressed in Å<sup><span
class="zptmcm7t-x-x-80">2</span></sup>.
<!--l. 1875--><p class="noindent" >
   <h5 class="subsubsectionHead"><span class="titlemark">5.1.18    </span> <a
 id="x1-380005.1.18"></a><span
class="ot1-zi4r-0-x-x-109">inertiaZ</span>: total moment of inertia of a group of atoms around a chosen axis.</h5>
<!--l. 1876--><p class="noindent" >The block <span
class="ot1-zi4r-0-x-x-109">inertiaZ</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="cmsy-10x-x-109">{</span><span
class="ot1-zi4r-0-x-x-109">...</span><span
class="cmsy-10x-x-109">} </span>defines the parameters for calculating the component along the axis <span
class="ptmb7t-x-x-109">e </span>of the moment of
inertia of a group of atomic positions <span
class="zptmcm7y-x-x-109">{</span><span
class="ptmb7t-x-x-109">x</span><sub><span
class="zptmcm7t-x-x-80">1</span></sub><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">t</span><span
class="zptmcm7t-x-x-109">)</span><span
class="zptmcm7m-x-x-109">,</span><span
class="ptmb7t-x-x-109">x</span><sub><span
class="zptmcm7t-x-x-80">2</span></sub><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">t</span><span
class="zptmcm7t-x-x-109">)</span><span
class="zptmcm7m-x-x-109">,</span><span
class="zptmcm7m-x-x-109">&#x2026;</span><span
class="ptmb7t-x-x-109">x</span><sub><span
class="zptmcm7m-x-x-80">N</span></sub><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">t</span><span
class="zptmcm7t-x-x-109">)</span><span
class="zptmcm7y-x-x-109">} </span>with respect to their center of geometry,
<span
class="ptmb7t-x-x-109">x</span><sub><span
class="zptmcm7t-x-x-80">cog</span></sub><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">t</span><span
class="zptmcm7t-x-x-109">)</span>:
   <table
class="equation"><tr><td><a
 id="x1-38001r9"></a>
   <center class="math-display" >
<img
src="colvars-refman-lammps13x.png" alt="     N ((           )  )
Ie = &#x2211;   xi(t)- xcog(t) &#x22C5;e 2
    i=1
" class="math-display" ></center></td><td class="equation-label">(9)</td></tr></table>
<!--l. 1885--><p class="nopar" >
<span
class="ptmri7t-x-x-109">Note that all atomic masses are set to 1 for simplicity. </span>This component must contain one <span
class="ot1-zi4r-0-x-x-109">atoms</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="cmsy-10x-x-109">{</span><span
class="ot1-zi4r-0-x-x-109">...</span><span
class="cmsy-10x-x-109">} </span>block to
define the atom group, and returns a positive number, expressed in Å<sup><span
class="zptmcm7t-x-x-80">2</span></sup>. The following option may also be


provided:
     <ul class="itemize1">
     <li class="itemize"><a
 id="dx1-38002"></a> <span
class="ot1-zi4r-0-x-x-120">axis </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Projection axis (Å)<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">inertiaZ</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span><span
class="ot1-zi4r-0-x-x-109">(x, y, z) </span>triplet <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">(0.0, 0.0, 1.0)</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>The three components of this vector define (when normalized) the projection axis <span
class="ptmb7t-x-x-109">e</span>.</li></ul>
<!--l. 1903--><p class="noindent" >
   <h5 class="subsubsectionHead"><span class="titlemark">5.1.19    </span> <a
 id="x1-390005.1.19"></a><span
class="ot1-zi4r-0-x-x-109">orientation</span>: orientation from reference coordinates.</h5>
<!--l. 1904--><p class="noindent" >The block <span
class="ot1-zi4r-0-x-x-109">orientation</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="cmsy-10x-x-109">{</span><span
class="ot1-zi4r-0-x-x-109">...</span><span
class="cmsy-10x-x-109">} </span>returns the same optimal rotation used in the <span
class="ot1-zi4r-0-x-x-109">rmsd </span>component to superimpose
the coordinates <span
class="zptmcm7y-x-x-109">{</span><span
class="ptmb7t-x-x-109">x</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">t</span><span
class="zptmcm7t-x-x-109">)</span><span
class="zptmcm7y-x-x-109">} </span>onto a set of reference coordinates <span
class="zptmcm7y-x-x-109">{</span><span
class="ptmb7t-x-x-109">x</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub><sup><span
class="zptmcm7t-x-x-80">(ref)</span></sup><span
class="zptmcm7y-x-x-109">}</span>. Such component returns a four
dimensional vector <span
class="cmss-10x-x-109">q</span> <span
class="zptmcm7t-x-x-109">= (</span><span
class="zptmcm7m-x-x-109">q</span><sub><span
class="zptmcm7t-x-x-80">0</span></sub><span
class="zptmcm7m-x-x-109">,q</span><sub><span
class="zptmcm7t-x-x-80">1</span></sub><span
class="zptmcm7m-x-x-109">,q</span><sub><span
class="zptmcm7t-x-x-80">2</span></sub><span
class="zptmcm7m-x-x-109">,q</span><sub><span
class="zptmcm7t-x-x-80">3</span></sub><span
class="zptmcm7t-x-x-109">)</span>, with <span
class="zptmcm7v-x-x-109">&#x2211;</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub><span
class="zptmcm7m-x-x-109">q</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub><sup><span
class="zptmcm7t-x-x-80">2</span></sup> <span
class="zptmcm7t-x-x-109">= 1</span>; this <span
class="ptmri7t-x-x-109">quaternion </span>expresses the optimal rotation
<span
class="zptmcm7y-x-x-109">{</span><span
class="ptmb7t-x-x-109">x</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">t</span><span
class="zptmcm7t-x-x-109">)</span><span
class="zptmcm7y-x-x-109">}&#x2192;{</span><span
class="ptmb7t-x-x-109">x</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub><sup><span
class="zptmcm7t-x-x-80">(ref)</span></sup><span
class="zptmcm7y-x-x-109">} </span>according to the formalism in reference&#x00A0;<span class="cite">[<a
href="#XCoutsias2004">3</a>]</span>. The quaternion <span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">q</span><sub><span
class="zptmcm7t-x-x-80">0</span></sub><span
class="zptmcm7m-x-x-109">,q</span><sub><span
class="zptmcm7t-x-x-80">1</span></sub><span
class="zptmcm7m-x-x-109">,q</span><sub><span
class="zptmcm7t-x-x-80">2</span></sub><span
class="zptmcm7m-x-x-109">,q</span><sub><span
class="zptmcm7t-x-x-80">3</span></sub><span
class="zptmcm7t-x-x-109">) </span>can also be written as
<img
src="colvars-refman-lammps14x.png" alt="(cos(&#x03B8;&#x2215;2),sin(&#x03B8; &#x2215;2)u)"  class="left" align="middle">, where <span
class="zptmcm7m-x-x-109">&#x03B8; </span>is the angle and <span
class="ptmb7t-x-x-109">u </span>the normalized axis of rotation; for example, a rotation of 90<sup><span
class="zptmcm7y-x-x-80">&#x2218;</span></sup>
around the <span
class="zptmcm7m-x-x-109">z </span>axis is expressed as &#8220;<span
class="ot1-zi4r-0-x-x-109">(0.707, 0.0, 0.0, 0.707)</span>&#8221;. The script <span
class="ot1-zi4r-0-x-x-109">quaternion2rmatrix.tcl</span>
provides Tcl functions for converting to and from a <span
class="zptmcm7t-x-x-109">4</span><span
class="zptmcm7y-x-x-109">&#x00D7;</span><span
class="zptmcm7t-x-x-109">4 </span>rotation matrix in a format suitable for usage in
VMD.
<!--l. 1922--><p class="indent" >   As for the component <span
class="ot1-zi4r-0-x-x-109">rmsd</span>, the available options are <span
class="ot1-zi4r-0-x-x-109">atoms </span>and <span
class="ot1-zi4r-0-x-x-109">refPositions</span>.
<!--l. 1924--><p class="indent" >   <span
class="ptmb7t-x-x-109">Note: </span><span
class="ot1-zi4r-0-x-x-109">refPositions </span>define the set of positions <span
class="ptmri7t-x-x-109">from which </span>the optimal rotation is calculated, but this
rotation is not applied to the coordinates of the atoms involved: it is used instead to define the variable
itself.
     <ul class="itemize1">
     <li class="itemize"><a
 id="dx1-39001"></a> <span
class="ot1-zi4r-0-x-x-120">closestToQuaternion </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Reference rotation<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">orientation</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>&#8220;<span
class="ot1-zi4r-0-x-x-109">(q0, q1, q2, q3)</span>&#8221; quadruplet <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">(1.0, 0.0, 0.0, 0.0) </span>(&#8220;null&#8221; rotation) <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>Between the two equivalent quaternions <span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">q</span><sub><span
class="zptmcm7t-x-x-80">0</span></sub><span
class="zptmcm7m-x-x-109">,q</span><sub><span
class="zptmcm7t-x-x-80">1</span></sub><span
class="zptmcm7m-x-x-109">,q</span><sub><span
class="zptmcm7t-x-x-80">2</span></sub><span
class="zptmcm7m-x-x-109">,q</span><sub><span
class="zptmcm7t-x-x-80">3</span></sub><span
class="zptmcm7t-x-x-109">) </span>and <span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7y-x-x-109">-</span><span
class="zptmcm7m-x-x-109">q</span><sub><span
class="zptmcm7t-x-x-80">0</span></sub><span
class="zptmcm7m-x-x-109">,</span><span
class="zptmcm7y-x-x-109">-</span><span
class="zptmcm7m-x-x-109">q</span><sub><span
class="zptmcm7t-x-x-80">1</span></sub><span
class="zptmcm7m-x-x-109">,</span><span
class="zptmcm7y-x-x-109">-</span><span
class="zptmcm7m-x-x-109">q</span><sub><span
class="zptmcm7t-x-x-80">2</span></sub><span
class="zptmcm7m-x-x-109">,</span><span
class="zptmcm7y-x-x-109">-</span><span
class="zptmcm7m-x-x-109">q</span><sub><span
class="zptmcm7t-x-x-80">3</span></sub><span
class="zptmcm7t-x-x-109">)</span>, the
     closer to <span
class="ot1-zi4r-0-x-x-109">(1.0, 0.0, 0.0, 0.0) </span>is chosen. This simplifies the visualization of the colvar trajectory
     when samples values are a smaller subset of all possible rotations. <span
class="ptmb7t-x-x-109">Note: </span><span
class="ptmri7t-x-x-109">this only affects the output,</span>
     <span
class="ptmri7t-x-x-109">never the dynamics</span>.
     </li></ul>
<!--l. 1944--><p class="indent" >   <span
class="ptmb7t-x-x-109">Hint: stopping the rotation of a protein. </span>To stop the rotation of an elongated macromolecule in solution (and
use an anisotropic box to save water molecules), it is possible to define a colvar with an <span
class="ot1-zi4r-0-x-x-109">orientation </span>component,
and restrain it throuh the <span
class="ot1-zi4r-0-x-x-109">harmonic </span>bias around the identity rotation, <span
class="ot1-zi4r-0-x-x-109">(1.0, 0.0, 0.0, 0.0)</span>. Only
the overall orientation of the macromolecule is affected, and <span
class="ptmri7t-x-x-109">not </span>its internal degrees of freedom. The
user should also take care that the macromolecule is composed by a single chain, or disable <span
class="ot1-zi4r-0-x-x-109">wrapAll</span>
otherwise.
<!--l. 1956--><p class="noindent" >
   <h5 class="subsubsectionHead"><span class="titlemark">5.1.20    </span> <a
 id="x1-400005.1.20"></a><span
class="ot1-zi4r-0-x-x-109">orientationAngle</span>: angle of rotation from reference coordinates.</h5>


<!--l. 1957--><p class="noindent" >The block <span
class="ot1-zi4r-0-x-x-109">orientationAngle</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="cmsy-10x-x-109">{</span><span
class="ot1-zi4r-0-x-x-109">...</span><span
class="cmsy-10x-x-109">} </span>accepts the same base options as the component <span
class="ot1-zi4r-0-x-x-109">orientation</span>: <span
class="ot1-zi4r-0-x-x-109">atoms </span>and
<span
class="ot1-zi4r-0-x-x-109">refPositions</span>. The returned value is the angle of rotation <span
class="zptmcm7m-x-x-109">&#x03B8; </span>between the current and the reference positions. This
angle is expressed in degrees within the range [0<sup><span
class="zptmcm7y-x-x-80">&#x2218;</span></sup>:180<sup><span
class="zptmcm7y-x-x-80">&#x2218;</span></sup>].
<!--l. 1963--><p class="noindent" >
   <h5 class="subsubsectionHead"><span class="titlemark">5.1.21    </span> <a
 id="x1-410005.1.21"></a><span
class="ot1-zi4r-0-x-x-109">orientationProj</span>: cosine of the angle of rotation from reference coordinates.</h5>
<!--l. 1964--><p class="noindent" >The block <span
class="ot1-zi4r-0-x-x-109">orientationProj</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="cmsy-10x-x-109">{</span><span
class="ot1-zi4r-0-x-x-109">...</span><span
class="cmsy-10x-x-109">} </span>accepts the same base options as the component <span
class="ot1-zi4r-0-x-x-109">orientation</span>: <span
class="ot1-zi4r-0-x-x-109">atoms </span>and
<span
class="ot1-zi4r-0-x-x-109">refPositions</span>. The returned value is the cosine of the angle of rotation <span
class="zptmcm7m-x-x-109">&#x03B8; </span>between the current and the reference
positions. The range of values is [-1:1].
<!--l. 1970--><p class="noindent" >
   <h5 class="subsubsectionHead"><span class="titlemark">5.1.22    </span> <a
 id="x1-420005.1.22"></a><span
class="ot1-zi4r-0-x-x-109">spinAngle</span>: angle of rotation around a given axis.</h5>
<!--l. 1971--><p class="noindent" >The complete rotation described by <span
class="ot1-zi4r-0-x-x-109">orientation </span>can optionally be decomposed into two sub-rotations: one is a
&#8220;<span
class="ptmri7t-x-x-109">spin</span>&#8221; rotation around <span
class="ptmb7t-x-x-109">e</span>, and the other a &#8220;<span
class="ptmri7t-x-x-109">tilt</span>&#8221; rotation around an axis orthogonal to <span
class="ptmb7t-x-x-109">e</span>. The component <span
class="ot1-zi4r-0-x-x-109">spinAngle</span>
measures the angle of the &#8220;spin&#8221; sub-rotation around <span
class="ptmb7t-x-x-109">e</span>. This can be defined using the same options as
the component <span
class="ot1-zi4r-0-x-x-109">orientation</span>: <span
class="ot1-zi4r-0-x-x-109">atoms </span>and <span
class="ot1-zi4r-0-x-x-109">refPositions</span>. In addition, <span
class="ot1-zi4r-0-x-x-109">spinAngle </span>accepts the <span
class="ot1-zi4r-0-x-x-109">axis</span>
option:
     <ul class="itemize1">
     <li class="itemize"><a
 id="dx1-42001"></a> <span
class="ot1-zi4r-0-x-x-120">axis </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Special rotation axis (Å)<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">tilt, spinAngle</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span><span
class="ot1-zi4r-0-x-x-109">(x, y, z) </span>triplet <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">(0.0, 0.0, 1.0)</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>The three components of this vector define (when normalized) the special rotation axis
     used to calculate the <span
class="ot1-zi4r-0-x-x-109">tilt </span>and <span
class="ot1-zi4r-0-x-x-109">spinAngle </span>components.</li></ul>
<!--l. 1985--><p class="noindent" >The component <span
class="ot1-zi4r-0-x-x-109">spinAngle </span>returns an angle (in degrees) within the periodic interval <span
class="zptmcm7t-x-x-109">[</span><span
class="zptmcm7y-x-x-109">-</span><span
class="zptmcm7t-x-x-109">180 : 180]</span>.
<!--l. 1987--><p class="indent" >   <span
class="ptmb7t-x-x-109">Note: </span>the value of <span
class="ot1-zi4r-0-x-x-109">spinAngle </span>is a continuous function almost everywhere, with the exception of configurations
with the corresponding &#8220;tilt&#8221; angle equal to 180<sup><span
class="zptmcm7y-x-x-80">&#x2218;</span></sup> (i.e.&#x00A0;the <span
class="ot1-zi4r-0-x-x-109">tilt </span>component is equal to <span
class="zptmcm7y-x-x-109">-</span><span
class="zptmcm7t-x-x-109">1</span>): in those cases,
<span
class="ot1-zi4r-0-x-x-109">spinAngle </span>is undefined. If such configurations are expected, consider defining a <span
class="ot1-zi4r-0-x-x-109">tilt </span>colvar using the same axis <span
class="ptmb7t-x-x-109">e</span>,
and restraining it with a lower wall away from <span
class="zptmcm7y-x-x-109">-</span><span
class="zptmcm7t-x-x-109">1</span>.
<!--l. 1990--><p class="noindent" >
   <h5 class="subsubsectionHead"><span class="titlemark">5.1.23    </span> <a
 id="x1-430005.1.23"></a><span
class="ot1-zi4r-0-x-x-109">tilt</span>: cosine of the rotation orthogonal to a given axis.</h5>
<!--l. 1991--><p class="noindent" >The component <span
class="ot1-zi4r-0-x-x-109">tilt </span>measures the cosine of the angle of the &#8220;tilt&#8221; sub-rotation, which combined with the &#8220;spin&#8221;
sub-rotation provides the complete rotation of a group of atoms. The cosine of the tilt angle rather than the tilt angle
itself is implemented, because the latter is unevenly distributed even for an isotropic system: consider as an analogy
the angle <span
class="zptmcm7m-x-x-109">&#x03B8; </span>in the spherical coordinate system. The component <span
class="ot1-zi4r-0-x-x-109">tilt </span>relies on the same options as <span
class="ot1-zi4r-0-x-x-109">spinAngle</span>,
including the definition of the axis <span
class="ptmb7t-x-x-109">e</span>. The values of <span
class="ot1-zi4r-0-x-x-109">tilt </span>are real numbers in the interval <span
class="zptmcm7t-x-x-109">[</span><span
class="zptmcm7y-x-x-109">-</span><span
class="zptmcm7t-x-x-109">1 : 1]</span>: the value <span
class="zptmcm7t-x-x-109">1</span>
represents an orientation fully parallel to <span
class="ptmb7t-x-x-109">e </span>(tilt angle = 0<sup><span
class="zptmcm7y-x-x-80">&#x2218;</span></sup>), and the value <span
class="zptmcm7y-x-x-109">-</span><span
class="zptmcm7t-x-x-109">1 </span>represents an anti-parallel
orientation.


<!--l. 2208--><p class="noindent" >
   <h4 class="subsectionHead"><span class="titlemark">5.2    </span> <a
 id="x1-440005.2"></a>Advanced usage and special considerations</h4>
<!--l. 2210--><p class="noindent" >
   <h5 class="subsubsectionHead"><span class="titlemark">5.2.1    </span> <a
 id="x1-450005.2.1"></a>Periodic components.</h5>
<!--l. 2211--><p class="noindent" >The following components returns real numbers that lie in a periodic interval:
     <ul class="itemize1">
     <li class="itemize"><span
class="ot1-zi4r-0-x-x-109">dihedral</span>: torsional angle between four groups;
     </li>
     <li class="itemize"><span
class="ot1-zi4r-0-x-x-109">spinAngle</span>:  angle  of  rotation  around  a  predefined  axis  in  the  best-fit  from  a  set  of  reference
     coordinates.</li></ul>
<!--l. 2218--><p class="noindent" >In certain conditions, <span
class="ot1-zi4r-0-x-x-109">distanceZ </span>can also be periodic, namely when periodic boundary conditions (PBCs) are defined in
the simulation and <span
class="ot1-zi4r-0-x-x-109">distanceZ</span>&#8217;s axis is parallel to a unit cell vector.
<!--l. 2222--><p class="indent" >   The following keywords can be used within periodic components (and are illegal elsewhere):
     <ul class="itemize1">
     <li class="itemize"><a
 id="dx1-45001"></a> <span
class="ot1-zi4r-0-x-x-120">period </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Period of the component<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">distanceZ</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>positive decimal <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span>0.0 <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:   </span>Setting  this  number  enables  the  treatment  of  <span
class="ot1-zi4r-0-x-x-109">distanceZ </span>as  a  periodic  component:
     by default, <span
class="ot1-zi4r-0-x-x-109">distanceZ </span>is not considered periodic. The keyword is supported, but irrelevant within
     <span
class="ot1-zi4r-0-x-x-109">dihedral </span>or <span
class="ot1-zi4r-0-x-x-109">spinAngle</span>, because their period is always 360&#x00A0;degrees.
     </li>
     <li class="itemize"><a
 id="dx1-45002"></a> <span
class="ot1-zi4r-0-x-x-120">wrapAround </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Center of the wrapping interval for periodic variables<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">distanceZ</span>, <span
class="ot1-zi4r-0-x-x-109">dihedral </span>or <span
class="ot1-zi4r-0-x-x-109">spinAngle</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>decimal <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span>0.0 <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>By default, values of the periodic components are centered around zero, ranging from
     <span
class="zptmcm7y-x-x-109">-</span><span
class="zptmcm7m-x-x-109">P&#x2215;</span><span
class="zptmcm7t-x-x-109">2 </span>to <span
class="zptmcm7m-x-x-109">P&#x2215;</span><span
class="zptmcm7t-x-x-109">2</span>, where <span
class="zptmcm7m-x-x-109">P </span>is the period. Setting this number centers the interval around this value. This
     can be useful for convenience of output, or to set <span
class="ot1-zi4r-0-x-x-109">lowerWall </span>and <span
class="ot1-zi4r-0-x-x-109">upperWall </span>in an order that would
     not otherwise be allowed.</li></ul>
<!--l. 2254--><p class="noindent" >Internally, all differences between two values of a periodic colvar follow the minimum image convention: they are
calculated based on the two periodic images that are closest to each other.
<!--l. 2262--><p class="indent" >   <span
class="ptmri7t-x-x-109">Note: linear or polynomial combinations of periodic components may become meaningless when components</span>
<span
class="ptmri7t-x-x-109">cross the periodic boundary. Use such combinations carefully: estimate the range of possible values of</span>
<span
class="ptmri7t-x-x-109">each component in a given simulation, and make use of  </span><span
class="ot1-zi4r-0-x-x-109">wrapAround </span><span
class="ptmri7t-x-x-109">to limit this problem whenever</span>
<span
class="ptmri7t-x-x-109">possible.</span>


<!--l. 2265--><p class="noindent" >
   <h5 class="subsubsectionHead"><span class="titlemark">5.2.2    </span> <a
 id="x1-460005.2.2"></a>Non-scalar components.</h5>
<!--l. 2266--><p class="noindent" >When one of the following components are used, the defined colvar returns a value that is not a scalar
number:
     <ul class="itemize1">
     <li class="itemize"><span
class="ot1-zi4r-0-x-x-109">distanceVec</span>: 3-dimensional vector of the distance between two groups;
     </li>
     <li class="itemize"><span
class="ot1-zi4r-0-x-x-109">distanceDir</span>: 3-dimensional unit vector of the distance between two groups;
     </li>
     <li class="itemize"><span
class="ot1-zi4r-0-x-x-109">orientation</span>: 4-dimensional unit quaternion representing the best-fit rotation from a set of reference
     coordinates.</li></ul>
<!--l. 2275--><p class="noindent" >The distance between two 3-dimensional unit vectors is computed as the angle between them. The distance between two
quaternions is computed as the angle between the two 4-dimensional unit vectors: because the orientation
represented by <span
class="cmss-10x-x-109">q </span>is the same as the one represented by <span
class="zptmcm7y-x-x-109">-</span><span
class="cmss-10x-x-109">q</span>, distances between two quaternions are computed
considering the closest of the two symmetric images.
<!--l. 2282--><p class="indent" >   Non-scalar components carry the following restrictions:
     <ul class="itemize1">
     <li class="itemize">Calculation of system forces (<span
class="ot1-zi4r-0-x-x-109">outputSystemForce </span>option) is currently not implemented.
     </li>
     <li class="itemize">Each colvar can only contain one non-scalar component.
     </li>
     <li class="itemize">Binning  on  a  grid  (<span
class="ot1-zi4r-0-x-x-109">abf</span>,  <span
class="ot1-zi4r-0-x-x-109">histogram </span>and  <span
class="ot1-zi4r-0-x-x-109">metadynamics </span>with  <span
class="ot1-zi4r-0-x-x-109">useGrids </span>enabled)  is  currently  not
     implemented for colvars based on such components.</li></ul>
<!--l. 2295--><p class="indent" >   <span
class="ptmri7t-x-x-109">Note: while these restrictions apply to individual colvars based on non-scalar components, no limit is set to the</span>
<span
class="ptmri7t-x-x-109">number of scalar colvars. To compute multi-dimensional histograms and PMFs, use sets of scalar colvars of</span>
<span
class="ptmri7t-x-x-109">arbitrary size.</span>
<!--l. 2298--><p class="noindent" >
   <h5 class="subsubsectionHead"><span class="titlemark">5.2.3    </span> <a
 id="x1-470005.2.3"></a>Calculating system forces.</h5>
<!--l. 2299--><p class="noindent" >In addition to the restrictions due to the type of value computed (scalar or non-scalar), a final restriction can arise
when calculating system force (<span
class="ot1-zi4r-0-x-x-109">outputSystemForce </span>option or application of a <span
class="ot1-zi4r-0-x-x-109">abf </span>bias). System forces are
available currently only for the following components: <span
class="ot1-zi4r-0-x-x-109">distance</span>, <span
class="ot1-zi4r-0-x-x-109">distanceZ</span>, <span
class="ot1-zi4r-0-x-x-109">distanceXY</span>, <span
class="ot1-zi4r-0-x-x-109">angle</span>, <span
class="ot1-zi4r-0-x-x-109">dihedral</span>,
<span
class="ot1-zi4r-0-x-x-109">rmsd</span>, <span
class="ot1-zi4r-0-x-x-109">eigenvector </span>and <span
class="ot1-zi4r-0-x-x-109">gyration</span>.
<!--l. 2309--><p class="noindent" >
   <h4 class="subsectionHead"><span class="titlemark">5.3    </span> <a
 id="x1-480005.3"></a>Linear and polynomial combinations of components</h4>
<!--l. 2312--><p class="noindent" >To extend the set of possible definitions of colvars <span
class="zptmcm7m-x-x-109">&#x03BE;</span><span
class="zptmcm7t-x-x-109">(</span><span
class="ptmb7t-x-x-109">r</span><span
class="zptmcm7t-x-x-109">)</span>, multiple components <span
class="zptmcm7m-x-x-109">q</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub><span
class="zptmcm7t-x-x-109">(</span><span
class="ptmb7t-x-x-109">r</span><span
class="zptmcm7t-x-x-109">) </span>can be summed with the
formula:
   <table
class="equation"><tr><td><a
 id="x1-48001r10"></a>


   <center class="math-display" >
<img
src="colvars-refman-lammps15x.png" alt="&#x03BE;(r)=  &#x2211; ci[qi(r)]ni
       i
" class="math-display" ></center></td><td class="equation-label">(10)</td></tr></table>
<!--l. 2317--><p class="nopar" >
where each component appears with a unique coefficient <span
class="zptmcm7m-x-x-109">c</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub> (1.0 by default) the positive integer exponent <span
class="zptmcm7m-x-x-109">n</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub> (1 by
default).
<!--l. 2321--><p class="indent" >   Any set of components can be combined within a colvar, provided that they return the same type of values
(scalar, unit vector, vector, or quaternion). By default, the colvar is the sum of its components. Linear or polynomial
combinations (following equation&#x00A0;(<a
href="#x1-48001r10">10<!--tex4ht:ref: eq:colvar_combination --></a>)) can be obtained by setting the following parameters, which are common to
all components:
     <ul class="itemize1">
     <li class="itemize"><a
 id="dx1-48002"></a> <span
class="ot1-zi4r-0-x-x-120">componentCoeff </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Coefficient of this component in the colvar<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span>any component <br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>decimal <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">1.0</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:   </span>Defines the coefficient by which this component is multiplied (after being raised to
     <span
class="ot1-zi4r-0-x-x-109">componentExp</span>) before being added to the sum.
     </li>
     <li class="itemize"><a
 id="dx1-48003"></a> <span
class="ot1-zi4r-0-x-x-120">componentExp </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Exponent of this component in the colvar<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span>any component <br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>integer <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">1</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>Defines the power at which the value of this component is raised before being added
     to the sum. When this exponent is different than 1 (non-linear sum), system forces and the Jacobian
     force are not available, making the colvar unsuitable for ABF calculations.</li></ul>
<!--l. 2352--><p class="indent" >   <span
class="ptmb7t-x-x-109">Example: </span>To define the <span
class="ptmri7t-x-x-109">average </span>of a colvar across different parts of the system, simply define within the same
colvar block a series of components of the same type (applied to different atom groups), and assign to each
component a <span
class="ot1-zi4r-0-x-x-109">componentCoeff </span>of <span
class="zptmcm7t-x-x-109">1</span><span
class="zptmcm7m-x-x-109">&#x2215;N</span>.
<!--l. 2429--><p class="noindent" >
   <h3 class="sectionHead"><span class="titlemark">6    </span> <a
 id="x1-490006"></a>Biasing and analysis methods</h3>
<!--l. 2432--><p class="noindent" >All of the biasing and analysis methods implemented (<span
class="ot1-zi4r-0-x-x-109">abf</span>, <span
class="ot1-zi4r-0-x-x-109">harmonic</span>, <span
class="ot1-zi4r-0-x-x-109">histogram </span>and <span
class="ot1-zi4r-0-x-x-109">metadynamics</span>) recognize the
following options:
     <ul class="itemize1">
     <li class="itemize"><a
 id="dx1-49001"></a> <span
class="ot1-zi4r-0-x-x-120">name </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Identifier for the bias<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span>colvar bias <br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>string <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="zptmcm7m-x-x-109">&#x003C;</span><span
class="ot1-zi4r-0-x-x-109">type of bias</span><span
class="zptmcm7m-x-x-109">&#x003E;&#x003C;</span><span
class="ot1-zi4r-0-x-x-109">bias index</span><span
class="zptmcm7m-x-x-109">&#x003E; </span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:   </span>This string is used to identify the bias or analysis method in output messages and to
     name some output files.
     </li>
     <li class="itemize"><a
 id="dx1-49002"></a> <span
class="ot1-zi4r-0-x-x-120">colvars </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Collective variables involved<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span>colvar bias <br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>space-separated list of colvar names <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>This option selects by name all the colvars to which this bias or analysis will be applied.
     </li>
     <li class="itemize"><a
 id="dx1-49003"></a> <span
class="ot1-zi4r-0-x-x-120">outputEnergy </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Write the current bias energy to the trajectory file<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span>colvar bias <br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">off</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>If this option is chosen and <span
class="ot1-zi4r-0-x-x-109">colvarsTrajFrequency </span>is not zero, the current value of
     the biasing energy will be written to the trajectory file during the simulation.
     </li></ul>
<!--l. 2470--><p class="noindent" >
   <h4 class="subsectionHead"><span class="titlemark">6.1    </span> <a
 id="x1-500006.1"></a>Adaptive Biasing Force</h4>
<!--l. 2474--><p class="noindent" >For a full description of the Adaptive Biasing Force method, see reference&#x00A0;<span class="cite">[<a
href="#XDarve2008">4</a>]</span>. For details about this
implementation, see references&#x00A0;<span class="cite">[<a
href="#XHenin2004">5</a>]</span> and <span class="cite">[<a
href="#XHenin2010">6</a>]</span>. <span
class="ptmb7t-x-x-109">When publishing research that makes use of this functionality,</span>
<span
class="ptmb7t-x-x-109">please cite references</span><span
class="ptmb7t-x-x-109">&#x00A0;</span><span class="cite"><span
class="ptmb7t-x-x-109">[</span><a
href="#XDarve2008"><span
class="ptmb7t-x-x-109">4</span></a><span
class="ptmb7t-x-x-109">]</span></span> <span
class="ptmb7t-x-x-109">and </span><span class="cite"><span
class="ptmb7t-x-x-109">[</span><a
href="#XHenin2010"><span
class="ptmb7t-x-x-109">6</span></a><span
class="ptmb7t-x-x-109">]</span></span><span
class="ptmb7t-x-x-109">.</span>
<!--l. 2480--><p class="indent" >   An alternate usage of this feature is the application of custom tabulated biasing potentials to one or more colvars.
See <span
class="ot1-zi4r-0-x-x-109">inputPrefix </span>and <span
class="ot1-zi4r-0-x-x-109">updateBias </span>below.
<!--l. 2484--><p class="indent" >   ABF is based on the thermodynamic integration (TI) scheme for computing free energy profiles. The free energy
as a function of a set of collective variables <span
class="zptmcm7m-x-x-109">&#x03BE;</span> <span
class="zptmcm7t-x-x-109">= (</span><span
class="zptmcm7m-x-x-109">&#x03BE;</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub><span
class="zptmcm7t-x-x-109">)</span><sub><span
class="zptmcm7m-x-x-80">i</span><span
class="zptmcm7y-x-x-80">&#x2208;</span><span
class="zptmcm7t-x-x-80">[1</span><span
class="zptmcm7m-x-x-80">,n</span><span
class="zptmcm7t-x-x-80">]</span></sub> is defined from the canonical distribution of <span
class="zptmcm7m-x-x-109">&#x03BE;</span> ,
<span
class="zptmcm7y-x-x-109"><img
src="zptmcm7y-50.png" alt="P" class="-109x-x-50" /></span><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">&#x03BE;</span><span
class="zptmcm7t-x-x-109">)</span>:
   <table
class="equation"><tr><td><a
 id="x1-50001r11"></a>
   <center class="math-display" >
<img
src="colvars-refman-lammps16x.png" alt="A(&#x03BE;)= - 1-lnP (&#x03BE;)+ A0
        &#x03B2;
" class="math-display" ></center></td><td class="equation-label">(11)</td></tr></table>
<!--l. 2492--><p class="nopar" >
<!--l. 2494--><p class="indent" >   In the TI formalism, the free energy is obtained from its gradient, which is generally calculated in the form of
the average of a force <span
class="zptmcm7m-x-x-109">F</span><sub><span
class="zptmcm7m-x-x-80">&#x03BE;</span></sub> exerted on <span
class="zptmcm7m-x-x-109">&#x03BE;</span> , taken over an iso-<span
class="zptmcm7m-x-x-109">&#x03BE;</span> surface:
   <table
class="equation"><tr><td><a
 id="x1-50002r12"></a>


   <center class="math-display" >
<img
src="colvars-refman-lammps17x.png" alt="         &#x27E8;    &#x27E9;
&#x2207; &#x03BE;A(&#x03BE;)=  - F&#x03BE; &#x03BE;
" class="math-display" ></center></td><td class="equation-label">(12)</td></tr></table>
<!--l. 2501--><p class="nopar" >
<!--l. 2503--><p class="indent" >   Several formulae that take the form of&#x00A0;(<a
href="#x1-50002r12">12<!--tex4ht:ref: eq:gradient --></a>) have been proposed. This implementation relies partly on the classic
formulation&#x00A0;<span class="cite">[<a
href="#XCarter1989">7</a>]</span>, and partly on a more versatile scheme originating in a work by Ruiz-Montero et al.&#x00A0;<span class="cite">[<a
href="#XRuiz-Montero1997">8</a>]</span>,
generalized by den Otter&#x00A0;<span class="cite">[<a
href="#XdenOtter2000">9</a>]</span> and extended to multiple variables by Ciccotti et al.&#x00A0;<span class="cite">[<a
href="#XCiccotti2005">10</a>]</span>. Consider a system subject to
constraints of the form <span
class="zptmcm7m-x-x-109">&#x03C3;</span><sub><span
class="zptmcm7m-x-x-80">k</span></sub><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">x</span><span
class="zptmcm7t-x-x-109">) = 0</span>. Let (<span
class="zptmcm7m-x-x-109">v</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub><span
class="zptmcm7t-x-x-109">)</span><sub><span
class="zptmcm7m-x-x-80">i</span><span
class="zptmcm7y-x-x-80">&#x2208;</span><span
class="zptmcm7t-x-x-80">[1</span><span
class="zptmcm7m-x-x-80">,n</span><span
class="zptmcm7t-x-x-80">]</span></sub> be arbitrarily chosen vector fields (<span
class="msbm-10x-x-109">&#x211D;</span><sup><span
class="zptmcm7t-x-x-80">3</span><span
class="zptmcm7m-x-x-80">N</span></sup> <span
class="zptmcm7y-x-x-109">&#x2192; </span><span
class="msbm-10x-x-109">&#x211D;</span><sup><span
class="zptmcm7t-x-x-80">3</span><span
class="zptmcm7m-x-x-80">N</span></sup>) verifying, for all <span
class="zptmcm7m-x-x-109">i</span>,
<span
class="zptmcm7m-x-x-109">j</span>, and <span
class="zptmcm7m-x-x-109">k</span>:
<!--l. 2514--><p class="indent" >   <div class="eqnarray">
   <center class="math-display" >
<img
src="colvars-refman-lammps18x.png" alt="v &#x22C5;&#x2207; &#x03BE;   =   &#x03B4;                                   (1 3)
 i   xj       ij
vi&#x22C5;&#x2207;x&#x03C3;k  =   0                                   (1 4)
" class="math-display" ><a
 id="x1-50003r13"></a><a
 id="x1-50003r14"></a></center>
</div>
<!--l. 2521--><p class="indent" >   then the following holds&#x00A0;<span class="cite">[<a
href="#XCiccotti2005">10</a>]</span>:
   <table
class="equation"><tr><td><a
 id="x1-50004r15"></a>
   <center class="math-display" >
<img
src="colvars-refman-lammps19x.png" alt="&#x2202;A-= &#x27E8;vi&#x22C5;&#x2207;xV - kBT&#x2207;x &#x22C5;vi&#x27E9;
&#x2202;&#x03BE;i                     &#x03BE;
" class="math-display" ></center></td><td class="equation-label">(15)</td></tr></table>
<!--l. 2527--><p class="nopar" >


<!--l. 2529--><p class="indent" >   where <span
class="zptmcm7m-x-x-109">V </span>is the potential energy function. <span
class="zptmcm7m-x-x-109">v</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub> can be interpreted as the direction along which the force acting on
variable <span
class="zptmcm7m-x-x-109">&#x03BE;</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub> is measured, whereas the second term in the average corresponds to the geometric entropy contribution
that appears as a Jacobian correction in the classic formalism&#x00A0;<span class="cite">[<a
href="#XCarter1989">7</a>]</span>. Condition&#x00A0;(<a
href="#x1-50003r13">13<!--tex4ht:ref: eq:ortho_gradient --></a>) states that the direction along
which the system force on <span
class="zptmcm7m-x-x-109">&#x03BE;</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub> is measured is orthogonal to the gradient of <span
class="zptmcm7m-x-x-109">&#x03BE;</span><sub><span
class="zptmcm7m-x-x-80">j</span></sub>, which means that the force measured
on <span
class="zptmcm7m-x-x-109">&#x03BE;</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub> does not act on <span
class="zptmcm7m-x-x-109">&#x03BE;</span><sub><span
class="zptmcm7m-x-x-80">j</span></sub>.
<!--l. 2539--><p class="indent" >   Equation&#x00A0;(<a
href="#x1-50003r14">14<!--tex4ht:ref: eq:ortho_constraints --></a>) implies that constraint forces are orthogonal to the directions along which the free energy
gradient is measured, so that the measurement is effectively performed on unconstrained degrees of
freedom.
<!--l. 2546--><p class="indent" >   In the framework of ABF, <span
class="zptmcm7m-x-x-109">F</span><sub><span
class="zptmcm7m-x-x-80">&#x03BE;</span></sub> is accumulated in bins of finite size <span
class="zptmcm7m-x-x-109">&#x03B4;&#x03BE; </span>, thereby providing an estimate of the free
energy gradient according to equation&#x00A0;(<a
href="#x1-50002r12">12<!--tex4ht:ref: eq:gradient --></a>). The biasing force applied along the collective variables to overcome
free energy barriers is calculated as:
   <table
class="equation"><tr><td><a
 id="x1-50005r16"></a>
   <center class="math-display" >
<img
src="colvars-refman-lammps20x.png" alt="FABF = &#x03B1;(N &#x03BE;)&#x00D7; &#x2207;x^A(&#x03BE;)
" class="math-display" ></center></td><td class="equation-label">(16)</td></tr></table>
<!--l. 2556--><p class="nopar" >
<!--l. 2558--><p class="indent" >   where <span
class="zptmcm7y-x-x-109">&#x2207;</span><sub><span style="margin-left:0.3em" class="thinspace"></span><span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7m-x-x-80">x</span></sub><span style="margin-left:0.3em" class="thinspace"></span><img
src="colvars-refman-lammps21x.png" alt="A^"  class="widetilde" > denotes the current estimate of the free energy gradient at the current point <span
class="zptmcm7m-x-x-109">&#x03BE;</span> in the collective
variable subspace, and <span
class="zptmcm7m-x-x-109">&#x03B1;</span><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">N</span><sub><span
class="zptmcm7m-x-x-80">&#x03BE;</span></sub><span
class="zptmcm7t-x-x-109">) </span>is a scaling factor that is ramped from 0 to 1 as the local number of samples <span
class="zptmcm7m-x-x-109">N</span><sub><span
class="zptmcm7m-x-x-80">&#x03BE;</span></sub>
increases to prevent nonequilibrium effects in the early phase of the simulation, when the gradient estimate has a
large variance. See the <span
class="ot1-zi4r-0-x-x-109">fullSamples </span>parameter below for details.
<!--l. 2566--><p class="indent" >   As sampling of the phase space proceeds, the estimate <span
class="zptmcm7y-x-x-109">&#x2207;</span><sub><span style="margin-left:0.3em" class="thinspace"></span><span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7m-x-x-80">x</span></sub><span style="margin-left:0.3em" class="thinspace"></span><img
src="colvars-refman-lammps22x.png" alt="^A"  class="widetilde" > is progressively refined. The biasing force
introduced in the equations of motion guarantees that in the bin centered around <span
class="zptmcm7m-x-x-109">&#x03BE;</span> , the forces acting along the
selected collective variables average to zero over time. Eventually, as the undelying free energy surface is canceled
by the adaptive bias, evolution of the system along <span
class="zptmcm7m-x-x-109">&#x03BE;</span> is governed mainly by diffusion. Although this
implementation of ABF can in principle be used in arbitrary dimension, a higher-dimension collective variable
space is likely to result in sampling difficulties. Most commonly, the number of variables is one or
two.
<!--l. 2581--><p class="noindent" >
   <h5 class="subsubsectionHead"><span class="titlemark">6.1.1    </span> <a
 id="x1-510006.1.1"></a>ABF requirements on collective variables</h5>
<!--l. 2584--><p class="noindent" >
     <ol  class="enumerate1" >
     <li
  class="enumerate" id="x1-51002x1"><span
class="ptmri7t-x-x-109">Only linear combinations </span>of colvar components can be used in ABF calculations.
     </li>
     <li
  class="enumerate" id="x1-51004x2"><span
class="ptmri7t-x-x-109">Availability  of  system  forces  </span>is  necessary.  The  following  colvar  components  can  be  used  in
     ABF  calculations:  <span
class="ot1-zi4r-0-x-x-109">distance</span>,  <span
class="ot1-zi4r-0-x-x-109">distance</span><span
class="ot1-zi4r-0-x-x-109">_xy</span>,  <span
class="ot1-zi4r-0-x-x-109">distance</span><span
class="ot1-zi4r-0-x-x-109">_z</span>,  <span
class="ot1-zi4r-0-x-x-109">angle</span>,  <span
class="ot1-zi4r-0-x-x-109">dihedral</span>,  <span
class="ot1-zi4r-0-x-x-109">gyration</span>,  <span
class="ot1-zi4r-0-x-x-109">rmsd </span>and


     <span
class="ot1-zi4r-0-x-x-109">eigenvector</span>. Atom groups may not be replaced by dummy atoms, unless they are excluded from the
     force measurement by specifying <span
class="ot1-zi4r-0-x-x-109">oneSiteSystemForce</span>, if available.
     </li>
     <li
  class="enumerate" id="x1-51006x3"><span
class="ptmri7t-x-x-109">Mutual orthogonality of colvars</span>. In a multidimensional ABF calculation, equation&#x00A0;(<a
href="#x1-50003r13">13<!--tex4ht:ref: eq:ortho_gradient --></a>) must be satisfied for
     any two colvars <span
class="zptmcm7m-x-x-109">&#x03BE;</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub> and <span
class="zptmcm7m-x-x-109">&#x03BE;</span><sub><span
class="zptmcm7m-x-x-80">j</span></sub>. Various cases fulfill this orthogonality condition:
          <ul class="itemize1">
          <li class="itemize"><span
class="zptmcm7m-x-x-109">&#x03BE;</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub> and <span
class="zptmcm7m-x-x-109">&#x03BE;</span><sub><span
class="zptmcm7m-x-x-80">j</span></sub> are based on non-overlapping sets of atoms.
          </li>
          <li class="itemize">atoms involved in the force measurement on <span
class="zptmcm7m-x-x-109">&#x03BE;</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub> do not participate in the definition of <span
class="zptmcm7m-x-x-109">&#x03BE;</span><sub><span
class="zptmcm7m-x-x-80">j</span></sub>. This
          can be obtained using the option <span
class="ot1-zi4r-0-x-x-109">oneSiteSystemForce </span>of the <span
class="ot1-zi4r-0-x-x-109">distance</span>, <span
class="ot1-zi4r-0-x-x-109">angle</span>, and <span
class="ot1-zi4r-0-x-x-109">dihedral</span>
          components (example: Ramachandran angles <span
class="zptmcm7m-x-x-109">&#x03D5;</span>, <span
class="zptmcm7m-x-x-109">&#x03C8;</span>).
          </li>
          <li class="itemize"><span
class="zptmcm7m-x-x-109">&#x03BE;</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub> and  <span
class="zptmcm7m-x-x-109">&#x03BE;</span><sub><span
class="zptmcm7m-x-x-80">j</span></sub> are  orthogonal  by  construction.  Useful  cases  are  the  sum  and  difference  of  two
          components, or <span
class="ot1-zi4r-0-x-x-109">distance</span><span
class="ot1-zi4r-0-x-x-109">_z </span>and <span
class="ot1-zi4r-0-x-x-109">distance</span><span
class="ot1-zi4r-0-x-x-109">_xy </span>using the same axis.</li></ul>
     </li>
     <li
  class="enumerate" id="x1-51008x4"><span
class="ptmri7t-x-x-109">Mutual orthogonality of components</span>: when several components are combined into a colvar, it is assumed that
     their vectors <span
class="zptmcm7m-x-x-109">v</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub> (equation&#x00A0;(<a
href="#x1-50004r15">15<!--tex4ht:ref: eq:gradient_vector --></a>)) are mutually orthogonal. The cases described for colvars in the previous
     paragraph apply.
     </li>
     <li
  class="enumerate" id="x1-51010x5"><span
class="ptmri7t-x-x-109">Orthogonality of colvars and constraints</span>: equation&#x00A0;<a
href="#x1-50003r14">14<!--tex4ht:ref: eq:ortho_constraints --></a> can be satisfied in two simple ways, if
     either no constrained atoms are involved in the force measurement (see point 3 above) or pairs of
     atoms joined by a constrained bond are part of an <span
class="ptmri7t-x-x-109">atom group </span>which only intervenes through its
     center (center of mass or geometric center) in the force measurement. In the latter case, the
     contributions of the two atoms to the left-hand side of equation&#x00A0;<a
href="#x1-50003r14">14<!--tex4ht:ref: eq:ortho_constraints --></a> cancel out. For example, all atoms
     of a rigid TIP3P water molecule can safely be included in an atom group used in a <span
class="ot1-zi4r-0-x-x-109">distance</span>
     component.</li></ol>
<!--l. 2618--><p class="noindent" >
   <h5 class="subsubsectionHead"><span class="titlemark">6.1.2    </span> <a
 id="x1-520006.1.2"></a>Parameters for ABF</h5>
<!--l. 2620--><p class="noindent" >ABF depends on parameters from collective variables to define the grid on which free energy gradients are
computed. In the direction of each colvar, the grid ranges from <span
class="ot1-zi4r-0-x-x-109">lowerBoundary </span>to <span
class="ot1-zi4r-0-x-x-109">upperBoundary</span>, and the
bin width (grid spacing) is set by the <span
class="ot1-zi4r-0-x-x-109">width </span>parameter (see&#x00A0;<a
href="#x1-90003.1">3.1<!--tex4ht:ref: sec:colvar_general --></a>). The following specific parameters
can be set in the ABF configuration block (in addition to generic bias parameters such as <span
class="ot1-zi4r-0-x-x-109">colvars </span>&#8211;
section&#x00A0;<a
href="#x1-490006">6<!--tex4ht:ref: sec:colvarbias --></a>):
     <ul class="itemize1">
     <li class="itemize"><a
 id="dx1-52001"></a> <span
class="ot1-zi4r-0-x-x-120">fullSamples </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Number of samples in a bin prior to application of the ABF<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">abf</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>positive integer <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span>200 <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>To avoid nonequilibrium effects due to large fluctuations of the force exerted along the
     colvars, it is recommended to apply a biasing force only after a the estimate has started converging. If


     <span
class="ot1-zi4r-0-x-x-109">fullSamples </span>is non-zero, the applied biasing force is scaled by a factor <span
class="zptmcm7m-x-x-109">&#x03B1;</span><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">N</span><sub><span
class="zptmcm7m-x-x-80">&#x03BE;</span></sub><span
class="zptmcm7t-x-x-109">) </span>between 0 and 1. If
     the number of samples <span
class="zptmcm7m-x-x-109">N</span><sub><span
class="zptmcm7m-x-x-80">&#x03BE;</span></sub> in the current bin is higher than <span
class="ot1-zi4r-0-x-x-109">fullSamples</span>, the factor is one. If it is less
     than half of <span
class="ot1-zi4r-0-x-x-109">fullSamples</span>, the factor is zero and no bias is applied. Between those two thresholds, the
     factor follows a linear ramp from 0 to 1: <span
class="zptmcm7m-x-x-109">&#x03B1;</span><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">N</span><sub><span
class="zptmcm7m-x-x-80">&#x03BE;</span></sub><span
class="zptmcm7t-x-x-109">) = (2</span><span
class="zptmcm7m-x-x-109">N</span><sub><span
class="zptmcm7m-x-x-80">&#x03BE;</span></sub><span
class="zptmcm7m-x-x-109">&#x2215;</span><span
class="zptmcm7t-x-x-109">fullSamples</span><span
class="zptmcm7t-x-x-109">)</span><span
class="zptmcm7y-x-x-109">-</span><span
class="zptmcm7t-x-x-109">1</span>.
     </li>
     <li class="itemize"><a
 id="dx1-52002"></a> <span
class="ot1-zi4r-0-x-x-120">maxForce </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Maximum magnitude of the ABF force<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">abf</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>positive decimals (one per colvar) <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span>disabled <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>This option enforces a cap on the magnitude of the biasing force effectively applied by
     this ABF bias on each colvar. This can be useful in the presence of singularities in the PMF such as
     hard walls, where the discretization of the average force becomes very inaccurate, causing the colvar&#8217;s
     diffusion to get &#8220;stuck&#8221; at the singularity. To enable this cap, provide one non-negative value for each
     colvar. The unit of force is the unit of energy specified by <span
class="ot1-zi4r-0-x-x-109">units </span>divided by the colvar unit.
     </li>
     <li class="itemize"><a
 id="dx1-52003"></a> <span
class="ot1-zi4r-0-x-x-120">hideJacobian </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Remove geometric entropy term from calculated free energy gradient?<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">abf</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">no</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>In a few special cases, most notably distance-based variables, an alternate definition of
     the potential of mean force is traditionally used, which excludes the Jacobian term describing the effect
     of geometric entropy on the distribution of the variable. This results, for example, in particle-particle
     potentials of mean force being flat at large separations. Setting this parameter to <span
class="ot1-zi4r-0-x-x-109">yes </span>causes the output
     data to follow that convention, by removing this contribution from the output gradients while applying
     internally the corresponding correction to ensure uniform sampling. It is not allowed for colvars with
     multiple components.
     </li>
     <li class="itemize"><a
 id="dx1-52004"></a> <span
class="ot1-zi4r-0-x-x-120">outputFreq </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Frequency (in timesteps) at which ABF data files are refreshed<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">abf</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>positive integer <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span>Colvar module restart frequency <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>The files containing the free energy gradient estimate and sampling histogram (and the
     PMF in one-dimensional calculations) are written on disk at the given time interval.
     </li>
     <li class="itemize"><a
 id="dx1-52005"></a> <span
class="ot1-zi4r-0-x-x-120">historyFreq </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Frequency (in timesteps) at which ABF history files are accumulated<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">abf</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>positive integer <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span>0 <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>If this number is non-zero, the free energy gradient estimate and sampling histogram
     (and the PMF in one-dimensional calculations) are appended to files on disk at the given time interval.
     History file names use the same prefix as output files, with &#8220;<span
class="ot1-zi4r-0-x-x-109">.hist</span>&#8221; appended.
     </li>
     <li class="itemize"><a
 id="dx1-52006"></a> <span
class="ot1-zi4r-0-x-x-120">inputPrefix </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Filename prefix for reading ABF data<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">abf</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>list of strings <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:    </span>If  this  parameter  is  set,  for  each  item  in  the  list,  ABF  tries  to  read  a  gradient
     and a sampling files named <span
class="zptmcm7m-x-x-109">&#x003C;</span><span
class="ot1-zi4r-0-x-x-109">inputPrefix</span><span
class="zptmcm7m-x-x-109">&#x003E;</span><span
class="ot1-zi4r-0-x-x-109">.grad </span>and <span
class="zptmcm7m-x-x-109">&#x003C;</span><span
class="ot1-zi4r-0-x-x-109">inputPrefix</span><span
class="zptmcm7m-x-x-109">&#x003E;</span><span
class="ot1-zi4r-0-x-x-109">.count</span>. This is done at
     startup and sets the initial state of the ABF algorithm. The data from all provided files is combined
     appropriately. Also, the grid definition (min and max values, width) need not be the same that for
     the current run. This command is useful to piece together data from simulations in different regions
     of  collective  variable  space,  or  change  the  colvar  boundary  values  and  widths.  Note  that  it  is  not
     recommended to use it to switch to a smaller width, as that will leave some bins empty in the finer
     data grid. This option is NOT compatible with reading the data from a restart file (<span
class="ot1-zi4r-0-x-x-109">input </span>keyword of
     the <span
class="ot1-zi4r-0-x-x-109">fix ID group-ID colvars </span>command).
     </li>
     <li class="itemize"><a
 id="dx1-52007"></a> <span
class="ot1-zi4r-0-x-x-120">applyBias </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Apply the ABF bias?<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">abf</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">yes</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:   </span>If this is set to no, the calculation proceeds normally but the adaptive biasing force is
     not applied. Data is still collected to compute the free energy gradient. This is mostly intended for
     testing purposes, and should not be used in routine simulations.
     </li>
     <li class="itemize"><a
 id="dx1-52008"></a> <span
class="ot1-zi4r-0-x-x-120">updateBias </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Update the ABF bias?<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">abf</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">yes</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:    </span>If this is set to no, the initial biasing force (e.g. read from a restart file or through
     <span
class="ot1-zi4r-0-x-x-109">inputPrefix</span>) is not updated during the simulation. As a result, a constant bias is applied. This can
     be used to apply a custom, tabulated biasing potential to any combination of colvars. To that effect,
     one should prepare a gradient file containing the gradient of the potential to be applied (negative of
     the bias force), and a count file containing only values greater than <span
class="ot1-zi4r-0-x-x-109">fullSamples</span>. These files must
     match the grid parameters of the colvars. </li></ul>
<!--l. 2758--><p class="noindent" >
   <h5 class="subsubsectionHead"><span class="titlemark">6.1.3    </span> <a
 id="x1-530006.1.3"></a>Output files</h5>
<!--l. 2760--><p class="noindent" >The ABF bias produces the following files, all in multicolumn ASCII format:
     <ul class="itemize1">
     <li class="itemize"><span
class="ptmri7t-x-x-109">output</span><span
class="ot1-zi4r-0-x-x-109">.grad</span>: current estimate of the free energy gradient (grid), in multicolumn;
     </li>
     <li class="itemize"><span
class="ptmri7t-x-x-109">output</span><span
class="ot1-zi4r-0-x-x-109">.count</span>: total number of samples collected, on the same grid;
     </li>
     <li class="itemize"><span
class="ptmri7t-x-x-109">output</span><span
class="ot1-zi4r-0-x-x-109">.pmf</span>: only for one-dimensional calculations, integrated free energy profile or PMF.</li></ul>
<!--l. 2769--><p class="indent" >   If several ABF biases are defined concurrently, their name is inserted to produce unique filenames for output, as
in <span
class="ptmri7t-x-x-109">output</span><span
class="ot1-zi4r-0-x-x-109">.abf1.grad</span>. This should not be done routinely and could lead to meaningless results: only do it if you
know what you are doing!


<!--l. 2774--><p class="indent" >   If the colvar space has been partitioned into sections (<span
class="ptmri7t-x-x-109">windows</span>) in which independent ABF simulations have
been run, the resulting data can be merged using the <span
class="ot1-zi4r-0-x-x-109">inputPrefix </span>option described above (a run of 0 steps is
enough).
<!--l. 2779--><p class="noindent" >
   <h5 class="subsubsectionHead"><span class="titlemark">6.1.4    </span> <a
 id="x1-540006.1.4"></a>Post-processing: reconstructing a multidimensional free energy surface</h5>
<!--l. 2781--><p class="noindent" >If a one-dimensional calculation is performed, the estimated free energy gradient is automatically integrated and a
potential of mean force is written under the file name <span
class="ot1-zi4r-0-x-x-109"><outputName>.pmf</span>, in a plain text format that can be read by
most data plotting and analysis programs (e.g. gnuplot).
<!--l. 2786--><p class="indent" >   In dimension 2 or greater, integrating the discretized gradient becomes non-trivial. The standalone utility
<span
class="ot1-zi4r-0-x-x-109">abf</span><span
class="ot1-zi4r-0-x-x-109">_integrate </span>is provided to perform that task. <span
class="ot1-zi4r-0-x-x-109">abf</span><span
class="ot1-zi4r-0-x-x-109">_integrate </span>reads the gradient data and uses it to perform a
Monte-Carlo (M-C) simulation in discretized collective variable space (specifically, on the same grid used by ABF
to discretize the free energy gradient). By default, a history-dependent bias (similar in spirit to metadynamics) is
used: at each M-C step, the bias at the current position is incremented by a preset amount (the <span
class="ptmri7t-x-x-109">hill height</span>). Upon
convergence, this bias counteracts optimally the underlying gradient; it is negated to obtain the estimate of the free
energy surface.
<!--l. 2797--><p class="indent" >   <span
class="ot1-zi4r-0-x-x-109">abf</span><span
class="ot1-zi4r-0-x-x-109">_integrate </span>is invoked using the command-line:


   <div class="verbatim" id="verbatim-1">
integrate&#x00A0;<gradient_file>&#x00A0;[-n&#x00A0;<nsteps>]&#x00A0;[-t&#x00A0;<temp>]&#x00A0;[-m&#x00A0;(0|1)]
&#x00A0;<br />&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;&#x00A0;[-h&#x00A0;<hill_height>]&#x00A0;[-f&#x00A0;<factor>]
</div>
<!--l. 2802--><p class="nopar" >
<!--l. 2805--><p class="indent" >   The gradient file name is provided first, followed by other parameters in any order. They are described below,
with their default value in square brackets:
     <ul class="itemize1">
     <li class="itemize"><span
class="ot1-zi4r-0-x-x-109">-n</span>: number of M-C steps to be performed; by default, a minimal number of steps is chosen based on
     the size of the grid, and the integration runs until a convergence criterion is satisfied (based on the
     RMSD between the target gradient and the real PMF gradient)
     </li>
     <li class="itemize"><span
class="ot1-zi4r-0-x-x-109">-t</span>: temperature for M-C sampling (unrelated to the simulation temperature) [500&#x00A0;K]
     </li>
     <li class="itemize"><span
class="ot1-zi4r-0-x-x-109">-m</span>: use metadynamics-like biased sampling? (0 = false) [1]
     </li>
     <li class="itemize"><span
class="ot1-zi4r-0-x-x-109">-h</span>: increment for the history-dependent bias (&#8220;hill height&#8221;) [0.01&#x00A0;kcal/mol]
     </li>
     <li class="itemize"><span
class="ot1-zi4r-0-x-x-109">-f</span>: if non-zero, this factor is used to scale the increment stepwise in the second half of the M-C
     sampling to refine the free energy estimate [0.5]</li></ul>
<!--l. 2820--><p class="indent" >   Using the default values of all parameters should give reasonable results in most cases.
<!--l. 2823--><p class="indent" >   <span
class="ot1-zi4r-0-x-x-109">abf</span><span
class="ot1-zi4r-0-x-x-109">_integrate </span>produces the following output files:
     <ul class="itemize1">
     <li class="itemize"><span
class="ot1-zi4r-0-x-x-109"><gradient</span><span
class="ot1-zi4r-0-x-x-109">_file>.pmf</span>: computed free energy surface
     </li>
     <li class="itemize"><span
class="ot1-zi4r-0-x-x-109"><gradient</span><span
class="ot1-zi4r-0-x-x-109">_file>.histo</span>: histogram of M-C sampling (not usable in a straightforward way if the
     history-dependent bias has been applied)
     </li>
     <li class="itemize"><span
class="ot1-zi4r-0-x-x-109"><gradient</span><span
class="ot1-zi4r-0-x-x-109">_file>.est</span>:  estimated  gradient  of  the  calculated  free  energy  surface  (from  finite
     differences)
     </li>
     <li class="itemize"><span
class="ot1-zi4r-0-x-x-109"><gradient</span><span
class="ot1-zi4r-0-x-x-109">_file>.dev</span>:  deviation  between  the  user-provided  numerical  gradient  and  the  actual
     gradient  of  the  calculated  free  energy  surface.  The  RMS  norm  of  this  vector  field  is  used  as  a
     convergence criteria and displayed periodically during the integration.</li></ul>
<!--l. 2836--><p class="indent" >   <span
class="ptmb7t-x-x-109">Note: </span>Typically, the &#8220;deviation&#8221; vector field does not vanish as the integration converges. This happens because
the numerical estimate of the gradient does not exactly derive from a potential, due to numerical approximations
used to obtain it (finite sampling and discretization on a grid).


<!--l. 2843--><p class="noindent" >
   <h4 class="subsectionHead"><span class="titlemark">6.2    </span> <a
 id="x1-550006.2"></a>Metadynamics</h4>
<!--l. 2846--><p class="noindent" >The metadynamics method uses a history-dependent potential <span class="cite">[<a
href="#XLaio2002">11</a>]</span> that generalizes to any type of colvars the
conformational flooding <span class="cite">[<a
href="#XGrubmuller1995">12</a>]</span> and local elevation <span class="cite">[<a
href="#XHuber1994">13</a>]</span> methods, originally formulated to use as colvars the principal
components of a covariance matrix or a set of dihedral angles, respectively. The metadynamics potential on the
colvars <span
class="zptmcm7m-x-x-109">&#x03BE;</span> <span
class="zptmcm7t-x-x-109">= (</span><span
class="zptmcm7m-x-x-109">&#x03BE;</span><sub><span
class="zptmcm7t-x-x-80">1</span></sub><span
class="zptmcm7m-x-x-109">,&#x03BE;</span><sub><span
class="zptmcm7t-x-x-80">2</span></sub><span
class="zptmcm7m-x-x-109">,</span><span
class="zptmcm7m-x-x-109">&#x2026;</span><span
class="zptmcm7m-x-x-109">,&#x03BE;</span><sub><span
class="zptmcm7m-x-x-80">N</span><sub><span
class="zptmcm7t-x-x-60">cv</span></sub></sub><span
class="zptmcm7t-x-x-109">) </span>is defined as:
   <table
class="equation"><tr><td><a
 id="x1-55001r17"></a>
   <center class="math-display" >
<img
src="colvars-refman-lammps23x.png" alt="                         (             )
            t&#x2032;&#x003C;t    Ncv      (&#x03BE;i- &#x03BE;i(t&#x2032;))2
Vmeta(&#x03BE; )=    &#x2211;    W &#x220F;  exp  -------2----  ,
         t&#x2032;=&#x03B4;t,2&#x03B4;t,... i=1          2&#x03B4;&#x03BE;i
" class="math-display" ></center></td><td class="equation-label">(17)</td></tr></table>
<!--l. 2856--><p class="nopar" >
where <span
class="zptmcm7m-x-x-109">V</span><sub><span
class="zptmcm7t-x-x-80">meta</span></sub> is the history-dependent potential acting on the <span
class="ptmri7t-x-x-109">current </span>values of the colvars <span
class="zptmcm7m-x-x-109">&#x03BE;</span> , and depends only
parametrically on the <span
class="ptmri7t-x-x-109">previous </span>values of the colvars. <span
class="zptmcm7m-x-x-109">V</span><sub><span
class="zptmcm7t-x-x-80">meta</span></sub> is constructed as a sum of <span
class="zptmcm7m-x-x-109">N</span><sub><span
class="zptmcm7t-x-x-80">cv</span></sub>-dimensional repulsive
Gaussian &#8220;hills&#8221;, whose height is a chosen energy constant <span
class="zptmcm7m-x-x-109">W </span>, and whose centers are the previously explored
configurations <img
src="colvars-refman-lammps24x.png" alt="(&#x03BE;(&#x03B4;t),&#x03BE; (2&#x03B4;t),...)"  class="left" align="middle">. Each Gaussian functions has a width of approximately <span
class="zptmcm7t-x-x-109">2</span><span
class="zptmcm7m-x-x-109">&#x03B4;</span><sub><span
class="zptmcm7m-x-x-80">&#x03BE;</span><sub><span
class="zptmcm7m-x-x-60">i</span></sub></sub> along the direction
of the <span
class="zptmcm7m-x-x-109">i</span>-th colvar.
<!--l. 2861--><p class="indent" >   During the simulation, the system evolves towards the nearest minimum of the &#8220;effective&#8221; potential of mean
force <span
class="zptmcm7m-x-x-109">Ã</span><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">&#x03BE;</span><span
class="zptmcm7t-x-x-109">)</span>, which is the sum of the &#8220;real&#8221; underlying potential of mean force <span
class="zptmcm7m-x-x-109">A</span><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">&#x03BE;</span><span
class="zptmcm7t-x-x-109">) </span>and the the metadynamics
potential <span
class="zptmcm7m-x-x-109">V</span><sub><span
class="zptmcm7t-x-x-80">meta</span></sub><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">&#x03BE;</span><span
class="zptmcm7t-x-x-109">)</span>. Therefore, at any given time the probability of observing the configuration <span
class="zptmcm7m-x-x-109">&#x03BE;</span><sup><span
class="zptmcm7y-x-x-80">*</span></sup> is proportional to
<span
class="zptmcm7t-x-x-109">exp</span><img
src="colvars-refman-lammps25x.png" alt="(-A~(&#x03BE; *)&#x2215;&#x03BA;BT )"  class="left" align="middle">: this is also the probability that a new Gaussian &#8220;hill&#8221; is added at that configuration. If the
simulation is run for a sufficiently long time, each local minimum is canceled out by the sum of the
Gaussian &#8220;hill&#8221; functions. At that stage the the &#8220;effective&#8221; potential of mean force <span
class="zptmcm7m-x-x-109">Ã</span><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">&#x03BE;</span><span
class="zptmcm7t-x-x-109">) </span>is constant, and
<span
class="zptmcm7y-x-x-109">-</span><span
class="zptmcm7m-x-x-109">V</span><sub><span
class="zptmcm7t-x-x-80">meta</span></sub><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">&#x03BE;</span><span
class="zptmcm7t-x-x-109">) </span>is an accurate estimator of the &#8220;real&#8221; potential of mean force <span
class="zptmcm7m-x-x-109">A</span><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">&#x03BE;</span><span
class="zptmcm7t-x-x-109">)</span>, save for an additive
constant:
   <table
class="equation"><tr><td><a
 id="x1-55002r18"></a>
   <center class="math-display" >
<img
src="colvars-refman-lammps26x.png" alt="A(&#x03BE;) &#x2243; - Vmeta(&#x03BE; )+ K
" class="math-display" ></center></td><td class="equation-label">(18)</td></tr></table>
<!--l. 2870--><p class="nopar" >


<!--l. 2872--><p class="indent" >   Assuming that the set of collective variables includes all relevant degrees of freedom, the predicted error of the
estimate is a simple function of the correlation times of the colvars <span
class="zptmcm7m-x-x-109">&#x03C4;</span><sub><span
class="zptmcm7m-x-x-80">&#x03BE;</span><sub><span
class="zptmcm7m-x-x-60">i</span></sub></sub>, and of the user-defined parameters <span
class="zptmcm7m-x-x-109">W </span>, <span
class="zptmcm7m-x-x-109">&#x03B4;</span><sub><span
class="zptmcm7m-x-x-80">&#x03BE;</span><sub><span
class="zptmcm7m-x-x-60">i</span></sub></sub>
and <span
class="zptmcm7m-x-x-109">&#x03B4;t </span><span class="cite">[<a
href="#XBussi2006">14</a>]</span>. In typical applications, a good rule of thumb can be to choose the ratio <span
class="zptmcm7m-x-x-109">W&#x2215;&#x03B4;t </span>much smaller than
<span
class="zptmcm7m-x-x-109">&#x03BA;</span><sub><span
class="zptmcm7t-x-x-80">B</span></sub><span
class="zptmcm7m-x-x-109">T &#x2215;&#x03C4;</span><sub><span
class="zptmcm7m-x-x-109">&#x03BE;</span></sub>, where <span
class="zptmcm7m-x-x-109">&#x03C4;</span><sub><span
class="zptmcm7m-x-x-109">&#x03BE;</span></sub> is the longest among <span
class="zptmcm7m-x-x-109">&#x03BE;</span> &#8217;s correlation times: <span
class="zptmcm7m-x-x-109">&#x03B4;</span><sub><span
class="zptmcm7m-x-x-80">&#x03BE;</span><sub><span
class="zptmcm7m-x-x-60">i</span></sub></sub> then dictates the resolution of the calculated
PMF.
<!--l. 2879--><p class="indent" >   To enable a metadynamics calculation, a <span
class="ot1-zi4r-0-x-x-109">metadynamics </span>block must be defined in the colvars configuration file.
Its only mandatory keyword is the <span
class="ot1-zi4r-0-x-x-109">colvars </span>option listing all the variables involved: multidimensional PMFs are
obtained by the same <span
class="ot1-zi4r-0-x-x-109">metadynamics </span>instance applied to all the colvars.
<!--l. 2882--><p class="indent" >   The parameters <span
class="zptmcm7m-x-x-109">W </span>and <span
class="zptmcm7m-x-x-109">&#x03B4;t </span>are specified by the keywords <span
class="ot1-zi4r-0-x-x-109">hillWeight </span>and <span
class="ot1-zi4r-0-x-x-109">newHillFrequency</span>, respectively. The
values of these options are optimal for colvars with correlation times <span
class="zptmcm7m-x-x-109">&#x03C4;</span><sub><span
class="zptmcm7m-x-x-109">&#x03BE;</span></sub> in the range of a few thousand simulation
steps, typical of many biomolecular simulations:
     <ul class="itemize1">
     <li class="itemize"><a
 id="dx1-55003"></a> <span
class="ot1-zi4r-0-x-x-120">hillWeight </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Height of each hill (unit of energy specified by <span
class="ot1-zi4r-0-x-x-109">units</span>)<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">metadynamics</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>positive decimal <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">0.01</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>This option sets the height <span
class="zptmcm7m-x-x-109">W </span>of the hills that are added during this run. Lower values
     provide more accurate sampling at the price of longer simulation times to complete a PMF calculation.
     </li>
     <li class="itemize"><a
 id="dx1-55004"></a> <span
class="ot1-zi4r-0-x-x-120">newHillFrequency </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Frequency of hill creation<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">metadynamics</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>positive integer <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">1000</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>This option sets the number of integration steps after which a new hill is added to the
     metadynamics potential. Its value determines the parameter <span
class="zptmcm7m-x-x-109">&#x03B4;t </span>in eq.&#x00A0;<a
href="#x1-55001r17">17<!--tex4ht:ref: eq:colvars_meta_pot --></a>. Higher values provide more
     accurate sampling at the price of longer simulation times to complete a PMF calculation.
     </li></ul>
<!--l. 2912--><p class="indent" >   It is the user&#8217;s responsibility to either leave <span
class="ot1-zi4r-0-x-x-109">hillWeight </span>and <span
class="ot1-zi4r-0-x-x-109">newHillFrequency </span>at their default values, or to
change them to match the specifics of each system. The parameter <span
class="zptmcm7m-x-x-109">&#x03B4;</span><sub><span
class="zptmcm7m-x-x-80">&#x03BE;</span><sub><span
class="zptmcm7m-x-x-60">i</span></sub></sub> is instead defined as approximately half the
<span
class="ot1-zi4r-0-x-x-109">width </span>of the corresponding colvar <span
class="zptmcm7m-x-x-109">&#x03BE;</span><sub><span
class="zptmcm7m-x-x-80">i</span></sub> (see <a
href="#x1-90003.1">3.1<!--tex4ht:ref: sec:colvar_general --></a>).
<!--l. 2917--><p class="noindent" >
   <h5 class="subsubsectionHead"><span class="titlemark">6.2.1    </span> <a
 id="x1-560006.2.1"></a>Output files</h5>
<!--l. 2919--><p class="noindent" >When interpolating grids are enabled (default behavior), the PMF is written every <span
class="ot1-zi4r-0-x-x-109">colvarsRestartFrequency </span>steps
to the file <span
class="ptmri7t-x-x-109">output</span><span
class="ot1-zi4r-0-x-x-109">.pmf</span>. The following two options allow to control this behavior and to visually track statistical
convergence:
     <ul class="itemize1">
     <li class="itemize"><a
 id="dx1-56001"></a> <span
class="ot1-zi4r-0-x-x-120">writeFreeEnergyFile </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Periodically write the PMF for visualization<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">metadynamics</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">on</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:       </span>When    <span
class="ot1-zi4r-0-x-x-109">useGrids  </span>and    this    option    are    <span
class="ot1-zi4r-0-x-x-109">on</span>,    the    PMF    is    written    every
     <span
class="ot1-zi4r-0-x-x-109">colvarsRestartFrequency </span>steps.


     </li>
     <li class="itemize"><a
 id="dx1-56002"></a> <span
class="ot1-zi4r-0-x-x-120">saveFreeEnergyFile </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Keep all the PMF files<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">metadynamics</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">off</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:   </span>When <span
class="ot1-zi4r-0-x-x-109">writeFreeEnergyFile </span>and this option are <span
class="ot1-zi4r-0-x-x-109">on</span>, the step number is included in
     the  file  name.  Activating  this  option  can  be  useful  to  follow  more  closely  the  convergence  of  the
     simulation, by comparing PMFs separated by short times.
     </li></ul>
<!--l. 2948--><p class="indent" >   <span
class="ptmb7t-x-x-109">Note: </span>when Gaussian hills are deposited near <span
class="ot1-zi4r-0-x-x-109">lowerBoundary </span>or <span
class="ot1-zi4r-0-x-x-109">upperBoundary </span>(see <a
href="#x1-90003.1">3.1<!--tex4ht:ref: sec:colvar_general --></a>) and interpolating
grids are used (default behavior), their truncation can give rise to accumulating errors. In these cases, as a measure
of fault-tolerance all Gaussian hills near the boundaries are included in the output state file, and are recalculated
analytically whenever the colvar falls outside the grid&#8217;s boundaries. (Such measure protects the accuracy
of the calculation, and can only be disabled by <span
class="ot1-zi4r-0-x-x-109">hardLowerBoundary </span>or <span
class="ot1-zi4r-0-x-x-109">hardUpperBoundary</span>.) To
avoid gradual loss of performance and growth of the state file, either one of the following solutions is
recommended:
     <ul class="itemize1">
     <li class="itemize">enabling the option <span
class="ot1-zi4r-0-x-x-109">expandBoundaries</span>, so that the grid&#8217;s boundaries are automatically recalculated
     whenever necessary; the resulting <span
class="ot1-zi4r-0-x-x-109">.pmf </span>will have its abscissas expanded accordingly;
     </li>
     <li class="itemize">setting  <span
class="ot1-zi4r-0-x-x-109">lowerWall </span>and  <span
class="ot1-zi4r-0-x-x-109">upperWall </span>well  within  the  interval  delimited  by  <span
class="ot1-zi4r-0-x-x-109">lowerBoundary </span>and
     <span
class="ot1-zi4r-0-x-x-109">upperBoundary</span>.</li></ul>
<!--l. 2958--><p class="noindent" >
   <h5 class="subsubsectionHead"><span class="titlemark">6.2.2    </span> <a
 id="x1-570006.2.2"></a>Performance tuning</h5>
<!--l. 2960--><p class="noindent" >The following options control the computational cost of metadynamics calculations, but do not affect results.
Default values are chosen to minimize such cost with no loss of accuracy.
     <ul class="itemize1">
     <li class="itemize"><a
 id="dx1-57001"></a> <span
class="ot1-zi4r-0-x-x-120">useGrids </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Interpolate the hills with grids<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">metadynamics</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">on</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:     </span>This  option  discretizes  all  hills  for  improved  performance,  accumulating  their
     energy  and  their  gradients  on  two  separate  grids  of  equal  spacing.  Grids  are  defined  by  the
     values  of  <span
class="ot1-zi4r-0-x-x-109">lowerBoundary</span>,  <span
class="ot1-zi4r-0-x-x-109">upperBoundary </span>and  <span
class="ot1-zi4r-0-x-x-109">width </span>for  each  colvar.  Currently,  this  option  is
     implemented for all types of variables except the non-scalar types (<span
class="ot1-zi4r-0-x-x-109">distanceDir </span>or <span
class="ot1-zi4r-0-x-x-109">orientation</span>).
     If <span
class="ot1-zi4r-0-x-x-109">expandBoundaries </span>is defined in one of the colvars, grids are automatically expanded along the
     direction of that colvar.
     </li>
     <li class="itemize"><a
 id="dx1-57002"></a> <span
class="ot1-zi4r-0-x-x-120">hillWidth </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Relative width of the hills<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">metadynamics</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>positive decimal <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:</span>  <img
src="colvars-refman-lammps27x.png" alt="&#x221A; ---
  2&#x03C0;"  class="sqrt" ><span
class="zptmcm7m-x-x-109">&#x2215;</span><span
class="zptmcm7t-x-x-109">2 </span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:   </span>Along  each  colvar,  the  width  of  each  Gaussian  hill  (<span
class="zptmcm7t-x-x-109">2</span><span
class="zptmcm7m-x-x-109">&#x03B4;</span><sub><span
class="zptmcm7m-x-x-80">&#x03BE;</span><sub><span
class="zptmcm7m-x-x-60">i</span></sub></sub>)  is  given  by  the  product
     between this number and the colvar&#8217;s <span
class="ot1-zi4r-0-x-x-109">width</span>. The default value gives hills whose volume is the product
     of <span
class="zptmcm7m-x-x-109">W </span>times the <span
class="ot1-zi4r-0-x-x-109">width </span>of all colvars. For a smoother visualization of the free energy plot, decrease
     <span
class="ot1-zi4r-0-x-x-109">width </span>and increase <span
class="ot1-zi4r-0-x-x-109">hillWidth </span>in the same proportion. <span
class="ptmb7t-x-x-109">Note: </span><span
class="ptmri7t-x-x-109">when </span><span
class="ot1-zi4r-0-x-x-109">useGrids </span><span
class="ptmri7t-x-x-109">is </span><span
class="ot1-zi4r-0-x-x-109">on </span><span
class="ptmri7t-x-x-109">(default in most</span>
     <span
class="ptmri7t-x-x-109">cases), values smaller than 1 should be avoided to avoid discretization errors</span>.
     </li>
     <li class="itemize"><a
 id="dx1-57003"></a> <span
class="ot1-zi4r-0-x-x-120">rebinGrids </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Recompute the grids when reading a state file<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">metadynamics</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">off</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>When restarting from a state file, the grid&#8217;s parameters (boundaries and widths) saved
     in the state file override those in the configuration file. Enabling this option forces the grids to match
     those in the current configuration file.
     </li></ul>
<!--l. 3015--><p class="noindent" >
   <h5 class="subsubsectionHead"><span class="titlemark">6.2.3    </span> <a
 id="x1-580006.2.3"></a>Well-tempered metadynamics</h5>
<!--l. 3017--><p class="noindent" >The following options define the configuration for the &#8220;well-tempered&#8221; metadynamics approach <span class="cite">[<a
href="#XBarducci2008">15</a>]</span>:
     <ul class="itemize1">
     <li class="itemize"><a
 id="dx1-58001"></a> <span
class="ot1-zi4r-0-x-x-120">wellTempered </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Perform well-tempered metadynamics<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">metadynamics</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">off</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>If enabled, this flag causes well-tempered metadynamics as described by Barducci et
     al.<span class="cite">[<a
href="#XBarducci2008">15</a>]</span> to be performed, rather than standard metadynamics. The parameter <span
class="ot1-zi4r-0-x-x-109">biasTemperature </span>is then
     required.This feature was contributed by Li Li (Luthey-Schulten group, Departement of Chemistry,
     UIUC).
     </li>
     <li class="itemize"><a
 id="dx1-58002"></a> <span
class="ot1-zi4r-0-x-x-120">biasTemperature </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Temperature bias for well-tempered metadynamics<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">metadynamics</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>positive decimal <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>When running metadynamics in the long time limit, collective variable space is sampled
     to a modified temperature <span
class="zptmcm7m-x-x-109">T </span><span
class="zptmcm7t-x-x-109">+&#x0394;</span><span
class="zptmcm7m-x-x-109">T </span>. In conventional metadynamics, the temperature &#8220;boost&#8221; <span
class="zptmcm7t-x-x-109">&#x0394;</span><span
class="zptmcm7m-x-x-109">T </span>would
     constantly increases with time. Instead, in well-tempered metadynamics <span
class="zptmcm7t-x-x-109">&#x0394;</span><span
class="zptmcm7m-x-x-109">T </span>must be defined by the
     user via <span
class="ot1-zi4r-0-x-x-109">biasTemperature</span>. If <span
class="ot1-zi4r-0-x-x-109">dumpFreeEnergyFile </span>is enabled, the written PMF includes the scaling
     factor <span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">T </span><span
class="zptmcm7t-x-x-109">+&#x0394;</span><span
class="zptmcm7m-x-x-109">T </span><span
class="zptmcm7t-x-x-109">)</span><span
class="zptmcm7m-x-x-109">&#x2215;</span><span
class="zptmcm7t-x-x-109">&#x0394;</span><span
class="zptmcm7m-x-x-109">T </span><span class="cite">[<a
href="#XBarducci2008">15</a>]</span>. A careful choice of <span
class="zptmcm7t-x-x-109">&#x0394;</span><span
class="zptmcm7m-x-x-109">T </span>determines the sampling and convergence rate, and
     is hence crucial to the success of a well-tempered metadynamics simulation.</li></ul>


<!--l. 3050--><p class="noindent" >
   <h5 class="subsubsectionHead"><span class="titlemark">6.2.4    </span> <a
 id="x1-590006.2.4"></a>Multiple-replicas metadynamics</h5>
<!--l. 3052--><p class="noindent" >The following options define metadynamics calculations with more than one replica:
     <ul class="itemize1">
     <li class="itemize"><a
 id="dx1-59001"></a> <span
class="ot1-zi4r-0-x-x-120">multipleReplicas </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Multiple replicas metadynamics<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">metadynamics</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">off</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>If this option is <span
class="ot1-zi4r-0-x-x-109">on</span>, multiple (independent) replica of the same system can be run at the
     same time, and their hills will be combined to obtain a single PMF <span class="cite">[<a
href="#XRaiteri2005">16</a>]</span>. Replicas are identified by
     the value of <span
class="ot1-zi4r-0-x-x-109">replicaID</span>. Communication is done by files: each replica must be able to read the files
     created by the others, whose paths are communicated through the file <span
class="ot1-zi4r-0-x-x-109">replicasRegistry</span>. This file,
     and the files listed in it, are read every <span
class="ot1-zi4r-0-x-x-109">replicaUpdateFrequency </span>steps. Every time the colvars state
     file is written (<span
class="ot1-zi4r-0-x-x-109">colvarsRestartFrequency</span>), the file:<br
class="newline" />&#8220;<span
class="ptmri7t-x-x-109">output</span><span
class="ot1-zi4r-0-x-x-109">.colvars.</span><span
class="ptmri7t-x-x-109">name</span><span
class="ot1-zi4r-0-x-x-109">.</span><span
class="ptmri7t-x-x-109">replicaID</span><span
class="ot1-zi4r-0-x-x-109">.state</span>&#8221; is also written, containing the state of the metadynamics
     bias for <span
class="ot1-zi4r-0-x-x-109">replicaID</span>. In the time steps between <span
class="ot1-zi4r-0-x-x-109">colvarsRestartFrequency</span>, new hills are temporarily
     written to the file:<br
class="newline" />&#8220;<span
class="ptmri7t-x-x-109">output</span><span
class="ot1-zi4r-0-x-x-109">.colvars.</span><span
class="ptmri7t-x-x-109">name</span><span
class="ot1-zi4r-0-x-x-109">.</span><span
class="ptmri7t-x-x-109">replicaID</span><span
class="ot1-zi4r-0-x-x-109">.hills</span>&#8221;, which serves as communication buffer. These files are
     only required for communication, and may be deleted after a new MD run is started with a different
     <span
class="ot1-zi4r-0-x-x-109">outputName</span>.
     </li>
     <li class="itemize"><a
 id="dx1-59002"></a> <span
class="ot1-zi4r-0-x-x-120">replicaID </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Set the identifier for this replica<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">metadynamics</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>string <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:   </span>If <span
class="ot1-zi4r-0-x-x-109">multipleReplicas </span>is <span
class="ot1-zi4r-0-x-x-109">on</span>, this option sets a unique identifier for this replica. All
     replicas should use identical collective variable configurations, except for the value of this option.
     </li>
     <li class="itemize"><a
 id="dx1-59003"></a> <span
class="ot1-zi4r-0-x-x-120">replicasRegistry </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Multiple replicas database file<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">metadynamics</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>UNIX filename <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span>&#8220;<span
class="ptmri7t-x-x-109">name</span><span
class="ot1-zi4r-0-x-x-109">.replica</span><span
class="ot1-zi4r-0-x-x-109">_files.txt</span>&#8221; <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>If <span
class="ot1-zi4r-0-x-x-109">multipleReplicas </span>is <span
class="ot1-zi4r-0-x-x-109">on</span>, this option sets the path to the replicas&#8217; database file.
     </li>
     <li class="itemize"><a
 id="dx1-59004"></a> <span
class="ot1-zi4r-0-x-x-120">replicaUpdateFrequency </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>How often hills are communicated between replicas<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">metadynamics</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>positive integer <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">newHillFrequency</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>If <span
class="ot1-zi4r-0-x-x-109">multipleReplicas </span>is <span
class="ot1-zi4r-0-x-x-109">on</span>, this option sets the number of steps after which each replica
     (re)reads the other replicas&#8217; files. The lowest meaningful value of this number is <span
class="ot1-zi4r-0-x-x-109">newHillFrequency</span>.
     If  access  to  the  file  system  is  significantly  affecting  the  simulation  performance,  this  number
     can  be  increased,  at  the  price  of  reduced  synchronization  between  replicas.  Values  higher  than
     <span
class="ot1-zi4r-0-x-x-109">colvarsRestartFrequency </span>may not improve performance significantly.
     </li>
     <li class="itemize"><a
 id="dx1-59005"></a> <span
class="ot1-zi4r-0-x-x-120">dumpPartialFreeEnergyFile </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Periodically write the contribution to the PMF from this replica<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span>


     <br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">metadynamics</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">on</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>When <span
class="ot1-zi4r-0-x-x-109">multipleReplicas </span>is <span
class="ot1-zi4r-0-x-x-109">on</span>, tje file <span
class="ptmri7t-x-x-109">output</span><span
class="ot1-zi4r-0-x-x-109">.pmf </span>contains the combined PMF from
     all replicas. Enabling this option produces an additional file <span
class="ptmri7t-x-x-109">output</span><span
class="ot1-zi4r-0-x-x-109">.partial.pmf</span>, which can be useful
     to quickly monitor the contribution of each replica to the PMF. The requirements for this option are
     the same as <span
class="ot1-zi4r-0-x-x-109">dumpFreeEnergyFile</span>.
     </li></ul>
<!--l. 3151--><p class="noindent" >
   <h5 class="subsubsectionHead"><span class="titlemark">6.2.5    </span> <a
 id="x1-600006.2.5"></a>Compatibility and post-processing</h5>
<!--l. 3152--><p class="noindent" >The following options may be useful only for applications that go beyond the calculation of a PMF by
metadynamics:
     <ul class="itemize1">
     <li class="itemize"><a
 id="dx1-60001"></a> <span
class="ot1-zi4r-0-x-x-120">name </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Name of this metadynamics instance<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">metadynamics</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>string <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span>&#8220;<span
class="ot1-zi4r-0-x-x-109">meta</span>&#8221; + rank number <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>This option sets the name for this metadynamics instance. While it is not advisable to
     use more than one metadynamics instance within the same simulation, this allows to distinguish each
     instance from the others. If there is more than one metadynamics instance, the name of this bias is
     included in the metadynamics output file names, such as e.g.&#x00A0;the <span
class="ot1-zi4r-0-x-x-109">.pmf </span>file.
     </li>
     <li class="itemize"><a
 id="dx1-60002"></a> <span
class="ot1-zi4r-0-x-x-120">keepHills </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Write each individual hill to the state file<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">metadynamics</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">off</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:   </span>When  <span
class="ot1-zi4r-0-x-x-109">useGrids </span>and  this  option  are  <span
class="ot1-zi4r-0-x-x-109">on</span>,  all  hills  are  saved  to  the  state  file  in  their
     analytic  form,  alongside  their  grids.  This  makes  it  possible  to  later  use  exact  analytic  Gaussians
     for  <span
class="ot1-zi4r-0-x-x-109">rebinGrids</span>.  To  only  keep  track  of  the  history  of  the  added  hills,  <span
class="ot1-zi4r-0-x-x-109">writeHillsTrajectory </span>is
     preferable.
     </li>
     <li class="itemize"><a
 id="dx1-60003"></a> <span
class="ot1-zi4r-0-x-x-120">writeHillsTrajectory </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Write a log of new hills<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">metadynamics</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">on</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:   </span>If  this  option  is  <span
class="ot1-zi4r-0-x-x-109">on</span>,  a  logfile  is  written  by  the  <span
class="ot1-zi4r-0-x-x-109">metadynamics </span>bias,  with  the  name
     &#8220;<span
class="ptmri7t-x-x-109">output</span><span
class="ot1-zi4r-0-x-x-109">.colvars.</span><span
class="zptmcm7m-x-x-109">&#x003C;</span><span
class="ot1-zi4r-0-x-x-109">name</span><span
class="zptmcm7m-x-x-109">&#x003E;</span><span
class="ot1-zi4r-0-x-x-109">.hills.traj</span>&#8221;, which can be useful to follow the time series of the hills.
     When <span
class="ot1-zi4r-0-x-x-109">multipleReplicas </span>is <span
class="ot1-zi4r-0-x-x-109">on</span>, its name changes to<br
class="newline" />&#8220;<span
class="ptmri7t-x-x-109">output</span><span
class="ot1-zi4r-0-x-x-109">.colvars.</span><span
class="zptmcm7m-x-x-109">&#x003C;</span><span
class="ot1-zi4r-0-x-x-109">name</span><span
class="zptmcm7m-x-x-109">&#x003E;</span><span
class="ot1-zi4r-0-x-x-109">.</span><span
class="zptmcm7m-x-x-109">&#x003C;</span><span
class="ot1-zi4r-0-x-x-109">replicaID</span><span
class="zptmcm7m-x-x-109">&#x003E;</span><span
class="ot1-zi4r-0-x-x-109">.hills.traj</span>&#8221;.   This   file   can   be   used   to   quickly
     visualize  the  positions  of  all  added  hills,  in  case  <span
class="ot1-zi4r-0-x-x-109">newHillFrequency </span>does  not  coincide  with
     <span
class="ot1-zi4r-0-x-x-109">colvarsRestartFrequency</span>.
     </li></ul>


<!--l. 3214--><p class="noindent" >
   <h4 class="subsectionHead"><span class="titlemark">6.3    </span> <a
 id="x1-610006.3"></a>Harmonic restraints</h4>
<!--l. 3217--><p class="noindent" >The harmonic biasing method may be used to enforce fixed or moving restraints, including variants of Steered and
Targeted MD. Within energy minimization runs, it allows for restrained minimization, e.g. to calculate relaxed
potential energy surfaces. In the context of the colvars module, harmonic potentials are meant according to their
textbook definition: <span
class="zptmcm7m-x-x-109">V </span><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">x</span><span
class="zptmcm7t-x-x-109">) =</span> <img
src="colvars-refman-lammps28x.png" alt="1-
2"  class="frac" align="middle"><span
class="zptmcm7m-x-x-109">k</span><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">x</span><span
class="zptmcm7y-x-x-109">-</span><span
class="zptmcm7m-x-x-109">x</span><sub><span
class="zptmcm7t-x-x-80">0</span></sub><span
class="zptmcm7t-x-x-109">)</span><sup><span
class="zptmcm7t-x-x-80">2</span></sup>. Note that this differs from harmonic bond and angle potentials in common
force fields, where the factor of one half is typically omitted, resulting in a non-standard definition of the force
constant.  A harmonic restraint is set up by a <span
class="ot1-zi4r-0-x-x-109">harmonic</span><span
class="ot1-zi4r-0-x-x-109">&#x00A0;</span><span
class="cmsy-10x-x-109">{</span><span
class="ot1-zi4r-0-x-x-109">...</span><span
class="cmsy-10x-x-109">} </span>block, which may contain (in addition to the standard
option <span
class="ot1-zi4r-0-x-x-109">colvars</span>) the following keywords:
     <ul class="itemize1">
     <li class="itemize"><a
 id="dx1-61001"></a> <span
class="ot1-zi4r-0-x-x-120">forceConstant </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Scaled force constant (unit of energy specified by <span
class="ot1-zi4r-0-x-x-109">units</span>)<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">harmonic</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>positive decimal <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">1.0</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>This defines a scaled force constant for the harmonic potential. To ensure consistency
     for multidimensional restraints, it is divided internally by the square of the specific <span
class="ot1-zi4r-0-x-x-109">width </span>for each
     colvar  involved  (which  is  1  by  default),  so  that  all  colvars  are  effectively  dimensionless  and  of
     commensurate size. For instance, setting a scaled force constant of 10&#x00A0;kcal/mol acting on two colvars,
     an angle with a <span
class="ot1-zi4r-0-x-x-109">width </span>of 5&#x00A0;degrees and a distance with a width of 0.5&#x00A0;Å, will apply actual force
     constants of 0.4&#x00A0;kcal/mol<span
class="zptmcm7y-x-x-109">&#x00D7;</span>degree<sup><span
class="zptmcm7y-x-x-80">-</span><span
class="zptmcm7t-x-x-80">2</span></sup> for the angle and 40&#x00A0;kcal/mol/Å<sup><span
class="zptmcm7t-x-x-80">2</span></sup> for the distance.
     </li>
     <li class="itemize"><a
 id="dx1-61002"></a> <span
class="ot1-zi4r-0-x-x-120">centers </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Initial harmonic restraint centers<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">harmonic</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>space-separated list of colvar values <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>The centers (equilibrium values) of the restraint are entered here. The number of values
     must be the number of requested colvars. Each value is a decimal number if the corresponding colvar
     returns a scalar, a &#8220;<span
class="ot1-zi4r-0-x-x-109">(x, y, z)</span>&#8221; triplet if it returns a unit vector or a vector, and a &#8220;<span
class="ot1-zi4r-0-x-x-109">q0, q1, q2, q3)</span>&#8221;
     quadruplet if it returns a rotational quaternion. If a colvar has periodicities or symmetries, its closest
     image to the restraint center is considered when calculating the harmonic potential.</li></ul>
<!--l. 3266--><p class="indent" >   <span
class="ptmb7t-x-x-109">Tip: </span>A complex set of restraints can be applied to a system, by defining several colvars, and applying one
or more harmonic restraints to different groups of colvars. In some cases, dozens of colvars can be
defined, but their value may not be relevant: to limit the size of the colvars trajectory file, it may be
wise to disable <span
class="ot1-zi4r-0-x-x-109">outputValue </span>for such &#8220;ancillary&#8221; variables, and leave it enabled only for &#8220;relevant&#8221;
ones.
<!--l. 3274--><p class="noindent" >
   <h5 class="subsubsectionHead"><span class="titlemark">6.3.1    </span> <a
 id="x1-620006.3.1"></a>Moving restraints: steered molecular dynamics</h5>
<!--l. 3277--><p class="noindent" >The following options allow to change gradually the centers of the harmonic restraints during a simulations.
When the centers are changed continuously, a steered MD in a collective variable space is carried
out.


     <ul class="itemize1">
     <li class="itemize"><a
 id="dx1-62001"></a> <span
class="ot1-zi4r-0-x-x-120">targetCenters </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Steer the restraint centers towards these targets<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">harmonic</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>space-separated list of colvar values <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:   </span>When defined, the current <span
class="ot1-zi4r-0-x-x-109">centers </span>will be moved towards these values during the
     simulation.  By  default,  the  centers  are  moved  over  a  total  of  <span
class="ot1-zi4r-0-x-x-109">targetNumSteps </span>steps  by  a  linear
     interpolation, in the spirit of Steered MD. If <span
class="ot1-zi4r-0-x-x-109">targetNumStages </span>is set to a nonzero value, the change
     is performed in discrete stages, lasting <span
class="ot1-zi4r-0-x-x-109">targetNumSteps </span>steps <span
class="ptmri7t-x-x-109">each</span>. This second mode may be used
     to sample successive windows in the context of an Umbrella Sampling simulation. When continuing
     a simulation run, the <span
class="ot1-zi4r-0-x-x-109">centers </span>specified in the configuration file <span
class="zptmcm7m-x-x-109">&#x003C;</span><span
class="ot1-zi4r-0-x-x-109">colvarsConfig</span><span
class="zptmcm7m-x-x-109">&#x003E; </span>are overridden
     by those saved in the restart file <span
class="zptmcm7m-x-x-109">&#x003C;</span><span
class="ot1-zi4r-0-x-x-109">colvarsInput</span><span
class="zptmcm7m-x-x-109">&#x003E;</span>. To perform Steered MD in an arbitrary space of
     colvars, it is sufficient to use this option and enable <span
class="ot1-zi4r-0-x-x-109">outputAppliedForce </span>within each of the colvars
     involved.
     </li>
     <li class="itemize"><a
 id="dx1-62002"></a> <span
class="ot1-zi4r-0-x-x-120">targetNumSteps </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Number of steps for steering<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">harmonic</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>positive integer <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>In single-stage (continuous) transformations, defines the number of MD steps required
     to move the restraint centers (or force constant) towards the values specified with <span
class="ot1-zi4r-0-x-x-109">targetCenters </span>or
     <span
class="ot1-zi4r-0-x-x-109">targetForceConstant</span>. After the target values have been reached, the centers (resp. force constant)
     are kept fixed. In multi-stage transformations, this sets the number of MD steps <span
class="ptmri7t-x-x-109">per stage</span>.
     </li>
     <li class="itemize"><a
 id="dx1-62003"></a> <span
class="ot1-zi4r-0-x-x-120">outputCenters </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Write the current centers to the trajectory file<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">harmonic</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">off</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:   </span>If this option is chosen and <span
class="ot1-zi4r-0-x-x-109">colvarsTrajFrequency </span>is not zero, the positions of the
     restraint  centers  will  be  written  to  the  trajectory  file  during  the  simulation.  This  option  allows  to
     conveniently extract the PMF from the colvars trajectory files in a steered MD calculation.
     </li>
     <li class="itemize"><a
 id="dx1-62004"></a> <span
class="ot1-zi4r-0-x-x-120">outputAccumulatedWork  </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Write the accumulated work of the moving restraint to the trajectory
     file<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">harmonic</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>boolean <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">off</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>If this option is chosen, <span
class="ot1-zi4r-0-x-x-109">targetCenters </span>is defined, and <span
class="ot1-zi4r-0-x-x-109">colvarsTrajFrequency </span>is not
     zero, the accumulated work from the beginning of the simulation will be written to the trajectory file.
     If the simulation has been continued from a previous state file, the previously accumulated work is
     included in the integral. This option allows to conveniently extract the PMF from the colvars trajectory
     files in a steered MD calculation.
     </li></ul>
<!--l. 3346--><p class="indent" >   <span
class="ptmb7t-x-x-109">Note on restarting moving restraint simulations: </span>Information about the current step and stage of a
simulation with moving restraints is stored in the restart file (state file). Thus, such simulations can be
run in several chunks, and restarted directly using the same colvars configuration file. In case of a
restart, the values of parameters such as <span
class="ot1-zi4r-0-x-x-109">targetCenters</span>, <span
class="ot1-zi4r-0-x-x-109">targetNumSteps</span>, etc. should not be changed


manually.
<!--l. 3355--><p class="noindent" >
   <h5 class="subsubsectionHead"><span class="titlemark">6.3.2    </span> <a
 id="x1-630006.3.2"></a>Moving restraints: umbrella sampling</h5>
<!--l. 3358--><p class="noindent" >The centers of the harmonic restraints can also be changed in discrete stages: in this cases a one-dimensional
umbrella sampling simulation is performed. The sampling windows in simulation are calculated in sequence. The
colvars trajectory file may then be used both to evaluate the correlation times between consecutive windows, and to
calculate the frequency distribution of the colvar of interest in each window. Furthermore, frequency
distributions on a predefined grid can be automatically obtained by using the <span
class="ot1-zi4r-0-x-x-109">histogram </span>bias (see
<a
href="#x1-670006.6">6.6<!--tex4ht:ref: sec:colvarbias_histogram --></a>).
<!--l. 3363--><p class="indent" >   To activate an umbrella sampling simulation, the same keywords as in the previous section can be used, with the
addition of the following:
     <ul class="itemize1">
     <li class="itemize"><a
 id="dx1-63001"></a> <span
class="ot1-zi4r-0-x-x-120">targetNumStages </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Number of stages for steering<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">harmonic</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>non-negative integer <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">0</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>If non-zero, sets the number of stages in which the restraint centers or force constant are
     changed to their target values. If zero, the change is continuous. Each stage lasts <span
class="ot1-zi4r-0-x-x-109">targetNumSteps </span>MD
     steps. To sample both ends of the transformation, the simulation should be run for <span
class="ot1-zi4r-0-x-x-109">targetNumSteps</span>
     <span
class="zptmcm7y-x-x-109">&#x00D7; </span>(<span
class="ot1-zi4r-0-x-x-109">targetNumStages </span>+ 1).
     </li></ul>
<!--l. 3382--><p class="noindent" >
   <h5 class="subsubsectionHead"><span class="titlemark">6.3.3    </span> <a
 id="x1-640006.3.3"></a>Changing force constant</h5>
<!--l. 3385--><p class="noindent" >The force constant of the harmonic restraint may also be changed to equilibrate <span class="cite">[<a
href="#XDeng2009">17</a>]</span>.
     <ul class="itemize1">
     <li class="itemize"><a
 id="dx1-64001"></a> <span
class="ot1-zi4r-0-x-x-120">targetForceConstant </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Change the force constant towards this value<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">harmonic</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>positive decimal <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>When defined, the current <span
class="ot1-zi4r-0-x-x-109">forceConstant </span>will be moved towards this value during the
     simulation. Time evolution of the force constant is dictated by the <span
class="ot1-zi4r-0-x-x-109">targetForceExponent </span>parameter
     (see below). By default, the force constant is changed smoothly over a total of <span
class="ot1-zi4r-0-x-x-109">targetNumSteps </span>steps.
     This is useful to introduce or remove restraints in a progressive manner. If <span
class="ot1-zi4r-0-x-x-109">targetNumStages </span>is set
     to a nonzero value, the change is performed in discrete stages, lasting <span
class="ot1-zi4r-0-x-x-109">targetNumSteps </span>steps <span
class="ptmri7t-x-x-109">each</span>.
     This second mode may be used to compute the conformational free energy change associated with the
     restraint, within the FEP or TI formalisms. For convenience, the code provides an estimate of the free
     energy derivative for use in TI. A more complete free energy calculation (particularly with regard to
     convergence analysis), while not handled by the colvars module, can be performed by post-processing
     the colvars trajectory, if <span
class="ot1-zi4r-0-x-x-109">colvarsTrajFrequency </span>is set to a suitably small value. It should be noted,
     however, that restraint free energy calculations may be handled more efficiently by an indirect route,
     through the determination of a PMF for the restrained coordinate.<span class="cite">[<a
href="#XDeng2009">17</a>]</span>


     </li>
     <li class="itemize"><a
 id="dx1-64002"></a> <span
class="ot1-zi4r-0-x-x-120">targetForceExponent </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Exponent in the time-dependence of the force constant<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">harmonic</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>decimal equal to or greater than 1.0 <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">1.0</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:   </span>Sets  the  exponent,  <span
class="zptmcm7m-x-x-109">&#x03B1;</span>,  in  the  function  used  to  vary  the  force  constant  as  a  function
     of  time.  The  force  is  varied  according  to  a  coupling  parameter  <span
class="zptmcm7m-x-x-109">&#x03BB;</span>,  raised  to  the  power  <span
class="zptmcm7m-x-x-109">&#x03B1;</span>:  <span
class="zptmcm7m-x-x-109">k</span><sub><span
class="zptmcm7m-x-x-80">&#x03BB;</span></sub> <span
class="zptmcm7t-x-x-109">=</span>
     <span
class="zptmcm7m-x-x-109">k</span><sub><span
class="zptmcm7t-x-x-80">0</span></sub> <span
class="zptmcm7t-x-x-109">+</span><span
class="zptmcm7m-x-x-109">&#x03BB;</span><sup><span
class="zptmcm7m-x-x-80">&#x03B1;</span></sup><span
class="zptmcm7t-x-x-109">(</span><span
class="zptmcm7m-x-x-109">k</span><sub><span
class="zptmcm7t-x-x-80">1</span></sub> <span
class="zptmcm7y-x-x-109">-</span><span
class="zptmcm7m-x-x-109">k</span><sub><span
class="zptmcm7t-x-x-80">0</span></sub><span
class="zptmcm7t-x-x-109">)</span>, where <span
class="zptmcm7m-x-x-109">k</span><sub><span
class="zptmcm7t-x-x-80">0</span></sub>, <span
class="zptmcm7m-x-x-109">k</span><sub><span
class="zptmcm7m-x-x-80">&#x03BB;</span></sub>, and <span
class="zptmcm7m-x-x-109">k</span><sub><span
class="zptmcm7t-x-x-80">1</span></sub> are the initial, current, and final values of the force constant.
     The parameter <span
class="zptmcm7m-x-x-109">&#x03BB; </span>evolves linearly from 0 to 1, either smoothly, or in <span
class="ot1-zi4r-0-x-x-109">targetNumStages </span>equally spaced
     discrete stages, or according to an arbitrary schedule set with <span
class="ot1-zi4r-0-x-x-109">lambdaSchedule</span>. When the initial value
     of the force constant is zero, an exponent greater than 1.0 distributes the effects of introducing the
     restraint more smoothly over time than a linear dependence, and ensures that there is no singularity in
     the derivative of the restraint free energy with respect to lambda. A value of 4 has been found to give
     good results in some tests.
     </li>
     <li class="itemize"><a
 id="dx1-64003"></a> <span
class="ot1-zi4r-0-x-x-120">targetEquilSteps </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Number of steps discarded from TI estimate<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">harmonic</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>positive integer <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:   </span>Defines the number of steps within each stage that are considered equilibration and
     discarded from the restraint free energy derivative estimate reported reported in the output.
     </li>
     <li class="itemize"><a
 id="dx1-64004"></a> <span
class="ot1-zi4r-0-x-x-120">lambdaSchedule </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Schedule of lambda-points for changing force constant<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">harmonic</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>list of real numbers between 0 and 1 <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>If specified together with targetForceConstant, sets the sequence of discrete <span
class="zptmcm7m-x-x-109">&#x03BB; </span>values
     that will be used for different stages.
     </li></ul>
<!--l. 3460--><p class="noindent" >
   <h4 class="subsectionHead"><span class="titlemark">6.4    </span> <a
 id="x1-650006.4"></a>Linear restraints</h4>
<!--l. 3463--><p class="noindent" >The linear restraint biasing method is used to minimally bias a simulation. There is generally a unique strength of
bias for each CV center, which means you must know the bias force constant specifically for the center of the CV.
This force constant may be found by using experiment directed simulation described in section <a
href="#x1-660006.5">6.5<!--tex4ht:ref: sec:colvarbias_alb --></a>. Please cite
Pitera and Chodera when using <span class="cite">[<a
href="#XPitera2012">18</a>]</span>.
     <ul class="itemize1">
     <li class="itemize"><a
 id="dx1-65001"></a> <span
class="ot1-zi4r-0-x-x-120">forceConstant </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Scaled force constant (unit of energy specified by <span
class="ot1-zi4r-0-x-x-109">units</span>)<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">linear</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>positive decimal <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span><span
class="ot1-zi4r-0-x-x-109">1.0</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:   </span>This  defines  a  scaled  force  constant  for  the  linear  bias.  To  ensure  consistency  for
     multidimensional restraints, it is divided internally by the specific <span
class="ot1-zi4r-0-x-x-109">width </span>for each colvar involved
     (which is 1 by default), so that all colvars are effectively dimensionless and of commensurate size.


     </li>
     <li class="itemize"><a
 id="dx1-65002"></a> <span
class="ot1-zi4r-0-x-x-120">centers </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Initial linear restraint centers<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">linear</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>space-separated list of colvar values <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>The centers (equilibrium values) of the restraint are entered here. The number of values
     must be the number of requested colvars. Each value is a decimal number if the corresponding colvar
     returns a scalar, a &#8220;<span
class="ot1-zi4r-0-x-x-109">(x, y, z)</span>&#8221; triplet if it returns a unit vector or a vector, and a &#8220;<span
class="ot1-zi4r-0-x-x-109">q0, q1, q2, q3)</span>&#8221;
     quadruplet if it returns a rotational quaternion. If a colvar has periodicities or symmetries, its closest
     image to the restraint center is considered when calculating the linear potential.</li></ul>
<!--l. 3502--><p class="noindent" >
   <h4 class="subsectionHead"><span class="titlemark">6.5    </span> <a
 id="x1-660006.5"></a>Adaptive Linear Bias/Experiment Directed Simulation</h4>
<!--l. 3505--><p class="noindent" >Experiment directed simulation applies a linear bias with a changing force constant. Please cite White and Voth <span class="cite">[<a
href="#XWhite2014">19</a>]</span>
when using this feature. As opposed to that reference, the force constant here is scaled by the <span
class="ot1-zi4r-0-x-x-109">width </span>corresponding
to the biased colvar. In White and Voth, each force constant is scaled by the colvars set center. The bias converges to
a linear bias, after which it will be the minimal possible bias. You may also stop the simulation, take the median of
the force constants (ForceConst) found in the colvars trajectory file, and then apply a linear bias with that constant.
All the notes about units described in sections <a
href="#x1-650006.4">6.4<!--tex4ht:ref: sec:colvarbias_linear --></a> and <a
href="#x1-610006.3">6.3<!--tex4ht:ref: sec:colvarbias_harmonic --></a> apply here as well. <span
class="ptmb7t-x-x-109">This is not a valid simulation</span>
<span
class="ptmb7t-x-x-109">of any particular statistical ensemble and is only an optimization algorithm until the bias has</span>
<span
class="ptmb7t-x-x-109">converged</span>.
     <ul class="itemize1">
     <li class="itemize"><a
 id="dx1-66001"></a> <span
class="ot1-zi4r-0-x-x-120">centers </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Collective variable centers<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">alb</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>space-separated list of colvar values <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>The desired center (equilibrium values) which will be sought during the adaptive linear
     biasing. The number of values must be the number of requested colvars. Each value is a decimal
     number if the corresponding colvar returns a scalar, a &#8220;<span
class="ot1-zi4r-0-x-x-109">(x, y, z)</span>&#8221; triplet if it returns a unit vector
     or a vector, and a &#8220;<span
class="ot1-zi4r-0-x-x-109">q0, q1, q2, q3)</span>&#8221; quadruplet if it returns a rotational quaternion. If a colvar has
     periodicities or symmetries, its closest image to the restraint center is considered when calculating the
     linear potential.
     </li>
     <li class="itemize"><a
 id="dx1-66002"></a> <span
class="ot1-zi4r-0-x-x-120">updateFrequency </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>The duration of updates<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">alb</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>An integer <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:   </span>This is, <span
class="zptmcm7m-x-x-109">N</span>, the number of simulation steps to use for each update to the bias. This
     determines how long the system requires to equilibrate after a change in force constant (<span
class="zptmcm7m-x-x-109">N&#x2215;</span><span
class="zptmcm7t-x-x-109">2</span>), how
     long statistics are collected for an iteration (<span
class="zptmcm7m-x-x-109">N&#x2215;</span><span
class="zptmcm7t-x-x-109">2</span>), and how quickly energy is added to the system
     (at most, <span
class="zptmcm7m-x-x-109">A&#x2215;</span><span
class="zptmcm7t-x-x-109">2</span><span
class="zptmcm7m-x-x-109">N</span>, where <span
class="zptmcm7m-x-x-109">A </span>is the <span
class="ot1-zi4r-0-x-x-109">forceRange</span>). Until the force constant has converged, the method as
     described is an optimization procedure and not an integration of a particular statistical ensemble. It
     is important that each step should be uncorrelated from the last so that iterations are independent.
     Therefore, <span
class="zptmcm7m-x-x-109">N </span>should be at least twice the autocorrelation time of the collective variable. The system
     should also be able to dissipate energy as fast as <span
class="zptmcm7m-x-x-109">N&#x2215;</span><span
class="zptmcm7t-x-x-109">2</span>, which can be done by adjusting thermostat
     parameters. Practically, <span
class="zptmcm7m-x-x-109">N </span>has been tested successfully at significantly shorter than the autocorrelation
     time of the collective variables being biased and still converge correctly.


     </li>
     <li class="itemize"><a
 id="dx1-66003"></a> <span
class="ot1-zi4r-0-x-x-120">forceRange </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>The expected range of the force constant in units of energy<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">alb</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>A space-separated list of decimal numbers <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span>3 <span
class="zptmcm7m-x-x-109">k</span><sub><span
class="zptmcm7m-x-x-80">b</span></sub><span
class="zptmcm7m-x-x-109">T </span><br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>This is largest magnitude of the force constant which one expects. If this parameter is
     too low, the simulation will not converge. If it is too high the simulation will waste time exploring
     values that are too large. A value of 3 <span
class="zptmcm7m-x-x-109">k</span><sub><span
class="zptmcm7m-x-x-80">b</span></sub><span
class="zptmcm7m-x-x-109">T </span>has worked well in the systems presented as a first choice.
     This parameter is dynamically adjusted over the course of a simulation. The benefit is that a bad guess
     for the forceRange can be corrected. However, this can lead to large amounts of energy being added
     over time to the system. To prevent this dynamic update, add <span
class="ot1-zi4r-0-x-x-109">hardForceRange yes </span>as a parameter
     </li>
     <li class="itemize"><a
 id="dx1-66004"></a> <span
class="ot1-zi4r-0-x-x-120">rateMax </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>The maximum rate of change of force constant<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">alb</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>A list of space-separated real numbers <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>This optional parameter controls how much energy is added to the system from this bias.
     Tuning this separately from the <span
class="ot1-zi4r-0-x-x-109">updateFrequency </span>and <span
class="ot1-zi4r-0-x-x-109">forceRange </span>can allow for large bias changes
     but with a low <span
class="ot1-zi4r-0-x-x-109">rateMax </span>prevents large energy changes that can lead to instability in the simulation.
     </li></ul>
<!--l. 3579--><p class="noindent" >
   <h4 class="subsectionHead"><span class="titlemark">6.6    </span> <a
 id="x1-670006.6"></a>Multidimensional histograms</h4>
<!--l. 3582--><p class="noindent" >The <span
class="ot1-zi4r-0-x-x-109">histogram </span>feature is used to record the distribution of a set of collective variables in the form of a
N-dimensional histogram. It functions as a &#8220;collective variable bias&#8221;, and is invoked by adding a <span
class="ot1-zi4r-0-x-x-109">histogram </span>block
to the <span
class="ptmri7t-x-x-109">colvars </span>configuration file.
<!--l. 3587--><p class="indent" >   In addition to the common parameters <span
class="ot1-zi4r-0-x-x-109">name </span>and <span
class="ot1-zi4r-0-x-x-109">colvars </span>described above, a <span
class="ot1-zi4r-0-x-x-109">histogram </span>block may define the
following parameter:
     <ul class="itemize1">
     <li class="itemize"><a
 id="dx1-67001"></a> <span
class="ot1-zi4r-0-x-x-120">outputFreq </span><span
class="zptmcm7y-x-x-109">&#x27E8;</span><span style="margin-left:0.3em" class="thinspace"></span>Frequency (in timesteps) at which the histogram file is refreshed<span style="margin-left:0.3em" class="thinspace"></span><span
class="zptmcm7y-x-x-109">&#x27E9;</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Context:  </span><span
class="ot1-zi4r-0-x-x-109">histogram</span><br
class="newline" /><span
class="ptmb7t-x-x-109">Acceptable values:  </span>positive integer <br
class="newline" /><span
class="ptmb7t-x-x-109">Default value:  </span>Colvar module restart frequency <br
class="newline" /><span
class="ptmb7t-x-x-109">Description:  </span>The file containing histogram data is written on disk at the given time interval.</li></ul>
<!--l. 3598--><p class="indent" >   Like the ABF and metadynamics biases, <span
class="ot1-zi4r-0-x-x-109">histogram </span>uses parameters from the colvars to define its
grid. The grid ranges from <span
class="ot1-zi4r-0-x-x-109">lowerBoundary </span>to <span
class="ot1-zi4r-0-x-x-109">upperBoundary</span>, and the bin width is set by the <span
class="ot1-zi4r-0-x-x-109">width</span>
parameter.
<!--l. 108--><p class="indent" >


   <h3 class="likesectionHead"><a
 id="x1-680006.6"></a>Index</h3>
   <div class="theindex"><span class="index-item"><span
class="ot1-zi4r-0-x-x-109">abf</span><br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">applyBias</span>, 33 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">fullSamples</span>, 32 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">hideJacobian</span>, 32 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">historyFreq</span>, 33 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">inputPrefix</span>, 33 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">maxForce</span>, 32 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">outputFreq</span>, 33 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">updateBias</span>, 33 <br /></span>
<span class="index-item"><span
class="ot1-zi4r-0-x-x-109">alb</span><br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">centers</span>, 43 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">forceRange</span>, 44 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">rateMax</span>, 44 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">updateFrequency</span>, 43 <br /></span>
<span class="index-item"><span
class="ot1-zi4r-0-x-x-109">angle</span>, <span
class="ot1-zi4r-0-x-x-109">dihedral</span><br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">oneSiteSystemForce</span>, 21 <br /></span>
<span class="index-item"><span
class="ot1-zi4r-0-x-x-109">cartesian</span><br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">atoms</span>, 21 <br /></span>
<span class="index-item"><span
class="ot1-zi4r-0-x-x-109">colvar</span><br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">corrFuncLength</span>, 13 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">corrFuncNormalize</span>, 13 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">corrFuncOffset</span>, 13 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">corrFuncOutputFile</span>, 13 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">corrFuncStride</span>, 13 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">corrFuncType</span>, 12 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">corrFuncWithColvar</span>, 12 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">corrFunc</span>, 12 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">expandBoundaries</span>, 9 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">extendedFluctuation</span>, 11 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">extendedLagrangian</span>, 11 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">extendedLangevinDamping</span>, 12 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">extendedTemp</span>, 12 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">extendedTimeConstant</span>, 11 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">hardLowerBoundary</span>, 9 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">hardUpperBoundary</span>, 9 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">lowerBoundary</span>, 9 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">lowerWallConstant</span>, 9 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">lowerWall</span>, 10 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">name</span>, 8 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">outputAppliedForce</span>, 11 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">outputEnergy</span>, 10 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">outputSystemForce</span>, 10 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">outputValue</span>, 10 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">outputVelocity</span>, 10 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">runAveLength</span>, 13 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">runAveOutputFile</span>, 14 <br /></span>


<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">runAveStride</span>, 13 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">runAve</span>, 13 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">upperBoundary</span>, 9 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">upperWallConstant</span>, 10 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">upperWall</span>, 10 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">width</span>, 8 <br /></span>
<span class="index-item"><span
class="ot1-zi4r-0-x-x-109">coordNum</span><br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">cutoff3</span>, 22 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">cutoff</span>, 22 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">expDenom</span>, 22 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">expNumer</span>, 22 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">group2CenterOnly</span>, 22 <br /></span>
<span class="index-item"><span
class="ot1-zi4r-0-x-x-109">distanceInv</span><br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">exponent</span>, 21 <br /></span>
<span class="index-item"><span
class="ot1-zi4r-0-x-x-109">distanceZ, distanceXY</span><br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">main</span>, 19 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">ref2</span>, 19 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">ref</span>, 19 <br /></span>
<span class="index-item"><span
class="ot1-zi4r-0-x-x-109">distanceZ, distanceXY</span><br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">forceNoPBC</span>, 20 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">oneSiteSystemForce</span>, 20 <br /></span>
<span class="index-item"><span
class="ot1-zi4r-0-x-x-109">distanceZ</span><br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">period</span>, 28 <br /></span>
<span class="index-item"><span
class="ot1-zi4r-0-x-x-109">distanceZ</span>, <span
class="ot1-zi4r-0-x-x-109">dihedral </span>or <span
class="ot1-zi4r-0-x-x-109">spinAngle</span><br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">wrapAround</span>, 28 <br /></span>
<span class="index-item"><span
class="ot1-zi4r-0-x-x-109">distanceZ</span>, <span
class="ot1-zi4r-0-x-x-109">distanceXY</span><br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">axis</span>, 20 <br /></span>
<span class="index-item"><span
class="ot1-zi4r-0-x-x-109">distance</span><br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">forceNoPBC</span>, 19 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">group1</span>, 18 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">group2</span>, 19 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">oneSiteSystemForce</span>, 19 <br /></span>
<span class="index-item"><span
class="ot1-zi4r-0-x-x-109">eigenvector</span><br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">differenceVector</span>, 25 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">vector</span>, 25 <br /></span>
<span class="index-item"><span
class="ot1-zi4r-0-x-x-109">harmonic</span><br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">centers</span>, 40 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">forceConstant</span>, 40 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">lambdaSchedule</span>, 42 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">outputAccumulatedWork</span>, 41 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">outputCenters</span>, 41 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">targetCenters</span>, 40 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">targetEquilSteps</span>, 42 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">targetForceConstant</span>, 42 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">targetForceExponent</span>, 42 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">targetNumStages</span>, 41 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">targetNumSteps</span>, 40 <br /></span>
<span class="index-item"><span
class="ot1-zi4r-0-x-x-109">histogram</span><br /></span>


<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">outputFreq</span>, 44 <br /></span>
<span class="index-item"><span
class="ot1-zi4r-0-x-x-109">inertiaZ</span><br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">axis</span>, 26 <br /></span>
<span class="index-item"><span
class="ot1-zi4r-0-x-x-109">linear</span><br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">centers</span>, 43 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">forceConstant</span>, 43 <br /></span>
<span class="index-item"><span
class="ot1-zi4r-0-x-x-109">metadynamics</span><br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">biasTemperature</span>, 37 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">dumpPartialFreeEnergyFile</span>, 39 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">hillWeight</span>, 35 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">hillWidth</span>, 37 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">keepHills</span>, 39 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">multipleReplicas</span>, 38 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">name</span>, 39 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">newHillFrequency</span>, 36 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">rebinGrids</span>, 37 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">replicaID</span>, 38 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">replicaUpdateFrequency</span>, 38 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">replicasRegistry</span>, 38 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">saveFreeEnergyFile</span>, 36 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">useGrids</span>, 37 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">wellTempered</span>, 37 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">writeFreeEnergyFile</span>, 36 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">writeHillsTrajectory</span>, 39 <br /></span>
<span class="index-item"><span
class="ot1-zi4r-0-x-x-109">orientation</span><br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">closestToQuaternion</span>, 26 <br /></span>
<span class="index-item"><span
class="ot1-zi4r-0-x-x-109">rmsd</span><br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">atoms</span>, 23 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">refPositions</span>, 24 <br /></span>
<span class="index-item"><span
class="ot1-zi4r-0-x-x-109">tilt, spinAngle</span><br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">axis</span>, 27 <br /></span>
<p class="theindex">
<span class="index-item">any component <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">componentCoeff</span>, 29 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">componentExp</span>, 29 <br /></span>
<span class="index-item">atom group <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">atomNumbersRange</span>, 15 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">atomNumbers</span>, 14 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">centerReference</span>, 15 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">dummyAtom</span>, 15 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">enableFitGradients</span>, 16 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">enableForces</span>, 17 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">indexGroup</span>, 14 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">refPositionsGroup</span>, 16 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">refPositions</span>, 16 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">rotateReference</span>, 15 <br /></span>
</p><p class="theindex">


<span class="index-item">colvar bias <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">colvars</span>, 30 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">name</span>, 30 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">outputEnergy</span>, 30 <br /></span>
</p><p class="theindex">
<span class="index-item">global <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">analysis</span>, 6 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">colvarsRestartFrequency</span>, 6 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">colvarsTrajAppend</span>, 6 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">colvarsTrajFrequency</span>, 5 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">indexFile</span>, 6 <br /></span>
</p><p class="theindex">
<span class="index-item">Keyword of the <span
class="ot1-zi4r-0-x-x-109">fix colvars </span>command <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">input</span>, 4 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">output</span>, 4 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">seed</span>, 5 <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">tstat</span>, 5 <br /></span>
<span class="index-item">keyword of the <span
class="ot1-zi4r-0-x-x-109">fix colvars </span>command <br /></span>
<span class="index-subitem">&#x00A0;&#x00A0;&#x00A0;&#x00A0;<span
class="ot1-zi4r-0-x-x-109">unwrap</span>, 4 <br /></span>
</p></div>
<!--l. 111--><p class="noindent" >


   <h3 class="likesectionHead"><a
 id="x1-690006.6"></a>References</h3>
<!--l. 1--><p class="noindent" >
     <div class="thebibliography">
     <p class="bibitem" ><span class="biblabel">
  [1]<span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span><a
 id="XFiorin2013"></a>G.&#x00A0;Fiorin, M.&#x00A0;L. Klein, and J.&#x00A0;Hénin.   Using collective variables to drive molecular dynamics
     simulations. <span
class="ptmri7t-x-x-109">Mol. Phys.</span>, 111(22-23):3345&#8211;3362, 2013.
     </p>
     <p class="bibitem" ><span class="biblabel">
  [2]<span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span><a
 id="XIannuzzi2003"></a>M.&#x00A0;Iannuzzi,  A.&#x00A0;Laio,  and  M.&#x00A0;Parrinello.    Efficient  exploration  of  reactive  potential  energy
     surfaces using car-parrinello molecular dynamics. <span
class="ptmri7t-x-x-109">Phys. Rev. Lett.</span>, 90(23):238302, 2003.
     </p>
     <p class="bibitem" ><span class="biblabel">
  [3]<span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span><a
 id="XCoutsias2004"></a>E&#x00A0;A Coutsias, C&#x00A0;Seok, and K&#x00A0;A Dill.  Using quaternions to calculate RMSD.  <span
class="ptmri7t-x-x-109">J. Comput. Chem.</span>,
     25(15):1849&#8211;1857, 2004.
     </p>
     <p class="bibitem" ><span class="biblabel">
  [4]<span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span><a
 id="XDarve2008"></a>Eric Darve, David Rodríguez-Gómez, and Andrew Pohorille.  Adaptive biasing force method for
     scalar and vector free energy calculations. <span
class="ptmri7t-x-x-109">J. Chem. Phys.</span>, 128(14):144120, 2008.
     </p>
     <p class="bibitem" ><span class="biblabel">
  [5]<span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span><a
 id="XHenin2004"></a>J.&#x00A0;Hénin  and  C.&#x00A0;Chipot.     Overcoming  free  energy  barriers  using  unconstrained  molecular
     dynamics simulations. <span
class="ptmri7t-x-x-109">J. Chem. Phys.</span>, 121:2904&#8211;2914, 2004.
     </p>
     <p class="bibitem" ><span class="biblabel">
  [6]<span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span><a
 id="XHenin2010"></a>J.&#x00A0;Hénin,  G.&#x00A0;Fiorin,  C.&#x00A0;Chipot,  and  M.&#x00A0;L.  Klein.     Exploring  multidimensional  free  energy
     landscapes using time-dependent biases on collective variables. <span
class="ptmri7t-x-x-109">J. Chem. Theory Comput.</span>, 6(1):35&#8211;47,
     2010.
     </p>
     <p class="bibitem" ><span class="biblabel">
  [7]<span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span><a
 id="XCarter1989"></a>A.&#x00A0;Carter, E, G.&#x00A0;Ciccotti, J.&#x00A0;T. Hynes, and R.&#x00A0;Kapral. Constrained reaction coordinate dynamics
     for the simulation of rare events. <span
class="ptmri7t-x-x-109">Chem. Phys. Lett.</span>, 156:472&#8211;477, 1989.
     </p>
     <p class="bibitem" ><span class="biblabel">
  [8]<span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span><a
 id="XRuiz-Montero1997"></a>M.&#x00A0;J. Ruiz-Montero, D.&#x00A0;Frenkel, and J.&#x00A0;J. Brey.  Efficient schemes to compute diffusive barrier
     crossing rates. <span
class="ptmri7t-x-x-109">Mol. Phys.</span>, 90:925&#8211;941, 1997.
     </p>
     <p class="bibitem" ><span class="biblabel">
  [9]<span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span><a
 id="XdenOtter2000"></a>W.&#x00A0;K. den Otter.   Thermodynamic integration of the free energy along a reaction coordinate in
     cartesian coordinates. <span
class="ptmri7t-x-x-109">J. Chem. Phys.</span>, 112:7283&#8211;7292, 2000.


     </p>
     <p class="bibitem" ><span class="biblabel">
 [10]<span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span><a
 id="XCiccotti2005"></a>Giovanni Ciccotti, Raymond Kapral, and Eric Vanden-Eijnden.   Blue moon sampling, vectorial
     reaction coordinates, and unbiased constrained dynamics. <span
class="ptmri7t-x-x-109">ChemPhysChem</span>, 6(9):1809&#8211;1814, 2005.
     </p>
     <p class="bibitem" ><span class="biblabel">
 [11]<span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span><a
 id="XLaio2002"></a>A.&#x00A0;Laio  and  M.&#x00A0;Parrinello.     Escaping  free-energy  minima.     <span
class="ptmri7t-x-x-109">Proc.  Natl.  Acad.  Sci.  USA</span>,
     99(20):12562&#8211;12566, 2002.
     </p>
     <p class="bibitem" ><span class="biblabel">
 [12]<span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span><a
 id="XGrubmuller1995"></a>Helmut   Grubmüller.       Predicting   slow   structural   transitions   in   macromolecular   systems:
     Conformational flooding. <span
class="ptmri7t-x-x-109">Phys. Rev. E</span>, 52(3):2893&#8211;2906, Sep 1995.
     </p>
     <p class="bibitem" ><span class="biblabel">
 [13]<span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span><a
 id="XHuber1994"></a>T.&#x00A0;Huber,  A.&#x00A0;E.  Torda,  and  W.F.  van  Gunsteren.    Local  elevation  -  A  method  for  improving
     the searching properties of molecular-dynamics simulation.   <span
class="ptmri7t-x-x-109">Journal of Computer-Aided Molecular</span>
     <span
class="ptmri7t-x-x-109">Design</span>, 8(6):695&#8211;708, DEC 1994.
     </p>
     <p class="bibitem" ><span class="biblabel">
 [14]<span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span><a
 id="XBussi2006"></a>G.&#x00A0;Bussi,   A.&#x00A0;Laio,   and   M.&#x00A0;Parrinello.       Equilibrium   free   energies   from   nonequilibrium
     metadynamics. <span
class="ptmri7t-x-x-109">Phys. Rev. Lett.</span>, 96(9):090601, 2006.
     </p>
     <p class="bibitem" ><span class="biblabel">
 [15]<span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span><a
 id="XBarducci2008"></a>Alessandro Barducci, Giovanni Bussi, and Michele Parrinello.  Well-tempered metadynamics: A
     smoothly converging and tunable free-energy method. <span
class="ptmri7t-x-x-109">Phys. Rev. Lett.</span>, 100:020603, 2008.
     </p>
     <p class="bibitem" ><span class="biblabel">
 [16]<span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span><a
 id="XRaiteri2005"></a>P.&#x00A0;Raiteri, A.&#x00A0;Laio, F.&#x00A0;L. Gervasio, C.&#x00A0;Micheletti, and M.&#x00A0;Parrinello.  Efficient reconstruction of
     complex free energy landscapes by multiple walkers metadynamics. <span
class="ptmri7t-x-x-109">J. Phys. Chem. B</span>, 110(8):3533&#8211;9,
     2005.
     </p>
     <p class="bibitem" ><span class="biblabel">
 [17]<span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span><a
 id="XDeng2009"></a>Yuqing Deng and Benoît Roux.  Computations of standard binding free energies with molecular
     dynamics simulations. <span
class="ptmri7t-x-x-109">J. Phys. Chem. B</span>, 113(8):2234&#8211;2246, 2009.
     </p>
     <p class="bibitem" ><span class="biblabel">
 [18]<span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span><a
 id="XPitera2012"></a>Jed&#x00A0;W. Pitera and John&#x00A0;D. Chodera.  On the use of experimental observations to bias simulated
     ensembles. <span
class="ptmri7t-x-x-109">J. Chem. Theory Comput.</span>, 8:3445&#8211;3451, 2012.


     </p>
     <p class="bibitem" ><span class="biblabel">
 [19]<span class="bibsp">&#x00A0;&#x00A0;&#x00A0;</span></span><a
 id="XWhite2014"></a>A.&#x00A0;D. White and G.&#x00A0;A. Voth.  Efficient and minimal method to bias molecular simulations with
     experimental data. <span
class="ptmri7t-x-x-109">J. Chem. Theory Comput.</span>, ASAP, 2014.
</p>
     </div>

</body></html>




